<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 20 Linear models for counts, binary responses, skewed responses, and ratios – Generalized Linear Models | Elements of Statistical Modeling for Experimental Biology</title>
  <meta name="description" content="A first course in statistical modeling for experimental biology researchers" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 20 Linear models for counts, binary responses, skewed responses, and ratios – Generalized Linear Models | Elements of Statistical Modeling for Experimental Biology" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A first course in statistical modeling for experimental biology researchers" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 20 Linear models for counts, binary responses, skewed responses, and ratios – Generalized Linear Models | Elements of Statistical Modeling for Experimental Biology" />
  
  <meta name="twitter:description" content="A first course in statistical modeling for experimental biology researchers" />
  

<meta name="author" content="Copyright 2018 Jeffrey A. Walker" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="pre-post.html"/>
<link rel="next" href="gls.html"/>
<script src="libs/header-attrs-2.8/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Elements of Applied Biostatistics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#why-bother-with-linear-models-arent-t-tests-and-anova-good-enough"><i class="fa fa-check"></i><b>0.1</b> Why bother with linear models – aren’t <em>t</em>-tests and ANOVA good enough?</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#what-is-unusual-about-this-book"><i class="fa fa-check"></i><b>0.2</b> What is unusual about this book?</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="a-table-of-models.html"><a href="a-table-of-models.html"><i class="fa fa-check"></i>A Table of Models</a>
<ul>
<li class="chapter" data-level="" data-path="a-table-of-models.html"><a href="a-table-of-models.html#including-mapping-between-linear-models-and-classical-tests"><i class="fa fa-check"></i>including mapping between linear models and classical tests</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-i-getting-started.html"><a href="part-i-getting-started.html"><i class="fa fa-check"></i>Part I: Getting Started</a></li>
<li class="chapter" data-level="1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html"><i class="fa fa-check"></i><b>1</b> Getting Started – R Projects and R Markdown</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#r-vs-r-studio"><i class="fa fa-check"></i><b>1.1</b> R vs R Studio</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#download-and-install-r-and-r-studio"><i class="fa fa-check"></i><b>1.2</b> Download and install R and R studio</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#open-r-studio-and-modify-the-workspace-preference"><i class="fa fa-check"></i><b>1.3</b> Open R Studio and modify the workspace preference</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#if-you-didnt-modify-the-workspace-preferences-from-the-previous-section-go-back-and-do-it"><i class="fa fa-check"></i><b>1.4</b> If you didn’t modify the workspace preferences from the previous section, go back and do it</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#r-markdown-in-a-nutshell"><i class="fa fa-check"></i><b>1.5</b> R Markdown in a nutshell</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#install-r-markdown"><i class="fa fa-check"></i><b>1.6</b> Install R Markdown</a></li>
<li class="chapter" data-level="1.7" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#importing-packages"><i class="fa fa-check"></i><b>1.7</b> Importing Packages</a></li>
<li class="chapter" data-level="1.8" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#setup-create-project"><i class="fa fa-check"></i><b>1.8</b> Create an R Studio Project for this textbook</a></li>
<li class="chapter" data-level="1.9" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#working-on-a-project-in-a-nutshell"><i class="fa fa-check"></i><b>1.9</b> Working on a project, in a nutshell</a></li>
<li class="chapter" data-level="1.10" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-an-r-markdown-file-for-this-chapter"><i class="fa fa-check"></i><b>1.10</b> Create an R Markdown file for this Chapter</a>
<ul>
<li class="chapter" data-level="1.10.1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#modify-the-yaml-header"><i class="fa fa-check"></i><b>1.10.1</b> Modify the yaml header</a></li>
<li class="chapter" data-level="1.10.2" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#modify-the-setup-chunk"><i class="fa fa-check"></i><b>1.10.2</b> Modify the “setup” chunk</a></li>
<li class="chapter" data-level="1.10.3" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-a-fake-data-chunk"><i class="fa fa-check"></i><b>1.10.3</b> Create a “fake-data” chunk</a></li>
<li class="chapter" data-level="1.10.4" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-a-plot-chunk"><i class="fa fa-check"></i><b>1.10.4</b> Create a “plot” chunk</a></li>
<li class="chapter" data-level="1.10.5" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#knit"><i class="fa fa-check"></i><b>1.10.5</b> Knit</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-ii-an-introduction-to-the-analysis-of-experimental-data-with-a-linear-model.html"><a href="part-ii-an-introduction-to-the-analysis-of-experimental-data-with-a-linear-model.html"><i class="fa fa-check"></i>Part II: An introduction to the analysis of experimental data with a linear model</a></li>
<li class="chapter" data-level="2" data-path="analyzing-experimental-data-with-a-linear-model.html"><a href="analyzing-experimental-data-with-a-linear-model.html"><i class="fa fa-check"></i><b>2</b> Analyzing experimental data with a linear model</a>
<ul>
<li class="chapter" data-level="2.1" data-path="analyzing-experimental-data-with-a-linear-model.html"><a href="analyzing-experimental-data-with-a-linear-model.html#this-text-is-about-using-linear-models-to-estimate-treatment-effects-and-the-uncertainty-in-our-estimates.-this-raises-the-question-what-is-an-effect"><i class="fa fa-check"></i><b>2.1</b> This text is about using linear models to estimate treatment effects and the uncertainty in our estimates. This, raises the question, what is “an effect?”</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="ask1-bio.html"><a href="ask1-bio.html"><i class="fa fa-check"></i><b>3</b> Background physiology to the experiments in Figure 2 of “ASK1 inhibits browning of white adipose tissue in obesity”</a></li>
<li class="chapter" data-level="4" data-path="ask1-report.html"><a href="ask1-report.html"><i class="fa fa-check"></i><b>4</b> Analyses for Figure 2 of “ASK1 inhibits browning of white adipose tissue in obesity”</a>
<ul>
<li class="chapter" data-level="4.1" data-path="ask1-report.html"><a href="ask1-report.html#setup"><i class="fa fa-check"></i><b>4.1</b> Setup</a></li>
<li class="chapter" data-level="4.2" data-path="ask1-report.html"><a href="ask1-report.html#data-source"><i class="fa fa-check"></i><b>4.2</b> Data source</a></li>
<li class="chapter" data-level="4.3" data-path="ask1-report.html"><a href="ask1-report.html#control-the-color-palette"><i class="fa fa-check"></i><b>4.3</b> control the color palette</a></li>
<li class="chapter" data-level="4.4" data-path="ask1-report.html"><a href="ask1-report.html#useful-functions"><i class="fa fa-check"></i><b>4.4</b> useful functions</a></li>
<li class="chapter" data-level="4.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2b-effect-of-ask1-deletion-on-growth-body-weight"><i class="fa fa-check"></i><b>4.5</b> figure 2b – effect of ASK1 deletion on growth (body weight)</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2b-import"><i class="fa fa-check"></i><b>4.5.1</b> figure 2b – import</a></li>
<li class="chapter" data-level="4.5.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2b-exploratory-plots"><i class="fa fa-check"></i><b>4.5.2</b> figure 2b – exploratory plots</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-effect-of-ask1-deletion-on-final-body-weight"><i class="fa fa-check"></i><b>4.6</b> Figure 2c – Effect of ASK1 deletion on final body weight</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-import"><i class="fa fa-check"></i><b>4.6.1</b> Figure 2c – import</a></li>
<li class="chapter" data-level="4.6.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-check-own-computation-of-weight-change-v-imported-value"><i class="fa fa-check"></i><b>4.6.2</b> Figure 2c – check own computation of weight change v imported value</a></li>
<li class="chapter" data-level="4.6.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-exploratory-plots"><i class="fa fa-check"></i><b>4.6.3</b> Figure 2c – exploratory plots</a></li>
<li class="chapter" data-level="4.6.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-fit-the-model-m1-lm"><i class="fa fa-check"></i><b>4.6.4</b> Figure 2c – fit the model: m1 (lm)</a></li>
<li class="chapter" data-level="4.6.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-check-the-model-m1"><i class="fa fa-check"></i><b>4.6.5</b> Figure 2c – check the model: m1</a></li>
<li class="chapter" data-level="4.6.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-fit-the-model-m2-gamma-glm"><i class="fa fa-check"></i><b>4.6.6</b> Figure 2c – fit the model: m2 (gamma glm)</a></li>
<li class="chapter" data-level="4.6.7" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-check-the-model-m2"><i class="fa fa-check"></i><b>4.6.7</b> Figure 2c – check the model, m2</a></li>
<li class="chapter" data-level="4.6.8" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-inference-from-the-model"><i class="fa fa-check"></i><b>4.6.8</b> Figure 2c – inference from the model</a></li>
<li class="chapter" data-level="4.6.9" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-plot-the-model"><i class="fa fa-check"></i><b>4.6.9</b> Figure 2c – plot the model</a></li>
<li class="chapter" data-level="4.6.10" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-report"><i class="fa fa-check"></i><b>4.6.10</b> Figure 2c – report</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve"><i class="fa fa-check"></i><b>4.7</b> Figure 2d – Effect of ASK1 KO on glucose tolerance (whole curve)</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-import"><i class="fa fa-check"></i><b>4.7.1</b> Figure 2d – Import</a></li>
<li class="chapter" data-level="4.7.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-exploratory-plots"><i class="fa fa-check"></i><b>4.7.2</b> Figure 2d – exploratory plots</a></li>
<li class="chapter" data-level="4.7.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-fit-the-model"><i class="fa fa-check"></i><b>4.7.3</b> Figure 2d – fit the model</a></li>
<li class="chapter" data-level="4.7.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-check-the-model"><i class="fa fa-check"></i><b>4.7.4</b> Figure 2d – check the model</a></li>
<li class="chapter" data-level="4.7.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-inference"><i class="fa fa-check"></i><b>4.7.5</b> Figure 2d – inference</a></li>
<li class="chapter" data-level="4.7.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-plot-the-model"><i class="fa fa-check"></i><b>4.7.6</b> Figure 2d – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure"><i class="fa fa-check"></i><b>4.8</b> Figure 2e – Effect of ASK1 deletion on glucose tolerance (summary measure)</a>
<ul>
<li class="chapter" data-level="4.8.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-message-the-data"><i class="fa fa-check"></i><b>4.8.1</b> Figure 2e – message the data</a></li>
<li class="chapter" data-level="4.8.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-exploratory-plots"><i class="fa fa-check"></i><b>4.8.2</b> Figure 2e – exploratory plots</a></li>
<li class="chapter" data-level="4.8.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-fit-the-model"><i class="fa fa-check"></i><b>4.8.3</b> Figure 2e – fit the model</a></li>
<li class="chapter" data-level="4.8.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-check-the-model"><i class="fa fa-check"></i><b>4.8.4</b> Figure 2e – check the model</a></li>
<li class="chapter" data-level="4.8.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-inference-from-the-model"><i class="fa fa-check"></i><b>4.8.5</b> Figure 2e – inference from the model</a></li>
<li class="chapter" data-level="4.8.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-plot-the-model"><i class="fa fa-check"></i><b>4.8.6</b> Figure 2e – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate"><i class="fa fa-check"></i><b>4.9</b> Figure 2f – Effect of ASK1 deletion on glucose infusion rate</a>
<ul>
<li class="chapter" data-level="4.9.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-import"><i class="fa fa-check"></i><b>4.9.1</b> Figure 2f – import</a></li>
<li class="chapter" data-level="4.9.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-exploratory-plots"><i class="fa fa-check"></i><b>4.9.2</b> Figure 2f – exploratory plots</a></li>
<li class="chapter" data-level="4.9.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-fit-the-model"><i class="fa fa-check"></i><b>4.9.3</b> Figure 2f – fit the model</a></li>
<li class="chapter" data-level="4.9.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-check-the-model"><i class="fa fa-check"></i><b>4.9.4</b> Figure 2f – check the model</a></li>
<li class="chapter" data-level="4.9.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-inference"><i class="fa fa-check"></i><b>4.9.5</b> Figure 2f – inference</a></li>
<li class="chapter" data-level="4.9.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-plot-the-model"><i class="fa fa-check"></i><b>4.9.6</b> Figure 2f – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.10" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake"><i class="fa fa-check"></i><b>4.10</b> Figure 2g – Effect of ASK1 deletion on tissue-specific glucose uptake</a>
<ul>
<li class="chapter" data-level="4.10.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-import"><i class="fa fa-check"></i><b>4.10.1</b> Figure 2g – import</a></li>
<li class="chapter" data-level="4.10.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-exploratory-plots"><i class="fa fa-check"></i><b>4.10.2</b> Figure 2g – exploratory plots</a></li>
<li class="chapter" data-level="4.10.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-fit-the-model"><i class="fa fa-check"></i><b>4.10.3</b> Figure 2g – fit the model</a></li>
<li class="chapter" data-level="4.10.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-check-the-model"><i class="fa fa-check"></i><b>4.10.4</b> Figure 2g – check the model</a></li>
<li class="chapter" data-level="4.10.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-inference"><i class="fa fa-check"></i><b>4.10.5</b> Figure 2g – inference</a></li>
<li class="chapter" data-level="4.10.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-plot-the-model"><i class="fa fa-check"></i><b>4.10.6</b> Figure 2g – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="ask1-report.html"><a href="ask1-report.html#figure-2h"><i class="fa fa-check"></i><b>4.11</b> Figure 2h</a></li>
<li class="chapter" data-level="4.12" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-effect-of-ask1-deletion-on-liver-tg"><i class="fa fa-check"></i><b>4.12</b> Figure 2i – Effect of ASK1 deletion on liver TG</a>
<ul>
<li class="chapter" data-level="4.12.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-fit-the-model"><i class="fa fa-check"></i><b>4.12.1</b> Figure 2i – fit the model</a></li>
<li class="chapter" data-level="4.12.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-check-the-model"><i class="fa fa-check"></i><b>4.12.2</b> Figure 2i – check the model</a></li>
<li class="chapter" data-level="4.12.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-inference"><i class="fa fa-check"></i><b>4.12.3</b> Figure 2i – inference</a></li>
<li class="chapter" data-level="4.12.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-plot-the-model"><i class="fa fa-check"></i><b>4.12.4</b> Figure 2i – plot the model</a></li>
<li class="chapter" data-level="4.12.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-report-the-model"><i class="fa fa-check"></i><b>4.12.5</b> Figure 2i – report the model</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="ask1-report.html"><a href="ask1-report.html#figure-2j"><i class="fa fa-check"></i><b>4.13</b> Figure 2j</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-iii-r-fundamentals.html"><a href="part-iii-r-fundamentals.html"><i class="fa fa-check"></i>Part III: R fundamentals</a></li>
<li class="chapter" data-level="5" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html"><i class="fa fa-check"></i><b>5</b> Data – Reading, Wrangling, and Writing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#learning-from-this-chapter"><i class="fa fa-check"></i><b>5.1</b> Learning from this chapter</a></li>
<li class="chapter" data-level="5.2" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#data-working-in-r"><i class="fa fa-check"></i><b>5.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#importing-data"><i class="fa fa-check"></i><b>5.2.1</b> Importing data</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#data-wrangling"><i class="fa fa-check"></i><b>5.3</b> Data wrangling</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#reshaping-data-wide-to-long"><i class="fa fa-check"></i><b>5.3.1</b> Reshaping data – Wide to long</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#reshaping-data-transpose-turning-the-columns-into-rows"><i class="fa fa-check"></i><b>5.3.2</b> Reshaping data – Transpose (turning the columns into rows)</a></li>
<li class="chapter" data-level="5.3.3" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#combining-data"><i class="fa fa-check"></i><b>5.3.3</b> Combining data</a></li>
<li class="chapter" data-level="5.3.4" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#subsetting-data"><i class="fa fa-check"></i><b>5.3.4</b> Subsetting data</a></li>
<li class="chapter" data-level="5.3.5" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#wrangling-columns"><i class="fa fa-check"></i><b>5.3.5</b> Wrangling columns</a></li>
<li class="chapter" data-level="5.3.6" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#missing-data"><i class="fa fa-check"></i><b>5.3.6</b> Missing data</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#saving-data"><i class="fa fa-check"></i><b>5.4</b> Saving data</a></li>
<li class="chapter" data-level="5.5" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#exercises"><i class="fa fa-check"></i><b>5.5</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="plotting-models.html"><a href="plotting-models.html"><i class="fa fa-check"></i><b>6</b> Plotting Models</a>
<ul>
<li class="chapter" data-level="6.1" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plots-show-the-model-and-the-data"><i class="fa fa-check"></i><b>6.1</b> Pretty good plots show the model and the data</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plot-component-1-modeled-effects-plot"><i class="fa fa-check"></i><b>6.1.1</b> Pretty good plot component 1: Modeled effects plot</a></li>
<li class="chapter" data-level="6.1.2" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plot-component-2-modeled-mean-and-ci-plot"><i class="fa fa-check"></i><b>6.1.2</b> Pretty good plot component 2: Modeled mean and CI plot</a></li>
<li class="chapter" data-level="6.1.3" data-path="plotting-models.html"><a href="plotting-models.html#combining-effects-and-modeled-mean-and-ci-plots-an-effects-and-response-plot."><i class="fa fa-check"></i><b>6.1.3</b> Combining Effects and Modeled mean and CI plots – an Effects and response plot.</a></li>
<li class="chapter" data-level="6.1.4" data-path="plotting-models.html"><a href="plotting-models.html#some-comments-on-plot-components"><i class="fa fa-check"></i><b>6.1.4</b> Some comments on plot components</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="plotting-models.html"><a href="plotting-models.html#working-in-r"><i class="fa fa-check"></i><b>6.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="plotting-models.html"><a href="plotting-models.html#source-data"><i class="fa fa-check"></i><b>6.2.1</b> Source data</a></li>
<li class="chapter" data-level="6.2.2" data-path="plotting-models.html"><a href="plotting-models.html#how-to-plot-the-model"><i class="fa fa-check"></i><b>6.2.2</b> How to plot the model</a></li>
<li class="chapter" data-level="6.2.3" data-path="plotting-models.html"><a href="plotting-models.html#be-sure-ggplot_the_model-is-in-your-r-folder"><i class="fa fa-check"></i><b>6.2.3</b> Be sure ggplot_the_model is in your R folder</a></li>
<li class="chapter" data-level="6.2.4" data-path="plotting-models.html"><a href="plotting-models.html#how-to-use-the-plot-the-model-functions"><i class="fa fa-check"></i><b>6.2.4</b> How to use the Plot the Model functions</a></li>
<li class="chapter" data-level="6.2.5" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-a-response-plot-using-ggpubr"><i class="fa fa-check"></i><b>6.2.5</b> How to generate a Response Plot using ggpubr</a></li>
<li class="chapter" data-level="6.2.6" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-a-response-plot-with-a-grid-of-treatments-using-ggplot2"><i class="fa fa-check"></i><b>6.2.6</b> How to generate a Response Plot with a grid of treatments using ggplot2</a></li>
<li class="chapter" data-level="6.2.7" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-an-effects-plot"><i class="fa fa-check"></i><b>6.2.7</b> How to generate an Effects Plot</a></li>
<li class="chapter" data-level="6.2.8" data-path="plotting-models.html"><a href="plotting-models.html#how-to-combine-the-response-and-effects-plots"><i class="fa fa-check"></i><b>6.2.8</b> How to combine the response and effects plots</a></li>
<li class="chapter" data-level="6.2.9" data-path="plotting-models.html"><a href="plotting-models.html#how-to-add-the-interaction-effect-to-response-and-effects-plots"><i class="fa fa-check"></i><b>6.2.9</b> How to add the interaction effect to response and effects plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-iv-some-fundamentals-of-statistical-modeling.html"><a href="part-iv-some-fundamentals-of-statistical-modeling.html"><i class="fa fa-check"></i>Part IV: Some Fundamentals of Statistical Modeling</a></li>
<li class="chapter" data-level="7" data-path="uncertainty.html"><a href="uncertainty.html"><i class="fa fa-check"></i><b>7</b> Variability and Uncertainty (Standard Deviations, Standard Errors, Confidence Intervals)</a>
<ul>
<li class="chapter" data-level="7.1" data-path="uncertainty.html"><a href="uncertainty.html#the-sample-standard-deviation-vs.-the-standard-error-of-the-mean"><i class="fa fa-check"></i><b>7.1</b> The sample standard deviation vs. the standard error of the mean</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="uncertainty.html"><a href="uncertainty.html#sample-standard-deviation"><i class="fa fa-check"></i><b>7.1.1</b> Sample standard deviation</a></li>
<li class="chapter" data-level="7.1.2" data-path="uncertainty.html"><a href="uncertainty.html#standard-error-of-the-mean"><i class="fa fa-check"></i><b>7.1.2</b> Standard error of the mean</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="uncertainty.html"><a href="uncertainty.html#using-google-sheets-to-generate-fake-data-to-explore-the-standard-error"><i class="fa fa-check"></i><b>7.2</b> Using Google Sheets to generate fake data to explore the standard error</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="uncertainty.html"><a href="uncertainty.html#steps"><i class="fa fa-check"></i><b>7.2.1</b> Steps</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="uncertainty.html"><a href="uncertainty.html#using-r-to-generate-fake-data-to-explore-the-standard-error"><i class="fa fa-check"></i><b>7.3</b> Using R to generate fake data to explore the standard error</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="uncertainty.html"><a href="uncertainty.html#part-i"><i class="fa fa-check"></i><b>7.3.1</b> part I</a></li>
<li class="chapter" data-level="7.3.2" data-path="uncertainty.html"><a href="uncertainty.html#part-ii---means"><i class="fa fa-check"></i><b>7.3.2</b> part II - means</a></li>
<li class="chapter" data-level="7.3.3" data-path="uncertainty.html"><a href="uncertainty.html#part-iii---how-do-sd-and-se-change-as-sample-size-n-increases"><i class="fa fa-check"></i><b>7.3.3</b> part III - how do SD and SE change as sample size (n) increases?</a></li>
<li class="chapter" data-level="7.3.4" data-path="uncertainty.html"><a href="uncertainty.html#part-iv-generating-fake-data-with-for-loops"><i class="fa fa-check"></i><b>7.3.4</b> Part IV – Generating fake data with for-loops</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="uncertainty.html"><a href="uncertainty.html#bootstrap"><i class="fa fa-check"></i><b>7.4</b> Bootstrapped standard errors</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="uncertainty.html"><a href="uncertainty.html#an-example-of-bootstrapped-standard-errors-using-vole-data"><i class="fa fa-check"></i><b>7.4.1</b> An example of bootstrapped standard errors using vole data</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="uncertainty.html"><a href="uncertainty.html#confidence-interval"><i class="fa fa-check"></i><b>7.5</b> Confidence Interval</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="uncertainty.html"><a href="uncertainty.html#interpretation-of-a-confidence-interval"><i class="fa fa-check"></i><b>7.5.1</b> Interpretation of a confidence interval</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="p-values.html"><a href="p-values.html"><i class="fa fa-check"></i><b>8</b> P-values</a>
<ul>
<li class="chapter" data-level="8.1" data-path="p-values.html"><a href="p-values.html#a-p-value-is-the-probability-of-sampling-a-value-as-or-more-extreme-than-the-test-statistic-if-sampling-from-a-null-distribution"><i class="fa fa-check"></i><b>8.1</b> A <em>p</em>-value is the probability of sampling a value as or more extreme than the test statistic if sampling from a null distribution</a></li>
<li class="chapter" data-level="8.2" data-path="p-values.html"><a href="p-values.html#pump-your-intuition-creating-a-null-distribution"><i class="fa fa-check"></i><b>8.2</b> Pump your intuition – Creating a null distribution</a></li>
<li class="chapter" data-level="8.3" data-path="p-values.html"><a href="p-values.html#a-null-distribution-of-t-values-the-t-distribution"><i class="fa fa-check"></i><b>8.3</b> A null distribution of <em>t</em>-values – the <em>t</em> distribution</a></li>
<li class="chapter" data-level="8.4" data-path="p-values.html"><a href="p-values.html#p-values-from-the-perspective-of-permutation"><i class="fa fa-check"></i><b>8.4</b> P-values from the perspective of permutation</a></li>
<li class="chapter" data-level="8.5" data-path="p-values.html"><a href="p-values.html#parametric-vs.-non-parametric-statistics"><i class="fa fa-check"></i><b>8.5</b> Parametric vs. non-parametric statistics</a></li>
<li class="chapter" data-level="8.6" data-path="p-values.html"><a href="p-values.html#frequentist-probability-and-the-interpretation-of-p-values"><i class="fa fa-check"></i><b>8.6</b> frequentist probability and the interpretation of p-values</a>
<ul>
<li class="chapter" data-level="8.6.1" data-path="p-values.html"><a href="p-values.html#background"><i class="fa fa-check"></i><b>8.6.1</b> Background</a></li>
<li class="chapter" data-level="8.6.2" data-path="p-values.html"><a href="p-values.html#this-book-covers-frequentist-approaches-to-statistical-modeling-and-when-a-probability-arises-such-as-the-p-value-of-a-test-statistic-this-will-be-a-frequentist-probability."><i class="fa fa-check"></i><b>8.6.2</b> This book covers frequentist approaches to statistical modeling and when a probability arises, such as the <em>p</em>-value of a test statistic, this will be a frequentist probability.</a></li>
<li class="chapter" data-level="8.6.3" data-path="p-values.html"><a href="p-values.html#two-interpretations-of-the-p-value"><i class="fa fa-check"></i><b>8.6.3</b> Two interpretations of the <em>p</em>-value</a></li>
<li class="chapter" data-level="8.6.4" data-path="p-values.html"><a href="p-values.html#nhst"><i class="fa fa-check"></i><b>8.6.4</b> NHST</a></li>
</ul></li>
<li class="chapter" data-level="8.7" data-path="p-values.html"><a href="p-values.html#some-major-misconceptions-of-the-p-value"><i class="fa fa-check"></i><b>8.7</b> Some major misconceptions of the <em>p</em>-value</a>
<ul>
<li class="chapter" data-level="8.7.1" data-path="p-values.html"><a href="p-values.html#misconception-p-is-the-probability-that-the-null-is-true-and-1-p-is-probability-that-the-alternative-is-true"><i class="fa fa-check"></i><b>8.7.1</b> Misconception: <em>p</em> is the probability that the null is true <em>and</em> <span class="math inline">\(1-p\)</span> is probability that the alternative is true</a></li>
<li class="chapter" data-level="8.7.2" data-path="p-values.html"><a href="p-values.html#misconception-a-p-value-is-repeatable"><i class="fa fa-check"></i><b>8.7.2</b> Misconception: a <em>p</em>-value is repeatable</a></li>
<li class="chapter" data-level="8.7.3" data-path="p-values.html"><a href="p-values.html#misconception-0.05-is-the-lifetime-rate-of-false-discoveries"><i class="fa fa-check"></i><b>8.7.3</b> Misconception: 0.05 is the lifetime rate of false discoveries</a></li>
<li class="chapter" data-level="8.7.4" data-path="p-values.html"><a href="p-values.html#misconception-a-low-p-value-indicates-an-important-effect"><i class="fa fa-check"></i><b>8.7.4</b> Misconception: a low <em>p</em>-value indicates an important effect</a></li>
<li class="chapter" data-level="8.7.5" data-path="p-values.html"><a href="p-values.html#misconception-a-low-p-value-indicates-high-model-fit-or-high-predictive-capacity"><i class="fa fa-check"></i><b>8.7.5</b> Misconception: a low <em>p</em>-value indicates high model fit or high predictive capacity</a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="p-values.html"><a href="p-values.html#what-the-p-value-does-not-mean"><i class="fa fa-check"></i><b>8.8</b> What the <em>p</em>-value does not mean</a></li>
<li class="chapter" data-level="8.9" data-path="p-values.html"><a href="p-values.html#recommendations"><i class="fa fa-check"></i><b>8.9</b> Recommendations</a>
<ul>
<li class="chapter" data-level="8.9.1" data-path="p-values.html"><a href="p-values.html#primary-sources-for-recommendations"><i class="fa fa-check"></i><b>8.9.1</b> Primary sources for recommendations</a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="p-values.html"><a href="p-values.html#problems"><i class="fa fa-check"></i><b>8.10</b> Problems</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="errors-in-inference.html"><a href="errors-in-inference.html"><i class="fa fa-check"></i><b>9</b> Errors in inference</a>
<ul>
<li class="chapter" data-level="9.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#classical-nhst-concepts-of-wrong"><i class="fa fa-check"></i><b>9.1</b> Classical NHST concepts of wrong</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-i-error"><i class="fa fa-check"></i><b>9.1.1</b> Type I error</a></li>
<li class="chapter" data-level="9.1.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#power"><i class="fa fa-check"></i><b>9.1.2</b> Power</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#a-non-neyman-pearson-concept-of-power"><i class="fa fa-check"></i><b>9.2</b> A non-Neyman-Pearson concept of power</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#estimation-error"><i class="fa fa-check"></i><b>9.2.1</b> Estimation error</a></li>
<li class="chapter" data-level="9.2.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#coverage"><i class="fa fa-check"></i><b>9.2.2</b> Coverage</a></li>
<li class="chapter" data-level="9.2.3" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-s-error"><i class="fa fa-check"></i><b>9.2.3</b> Type S error</a></li>
<li class="chapter" data-level="9.2.4" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-m-error"><i class="fa fa-check"></i><b>9.2.4</b> Type M error</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-v-introduction-to-linear-models.html"><a href="part-v-introduction-to-linear-models.html"><i class="fa fa-check"></i>Part V: Introduction to Linear Models</a></li>
<li class="chapter" data-level="10" data-path="intro-linear-models.html"><a href="intro-linear-models.html"><i class="fa fa-check"></i><b>10</b> An introduction to linear models</a>
<ul>
<li class="chapter" data-level="10.1" data-path="intro-linear-models.html"><a href="intro-linear-models.html#lm-specifications"><i class="fa fa-check"></i><b>10.1</b> Two specifications of a linear model</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="intro-linear-models.html"><a href="intro-linear-models.html#the-error-draw-specification"><i class="fa fa-check"></i><b>10.1.1</b> The “error draw” specification</a></li>
<li class="chapter" data-level="10.1.2" data-path="intro-linear-models.html"><a href="intro-linear-models.html#the-conditional-draw-specification"><i class="fa fa-check"></i><b>10.1.2</b> The “conditional draw” specification</a></li>
<li class="chapter" data-level="10.1.3" data-path="intro-linear-models.html"><a href="intro-linear-models.html#comparing-the-error-draw-and-conditional-draw-ways-of-specifying-the-linear-model"><i class="fa fa-check"></i><b>10.1.3</b> Comparing the error-draw and conditional-draw ways of specifying the linear model</a></li>
<li class="chapter" data-level="10.1.4" data-path="intro-linear-models.html"><a href="intro-linear-models.html#anova-notation-of-a-linear-model"><i class="fa fa-check"></i><b>10.1.4</b> ANOVA notation of a linear model</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="intro-linear-models.html"><a href="intro-linear-models.html#a-linear-model-can-be-fit-to-data-with-continuous-discrete-or-categorical-x-variables"><i class="fa fa-check"></i><b>10.2</b> A linear model can be fit to data with continuous, discrete, or categorical <span class="math inline">\(X\)</span> variables</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="intro-linear-models.html"><a href="intro-linear-models.html#fitting-linear-models-to-experimental-data-in-which-the-x-variable-is-continuous-or-discrete"><i class="fa fa-check"></i><b>10.2.1</b> Fitting linear models to experimental data in which the <span class="math inline">\(X\)</span> variable is continuous or discrete</a></li>
<li class="chapter" data-level="10.2.2" data-path="intro-linear-models.html"><a href="intro-linear-models.html#slope"><i class="fa fa-check"></i><b>10.2.2</b> Fitting linear models to experimental data in which the <span class="math inline">\(X\)</span> variable is categorical</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="intro-linear-models.html"><a href="intro-linear-models.html#statistical-models-are-used-for-prediction-explanation-and-description"><i class="fa fa-check"></i><b>10.3</b> Statistical models are used for prediction, explanation, and description</a></li>
<li class="chapter" data-level="10.4" data-path="intro-linear-models.html"><a href="intro-linear-models.html#what-do-we-call-the-x-and-y-variables"><i class="fa fa-check"></i><b>10.4</b> What do we call the <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span> variables?</a></li>
<li class="chapter" data-level="10.5" data-path="intro-linear-models.html"><a href="intro-linear-models.html#modeling-strategy"><i class="fa fa-check"></i><b>10.5</b> Modeling strategy</a></li>
<li class="chapter" data-level="10.6" data-path="intro-linear-models.html"><a href="intro-linear-models.html#predictions-from-the-model"><i class="fa fa-check"></i><b>10.6</b> Predictions from the model</a></li>
<li class="chapter" data-level="10.7" data-path="intro-linear-models.html"><a href="intro-linear-models.html#inference-from-the-model"><i class="fa fa-check"></i><b>10.7</b> Inference from the model</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="intro-linear-models.html"><a href="intro-linear-models.html#assumptions-for-inference-with-a-statistical-model"><i class="fa fa-check"></i><b>10.7.1</b> Assumptions for inference with a statistical model</a></li>
<li class="chapter" data-level="10.7.2" data-path="intro-linear-models.html"><a href="intro-linear-models.html#specific-assumptions-for-inference-with-a-linear-model"><i class="fa fa-check"></i><b>10.7.2</b> Specific assumptions for inference with a linear model</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="intro-linear-models.html"><a href="intro-linear-models.html#linear-modelregression-model-orstatistical-model"><i class="fa fa-check"></i><b>10.8</b> “linear model,”regression model“, or”statistical model"?</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="regression.html"><a href="regression.html"><i class="fa fa-check"></i><b>11</b> Linear models with a single, continuous <em>X</em> (“regression”)</a>
<ul>
<li class="chapter" data-level="11.1" data-path="regression.html"><a href="regression.html#a-linear-model-with-a-single-continuous-x-is-classical-regression"><i class="fa fa-check"></i><b>11.1</b> A linear model with a single, continuous <em>X</em> is classical “regression”</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="regression.html"><a href="regression.html#analysis-of-green-down-data"><i class="fa fa-check"></i><b>11.1.1</b> Analysis of “green-down” data</a></li>
<li class="chapter" data-level="11.1.2" data-path="regression.html"><a href="regression.html#learning-from-the-green-down-example"><i class="fa fa-check"></i><b>11.1.2</b> Learning from the green-down example</a></li>
<li class="chapter" data-level="11.1.3" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-explanation-causal-models"><i class="fa fa-check"></i><b>11.1.3</b> Using a regression model for “explanation” – causal models</a></li>
<li class="chapter" data-level="11.1.4" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-prediction-prediction-models"><i class="fa fa-check"></i><b>11.1.4</b> Using a regression model for prediction – prediction models</a></li>
<li class="chapter" data-level="11.1.5" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-creating-a-new-response-variable-comparing-slopes-of-longitudinal-data"><i class="fa fa-check"></i><b>11.1.5</b> Using a regression model for creating a new response variable – comparing slopes of longitudinal data</a></li>
<li class="chapter" data-level="11.1.6" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-for-calibration"><i class="fa fa-check"></i><b>11.1.6</b> Using a regression model for for calibration</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="regression.html"><a href="regression.html#working-in-r-1"><i class="fa fa-check"></i><b>11.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="plotting-models.html"><a href="plotting-models.html#fit-the-model"><i class="fa fa-check"></i><b>11.2.1</b> Fitting the linear model</a></li>
<li class="chapter" data-level="11.2.2" data-path="regression.html"><a href="regression.html#getting-to-know-the-linear-model-the-summary-function"><i class="fa fa-check"></i><b>11.2.2</b> Getting to know the linear model: the <code>summary</code> function</a></li>
<li class="chapter" data-level="11.2.3" data-path="regression.html"><a href="regression.html#inference-the-coefficient-table"><i class="fa fa-check"></i><b>11.2.3</b> Inference – the coefficient table</a></li>
<li class="chapter" data-level="11.2.4" data-path="regression.html"><a href="regression.html#how-good-is-our-model-model-checking"><i class="fa fa-check"></i><b>11.2.4</b> How good is our model? – Model checking</a></li>
<li class="chapter" data-level="11.2.5" data-path="regression.html"><a href="regression.html#plotting-models-with-continuous-x"><i class="fa fa-check"></i><b>11.2.5</b> Plotting models with continuous <em>X</em></a></li>
<li class="chapter" data-level="11.2.6" data-path="regression.html"><a href="regression.html#creating-a-table-of-predicted-values-and-95-prediction-intervals"><i class="fa fa-check"></i><b>11.2.6</b> Creating a table of predicted values and 95% prediction intervals</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="regression.html"><a href="regression.html#hidden-code"><i class="fa fa-check"></i><b>11.3</b> Hidden code</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="regression.html"><a href="regression.html#import-and-plot-of-fig2c-ecosystem-warming-experimental-data"><i class="fa fa-check"></i><b>11.3.1</b> Import and plot of fig2c (ecosystem warming experimental) data</a></li>
<li class="chapter" data-level="11.3.2" data-path="regression.html"><a href="regression.html#import-and-plot-efig_3d-ecosysem-warming-observational-data"><i class="fa fa-check"></i><b>11.3.2</b> Import and plot efig_3d (Ecosysem warming observational) data</a></li>
<li class="chapter" data-level="11.3.3" data-path="regression.html"><a href="regression.html#import-and-plot-of-fig1f-methionine-restriction-data"><i class="fa fa-check"></i><b>11.3.3</b> Import and plot of fig1f (methionine restriction) data</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="regression.html"><a href="regression.html#try-it"><i class="fa fa-check"></i><b>11.4</b> Try it</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="regression.html"><a href="regression.html#a-prediction-model-from-the-literature"><i class="fa fa-check"></i><b>11.4.1</b> A prediction model from the literature</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="regression.html"><a href="regression.html#intuition-pumps"><i class="fa fa-check"></i><b>11.5</b> Intuition pumps</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="regression.html"><a href="regression.html#correlation-and-r2"><i class="fa fa-check"></i><b>11.5.1</b> Correlation and $R^2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="oneway.html"><a href="oneway.html"><i class="fa fa-check"></i><b>12</b> Linear models with a single, categorical <em>X</em> (“t-tests” and “ANOVA”)</a>
<ul>
<li class="chapter" data-level="12.1" data-path="oneway.html"><a href="oneway.html#oneway-data"><i class="fa fa-check"></i><b>12.1</b> A linear model with a single, categorical <em>X</em> variable estimates the effects of the levels of <em>X</em> on the response</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="oneway.html"><a href="oneway.html#oneway-example1"><i class="fa fa-check"></i><b>12.1.1</b> Example 1 (fig3d) – two treatment levels (“groups”)</a></li>
<li class="chapter" data-level="12.1.2" data-path="oneway.html"><a href="oneway.html#understanding-the-analysis-with-two-treatment-levels"><i class="fa fa-check"></i><b>12.1.2</b> Understanding the analysis with two treatment levels</a></li>
<li class="chapter" data-level="12.1.3" data-path="oneway.html"><a href="oneway.html#oneway-example2"><i class="fa fa-check"></i><b>12.1.3</b> Example 2 – three treatment levels (“groups”)</a></li>
<li class="chapter" data-level="12.1.4" data-path="oneway.html"><a href="oneway.html#understanding-the-analysis-with-three-or-more-treatment-levels"><i class="fa fa-check"></i><b>12.1.4</b> Understanding the analysis with three (or more) treatment levels</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="oneway.html"><a href="oneway.html#working-in-r-2"><i class="fa fa-check"></i><b>12.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="oneway.html"><a href="oneway.html#fit-the-model-1"><i class="fa fa-check"></i><b>12.2.1</b> Fit the model</a></li>
<li class="chapter" data-level="12.2.2" data-path="oneway.html"><a href="oneway.html#controlling-the-output-in-tables-using-the-coefficient-table-as-an-example"><i class="fa fa-check"></i><b>12.2.2</b> Controlling the output in tables using the coefficient table as an example</a></li>
<li class="chapter" data-level="12.2.3" data-path="oneway.html"><a href="oneway.html#using-the-emmeans-function"><i class="fa fa-check"></i><b>12.2.3</b> Using the emmeans function</a></li>
<li class="chapter" data-level="12.2.4" data-path="oneway.html"><a href="oneway.html#using-the-contrast-function"><i class="fa fa-check"></i><b>12.2.4</b> Using the contrast function</a></li>
<li class="chapter" data-level="12.2.5" data-path="oneway.html"><a href="oneway.html#how-to-generate-anova-tables"><i class="fa fa-check"></i><b>12.2.5</b> How to generate ANOVA tables</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="oneway.html"><a href="oneway.html#hidden-code-1"><i class="fa fa-check"></i><b>12.3</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="oneway.html"><a href="oneway.html#importing-and-wrangling-the-fig_3d-data-for-example-1"><i class="fa fa-check"></i><b>12.3.1</b> Importing and wrangling the fig_3d data for example 1</a></li>
<li class="chapter" data-level="12.3.2" data-path="oneway.html"><a href="oneway.html#importing-and-wrangling-the-fig2a-data-for-example-2"><i class="fa fa-check"></i><b>12.3.2</b> Importing and wrangling the fig2a data for example 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="model-checking.html"><a href="model-checking.html"><i class="fa fa-check"></i><b>13</b> Model Checking</a>
<ul>
<li class="chapter" data-level="13.1" data-path="model-checking.html"><a href="model-checking.html#check-check"><i class="fa fa-check"></i><b>13.1</b> All statistical analyses should be followed by model checking</a></li>
<li class="chapter" data-level="13.2" data-path="model-checking.html"><a href="model-checking.html#linear-model-assumptions"><i class="fa fa-check"></i><b>13.2</b> Linear model assumptions</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="model-checking.html"><a href="model-checking.html#a-bit-about-iid"><i class="fa fa-check"></i><b>13.2.1</b> A bit about IID</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="model-checking.html"><a href="model-checking.html#diagnostic-plots-use-the-residuals-from-the-model-fit"><i class="fa fa-check"></i><b>13.3</b> Diagnostic plots use the residuals from the model fit</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="model-checking.html"><a href="model-checking.html#residuals"><i class="fa fa-check"></i><b>13.3.1</b> Residuals</a></li>
<li class="chapter" data-level="13.3.2" data-path="model-checking.html"><a href="model-checking.html#normal-qq"><i class="fa fa-check"></i><b>13.3.2</b> A Normal Q-Q plot is used to check for characteristic departures from Normality</a></li>
<li class="chapter" data-level="13.3.3" data-path="model-checking.html"><a href="model-checking.html#mapping-qq-plot-departures-from-normality"><i class="fa fa-check"></i><b>13.3.3</b> Mapping QQ-plot departures from Normality</a></li>
<li class="chapter" data-level="13.3.4" data-path="model-checking.html"><a href="model-checking.html#model-checking-spread-level"><i class="fa fa-check"></i><b>13.3.4</b> Model checking homoskedasticity</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="model-checking.html"><a href="model-checking.html#using-r"><i class="fa fa-check"></i><b>13.4</b> Using R</a></li>
<li class="chapter" data-level="13.5" data-path="model-checking.html"><a href="model-checking.html#hidden-code-2"><i class="fa fa-check"></i><b>13.5</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="model-checking.html"><a href="model-checking.html#normal-q-q-plots"><i class="fa fa-check"></i><b>13.5.1</b> Normal Q-Q plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="violations.html"><a href="violations.html"><i class="fa fa-check"></i><b>14</b> Violations of independence, homogeneity, or Normality</a>
<ul>
<li class="chapter" data-level="14.1" data-path="violations.html"><a href="violations.html#oneway-paired-t"><i class="fa fa-check"></i><b>14.1</b> Lack of independence</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="violations.html"><a href="violations.html#violations-paired-t"><i class="fa fa-check"></i><b>14.1.1</b> Example 1 (exp1b) – a paired t-test is a special case of a linear mixed model</a></li>
<li class="chapter" data-level="14.1.2" data-path="violations.html"><a href="violations.html#violations-rmanova"><i class="fa fa-check"></i><b>14.1.2</b> Example 2 (diHOME exp2a) – A repeated measures ANOVA is a special case of a linear mixed model</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="violations.html"><a href="violations.html#oneway-welch"><i class="fa fa-check"></i><b>14.2</b> Heterogeneity of variances</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="violations.html"><a href="violations.html#when-groups-of-the-focal-test-have-variance"><i class="fa fa-check"></i><b>14.2.1</b> When groups of the focal test have &gt;&gt; variance</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="violations.html"><a href="violations.html#the-conditional-response-isnt-normal"><i class="fa fa-check"></i><b>14.3</b> The conditional response isn’t Normal</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="violations.html"><a href="violations.html#violations-6f"><i class="fa fa-check"></i><b>14.3.1</b> Example 1 (fig6f) – Linear models for non-normal count data</a></li>
<li class="chapter" data-level="14.3.2" data-path="violations.html"><a href="violations.html#my-data-arent-normal-what-is-the-best-practice"><i class="fa fa-check"></i><b>14.3.2</b> My data aren’t normal, what is the best practice?</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="violations.html"><a href="violations.html#hidden-code-3"><i class="fa fa-check"></i><b>14.4</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-exp1b-data"><i class="fa fa-check"></i><b>14.4.1</b> Importing and wrangling the exp1b data</a></li>
<li class="chapter" data-level="14.4.2" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-exp2a-data"><i class="fa fa-check"></i><b>14.4.2</b> Importing and wrangling the exp2a data</a></li>
<li class="chapter" data-level="14.4.3" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-fig6f-data"><i class="fa fa-check"></i><b>14.4.3</b> Importing and wrangling the fig6f data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="issues.html"><a href="issues.html"><i class="fa fa-check"></i><b>15</b> Issues in inference</a>
<ul>
<li class="chapter" data-level="15.1" data-path="issues.html"><a href="issues.html#replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means"><i class="fa fa-check"></i><b>15.1</b> Replicated experiments – include <span class="math inline">\(\texttt{Experiment}\)</span> as a random factor (better than one-way ANOVA of means)</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="issues.html"><a href="issues.html#issues-exp4d"><i class="fa fa-check"></i><b>15.1.1</b> Multiple experiments Example 1 (wound healing Exp4d)</a></li>
<li class="chapter" data-level="15.1.2" data-path="issues.html"><a href="issues.html#issues-combined-experiments"><i class="fa fa-check"></i><b>15.1.2</b> Models for combining replicated experiments</a></li>
<li class="chapter" data-level="15.1.3" data-path="issues.html"><a href="issues.html#understanding-model-exp4d_m1"><i class="fa fa-check"></i><b>15.1.3</b> Understanding Model <code>exp4d_m1</code></a></li>
<li class="chapter" data-level="15.1.4" data-path="issues.html"><a href="issues.html#the-univariate-model-is-equivalent-to-a-linear-mixed-model-of-the-aggregated-data-model-exp4d_m2"><i class="fa fa-check"></i><b>15.1.4</b> The univariate model is equivalent to a linear mixed model of the aggregated data (Model exp4d_m2)</a></li>
<li class="chapter" data-level="15.1.5" data-path="issues.html"><a href="issues.html#a-linear-mixed-model-of-the-full-data"><i class="fa fa-check"></i><b>15.1.5</b> A linear mixed model of the full data</a></li>
<li class="chapter" data-level="15.1.6" data-path="issues.html"><a href="issues.html#analysis-of-the-experiment-means-has-less-precision-and-power"><i class="fa fa-check"></i><b>15.1.6</b> Analysis of the experiment means has less precision and power</a></li>
<li class="chapter" data-level="15.1.7" data-path="issues.html"><a href="issues.html#dont-do-this-a-t-testfixed-effect-anova-of-the-full-data"><i class="fa fa-check"></i><b>15.1.7</b> Don’t do this – a t-test/fixed-effect ANOVA of the full data</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="issues.html"><a href="issues.html#issues-pre-post"><i class="fa fa-check"></i><b>15.2</b> Comparing change from baseline (pre-post)</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="issues.html"><a href="issues.html#pre-post-example-1-dpp4-fig4c"><i class="fa fa-check"></i><b>15.2.1</b> Pre-post example 1 (DPP4 fig4c)</a></li>
<li class="chapter" data-level="15.2.2" data-path="issues.html"><a href="issues.html#what-if-the-data-in-example-1-were-from-from-an-experiment-where-the-treatment-was-applied-prior-to-the-baseline-measure"><i class="fa fa-check"></i><b>15.2.2</b> What if the data in example 1 were from from an experiment where the treatment was applied prior to the baseline measure?</a></li>
<li class="chapter" data-level="15.2.3" data-path="issues.html"><a href="issues.html#pre-post-example-2-xx-males-fig1c"><i class="fa fa-check"></i><b>15.2.3</b> Pre-post example 2 (XX males fig1c)</a></li>
<li class="chapter" data-level="15.2.4" data-path="issues.html"><a href="issues.html#issues-regression-to-mean"><i class="fa fa-check"></i><b>15.2.4</b> Regression to the mean</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="issues.html"><a href="issues.html#longitudinal-designs-with-more-than-one-post-baseline-measure"><i class="fa fa-check"></i><b>15.3</b> Longitudinal designs with more than one-post baseline measure</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="issues.html"><a href="issues.html#issues-auc"><i class="fa fa-check"></i><b>15.3.1</b> Area under the curve (AUC)</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="issues.html"><a href="issues.html#normalization-the-analysis-of-ratios"><i class="fa fa-check"></i><b>15.4</b> Normalization – the analysis of ratios</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="issues.html"><a href="issues.html#kinds-of-ratios-in-experimental-biology"><i class="fa fa-check"></i><b>15.4.1</b> Kinds of ratios in experimental biology</a></li>
<li class="chapter" data-level="15.4.2" data-path="issues.html"><a href="issues.html#issues-offset"><i class="fa fa-check"></i><b>15.4.2</b> Example 1 – The ratio is a density (number of something per area)</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="issues.html"><a href="issues.html#dont-do-this-stuff"><i class="fa fa-check"></i><b>15.5</b> Don’t do this stuff</a>
<ul>
<li class="chapter" data-level="15.5.1" data-path="issues.html"><a href="issues.html#normalize-the-response-so-that-all-control-values-are-equal-to-1."><i class="fa fa-check"></i><b>15.5.1</b> Normalize the response so that all control values are equal to 1.</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="issues.html"><a href="issues.html#a-difference-in-significance-is-not-necessarily-significant"><i class="fa fa-check"></i><b>15.6</b> A difference in significance is not necessarily significant</a></li>
<li class="chapter" data-level="15.7" data-path="issues.html"><a href="issues.html#researcher-degrees-of-freedom"><i class="fa fa-check"></i><b>15.7</b> Researcher degrees of freedom</a></li>
<li class="chapter" data-level="15.8" data-path="issues.html"><a href="issues.html#hidden-code-4"><i class="fa fa-check"></i><b>15.8</b> Hidden code</a>
<ul>
<li class="chapter" data-level="15.8.1" data-path="issues.html"><a href="issues.html#import-exp4d-vimentin-cell-count-data-replicate-experiments-example"><i class="fa fa-check"></i><b>15.8.1</b> Import exp4d vimentin cell count data (replicate experiments example)</a></li>
<li class="chapter" data-level="15.8.2" data-path="issues.html"><a href="issues.html#import-fig4c-data"><i class="fa fa-check"></i><b>15.8.2</b> Import Fig4c data</a></li>
<li class="chapter" data-level="15.8.3" data-path="issues.html"><a href="issues.html#xx-males-fig1c"><i class="fa fa-check"></i><b>15.8.3</b> XX males fig1c</a></li>
<li class="chapter" data-level="15.8.4" data-path="issues.html"><a href="issues.html#generation-of-fake-data-to-illustrate-regression-to-the-mean"><i class="fa fa-check"></i><b>15.8.4</b> Generation of fake data to illustrate regression to the mean</a></li>
<li class="chapter" data-level="15.8.5" data-path="issues.html"><a href="issues.html#import-fig3f"><i class="fa fa-check"></i><b>15.8.5</b> Import fig3f</a></li>
<li class="chapter" data-level="15.8.6" data-path="issues.html"><a href="issues.html#issues-import-exp3b"><i class="fa fa-check"></i><b>15.8.6</b> Import exp3b</a></li>
<li class="chapter" data-level="15.8.7" data-path="issues.html"><a href="issues.html#issues-exp3b-plot"><i class="fa fa-check"></i><b>15.8.7</b> Plot the model of exp3b (glm offset data)</a></li>
</ul></li>
</ul></li>
<li><a href="part-vi-more-than-one-x-multivariable-models.html#part-vi-more-than-one-x-multivariable-models">Part VI: More than one <span class="math inline">\(X\)</span> – Multivariable Models</a></li>
<li class="chapter" data-level="16" data-path="covariates.html"><a href="covariates.html"><i class="fa fa-check"></i><b>16</b> Linear models with added covariates (“ANCOVA”)</a>
<ul>
<li class="chapter" data-level="16.1" data-path="covariates.html"><a href="covariates.html#adding-covariates-can-increases-the-precision-of-the-effect-of-interest"><i class="fa fa-check"></i><b>16.1</b> Adding covariates can increases the precision of the effect of interest</a></li>
<li class="chapter" data-level="16.2" data-path="covariates.html"><a href="covariates.html#understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data"><i class="fa fa-check"></i><b>16.2</b> Understanding a linear model with an added covariate – heart necrosis data</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="covariates.html"><a href="covariates.html#fit-the-model-5"><i class="fa fa-check"></i><b>16.2.1</b> Fit the model</a></li>
<li class="chapter" data-level="16.2.2" data-path="covariates.html"><a href="covariates.html#plot-the-model-2"><i class="fa fa-check"></i><b>16.2.2</b> Plot the model</a></li>
<li class="chapter" data-level="16.2.3" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-model-coefficients"><i class="fa fa-check"></i><b>16.2.3</b> Interpretation of the model coefficients</a></li>
<li class="chapter" data-level="16.2.4" data-path="covariates.html"><a href="covariates.html#everything-adds-up"><i class="fa fa-check"></i><b>16.2.4</b> Everything adds up</a></li>
<li class="chapter" data-level="16.2.5" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-estimated-marginal-means"><i class="fa fa-check"></i><b>16.2.5</b> Interpretation of the estimated marginal means</a></li>
<li class="chapter" data-level="16.2.6" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-contrasts"><i class="fa fa-check"></i><b>16.2.6</b> Interpretation of the contrasts</a></li>
<li class="chapter" data-level="16.2.7" data-path="covariates.html"><a href="covariates.html#adding-the-covariate-improves-inference"><i class="fa fa-check"></i><b>16.2.7</b> Adding the covariate improves inference</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="covariates.html"><a href="covariates.html#understanding-interaction-effects-with-covariates"><i class="fa fa-check"></i><b>16.3</b> Understanding interaction effects with covariates</a>
<ul>
<li class="chapter" data-level="16.3.1" data-path="covariates.html"><a href="covariates.html#fit-the-model-6"><i class="fa fa-check"></i><b>16.3.1</b> Fit the model</a></li>
<li class="chapter" data-level="16.3.2" data-path="covariates.html"><a href="covariates.html#plot-the-model-with-interaction-effect"><i class="fa fa-check"></i><b>16.3.2</b> Plot the model with interaction effect</a></li>
<li class="chapter" data-level="16.3.3" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-model-coefficients-1"><i class="fa fa-check"></i><b>16.3.3</b> Interpretation of the model coefficients</a></li>
<li class="chapter" data-level="16.3.4" data-path="covariates.html"><a href="covariates.html#what-is-the-effect-of-a-treatment-if-interactions-are-modeled-it-depends."><i class="fa fa-check"></i><b>16.3.4</b> What is the effect of a treatment, if interactions are modeled? – it depends.</a></li>
<li class="chapter" data-level="16.3.5" data-path="covariates.html"><a href="covariates.html#which-model-do-we-use-mathcalm_1-or-mathcalm_2"><i class="fa fa-check"></i><b>16.3.5</b> Which model do we use, <span class="math inline">\(\mathcal{M}_1\)</span> or <span class="math inline">\(\mathcal{M}_2\)</span>?</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="covariates.html"><a href="covariates.html#understanding-ancova-tables"><i class="fa fa-check"></i><b>16.4</b> Understanding ANCOVA tables</a></li>
<li class="chapter" data-level="16.5" data-path="covariates.html"><a href="covariates.html#working-in-r-3"><i class="fa fa-check"></i><b>16.5</b> Working in R</a>
<ul>
<li class="chapter" data-level="16.5.1" data-path="covariates.html"><a href="covariates.html#importing-the-heart-necrosis-data"><i class="fa fa-check"></i><b>16.5.1</b> Importing the heart necrosis data</a></li>
<li class="chapter" data-level="16.5.2" data-path="covariates.html"><a href="covariates.html#fitting-the-model"><i class="fa fa-check"></i><b>16.5.2</b> Fitting the model</a></li>
<li class="chapter" data-level="16.5.3" data-path="covariates.html"><a href="covariates.html#using-the-emmeans-function-1"><i class="fa fa-check"></i><b>16.5.3</b> Using the emmeans function</a></li>
<li class="chapter" data-level="16.5.4" data-path="covariates.html"><a href="covariates.html#ancova-tables"><i class="fa fa-check"></i><b>16.5.4</b> ANCOVA tables</a></li>
<li class="chapter" data-level="16.5.5" data-path="covariates.html"><a href="covariates.html#plotting-the-model"><i class="fa fa-check"></i><b>16.5.5</b> Plotting the model</a></li>
</ul></li>
<li class="chapter" data-level="16.6" data-path="covariates.html"><a href="covariates.html#best-practices"><i class="fa fa-check"></i><b>16.6</b> Best practices</a>
<ul>
<li class="chapter" data-level="16.6.1" data-path="covariates.html"><a href="covariates.html#do-not-use-a-ratio-of-partwhole-as-a-response-variable-instead-add-the-denominator-as-a-covariate"><i class="fa fa-check"></i><b>16.6.1</b> Do not use a ratio of part:whole as a response variable – instead add the denominator as a covariate</a></li>
<li class="chapter" data-level="16.6.2" data-path="covariates.html"><a href="covariates.html#do-not-use-change-from-baseline-as-a-response-variable-instead-add-the-baseline-measure-as-a-covariate"><i class="fa fa-check"></i><b>16.6.2</b> Do not use change from baseline as a response variable – instead add the baseline measure as a covariate</a></li>
<li class="chapter" data-level="16.6.3" data-path="covariates.html"><a href="covariates.html#do-not-test-for-balance-of-baseline-measures"><i class="fa fa-check"></i><b>16.6.3</b> Do not “test for balance” of baseline measures</a></li>
</ul></li>
<li class="chapter" data-level="16.7" data-path="covariates.html"><a href="covariates.html#best-practices-2-use-a-covariate-instead-of-normalizing-a-response"><i class="fa fa-check"></i><b>16.7</b> Best practices 2: Use a covariate instead of normalizing a response</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="factorial.html"><a href="factorial.html"><i class="fa fa-check"></i><b>17</b> Linear models with two categorical <span class="math inline">\(X\)</span> – Factorial linear models (“two-way ANOVA”)</a>
<ul>
<li class="chapter" data-level="17.1" data-path="factorial.html"><a href="factorial.html#a-linear-model-with-crossed-factors-estimates-interaction-effects"><i class="fa fa-check"></i><b>17.1</b> A linear model with crossed factors estimates interaction effects</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="factorial.html"><a href="factorial.html#an-interaction-is-a-difference-in-simple-effects"><i class="fa fa-check"></i><b>17.1.1</b> An interaction is a difference in simple effects</a></li>
<li class="chapter" data-level="17.1.2" data-path="factorial.html"><a href="factorial.html#a-linear-model-with-crossed-factors-includes-interaction-effects"><i class="fa fa-check"></i><b>17.1.2</b> A linear model with crossed factors includes interaction effects</a></li>
<li class="chapter" data-level="17.1.3" data-path="factorial.html"><a href="factorial.html#factorial-experiments-are-frequently-analyzed-as-flattened-linear-models-in-the-experimental-biology-literature"><i class="fa fa-check"></i><b>17.1.3</b> factorial experiments are frequently analyzed as flattened linear models in the experimental biology literature</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="factorial.html"><a href="factorial.html#example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data"><i class="fa fa-check"></i><b>17.2</b> Example 1 – Estimation of a treatment effect relative to a control effect (“Something different”) (Experiment 2j glucose uptake data)</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="factorial.html"><a href="factorial.html#twoway-understand1"><i class="fa fa-check"></i><b>17.2.1</b> Understand the experimental design</a></li>
<li class="chapter" data-level="17.2.2" data-path="factorial.html"><a href="factorial.html#fit-the-linear-model"><i class="fa fa-check"></i><b>17.2.2</b> Fit the linear model</a></li>
<li class="chapter" data-level="17.2.3" data-path="factorial.html"><a href="factorial.html#twoway-exp2j-inf"><i class="fa fa-check"></i><b>17.2.3</b> Inference</a></li>
<li class="chapter" data-level="17.2.4" data-path="factorial.html"><a href="factorial.html#plot-the-model-3"><i class="fa fa-check"></i><b>17.2.4</b> Plot the model</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-1"><i class="fa fa-check"></i><b>17.3</b> Understanding the linear model with crossed factors 1</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="factorial.html"><a href="factorial.html#twoway-what-coefs-are"><i class="fa fa-check"></i><b>17.3.1</b> What the coefficients are</a></li>
<li class="chapter" data-level="17.3.2" data-path="factorial.html"><a href="factorial.html#the-interaction-effect-is-something-different"><i class="fa fa-check"></i><b>17.3.2</b> The interaction effect is something different</a></li>
<li class="chapter" data-level="17.3.3" data-path="factorial.html"><a href="factorial.html#why-we-want-to-compare-the-treatment-effect-to-a-control-effect"><i class="fa fa-check"></i><b>17.3.3</b> Why we want to compare the treatment effect to a control effect</a></li>
<li class="chapter" data-level="17.3.4" data-path="factorial.html"><a href="factorial.html#the-order-of-the-factors-in-the-model-tells-the-same-story-differently"><i class="fa fa-check"></i><b>17.3.4</b> The order of the factors in the model tells the same story differently</a></li>
<li class="chapter" data-level="17.3.5" data-path="factorial.html"><a href="factorial.html#power-for-the-interaction-effect-is-less-than-that-for-simple-effects"><i class="fa fa-check"></i><b>17.3.5</b> Power for the interaction effect is less than that for simple effects</a></li>
<li class="chapter" data-level="17.3.6" data-path="factorial.html"><a href="factorial.html#planned-comparisons-vs.-post-hoc-tests"><i class="fa fa-check"></i><b>17.3.6</b> Planned comparisons vs. post-hoc tests</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="factorial.html"><a href="factorial.html#example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data"><i class="fa fa-check"></i><b>17.4</b> Example 2: Estimation of the effect of background condition on an effect (“it depends”) (Experiment 3e lesian area data)</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="factorial.html"><a href="factorial.html#understand-the-experimental-design"><i class="fa fa-check"></i><b>17.4.1</b> Understand the experimental design</a></li>
<li class="chapter" data-level="17.4.2" data-path="factorial.html"><a href="factorial.html#fit-the-linear-model-1"><i class="fa fa-check"></i><b>17.4.2</b> Fit the linear model</a></li>
<li class="chapter" data-level="17.4.3" data-path="factorial.html"><a href="factorial.html#check-the-model-1"><i class="fa fa-check"></i><b>17.4.3</b> Check the model</a></li>
<li class="chapter" data-level="17.4.4" data-path="factorial.html"><a href="factorial.html#inference-from-the-model-2"><i class="fa fa-check"></i><b>17.4.4</b> Inference from the model</a></li>
<li class="chapter" data-level="17.4.5" data-path="factorial.html"><a href="factorial.html#plot-the-model-4"><i class="fa fa-check"></i><b>17.4.5</b> Plot the model</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-2"><i class="fa fa-check"></i><b>17.5</b> Understanding the linear model with crossed factors 2</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="factorial.html"><a href="factorial.html#twoway-marginal-means"><i class="fa fa-check"></i><b>17.5.1</b> Conditional and marginal means</a></li>
<li class="chapter" data-level="17.5.2" data-path="factorial.html"><a href="factorial.html#simple-conditional-effects"><i class="fa fa-check"></i><b>17.5.2</b> Simple (conditional) effects</a></li>
<li class="chapter" data-level="17.5.3" data-path="factorial.html"><a href="factorial.html#twoway-marginal-effects"><i class="fa fa-check"></i><b>17.5.3</b> Marginal effects</a></li>
<li class="chapter" data-level="17.5.4" data-path="factorial.html"><a href="factorial.html#the-additive-model"><i class="fa fa-check"></i><b>17.5.4</b> The additive model</a></li>
<li class="chapter" data-level="17.5.5" data-path="factorial.html"><a href="factorial.html#twoway-reduce"><i class="fa fa-check"></i><b>17.5.5</b> Reduce models for the right reason</a></li>
<li class="chapter" data-level="17.5.6" data-path="factorial.html"><a href="factorial.html#the-marginal-means-of-an-additive-linear-model-with-two-factors-can-be-weird"><i class="fa fa-check"></i><b>17.5.6</b> The marginal means of an additive linear model with two factors can be weird</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="factorial.html"><a href="factorial.html#example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data"><i class="fa fa-check"></i><b>17.6</b> Example 3: Estimation of synergy (“More than the sum of the parts”) (Experiment 1c JA data)</a>
<ul>
<li class="chapter" data-level="17.6.1" data-path="factorial.html"><a href="factorial.html#examine-the-data"><i class="fa fa-check"></i><b>17.6.1</b> Examine the data</a></li>
<li class="chapter" data-level="17.6.2" data-path="factorial.html"><a href="factorial.html#fit-the-model-7"><i class="fa fa-check"></i><b>17.6.2</b> Fit the model</a></li>
<li class="chapter" data-level="17.6.3" data-path="factorial.html"><a href="factorial.html#model-check"><i class="fa fa-check"></i><b>17.6.3</b> Model check</a></li>
<li class="chapter" data-level="17.6.4" data-path="factorial.html"><a href="factorial.html#inference-from-the-model-3"><i class="fa fa-check"></i><b>17.6.4</b> Inference from the model</a></li>
<li class="chapter" data-level="17.6.5" data-path="factorial.html"><a href="factorial.html#plot-the-model-5"><i class="fa fa-check"></i><b>17.6.5</b> Plot the model</a></li>
<li class="chapter" data-level="17.6.6" data-path="factorial.html"><a href="factorial.html#alternative-plot"><i class="fa fa-check"></i><b>17.6.6</b> Alternative plot</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-3"><i class="fa fa-check"></i><b>17.7</b> Understanding the linear model with crossed factors 3</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="factorial.html"><a href="factorial.html#thinking-about-the-coefficients-of-the-linear-model"><i class="fa fa-check"></i><b>17.7.1</b> Thinking about the coefficients of the linear model</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="factorial.html"><a href="factorial.html#issues-in-inference"><i class="fa fa-check"></i><b>17.8</b> Issues in inference</a>
<ul>
<li class="chapter" data-level="17.8.1" data-path="factorial.html"><a href="factorial.html#for-pairwise-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model"><i class="fa fa-check"></i><b>17.8.1</b> For pairwise contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</a></li>
<li class="chapter" data-level="17.8.2" data-path="factorial.html"><a href="factorial.html#for-interaction-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model"><i class="fa fa-check"></i><b>17.8.2</b> For interaction contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</a></li>
<li class="chapter" data-level="17.8.3" data-path="factorial.html"><a href="factorial.html#twoway-multiple-tests"><i class="fa fa-check"></i><b>17.8.3</b> Adjusting <em>p</em>-values for multiple tests</a></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="factorial.html"><a href="factorial.html#two-way-anova"><i class="fa fa-check"></i><b>17.9</b> Two-way ANOVA</a>
<ul>
<li class="chapter" data-level="17.9.1" data-path="factorial.html"><a href="factorial.html#how-to-read-a-two-way-anova-table"><i class="fa fa-check"></i><b>17.9.1</b> How to read a two-way ANOVA table</a></li>
<li class="chapter" data-level="17.9.2" data-path="factorial.html"><a href="factorial.html#what-do-the-main-effects-in-an-anova-table-mean"><i class="fa fa-check"></i><b>17.9.2</b> What do the main effects in an ANOVA table mean?</a></li>
</ul></li>
<li class="chapter" data-level="17.10" data-path="factorial.html"><a href="factorial.html#more-issues-in-inference"><i class="fa fa-check"></i><b>17.10</b> More issues in inference</a>
<ul>
<li class="chapter" data-level="17.10.1" data-path="factorial.html"><a href="factorial.html#longitudinal-experiments-include-time-as-a-random-factor-better-than-repeated-measures-anova"><i class="fa fa-check"></i><b>17.10.1</b> Longitudinal experiments – include Time as a random factor (better than repeated measures ANOVA)</a></li>
</ul></li>
<li class="chapter" data-level="17.11" data-path="factorial.html"><a href="factorial.html#working-in-r-4"><i class="fa fa-check"></i><b>17.11</b> Working in R</a>
<ul>
<li class="chapter" data-level="17.11.1" data-path="factorial.html"><a href="factorial.html#model-formula"><i class="fa fa-check"></i><b>17.11.1</b> Model formula</a></li>
<li class="chapter" data-level="17.11.2" data-path="factorial.html"><a href="factorial.html#using-the-emmeans-function-2"><i class="fa fa-check"></i><b>17.11.2</b> Using the emmeans function</a></li>
<li class="chapter" data-level="17.11.3" data-path="factorial.html"><a href="factorial.html#contrasts"><i class="fa fa-check"></i><b>17.11.3</b> Contrasts</a></li>
<li class="chapter" data-level="17.11.4" data-path="factorial.html"><a href="factorial.html#practice-safe-anova"><i class="fa fa-check"></i><b>17.11.4</b> Practice safe ANOVA</a></li>
<li class="chapter" data-level="17.11.5" data-path="factorial.html"><a href="factorial.html#better-to-avoid-these"><i class="fa fa-check"></i><b>17.11.5</b> Better to avoid these</a></li>
</ul></li>
<li class="chapter" data-level="17.12" data-path="factorial.html"><a href="factorial.html#hidden-code-5"><i class="fa fa-check"></i><b>17.12</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="17.12.1" data-path="factorial.html"><a href="factorial.html#import-exp2j-example-1"><i class="fa fa-check"></i><b>17.12.1</b> Import exp2j (Example 1)</a></li>
<li class="chapter" data-level="17.12.2" data-path="factorial.html"><a href="factorial.html#import-exp3e-lesian-area-data-example-2"><i class="fa fa-check"></i><b>17.12.2</b> Import exp3e lesian area data (Example 2)</a></li>
<li class="chapter" data-level="17.12.3" data-path="factorial.html"><a href="factorial.html#import-exp1c-ja-data-example-3"><i class="fa fa-check"></i><b>17.12.3</b> Import Exp1c JA data (Example 3)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-vii-expanding-the-linear-model.html"><a href="part-vii-expanding-the-linear-model.html"><i class="fa fa-check"></i>Part VII – Expanding the Linear Model</a></li>
<li class="chapter" data-level="18" data-path="lmm.html"><a href="lmm.html"><i class="fa fa-check"></i><b>18</b> Models with random factors – linear mixed models</a>
<ul>
<li class="chapter" data-level="18.1" data-path="lmm.html"><a href="lmm.html#lmm-demo"><i class="fa fa-check"></i><b>18.1</b> Example 1 – A random intercepts and slopes explainer (demo1)</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="lmm.html"><a href="lmm.html#batched-measurements-result-in-clustered-residuals"><i class="fa fa-check"></i><b>18.1.1</b> Batched measurements result in clustered residuals</a></li>
<li class="chapter" data-level="18.1.2" data-path="lmm.html"><a href="lmm.html#lmm-corerr"><i class="fa fa-check"></i><b>18.1.2</b> Clustered residuals result in correlated error</a></li>
<li class="chapter" data-level="18.1.3" data-path="lmm.html"><a href="lmm.html#in-blocked-designs-clustered-residuals-adds-a-variance-component-that-masks-treatment-effects"><i class="fa fa-check"></i><b>18.1.3</b> In blocked designs, clustered residuals adds a variance component that masks treatment effects</a></li>
<li class="chapter" data-level="18.1.4" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-are-linear-models-with-added-random-factors"><i class="fa fa-check"></i><b>18.1.4</b> Linear mixed models are linear models with added random factors</a></li>
<li class="chapter" data-level="18.1.5" data-path="lmm.html"><a href="lmm.html#what-the-random-effects-are"><i class="fa fa-check"></i><b>18.1.5</b> What the random effects are</a></li>
<li class="chapter" data-level="18.1.6" data-path="lmm.html"><a href="lmm.html#in-a-blocked-design-a-linear-model-with-added-random-effects-increases-precision-of-treatment-effects"><i class="fa fa-check"></i><b>18.1.6</b> In a blocked design, a linear model with added random effects increases precision of treatment effects</a></li>
<li class="chapter" data-level="18.1.7" data-path="lmm.html"><a href="lmm.html#lmm-varcorr"><i class="fa fa-check"></i><b>18.1.7</b> The correlation among random intercepts and slopes</a></li>
<li class="chapter" data-level="18.1.8" data-path="lmm.html"><a href="lmm.html#clustered-residuals-create-heterogeneity-among-treatments"><i class="fa fa-check"></i><b>18.1.8</b> Clustered residuals create heterogeneity among treatments</a></li>
<li class="chapter" data-level="18.1.9" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-are-flexible"><i class="fa fa-check"></i><b>18.1.9</b> Linear mixed models are flexible</a></li>
<li class="chapter" data-level="18.1.10" data-path="lmm.html"><a href="lmm.html#a-random-intercept-only-model"><i class="fa fa-check"></i><b>18.1.10</b> A random intercept only model</a></li>
<li class="chapter" data-level="18.1.11" data-path="lmm.html"><a href="lmm.html#a-model-including-an-interaction-intercept"><i class="fa fa-check"></i><b>18.1.11</b> A model including an interaction intercept</a></li>
<li class="chapter" data-level="18.1.12" data-path="lmm.html"><a href="lmm.html#lmm-demo-aic"><i class="fa fa-check"></i><b>18.1.12</b> AIC and model selection – which model to report?</a></li>
<li class="chapter" data-level="18.1.13" data-path="lmm.html"><a href="lmm.html#the-specification-of-random-effects-matters"><i class="fa fa-check"></i><b>18.1.13</b> The specification of random effects matters</a></li>
<li class="chapter" data-level="18.1.14" data-path="lmm.html"><a href="lmm.html#mixed-effect-and-repeated-measures-anova"><i class="fa fa-check"></i><b>18.1.14</b> Mixed Effect and Repeated Measures ANOVA</a></li>
<li class="chapter" data-level="18.1.15" data-path="lmm.html"><a href="lmm.html#pseudoreplication"><i class="fa fa-check"></i><b>18.1.15</b> Pseudoreplication</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="lmm.html"><a href="lmm.html#lmm-example2"><i class="fa fa-check"></i><b>18.2</b> Example 2 – experiments without subsampling replication (exp6g)</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="lmm.html"><a href="lmm.html#understand-the-data"><i class="fa fa-check"></i><b>18.2.1</b> Understand the data</a></li>
<li class="chapter" data-level="18.2.2" data-path="lmm.html"><a href="lmm.html#model-fit-and-inference"><i class="fa fa-check"></i><b>18.2.2</b> Model fit and inference</a></li>
<li class="chapter" data-level="18.2.3" data-path="lmm.html"><a href="lmm.html#the-model-exp6g_m1-adds-a-random-intercept-but-not-a-random-slope"><i class="fa fa-check"></i><b>18.2.3</b> The model exp6g_m1 adds a random intercept but not a random slope</a></li>
<li class="chapter" data-level="18.2.4" data-path="lmm.html"><a href="lmm.html#the-fixed-effect-coefficients-of-model-exp6g_m1"><i class="fa fa-check"></i><b>18.2.4</b> The fixed effect coefficients of model exp6g_m1</a></li>
<li class="chapter" data-level="18.2.5" data-path="lmm.html"><a href="lmm.html#the-random-intercept-coefficients-of-exp6g_m1"><i class="fa fa-check"></i><b>18.2.5</b> The random intercept coefficients of exp6g_m1</a></li>
<li class="chapter" data-level="18.2.6" data-path="lmm.html"><a href="lmm.html#the-random-and-residual-variance-and-the-intraclass-correlation-of-model-exp6g_m1"><i class="fa fa-check"></i><b>18.2.6</b> The random and residual variance and the intraclass correlation of model exp6g_m1</a></li>
<li class="chapter" data-level="18.2.7" data-path="lmm.html"><a href="lmm.html#the-linear-mixed-model-exp6g_m1-increases-precision-of-treatment-effects-relative-to-a-fixed-effects-model"><i class="fa fa-check"></i><b>18.2.7</b> The linear mixed model exp6g_m1 increases precision of treatment effects, relative to a fixed effects model</a></li>
<li class="chapter" data-level="18.2.8" data-path="lmm.html"><a href="lmm.html#lmm-exp6g-alt"><i class="fa fa-check"></i><b>18.2.8</b> Alternative models for exp6g</a></li>
<li class="chapter" data-level="18.2.9" data-path="lmm.html"><a href="lmm.html#paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models"><i class="fa fa-check"></i><b>18.2.9</b> Paired t-tests and repeated measures ANOVA are special cases of linear mixed models</a></li>
<li class="chapter" data-level="18.2.10" data-path="lmm.html"><a href="lmm.html#lmm-exp6g-rmanova"><i class="fa fa-check"></i><b>18.2.10</b> Classical (“univariate model”) repeated measures ANOVA of exp6g</a></li>
<li class="chapter" data-level="18.2.11" data-path="lmm.html"><a href="lmm.html#lmm-exp6g-rmanova-multi"><i class="fa fa-check"></i><b>18.2.11</b> “Multivariate model” repeated measures ANOVA</a></li>
<li class="chapter" data-level="18.2.12" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-vs-repeated-measures-anova"><i class="fa fa-check"></i><b>18.2.12</b> Linear mixed models vs repeated measures ANOVA</a></li>
<li class="chapter" data-level="18.2.13" data-path="lmm.html"><a href="lmm.html#modeling-textttmouse_id-as-a-fixed-effect"><i class="fa fa-check"></i><b>18.2.13</b> Modeling <span class="math inline">\(\texttt{mouse_id}\)</span> as a fixed effect</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="lmm.html"><a href="lmm.html#example-3-factorial-experiments-and-no-subsampling-replicates-exp5c"><i class="fa fa-check"></i><b>18.3</b> Example 3 – Factorial experiments and no subsampling replicates (exp5c)</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="lmm.html"><a href="lmm.html#understand-the-data-1"><i class="fa fa-check"></i><b>18.3.1</b> Understand the data</a></li>
<li class="chapter" data-level="18.3.2" data-path="lmm.html"><a href="lmm.html#examine-the-data-1"><i class="fa fa-check"></i><b>18.3.2</b> Examine the data</a></li>
<li class="chapter" data-level="18.3.3" data-path="lmm.html"><a href="lmm.html#lmm-exp5c-m1"><i class="fa fa-check"></i><b>18.3.3</b> Model fit and inference</a></li>
<li class="chapter" data-level="18.3.4" data-path="lmm.html"><a href="lmm.html#why-we-care-about-modeling-batch-in-exp5c"><i class="fa fa-check"></i><b>18.3.4</b> Why we care about modeling batch in exp5c</a></li>
<li class="chapter" data-level="18.3.5" data-path="lmm.html"><a href="lmm.html#the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts"><i class="fa fa-check"></i><b>18.3.5</b> The linear mixed model exp5c_m1 adds two random intercepts</a></li>
<li class="chapter" data-level="18.3.6" data-path="lmm.html"><a href="lmm.html#the-fixed-effect-coefficients-of-model-exp5c_m1"><i class="fa fa-check"></i><b>18.3.6</b> The fixed effect coefficients of model exp5c_m1</a></li>
<li class="chapter" data-level="18.3.7" data-path="lmm.html"><a href="lmm.html#the-random-effect-coefficients-of-model-exp5c_m1"><i class="fa fa-check"></i><b>18.3.7</b> The random effect coefficients of model exp5c_m1</a></li>
<li class="chapter" data-level="18.3.8" data-path="lmm.html"><a href="lmm.html#lmm-exp5c-alt"><i class="fa fa-check"></i><b>18.3.8</b> Alternative models for exp5c</a></li>
<li class="chapter" data-level="18.3.9" data-path="lmm.html"><a href="lmm.html#lmm-rmanova-5c"><i class="fa fa-check"></i><b>18.3.9</b> Classical (“univariate model”) repeated measures ANOVA</a></li>
<li class="chapter" data-level="18.3.10" data-path="lmm.html"><a href="lmm.html#lmm-rmanova-5c-multi"><i class="fa fa-check"></i><b>18.3.10</b> “Multivariate model” repeated measures ANOVA of exp5c</a></li>
<li class="chapter" data-level="18.3.11" data-path="lmm.html"><a href="lmm.html#modeling-textttdonor-as-a-fixed-effect"><i class="fa fa-check"></i><b>18.3.11</b> Modeling <span class="math inline">\(\texttt{donor}\)</span> as a fixed effect</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="lmm.html"><a href="lmm.html#lmm-example4"><i class="fa fa-check"></i><b>18.4</b> Example 4 – Experiments with subsampling replication (exp1g)</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="lmm.html"><a href="lmm.html#understand-the-data-2"><i class="fa fa-check"></i><b>18.4.1</b> Understand the data</a></li>
<li class="chapter" data-level="18.4.2" data-path="lmm.html"><a href="lmm.html#examine-the-data-2"><i class="fa fa-check"></i><b>18.4.2</b> Examine the data</a></li>
<li class="chapter" data-level="18.4.3" data-path="lmm.html"><a href="lmm.html#fit-the-model-9"><i class="fa fa-check"></i><b>18.4.3</b> Fit the model</a></li>
<li class="chapter" data-level="18.4.4" data-path="lmm.html"><a href="lmm.html#inference-from-the-model-6"><i class="fa fa-check"></i><b>18.4.4</b> Inference from the model</a></li>
<li class="chapter" data-level="18.4.5" data-path="lmm.html"><a href="lmm.html#plot-the-model-6"><i class="fa fa-check"></i><b>18.4.5</b> Plot the model</a></li>
<li class="chapter" data-level="18.4.6" data-path="lmm.html"><a href="lmm.html#alternaplot-the-model-2"><i class="fa fa-check"></i><b>18.4.6</b> Alternaplot the model</a></li>
<li class="chapter" data-level="18.4.7" data-path="lmm.html"><a href="lmm.html#understanding-the-alternative-models"><i class="fa fa-check"></i><b>18.4.7</b> Understanding the alternative models</a></li>
<li class="chapter" data-level="18.4.8" data-path="lmm.html"><a href="lmm.html#the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b"><i class="fa fa-check"></i><b>18.4.8</b> The VarCorr matrix of models exp1g_m1a and exp1g_m1b</a></li>
<li class="chapter" data-level="18.4.9" data-path="lmm.html"><a href="lmm.html#the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means"><i class="fa fa-check"></i><b>18.4.9</b> The linear mixed model has more precision and power than the fixed effect model of batch means</a></li>
<li class="chapter" data-level="18.4.10" data-path="lmm.html"><a href="lmm.html#fixed-effect-models-and-pseudoreplication"><i class="fa fa-check"></i><b>18.4.10</b> Fixed effect models and pseudoreplication</a></li>
<li class="chapter" data-level="18.4.11" data-path="lmm.html"><a href="lmm.html#mixed-effect-anova"><i class="fa fa-check"></i><b>18.4.11</b> Mixed-effect ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="lmm.html"><a href="lmm.html#working-in-r-5"><i class="fa fa-check"></i><b>18.5</b> Working in R</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="lmm.html"><a href="lmm.html#fitting-linear-mixed-models"><i class="fa fa-check"></i><b>18.5.1</b> Fitting linear mixed models</a></li>
<li class="chapter" data-level="18.5.2" data-path="lmm.html"><a href="lmm.html#plotting-models-fit-to-batched-data"><i class="fa fa-check"></i><b>18.5.2</b> Plotting models fit to batched data</a></li>
<li class="chapter" data-level="18.5.3" data-path="lmm.html"><a href="lmm.html#repeated-measures-anova-randomized-complete-block-with-no-subsampling"><i class="fa fa-check"></i><b>18.5.3</b> Repeated measures ANOVA (randomized complete block with no subsampling)</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="lmm.html"><a href="lmm.html#hidden-code-6"><i class="fa fa-check"></i><b>18.6</b> Hidden code</a>
<ul>
<li class="chapter" data-level="18.6.1" data-path="lmm.html"><a href="lmm.html#import-exp5c"><i class="fa fa-check"></i><b>18.6.1</b> Import exp5c</a></li>
<li class="chapter" data-level="18.6.2" data-path="lmm.html"><a href="lmm.html#import-exp1g"><i class="fa fa-check"></i><b>18.6.2</b> Import exp1g</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="pre-post.html"><a href="pre-post.html"><i class="fa fa-check"></i><b>19</b> Linear models for longitudinal experiments – I. pre-post designs</a>
<ul>
<li class="chapter" data-level="19.1" data-path="pre-post.html"><a href="pre-post.html#best-practice-models"><i class="fa fa-check"></i><b>19.1</b> Best practice models</a></li>
<li class="chapter" data-level="19.2" data-path="pre-post.html"><a href="pre-post.html#common-alternatives-that-are-not-recommended"><i class="fa fa-check"></i><b>19.2</b> Common alternatives that are not recommended</a></li>
<li class="chapter" data-level="19.3" data-path="pre-post.html"><a href="pre-post.html#advanced-models"><i class="fa fa-check"></i><b>19.3</b> Advanced models</a></li>
<li class="chapter" data-level="19.4" data-path="pre-post.html"><a href="pre-post.html#understanding-the-alternative-models-1"><i class="fa fa-check"></i><b>19.4</b> Understanding the alternative models</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="pre-post.html"><a href="pre-post.html#m1-linear-model-with-the-baseline-measure-as-the-covariate-ancova-model"><i class="fa fa-check"></i><b>19.4.1</b> (M1) Linear model with the baseline measure as the covariate (ANCOVA model)</a></li>
<li class="chapter" data-level="19.4.2" data-path="pre-post.html"><a href="pre-post.html#m2-linear-model-of-the-change-score-change-score-model"><i class="fa fa-check"></i><b>19.4.2</b> (M2) Linear model of the change score (change-score model)</a></li>
<li class="chapter" data-level="19.4.3" data-path="pre-post.html"><a href="pre-post.html#m3-linear-model-of-post-baseline-values-without-the-baseline-as-a-covariate-post-model"><i class="fa fa-check"></i><b>19.4.3</b> (M3) Linear model of post-baseline values without the baseline as a covariate (post model)</a></li>
<li class="chapter" data-level="19.4.4" data-path="pre-post.html"><a href="pre-post.html#m4-linear-model-with-factorial-fixed-effects-fixed-effects-model"><i class="fa fa-check"></i><b>19.4.4</b> (M4) Linear model with factorial fixed effects (fixed-effects model)</a></li>
<li class="chapter" data-level="19.4.5" data-path="pre-post.html"><a href="pre-post.html#m5-repeated-measures-anova"><i class="fa fa-check"></i><b>19.4.5</b> (M5) Repeated measures ANOVA</a></li>
<li class="chapter" data-level="19.4.6" data-path="pre-post.html"><a href="pre-post.html#m6-linear-mixed-model"><i class="fa fa-check"></i><b>19.4.6</b> (M6) Linear mixed model</a></li>
<li class="chapter" data-level="19.4.7" data-path="pre-post.html"><a href="pre-post.html#m7-linear-model-with-correlated-error"><i class="fa fa-check"></i><b>19.4.7</b> (M7) Linear model with correlated error</a></li>
<li class="chapter" data-level="19.4.8" data-path="pre-post.html"><a href="pre-post.html#m8-constrained-fixed-effects-model-with-correlated-error-clda-model"><i class="fa fa-check"></i><b>19.4.8</b> (M8) Constrained fixed effects model with correlated error (cLDA model)</a></li>
<li class="chapter" data-level="19.4.9" data-path="pre-post.html"><a href="pre-post.html#comparison-table"><i class="fa fa-check"></i><b>19.4.9</b> Comparison table</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="pre-post.html"><a href="pre-post.html#example-1-a-single-post-baseline-measure-pre-post-design"><i class="fa fa-check"></i><b>19.5</b> Example 1 – a single post-baseline measure (pre-post design)</a></li>
<li class="chapter" data-level="19.6" data-path="pre-post.html"><a href="pre-post.html#working-in-r-6"><i class="fa fa-check"></i><b>19.6</b> Working in R</a></li>
<li class="chapter" data-level="19.7" data-path="pre-post.html"><a href="pre-post.html#hidden-code-7"><i class="fa fa-check"></i><b>19.7</b> Hidden code</a>
<ul>
<li class="chapter" data-level="19.7.1" data-path="pre-post.html"><a href="pre-post.html#import-and-wrangle-mouse-sociability-data"><i class="fa fa-check"></i><b>19.7.1</b> Import and wrangle mouse sociability data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="counts.html"><a href="counts.html"><i class="fa fa-check"></i><b>20</b> Linear models for counts, binary responses, skewed responses, and ratios – Generalized Linear Models</a>
<ul>
<li class="chapter" data-level="20.1" data-path="counts.html"><a href="counts.html#glm-counts"><i class="fa fa-check"></i><b>20.1</b> Introducing Generalized Linear Models using count data examples</a>
<ul>
<li class="chapter" data-level="20.1.1" data-path="counts.html"><a href="counts.html#the-generalized-linear-model-glm"><i class="fa fa-check"></i><b>20.1.1</b> The Generalized Linear Model (GLM)</a></li>
<li class="chapter" data-level="20.1.2" data-path="counts.html"><a href="counts.html#kinds-of-data-that-are-modeled-by-a-glm"><i class="fa fa-check"></i><b>20.1.2</b> Kinds of data that are modeled by a GLM</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="counts.html"><a href="counts.html#glm-1"><i class="fa fa-check"></i><b>20.2</b> Example 1 – GLM models for count responses (“angiogenic sprouts” exp3a)</a>
<ul>
<li class="chapter" data-level="20.2.1" data-path="counts.html"><a href="counts.html#understand-the-data-3"><i class="fa fa-check"></i><b>20.2.1</b> Understand the data</a></li>
<li class="chapter" data-level="20.2.2" data-path="counts.html"><a href="counts.html#model-fit-and-inference-1"><i class="fa fa-check"></i><b>20.2.2</b> Model fit and inference</a></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="counts.html"><a href="counts.html#understanding-example-1"><i class="fa fa-check"></i><b>20.3</b> Understanding Example 1</a>
<ul>
<li class="chapter" data-level="20.3.1" data-path="counts.html"><a href="counts.html#modeling-strategy-1"><i class="fa fa-check"></i><b>20.3.1</b> Modeling strategy</a></li>
<li class="chapter" data-level="20.3.2" data-path="counts.html"><a href="counts.html#model-checking-fits-to-count-data"><i class="fa fa-check"></i><b>20.3.2</b> Model checking fits to count data</a></li>
<li class="chapter" data-level="20.3.3" data-path="counts.html"><a href="counts.html#biological-count-data-are-rarely-fit-well-by-a-poisson-glm.-instead-fit-a-quasi-poisson-or-negative-binomial-glm-model."><i class="fa fa-check"></i><b>20.3.3</b> Biological count data are rarely fit well by a Poisson GLM. Instead, fit a quasi-poisson or negative binomial GLM model.</a></li>
<li class="chapter" data-level="20.3.4" data-path="counts.html"><a href="counts.html#a-glm-is-a-linear-model-on-the-link-scale"><i class="fa fa-check"></i><b>20.3.4</b> A GLM is a linear model on the link scale</a></li>
<li class="chapter" data-level="20.3.5" data-path="counts.html"><a href="counts.html#coeffecients-of-a-generalized-linear-model-with-a-log-link-function-are-on-the-link-scale."><i class="fa fa-check"></i><b>20.3.5</b> Coeffecients of a Generalized Linear Model with a log-link function are on the link scale.</a></li>
<li class="chapter" data-level="20.3.6" data-path="counts.html"><a href="counts.html#modeled-means-in-the-emmeans-table-of-a-generalized-linear-model-can-be-on-the-link-scale-or-response-scale-report-the-response-scale"><i class="fa fa-check"></i><b>20.3.6</b> Modeled means in the emmeans table of a Generalized Linear Model can be on the link scale or response scale – Report the response scale</a></li>
<li class="chapter" data-level="20.3.7" data-path="counts.html"><a href="counts.html#some-consequences-of-fitting-a-linear-model-to-count-data"><i class="fa fa-check"></i><b>20.3.7</b> Some consequences of fitting a linear model to count data</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="counts.html"><a href="counts.html#example-2-use-a-glm-with-an-offset-instead-of-a-ratio-of-some-measurement-per-area-dna-damage-data-exp3b"><i class="fa fa-check"></i><b>20.4</b> Example 2 – Use a GLM with an offset instead of a ratio of some measurement per area (“dna damage” data exp3b)</a>
<ul>
<li class="chapter" data-level="20.4.1" data-path="counts.html"><a href="counts.html#exp3b-dna-damage-data"><i class="fa fa-check"></i><b>20.4.1</b> exp3b (“dna damage”) data</a></li>
<li class="chapter" data-level="20.4.2" data-path="counts.html"><a href="counts.html#understand-the-data-4"><i class="fa fa-check"></i><b>20.4.2</b> Understand the data</a></li>
<li class="chapter" data-level="20.4.3" data-path="counts.html"><a href="counts.html#model-fit-and-inference-2"><i class="fa fa-check"></i><b>20.4.3</b> Model fit and inference</a></li>
</ul></li>
<li class="chapter" data-level="20.5" data-path="counts.html"><a href="counts.html#understanding-example-2"><i class="fa fa-check"></i><b>20.5</b> Understanding Example 2</a>
<ul>
<li class="chapter" data-level="20.5.1" data-path="counts.html"><a href="counts.html#an-offset-is-an-added-covariate-with-a-coefficient-fixed-at-1"><i class="fa fa-check"></i><b>20.5.1</b> An offset is an added covariate with a coefficient fixed at 1</a></li>
<li class="chapter" data-level="20.5.2" data-path="counts.html"><a href="counts.html#glm-rate"><i class="fa fa-check"></i><b>20.5.2</b> A count GLM with an offset models the area-normalized means</a></li>
<li class="chapter" data-level="20.5.3" data-path="counts.html"><a href="counts.html#compare-an-offset-to-an-added-covariate-with-an-estimated-coefficient"><i class="fa fa-check"></i><b>20.5.3</b> Compare an offset to an added covariate with an estimated coefficient</a></li>
<li class="chapter" data-level="20.5.4" data-path="counts.html"><a href="counts.html#issues-with-plotting"><i class="fa fa-check"></i><b>20.5.4</b> Issues with plotting</a></li>
</ul></li>
<li class="chapter" data-level="20.6" data-path="counts.html"><a href="counts.html#example-3-glm-models-for-binary-responses"><i class="fa fa-check"></i><b>20.6</b> Example 3 – GLM models for binary responses</a></li>
<li class="chapter" data-level="20.7" data-path="counts.html"><a href="counts.html#working-in-r-7"><i class="fa fa-check"></i><b>20.7</b> Working in R</a>
<ul>
<li class="chapter" data-level="20.7.1" data-path="counts.html"><a href="counts.html#fitting-glms-to-count-data"><i class="fa fa-check"></i><b>20.7.1</b> Fitting GLMs to count data</a></li>
<li class="chapter" data-level="20.7.2" data-path="counts.html"><a href="counts.html#fitting-a-glm-to-a-continuous-conditional-response-with-right-skew."><i class="fa fa-check"></i><b>20.7.2</b> Fitting a GLM to a continuous conditional response with right skew.</a></li>
<li class="chapter" data-level="20.7.3" data-path="counts.html"><a href="counts.html#fitting-a-glm-to-a-binary-success-or-failure-presence-or-absence-survived-or-died-response"><i class="fa fa-check"></i><b>20.7.3</b> Fitting a GLM to a binary (success or failure, presence or absence, survived or died) response</a></li>
<li class="chapter" data-level="20.7.4" data-path="counts.html"><a href="counts.html#fitting-generalized-linear-mixed-models"><i class="fa fa-check"></i><b>20.7.4</b> Fitting Generalized Linear Mixed Models</a></li>
</ul></li>
<li class="chapter" data-level="20.8" data-path="counts.html"><a href="counts.html#model-checking-glms"><i class="fa fa-check"></i><b>20.8</b> Model checking GLMs</a></li>
<li class="chapter" data-level="20.9" data-path="counts.html"><a href="counts.html#hidden-code-8"><i class="fa fa-check"></i><b>20.9</b> Hidden code</a>
<ul>
<li class="chapter" data-level="20.9.1" data-path="counts.html"><a href="counts.html#import-example-1-data-exp3a-angiogenic-sprouts"><i class="fa fa-check"></i><b>20.9.1</b> Import Example 1 data (exp3a – “angiogenic sprouts”)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="gls.html"><a href="gls.html"><i class="fa fa-check"></i><b>21</b> Linear models with heterogenous variance</a></li>
<li class="chapter" data-level="22" data-path="count-sim.html"><a href="count-sim.html"><i class="fa fa-check"></i><b>22</b> Simulations – Count data (alternatives to a t-test)</a>
<ul>
<li class="chapter" data-level="22.1" data-path="count-sim.html"><a href="count-sim.html#use-data-similar-to-figure-6f-from-example-1"><i class="fa fa-check"></i><b>22.1</b> Use data similar to Figure 6f from Example 1</a></li>
<li class="chapter" data-level="22.2" data-path="count-sim.html"><a href="count-sim.html#functions"><i class="fa fa-check"></i><b>22.2</b> Functions</a></li>
<li class="chapter" data-level="22.3" data-path="count-sim.html"><a href="count-sim.html#simulations"><i class="fa fa-check"></i><b>22.3</b> Simulations</a>
<ul>
<li class="chapter" data-level="22.3.1" data-path="count-sim.html"><a href="count-sim.html#type-i-pseudo-normal-distribution"><i class="fa fa-check"></i><b>22.3.1</b> Type I, Pseudo-Normal distribution</a></li>
<li class="chapter" data-level="22.3.2" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-equal-n"><i class="fa fa-check"></i><b>22.3.2</b> Type I, neg binom, equal n</a></li>
<li class="chapter" data-level="22.3.3" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-equal-n-small-theta"><i class="fa fa-check"></i><b>22.3.3</b> Type I, neg binom, equal n, small theta</a></li>
<li class="chapter" data-level="22.3.4" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-unequal-n"><i class="fa fa-check"></i><b>22.3.4</b> Type I, neg binom, unequal n</a></li>
<li class="chapter" data-level="22.3.5" data-path="count-sim.html"><a href="count-sim.html#power-pseudo-normal-distribution-equal-n"><i class="fa fa-check"></i><b>22.3.5</b> Power, Pseudo-Normal distribution, equal n</a></li>
<li class="chapter" data-level="22.3.6" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-equal-n"><i class="fa fa-check"></i><b>22.3.6</b> Power, neg binom, equal n</a></li>
<li class="chapter" data-level="22.3.7" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-small-theta"><i class="fa fa-check"></i><b>22.3.7</b> Power, neg binom, small theta</a></li>
<li class="chapter" data-level="22.3.8" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-unequal-n"><i class="fa fa-check"></i><b>22.3.8</b> Power, neg binom, unequal n</a></li>
<li class="chapter" data-level="22.3.9" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-unequal-n-unequal-theta"><i class="fa fa-check"></i><b>22.3.9</b> Power, neg binom, unequal n, unequal theta</a></li>
<li class="chapter" data-level="22.3.10" data-path="count-sim.html"><a href="count-sim.html#type-1-neg-binom-equal-n-unequal-theta"><i class="fa fa-check"></i><b>22.3.10</b> Type 1, neg binom, equal n, unequal theta</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="count-sim.html"><a href="count-sim.html#save-it-read-it"><i class="fa fa-check"></i><b>22.4</b> Save it, Read it</a></li>
<li class="chapter" data-level="22.5" data-path="count-sim.html"><a href="count-sim.html#analysis"><i class="fa fa-check"></i><b>22.5</b> Analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html"><i class="fa fa-check"></i>Appendix 1: Getting Started with R</a>
<ul>
<li class="chapter" data-level="22.6" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#get-your-computer-ready"><i class="fa fa-check"></i><b>22.6</b> Get your computer ready</a>
<ul>
<li class="chapter" data-level="22.6.1" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#start-here"><i class="fa fa-check"></i><b>22.6.1</b> Start here</a></li>
<li class="chapter" data-level="22.6.2" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r"><i class="fa fa-check"></i><b>22.6.2</b> Install R</a></li>
<li class="chapter" data-level="22.6.3" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r-studio"><i class="fa fa-check"></i><b>22.6.3</b> Install R Studio</a></li>
<li class="chapter" data-level="22.6.4" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r-markdown-1"><i class="fa fa-check"></i><b>22.6.4</b> Install R Markdown</a></li>
<li class="chapter" data-level="22.6.5" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#optional-alternative-latex-installations"><i class="fa fa-check"></i><b>22.6.5</b> (optional) Alternative LaTeX installations</a></li>
</ul></li>
<li class="chapter" data-level="22.7" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#start-learning-r-studio"><i class="fa fa-check"></i><b>22.7</b> Start learning R Studio</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix-2-online-resources-for-getting-started-with-statistical-modeling-in-r.html"><a href="appendix-2-online-resources-for-getting-started-with-statistical-modeling-in-r.html"><i class="fa fa-check"></i>Appendix 2: Online Resources for Getting Started with Statistical Modeling in R</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Elements of Statistical Modeling for Experimental Biology</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="counts" class="section level1" number="20">
<h1><span class="header-section-number">Chapter 20</span> Linear models for counts, binary responses, skewed responses, and ratios – Generalized Linear Models</h1>
<div id="glm-counts" class="section level2" number="20.1">
<h2><span class="header-section-number">20.1</span> Introducing Generalized Linear Models using count data examples</h2>
<p>Biologists frequently count stuff, and design experiments to estimate the effects of different factors on these counts. Count data can cause numerous problems with linear models that assume a normal, conditional distribution, including 1) counts are discrete, and can be zero or positive integers only, 2) counts tend to bunch up on the small side of the range, creating a distribution with a positive skew, 3) a sample of counts can have an abundance of zeros, and 4) the variance of counts increases with the mean (see Figure <a href="counts.html#fig:glm1-plot1">20.1</a> for some of these properties). Some count data can be approximated by and reasonably modeled with a normal distribution. More often, count data should modeled with a <a href="https://en.wikipedia.org/wiki/Poisson_distribution" target="_blank">Poisson distribution</a> or <a href="https://en.wikipedia.org/wiki/Negative_binomial_distribution" target="_blank">negative binomial distribution</a> using a <a href="https://en.wikipedia.org/wiki/Generalized_linear_model%20(GLM)" target="_blank">generalized linear model</a>. Poisson and negative binomial distributions are <strong>discrete probability distributions</strong> with two important properties: 1) the distribution contains only zero and positive integers and 2) the variance is a function of the mean. Back before modern computing and fast processors, count data were often analyzed by either <strong>transforming</strong> the response or by <strong>non-parametric hypothesis tests</strong>. One reason to prefer a statistical modeling approach with a GLM is that we can get interpretable parameter estimates. By contrast, both the analysis of transformed data and non-parametric hypothesis tests are really tools for computing “correct” <em>p</em>-values.</p>
<div class="figure"><span id="fig:glm1-plot1"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm1-plot1-1.png" alt="Histogram of the number of angiogenic sprouts in response to two of the four treatment combinations for the experiment in Example 1." width="576" />
<p class="caption">
Figure 20.1: Histogram of the number of angiogenic sprouts in response to two of the four treatment combinations for the experiment in Example 1.
</p>
</div>
<div id="the-generalized-linear-model-glm" class="section level3" number="20.1.1">
<h3><span class="header-section-number">20.1.1</span> The Generalized Linear Model (GLM)</h3>
<p>As outlined in <a href="intro-linear-models.html#lm-specifications">Two specifications of a linear model</a>, a common way that biological researchers are taught to think about a response variable is</p>
<p><span class="math display">\[
response = expected + error
\]</span></p>
<p>or, using the notation of this text,</p>
<p><span class="math display" id="eq:spec1">\[
\begin{align}
y &amp;= \beta_0 + \beta_1 \texttt{treatment} + \varepsilon \\
\varepsilon &amp;\sim \operatorname{Normal}(0, \sigma^2)
\tag{20.1}
\end{align}
\]</span></p>
<p>That is, we can think of a response as the sum of some systematic part (<span class="math inline">\(\beta_0 + \beta_1 \texttt{treatment}\)</span>) and a stochastic (“random error”) part (<span class="math inline">\(\varepsilon\)</span>), where the stochastic part is a random draw from a normal distribution with mean zero and variance <span class="math inline">\(\sigma^2\)</span>. This way of thinking about the generation of the response is useful for linear models, and model checking linear models, but is much less useful for thinking about generalized linear models or model checking generalized liner models. For example, if we want to model the number of angiogenic sprouts in response to some combination of GAS6 and treatment using a Poisson distribution, the following is the <strong>wrong</strong> way to think about the statistical model</p>
<p><span class="math display" id="eq:spec1-poisson">\[
\begin{align}
\texttt{sprouts} = &amp;\ \beta_0 + \beta_1 
\texttt{treatment}_{\texttt{GAS6}} + \beta_2 
\texttt{genotype}_{\texttt{FAK_ko}} + \\
&amp;\ \beta_3 \texttt{treatment}_{\texttt{GAS6}}:\texttt{genotype}_{\texttt{FAK_ko}} + \varepsilon_i\\
\varepsilon \sim &amp;\ \operatorname{Poisson}(\lambda)
\tag{20.2}
\end{align}
\]</span>
That is, we should not think of a count as the sum of a systematic part and a random draw from a Poisson distribution. Why? Because it is the counts, conditional on the <span class="math inline">\(\treattt{treatment\)</span> and <span class="math inline">\(\treattt{genotype\)</span>, that are poisson distributed, not the residuals from the fit model.</p>
<p>Thinking about the distribution of count data using model <a href="counts.html#eq:spec1-poisson">(20.2)</a> leads to absurd consequences. For example, if we set the mean of the Poisson “error” to zero (like with a normal distribution), then the error term for every observation would <em>have to</em> be zero (because the only way to get a mean of zero with non-negative integers is if every value is zero). Or, if the study is modeling the effect of a treatment on the counts (that is, the <span class="math inline">\(X\)</span> are dummy variables) then <span class="math inline">\(\beta_0\)</span> is the expected mean count of the control (or reference) group. But if we add non-zero Poisson error to this, then the mean of the control group would be larger than <span class="math inline">\(\beta_0\)</span>. This doesn’t make sense. And finally, equation <a href="counts.html#eq:spec1-poisson">(20.2)</a> generates a continuous response, instead of an integer, because <span class="math inline">\(\beta_0\)</span> and <span class="math inline">\(\beta_1\)</span> are continuous.</p>
<p>A better way to think about the data generation for a linear model that naturally leads to the <em>correct</em> way to think about data generation for a generalized linear model, is</p>
<p><span class="math display" id="eq:lm-spec2">\[
\begin{align}
y_i &amp;\sim N(\mu_i, \sigma^2)\\
\mathrm{E}(y_i| \texttt{treatment}) &amp;= \mu_i\\
\mu_i &amp;= \beta_0 + \beta_1 \texttt{treatment}_i
\tag{20.3}
\end{align}
\]</span></p>
<p>That is, a response is a random draw from a normal distribution with mean <span class="math inline">\(mu\)</span> (not zero!) and variance <span class="math inline">\(\sigma^2\)</span>. Line 1 is the stochastic part of this specification. Line 3 is the systematic part.</p>
<p>The specification of a generalized linear model has both stochastic and systematic parts but adds a third part, which is a <strong>link function</strong> connecting the stochastic and systematic parts.</p>
<ol style="list-style-type: decimal">
<li><strong>The stochastic part</strong>, which is a probability distribution from the exponential family (this is sometimes called the “random part”)
<span class="math display">\[
\begin{equation}
y_i \sim \operatorname{Prob}(\mu_i)
\end{equation}
\]</span></li>
<li><strong>the systematic part</strong>, which is a linear predictor (I like to think about this as the deterministic part)
<span class="math display">\[
\begin{equation}
\eta = \beta_0 + \beta_1 \texttt{treatment}_i
\end{equation}
\]</span></li>
<li>a <strong>link function</strong> connecting the two parts
<span class="math display">\[
\begin{equation}
\eta_i = g(\mu_i)
\end{equation}
\]</span></li>
</ol>
<p><span class="math inline">\(\mu\)</span> (the Greek symbol mu) is the conditional mean (or expectation <span class="math inline">\(\mathrm{E}(Y|X)\)</span>) of the response on the <strong>response scale</strong> and <span class="math inline">\(\eta\)</span> (the Greek symbol eta) is the conditional mean of the response on the <strong>link scale</strong>. The response scale is the scale of the raw measurements and has units of the raw measurements. The link scale is the scale of the transformed mean. A GLM models the response with a distribution specified in the stochastic part. The probability distributions introduced in this chapter are the Poisson and Negative Binomial. The natural link function, and default link function in R, for the Poisson and Negative Binomial is the “log link,” <span class="math inline">\(\eta = log(\mu)\)</span>. More generally, while each distribution has a natural (or, “canonical”) link function, one can use alternatives. Given this definition of a generalized linear model, a linear model is a GLM with a normal distribution and an <strong>Identity</strong> link (<span class="math inline">\(\eta = \mu\)</span>).</p>
<p><span class="math display">\[
\begin{align}
y_i &amp;\sim \operatorname{Normal}(\mu_i, \sigma^2)\\
\mathrm{E}(y_i| \texttt{treatment}) &amp;= \mu_i\\
\eta &amp;= \beta_0 + \beta_1 \texttt{treatment}\\
\mu_i &amp;= \eta_i
\end{align}
\]</span></p>
<p>Think about the link function and GLM more generally like this: the GLM constructs a linear model that predicts the conditional means <em>on the link scale</em>. If the model uses a “log link” (<span class="math inline">\(\eta_i = \mathrm{log}(\mu_i)\)</span>), then <span class="math inline">\(\eta_i\)</span> – the conditional mean on the link scale – is the <em>log of the modeled mean on the response scale</em>. The modeled mean on the response scale is the inverse link function.</p>
<p><span class="math display">\[
\begin{equation}
\mu_i = g^{-1}(\eta_i)
\end{equation}
\]</span></p>
<p>For a log link, a modeled mean is <span class="math inline">\(\mathrm{exp}(\eta_i)\)</span>)$.</p>
<p>Importantly, in a GLM, the individual data values <em>are not transformed</em>. A GLM with a log link is not the same as a linear model on log transformed data.</p>
<p>When modeling counts using the Poisson or negative binomial distributions with a log link, the link scale is linear, and so the effects are additive on the link scale, while the response scale is nonlinear (it is the exponent of the link scale), and so the <em>effects are multiplicative on the response scale</em>. If this doesn’t make sense now, an example is worked out below. The inverse of the link function backtransforms the parameters from the link scale back to the response scale. So, for example, a prediction on the response sale is <span class="math inline">\(\mathrm{exp}(\hat{\eta})\)</span> and a coefficient on the response scale is <span class="math inline">\(\mathrm{exp}(b_j)\)</span>.</p>
</div>
<div id="kinds-of-data-that-are-modeled-by-a-glm" class="section level3" number="20.1.2">
<h3><span class="header-section-number">20.1.2</span> Kinds of data that are modeled by a GLM</h3>
<ol style="list-style-type: decimal">
<li>If a response is a count then use a Poisson, quasi-Poisson or negative binomial family with a log link.</li>
<li>If a response is binary response, for example, presence/absence or success/failure or survive/die, then use the binomial family with a logistic link (other link functions are also useful). This is classically known as <strong>logistic regression</strong>.</li>
<li>If a response is a fraction that is a ratio of counts, for example the fraction of total cells that express some marker, then use the binomial family with a logistic link. Think of each count as a “success.”</li>
<li>If a response is a fraction of counts per “effort” (cells per area or volume or time), then use a Poisson, quasi-Poisson or negative binomial family with a log link on the raw count and include the measure of effort as an offset.</li>
<li>If a response is continuous but the variance increases with mean then use the gamma family</li>
<li>If a response is a fraction of a continuous measure per “effort” (tumor area per total area or volume or time), then use a gamma family on the raw measure and include the measure of effort as an offset.</li>
</ol>
</div>
</div>
<div id="glm-1" class="section level2" number="20.2">
<h2><span class="header-section-number">20.2</span> Example 1 – GLM models for count responses (“angiogenic sprouts” exp3a)</h2>
<div id="understand-the-data-3" class="section level3" number="20.2.1">
<h3><span class="header-section-number">20.2.1</span> Understand the data</h3>
<p>Article source <a href="https://www.nature.com/articles/s41467-020-16618-6" target="_blank">Lechertier, Tanguy, et al. “Pericyte FAK negatively regulates Gas6/Axl signalling to suppress tumour angiogenesis and tumour growth.” Nature communications 11.1 (2020): 1-14.</a></p>
<p><a href="https://www.nature.com/articles/s41467-020-16618-6#Sec34" target="_blank">data source</a></p>
<p>The researchers designed a set of experiments to investigate the effects of pericyte derived FAK (focal adhesion kinase, a <a href="https://en.wikipedia.org/wiki/Tyrosine_kinase" target="_blank">protein tyrosine kinase</a>) on the Gas6/Axl pathway regulating angiogenesis promoting tumor growth. Pericytes are cells immediately deep to the endothelium (the epithelial lining) of the smallest blood vessels, including capillaries, arterioles, and venules. Angiogenesis is the growth of new blood vessels. GAS6 (growth arrest specific 6) is a protein commonly expressed in tumors.</p>
<p>The example data is from the experiment for Figure 3a. The design is <span class="math inline">\(2 \times 2\)</span> – two factors (<span class="math inline">\(\texttt{treatment}\)</span> and <span class="math inline">\(\texttt{genotype}\)</span>) each with two levels.</p>
<ol style="list-style-type: decimal">
<li>Factor 1: <span class="math inline">\(\texttt{treatment}\)</span></li>
</ol>
<ul>
<li>reference level: “PBS.” Phosphate buffered saline added to tissue. This is the control treatment.</li>
<li>treatment level: “GAS6.” Added to tissue in solution. The experiment is designed to test its effect on promoting angiogenesis in the development of tumors.</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Factor 2: <span class="math inline">\(\texttt{genotype}\)</span></li>
</ol>
<ul>
<li>reference level: “FAK_wt.” The functional genotype.</li>
<li>treatment level: “FAK_ko.” Tissue-specific FAK deletion in pericytes. The experiment is designed to test the effect of pericyte-derived FAK on slowing angiogenesis in the development of tumors. If true, then its deletion should result in increased tumor development.</li>
</ul>
<p>The four treatment by genotype combinations are</p>
<ol style="list-style-type: decimal">
<li>Control (“PBS FAK_wt”) – negative control</li>
<li>FAK_ko (“PBS FAK_ko”) – PBS control. Unknown response given deletion of putative angiogenesis inhibitor but no added putative angiogenesis promotor</li>
<li>GAS6 (“GAS6 FAK_wt”) – GAS6 control. GAS6 expected to promote angiogenesis but this the response is expected to be inhibited by some amount by FAK</li>
<li>GAS6+FAK_ko (“GAS6 FAK_ko”) – focal treatment. Expected positive angiogenesis</li>
</ol>
<p>The planned contrasts are</p>
<ol style="list-style-type: decimal">
<li>(PBS FAK_ko) - (PBS FAK_wt) – the effect of FAK deletion given the control treatment</li>
<li>(GAS6 FAK_ko) - (GAS6 FAK_wt) – the effect of FAK deletion given the GAS6 treatment</li>
<li>((PBS FAK_ko) - (PBS FAK_wt)) - ((GAS6 FAK_ko) - (GAS6 FAK_wt)). The interaction effect giving the effect of the combined treatment relative to the individual effects.</li>
</ol>
</div>
<div id="model-fit-and-inference-1" class="section level3" number="20.2.2">
<h3><span class="header-section-number">20.2.2</span> Model fit and inference</h3>
<div id="fit-the-models-1" class="section level4" number="20.2.2.1">
<h4><span class="header-section-number">20.2.2.1</span> Fit the models</h4>
<div class="sourceCode" id="cb643"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb643-1"><a href="counts.html#cb643-1" aria-hidden="true" tabindex="-1"></a>exp3a_m1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(sprouts <span class="sc">~</span> treatment <span class="sc">*</span> genotype, <span class="at">data =</span> exp3a)</span>
<span id="cb643-2"><a href="counts.html#cb643-2" aria-hidden="true" tabindex="-1"></a>exp3a_m2 <span class="ot">&lt;-</span> <span class="fu">glm</span>(sprouts <span class="sc">~</span> treatment <span class="sc">*</span> genotype,</span>
<span id="cb643-3"><a href="counts.html#cb643-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">family =</span> <span class="st">&quot;poisson&quot;</span>,</span>
<span id="cb643-4"><a href="counts.html#cb643-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp3a)</span>
<span id="cb643-5"><a href="counts.html#cb643-5" aria-hidden="true" tabindex="-1"></a>exp3a_m3 <span class="ot">&lt;-</span> <span class="fu">glm.nb</span>(sprouts <span class="sc">~</span> treatment <span class="sc">*</span> genotype,</span>
<span id="cb643-6"><a href="counts.html#cb643-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp3a)</span></code></pre></div>
</div>
<div id="check-the-linear-model" class="section level4" number="20.2.2.2">
<h4><span class="header-section-number">20.2.2.2</span> Check the linear model</h4>
<div class="sourceCode" id="cb644"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb644-1"><a href="counts.html#cb644-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcheck_the_model</span>(exp3a_m1)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3a_m1-check-lm-1.png" width="576" />
Notes</p>
<ol style="list-style-type: decimal">
<li>left panel shows classic right skew conditional distribution with larger values much larger than expected with a normal distribution.</li>
<li>right panel shows heterogeneity and specifically the variance increasing with the mean.</li>
</ol>
</div>
<div id="check-the-poisson-model" class="section level4" number="20.2.2.3">
<h4><span class="header-section-number">20.2.2.3</span> Check the poisson model</h4>
<p><strong>Check shape and homogeneity</strong></p>
<div class="sourceCode" id="cb645"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb645-1"><a href="counts.html#cb645-1" aria-hidden="true" tabindex="-1"></a><span class="co"># from the DHARMa package</span></span>
<span id="cb645-2"><a href="counts.html#cb645-2" aria-hidden="true" tabindex="-1"></a>  exp3a_m2_simulation <span class="ot">&lt;-</span> <span class="fu">simulateResiduals</span>(<span class="at">fittedModel =</span> exp3a_m2, <span class="at">n =</span> <span class="dv">250</span>)</span>
<span id="cb645-3"><a href="counts.html#cb645-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(exp3a_m2_simulation, <span class="at">asFactor =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3a_m2-check-poisson-1.png" width="576" /></p>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>poisson glm fails to generated scaled residuals approximating uniform distribution.</li>
</ol>
<p><strong>Check dispersion</strong></p>
<div class="sourceCode" id="cb646"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb646-1"><a href="counts.html#cb646-1" aria-hidden="true" tabindex="-1"></a>exp3a_m2_simulation_refit <span class="ot">&lt;-</span> <span class="fu">simulateResiduals</span>(<span class="at">fittedModel =</span> exp3a_m2,</span>
<span id="cb646-2"><a href="counts.html#cb646-2" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">n =</span> <span class="dv">250</span>,</span>
<span id="cb646-3"><a href="counts.html#cb646-3" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">refit =</span> <span class="cn">TRUE</span>)</span>
<span id="cb646-4"><a href="counts.html#cb646-4" aria-hidden="true" tabindex="-1"></a>exp3a_m2_test_dispersion <span class="ot">&lt;-</span> <span class="fu">testDispersion</span>(exp3a_m2_simulation_refit)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3a_m2-check-dispersion-1.png" width="576" /></p>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>large overdispersion</li>
</ol>
<p><strong>Check zero inflation</strong></p>
<div class="sourceCode" id="cb647"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb647-1"><a href="counts.html#cb647-1" aria-hidden="true" tabindex="-1"></a>exp3a_m2_test_zi <span class="ot">&lt;-</span> <span class="fu">testZeroInflation</span>(exp3a_m2_simulation_refit)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3a_m2-check-zero-1.png" width="576" /></p>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The data has too many zeros relative to the expected number from a poisson GLM.</li>
</ol>
</div>
<div id="check-the-negative-binomial-model" class="section level4" number="20.2.2.4">
<h4><span class="header-section-number">20.2.2.4</span> Check the negative binomial model</h4>
<p><strong>Check shape and homogeneity</strong></p>
<div class="sourceCode" id="cb648"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb648-1"><a href="counts.html#cb648-1" aria-hidden="true" tabindex="-1"></a><span class="co"># from the DHARMa package</span></span>
<span id="cb648-2"><a href="counts.html#cb648-2" aria-hidden="true" tabindex="-1"></a>  exp3a_m3_simulation <span class="ot">&lt;-</span> <span class="fu">simulateResiduals</span>(<span class="at">fittedModel =</span> exp3a_m3,</span>
<span id="cb648-3"><a href="counts.html#cb648-3" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">n =</span> <span class="dv">250</span>)</span>
<span id="cb648-4"><a href="counts.html#cb648-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(exp3a_m3_simulation, <span class="at">asFactor =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3a_m3-check-nb-1.png" width="576" /></p>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>uniform q-q for negative binomial GLM looks good.</li>
<li>spread-location plot looks good.</li>
</ol>
<p><strong>Check dispersion</strong></p>
<div class="sourceCode" id="cb649"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb649-1"><a href="counts.html#cb649-1" aria-hidden="true" tabindex="-1"></a>exp3a_m3_simulation_refit <span class="ot">&lt;-</span> <span class="fu">simulateResiduals</span>(<span class="at">fittedModel =</span> exp3a_m3,</span>
<span id="cb649-2"><a href="counts.html#cb649-2" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">n =</span> <span class="dv">250</span>,</span>
<span id="cb649-3"><a href="counts.html#cb649-3" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">refit =</span> <span class="cn">TRUE</span>)</span>
<span id="cb649-4"><a href="counts.html#cb649-4" aria-hidden="true" tabindex="-1"></a>exp3a_m3_test_dispersion <span class="ot">&lt;-</span> <span class="fu">testDispersion</span>(exp3a_m3_simulation_refit)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3a_m3_test_dispersion-1.png" width="576" /></p>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>good</li>
</ol>
<p><strong>Check zero inflation</strong></p>
<div class="sourceCode" id="cb650"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb650-1"><a href="counts.html#cb650-1" aria-hidden="true" tabindex="-1"></a>exp3a_m3_test_zi <span class="ot">&lt;-</span> <span class="fu">testZeroInflation</span>(exp3a_m3_simulation_refit)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3a_m3_test_zi-1.png" width="576" /></p>
</div>
<div id="glm-exp3a-m3-inference" class="section level4" number="20.2.2.5">
<h4><span class="header-section-number">20.2.2.5</span> Inference from the model</h4>
<div class="sourceCode" id="cb651"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb651-1"><a href="counts.html#cb651-1" aria-hidden="true" tabindex="-1"></a>exp3a_m3_coef <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">coef</span>(<span class="fu">summary</span>(exp3a_m3)),</span>
<span id="cb651-2"><a href="counts.html#cb651-2" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">confint</span>(exp3a_m3))</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
z value
</th>
<th style="text-align:right;">
Pr(&gt;|z|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
0.90
</td>
<td style="text-align:right;">
0.231
</td>
<td style="text-align:right;">
3.9
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
1.36
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentGAS6
</td>
<td style="text-align:right;">
1.30
</td>
<td style="text-align:right;">
0.287
</td>
<td style="text-align:right;">
4.5
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.74
</td>
<td style="text-align:right;">
1.86
</td>
</tr>
<tr>
<td style="text-align:left;">
genotypeFAK_ko
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
0.357
</td>
<td style="text-align:right;">
0.1
</td>
<td style="text-align:right;">
0.900
</td>
<td style="text-align:right;">
-0.65
</td>
<td style="text-align:right;">
0.75
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentGAS6:genotypeFAK_ko
</td>
<td style="text-align:right;">
0.46
</td>
<td style="text-align:right;">
0.425
</td>
<td style="text-align:right;">
1.1
</td>
<td style="text-align:right;">
0.275
</td>
<td style="text-align:right;">
-0.37
</td>
<td style="text-align:right;">
1.30
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb652"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb652-1"><a href="counts.html#cb652-1" aria-hidden="true" tabindex="-1"></a>exp3a_m3_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp3a_m3,</span>
<span id="cb652-2"><a href="counts.html#cb652-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;genotype&quot;</span>),</span>
<span id="cb652-3"><a href="counts.html#cb652-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type=</span><span class="st">&quot;response&quot;</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:left;">
genotype
</th>
<th style="text-align:right;">
response
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
asymp.LCL
</th>
<th style="text-align:right;">
asymp.UCL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PBS
</td>
<td style="text-align:left;">
FAK_wt
</td>
<td style="text-align:right;">
2.47
</td>
<td style="text-align:right;">
0.6
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
1.57
</td>
<td style="text-align:right;">
3.88
</td>
</tr>
<tr>
<td style="text-align:left;">
GAS6
</td>
<td style="text-align:left;">
FAK_wt
</td>
<td style="text-align:right;">
9.05
</td>
<td style="text-align:right;">
1.5
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
6.48
</td>
<td style="text-align:right;">
12.64
</td>
</tr>
<tr>
<td style="text-align:left;">
PBS
</td>
<td style="text-align:left;">
FAK_ko
</td>
<td style="text-align:right;">
2.58
</td>
<td style="text-align:right;">
0.7
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
1.52
</td>
<td style="text-align:right;">
4.40
</td>
</tr>
<tr>
<td style="text-align:left;">
GAS6
</td>
<td style="text-align:left;">
FAK_ko
</td>
<td style="text-align:right;">
15.04
</td>
<td style="text-align:right;">
2.4
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
11.06
</td>
<td style="text-align:right;">
20.46
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb653"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb653-1"><a href="counts.html#cb653-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exp3a_m3_emm # print in console to get row numbers</span></span>
<span id="cb653-2"><a href="counts.html#cb653-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb653-3"><a href="counts.html#cb653-3" aria-hidden="true" tabindex="-1"></a>pbs_fak_wt <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb653-4"><a href="counts.html#cb653-4" aria-hidden="true" tabindex="-1"></a>gas6_fak_wt <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb653-5"><a href="counts.html#cb653-5" aria-hidden="true" tabindex="-1"></a>pbs_fak_ko <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb653-6"><a href="counts.html#cb653-6" aria-hidden="true" tabindex="-1"></a>gas6_fak_ko <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb653-7"><a href="counts.html#cb653-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb653-8"><a href="counts.html#cb653-8" aria-hidden="true" tabindex="-1"></a><span class="co">#1. (PBS FAK_ko) - (PBS FAK_wt)</span></span>
<span id="cb653-9"><a href="counts.html#cb653-9" aria-hidden="true" tabindex="-1"></a><span class="co">#2. (GAS6 FAK_ko) - (GAS6 FAK_wt) </span></span>
<span id="cb653-10"><a href="counts.html#cb653-10" aria-hidden="true" tabindex="-1"></a><span class="co">#3. ((PBS FAK_ko) - (PBS FAK_wt)) - (GAS6 FAK_ko) - (GAS6 FAK_wt). </span></span>
<span id="cb653-11"><a href="counts.html#cb653-11" aria-hidden="true" tabindex="-1"></a>exp3a_m3_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(</span>
<span id="cb653-12"><a href="counts.html#cb653-12" aria-hidden="true" tabindex="-1"></a>  exp3a_m3_emm,</span>
<span id="cb653-13"><a href="counts.html#cb653-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb653-14"><a href="counts.html#cb653-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;(PBS FAK_ko) - (PBS FAK_wt)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(pbs_fak_ko <span class="sc">-</span> pbs_fak_wt),</span>
<span id="cb653-15"><a href="counts.html#cb653-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;(GAS6 FAK_ko) - (GAS6 FAK_wt)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(gas6_fak_ko <span class="sc">-</span> gas6_fak_wt),</span>
<span id="cb653-16"><a href="counts.html#cb653-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;Interaction&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(gas6_fak_ko <span class="sc">-</span> gas6_fak_wt) <span class="sc">-</span></span>
<span id="cb653-17"><a href="counts.html#cb653-17" aria-hidden="true" tabindex="-1"></a>      <span class="fu">c</span>(pbs_fak_ko <span class="sc">-</span> pbs_fak_wt)</span>
<span id="cb653-18"><a href="counts.html#cb653-18" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb653-19"><a href="counts.html#cb653-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb653-20"><a href="counts.html#cb653-20" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb653-21"><a href="counts.html#cb653-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
ratio
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
asymp.LCL
</th>
<th style="text-align:right;">
asymp.UCL
</th>
<th style="text-align:right;">
z.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(PBS FAK_ko) / (PBS FAK_wt)
</td>
<td style="text-align:right;">
1.05
</td>
<td style="text-align:right;">
0.373
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.52
</td>
<td style="text-align:right;">
2.10
</td>
<td style="text-align:right;">
0.13
</td>
<td style="text-align:right;">
0.90040
</td>
</tr>
<tr>
<td style="text-align:left;">
(GAS6 FAK_ko) / (GAS6 FAK_wt)
</td>
<td style="text-align:right;">
1.66
</td>
<td style="text-align:right;">
0.385
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
1.06
</td>
<td style="text-align:right;">
2.62
</td>
<td style="text-align:right;">
2.19
</td>
<td style="text-align:right;">
0.02820
</td>
</tr>
<tr>
<td style="text-align:left;">
Interaction
</td>
<td style="text-align:right;">
1.59
</td>
<td style="text-align:right;">
0.676
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.69
</td>
<td style="text-align:right;">
3.66
</td>
<td style="text-align:right;">
1.09
</td>
<td style="text-align:right;">
0.27535
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
</div>
<div id="lmm-exp5c-plotthemodel" class="section level4" number="20.2.2.6">
<h4><span class="header-section-number">20.2.2.6</span> Plot the model</h4>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/unnamed-chunk-283-1.png" width="576" /></p>
</div>
<div id="alternaplot-the-model-3" class="section level4" number="20.2.2.7">
<h4><span class="header-section-number">20.2.2.7</span> Alternaplot the model</h4>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/unnamed-chunk-284-1.png" width="576" /></p>
</div>
</div>
</div>
<div id="understanding-example-1" class="section level2" number="20.3">
<h2><span class="header-section-number">20.3</span> Understanding Example 1</h2>
<div id="modeling-strategy-1" class="section level3" number="20.3.1">
<h3><span class="header-section-number">20.3.1</span> Modeling strategy</h3>
<p>Instead of testing assumptions of a model using formal hypothesis tests <em>before</em> fitting the model, a better strategy is to 1) fit one or more models based on initial evaluation of the data, and then do 2) <strong>model checking</strong> using <strong>diagnostic plots</strong>, diagnostic statistics, and simulation (see Section <a href="model-checking.html#check-check">All statistical analyses should be followed by model checking</a>).</p>
<p>For the exp3a data, I fit a linear model, a Poisson GLM, and a negative binomial GLM. I use the diagnostic plots and statistics to help me decide which model to report.</p>
</div>
<div id="model-checking-fits-to-count-data" class="section level3" number="20.3.2">
<h3><span class="header-section-number">20.3.2</span> Model checking fits to count data</h3>
<p>We use the fit models to check</p>
<ol style="list-style-type: decimal">
<li>the compatibility between the quantiles of the observed residuals and the distribution of expected quantiles from the family in the model fit</li>
<li>if the observed distribution is over or under dispersed</li>
<li>if there are more zeros than expected by the theoretical distribution. If so, the observed distribution is <strong>zero-inflated</strong></li>
</ol>
<div id="glm-exp3a-normal-qq" class="section level4" number="20.3.2.1">
<h4><span class="header-section-number">20.3.2.1</span> Checking the linear model exp3a_m1 – a Normal-QQ plot</h4>
<p>Figure <a href="#fig:glm1-plot2"><strong>??</strong></a>A shows a histogram of the residuals from the fit linear model. The plot shows that the residuals seem to be clumped at the negative end of the range, which suggests that a model with a normally distributed conditional outcome (or normal error) is not well approximated.</p>
<div class="figure"><span id="fig:glm1-exp3a-qq"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm1-exp3a-qq-1.png" alt="Diagnostic plots of angiogenic sprout data (exp3a). A) Distribution of the residuals of the fit linear model. B) Normal-QQ plot of the residuals of the fit linear model." width="672" />
<p class="caption">
Figure 20.2: Diagnostic plots of angiogenic sprout data (exp3a). A) Distribution of the residuals of the fit linear model. B) Normal-QQ plot of the residuals of the fit linear model.
</p>
</div>
<p>A better way to investigate this is with the <strong>Normal-QQ</strong> plot in Figure <a href="#fig:glm1-plot2"><strong>??</strong></a>B, which plots the sample quantiles for a variable against their theoretical quantiles. If the conditional outcome approximates a normal distribution,</p>
<ol style="list-style-type: decimal">
<li>the points should roughly follow the robust regression line,</li>
<li>the points should be largely inside the 95% CI (gray) bounds for sampling a normal distribution with the variance estimated by the model, and</li>
<li>the robust regression line should be largely inside the CI bounds.</li>
</ol>
<p>For the sprout data, the points are above the line at the positive end, hug the upper bound of the 95% CI at the negative end, are well above the 95% CI at the positive end, and the robust regression is distinctly shallower than a line bisecting the CI bounds. At the left (negative) end, the observed values are more positive than the theoretical values. Remembering that this plot is of residuals, if we think about this as counts, this means that our smallest counts are not as small as we would expect given the mean, the variance, and a normal distribution. This shouldn’t be surprising – the counts range down to zero and counts cannot be below zero. At the positive end, the sample values are again more positive than the theoretical values. Thinking about this as counts, this means that are largest counts are larger than expected given the mean, the variance, and a normal distribution. This pattern is what we’d expect of count data.</p>
</div>
<div id="glm-exp3a_m1-spreadlevel" class="section level4" number="20.3.2.2">
<h4><span class="header-section-number">20.3.2.2</span> Checking the linear model exp3a_m1 – Spread-level plot for checking homoskedasticity</h4>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm1-exp3a-qq-scale-location-1.png" width="576" /></p>
<p>A linear model also assumes that the error is <strong>homoskedastic</strong> – the error variance is not a function of the value of the <span class="math inline">\(X\)</span> variables). Non-homoskedastic error is <strong>heteroskedastic</strong>. I will typically use “homogenous variance” and “heterogenous variance” since these terms are more familiar to biologists. The fit model can be checked for homogeneity using a spread-level (also known as a scale-location) plot, which comes in several forms. I like a spread-level plot that is a scatterplot of the positive square-root of the standardized residuals against the fitted values (remember that the fitted values are the values computed by the linear predictor of the model – they are the “predicted values” of the observed data). If the residuals approximate a normal distribution, then a regression line through the scatter should be close to horizontal. The regression line in the spread-level plot of the fit of the linear model to the sprout data shows a distinct increase in the “scale” (the square root of the standardized residuals) with increased fitted value, which is expected of data sampled from a distribution in which the variance increases the mean.</p>
</div>
<div id="two-distributions-for-count-data-poisson-and-negative-binomial" class="section level4" number="20.3.2.3">
<h4><span class="header-section-number">20.3.2.3</span> Two distributions for count data – Poisson and Negative Binomial</h4>
<p>The pattern in the Normal-QQ plot in Figure <a href="#fig:glm1-plot2"><strong>??</strong></a>B should discourage a researcher from modeling the data with a normal distribution and instead model the data with an alternative distribution using a Generalized Linear Model. There is no unique mapping between observed data and a data generating mechanism with a specific distribution, so this decision is not as easy as thinking about the data generation mechanism and then simply choosing the “correct” distribution. Section 4.5 in Bolker (xxx) is an excellent summary of how to think about the generating processes for different distributions in the context of ecological data. Since the response in the angiogenic sprouts data are counts, we need to choose a distribution that generates integer values, such as the Poisson or the negative binomial.</p>
<ol style="list-style-type: decimal">
<li>Poisson – A Poisson distribution is the probability distribution of the number of occurrences of some thing (a white blood cell, a tumor, or a specific mRNA transcript) generated by a process that generates the thing at a constant rate per unit effort (duration or space). This constant rate is the parameter <span class="math inline">\(\lambda\)</span>, which is the expectation (the expected mean of the counts), so <span class="math inline">\(\mathrm{E}(Y) = \mu = \lambda\)</span>. Because the rate per effort is constant, <em>the variance of a Poisson variable equals the mean</em>, <span class="math inline">\(\sigma^2 = \mu = \lambda\)</span>. Figure <a href="#fig:glm1-poisson"><strong>??</strong></a> shows three samples from a Poisson distribution with <span class="math inline">\(\lambda\)</span> set to 1, 5, and 10. The plots show that, as the mean count (<span class="math inline">\(\lambda\)</span>) moves away from zero, a Poisson distribution 1) becomes less skewed and more closely approximates a normal distribution and 2) has an increasingly low probability of including zero (less than 1% zeros when the mean is 5).</li>
</ol>
<p>A Poisson distribution, then, is useful for count data in which the conditional variance is close to the conditional mean. Very often, biological count data are not well approximated by a Poisson distribution because the variance is either less than the mean, an example of <strong>underdispersion</strong><a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>, or greater than the mean, an example of <strong>overdispersion</strong><a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a>. A useful distribution for count data with overdispersion is the negative binomial.</p>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm1-fake-poisson-1.png" width="672" /></p>
<ol start="2" style="list-style-type: decimal">
<li>Negative Binomial – The negative binomial distribution is a discrete probability distribution of the number of successes that occur before a specified number of failures <span class="math inline">\(k\)</span> given a probability <em>p</em> of success. This isn’t a very useful way of thinking about modeling count data in biology. What is useful is that the Negative Binomial distribution can be used simply as way of modeling an “overdispersed” Poisson process. Using the parameterization in the <code>MASS::glm.nb</code> function, the mean of a negative binomial variable is <span class="math inline">\(\mu\)</span> and the variance is <span class="math inline">\(\sigma^2 = \mu + \frac{\mu^2}{\theta}\)</span>. As a method for modeling an overdispersed Poisson variable, <span class="math inline">\(\theta\)</span> functions as a **dispersion parameter* controlling the amount of overdispersion and can be any real, positive value (not simply a positive integer), including values less than 1. As <span class="math inline">\(\theta\)</span> approaches positive infinity, the “overdispersion” bit <span class="math inline">\(\frac{\mu^2}{\theta}\)</span> goes to zero and the variance goes to <span class="math inline">\(\mu\)</span>, which is the same as the Poisson.</li>
</ol>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm1-fake-nb-1.png" width="672" /></p>
</div>
<div id="model-checking-a-glm-i-the-quantile-residual-uniform-qq-plot" class="section level4" number="20.3.2.4">
<h4><span class="header-section-number">20.3.2.4</span> Model checking a GLM I – the quantile-residual uniform-QQ plot</h4>
<p>Normal-QQ plots were introduced in Section @ref{normal-qq} of the Model Checking chapter and applied to the linear model fit of the angiogenic sprout data in Section <a href="counts.html#glm-exp3a-normal-qq">20.3.2.1</a> above. We cannot use a Normal-QQ plot with a Poisson or negative binomial GLM fit because the residuals from this fit are not expected to be normally distributed. An alternative to a Normal-QQ plot for a GLM fit is a <strong>quantile-residual uniform-QQ</strong> plot of observed <strong>quantile residuals</strong>.</p>
<div class="figure">
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3a_m2-check-poisson-again-1.png" alt="Quantile-residual uniform-QQ plot of the Poisson GLM fit to the angiogenic sprouts (exp3a) data." width="576" />
<p class="caption">
(#fig:glm-exp3a_m2-check-poisson-again)Quantile-residual uniform-QQ plot of the Poisson GLM fit to the angiogenic sprouts (exp3a) data.
</p>
</div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The x-axis (“Expected”) contains the expected quantiles from a uniform distribution.</li>
<li>The y-axis (“Observed”) contains the observed quantile residuals from a GLM fit, which are the residuals from the fit model that are transformed in a way that the expected distribution is uniform under the fit model family. This means that we’d expect the quantile residuals to closely approximate the expected quantiles from a uniform distribution. If the approximation is close, the points will fall along the “y = x” line in the plot.</li>
<li>The gray shaded area is a 95% confidence interval computed using a parametric bootstrap. At any value of the expected quantile, the interval will include an observed quantile 95% of the time. This gray area gives us a sense of the variability we’d get when we fit models to random samples from the specified model.</li>
<li>In the quantile-residual QQ plot for Model <code>exp3a_m2</code>, the observed residuals are far outside the 95% boundary. The observed residuals are smaller than expected at the negative (left) end and larger than expected at the right (high) end. This means the residuals are more spread out than expected for a Poisson sample. The data are overdispersed <em>for this model</em>. Understand that overdispersion is not a property of data but of the residuals from a specific model fit to the data.</li>
</ol>
<div style="background-color:#cccccc; text-align:left; vertical-align: middle; padding:20px 47px;">
<p><strong>Misconceivable</strong> – A common misconception is that if the distribution of the response approximates a Poisson distribution, then the residuals of a GLM fit with a Poisson distribution should be normally distributed, which could then be checked with a Normal-QQ plot, and homoskedastic, which could be checked with a scale-location plot. Neither of these is true because a GLM does not transform the data and, in fact, the model definition does not specify anything about the distribution of an “error” term – there is no <span class="math inline">\(\varepsilon\)</span> in the model definition above! This is why thinking about the definition of a linear model by specifying an error term with a normal distribution can be confusing and lead to misconceptions when learning GLMs.</p>
</div>
</div>
<div id="model-checking-a-glm-ii-spread-level-plot-for-checking-homoskedasticity" class="section level4" number="20.3.2.5">
<h4><span class="header-section-number">20.3.2.5</span> Model checking a GLM II – Spread-level plot for checking homoskedasticity</h4>
<div class="figure">
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3a_m2-spreadlevel-again-1.png" alt="Spread-level plot of the Poisson GLM fit to the angiogenic sprouts (exp3a) data." width="576" />
<p class="caption">
(#fig:glm-exp3a_m2-spreadlevel-again)Spread-level plot of the Poisson GLM fit to the angiogenic sprouts (exp3a) data.
</p>
</div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The three curves are <strong>quantile regressions</strong> fit to the 25% (the first quartile), 50% (the second quartile or median), and 75% (the third quartile) quantiles conditional on the fitted value (the x-axis). The <span class="math inline">\(p\%\)</span> quantile regression line goes through the <span class="math inline">\(p\%\)</span> quantile of the <span class="math inline">\(Y\)</span> values at each value of <span class="math inline">\(X\)</span>. Think of it like this: at any location along the x-axis, there is a spread of points on the y-axis that contains a first quartile, a median, and a third quartile. The 25%, 50%, and 75% quantile regression lines go through these points.</li>
<li>Why fit the three quartiles? The classic spread-level plot typically fits a regression, which is the expected (mean) residual, conditional on the fitted value (for example, the plot in section @ref(glm-exp3a_m1-spreadlevel)). Researchers typically look at this and think, “the whole spread of points is becoming more variable as the fitted value increases” but the regression is only modeling the mean. By fitting the first, second, and third quartile regressions, this spread-level plot evaluates deviations from expected spread not just at the mean but in the lower and upper halves of the spread.</li>
</ol>
</div>
<div id="model-checking-a-glm-iii-checking-dispersion" class="section level4" number="20.3.2.6">
<h4><span class="header-section-number">20.3.2.6</span> Model checking a GLM III – Checking dispersion</h4>
<div class="figure">
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3a_m2-check-dispersion-again-1.png" alt="Dispersion plot of the negative binomial GLM fit to the angiogenic sprouts (exp3a) data." width="576" />
<p class="caption">
(#fig:glm-exp3a_m2-check-dispersion-again)Dispersion plot of the negative binomial GLM fit to the angiogenic sprouts (exp3a) data.
</p>
</div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>This plot is a histogram of the sum of squared <strong>Pearson residuals</strong> of fake data sampled from the fit model. Pearson residuals are the raw residuals divided by the square root of the fitted value. Remember that in the Poisson distribution, the variance is equal to the expectation (mean), so a Pearson residual is the raw residual divided by the standard deviation of the residual. A way to think about this is, Pearson residuals “correct” for the heterogeneity in variance that arises among groups with different mean counts.</li>
<li>The sum of squared <strong>Pearson residuals</strong> is a measure of the dispersion of the residuals.</li>
<li>The red line is the observed sum of the squared Pearson residuals of the fit model.</li>
<li>If the observed dispersion approximates that expected from sampling from the fit model, the red line will be within the histogram.</li>
<li>The red line here is far larger than expected given the histogram, which indicates that the residuals are overdispersed given the fit model.</li>
<li>Overdispersion will be common with Poisson GLM fits to biological data.</li>
</ol>
</div>
<div id="model-checking-a-count-glm-check-zero-inflation" class="section level4" number="20.3.2.7">
<h4><span class="header-section-number">20.3.2.7</span> Model Checking a count GLM – Check zero inflation</h4>
<p>Counts can have the value zero. Data that have more zeros than expected given a fit count GLM model (Poisson, quasi-Poisson, negative binomial) is <strong>zero-inflated</strong>.</p>
<div class="sourceCode" id="cb654"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb654-1"><a href="counts.html#cb654-1" aria-hidden="true" tabindex="-1"></a>zero_inflation_test <span class="ot">&lt;-</span> <span class="fu">testZeroInflation</span>(simulation_output)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/unnamed-chunk-286-1.png" width="576" /></p>
<ol style="list-style-type: decimal">
<li>This plot is a histogram of the number of zeros in each of the fake data sets generated by the fit model. An observed number of zeros at the extremes of this distribution are unlikely given the fit model. The number of zeros in the observed data is greater than expected by the model.</li>
</ol>
</div>
</div>
<div id="biological-count-data-are-rarely-fit-well-by-a-poisson-glm.-instead-fit-a-quasi-poisson-or-negative-binomial-glm-model." class="section level3" number="20.3.3">
<h3><span class="header-section-number">20.3.3</span> Biological count data are rarely fit well by a Poisson GLM. Instead, fit a quasi-poisson or negative binomial GLM model.</h3>
<p>Here are the diagnostic plots of the negative binomial GLM fit to the exp3a data</p>
<div class="figure">
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3a_m3-qq-again-1.png" alt="Quantile-residual uniform-QQ plot of the negative binomial GLM fit to the angiogenic sprouts (exp3a) data." width="576" />
<p class="caption">
(#fig:glm-exp3a_m3-qq-again)Quantile-residual uniform-QQ plot of the negative binomial GLM fit to the angiogenic sprouts (exp3a) data.
</p>
</div>
<div class="figure">
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3a_m3-spreadlevel-again-1.png" alt="Spread-level plot of the negative binomial GLM fit to the angiogenic sprouts (exp3a) data." width="576" />
<p class="caption">
(#fig:glm-exp3a_m3-spreadlevel-again)Spread-level plot of the negative binomial GLM fit to the angiogenic sprouts (exp3a) data.
</p>
</div>
<div class="figure">
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3a_m3-check-dispersion-again-1.png" alt="Dispersion plot of the negative binomial GLM fit to the angiogenic sprouts (exp3a) data." width="576" />
<p class="caption">
(#fig:glm-exp3a_m3-check-dispersion-again)Dispersion plot of the negative binomial GLM fit to the angiogenic sprouts (exp3a) data.
</p>
</div>
</div>
<div id="a-glm-is-a-linear-model-on-the-link-scale" class="section level3" number="20.3.4">
<h3><span class="header-section-number">20.3.4</span> A GLM is a linear model on the link scale</h3>
<p>The negative-binomial GLM fit to the angiogenic sprout data (exp3a) is</p>
<p><span class="math display" id="eq:nb-glm-exp3a">\[
\begin{align}
\texttt{sprouts}_i &amp;\sim \operatorname{NB}(\mu_i, \theta)\\
\mathrm{E}(\texttt{sprouts}\ | \ \texttt{treatment, genotype}) &amp;= \mu\\
\mu_i &amp;= \mathrm{exp}(\eta_i)\\
\eta_i &amp;= \beta_0 \ + \\
&amp; \quad \; \beta_1 \texttt{treatment}_{\texttt{GAS6}} \ + \\
&amp; \quad \; \beta_2 \texttt{genotype}_{\texttt{FAK_ko}} \ + \\
&amp; \quad \; \beta_3 \texttt{treatment}_{\texttt{GAS6}}:\texttt{genotype}_{\texttt{FAK_ko}}
\tag{20.4}
\end{align}
\]</span></p>
<ol style="list-style-type: decimal">
<li>The first line of Model <a href="counts.html#eq:nb-glm-exp3a">(20.4)</a> is the stochastic part stating the response is modeled as a random Negative Binomial variable with conditional mean <span class="math inline">\(\mu_i\)</span>, and variance <span class="math inline">\(\mu + \frac{\mu^2}{\theta}\)</span>. Fitting the model to the data estimates <span class="math inline">\(\mu_i\)</span> for all <em>i</em>. <span class="math inline">\(\mu_i\)</span> will be the same for all mice within each treatment by genotype combination because they share the same conditions (the values of the indicator variables for treatment, genotype, and their interaction).</li>
<li>The second line states the <span class="math inline">\(\mu\)</span> is the mean conditional on the value of <span class="math inline">\(\texttt{treatment}\)</span> and <span class="math inline">\(\texttt{genotype}\)</span></li>
<li>The third line connects the conditional mean on the link scale (<span class="math inline">\(\eta\)</span>) with the conditional mean on the response scale (<span class="math inline">\(\mu\)</span>). This is the backtransformation.</li>
<li>The fourth line is the <strong>linear predictor</strong> – it is a linear model on the link scale. The linear predictor includes three indicator variables.</li>
<li>Remember that a linear model is a model in which the coefficients are additive, meaning that the coefficients do not have exponents or are not multiplied by each other.</li>
</ol>
</div>
<div id="coeffecients-of-a-generalized-linear-model-with-a-log-link-function-are-on-the-link-scale." class="section level3" number="20.3.5">
<h3><span class="header-section-number">20.3.5</span> Coeffecients of a Generalized Linear Model with a log-link function are on the link scale.</h3>
<p>The coefficients of the fit negative binomial model are</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
z value
</th>
<th style="text-align:right;">
Pr(&gt;|z|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
0.90446
</td>
<td style="text-align:right;">
0.231
</td>
<td style="text-align:right;">
3.9
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
1.36
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentGAS6
</td>
<td style="text-align:right;">
1.29805
</td>
<td style="text-align:right;">
0.287
</td>
<td style="text-align:right;">
4.5
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.74
</td>
<td style="text-align:right;">
1.86
</td>
</tr>
<tr>
<td style="text-align:left;">
genotypeFAK_ko
</td>
<td style="text-align:right;">
0.04462
</td>
<td style="text-align:right;">
0.357
</td>
<td style="text-align:right;">
0.1
</td>
<td style="text-align:right;">
0.900
</td>
<td style="text-align:right;">
-0.65
</td>
<td style="text-align:right;">
0.75
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentGAS6:genotypeFAK_ko
</td>
<td style="text-align:right;">
0.46382
</td>
<td style="text-align:right;">
0.425
</td>
<td style="text-align:right;">
1.1
</td>
<td style="text-align:right;">
0.275
</td>
<td style="text-align:right;">
-0.37
</td>
<td style="text-align:right;">
1.30
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The coefficients are on the link scale, which is a linear (additive) scale.</li>
</ol>
<div id="the-intercept-of-a-generalized-linear-model-with-a-log-link-function-is-the-mean-of-the-reference-on-the-link-scale" class="section level4" number="20.3.5.1">
<h4><span class="header-section-number">20.3.5.1</span> The intercept of a Generalized Linear Model with a log-link function is the mean of the reference on the link scale</h4>
<ol style="list-style-type: decimal">
<li>In the linear model <code>exp3a_m1</code>, the intercept is the modeled mean of the reference group. In a GLM with a log-link (including Models <code>exp3a_m2</code> and <code>exp3a_m3</code>), the intercept is the mean of the reference group <em>on the link scale</em>. The modeled mean of the reference on the response scale is computed by the backtransformaiton <span class="math inline">\(\exp(b_1)\)</span>. The function “exp(x)” is the exponent, which is often written using the notation <span class="math inline">\(e^{x}\)</span>.</li>
<li>The transformation between link scale and response scale is part of the specification of a Generalized Linear Model. For the negative binomial GLM fit to the exp3a data, this specification is in line 3 of the specification (Model <a href="counts.html#eq:nb-glm-exp3a">(20.4)</a>).</li>
</ol>
<p>Compare the computation here with the modeled mean in the emmeans table (Section <a href="counts.html#glm-exp3a-m3-inference">20.2.2.5</a> or with a computation of the sample mean.</p>
<div class="sourceCode" id="cb655"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb655-1"><a href="counts.html#cb655-1" aria-hidden="true" tabindex="-1"></a>b1 <span class="ot">&lt;-</span> <span class="fu">coef</span>(exp3a_m3)[<span class="dv">1</span>]</span>
<span id="cb655-2"><a href="counts.html#cb655-2" aria-hidden="true" tabindex="-1"></a><span class="fu">exp</span>(b1)</span></code></pre></div>
<pre><code>## (Intercept) 
##    2.470588</code></pre>
<div class="sourceCode" id="cb657"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb657-1"><a href="counts.html#cb657-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(exp3a[t_by_g <span class="sc">==</span> <span class="st">&quot;PBS FAK_wt&quot;</span>, sprouts])</span></code></pre></div>
<pre><code>## [1] 2.470588</code></pre>
</div>
<div id="the-coefficients-of-the-indicator-variables-of-a-generalized-linear-model-with-a-log-link-function-are-effect-ratios-on-the-response-scale" class="section level4" number="20.3.5.2">
<h4><span class="header-section-number">20.3.5.2</span> The coefficients of the indicator variables of a Generalized Linear Model with a log-link function are effect-ratios on the response scale</h4>
<ol style="list-style-type: decimal">
<li>In the linear model <code>exp3a_m1</code>, the coefficient of an indicator variable is a difference in means. In a GLM with a log-link, the coefficient of an indicator variable is the difference (group minus reference) of the modeled means <em>on the link scale</em>. Let’s check this with some computations.</li>
</ol>
<p>The modeled means on the link scale are in the emmeans table</p>
<div class="sourceCode" id="cb659"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb659-1"><a href="counts.html#cb659-1" aria-hidden="true" tabindex="-1"></a>exp3a_m3_emm_link <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp3a_m3,</span>
<span id="cb659-2"><a href="counts.html#cb659-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;genotype&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb659-3"><a href="counts.html#cb659-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>() <span class="sc">%&gt;%</span></span>
<span id="cb659-4"><a href="counts.html#cb659-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span></code></pre></div>
<p>Using this table, the difference between the link-scale mean of “GAS6 FAK_wt” and “PBS FAK_wt” (the reference) is</p>
<div class="sourceCode" id="cb660"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb660-1"><a href="counts.html#cb660-1" aria-hidden="true" tabindex="-1"></a>exp3a_m3_emm_link[treatment <span class="sc">==</span> <span class="st">&quot;GAS6&quot;</span> <span class="sc">&amp;</span></span>
<span id="cb660-2"><a href="counts.html#cb660-2" aria-hidden="true" tabindex="-1"></a>                    genotype <span class="sc">==</span> <span class="st">&quot;FAK_wt&quot;</span>, emmean] <span class="sc">-</span></span>
<span id="cb660-3"><a href="counts.html#cb660-3" aria-hidden="true" tabindex="-1"></a>  exp3a_m3_emm_link[treatment <span class="sc">==</span> <span class="st">&quot;PBS&quot;</span> <span class="sc">&amp;</span> genotype <span class="sc">==</span> <span class="st">&quot;FAK_wt&quot;</span>, emmean]</span></code></pre></div>
<pre><code>## [1] 1.298045</code></pre>
<p>And the coefficient for the <span class="math inline">\(\texttt{treatment_gas6}\)</span> indicator variable is</p>
<div class="sourceCode" id="cb662"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb662-1"><a href="counts.html#cb662-1" aria-hidden="true" tabindex="-1"></a><span class="co"># coefficient for treatment_gas6</span></span>
<span id="cb662-2"><a href="counts.html#cb662-2" aria-hidden="true" tabindex="-1"></a>b2 <span class="ot">&lt;-</span> <span class="fu">coef</span>(exp3a_m3)[<span class="dv">2</span>]</span>
<span id="cb662-3"><a href="counts.html#cb662-3" aria-hidden="true" tabindex="-1"></a>b2</span></code></pre></div>
<pre><code>## treatmentGAS6 
##      1.298045</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>In a GLM with a log-link, the exponent of the coefficient of an indicator variable is the ratio of the modeled means <em>on the response scale</em>. Let’s check this with some computations.</li>
</ol>
<p>The modeled means on the response scale are in the emmeans table</p>
<div class="sourceCode" id="cb664"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb664-1"><a href="counts.html#cb664-1" aria-hidden="true" tabindex="-1"></a>exp3a_m3_emm_response <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp3a_m3,</span>
<span id="cb664-2"><a href="counts.html#cb664-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;genotype&quot;</span>),</span>
<span id="cb664-3"><a href="counts.html#cb664-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">type =</span> <span class="st">&quot;response&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb664-4"><a href="counts.html#cb664-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>() <span class="sc">%&gt;%</span></span>
<span id="cb664-5"><a href="counts.html#cb664-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span></code></pre></div>
<p>Using this table, the ratio of the response-scale mean of “GAS6 FAK_wt” to the response-scale mean of “PBS FAK_wt” (the reference) is</p>
<div class="sourceCode" id="cb665"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb665-1"><a href="counts.html#cb665-1" aria-hidden="true" tabindex="-1"></a>exp3a_m3_emm_response[treatment <span class="sc">==</span> <span class="st">&quot;GAS6&quot;</span> <span class="sc">&amp;</span></span>
<span id="cb665-2"><a href="counts.html#cb665-2" aria-hidden="true" tabindex="-1"></a>                    genotype <span class="sc">==</span> <span class="st">&quot;FAK_wt&quot;</span>, response]<span class="sc">/</span></span>
<span id="cb665-3"><a href="counts.html#cb665-3" aria-hidden="true" tabindex="-1"></a>  exp3a_m3_emm_response[treatment <span class="sc">==</span> <span class="st">&quot;PBS&quot;</span> <span class="sc">&amp;</span> genotype <span class="sc">==</span> <span class="st">&quot;FAK_wt&quot;</span>, response]</span></code></pre></div>
<pre><code>## [1] 3.662132</code></pre>
<p>And the coefficient for the <span class="math inline">\(\texttt{treatment_gas6}\)</span> indicator variable backtransformed to the response scale is</p>
<div class="sourceCode" id="cb667"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb667-1"><a href="counts.html#cb667-1" aria-hidden="true" tabindex="-1"></a>exp_b2 <span class="ot">&lt;-</span> <span class="fu">exp</span>(b2)</span>
<span id="cb667-2"><a href="counts.html#cb667-2" aria-hidden="true" tabindex="-1"></a>exp_b2</span></code></pre></div>
<pre><code>## treatmentGAS6 
##      3.662132</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Since this backtransformed coefficient is both an effect and a ratio, I call it an <strong>effect ratio</strong>. It’s value is how many times bigger (or smaller if less than one) the non-reference response is relative to the reference response. The response of the GAS6 treatment is 3.66<span class="math inline">\(\times\)</span> that of the reference treatment.</li>
</ol>
</div>
</div>
<div id="modeled-means-in-the-emmeans-table-of-a-generalized-linear-model-can-be-on-the-link-scale-or-response-scale-report-the-response-scale" class="section level3" number="20.3.6">
<h3><span class="header-section-number">20.3.6</span> Modeled means in the emmeans table of a Generalized Linear Model can be on the link scale or response scale – Report the response scale</h3>
<div id="glm-exp3a-emmans-link" class="section level4" number="20.3.6.1">
<h4><span class="header-section-number">20.3.6.1</span> The emmeans table on the link scale contains the log means</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
(#tab:exp3a_m3_emm_link)Emmeans table on the link scale
</caption>
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:left;">
genotype
</th>
<th style="text-align:right;">
emmean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
asymp.LCL
</th>
<th style="text-align:right;">
asymp.UCL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PBS
</td>
<td style="text-align:left;">
FAK_wt
</td>
<td style="text-align:right;">
0.90446
</td>
<td style="text-align:right;">
0.23
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
1.36
</td>
</tr>
<tr>
<td style="text-align:left;">
GAS6
</td>
<td style="text-align:left;">
FAK_wt
</td>
<td style="text-align:right;">
2.20250
</td>
<td style="text-align:right;">
0.17
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
1.87
</td>
<td style="text-align:right;">
2.54
</td>
</tr>
<tr>
<td style="text-align:left;">
PBS
</td>
<td style="text-align:left;">
FAK_ko
</td>
<td style="text-align:right;">
0.94908
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.42
</td>
<td style="text-align:right;">
1.48
</td>
</tr>
<tr>
<td style="text-align:left;">
GAS6
</td>
<td style="text-align:left;">
FAK_ko
</td>
<td style="text-align:right;">
2.71094
</td>
<td style="text-align:right;">
0.16
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
2.40
</td>
<td style="text-align:right;">
3.02
</td>
</tr>
</tbody>
</table>
<ol style="list-style-type: decimal">
<li>The values are the statistics on the link scale. The mean and CI, but not the SE, can be meaningfully backtransformed to the response scale using the exponent function to get the mean and CI on the response scale.</li>
<li>The CIs are “asymptotic,” meaning they are computed using infinite degrees of freedom. The consequences is that the CIs will be too narrow, especially for small <em>n</em>.</li>
<li>Check the math! Given asymptotic CIs, the lower CI should be (mean - 1.96 <span class="math inline">\(\times\)</span> SE) and the upper should be (mean + 1.96 <span class="math inline">\(\times\)</span> SE).</li>
</ol>
<div class="sourceCode" id="cb669"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb669-1"><a href="counts.html#cb669-1" aria-hidden="true" tabindex="-1"></a><span class="fl">0.90446</span> <span class="sc">-</span> <span class="fl">1.96</span><span class="sc">*</span><span class="fl">0.23</span></span></code></pre></div>
<pre><code>## [1] 0.45366</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>A GLM is linear on the link scale. This means the model is additive on the link scale. Modeled means on the link scale are computed by <em>adding</em> the coefficients.</li>
</ol>
<div class="sourceCode" id="cb671"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb671-1"><a href="counts.html#cb671-1" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">coef</span>(exp3a_m3) <span class="co"># the model coefficients</span></span>
<span id="cb671-2"><a href="counts.html#cb671-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb671-3"><a href="counts.html#cb671-3" aria-hidden="true" tabindex="-1"></a><span class="co"># the way I typically compute these</span></span>
<span id="cb671-4"><a href="counts.html#cb671-4" aria-hidden="true" tabindex="-1"></a>pbs_fak_wt_link <span class="ot">&lt;-</span> b[<span class="dv">1</span>] </span>
<span id="cb671-5"><a href="counts.html#cb671-5" aria-hidden="true" tabindex="-1"></a>gas6_fak_wt_link <span class="ot">&lt;-</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] </span>
<span id="cb671-6"><a href="counts.html#cb671-6" aria-hidden="true" tabindex="-1"></a>pbs_fak_ko_link <span class="ot">&lt;-</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">3</span>]</span>
<span id="cb671-7"><a href="counts.html#cb671-7" aria-hidden="true" tabindex="-1"></a>gas6_fak_ko_link <span class="ot">&lt;-</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>] <span class="sc">+</span> b[<span class="dv">4</span>]</span>
<span id="cb671-8"><a href="counts.html#cb671-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb671-9"><a href="counts.html#cb671-9" aria-hidden="true" tabindex="-1"></a><span class="co"># but this is the algebra more consistent with the linear model math</span></span>
<span id="cb671-10"><a href="counts.html#cb671-10" aria-hidden="true" tabindex="-1"></a>pbs_fak_wt_link <span class="ot">&lt;-</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>]<span class="sc">*</span><span class="dv">0</span> <span class="sc">+</span> b[<span class="dv">3</span>]<span class="sc">*</span><span class="dv">0</span> <span class="sc">+</span> b[<span class="dv">4</span>]<span class="sc">*</span><span class="dv">0</span></span>
<span id="cb671-11"><a href="counts.html#cb671-11" aria-hidden="true" tabindex="-1"></a>gas6_fak_wt_link <span class="ot">&lt;-</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>]<span class="sc">*</span><span class="dv">1</span> <span class="sc">+</span> b[<span class="dv">3</span>]<span class="sc">*</span><span class="dv">0</span> <span class="sc">+</span> b[<span class="dv">4</span>]<span class="sc">*</span><span class="dv">0</span></span>
<span id="cb671-12"><a href="counts.html#cb671-12" aria-hidden="true" tabindex="-1"></a>pbs_fak_ko_link <span class="ot">&lt;-</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>]<span class="sc">*</span><span class="dv">0</span> <span class="sc">+</span> b[<span class="dv">3</span>]<span class="sc">*</span><span class="dv">1</span> <span class="sc">+</span> b[<span class="dv">4</span>]<span class="sc">*</span><span class="dv">0</span></span>
<span id="cb671-13"><a href="counts.html#cb671-13" aria-hidden="true" tabindex="-1"></a>gas6_fak_ko_link <span class="ot">&lt;-</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>]<span class="sc">*</span><span class="dv">1</span> <span class="sc">+</span> b[<span class="dv">3</span>]<span class="sc">*</span><span class="dv">1</span> <span class="sc">+</span> b[<span class="dv">4</span>]<span class="sc">*</span><span class="dv">1</span></span>
<span id="cb671-14"><a href="counts.html#cb671-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb671-15"><a href="counts.html#cb671-15" aria-hidden="true" tabindex="-1"></a><span class="co"># combine into a table</span></span>
<span id="cb671-16"><a href="counts.html#cb671-16" aria-hidden="true" tabindex="-1"></a>exp_3a_means <span class="ot">&lt;-</span> <span class="fu">data.table</span>(</span>
<span id="cb671-17"><a href="counts.html#cb671-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">group =</span> t_by_g_levels,</span>
<span id="cb671-18"><a href="counts.html#cb671-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;mean (link scale)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(pbs_fak_wt_link,</span>
<span id="cb671-19"><a href="counts.html#cb671-19" aria-hidden="true" tabindex="-1"></a>                          gas6_fak_wt_link,</span>
<span id="cb671-20"><a href="counts.html#cb671-20" aria-hidden="true" tabindex="-1"></a>                          pbs_fak_ko_link,</span>
<span id="cb671-21"><a href="counts.html#cb671-21" aria-hidden="true" tabindex="-1"></a>                          gas6_fak_ko_link)</span>
<span id="cb671-22"><a href="counts.html#cb671-22" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-288">Table 20.1: </span>Table of group treatment by genotype means on the link scale
</caption>
<thead>
<tr>
<th style="text-align:left;">
group
</th>
<th style="text-align:right;">
mean (link scale)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PBS FAK_wt
</td>
<td style="text-align:right;">
0.90446
</td>
</tr>
<tr>
<td style="text-align:left;">
GAS6 FAK_wt
</td>
<td style="text-align:right;">
2.20250
</td>
</tr>
<tr>
<td style="text-align:left;">
PBS FAK_ko
</td>
<td style="text-align:right;">
0.94908
</td>
</tr>
<tr>
<td style="text-align:left;">
GAS6 FAK_ko
</td>
<td style="text-align:right;">
2.71094
</td>
</tr>
</tbody>
</table>
<p>Compare the values here to those in the emmeans table on the link scale (Table @ref(tab:exp3a_m3_emm_link)).</p>
</div>
<div id="the-emmeans-table-on-the-response-scale-contains-more-readily-interpretable-means" class="section level4" number="20.3.6.2">
<h4><span class="header-section-number">20.3.6.2</span> The emmeans table on the response scale contains more readily interpretable means</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
(#tab:exp3a_m3_emm_response)Emmeans table on the response scale
</caption>
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:left;">
genotype
</th>
<th style="text-align:right;">
response
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
asymp.LCL
</th>
<th style="text-align:right;">
asymp.UCL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PBS
</td>
<td style="text-align:left;">
FAK_wt
</td>
<td style="text-align:right;">
2.47059
</td>
<td style="text-align:right;">
0.57
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
1.57
</td>
<td style="text-align:right;">
3.88
</td>
</tr>
<tr>
<td style="text-align:left;">
GAS6
</td>
<td style="text-align:left;">
FAK_wt
</td>
<td style="text-align:right;">
9.04762
</td>
<td style="text-align:right;">
1.54
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
6.48
</td>
<td style="text-align:right;">
12.64
</td>
</tr>
<tr>
<td style="text-align:left;">
PBS
</td>
<td style="text-align:left;">
FAK_ko
</td>
<td style="text-align:right;">
2.58333
</td>
<td style="text-align:right;">
0.70
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
1.52
</td>
<td style="text-align:right;">
4.40
</td>
</tr>
<tr>
<td style="text-align:left;">
GAS6
</td>
<td style="text-align:left;">
FAK_ko
</td>
<td style="text-align:right;">
15.04348
</td>
<td style="text-align:right;">
2.36
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
11.06
</td>
<td style="text-align:right;">
20.46
</td>
</tr>
</tbody>
</table>
<ol style="list-style-type: decimal">
<li>The values in the column “response” are the modeled means on the response scale. These values are the exponent of the values in the “emmean” column of the emmeans table on the link scale.</li>
<li>The values in the columns “asymp.LCL” and “asymp.UCL” are the 95% confidence intervals on the response scale. These values are the exponent of the values in the same columns of the emmeans table on the link scale.</li>
<li>Don’t do additive math on the response scale! Remember, a GLM is linear on the link scale. The CIs on the response scale are not the mean plus or minus 1.96 <span class="math inline">\(\times\)</span> SE!</li>
</ol>
<div class="sourceCode" id="cb672"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb672-1"><a href="counts.html#cb672-1" aria-hidden="true" tabindex="-1"></a><span class="fl">2.47059</span> <span class="sc">-</span> <span class="fl">1.96</span><span class="sc">*</span><span class="fl">0.57</span></span></code></pre></div>
<pre><code>## [1] 1.35339</code></pre>
<div class="sourceCode" id="cb674"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb674-1"><a href="counts.html#cb674-1" aria-hidden="true" tabindex="-1"></a><span class="co"># oops</span></span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Modeled means on the response scale are computed by backtransforming the modeled means on the link scale. Since the fit model used a log-link, the backtransformation is the exponent.</li>
</ol>
<div class="sourceCode" id="cb675"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb675-1"><a href="counts.html#cb675-1" aria-hidden="true" tabindex="-1"></a>pbs_fak_wt_response <span class="ot">&lt;-</span> <span class="fu">exp</span>(pbs_fak_wt_link)</span>
<span id="cb675-2"><a href="counts.html#cb675-2" aria-hidden="true" tabindex="-1"></a>gas6_fak_wt_response <span class="ot">&lt;-</span> <span class="fu">exp</span>(gas6_fak_wt_link)</span>
<span id="cb675-3"><a href="counts.html#cb675-3" aria-hidden="true" tabindex="-1"></a>pbs_fak_ko_response <span class="ot">&lt;-</span> <span class="fu">exp</span>(pbs_fak_ko_link)</span>
<span id="cb675-4"><a href="counts.html#cb675-4" aria-hidden="true" tabindex="-1"></a>gas6_fak_ko_response <span class="ot">&lt;-</span> <span class="fu">exp</span>(gas6_fak_ko_link)</span>
<span id="cb675-5"><a href="counts.html#cb675-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb675-6"><a href="counts.html#cb675-6" aria-hidden="true" tabindex="-1"></a><span class="co"># add a column to the exp_3a_means data.table</span></span>
<span id="cb675-7"><a href="counts.html#cb675-7" aria-hidden="true" tabindex="-1"></a>exp_3a_means[, <span class="st">&quot;mean (response scale)&quot;</span> <span class="sc">:</span><span class="er">=</span> <span class="fu">c</span>(pbs_fak_wt_response,</span>
<span id="cb675-8"><a href="counts.html#cb675-8" aria-hidden="true" tabindex="-1"></a>                                            gas6_fak_wt_response,</span>
<span id="cb675-9"><a href="counts.html#cb675-9" aria-hidden="true" tabindex="-1"></a>                                            pbs_fak_ko_response,</span>
<span id="cb675-10"><a href="counts.html#cb675-10" aria-hidden="true" tabindex="-1"></a>                                            gas6_fak_ko_response)]</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-291">Table 20.2: </span>Table of group treatment by genotype means on both the link and response scale.
</caption>
<thead>
<tr>
<th style="text-align:left;">
group
</th>
<th style="text-align:right;">
mean (link scale)
</th>
<th style="text-align:right;">
mean (response scale)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PBS FAK_wt
</td>
<td style="text-align:right;">
0.90446
</td>
<td style="text-align:right;">
2.47059
</td>
</tr>
<tr>
<td style="text-align:left;">
GAS6 FAK_wt
</td>
<td style="text-align:right;">
2.20250
</td>
<td style="text-align:right;">
9.04762
</td>
</tr>
<tr>
<td style="text-align:left;">
PBS FAK_ko
</td>
<td style="text-align:right;">
0.94908
</td>
<td style="text-align:right;">
2.58333
</td>
</tr>
<tr>
<td style="text-align:left;">
GAS6 FAK_ko
</td>
<td style="text-align:right;">
2.71094
</td>
<td style="text-align:right;">
15.04348
</td>
</tr>
</tbody>
</table>
<p>Compare the values here to those in the emmeans table on the response scale (Table @ref(tab:exp3a_m3_emm_response)).</p>
</div>
</div>
<div id="some-consequences-of-fitting-a-linear-model-to-count-data" class="section level3" number="20.3.7">
<h3><span class="header-section-number">20.3.7</span> Some consequences of fitting a linear model to count data</h3>
<div id="one-linear-models-can-make-absurd-predictions" class="section level4" number="20.3.7.1">
<h4><span class="header-section-number">20.3.7.1</span> One – linear models can make absurd predictions</h4>
<div class="sourceCode" id="cb676"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb676-1"><a href="counts.html#cb676-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(gg1, gg2, gg3, <span class="at">ncol=</span><span class="dv">3</span>, <span class="at">labels =</span> <span class="st">&quot;AUTO&quot;</span>)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/unnamed-chunk-292-1.png" width="1066.66666666667" /></p>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>A prediction interval is a confidence interval of a prediction – using the fit model to predict future responses given the same conditions (here, assignment to one of the four different treatment combinations).</li>
<li>Left panel: The prediction intervals from the linear model imply that negative sprouts could be sampled. This is absurd.</li>
<li>Middle panel: The fit linear model is used to make 100 fake predictions in each group.</li>
<li>Right panel: The fit negative binomial GLM is used to make 100 fake predictions in each group. Nothing absurd here.</li>
</ol>
</div>
<div id="two-linear-models-can-perform-surprisingly-well-if-one-is-only-interested-in-p-values" class="section level4" number="20.3.7.2">
<h4><span class="header-section-number">20.3.7.2</span> Two – linear models can perform surprisingly well if one is only interested in <em>p</em>-values</h4>
<p><em>P</em>-values are a function of the sampling distribution of group means and differences in means, and, due to the magic of the <a href="https://en.wikipedia.org/wiki/Central_limit_theorem" target="_blank">central limit theorem</a>, linear models fit to count data perform surprisingly well in the sense of</p>
<ol style="list-style-type: decimal">
<li>Type I error that approximates the nominal value</li>
<li>Reasonable power compared to GLM models and many non-parametric tests.</li>
</ol>
</div>
</div>
</div>
<div id="example-2-use-a-glm-with-an-offset-instead-of-a-ratio-of-some-measurement-per-area-dna-damage-data-exp3b" class="section level2" number="20.4">
<h2><span class="header-section-number">20.4</span> Example 2 – Use a GLM with an offset instead of a ratio of some measurement per area (“dna damage” data exp3b)</h2>
<p>A problem that often arises in count data (and other kinds of measures) are counts that are taken in samples with different areas or volumes of tissue. As a consequence, size and treatment response are confounded – samples with higher counts may have higher counts because of a different response to treatment, a larger amount of tissue, or some combination. The common practice in experimental biology is to adjust for tissue size variation by area-normalizing the count using the ratio <span class="math inline">\(\frac{count}{area}\)</span> and then testing for a difference in the normalized count using either a linear model NHST (<em>t</em>-test/ANOVA) or a non-parametric NHST (Mann-Whitney-Wilcoxan). These practices raise at least two statistical issues.</p>
<ol style="list-style-type: decimal">
<li>The ratio will have some kind of <a href="https://en.wikipedia.org/wiki/Ratio_distribution" target="_blank">ratio distribution</a> that violates the normal distribution assumption of the linear model.</li>
<li>A Mann-Whitney-Wilcoxan does not estimate meaningful effects, is less powerful than GLM models, and is not flexible for complex designs including factorial or covariate models.</li>
</ol>
<p>A better practice is to model the count using a GLM that adds the denominator of the ratio (the area or volume) as an <strong>offset</strong> in the model. A count GLM with offset also models the ratio (see section <a href="counts.html#glm-rate">20.5.2</a> below), but in a way that is likely to be much more compatible with the data.</p>
<p>NHST of the ratios will perform okay in the sense of Type I error that is close to nominal but will have relatively low power compared to a generalized linear model with offset. If the researcher is interested in best practices including the reporting of uncertainty of estimated effects, a GLM with offset will have more useful confidence intervals – for example CIs from linear model assuming Normal error can often include absurd values such as ratios less than zero.
Source article (Fernández, Álvaro F., et al. “Disruption of the beclin 1–BCL2 autophagy regulatory complex promotes longevity in mice.” Nature 558.7708 (2018): 136-140.)<span target="_blank"><a href="https://www.nature.com/articles/s41586-018-0162-7" class="uri">https://www.nature.com/articles/s41586-018-0162-7</a></span></p>
<div id="exp3b-dna-damage-data" class="section level3" number="20.4.1">
<h3><span class="header-section-number">20.4.1</span> exp3b (“dna damage”) data</h3>
<p>These data were first introduced in the Issues chapter, Section <a href="issues.html#issues-offset">15.4.2</a>. There only the age 20 month data were analyzed. Here the full dataset is analyzed.</p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5992097/" target="_blank">Public source</a></p>
<p><a href="https://www.nature.com/articles/s41586-018-0162-7#Sec15" target="_blank">Source data for Fig. 3</a></p>
<p>The example here is from Fig 3b.</p>
</div>
<div id="understand-the-data-4" class="section level3" number="20.4.2">
<h3><span class="header-section-number">20.4.2</span> Understand the data</h3>
<p>Response variable – number of <a href="https://en.wikipedia.org/wiki/TUNEL_assay" target="_blank">TUNEL+</a> cells measured in kidney tissue, where a positive marker indicates nuclear DNA damage.</p>
<p>Background. The experiments in Figure 3 were designed to measure the effect of a knock-in mutation in the gene for the <a href="https://en.wikipedia.org/wiki/BECN1" target="_blank">beclin 1 protein</a> on autophagy and tissue health in the kidney and heart. The researchers were interested in autophagy because there is evidence in many non-mammalian model organisms that increased autophagy reduces age-related damage to tissues and increases health and lifespan. <a href="https://en.wikipedia.org/wiki/Bcl-2" target="_blank">BCL2</a> is an autophagy inhibitor. Initial experiments showed that the knock-in mutation in beclin 1 inhibits BCL2. Inhibiting BCL2 with the knock-in mutation should increase autophagy and, as a consequence, reduce age-related tissue damage.</p>
<p>Design - <span class="math inline">\(2 \times 2\)</span> factorial with offset</p>
<ol style="list-style-type: decimal">
<li>Factor 1: <span class="math inline">\(\texttt{age}\)</span> with levels “Young” (two months) and “Old” (twenty months).</li>
<li>Factor 2: <span class="math inline">\(\texttt{genotype}\)</span> with levels “WT” (wildtype) and “KI” (knock-in).</li>
<li>Offset: <span class="math inline">\(\texttt{area_mm2}\)</span>. The area of the kidney tissue containing the counted cells.</li>
</ol>
<p>The planned contrasts are</p>
<ol style="list-style-type: decimal">
<li>(Young KI) - (Young WT) – the effect of the beclin 1 knock-in mutation in the 2 month old mice. We expect this effect to be negative (more damage in the WT) but small, assuming there is little BCL2-related DNA damage by 2 months.</li>
<li>(Old KI) - (Old WT) – the effect of the beclin 1 knock-in mutation in the 20 month old mice. We expect this effect to be negative (more damage in the WT) but large assuming there is substantial BCL2-related DNA damage by 20 months.</li>
<li>Interaction. The magnitude of the interaction will largely be a function of the difference in BCL2-related DNA damage between 2 and 20 months and not really a function with how the knockin functions at these two ages.</li>
</ol>
</div>
<div id="model-fit-and-inference-2" class="section level3" number="20.4.3">
<h3><span class="header-section-number">20.4.3</span> Model fit and inference</h3>
<div id="fit-the-models-2" class="section level4" number="20.4.3.1">
<h4><span class="header-section-number">20.4.3.1</span> Fit the models</h4>
<div class="sourceCode" id="cb677"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb677-1"><a href="counts.html#cb677-1" aria-hidden="true" tabindex="-1"></a><span class="co"># lm with ratio response</span></span>
<span id="cb677-2"><a href="counts.html#cb677-2" aria-hidden="true" tabindex="-1"></a>exp3b_m1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(count_per_area <span class="sc">~</span> age <span class="sc">*</span> genotype,</span>
<span id="cb677-3"><a href="counts.html#cb677-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> exp3b)</span>
<span id="cb677-4"><a href="counts.html#cb677-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb677-5"><a href="counts.html#cb677-5" aria-hidden="true" tabindex="-1"></a><span class="co"># poisson offset</span></span>
<span id="cb677-6"><a href="counts.html#cb677-6" aria-hidden="true" tabindex="-1"></a>exp3b_m2 <span class="ot">&lt;-</span> <span class="fu">glm</span>(positive_nuclei <span class="sc">~</span> age <span class="sc">*</span> genotype <span class="sc">+</span></span>
<span id="cb677-7"><a href="counts.html#cb677-7" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">offset</span>(<span class="fu">log</span>(area_mm2)),</span>
<span id="cb677-8"><a href="counts.html#cb677-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">family =</span> <span class="st">&quot;poisson&quot;</span>,</span>
<span id="cb677-9"><a href="counts.html#cb677-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp3b)</span>
<span id="cb677-10"><a href="counts.html#cb677-10" aria-hidden="true" tabindex="-1"></a><span class="co"># nb offset</span></span>
<span id="cb677-11"><a href="counts.html#cb677-11" aria-hidden="true" tabindex="-1"></a>exp3b_m3 <span class="ot">&lt;-</span> <span class="fu">glm.nb</span>(positive_nuclei <span class="sc">~</span> age <span class="sc">*</span> genotype <span class="sc">+</span></span>
<span id="cb677-12"><a href="counts.html#cb677-12" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">offset</span>(<span class="fu">log</span>(area_mm2)),</span>
<span id="cb677-13"><a href="counts.html#cb677-13" aria-hidden="true" tabindex="-1"></a>                   <span class="at">data =</span> exp3b)</span>
<span id="cb677-14"><a href="counts.html#cb677-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb677-15"><a href="counts.html#cb677-15" aria-hidden="true" tabindex="-1"></a><span class="co"># non-parametric</span></span>
<span id="cb677-16"><a href="counts.html#cb677-16" aria-hidden="true" tabindex="-1"></a>exp3b_m4_2mo <span class="ot">&lt;-</span> <span class="fu">wilcox.test</span>(count_per_area <span class="sc">~</span> genotype,</span>
<span id="cb677-17"><a href="counts.html#cb677-17" aria-hidden="true" tabindex="-1"></a>                            <span class="at">data =</span> exp3b[age <span class="sc">==</span> <span class="st">&quot;Young&quot;</span>])</span>
<span id="cb677-18"><a href="counts.html#cb677-18" aria-hidden="true" tabindex="-1"></a>exp3b_m4_20mo <span class="ot">&lt;-</span> <span class="fu">wilcox.test</span>(count_per_area <span class="sc">~</span> genotype,</span>
<span id="cb677-19"><a href="counts.html#cb677-19" aria-hidden="true" tabindex="-1"></a>                             <span class="at">data =</span> exp3b[age <span class="sc">==</span> <span class="st">&quot;Old&quot;</span>])</span></code></pre></div>
</div>
<div id="check-the-models-1" class="section level4" number="20.4.3.2">
<h4><span class="header-section-number">20.4.3.2</span> Check the models</h4>
<div class="sourceCode" id="cb678"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb678-1"><a href="counts.html#cb678-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcheck_the_model</span>(exp3b_m1)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3b_m1-check-1.png" width="576" /></p>
<div class="sourceCode" id="cb679"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb679-1"><a href="counts.html#cb679-1" aria-hidden="true" tabindex="-1"></a>exp3b_m2_simulation <span class="ot">&lt;-</span> <span class="fu">simulateResiduals</span>(<span class="at">fittedModel =</span> exp3b_m2, <span class="at">n =</span> <span class="dv">250</span>)</span>
<span id="cb679-2"><a href="counts.html#cb679-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(exp3b_m2_simulation, <span class="at">asFactor =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3b_m2_simulation-1.png" width="576" /></p>
<div class="sourceCode" id="cb680"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb680-1"><a href="counts.html#cb680-1" aria-hidden="true" tabindex="-1"></a>exp3b_m3_simulation <span class="ot">&lt;-</span> <span class="fu">simulateResiduals</span>(<span class="at">fittedModel =</span> exp3b_m3,</span>
<span id="cb680-2"><a href="counts.html#cb680-2" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">n =</span> <span class="dv">250</span>)</span>
<span id="cb680-3"><a href="counts.html#cb680-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(exp3b_m3_simulation, <span class="at">asFactor =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3b_m3_simulation-1.png" width="576" /></p>
</div>
<div id="glm-exp3b-m3-inference" class="section level4" number="20.4.3.3">
<h4><span class="header-section-number">20.4.3.3</span> Inference from the model</h4>
<div class="sourceCode" id="cb681"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb681-1"><a href="counts.html#cb681-1" aria-hidden="true" tabindex="-1"></a>exp3b_m3_coef <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">coef</span>(<span class="fu">summary</span>(exp3b_m3)),</span>
<span id="cb681-2"><a href="counts.html#cb681-2" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">confint</span>(exp3b_m3))</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
z value
</th>
<th style="text-align:right;">
Pr(&gt;|z|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
-0.07
</td>
<td style="text-align:right;">
0.508
</td>
<td style="text-align:right;">
-0.1
</td>
<td style="text-align:right;">
0.887
</td>
<td style="text-align:right;">
-0.95
</td>
<td style="text-align:right;">
1.09
</td>
</tr>
<tr>
<td style="text-align:left;">
ageOld
</td>
<td style="text-align:right;">
0.86
</td>
<td style="text-align:right;">
0.575
</td>
<td style="text-align:right;">
1.5
</td>
<td style="text-align:right;">
0.133
</td>
<td style="text-align:right;">
-0.39
</td>
<td style="text-align:right;">
1.91
</td>
</tr>
<tr>
<td style="text-align:left;">
genotypeKI
</td>
<td style="text-align:right;">
-0.10
</td>
<td style="text-align:right;">
0.722
</td>
<td style="text-align:right;">
-0.1
</td>
<td style="text-align:right;">
0.887
</td>
<td style="text-align:right;">
-1.56
</td>
<td style="text-align:right;">
1.36
</td>
</tr>
<tr>
<td style="text-align:left;">
ageOld:genotypeKI
</td>
<td style="text-align:right;">
-0.66
</td>
<td style="text-align:right;">
0.806
</td>
<td style="text-align:right;">
-0.8
</td>
<td style="text-align:right;">
0.416
</td>
<td style="text-align:right;">
-2.27
</td>
<td style="text-align:right;">
0.96
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb682"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb682-1"><a href="counts.html#cb682-1" aria-hidden="true" tabindex="-1"></a>mean_area <span class="ot">&lt;-</span> <span class="fu">mean</span>(exp3b[age <span class="sc">==</span> <span class="st">&quot;Old&quot;</span> <span class="sc">&amp;</span> genotype <span class="sc">==</span> <span class="st">&quot;WT&quot;</span>,</span>
<span id="cb682-2"><a href="counts.html#cb682-2" aria-hidden="true" tabindex="-1"></a>                        positive_nuclei])</span>
<span id="cb682-3"><a href="counts.html#cb682-3" aria-hidden="true" tabindex="-1"></a>exp3b_m3_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp3b_m3,</span>
<span id="cb682-4"><a href="counts.html#cb682-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;age&quot;</span>, <span class="st">&quot;genotype&quot;</span>),</span>
<span id="cb682-5"><a href="counts.html#cb682-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type=</span><span class="st">&quot;response&quot;</span>,</span>
<span id="cb682-6"><a href="counts.html#cb682-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">offset =</span> <span class="fu">log</span>(mean_area))</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
age
</th>
<th style="text-align:left;">
genotype
</th>
<th style="text-align:right;">
response
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
asymp.LCL
</th>
<th style="text-align:right;">
asymp.UCL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Young
</td>
<td style="text-align:left;">
WT
</td>
<td style="text-align:right;">
58.20
</td>
<td style="text-align:right;">
29.6
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
21.51
</td>
<td style="text-align:right;">
157.48
</td>
</tr>
<tr>
<td style="text-align:left;">
Old
</td>
<td style="text-align:left;">
WT
</td>
<td style="text-align:right;">
137.86
</td>
<td style="text-align:right;">
37.0
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
81.42
</td>
<td style="text-align:right;">
233.40
</td>
</tr>
<tr>
<td style="text-align:left;">
Young
</td>
<td style="text-align:left;">
KI
</td>
<td style="text-align:right;">
52.53
</td>
<td style="text-align:right;">
27.0
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
19.20
</td>
<td style="text-align:right;">
143.72
</td>
</tr>
<tr>
<td style="text-align:left;">
Old
</td>
<td style="text-align:left;">
KI
</td>
<td style="text-align:right;">
64.57
</td>
<td style="text-align:right;">
15.3
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
40.55
</td>
<td style="text-align:right;">
102.80
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb683"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb683-1"><a href="counts.html#cb683-1" aria-hidden="true" tabindex="-1"></a>exp3b_m3_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp3b_m3_emm,</span>
<span id="cb683-2"><a href="counts.html#cb683-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb683-3"><a href="counts.html#cb683-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb683-4"><a href="counts.html#cb683-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb683-5"><a href="counts.html#cb683-5" aria-hidden="true" tabindex="-1"></a><span class="co"># exp3b_m3_emm # print in console to get row numbers</span></span>
<span id="cb683-6"><a href="counts.html#cb683-6" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb683-7"><a href="counts.html#cb683-7" aria-hidden="true" tabindex="-1"></a>young_wt <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb683-8"><a href="counts.html#cb683-8" aria-hidden="true" tabindex="-1"></a>old_wt <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb683-9"><a href="counts.html#cb683-9" aria-hidden="true" tabindex="-1"></a>young_ki <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb683-10"><a href="counts.html#cb683-10" aria-hidden="true" tabindex="-1"></a>old_ki <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb683-11"><a href="counts.html#cb683-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb683-12"><a href="counts.html#cb683-12" aria-hidden="true" tabindex="-1"></a><span class="co">#1. (Young KI) - (Young WT)</span></span>
<span id="cb683-13"><a href="counts.html#cb683-13" aria-hidden="true" tabindex="-1"></a><span class="co">#2. (Old   KI) - (Old   WT)</span></span>
<span id="cb683-14"><a href="counts.html#cb683-14" aria-hidden="true" tabindex="-1"></a><span class="co">#3. Interaction</span></span>
<span id="cb683-15"><a href="counts.html#cb683-15" aria-hidden="true" tabindex="-1"></a>exp3b_m3_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(</span>
<span id="cb683-16"><a href="counts.html#cb683-16" aria-hidden="true" tabindex="-1"></a>  exp3b_m3_emm,</span>
<span id="cb683-17"><a href="counts.html#cb683-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb683-18"><a href="counts.html#cb683-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;(Young KI) - (Young WT)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(young_ki <span class="sc">-</span> young_wt),</span>
<span id="cb683-19"><a href="counts.html#cb683-19" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;(Old   KI) - (Old   WT)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(old_ki <span class="sc">-</span> old_wt),</span>
<span id="cb683-20"><a href="counts.html#cb683-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;Interaction&quot;</span> <span class="ot">=</span> <span class="fu">c</span>((old_ki <span class="sc">-</span> old_wt) <span class="sc">-</span> (young_ki <span class="sc">-</span> young_wt))</span>
<span id="cb683-21"><a href="counts.html#cb683-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># &quot;Interaction&quot; = c(old_ki - old_wt) -</span></span>
<span id="cb683-22"><a href="counts.html#cb683-22" aria-hidden="true" tabindex="-1"></a>    <span class="co">#   c(young_ki - young_wt)</span></span>
<span id="cb683-23"><a href="counts.html#cb683-23" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb683-24"><a href="counts.html#cb683-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb683-25"><a href="counts.html#cb683-25" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb683-26"><a href="counts.html#cb683-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
ratio
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
asymp.LCL
</th>
<th style="text-align:right;">
asymp.UCL
</th>
<th style="text-align:right;">
z.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Young KI) / (Young WT)
</td>
<td style="text-align:right;">
0.90
</td>
<td style="text-align:right;">
0.652
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.22
</td>
<td style="text-align:right;">
3.72
</td>
<td style="text-align:right;">
-0.14
</td>
<td style="text-align:right;">
0.88708
</td>
</tr>
<tr>
<td style="text-align:left;">
(Old KI) / (Old WT)
</td>
<td style="text-align:right;">
0.47
</td>
<td style="text-align:right;">
0.168
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.23
</td>
<td style="text-align:right;">
0.95
</td>
<td style="text-align:right;">
-2.12
</td>
<td style="text-align:right;">
0.03434
</td>
</tr>
<tr>
<td style="text-align:left;">
Interaction
</td>
<td style="text-align:right;">
0.52
</td>
<td style="text-align:right;">
0.418
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.11
</td>
<td style="text-align:right;">
2.52
</td>
<td style="text-align:right;">
-0.81
</td>
<td style="text-align:right;">
0.41592
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
</div>
<div id="plot-the-model-8" class="section level4" number="20.4.3.4">
<h4><span class="header-section-number">20.4.3.4</span> Plot the model</h4>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-eg2-exp3b-m3-gg1-1.png" width="576" /></p>
</div>
</div>
</div>
<div id="understanding-example-2" class="section level2" number="20.5">
<h2><span class="header-section-number">20.5</span> Understanding Example 2</h2>
<div id="an-offset-is-an-added-covariate-with-a-coefficient-fixed-at-1" class="section level3" number="20.5.1">
<h3><span class="header-section-number">20.5.1</span> An offset is an added covariate with a coefficient fixed at 1</h3>
<p>The systematic part (the linear predictor) of the negative binomial model fit to the exp3b data is</p>
<p><span class="math display">\[
\begin{align}
\texttt{positive_nuclei} = \ &amp;\beta_0 + \beta_1 \texttt{age}_\texttt{Old} + \beta_2 \texttt{genotype}_\texttt{KI} \ + \\ &amp;\beta_3 \texttt{age}_\texttt{Old}\texttt{:genotype}_\texttt{KI} \ + \\ 
&amp;1.0 \times \textrm{log}(\texttt{area_mm2})
\end{align}
\]</span></p>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The offset variable is <span class="math inline">\(\textrm{log}(\texttt{area_mm2})\)</span>. An offset is a covariate with a fixed coefficient of 1.0 – this coefficient is not estimated.</li>
<li>Figure <a href="counts.html#fig:glm-eg2-exp3b-m3-plot-offset">20.3</a> illustrates the offset on the response and the log scale.</li>
</ol>
<div class="figure"><span id="fig:glm-eg2-exp3b-m3-plot-offset"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-eg2-exp3b-m3-plot-offset-1.png" alt="What an offset looks like. A. The offset on the response scale. B. The offset on the log scale. In A and B, only the lines for the 20 month old mice are shown." width="576" />
<p class="caption">
Figure 20.3: What an offset looks like. A. The offset on the response scale. B. The offset on the log scale. In A and B, only the lines for the 20 month old mice are shown.
</p>
</div>
<ol start="3" style="list-style-type: decimal">
<li>On the log scale (Figure <a href="counts.html#fig:glm-eg2-exp3b-m3-plot-offset">20.3</a>B), the offset curves are parallel because each is forced to have a slope of one on the link scale. On the response scale, these curves have different slopes – this slope is <span class="math inline">\(\mathrm{exp}(b_0)\)</span>, that is, the exponent of the intercept on the log scale).</li>
<li>The plot on the log scale (Figure <a href="counts.html#fig:glm-eg2-exp3b-m3-plot-offset">20.3</a>B) is easy to misinterpret as a regression with a slope of 1 fit to the log transformed values. It isn’t, because individual values are not transformed in a GLM. I’m specifically referring to this space as the log and not link scale to try to avoid this misinterpretation.</li>
</ol>
</div>
<div id="glm-rate" class="section level3" number="20.5.2">
<h3><span class="header-section-number">20.5.2</span> A count GLM with an offset models the area-normalized means</h3>
<p>In the <em>t</em>-test or ANOVA of the difference in the area-normalized count of TUNEL+ nuclei, the systematic component of the equivalent linear model is (using only the 20 month old mice for simplicity)</p>
<p><span class="math display">\[
\frac{\mu}{\texttt{area_mm2}} = \beta_0 + \beta_1 \texttt{genotype}_\texttt{KI}
\]</span></p>
<p>where <span class="math inline">\(\mu\)</span> is the mean response (<span class="math inline">\(\texttt{positive_nuclei}\)</span>) conditional on the level of genotype. Statisticians typically refer to the ratio <span class="math inline">\(\frac{\mu}{\texttt{area_mm2}}\)</span> as a rate – here, it is the expected rate of observing a TUNEL+ nucleus as more area is searched.</p>
<p>Let’s compare this to the GLM model recommended here. Recall that the conditional expecation on the link scale is <span class="math inline">\(\eta\)</span> and, using the inverse link function, <span class="math inline">\(\mathrm{log}(\mu) = \eta\)</span>, the systematic part of the count GLM model with offset is</p>
<p><span class="math display">\[
\begin{align}
\eta &amp;= \beta_0 + \beta_1 \texttt{genotype}_\texttt{KI} + \mathrm{log}(\texttt{area_mm2}) \\
\mathrm{log}(\mu) &amp;= \beta_0 + \beta_1 \texttt{genotype}_\texttt{KI} + \mathrm{log}(\texttt{area_mm2}) \\
\mathrm{log}(\mu) - \mathrm{log}(\texttt{area_mm2}) &amp;= \beta_0 + \beta_1 \texttt{genotype}_\texttt{KI} \\
\mathrm{log}(\frac{\mu}{\texttt{area_mm2}}) &amp;= \beta_0 + \beta_1 \texttt{genotype}_\texttt{KI} \\
\frac{\mu}{\texttt{area_mm2}} &amp;= \mathrm{exp}(\beta_0 + \beta_1 \texttt{genotype}_\texttt{KI}) \\
\end{align}
\]</span></p>
<p>That is, the GLM with offset also models the conditional mean of the area-normalized count, but in a way where the data are much more compatible with the assumptions of the statistical model (in the sense that model checks show that the data look like a sample from the statistical model).</p>
<p>If you want the area-normalized means and their uncertainty under the assumptions of the count GLM offset model, here is some code.</p>
<div class="sourceCode" id="cb684"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb684-1"><a href="counts.html#cb684-1" aria-hidden="true" tabindex="-1"></a>mean_area <span class="ot">&lt;-</span> <span class="fu">mean</span>(exp3b[, area_mm2])</span>
<span id="cb684-2"><a href="counts.html#cb684-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb684-3"><a href="counts.html#cb684-3" aria-hidden="true" tabindex="-1"></a><span class="co"># link scale</span></span>
<span id="cb684-4"><a href="counts.html#cb684-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb684-5"><a href="counts.html#cb684-5" aria-hidden="true" tabindex="-1"></a><span class="co"># compute rate or normalized count using emmeans</span></span>
<span id="cb684-6"><a href="counts.html#cb684-6" aria-hidden="true" tabindex="-1"></a><span class="co"># setting the offset will compute mean at the specified</span></span>
<span id="cb684-7"><a href="counts.html#cb684-7" aria-hidden="true" tabindex="-1"></a><span class="co"># value of the offset</span></span>
<span id="cb684-8"><a href="counts.html#cb684-8" aria-hidden="true" tabindex="-1"></a><span class="co"># be sure the stats are on the link scale</span></span>
<span id="cb684-9"><a href="counts.html#cb684-9" aria-hidden="true" tabindex="-1"></a>m3_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp3b_m3,</span>
<span id="cb684-10"><a href="counts.html#cb684-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;age&quot;</span>, <span class="st">&quot;genotype&quot;</span>),</span>
<span id="cb684-11"><a href="counts.html#cb684-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">offset =</span> <span class="fu">log</span>(mean_area)) <span class="sc">%&gt;%</span></span>
<span id="cb684-12"><a href="counts.html#cb684-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>() <span class="sc">%&gt;%</span></span>
<span id="cb684-13"><a href="counts.html#cb684-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span>
<span id="cb684-14"><a href="counts.html#cb684-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb684-15"><a href="counts.html#cb684-15" aria-hidden="true" tabindex="-1"></a><span class="co"># emmeans is returning the first line in equations above.</span></span>
<span id="cb684-16"><a href="counts.html#cb684-16" aria-hidden="true" tabindex="-1"></a><span class="co"># We want the last line. So subtract the log of the offset value.</span></span>
<span id="cb684-17"><a href="counts.html#cb684-17" aria-hidden="true" tabindex="-1"></a>m3_emm[, log_rate <span class="sc">:</span><span class="er">=</span> emmean <span class="sc">-</span> <span class="fu">log</span>(mean_area)]</span>
<span id="cb684-18"><a href="counts.html#cb684-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb684-19"><a href="counts.html#cb684-19" aria-hidden="true" tabindex="-1"></a><span class="co"># and compute the 95% asymptotic CIs</span></span>
<span id="cb684-20"><a href="counts.html#cb684-20" aria-hidden="true" tabindex="-1"></a>m3_emm[, log_rate_lcl <span class="sc">:</span><span class="er">=</span> log_rate <span class="sc">-</span> <span class="fl">1.96</span><span class="sc">*</span>SE]</span>
<span id="cb684-21"><a href="counts.html#cb684-21" aria-hidden="true" tabindex="-1"></a>m3_emm[, log_rate_ucl <span class="sc">:</span><span class="er">=</span> log_rate <span class="sc">+</span> <span class="fl">1.96</span><span class="sc">*</span>SE]</span>
<span id="cb684-22"><a href="counts.html#cb684-22" aria-hidden="true" tabindex="-1"></a><span class="co"># m3_emm</span></span>
<span id="cb684-23"><a href="counts.html#cb684-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb684-24"><a href="counts.html#cb684-24" aria-hidden="true" tabindex="-1"></a><span class="co"># response scale</span></span>
<span id="cb684-25"><a href="counts.html#cb684-25" aria-hidden="true" tabindex="-1"></a>m3_emm_response <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp3b_m3,</span>
<span id="cb684-26"><a href="counts.html#cb684-26" aria-hidden="true" tabindex="-1"></a>                  <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;age&quot;</span>, <span class="st">&quot;genotype&quot;</span>),</span>
<span id="cb684-27"><a href="counts.html#cb684-27" aria-hidden="true" tabindex="-1"></a>                  <span class="at">offset =</span> <span class="fu">log</span>(mean_area),</span>
<span id="cb684-28"><a href="counts.html#cb684-28" aria-hidden="true" tabindex="-1"></a>                  <span class="at">type =</span> <span class="st">&quot;response&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb684-29"><a href="counts.html#cb684-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>() <span class="sc">%&gt;%</span></span>
<span id="cb684-30"><a href="counts.html#cb684-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span>
<span id="cb684-31"><a href="counts.html#cb684-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb684-32"><a href="counts.html#cb684-32" aria-hidden="true" tabindex="-1"></a><span class="co"># replace the mean and CIs with the backtransformed values from the</span></span>
<span id="cb684-33"><a href="counts.html#cb684-33" aria-hidden="true" tabindex="-1"></a><span class="co"># link scale table</span></span>
<span id="cb684-34"><a href="counts.html#cb684-34" aria-hidden="true" tabindex="-1"></a>m3_emm_rates <span class="ot">&lt;-</span> m3_emm_response</span>
<span id="cb684-35"><a href="counts.html#cb684-35" aria-hidden="true" tabindex="-1"></a>m3_emm_rates[, response <span class="sc">:</span><span class="er">=</span> <span class="fu">exp</span>(m3_emm<span class="sc">$</span>log_rate)]</span>
<span id="cb684-36"><a href="counts.html#cb684-36" aria-hidden="true" tabindex="-1"></a>m3_emm_rates[, asymp.LCL <span class="sc">:</span><span class="er">=</span> <span class="fu">exp</span>(m3_emm<span class="sc">$</span>log_rate_lcl)]</span>
<span id="cb684-37"><a href="counts.html#cb684-37" aria-hidden="true" tabindex="-1"></a>m3_emm_rates[, asymp.UCL <span class="sc">:</span><span class="er">=</span> <span class="fu">exp</span>(m3_emm<span class="sc">$</span>log_rate_ucl)]</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:glm-exp3b-rate-show">Table 20.3: </span>Modeled mean area-normalized counts of TUNEL+ cells, estimated using negative binomial GLM.
</caption>
<thead>
<tr>
<th style="text-align:left;">
age
</th>
<th style="text-align:left;">
genotype
</th>
<th style="text-align:right;">
response
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
asymp.LCL
</th>
<th style="text-align:right;">
asymp.UCL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Young
</td>
<td style="text-align:left;">
WT
</td>
<td style="text-align:right;">
0.93
</td>
<td style="text-align:right;">
10.79
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.34
</td>
<td style="text-align:right;">
2.5
</td>
</tr>
<tr>
<td style="text-align:left;">
Old
</td>
<td style="text-align:left;">
WT
</td>
<td style="text-align:right;">
2.20
</td>
<td style="text-align:right;">
13.52
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
1.30
</td>
<td style="text-align:right;">
3.7
</td>
</tr>
<tr>
<td style="text-align:left;">
Young
</td>
<td style="text-align:left;">
KI
</td>
<td style="text-align:right;">
0.84
</td>
<td style="text-align:right;">
9.85
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.31
</td>
<td style="text-align:right;">
2.3
</td>
</tr>
<tr>
<td style="text-align:left;">
Old
</td>
<td style="text-align:left;">
KI
</td>
<td style="text-align:right;">
1.03
</td>
<td style="text-align:right;">
5.59
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.65
</td>
<td style="text-align:right;">
1.6
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Again, no ratios were computed. The mean area-normalized counts were computed using the coefficients of the GLM. Here, we used the emmeans package to do this. See section xxx below for code that does this withouth the emmeans package.</li>
</ol>
<div class="figure"><span id="fig:glm-exp3b-rate-plot-show"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3b-rate-plot-show-1.png" alt="Area-normalized counts modeled using negative binomial GLM with offset, with the raw count as the response (A) and linear model with the area-normalized count as the response." width="672" />
<p class="caption">
Figure 20.4: Area-normalized counts modeled using negative binomial GLM with offset, with the raw count as the response (A) and linear model with the area-normalized count as the response.
</p>
</div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>In Panel B, the CIs of the two Young treatment groups have lower bounds that extend below zero. This is absurd – negative area-normalized-count means cannot exist. But this inference is implied by an ANOVA of these data.</li>
<li>Panel A has the potential to lead to misconceptions because the GLM model with offset is not fit to the area-normalized data but to the raw counts. Regardless, the GLM model with offset is modeling the means of these ratios.</li>
</ol>
<div style="background-color:#cccccc; text-align:left; vertical-align: middle; padding:20px 47px;">
<p><strong>More than I need to know</strong>. The modeled means of a count GLM on the link scale are <span class="math inline">\(\mathbf{X} \mathbf{b}\)</span>, where <span class="math inline">\(\mathbf{X}\)</span> is the model matrix and <span class="math inline">\(\mathbf{b}\)</span> is the vector of model coefficients. In a model with an offset, the values of the offset variable are not in the model matrix and the coefficient (1.0) is not in the vector of coefficients. Following the math above, this expectation is the modeled mean normalized-count (on the link scale) and not the modeled mean count conditional on some offset value (on the link scale). Again, following the math above, to compute the expected count conditional on some value of the offset, use <span class="math inline">\(\mathbf{X} \mathbf{b} + \mathrm{log}(offset)\)</span>. But we want the rate not the mean, and we want the rate on the response scale, so simply use <span class="math inline">\(\mathrm{exp}(\mathbf{X} \mathbf{b})\)</span>.</p>
<div class="sourceCode" id="cb685"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb685-1"><a href="counts.html#cb685-1" aria-hidden="true" tabindex="-1"></a><span class="co"># compute rate or normalized count using matrix algebra using</span></span>
<span id="cb685-2"><a href="counts.html#cb685-2" aria-hidden="true" tabindex="-1"></a><span class="co"># final line of equations above</span></span>
<span id="cb685-3"><a href="counts.html#cb685-3" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">coef</span>(exp3b_m3)</span>
<span id="cb685-4"><a href="counts.html#cb685-4" aria-hidden="true" tabindex="-1"></a>X <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(exp3b_m3)</span>
<span id="cb685-5"><a href="counts.html#cb685-5" aria-hidden="true" tabindex="-1"></a>exp3b[, rate <span class="sc">:</span><span class="er">=</span> <span class="fu">exp</span>((X <span class="sc">%*%</span> b)[,<span class="dv">1</span>])]</span>
<span id="cb685-6"><a href="counts.html#cb685-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb685-7"><a href="counts.html#cb685-7" aria-hidden="true" tabindex="-1"></a><span class="co"># compare rate to area-normalized count</span></span>
<span id="cb685-8"><a href="counts.html#cb685-8" aria-hidden="true" tabindex="-1"></a>rates <span class="ot">&lt;-</span> exp3b[, .(<span class="at">rate_manual =</span> <span class="fu">mean</span>(rate),</span>
<span id="cb685-9"><a href="counts.html#cb685-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">norm_count =</span> <span class="fu">mean</span>(count_per_area)),</span>
<span id="cb685-10"><a href="counts.html#cb685-10" aria-hidden="true" tabindex="-1"></a>                    by <span class="ot">=</span> .(age, genotype)]</span>
<span id="cb685-11"><a href="counts.html#cb685-11" aria-hidden="true" tabindex="-1"></a>m3_emm <span class="ot">&lt;-</span> <span class="fu">merge</span>(m3_emm_rates, rates, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;age&quot;</span>, <span class="st">&quot;genotype&quot;</span>))</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:glm-rate-algebra-table">Table 20.4: </span>Mean area-normalized-counts computed using the emmeans package (<span class="math inline">\(\texttt{response}\)</span>), the matrix algebra code in this section (<span class="math inline">\(\texttt{rate_manual}\)</span>), and as the simple average of the normalized counts (<span class="math inline">\(\texttt{norm_count}\)</span>)
</caption>
<thead>
<tr>
<th style="text-align:left;">
age
</th>
<th style="text-align:left;">
genotype
</th>
<th style="text-align:right;">
response
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
asymp.LCL
</th>
<th style="text-align:right;">
asymp.UCL
</th>
<th style="text-align:left;">
g_by_age
</th>
<th style="text-align:right;">
rate_manual
</th>
<th style="text-align:right;">
norm_count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Young
</td>
<td style="text-align:left;">
WT
</td>
<td style="text-align:right;">
0.9304
</td>
<td style="text-align:right;">
10.79
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.34
</td>
<td style="text-align:right;">
2.52
</td>
<td style="text-align:left;">
WT Young
</td>
<td style="text-align:right;">
0.9304
</td>
<td style="text-align:right;">
0.9269
</td>
</tr>
<tr>
<td style="text-align:left;">
Young
</td>
<td style="text-align:left;">
KI
</td>
<td style="text-align:right;">
0.8397
</td>
<td style="text-align:right;">
9.85
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.31
</td>
<td style="text-align:right;">
2.30
</td>
<td style="text-align:left;">
KI Young
</td>
<td style="text-align:right;">
0.8397
</td>
<td style="text-align:right;">
0.8432
</td>
</tr>
<tr>
<td style="text-align:left;">
Old
</td>
<td style="text-align:left;">
WT
</td>
<td style="text-align:right;">
2.2039
</td>
<td style="text-align:right;">
13.52
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
1.30
</td>
<td style="text-align:right;">
3.73
</td>
<td style="text-align:left;">
WT Old
</td>
<td style="text-align:right;">
2.2039
</td>
<td style="text-align:right;">
2.2027
</td>
</tr>
<tr>
<td style="text-align:left;">
Old
</td>
<td style="text-align:left;">
KI
</td>
<td style="text-align:right;">
1.0322
</td>
<td style="text-align:right;">
5.59
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.65
</td>
<td style="text-align:right;">
1.64
</td>
<td style="text-align:left;">
KI Old
</td>
<td style="text-align:right;">
1.0322
</td>
<td style="text-align:right;">
1.0284
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="compare-an-offset-to-an-added-covariate-with-an-estimated-coefficient" class="section level3" number="20.5.3">
<h3><span class="header-section-number">20.5.3</span> Compare an offset to an added covariate with an estimated coefficient</h3>
<div class="sourceCode" id="cb686"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb686-1"><a href="counts.html#cb686-1" aria-hidden="true" tabindex="-1"></a>exp3b_m5 <span class="ot">&lt;-</span> <span class="fu">glm.nb</span>(positive_nuclei <span class="sc">~</span> age <span class="sc">*</span> genotype <span class="sc">+</span></span>
<span id="cb686-2"><a href="counts.html#cb686-2" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">log</span>(area_mm2),</span>
<span id="cb686-3"><a href="counts.html#cb686-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">data =</span> exp3b)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:glm-exp3b-m5-coef">Table 20.5: </span>Coefficients from model exp3b_m5 (covariate) and model exp3b_m3 (offset).
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
z value
</th>
<th style="text-align:right;">
Pr(&gt;|z|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr grouplength="5">
<td colspan="7" style="border-bottom: 1px solid;">
<strong>exp3b_m5 (covariate)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Intercept)
</td>
<td style="text-align:right;">
-3.36
</td>
<td style="text-align:right;">
1.48
</td>
<td style="text-align:right;">
-2.27
</td>
<td style="text-align:right;">
0.0232
</td>
<td style="text-align:right;">
-6.27
</td>
<td style="text-align:right;">
-0.33
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
ageOld
</td>
<td style="text-align:right;">
-0.67
</td>
<td style="text-align:right;">
0.89
</td>
<td style="text-align:right;">
-0.75
</td>
<td style="text-align:right;">
0.4528
</td>
<td style="text-align:right;">
-2.33
</td>
<td style="text-align:right;">
1.00
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
genotypeKI
</td>
<td style="text-align:right;">
0.24
</td>
<td style="text-align:right;">
0.70
</td>
<td style="text-align:right;">
0.35
</td>
<td style="text-align:right;">
0.7299
</td>
<td style="text-align:right;">
-1.20
</td>
<td style="text-align:right;">
1.69
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
log(area_mm2)
</td>
<td style="text-align:right;">
2.48
</td>
<td style="text-align:right;">
0.64
</td>
<td style="text-align:right;">
3.89
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
1.19
</td>
<td style="text-align:right;">
3.74
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
ageOld:genotypeKI
</td>
<td style="text-align:right;">
-1.16
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
-1.48
</td>
<td style="text-align:right;">
0.1390
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
0.47
</td>
</tr>
<tr grouplength="4">
<td colspan="7" style="border-bottom: 1px solid;">
<strong>exp3b_m3 (offset)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Intercept)
</td>
<td style="text-align:right;">
-0.07
</td>
<td style="text-align:right;">
0.51
</td>
<td style="text-align:right;">
-0.14
</td>
<td style="text-align:right;">
0.8871
</td>
<td style="text-align:right;">
-0.95
</td>
<td style="text-align:right;">
1.09
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
ageOld
</td>
<td style="text-align:right;">
0.86
</td>
<td style="text-align:right;">
0.57
</td>
<td style="text-align:right;">
1.50
</td>
<td style="text-align:right;">
0.1334
</td>
<td style="text-align:right;">
-0.39
</td>
<td style="text-align:right;">
1.91
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
genotypeKI
</td>
<td style="text-align:right;">
-0.10
</td>
<td style="text-align:right;">
0.72
</td>
<td style="text-align:right;">
-0.14
</td>
<td style="text-align:right;">
0.8871
</td>
<td style="text-align:right;">
-1.56
</td>
<td style="text-align:right;">
1.36
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
ageOld:genotypeKI
</td>
<td style="text-align:right;">
-0.66
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
-0.81
</td>
<td style="text-align:right;">
0.4159
</td>
<td style="text-align:right;">
-2.27
</td>
<td style="text-align:right;">
0.96
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The coefficient table for Model exp3b_m5 has a row for the added covariate <span class="math inline">\(\texttt{area_mm2}\)</span>. This row is absent for Model exp3b_m3 since the coefficient is fixed at 1.0.</li>
<li>The estimate of the coefficient of <span class="math inline">\(\texttt{area_mm2}\)</span> in Model exp3b_m5 is 2.48. A value of 1.0 – the value assumed by the offset model or an analysis of the ratio – is not very compatible with the data.</li>
<li>A coefficient greater than 1.0 means that the rate of DNA damage (the number of TUNEL positive cells per area) increases with the size of the area measured. Does this make any biological sense? The area measured should not have a biological component unless it is a function of the size of the organ. If the area measured is a function of the size of the organ, then we would want to use the ANCOVA model and not the offset model.</li>
<li>This increase in the rate of TUNEL positive cells is seen in the response-scale plot of the fit model in Figure <a href="#fig:glm-exp3b-m5-plot"><strong>??</strong></a>A, where the slope of the regression line increases as <span class="math inline">\(\texttt{area_mm2}\)</span> increases.</li>
</ol>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/glm-exp3b-m5-plot-1.png" width="576" /></p>
<ol start="5" style="list-style-type: decimal">
<li>Note that on the log scale (Figure <a href="#fig:glm-exp3b-m5-plot"><strong>??</strong></a>B), the two regression lines are parallel, as they must be in an ANCOVA linear model.</li>
</ol>
</div>
<div id="issues-with-plotting" class="section level3" number="20.5.4">
<h3><span class="header-section-number">20.5.4</span> Issues with plotting</h3>
</div>
</div>
<div id="example-3-glm-models-for-binary-responses" class="section level2" number="20.6">
<h2><span class="header-section-number">20.6</span> Example 3 – GLM models for binary responses</h2>
</div>
<div id="working-in-r-7" class="section level2" number="20.7">
<h2><span class="header-section-number">20.7</span> Working in R</h2>
<div id="fitting-glms-to-count-data" class="section level3" number="20.7.1">
<h3><span class="header-section-number">20.7.1</span> Fitting GLMs to count data</h3>
<p>The Poisson family is specified with the base R <code>glm()</code> function. For negative binomial, use <code>glm.nb</code> from the MASS package</p>
<div class="sourceCode" id="cb687"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb687-1"><a href="counts.html#cb687-1" aria-hidden="true" tabindex="-1"></a><span class="co"># poisson - less likely to fit to real biological data well </span></span>
<span id="cb687-2"><a href="counts.html#cb687-2" aria-hidden="true" tabindex="-1"></a><span class="co"># because of overdispersion</span></span>
<span id="cb687-3"><a href="counts.html#cb687-3" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glm</span>(y <span class="sc">~</span> treatment, <span class="at">family =</span> <span class="st">&quot;poisson&quot;</span>, <span class="at">data =</span> dt)</span>
<span id="cb687-4"><a href="counts.html#cb687-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb687-5"><a href="counts.html#cb687-5" aria-hidden="true" tabindex="-1"></a><span class="co"># two alternatives to overdispersed poisson fit</span></span>
<span id="cb687-6"><a href="counts.html#cb687-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb687-7"><a href="counts.html#cb687-7" aria-hidden="true" tabindex="-1"></a><span class="co"># quasipoisson</span></span>
<span id="cb687-8"><a href="counts.html#cb687-8" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glm</span>(y <span class="sc">~</span> treatment, <span class="at">family =</span> <span class="st">&quot;quasipoisson&quot;</span>, <span class="at">data=</span>dt)</span>
<span id="cb687-9"><a href="counts.html#cb687-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb687-10"><a href="counts.html#cb687-10" aria-hidden="true" tabindex="-1"></a><span class="co"># note that &quot;family&quot; is not an argument since this function is used only to fit a negative binomial distribution!</span></span>
<span id="cb687-11"><a href="counts.html#cb687-11" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glm.nb</span>(y <span class="sc">~</span> treatment, <span class="at">data =</span> dt)</span></code></pre></div>
</div>
<div id="fitting-a-glm-to-a-continuous-conditional-response-with-right-skew." class="section level3" number="20.7.2">
<h3><span class="header-section-number">20.7.2</span> Fitting a GLM to a continuous conditional response with right skew.</h3>
<p>The Gamma family is specified with the base R <code>glm()</code> function.</p>
<div class="sourceCode" id="cb688"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb688-1"><a href="counts.html#cb688-1" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glm</span>(y <span class="sc">~</span> treatment, <span class="at">family =</span> <span class="fu">Gamma</span>(<span class="at">link =</span> <span class="st">&quot;log&quot;</span>), <span class="at">data =</span> dt)</span></code></pre></div>
</div>
<div id="fitting-a-glm-to-a-binary-success-or-failure-presence-or-absence-survived-or-died-response" class="section level3" number="20.7.3">
<h3><span class="header-section-number">20.7.3</span> Fitting a GLM to a binary (success or failure, presence or absence, survived or died) response</h3>
<p>The binomial family is specified with base R <code>glm()</code> function.</p>
<div class="sourceCode" id="cb689"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb689-1"><a href="counts.html#cb689-1" aria-hidden="true" tabindex="-1"></a><span class="co"># if the data includes a 0 or 1 for every observation of y</span></span>
<span id="cb689-2"><a href="counts.html#cb689-2" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glm</span>(y <span class="sc">~</span> treatment, <span class="at">family =</span> <span class="st">&quot;binomial&quot;</span>, <span class="at">data =</span> dt)</span>
<span id="cb689-3"><a href="counts.html#cb689-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb689-4"><a href="counts.html#cb689-4" aria-hidden="true" tabindex="-1"></a><span class="co"># if the data includes the frequency of success AND there is a measure of the total n</span></span>
<span id="cb689-5"><a href="counts.html#cb689-5" aria-hidden="true" tabindex="-1"></a>dt[ , failure <span class="sc">:</span><span class="er">=</span> n <span class="sc">-</span> success]</span>
<span id="cb689-6"><a href="counts.html#cb689-6" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glm</span>(<span class="fu">cbind</span>(success, failure) <span class="sc">~</span> treatment, <span class="at">family =</span> <span class="st">&quot;binomial&quot;</span>, <span class="at">data =</span> dt)</span></code></pre></div>
</div>
<div id="fitting-generalized-linear-mixed-models" class="section level3" number="20.7.4">
<h3><span class="header-section-number">20.7.4</span> Fitting Generalized Linear Mixed Models</h3>
<p>Generalized linear mixed models are fit with <code>glmer</code> from the lmer package.</p>
<div class="sourceCode" id="cb690"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb690-1"><a href="counts.html#cb690-1" aria-hidden="true" tabindex="-1"></a><span class="co"># random intercept of factor &quot;id&quot;</span></span>
<span id="cb690-2"><a href="counts.html#cb690-2" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glmer</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>id), <span class="at">family =</span> <span class="st">&quot;poisson&quot;</span>, <span class="at">data =</span> dt)</span>
<span id="cb690-3"><a href="counts.html#cb690-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb690-4"><a href="counts.html#cb690-4" aria-hidden="true" tabindex="-1"></a><span class="co"># random intercept and slope of factor &quot;id&quot;</span></span>
<span id="cb690-5"><a href="counts.html#cb690-5" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glmer</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (treatment<span class="sc">|</span>id), <span class="at">family =</span> <span class="fu">Gamma</span>(<span class="at">link =</span> <span class="st">&quot;log&quot;</span>), <span class="at">data =</span> dt)</span>
<span id="cb690-6"><a href="counts.html#cb690-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb690-7"><a href="counts.html#cb690-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Again, negative binomial uses a special function</span></span>
<span id="cb690-8"><a href="counts.html#cb690-8" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glmer.nb</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (treatment<span class="sc">|</span>id), <span class="at">data =</span> dt)</span></code></pre></div>
<p>Another good package for GLMMs is <code>glmmTMB</code> from the glmmTMB package</p>
<div class="sourceCode" id="cb691"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb691-1"><a href="counts.html#cb691-1" aria-hidden="true" tabindex="-1"></a><span class="co"># negative binomial</span></span>
<span id="cb691-2"><a href="counts.html#cb691-2" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glmmTMB</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>id), <span class="at">family=</span><span class="st">&quot;nbinom2&quot;</span>, <span class="at">data =</span> dt)</span>
<span id="cb691-3"><a href="counts.html#cb691-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb691-4"><a href="counts.html#cb691-4" aria-hidden="true" tabindex="-1"></a><span class="co"># nbinom1, the mean variance relationship is that of quasipoisson</span></span>
<span id="cb691-5"><a href="counts.html#cb691-5" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glmmTMB</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>id), <span class="at">family=</span><span class="st">&quot;nbinom1&quot;</span>, <span class="at">data =</span> dt)</span></code></pre></div>
</div>
</div>
<div id="model-checking-glms" class="section level2" number="20.8">
<h2><span class="header-section-number">20.8</span> Model checking GLMs</h2>
<p>The <code>DHARMa</code> package has an excellent set of model checking tools. The DHARMa package uses simulation to generate fake data sampled from the fit model using the function <code>simulateResiduals</code>.</p>
<div class="sourceCode" id="cb692"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb692-1"><a href="counts.html#cb692-1" aria-hidden="true" tabindex="-1"></a>simulation_output <span class="ot">&lt;-</span> <span class="fu">simulateResiduals</span>(<span class="at">fittedModel =</span> exp3a_m3,</span>
<span id="cb692-2"><a href="counts.html#cb692-2" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">n =</span> <span class="dv">250</span>,</span>
<span id="cb692-3"><a href="counts.html#cb692-3" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">refit =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb693"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb693-1"><a href="counts.html#cb693-1" aria-hidden="true" tabindex="-1"></a>simulation_output <span class="ot">&lt;-</span> <span class="fu">simulateResiduals</span>(<span class="at">fittedModel =</span> exp3a_m2,</span>
<span id="cb693-2"><a href="counts.html#cb693-2" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">n =</span> <span class="dv">250</span>,</span>
<span id="cb693-3"><a href="counts.html#cb693-3" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">refit =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<ol style="list-style-type: decimal">
<li>The DHARMa package uses simulation to</li>
</ol>
<div class="sourceCode" id="cb694"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb694-1"><a href="counts.html#cb694-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(simulation_output)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/unnamed-chunk-300-1.png" width="576" /></p>
<div class="sourceCode" id="cb695"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb695-1"><a href="counts.html#cb695-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQQunif</span>(simulation_output)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/unnamed-chunk-301-1.png" width="576" /></p>
<ol start="4" style="list-style-type: decimal">
<li>Three test statistics are superimposed. Use these <em>p</em>-values cautiously – they are guides and not thresholds of demarcation. The two we care about here are</li>
</ol>
<ul>
<li>The KS statistic indicates that the quantile residuals are not very compatible with a Poisson model – think of this as having a very low probability of sampling these counts from a Poisson with the estimated <span class="math inline">\(\lambda\)</span>.</li>
<li>The dispersion statistic indicates that the value of the dispersion of the quantile residuals is not very compatible with a Poisson model – think of this as having a very low probability of sampling counts with this dispersion from a Poisson with the estimated <span class="math inline">\(\lambda\)</span>.</li>
</ul>
</div>
<div id="hidden-code-8" class="section level2" number="20.9">
<h2><span class="header-section-number">20.9</span> Hidden code</h2>
<div id="import-example-1-data-exp3a-angiogenic-sprouts" class="section level3" number="20.9.1">
<h3><span class="header-section-number">20.9.1</span> Import Example 1 data (exp3a – “angiogenic sprouts”)</h3>
<div class="sourceCode" id="cb696"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb696-1"><a href="counts.html#cb696-1" aria-hidden="true" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">&quot;Pericyte FAK negatively regulates Gas6-Axl signalling to suppress tumour angiogenesis and tumour growth&quot;</span></span>
<span id="cb696-2"><a href="counts.html#cb696-2" aria-hidden="true" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">&quot;41467_2020_16618_MOESM3_ESM.xlsx&quot;</span></span>
<span id="cb696-3"><a href="counts.html#cb696-3" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb696-4"><a href="counts.html#cb696-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb696-5"><a href="counts.html#cb696-5" aria-hidden="true" tabindex="-1"></a>exp3a_wide <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb696-6"><a href="counts.html#cb696-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sheet =</span> <span class="st">&quot;Figure 3&quot;</span>,</span>
<span id="cb696-7"><a href="counts.html#cb696-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">range =</span> <span class="st">&quot;B4:E26&quot;</span>,</span>
<span id="cb696-8"><a href="counts.html#cb696-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb696-9"><a href="counts.html#cb696-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span>
<span id="cb696-10"><a href="counts.html#cb696-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb696-11"><a href="counts.html#cb696-11" aria-hidden="true" tabindex="-1"></a>input_labels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;PBS FAK_wt&quot;</span>, <span class="st">&quot;PBS FAK_ko&quot;</span>, <span class="st">&quot;GAS6 FAK_wt&quot;</span>, <span class="st">&quot;GAS6 FAK_ko&quot;</span>)</span>
<span id="cb696-12"><a href="counts.html#cb696-12" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(exp3a_wide) <span class="ot">&lt;-</span> input_labels</span>
<span id="cb696-13"><a href="counts.html#cb696-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb696-14"><a href="counts.html#cb696-14" aria-hidden="true" tabindex="-1"></a>exp3a <span class="ot">&lt;-</span> <span class="fu">melt</span>(exp3a_wide,</span>
<span id="cb696-15"><a href="counts.html#cb696-15" aria-hidden="true" tabindex="-1"></a>              <span class="at">measure.vars =</span> input_labels,</span>
<span id="cb696-16"><a href="counts.html#cb696-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">variable.name =</span> <span class="st">&quot;t_by_g&quot;</span>,</span>
<span id="cb696-17"><a href="counts.html#cb696-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">value.name =</span> <span class="st">&quot;sprouts&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb696-18"><a href="counts.html#cb696-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb696-19"><a href="counts.html#cb696-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb696-20"><a href="counts.html#cb696-20" aria-hidden="true" tabindex="-1"></a><span class="co"># change order of factor levels</span></span>
<span id="cb696-21"><a href="counts.html#cb696-21" aria-hidden="true" tabindex="-1"></a>t_by_g_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;PBS FAK_wt&quot;</span>, <span class="st">&quot;GAS6 FAK_wt&quot;</span>, <span class="st">&quot;PBS FAK_ko&quot;</span>, <span class="st">&quot;GAS6 FAK_ko&quot;</span>)</span>
<span id="cb696-22"><a href="counts.html#cb696-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb696-23"><a href="counts.html#cb696-23" aria-hidden="true" tabindex="-1"></a>exp3a[, <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;genotype&quot;</span>)<span class="sc">:</span><span class="er">=</span> <span class="fu">tstrsplit</span>(t_by_g,</span>
<span id="cb696-24"><a href="counts.html#cb696-24" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">&quot; &quot;</span>,</span>
<span id="cb696-25"><a href="counts.html#cb696-25" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">fixed =</span> <span class="cn">TRUE</span>)]</span>
<span id="cb696-26"><a href="counts.html#cb696-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb696-27"><a href="counts.html#cb696-27" aria-hidden="true" tabindex="-1"></a>exp3a[, t_by_g <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(t_by_g, <span class="at">levels =</span> t_by_g_levels)]</span>
<span id="cb696-28"><a href="counts.html#cb696-28" aria-hidden="true" tabindex="-1"></a>treatment_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;PBS&quot;</span>, <span class="st">&quot;GAS6&quot;</span>)</span>
<span id="cb696-29"><a href="counts.html#cb696-29" aria-hidden="true" tabindex="-1"></a>exp3a[, treatment <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(treatment, <span class="at">levels =</span> treatment_levels)]</span>
<span id="cb696-30"><a href="counts.html#cb696-30" aria-hidden="true" tabindex="-1"></a>genotype_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;FAK_wt&quot;</span>, <span class="st">&quot;FAK_ko&quot;</span>)</span>
<span id="cb696-31"><a href="counts.html#cb696-31" aria-hidden="true" tabindex="-1"></a>exp3a[, genotype <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(genotype, <span class="at">levels =</span> genotype_levels)]</span></code></pre></div>

</div>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="4">
<li id="fn4"><p>the variance is less than that expected by the probability model<a href="counts.html#fnref4" class="footnote-back">↩︎</a></p></li>
<li id="fn5"><p>the variance is greater than that expected by the probability model<a href="counts.html#fnref5" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="pre-post.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gls.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/chapters/60-glm01-counts.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Walker-elementary-statistical-modeling-draft.pdf", "Walker-elementary-statistical-modeling-draft.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
