<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 17 Linear models with two categorical \(X\) – Factorial linear models (“two-way ANOVA”) | Elements of Statistical Modeling for Experimental Biology</title>
  <meta name="description" content="A first course in statistical modeling for experimental biology researchers" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 17 Linear models with two categorical \(X\) – Factorial linear models (“two-way ANOVA”) | Elements of Statistical Modeling for Experimental Biology" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A first course in statistical modeling for experimental biology researchers" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 17 Linear models with two categorical \(X\) – Factorial linear models (“two-way ANOVA”) | Elements of Statistical Modeling for Experimental Biology" />
  
  <meta name="twitter:description" content="A first course in statistical modeling for experimental biology researchers" />
  

<meta name="author" content="Copyright 2018 Jeffrey A. Walker" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="covariates.html"/>
<link rel="next" href="part-vii-expanding-the-linear-model.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="libs/d3-3.3.8/d3.min.js"></script>
<script src="libs/dagre-0.4.0/dagre-d3.min.js"></script>
<link href="libs/mermaid-0.3.0/dist/mermaid.css" rel="stylesheet" />
<script src="libs/mermaid-0.3.0/dist/mermaid.slim.min.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/chromatography-0.1/chromatography.js"></script>
<script src="libs/DiagrammeR-binding-1.0.6.1/DiagrammeR.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Elements of Applied Biostatistics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#math"><i class="fa fa-check"></i><b>0.1</b> Math</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#r-and-programming"><i class="fa fa-check"></i><b>0.2</b> R and programming</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="a-table-of-models.html"><a href="a-table-of-models.html"><i class="fa fa-check"></i>A Table of Models</a>
<ul>
<li class="chapter" data-level="" data-path="a-table-of-models.html"><a href="a-table-of-models.html#including-mapping-between-linear-models-and-classical-tests"><i class="fa fa-check"></i>including mapping between linear models and classical tests</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-i-getting-started.html"><a href="part-i-getting-started.html"><i class="fa fa-check"></i>Part I: Getting Started</a></li>
<li class="chapter" data-level="1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html"><i class="fa fa-check"></i><b>1</b> Getting Started – R Projects and R Markdown</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#r-vs-r-studio"><i class="fa fa-check"></i><b>1.1</b> R vs R Studio</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#download-and-install-r-and-r-studio"><i class="fa fa-check"></i><b>1.2</b> Download and install R and R studio</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#open-r-studio-and-modify-the-workspace-preference"><i class="fa fa-check"></i><b>1.3</b> Open R Studio and modify the workspace preference</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#if-you-didnt-modify-the-workspace-preferences-from-the-previous-section-go-back-and-do-it"><i class="fa fa-check"></i><b>1.4</b> If you didn’t modify the workspace preferences from the previous section, go back and do it</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#r-markdown-in-a-nutshell"><i class="fa fa-check"></i><b>1.5</b> R Markdown in a nutshell</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#install-r-markdown"><i class="fa fa-check"></i><b>1.6</b> Install R Markdown</a></li>
<li class="chapter" data-level="1.7" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#importing-packages"><i class="fa fa-check"></i><b>1.7</b> Importing Packages</a></li>
<li class="chapter" data-level="1.8" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#setup-create-project"><i class="fa fa-check"></i><b>1.8</b> Create an R Studio Project for this textbook</a></li>
<li class="chapter" data-level="1.9" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#working-on-a-project-in-a-nutshell"><i class="fa fa-check"></i><b>1.9</b> Working on a project, in a nutshell</a></li>
<li class="chapter" data-level="1.10" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-an-r-markdown-file-for-this-chapter"><i class="fa fa-check"></i><b>1.10</b> Create an R Markdown file for this Chapter</a>
<ul>
<li class="chapter" data-level="1.10.1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#modify-the-yaml-header"><i class="fa fa-check"></i><b>1.10.1</b> Modify the yaml header</a></li>
<li class="chapter" data-level="1.10.2" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#modify-the-setup-chunk"><i class="fa fa-check"></i><b>1.10.2</b> Modify the “setup” chunk</a></li>
<li class="chapter" data-level="1.10.3" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-a-fake-data-chunk"><i class="fa fa-check"></i><b>1.10.3</b> Create a “fake-data” chunk</a></li>
<li class="chapter" data-level="1.10.4" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-a-plot-chunk"><i class="fa fa-check"></i><b>1.10.4</b> Create a “plot” chunk</a></li>
<li class="chapter" data-level="1.10.5" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#knit"><i class="fa fa-check"></i><b>1.10.5</b> Knit</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-ii-an-introduction-to-the-analysis-of-experimental-data-with-a-linear-model.html"><a href="part-ii-an-introduction-to-the-analysis-of-experimental-data-with-a-linear-model.html"><i class="fa fa-check"></i>Part II: An introduction to the analysis of experimental data with a linear model</a></li>
<li class="chapter" data-level="2" data-path="analyzing-experimental-data-with-a-linear-model.html"><a href="analyzing-experimental-data-with-a-linear-model.html"><i class="fa fa-check"></i><b>2</b> Analyzing experimental data with a linear model</a>
<ul>
<li class="chapter" data-level="2.1" data-path="analyzing-experimental-data-with-a-linear-model.html"><a href="analyzing-experimental-data-with-a-linear-model.html#this-text-is-about-using-linear-models-to-estimate-treatment-effects-and-the-uncertainty-in-our-estimates.-this-raises-the-question-what-is-an-effect"><i class="fa fa-check"></i><b>2.1</b> This text is about using linear models to estimate treatment effects and the uncertainty in our estimates. This, raises the question, what is “an effect?”</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="ask1-bio.html"><a href="ask1-bio.html"><i class="fa fa-check"></i><b>3</b> Background physiology to the experiments in Figure 2 of “ASK1 inhibits browning of white adipose tissue in obesity”</a></li>
<li class="chapter" data-level="4" data-path="ask1-report.html"><a href="ask1-report.html"><i class="fa fa-check"></i><b>4</b> Analyses for Figure 2 of “ASK1 inhibits browning of white adipose tissue in obesity”</a>
<ul>
<li class="chapter" data-level="4.1" data-path="ask1-report.html"><a href="ask1-report.html#setup"><i class="fa fa-check"></i><b>4.1</b> Setup</a></li>
<li class="chapter" data-level="4.2" data-path="ask1-report.html"><a href="ask1-report.html#data-source"><i class="fa fa-check"></i><b>4.2</b> Data source</a></li>
<li class="chapter" data-level="4.3" data-path="ask1-report.html"><a href="ask1-report.html#control-the-color-palette"><i class="fa fa-check"></i><b>4.3</b> control the color palette</a></li>
<li class="chapter" data-level="4.4" data-path="ask1-report.html"><a href="ask1-report.html#useful-functions"><i class="fa fa-check"></i><b>4.4</b> useful functions</a></li>
<li class="chapter" data-level="4.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2b-effect-of-ask1-deletion-on-growth-body-weight"><i class="fa fa-check"></i><b>4.5</b> figure 2b – effect of ASK1 deletion on growth (body weight)</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2b-import"><i class="fa fa-check"></i><b>4.5.1</b> figure 2b – import</a></li>
<li class="chapter" data-level="4.5.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2b-exploratory-plots"><i class="fa fa-check"></i><b>4.5.2</b> figure 2b – exploratory plots</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-effect-of-ask1-deletion-on-final-body-weight"><i class="fa fa-check"></i><b>4.6</b> Figure 2c – Effect of ASK1 deletion on final body weight</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-import"><i class="fa fa-check"></i><b>4.6.1</b> Figure 2c – import</a></li>
<li class="chapter" data-level="4.6.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-check-own-computation-of-weight-change-v-imported-value"><i class="fa fa-check"></i><b>4.6.2</b> Figure 2c – check own computation of weight change v imported value</a></li>
<li class="chapter" data-level="4.6.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-exploratory-plots"><i class="fa fa-check"></i><b>4.6.3</b> Figure 2c – exploratory plots</a></li>
<li class="chapter" data-level="4.6.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-fit-the-model-m1-lm"><i class="fa fa-check"></i><b>4.6.4</b> Figure 2c – fit the model: m1 (lm)</a></li>
<li class="chapter" data-level="4.6.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-check-the-model-m1"><i class="fa fa-check"></i><b>4.6.5</b> Figure 2c – check the model: m1</a></li>
<li class="chapter" data-level="4.6.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-fit-the-model-m2-gamma-glm"><i class="fa fa-check"></i><b>4.6.6</b> Figure 2c – fit the model: m2 (gamma glm)</a></li>
<li class="chapter" data-level="4.6.7" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-check-the-model-m2"><i class="fa fa-check"></i><b>4.6.7</b> Figure 2c – check the model, m2</a></li>
<li class="chapter" data-level="4.6.8" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-inference-from-the-model"><i class="fa fa-check"></i><b>4.6.8</b> Figure 2c – inference from the model</a></li>
<li class="chapter" data-level="4.6.9" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-plot-the-model"><i class="fa fa-check"></i><b>4.6.9</b> Figure 2c – plot the model</a></li>
<li class="chapter" data-level="4.6.10" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-report"><i class="fa fa-check"></i><b>4.6.10</b> Figure 2c – report</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve"><i class="fa fa-check"></i><b>4.7</b> Figure 2d – Effect of ASK1 KO on glucose tolerance (whole curve)</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-import"><i class="fa fa-check"></i><b>4.7.1</b> Figure 2d – Import</a></li>
<li class="chapter" data-level="4.7.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-exploratory-plots"><i class="fa fa-check"></i><b>4.7.2</b> Figure 2d – exploratory plots</a></li>
<li class="chapter" data-level="4.7.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-fit-the-model"><i class="fa fa-check"></i><b>4.7.3</b> Figure 2d – fit the model</a></li>
<li class="chapter" data-level="4.7.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-check-the-model"><i class="fa fa-check"></i><b>4.7.4</b> Figure 2d – check the model</a></li>
<li class="chapter" data-level="4.7.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-inference"><i class="fa fa-check"></i><b>4.7.5</b> Figure 2d – inference</a></li>
<li class="chapter" data-level="4.7.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-plot-the-model"><i class="fa fa-check"></i><b>4.7.6</b> Figure 2d – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure"><i class="fa fa-check"></i><b>4.8</b> Figure 2e – Effect of ASK1 deletion on glucose tolerance (summary measure)</a>
<ul>
<li class="chapter" data-level="4.8.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-message-the-data"><i class="fa fa-check"></i><b>4.8.1</b> Figure 2e – message the data</a></li>
<li class="chapter" data-level="4.8.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-exploratory-plots"><i class="fa fa-check"></i><b>4.8.2</b> Figure 2e – exploratory plots</a></li>
<li class="chapter" data-level="4.8.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-fit-the-model"><i class="fa fa-check"></i><b>4.8.3</b> Figure 2e – fit the model</a></li>
<li class="chapter" data-level="4.8.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-check-the-model"><i class="fa fa-check"></i><b>4.8.4</b> Figure 2e – check the model</a></li>
<li class="chapter" data-level="4.8.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-inference-from-the-model"><i class="fa fa-check"></i><b>4.8.5</b> Figure 2e – inference from the model</a></li>
<li class="chapter" data-level="4.8.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-plot-the-model"><i class="fa fa-check"></i><b>4.8.6</b> Figure 2e – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate"><i class="fa fa-check"></i><b>4.9</b> Figure 2f – Effect of ASK1 deletion on glucose infusion rate</a>
<ul>
<li class="chapter" data-level="4.9.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-import"><i class="fa fa-check"></i><b>4.9.1</b> Figure 2f – import</a></li>
<li class="chapter" data-level="4.9.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-exploratory-plots"><i class="fa fa-check"></i><b>4.9.2</b> Figure 2f – exploratory plots</a></li>
<li class="chapter" data-level="4.9.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-fit-the-model"><i class="fa fa-check"></i><b>4.9.3</b> Figure 2f – fit the model</a></li>
<li class="chapter" data-level="4.9.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-check-the-model"><i class="fa fa-check"></i><b>4.9.4</b> Figure 2f – check the model</a></li>
<li class="chapter" data-level="4.9.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-inference"><i class="fa fa-check"></i><b>4.9.5</b> Figure 2f – inference</a></li>
<li class="chapter" data-level="4.9.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-plot-the-model"><i class="fa fa-check"></i><b>4.9.6</b> Figure 2f – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.10" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake"><i class="fa fa-check"></i><b>4.10</b> Figure 2g – Effect of ASK1 deletion on tissue-specific glucose uptake</a>
<ul>
<li class="chapter" data-level="4.10.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-import"><i class="fa fa-check"></i><b>4.10.1</b> Figure 2g – import</a></li>
<li class="chapter" data-level="4.10.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-exploratory-plots"><i class="fa fa-check"></i><b>4.10.2</b> Figure 2g – exploratory plots</a></li>
<li class="chapter" data-level="4.10.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-fit-the-model"><i class="fa fa-check"></i><b>4.10.3</b> Figure 2g – fit the model</a></li>
<li class="chapter" data-level="4.10.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-check-the-model"><i class="fa fa-check"></i><b>4.10.4</b> Figure 2g – check the model</a></li>
<li class="chapter" data-level="4.10.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-inference"><i class="fa fa-check"></i><b>4.10.5</b> Figure 2g – inference</a></li>
<li class="chapter" data-level="4.10.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-plot-the-model"><i class="fa fa-check"></i><b>4.10.6</b> Figure 2g – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="ask1-report.html"><a href="ask1-report.html#figure-2h"><i class="fa fa-check"></i><b>4.11</b> Figure 2h</a></li>
<li class="chapter" data-level="4.12" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-effect-of-ask1-deletion-on-liver-tg"><i class="fa fa-check"></i><b>4.12</b> Figure 2i – Effect of ASK1 deletion on liver TG</a>
<ul>
<li class="chapter" data-level="4.12.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-fit-the-model"><i class="fa fa-check"></i><b>4.12.1</b> Figure 2i – fit the model</a></li>
<li class="chapter" data-level="4.12.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-check-the-model"><i class="fa fa-check"></i><b>4.12.2</b> Figure 2i – check the model</a></li>
<li class="chapter" data-level="4.12.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-inference"><i class="fa fa-check"></i><b>4.12.3</b> Figure 2i – inference</a></li>
<li class="chapter" data-level="4.12.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-plot-the-model"><i class="fa fa-check"></i><b>4.12.4</b> Figure 2i – plot the model</a></li>
<li class="chapter" data-level="4.12.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-report-the-model"><i class="fa fa-check"></i><b>4.12.5</b> Figure 2i – report the model</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="ask1-report.html"><a href="ask1-report.html#figure-2j"><i class="fa fa-check"></i><b>4.13</b> Figure 2j</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-iii-r-fundamentals.html"><a href="part-iii-r-fundamentals.html"><i class="fa fa-check"></i>Part III: R fundamentals</a></li>
<li class="chapter" data-level="5" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html"><i class="fa fa-check"></i><b>5</b> Data – Reading, Wrangling, and Writing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#learning-from-this-chapter"><i class="fa fa-check"></i><b>5.1</b> Learning from this chapter</a></li>
<li class="chapter" data-level="5.2" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#data-working-in-r"><i class="fa fa-check"></i><b>5.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#importing-data"><i class="fa fa-check"></i><b>5.2.1</b> Importing data</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#data-wrangling"><i class="fa fa-check"></i><b>5.3</b> Data wrangling</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#reshaping-data-wide-to-long"><i class="fa fa-check"></i><b>5.3.1</b> Reshaping data – Wide to long</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#reshaping-data-transpose-turning-the-columns-into-rows"><i class="fa fa-check"></i><b>5.3.2</b> Reshaping data – Transpose (turning the columns into rows)</a></li>
<li class="chapter" data-level="5.3.3" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#combining-data"><i class="fa fa-check"></i><b>5.3.3</b> Combining data</a></li>
<li class="chapter" data-level="5.3.4" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#subsetting-data"><i class="fa fa-check"></i><b>5.3.4</b> Subsetting data</a></li>
<li class="chapter" data-level="5.3.5" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#wrangling-columns"><i class="fa fa-check"></i><b>5.3.5</b> Wrangling columns</a></li>
<li class="chapter" data-level="5.3.6" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#missing-data"><i class="fa fa-check"></i><b>5.3.6</b> Missing data</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#saving-data"><i class="fa fa-check"></i><b>5.4</b> Saving data</a></li>
<li class="chapter" data-level="5.5" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#exercises"><i class="fa fa-check"></i><b>5.5</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="plotting-models.html"><a href="plotting-models.html"><i class="fa fa-check"></i><b>6</b> Plotting Models</a>
<ul>
<li class="chapter" data-level="6.1" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plots-show-the-model-and-the-data"><i class="fa fa-check"></i><b>6.1</b> Pretty good plots show the model and the data</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plot-component-1-modeled-effects-plot"><i class="fa fa-check"></i><b>6.1.1</b> Pretty good plot component 1: Modeled effects plot</a></li>
<li class="chapter" data-level="6.1.2" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plot-component-2-modeled-mean-and-ci-plot"><i class="fa fa-check"></i><b>6.1.2</b> Pretty good plot component 2: Modeled mean and CI plot</a></li>
<li class="chapter" data-level="6.1.3" data-path="plotting-models.html"><a href="plotting-models.html#combining-effects-and-modeled-mean-and-ci-plots-an-effects-and-response-plot."><i class="fa fa-check"></i><b>6.1.3</b> Combining Effects and Modeled mean and CI plots – an Effects and response plot.</a></li>
<li class="chapter" data-level="6.1.4" data-path="plotting-models.html"><a href="plotting-models.html#some-comments-on-plot-components"><i class="fa fa-check"></i><b>6.1.4</b> Some comments on plot components</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="plotting-models.html"><a href="plotting-models.html#working-in-r"><i class="fa fa-check"></i><b>6.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="plotting-models.html"><a href="plotting-models.html#source-data"><i class="fa fa-check"></i><b>6.2.1</b> Source data</a></li>
<li class="chapter" data-level="6.2.2" data-path="plotting-models.html"><a href="plotting-models.html#how-to-plot-the-model"><i class="fa fa-check"></i><b>6.2.2</b> How to plot the model</a></li>
<li class="chapter" data-level="6.2.3" data-path="plotting-models.html"><a href="plotting-models.html#be-sure-ggplot_the_model-is-in-your-r-folder"><i class="fa fa-check"></i><b>6.2.3</b> Be sure ggplot_the_model is in your R folder</a></li>
<li class="chapter" data-level="6.2.4" data-path="plotting-models.html"><a href="plotting-models.html#how-to-use-the-plot-the-model-functions"><i class="fa fa-check"></i><b>6.2.4</b> How to use the Plot the Model functions</a></li>
<li class="chapter" data-level="6.2.5" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-a-response-plot-using-ggpubr"><i class="fa fa-check"></i><b>6.2.5</b> How to generate a Response Plot using ggpubr</a></li>
<li class="chapter" data-level="6.2.6" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-a-response-plot-with-a-grid-of-treatments-using-ggplot2"><i class="fa fa-check"></i><b>6.2.6</b> How to generate a Response Plot with a grid of treatments using ggplot2</a></li>
<li class="chapter" data-level="6.2.7" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-an-effects-plot"><i class="fa fa-check"></i><b>6.2.7</b> How to generate an Effects Plot</a></li>
<li class="chapter" data-level="6.2.8" data-path="plotting-models.html"><a href="plotting-models.html#how-to-combine-the-response-and-effects-plots"><i class="fa fa-check"></i><b>6.2.8</b> How to combine the response and effects plots</a></li>
<li class="chapter" data-level="6.2.9" data-path="plotting-models.html"><a href="plotting-models.html#how-to-add-the-interaction-effect-to-response-and-effects-plots"><i class="fa fa-check"></i><b>6.2.9</b> How to add the interaction effect to response and effects plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-iv-some-fundamentals-of-statistical-modeling.html"><a href="part-iv-some-fundamentals-of-statistical-modeling.html"><i class="fa fa-check"></i>Part IV: Some Fundamentals of Statistical Modeling</a></li>
<li class="chapter" data-level="7" data-path="uncertainty.html"><a href="uncertainty.html"><i class="fa fa-check"></i><b>7</b> Variability and Uncertainty (Standard Deviations, Standard Errors, Confidence Intervals)</a>
<ul>
<li class="chapter" data-level="7.1" data-path="uncertainty.html"><a href="uncertainty.html#the-sample-standard-deviation-vs.-the-standard-error-of-the-mean"><i class="fa fa-check"></i><b>7.1</b> The sample standard deviation vs. the standard error of the mean</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="uncertainty.html"><a href="uncertainty.html#sample-standard-deviation"><i class="fa fa-check"></i><b>7.1.1</b> Sample standard deviation</a></li>
<li class="chapter" data-level="7.1.2" data-path="uncertainty.html"><a href="uncertainty.html#standard-error-of-the-mean"><i class="fa fa-check"></i><b>7.1.2</b> Standard error of the mean</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="uncertainty.html"><a href="uncertainty.html#using-google-sheets-to-generate-fake-data-to-explore-the-standard-error"><i class="fa fa-check"></i><b>7.2</b> Using Google Sheets to generate fake data to explore the standard error</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="uncertainty.html"><a href="uncertainty.html#steps"><i class="fa fa-check"></i><b>7.2.1</b> Steps</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="uncertainty.html"><a href="uncertainty.html#using-r-to-generate-fake-data-to-explore-the-standard-error"><i class="fa fa-check"></i><b>7.3</b> Using R to generate fake data to explore the standard error</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="uncertainty.html"><a href="uncertainty.html#part-i"><i class="fa fa-check"></i><b>7.3.1</b> part I</a></li>
<li class="chapter" data-level="7.3.2" data-path="uncertainty.html"><a href="uncertainty.html#part-ii---means"><i class="fa fa-check"></i><b>7.3.2</b> part II - means</a></li>
<li class="chapter" data-level="7.3.3" data-path="uncertainty.html"><a href="uncertainty.html#part-iii---how-do-sd-and-se-change-as-sample-size-n-increases"><i class="fa fa-check"></i><b>7.3.3</b> part III - how do SD and SE change as sample size (n) increases?</a></li>
<li class="chapter" data-level="7.3.4" data-path="uncertainty.html"><a href="uncertainty.html#part-iv-generating-fake-data-with-for-loops"><i class="fa fa-check"></i><b>7.3.4</b> Part IV – Generating fake data with for-loops</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="uncertainty.html"><a href="uncertainty.html#bootstrap"><i class="fa fa-check"></i><b>7.4</b> Bootstrapped standard errors</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="uncertainty.html"><a href="uncertainty.html#an-example-of-bootstrapped-standard-errors-using-vole-data"><i class="fa fa-check"></i><b>7.4.1</b> An example of bootstrapped standard errors using vole data</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="uncertainty.html"><a href="uncertainty.html#confidence-interval"><i class="fa fa-check"></i><b>7.5</b> Confidence Interval</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="uncertainty.html"><a href="uncertainty.html#interpretation-of-a-confidence-interval"><i class="fa fa-check"></i><b>7.5.1</b> Interpretation of a confidence interval</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="p-values.html"><a href="p-values.html"><i class="fa fa-check"></i><b>8</b> P-values</a>
<ul>
<li class="chapter" data-level="8.1" data-path="p-values.html"><a href="p-values.html#a-p-value-is-the-probability-of-sampling-a-value-as-or-more-extreme-than-the-test-statistic-if-sampling-from-a-null-distribution"><i class="fa fa-check"></i><b>8.1</b> A <em>p</em>-value is the probability of sampling a value as or more extreme than the test statistic if sampling from a null distribution</a></li>
<li class="chapter" data-level="8.2" data-path="p-values.html"><a href="p-values.html#pump-your-intuition-creating-a-null-distribution"><i class="fa fa-check"></i><b>8.2</b> Pump your intuition – Creating a null distribution</a></li>
<li class="chapter" data-level="8.3" data-path="p-values.html"><a href="p-values.html#a-null-distribution-of-t-values-the-t-distribution"><i class="fa fa-check"></i><b>8.3</b> A null distribution of <em>t</em>-values – the <em>t</em> distribution</a></li>
<li class="chapter" data-level="8.4" data-path="p-values.html"><a href="p-values.html#p-values-from-the-perspective-of-permutation"><i class="fa fa-check"></i><b>8.4</b> P-values from the perspective of permutation</a></li>
<li class="chapter" data-level="8.5" data-path="p-values.html"><a href="p-values.html#parametric-vs.-non-parametric-statistics"><i class="fa fa-check"></i><b>8.5</b> Parametric vs. non-parametric statistics</a></li>
<li class="chapter" data-level="8.6" data-path="p-values.html"><a href="p-values.html#frequentist-probability-and-the-interpretation-of-p-values"><i class="fa fa-check"></i><b>8.6</b> frequentist probability and the interpretation of p-values</a>
<ul>
<li class="chapter" data-level="8.6.1" data-path="p-values.html"><a href="p-values.html#background"><i class="fa fa-check"></i><b>8.6.1</b> Background</a></li>
<li class="chapter" data-level="8.6.2" data-path="p-values.html"><a href="p-values.html#this-book-covers-frequentist-approaches-to-statistical-modeling-and-when-a-probability-arises-such-as-the-p-value-of-a-test-statistic-this-will-be-a-frequentist-probability."><i class="fa fa-check"></i><b>8.6.2</b> This book covers frequentist approaches to statistical modeling and when a probability arises, such as the <em>p</em>-value of a test statistic, this will be a frequentist probability.</a></li>
<li class="chapter" data-level="8.6.3" data-path="p-values.html"><a href="p-values.html#two-interpretations-of-the-p-value"><i class="fa fa-check"></i><b>8.6.3</b> Two interpretations of the <em>p</em>-value</a></li>
<li class="chapter" data-level="8.6.4" data-path="p-values.html"><a href="p-values.html#nhst"><i class="fa fa-check"></i><b>8.6.4</b> NHST</a></li>
</ul></li>
<li class="chapter" data-level="8.7" data-path="p-values.html"><a href="p-values.html#some-major-misconceptions-of-the-p-value"><i class="fa fa-check"></i><b>8.7</b> Some major misconceptions of the <em>p</em>-value</a>
<ul>
<li class="chapter" data-level="8.7.1" data-path="p-values.html"><a href="p-values.html#misconception-p-is-the-probability-that-the-null-is-true-and-1-p-is-probability-that-the-alternative-is-true"><i class="fa fa-check"></i><b>8.7.1</b> Misconception: <em>p</em> is the probability that the null is true <em>and</em> <span class="math inline">\(1-p\)</span> is probability that the alternative is true</a></li>
<li class="chapter" data-level="8.7.2" data-path="p-values.html"><a href="p-values.html#misconception-a-p-value-is-repeatable"><i class="fa fa-check"></i><b>8.7.2</b> Misconception: a <em>p</em>-value is repeatable</a></li>
<li class="chapter" data-level="8.7.3" data-path="p-values.html"><a href="p-values.html#misconception-0.05-is-the-lifetime-rate-of-false-discoveries"><i class="fa fa-check"></i><b>8.7.3</b> Misconception: 0.05 is the lifetime rate of false discoveries</a></li>
<li class="chapter" data-level="8.7.4" data-path="p-values.html"><a href="p-values.html#misconception-a-low-p-value-indicates-an-important-effect"><i class="fa fa-check"></i><b>8.7.4</b> Misconception: a low <em>p</em>-value indicates an important effect</a></li>
<li class="chapter" data-level="8.7.5" data-path="p-values.html"><a href="p-values.html#misconception-a-low-p-value-indicates-high-model-fit-or-high-predictive-capacity"><i class="fa fa-check"></i><b>8.7.5</b> Misconception: a low <em>p</em>-value indicates high model fit or high predictive capacity</a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="p-values.html"><a href="p-values.html#what-the-p-value-does-not-mean"><i class="fa fa-check"></i><b>8.8</b> What the <em>p</em>-value does not mean</a></li>
<li class="chapter" data-level="8.9" data-path="p-values.html"><a href="p-values.html#recommendations"><i class="fa fa-check"></i><b>8.9</b> Recommendations</a>
<ul>
<li class="chapter" data-level="8.9.1" data-path="p-values.html"><a href="p-values.html#primary-sources-for-recommendations"><i class="fa fa-check"></i><b>8.9.1</b> Primary sources for recommendations</a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="p-values.html"><a href="p-values.html#problems"><i class="fa fa-check"></i><b>8.10</b> Problems</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="errors-in-inference.html"><a href="errors-in-inference.html"><i class="fa fa-check"></i><b>9</b> Errors in inference</a>
<ul>
<li class="chapter" data-level="9.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#classical-nhst-concepts-of-wrong"><i class="fa fa-check"></i><b>9.1</b> Classical NHST concepts of wrong</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-i-error"><i class="fa fa-check"></i><b>9.1.1</b> Type I error</a></li>
<li class="chapter" data-level="9.1.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#power"><i class="fa fa-check"></i><b>9.1.2</b> Power</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#a-non-neyman-pearson-concept-of-power"><i class="fa fa-check"></i><b>9.2</b> A non-Neyman-Pearson concept of power</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#estimation-error"><i class="fa fa-check"></i><b>9.2.1</b> Estimation error</a></li>
<li class="chapter" data-level="9.2.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#coverage"><i class="fa fa-check"></i><b>9.2.2</b> Coverage</a></li>
<li class="chapter" data-level="9.2.3" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-s-error"><i class="fa fa-check"></i><b>9.2.3</b> Type S error</a></li>
<li class="chapter" data-level="9.2.4" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-m-error"><i class="fa fa-check"></i><b>9.2.4</b> Type M error</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-v-introduction-to-linear-models.html"><a href="part-v-introduction-to-linear-models.html"><i class="fa fa-check"></i>Part V: Introduction to Linear Models</a></li>
<li class="chapter" data-level="10" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html"><i class="fa fa-check"></i><b>10</b> An introduction to linear models</a>
<ul>
<li class="chapter" data-level="10.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#lm-specifications"><i class="fa fa-check"></i><b>10.1</b> Two specifications of a linear model</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#the-error-draw-specification"><i class="fa fa-check"></i><b>10.1.1</b> The “error draw” specification</a></li>
<li class="chapter" data-level="10.1.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#the-conditional-draw-specification"><i class="fa fa-check"></i><b>10.1.2</b> The “conditional draw” specification</a></li>
<li class="chapter" data-level="10.1.3" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#comparing-the-error-draw-and-conditional-draw-ways-of-specifying-the-linear-model"><i class="fa fa-check"></i><b>10.1.3</b> Comparing the error-draw and conditional-draw ways of specifying the linear model</a></li>
<li class="chapter" data-level="10.1.4" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#anova-notation-of-a-linear-model"><i class="fa fa-check"></i><b>10.1.4</b> ANOVA notation of a linear model</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#a-linear-model-can-be-fit-to-data-with-continuous-discrete-or-categorical-x-variables"><i class="fa fa-check"></i><b>10.2</b> A linear model can be fit to data with continuous, discrete, or categorical <span class="math inline">\(X\)</span> variables</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#fitting-linear-models-to-experimental-data-in-which-the-x-variable-is-continuous-or-discrete"><i class="fa fa-check"></i><b>10.2.1</b> Fitting linear models to experimental data in which the <span class="math inline">\(X\)</span> variable is continuous or discrete</a></li>
<li class="chapter" data-level="10.2.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#slope"><i class="fa fa-check"></i><b>10.2.2</b> Fitting linear models to experimental data in which the <span class="math inline">\(X\)</span> variable is categorical</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#statistical-models-are-used-for-prediction-explanation-and-description"><i class="fa fa-check"></i><b>10.3</b> Statistical models are used for prediction, explanation, and description</a></li>
<li class="chapter" data-level="10.4" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#what-do-we-call-the-x-and-y-variables"><i class="fa fa-check"></i><b>10.4</b> What do we call the <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span> variables?</a></li>
<li class="chapter" data-level="10.5" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#modeling-strategy"><i class="fa fa-check"></i><b>10.5</b> Modeling strategy</a></li>
<li class="chapter" data-level="10.6" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#predictions-from-the-model"><i class="fa fa-check"></i><b>10.6</b> Predictions from the model</a></li>
<li class="chapter" data-level="10.7" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#inference-from-the-model"><i class="fa fa-check"></i><b>10.7</b> Inference from the model</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#assumptions-for-inference-with-a-statistical-model"><i class="fa fa-check"></i><b>10.7.1</b> Assumptions for inference with a statistical model</a></li>
<li class="chapter" data-level="10.7.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#specific-assumptions-for-inference-with-a-linear-model"><i class="fa fa-check"></i><b>10.7.2</b> Specific assumptions for inference with a linear model</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#linear-modelregression-model-orstatistical-model"><i class="fa fa-check"></i><b>10.8</b> “linear model,”regression model“, or”statistical model"?</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="regression.html"><a href="regression.html"><i class="fa fa-check"></i><b>11</b> Linear models with a single, continuous <em>X</em> (“regression”)</a>
<ul>
<li class="chapter" data-level="11.1" data-path="regression.html"><a href="regression.html#a-linear-model-with-a-single-continuous-x-is-classical-regression"><i class="fa fa-check"></i><b>11.1</b> A linear model with a single, continuous <em>X</em> is classical “regression”</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="regression.html"><a href="regression.html#analysis-of-green-down-data"><i class="fa fa-check"></i><b>11.1.1</b> Analysis of “green-down” data</a></li>
<li class="chapter" data-level="11.1.2" data-path="regression.html"><a href="regression.html#learning-from-the-green-down-example"><i class="fa fa-check"></i><b>11.1.2</b> Learning from the green-down example</a></li>
<li class="chapter" data-level="11.1.3" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-explanation-causal-models"><i class="fa fa-check"></i><b>11.1.3</b> Using a regression model for “explanation” – causal models</a></li>
<li class="chapter" data-level="11.1.4" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-prediction-prediction-models"><i class="fa fa-check"></i><b>11.1.4</b> Using a regression model for prediction – prediction models</a></li>
<li class="chapter" data-level="11.1.5" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-creating-a-new-response-variable-comparing-slopes-of-longitudinal-data"><i class="fa fa-check"></i><b>11.1.5</b> Using a regression model for creating a new response variable – comparing slopes of longitudinal data</a></li>
<li class="chapter" data-level="11.1.6" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-for-calibration"><i class="fa fa-check"></i><b>11.1.6</b> Using a regression model for for calibration</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="regression.html"><a href="regression.html#working-in-r-1"><i class="fa fa-check"></i><b>11.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="plotting-models.html"><a href="plotting-models.html#fit-the-model"><i class="fa fa-check"></i><b>11.2.1</b> Fitting the linear model</a></li>
<li class="chapter" data-level="11.2.2" data-path="regression.html"><a href="regression.html#getting-to-know-the-linear-model-the-summary-function"><i class="fa fa-check"></i><b>11.2.2</b> Getting to know the linear model: the <code>summary</code> function</a></li>
<li class="chapter" data-level="11.2.3" data-path="regression.html"><a href="regression.html#inference-the-coefficient-table"><i class="fa fa-check"></i><b>11.2.3</b> Inference – the coefficient table</a></li>
<li class="chapter" data-level="11.2.4" data-path="regression.html"><a href="regression.html#how-good-is-our-model-model-checking"><i class="fa fa-check"></i><b>11.2.4</b> How good is our model? – Model checking</a></li>
<li class="chapter" data-level="11.2.5" data-path="regression.html"><a href="regression.html#plotting-models-with-continuous-x"><i class="fa fa-check"></i><b>11.2.5</b> Plotting models with continuous <em>X</em></a></li>
<li class="chapter" data-level="11.2.6" data-path="regression.html"><a href="regression.html#creating-a-table-of-predicted-values-and-95-prediction-intervals"><i class="fa fa-check"></i><b>11.2.6</b> Creating a table of predicted values and 95% prediction intervals</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="regression.html"><a href="regression.html#hidden-code"><i class="fa fa-check"></i><b>11.3</b> Hidden code</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="regression.html"><a href="regression.html#import-and-plot-of-fig2c-ecosystem-warming-experimental-data"><i class="fa fa-check"></i><b>11.3.1</b> Import and plot of fig2c (ecosystem warming experimental) data</a></li>
<li class="chapter" data-level="11.3.2" data-path="regression.html"><a href="regression.html#import-and-plot-efig_3d-ecosysem-warming-observational-data"><i class="fa fa-check"></i><b>11.3.2</b> Import and plot efig_3d (Ecosysem warming observational) data</a></li>
<li class="chapter" data-level="11.3.3" data-path="regression.html"><a href="regression.html#import-and-plot-of-fig1f-methionine-restriction-data"><i class="fa fa-check"></i><b>11.3.3</b> Import and plot of fig1f (methionine restriction) data</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="regression.html"><a href="regression.html#try-it"><i class="fa fa-check"></i><b>11.4</b> Try it</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="regression.html"><a href="regression.html#a-prediction-model-from-the-literature"><i class="fa fa-check"></i><b>11.4.1</b> A prediction model from the literature</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="regression.html"><a href="regression.html#intuition-pumps"><i class="fa fa-check"></i><b>11.5</b> Intuition pumps</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="regression.html"><a href="regression.html#correlation-and-r2"><i class="fa fa-check"></i><b>11.5.1</b> Correlation and $R^2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="oneway.html"><a href="oneway.html"><i class="fa fa-check"></i><b>12</b> Linear models with a single, categorical <em>X</em> (“t-tests” and “ANOVA”)</a>
<ul>
<li class="chapter" data-level="12.1" data-path="oneway.html"><a href="oneway.html#oneway-data"><i class="fa fa-check"></i><b>12.1</b> A linear model with a single, categorical <em>X</em> variable estimates the effects of the levels of <em>X</em> on the response</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="oneway.html"><a href="oneway.html#oneway-example1"><i class="fa fa-check"></i><b>12.1.1</b> Example 1 (fig3d) – two treatment levels (“groups”)</a></li>
<li class="chapter" data-level="12.1.2" data-path="oneway.html"><a href="oneway.html#understanding-the-analysis-with-two-treatment-levels"><i class="fa fa-check"></i><b>12.1.2</b> Understanding the analysis with two treatment levels</a></li>
<li class="chapter" data-level="12.1.3" data-path="oneway.html"><a href="oneway.html#oneway-example2"><i class="fa fa-check"></i><b>12.1.3</b> Example 2 – three treatment levels (“groups”)</a></li>
<li class="chapter" data-level="12.1.4" data-path="oneway.html"><a href="oneway.html#understanding-the-analysis-with-three-or-more-treatment-levels"><i class="fa fa-check"></i><b>12.1.4</b> Understanding the analysis with three (or more) treatment levels</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="oneway.html"><a href="oneway.html#working-in-r-2"><i class="fa fa-check"></i><b>12.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="oneway.html"><a href="oneway.html#fit-the-model-1"><i class="fa fa-check"></i><b>12.2.1</b> Fit the model</a></li>
<li class="chapter" data-level="12.2.2" data-path="oneway.html"><a href="oneway.html#controlling-the-output-in-tables-using-the-coefficient-table-as-an-example"><i class="fa fa-check"></i><b>12.2.2</b> Controlling the output in tables using the coefficient table as an example</a></li>
<li class="chapter" data-level="12.2.3" data-path="oneway.html"><a href="oneway.html#using-the-emmeans-function"><i class="fa fa-check"></i><b>12.2.3</b> Using the emmeans function</a></li>
<li class="chapter" data-level="12.2.4" data-path="oneway.html"><a href="oneway.html#using-the-contrast-function"><i class="fa fa-check"></i><b>12.2.4</b> Using the contrast function</a></li>
<li class="chapter" data-level="12.2.5" data-path="oneway.html"><a href="oneway.html#how-to-generate-anova-tables"><i class="fa fa-check"></i><b>12.2.5</b> How to generate ANOVA tables</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="oneway.html"><a href="oneway.html#hidden-code-1"><i class="fa fa-check"></i><b>12.3</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="oneway.html"><a href="oneway.html#importing-and-wrangling-the-fig_3d-data-for-example-1"><i class="fa fa-check"></i><b>12.3.1</b> Importing and wrangling the fig_3d data for example 1</a></li>
<li class="chapter" data-level="12.3.2" data-path="oneway.html"><a href="oneway.html#importing-and-wrangling-the-fig2a-data-for-example-2"><i class="fa fa-check"></i><b>12.3.2</b> Importing and wrangling the fig2a data for example 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="model-checking.html"><a href="model-checking.html"><i class="fa fa-check"></i><b>13</b> Model Checking</a>
<ul>
<li class="chapter" data-level="13.1" data-path="model-checking.html"><a href="model-checking.html#all-statistical-analyses-should-be-followed-by-model-checking"><i class="fa fa-check"></i><b>13.1</b> All statistical analyses should be followed by model checking</a></li>
<li class="chapter" data-level="13.2" data-path="model-checking.html"><a href="model-checking.html#linear-model-assumptions"><i class="fa fa-check"></i><b>13.2</b> Linear model assumptions</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="model-checking.html"><a href="model-checking.html#a-bit-about-iid"><i class="fa fa-check"></i><b>13.2.1</b> A bit about IID</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="model-checking.html"><a href="model-checking.html#diagnostic-plots-use-the-residuals-from-the-model-fit"><i class="fa fa-check"></i><b>13.3</b> Diagnostic plots use the residuals from the model fit</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="model-checking.html"><a href="model-checking.html#residuals"><i class="fa fa-check"></i><b>13.3.1</b> Residuals</a></li>
<li class="chapter" data-level="13.3.2" data-path="model-checking.html"><a href="model-checking.html#normal-qq"><i class="fa fa-check"></i><b>13.3.2</b> A Normal Q-Q plot is used to check for characteristic departures from Normality</a></li>
<li class="chapter" data-level="13.3.3" data-path="model-checking.html"><a href="model-checking.html#mapping-qq-plot-departures-from-normality"><i class="fa fa-check"></i><b>13.3.3</b> Mapping QQ-plot departures from Normality</a></li>
<li class="chapter" data-level="13.3.4" data-path="model-checking.html"><a href="model-checking.html#model-checking-homoskedasticity"><i class="fa fa-check"></i><b>13.3.4</b> Model checking homoskedasticity</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="model-checking.html"><a href="model-checking.html#using-r"><i class="fa fa-check"></i><b>13.4</b> Using R</a>
<ul>
<li class="chapter" data-level="13.4.1" data-path="model-checking.html"><a href="model-checking.html#normal-q-q-plots"><i class="fa fa-check"></i><b>13.4.1</b> Normal Q-Q plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="violations.html"><a href="violations.html"><i class="fa fa-check"></i><b>14</b> Violations of independence, homogeneity, or Normality</a>
<ul>
<li class="chapter" data-level="14.1" data-path="violations.html"><a href="violations.html#oneway-paired-t"><i class="fa fa-check"></i><b>14.1</b> Lack of independence</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="violations.html"><a href="violations.html#violations-paired-t"><i class="fa fa-check"></i><b>14.1.1</b> Example 1 (exp1b) – a paired t-test is a special case of a linear mixed model</a></li>
<li class="chapter" data-level="14.1.2" data-path="violations.html"><a href="violations.html#violations-rmanova"><i class="fa fa-check"></i><b>14.1.2</b> Example 2 (diHOME exp2a) – A repeated measures ANOVA is a special case of a linear mixed model</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="violations.html"><a href="violations.html#oneway-welch"><i class="fa fa-check"></i><b>14.2</b> Heterogeneity of variances</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="violations.html"><a href="violations.html#when-groups-of-the-focal-test-have-variance"><i class="fa fa-check"></i><b>14.2.1</b> When groups of the focal test have &gt;&gt; variance</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="violations.html"><a href="violations.html#the-conditional-response-isnt-normal"><i class="fa fa-check"></i><b>14.3</b> The conditional response isn’t Normal</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="violations.html"><a href="violations.html#violations-6f"><i class="fa fa-check"></i><b>14.3.1</b> Example 1 (fig6f) – Linear models for non-normal count data</a></li>
<li class="chapter" data-level="14.3.2" data-path="violations.html"><a href="violations.html#my-data-arent-normal-what-is-the-best-practice"><i class="fa fa-check"></i><b>14.3.2</b> My data aren’t normal, what is the best practice?</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="violations.html"><a href="violations.html#hidden-code-2"><i class="fa fa-check"></i><b>14.4</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-exp1b-data"><i class="fa fa-check"></i><b>14.4.1</b> Importing and wrangling the exp1b data</a></li>
<li class="chapter" data-level="14.4.2" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-exp2a-data"><i class="fa fa-check"></i><b>14.4.2</b> Importing and wrangling the exp2a data</a></li>
<li class="chapter" data-level="14.4.3" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-fig6f-data"><i class="fa fa-check"></i><b>14.4.3</b> Importing and wrangling the fig6f data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="issues.html"><a href="issues.html"><i class="fa fa-check"></i><b>15</b> Issues in inference</a>
<ul>
<li class="chapter" data-level="15.1" data-path="issues.html"><a href="issues.html#replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means"><i class="fa fa-check"></i><b>15.1</b> Replicated experiments – include <span class="math inline">\(\texttt{Experiment}\)</span> as a random factor (better than one-way ANOVA of means)</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="issues.html"><a href="issues.html#issues-exp4d"><i class="fa fa-check"></i><b>15.1.1</b> Multiple experiments Example 1 (wound healing Exp4d)</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="issues.html"><a href="issues.html#issues-pre-post"><i class="fa fa-check"></i><b>15.2</b> Comparing change from baseline (pre-post)</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="issues.html"><a href="issues.html#pre-post-example-1-dpp4-fig4c"><i class="fa fa-check"></i><b>15.2.1</b> Pre-post example 1 (DPP4 fig4c)</a></li>
<li class="chapter" data-level="15.2.2" data-path="issues.html"><a href="issues.html#what-if-the-data-in-example-1-were-from-from-an-experiment-where-the-treatment-was-applied-prior-to-the-baseline-measure"><i class="fa fa-check"></i><b>15.2.2</b> What if the data in example 1 were from from an experiment where the treatment was applied prior to the baseline measure?</a></li>
<li class="chapter" data-level="15.2.3" data-path="issues.html"><a href="issues.html#pre-post-example-2-xx-males-fig1c"><i class="fa fa-check"></i><b>15.2.3</b> Pre-post example 2 (XX males fig1c)</a></li>
<li class="chapter" data-level="15.2.4" data-path="issues.html"><a href="issues.html#issues-regression-to-mean"><i class="fa fa-check"></i><b>15.2.4</b> Regression to the mean</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="issues.html"><a href="issues.html#longitudinal-designs-with-more-than-one-post-baseline-measure"><i class="fa fa-check"></i><b>15.3</b> Longitudinal designs with more than one-post baseline measure</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="issues.html"><a href="issues.html#issues-auc"><i class="fa fa-check"></i><b>15.3.1</b> Area under the curve (AUC)</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="issues.html"><a href="issues.html#comparing-ratios"><i class="fa fa-check"></i><b>15.4</b> Comparing ratios</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="issues.html"><a href="issues.html#the-ratio-is-a-density"><i class="fa fa-check"></i><b>15.4.1</b> The ratio is a density</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="issues.html"><a href="issues.html#dont-do-this-stuff"><i class="fa fa-check"></i><b>15.5</b> Don’t do this stuff</a>
<ul>
<li class="chapter" data-level="15.5.1" data-path="issues.html"><a href="issues.html#normalize-the-response-so-that-all-control-values-are-equal-to-1."><i class="fa fa-check"></i><b>15.5.1</b> Normalize the response so that all control values are equal to 1.</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="issues.html"><a href="issues.html#a-difference-in-significance-is-not-necessarily-significant"><i class="fa fa-check"></i><b>15.6</b> A difference in significance is not necessarily significant</a></li>
<li class="chapter" data-level="15.7" data-path="issues.html"><a href="issues.html#researcher-degrees-of-freedom"><i class="fa fa-check"></i><b>15.7</b> Researcher degrees of freedom</a></li>
<li class="chapter" data-level="15.8" data-path="issues.html"><a href="issues.html#hidden-code-3"><i class="fa fa-check"></i><b>15.8</b> Hidden code</a>
<ul>
<li class="chapter" data-level="15.8.1" data-path="issues.html"><a href="issues.html#import-exp4d-vimentin-cell-count-data-replicate-experiments-example"><i class="fa fa-check"></i><b>15.8.1</b> Import exp4d vimentin cell count data (replicate experiments example)</a></li>
<li class="chapter" data-level="15.8.2" data-path="issues.html"><a href="issues.html#import-fig4c-data"><i class="fa fa-check"></i><b>15.8.2</b> Import Fig4c data</a></li>
<li class="chapter" data-level="15.8.3" data-path="issues.html"><a href="issues.html#xx-males-fig1c"><i class="fa fa-check"></i><b>15.8.3</b> XX males fig1c</a></li>
<li class="chapter" data-level="15.8.4" data-path="issues.html"><a href="issues.html#generation-of-fake-data-to-illustrate-regression-to-the-mean"><i class="fa fa-check"></i><b>15.8.4</b> Generation of fake data to illustrate regression to the mean</a></li>
<li class="chapter" data-level="15.8.5" data-path="issues.html"><a href="issues.html#import-fig3f"><i class="fa fa-check"></i><b>15.8.5</b> Import fig3f</a></li>
</ul></li>
</ul></li>
<li><a href="part-vi-more-than-one-x-multivariable-models.html#part-vi-more-than-one-x-multivariable-models">Part VI: More than one <span class="math inline">\(X\)</span> – Multivariable Models</a></li>
<li class="chapter" data-level="16" data-path="covariates.html"><a href="covariates.html"><i class="fa fa-check"></i><b>16</b> Linear models with added covariates (“ANCOVA”)</a>
<ul>
<li class="chapter" data-level="16.1" data-path="covariates.html"><a href="covariates.html#adding-covariates-can-increases-the-precision-of-the-effect-of-interest"><i class="fa fa-check"></i><b>16.1</b> Adding covariates can increases the precision of the effect of interest</a></li>
<li class="chapter" data-level="16.2" data-path="covariates.html"><a href="covariates.html#understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data"><i class="fa fa-check"></i><b>16.2</b> Understanding a linear model with an added covariate – heart necrosis data</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="covariates.html"><a href="covariates.html#fit-the-model-5"><i class="fa fa-check"></i><b>16.2.1</b> Fit the model</a></li>
<li class="chapter" data-level="16.2.2" data-path="covariates.html"><a href="covariates.html#plot-the-model-1"><i class="fa fa-check"></i><b>16.2.2</b> Plot the model</a></li>
<li class="chapter" data-level="16.2.3" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-model-coefficients"><i class="fa fa-check"></i><b>16.2.3</b> Interpretation of the model coefficients</a></li>
<li class="chapter" data-level="16.2.4" data-path="covariates.html"><a href="covariates.html#everything-adds-up"><i class="fa fa-check"></i><b>16.2.4</b> Everything adds up</a></li>
<li class="chapter" data-level="16.2.5" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-estimated-marginal-means"><i class="fa fa-check"></i><b>16.2.5</b> Interpretation of the estimated marginal means</a></li>
<li class="chapter" data-level="16.2.6" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-contrasts"><i class="fa fa-check"></i><b>16.2.6</b> Interpretation of the contrasts</a></li>
<li class="chapter" data-level="16.2.7" data-path="covariates.html"><a href="covariates.html#adding-the-covariate-improves-inference"><i class="fa fa-check"></i><b>16.2.7</b> Adding the covariate improves inference</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="covariates.html"><a href="covariates.html#understanding-interaction-effects-with-covariates"><i class="fa fa-check"></i><b>16.3</b> Understanding interaction effects with covariates</a>
<ul>
<li class="chapter" data-level="16.3.1" data-path="covariates.html"><a href="covariates.html#fit-the-model-6"><i class="fa fa-check"></i><b>16.3.1</b> Fit the model</a></li>
<li class="chapter" data-level="16.3.2" data-path="covariates.html"><a href="covariates.html#plot-the-model-with-interaction-effect"><i class="fa fa-check"></i><b>16.3.2</b> Plot the model with interaction effect</a></li>
<li class="chapter" data-level="16.3.3" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-model-coefficients-1"><i class="fa fa-check"></i><b>16.3.3</b> Interpretation of the model coefficients</a></li>
<li class="chapter" data-level="16.3.4" data-path="covariates.html"><a href="covariates.html#what-is-the-effect-of-a-treatment-if-interactions-are-modeled-it-depends."><i class="fa fa-check"></i><b>16.3.4</b> What is the effect of a treatment, if interactions are modeled? – it depends.</a></li>
<li class="chapter" data-level="16.3.5" data-path="covariates.html"><a href="covariates.html#which-model-do-we-use-mathcalm_1-or-mathcalm_2"><i class="fa fa-check"></i><b>16.3.5</b> Which model do we use, <span class="math inline">\(\mathcal{M}_1\)</span> or <span class="math inline">\(\mathcal{M}_2\)</span>?</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="covariates.html"><a href="covariates.html#understanding-ancova-tables"><i class="fa fa-check"></i><b>16.4</b> Understanding ANCOVA tables</a></li>
<li class="chapter" data-level="16.5" data-path="covariates.html"><a href="covariates.html#working-in-r-3"><i class="fa fa-check"></i><b>16.5</b> Working in R</a>
<ul>
<li class="chapter" data-level="16.5.1" data-path="covariates.html"><a href="covariates.html#importing-the-heart-necrosis-data"><i class="fa fa-check"></i><b>16.5.1</b> Importing the heart necrosis data</a></li>
<li class="chapter" data-level="16.5.2" data-path="covariates.html"><a href="covariates.html#fitting-the-model"><i class="fa fa-check"></i><b>16.5.2</b> Fitting the model</a></li>
<li class="chapter" data-level="16.5.3" data-path="covariates.html"><a href="covariates.html#using-the-emmeans-function-1"><i class="fa fa-check"></i><b>16.5.3</b> Using the emmeans function</a></li>
<li class="chapter" data-level="16.5.4" data-path="covariates.html"><a href="covariates.html#ancova-tables"><i class="fa fa-check"></i><b>16.5.4</b> ANCOVA tables</a></li>
<li class="chapter" data-level="16.5.5" data-path="covariates.html"><a href="covariates.html#plotting-the-model"><i class="fa fa-check"></i><b>16.5.5</b> Plotting the model</a></li>
</ul></li>
<li class="chapter" data-level="16.6" data-path="covariates.html"><a href="covariates.html#best-practices"><i class="fa fa-check"></i><b>16.6</b> Best practices</a>
<ul>
<li class="chapter" data-level="16.6.1" data-path="covariates.html"><a href="covariates.html#do-not-use-a-ratio-of-partwhole-as-a-response-variable-instead-add-the-denominator-as-a-covariate"><i class="fa fa-check"></i><b>16.6.1</b> Do not use a ratio of part:whole as a response variable – instead add the denominator as a covariate</a></li>
<li class="chapter" data-level="16.6.2" data-path="covariates.html"><a href="covariates.html#do-not-use-change-from-baseline-as-a-response-variable-instead-add-the-baseline-measure-as-a-covariate"><i class="fa fa-check"></i><b>16.6.2</b> Do not use change from baseline as a response variable – instead add the baseline measure as a covariate</a></li>
<li class="chapter" data-level="16.6.3" data-path="covariates.html"><a href="covariates.html#do-not-test-for-balance-of-baseline-measures"><i class="fa fa-check"></i><b>16.6.3</b> Do not “test for balance” of baseline measures</a></li>
</ul></li>
<li class="chapter" data-level="16.7" data-path="covariates.html"><a href="covariates.html#best-practices-2-use-a-covariate-instead-of-normalizing-a-response"><i class="fa fa-check"></i><b>16.7</b> Best practices 2: Use a covariate instead of normalizing a response</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="factorial.html"><a href="factorial.html"><i class="fa fa-check"></i><b>17</b> Linear models with two categorical <span class="math inline">\(X\)</span> – Factorial linear models (“two-way ANOVA”)</a>
<ul>
<li class="chapter" data-level="17.1" data-path="factorial.html"><a href="factorial.html#a-linear-model-with-crossed-factors-estimates-interaction-effects"><i class="fa fa-check"></i><b>17.1</b> A linear model with crossed factors estimates interaction effects</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="factorial.html"><a href="factorial.html#an-interaction-is-a-difference-in-simple-effects"><i class="fa fa-check"></i><b>17.1.1</b> An interaction is a difference in simple effects</a></li>
<li class="chapter" data-level="17.1.2" data-path="factorial.html"><a href="factorial.html#a-linear-model-with-crossed-factors-includes-interaction-effects"><i class="fa fa-check"></i><b>17.1.2</b> A linear model with crossed factors includes interaction effects</a></li>
<li class="chapter" data-level="17.1.3" data-path="factorial.html"><a href="factorial.html#factorial-experiments-are-frequently-analyzed-as-flattened-linear-models-in-the-experimental-biology-literature"><i class="fa fa-check"></i><b>17.1.3</b> factorial experiments are frequently analyzed as flattened linear models in the experimental biology literature</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="factorial.html"><a href="factorial.html#example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data"><i class="fa fa-check"></i><b>17.2</b> Example 1 – Estimation of a treatment effect relative to a control effect (“Something different”) (Experiment 2j glucose uptake data)</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="factorial.html"><a href="factorial.html#twoway-understand1"><i class="fa fa-check"></i><b>17.2.1</b> Understand the experimental design</a></li>
<li class="chapter" data-level="17.2.2" data-path="factorial.html"><a href="factorial.html#fit-the-linear-model"><i class="fa fa-check"></i><b>17.2.2</b> Fit the linear model</a></li>
<li class="chapter" data-level="17.2.3" data-path="factorial.html"><a href="factorial.html#twoway-exp2j-inf"><i class="fa fa-check"></i><b>17.2.3</b> Inference</a></li>
<li class="chapter" data-level="17.2.4" data-path="factorial.html"><a href="factorial.html#plot-the-model-2"><i class="fa fa-check"></i><b>17.2.4</b> Plot the model</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-1"><i class="fa fa-check"></i><b>17.3</b> Understanding the linear model with crossed factors 1</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="factorial.html"><a href="factorial.html#twoway-what-coefs-are"><i class="fa fa-check"></i><b>17.3.1</b> What the coefficients are</a></li>
<li class="chapter" data-level="17.3.2" data-path="factorial.html"><a href="factorial.html#the-interaction-effect-is-something-different"><i class="fa fa-check"></i><b>17.3.2</b> The interaction effect is something different</a></li>
<li class="chapter" data-level="17.3.3" data-path="factorial.html"><a href="factorial.html#why-we-want-to-compare-the-treatment-effect-to-a-control-effect"><i class="fa fa-check"></i><b>17.3.3</b> Why we want to compare the treatment effect to a control effect</a></li>
<li class="chapter" data-level="17.3.4" data-path="factorial.html"><a href="factorial.html#the-order-of-the-factors-in-the-model-tells-the-same-story-differently"><i class="fa fa-check"></i><b>17.3.4</b> The order of the factors in the model tells the same story differently</a></li>
<li class="chapter" data-level="17.3.5" data-path="factorial.html"><a href="factorial.html#power-for-the-interaction-effect-is-less-than-that-for-simple-effects"><i class="fa fa-check"></i><b>17.3.5</b> Power for the interaction effect is less than that for simple effects</a></li>
<li class="chapter" data-level="17.3.6" data-path="factorial.html"><a href="factorial.html#planned-comparisons-vs.-post-hoc-tests"><i class="fa fa-check"></i><b>17.3.6</b> Planned comparisons vs. post-hoc tests</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="factorial.html"><a href="factorial.html#example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data"><i class="fa fa-check"></i><b>17.4</b> Example 2: Estimation of the effect of background condition on an effect (“it depends”) (Experiment 3e lesian area data)</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="factorial.html"><a href="factorial.html#understand-the-experimental-design"><i class="fa fa-check"></i><b>17.4.1</b> Understand the experimental design</a></li>
<li class="chapter" data-level="17.4.2" data-path="factorial.html"><a href="factorial.html#fit-the-linear-model-1"><i class="fa fa-check"></i><b>17.4.2</b> Fit the linear model</a></li>
<li class="chapter" data-level="17.4.3" data-path="factorial.html"><a href="factorial.html#check-the-model-1"><i class="fa fa-check"></i><b>17.4.3</b> Check the model</a></li>
<li class="chapter" data-level="17.4.4" data-path="factorial.html"><a href="factorial.html#inference-from-the-model-2"><i class="fa fa-check"></i><b>17.4.4</b> Inference from the model</a></li>
<li class="chapter" data-level="17.4.5" data-path="factorial.html"><a href="factorial.html#plot-the-model-3"><i class="fa fa-check"></i><b>17.4.5</b> Plot the model</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-2"><i class="fa fa-check"></i><b>17.5</b> Understanding the linear model with crossed factors 2</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="factorial.html"><a href="factorial.html#twoway-marginal-means"><i class="fa fa-check"></i><b>17.5.1</b> Conditional and marginal means</a></li>
<li class="chapter" data-level="17.5.2" data-path="factorial.html"><a href="factorial.html#simple-conditional-effects"><i class="fa fa-check"></i><b>17.5.2</b> Simple (conditional) effects</a></li>
<li class="chapter" data-level="17.5.3" data-path="factorial.html"><a href="factorial.html#twoway-marginal-effects"><i class="fa fa-check"></i><b>17.5.3</b> Marginal effects</a></li>
<li class="chapter" data-level="17.5.4" data-path="factorial.html"><a href="factorial.html#the-additive-model"><i class="fa fa-check"></i><b>17.5.4</b> The additive model</a></li>
<li class="chapter" data-level="17.5.5" data-path="factorial.html"><a href="factorial.html#twoway-reduce"><i class="fa fa-check"></i><b>17.5.5</b> Reduce models for the right reason</a></li>
<li class="chapter" data-level="17.5.6" data-path="factorial.html"><a href="factorial.html#the-marginal-means-of-an-additive-linear-model-with-two-factors-can-be-weird"><i class="fa fa-check"></i><b>17.5.6</b> The marginal means of an additive linear model with two factors can be weird</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="factorial.html"><a href="factorial.html#example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data"><i class="fa fa-check"></i><b>17.6</b> Example 3: Estimation of synergy (“More than the sum of the parts”) (Experiment 1c JA data)</a>
<ul>
<li class="chapter" data-level="17.6.1" data-path="factorial.html"><a href="factorial.html#examine-the-data"><i class="fa fa-check"></i><b>17.6.1</b> Examine the data</a></li>
<li class="chapter" data-level="17.6.2" data-path="factorial.html"><a href="factorial.html#fit-the-model-7"><i class="fa fa-check"></i><b>17.6.2</b> Fit the model</a></li>
<li class="chapter" data-level="17.6.3" data-path="factorial.html"><a href="factorial.html#model-check"><i class="fa fa-check"></i><b>17.6.3</b> Model check</a></li>
<li class="chapter" data-level="17.6.4" data-path="factorial.html"><a href="factorial.html#inference-from-the-model-3"><i class="fa fa-check"></i><b>17.6.4</b> Inference from the model</a></li>
<li class="chapter" data-level="17.6.5" data-path="factorial.html"><a href="factorial.html#plot-the-model-4"><i class="fa fa-check"></i><b>17.6.5</b> Plot the model</a></li>
<li class="chapter" data-level="17.6.6" data-path="factorial.html"><a href="factorial.html#alternative-plot"><i class="fa fa-check"></i><b>17.6.6</b> Alternative plot</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-3"><i class="fa fa-check"></i><b>17.7</b> Understanding the linear model with crossed factors 3</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="factorial.html"><a href="factorial.html#thinking-about-the-coefficients-of-the-linear-model"><i class="fa fa-check"></i><b>17.7.1</b> Thinking about the coefficients of the linear model</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="factorial.html"><a href="factorial.html#issues-in-inference"><i class="fa fa-check"></i><b>17.8</b> Issues in inference</a>
<ul>
<li class="chapter" data-level="17.8.1" data-path="factorial.html"><a href="factorial.html#for-pairwise-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model"><i class="fa fa-check"></i><b>17.8.1</b> For pairwise contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</a></li>
<li class="chapter" data-level="17.8.2" data-path="factorial.html"><a href="factorial.html#for-interaction-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model"><i class="fa fa-check"></i><b>17.8.2</b> For interaction contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</a></li>
<li class="chapter" data-level="17.8.3" data-path="factorial.html"><a href="factorial.html#twoway-multiple-tests"><i class="fa fa-check"></i><b>17.8.3</b> Adjusting <em>p</em>-values for multiple tests</a></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="factorial.html"><a href="factorial.html#two-way-anova"><i class="fa fa-check"></i><b>17.9</b> Two-way ANOVA</a>
<ul>
<li class="chapter" data-level="17.9.1" data-path="factorial.html"><a href="factorial.html#how-to-read-a-two-way-anova-table"><i class="fa fa-check"></i><b>17.9.1</b> How to read a two-way ANOVA table</a></li>
<li class="chapter" data-level="17.9.2" data-path="factorial.html"><a href="factorial.html#what-do-the-main-effects-in-an-anova-table-mean"><i class="fa fa-check"></i><b>17.9.2</b> What do the main effects in an ANOVA table mean?</a></li>
</ul></li>
<li class="chapter" data-level="17.10" data-path="factorial.html"><a href="factorial.html#more-issues-in-inference"><i class="fa fa-check"></i><b>17.10</b> More issues in inference</a>
<ul>
<li class="chapter" data-level="17.10.1" data-path="factorial.html"><a href="factorial.html#longitudinal-experiments-include-time-as-a-random-factor-better-than-repeated-measures-anova"><i class="fa fa-check"></i><b>17.10.1</b> Longitudinal experiments – include Time as a random factor (better than repeated measures ANOVA)</a></li>
</ul></li>
<li class="chapter" data-level="17.11" data-path="factorial.html"><a href="factorial.html#working-in-r-4"><i class="fa fa-check"></i><b>17.11</b> Working in R</a>
<ul>
<li class="chapter" data-level="17.11.1" data-path="factorial.html"><a href="factorial.html#model-formula"><i class="fa fa-check"></i><b>17.11.1</b> Model formula</a></li>
<li class="chapter" data-level="17.11.2" data-path="factorial.html"><a href="factorial.html#using-the-emmeans-function-2"><i class="fa fa-check"></i><b>17.11.2</b> Using the emmeans function</a></li>
<li class="chapter" data-level="17.11.3" data-path="factorial.html"><a href="factorial.html#contrasts"><i class="fa fa-check"></i><b>17.11.3</b> Contrasts</a></li>
<li class="chapter" data-level="17.11.4" data-path="factorial.html"><a href="factorial.html#practice-safe-anova"><i class="fa fa-check"></i><b>17.11.4</b> Practice safe ANOVA</a></li>
<li class="chapter" data-level="17.11.5" data-path="factorial.html"><a href="factorial.html#better-to-avoid-these"><i class="fa fa-check"></i><b>17.11.5</b> Better to avoid these</a></li>
</ul></li>
<li class="chapter" data-level="17.12" data-path="factorial.html"><a href="factorial.html#hidden-code-4"><i class="fa fa-check"></i><b>17.12</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="17.12.1" data-path="factorial.html"><a href="factorial.html#import-exp2j-example-1"><i class="fa fa-check"></i><b>17.12.1</b> Import exp2j (Example 1)</a></li>
<li class="chapter" data-level="17.12.2" data-path="factorial.html"><a href="factorial.html#import-exp3e-lesian-area-data-example-2"><i class="fa fa-check"></i><b>17.12.2</b> Import exp3e lesian area data (Example 2)</a></li>
<li class="chapter" data-level="17.12.3" data-path="factorial.html"><a href="factorial.html#import-exp1c-ja-data-example-3"><i class="fa fa-check"></i><b>17.12.3</b> Import Exp1c JA data (Example 3)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-vii-expanding-the-linear-model.html"><a href="part-vii-expanding-the-linear-model.html"><i class="fa fa-check"></i>Part VII – Expanding the Linear Model</a></li>
<li class="chapter" data-level="18" data-path="lmm.html"><a href="lmm.html"><i class="fa fa-check"></i><b>18</b> Models with random factors – linear mixed models</a>
<ul>
<li class="chapter" data-level="18.1" data-path="lmm.html"><a href="lmm.html#lmm-demo"><i class="fa fa-check"></i><b>18.1</b> Example 1 – A random intercepts and slopes explainer (demo1)</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="lmm.html"><a href="lmm.html#batched-measurements-result-in-clustered-residuals"><i class="fa fa-check"></i><b>18.1.1</b> Batched measurements result in clustered residuals</a></li>
<li class="chapter" data-level="18.1.2" data-path="lmm.html"><a href="lmm.html#lmm-corerr"><i class="fa fa-check"></i><b>18.1.2</b> Clustered residuals result in correlated error</a></li>
<li class="chapter" data-level="18.1.3" data-path="lmm.html"><a href="lmm.html#in-blocked-designs-clustered-residuals-adds-a-variance-component-that-masks-treatment-effects"><i class="fa fa-check"></i><b>18.1.3</b> In blocked designs, clustered residuals adds a variance component that masks treatment effects</a></li>
<li class="chapter" data-level="18.1.4" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-are-linear-models-with-added-random-factors"><i class="fa fa-check"></i><b>18.1.4</b> Linear mixed models are linear models with added random factors</a></li>
<li class="chapter" data-level="18.1.5" data-path="lmm.html"><a href="lmm.html#what-the-random-effects-are"><i class="fa fa-check"></i><b>18.1.5</b> What the random effects are</a></li>
<li class="chapter" data-level="18.1.6" data-path="lmm.html"><a href="lmm.html#in-a-blocked-design-a-linear-model-with-added-random-effects-increases-precision-of-treatment-effects"><i class="fa fa-check"></i><b>18.1.6</b> In a blocked design, a linear model with added random effects increases precision of treatment effects</a></li>
<li class="chapter" data-level="18.1.7" data-path="lmm.html"><a href="lmm.html#lmm-varcorr"><i class="fa fa-check"></i><b>18.1.7</b> The correlation among random intercepts and slopes</a></li>
<li class="chapter" data-level="18.1.8" data-path="lmm.html"><a href="lmm.html#clustered-residuals-create-heterogeneity-among-treatments"><i class="fa fa-check"></i><b>18.1.8</b> Clustered residuals create heterogeneity among treatments</a></li>
<li class="chapter" data-level="18.1.9" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-are-flexible"><i class="fa fa-check"></i><b>18.1.9</b> Linear mixed models are flexible</a></li>
<li class="chapter" data-level="18.1.10" data-path="lmm.html"><a href="lmm.html#a-random-intercept-only-model"><i class="fa fa-check"></i><b>18.1.10</b> A random intercept only model</a></li>
<li class="chapter" data-level="18.1.11" data-path="lmm.html"><a href="lmm.html#a-model-including-an-interaction-intercept"><i class="fa fa-check"></i><b>18.1.11</b> A model including an interaction intercept</a></li>
<li class="chapter" data-level="18.1.12" data-path="lmm.html"><a href="lmm.html#lmm-demo-aic"><i class="fa fa-check"></i><b>18.1.12</b> AIC and model selection – which model to report?</a></li>
<li class="chapter" data-level="18.1.13" data-path="lmm.html"><a href="lmm.html#the-specification-of-random-effects-matters"><i class="fa fa-check"></i><b>18.1.13</b> The specification of random effects matters</a></li>
<li class="chapter" data-level="18.1.14" data-path="lmm.html"><a href="lmm.html#mixed-effect-and-repeated-measures-anova"><i class="fa fa-check"></i><b>18.1.14</b> Mixed Effect and Repeated Measures ANOVA</a></li>
<li class="chapter" data-level="18.1.15" data-path="lmm.html"><a href="lmm.html#pseudoreplication"><i class="fa fa-check"></i><b>18.1.15</b> Pseudoreplication</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="lmm.html"><a href="lmm.html#lmm-example2"><i class="fa fa-check"></i><b>18.2</b> Example 2 – experiments without subsampling replication (exp6g)</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="lmm.html"><a href="lmm.html#understand-the-data"><i class="fa fa-check"></i><b>18.2.1</b> Understand the data</a></li>
<li class="chapter" data-level="18.2.2" data-path="lmm.html"><a href="lmm.html#model-fit-and-inference"><i class="fa fa-check"></i><b>18.2.2</b> Model fit and inference</a></li>
<li class="chapter" data-level="18.2.3" data-path="lmm.html"><a href="lmm.html#the-model-exp6g_m1-adds-a-random-intercept-but-not-a-random-slope"><i class="fa fa-check"></i><b>18.2.3</b> The model exp6g_m1 adds a random intercept but not a random slope</a></li>
<li class="chapter" data-level="18.2.4" data-path="lmm.html"><a href="lmm.html#the-fixed-effect-coefficients-of-model-exp6g_m1"><i class="fa fa-check"></i><b>18.2.4</b> The fixed effect coefficients of model exp6g_m1</a></li>
<li class="chapter" data-level="18.2.5" data-path="lmm.html"><a href="lmm.html#the-random-intercept-coefficients-of-exp6g_m1"><i class="fa fa-check"></i><b>18.2.5</b> The random intercept coefficients of exp6g_m1</a></li>
<li class="chapter" data-level="18.2.6" data-path="lmm.html"><a href="lmm.html#the-random-and-residual-variance-and-the-intraclass-correlation-of-model-exp6g_m1"><i class="fa fa-check"></i><b>18.2.6</b> The random and residual variance and the intraclass correlation of model exp6g_m1</a></li>
<li class="chapter" data-level="18.2.7" data-path="lmm.html"><a href="lmm.html#the-linear-mixed-model-exp6g_m1-increases-precision-of-treatment-effects-relative-to-a-fixed-effects-model"><i class="fa fa-check"></i><b>18.2.7</b> The linear mixed model exp6g_m1 increases precision of treatment effects, relative to a fixed effects model</a></li>
<li class="chapter" data-level="18.2.8" data-path="lmm.html"><a href="lmm.html#lmm-exp6g-alt"><i class="fa fa-check"></i><b>18.2.8</b> Alternative models for exp6g</a></li>
<li class="chapter" data-level="18.2.9" data-path="lmm.html"><a href="lmm.html#paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models"><i class="fa fa-check"></i><b>18.2.9</b> Paired t-tests and repeated measures ANOVA are special cases of linear mixed models</a></li>
<li class="chapter" data-level="18.2.10" data-path="lmm.html"><a href="lmm.html#lmm-exp6g-rmanova"><i class="fa fa-check"></i><b>18.2.10</b> Classical (“univariate model”) repeated measures ANOVA of exp6g</a></li>
<li class="chapter" data-level="18.2.11" data-path="lmm.html"><a href="lmm.html#lmm-exp6g-rmanova-multi"><i class="fa fa-check"></i><b>18.2.11</b> “Multivariate model” repeated measures ANOVA</a></li>
<li class="chapter" data-level="18.2.12" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-vs-repeated-measures-anova"><i class="fa fa-check"></i><b>18.2.12</b> Linear mixed models vs repeated measures ANOVA</a></li>
<li class="chapter" data-level="18.2.13" data-path="lmm.html"><a href="lmm.html#modeling-textttmouse_id-as-a-fixed-effect"><i class="fa fa-check"></i><b>18.2.13</b> Modeling <span class="math inline">\(\texttt{mouse_id}\)</span> as a fixed effect</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="lmm.html"><a href="lmm.html#example-3-factorial-experiments-and-no-subsampling-replicates-exp5c"><i class="fa fa-check"></i><b>18.3</b> Example 3 – Factorial experiments and no subsampling replicates (exp5c)</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="lmm.html"><a href="lmm.html#understand-the-data-1"><i class="fa fa-check"></i><b>18.3.1</b> Understand the data</a></li>
<li class="chapter" data-level="18.3.2" data-path="lmm.html"><a href="lmm.html#examine-the-data-1"><i class="fa fa-check"></i><b>18.3.2</b> Examine the data</a></li>
<li class="chapter" data-level="18.3.3" data-path="lmm.html"><a href="lmm.html#lmm-exp5c-m1"><i class="fa fa-check"></i><b>18.3.3</b> Model fit and inference</a></li>
<li class="chapter" data-level="18.3.4" data-path="lmm.html"><a href="lmm.html#why-we-care-about-modeling-batch-in-exp5c"><i class="fa fa-check"></i><b>18.3.4</b> Why we care about modeling batch in exp5c</a></li>
<li class="chapter" data-level="18.3.5" data-path="lmm.html"><a href="lmm.html#the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts"><i class="fa fa-check"></i><b>18.3.5</b> The linear mixed model exp5c_m1 adds two random intercepts</a></li>
<li class="chapter" data-level="18.3.6" data-path="lmm.html"><a href="lmm.html#the-fixed-effect-coefficients-of-model-exp5c_m1"><i class="fa fa-check"></i><b>18.3.6</b> The fixed effect coefficients of model exp5c_m1</a></li>
<li class="chapter" data-level="18.3.7" data-path="lmm.html"><a href="lmm.html#the-random-effect-coefficients-of-model-exp5c_m1"><i class="fa fa-check"></i><b>18.3.7</b> The random effect coefficients of model exp5c_m1</a></li>
<li class="chapter" data-level="18.3.8" data-path="lmm.html"><a href="lmm.html#lmm-exp5c-alt"><i class="fa fa-check"></i><b>18.3.8</b> Alternative models for exp5c</a></li>
<li class="chapter" data-level="18.3.9" data-path="lmm.html"><a href="lmm.html#lmm-rmanova-5c"><i class="fa fa-check"></i><b>18.3.9</b> Classical (“univariate model”) repeated measures ANOVA</a></li>
<li class="chapter" data-level="18.3.10" data-path="lmm.html"><a href="lmm.html#lmm-rmanova-5c-multi"><i class="fa fa-check"></i><b>18.3.10</b> “Multivariate model” repeated measures ANOVA of exp5c</a></li>
<li class="chapter" data-level="18.3.11" data-path="lmm.html"><a href="lmm.html#modeling-textttdonor-as-a-fixed-effect"><i class="fa fa-check"></i><b>18.3.11</b> Modeling <span class="math inline">\(\texttt{donor}\)</span> as a fixed effect</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="lmm.html"><a href="lmm.html#lmm-example4"><i class="fa fa-check"></i><b>18.4</b> Example 4 – Experiments with subsampling replication (exp1g)</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="lmm.html"><a href="lmm.html#understand-the-data-2"><i class="fa fa-check"></i><b>18.4.1</b> Understand the data</a></li>
<li class="chapter" data-level="18.4.2" data-path="lmm.html"><a href="lmm.html#examine-the-data-2"><i class="fa fa-check"></i><b>18.4.2</b> Examine the data</a></li>
<li class="chapter" data-level="18.4.3" data-path="lmm.html"><a href="lmm.html#fit-the-model-9"><i class="fa fa-check"></i><b>18.4.3</b> Fit the model</a></li>
<li class="chapter" data-level="18.4.4" data-path="lmm.html"><a href="lmm.html#inference-from-the-model-6"><i class="fa fa-check"></i><b>18.4.4</b> Inference from the model</a></li>
<li class="chapter" data-level="18.4.5" data-path="lmm.html"><a href="lmm.html#plot-the-model-5"><i class="fa fa-check"></i><b>18.4.5</b> Plot the model</a></li>
<li class="chapter" data-level="18.4.6" data-path="lmm.html"><a href="lmm.html#alternaplot-the-model-2"><i class="fa fa-check"></i><b>18.4.6</b> Alternaplot the model</a></li>
<li class="chapter" data-level="18.4.7" data-path="lmm.html"><a href="lmm.html#understanding-the-alternative-models"><i class="fa fa-check"></i><b>18.4.7</b> Understanding the alternative models</a></li>
<li class="chapter" data-level="18.4.8" data-path="lmm.html"><a href="lmm.html#the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b"><i class="fa fa-check"></i><b>18.4.8</b> The VarCorr matrix of models exp1g_m1a and exp1g_m1b</a></li>
<li class="chapter" data-level="18.4.9" data-path="lmm.html"><a href="lmm.html#a"><i class="fa fa-check"></i><b>18.4.9</b> A</a></li>
<li class="chapter" data-level="18.4.10" data-path="lmm.html"><a href="lmm.html#fixed-effect-models-and-pseudoreplication"><i class="fa fa-check"></i><b>18.4.10</b> Fixed effect models and pseudoreplication</a></li>
<li class="chapter" data-level="18.4.11" data-path="lmm.html"><a href="lmm.html#mixed-effect-anova"><i class="fa fa-check"></i><b>18.4.11</b> Mixed effect ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="lmm.html"><a href="lmm.html#working-in-r-5"><i class="fa fa-check"></i><b>18.5</b> Working in R</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="lmm.html"><a href="lmm.html#fitting-linear-mixed-models-with-lme4"><i class="fa fa-check"></i><b>18.5.1</b> Fitting linear mixed models with lme4</a></li>
<li class="chapter" data-level="18.5.2" data-path="lmm.html"><a href="lmm.html#repeated-measures-anova-randomized-complete-block-with-no-subsampling"><i class="fa fa-check"></i><b>18.5.2</b> Repeated measures ANOVA (randomized complete block with no subsampling)</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="lmm.html"><a href="lmm.html#hidden-code-5"><i class="fa fa-check"></i><b>18.6</b> Hidden code</a>
<ul>
<li class="chapter" data-level="18.6.1" data-path="lmm.html"><a href="lmm.html#import-exp5c"><i class="fa fa-check"></i><b>18.6.1</b> Import exp5c</a></li>
<li class="chapter" data-level="18.6.2" data-path="lmm.html"><a href="lmm.html#import-exp1g"><i class="fa fa-check"></i><b>18.6.2</b> Import exp1g</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="pre-post.html"><a href="pre-post.html"><i class="fa fa-check"></i><b>19</b> Linear models for longitudinal experiments – I. pre-post designs</a>
<ul>
<li class="chapter" data-level="19.1" data-path="pre-post.html"><a href="pre-post.html#best-practice-models"><i class="fa fa-check"></i><b>19.1</b> Best practice models</a></li>
<li class="chapter" data-level="19.2" data-path="pre-post.html"><a href="pre-post.html#common-alternatives-that-are-not-recommended"><i class="fa fa-check"></i><b>19.2</b> Common alternatives that are not recommended</a></li>
<li class="chapter" data-level="19.3" data-path="pre-post.html"><a href="pre-post.html#advanced-models"><i class="fa fa-check"></i><b>19.3</b> Advanced models</a></li>
<li class="chapter" data-level="19.4" data-path="pre-post.html"><a href="pre-post.html#understanding-the-alternative-models-1"><i class="fa fa-check"></i><b>19.4</b> Understanding the alternative models</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="pre-post.html"><a href="pre-post.html#m1-linear-model-with-the-baseline-measure-as-the-covariate-ancova-model"><i class="fa fa-check"></i><b>19.4.1</b> (M1) Linear model with the baseline measure as the covariate (ANCOVA model)</a></li>
<li class="chapter" data-level="19.4.2" data-path="pre-post.html"><a href="pre-post.html#m2-linear-model-of-the-change-score-change-score-model"><i class="fa fa-check"></i><b>19.4.2</b> (M2) Linear model of the change score (change-score model)</a></li>
<li class="chapter" data-level="19.4.3" data-path="pre-post.html"><a href="pre-post.html#m3-linear-model-of-post-baseline-values-without-the-baseline-as-a-covariate-post-model"><i class="fa fa-check"></i><b>19.4.3</b> (M3) Linear model of post-baseline values without the baseline as a covariate (post model)</a></li>
<li class="chapter" data-level="19.4.4" data-path="pre-post.html"><a href="pre-post.html#m4-linear-model-with-factorial-fixed-effects-fixed-effects-model"><i class="fa fa-check"></i><b>19.4.4</b> (M4) Linear model with factorial fixed effects (fixed-effects model)</a></li>
<li class="chapter" data-level="19.4.5" data-path="pre-post.html"><a href="pre-post.html#m5-repeated-measures-anova"><i class="fa fa-check"></i><b>19.4.5</b> (M5) Repeated measures ANOVA</a></li>
<li class="chapter" data-level="19.4.6" data-path="pre-post.html"><a href="pre-post.html#m6-linear-mixed-model"><i class="fa fa-check"></i><b>19.4.6</b> (M6) Linear mixed model</a></li>
<li class="chapter" data-level="19.4.7" data-path="pre-post.html"><a href="pre-post.html#m7-linear-model-with-correlated-error"><i class="fa fa-check"></i><b>19.4.7</b> (M7) Linear model with correlated error</a></li>
<li class="chapter" data-level="19.4.8" data-path="pre-post.html"><a href="pre-post.html#m8-constrained-fixed-effects-model-with-correlated-error-clda-model"><i class="fa fa-check"></i><b>19.4.8</b> (M8) Constrained fixed effects model with correlated error (cLDA model)</a></li>
<li class="chapter" data-level="19.4.9" data-path="pre-post.html"><a href="pre-post.html#comparison-table"><i class="fa fa-check"></i><b>19.4.9</b> Comparison table</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="pre-post.html"><a href="pre-post.html#example-1-a-single-post-baseline-measure-pre-post-design"><i class="fa fa-check"></i><b>19.5</b> Example 1 – a single post-baseline measure (pre-post design)</a></li>
<li class="chapter" data-level="19.6" data-path="pre-post.html"><a href="pre-post.html#working-in-r-6"><i class="fa fa-check"></i><b>19.6</b> Working in R</a></li>
<li class="chapter" data-level="19.7" data-path="pre-post.html"><a href="pre-post.html#hidden-code-6"><i class="fa fa-check"></i><b>19.7</b> Hidden code</a>
<ul>
<li class="chapter" data-level="19.7.1" data-path="pre-post.html"><a href="pre-post.html#import-and-wrangle-mouse-sociability-data"><i class="fa fa-check"></i><b>19.7.1</b> Import and wrangle mouse sociability data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="counts.html"><a href="counts.html"><i class="fa fa-check"></i><b>20</b> Linear models for count data – Generalized Linear Models I</a>
<ul>
<li class="chapter" data-level="20.1" data-path="counts.html"><a href="counts.html#the-generalized-linear-model"><i class="fa fa-check"></i><b>20.1</b> The generalized linear model</a></li>
<li class="chapter" data-level="20.2" data-path="counts.html"><a href="counts.html#count-data-example-number-of-trematode-worm-larvae-in-eyes-of-threespine-stickleback-fish"><i class="fa fa-check"></i><b>20.2</b> Count data example – number of trematode worm larvae in eyes of threespine stickleback fish</a>
<ul>
<li class="chapter" data-level="20.2.1" data-path="counts.html"><a href="counts.html#modeling-strategy-1"><i class="fa fa-check"></i><b>20.2.1</b> Modeling strategy</a></li>
<li class="chapter" data-level="20.2.2" data-path="counts.html"><a href="counts.html#checking-the-model-i-a-normal-q-q-plot"><i class="fa fa-check"></i><b>20.2.2</b> Checking the model I – a Normal Q-Q plot</a></li>
<li class="chapter" data-level="20.2.3" data-path="counts.html"><a href="counts.html#checking-the-model-ii-scale-location-plot-for-checking-homoskedasticity"><i class="fa fa-check"></i><b>20.2.3</b> Checking the model II – scale-location plot for checking homoskedasticity</a></li>
<li class="chapter" data-level="20.2.4" data-path="counts.html"><a href="counts.html#two-distributions-for-count-data-poisson-and-negative-binomial"><i class="fa fa-check"></i><b>20.2.4</b> Two distributions for count data – Poisson and Negative Binomial</a></li>
<li class="chapter" data-level="20.2.5" data-path="counts.html"><a href="counts.html#fitting-a-glm-with-a-poisson-distribution-to-the-worm-data"><i class="fa fa-check"></i><b>20.2.5</b> Fitting a GLM with a Poisson distribution to the worm data</a></li>
<li class="chapter" data-level="20.2.6" data-path="counts.html"><a href="counts.html#model-checking-fits-to-count-data"><i class="fa fa-check"></i><b>20.2.6</b> Model checking fits to count data</a></li>
<li class="chapter" data-level="20.2.7" data-path="counts.html"><a href="counts.html#fitting-a-glm-with-a-negative-binomial-distribution-to-the-worm-data"><i class="fa fa-check"></i><b>20.2.7</b> Fitting a GLM with a Negative Binomial distribution to the worm data</a></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="counts.html"><a href="counts.html#working-in-r-7"><i class="fa fa-check"></i><b>20.3</b> Working in R</a>
<ul>
<li class="chapter" data-level="20.3.1" data-path="counts.html"><a href="counts.html#fitting-a-glm-to-count-data"><i class="fa fa-check"></i><b>20.3.1</b> Fitting a GLM to count data</a></li>
<li class="chapter" data-level="20.3.2" data-path="counts.html"><a href="counts.html#fitting-a-generalized-linear-mixed-model-glmm-to-count-data"><i class="fa fa-check"></i><b>20.3.2</b> Fitting a generalized linear mixed model (GLMM) to count data</a></li>
<li class="chapter" data-level="20.3.3" data-path="counts.html"><a href="counts.html#fitting-a-generalized-linear-model-to-continouus-data"><i class="fa fa-check"></i><b>20.3.3</b> Fitting a generalized linear model to continouus data</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="counts.html"><a href="counts.html#problems-1"><i class="fa fa-check"></i><b>20.4</b> Problems</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="gls.html"><a href="gls.html"><i class="fa fa-check"></i><b>21</b> Linear models with heterogenous variance</a></li>
<li class="chapter" data-level="22" data-path="count-sim.html"><a href="count-sim.html"><i class="fa fa-check"></i><b>22</b> Simulations – Count data (alternatives to a t-test)</a>
<ul>
<li class="chapter" data-level="22.1" data-path="count-sim.html"><a href="count-sim.html#use-data-similar-to-figure-6f-from-example-1"><i class="fa fa-check"></i><b>22.1</b> Use data similar to Figure 6f from Example 1</a></li>
<li class="chapter" data-level="22.2" data-path="count-sim.html"><a href="count-sim.html#functions"><i class="fa fa-check"></i><b>22.2</b> Functions</a></li>
<li class="chapter" data-level="22.3" data-path="count-sim.html"><a href="count-sim.html#simulations"><i class="fa fa-check"></i><b>22.3</b> Simulations</a>
<ul>
<li class="chapter" data-level="22.3.1" data-path="count-sim.html"><a href="count-sim.html#type-i-pseudo-normal-distribution"><i class="fa fa-check"></i><b>22.3.1</b> Type I, Pseudo-Normal distribution</a></li>
<li class="chapter" data-level="22.3.2" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-equal-n"><i class="fa fa-check"></i><b>22.3.2</b> Type I, neg binom, equal n</a></li>
<li class="chapter" data-level="22.3.3" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-equal-n-small-theta"><i class="fa fa-check"></i><b>22.3.3</b> Type I, neg binom, equal n, small theta</a></li>
<li class="chapter" data-level="22.3.4" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-unequal-n"><i class="fa fa-check"></i><b>22.3.4</b> Type I, neg binom, unequal n</a></li>
<li class="chapter" data-level="22.3.5" data-path="count-sim.html"><a href="count-sim.html#power-pseudo-normal-distribution-equal-n"><i class="fa fa-check"></i><b>22.3.5</b> Power, Pseudo-Normal distribution, equal n</a></li>
<li class="chapter" data-level="22.3.6" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-equal-n"><i class="fa fa-check"></i><b>22.3.6</b> Power, neg binom, equal n</a></li>
<li class="chapter" data-level="22.3.7" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-small-theta"><i class="fa fa-check"></i><b>22.3.7</b> Power, neg binom, small theta</a></li>
<li class="chapter" data-level="22.3.8" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-unequal-n"><i class="fa fa-check"></i><b>22.3.8</b> Power, neg binom, unequal n</a></li>
<li class="chapter" data-level="22.3.9" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-unequal-n-unequal-theta"><i class="fa fa-check"></i><b>22.3.9</b> Power, neg binom, unequal n, unequal theta</a></li>
<li class="chapter" data-level="22.3.10" data-path="count-sim.html"><a href="count-sim.html#type-1-neg-binom-equal-n-unequal-theta"><i class="fa fa-check"></i><b>22.3.10</b> Type 1, neg binom, equal n, unequal theta</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="count-sim.html"><a href="count-sim.html#save-it-read-it"><i class="fa fa-check"></i><b>22.4</b> Save it, Read it</a></li>
<li class="chapter" data-level="22.5" data-path="count-sim.html"><a href="count-sim.html#analysis"><i class="fa fa-check"></i><b>22.5</b> Analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html"><i class="fa fa-check"></i>Appendix 1: Getting Started with R</a>
<ul>
<li class="chapter" data-level="22.6" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#get-your-computer-ready"><i class="fa fa-check"></i><b>22.6</b> Get your computer ready</a>
<ul>
<li class="chapter" data-level="22.6.1" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#start-here"><i class="fa fa-check"></i><b>22.6.1</b> Start here</a></li>
<li class="chapter" data-level="22.6.2" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r"><i class="fa fa-check"></i><b>22.6.2</b> Install R</a></li>
<li class="chapter" data-level="22.6.3" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r-studio"><i class="fa fa-check"></i><b>22.6.3</b> Install R Studio</a></li>
<li class="chapter" data-level="22.6.4" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r-markdown-1"><i class="fa fa-check"></i><b>22.6.4</b> Install R Markdown</a></li>
<li class="chapter" data-level="22.6.5" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#optional-alternative-latex-installations"><i class="fa fa-check"></i><b>22.6.5</b> (optional) Alternative LaTeX installations</a></li>
</ul></li>
<li class="chapter" data-level="22.7" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#start-learning-r-studio"><i class="fa fa-check"></i><b>22.7</b> Start learning R Studio</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix-2-online-resources-for-getting-started-with-statistical-modeling-in-r.html"><a href="appendix-2-online-resources-for-getting-started-with-statistical-modeling-in-r.html"><i class="fa fa-check"></i>Appendix 2: Online Resources for Getting Started with Statistical Modeling in R</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Elements of Statistical Modeling for Experimental Biology</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="factorial" class="section level1" number="17">
<h1><span class="header-section-number">Chapter 17</span> Linear models with two categorical <span class="math inline">\(X\)</span> – Factorial linear models (“two-way ANOVA”)</h1>
<div class="sourceCode" id="cb469"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb469-1"><a href="factorial.html#cb469-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># X &lt;- model.matrix(m1)</span></span>
<span id="cb469-2"><a href="factorial.html#cb469-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># y_col &lt;- insight::find_response(m1)</span></span>
<span id="cb469-3"><a href="factorial.html#cb469-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># y &lt;- model.frame(m1)[, y_col]</span></span>
<span id="cb469-4"><a href="factorial.html#cb469-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># b &lt;- (solve(t(X)%*%X)%*%t(X)%*%y)[,1]</span></span>
<span id="cb469-5"><a href="factorial.html#cb469-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb469-6"><a href="factorial.html#cb469-6" aria-hidden="true" tabindex="-1"></a>  emm_b <span class="ot">&lt;-</span> <span class="cf">function</span>(m1_emm){</span>
<span id="cb469-7"><a href="factorial.html#cb469-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># works only for 2 x 2 and specs have to be in</span></span>
<span id="cb469-8"><a href="factorial.html#cb469-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># same order as model</span></span>
<span id="cb469-9"><a href="factorial.html#cb469-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.data.frame</span>(m1_emm)){</span>
<span id="cb469-10"><a href="factorial.html#cb469-10" aria-hidden="true" tabindex="-1"></a>      m1_emm <span class="ot">&lt;-</span> <span class="fu">summary</span>(m1_emm)</span>
<span id="cb469-11"><a href="factorial.html#cb469-11" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb469-12"><a href="factorial.html#cb469-12" aria-hidden="true" tabindex="-1"></a>    mu <span class="ot">&lt;-</span> m1_emm[, <span class="st">&quot;emmean&quot;</span>]</span>
<span id="cb469-13"><a href="factorial.html#cb469-13" aria-hidden="true" tabindex="-1"></a>    b <span class="ot">&lt;-</span> <span class="fu">c</span>(mu[<span class="dv">1</span>],</span>
<span id="cb469-14"><a href="factorial.html#cb469-14" aria-hidden="true" tabindex="-1"></a>            mu[<span class="dv">2</span>] <span class="sc">-</span> mu[<span class="dv">1</span>],</span>
<span id="cb469-15"><a href="factorial.html#cb469-15" aria-hidden="true" tabindex="-1"></a>            mu[<span class="dv">3</span>] <span class="sc">-</span> mu[<span class="dv">1</span>],</span>
<span id="cb469-16"><a href="factorial.html#cb469-16" aria-hidden="true" tabindex="-1"></a>            mu[<span class="dv">4</span>])</span>
<span id="cb469-17"><a href="factorial.html#cb469-17" aria-hidden="true" tabindex="-1"></a>    b[<span class="dv">4</span>] <span class="ot">&lt;-</span> b[<span class="dv">4</span>] <span class="sc">-</span> (b[<span class="dv">1</span>]<span class="sc">+</span>b[<span class="dv">2</span>]<span class="sc">+</span>b[<span class="dv">3</span>])</span>
<span id="cb469-18"><a href="factorial.html#cb469-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(b)</span>
<span id="cb469-19"><a href="factorial.html#cb469-19" aria-hidden="true" tabindex="-1"></a>  }</span></code></pre></div>
<div id="a-linear-model-with-crossed-factors-estimates-interaction-effects" class="section level2" number="17.1">
<h2><span class="header-section-number">17.1</span> A linear model with crossed factors estimates interaction effects</h2>
<p>A factorial experiment is one in which there are two or more factor variables (categorical <span class="math inline">\(X\)</span>) that are <strong>crossed</strong>, resulting in a group for each combination of the levels of each factor. Each specific combination is a different treatment. A <strong>linear model with crossed factors</strong> is used to estimate <strong>interaction effects</strong>, which occur when the effect of the level of one factor is conditional on the level of the other factors. Estimation of the interaction effect is necessary for inferences about</p>
<ol style="list-style-type: decimal">
<li>“Something different” – Estimation of a treatment effect relative to a control effect (Example 1 – TLR9-/- mice)</li>
<li>“It depends” – Estimation of the effect of background condition on an effect (Example 2 – XX mice)</li>
<li>“More than the sum of the parts” – Estimation of <strong>synergy</strong>, a non-additive effect (Example 3 – plant root growth)</li>
</ol>
<p>Inferences like these are common in the experimental biology literature but they are made using the wrong statistics. The correct statistic – the interaction effect – is easy to compute but rarely computed.</p>
<div style="background-color:#cccccc; text-align:left; vertical-align: middle; padding:20px 47px;">
<p>Alert. NHST encourages thinking such as “there is or isn’t an interaction effect” This dichotomous thinking can only lead to disasters of biblical proportion, dogs and cats living together. Interaction effects in biology are ubiquitous. Sometimes they are small enough to ignore.</p>
</div>
<div id="an-interaction-is-a-difference-in-simple-effects" class="section level3" number="17.1.1">
<h3><span class="header-section-number">17.1.1</span> An interaction is a difference in simple effects</h3>
<p>In this chapter, I’ll describe an interaction effect using different descriptions (Examples 1-3). Here, I want to emphasize that an interaction effect is a difference of differences. To clarify this, I’ll introduce a fake experiment. A research group has evidence that a certain gene product (CGP) is an intermediary between intestinal microbiota and obesity. The researchers transfer feces from either lean mice or obese mice into either wildtype mice or CGP-/- mice to investigate the effect of microbiota and CGP on body weight. The design has two factors (donor and genotype), each with two levels (donor: “Lean,” “Obese; genotype:”WT“,”KO"), which makes this a <span class="math inline">\(2 \times 2\)</span> (crossed or factorial) design. There are <span class="math inline">\(n=6\)</span> replicates for each treatment.</p>
<p>A good way to visualize the treatment combinations in a crossed design is with a <span class="math inline">\(m \times p\)</span> table showing all combinations of the <span class="math inline">\(m\)</span> levels of factor 1 (<span class="math inline">\(\texttt{donor}\)</span>) against the <span class="math inline">\(p\)</span> levels of factor 2 (<span class="math inline">\(\texttt{genotype}\)</span>) (<a href="factorial.html#tab:twoway-factorial-table">17.1</a>).</p>
<table class=" lightable-paper lightable-hover" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:twoway-factorial-table">Table 17.1: </span>Mean body weight of mice in four treatment groups of the 2 x 2 factorial experiment.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
WT
</th>
<th style="text-align:right;">
KO
</th>
<th style="text-align:right;">
Effect
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
Lean
</td>
<td style="text-align:right;">
33.2
</td>
<td style="text-align:right;">
34.1
</td>
<td style="text-align:right;font-weight: bold;color: #696969 !important;">
0.9
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Obese
</td>
<td style="text-align:right;">
41.9
</td>
<td style="text-align:right;">
32.7
</td>
<td style="text-align:right;font-weight: bold;color: #696969 !important;">
-9.2
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;font-weight: bold;">
Effect
</td>
<td style="text-align:right;font-weight: bold;">
8.7
</td>
<td style="text-align:right;font-weight: bold;">
-1.4
</td>
<td style="text-align:right;font-weight: bold;color: #FF0000 !important;font-weight: bold;">
-10.1
</td>
</tr>
</tbody>
</table>
<ol style="list-style-type: decimal">
<li>The upper-left <span class="math inline">\(2 \times 2\)</span> part of the table contains the fake mean body weight of each treatment group. These means are known as <strong>cell means</strong>.</li>
<li>The first two elements in the “Effect” column contains the difference of the two cells to the left – these are the <strong>effects of genotype conditional on the level of donor</strong>.</li>
<li>The first two elements in the “Effect” row contains the difference of the two cells above – these are the <strong>effects of donor conditional on the level of genotype</strong>.</li>
<li>The effects described in items 2 and 3 are known as the <strong>simple effects</strong>.</li>
<li>The value in red is the difference of the two simple effects above it. It is also the difference of the two simple effects to the left. These differences are equal. This is the interaction effect.</li>
<li>In an experiment with a single factor with four levels, all six pairwise comparisons may be of interest. In a <span class="math inline">\(2 \times 2\)</span> factorial experiment, it is the four simple effects and the interaction effect that should pique the interest of the researcher.</li>
</ol>
<p>In this fake experiment, we want to know the effect of obese donor treatment in the KO mice <em>compared to</em> the effect of obese donor treatment in the WT mice. That is, we want the contrast of these two simple effects.</p>
<p><span class="math display">\[
(\operatorname{Obese/KO} - \operatorname{Lean/KO}) - (\operatorname{Obese/WT} - \operatorname{Lean/WT})
\]</span></p>
<p>This contrast is the interaction effect. <strong>An interaction effect is a difference of differences</strong>.</p>
</div>
<div id="a-linear-model-with-crossed-factors-includes-interaction-effects" class="section level3" number="17.1.2">
<h3><span class="header-section-number">17.1.2</span> A linear model with crossed factors includes interaction effects</h3>
<p>The factorial linear model for the fake data is</p>
<p><span class="math display">\[
\begin{align}
\texttt{body_weight} &amp;= \beta_0 + \beta_1 (\texttt{donor}_{\texttt{Obese}}) + \beta_2 (\texttt{genotype}_{\texttt{KO}})\ + \\
&amp;\quad \ \beta_3 (\texttt{donor}_{\texttt{Obese}} : \texttt{genotype}_{\texttt{KO}}) + \varepsilon
\end{align}
\]</span>
What are the variables?</p>
<ol style="list-style-type: decimal">
<li><span class="math inline">\(\texttt{donor}_{\texttt{Obese}}\)</span> is the indicator variable for the “Obese” level of the factor <span class="math inline">\(\texttt{donor}\)</span>. It contains the value 1 if the donor was “Obese” and 0 otherwise.</li>
<li><span class="math inline">\(\texttt{genotype}_{\texttt{KO}}\)</span> is the indicator variable for the “KO” level of the factor <span class="math inline">\(\texttt{genotype}\)</span>. It contains the value 1 if the genotype is “KO” and 0 otherwise.</li>
<li><span class="math inline">\(\texttt{donor}_{\texttt{Obese}} : \texttt{genotype}_{\texttt{KO}}\)</span> is the indicator variable for the interaction between the “Obese” level of <span class="math inline">\(\texttt{donor}\)</span> and the “KO” level of <span class="math inline">\(\texttt{genotype}\)</span>. This “:” character to indicate interaction follows the R formula convention in the LME4 package. Many sources use a <span class="math inline">\(\times\)</span> symbol instead of the colon. The variable contains the value 1 if the mouse is assigned to both “Obese” and “KO” and 0 otherwise. This value is the product of the values in <span class="math inline">\(\texttt{donor}_{\texttt{Obese}}\)</span> and <span class="math inline">\(\texttt{genotype}_{\texttt{KO}}\)</span>.</li>
</ol>
<p>What are the parameters?</p>
<p>This linear model has set “Lean” as the reference level of <span class="math inline">\(\texttt{donor}\)</span> and “WT” as the reference level of <span class="math inline">\(\texttt{genotype}\)</span>. This make the Lean/WT mice the control.</p>
<ol style="list-style-type: decimal">
<li><span class="math inline">\(\beta_0\)</span> is the true mean of the control, which is the “Lean/WT” group.</li>
<li><span class="math inline">\(\beta_1\)</span> is the true effect of donor in the WT mice (the effect of manipulating the donor factor but not the genotype factor). It is the difference between the true means of the “Obese/WT” group and the “Lean/WT” group.</li>
</ol>
<p>The mean of the “Obese/WT” group is <span class="math inline">\(\beta_0 + \beta_1\)</span>. This is the expectation if we start with the control and then add the effect of Obese donor.</p>
<ol start="3" style="list-style-type: decimal">
<li><span class="math inline">\(\beta_2\)</span> is the true effect of genotype in the mice given feces from lean donors (the effect of manipulating the genotype factor but not the donor factor). It is the difference between the true means of the “Lean/KO” group and the “Lean/WT” group.</li>
</ol>
<p>The mean of the “Lean/KO” group is <span class="math inline">\(\beta_0 + \beta_2\)</span>. This is the expectation if we start with the control and then add the effect of KO genotype.</p>
<ol start="4" style="list-style-type: decimal">
<li><span class="math inline">\(\beta_3\)</span> is the true interaction effect of <span class="math inline">\(\texttt{donor}_{\texttt{Obese}} : \texttt{genotype}_{\texttt{KO}}\)</span></li>
</ol>
<p>The mean of the Obese/KO group is <span class="math inline">\(\beta_0 + \beta_1 + \beta_2 + \beta_3\)</span>. The expected mean of the Obese/KO group, <em>if the factors are additive, which means the interaction effect is zero</em>, is <span class="math inline">\(\beta_0 + \beta_1 + \beta_2\)</span>. <strong>The interaction effect is the difference between the actual mean of the Obese/KO group and this additive mean</strong>. The interaction effect is what’s left-over, after you’ve added Obese effect and the KO effect to the control.</p>
</div>
<div id="factorial-experiments-are-frequently-analyzed-as-flattened-linear-models-in-the-experimental-biology-literature" class="section level3" number="17.1.3">
<h3><span class="header-section-number">17.1.3</span> factorial experiments are frequently analyzed as flattened linear models in the experimental biology literature</h3>
<p>Often, researchers analyze data from a factorial experiment with a one-way ANOVA followed by pairwise tests (or by a simple series of separate <em>t</em>-tests). For the fake experiment, this linear model is</p>
<p><span class="math display">\[
\begin{align}
\texttt{body_weight} &amp;= \beta_0 + \beta_1 (\texttt{treatment}_{\texttt{Obese}}) + \beta_2 (\texttt{treatment}_{\texttt{KO}}) \ + \\
&amp; \quad \ \beta_3 (\texttt{treatment}_{\texttt{Obese + KO}}) + \varepsilon
\end{align}
\]</span></p>
<p>I refer to this as a <strong>flattened</strong> model (the table of treatment combinations has been flattened into a single row). Inference from a factorial or flattened model is the same. We can get the same pairwise contrasts or interaction contrasts from either model. That said, a factorial model nudges researchers to think about the analysis as a factorial design, which is good, because interaction effects are an explicit component of a factorial model.</p>
</div>
</div>
<div id="example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data" class="section level2" number="17.2">
<h2><span class="header-section-number">17.2</span> Example 1 – Estimation of a treatment effect relative to a control effect (“Something different”) (Experiment 2j glucose uptake data)</h2>
<p>To introduce a linear model with crossed factors (categorical <span class="math inline">\(X\)</span> variables), I’ll use data from a set of experiments designed to measure the effect of the toll-like receptor protein TLR9 on the activation of excercise-induced AMP-activated protein kinase (AMPK) and downstream sequelae of this activation, including glucose transport (from outside to inside the cell) by skeletal muscle cells.</p>
<p>Article source: <a href="https://www.nature.com/articles/s41586-020-1992-7" target="_blank">TLR9 and beclin 1 crosstalk regulates muscle AMPK activation in exercise</a></p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7047589/" target="_blank">Public source</a></p>
<p>The data are from multiple experiments in Figure 2.</p>
<p><a href="https://www.nature.com/articles/s41586-020-1992-7#Sec27" target="_blank">Data source</a></p>
<div id="twoway-understand1" class="section level3" number="17.2.1">
<h3><span class="header-section-number">17.2.1</span> Understand the experimental design</h3>
<p><strong>Background</strong>. Exercise (muscle activity) stimulates AMPK activated glucose uptake (transport from outside to inside the muscle cell). Instead of natural exercise, which can stimulate multiple systems, the researchers directly activate the muscle with electrical stimulation.</p>
<p><strong>Research question</strong> There are two ways to think about the question – both produce the same answer.</p>
<ol style="list-style-type: decimal">
<li>How much does TLR9 knockout inhibit the expected increase of glucose uptake following electrical stimulation? The expected increase comes from the contrast of the positive and negative controls. Call this the knockout-induced stimulation effect.</li>
<li>How much does TLR9 knockout inhibit glucose uptake during muscle stimulation compared to effect of TLR9 knockuot during muscle rest? Call this the <strong>stimulation-induced TLR9-/- effect</strong>.</li>
</ol>
<p><strong>Response variable</strong> <span class="math inline">\(\texttt{glucose_uptake}\)</span> (nmol per mg protein per 15 min) – the rate of glucose transported into the cell.</p>
<p><strong>Factor 1</strong> – <span class="math inline">\(\texttt{genotype}\)</span> (“WT,” “KO”).</p>
<ul>
<li>“WT” (reference level) – C57BL/6J mice with intact TLR gene (TLR+/+)</li>
<li>“KO” – TLR-/- mice on a C57BL/6J background</li>
</ul>
<p><strong>Factor 2</strong> <span class="math inline">\(\texttt{stimulation}\)</span> (“Rest,” “Active”) – Two levels:</p>
<ul>
<li>“Rest” (reference level) – muscle that has not been stimulated.</li>
<li>“Active” – electrical stimulation of muscle to induce contraction and contractile-related cell changes</li>
</ul>
<p>These two factors create the three control treatments and the one focal treatment:</p>
<ul>
<li><p>WT/Rest – <strong>Negative control</strong>. Expect non-exercise (low) level uptake.</p></li>
<li><p>WT/Active – <strong>Positive control</strong>. Expect high uptake.</p></li>
<li><p>KO/Rest – <strong>Method control</strong>. Unsure of KO effect on uptake in Rest, which is why we need this control.</p></li>
<li><p>KO/Active – <strong>Focal treatment</strong>. At about same level of KO/Rest if TLR9-/- completely inhibits electrical stimulation of glucose uptake</p></li>
</ul>
<p><strong>Design</strong> – <span class="math inline">\(2 \times 2\)</span>, that is, two crossed factors each with two levels. This results in four groups, each with a unique combination of the levels from each factor.</p>
<p><strong>Planned Contrasts</strong></p>
<p>The two ways of framing the research question suggest either of the following sets of contrasts. Both ways of framing generate a <strong>treatment</strong> contrast that includes the focal treatment and a <strong>control contrast</strong> that does not include the focal treatment. The question pursued by the experiment is addressed with the contrast of the treatment and control contrasts, which is is the estimate of the interaction effect. There is no difference in inference between the two ways of framing the question – the interaction effects are equivalent. The two framings simply give two different ways of viewing an interaction effect.</p>
<p><strong>Framing 1</strong>: How much does TLR9 knockout inhibit the expected increase of glucose uptake during stimulation?</p>
<ol style="list-style-type: decimal">
<li><p>(KO/Active - KO/Rest) – effect of Stimulation in KO mice. This is our <strong>treatment contrast</strong>. If TLR9 is necessary for glucose uptake, then this should be zero. If positive, there are non-TLR9 paths.</p></li>
<li><p>(WT/Active - WT/Rest) – This is the <strong>positive control contrast</strong> – it is what we know based on prior knowledge and what we want to compare the treatment effect to. This should be positive based on prior knowledge.</p></li>
</ol>
<p>We need to control for the expected increase in 2 using the contrast:</p>
<ol start="3" style="list-style-type: decimal">
<li>(KO/Active - KO/Rest) - (WT/Active - WT/Rest) – This contrast is the interaction effect. This <strong>focal contrast</strong> is what we need to estimate the knockout-induced stimulation effect.</li>
</ol>
<p><strong>Framing 2</strong>: What is stimulation-induced TLR9-/- effect?</p>
<ol style="list-style-type: decimal">
<li><p>(KO/Active - WT/Active) – effect of KO during muscle stimulation. This is our <strong>treatment contrast</strong>. If TLR9 is necessary for glucose uptake, then this should be big and negative <em>if</em> the KO effect at rest is small.</p></li>
<li><p>(KO/Rest - WT/Rest) – effect of KO when the muscle is not stimulated. This is the <strong>methodological control contrast</strong>. If TLR9-/- KO has non-muscle-stimulation paths to glucose uptake, this will be something other than zero.</p></li>
</ol>
<p>We need to control for any (KO/Rest - WT/Rest) effect using the contrast:</p>
<ol start="3" style="list-style-type: decimal">
<li>(KO/Active - WT/Active) - (KO/Rest - WT/Rest) – This contrast is the interaction effect. This <strong>focal contrast</strong> is what we need to estimate the stimulation-induced TLR9-/- effect.</li>
</ol>
<p>Our planned contrasts are:</p>
<ol style="list-style-type: decimal">
<li>(WT/Active - WT/Rest) – positive control contrast</li>
<li>(KO/Rest - WT/Rest) – methodological control contrast</li>
<li>(KO/Active - WT/Active) – treatment contrast</li>
<li>(KO/Active - WT/Active) - (KO/Rest - WT/Rest) – interaction contrast</li>
</ol>
</div>
<div id="fit-the-linear-model" class="section level3" number="17.2.2">
<h3><span class="header-section-number">17.2.2</span> Fit the linear model</h3>
<div class="sourceCode" id="cb470"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb470-1"><a href="factorial.html#cb470-1" aria-hidden="true" tabindex="-1"></a>exp2j_m1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> stimulation <span class="sc">*</span> genotype,</span>
<span id="cb470-2"><a href="factorial.html#cb470-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> exp2j)</span></code></pre></div>
</div>
<div id="twoway-exp2j-inf" class="section level3" number="17.2.3">
<h3><span class="header-section-number">17.2.3</span> Inference</h3>
<p><strong>The coefficient table</strong></p>
<div class="sourceCode" id="cb471"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb471-1"><a href="factorial.html#cb471-1" aria-hidden="true" tabindex="-1"></a>exp2j_m1_coef <span class="ot">&lt;-</span> <span class="fu">tidy</span>(exp2j_m1, <span class="at">conf.int =</span> <span class="cn">TRUE</span>)</span>
<span id="cb471-2"><a href="factorial.html#cb471-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb471-3"><a href="factorial.html#cb471-3" aria-hidden="true" tabindex="-1"></a>exp2j_m1_coef <span class="sc">%&gt;%</span></span>
<span id="cb471-4"><a href="factorial.html#cb471-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb471-5"><a href="factorial.html#cb471-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
std.error
</th>
<th style="text-align:right;">
statistic
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
conf.low
</th>
<th style="text-align:right;">
conf.high
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
6.75
</td>
<td style="text-align:right;">
0.419
</td>
<td style="text-align:right;">
16.1
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
5.89
</td>
<td style="text-align:right;">
7.61
</td>
</tr>
<tr>
<td style="text-align:left;">
stimulationActive
</td>
<td style="text-align:right;">
3.45
</td>
<td style="text-align:right;">
0.592
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
2.23
</td>
<td style="text-align:right;">
4.67
</td>
</tr>
<tr>
<td style="text-align:left;">
genotypeKO
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.639
</td>
<td style="text-align:right;">
1.2
</td>
<td style="text-align:right;">
0.2338
</td>
<td style="text-align:right;">
-0.54
</td>
<td style="text-align:right;">
2.10
</td>
</tr>
<tr>
<td style="text-align:left;">
stimulationActive:genotypeKO
</td>
<td style="text-align:right;">
-2.31
</td>
<td style="text-align:right;">
0.885
</td>
<td style="text-align:right;">
-2.6
</td>
<td style="text-align:right;">
0.0152
</td>
<td style="text-align:right;">
-4.13
</td>
<td style="text-align:right;">
-0.48
</td>
</tr>
</tbody>
</table>
<p><strong>emmeans table</strong></p>
<div class="sourceCode" id="cb472"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb472-1"><a href="factorial.html#cb472-1" aria-hidden="true" tabindex="-1"></a>exp2j_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp2j_m1,</span>
<span id="cb472-2"><a href="factorial.html#cb472-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;stimulation&quot;</span>, <span class="st">&quot;genotype&quot;</span>))</span>
<span id="cb472-3"><a href="factorial.html#cb472-3" aria-hidden="true" tabindex="-1"></a>exp2j_m1_emm <span class="sc">%&gt;%</span></span>
<span id="cb472-4"><a href="factorial.html#cb472-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb472-5"><a href="factorial.html#cb472-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
stimulation
</th>
<th style="text-align:left;">
genotype
</th>
<th style="text-align:right;">
emmean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rest
</td>
<td style="text-align:left;">
WT
</td>
<td style="text-align:right;">
6.75
</td>
<td style="text-align:right;">
0.419
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
5.89
</td>
<td style="text-align:right;">
7.61
</td>
</tr>
<tr>
<td style="text-align:left;">
Active
</td>
<td style="text-align:left;">
WT
</td>
<td style="text-align:right;">
10.20
</td>
<td style="text-align:right;">
0.419
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
9.34
</td>
<td style="text-align:right;">
11.06
</td>
</tr>
<tr>
<td style="text-align:left;">
Rest
</td>
<td style="text-align:left;">
KO
</td>
<td style="text-align:right;">
7.53
</td>
<td style="text-align:right;">
0.483
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
6.53
</td>
<td style="text-align:right;">
8.53
</td>
</tr>
<tr>
<td style="text-align:left;">
Active
</td>
<td style="text-align:left;">
KO
</td>
<td style="text-align:right;">
8.67
</td>
<td style="text-align:right;">
0.447
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
7.75
</td>
<td style="text-align:right;">
9.59
</td>
</tr>
</tbody>
</table>
<p><strong>The contrasts table</strong></p>
<div class="sourceCode" id="cb473"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb473-1"><a href="factorial.html#cb473-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exp2j_m1_emm # print in console to get row numbers</span></span>
<span id="cb473-2"><a href="factorial.html#cb473-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb473-3"><a href="factorial.html#cb473-3" aria-hidden="true" tabindex="-1"></a>wt_rest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb473-4"><a href="factorial.html#cb473-4" aria-hidden="true" tabindex="-1"></a>wt_active <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb473-5"><a href="factorial.html#cb473-5" aria-hidden="true" tabindex="-1"></a>ko_rest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb473-6"><a href="factorial.html#cb473-6" aria-hidden="true" tabindex="-1"></a>ko_active <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb473-7"><a href="factorial.html#cb473-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-8"><a href="factorial.html#cb473-8" aria-hidden="true" tabindex="-1"></a><span class="co"># contrasts are the difference in the vectors created above</span></span>
<span id="cb473-9"><a href="factorial.html#cb473-9" aria-hidden="true" tabindex="-1"></a><span class="co"># these planned contrasts are described above</span></span>
<span id="cb473-10"><a href="factorial.html#cb473-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. (WT/Active - WT/Rest) -- positive control contrast</span></span>
<span id="cb473-11"><a href="factorial.html#cb473-11" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. (KO/Rest - WT/Rest) -- methodological control contrast</span></span>
<span id="cb473-12"><a href="factorial.html#cb473-12" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. (KO/Active - WT/Active) -- treatment contrast</span></span>
<span id="cb473-13"><a href="factorial.html#cb473-13" aria-hidden="true" tabindex="-1"></a><span class="co"># 4. (KO/Active - WT/Active) - (KO/Rest - WT/Rest) -- interaction </span></span>
<span id="cb473-14"><a href="factorial.html#cb473-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-15"><a href="factorial.html#cb473-15" aria-hidden="true" tabindex="-1"></a>exp2j_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(</span>
<span id="cb473-16"><a href="factorial.html#cb473-16" aria-hidden="true" tabindex="-1"></a>  exp2j_m1_emm,</span>
<span id="cb473-17"><a href="factorial.html#cb473-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb473-18"><a href="factorial.html#cb473-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;(WT/Active - WT/Rest)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(wt_active <span class="sc">-</span> wt_rest),</span>
<span id="cb473-19"><a href="factorial.html#cb473-19" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;(KO/Rest - WT/Rest)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(ko_rest <span class="sc">-</span> wt_rest),</span>
<span id="cb473-20"><a href="factorial.html#cb473-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;(KO/Active - WT/Active)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(ko_active <span class="sc">-</span> wt_active),</span>
<span id="cb473-21"><a href="factorial.html#cb473-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;KO:Active Ixn&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(ko_active <span class="sc">-</span> wt_active) <span class="sc">-</span></span>
<span id="cb473-22"><a href="factorial.html#cb473-22" aria-hidden="true" tabindex="-1"></a>      (ko_rest <span class="sc">-</span> wt_rest)</span>
<span id="cb473-23"><a href="factorial.html#cb473-23" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb473-24"><a href="factorial.html#cb473-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb473-25"><a href="factorial.html#cb473-25" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb473-26"><a href="factorial.html#cb473-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb473-27"><a href="factorial.html#cb473-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-28"><a href="factorial.html#cb473-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-29"><a href="factorial.html#cb473-29" aria-hidden="true" tabindex="-1"></a>exp2j_m1_planned <span class="sc">%&gt;%</span></span>
<span id="cb473-30"><a href="factorial.html#cb473-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">0</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">5</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb473-31"><a href="factorial.html#cb473-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(WT/Active - WT/Rest)
</td>
<td style="text-align:right;">
3.448
</td>
<td style="text-align:right;">
0.5919
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
2.229
</td>
<td style="text-align:right;">
4.667
</td>
<td style="text-align:right;">
5.83
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;">
(KO/Rest - WT/Rest)
</td>
<td style="text-align:right;">
0.780
</td>
<td style="text-align:right;">
0.6393
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-0.537
</td>
<td style="text-align:right;">
2.097
</td>
<td style="text-align:right;">
1.22
</td>
<td style="text-align:right;">
0.23379
</td>
</tr>
<tr>
<td style="text-align:left;">
(KO/Active - WT/Active)
</td>
<td style="text-align:right;">
-1.527
</td>
<td style="text-align:right;">
0.6127
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-2.789
</td>
<td style="text-align:right;">
-0.265
</td>
<td style="text-align:right;">
-2.49
</td>
<td style="text-align:right;">
0.01967
</td>
</tr>
<tr>
<td style="text-align:left;">
KO:Active Ixn
</td>
<td style="text-align:right;">
-2.307
</td>
<td style="text-align:right;">
0.8855
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-4.131
</td>
<td style="text-align:right;">
-0.484
</td>
<td style="text-align:right;">
-2.61
</td>
<td style="text-align:right;">
0.01523
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb474"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb474-1"><a href="factorial.html#cb474-1" aria-hidden="true" tabindex="-1"></a><span class="co"># double check with automated contrasts</span></span>
<span id="cb474-2"><a href="factorial.html#cb474-2" aria-hidden="true" tabindex="-1"></a><span class="co"># contrast(exp2b.2_m1_emm, method = c(&quot;revpairwise&quot;))</span></span></code></pre></div>
</div>
<div id="plot-the-model-2" class="section level3" number="17.2.4">
<h3><span class="header-section-number">17.2.4</span> Plot the model</h3>
<div class="sourceCode" id="cb475"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb475-1"><a href="factorial.html#cb475-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot_the_model</span>(</span>
<span id="cb475-2"><a href="factorial.html#cb475-2" aria-hidden="true" tabindex="-1"></a>  exp2j_m1,</span>
<span id="cb475-3"><a href="factorial.html#cb475-3" aria-hidden="true" tabindex="-1"></a>  exp2j_m1_emm,</span>
<span id="cb475-4"><a href="factorial.html#cb475-4" aria-hidden="true" tabindex="-1"></a>  exp2j_m1_planned,</span>
<span id="cb475-5"><a href="factorial.html#cb475-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> pal_okabe_ito_blue,</span>
<span id="cb475-6"><a href="factorial.html#cb475-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend_position =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb475-7"><a href="factorial.html#cb475-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_label =</span> <span class="st">&quot;Glucose uptake</span><span class="sc">\n</span><span class="st">(nmol per mg protein</span><span class="sc">\n</span><span class="st">per 15 min)&quot;</span>,</span>
<span id="cb475-8"><a href="factorial.html#cb475-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">effect_label =</span> <span class="st">&quot;Difference in glucose uptake&quot;</span>,</span>
<span id="cb475-9"><a href="factorial.html#cb475-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="fl">0.44</span>,<span class="dv">1</span>),</span>
<span id="cb475-10"><a href="factorial.html#cb475-10" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/twoway-exp2j-plot-1.png" width="576" />
### alternaPlot the model</p>
<div class="sourceCode" id="cb476"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb476-1"><a href="factorial.html#cb476-1" aria-hidden="true" tabindex="-1"></a>exp2j_m1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp2j_m1_emm,</span>
<span id="cb476-2"><a href="factorial.html#cb476-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb476-3"><a href="factorial.html#cb476-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb476-4"><a href="factorial.html#cb476-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb476-5"><a href="factorial.html#cb476-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb476-6"><a href="factorial.html#cb476-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb476-7"><a href="factorial.html#cb476-7" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">emm_b</span>(exp2j_m1_emm)</span>
<span id="cb476-8"><a href="factorial.html#cb476-8" aria-hidden="true" tabindex="-1"></a>dodge_width <span class="ot">&lt;-</span> <span class="fl">0.4</span></span>
<span id="cb476-9"><a href="factorial.html#cb476-9" aria-hidden="true" tabindex="-1"></a>gg <span class="ot">&lt;-</span> <span class="fu">ggplot_the_response</span>(</span>
<span id="cb476-10"><a href="factorial.html#cb476-10" aria-hidden="true" tabindex="-1"></a>  exp2j_m1,</span>
<span id="cb476-11"><a href="factorial.html#cb476-11" aria-hidden="true" tabindex="-1"></a>  exp2j_m1_emm,</span>
<span id="cb476-12"><a href="factorial.html#cb476-12" aria-hidden="true" tabindex="-1"></a>  exp2j_m1_pairs[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">4</span>),],</span>
<span id="cb476-13"><a href="factorial.html#cb476-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> pal_okabe_ito_blue,</span>
<span id="cb476-14"><a href="factorial.html#cb476-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend_position =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb476-15"><a href="factorial.html#cb476-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_label =</span> <span class="st">&quot;Glucose uptake</span><span class="sc">\n</span><span class="st">(nmol per mg protein</span><span class="sc">\n</span><span class="st">per 15 min)&quot;</span>,</span>
<span id="cb476-16"><a href="factorial.html#cb476-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_pos =</span> <span class="fu">c</span>(<span class="fl">13.6</span>, <span class="fl">13.2</span>, <span class="fl">13.3</span>)</span>
<span id="cb476-17"><a href="factorial.html#cb476-17" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb476-18"><a href="factorial.html#cb476-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">x =</span> <span class="dv">2</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="dv">2</span> <span class="sc">-</span> <span class="fl">0.05</span>,</span>
<span id="cb476-19"><a href="factorial.html#cb476-19" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb476-20"><a href="factorial.html#cb476-20" aria-hidden="true" tabindex="-1"></a>               <span class="at">xend =</span> <span class="dv">2</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="dv">2</span> <span class="sc">+</span> <span class="fl">0.05</span>,</span>
<span id="cb476-21"><a href="factorial.html#cb476-21" aria-hidden="true" tabindex="-1"></a>               <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb476-22"><a href="factorial.html#cb476-22" aria-hidden="true" tabindex="-1"></a>               <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb476-23"><a href="factorial.html#cb476-23" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;gray&quot;</span>) <span class="sc">+</span></span>
<span id="cb476-24"><a href="factorial.html#cb476-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bracket</span>(</span>
<span id="cb476-25"><a href="factorial.html#cb476-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fl">2.25</span>,</span>
<span id="cb476-26"><a href="factorial.html#cb476-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb476-27"><a href="factorial.html#cb476-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>] <span class="sc">+</span> b[<span class="dv">4</span>],</span>
<span id="cb476-28"><a href="factorial.html#cb476-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;ixn p = &quot;</span>,</span>
<span id="cb476-29"><a href="factorial.html#cb476-29" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">fmt_p_value_rmd</span>(exp2j_m1_planned[<span class="dv">4</span>,<span class="st">&quot;p.value&quot;</span>])),</span>
<span id="cb476-30"><a href="factorial.html#cb476-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.size =</span> <span class="dv">3</span>,</span>
<span id="cb476-31"><a href="factorial.html#cb476-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.hjust =</span> <span class="dv">0</span>,</span>
<span id="cb476-32"><a href="factorial.html#cb476-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb476-33"><a href="factorial.html#cb476-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-34"><a href="factorial.html#cb476-34" aria-hidden="true" tabindex="-1"></a>gg</span></code></pre></div>
<div class="figure"><span id="fig:twoway-exp2j-altplot"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/twoway-exp2j-altplot-1.png" alt="Dashed gray line is expected additive mean of KO/Active" width="576" />
<p class="caption">
Figure 17.1: Dashed gray line is expected additive mean of KO/Active
</p>
</div>
</div>
</div>
<div id="understanding-the-linear-model-with-crossed-factors-1" class="section level2" number="17.3">
<h2><span class="header-section-number">17.3</span> Understanding the linear model with crossed factors 1</h2>
<p>Researchers in experimental biology report almost exclusively the <em>p</em>-values of the simple effects (Table <a href="factorial.html#tab:twoway-factorial-table">17.1</a>) in data from factorial experiments. These can be computed even if the data are flattened into a single “treatment” factor. But it is the interaction effect that is necessary to make many of the inferences made by the researchers. To see why, we need to understand what the coefficients in the coefficient table are.</p>
<div id="twoway-what-coefs-are" class="section level3" number="17.3.1">
<h3><span class="header-section-number">17.3.1</span> What the coefficients are</h3>
<p>To understand the coefficients, it helps to use the means in the emmeans table to construct a factorial table of the cell means</p>
<table class=" lightable-paper lightable-hover" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:twoway-exp2j-cellmeans">Table 17.2: </span>Cell mean table
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Rest
</th>
<th style="text-align:right;">
Active
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
WT
</td>
<td style="text-align:right;">
6.75
</td>
<td style="text-align:right;">
7.53
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
KO
</td>
<td style="text-align:right;">
10.20
</td>
<td style="text-align:right;">
8.67
</td>
</tr>
</tbody>
</table>
<p>The linear model fit to the <span class="math inline">\(\texttt{exp2j}\)</span> data is</p>
<p><span class="math display">\[
\begin{align}
\texttt{glucose_uptake} &amp;= \beta_{0} + \beta_{1}(\texttt{stimulation}_{\texttt{Active}}) + \beta_{2}(\texttt{genotype}_{\texttt{KO}}) \ + \\
&amp;\quad \ \beta_{3}(\texttt{stimulation}_{\texttt{Active}} : \texttt{genotype}_{\texttt{KO}}) + \varepsilon\
\end{align}
\]</span></p>
<p>and the fit coefficients are</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
std.error
</th>
<th style="text-align:right;">
statistic
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
conf.low
</th>
<th style="text-align:right;">
conf.high
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
6.7501
</td>
<td style="text-align:right;">
0.42
</td>
<td style="text-align:right;">
16.1280
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
5.888
</td>
<td style="text-align:right;">
7.612
</td>
</tr>
<tr>
<td style="text-align:left;">
stimulationActive
</td>
<td style="text-align:right;">
3.4482
</td>
<td style="text-align:right;">
0.59
</td>
<td style="text-align:right;">
5.8258
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
2.229
</td>
<td style="text-align:right;">
4.667
</td>
</tr>
<tr>
<td style="text-align:left;">
genotypeKO
</td>
<td style="text-align:right;">
0.7801
</td>
<td style="text-align:right;">
0.64
</td>
<td style="text-align:right;">
1.2202
</td>
<td style="text-align:right;">
0.234
</td>
<td style="text-align:right;">
-0.537
</td>
<td style="text-align:right;">
2.097
</td>
</tr>
<tr>
<td style="text-align:left;">
stimulationActive:genotypeKO
</td>
<td style="text-align:right;">
-2.3072
</td>
<td style="text-align:right;">
0.89
</td>
<td style="text-align:right;">
-2.6056
</td>
<td style="text-align:right;">
0.015
</td>
<td style="text-align:right;">
-4.131
</td>
<td style="text-align:right;">
-0.484
</td>
</tr>
</tbody>
</table>
<p>Explainer</p>
<ol style="list-style-type: decimal">
<li>Understand what the rows of the coefficient table are. There are four parameters in the fit linear model – the rows are the statistics for the estimates of these parameters. These estimates are the <strong>coefficients</strong> of the model.</li>
<li>The <span class="math inline">\(\texttt{(Intercept)}\)</span> (<span class="math inline">\(b_0\)</span>) is the mean <span class="math inline">\(\texttt{glucose_uptake}\)</span> of the reference level, which was set to the WT/Rest group (Figure <a href="factorial.html#fig:twoway-what-coefs-are">17.2</a>). This is the mean in the upper left cell in Table <a href="factorial.html#tab:twoway-exp2j-cellmeans">17.2</a>.</li>
<li>The <span class="math inline">\(\texttt{stimulationActive}\)</span> coefficient (<span class="math inline">\(b_2\)</span>) is the estimate of the <strong>the added effect of Stimulation when the genotype factor is at its reference level</strong>, and so is the mean in the upper right cell minus the mean in the upper left cell in Table <a href="factorial.html#tab:twoway-exp2j-cellmeans">17.2</a>. This coefficient is not a mean – it is a difference of means (Figure <a href="factorial.html#fig:twoway-what-coefs-are">17.2</a>). The mean of the WT/Active group is <span class="math inline">\(b_0 + b_2\)</span>.</li>
<li>The <span class="math inline">\(\texttt{genotypeKO}\)</span> coefficient (<span class="math inline">\(b_1\)</span>) is the estimate of the <strong>the added effect of knocking out TLR9 when the stimulation factor is at its reference level</strong>, and so is the mean in the lower left cell minus the mean in the upper left cell in Table <a href="factorial.html#tab:twoway-exp2j-cellmeans">17.2</a>. This coefficient is not a mean – it is a difference of means (Figure <a href="factorial.html#fig:twoway-what-coefs-are">17.2</a>). The mean of the KO/Rest group is <span class="math inline">\(b_0 + b_1\)</span>.</li>
<li>The <span class="math inline">\(\texttt{stimulationActive:genotypeKO}\)</span> coefficient (<span class="math inline">\(b_3\)</span>) is the estimate of the interaction effect between <span class="math inline">\(\texttt{genotype}\)</span> and <span class="math inline">\(\texttt{stimulation}\)</span>. It is <em>not</em> the mean in the lower right cell minus the mean in the upper left cell. Instead, <strong>it is the mean in the lower right cell minus the expected mean in the lower right cell if the genotype and activity treatment effects were additive</strong>. The expected additive-mean in the lower right cell (KO/Active) is <span class="math inline">\(b_0 + b_1 + b_2\)</span>. The mean of KO/stimulation is <span class="math inline">\(b_0 + b_1 + b_2 + b_3\)</span> (Figure <a href="factorial.html#fig:twoway-what-coefs-are">17.2</a>).</li>
<li>The interaction effect is a <strong>non-additive effect</strong>. Think about this. Adding the Stimulation alone adds 3.45 nmol glucose per protein per 15 min to the uptake rate of the reference. Adding “KO” alone adds 0.78 glucose per protein per 15 min to the uptake rate of the reference. If these effects were purely additive, then adding both Stimulation and KO to the reference rate should result in a mean of 6.75 + 3.45 + 0.78 = 10.98 glucose per protein per 15 min. The modeled mean for KO/Active is 8.67 glucose per protein per 15 min. The difference observed - additive is 8.67 - 10.98 = -2.31 glucose per protein per 15 min. Compare this difference to the interaction coefficient in the coefficient table.</li>
<li>Reinforce your understanding of “non-additive” in item 5. The interaction is a non-additive effect because the mean of the combined treatment is something different than if we were to just add the KO and Active effects. But this effect is additive in the linear model. This is what linear models are – a reference mean plus the sum of a bunch of effects.</li>
<li>Understand what these rows <strong>are not</strong>. The <span class="math inline">\(\texttt{stimulationActive}\)</span> row is not the same as the “stimulation” term in a Type III ANOVA table (the ANOVA table produced in GraphPad Prism or JMP). The <em>p</em>-values will be different because the <em>p</em>-values are testing different hypotheses. In the coefficient table, the <span class="math inline">\(\texttt{stimulationActive}\)</span> <em>p</em>-value is testing the difference of the means of WT/Active and WT/Rest. The stimulation term in a Type III ANOVA table is testing if there is an overall stimulation effect, which is estimated as the average of the two stimulation contrasts (WT/Active - WT/Rest) and (KO/Active - KO/Rest). The average of these two contrasts is not often of interest (but sometimes is – see below).</li>
</ol>
<div class="figure"><span id="fig:twoway-what-coefs-are"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/twoway-what-coefs-are-1.png" alt="The coefficients of a linear model with two crossed factors, explained." width="672" />
<p class="caption">
Figure 17.2: The coefficients of a linear model with two crossed factors, explained.
</p>
</div>
</div>
<div id="the-interaction-effect-is-something-different" class="section level3" number="17.3.2">
<h3><span class="header-section-number">17.3.2</span> The interaction effect is something different</h3>
<p>Item 6 in the coefficient table explainer stated “The interaction is a non-additive effect because the mean of the combined treatment is <strong>something different</strong> than if we were to just add the KO and Active effects. The something different is the interaction effect. If the interaction effect were zero, the expected effect of stimulation in the KO mice would be the same as the expected effect of stimulation in the WT mice (Figure <a href="factorial.html#fig:twoway-something-different">17.3</a>. This would suggest the underlying physiological changes between Rest and Active in the KO mice is”more of the same" physiological changes in the WT mice. But, because of the interaction, the underlying physiological changes between Rest and Active in the KO mice is “something different” to that of physiological changes in the WT mice (Figure <a href="factorial.html#fig:twoway-something-different">17.3</a>).</p>
<p>The biological reasons causing interaction effects are highly variable and are what makes Biology fun. Additive effects (no interaction) may occur when combined treatments act independently of each other. This might occur in the glucose uptake response if knocking out TLR9 opens a path to glucose uptake that is different from and independent of the paths activated by electrial stimulation. Positive, or <strong>synergistic</strong> interaction effects may occur when combined treatments augment each other’s ability to affect the response (see Example 3 below). This could occur in the glucose uptake response if knocking out TLR9 opens a path to glucose uptake that is different the paths activated by electrial stimulation but also makes the paths activated by stimulation more sensitive to stimulation. Negative, or <strong>antagonistic</strong> interaction effects may occur when combined treatments interfere with each other’s ability to affect the response. This could occur in the glucose uptake response if TLR9 is on the path from stimulation to glucose uptake. Knocking out TLR9 interferes with this path. If TLR9 is required, we’d expect the interaction effect to be the same magnitude but opposite sign of the control effect – that is, complete antagonism of the control effect. Previous experiments suggested this negative interaction. The measurement of this effect was the purpose of experiment <span class="math inline">\(\texttt{exp2j}\)</span>.</p>
<div class="figure"><span id="fig:twoway-something-different"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/twoway-something-different-1.png" alt="The interaction is something different, not more of the same. To get the observed difference (4), take the expected difference (3) and add something different (5)." width="672" />
<p class="caption">
Figure 17.3: The interaction is something different, not more of the same. To get the observed difference (4), take the expected difference (3) and add something different (5).
</p>
</div>
</div>
<div id="why-we-want-to-compare-the-treatment-effect-to-a-control-effect" class="section level3" number="17.3.3">
<h3><span class="header-section-number">17.3.3</span> Why we want to compare the treatment effect to a control effect</h3>
<p>The purpose of the experiment is to infer a TLR9 role in the regulation of muscle-stimulated glucose transport, that is, a <strong>stimulation-induced TLR9-/- effect</strong>. If TLR9 is in the pathway from muscle activity to glucose transport, we expect some kind of recovery to baseline (Rest) values in the KO/Active group. But TLR9 may also (or alternatively) have a role in non-stimulation-induced glucose uptake.</p>
<p>How do we make an inference about a stimulation-induced TLR9-/- effect using this experiment? Researchers typically look at the <strong>treatment effect</strong></p>
<p>(KO/Active - WT/Active)</p>
<p>and conclude a stimulation-induced TLR9-/- effect if it’s big (and negative)</p>
<p>If the treatment effect is the correct contrast for inference, why bother with the measures of glucose uptake during Rest? The reason the treatment effect alone is the wrong contrast for inferring the <em>stimulation-induced</em> TLR9 effect is because it is confounded by the <strong>control effect</strong>, which is the contrast (KO/Rest - WT/Rest) (Figure <a href="factorial.html#fig:twoway-what-we-want">17.4</a>).</p>
<div class="figure"><span id="fig:twoway-what-we-want"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/twoway-what-we-want-1.png" alt="Why the interaction effect is the stimulation-induced TLR9-/- effect" width="672" />
<p class="caption">
Figure 17.4: Why the interaction effect is the stimulation-induced TLR9-/- effect
</p>
</div>
<p>To unconfound the inference, subtract the confounder:</p>
<p>(KO/Active - WT/Active) - (KO/Rest - WT/Rest)</p>
<p>This is the interaction effect. Consequences of interpreting the treatment effect KO/Active - WT/Active as the stimulation-induced TLR9-/- effect are highlighted in Figure <a href="factorial.html#fig:twoway-scenarios">17.5</a>. In these plots, the data are those from <span class="math inline">\(\texttt{exp2j}\)</span> but with the values in the KO groups shifted up or down to create the scenarios.</p>
<p>Scenario 1. The positive control has big effect AND KO has no effect during rest AND the stimulation-induced TLR9-/- effect is equal in magnitude but opposite direction to the Active effect in WT (Figure <a href="factorial.html#fig:twoway-scenarios">17.5</a>A). The stimulation-induced TLR9-/- effect is conspicuous from the plot, if our sample means are close to the true means and we have high precision. The simple effect measures the stimulation-induced TLR9-/- effect correctly – but this is because the simple effect is equal in magnitude (but opposite sign) to the interaction effect. Many experiments in the literature are pretty similar to this scenario.</p>
<p>Scenario 2. The positive control has big effect AND KO has a negative effect during rest BUT there is zero stimulation-induced TLR9-/- effect – that is, the interaction is zero (Figure <a href="factorial.html#fig:twoway-scenarios">17.5</a>B). There is an effect of KO during stimulation but this effect is no different that that occuring during Rest. So this cannot be a <em>contraction induced</em> TLR9 effect. Simple effect 1 <em>inflates</em> the effect.</p>
<p>Scenario 3. The positive control has big effect AND KO has a positive effect during rest AND there is a stimulation-induced TLR9-/- effect that is equal to that in scenario 1 (Figure <a href="factorial.html#fig:twoway-scenarios">17.5</a>C). The positive effect of KO during Rest masks the stimulation-induced TLR9-/- effect. The simple effect <em>underestimates</em> the stimulation-induced TLR9-/- effect. This is similar to what is happening in Experiment 2j.</p>
<div class="figure"><span id="fig:twoway-scenarios"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/twoway-scenarios-1.png" alt="Scenarios to show the consequence of inferring the stimulation-induced TLR9-/- effect from the simple effect (KO/Active - WT/Active). The simple effect and interaction effect lines extend from the KO/Active mean to either the KO/Rest mean (simple) or the expected mean of KO/Active if the two factors were additive (interaction)." width="1066.66666666667" />
<p class="caption">
Figure 17.5: Scenarios to show the consequence of inferring the stimulation-induced TLR9-/- effect from the simple effect (KO/Active - WT/Active). The simple effect and interaction effect lines extend from the KO/Active mean to either the KO/Rest mean (simple) or the expected mean of KO/Active if the two factors were additive (interaction).
</p>
</div>
</div>
<div id="the-order-of-the-factors-in-the-model-tells-the-same-story-differently" class="section level3" number="17.3.4">
<h3><span class="header-section-number">17.3.4</span> The order of the factors in the model tells the same story differently</h3>
<p>The order of the factors in the model formula doesn’t matter for the coefficients, the estimated marginal means, or the contrasts. It can matter for ANOVA (more on this below) but not “tests after ANOVA.” But the order does matter to how the researchers communicates the results to themselves and in the report (Figure <a href="factorial.html#fig:twoway-same-story">17.6</a>). The order is also a natural consequence of the <a href="factorial.html#twoway-understand1">two different ways of framing the research question</a>.</p>
<div class="figure"><span id="fig:twoway-same-story"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/twoway-same-story-1.png" alt="It may take some work but these plots show the same four means and effects. The only difference is how we communicate the story to ourselves and to others. A) the order of the factors in the model is stimulation * genotype. B) the order of the factors in the model is genotype * stimulation. " width="1066.66666666667" />
<p class="caption">
Figure 17.6: It may take some work but these plots show the same four means and effects. The only difference is how we communicate the story to ourselves and to others. A) the order of the factors in the model is stimulation * genotype. B) the order of the factors in the model is genotype * stimulation.
</p>
</div>
</div>
<div id="power-for-the-interaction-effect-is-less-than-that-for-simple-effects" class="section level3" number="17.3.5">
<h3><span class="header-section-number">17.3.5</span> Power for the interaction effect is less than that for simple effects</h3>
<p>The interaction contrast is the difference of two simple contrasts and, consequently, the variance of the interaction contrast is <a href="https://en.wikipedia.org/wiki/Variance#Sum_of_uncorrelated_variables_(Bienaymé_formula)" target="_blank">twice that of the simple contrasts</a>. And, consequently, the SE of the interaction estimate is <span class="math inline">\(\sim 1.4 \times\)</span> larger than the precision of the two simple effects that form the contrast (1.4 is $sqrt{2}. The exact value will depend on differences in sample size among groups). The consequence of this is wider confidence intervals and larger <em>p</em>-values for the interaction contrast compared to a simple contrast of the same effect size.</p>
</div>
<div id="planned-comparisons-vs.-post-hoc-tests" class="section level3" number="17.3.6">
<h3><span class="header-section-number">17.3.6</span> Planned comparisons vs. post-hoc tests</h3>
<p>The contrasts computed above were planned based on questions motivating the experiment. The planned contrasts contain three of the four simple effects. The four simple effects are</p>
<div class="sourceCode" id="cb477"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb477-1"><a href="factorial.html#cb477-1" aria-hidden="true" tabindex="-1"></a>exp2j_m1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp2j_m1_emm,</span>
<span id="cb477-2"><a href="factorial.html#cb477-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb477-3"><a href="factorial.html#cb477-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb477-4"><a href="factorial.html#cb477-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb477-5"><a href="factorial.html#cb477-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb477-6"><a href="factorial.html#cb477-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb477-7"><a href="factorial.html#cb477-7" aria-hidden="true" tabindex="-1"></a>exp2j_m1_pairs <span class="sc">%&gt;%</span></span>
<span id="cb477-8"><a href="factorial.html#cb477-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">6</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb477-9"><a href="factorial.html#cb477-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
genotype
</th>
<th style="text-align:left;">
stimulation
</th>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Active - Rest
</td>
<td style="text-align:right;">
3.45
</td>
<td style="text-align:right;">
0.592
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
2.23
</td>
<td style="text-align:right;">
4.67
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
0.000004
</td>
</tr>
<tr>
<td style="text-align:left;">
KO
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Active - Rest
</td>
<td style="text-align:right;">
1.14
</td>
<td style="text-align:right;">
0.659
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-0.22
</td>
<td style="text-align:right;">
2.50
</td>
<td style="text-align:right;">
1.7
</td>
<td style="text-align:right;">
0.095500
</td>
</tr>
<tr>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Rest
</td>
<td style="text-align:left;">
KO - WT
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.639
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-0.54
</td>
<td style="text-align:right;">
2.10
</td>
<td style="text-align:right;">
1.2
</td>
<td style="text-align:right;">
0.233785
</td>
</tr>
<tr>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Active
</td>
<td style="text-align:left;">
KO - WT
</td>
<td style="text-align:right;">
-1.53
</td>
<td style="text-align:right;">
0.613
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-2.79
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
-2.5
</td>
<td style="text-align:right;">
0.019668
</td>
</tr>
</tbody>
</table>
<p>There are six pairwise comparisons for this experiment. Two of these are not a simple effect:</p>
<ol style="list-style-type: decimal">
<li>KO/Active - WT/Rest</li>
<li>WT/Active - KO/Rest</li>
</ol>
<p>These are the contrasts of the diagonal pairs in the cell-means table (Table <a href="factorial.html#tab:twoway-exp2j-cellmeans">17.2</a>). In a factorial design, we generally are not interested in these diagonal contrasts. They could be reported in a supplement. Recognize that if you are adjusting <em>p</em>-values for multiple tests, and you do not care about these contrasts but have included them in the computation of the adjustment, then your adjusted <em>p</em>-values are conservative.</p>
</div>
</div>
<div id="example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data" class="section level2" number="17.4">
<h2><span class="header-section-number">17.4</span> Example 2: Estimation of the effect of background condition on an effect (“it depends”) (Experiment 3e lesian area data)</h2>
<div id="understand-the-experimental-design" class="section level3" number="17.4.1">
<h3><span class="header-section-number">17.4.1</span> Understand the experimental design</h3>
<p><strong>Research question</strong>
1. What is the effect of the X chromosome complement on lipid-related disease markers?
2. What is the effect of sex (female or male gonad) on lipid-related disease markers?
3. How conditional is the sex effect on the chromosome complement?
4. How conditional is the effect of X chromosome complement on the level of sex?</p>
<p><strong>Response variable</strong> <span class="math inline">\(\texttt{lesian_area}\)</span> – atherosclerotic lesian area in aortic sinus.</p>
<p><strong>Factor 1</strong> – <span class="math inline">\(\texttt{sex}\)</span> (“Female,” “Male”). This is not the typical sex factor that is merely observed but is a manipulated factor. Sex is determined by the presence or absence of <em>SRY</em> on an autosome using the Four Core Genotype mouse model. <em>SRY</em> determines the gonad that develops (ovary or testis). “Female” does not have the autosome with <em>SRY</em>. “Male” has the autosome with <em>SRY</em>.</p>
<p><strong>Factor 2</strong> – <span class="math inline">\(\texttt{chromosome}\)</span> (“XX,” “XY”). The sex chromosome complement is not observed but manipulated. In “XX,” neither sex chromosome has <em>SRY</em> as the natural condition. In “XY,” <em>SRY</em> has been removed from the Y chromosome.</p>
<p><strong>Design</strong> – <span class="math inline">\(2 \times 2\)</span>, that is, two crossed factors each with two levels. This results in four groups, each with a unique combination of the levels from each factor. “Female/XX” is the control. “Male/XX” adds the autosomal <em>SRY</em> gene (and testes instead of an ovary). “Female/XY” replaces the “X” complement with the engineered Y complement. “Male/XY” has both the autosomal <em>SRY</em> and the engineered Y complement.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
sex
</th>
<th style="text-align:left;">
chromosome
</th>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:left;">
chromosome complement
</th>
<th style="text-align:left;">
autosome
</th>
<th style="text-align:left;">
gonad
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Female
</td>
<td style="text-align:left;">
XX
</td>
<td style="text-align:left;">
FXX
</td>
<td style="text-align:left;">
X
</td>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
ovary
</td>
</tr>
<tr>
<td style="text-align:left;">
Male
</td>
<td style="text-align:left;">
XX
</td>
<td style="text-align:left;">
MXX
</td>
<td style="text-align:left;">
X
</td>
<td style="text-align:left;">
SRY+
</td>
<td style="text-align:left;">
testis
</td>
</tr>
<tr>
<td style="text-align:left;">
Female
</td>
<td style="text-align:left;">
XY
</td>
<td style="text-align:left;">
FXY
</td>
<td style="text-align:left;">
Y (sry-)
</td>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
ovary
</td>
</tr>
<tr>
<td style="text-align:left;">
Male
</td>
<td style="text-align:left;">
XY
</td>
<td style="text-align:left;">
MXY
</td>
<td style="text-align:left;">
Y (sry-)
</td>
<td style="text-align:left;">
SRY+
</td>
<td style="text-align:left;">
testis
</td>
</tr>
</tbody>
</table>
<p>The research question suggest the following planned contrasts</p>
<p>What is the effect of <span class="math inline">\(\texttt{chromosome}\)</span> on lipid-related disease markers?</p>
<ol style="list-style-type: decimal">
<li>(Female/XX - Female/XY) – effect of XX in mice with female gonad. If hypothesis is true, this should be large, negative.</li>
<li>(Male/XX - Male/XY) – effect of XX in mice with male gonad. If hypothesis is true, this should be large, negative.</li>
</ol>
<p>What is the effect of <span class="math inline">\(\texttt{sex}\)</span> on lipid-related disease markers?</p>
<ol start="3" style="list-style-type: decimal">
<li>(Male/XX - Female/XX) – effect of sex in XX mice.</li>
<li>(Male/XY - Female/XY) – effect of sex in XY in mice.</li>
</ol>
<p>How conditional are the effects?</p>
<ol start="5" style="list-style-type: decimal">
<li>(Male/XX - Male/XY) - (Female/XX - Female/XY) – Interaction.</li>
</ol>
<p>In this experiment, there is no treatment contrast or control contrast. Instead, all four simple contrasts are of equal interest.</p>
</div>
<div id="fit-the-linear-model-1" class="section level3" number="17.4.2">
<h3><span class="header-section-number">17.4.2</span> Fit the linear model</h3>
<div class="sourceCode" id="cb478"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb478-1"><a href="factorial.html#cb478-1" aria-hidden="true" tabindex="-1"></a>exp3e_m1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(lesian_area <span class="sc">~</span> sex<span class="sc">*</span>chromosome, <span class="at">data =</span> exp3e)</span></code></pre></div>
</div>
<div id="check-the-model-1" class="section level3" number="17.4.3">
<h3><span class="header-section-number">17.4.3</span> Check the model</h3>
<div class="sourceCode" id="cb479"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb479-1"><a href="factorial.html#cb479-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcheck_the_model</span>(exp3e_m1)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/unnamed-chunk-221-1.png" width="576" /></p>
</div>
<div id="inference-from-the-model-2" class="section level3" number="17.4.4">
<h3><span class="header-section-number">17.4.4</span> Inference from the model</h3>
<p><strong>The coefficient table</strong></p>
<div class="sourceCode" id="cb480"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb480-1"><a href="factorial.html#cb480-1" aria-hidden="true" tabindex="-1"></a><span class="co"># step 2 - get the coefficient table</span></span>
<span id="cb480-2"><a href="factorial.html#cb480-2" aria-hidden="true" tabindex="-1"></a>exp3e_m1_coef <span class="ot">&lt;-</span> <span class="fu">tidy</span>(exp3e_m1, <span class="at">conf.int =</span> <span class="cn">TRUE</span>)</span>
<span id="cb480-3"><a href="factorial.html#cb480-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb480-4"><a href="factorial.html#cb480-4" aria-hidden="true" tabindex="-1"></a>exp3e_m1_coef <span class="sc">%&gt;%</span></span>
<span id="cb480-5"><a href="factorial.html#cb480-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">3</span>,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb480-6"><a href="factorial.html#cb480-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
std.error
</th>
<th style="text-align:right;">
statistic
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
conf.low
</th>
<th style="text-align:right;">
conf.high
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
0.540
</td>
<td style="text-align:right;">
0.0353
</td>
<td style="text-align:right;">
15.29
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
0.464
</td>
<td style="text-align:right;">
0.616
</td>
</tr>
<tr>
<td style="text-align:left;">
sexMale
</td>
<td style="text-align:right;">
-0.018
</td>
<td style="text-align:right;">
0.0500
</td>
<td style="text-align:right;">
-0.37
</td>
<td style="text-align:right;">
0.7172
</td>
<td style="text-align:right;">
-0.126
</td>
<td style="text-align:right;">
0.089
</td>
</tr>
<tr>
<td style="text-align:left;">
chromosomeXY
</td>
<td style="text-align:right;">
-0.290
</td>
<td style="text-align:right;">
0.0530
</td>
<td style="text-align:right;">
-5.48
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
-0.404
</td>
<td style="text-align:right;">
-0.177
</td>
</tr>
<tr>
<td style="text-align:left;">
sexMale:chromosomeXY
</td>
<td style="text-align:right;">
0.143
</td>
<td style="text-align:right;">
0.0749
</td>
<td style="text-align:right;">
1.90
</td>
<td style="text-align:right;">
0.0776
</td>
<td style="text-align:right;">
-0.018
</td>
<td style="text-align:right;">
0.303
</td>
</tr>
</tbody>
</table>
<p><strong>The emmeans table</strong></p>
<div class="sourceCode" id="cb481"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb481-1"><a href="factorial.html#cb481-1" aria-hidden="true" tabindex="-1"></a><span class="co"># step 3 - get the modeled means</span></span>
<span id="cb481-2"><a href="factorial.html#cb481-2" aria-hidden="true" tabindex="-1"></a>exp3e_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp3e_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;chromosome&quot;</span>))</span>
<span id="cb481-3"><a href="factorial.html#cb481-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb481-4"><a href="factorial.html#cb481-4" aria-hidden="true" tabindex="-1"></a>exp3e_m1_emm <span class="sc">%&gt;%</span></span>
<span id="cb481-5"><a href="factorial.html#cb481-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>() <span class="sc">%&gt;%</span></span>
<span id="cb481-6"><a href="factorial.html#cb481-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb481-7"><a href="factorial.html#cb481-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
sex
</th>
<th style="text-align:left;">
chromosome
</th>
<th style="text-align:right;">
emmean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Female
</td>
<td style="text-align:left;">
XX
</td>
<td style="text-align:right;">
0.540
</td>
<td style="text-align:right;">
0.0353
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
0.464
</td>
<td style="text-align:right;">
0.616
</td>
</tr>
<tr>
<td style="text-align:left;">
Male
</td>
<td style="text-align:left;">
XX
</td>
<td style="text-align:right;">
0.522
</td>
<td style="text-align:right;">
0.0353
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
0.446
</td>
<td style="text-align:right;">
0.597
</td>
</tr>
<tr>
<td style="text-align:left;">
Female
</td>
<td style="text-align:left;">
XY
</td>
<td style="text-align:right;">
0.250
</td>
<td style="text-align:right;">
0.0395
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
0.165
</td>
<td style="text-align:right;">
0.335
</td>
</tr>
<tr>
<td style="text-align:left;">
Male
</td>
<td style="text-align:left;">
XY
</td>
<td style="text-align:right;">
0.374
</td>
<td style="text-align:right;">
0.0395
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
0.289
</td>
<td style="text-align:right;">
0.459
</td>
</tr>
</tbody>
</table>
<p><strong>The contrasts table</strong></p>
<div class="sourceCode" id="cb482"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb482-1"><a href="factorial.html#cb482-1" aria-hidden="true" tabindex="-1"></a><span class="co"># m1_emm # print in console to get row numbers</span></span>
<span id="cb482-2"><a href="factorial.html#cb482-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb482-3"><a href="factorial.html#cb482-3" aria-hidden="true" tabindex="-1"></a>fxx <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb482-4"><a href="factorial.html#cb482-4" aria-hidden="true" tabindex="-1"></a>mxx <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb482-5"><a href="factorial.html#cb482-5" aria-hidden="true" tabindex="-1"></a>fxy <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb482-6"><a href="factorial.html#cb482-6" aria-hidden="true" tabindex="-1"></a>mxy <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb482-7"><a href="factorial.html#cb482-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-8"><a href="factorial.html#cb482-8" aria-hidden="true" tabindex="-1"></a><span class="co"># contrasts are the difference in the vectors created above</span></span>
<span id="cb482-9"><a href="factorial.html#cb482-9" aria-hidden="true" tabindex="-1"></a><span class="co"># the focal contrasts are in the understand the experiment section</span></span>
<span id="cb482-10"><a href="factorial.html#cb482-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. (FXY - FXX) </span></span>
<span id="cb482-11"><a href="factorial.html#cb482-11" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. (MXY - MXX)</span></span>
<span id="cb482-12"><a href="factorial.html#cb482-12" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. (MXX - FXX)</span></span>
<span id="cb482-13"><a href="factorial.html#cb482-13" aria-hidden="true" tabindex="-1"></a><span class="co"># 4. (MXY - FXY)</span></span>
<span id="cb482-14"><a href="factorial.html#cb482-14" aria-hidden="true" tabindex="-1"></a><span class="co"># 5. Interaction</span></span>
<span id="cb482-15"><a href="factorial.html#cb482-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-16"><a href="factorial.html#cb482-16" aria-hidden="true" tabindex="-1"></a>exp3e_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp3e_m1_emm,</span>
<span id="cb482-17"><a href="factorial.html#cb482-17" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb482-18"><a href="factorial.html#cb482-18" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;FXY - FXX&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(fxy <span class="sc">-</span> fxx),</span>
<span id="cb482-19"><a href="factorial.html#cb482-19" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;MXY - MXX&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(mxy <span class="sc">-</span> mxx),</span>
<span id="cb482-20"><a href="factorial.html#cb482-20" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;MXX - FXX&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(mxx <span class="sc">-</span> fxx),</span>
<span id="cb482-21"><a href="factorial.html#cb482-21" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;MXY - FXY&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(mxy <span class="sc">-</span> fxy),</span>
<span id="cb482-22"><a href="factorial.html#cb482-22" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;Interaction&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(mxy <span class="sc">-</span> mxx) <span class="sc">-</span></span>
<span id="cb482-23"><a href="factorial.html#cb482-23" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">c</span>(fxy <span class="sc">-</span> fxx)</span>
<span id="cb482-24"><a href="factorial.html#cb482-24" aria-hidden="true" tabindex="-1"></a>                           </span>
<span id="cb482-25"><a href="factorial.html#cb482-25" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb482-26"><a href="factorial.html#cb482-26" aria-hidden="true" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb482-27"><a href="factorial.html#cb482-27" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb482-28"><a href="factorial.html#cb482-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb482-29"><a href="factorial.html#cb482-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-30"><a href="factorial.html#cb482-30" aria-hidden="true" tabindex="-1"></a><span class="co"># check</span></span>
<span id="cb482-31"><a href="factorial.html#cb482-31" aria-hidden="true" tabindex="-1"></a> <span class="co"># exp3e_m1_ixn &lt;- contrast(exp3e_m1_emm,</span></span>
<span id="cb482-32"><a href="factorial.html#cb482-32" aria-hidden="true" tabindex="-1"></a> <span class="co">#                           interaction = c(&quot;revpairwise&quot;),</span></span>
<span id="cb482-33"><a href="factorial.html#cb482-33" aria-hidden="true" tabindex="-1"></a> <span class="co">#                           by = NULL)</span></span>
<span id="cb482-34"><a href="factorial.html#cb482-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-35"><a href="factorial.html#cb482-35" aria-hidden="true" tabindex="-1"></a>exp3e_m1_planned <span class="sc">%&gt;%</span></span>
<span id="cb482-36"><a href="factorial.html#cb482-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">0</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">5</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb482-37"><a href="factorial.html#cb482-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FXY - FXX
</td>
<td style="text-align:right;">
-0.290
</td>
<td style="text-align:right;">
0.0530
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
-0.404
</td>
<td style="text-align:right;">
-0.177
</td>
<td style="text-align:right;">
-5.48
</td>
<td style="text-align:right;">
0.00008
</td>
</tr>
<tr>
<td style="text-align:left;">
MXY - MXX
</td>
<td style="text-align:right;">
-0.148
</td>
<td style="text-align:right;">
0.0530
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
-0.261
</td>
<td style="text-align:right;">
-0.034
</td>
<td style="text-align:right;">
-2.79
</td>
<td style="text-align:right;">
0.01458
</td>
</tr>
<tr>
<td style="text-align:left;">
MXX - FXX
</td>
<td style="text-align:right;">
-0.018
</td>
<td style="text-align:right;">
0.0500
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
-0.126
</td>
<td style="text-align:right;">
0.089
</td>
<td style="text-align:right;">
-0.37
</td>
<td style="text-align:right;">
0.71716
</td>
</tr>
<tr>
<td style="text-align:left;">
MXY - FXY
</td>
<td style="text-align:right;">
0.124
</td>
<td style="text-align:right;">
0.0559
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
0.004
</td>
<td style="text-align:right;">
0.244
</td>
<td style="text-align:right;">
2.22
</td>
<td style="text-align:right;">
0.04309
</td>
</tr>
<tr>
<td style="text-align:left;">
Interaction
</td>
<td style="text-align:right;">
0.143
</td>
<td style="text-align:right;">
0.0749
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
-0.018
</td>
<td style="text-align:right;">
0.303
</td>
<td style="text-align:right;">
1.90
</td>
<td style="text-align:right;">
0.07762
</td>
</tr>
</tbody>
</table>
<p>Explainer</p>
<ol style="list-style-type: decimal">
<li>The two simple effects and the interaction are computed separately. If we want to adjust for three comparisons, I would use the Holm method.</li>
<li>The magnitude of the estimated effect of XX in mice with male gonads is about 1/2 that in mice with female gonads. This difference is the magnitude of the interaction.</li>
<li>Don’t infer “no effect” given the p-value of the interaction. The estimate of the interaction effect has the same magnitude as the estimated effect of XX in mice with male gonads and about 1/2 the magnitude as the estimated effect of XX in mice with female gonads. The interaction <em>p</em>-value suggests caution in our confidence of the sign of this effect.</li>
</ol>
</div>
<div id="plot-the-model-3" class="section level3" number="17.4.5">
<h3><span class="header-section-number">17.4.5</span> Plot the model</h3>
<div class="sourceCode" id="cb483"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb483-1"><a href="factorial.html#cb483-1" aria-hidden="true" tabindex="-1"></a>exp3e_m1_plot <span class="ot">&lt;-</span> <span class="fu">ggplot_the_model</span>(</span>
<span id="cb483-2"><a href="factorial.html#cb483-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">fit =</span> exp3e_m1,</span>
<span id="cb483-3"><a href="factorial.html#cb483-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">fit_emm =</span> exp3e_m1_emm,</span>
<span id="cb483-4"><a href="factorial.html#cb483-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">fit_pairs =</span> exp3e_m1_planned,</span>
<span id="cb483-5"><a href="factorial.html#cb483-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> pal_okabe_ito_blue,</span>
<span id="cb483-6"><a href="factorial.html#cb483-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend_position =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb483-7"><a href="factorial.html#cb483-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_label =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">&quot;Lesian area (&quot;</span>, mm<span class="sc">^</span><span class="dv">2</span>, <span class="st">&quot;)&quot;</span>)),</span>
<span id="cb483-8"><a href="factorial.html#cb483-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">effect_label =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">&quot;Effect (&quot;</span>, mm<span class="sc">^</span><span class="dv">2</span>, <span class="st">&quot;)&quot;</span>)),</span>
<span id="cb483-9"><a href="factorial.html#cb483-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">contrast_rows =</span> <span class="st">&quot;all&quot;</span>,</span>
<span id="cb483-10"><a href="factorial.html#cb483-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)</span>
<span id="cb483-11"><a href="factorial.html#cb483-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb483-12"><a href="factorial.html#cb483-12" aria-hidden="true" tabindex="-1"></a>exp3e_m1_plot</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/plot_the_model-1.png" width="576" />
### alternaPlot the model</p>
<div class="sourceCode" id="cb484"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb484-1"><a href="factorial.html#cb484-1" aria-hidden="true" tabindex="-1"></a>exp3e_m1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp3e_m1_emm,</span>
<span id="cb484-2"><a href="factorial.html#cb484-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb484-3"><a href="factorial.html#cb484-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb484-4"><a href="factorial.html#cb484-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb484-5"><a href="factorial.html#cb484-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb484-6"><a href="factorial.html#cb484-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb484-7"><a href="factorial.html#cb484-7" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">emm_b</span>(exp3e_m1_emm)</span>
<span id="cb484-8"><a href="factorial.html#cb484-8" aria-hidden="true" tabindex="-1"></a>dodge_width <span class="ot">&lt;-</span> <span class="fl">0.4</span></span>
<span id="cb484-9"><a href="factorial.html#cb484-9" aria-hidden="true" tabindex="-1"></a>gg <span class="ot">&lt;-</span> <span class="fu">ggplot_the_response</span>(</span>
<span id="cb484-10"><a href="factorial.html#cb484-10" aria-hidden="true" tabindex="-1"></a>  exp3e_m1,</span>
<span id="cb484-11"><a href="factorial.html#cb484-11" aria-hidden="true" tabindex="-1"></a>  exp3e_m1_emm,</span>
<span id="cb484-12"><a href="factorial.html#cb484-12" aria-hidden="true" tabindex="-1"></a>  exp3e_m1_pairs[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,],</span>
<span id="cb484-13"><a href="factorial.html#cb484-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> pal_okabe_ito_blue,</span>
<span id="cb484-14"><a href="factorial.html#cb484-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend_position =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb484-15"><a href="factorial.html#cb484-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_label =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">&quot;Lesian area (&quot;</span>, mm<span class="sc">^</span><span class="dv">2</span>, <span class="st">&quot;)&quot;</span>)),</span>
<span id="cb484-16"><a href="factorial.html#cb484-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_pos =</span> <span class="fu">c</span>(<span class="fl">0.75</span>, <span class="fl">0.78</span>, <span class="fl">0.72</span>, <span class="fl">0.72</span>)</span>
<span id="cb484-17"><a href="factorial.html#cb484-17" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb484-18"><a href="factorial.html#cb484-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">x =</span> <span class="dv">2</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="dv">2</span> <span class="sc">-</span> <span class="fl">0.05</span>,</span>
<span id="cb484-19"><a href="factorial.html#cb484-19" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb484-20"><a href="factorial.html#cb484-20" aria-hidden="true" tabindex="-1"></a>               <span class="at">xend =</span> <span class="dv">2</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="dv">2</span> <span class="sc">+</span> <span class="fl">0.05</span>,</span>
<span id="cb484-21"><a href="factorial.html#cb484-21" aria-hidden="true" tabindex="-1"></a>               <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb484-22"><a href="factorial.html#cb484-22" aria-hidden="true" tabindex="-1"></a>               <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb484-23"><a href="factorial.html#cb484-23" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;gray&quot;</span>) <span class="sc">+</span></span>
<span id="cb484-24"><a href="factorial.html#cb484-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bracket</span>(</span>
<span id="cb484-25"><a href="factorial.html#cb484-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fl">2.25</span>,</span>
<span id="cb484-26"><a href="factorial.html#cb484-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb484-27"><a href="factorial.html#cb484-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>] <span class="sc">+</span> b[<span class="dv">4</span>],</span>
<span id="cb484-28"><a href="factorial.html#cb484-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;ixn p = &quot;</span>,</span>
<span id="cb484-29"><a href="factorial.html#cb484-29" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">fmt_p_value_rmd</span>(exp3e_m1_planned[<span class="dv">5</span>,<span class="st">&quot;p.value&quot;</span>])),</span>
<span id="cb484-30"><a href="factorial.html#cb484-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.size =</span> <span class="dv">3</span>,</span>
<span id="cb484-31"><a href="factorial.html#cb484-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.hjust =</span> <span class="dv">0</span>,</span>
<span id="cb484-32"><a href="factorial.html#cb484-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb484-33"><a href="factorial.html#cb484-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb484-34"><a href="factorial.html#cb484-34" aria-hidden="true" tabindex="-1"></a>gg</span></code></pre></div>
<div class="figure"><span id="fig:twoway-exp3e-altplot"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/twoway-exp3e-altplot-1.png" alt="Dashed gray line is expected additive mean of KO/Active" width="576" />
<p class="caption">
Figure 17.7: Dashed gray line is expected additive mean of KO/Active
</p>
</div>
</div>
</div>
<div id="understanding-the-linear-model-with-crossed-factors-2" class="section level2" number="17.5">
<h2><span class="header-section-number">17.5</span> Understanding the linear model with crossed factors 2</h2>
<div id="twoway-marginal-means" class="section level3" number="17.5.1">
<h3><span class="header-section-number">17.5.1</span> Conditional and marginal means</h3>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:twoway-conditional-means">Table 17.3: </span>Conditional (white background) and marginal (color background) means from full factorial model fit to lesian area data
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Female
</th>
<th style="text-align:right;">
Male
</th>
<th style="text-align:right;">
mean
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
XX
</td>
<td style="text-align:right;background-color: white !important;">
0.540
</td>
<td style="text-align:right;background-color: white !important;">
0.522
</td>
<td style="text-align:right;background-color: #E69F00 !important;">
0.531
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
XY
</td>
<td style="text-align:right;background-color: white !important;">
0.250
</td>
<td style="text-align:right;background-color: white !important;">
0.374
</td>
<td style="text-align:right;background-color: #E69F00 !important;">
0.312
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
mean
</td>
<td style="text-align:right;background-color: #56B4E9 !important;">
0.395
</td>
<td style="text-align:right;background-color: #56B4E9 !important;">
0.448
</td>
<td style="text-align:right;background-color: white !important;">
</td>
</tr>
</tbody>
</table>
<p>The conditional means from the fit model are shown in the upper left <span class="math inline">\(2 \times 2\)</span> block (white background) of Table <a href="factorial.html#tab:twoway-conditional-means">17.3</a>. These means are conditional on the level of <span class="math inline">\(\texttt{sex}\)</span> and <span class="math inline">\(\texttt{chromosome}\)</span>. For the linear model with two crossed factor here (with no continuous covariates), these conditional means are equal to the sample means of the treatment. The values in the last row and column are the <strong>marginal means</strong>, which are the means of the associated row or column cells (these values are in the margins of the table). More generally, <em>marginal</em> refers to a statistic averaged across multiple levels of another variable. The marginal means of the <span class="math inline">\(\texttt{chromosome}\)</span> levels (orange background) are the means of the “XX” and “XY” rows. The marginal means of the <span class="math inline">\(\texttt{sex}\)</span> levels (blue background) are the means of the “Female” and “Male” columns. Note that the marginal means are simple averages across cell means and not weighted averages where the weights are the sample size used to compute the conditional (cell) means.</p>
</div>
<div id="simple-conditional-effects" class="section level3" number="17.5.2">
<h3><span class="header-section-number">17.5.2</span> Simple (conditional) effects</h3>
<p>In a factorial experiment with crossed A and B factors, there is an effect of factor A (relative to the reference, or another level of factor A) for each of the <em>p</em> levels of factor B. And, there is an effect of factor B (relative to the reference, or another level of factor B) for each of the <em>m</em> levels of factor A. These effects of one factor at each of the levels of the other factor are called the <strong>simple effects</strong>. I prefer <strong>conditional effects</strong>, since the value of the effect is conditional on the level of the other factor.</p>
<p>For the mouse lesian area experiment, there is a chromosome effect at <code>sex = Female</code> and a different effect at <code>sex = Male</code>. Similarly, there is a sex effect at <code>chromosome = XX</code> and a different effect at <code>chromosome = XY</code>.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:twoway-conditional-effects">Table 17.4: </span>Conditional (or “simple”) effects of the factorial model.
</caption>
<thead>
<tr>
<th style="text-align:left;">
chromosome
</th>
<th style="text-align:left;">
sex
</th>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
XX
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Male - Female
</td>
<td style="text-align:right;">
-0.018
</td>
<td style="text-align:right;">
0.050
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
-0.126
</td>
<td style="text-align:right;">
0.089
</td>
<td style="text-align:right;">
-0.370
</td>
<td style="text-align:right;">
0.71716
</td>
</tr>
<tr>
<td style="text-align:left;">
XY
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Male - Female
</td>
<td style="text-align:right;">
0.124
</td>
<td style="text-align:right;">
0.056
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
0.004
</td>
<td style="text-align:right;">
0.244
</td>
<td style="text-align:right;">
2.224
</td>
<td style="text-align:right;">
0.04309
</td>
</tr>
<tr>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Female
</td>
<td style="text-align:left;">
XY - XX
</td>
<td style="text-align:right;">
-0.290
</td>
<td style="text-align:right;">
0.053
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
-0.404
</td>
<td style="text-align:right;">
-0.177
</td>
<td style="text-align:right;">
-5.479
</td>
<td style="text-align:right;">
0.00008
</td>
</tr>
<tr>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Male
</td>
<td style="text-align:left;">
XY - XX
</td>
<td style="text-align:right;">
-0.148
</td>
<td style="text-align:right;">
0.053
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
-0.261
</td>
<td style="text-align:right;">
-0.034
</td>
<td style="text-align:right;">
-2.786
</td>
<td style="text-align:right;">
0.01458
</td>
</tr>
</tbody>
</table>
<p>The first two rows are the conditional effects of <span class="math inline">\(\texttt{sex}\)</span> in each of the levels of <span class="math inline">\(\texttt{chromosome}\)</span>. The last two rows are the conditional effects of <span class="math inline">\(\texttt{chromosome}\)</span> in each of the levels of <span class="math inline">\(\texttt{sex}\)</span>.</p>
<p>To help understand conditional effects, I add these to the <span class="math inline">\(m \times p\)</span> table of treatment combination means (Table <a href="factorial.html#tab:twoway-simple-effects">17.5</a>). The values in the right-side column (orange) are the conditional effects of <span class="math inline">\(\texttt{sex}\)</span> at each level of <span class="math inline">\(\texttt{chromosome}\)</span> These values are the difference of the means in the associated row. For example, the conditional effect of <span class="math inline">\(\texttt{sex}\)</span> when <code>chromosome = XY</code> is 0.124 (second value in orange column). The values in the bottom row (blue) are the conditional effects of <span class="math inline">\(\texttt{chromosome}\)</span> at each level of <span class="math inline">\(\texttt{sex}\)</span>. These values are the difference of the means in the associated column. For example, the conditional effect of <span class="math inline">\(\texttt{chromosome}\)</span> when <code>sex = Female</code> is -0.29 (first value in blue row). Note that the first conditional effect for each factor has a corresponding row in the table of coefficients of the fit model because these are the effects for that factor at the reference level of the other factor.</p>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:twoway-simple-effects">Table 17.5: </span>Conditional (“simple”) effects of sex (orange background) and chromosome (blue background) on lesian area. The conditional means of each combination of the factor levels are in the cells with the white background. The simple effect is the difference in the means of the associated row or column.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Female
</th>
<th style="text-align:right;">
Male
</th>
<th style="text-align:right;">
simple
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
XX
</td>
<td style="text-align:right;background-color: white !important;">
0.54
</td>
<td style="text-align:right;background-color: white !important;">
0.522
</td>
<td style="text-align:right;background-color: #E69F00 !important;">
-0.018
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
XY
</td>
<td style="text-align:right;background-color: white !important;">
0.25
</td>
<td style="text-align:right;background-color: white !important;">
0.374
</td>
<td style="text-align:right;background-color: #E69F00 !important;">
0.124
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
simple
</td>
<td style="text-align:right;background-color: #56B4E9 !important;">
-0.29
</td>
<td style="text-align:right;background-color: #56B4E9 !important;">
-0.148
</td>
<td style="text-align:right;background-color: white !important;">
</td>
</tr>
</tbody>
</table>
</div>
<div id="twoway-marginal-effects" class="section level3" number="17.5.3">
<h3><span class="header-section-number">17.5.3</span> Marginal effects</h3>
<p>The average of the conditional effects for a factor are the <strong>marginal effects</strong>, or the <strong>main effects</strong> in ANOVA terminology.</p>
<pre><code>## NOTE: Results may be misleading due to involvement in interactions</code></pre>
<pre><code>##  contrast      estimate     SE df t.ratio p.value
##  Male - Female   0.0529 0.0375 14 1.411   0.1800 
## 
## Results are averaged over the levels of: chromosome</code></pre>
<pre><code>## NOTE: Results may be misleading due to involvement in interactions</code></pre>
<pre><code>##  contrast estimate     SE df t.ratio p.value
##  XY - XX    -0.219 0.0375 14 -5.844  &lt;.0001 
## 
## Results are averaged over the levels of: sex</code></pre>
<p>I’m showing the full output from the emmeans package output to highlight the warning that the inference “may be misleading” because of the interaction effect in the linear model. This is a healthy warning that I follow up on below.</p>
<p>In Table <a href="factorial.html#tab:twoway-main">17.6</a>, I add the marginal effects to the table of conditional effects from above (Table <a href="factorial.html#tab:twoway-simple-effects">17.5</a>)</p>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:twoway-main">Table 17.6: </span>Marginal effects of sex (orange) and chromosome (blue) on lesian area. Simple effects are in grey cells. The conditional means of each combination of the factor levels are in the white cells.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Female
</th>
<th style="text-align:right;">
Male
</th>
<th style="text-align:right;">
simple
</th>
<th style="text-align:right;">
marginal
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
XX
</td>
<td style="text-align:right;background-color: white !important;">
0.54
</td>
<td style="text-align:right;background-color: white !important;">
0.522
</td>
<td style="text-align:right;background-color: rgba(134, 134, 134, 0.6) !important;">
-0.018
</td>
<td style="text-align:right;background-color: white !important;">
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
XY
</td>
<td style="text-align:right;background-color: white !important;">
0.25
</td>
<td style="text-align:right;background-color: white !important;">
0.374
</td>
<td style="text-align:right;background-color: rgba(134, 134, 134, 0.6) !important;">
0.124
</td>
<td style="text-align:right;background-color: white !important;">
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
simple
</td>
<td style="text-align:right;background-color: rgba(134, 134, 134, 0.6) !important;">
-0.29
</td>
<td style="text-align:right;background-color: rgba(134, 134, 134, 0.6) !important;">
-0.148
</td>
<td style="text-align:right;background-color: white !important;">
</td>
<td style="text-align:right;background-color: #56B4E9 !important;">
-0.219
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
marginal
</td>
<td style="text-align:right;background-color: white !important;">
</td>
<td style="text-align:right;background-color: white !important;">
</td>
<td style="text-align:right;background-color: #E69F00 !important;">
0.053
</td>
<td style="text-align:right;background-color: white !important;">
</td>
</tr>
</tbody>
</table>
</div>
<div id="the-additive-model" class="section level3" number="17.5.4">
<h3><span class="header-section-number">17.5.4</span> The additive model</h3>
<p>Marginal effects can be useful for summarizing a general trend, but, like any average, might not be especially meaningful if there is large heterogeneity of the simple effects, which occurs when the interaction effect is large.</p>
<p>If an interaction effect is small, and we want to summarize the results as general trends (“sex does this, chromosome does that”), then the best practice strategy is to refit a new linear model that estimates the effects of the two factors as if the interaction were equal to zero.</p>
<p><span class="math display" id="eq:twoway-reduced">\[
\begin{equation}
\texttt{lesian_area} = \beta_0 + \beta_1 \texttt{sex}_\texttt{Male} + \beta_2 \texttt{chromosome}_\texttt{XY} + \varepsilon
\tag{17.1}
\end{equation}
\]</span></p>
<p>Model @ref{eq:twoway-reduced} is a <strong>reduced model</strong> because one of the terms has been removed from the model. This particular reduced model is often referred to as the <strong>additive model</strong>, since it excludes the interaction term, which is non-additive effect (the indicator variable is the product of two “main” indicator variables). In R, this model is</p>
<div class="sourceCode" id="cb489"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb489-1"><a href="factorial.html#cb489-1" aria-hidden="true" tabindex="-1"></a>exp3e_m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(lesian_area <span class="sc">~</span> sex <span class="sc">+</span> chromosome, <span class="at">data =</span> exp3e)</span></code></pre></div>
<p>The model coefficients of the additive model are</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:twoway-exp3e-m2-coef">Table 17.7: </span>Model coefficients of additive model.
</caption>
<thead>
<tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
std.error
</th>
<th style="text-align:right;">
statistic
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
conf.low
</th>
<th style="text-align:right;">
conf.high
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
0.508
</td>
<td style="text-align:right;">
0.034
</td>
<td style="text-align:right;">
15.06
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.436
</td>
<td style="text-align:right;">
0.580
</td>
</tr>
<tr>
<td style="text-align:left;">
sexMale
</td>
<td style="text-align:right;">
0.045
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
1.11
</td>
<td style="text-align:right;">
0.283
</td>
<td style="text-align:right;">
-0.041
</td>
<td style="text-align:right;">
0.131
</td>
</tr>
<tr>
<td style="text-align:left;">
chromosomeXY
</td>
<td style="text-align:right;">
-0.219
</td>
<td style="text-align:right;">
0.041
</td>
<td style="text-align:right;">
-5.39
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
-0.306
</td>
<td style="text-align:right;">
-0.132
</td>
</tr>
</tbody>
</table>
<p>Explainer</p>
<ol style="list-style-type: decimal">
<li><span class="math inline">\(\texttt{sexMale}\)</span> is the average of the two conditional effects of “Male” (one at <code>chromosome = "XX"</code> and one at <code>chromosome = "XY</code>).</li>
<li><span class="math inline">\(\texttt{chromosomeXY}\)</span> is the average of the two conditional effects of “XY” (one at <code>sex = "Female"</code> and one at <code>sex = "Male</code>).</li>
<li><span class="math inline">\(\texttt{(Intercept)}\)</span> is the expected value without the added <span class="math inline">\(\texttt{sexMale}\)</span> or <span class="math inline">\(\texttt{chromosomeXY}\)</span> effects. This is a very abstract “average” of Female/XX and is not the average value in the Female/XX group.</li>
</ol>
<p>The conditional effects of the reduced model are</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:twoway-exp3e-m2-pairs">Table 17.8: </span>Conditional effects of additive model.
</caption>
<thead>
<tr>
<th style="text-align:left;">
chromosome
</th>
<th style="text-align:left;">
sex
</th>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
XX
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Male - Female
</td>
<td style="text-align:right;">
0.04495
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.041
</td>
<td style="text-align:right;">
0.131
</td>
<td style="text-align:right;">
1.11
</td>
<td style="text-align:right;">
0.28292
</td>
</tr>
<tr>
<td style="text-align:left;">
XY
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Male - Female
</td>
<td style="text-align:right;">
0.04495
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.041
</td>
<td style="text-align:right;">
0.131
</td>
<td style="text-align:right;">
1.11
</td>
<td style="text-align:right;">
0.28292
</td>
</tr>
<tr>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Female
</td>
<td style="text-align:left;">
XY - XX
</td>
<td style="text-align:right;">
-0.21895
</td>
<td style="text-align:right;">
0.041
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.306
</td>
<td style="text-align:right;">
-0.132
</td>
<td style="text-align:right;">
-5.39
</td>
<td style="text-align:right;">
0.00007
</td>
</tr>
<tr>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Male
</td>
<td style="text-align:left;">
XY - XX
</td>
<td style="text-align:right;">
-0.21895
</td>
<td style="text-align:right;">
0.041
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.306
</td>
<td style="text-align:right;">
-0.132
</td>
<td style="text-align:right;">
-5.39
</td>
<td style="text-align:right;">
0.00007
</td>
</tr>
</tbody>
</table>
<p>Explainer</p>
<ol style="list-style-type: decimal">
<li>In an additive model, all conditional effects for one factor are the same for each level of the other factor. This makes sense. If the model fit is additive, the interaction effect is set to zero by the model and there cannot be differences in conditional effects among the contrasts at each of the levels of the other factor (otherwise, there would be an interaction). A more sensible way of thinking about this is, it doesn’t make sense to compute or discuss conditional effects in an additive model. Instead, an additive model automatically estimates marginal effects.</li>
<li>Compare the table of marginal effects of the additive model to the table of marginal effects of the full model. The estimates for the chromosome effect are the same but the <em>t</em>-values and <em>p</em>-values differ because of different degrees of freedom (the full model estimates one more parameter, the interaction effect). The estimates for the sex effect are not the same between the two tables because of an imbalance of sample size. In the computation of the marginal effect of chromosome, the two simple effects both have sample size of 5 and 4. But in the computation of the marginal effect of sex, one simple effect has sample size of 5 and 5 while the other has a simple effect of 4 and 4.</li>
</ol>
</div>
<div id="twoway-reduce" class="section level3" number="17.5.5">
<h3><span class="header-section-number">17.5.5</span> Reduce models for the right reason</h3>
<p>Unless one factor truly has no effect, there will always be an interaction. As stated above, interactions are ubiquitous. If an interaction is small, it can make sense to drop the interaction term and re-fit an additive model to estimate marginal effects in order to present a simplified picture of what is going on, with the recognition that these estimates are smoothing over the heterogenity in conditional (simple) effects that truly exist.</p>
<p>Aided and abetted by statistics textbooks for biologists, there is a long history of researchers dropping an interaction effect because the interaction <span class="math inline">\(p&gt;0.05\)</span>. A good rule of thumb is, don’t make model decisions based on <em>p</em>-values. It doesn’t make any sense.</p>
<ol style="list-style-type: decimal">
<li>The <span class="math inline">\(p\)</span>-value is an arbitrary dichotomization of a continuous variable. Would it make sense to behave differently if the interaction were <span class="math inline">\(p=0.06\)</span> vs. <span class="math inline">\(p=0.04\)</span>, given that these two p-values are effectively identical?</li>
<li>A <span class="math inline">\(p\)</span>-value is not evidence that an effect is zero, or “doesn’t exist,” or even that an effect is “trivially small.” This is because <span class="math inline">\(p\)</span>-values are a function of measurement error, sampling error, and sample size, in addition to effect size.</li>
</ol>
<p>The interaction <em>p</em>-value for the lesion-area data is 0.078. Should we refit the additive model and report a simpler story of “a” chromosome effect and “a” sex effect? This reduced model isn’t invalid and it is useful. Some considerations.</p>
<ol style="list-style-type: decimal">
<li>there is certainly a real interaction between these two factors and this interaction reflects interesting biology.</li>
<li>For this example, I might report both – the additive effect in the main paper (since the big chromosome complement effect is the story) and the conditional effects in the supplement, which might further work on investigating the underlying biology. Or maybe two sets of <em>p</em>-values on the plot? There are lots of unexplored ways to provide more “ways” of looking at the results.</li>
<li>Regardless, for this example, I would not avoid reporting the interaction effect and the conditional effects somewhere. The estimated interaction effect (0.14 mm<span class="math inline">\(^2\)</span>) is moderately large relative to the four simple effects. It’s much bigger than the marginal effect of sex (Table <a href="factorial.html#tab:twoway-exp3e-m2-pairs">17.8</a>) and about 2/3 the size of the marginal effect of chromosome (Table <a href="factorial.html#tab:twoway-exp3e-m2-pairs">17.8</a>).</li>
<li>A response plot of both models (Figure <a href="factorial.html#fig:twoway-additive-plot">17.8</a>) can help understanding and the decision of which model to report.</li>
</ol>
<div class="figure"><span id="fig:twoway-additive-plot"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/twoway-additive-plot-1.png" alt="A. Conditional means and p-values of conditional effects. B. Marginal means and p-values of marginal effects." width="1066.66666666667" />
<p class="caption">
Figure 17.8: A. Conditional means and p-values of conditional effects. B. Marginal means and p-values of marginal effects.
</p>
</div>
</div>
<div id="the-marginal-means-of-an-additive-linear-model-with-two-factors-can-be-weird" class="section level3" number="17.5.6">
<h3><span class="header-section-number">17.5.6</span> The marginal means of an additive linear model with two factors can be weird</h3>
<p>To better understand the marginal effects computed from the additive model, let’s compare the emmeans table of the factorial and additive models.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
(#tab:twoway-exp3e_m1_emm)Conditional means of the lesian area data computed from the factorial model.
</caption>
<thead>
<tr>
<th style="text-align:left;">
sex
</th>
<th style="text-align:left;">
chromosome
</th>
<th style="text-align:right;">
emmean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Female
</td>
<td style="text-align:left;">
XX
</td>
<td style="text-align:right;">
0.540
</td>
<td style="text-align:right;">
0.035
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
0.464
</td>
<td style="text-align:right;">
0.616
</td>
</tr>
<tr>
<td style="text-align:left;">
Male
</td>
<td style="text-align:left;">
XX
</td>
<td style="text-align:right;">
0.522
</td>
<td style="text-align:right;">
0.035
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
0.446
</td>
<td style="text-align:right;">
0.597
</td>
</tr>
<tr>
<td style="text-align:left;">
Female
</td>
<td style="text-align:left;">
XY
</td>
<td style="text-align:right;">
0.250
</td>
<td style="text-align:right;">
0.039
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
0.165
</td>
<td style="text-align:right;">
0.335
</td>
</tr>
<tr>
<td style="text-align:left;">
Male
</td>
<td style="text-align:left;">
XY
</td>
<td style="text-align:right;">
0.374
</td>
<td style="text-align:right;">
0.039
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
0.289
</td>
<td style="text-align:right;">
0.459
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
(#tab:twoway-exp3e_m2_emm)Marginal means of the lesian area data computed from the additive model.
</caption>
<thead>
<tr>
<th style="text-align:left;">
sex
</th>
<th style="text-align:left;">
chromosome
</th>
<th style="text-align:right;">
emmean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Female
</td>
<td style="text-align:left;">
XX
</td>
<td style="text-align:right;">
0.508
</td>
<td style="text-align:right;">
0.034
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.436
</td>
<td style="text-align:right;">
0.580
</td>
</tr>
<tr>
<td style="text-align:left;">
Male
</td>
<td style="text-align:left;">
XX
</td>
<td style="text-align:right;">
0.553
</td>
<td style="text-align:right;">
0.034
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.481
</td>
<td style="text-align:right;">
0.625
</td>
</tr>
<tr>
<td style="text-align:left;">
Female
</td>
<td style="text-align:left;">
XY
</td>
<td style="text-align:right;">
0.289
</td>
<td style="text-align:right;">
0.036
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.212
</td>
<td style="text-align:right;">
0.367
</td>
</tr>
<tr>
<td style="text-align:left;">
Male
</td>
<td style="text-align:left;">
XY
</td>
<td style="text-align:right;">
0.334
</td>
<td style="text-align:right;">
0.036
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.257
</td>
<td style="text-align:right;">
0.412
</td>
</tr>
</tbody>
</table>
<p>Explainer</p>
<ol style="list-style-type: decimal">
<li>The means in the conditional means table (Table @ref(tab:twoway-exp3e_m1_emm)) are equal to the sample means. These are conditional on <span class="math inline">\(\texttt{sex}\)</span> and <span class="math inline">\(\texttt{chromosome}\)</span>.</li>
<li>The means in the marginal means table (Table @ref(tab:twoway-exp3e_m2_emm)) are not equal to the sample means. These are modeled means of the four groups from a model in which there is no interaction effect, so all conditional effects for one factor are the same for each level of the other factor. If you take the difference of Male - Female for both the XX and the XY groups, these will be the same. All data has some measured interaction (even if there is no true interaction. But remember, interaction is ubiquitous in biology). The larger this interaction, the more weird the marginal means because these are less compatible with the data.</li>
</ol>
</div>
</div>
<div id="example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data" class="section level2" number="17.6">
<h2><span class="header-section-number">17.6</span> Example 3: Estimation of synergy (“More than the sum of the parts”) (Experiment 1c JA data)</h2>
<p>To explain what synergy is and why it is estimated by the interaction effect, this example uses data from an experiment measuring the effect of two defense signals on the defense response in Maize plants. In response to herbivory from insects, maize, and other plants, release multiple, chemical signals into the air (chemicals that evaporate into the air are known as volatile compounds). These chemicals signal the plant, and neighboring plants, to secrete anti-herbivory hormones, including abcisic acid and jasmonic acid. The researchers investigated the effects of two volatile compounds, (Z)‐3‐hexenyl acetate (HAC) and Indole, on the defense response both each without the other and in combination.</p>
<p>The example data come from Figure 1c, which is the effect of HAC and Indole on tissue concentrations of the hormone jasmonic acid (JA). The design is fully crossed with two factors, each with two levels: <span class="math inline">\(\texttt{hac}\)</span>, with levels “HAC-” and “HAC+,” and <span class="math inline">\(\texttt{indole}\)</span>, with levels (“Indole-” and “Indole+”).</p>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
HAC-
</th>
<th style="text-align:left;">
HAC+
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Indole-
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:left;">
HAC
</td>
</tr>
<tr>
<td style="text-align:left;">
Indole+
</td>
<td style="text-align:left;">
Indole
</td>
<td style="text-align:left;">
HAC+Indole
</td>
</tr>
</tbody>
</table>
<div id="examine-the-data" class="section level3" number="17.6.1">
<h3><span class="header-section-number">17.6.1</span> Examine the data</h3>
<div class="sourceCode" id="cb490"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb490-1"><a href="factorial.html#cb490-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qplot</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> ja, <span class="at">data =</span> exp1)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/unnamed-chunk-222-1.png" width="576" /></p>
<p>Too few points for box plot. control variance is small. No obvious implausible points. fit with lm but recognize small n warning for any inference.</p>
</div>
<div id="fit-the-model-7" class="section level3" number="17.6.2">
<h3><span class="header-section-number">17.6.2</span> Fit the model</h3>
<div class="sourceCode" id="cb491"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb491-1"><a href="factorial.html#cb491-1" aria-hidden="true" tabindex="-1"></a>exp1c_m1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(ja <span class="sc">~</span> hac <span class="sc">*</span> indole, <span class="at">data =</span> exp1)</span></code></pre></div>
</div>
<div id="model-check" class="section level3" number="17.6.3">
<h3><span class="header-section-number">17.6.3</span> Model check</h3>
<div class="sourceCode" id="cb492"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb492-1"><a href="factorial.html#cb492-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcheck_the_model</span>(exp1c_m1)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/twoway-ja-check-1.png" width="576" /></p>
<p>The distribution looks like a sample from a Normal. The variance looks like it increases with the mean. This suggest gls modeling heterogeneity.</p>
</div>
<div id="inference-from-the-model-3" class="section level3" number="17.6.4">
<h3><span class="header-section-number">17.6.4</span> Inference from the model</h3>
<div class="sourceCode" id="cb493"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb493-1"><a href="factorial.html#cb493-1" aria-hidden="true" tabindex="-1"></a>exp1c_m1_coef <span class="ot">&lt;-</span> <span class="fu">tidy</span>(exp1c_m1, <span class="at">conf.int =</span> <span class="cn">TRUE</span>)</span>
<span id="cb493-2"><a href="factorial.html#cb493-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb493-3"><a href="factorial.html#cb493-3" aria-hidden="true" tabindex="-1"></a>exp1c_m1_coef <span class="sc">%&gt;%</span></span>
<span id="cb493-4"><a href="factorial.html#cb493-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb493-5"><a href="factorial.html#cb493-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
std.error
</th>
<th style="text-align:right;">
statistic
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
conf.low
</th>
<th style="text-align:right;">
conf.high
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
43.862
</td>
<td style="text-align:right;">
4.881
</td>
<td style="text-align:right;">
8.986
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
33.515
</td>
<td style="text-align:right;">
54.210
</td>
</tr>
<tr>
<td style="text-align:left;">
hacHAC+
</td>
<td style="text-align:right;">
15.615
</td>
<td style="text-align:right;">
6.903
</td>
<td style="text-align:right;">
2.262
</td>
<td style="text-align:right;">
0.038
</td>
<td style="text-align:right;">
0.982
</td>
<td style="text-align:right;">
30.249
</td>
</tr>
<tr>
<td style="text-align:left;">
indoleIndole+
</td>
<td style="text-align:right;">
13.104
</td>
<td style="text-align:right;">
6.903
</td>
<td style="text-align:right;">
1.898
</td>
<td style="text-align:right;">
0.076
</td>
<td style="text-align:right;">
-1.529
</td>
<td style="text-align:right;">
27.738
</td>
</tr>
<tr>
<td style="text-align:left;">
hacHAC+:indoleIndole+
</td>
<td style="text-align:right;">
29.813
</td>
<td style="text-align:right;">
9.762
</td>
<td style="text-align:right;">
3.054
</td>
<td style="text-align:right;">
0.008
</td>
<td style="text-align:right;">
9.118
</td>
<td style="text-align:right;">
50.508
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb494"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb494-1"><a href="factorial.html#cb494-1" aria-hidden="true" tabindex="-1"></a>exp1c_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp1c_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;hac&quot;</span>, <span class="st">&quot;indole&quot;</span>))</span>
<span id="cb494-2"><a href="factorial.html#cb494-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb494-3"><a href="factorial.html#cb494-3" aria-hidden="true" tabindex="-1"></a>exp1c_m1_emm <span class="sc">%&gt;%</span></span>
<span id="cb494-4"><a href="factorial.html#cb494-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb494-5"><a href="factorial.html#cb494-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
hac
</th>
<th style="text-align:left;">
indole
</th>
<th style="text-align:right;">
emmean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
HAC-
</td>
<td style="text-align:left;">
Indole-
</td>
<td style="text-align:right;">
43.9
</td>
<td style="text-align:right;">
4.88
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
33.5
</td>
<td style="text-align:right;">
54.2
</td>
</tr>
<tr>
<td style="text-align:left;">
HAC+
</td>
<td style="text-align:left;">
Indole-
</td>
<td style="text-align:right;">
59.5
</td>
<td style="text-align:right;">
4.88
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
49.1
</td>
<td style="text-align:right;">
69.8
</td>
</tr>
<tr>
<td style="text-align:left;">
HAC-
</td>
<td style="text-align:left;">
Indole+
</td>
<td style="text-align:right;">
57.0
</td>
<td style="text-align:right;">
4.88
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
46.6
</td>
<td style="text-align:right;">
67.3
</td>
</tr>
<tr>
<td style="text-align:left;">
HAC+
</td>
<td style="text-align:left;">
Indole+
</td>
<td style="text-align:right;">
102.4
</td>
<td style="text-align:right;">
4.88
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
92.0
</td>
<td style="text-align:right;">
112.7
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb495"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb495-1"><a href="factorial.html#cb495-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exp1c_m1_emm # print in console to get row numbers</span></span>
<span id="cb495-2"><a href="factorial.html#cb495-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb495-3"><a href="factorial.html#cb495-3" aria-hidden="true" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb495-4"><a href="factorial.html#cb495-4" aria-hidden="true" tabindex="-1"></a>hac <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb495-5"><a href="factorial.html#cb495-5" aria-hidden="true" tabindex="-1"></a>indole <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb495-6"><a href="factorial.html#cb495-6" aria-hidden="true" tabindex="-1"></a>hac_indole <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb495-7"><a href="factorial.html#cb495-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb495-8"><a href="factorial.html#cb495-8" aria-hidden="true" tabindex="-1"></a><span class="co"># contrasts are the difference in the vectors created above</span></span>
<span id="cb495-9"><a href="factorial.html#cb495-9" aria-hidden="true" tabindex="-1"></a><span class="co"># these planned contrasts are described above</span></span>
<span id="cb495-10"><a href="factorial.html#cb495-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. (hac+/indole- - hac-/indole-) # add hac</span></span>
<span id="cb495-11"><a href="factorial.html#cb495-11" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. (hac-/indole+ -  hac-/indole-) # add indole</span></span>
<span id="cb495-12"><a href="factorial.html#cb495-12" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. (hac+/indole+ - hac-/indole+) - (hac+/indole- - hac-/indole-) # Interaction</span></span>
<span id="cb495-13"><a href="factorial.html#cb495-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb495-14"><a href="factorial.html#cb495-14" aria-hidden="true" tabindex="-1"></a>exp1c_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(</span>
<span id="cb495-15"><a href="factorial.html#cb495-15" aria-hidden="true" tabindex="-1"></a>  exp1c_m1_emm,</span>
<span id="cb495-16"><a href="factorial.html#cb495-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb495-17"><a href="factorial.html#cb495-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;hac+&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(hac <span class="sc">-</span> ref),</span>
<span id="cb495-18"><a href="factorial.html#cb495-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;indole+&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(indole <span class="sc">-</span> ref),</span>
<span id="cb495-19"><a href="factorial.html#cb495-19" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;HAC/Indole Ixn&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(hac_indole <span class="sc">-</span> indole) <span class="sc">-</span></span>
<span id="cb495-20"><a href="factorial.html#cb495-20" aria-hidden="true" tabindex="-1"></a>      (hac <span class="sc">-</span> ref)</span>
<span id="cb495-21"><a href="factorial.html#cb495-21" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb495-22"><a href="factorial.html#cb495-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb495-23"><a href="factorial.html#cb495-23" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb495-24"><a href="factorial.html#cb495-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb495-25"><a href="factorial.html#cb495-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb495-26"><a href="factorial.html#cb495-26" aria-hidden="true" tabindex="-1"></a>exp1c_m1_planned <span class="sc">%&gt;%</span></span>
<span id="cb495-27"><a href="factorial.html#cb495-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">5</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb495-28"><a href="factorial.html#cb495-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
hac+
</td>
<td style="text-align:right;">
15.62
</td>
<td style="text-align:right;">
6.903
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
0.98
</td>
<td style="text-align:right;">
30.25
</td>
<td style="text-align:right;">
2.26
</td>
<td style="text-align:right;">
0.03796
</td>
</tr>
<tr>
<td style="text-align:left;">
indole+
</td>
<td style="text-align:right;">
13.10
</td>
<td style="text-align:right;">
6.903
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
-1.53
</td>
<td style="text-align:right;">
27.74
</td>
<td style="text-align:right;">
1.90
</td>
<td style="text-align:right;">
0.07583
</td>
</tr>
<tr>
<td style="text-align:left;">
HAC/Indole Ixn
</td>
<td style="text-align:right;">
29.81
</td>
<td style="text-align:right;">
9.762
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
9.12
</td>
<td style="text-align:right;">
50.51
</td>
<td style="text-align:right;">
3.05
</td>
<td style="text-align:right;">
0.00758
</td>
</tr>
</tbody>
</table>
</div>
<div id="plot-the-model-4" class="section level3" number="17.6.5">
<h3><span class="header-section-number">17.6.5</span> Plot the model</h3>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/twoway-ja-plot-1.png" width="576" /></p>
</div>
<div id="alternative-plot" class="section level3" number="17.6.6">
<h3><span class="header-section-number">17.6.6</span> Alternative plot</h3>
<div class="figure"><span id="fig:twoway-ja-alternaplot"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/twoway-ja-alternaplot-1.png" alt="An alternative plot for showing the estimate of synergy. Gray, dashed line is the expected mean of the HAC + Indole group if the interaction is zero." width="576" />
<p class="caption">
Figure 17.9: An alternative plot for showing the estimate of synergy. Gray, dashed line is the expected mean of the HAC + Indole group if the interaction is zero.
</p>
</div>
</div>
</div>
<div id="understanding-the-linear-model-with-crossed-factors-3" class="section level2" number="17.7">
<h2><span class="header-section-number">17.7</span> Understanding the linear model with crossed factors 3</h2>
<div id="thinking-about-the-coefficients-of-the-linear-model" class="section level3" number="17.7.1">
<h3><span class="header-section-number">17.7.1</span> Thinking about the coefficients of the linear model</h3>
<p><span class="math display" id="eq:factorial-full">\[
\begin{equation}
\texttt{ja} = \beta_0 + \beta_1 (\texttt{hac}_\texttt{HAC+}) + \beta_2 (\texttt{indole}_\texttt{Indole+}) + \beta_3 (\texttt{hac}_\texttt{HAC+}:\texttt{indole}_\texttt{Indole+}) +\varepsilon
\tag{17.2}
\end{equation}
\]</span></p>
<p>The linear model makes it easy to understand synergy in the <span class="math inline">\(2 \times 2\)</span> design. If we start with the mean of the reference (the group without the added HAC or Indole), then <span class="math inline">\(\beta_1\)</span> is the extra bit due to adding HAC, <span class="math inline">\(\beta_2\)</span> is the extra bit due to adding Indole, and <span class="math inline">\(\beta_3\)</span> is the extra bit due to synergy between HAC and Indole. A positive <span class="math inline">\(\beta_3\)</span> means the combined treatment effect is more than the sum of the parts.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:factorial-coef">Table 17.9: </span>Coefficient table of the factorial model
</caption>
<thead>
<tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
std.error
</th>
<th style="text-align:right;">
statistic
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
conf.low
</th>
<th style="text-align:right;">
conf.high
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
43.86
</td>
<td style="text-align:right;">
4.9
</td>
<td style="text-align:right;">
8.986
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
33.51
</td>
<td style="text-align:right;">
54.21
</td>
</tr>
<tr>
<td style="text-align:left;">
hacHAC+
</td>
<td style="text-align:right;">
15.62
</td>
<td style="text-align:right;">
6.9
</td>
<td style="text-align:right;">
2.262
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
0.98
</td>
<td style="text-align:right;">
30.25
</td>
</tr>
<tr>
<td style="text-align:left;">
indoleIndole+
</td>
<td style="text-align:right;">
13.10
</td>
<td style="text-align:right;">
6.9
</td>
<td style="text-align:right;">
1.898
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
-1.53
</td>
<td style="text-align:right;">
27.74
</td>
</tr>
<tr>
<td style="text-align:left;">
hacHAC+:indoleIndole+
</td>
<td style="text-align:right;">
29.81
</td>
<td style="text-align:right;">
9.8
</td>
<td style="text-align:right;">
3.054
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
9.12
</td>
<td style="text-align:right;">
50.51
</td>
</tr>
</tbody>
</table>
<p>Again, the interaction is a <strong>non-additive effect</strong>. Adding HAC alone increases JA concentration by 15.6 ng per g FW. Adding Indole alone increases JA concentration by 13.1 ng per g FW. If these effects were purely additive, then adding both HAC and Indole to the Control mean should result in a mean of 43.9 + 15.6 + 13.1 = 72.6 ng per g FW in the HAC+Indole group. The modeled mean is 102.4 ng per g FW. The difference (observed - additive) is 102.4 - 72.6 = 29.8 ng per g FW. This is the estimated interaction effect in the coefficient table.</p>
<div class="figure"><span id="fig:twoway-synergy"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/twoway-synergy-1.png" alt="Synergy is the bit needed to get to the HAC + Indole mean after adding the HAC effect and the Indole effect to the Control mean" width="672" />
<p class="caption">
Figure 17.10: Synergy is the bit needed to get to the HAC + Indole mean after adding the HAC effect and the Indole effect to the Control mean
</p>
</div>
</div>
</div>
<div id="issues-in-inference" class="section level2" number="17.8">
<h2><span class="header-section-number">17.8</span> Issues in inference</h2>
<div id="for-pairwise-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model" class="section level3" number="17.8.1">
<h3><span class="header-section-number">17.8.1</span> For pairwise contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</h3>
<p>Compare the pairwise comparisons of the Experiment 2j glucose uptake data using a factorial linear model and a flattened linear model.</p>
<p>Factorial:</p>
<div class="sourceCode" id="cb496"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb496-1"><a href="factorial.html#cb496-1" aria-hidden="true" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> stimulation <span class="sc">*</span> genotype, <span class="at">data =</span> exp2j)</span>
<span id="cb496-2"><a href="factorial.html#cb496-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb496-3"><a href="factorial.html#cb496-3" aria-hidden="true" tabindex="-1"></a>m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;stimulation&quot;</span>, <span class="st">&quot;genotype&quot;</span>))</span>
<span id="cb496-4"><a href="factorial.html#cb496-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb496-5"><a href="factorial.html#cb496-5" aria-hidden="true" tabindex="-1"></a>m1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(m1_emm,</span>
<span id="cb496-6"><a href="factorial.html#cb496-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb496-7"><a href="factorial.html#cb496-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">adjust =</span> <span class="st">&quot;tukey&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb496-8"><a href="factorial.html#cb496-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb496-9"><a href="factorial.html#cb496-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb496-10"><a href="factorial.html#cb496-10" aria-hidden="true" tabindex="-1"></a>m1_pairs <span class="sc">%&gt;%</span></span>
<span id="cb496-11"><a href="factorial.html#cb496-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb496-12"><a href="factorial.html#cb496-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Active WT - Rest WT
</td>
<td style="text-align:right;">
3.448
</td>
<td style="text-align:right;">
0.592
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
1.820
</td>
<td style="text-align:right;">
5.076
</td>
<td style="text-align:right;">
5.826
</td>
<td style="text-align:right;">
0.000
</td>
</tr>
<tr>
<td style="text-align:left;">
Rest KO - Rest WT
</td>
<td style="text-align:right;">
0.780
</td>
<td style="text-align:right;">
0.639
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-0.978
</td>
<td style="text-align:right;">
2.539
</td>
<td style="text-align:right;">
1.220
</td>
<td style="text-align:right;">
0.620
</td>
</tr>
<tr>
<td style="text-align:left;">
Rest KO - Active WT
</td>
<td style="text-align:right;">
-2.668
</td>
<td style="text-align:right;">
0.639
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-4.427
</td>
<td style="text-align:right;">
-0.910
</td>
<td style="text-align:right;">
-4.173
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
<tr>
<td style="text-align:left;">
Active KO - Rest WT
</td>
<td style="text-align:right;">
1.921
</td>
<td style="text-align:right;">
0.613
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
0.236
</td>
<td style="text-align:right;">
3.606
</td>
<td style="text-align:right;">
3.136
</td>
<td style="text-align:right;">
0.021
</td>
</tr>
<tr>
<td style="text-align:left;">
Active KO - Active WT
</td>
<td style="text-align:right;">
-1.527
</td>
<td style="text-align:right;">
0.613
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-3.212
</td>
<td style="text-align:right;">
0.158
</td>
<td style="text-align:right;">
-2.493
</td>
<td style="text-align:right;">
0.086
</td>
</tr>
<tr>
<td style="text-align:left;">
Active KO - Rest KO
</td>
<td style="text-align:right;">
1.141
</td>
<td style="text-align:right;">
0.659
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-0.671
</td>
<td style="text-align:right;">
2.953
</td>
<td style="text-align:right;">
1.733
</td>
<td style="text-align:right;">
0.329
</td>
</tr>
</tbody>
</table>
<p>Flattened:</p>
<div class="sourceCode" id="cb497"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb497-1"><a href="factorial.html#cb497-1" aria-hidden="true" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> treatment, <span class="at">data =</span> exp2j)</span>
<span id="cb497-2"><a href="factorial.html#cb497-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb497-3"><a href="factorial.html#cb497-3" aria-hidden="true" tabindex="-1"></a>m2_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(m2, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>))</span>
<span id="cb497-4"><a href="factorial.html#cb497-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb497-5"><a href="factorial.html#cb497-5" aria-hidden="true" tabindex="-1"></a>m2_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(m2_emm,</span>
<span id="cb497-6"><a href="factorial.html#cb497-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb497-7"><a href="factorial.html#cb497-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">adjust =</span> <span class="st">&quot;tukey&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb497-8"><a href="factorial.html#cb497-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb497-9"><a href="factorial.html#cb497-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb497-10"><a href="factorial.html#cb497-10" aria-hidden="true" tabindex="-1"></a>m2_pairs <span class="sc">%&gt;%</span></span>
<span id="cb497-11"><a href="factorial.html#cb497-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb497-12"><a href="factorial.html#cb497-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
WT Active - WT Rest
</td>
<td style="text-align:right;">
3.448
</td>
<td style="text-align:right;">
0.592
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
1.820
</td>
<td style="text-align:right;">
5.076
</td>
<td style="text-align:right;">
5.826
</td>
<td style="text-align:right;">
0.000
</td>
</tr>
<tr>
<td style="text-align:left;">
KO Rest - WT Rest
</td>
<td style="text-align:right;">
0.780
</td>
<td style="text-align:right;">
0.639
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-0.978
</td>
<td style="text-align:right;">
2.539
</td>
<td style="text-align:right;">
1.220
</td>
<td style="text-align:right;">
0.620
</td>
</tr>
<tr>
<td style="text-align:left;">
KO Rest - WT Active
</td>
<td style="text-align:right;">
-2.668
</td>
<td style="text-align:right;">
0.639
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-4.427
</td>
<td style="text-align:right;">
-0.910
</td>
<td style="text-align:right;">
-4.173
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
<tr>
<td style="text-align:left;">
KO Active - WT Rest
</td>
<td style="text-align:right;">
1.921
</td>
<td style="text-align:right;">
0.613
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
0.236
</td>
<td style="text-align:right;">
3.606
</td>
<td style="text-align:right;">
3.136
</td>
<td style="text-align:right;">
0.021
</td>
</tr>
<tr>
<td style="text-align:left;">
KO Active - WT Active
</td>
<td style="text-align:right;">
-1.527
</td>
<td style="text-align:right;">
0.613
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-3.212
</td>
<td style="text-align:right;">
0.158
</td>
<td style="text-align:right;">
-2.493
</td>
<td style="text-align:right;">
0.086
</td>
</tr>
<tr>
<td style="text-align:left;">
KO Active - KO Rest
</td>
<td style="text-align:right;">
1.141
</td>
<td style="text-align:right;">
0.659
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-0.671
</td>
<td style="text-align:right;">
2.953
</td>
<td style="text-align:right;">
1.733
</td>
<td style="text-align:right;">
0.329
</td>
</tr>
</tbody>
</table>
</div>
<div id="for-interaction-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model" class="section level3" number="17.8.2">
<h3><span class="header-section-number">17.8.2</span> For interaction contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</h3>
<p>Factorial model:</p>
<div class="sourceCode" id="cb498"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb498-1"><a href="factorial.html#cb498-1" aria-hidden="true" tabindex="-1"></a><span class="co"># using m1 from above</span></span>
<span id="cb498-2"><a href="factorial.html#cb498-2" aria-hidden="true" tabindex="-1"></a>m1_ixn <span class="ot">&lt;-</span> <span class="fu">contrast</span>(m1_emm,</span>
<span id="cb498-3"><a href="factorial.html#cb498-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">interaction =</span> <span class="st">&quot;revpairwise&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb498-4"><a href="factorial.html#cb498-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb498-5"><a href="factorial.html#cb498-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb498-6"><a href="factorial.html#cb498-6" aria-hidden="true" tabindex="-1"></a>m1_ixn <span class="sc">%&gt;%</span></span>
<span id="cb498-7"><a href="factorial.html#cb498-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb498-8"><a href="factorial.html#cb498-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
stimulation_revpairwise
</th>
<th style="text-align:left;">
genotype_revpairwise
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Active - Rest
</td>
<td style="text-align:left;">
KO - WT
</td>
<td style="text-align:right;">
-2.307
</td>
<td style="text-align:right;">
0.885
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-4.131
</td>
<td style="text-align:right;">
-0.484
</td>
<td style="text-align:right;">
-2.606
</td>
<td style="text-align:right;">
0.015
</td>
</tr>
</tbody>
</table>
<p>Flattened model:</p>
<div class="sourceCode" id="cb499"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb499-1"><a href="factorial.html#cb499-1" aria-hidden="true" tabindex="-1"></a><span class="co"># need to compute the interaction as a contrast</span></span>
<span id="cb499-2"><a href="factorial.html#cb499-2" aria-hidden="true" tabindex="-1"></a><span class="co"># using m2 from the previous chunk</span></span>
<span id="cb499-3"><a href="factorial.html#cb499-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb499-4"><a href="factorial.html#cb499-4" aria-hidden="true" tabindex="-1"></a><span class="co"># m2_emm # print in console to get row numbers</span></span>
<span id="cb499-5"><a href="factorial.html#cb499-5" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb499-6"><a href="factorial.html#cb499-6" aria-hidden="true" tabindex="-1"></a>wt_rest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb499-7"><a href="factorial.html#cb499-7" aria-hidden="true" tabindex="-1"></a>wt_active <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb499-8"><a href="factorial.html#cb499-8" aria-hidden="true" tabindex="-1"></a>ko_rest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb499-9"><a href="factorial.html#cb499-9" aria-hidden="true" tabindex="-1"></a>ko_active <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb499-10"><a href="factorial.html#cb499-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb499-11"><a href="factorial.html#cb499-11" aria-hidden="true" tabindex="-1"></a><span class="co"># contrasts are the difference in the vectors created above</span></span>
<span id="cb499-12"><a href="factorial.html#cb499-12" aria-hidden="true" tabindex="-1"></a><span class="co"># 4. (KO/Active - WT/Active) - (KO/Rest - WT/Rest) -- interaction </span></span>
<span id="cb499-13"><a href="factorial.html#cb499-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb499-14"><a href="factorial.html#cb499-14" aria-hidden="true" tabindex="-1"></a>m2_ixn <span class="ot">&lt;-</span> <span class="fu">contrast</span>(m2_emm,</span>
<span id="cb499-15"><a href="factorial.html#cb499-15" aria-hidden="true" tabindex="-1"></a>                   <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb499-16"><a href="factorial.html#cb499-16" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;KO:Active Ixn&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(ko_active <span class="sc">-</span> wt_active) <span class="sc">-</span></span>
<span id="cb499-17"><a href="factorial.html#cb499-17" aria-hidden="true" tabindex="-1"></a>                       (ko_rest <span class="sc">-</span> wt_rest)</span>
<span id="cb499-18"><a href="factorial.html#cb499-18" aria-hidden="true" tabindex="-1"></a>                   )) <span class="sc">%&gt;%</span></span>
<span id="cb499-19"><a href="factorial.html#cb499-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb499-20"><a href="factorial.html#cb499-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb499-21"><a href="factorial.html#cb499-21" aria-hidden="true" tabindex="-1"></a>m2_ixn <span class="sc">%&gt;%</span></span>
<span id="cb499-22"><a href="factorial.html#cb499-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb499-23"><a href="factorial.html#cb499-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
KO:Active Ixn
</td>
<td style="text-align:right;">
-2.307
</td>
<td style="text-align:right;">
0.885
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-4.131
</td>
<td style="text-align:right;">
-0.484
</td>
<td style="text-align:right;">
-2.606
</td>
<td style="text-align:right;">
0.015
</td>
</tr>
</tbody>
</table>
</div>
<div id="twoway-multiple-tests" class="section level3" number="17.8.3">
<h3><span class="header-section-number">17.8.3</span> Adjusting <em>p</em>-values for multiple tests</h3>
<p>Inflated Type I error in pairwise tests is a concern among some statisticians and many reviewers, readers, and colleagues. It is not the frequency of individual tests that is inflated in a batch of pairwise tests but the “family-wise” error rate – given the four simple-effect tests or the six pairwise tests in a <span class="math inline">\(2 \times 2\)</span> design, what is the frequency of at least one Type I error within this batch of tests.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:twoway-exp2j-p-adjust">Table 17.10: </span>P-values from seven different methods of adjustment. The Tukey-all p-value is from all six pairwise comparisons. All other p-values are from the four simple effects.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
Tukey-all
</th>
<th style="text-align:right;">
Tukey
</th>
<th style="text-align:right;">
Bonferroni
</th>
<th style="text-align:right;">
Holm
</th>
<th style="text-align:right;">
MvT
</th>
<th style="text-align:right;">
FDR
</th>
<th style="text-align:right;">
None
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
WT: Active - Rest
</td>
<td style="text-align:right;">
3.45
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
</tr>
<tr>
<td style="text-align:left;">
KO: Active - Rest
</td>
<td style="text-align:right;">
1.14
</td>
<td style="text-align:right;">
0.329
</td>
<td style="text-align:right;">
0.257
</td>
<td style="text-align:right;">
0.382
</td>
<td style="text-align:right;">
0.191
</td>
<td style="text-align:right;">
0.275
</td>
<td style="text-align:right;">
0.127
</td>
<td style="text-align:right;">
0.095
</td>
</tr>
<tr>
<td style="text-align:left;">
Rest: KO - WT
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.620
</td>
<td style="text-align:right;">
0.521
</td>
<td style="text-align:right;">
0.935
</td>
<td style="text-align:right;">
0.234
</td>
<td style="text-align:right;">
0.561
</td>
<td style="text-align:right;">
0.234
</td>
<td style="text-align:right;">
0.234
</td>
</tr>
<tr>
<td style="text-align:left;background-color: #E69F00 !important;">
Active: KO - WT
</td>
<td style="text-align:right;background-color: #E69F00 !important;">
-1.53
</td>
<td style="text-align:right;background-color: #E69F00 !important;">
0.086
</td>
<td style="text-align:right;background-color: #E69F00 !important;">
0.063
</td>
<td style="text-align:right;background-color: #E69F00 !important;">
0.079
</td>
<td style="text-align:right;background-color: #E69F00 !important;">
0.059
</td>
<td style="text-align:right;background-color: #E69F00 !important;">
0.066
</td>
<td style="text-align:right;background-color: #E69F00 !important;">
0.039
</td>
<td style="text-align:right;background-color: #E69F00 !important;">
0.020
</td>
</tr>
</tbody>
</table>
<p>Table <a href="factorial.html#tab:twoway-exp2j-p-adjust">17.10</a> shows <em>p</em>-values adjusted using seven different methods (including no adjustment) for the Experiment 2j glucose uptake data.</p>
<ol style="list-style-type: decimal">
<li><strong>Bonferroni</strong> – The Bonferroni adjustment assumes independence among the tests. This isn’t true for posthoc tests because every contrast includes a group that is in other tests (that is, the different contrasts have common data). The consequence of the correlation among the tests is that the Bonferroni adjustment is too conservative (the actual rate of Type I error will be smaller than the nominal rate).</li>
<li><strong>Tukey HSD</strong> – The Tukey adjustment corrects for the expected correlation among the tests.</li>
<li><strong>Holm</strong> – The Holm variant of the Bonferroni adjustment has increased power relative to the Bonferroni but does explicitly account for correlated tests. It’s type I error rate is too conservative.</li>
<li><strong>Mvt</strong> – The multivariate t adjustment corrects for the empirically measured correlation.</li>
<li><strong>FDR</strong> – The FDR adjustment does not attempt to correct for Type I error but for the False Discovery Rate. The difference between these concepts is important. One can have a Fisherian world-view that rejects controlling for Type I error as something to be done, and find the FDR logically satisfying.</li>
</ol>
<p>Further thoughts on multiple adjustment, focussing on the highlighted row in the table.</p>
<ol style="list-style-type: decimal">
<li>The most transparent best practice is to report all, unadjusted <em>p</em>-values. Reporting all unadjusted <em>p</em>-values allows a reader to compute whatever adjustment they deem appropriate for the question they are addressing. Reporting all <em>p</em>-values could overwhelm the principal message of a figure. Instead, report focal (planned comparison) <em>p</em>-values in the figure and a table of all <em>p</em>-values in the supplement. There should be a table of these for every experiment in the supplement. This point raises the question of what “all” means. In some designs, “all” could mean all treatment vs. control comparisons. In others, “all” could mean all pairwise comparisons. In others, “all” could mean all pairwise comparisons and all interactions.</li>
<li>In a Fisherian world-view of <em>p</em>-values, there just isn’t any real difference in inference between <span class="math inline">\(p = 0.02\)</span> (no adjustment) and <span class="math inline">\(p = 0.086\)</span> (Tukey adjustment from all pairwise comparisons). Fisher expected decisions based on multiple replications of the experiment, all (or nearly) with <span class="math inline">\(p &lt; 0.05\)</span>. From a Fisherian-world-view, then, we have bigger how-to-do-science issues than multiple testing if we are making decisions entirely on <span class="math inline">\(p = 0.02\)</span> versus <span class="math inline">\(p = 0.086\)</span>.</li>
<li>From a Neyman-Pearson testing world-view, what is important is the family of tests. Using all six pairwise comparisons or all four simple effects as the family because this is how the software assigned it is rather mindless statistics. Many statisticians in favor of “family-wise error control” distinguish between planned and unplanned (“post-hoc”) comparisons with no adjustment for planned (“we have distinct hypotheses for each of these tests”) comparisons and adjustment for post-hoc (“is there a difference anywhere in this set of tests”) comparisons. The FDR was invented for “is there a difference anywhere in this set of tests” kinds of questions.</li>
</ol>
</div>
</div>
<div id="two-way-anova" class="section level2" number="17.9">
<h2><span class="header-section-number">17.9</span> Two-way ANOVA</h2>
<p>Both wet-bench and ecology/evolution experimental biologists do ANOVA. Many textbooks, advisors, and colleagues tell researchers to report “tests after ANOVA” – what I call pairwise contrasts – but in modern computing, pairwise contrasts and ANOVA tables are both computed from the same underlying regression model. The ANOVA is not necessary for any inference, a point I return to below.</p>
<p>Let’s use the glucose uptake experiment (Example 1) to explore the ANOVA table.</p>
<div id="how-to-read-a-two-way-anova-table" class="section level3" number="17.9.1">
<h3><span class="header-section-number">17.9.1</span> How to read a two-way ANOVA table</h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:twoway-anova-explainer">Table 17.11: </span>ANOVA table of glucose uptake experiment (Example 1).
</caption>
<thead>
<tr>
<th style="text-align:left;">
Effect
</th>
<th style="text-align:left;">
df
</th>
<th style="text-align:left;">
F
</th>
<th style="text-align:left;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
stimulation
</td>
<td style="text-align:left;">
1, 25
</td>
<td style="text-align:left;">
26.86
</td>
<td style="text-align:left;">
&lt;.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
1, 25
</td>
<td style="text-align:left;">
0.71
</td>
<td style="text-align:left;">
.41
</td>
</tr>
<tr>
<td style="text-align:left;">
stimulation:genotype
</td>
<td style="text-align:left;">
1, 25
</td>
<td style="text-align:left;">
6.79
</td>
<td style="text-align:left;">
.02
</td>
</tr>
</tbody>
</table>
<p>Explainer</p>
<ol style="list-style-type: decimal">
<li>This ANOVA table is computed using Type III sum of squares and should match the output from most statistics software including Graphpad Prism. The relevance of this is addressed below.</li>
<li>In general, an ANOVA table has a row for each term in the underlying linear model. ANOVA is a method for decomposing the total variance into batches and each of these terms is a batch. For the linear model with two crossed factors, there is a term for each factor and a term for the interaction. <span class="math inline">\(\texttt{stimulation}\)</span> and <span class="math inline">\(\texttt{genotype}\)</span> are the <strong>main</strong> (or 1st order) effects. <span class="math inline">\(\texttt{stimulation:genotype}\)</span> is the interaction (or 2nd order) effect. These terms are the row names of the table.</li>
<li>Some ANOVA functions in R also include a row for the variance of the residuals from the fit model. This row can be useful for learning what the values in ANOVA table are but only one statistic from the row (the residual df) is especially useful for reporting. In Table <a href="factorial.html#tab:twoway-anova-explainer">17.11</a>, this residual df is included in the statistics for each term.</li>
<li>Many ANOVA tables contain additional SS (sum of squares) and MSE (mean square error) columns. The MSE is the variance of the term and used to compute the <em>F</em> statistic. The SS is used to compute the MSE. Since these columns are used for computation but not reporting, Table <a href="factorial.html#tab:twoway-anova-explainer">17.11</a> excludes these.</li>
<li><strong>df</strong> (Degree of freedom) – If the term is a factor, the df will equal the number of levels (<span class="math inline">\(k\)</span>) for the factor minus 1. Think of it this way: the contribution of the variance due to a factor is a function of the variability of the <span class="math inline">\(k\)</span> level means around the grand mean. How many degrees of independent variation do these level means have, given that we know the grand mean? The answer is <span class="math inline">\(k-1\)</span> – once the values of <span class="math inline">\(k-1\)</span> level means are written down, the <span class="math inline">\(k\)</span>th level mean has no freedom to vary; its value has to be <span class="math inline">\(k\bar{\bar{Y}} - \sum_i^{k-1}{Y_i}\)</span>. For an interaction term, the df is the product of the df of each of the factors in the interaction.</li>
<li><strong>F</strong> (<em>F</em>-value or <em>F</em>-ratio) – This is the test statistic. It is the ratio of the MSE for the term divided by the MSE of the residual (this is strictly true for only “Fixed effects” ANOVA, which we have here).</li>
<li><strong>p.value</strong> – the <em>p</em>-value for the test statistic. <em>F</em> is compared to an <em>F</em>-distribution, which is a distribution of <em>F</em>-values under the null.</li>
</ol>
</div>
<div id="what-do-the-main-effects-in-an-anova-table-mean" class="section level3" number="17.9.2">
<h3><span class="header-section-number">17.9.2</span> What do the main effects in an ANOVA table mean?</h3>
<p>It is very common in the literature to see researchers report the rows of an ANOVA table by stating something like “We found an effect of stimulation on glucose uptake (<span class="math inline">\(F_{1,25} = 26.9\)</span>, <span class="math inline">\(p &lt; 0.0001\)</span>).” Thinking of a main effect term in an ANOVA table as “an effect” can be misleading. What does “an effect” mean? Afterall, there are <em>two</em> effects of <span class="math inline">\(\texttt{stimulation}\)</span> in this experiment, one in the wildtype mice and one in the TLR9-/- mice.</p>
<p>Main effects in ANOVA tables are about average effects. A main effect in a two-way ANOVA table is an “overall” marginal effect. Recall that a marginal effect of the level of Factor A is the average of the conditional effects at each level of B (Section <a href="factorial.html#twoway-marginal-effects">17.5.3</a>). For a <span class="math inline">\(2 \times 2\)</span> table, there is only one marginal effect for each factor and, as a consquence, the <em>p</em> value for the main effect term of <span class="math inline">\(\texttt{stimulation}\)</span> in the ANOVA table (Table <a href="factorial.html#tab:twoway-anova-table">17.12</a>) is equal to the <em>p</em>-value of the marginal effect of <span class="math inline">\(\texttt{stimulation}\)</span> in the contrast table (Table <a href="factorial.html#tab:twoway-marginal-table">17.13</a>). The “main term” effect of stimulation is illustrated in Figure <a href="#twoway-main-1"><strong>??</strong></a>.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:twoway-anova-table">Table 17.12: </span>Reprinting the ANOVA table of the glucose uptake experiment (Example 1) to show more decimal places of the p-value.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
num Df
</th>
<th style="text-align:right;">
den Df
</th>
<th style="text-align:right;">
F
</th>
<th style="text-align:right;">
Pr(&gt;F)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
stimulation
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
26.86
</td>
<td style="text-align:right;">
0.0000232
</td>
</tr>
<tr>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
0.71
</td>
<td style="text-align:right;">
0.4068630
</td>
</tr>
<tr>
<td style="text-align:left;">
stimulation:genotype
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
6.79
</td>
<td style="text-align:right;">
0.0152305
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:twoway-marginal-table">Table 17.13: </span>Marginal effect of stimulation on glucose uptake.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Active - Rest
</td>
<td style="text-align:right;">
2.29
</td>
<td style="text-align:right;">
0.443
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
1.38
</td>
<td style="text-align:right;">
3.21
</td>
<td style="text-align:right;">
5.183
</td>
<td style="text-align:right;">
0.0000232
</td>
</tr>
</tbody>
</table>
<p>Again, main effects in ANOVA tables are about average effects. Is the average effect what we want to report? The answer is, it depends. It does <em>not</em> depend on the <em>p</em>-value of the interaction (the answer in many textbooks and websites) although the <em>p</em>-value is, perhaps, not irrelevant. Rather, it depends on the research question motivating the experiment. The research question in Experiment 2j specifically predicted a different treatment (KO/Active - KO/Rest) and control effect (WT/Active - WT/Rest). The average of these two effects isn’t of interest. Any inference about the average effect of stimulation (the <em>p</em>-value from the ANOVA table or the marginal effect size or CIs) doesn’t answer any question motivating the experiment. A more general discussion of when we might be interested in the main effect term of an ANOVA table (or better, marginal effects) was in section xxx and below in section xxx.</p>
<div class="figure"><span id="fig:twoway-anova-explainer-plot"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/twoway-anova-explainer-plot-1.png" alt="The main effect of stimulation in the ANOVA table tests the average of the conditional effects of stimulation. The average of the conditional effects is the difference between the marginal means of Active and of Rest." width="672" />
<p class="caption">
Figure 17.11: The main effect of stimulation in the ANOVA table tests the average of the conditional effects of stimulation. The average of the conditional effects is the difference between the marginal means of Active and of Rest.
</p>
</div>
</div>
</div>
<div id="more-issues-in-inference" class="section level2" number="17.10">
<h2><span class="header-section-number">17.10</span> More issues in inference</h2>
<div id="longitudinal-experiments-include-time-as-a-random-factor-better-than-repeated-measures-anova" class="section level3" number="17.10.1">
<h3><span class="header-section-number">17.10.1</span> Longitudinal experiments – include Time as a random factor (better than repeated measures ANOVA)</h3>
</div>
</div>
<div id="working-in-r-4" class="section level2" number="17.11">
<h2><span class="header-section-number">17.11</span> Working in R</h2>
<div id="model-formula" class="section level3" number="17.11.1">
<h3><span class="header-section-number">17.11.1</span> Model formula</h3>
<p>A linear model with two crossed factors is specified in the model formula as <code>y ~ A*B</code> where <span class="math inline">\(\texttt{A}\)</span> is the first factor, and <span class="math inline">\(\texttt{B}\)</span> is the second factor. R expands this formula to <code>y ~ 1 + A + B + A:B</code> where the colon indicates an interaction (multiplicative) effect.</p>
<div class="sourceCode" id="cb500"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb500-1"><a href="factorial.html#cb500-1" aria-hidden="true" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> stimulation <span class="sc">*</span> genotype, <span class="at">data =</span> exp2j)</span>
<span id="cb500-2"><a href="factorial.html#cb500-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb500-3"><a href="factorial.html#cb500-3" aria-hidden="true" tabindex="-1"></a><span class="fu">tidy</span>(m1, <span class="at">conf.int =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb500-4"><a href="factorial.html#cb500-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">7</span>,<span class="dv">2</span>,<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb500-5"><a href="factorial.html#cb500-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
std.error
</th>
<th style="text-align:right;">
statistic
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
conf.low
</th>
<th style="text-align:right;">
conf.high
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
6.75
</td>
<td style="text-align:right;">
0.419
</td>
<td style="text-align:right;">
16.13
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
5.89
</td>
<td style="text-align:right;">
7.61
</td>
</tr>
<tr>
<td style="text-align:left;">
stimulationActive
</td>
<td style="text-align:right;">
3.45
</td>
<td style="text-align:right;">
0.592
</td>
<td style="text-align:right;">
5.83
</td>
<td style="text-align:right;">
0.0000045
</td>
<td style="text-align:right;">
2.23
</td>
<td style="text-align:right;">
4.67
</td>
</tr>
<tr>
<td style="text-align:left;">
genotypeKO
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.639
</td>
<td style="text-align:right;">
1.22
</td>
<td style="text-align:right;">
0.2337850
</td>
<td style="text-align:right;">
-0.54
</td>
<td style="text-align:right;">
2.10
</td>
</tr>
<tr>
<td style="text-align:left;">
stimulationActive:genotypeKO
</td>
<td style="text-align:right;">
-2.31
</td>
<td style="text-align:right;">
0.885
</td>
<td style="text-align:right;">
-2.61
</td>
<td style="text-align:right;">
0.0152305
</td>
<td style="text-align:right;">
-4.13
</td>
<td style="text-align:right;">
-0.48
</td>
</tr>
</tbody>
</table>
<p>The order of the factors in the model formula doesn’t matter for the coefficients, the estimated marginal means, or the contrasts. It can matter for ANOVA (more on this below) but not “tests after ANOVA.”</p>
<div class="sourceCode" id="cb501"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb501-1"><a href="factorial.html#cb501-1" aria-hidden="true" tabindex="-1"></a>m1_b <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> genotype <span class="sc">*</span> stimulation, <span class="at">data =</span> exp2j)</span>
<span id="cb501-2"><a href="factorial.html#cb501-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb501-3"><a href="factorial.html#cb501-3" aria-hidden="true" tabindex="-1"></a><span class="fu">tidy</span>(m1_b, <span class="at">conf.int =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb501-4"><a href="factorial.html#cb501-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">7</span>,<span class="dv">2</span>,<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb501-5"><a href="factorial.html#cb501-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
std.error
</th>
<th style="text-align:right;">
statistic
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
conf.low
</th>
<th style="text-align:right;">
conf.high
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
6.75
</td>
<td style="text-align:right;">
0.419
</td>
<td style="text-align:right;">
16.13
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
5.89
</td>
<td style="text-align:right;">
7.61
</td>
</tr>
<tr>
<td style="text-align:left;">
genotypeKO
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.639
</td>
<td style="text-align:right;">
1.22
</td>
<td style="text-align:right;">
0.2337850
</td>
<td style="text-align:right;">
-0.54
</td>
<td style="text-align:right;">
2.10
</td>
</tr>
<tr>
<td style="text-align:left;">
stimulationActive
</td>
<td style="text-align:right;">
3.45
</td>
<td style="text-align:right;">
0.592
</td>
<td style="text-align:right;">
5.83
</td>
<td style="text-align:right;">
0.0000045
</td>
<td style="text-align:right;">
2.23
</td>
<td style="text-align:right;">
4.67
</td>
</tr>
<tr>
<td style="text-align:left;">
genotypeKO:stimulationActive
</td>
<td style="text-align:right;">
-2.31
</td>
<td style="text-align:right;">
0.885
</td>
<td style="text-align:right;">
-2.61
</td>
<td style="text-align:right;">
0.0152305
</td>
<td style="text-align:right;">
-4.13
</td>
<td style="text-align:right;">
-0.48
</td>
</tr>
</tbody>
</table>
<p>The additive model is specified by the formula <code>y ~ A + B</code></p>
<div class="sourceCode" id="cb502"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb502-1"><a href="factorial.html#cb502-1" aria-hidden="true" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> stimulation <span class="sc">+</span> genotype, <span class="at">data =</span> exp2j)</span>
<span id="cb502-2"><a href="factorial.html#cb502-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb502-3"><a href="factorial.html#cb502-3" aria-hidden="true" tabindex="-1"></a><span class="fu">tidy</span>(m2, <span class="at">conf.int =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb502-4"><a href="factorial.html#cb502-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">7</span>,<span class="dv">2</span>,<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb502-5"><a href="factorial.html#cb502-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
term
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
std.error
</th>
<th style="text-align:right;">
statistic
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
conf.low
</th>
<th style="text-align:right;">
conf.high
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
7.27
</td>
<td style="text-align:right;">
0.408
</td>
<td style="text-align:right;">
17.82
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
6.43
</td>
<td style="text-align:right;">
8.10
</td>
</tr>
<tr>
<td style="text-align:left;">
stimulationActive
</td>
<td style="text-align:right;">
2.42
</td>
<td style="text-align:right;">
0.487
</td>
<td style="text-align:right;">
4.97
</td>
<td style="text-align:right;">
0.0000368
</td>
<td style="text-align:right;">
1.42
</td>
<td style="text-align:right;">
3.42
</td>
</tr>
<tr>
<td style="text-align:left;">
genotypeKO
</td>
<td style="text-align:right;">
-0.42
</td>
<td style="text-align:right;">
0.489
</td>
<td style="text-align:right;">
-0.86
</td>
<td style="text-align:right;">
0.3954703
</td>
<td style="text-align:right;">
-1.43
</td>
<td style="text-align:right;">
0.58
</td>
</tr>
</tbody>
</table>
</div>
<div id="using-the-emmeans-function-2" class="section level3" number="17.11.2">
<h3><span class="header-section-number">17.11.2</span> Using the emmeans function</h3>
<div id="conditional-means-table" class="section level4" number="17.11.2.1">
<h4><span class="header-section-number">17.11.2.1</span> Conditional means table</h4>
<div class="sourceCode" id="cb503"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb503-1"><a href="factorial.html#cb503-1" aria-hidden="true" tabindex="-1"></a><span class="co"># model m1 fit above</span></span>
<span id="cb503-2"><a href="factorial.html#cb503-2" aria-hidden="true" tabindex="-1"></a><span class="co"># m1 &lt;- lm(glucose_uptake ~ stimulation * genotype, data = exp2j)</span></span>
<span id="cb503-3"><a href="factorial.html#cb503-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb503-4"><a href="factorial.html#cb503-4" aria-hidden="true" tabindex="-1"></a>m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;stimulation&quot;</span>, <span class="st">&quot;genotype&quot;</span>))</span>
<span id="cb503-5"><a href="factorial.html#cb503-5" aria-hidden="true" tabindex="-1"></a>m1_emm</span></code></pre></div>
<pre><code>##  stimulation genotype emmean    SE df lower.CL upper.CL
##  Rest        WT         6.75 0.419 25     5.89     7.61
##  Active      WT        10.20 0.419 25     9.34    11.06
##  Rest        KO         7.53 0.483 25     6.53     8.53
##  Active      KO         8.67 0.447 25     7.75     9.59
## 
## Confidence level used: 0.95</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Printing the <code>emmeans</code> object displays useful information. Here, this information includes the confidence level used. If the object is printed using <code>kable() %&gt;% kable_styling()</code> (as in the “Inference” and “Understanding” sections above), only the table is printed and the additional information is lost.</li>
<li><code>emmeans</code> computes the modeled means of all combinations of the levels of the factor variables specified in <code>specs</code>.</li>
<li>If there are two factor variables in the model, and both are passed to <code>specs</code>, then the modeled means of all combinations of the levels of the two variables are computed. If only one factor variable is passed, then the marginal means (averaged over all levels of the missing factor) are computed (see below).</li>
</ol>
</div>
<div id="marginal-means" class="section level4" number="17.11.2.2">
<h4><span class="header-section-number">17.11.2.2</span> Marginal means</h4>
<div class="sourceCode" id="cb505"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb505-1"><a href="factorial.html#cb505-1" aria-hidden="true" tabindex="-1"></a>m1_emm_stimulation <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;stimulation&quot;</span>))</span></code></pre></div>
<pre><code>## NOTE: Results may be misleading due to involvement in interactions</code></pre>
<div class="sourceCode" id="cb507"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb507-1"><a href="factorial.html#cb507-1" aria-hidden="true" tabindex="-1"></a>m1_emm_stimulation</span></code></pre></div>
<pre><code>##  stimulation emmean    SE df lower.CL upper.CL
##  Rest          7.14 0.320 25     6.48      7.8
##  Active        9.43 0.306 25     8.80     10.1
## 
## Results are averaged over the levels of: genotype 
## Confidence level used: 0.95</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>In a model with two crossed factors, <code>y ~ A * B</code>, the marginal means of the levels of <span class="math inline">\(\texttt{A}\)</span>, averaged over all levels of <span class="math inline">\(\texttt{B}\)</span> are computed by setting <code>specs =</code> to <span class="math inline">\(\texttt{A}\)</span> only.</li>
<li>Remember that the <code>specs</code> argument sets the values of the predictors for which we want a mean. By excluding <span class="math inline">\(\texttt{B}\)</span>, we don’t get the means of the levels of <span class="math inline">\(\texttt{A}\)</span> at each level of <span class="math inline">\(\texttt{B}\)</span> but averaged across the levels of <span class="math inline">\(\texttt{B}\)</span>.<code>emmeans</code> “knows” to average across the levels of <span class="math inline">\(\texttt{B}\)</span> because <span class="math inline">\(\texttt{B}\)</span> is in the model.</li>
</ol>
</div>
</div>
<div id="contrasts" class="section level3" number="17.11.3">
<h3><span class="header-section-number">17.11.3</span> Contrasts</h3>
<div id="all-pairwise-effects" class="section level4" number="17.11.3.1">
<h4><span class="header-section-number">17.11.3.1</span> All pairwise effects</h4>
<div class="sourceCode" id="cb509"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb509-1"><a href="factorial.html#cb509-1" aria-hidden="true" tabindex="-1"></a>m1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(m1_emm,</span>
<span id="cb509-2"><a href="factorial.html#cb509-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb509-3"><a href="factorial.html#cb509-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">adjust =</span> <span class="st">&quot;tukey&quot;</span>,</span>
<span id="cb509-4"><a href="factorial.html#cb509-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">level =</span> <span class="fl">0.95</span>) <span class="sc">%&gt;%</span></span>
<span id="cb509-5"><a href="factorial.html#cb509-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>) <span class="co"># add the 95% CIs</span></span>
<span id="cb509-6"><a href="factorial.html#cb509-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb509-7"><a href="factorial.html#cb509-7" aria-hidden="true" tabindex="-1"></a>m1_pairs</span></code></pre></div>
<pre><code>##  contrast              estimate    SE df lower.CL upper.CL t.ratio p.value
##  Active WT - Rest WT       3.45 0.592 25    1.820    5.076  5.826  &lt;.0001 
##  Rest KO - Rest WT         0.78 0.639 25   -0.978    2.539  1.220  0.6202 
##  Rest KO - Active WT      -2.67 0.639 25   -4.427   -0.910 -4.173  0.0017 
##  Active KO - Rest WT       1.92 0.613 25    0.236    3.606  3.136  0.0212 
##  Active KO - Active WT    -1.53 0.613 25   -3.212    0.158 -2.493  0.0857 
##  Active KO - Rest KO       1.14 0.659 25   -0.671    2.953  1.733  0.3287 
## 
## Confidence level used: 0.95 
## Conf-level adjustment: tukey method for comparing a family of 4 estimates 
## P value adjustment: tukey method for comparing a family of 4 estimates</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Note that printing the <code>contrast</code> object displays useful information, including the method of adjustment for multiple tests. If the object is printed using <code>kable() %&gt;% kable_styling()</code> (as in the “Inference” and “Understanding” sections above), only the table is printed and the additional information is lost.</li>
<li>The <code>method</code> argument is used to control the set of contrasts that are computed. See below.</li>
<li>The <code>adjust</code> argument controls if and how to adjust for multiple tests. Each <code>method</code> has a default adjustment method. See below.</li>
<li>The <code>level</code> argument controls the percentile boundaries of the confidence interval. The default is 0.95. Including this argument with this value makes this level transparent.</li>
</ol>
</div>
<div id="simple-effects" class="section level4" number="17.11.3.2">
<h4><span class="header-section-number">17.11.3.2</span> Simple effects</h4>
<div class="sourceCode" id="cb511"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb511-1"><a href="factorial.html#cb511-1" aria-hidden="true" tabindex="-1"></a>m1_simple <span class="ot">&lt;-</span> <span class="fu">contrast</span>(m1_emm,</span>
<span id="cb511-2"><a href="factorial.html#cb511-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb511-3"><a href="factorial.html#cb511-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb511-4"><a href="factorial.html#cb511-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb511-5"><a href="factorial.html#cb511-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">adjust =</span> <span class="st">&quot;fdr&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb511-6"><a href="factorial.html#cb511-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>) <span class="co"># add the 95% CIs</span></span>
<span id="cb511-7"><a href="factorial.html#cb511-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb511-8"><a href="factorial.html#cb511-8" aria-hidden="true" tabindex="-1"></a>m1_simple <span class="sc">%&gt;%</span></span>
<span id="cb511-9"><a href="factorial.html#cb511-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  mutate(p.value = fmt_p_value_rmd(p.value)) %&gt;%</span></span>
<span id="cb511-10"><a href="factorial.html#cb511-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">4</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb511-11"><a href="factorial.html#cb511-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
genotype
</th>
<th style="text-align:left;">
stimulation
</th>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
WT
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Active - Rest
</td>
<td style="text-align:right;">
3.45
</td>
<td style="text-align:right;">
0.592
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
1.86
</td>
<td style="text-align:right;">
5.04
</td>
<td style="text-align:right;">
5.83
</td>
<td style="text-align:right;">
0.0000
</td>
</tr>
<tr>
<td style="text-align:left;">
KO
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Active - Rest
</td>
<td style="text-align:right;">
1.14
</td>
<td style="text-align:right;">
0.659
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-0.63
</td>
<td style="text-align:right;">
2.91
</td>
<td style="text-align:right;">
1.73
</td>
<td style="text-align:right;">
0.1273
</td>
</tr>
<tr>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Rest
</td>
<td style="text-align:left;">
KO - WT
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.639
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-0.94
</td>
<td style="text-align:right;">
2.50
</td>
<td style="text-align:right;">
1.22
</td>
<td style="text-align:right;">
0.2338
</td>
</tr>
<tr>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
Active
</td>
<td style="text-align:left;">
KO - WT
</td>
<td style="text-align:right;">
-1.53
</td>
<td style="text-align:right;">
0.613
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-3.18
</td>
<td style="text-align:right;">
0.12
</td>
<td style="text-align:right;">
-2.49
</td>
<td style="text-align:right;">
0.0393
</td>
</tr>
</tbody>
</table>
</div>
<div id="interaction-effects" class="section level4" number="17.11.3.3">
<h4><span class="header-section-number">17.11.3.3</span> Interaction effects</h4>
<div class="sourceCode" id="cb512"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb512-1"><a href="factorial.html#cb512-1" aria-hidden="true" tabindex="-1"></a>m1_ixn <span class="ot">&lt;-</span> <span class="fu">contrast</span>(m1_emm,</span>
<span id="cb512-2"><a href="factorial.html#cb512-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">interaction =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb512-3"><a href="factorial.html#cb512-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb512-4"><a href="factorial.html#cb512-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>) <span class="co"># add the 95% CIs</span></span>
<span id="cb512-5"><a href="factorial.html#cb512-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb512-6"><a href="factorial.html#cb512-6" aria-hidden="true" tabindex="-1"></a>m1_ixn <span class="sc">%&gt;%</span></span>
<span id="cb512-7"><a href="factorial.html#cb512-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">4</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb512-8"><a href="factorial.html#cb512-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
stimulation_revpairwise
</th>
<th style="text-align:left;">
genotype_revpairwise
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Active - Rest
</td>
<td style="text-align:left;">
KO - WT
</td>
<td style="text-align:right;">
-2.31
</td>
<td style="text-align:right;">
0.885
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-4.13
</td>
<td style="text-align:right;">
-0.48
</td>
<td style="text-align:right;">
-2.61
</td>
<td style="text-align:right;">
0.0152
</td>
</tr>
</tbody>
</table>
</div>
<div id="method-argument-1" class="section level4" number="17.11.3.4">
<h4><span class="header-section-number">17.11.3.4</span> Method argument</h4>
<div class="sourceCode" id="cb513"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb513-1"><a href="factorial.html#cb513-1" aria-hidden="true" tabindex="-1"></a>m1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(m1_emm,</span>
<span id="cb513-2"><a href="factorial.html#cb513-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;pairwise&quot;</span>,</span>
<span id="cb513-3"><a href="factorial.html#cb513-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">adjust =</span> <span class="st">&quot;fdr&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb513-4"><a href="factorial.html#cb513-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>) <span class="co"># add the 95% CIs</span></span>
<span id="cb513-5"><a href="factorial.html#cb513-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb513-6"><a href="factorial.html#cb513-6" aria-hidden="true" tabindex="-1"></a>m1_pairs <span class="sc">%&gt;%</span></span>
<span id="cb513-7"><a href="factorial.html#cb513-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb513-8"><a href="factorial.html#cb513-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rest WT - Active WT
</td>
<td style="text-align:right;">
-3.4482471
</td>
<td style="text-align:right;">
0.5918936
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-5.1439551
</td>
<td style="text-align:right;">
-1.7525392
</td>
<td style="text-align:right;">
-5.825789
</td>
<td style="text-align:right;">
0.0000269
</td>
</tr>
<tr>
<td style="text-align:left;">
Rest WT - Rest KO
</td>
<td style="text-align:right;">
-0.7800781
</td>
<td style="text-align:right;">
0.6393182
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-2.6116520
</td>
<td style="text-align:right;">
1.0514958
</td>
<td style="text-align:right;">
-1.220172
</td>
<td style="text-align:right;">
0.2337850
</td>
</tr>
<tr>
<td style="text-align:left;">
Rest WT - Active KO
</td>
<td style="text-align:right;">
-1.9211230
</td>
<td style="text-align:right;">
0.6126681
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-3.6763474
</td>
<td style="text-align:right;">
-0.1658985
</td>
<td style="text-align:right;">
-3.135667
</td>
<td style="text-align:right;">
0.0086970
</td>
</tr>
<tr>
<td style="text-align:left;">
Active WT - Rest KO
</td>
<td style="text-align:right;">
2.6681690
</td>
<td style="text-align:right;">
0.6393182
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
0.8365951
</td>
<td style="text-align:right;">
4.4997430
</td>
<td style="text-align:right;">
4.173460
</td>
<td style="text-align:right;">
0.0009510
</td>
</tr>
<tr>
<td style="text-align:left;">
Active WT - Active KO
</td>
<td style="text-align:right;">
1.5271242
</td>
<td style="text-align:right;">
0.6126681
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-0.2281003
</td>
<td style="text-align:right;">
3.2823486
</td>
<td style="text-align:right;">
2.492580
</td>
<td style="text-align:right;">
0.0295018
</td>
</tr>
<tr>
<td style="text-align:left;">
Rest KO - Active KO
</td>
<td style="text-align:right;">
-1.1410449
</td>
<td style="text-align:right;">
0.6585984
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-3.0278545
</td>
<td style="text-align:right;">
0.7457647
</td>
<td style="text-align:right;">
-1.732535
</td>
<td style="text-align:right;">
0.1145997
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb514"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb514-1"><a href="factorial.html#cb514-1" aria-hidden="true" tabindex="-1"></a>m1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(m1_emm,</span>
<span id="cb514-2"><a href="factorial.html#cb514-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;trt.vs.ctrl&quot;</span>,</span>
<span id="cb514-3"><a href="factorial.html#cb514-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb514-4"><a href="factorial.html#cb514-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>) <span class="co"># add the 95% CIs</span></span>
<span id="cb514-5"><a href="factorial.html#cb514-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb514-6"><a href="factorial.html#cb514-6" aria-hidden="true" tabindex="-1"></a>m1_pairs <span class="sc">%&gt;%</span></span>
<span id="cb514-7"><a href="factorial.html#cb514-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb514-8"><a href="factorial.html#cb514-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Active WT - Rest WT
</td>
<td style="text-align:right;">
3.4482471
</td>
<td style="text-align:right;">
0.5918936
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
2.2292194
</td>
<td style="text-align:right;">
4.667275
</td>
<td style="text-align:right;">
5.825789
</td>
<td style="text-align:right;">
0.0000045
</td>
</tr>
<tr>
<td style="text-align:left;">
Rest KO - Rest WT
</td>
<td style="text-align:right;">
0.7800781
</td>
<td style="text-align:right;">
0.6393182
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
-0.5366223
</td>
<td style="text-align:right;">
2.096779
</td>
<td style="text-align:right;">
1.220172
</td>
<td style="text-align:right;">
0.2337850
</td>
</tr>
<tr>
<td style="text-align:left;">
Active KO - Rest WT
</td>
<td style="text-align:right;">
1.9211230
</td>
<td style="text-align:right;">
0.6126681
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
0.6593094
</td>
<td style="text-align:right;">
3.182936
</td>
<td style="text-align:right;">
3.135667
</td>
<td style="text-align:right;">
0.0043485
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li><code>method = "pairwise"</code> and <code>method = "revpairwise"</code> compute all pairwise comparisons. I prefer “revpairwise” because the contrasts that include the reference are in the direction non-reference minus reference.</li>
<li><code>method = "trt.vs.ctrl"</code> gives a very flattened picture of the model results and constrains what we can infer from the results.</li>
</ol>
</div>
<div id="adjustment-for-multiple-tests-1" class="section level4" number="17.11.3.5">
<h4><span class="header-section-number">17.11.3.5</span> Adjustment for multiple tests</h4>
<p>set the <code>adjust =</code> argument to</p>
<ol style="list-style-type: decimal">
<li>“none” – no adjustment</li>
<li>“dunnettx” – <a href="https://en.wikipedia.org/wiki/Dunnett%27s_test" target="_blank">Dunnett’s test</a> is a method used when comparing all treatments to a single control. For a factorial design, this only makes sense for a flattened analysis.</li>
<li>“tukey” – <a href="https://en.wikipedia.org/wiki/Tukey%27s_range_test" target="_blank">Tukey’s HSD method</a> is a method used to compare all pairwise comparisons.</li>
<li>“bonferroni” – <a href="https://en.wikipedia.org/wiki/Bonferroni_correction" target="_blank">Bonferroni</a> is a general purpose method to compare any set of multiple tests. The test is conservative. A better method is “holm”</li>
<li>“holm” – <a href="https://en.wikipedia.org/wiki/Holm–Bonferroni_method" target="_blank">Holm-Bonferroni</a> is a general purpose method like the Bonferroni but is more powerful.</li>
<li>“fdr” – controls the <a href="https://en.wikipedia.org/wiki/False_discovery_rate" target="_blank">false discovery rate</a> not the Type I error rate for a family of tests. One might use this in an exploratory experiment.</li>
<li>“mvt” – based on the multivariate <em>t</em> distribution and using covariance structure of the variables.</li>
</ol>
</div>
</div>
<div id="practice-safe-anova" class="section level3" number="17.11.4">
<h3><span class="header-section-number">17.11.4</span> Practice safe ANOVA</h3>
<p>Many researchers are unaware of the different ways of computing the sum of squares in an ANOVA table. The <em>F</em>-value and <em>p</em>-value are both functions of the sum of squares. When the cells of a two-way ANOVA are not balanced (different sample sizes among the cells), the different ways of computing the sum of squares can matter. There is a great deal of sound and fury about which methods should be used and which avoided. For many research questions and the experiments used to address these, these arguments are moot because the ANOVA table is simply unnecessary for inference. Instead, fit the linear model and compute the contrasts of interest.</p>
<p>If your reviewer/advisor/boss wants an ANOVA table and wants this table to match that in Graphpad Prism or JMP, then you need a table computed from Type III sum of squares. If using a linear model to compute the ANOVA (remember that ANOVA was developed without fitting linear models), then the linear model needs to be fit using a model matrix constructed with indicator variables using effects coding. There are two safe ways to do this in R.</p>
<div id="the-afex-aov_4-function-1" class="section level4" number="17.11.4.1">
<h4><span class="header-section-number">17.11.4.1</span> The afex aov_4 function</h4>
<div class="sourceCode" id="cb515"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb515-1"><a href="factorial.html#cb515-1" aria-hidden="true" tabindex="-1"></a><span class="co"># .I is a data.table function that returns the row number</span></span>
<span id="cb515-2"><a href="factorial.html#cb515-2" aria-hidden="true" tabindex="-1"></a>exp2j[, fake_id <span class="sc">:</span><span class="er">=</span> <span class="fu">paste</span>(<span class="st">&quot;mouse&quot;</span>, .I)]</span>
<span id="cb515-3"><a href="factorial.html#cb515-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb515-4"><a href="factorial.html#cb515-4" aria-hidden="true" tabindex="-1"></a>m1_aov4 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> stimulation <span class="sc">*</span> genotype <span class="sc">+</span></span>
<span id="cb515-5"><a href="factorial.html#cb515-5" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span><span class="sc">|</span>fake_id),</span>
<span id="cb515-6"><a href="factorial.html#cb515-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp2j)</span>
<span id="cb515-7"><a href="factorial.html#cb515-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb515-8"><a href="factorial.html#cb515-8" aria-hidden="true" tabindex="-1"></a><span class="fu">nice</span>(m1_aov4, <span class="at">MSE =</span> <span class="cn">FALSE</span>)[,<span class="sc">-</span><span class="dv">4</span>] <span class="sc">%&gt;%</span> <span class="co"># delete ges column</span></span>
<span id="cb515-9"><a href="factorial.html#cb515-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb515-10"><a href="factorial.html#cb515-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Effect
</th>
<th style="text-align:left;">
df
</th>
<th style="text-align:left;">
F
</th>
<th style="text-align:left;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
stimulation
</td>
<td style="text-align:left;">
1, 25
</td>
<td style="text-align:left;">
26.86 ***
</td>
<td style="text-align:left;">
&lt;.0001
</td>
</tr>
<tr>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
1, 25
</td>
<td style="text-align:left;">
0.71
</td>
<td style="text-align:left;">
.41
</td>
</tr>
<tr>
<td style="text-align:left;">
stimulation:genotype
</td>
<td style="text-align:left;">
1, 25
</td>
<td style="text-align:left;">
6.79 *
</td>
<td style="text-align:left;">
.02
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The afex package has three function names for generating the same ANOVA table and statistics – here I’m using <code>aov_4</code> because this functions uses a linear model formula argument (specifically, that used in the lme4 package), which is consistent with the rest of this text.</li>
<li>The formula includes the addition of a random factor (<code>(1|id)</code>) even though there really is no random factor in this model. See <a href="lmm.html#lmm">Models with random factors – Blocking and pseudoreplication</a> for more information on random factors. The random factor (the factor variable “id” created in the line before the fit model line) identifies the individual mouse from which the response variable was measured. Because the response was only measured once on each individual mouse, “id” is not really a random factor but the addition of this in the model formula is necessary for the <code>aov_4</code> function to work.</li>
</ol>
</div>
<div id="twoway-car-anova" class="section level4" number="17.11.4.2">
<h4><span class="header-section-number">17.11.4.2</span> The car Anova function</h4>
<div class="sourceCode" id="cb516"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb516-1"><a href="factorial.html#cb516-1" aria-hidden="true" tabindex="-1"></a>type3 <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">stimulation =</span> contr.sum,</span>
<span id="cb516-2"><a href="factorial.html#cb516-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">genotype =</span> contr.sum)</span>
<span id="cb516-3"><a href="factorial.html#cb516-3" aria-hidden="true" tabindex="-1"></a>m1_type3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> stimulation <span class="sc">*</span> genotype,</span>
<span id="cb516-4"><a href="factorial.html#cb516-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> exp2j,</span>
<span id="cb516-5"><a href="factorial.html#cb516-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">contrasts =</span> type3)</span>
<span id="cb516-6"><a href="factorial.html#cb516-6" aria-hidden="true" tabindex="-1"></a><span class="fu">Anova</span>(m1_type3, <span class="at">type=</span><span class="st">&quot;3&quot;</span>)[<span class="sc">-</span><span class="dv">1</span>, <span class="sc">-</span><span class="dv">1</span>] <span class="sc">%&gt;%</span> <span class="co"># delete row 1 and col 1</span></span>
<span id="cb516-7"><a href="factorial.html#cb516-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb516-8"><a href="factorial.html#cb516-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Df
</th>
<th style="text-align:right;">
F value
</th>
<th style="text-align:right;">
Pr(&gt;F)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
stimulation
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
26.86
</td>
<td style="text-align:right;">
0.00002
</td>
</tr>
<tr>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.71
</td>
<td style="text-align:right;">
0.40686
</td>
</tr>
<tr>
<td style="text-align:left;">
stimulation:genotype
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
6.79
</td>
<td style="text-align:right;">
0.01523
</td>
</tr>
<tr>
<td style="text-align:left;">
Residuals
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li><code>car::Anova</code> has arguments for reporting the Type II and Type III sum of squares.</li>
<li>Background: The default model matrix in the <code>lm</code> function uses dummy (or treatment) coding. For a Type 3 SS ANOVA (the kind that matches that in Graphpad Prism or JMP), we need to tell <code>lm</code> to use sum (or deviation) coding.</li>
<li>The best practice method for changing the contrasts in the model matrix is using the <code>contrasts</code> argument within the <code>lm</code> function, as in the code above to fit <code>m1_type3</code>. This is the safest practice because this sets the contrasts only for this specific fit.</li>
<li>The coefficients of <code>m1_type3</code> will be different from those using the default contrasts. Except when computing a Type III ANOVA, this text uses the default contrasts throughout because the coefficients make sense for the kinds of experiments in experimental biology. If using sum (“type III”) coding, the intercept will be the grand mean and the coefficients of the non-reference levels (the effects) will be their deviations from the grand mean. I don’t find this definition of “effects” very useful for most experiments in biology (but “useful” is largely a function of “used to”).</li>
<li>The contrasts (differences in the means among pairs of groups) in the contrast table will be the same, regardless of the contrast coding.</li>
</ol>
<div style="background-color:#cccccc; text-align:left; vertical-align: middle; padding:20px 47px;">
<p>Danger!. Many online sites suggest this bit of code before a Type III ANOVA using car::Anova:</p>
<p><code>options(contrasts = c("contr.sum", "contr.poly")</code></p>
<p>If you’re reading this book, you almost certainly don’t want to do this because this code resets how R computes coefficients of linear models and SS of ANOVA tables. This will effect all future analysis until the contrasts are set to something else or a new R session is started.</p>
</div>
</div>
</div>
<div id="better-to-avoid-these" class="section level3" number="17.11.5">
<h3><span class="header-section-number">17.11.5</span> Better to avoid these</h3>
<div class="sourceCode" id="cb517"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb517-1"><a href="factorial.html#cb517-1" aria-hidden="true" tabindex="-1"></a>m1_aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(glucose_uptake <span class="sc">~</span> stimulation <span class="sc">*</span> genotype,</span>
<span id="cb517-2"><a href="factorial.html#cb517-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> exp2j)</span>
<span id="cb517-3"><a href="factorial.html#cb517-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m1_aov)</span></code></pre></div>
<pre><code>##                      Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## stimulation           1  41.75   41.75  29.796 1.14e-05 ***
## genotype              1   1.28    1.28   0.913   0.3485    
## stimulation:genotype  1   9.51    9.51   6.789   0.0152 *  
## Residuals            25  35.03    1.40                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb519"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb519-1"><a href="factorial.html#cb519-1" aria-hidden="true" tabindex="-1"></a><span class="co"># same as exp2j_m1 in the Example 1 section</span></span>
<span id="cb519-2"><a href="factorial.html#cb519-2" aria-hidden="true" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> stimulation <span class="sc">*</span> genotype,</span>
<span id="cb519-3"><a href="factorial.html#cb519-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> exp2j)</span>
<span id="cb519-4"><a href="factorial.html#cb519-4" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(m1)</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Response: glucose_uptake
##                      Df Sum Sq Mean Sq F value    Pr(&gt;F)    
## stimulation           1 41.755  41.755 29.7959 1.143e-05 ***
## genotype              1  1.279   1.279  0.9128   0.34853    
## stimulation:genotype  1  9.514   9.514  6.7890   0.01523 *  
## Residuals            25 35.034   1.401                      
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Note the differences from the the afex table and the car Anova table. The <em>p</em>-values for the interaction term is the same – this will always be the case for the highest order interaction term in the model. The <em>p</em>-values for <span class="math inline">\(\texttt{stimulation}\)</span> and <span class="math inline">\(\texttt{genotype}\)</span> terms are different. The difference will be a function of the degree of imbalance. Here, this difference doesn’t make a difference for inference. In some data sets, the difference is catastrophic.</li>
<li>Many introduction to statistics textbooks and websites teach the base R <code>aov</code> function. This function is only useful if 1) the data are balanced and 2) we care about ANOVA. Don’t use this. If you need the ANOVA table use either afex or car Anova.</li>
<li>The base R <code>anova</code> is useful if you know what you are doing with it. Otherwise, use either afex or car Anova.</li>
</ol>
</div>
</div>
<div id="hidden-code-4" class="section level2" number="17.12">
<h2><span class="header-section-number">17.12</span> Hidden Code</h2>
<div id="import-exp2j-example-1" class="section level3" number="17.12.1">
<h3><span class="header-section-number">17.12.1</span> Import exp2j (Example 1)</h3>
<div class="sourceCode" id="cb521"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb521-1"><a href="factorial.html#cb521-1" aria-hidden="true" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">&quot;TLR9 and beclin 1 crosstalk regulates muscle AMPK activation in exercise&quot;</span></span>
<span id="cb521-2"><a href="factorial.html#cb521-2" aria-hidden="true" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">&quot;41586_2020_1992_MOESM4_ESM.xlsx&quot;</span></span>
<span id="cb521-3"><a href="factorial.html#cb521-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb521-4"><a href="factorial.html#cb521-4" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb521-5"><a href="factorial.html#cb521-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb521-6"><a href="factorial.html#cb521-6" aria-hidden="true" tabindex="-1"></a>treatment_levels  <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;WT Rest&quot;</span>,</span>
<span id="cb521-7"><a href="factorial.html#cb521-7" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&quot;WT Active&quot;</span>,</span>
<span id="cb521-8"><a href="factorial.html#cb521-8" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&quot;KO Rest&quot;</span>,</span>
<span id="cb521-9"><a href="factorial.html#cb521-9" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&quot;KO Active&quot;</span>)</span>
<span id="cb521-10"><a href="factorial.html#cb521-10" aria-hidden="true" tabindex="-1"></a>exp2j_wide <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb521-11"><a href="factorial.html#cb521-11" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sheet =</span> <span class="st">&quot;2j&quot;</span>,</span>
<span id="cb521-12"><a href="factorial.html#cb521-12" aria-hidden="true" tabindex="-1"></a>                         <span class="at">range =</span> <span class="st">&quot;A5:D13&quot;</span>,</span>
<span id="cb521-13"><a href="factorial.html#cb521-13" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_names =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb521-14"><a href="factorial.html#cb521-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span>
<span id="cb521-15"><a href="factorial.html#cb521-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb521-16"><a href="factorial.html#cb521-16" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(exp2j_wide) <span class="ot">&lt;-</span> treatment_levels</span>
<span id="cb521-17"><a href="factorial.html#cb521-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb521-18"><a href="factorial.html#cb521-18" aria-hidden="true" tabindex="-1"></a>exp2j <span class="ot">&lt;-</span> <span class="fu">melt</span>(exp2j_wide,</span>
<span id="cb521-19"><a href="factorial.html#cb521-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">measure.vars =</span> treatment_levels,</span>
<span id="cb521-20"><a href="factorial.html#cb521-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">variable.name =</span> <span class="st">&quot;treatment&quot;</span>,</span>
<span id="cb521-21"><a href="factorial.html#cb521-21" aria-hidden="true" tabindex="-1"></a>              <span class="at">value.name =</span> <span class="st">&quot;glucose_uptake&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb521-22"><a href="factorial.html#cb521-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="co"># danger!</span></span>
<span id="cb521-23"><a href="factorial.html#cb521-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb521-24"><a href="factorial.html#cb521-24" aria-hidden="true" tabindex="-1"></a>exp2j[, <span class="fu">c</span>(<span class="st">&quot;genotype&quot;</span>, <span class="st">&quot;stimulation&quot;</span>) <span class="sc">:</span><span class="er">=</span> <span class="fu">tstrsplit</span>(treatment,</span>
<span id="cb521-25"><a href="factorial.html#cb521-25" aria-hidden="true" tabindex="-1"></a>                                                  <span class="st">&quot; &quot;</span>,</span>
<span id="cb521-26"><a href="factorial.html#cb521-26" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">fixed =</span> <span class="cn">TRUE</span>)]</span>
<span id="cb521-27"><a href="factorial.html#cb521-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb521-28"><a href="factorial.html#cb521-28" aria-hidden="true" tabindex="-1"></a>genotype_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;WT&quot;</span>, <span class="st">&quot;KO&quot;</span>)</span>
<span id="cb521-29"><a href="factorial.html#cb521-29" aria-hidden="true" tabindex="-1"></a>stimulation_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Rest&quot;</span>, <span class="st">&quot;Active&quot;</span>)</span>
<span id="cb521-30"><a href="factorial.html#cb521-30" aria-hidden="true" tabindex="-1"></a>exp2j[, genotype <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(genotype,</span>
<span id="cb521-31"><a href="factorial.html#cb521-31" aria-hidden="true" tabindex="-1"></a>                           <span class="at">levels =</span> genotype_levels)]</span>
<span id="cb521-32"><a href="factorial.html#cb521-32" aria-hidden="true" tabindex="-1"></a>exp2j[, stimulation <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(stimulation,</span>
<span id="cb521-33"><a href="factorial.html#cb521-33" aria-hidden="true" tabindex="-1"></a>                              <span class="at">levels =</span> stimulation_levels)]</span>
<span id="cb521-34"><a href="factorial.html#cb521-34" aria-hidden="true" tabindex="-1"></a><span class="co"># View(exp2j)</span></span></code></pre></div>
</div>
<div id="import-exp3e-lesian-area-data-example-2" class="section level3" number="17.12.2">
<h3><span class="header-section-number">17.12.2</span> Import exp3e lesian area data (Example 2)</h3>
<div class="sourceCode" id="cb522"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb522-1"><a href="factorial.html#cb522-1" aria-hidden="true" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">&quot;XX sex chromosome complement promotes atherosclerosis in mice&quot;</span></span>
<span id="cb522-2"><a href="factorial.html#cb522-2" aria-hidden="true" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">&quot;41467_2019_10462_MOESM6_ESM.xlsx&quot;</span></span>
<span id="cb522-3"><a href="factorial.html#cb522-3" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb522-4"><a href="factorial.html#cb522-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb522-5"><a href="factorial.html#cb522-5" aria-hidden="true" tabindex="-1"></a><span class="co"># fig 3e</span></span>
<span id="cb522-6"><a href="factorial.html#cb522-6" aria-hidden="true" tabindex="-1"></a>exp3e_wide <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb522-7"><a href="factorial.html#cb522-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">sheet =</span> <span class="st">&quot;Figure 3E&quot;</span>,</span>
<span id="cb522-8"><a href="factorial.html#cb522-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">range =</span> <span class="st">&quot;A3:K4&quot;</span>,</span>
<span id="cb522-9"><a href="factorial.html#cb522-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb522-10"><a href="factorial.html#cb522-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>() <span class="sc">%&gt;%</span></span>
<span id="cb522-11"><a href="factorial.html#cb522-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transpose</span>(<span class="at">make.names =</span> <span class="dv">1</span>)</span>
<span id="cb522-12"><a href="factorial.html#cb522-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb522-13"><a href="factorial.html#cb522-13" aria-hidden="true" tabindex="-1"></a>sex_levels <span class="ot">&lt;-</span> <span class="fu">colnames</span>(exp3e_wide)</span>
<span id="cb522-14"><a href="factorial.html#cb522-14" aria-hidden="true" tabindex="-1"></a>exp3e <span class="ot">&lt;-</span> <span class="fu">melt</span>(exp3e_wide,</span>
<span id="cb522-15"><a href="factorial.html#cb522-15" aria-hidden="true" tabindex="-1"></a>              <span class="at">measure.vars =</span> sex_levels,</span>
<span id="cb522-16"><a href="factorial.html#cb522-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">variable.name =</span> <span class="st">&quot;sex&quot;</span>,</span>
<span id="cb522-17"><a href="factorial.html#cb522-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">value.name =</span> <span class="st">&quot;lesian_area&quot;</span>)</span>
<span id="cb522-18"><a href="factorial.html#cb522-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb522-19"><a href="factorial.html#cb522-19" aria-hidden="true" tabindex="-1"></a><span class="co"># convert lesian_area to mm^2 from µm^2</span></span>
<span id="cb522-20"><a href="factorial.html#cb522-20" aria-hidden="true" tabindex="-1"></a>exp3e[, lesian_area <span class="sc">:</span><span class="er">=</span> lesian_area<span class="sc">/</span><span class="dv">10</span><span class="sc">^</span><span class="dv">6</span>]</span>
<span id="cb522-21"><a href="factorial.html#cb522-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb522-22"><a href="factorial.html#cb522-22" aria-hidden="true" tabindex="-1"></a><span class="co"># convert sex variable to factor</span></span>
<span id="cb522-23"><a href="factorial.html#cb522-23" aria-hidden="true" tabindex="-1"></a>exp3e[, sex <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(sex,</span>
<span id="cb522-24"><a href="factorial.html#cb522-24" aria-hidden="true" tabindex="-1"></a>                        <span class="at">levels =</span> sex_levels)]</span>
<span id="cb522-25"><a href="factorial.html#cb522-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb522-26"><a href="factorial.html#cb522-26" aria-hidden="true" tabindex="-1"></a><span class="co"># create chromosome column and convert to factor</span></span>
<span id="cb522-27"><a href="factorial.html#cb522-27" aria-hidden="true" tabindex="-1"></a>chromosome_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;XX&quot;</span>, <span class="st">&quot;XY&quot;</span>)</span>
<span id="cb522-28"><a href="factorial.html#cb522-28" aria-hidden="true" tabindex="-1"></a>exp3e[, chromosome <span class="sc">:</span><span class="er">=</span> <span class="fu">rep</span>(<span class="fu">rep</span>(chromosome_levels, <span class="at">each =</span> <span class="dv">5</span>), <span class="dv">2</span>)]</span>
<span id="cb522-29"><a href="factorial.html#cb522-29" aria-hidden="true" tabindex="-1"></a>exp3e[, chromosome <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(chromosome,</span>
<span id="cb522-30"><a href="factorial.html#cb522-30" aria-hidden="true" tabindex="-1"></a>                             <span class="at">levels =</span> chromosome_levels)]</span>
<span id="cb522-31"><a href="factorial.html#cb522-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb522-32"><a href="factorial.html#cb522-32" aria-hidden="true" tabindex="-1"></a><span class="co"># researchers treatment levels</span></span>
<span id="cb522-33"><a href="factorial.html#cb522-33" aria-hidden="true" tabindex="-1"></a>exp3e[, treatment <span class="sc">:</span><span class="er">=</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&quot;FXX&quot;</span>, <span class="st">&quot;FXY&quot;</span>, <span class="st">&quot;MXX&quot;</span>, <span class="st">&quot;MXY&quot;</span>), <span class="at">each =</span> <span class="dv">5</span>)]</span>
<span id="cb522-34"><a href="factorial.html#cb522-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb522-35"><a href="factorial.html#cb522-35" aria-hidden="true" tabindex="-1"></a><span class="co"># two rows with missing response so delte these rows as there is </span></span>
<span id="cb522-36"><a href="factorial.html#cb522-36" aria-hidden="true" tabindex="-1"></a><span class="co"># no useful information in them</span></span>
<span id="cb522-37"><a href="factorial.html#cb522-37" aria-hidden="true" tabindex="-1"></a>exp3e <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(exp3e) <span class="co"># be careful with a global na.omit</span></span>
<span id="cb522-38"><a href="factorial.html#cb522-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb522-39"><a href="factorial.html#cb522-39" aria-hidden="true" tabindex="-1"></a><span class="co"># View(exp3e) # highlight and run to see data</span></span></code></pre></div>
</div>
<div id="import-exp1c-ja-data-example-3" class="section level3" number="17.12.3">
<h3><span class="header-section-number">17.12.3</span> Import Exp1c JA data (Example 3)</h3>
<div class="sourceCode" id="cb523"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb523-1"><a href="factorial.html#cb523-1" aria-hidden="true" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">&quot;Integration of two herbivore-induced plant volatiles results in synergistic effects on plant defense and resistance&quot;</span></span>
<span id="cb523-2"><a href="factorial.html#cb523-2" aria-hidden="true" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">&quot;Data for Dryad.xlsx&quot;</span></span>
<span id="cb523-3"><a href="factorial.html#cb523-3" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb523-4"><a href="factorial.html#cb523-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb523-5"><a href="factorial.html#cb523-5" aria-hidden="true" tabindex="-1"></a>exp1 <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb523-6"><a href="factorial.html#cb523-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">sheet =</span> <span class="st">&quot;Fig. 1&quot;</span>,</span>
<span id="cb523-7"><a href="factorial.html#cb523-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">range =</span> <span class="st">&quot;A3:K23&quot;</span>, <span class="co"># 1 blank column</span></span>
<span id="cb523-8"><a href="factorial.html#cb523-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">col_names =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb523-9"><a href="factorial.html#cb523-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>() <span class="sc">%&gt;%</span></span>
<span id="cb523-10"><a href="factorial.html#cb523-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">clean_names</span>()</span>
<span id="cb523-11"><a href="factorial.html#cb523-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb523-12"><a href="factorial.html#cb523-12" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(exp1, <span class="at">old =</span> <span class="st">&quot;treat&quot;</span>, <span class="at">new =</span> <span class="st">&quot;treatment&quot;</span>)</span>
<span id="cb523-13"><a href="factorial.html#cb523-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb523-14"><a href="factorial.html#cb523-14" aria-hidden="true" tabindex="-1"></a>exp1[treatment <span class="sc">==</span> <span class="st">&quot;Control&quot;</span>, <span class="fu">c</span>(<span class="st">&quot;hac&quot;</span>, <span class="st">&quot;indole&quot;</span>) <span class="sc">:</span><span class="er">=</span> <span class="fu">list</span>(<span class="st">&quot;HAC-&quot;</span>, <span class="st">&quot;Indole-&quot;</span>)]</span>
<span id="cb523-15"><a href="factorial.html#cb523-15" aria-hidden="true" tabindex="-1"></a>exp1[treatment <span class="sc">==</span> <span class="st">&quot;HAC&quot;</span>, <span class="fu">c</span>(<span class="st">&quot;hac&quot;</span>, <span class="st">&quot;indole&quot;</span>) <span class="sc">:</span><span class="er">=</span> <span class="fu">list</span>(<span class="st">&quot;HAC+&quot;</span>, <span class="st">&quot;Indole-&quot;</span>)]</span>
<span id="cb523-16"><a href="factorial.html#cb523-16" aria-hidden="true" tabindex="-1"></a>exp1[treatment <span class="sc">==</span> <span class="st">&quot;Indole&quot;</span>, <span class="fu">c</span>(<span class="st">&quot;hac&quot;</span>, <span class="st">&quot;indole&quot;</span>) <span class="sc">:</span><span class="er">=</span> <span class="fu">list</span>(<span class="st">&quot;HAC-&quot;</span>, <span class="st">&quot;Indole+&quot;</span>)]</span>
<span id="cb523-17"><a href="factorial.html#cb523-17" aria-hidden="true" tabindex="-1"></a>exp1[treatment <span class="sc">==</span> <span class="st">&quot;HAC+Indole&quot;</span>, <span class="fu">c</span>(<span class="st">&quot;hac&quot;</span>, <span class="st">&quot;indole&quot;</span>) <span class="sc">:</span><span class="er">=</span> <span class="fu">list</span>(<span class="st">&quot;HAC+&quot;</span>, <span class="st">&quot;Indole+&quot;</span>)]</span>
<span id="cb523-18"><a href="factorial.html#cb523-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb523-19"><a href="factorial.html#cb523-19" aria-hidden="true" tabindex="-1"></a>treatment_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Control&quot;</span>, <span class="st">&quot;HAC&quot;</span>, <span class="st">&quot;Indole&quot;</span>, <span class="st">&quot;HAC+Indole&quot;</span>)</span>
<span id="cb523-20"><a href="factorial.html#cb523-20" aria-hidden="true" tabindex="-1"></a>exp1[, treatment <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(treatment,</span>
<span id="cb523-21"><a href="factorial.html#cb523-21" aria-hidden="true" tabindex="-1"></a>                           <span class="at">levels =</span> treatment_levels)]</span>
<span id="cb523-22"><a href="factorial.html#cb523-22" aria-hidden="true" tabindex="-1"></a>exp1[, hac <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(hac)] <span class="co"># levels in correct order</span></span>
<span id="cb523-23"><a href="factorial.html#cb523-23" aria-hidden="true" tabindex="-1"></a>exp1[, indole <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(indole)] <span class="co"># levels in correct order</span></span>
<span id="cb523-24"><a href="factorial.html#cb523-24" aria-hidden="true" tabindex="-1"></a><span class="co"># View(exp1)</span></span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="covariates.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="part-vii-expanding-the-linear-model.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/chapters/34-factorial.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Walker-elementary-statistical-modeling-draft.pdf", "Walker-elementary-statistical-modeling-draft.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
