<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 18 Models with random factors – linear mixed models | Elements of Statistical Modeling for Experimental Biology</title>
  <meta name="description" content="A first course in statistical modeling for experimental biology researchers" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 18 Models with random factors – linear mixed models | Elements of Statistical Modeling for Experimental Biology" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A first course in statistical modeling for experimental biology researchers" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 18 Models with random factors – linear mixed models | Elements of Statistical Modeling for Experimental Biology" />
  
  <meta name="twitter:description" content="A first course in statistical modeling for experimental biology researchers" />
  

<meta name="author" content="Copyright 2018 Jeffrey A. Walker" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="part-vii-expanding-the-linear-model.html"/>
<link rel="next" href="pre-post.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="libs/d3-3.3.8/d3.min.js"></script>
<script src="libs/dagre-0.4.0/dagre-d3.min.js"></script>
<link href="libs/mermaid-0.3.0/dist/mermaid.css" rel="stylesheet" />
<script src="libs/mermaid-0.3.0/dist/mermaid.slim.min.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/chromatography-0.1/chromatography.js"></script>
<script src="libs/DiagrammeR-binding-1.0.6.1/DiagrammeR.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Elements of Applied Biostatistics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#math"><i class="fa fa-check"></i><b>0.1</b> Math</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#r-and-programming"><i class="fa fa-check"></i><b>0.2</b> R and programming</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="a-table-of-models.html"><a href="a-table-of-models.html"><i class="fa fa-check"></i>A Table of Models</a>
<ul>
<li class="chapter" data-level="" data-path="a-table-of-models.html"><a href="a-table-of-models.html#including-mapping-between-linear-models-and-classical-tests"><i class="fa fa-check"></i>including mapping between linear models and classical tests</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-i-getting-started.html"><a href="part-i-getting-started.html"><i class="fa fa-check"></i>Part I: Getting Started</a></li>
<li class="chapter" data-level="1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html"><i class="fa fa-check"></i><b>1</b> Getting Started – R Projects and R Markdown</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#r-vs-r-studio"><i class="fa fa-check"></i><b>1.1</b> R vs R Studio</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#download-and-install-r-and-r-studio"><i class="fa fa-check"></i><b>1.2</b> Download and install R and R studio</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#open-r-studio-and-modify-the-workspace-preference"><i class="fa fa-check"></i><b>1.3</b> Open R Studio and modify the workspace preference</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#if-you-didnt-modify-the-workspace-preferences-from-the-previous-section-go-back-and-do-it"><i class="fa fa-check"></i><b>1.4</b> If you didn’t modify the workspace preferences from the previous section, go back and do it</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#r-markdown-in-a-nutshell"><i class="fa fa-check"></i><b>1.5</b> R Markdown in a nutshell</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#install-r-markdown"><i class="fa fa-check"></i><b>1.6</b> Install R Markdown</a></li>
<li class="chapter" data-level="1.7" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#importing-packages"><i class="fa fa-check"></i><b>1.7</b> Importing Packages</a></li>
<li class="chapter" data-level="1.8" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#setup-create-project"><i class="fa fa-check"></i><b>1.8</b> Create an R Studio Project for this textbook</a></li>
<li class="chapter" data-level="1.9" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#working-on-a-project-in-a-nutshell"><i class="fa fa-check"></i><b>1.9</b> Working on a project, in a nutshell</a></li>
<li class="chapter" data-level="1.10" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-an-r-markdown-file-for-this-chapter"><i class="fa fa-check"></i><b>1.10</b> Create an R Markdown file for this Chapter</a>
<ul>
<li class="chapter" data-level="1.10.1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#modify-the-yaml-header"><i class="fa fa-check"></i><b>1.10.1</b> Modify the yaml header</a></li>
<li class="chapter" data-level="1.10.2" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#modify-the-setup-chunk"><i class="fa fa-check"></i><b>1.10.2</b> Modify the “setup” chunk</a></li>
<li class="chapter" data-level="1.10.3" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-a-fake-data-chunk"><i class="fa fa-check"></i><b>1.10.3</b> Create a “fake-data” chunk</a></li>
<li class="chapter" data-level="1.10.4" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-a-plot-chunk"><i class="fa fa-check"></i><b>1.10.4</b> Create a “plot” chunk</a></li>
<li class="chapter" data-level="1.10.5" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#knit"><i class="fa fa-check"></i><b>1.10.5</b> Knit</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-ii-an-introduction-to-the-analysis-of-experimental-data-with-a-linear-model.html"><a href="part-ii-an-introduction-to-the-analysis-of-experimental-data-with-a-linear-model.html"><i class="fa fa-check"></i>Part II: An introduction to the analysis of experimental data with a linear model</a></li>
<li class="chapter" data-level="2" data-path="analyzing-experimental-data-with-a-linear-model.html"><a href="analyzing-experimental-data-with-a-linear-model.html"><i class="fa fa-check"></i><b>2</b> Analyzing experimental data with a linear model</a>
<ul>
<li class="chapter" data-level="2.1" data-path="analyzing-experimental-data-with-a-linear-model.html"><a href="analyzing-experimental-data-with-a-linear-model.html#this-text-is-about-using-linear-models-to-estimate-treatment-effects-and-the-uncertainty-in-our-estimates.-this-raises-the-question-what-is-an-effect"><i class="fa fa-check"></i><b>2.1</b> This text is about using linear models to estimate treatment effects and the uncertainty in our estimates. This, raises the question, what is “an effect?”</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="ask1-bio.html"><a href="ask1-bio.html"><i class="fa fa-check"></i><b>3</b> Background physiology to the experiments in Figure 2 of “ASK1 inhibits browning of white adipose tissue in obesity”</a></li>
<li class="chapter" data-level="4" data-path="ask1-report.html"><a href="ask1-report.html"><i class="fa fa-check"></i><b>4</b> Analyses for Figure 2 of “ASK1 inhibits browning of white adipose tissue in obesity”</a>
<ul>
<li class="chapter" data-level="4.1" data-path="ask1-report.html"><a href="ask1-report.html#setup"><i class="fa fa-check"></i><b>4.1</b> Setup</a></li>
<li class="chapter" data-level="4.2" data-path="ask1-report.html"><a href="ask1-report.html#data-source"><i class="fa fa-check"></i><b>4.2</b> Data source</a></li>
<li class="chapter" data-level="4.3" data-path="ask1-report.html"><a href="ask1-report.html#control-the-color-palette"><i class="fa fa-check"></i><b>4.3</b> control the color palette</a></li>
<li class="chapter" data-level="4.4" data-path="ask1-report.html"><a href="ask1-report.html#useful-functions"><i class="fa fa-check"></i><b>4.4</b> useful functions</a></li>
<li class="chapter" data-level="4.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2b-effect-of-ask1-deletion-on-growth-body-weight"><i class="fa fa-check"></i><b>4.5</b> figure 2b – effect of ASK1 deletion on growth (body weight)</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2b-import"><i class="fa fa-check"></i><b>4.5.1</b> figure 2b – import</a></li>
<li class="chapter" data-level="4.5.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2b-exploratory-plots"><i class="fa fa-check"></i><b>4.5.2</b> figure 2b – exploratory plots</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-effect-of-ask1-deletion-on-final-body-weight"><i class="fa fa-check"></i><b>4.6</b> Figure 2c – Effect of ASK1 deletion on final body weight</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-import"><i class="fa fa-check"></i><b>4.6.1</b> Figure 2c – import</a></li>
<li class="chapter" data-level="4.6.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-check-own-computation-of-weight-change-v-imported-value"><i class="fa fa-check"></i><b>4.6.2</b> Figure 2c – check own computation of weight change v imported value</a></li>
<li class="chapter" data-level="4.6.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-exploratory-plots"><i class="fa fa-check"></i><b>4.6.3</b> Figure 2c – exploratory plots</a></li>
<li class="chapter" data-level="4.6.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-fit-the-model-m1-lm"><i class="fa fa-check"></i><b>4.6.4</b> Figure 2c – fit the model: m1 (lm)</a></li>
<li class="chapter" data-level="4.6.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-check-the-model-m1"><i class="fa fa-check"></i><b>4.6.5</b> Figure 2c – check the model: m1</a></li>
<li class="chapter" data-level="4.6.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-fit-the-model-m2-gamma-glm"><i class="fa fa-check"></i><b>4.6.6</b> Figure 2c – fit the model: m2 (gamma glm)</a></li>
<li class="chapter" data-level="4.6.7" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-check-the-model-m2"><i class="fa fa-check"></i><b>4.6.7</b> Figure 2c – check the model, m2</a></li>
<li class="chapter" data-level="4.6.8" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-inference-from-the-model"><i class="fa fa-check"></i><b>4.6.8</b> Figure 2c – inference from the model</a></li>
<li class="chapter" data-level="4.6.9" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-plot-the-model"><i class="fa fa-check"></i><b>4.6.9</b> Figure 2c – plot the model</a></li>
<li class="chapter" data-level="4.6.10" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-report"><i class="fa fa-check"></i><b>4.6.10</b> Figure 2c – report</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve"><i class="fa fa-check"></i><b>4.7</b> Figure 2d – Effect of ASK1 KO on glucose tolerance (whole curve)</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-import"><i class="fa fa-check"></i><b>4.7.1</b> Figure 2d – Import</a></li>
<li class="chapter" data-level="4.7.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-exploratory-plots"><i class="fa fa-check"></i><b>4.7.2</b> Figure 2d – exploratory plots</a></li>
<li class="chapter" data-level="4.7.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-fit-the-model"><i class="fa fa-check"></i><b>4.7.3</b> Figure 2d – fit the model</a></li>
<li class="chapter" data-level="4.7.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-check-the-model"><i class="fa fa-check"></i><b>4.7.4</b> Figure 2d – check the model</a></li>
<li class="chapter" data-level="4.7.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-inference"><i class="fa fa-check"></i><b>4.7.5</b> Figure 2d – inference</a></li>
<li class="chapter" data-level="4.7.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-plot-the-model"><i class="fa fa-check"></i><b>4.7.6</b> Figure 2d – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure"><i class="fa fa-check"></i><b>4.8</b> Figure 2e – Effect of ASK1 deletion on glucose tolerance (summary measure)</a>
<ul>
<li class="chapter" data-level="4.8.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-message-the-data"><i class="fa fa-check"></i><b>4.8.1</b> Figure 2e – message the data</a></li>
<li class="chapter" data-level="4.8.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-exploratory-plots"><i class="fa fa-check"></i><b>4.8.2</b> Figure 2e – exploratory plots</a></li>
<li class="chapter" data-level="4.8.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-fit-the-model"><i class="fa fa-check"></i><b>4.8.3</b> Figure 2e – fit the model</a></li>
<li class="chapter" data-level="4.8.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-check-the-model"><i class="fa fa-check"></i><b>4.8.4</b> Figure 2e – check the model</a></li>
<li class="chapter" data-level="4.8.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-inference-from-the-model"><i class="fa fa-check"></i><b>4.8.5</b> Figure 2e – inference from the model</a></li>
<li class="chapter" data-level="4.8.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-plot-the-model"><i class="fa fa-check"></i><b>4.8.6</b> Figure 2e – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate"><i class="fa fa-check"></i><b>4.9</b> Figure 2f – Effect of ASK1 deletion on glucose infusion rate</a>
<ul>
<li class="chapter" data-level="4.9.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-import"><i class="fa fa-check"></i><b>4.9.1</b> Figure 2f – import</a></li>
<li class="chapter" data-level="4.9.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-exploratory-plots"><i class="fa fa-check"></i><b>4.9.2</b> Figure 2f – exploratory plots</a></li>
<li class="chapter" data-level="4.9.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-fit-the-model"><i class="fa fa-check"></i><b>4.9.3</b> Figure 2f – fit the model</a></li>
<li class="chapter" data-level="4.9.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-check-the-model"><i class="fa fa-check"></i><b>4.9.4</b> Figure 2f – check the model</a></li>
<li class="chapter" data-level="4.9.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-inference"><i class="fa fa-check"></i><b>4.9.5</b> Figure 2f – inference</a></li>
<li class="chapter" data-level="4.9.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-plot-the-model"><i class="fa fa-check"></i><b>4.9.6</b> Figure 2f – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.10" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake"><i class="fa fa-check"></i><b>4.10</b> Figure 2g – Effect of ASK1 deletion on tissue-specific glucose uptake</a>
<ul>
<li class="chapter" data-level="4.10.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-import"><i class="fa fa-check"></i><b>4.10.1</b> Figure 2g – import</a></li>
<li class="chapter" data-level="4.10.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-exploratory-plots"><i class="fa fa-check"></i><b>4.10.2</b> Figure 2g – exploratory plots</a></li>
<li class="chapter" data-level="4.10.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-fit-the-model"><i class="fa fa-check"></i><b>4.10.3</b> Figure 2g – fit the model</a></li>
<li class="chapter" data-level="4.10.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-check-the-model"><i class="fa fa-check"></i><b>4.10.4</b> Figure 2g – check the model</a></li>
<li class="chapter" data-level="4.10.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-inference"><i class="fa fa-check"></i><b>4.10.5</b> Figure 2g – inference</a></li>
<li class="chapter" data-level="4.10.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-plot-the-model"><i class="fa fa-check"></i><b>4.10.6</b> Figure 2g – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="ask1-report.html"><a href="ask1-report.html#figure-2h"><i class="fa fa-check"></i><b>4.11</b> Figure 2h</a></li>
<li class="chapter" data-level="4.12" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-effect-of-ask1-deletion-on-liver-tg"><i class="fa fa-check"></i><b>4.12</b> Figure 2i – Effect of ASK1 deletion on liver TG</a>
<ul>
<li class="chapter" data-level="4.12.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-fit-the-model"><i class="fa fa-check"></i><b>4.12.1</b> Figure 2i – fit the model</a></li>
<li class="chapter" data-level="4.12.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-check-the-model"><i class="fa fa-check"></i><b>4.12.2</b> Figure 2i – check the model</a></li>
<li class="chapter" data-level="4.12.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-inference"><i class="fa fa-check"></i><b>4.12.3</b> Figure 2i – inference</a></li>
<li class="chapter" data-level="4.12.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-plot-the-model"><i class="fa fa-check"></i><b>4.12.4</b> Figure 2i – plot the model</a></li>
<li class="chapter" data-level="4.12.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-report-the-model"><i class="fa fa-check"></i><b>4.12.5</b> Figure 2i – report the model</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="ask1-report.html"><a href="ask1-report.html#figure-2j"><i class="fa fa-check"></i><b>4.13</b> Figure 2j</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-iii-r-fundamentals.html"><a href="part-iii-r-fundamentals.html"><i class="fa fa-check"></i>Part III: R fundamentals</a></li>
<li class="chapter" data-level="5" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html"><i class="fa fa-check"></i><b>5</b> Data – Reading, Wrangling, and Writing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#learning-from-this-chapter"><i class="fa fa-check"></i><b>5.1</b> Learning from this chapter</a></li>
<li class="chapter" data-level="5.2" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#data-working-in-r"><i class="fa fa-check"></i><b>5.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#importing-data"><i class="fa fa-check"></i><b>5.2.1</b> Importing data</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#data-wrangling"><i class="fa fa-check"></i><b>5.3</b> Data wrangling</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#reshaping-data-wide-to-long"><i class="fa fa-check"></i><b>5.3.1</b> Reshaping data – Wide to long</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#reshaping-data-transpose-turning-the-columns-into-rows"><i class="fa fa-check"></i><b>5.3.2</b> Reshaping data – Transpose (turning the columns into rows)</a></li>
<li class="chapter" data-level="5.3.3" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#combining-data"><i class="fa fa-check"></i><b>5.3.3</b> Combining data</a></li>
<li class="chapter" data-level="5.3.4" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#subsetting-data"><i class="fa fa-check"></i><b>5.3.4</b> Subsetting data</a></li>
<li class="chapter" data-level="5.3.5" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#wrangling-columns"><i class="fa fa-check"></i><b>5.3.5</b> Wrangling columns</a></li>
<li class="chapter" data-level="5.3.6" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#missing-data"><i class="fa fa-check"></i><b>5.3.6</b> Missing data</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#saving-data"><i class="fa fa-check"></i><b>5.4</b> Saving data</a></li>
<li class="chapter" data-level="5.5" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#exercises"><i class="fa fa-check"></i><b>5.5</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="plotting-models.html"><a href="plotting-models.html"><i class="fa fa-check"></i><b>6</b> Plotting Models</a>
<ul>
<li class="chapter" data-level="6.1" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plots-show-the-model-and-the-data"><i class="fa fa-check"></i><b>6.1</b> Pretty good plots show the model and the data</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plot-component-1-modeled-effects-plot"><i class="fa fa-check"></i><b>6.1.1</b> Pretty good plot component 1: Modeled effects plot</a></li>
<li class="chapter" data-level="6.1.2" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plot-component-2-modeled-mean-and-ci-plot"><i class="fa fa-check"></i><b>6.1.2</b> Pretty good plot component 2: Modeled mean and CI plot</a></li>
<li class="chapter" data-level="6.1.3" data-path="plotting-models.html"><a href="plotting-models.html#combining-effects-and-modeled-mean-and-ci-plots-an-effects-and-response-plot."><i class="fa fa-check"></i><b>6.1.3</b> Combining Effects and Modeled mean and CI plots – an Effects and response plot.</a></li>
<li class="chapter" data-level="6.1.4" data-path="plotting-models.html"><a href="plotting-models.html#some-comments-on-plot-components"><i class="fa fa-check"></i><b>6.1.4</b> Some comments on plot components</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="plotting-models.html"><a href="plotting-models.html#working-in-r"><i class="fa fa-check"></i><b>6.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="plotting-models.html"><a href="plotting-models.html#source-data"><i class="fa fa-check"></i><b>6.2.1</b> Source data</a></li>
<li class="chapter" data-level="6.2.2" data-path="plotting-models.html"><a href="plotting-models.html#how-to-plot-the-model"><i class="fa fa-check"></i><b>6.2.2</b> How to plot the model</a></li>
<li class="chapter" data-level="6.2.3" data-path="plotting-models.html"><a href="plotting-models.html#be-sure-ggplot_the_model-is-in-your-r-folder"><i class="fa fa-check"></i><b>6.2.3</b> Be sure ggplot_the_model is in your R folder</a></li>
<li class="chapter" data-level="6.2.4" data-path="plotting-models.html"><a href="plotting-models.html#how-to-use-the-plot-the-model-functions"><i class="fa fa-check"></i><b>6.2.4</b> How to use the Plot the Model functions</a></li>
<li class="chapter" data-level="6.2.5" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-a-response-plot-using-ggpubr"><i class="fa fa-check"></i><b>6.2.5</b> How to generate a Response Plot using ggpubr</a></li>
<li class="chapter" data-level="6.2.6" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-a-response-plot-with-a-grid-of-treatments-using-ggplot2"><i class="fa fa-check"></i><b>6.2.6</b> How to generate a Response Plot with a grid of treatments using ggplot2</a></li>
<li class="chapter" data-level="6.2.7" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-an-effects-plot"><i class="fa fa-check"></i><b>6.2.7</b> How to generate an Effects Plot</a></li>
<li class="chapter" data-level="6.2.8" data-path="plotting-models.html"><a href="plotting-models.html#how-to-combine-the-response-and-effects-plots"><i class="fa fa-check"></i><b>6.2.8</b> How to combine the response and effects plots</a></li>
<li class="chapter" data-level="6.2.9" data-path="plotting-models.html"><a href="plotting-models.html#how-to-add-the-interaction-effect-to-response-and-effects-plots"><i class="fa fa-check"></i><b>6.2.9</b> How to add the interaction effect to response and effects plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-iv-some-fundamentals-of-statistical-modeling.html"><a href="part-iv-some-fundamentals-of-statistical-modeling.html"><i class="fa fa-check"></i>Part IV: Some Fundamentals of Statistical Modeling</a></li>
<li class="chapter" data-level="7" data-path="uncertainty.html"><a href="uncertainty.html"><i class="fa fa-check"></i><b>7</b> Variability and Uncertainty (Standard Deviations, Standard Errors, Confidence Intervals)</a>
<ul>
<li class="chapter" data-level="7.1" data-path="uncertainty.html"><a href="uncertainty.html#the-sample-standard-deviation-vs.-the-standard-error-of-the-mean"><i class="fa fa-check"></i><b>7.1</b> The sample standard deviation vs. the standard error of the mean</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="uncertainty.html"><a href="uncertainty.html#sample-standard-deviation"><i class="fa fa-check"></i><b>7.1.1</b> Sample standard deviation</a></li>
<li class="chapter" data-level="7.1.2" data-path="uncertainty.html"><a href="uncertainty.html#standard-error-of-the-mean"><i class="fa fa-check"></i><b>7.1.2</b> Standard error of the mean</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="uncertainty.html"><a href="uncertainty.html#using-google-sheets-to-generate-fake-data-to-explore-the-standard-error"><i class="fa fa-check"></i><b>7.2</b> Using Google Sheets to generate fake data to explore the standard error</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="uncertainty.html"><a href="uncertainty.html#steps"><i class="fa fa-check"></i><b>7.2.1</b> Steps</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="uncertainty.html"><a href="uncertainty.html#using-r-to-generate-fake-data-to-explore-the-standard-error"><i class="fa fa-check"></i><b>7.3</b> Using R to generate fake data to explore the standard error</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="uncertainty.html"><a href="uncertainty.html#part-i"><i class="fa fa-check"></i><b>7.3.1</b> part I</a></li>
<li class="chapter" data-level="7.3.2" data-path="uncertainty.html"><a href="uncertainty.html#part-ii---means"><i class="fa fa-check"></i><b>7.3.2</b> part II - means</a></li>
<li class="chapter" data-level="7.3.3" data-path="uncertainty.html"><a href="uncertainty.html#part-iii---how-do-sd-and-se-change-as-sample-size-n-increases"><i class="fa fa-check"></i><b>7.3.3</b> part III - how do SD and SE change as sample size (n) increases?</a></li>
<li class="chapter" data-level="7.3.4" data-path="uncertainty.html"><a href="uncertainty.html#part-iv-generating-fake-data-with-for-loops"><i class="fa fa-check"></i><b>7.3.4</b> Part IV – Generating fake data with for-loops</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="uncertainty.html"><a href="uncertainty.html#bootstrap"><i class="fa fa-check"></i><b>7.4</b> Bootstrapped standard errors</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="uncertainty.html"><a href="uncertainty.html#an-example-of-bootstrapped-standard-errors-using-vole-data"><i class="fa fa-check"></i><b>7.4.1</b> An example of bootstrapped standard errors using vole data</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="uncertainty.html"><a href="uncertainty.html#confidence-interval"><i class="fa fa-check"></i><b>7.5</b> Confidence Interval</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="uncertainty.html"><a href="uncertainty.html#interpretation-of-a-confidence-interval"><i class="fa fa-check"></i><b>7.5.1</b> Interpretation of a confidence interval</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="p-values.html"><a href="p-values.html"><i class="fa fa-check"></i><b>8</b> P-values</a>
<ul>
<li class="chapter" data-level="8.1" data-path="p-values.html"><a href="p-values.html#a-p-value-is-the-probability-of-sampling-a-value-as-or-more-extreme-than-the-test-statistic-if-sampling-from-a-null-distribution"><i class="fa fa-check"></i><b>8.1</b> A <em>p</em>-value is the probability of sampling a value as or more extreme than the test statistic if sampling from a null distribution</a></li>
<li class="chapter" data-level="8.2" data-path="p-values.html"><a href="p-values.html#pump-your-intuition-creating-a-null-distribution"><i class="fa fa-check"></i><b>8.2</b> Pump your intuition – Creating a null distribution</a></li>
<li class="chapter" data-level="8.3" data-path="p-values.html"><a href="p-values.html#a-null-distribution-of-t-values-the-t-distribution"><i class="fa fa-check"></i><b>8.3</b> A null distribution of <em>t</em>-values – the <em>t</em> distribution</a></li>
<li class="chapter" data-level="8.4" data-path="p-values.html"><a href="p-values.html#p-values-from-the-perspective-of-permutation"><i class="fa fa-check"></i><b>8.4</b> P-values from the perspective of permutation</a></li>
<li class="chapter" data-level="8.5" data-path="p-values.html"><a href="p-values.html#parametric-vs.-non-parametric-statistics"><i class="fa fa-check"></i><b>8.5</b> Parametric vs. non-parametric statistics</a></li>
<li class="chapter" data-level="8.6" data-path="p-values.html"><a href="p-values.html#frequentist-probability-and-the-interpretation-of-p-values"><i class="fa fa-check"></i><b>8.6</b> frequentist probability and the interpretation of p-values</a>
<ul>
<li class="chapter" data-level="8.6.1" data-path="p-values.html"><a href="p-values.html#background"><i class="fa fa-check"></i><b>8.6.1</b> Background</a></li>
<li class="chapter" data-level="8.6.2" data-path="p-values.html"><a href="p-values.html#this-book-covers-frequentist-approaches-to-statistical-modeling-and-when-a-probability-arises-such-as-the-p-value-of-a-test-statistic-this-will-be-a-frequentist-probability."><i class="fa fa-check"></i><b>8.6.2</b> This book covers frequentist approaches to statistical modeling and when a probability arises, such as the <em>p</em>-value of a test statistic, this will be a frequentist probability.</a></li>
<li class="chapter" data-level="8.6.3" data-path="p-values.html"><a href="p-values.html#two-interpretations-of-the-p-value"><i class="fa fa-check"></i><b>8.6.3</b> Two interpretations of the <em>p</em>-value</a></li>
<li class="chapter" data-level="8.6.4" data-path="p-values.html"><a href="p-values.html#nhst"><i class="fa fa-check"></i><b>8.6.4</b> NHST</a></li>
</ul></li>
<li class="chapter" data-level="8.7" data-path="p-values.html"><a href="p-values.html#some-major-misconceptions-of-the-p-value"><i class="fa fa-check"></i><b>8.7</b> Some major misconceptions of the <em>p</em>-value</a>
<ul>
<li class="chapter" data-level="8.7.1" data-path="p-values.html"><a href="p-values.html#misconception-p-is-the-probability-that-the-null-is-true-and-1-p-is-probability-that-the-alternative-is-true"><i class="fa fa-check"></i><b>8.7.1</b> Misconception: <em>p</em> is the probability that the null is true <em>and</em> <span class="math inline">\(1-p\)</span> is probability that the alternative is true</a></li>
<li class="chapter" data-level="8.7.2" data-path="p-values.html"><a href="p-values.html#misconception-a-p-value-is-repeatable"><i class="fa fa-check"></i><b>8.7.2</b> Misconception: a <em>p</em>-value is repeatable</a></li>
<li class="chapter" data-level="8.7.3" data-path="p-values.html"><a href="p-values.html#misconception-0.05-is-the-lifetime-rate-of-false-discoveries"><i class="fa fa-check"></i><b>8.7.3</b> Misconception: 0.05 is the lifetime rate of false discoveries</a></li>
<li class="chapter" data-level="8.7.4" data-path="p-values.html"><a href="p-values.html#misconception-a-low-p-value-indicates-an-important-effect"><i class="fa fa-check"></i><b>8.7.4</b> Misconception: a low <em>p</em>-value indicates an important effect</a></li>
<li class="chapter" data-level="8.7.5" data-path="p-values.html"><a href="p-values.html#misconception-a-low-p-value-indicates-high-model-fit-or-high-predictive-capacity"><i class="fa fa-check"></i><b>8.7.5</b> Misconception: a low <em>p</em>-value indicates high model fit or high predictive capacity</a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="p-values.html"><a href="p-values.html#what-the-p-value-does-not-mean"><i class="fa fa-check"></i><b>8.8</b> What the <em>p</em>-value does not mean</a></li>
<li class="chapter" data-level="8.9" data-path="p-values.html"><a href="p-values.html#recommendations"><i class="fa fa-check"></i><b>8.9</b> Recommendations</a>
<ul>
<li class="chapter" data-level="8.9.1" data-path="p-values.html"><a href="p-values.html#primary-sources-for-recommendations"><i class="fa fa-check"></i><b>8.9.1</b> Primary sources for recommendations</a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="p-values.html"><a href="p-values.html#problems"><i class="fa fa-check"></i><b>8.10</b> Problems</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="errors-in-inference.html"><a href="errors-in-inference.html"><i class="fa fa-check"></i><b>9</b> Errors in inference</a>
<ul>
<li class="chapter" data-level="9.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#classical-nhst-concepts-of-wrong"><i class="fa fa-check"></i><b>9.1</b> Classical NHST concepts of wrong</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-i-error"><i class="fa fa-check"></i><b>9.1.1</b> Type I error</a></li>
<li class="chapter" data-level="9.1.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#power"><i class="fa fa-check"></i><b>9.1.2</b> Power</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#a-non-neyman-pearson-concept-of-power"><i class="fa fa-check"></i><b>9.2</b> A non-Neyman-Pearson concept of power</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#estimation-error"><i class="fa fa-check"></i><b>9.2.1</b> Estimation error</a></li>
<li class="chapter" data-level="9.2.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#coverage"><i class="fa fa-check"></i><b>9.2.2</b> Coverage</a></li>
<li class="chapter" data-level="9.2.3" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-s-error"><i class="fa fa-check"></i><b>9.2.3</b> Type S error</a></li>
<li class="chapter" data-level="9.2.4" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-m-error"><i class="fa fa-check"></i><b>9.2.4</b> Type M error</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-v-introduction-to-linear-models.html"><a href="part-v-introduction-to-linear-models.html"><i class="fa fa-check"></i>Part V: Introduction to Linear Models</a></li>
<li class="chapter" data-level="10" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html"><i class="fa fa-check"></i><b>10</b> An introduction to linear models</a>
<ul>
<li class="chapter" data-level="10.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#lm-specifications"><i class="fa fa-check"></i><b>10.1</b> Two specifications of a linear model</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#the-error-draw-specification"><i class="fa fa-check"></i><b>10.1.1</b> The “error draw” specification</a></li>
<li class="chapter" data-level="10.1.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#the-conditional-draw-specification"><i class="fa fa-check"></i><b>10.1.2</b> The “conditional draw” specification</a></li>
<li class="chapter" data-level="10.1.3" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#comparing-the-error-draw-and-conditional-draw-ways-of-specifying-the-linear-model"><i class="fa fa-check"></i><b>10.1.3</b> Comparing the error-draw and conditional-draw ways of specifying the linear model</a></li>
<li class="chapter" data-level="10.1.4" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#anova-notation-of-a-linear-model"><i class="fa fa-check"></i><b>10.1.4</b> ANOVA notation of a linear model</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#a-linear-model-can-be-fit-to-data-with-continuous-discrete-or-categorical-x-variables"><i class="fa fa-check"></i><b>10.2</b> A linear model can be fit to data with continuous, discrete, or categorical <span class="math inline">\(X\)</span> variables</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#fitting-linear-models-to-experimental-data-in-which-the-x-variable-is-continuous-or-discrete"><i class="fa fa-check"></i><b>10.2.1</b> Fitting linear models to experimental data in which the <span class="math inline">\(X\)</span> variable is continuous or discrete</a></li>
<li class="chapter" data-level="10.2.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#slope"><i class="fa fa-check"></i><b>10.2.2</b> Fitting linear models to experimental data in which the <span class="math inline">\(X\)</span> variable is categorical</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#statistical-models-are-used-for-prediction-explanation-and-description"><i class="fa fa-check"></i><b>10.3</b> Statistical models are used for prediction, explanation, and description</a></li>
<li class="chapter" data-level="10.4" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#what-do-we-call-the-x-and-y-variables"><i class="fa fa-check"></i><b>10.4</b> What do we call the <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span> variables?</a></li>
<li class="chapter" data-level="10.5" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#modeling-strategy"><i class="fa fa-check"></i><b>10.5</b> Modeling strategy</a></li>
<li class="chapter" data-level="10.6" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#predictions-from-the-model"><i class="fa fa-check"></i><b>10.6</b> Predictions from the model</a></li>
<li class="chapter" data-level="10.7" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#inference-from-the-model"><i class="fa fa-check"></i><b>10.7</b> Inference from the model</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#assumptions-for-inference-with-a-statistical-model"><i class="fa fa-check"></i><b>10.7.1</b> Assumptions for inference with a statistical model</a></li>
<li class="chapter" data-level="10.7.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#specific-assumptions-for-inference-with-a-linear-model"><i class="fa fa-check"></i><b>10.7.2</b> Specific assumptions for inference with a linear model</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#linear-modelregression-model-orstatistical-model"><i class="fa fa-check"></i><b>10.8</b> “linear model,”regression model“, or”statistical model"?</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="regression.html"><a href="regression.html"><i class="fa fa-check"></i><b>11</b> Linear models with a single, continuous <em>X</em> (“regression”)</a>
<ul>
<li class="chapter" data-level="11.1" data-path="regression.html"><a href="regression.html#a-linear-model-with-a-single-continuous-x-is-classical-regression"><i class="fa fa-check"></i><b>11.1</b> A linear model with a single, continuous <em>X</em> is classical “regression”</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="regression.html"><a href="regression.html#analysis-of-green-down-data"><i class="fa fa-check"></i><b>11.1.1</b> Analysis of “green-down” data</a></li>
<li class="chapter" data-level="11.1.2" data-path="regression.html"><a href="regression.html#learning-from-the-green-down-example"><i class="fa fa-check"></i><b>11.1.2</b> Learning from the green-down example</a></li>
<li class="chapter" data-level="11.1.3" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-explanation-causal-models"><i class="fa fa-check"></i><b>11.1.3</b> Using a regression model for “explanation” – causal models</a></li>
<li class="chapter" data-level="11.1.4" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-prediction-prediction-models"><i class="fa fa-check"></i><b>11.1.4</b> Using a regression model for prediction – prediction models</a></li>
<li class="chapter" data-level="11.1.5" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-creating-a-new-response-variable-comparing-slopes-of-longitudinal-data"><i class="fa fa-check"></i><b>11.1.5</b> Using a regression model for creating a new response variable – comparing slopes of longitudinal data</a></li>
<li class="chapter" data-level="11.1.6" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-for-calibration"><i class="fa fa-check"></i><b>11.1.6</b> Using a regression model for for calibration</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="regression.html"><a href="regression.html#working-in-r-1"><i class="fa fa-check"></i><b>11.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="plotting-models.html"><a href="plotting-models.html#fit-the-model"><i class="fa fa-check"></i><b>11.2.1</b> Fitting the linear model</a></li>
<li class="chapter" data-level="11.2.2" data-path="regression.html"><a href="regression.html#getting-to-know-the-linear-model-the-summary-function"><i class="fa fa-check"></i><b>11.2.2</b> Getting to know the linear model: the <code>summary</code> function</a></li>
<li class="chapter" data-level="11.2.3" data-path="regression.html"><a href="regression.html#inference-the-coefficient-table"><i class="fa fa-check"></i><b>11.2.3</b> Inference – the coefficient table</a></li>
<li class="chapter" data-level="11.2.4" data-path="regression.html"><a href="regression.html#how-good-is-our-model-model-checking"><i class="fa fa-check"></i><b>11.2.4</b> How good is our model? – Model checking</a></li>
<li class="chapter" data-level="11.2.5" data-path="regression.html"><a href="regression.html#plotting-models-with-continuous-x"><i class="fa fa-check"></i><b>11.2.5</b> Plotting models with continuous <em>X</em></a></li>
<li class="chapter" data-level="11.2.6" data-path="regression.html"><a href="regression.html#creating-a-table-of-predicted-values-and-95-prediction-intervals"><i class="fa fa-check"></i><b>11.2.6</b> Creating a table of predicted values and 95% prediction intervals</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="regression.html"><a href="regression.html#hidden-code"><i class="fa fa-check"></i><b>11.3</b> Hidden code</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="regression.html"><a href="regression.html#import-and-plot-of-fig2c-ecosystem-warming-experimental-data"><i class="fa fa-check"></i><b>11.3.1</b> Import and plot of fig2c (ecosystem warming experimental) data</a></li>
<li class="chapter" data-level="11.3.2" data-path="regression.html"><a href="regression.html#import-and-plot-efig_3d-ecosysem-warming-observational-data"><i class="fa fa-check"></i><b>11.3.2</b> Import and plot efig_3d (Ecosysem warming observational) data</a></li>
<li class="chapter" data-level="11.3.3" data-path="regression.html"><a href="regression.html#import-and-plot-of-fig1f-methionine-restriction-data"><i class="fa fa-check"></i><b>11.3.3</b> Import and plot of fig1f (methionine restriction) data</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="regression.html"><a href="regression.html#try-it"><i class="fa fa-check"></i><b>11.4</b> Try it</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="regression.html"><a href="regression.html#a-prediction-model-from-the-literature"><i class="fa fa-check"></i><b>11.4.1</b> A prediction model from the literature</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="regression.html"><a href="regression.html#intuition-pumps"><i class="fa fa-check"></i><b>11.5</b> Intuition pumps</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="regression.html"><a href="regression.html#correlation-and-r2"><i class="fa fa-check"></i><b>11.5.1</b> Correlation and $R^2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="oneway.html"><a href="oneway.html"><i class="fa fa-check"></i><b>12</b> Linear models with a single, categorical <em>X</em> (“t-tests” and “ANOVA”)</a>
<ul>
<li class="chapter" data-level="12.1" data-path="oneway.html"><a href="oneway.html#oneway-data"><i class="fa fa-check"></i><b>12.1</b> A linear model with a single, categorical <em>X</em> variable estimates the effects of the levels of <em>X</em> on the response</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="oneway.html"><a href="oneway.html#oneway-example1"><i class="fa fa-check"></i><b>12.1.1</b> Example 1 (fig3d) – two treatment levels (“groups”)</a></li>
<li class="chapter" data-level="12.1.2" data-path="oneway.html"><a href="oneway.html#understanding-the-analysis-with-two-treatment-levels"><i class="fa fa-check"></i><b>12.1.2</b> Understanding the analysis with two treatment levels</a></li>
<li class="chapter" data-level="12.1.3" data-path="oneway.html"><a href="oneway.html#oneway-example2"><i class="fa fa-check"></i><b>12.1.3</b> Example 2 – three treatment levels (“groups”)</a></li>
<li class="chapter" data-level="12.1.4" data-path="oneway.html"><a href="oneway.html#understanding-the-analysis-with-three-or-more-treatment-levels"><i class="fa fa-check"></i><b>12.1.4</b> Understanding the analysis with three (or more) treatment levels</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="oneway.html"><a href="oneway.html#working-in-r-2"><i class="fa fa-check"></i><b>12.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="oneway.html"><a href="oneway.html#fit-the-model-1"><i class="fa fa-check"></i><b>12.2.1</b> Fit the model</a></li>
<li class="chapter" data-level="12.2.2" data-path="oneway.html"><a href="oneway.html#controlling-the-output-in-tables-using-the-coefficient-table-as-an-example"><i class="fa fa-check"></i><b>12.2.2</b> Controlling the output in tables using the coefficient table as an example</a></li>
<li class="chapter" data-level="12.2.3" data-path="oneway.html"><a href="oneway.html#using-the-emmeans-function"><i class="fa fa-check"></i><b>12.2.3</b> Using the emmeans function</a></li>
<li class="chapter" data-level="12.2.4" data-path="oneway.html"><a href="oneway.html#using-the-contrast-function"><i class="fa fa-check"></i><b>12.2.4</b> Using the contrast function</a></li>
<li class="chapter" data-level="12.2.5" data-path="oneway.html"><a href="oneway.html#how-to-generate-anova-tables"><i class="fa fa-check"></i><b>12.2.5</b> How to generate ANOVA tables</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="oneway.html"><a href="oneway.html#hidden-code-1"><i class="fa fa-check"></i><b>12.3</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="oneway.html"><a href="oneway.html#importing-and-wrangling-the-fig_3d-data-for-example-1"><i class="fa fa-check"></i><b>12.3.1</b> Importing and wrangling the fig_3d data for example 1</a></li>
<li class="chapter" data-level="12.3.2" data-path="oneway.html"><a href="oneway.html#importing-and-wrangling-the-fig2a-data-for-example-2"><i class="fa fa-check"></i><b>12.3.2</b> Importing and wrangling the fig2a data for example 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="model-checking.html"><a href="model-checking.html"><i class="fa fa-check"></i><b>13</b> Model Checking</a>
<ul>
<li class="chapter" data-level="13.1" data-path="model-checking.html"><a href="model-checking.html#all-statistical-analyses-should-be-followed-by-model-checking"><i class="fa fa-check"></i><b>13.1</b> All statistical analyses should be followed by model checking</a></li>
<li class="chapter" data-level="13.2" data-path="model-checking.html"><a href="model-checking.html#linear-model-assumptions"><i class="fa fa-check"></i><b>13.2</b> Linear model assumptions</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="model-checking.html"><a href="model-checking.html#a-bit-about-iid"><i class="fa fa-check"></i><b>13.2.1</b> A bit about IID</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="model-checking.html"><a href="model-checking.html#diagnostic-plots-use-the-residuals-from-the-model-fit"><i class="fa fa-check"></i><b>13.3</b> Diagnostic plots use the residuals from the model fit</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="model-checking.html"><a href="model-checking.html#residuals"><i class="fa fa-check"></i><b>13.3.1</b> Residuals</a></li>
<li class="chapter" data-level="13.3.2" data-path="model-checking.html"><a href="model-checking.html#normal-qq"><i class="fa fa-check"></i><b>13.3.2</b> A Normal Q-Q plot is used to check for characteristic departures from Normality</a></li>
<li class="chapter" data-level="13.3.3" data-path="model-checking.html"><a href="model-checking.html#mapping-qq-plot-departures-from-normality"><i class="fa fa-check"></i><b>13.3.3</b> Mapping QQ-plot departures from Normality</a></li>
<li class="chapter" data-level="13.3.4" data-path="model-checking.html"><a href="model-checking.html#model-checking-homoskedasticity"><i class="fa fa-check"></i><b>13.3.4</b> Model checking homoskedasticity</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="model-checking.html"><a href="model-checking.html#using-r"><i class="fa fa-check"></i><b>13.4</b> Using R</a>
<ul>
<li class="chapter" data-level="13.4.1" data-path="model-checking.html"><a href="model-checking.html#normal-q-q-plots"><i class="fa fa-check"></i><b>13.4.1</b> Normal Q-Q plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="violations.html"><a href="violations.html"><i class="fa fa-check"></i><b>14</b> Violations of independence, homogeneity, or Normality</a>
<ul>
<li class="chapter" data-level="14.1" data-path="violations.html"><a href="violations.html#oneway-paired-t"><i class="fa fa-check"></i><b>14.1</b> Lack of independence</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="violations.html"><a href="violations.html#violations-paired-t"><i class="fa fa-check"></i><b>14.1.1</b> Example 1 (exp1b) – a paired t-test is a special case of a linear mixed model</a></li>
<li class="chapter" data-level="14.1.2" data-path="violations.html"><a href="violations.html#violations-rmanova"><i class="fa fa-check"></i><b>14.1.2</b> Example 2 (diHOME exp2a) – A repeated measures ANOVA is a special case of a linear mixed model</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="violations.html"><a href="violations.html#oneway-welch"><i class="fa fa-check"></i><b>14.2</b> Heterogeneity of variances</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="violations.html"><a href="violations.html#when-groups-of-the-focal-test-have-variance"><i class="fa fa-check"></i><b>14.2.1</b> When groups of the focal test have &gt;&gt; variance</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="violations.html"><a href="violations.html#the-conditional-response-isnt-normal"><i class="fa fa-check"></i><b>14.3</b> The conditional response isn’t Normal</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="violations.html"><a href="violations.html#violations-6f"><i class="fa fa-check"></i><b>14.3.1</b> Example 1 (fig6f) – Linear models for non-normal count data</a></li>
<li class="chapter" data-level="14.3.2" data-path="violations.html"><a href="violations.html#my-data-arent-normal-what-is-the-best-practice"><i class="fa fa-check"></i><b>14.3.2</b> My data aren’t normal, what is the best practice?</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="violations.html"><a href="violations.html#hidden-code-2"><i class="fa fa-check"></i><b>14.4</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-exp1b-data"><i class="fa fa-check"></i><b>14.4.1</b> Importing and wrangling the exp1b data</a></li>
<li class="chapter" data-level="14.4.2" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-exp2a-data"><i class="fa fa-check"></i><b>14.4.2</b> Importing and wrangling the exp2a data</a></li>
<li class="chapter" data-level="14.4.3" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-fig6f-data"><i class="fa fa-check"></i><b>14.4.3</b> Importing and wrangling the fig6f data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="issues.html"><a href="issues.html"><i class="fa fa-check"></i><b>15</b> Issues in inference</a>
<ul>
<li class="chapter" data-level="15.1" data-path="issues.html"><a href="issues.html#replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means"><i class="fa fa-check"></i><b>15.1</b> Replicated experiments – include <span class="math inline">\(\texttt{Experiment}\)</span> as a random factor (better than one-way ANOVA of means)</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="issues.html"><a href="issues.html#issues-exp4d"><i class="fa fa-check"></i><b>15.1.1</b> Multiple experiments Example 1 (wound healing Exp4d)</a></li>
<li class="chapter" data-level="15.1.2" data-path="issues.html"><a href="issues.html#issues-combined-experiments"><i class="fa fa-check"></i><b>15.1.2</b> Models for combining replicated experiments</a></li>
<li class="chapter" data-level="15.1.3" data-path="issues.html"><a href="issues.html#understanding-model-exp4d_m1"><i class="fa fa-check"></i><b>15.1.3</b> Understanding Model <code>exp4d_m1</code></a></li>
<li class="chapter" data-level="15.1.4" data-path="issues.html"><a href="issues.html#the-univariate-model-is-equivalent-to-a-linear-mixed-model-of-the-aggregated-data-model-exp4d_m2"><i class="fa fa-check"></i><b>15.1.4</b> The univariate model is equivalent to a linear mixed model of the aggregated data (Model exp4d_m2)</a></li>
<li class="chapter" data-level="15.1.5" data-path="issues.html"><a href="issues.html#a-linear-mixed-model-of-the-full-data"><i class="fa fa-check"></i><b>15.1.5</b> A linear mixed model of the full data</a></li>
<li class="chapter" data-level="15.1.6" data-path="issues.html"><a href="issues.html#analysis-of-the-experiment-means-has-less-precision-and-power"><i class="fa fa-check"></i><b>15.1.6</b> Analysis of the experiment means has less precision and power</a></li>
<li class="chapter" data-level="15.1.7" data-path="issues.html"><a href="issues.html#dont-do-this-a-t-testfixed-effect-anova-of-the-full-data"><i class="fa fa-check"></i><b>15.1.7</b> Don’t do this – a t-test/fixed-effect ANOVA of the full data</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="issues.html"><a href="issues.html#issues-pre-post"><i class="fa fa-check"></i><b>15.2</b> Comparing change from baseline (pre-post)</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="issues.html"><a href="issues.html#pre-post-example-1-dpp4-fig4c"><i class="fa fa-check"></i><b>15.2.1</b> Pre-post example 1 (DPP4 fig4c)</a></li>
<li class="chapter" data-level="15.2.2" data-path="issues.html"><a href="issues.html#what-if-the-data-in-example-1-were-from-from-an-experiment-where-the-treatment-was-applied-prior-to-the-baseline-measure"><i class="fa fa-check"></i><b>15.2.2</b> What if the data in example 1 were from from an experiment where the treatment was applied prior to the baseline measure?</a></li>
<li class="chapter" data-level="15.2.3" data-path="issues.html"><a href="issues.html#pre-post-example-2-xx-males-fig1c"><i class="fa fa-check"></i><b>15.2.3</b> Pre-post example 2 (XX males fig1c)</a></li>
<li class="chapter" data-level="15.2.4" data-path="issues.html"><a href="issues.html#issues-regression-to-mean"><i class="fa fa-check"></i><b>15.2.4</b> Regression to the mean</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="issues.html"><a href="issues.html#longitudinal-designs-with-more-than-one-post-baseline-measure"><i class="fa fa-check"></i><b>15.3</b> Longitudinal designs with more than one-post baseline measure</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="issues.html"><a href="issues.html#issues-auc"><i class="fa fa-check"></i><b>15.3.1</b> Area under the curve (AUC)</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="issues.html"><a href="issues.html#comparing-ratios"><i class="fa fa-check"></i><b>15.4</b> Comparing ratios</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="issues.html"><a href="issues.html#the-ratio-is-a-density"><i class="fa fa-check"></i><b>15.4.1</b> The ratio is a density</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="issues.html"><a href="issues.html#dont-do-this-stuff"><i class="fa fa-check"></i><b>15.5</b> Don’t do this stuff</a>
<ul>
<li class="chapter" data-level="15.5.1" data-path="issues.html"><a href="issues.html#normalize-the-response-so-that-all-control-values-are-equal-to-1."><i class="fa fa-check"></i><b>15.5.1</b> Normalize the response so that all control values are equal to 1.</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="issues.html"><a href="issues.html#a-difference-in-significance-is-not-necessarily-significant"><i class="fa fa-check"></i><b>15.6</b> A difference in significance is not necessarily significant</a></li>
<li class="chapter" data-level="15.7" data-path="issues.html"><a href="issues.html#researcher-degrees-of-freedom"><i class="fa fa-check"></i><b>15.7</b> Researcher degrees of freedom</a></li>
<li class="chapter" data-level="15.8" data-path="issues.html"><a href="issues.html#hidden-code-3"><i class="fa fa-check"></i><b>15.8</b> Hidden code</a>
<ul>
<li class="chapter" data-level="15.8.1" data-path="issues.html"><a href="issues.html#import-exp4d-vimentin-cell-count-data-replicate-experiments-example"><i class="fa fa-check"></i><b>15.8.1</b> Import exp4d vimentin cell count data (replicate experiments example)</a></li>
<li class="chapter" data-level="15.8.2" data-path="issues.html"><a href="issues.html#import-fig4c-data"><i class="fa fa-check"></i><b>15.8.2</b> Import Fig4c data</a></li>
<li class="chapter" data-level="15.8.3" data-path="issues.html"><a href="issues.html#xx-males-fig1c"><i class="fa fa-check"></i><b>15.8.3</b> XX males fig1c</a></li>
<li class="chapter" data-level="15.8.4" data-path="issues.html"><a href="issues.html#generation-of-fake-data-to-illustrate-regression-to-the-mean"><i class="fa fa-check"></i><b>15.8.4</b> Generation of fake data to illustrate regression to the mean</a></li>
<li class="chapter" data-level="15.8.5" data-path="issues.html"><a href="issues.html#import-fig3f"><i class="fa fa-check"></i><b>15.8.5</b> Import fig3f</a></li>
</ul></li>
</ul></li>
<li><a href="part-vi-more-than-one-x-multivariable-models.html#part-vi-more-than-one-x-multivariable-models">Part VI: More than one <span class="math inline">\(X\)</span> – Multivariable Models</a></li>
<li class="chapter" data-level="16" data-path="covariates.html"><a href="covariates.html"><i class="fa fa-check"></i><b>16</b> Linear models with added covariates (“ANCOVA”)</a>
<ul>
<li class="chapter" data-level="16.1" data-path="covariates.html"><a href="covariates.html#adding-covariates-can-increases-the-precision-of-the-effect-of-interest"><i class="fa fa-check"></i><b>16.1</b> Adding covariates can increases the precision of the effect of interest</a></li>
<li class="chapter" data-level="16.2" data-path="covariates.html"><a href="covariates.html#understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data"><i class="fa fa-check"></i><b>16.2</b> Understanding a linear model with an added covariate – heart necrosis data</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="covariates.html"><a href="covariates.html#fit-the-model-5"><i class="fa fa-check"></i><b>16.2.1</b> Fit the model</a></li>
<li class="chapter" data-level="16.2.2" data-path="covariates.html"><a href="covariates.html#plot-the-model-1"><i class="fa fa-check"></i><b>16.2.2</b> Plot the model</a></li>
<li class="chapter" data-level="16.2.3" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-model-coefficients"><i class="fa fa-check"></i><b>16.2.3</b> Interpretation of the model coefficients</a></li>
<li class="chapter" data-level="16.2.4" data-path="covariates.html"><a href="covariates.html#everything-adds-up"><i class="fa fa-check"></i><b>16.2.4</b> Everything adds up</a></li>
<li class="chapter" data-level="16.2.5" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-estimated-marginal-means"><i class="fa fa-check"></i><b>16.2.5</b> Interpretation of the estimated marginal means</a></li>
<li class="chapter" data-level="16.2.6" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-contrasts"><i class="fa fa-check"></i><b>16.2.6</b> Interpretation of the contrasts</a></li>
<li class="chapter" data-level="16.2.7" data-path="covariates.html"><a href="covariates.html#adding-the-covariate-improves-inference"><i class="fa fa-check"></i><b>16.2.7</b> Adding the covariate improves inference</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="covariates.html"><a href="covariates.html#understanding-interaction-effects-with-covariates"><i class="fa fa-check"></i><b>16.3</b> Understanding interaction effects with covariates</a>
<ul>
<li class="chapter" data-level="16.3.1" data-path="covariates.html"><a href="covariates.html#fit-the-model-6"><i class="fa fa-check"></i><b>16.3.1</b> Fit the model</a></li>
<li class="chapter" data-level="16.3.2" data-path="covariates.html"><a href="covariates.html#plot-the-model-with-interaction-effect"><i class="fa fa-check"></i><b>16.3.2</b> Plot the model with interaction effect</a></li>
<li class="chapter" data-level="16.3.3" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-model-coefficients-1"><i class="fa fa-check"></i><b>16.3.3</b> Interpretation of the model coefficients</a></li>
<li class="chapter" data-level="16.3.4" data-path="covariates.html"><a href="covariates.html#what-is-the-effect-of-a-treatment-if-interactions-are-modeled-it-depends."><i class="fa fa-check"></i><b>16.3.4</b> What is the effect of a treatment, if interactions are modeled? – it depends.</a></li>
<li class="chapter" data-level="16.3.5" data-path="covariates.html"><a href="covariates.html#which-model-do-we-use-mathcalm_1-or-mathcalm_2"><i class="fa fa-check"></i><b>16.3.5</b> Which model do we use, <span class="math inline">\(\mathcal{M}_1\)</span> or <span class="math inline">\(\mathcal{M}_2\)</span>?</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="covariates.html"><a href="covariates.html#understanding-ancova-tables"><i class="fa fa-check"></i><b>16.4</b> Understanding ANCOVA tables</a></li>
<li class="chapter" data-level="16.5" data-path="covariates.html"><a href="covariates.html#working-in-r-3"><i class="fa fa-check"></i><b>16.5</b> Working in R</a>
<ul>
<li class="chapter" data-level="16.5.1" data-path="covariates.html"><a href="covariates.html#importing-the-heart-necrosis-data"><i class="fa fa-check"></i><b>16.5.1</b> Importing the heart necrosis data</a></li>
<li class="chapter" data-level="16.5.2" data-path="covariates.html"><a href="covariates.html#fitting-the-model"><i class="fa fa-check"></i><b>16.5.2</b> Fitting the model</a></li>
<li class="chapter" data-level="16.5.3" data-path="covariates.html"><a href="covariates.html#using-the-emmeans-function-1"><i class="fa fa-check"></i><b>16.5.3</b> Using the emmeans function</a></li>
<li class="chapter" data-level="16.5.4" data-path="covariates.html"><a href="covariates.html#ancova-tables"><i class="fa fa-check"></i><b>16.5.4</b> ANCOVA tables</a></li>
<li class="chapter" data-level="16.5.5" data-path="covariates.html"><a href="covariates.html#plotting-the-model"><i class="fa fa-check"></i><b>16.5.5</b> Plotting the model</a></li>
</ul></li>
<li class="chapter" data-level="16.6" data-path="covariates.html"><a href="covariates.html#best-practices"><i class="fa fa-check"></i><b>16.6</b> Best practices</a>
<ul>
<li class="chapter" data-level="16.6.1" data-path="covariates.html"><a href="covariates.html#do-not-use-a-ratio-of-partwhole-as-a-response-variable-instead-add-the-denominator-as-a-covariate"><i class="fa fa-check"></i><b>16.6.1</b> Do not use a ratio of part:whole as a response variable – instead add the denominator as a covariate</a></li>
<li class="chapter" data-level="16.6.2" data-path="covariates.html"><a href="covariates.html#do-not-use-change-from-baseline-as-a-response-variable-instead-add-the-baseline-measure-as-a-covariate"><i class="fa fa-check"></i><b>16.6.2</b> Do not use change from baseline as a response variable – instead add the baseline measure as a covariate</a></li>
<li class="chapter" data-level="16.6.3" data-path="covariates.html"><a href="covariates.html#do-not-test-for-balance-of-baseline-measures"><i class="fa fa-check"></i><b>16.6.3</b> Do not “test for balance” of baseline measures</a></li>
</ul></li>
<li class="chapter" data-level="16.7" data-path="covariates.html"><a href="covariates.html#best-practices-2-use-a-covariate-instead-of-normalizing-a-response"><i class="fa fa-check"></i><b>16.7</b> Best practices 2: Use a covariate instead of normalizing a response</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="factorial.html"><a href="factorial.html"><i class="fa fa-check"></i><b>17</b> Linear models with two categorical <span class="math inline">\(X\)</span> – Factorial linear models (“two-way ANOVA”)</a>
<ul>
<li class="chapter" data-level="17.1" data-path="factorial.html"><a href="factorial.html#a-linear-model-with-crossed-factors-estimates-interaction-effects"><i class="fa fa-check"></i><b>17.1</b> A linear model with crossed factors estimates interaction effects</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="factorial.html"><a href="factorial.html#an-interaction-is-a-difference-in-simple-effects"><i class="fa fa-check"></i><b>17.1.1</b> An interaction is a difference in simple effects</a></li>
<li class="chapter" data-level="17.1.2" data-path="factorial.html"><a href="factorial.html#a-linear-model-with-crossed-factors-includes-interaction-effects"><i class="fa fa-check"></i><b>17.1.2</b> A linear model with crossed factors includes interaction effects</a></li>
<li class="chapter" data-level="17.1.3" data-path="factorial.html"><a href="factorial.html#factorial-experiments-are-frequently-analyzed-as-flattened-linear-models-in-the-experimental-biology-literature"><i class="fa fa-check"></i><b>17.1.3</b> factorial experiments are frequently analyzed as flattened linear models in the experimental biology literature</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="factorial.html"><a href="factorial.html#example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data"><i class="fa fa-check"></i><b>17.2</b> Example 1 – Estimation of a treatment effect relative to a control effect (“Something different”) (Experiment 2j glucose uptake data)</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="factorial.html"><a href="factorial.html#twoway-understand1"><i class="fa fa-check"></i><b>17.2.1</b> Understand the experimental design</a></li>
<li class="chapter" data-level="17.2.2" data-path="factorial.html"><a href="factorial.html#fit-the-linear-model"><i class="fa fa-check"></i><b>17.2.2</b> Fit the linear model</a></li>
<li class="chapter" data-level="17.2.3" data-path="factorial.html"><a href="factorial.html#twoway-exp2j-inf"><i class="fa fa-check"></i><b>17.2.3</b> Inference</a></li>
<li class="chapter" data-level="17.2.4" data-path="factorial.html"><a href="factorial.html#plot-the-model-2"><i class="fa fa-check"></i><b>17.2.4</b> Plot the model</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-1"><i class="fa fa-check"></i><b>17.3</b> Understanding the linear model with crossed factors 1</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="factorial.html"><a href="factorial.html#twoway-what-coefs-are"><i class="fa fa-check"></i><b>17.3.1</b> What the coefficients are</a></li>
<li class="chapter" data-level="17.3.2" data-path="factorial.html"><a href="factorial.html#the-interaction-effect-is-something-different"><i class="fa fa-check"></i><b>17.3.2</b> The interaction effect is something different</a></li>
<li class="chapter" data-level="17.3.3" data-path="factorial.html"><a href="factorial.html#why-we-want-to-compare-the-treatment-effect-to-a-control-effect"><i class="fa fa-check"></i><b>17.3.3</b> Why we want to compare the treatment effect to a control effect</a></li>
<li class="chapter" data-level="17.3.4" data-path="factorial.html"><a href="factorial.html#the-order-of-the-factors-in-the-model-tells-the-same-story-differently"><i class="fa fa-check"></i><b>17.3.4</b> The order of the factors in the model tells the same story differently</a></li>
<li class="chapter" data-level="17.3.5" data-path="factorial.html"><a href="factorial.html#power-for-the-interaction-effect-is-less-than-that-for-simple-effects"><i class="fa fa-check"></i><b>17.3.5</b> Power for the interaction effect is less than that for simple effects</a></li>
<li class="chapter" data-level="17.3.6" data-path="factorial.html"><a href="factorial.html#planned-comparisons-vs.-post-hoc-tests"><i class="fa fa-check"></i><b>17.3.6</b> Planned comparisons vs. post-hoc tests</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="factorial.html"><a href="factorial.html#example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data"><i class="fa fa-check"></i><b>17.4</b> Example 2: Estimation of the effect of background condition on an effect (“it depends”) (Experiment 3e lesian area data)</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="factorial.html"><a href="factorial.html#understand-the-experimental-design"><i class="fa fa-check"></i><b>17.4.1</b> Understand the experimental design</a></li>
<li class="chapter" data-level="17.4.2" data-path="factorial.html"><a href="factorial.html#fit-the-linear-model-1"><i class="fa fa-check"></i><b>17.4.2</b> Fit the linear model</a></li>
<li class="chapter" data-level="17.4.3" data-path="factorial.html"><a href="factorial.html#check-the-model-1"><i class="fa fa-check"></i><b>17.4.3</b> Check the model</a></li>
<li class="chapter" data-level="17.4.4" data-path="factorial.html"><a href="factorial.html#inference-from-the-model-2"><i class="fa fa-check"></i><b>17.4.4</b> Inference from the model</a></li>
<li class="chapter" data-level="17.4.5" data-path="factorial.html"><a href="factorial.html#plot-the-model-3"><i class="fa fa-check"></i><b>17.4.5</b> Plot the model</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-2"><i class="fa fa-check"></i><b>17.5</b> Understanding the linear model with crossed factors 2</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="factorial.html"><a href="factorial.html#twoway-marginal-means"><i class="fa fa-check"></i><b>17.5.1</b> Conditional and marginal means</a></li>
<li class="chapter" data-level="17.5.2" data-path="factorial.html"><a href="factorial.html#simple-conditional-effects"><i class="fa fa-check"></i><b>17.5.2</b> Simple (conditional) effects</a></li>
<li class="chapter" data-level="17.5.3" data-path="factorial.html"><a href="factorial.html#twoway-marginal-effects"><i class="fa fa-check"></i><b>17.5.3</b> Marginal effects</a></li>
<li class="chapter" data-level="17.5.4" data-path="factorial.html"><a href="factorial.html#the-additive-model"><i class="fa fa-check"></i><b>17.5.4</b> The additive model</a></li>
<li class="chapter" data-level="17.5.5" data-path="factorial.html"><a href="factorial.html#twoway-reduce"><i class="fa fa-check"></i><b>17.5.5</b> Reduce models for the right reason</a></li>
<li class="chapter" data-level="17.5.6" data-path="factorial.html"><a href="factorial.html#the-marginal-means-of-an-additive-linear-model-with-two-factors-can-be-weird"><i class="fa fa-check"></i><b>17.5.6</b> The marginal means of an additive linear model with two factors can be weird</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="factorial.html"><a href="factorial.html#example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data"><i class="fa fa-check"></i><b>17.6</b> Example 3: Estimation of synergy (“More than the sum of the parts”) (Experiment 1c JA data)</a>
<ul>
<li class="chapter" data-level="17.6.1" data-path="factorial.html"><a href="factorial.html#examine-the-data"><i class="fa fa-check"></i><b>17.6.1</b> Examine the data</a></li>
<li class="chapter" data-level="17.6.2" data-path="factorial.html"><a href="factorial.html#fit-the-model-7"><i class="fa fa-check"></i><b>17.6.2</b> Fit the model</a></li>
<li class="chapter" data-level="17.6.3" data-path="factorial.html"><a href="factorial.html#model-check"><i class="fa fa-check"></i><b>17.6.3</b> Model check</a></li>
<li class="chapter" data-level="17.6.4" data-path="factorial.html"><a href="factorial.html#inference-from-the-model-3"><i class="fa fa-check"></i><b>17.6.4</b> Inference from the model</a></li>
<li class="chapter" data-level="17.6.5" data-path="factorial.html"><a href="factorial.html#plot-the-model-4"><i class="fa fa-check"></i><b>17.6.5</b> Plot the model</a></li>
<li class="chapter" data-level="17.6.6" data-path="factorial.html"><a href="factorial.html#alternative-plot"><i class="fa fa-check"></i><b>17.6.6</b> Alternative plot</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-3"><i class="fa fa-check"></i><b>17.7</b> Understanding the linear model with crossed factors 3</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="factorial.html"><a href="factorial.html#thinking-about-the-coefficients-of-the-linear-model"><i class="fa fa-check"></i><b>17.7.1</b> Thinking about the coefficients of the linear model</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="factorial.html"><a href="factorial.html#issues-in-inference"><i class="fa fa-check"></i><b>17.8</b> Issues in inference</a>
<ul>
<li class="chapter" data-level="17.8.1" data-path="factorial.html"><a href="factorial.html#for-pairwise-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model"><i class="fa fa-check"></i><b>17.8.1</b> For pairwise contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</a></li>
<li class="chapter" data-level="17.8.2" data-path="factorial.html"><a href="factorial.html#for-interaction-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model"><i class="fa fa-check"></i><b>17.8.2</b> For interaction contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</a></li>
<li class="chapter" data-level="17.8.3" data-path="factorial.html"><a href="factorial.html#twoway-multiple-tests"><i class="fa fa-check"></i><b>17.8.3</b> Adjusting <em>p</em>-values for multiple tests</a></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="factorial.html"><a href="factorial.html#two-way-anova"><i class="fa fa-check"></i><b>17.9</b> Two-way ANOVA</a>
<ul>
<li class="chapter" data-level="17.9.1" data-path="factorial.html"><a href="factorial.html#how-to-read-a-two-way-anova-table"><i class="fa fa-check"></i><b>17.9.1</b> How to read a two-way ANOVA table</a></li>
<li class="chapter" data-level="17.9.2" data-path="factorial.html"><a href="factorial.html#what-do-the-main-effects-in-an-anova-table-mean"><i class="fa fa-check"></i><b>17.9.2</b> What do the main effects in an ANOVA table mean?</a></li>
</ul></li>
<li class="chapter" data-level="17.10" data-path="factorial.html"><a href="factorial.html#more-issues-in-inference"><i class="fa fa-check"></i><b>17.10</b> More issues in inference</a>
<ul>
<li class="chapter" data-level="17.10.1" data-path="factorial.html"><a href="factorial.html#longitudinal-experiments-include-time-as-a-random-factor-better-than-repeated-measures-anova"><i class="fa fa-check"></i><b>17.10.1</b> Longitudinal experiments – include Time as a random factor (better than repeated measures ANOVA)</a></li>
</ul></li>
<li class="chapter" data-level="17.11" data-path="factorial.html"><a href="factorial.html#working-in-r-4"><i class="fa fa-check"></i><b>17.11</b> Working in R</a>
<ul>
<li class="chapter" data-level="17.11.1" data-path="factorial.html"><a href="factorial.html#model-formula"><i class="fa fa-check"></i><b>17.11.1</b> Model formula</a></li>
<li class="chapter" data-level="17.11.2" data-path="factorial.html"><a href="factorial.html#using-the-emmeans-function-2"><i class="fa fa-check"></i><b>17.11.2</b> Using the emmeans function</a></li>
<li class="chapter" data-level="17.11.3" data-path="factorial.html"><a href="factorial.html#contrasts"><i class="fa fa-check"></i><b>17.11.3</b> Contrasts</a></li>
<li class="chapter" data-level="17.11.4" data-path="factorial.html"><a href="factorial.html#practice-safe-anova"><i class="fa fa-check"></i><b>17.11.4</b> Practice safe ANOVA</a></li>
<li class="chapter" data-level="17.11.5" data-path="factorial.html"><a href="factorial.html#better-to-avoid-these"><i class="fa fa-check"></i><b>17.11.5</b> Better to avoid these</a></li>
</ul></li>
<li class="chapter" data-level="17.12" data-path="factorial.html"><a href="factorial.html#hidden-code-4"><i class="fa fa-check"></i><b>17.12</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="17.12.1" data-path="factorial.html"><a href="factorial.html#import-exp2j-example-1"><i class="fa fa-check"></i><b>17.12.1</b> Import exp2j (Example 1)</a></li>
<li class="chapter" data-level="17.12.2" data-path="factorial.html"><a href="factorial.html#import-exp3e-lesian-area-data-example-2"><i class="fa fa-check"></i><b>17.12.2</b> Import exp3e lesian area data (Example 2)</a></li>
<li class="chapter" data-level="17.12.3" data-path="factorial.html"><a href="factorial.html#import-exp1c-ja-data-example-3"><i class="fa fa-check"></i><b>17.12.3</b> Import Exp1c JA data (Example 3)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-vii-expanding-the-linear-model.html"><a href="part-vii-expanding-the-linear-model.html"><i class="fa fa-check"></i>Part VII – Expanding the Linear Model</a></li>
<li class="chapter" data-level="18" data-path="lmm.html"><a href="lmm.html"><i class="fa fa-check"></i><b>18</b> Models with random factors – linear mixed models</a>
<ul>
<li class="chapter" data-level="18.1" data-path="lmm.html"><a href="lmm.html#lmm-demo"><i class="fa fa-check"></i><b>18.1</b> Example 1 – A random intercepts and slopes explainer (demo1)</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="lmm.html"><a href="lmm.html#batched-measurements-result-in-clustered-residuals"><i class="fa fa-check"></i><b>18.1.1</b> Batched measurements result in clustered residuals</a></li>
<li class="chapter" data-level="18.1.2" data-path="lmm.html"><a href="lmm.html#lmm-corerr"><i class="fa fa-check"></i><b>18.1.2</b> Clustered residuals result in correlated error</a></li>
<li class="chapter" data-level="18.1.3" data-path="lmm.html"><a href="lmm.html#in-blocked-designs-clustered-residuals-adds-a-variance-component-that-masks-treatment-effects"><i class="fa fa-check"></i><b>18.1.3</b> In blocked designs, clustered residuals adds a variance component that masks treatment effects</a></li>
<li class="chapter" data-level="18.1.4" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-are-linear-models-with-added-random-factors"><i class="fa fa-check"></i><b>18.1.4</b> Linear mixed models are linear models with added random factors</a></li>
<li class="chapter" data-level="18.1.5" data-path="lmm.html"><a href="lmm.html#what-the-random-effects-are"><i class="fa fa-check"></i><b>18.1.5</b> What the random effects are</a></li>
<li class="chapter" data-level="18.1.6" data-path="lmm.html"><a href="lmm.html#in-a-blocked-design-a-linear-model-with-added-random-effects-increases-precision-of-treatment-effects"><i class="fa fa-check"></i><b>18.1.6</b> In a blocked design, a linear model with added random effects increases precision of treatment effects</a></li>
<li class="chapter" data-level="18.1.7" data-path="lmm.html"><a href="lmm.html#lmm-varcorr"><i class="fa fa-check"></i><b>18.1.7</b> The correlation among random intercepts and slopes</a></li>
<li class="chapter" data-level="18.1.8" data-path="lmm.html"><a href="lmm.html#clustered-residuals-create-heterogeneity-among-treatments"><i class="fa fa-check"></i><b>18.1.8</b> Clustered residuals create heterogeneity among treatments</a></li>
<li class="chapter" data-level="18.1.9" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-are-flexible"><i class="fa fa-check"></i><b>18.1.9</b> Linear mixed models are flexible</a></li>
<li class="chapter" data-level="18.1.10" data-path="lmm.html"><a href="lmm.html#a-random-intercept-only-model"><i class="fa fa-check"></i><b>18.1.10</b> A random intercept only model</a></li>
<li class="chapter" data-level="18.1.11" data-path="lmm.html"><a href="lmm.html#a-model-including-an-interaction-intercept"><i class="fa fa-check"></i><b>18.1.11</b> A model including an interaction intercept</a></li>
<li class="chapter" data-level="18.1.12" data-path="lmm.html"><a href="lmm.html#lmm-demo-aic"><i class="fa fa-check"></i><b>18.1.12</b> AIC and model selection – which model to report?</a></li>
<li class="chapter" data-level="18.1.13" data-path="lmm.html"><a href="lmm.html#the-specification-of-random-effects-matters"><i class="fa fa-check"></i><b>18.1.13</b> The specification of random effects matters</a></li>
<li class="chapter" data-level="18.1.14" data-path="lmm.html"><a href="lmm.html#mixed-effect-and-repeated-measures-anova"><i class="fa fa-check"></i><b>18.1.14</b> Mixed Effect and Repeated Measures ANOVA</a></li>
<li class="chapter" data-level="18.1.15" data-path="lmm.html"><a href="lmm.html#pseudoreplication"><i class="fa fa-check"></i><b>18.1.15</b> Pseudoreplication</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="lmm.html"><a href="lmm.html#lmm-example2"><i class="fa fa-check"></i><b>18.2</b> Example 2 – experiments without subsampling replication (exp6g)</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="lmm.html"><a href="lmm.html#understand-the-data"><i class="fa fa-check"></i><b>18.2.1</b> Understand the data</a></li>
<li class="chapter" data-level="18.2.2" data-path="lmm.html"><a href="lmm.html#model-fit-and-inference"><i class="fa fa-check"></i><b>18.2.2</b> Model fit and inference</a></li>
<li class="chapter" data-level="18.2.3" data-path="lmm.html"><a href="lmm.html#the-model-exp6g_m1-adds-a-random-intercept-but-not-a-random-slope"><i class="fa fa-check"></i><b>18.2.3</b> The model exp6g_m1 adds a random intercept but not a random slope</a></li>
<li class="chapter" data-level="18.2.4" data-path="lmm.html"><a href="lmm.html#the-fixed-effect-coefficients-of-model-exp6g_m1"><i class="fa fa-check"></i><b>18.2.4</b> The fixed effect coefficients of model exp6g_m1</a></li>
<li class="chapter" data-level="18.2.5" data-path="lmm.html"><a href="lmm.html#the-random-intercept-coefficients-of-exp6g_m1"><i class="fa fa-check"></i><b>18.2.5</b> The random intercept coefficients of exp6g_m1</a></li>
<li class="chapter" data-level="18.2.6" data-path="lmm.html"><a href="lmm.html#the-random-and-residual-variance-and-the-intraclass-correlation-of-model-exp6g_m1"><i class="fa fa-check"></i><b>18.2.6</b> The random and residual variance and the intraclass correlation of model exp6g_m1</a></li>
<li class="chapter" data-level="18.2.7" data-path="lmm.html"><a href="lmm.html#the-linear-mixed-model-exp6g_m1-increases-precision-of-treatment-effects-relative-to-a-fixed-effects-model"><i class="fa fa-check"></i><b>18.2.7</b> The linear mixed model exp6g_m1 increases precision of treatment effects, relative to a fixed effects model</a></li>
<li class="chapter" data-level="18.2.8" data-path="lmm.html"><a href="lmm.html#lmm-exp6g-alt"><i class="fa fa-check"></i><b>18.2.8</b> Alternative models for exp6g</a></li>
<li class="chapter" data-level="18.2.9" data-path="lmm.html"><a href="lmm.html#paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models"><i class="fa fa-check"></i><b>18.2.9</b> Paired t-tests and repeated measures ANOVA are special cases of linear mixed models</a></li>
<li class="chapter" data-level="18.2.10" data-path="lmm.html"><a href="lmm.html#lmm-exp6g-rmanova"><i class="fa fa-check"></i><b>18.2.10</b> Classical (“univariate model”) repeated measures ANOVA of exp6g</a></li>
<li class="chapter" data-level="18.2.11" data-path="lmm.html"><a href="lmm.html#lmm-exp6g-rmanova-multi"><i class="fa fa-check"></i><b>18.2.11</b> “Multivariate model” repeated measures ANOVA</a></li>
<li class="chapter" data-level="18.2.12" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-vs-repeated-measures-anova"><i class="fa fa-check"></i><b>18.2.12</b> Linear mixed models vs repeated measures ANOVA</a></li>
<li class="chapter" data-level="18.2.13" data-path="lmm.html"><a href="lmm.html#modeling-textttmouse_id-as-a-fixed-effect"><i class="fa fa-check"></i><b>18.2.13</b> Modeling <span class="math inline">\(\texttt{mouse_id}\)</span> as a fixed effect</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="lmm.html"><a href="lmm.html#example-3-factorial-experiments-and-no-subsampling-replicates-exp5c"><i class="fa fa-check"></i><b>18.3</b> Example 3 – Factorial experiments and no subsampling replicates (exp5c)</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="lmm.html"><a href="lmm.html#understand-the-data-1"><i class="fa fa-check"></i><b>18.3.1</b> Understand the data</a></li>
<li class="chapter" data-level="18.3.2" data-path="lmm.html"><a href="lmm.html#examine-the-data-1"><i class="fa fa-check"></i><b>18.3.2</b> Examine the data</a></li>
<li class="chapter" data-level="18.3.3" data-path="lmm.html"><a href="lmm.html#lmm-exp5c-m1"><i class="fa fa-check"></i><b>18.3.3</b> Model fit and inference</a></li>
<li class="chapter" data-level="18.3.4" data-path="lmm.html"><a href="lmm.html#why-we-care-about-modeling-batch-in-exp5c"><i class="fa fa-check"></i><b>18.3.4</b> Why we care about modeling batch in exp5c</a></li>
<li class="chapter" data-level="18.3.5" data-path="lmm.html"><a href="lmm.html#the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts"><i class="fa fa-check"></i><b>18.3.5</b> The linear mixed model exp5c_m1 adds two random intercepts</a></li>
<li class="chapter" data-level="18.3.6" data-path="lmm.html"><a href="lmm.html#the-fixed-effect-coefficients-of-model-exp5c_m1"><i class="fa fa-check"></i><b>18.3.6</b> The fixed effect coefficients of model exp5c_m1</a></li>
<li class="chapter" data-level="18.3.7" data-path="lmm.html"><a href="lmm.html#the-random-effect-coefficients-of-model-exp5c_m1"><i class="fa fa-check"></i><b>18.3.7</b> The random effect coefficients of model exp5c_m1</a></li>
<li class="chapter" data-level="18.3.8" data-path="lmm.html"><a href="lmm.html#lmm-exp5c-alt"><i class="fa fa-check"></i><b>18.3.8</b> Alternative models for exp5c</a></li>
<li class="chapter" data-level="18.3.9" data-path="lmm.html"><a href="lmm.html#lmm-rmanova-5c"><i class="fa fa-check"></i><b>18.3.9</b> Classical (“univariate model”) repeated measures ANOVA</a></li>
<li class="chapter" data-level="18.3.10" data-path="lmm.html"><a href="lmm.html#lmm-rmanova-5c-multi"><i class="fa fa-check"></i><b>18.3.10</b> “Multivariate model” repeated measures ANOVA of exp5c</a></li>
<li class="chapter" data-level="18.3.11" data-path="lmm.html"><a href="lmm.html#modeling-textttdonor-as-a-fixed-effect"><i class="fa fa-check"></i><b>18.3.11</b> Modeling <span class="math inline">\(\texttt{donor}\)</span> as a fixed effect</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="lmm.html"><a href="lmm.html#lmm-example4"><i class="fa fa-check"></i><b>18.4</b> Example 4 – Experiments with subsampling replication (exp1g)</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="lmm.html"><a href="lmm.html#understand-the-data-2"><i class="fa fa-check"></i><b>18.4.1</b> Understand the data</a></li>
<li class="chapter" data-level="18.4.2" data-path="lmm.html"><a href="lmm.html#examine-the-data-2"><i class="fa fa-check"></i><b>18.4.2</b> Examine the data</a></li>
<li class="chapter" data-level="18.4.3" data-path="lmm.html"><a href="lmm.html#fit-the-model-9"><i class="fa fa-check"></i><b>18.4.3</b> Fit the model</a></li>
<li class="chapter" data-level="18.4.4" data-path="lmm.html"><a href="lmm.html#inference-from-the-model-6"><i class="fa fa-check"></i><b>18.4.4</b> Inference from the model</a></li>
<li class="chapter" data-level="18.4.5" data-path="lmm.html"><a href="lmm.html#plot-the-model-5"><i class="fa fa-check"></i><b>18.4.5</b> Plot the model</a></li>
<li class="chapter" data-level="18.4.6" data-path="lmm.html"><a href="lmm.html#alternaplot-the-model-2"><i class="fa fa-check"></i><b>18.4.6</b> Alternaplot the model</a></li>
<li class="chapter" data-level="18.4.7" data-path="lmm.html"><a href="lmm.html#understanding-the-alternative-models"><i class="fa fa-check"></i><b>18.4.7</b> Understanding the alternative models</a></li>
<li class="chapter" data-level="18.4.8" data-path="lmm.html"><a href="lmm.html#the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b"><i class="fa fa-check"></i><b>18.4.8</b> The VarCorr matrix of models exp1g_m1a and exp1g_m1b</a></li>
<li class="chapter" data-level="18.4.9" data-path="lmm.html"><a href="lmm.html#the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means"><i class="fa fa-check"></i><b>18.4.9</b> The linear mixed model has more precision and power than the fixed effect model of batch means</a></li>
<li class="chapter" data-level="18.4.10" data-path="lmm.html"><a href="lmm.html#fixed-effect-models-and-pseudoreplication"><i class="fa fa-check"></i><b>18.4.10</b> Fixed effect models and pseudoreplication</a></li>
<li class="chapter" data-level="18.4.11" data-path="lmm.html"><a href="lmm.html#mixed-effect-anova"><i class="fa fa-check"></i><b>18.4.11</b> Mixed-effect ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="lmm.html"><a href="lmm.html#working-in-r-5"><i class="fa fa-check"></i><b>18.5</b> Working in R</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="lmm.html"><a href="lmm.html#fitting-linear-mixed-models-with-lme4"><i class="fa fa-check"></i><b>18.5.1</b> Fitting linear mixed models with lme4</a></li>
<li class="chapter" data-level="18.5.2" data-path="lmm.html"><a href="lmm.html#repeated-measures-anova-randomized-complete-block-with-no-subsampling"><i class="fa fa-check"></i><b>18.5.2</b> Repeated measures ANOVA (randomized complete block with no subsampling)</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="lmm.html"><a href="lmm.html#hidden-code-5"><i class="fa fa-check"></i><b>18.6</b> Hidden code</a>
<ul>
<li class="chapter" data-level="18.6.1" data-path="lmm.html"><a href="lmm.html#import-exp5c"><i class="fa fa-check"></i><b>18.6.1</b> Import exp5c</a></li>
<li class="chapter" data-level="18.6.2" data-path="lmm.html"><a href="lmm.html#import-exp1g"><i class="fa fa-check"></i><b>18.6.2</b> Import exp1g</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="pre-post.html"><a href="pre-post.html"><i class="fa fa-check"></i><b>19</b> Linear models for longitudinal experiments – I. pre-post designs</a>
<ul>
<li class="chapter" data-level="19.1" data-path="pre-post.html"><a href="pre-post.html#best-practice-models"><i class="fa fa-check"></i><b>19.1</b> Best practice models</a></li>
<li class="chapter" data-level="19.2" data-path="pre-post.html"><a href="pre-post.html#common-alternatives-that-are-not-recommended"><i class="fa fa-check"></i><b>19.2</b> Common alternatives that are not recommended</a></li>
<li class="chapter" data-level="19.3" data-path="pre-post.html"><a href="pre-post.html#advanced-models"><i class="fa fa-check"></i><b>19.3</b> Advanced models</a></li>
<li class="chapter" data-level="19.4" data-path="pre-post.html"><a href="pre-post.html#understanding-the-alternative-models-1"><i class="fa fa-check"></i><b>19.4</b> Understanding the alternative models</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="pre-post.html"><a href="pre-post.html#m1-linear-model-with-the-baseline-measure-as-the-covariate-ancova-model"><i class="fa fa-check"></i><b>19.4.1</b> (M1) Linear model with the baseline measure as the covariate (ANCOVA model)</a></li>
<li class="chapter" data-level="19.4.2" data-path="pre-post.html"><a href="pre-post.html#m2-linear-model-of-the-change-score-change-score-model"><i class="fa fa-check"></i><b>19.4.2</b> (M2) Linear model of the change score (change-score model)</a></li>
<li class="chapter" data-level="19.4.3" data-path="pre-post.html"><a href="pre-post.html#m3-linear-model-of-post-baseline-values-without-the-baseline-as-a-covariate-post-model"><i class="fa fa-check"></i><b>19.4.3</b> (M3) Linear model of post-baseline values without the baseline as a covariate (post model)</a></li>
<li class="chapter" data-level="19.4.4" data-path="pre-post.html"><a href="pre-post.html#m4-linear-model-with-factorial-fixed-effects-fixed-effects-model"><i class="fa fa-check"></i><b>19.4.4</b> (M4) Linear model with factorial fixed effects (fixed-effects model)</a></li>
<li class="chapter" data-level="19.4.5" data-path="pre-post.html"><a href="pre-post.html#m5-repeated-measures-anova"><i class="fa fa-check"></i><b>19.4.5</b> (M5) Repeated measures ANOVA</a></li>
<li class="chapter" data-level="19.4.6" data-path="pre-post.html"><a href="pre-post.html#m6-linear-mixed-model"><i class="fa fa-check"></i><b>19.4.6</b> (M6) Linear mixed model</a></li>
<li class="chapter" data-level="19.4.7" data-path="pre-post.html"><a href="pre-post.html#m7-linear-model-with-correlated-error"><i class="fa fa-check"></i><b>19.4.7</b> (M7) Linear model with correlated error</a></li>
<li class="chapter" data-level="19.4.8" data-path="pre-post.html"><a href="pre-post.html#m8-constrained-fixed-effects-model-with-correlated-error-clda-model"><i class="fa fa-check"></i><b>19.4.8</b> (M8) Constrained fixed effects model with correlated error (cLDA model)</a></li>
<li class="chapter" data-level="19.4.9" data-path="pre-post.html"><a href="pre-post.html#comparison-table"><i class="fa fa-check"></i><b>19.4.9</b> Comparison table</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="pre-post.html"><a href="pre-post.html#example-1-a-single-post-baseline-measure-pre-post-design"><i class="fa fa-check"></i><b>19.5</b> Example 1 – a single post-baseline measure (pre-post design)</a></li>
<li class="chapter" data-level="19.6" data-path="pre-post.html"><a href="pre-post.html#working-in-r-6"><i class="fa fa-check"></i><b>19.6</b> Working in R</a></li>
<li class="chapter" data-level="19.7" data-path="pre-post.html"><a href="pre-post.html#hidden-code-6"><i class="fa fa-check"></i><b>19.7</b> Hidden code</a>
<ul>
<li class="chapter" data-level="19.7.1" data-path="pre-post.html"><a href="pre-post.html#import-and-wrangle-mouse-sociability-data"><i class="fa fa-check"></i><b>19.7.1</b> Import and wrangle mouse sociability data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="counts.html"><a href="counts.html"><i class="fa fa-check"></i><b>20</b> Linear models for count data – Generalized Linear Models I</a>
<ul>
<li class="chapter" data-level="20.1" data-path="counts.html"><a href="counts.html#the-generalized-linear-model"><i class="fa fa-check"></i><b>20.1</b> The generalized linear model</a></li>
<li class="chapter" data-level="20.2" data-path="counts.html"><a href="counts.html#count-data-example-number-of-trematode-worm-larvae-in-eyes-of-threespine-stickleback-fish"><i class="fa fa-check"></i><b>20.2</b> Count data example – number of trematode worm larvae in eyes of threespine stickleback fish</a>
<ul>
<li class="chapter" data-level="20.2.1" data-path="counts.html"><a href="counts.html#modeling-strategy-1"><i class="fa fa-check"></i><b>20.2.1</b> Modeling strategy</a></li>
<li class="chapter" data-level="20.2.2" data-path="counts.html"><a href="counts.html#checking-the-model-i-a-normal-q-q-plot"><i class="fa fa-check"></i><b>20.2.2</b> Checking the model I – a Normal Q-Q plot</a></li>
<li class="chapter" data-level="20.2.3" data-path="counts.html"><a href="counts.html#checking-the-model-ii-scale-location-plot-for-checking-homoskedasticity"><i class="fa fa-check"></i><b>20.2.3</b> Checking the model II – scale-location plot for checking homoskedasticity</a></li>
<li class="chapter" data-level="20.2.4" data-path="counts.html"><a href="counts.html#two-distributions-for-count-data-poisson-and-negative-binomial"><i class="fa fa-check"></i><b>20.2.4</b> Two distributions for count data – Poisson and Negative Binomial</a></li>
<li class="chapter" data-level="20.2.5" data-path="counts.html"><a href="counts.html#fitting-a-glm-with-a-poisson-distribution-to-the-worm-data"><i class="fa fa-check"></i><b>20.2.5</b> Fitting a GLM with a Poisson distribution to the worm data</a></li>
<li class="chapter" data-level="20.2.6" data-path="counts.html"><a href="counts.html#model-checking-fits-to-count-data"><i class="fa fa-check"></i><b>20.2.6</b> Model checking fits to count data</a></li>
<li class="chapter" data-level="20.2.7" data-path="counts.html"><a href="counts.html#fitting-a-glm-with-a-negative-binomial-distribution-to-the-worm-data"><i class="fa fa-check"></i><b>20.2.7</b> Fitting a GLM with a Negative Binomial distribution to the worm data</a></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="counts.html"><a href="counts.html#working-in-r-7"><i class="fa fa-check"></i><b>20.3</b> Working in R</a>
<ul>
<li class="chapter" data-level="20.3.1" data-path="counts.html"><a href="counts.html#fitting-a-glm-to-count-data"><i class="fa fa-check"></i><b>20.3.1</b> Fitting a GLM to count data</a></li>
<li class="chapter" data-level="20.3.2" data-path="counts.html"><a href="counts.html#fitting-a-generalized-linear-mixed-model-glmm-to-count-data"><i class="fa fa-check"></i><b>20.3.2</b> Fitting a generalized linear mixed model (GLMM) to count data</a></li>
<li class="chapter" data-level="20.3.3" data-path="counts.html"><a href="counts.html#fitting-a-generalized-linear-model-to-continouus-data"><i class="fa fa-check"></i><b>20.3.3</b> Fitting a generalized linear model to continouus data</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="counts.html"><a href="counts.html#problems-1"><i class="fa fa-check"></i><b>20.4</b> Problems</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="gls.html"><a href="gls.html"><i class="fa fa-check"></i><b>21</b> Linear models with heterogenous variance</a></li>
<li class="chapter" data-level="22" data-path="count-sim.html"><a href="count-sim.html"><i class="fa fa-check"></i><b>22</b> Simulations – Count data (alternatives to a t-test)</a>
<ul>
<li class="chapter" data-level="22.1" data-path="count-sim.html"><a href="count-sim.html#use-data-similar-to-figure-6f-from-example-1"><i class="fa fa-check"></i><b>22.1</b> Use data similar to Figure 6f from Example 1</a></li>
<li class="chapter" data-level="22.2" data-path="count-sim.html"><a href="count-sim.html#functions"><i class="fa fa-check"></i><b>22.2</b> Functions</a></li>
<li class="chapter" data-level="22.3" data-path="count-sim.html"><a href="count-sim.html#simulations"><i class="fa fa-check"></i><b>22.3</b> Simulations</a>
<ul>
<li class="chapter" data-level="22.3.1" data-path="count-sim.html"><a href="count-sim.html#type-i-pseudo-normal-distribution"><i class="fa fa-check"></i><b>22.3.1</b> Type I, Pseudo-Normal distribution</a></li>
<li class="chapter" data-level="22.3.2" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-equal-n"><i class="fa fa-check"></i><b>22.3.2</b> Type I, neg binom, equal n</a></li>
<li class="chapter" data-level="22.3.3" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-equal-n-small-theta"><i class="fa fa-check"></i><b>22.3.3</b> Type I, neg binom, equal n, small theta</a></li>
<li class="chapter" data-level="22.3.4" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-unequal-n"><i class="fa fa-check"></i><b>22.3.4</b> Type I, neg binom, unequal n</a></li>
<li class="chapter" data-level="22.3.5" data-path="count-sim.html"><a href="count-sim.html#power-pseudo-normal-distribution-equal-n"><i class="fa fa-check"></i><b>22.3.5</b> Power, Pseudo-Normal distribution, equal n</a></li>
<li class="chapter" data-level="22.3.6" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-equal-n"><i class="fa fa-check"></i><b>22.3.6</b> Power, neg binom, equal n</a></li>
<li class="chapter" data-level="22.3.7" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-small-theta"><i class="fa fa-check"></i><b>22.3.7</b> Power, neg binom, small theta</a></li>
<li class="chapter" data-level="22.3.8" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-unequal-n"><i class="fa fa-check"></i><b>22.3.8</b> Power, neg binom, unequal n</a></li>
<li class="chapter" data-level="22.3.9" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-unequal-n-unequal-theta"><i class="fa fa-check"></i><b>22.3.9</b> Power, neg binom, unequal n, unequal theta</a></li>
<li class="chapter" data-level="22.3.10" data-path="count-sim.html"><a href="count-sim.html#type-1-neg-binom-equal-n-unequal-theta"><i class="fa fa-check"></i><b>22.3.10</b> Type 1, neg binom, equal n, unequal theta</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="count-sim.html"><a href="count-sim.html#save-it-read-it"><i class="fa fa-check"></i><b>22.4</b> Save it, Read it</a></li>
<li class="chapter" data-level="22.5" data-path="count-sim.html"><a href="count-sim.html#analysis"><i class="fa fa-check"></i><b>22.5</b> Analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html"><i class="fa fa-check"></i>Appendix 1: Getting Started with R</a>
<ul>
<li class="chapter" data-level="22.6" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#get-your-computer-ready"><i class="fa fa-check"></i><b>22.6</b> Get your computer ready</a>
<ul>
<li class="chapter" data-level="22.6.1" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#start-here"><i class="fa fa-check"></i><b>22.6.1</b> Start here</a></li>
<li class="chapter" data-level="22.6.2" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r"><i class="fa fa-check"></i><b>22.6.2</b> Install R</a></li>
<li class="chapter" data-level="22.6.3" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r-studio"><i class="fa fa-check"></i><b>22.6.3</b> Install R Studio</a></li>
<li class="chapter" data-level="22.6.4" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r-markdown-1"><i class="fa fa-check"></i><b>22.6.4</b> Install R Markdown</a></li>
<li class="chapter" data-level="22.6.5" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#optional-alternative-latex-installations"><i class="fa fa-check"></i><b>22.6.5</b> (optional) Alternative LaTeX installations</a></li>
</ul></li>
<li class="chapter" data-level="22.7" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#start-learning-r-studio"><i class="fa fa-check"></i><b>22.7</b> Start learning R Studio</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix-2-online-resources-for-getting-started-with-statistical-modeling-in-r.html"><a href="appendix-2-online-resources-for-getting-started-with-statistical-modeling-in-r.html"><i class="fa fa-check"></i>Appendix 2: Online Resources for Getting Started with Statistical Modeling in R</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Elements of Statistical Modeling for Experimental Biology</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="lmm" class="section level1" number="18">
<h1><span class="header-section-number">Chapter 18</span> Models with random factors – linear mixed models</h1>
<p>This chapter is about linear models with added random factors, or <strong>linear mixed models</strong>. In classical hypothesis testing, a <strong>paired t-test</strong>, <strong>repeated measures ANOVA</strong>, and <strong>mixed-effect ANOVA</strong> are equivalent to specific cases of linear mixed models. Linear mixed models are for analyzing data composed of subsets – or <strong>batches</strong> – of data that were measured from the “same thing.” Batched data results in <strong>correlated error</strong> that muddles statistical inference unless the correlated error is modeled, explicitly or implicitly. In some designs, modeling the correlated error increases precision and power, and ultimately decreases false discovery. In other designs, failure to model the correlated error results in incorrectly high precision and incorrectly low <em>p</em>-values, leading to increased rates of false discovery. I think it’s fair to infer from the experimental biology literature, that experimental biologists don’t recognize the ubiquitousness of batched data and correlated error. This is probably the biggest issue in inference in the field (far more of an issue than say, a <em>t</em>-test on non-normal data).</p>
<p>What do I mean by “batch” and how can correlated error both increase and decrease false discovery? Consider these experiments:</p>
<ol style="list-style-type: decimal">
<li>The experiment in Figure <a href="#fig:lmm-biological-replicates"><strong>??</strong></a> is a factorial design with two factors, <span class="math inline">\(\texttt{genotype}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>, each with two levels. One mice of each treatment combination is randomly assigned to a cage. There are five replicates of each cage. The five replicate mice per treatment combination are <strong>biological replicates</strong>. Each cage is a batch. Each cage has a unique set of factors that contribute to the error variance of the measure of the response. Each cage shares a cage-specific history of temperature, humidity, food, light, interactions with animal facilities staff and behavioral interactions among the mice. All response measures within a cage share the component of the error variance unique to that cage and, as a consequence, the error (residuals) within a cage are more similar to each other than they are to the residuals between cages.</li>
</ol>
<p><img src="images/lmm-biological-replicates-435-273.png" width="63%" /></p>
<ol start="2" style="list-style-type: decimal">
<li>The experiment in Figure <a href="#fig:lmm-technical-replicates"><strong>??</strong></a> is a factorial design with two factors, <span class="math inline">\(\texttt{genotype}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>, each with two levels. One mice of each treatment combination is randomly assigned to a cage. There are five replicates of each cage. The researchers take three measures of the response variable per mouse. The three measures are <strong>subsampled replicates</strong> (most often referred to as <strong>subsamples</strong>). The subsamples could be <strong>technical replicates</strong> if multiple measures are taken from the same prep or <strong>repeated measures</strong> if the the multiple measures are taken at different time points. In addition to each cage being a batch, each mouse is a batch. Each mouse has a unique set of factors that contribute to the error variance of the measures of the response in that mouse. All response measures within a mouse share the component of the error variance unique to that mouse and, as a consequence, the error (residuals) within a mouse are more similar to each other than they are to the residuals between mice</li>
</ol>
<p><img src="images/lmm-technical-replicates-435-273.png" width="63%" /></p>
<ol start="3" style="list-style-type: decimal">
<li>The experiment in Figure <a href="#fig:lmm-segregated"><strong>??</strong></a> has a single factor <span class="math inline">\(\texttt{treatment}\)</span> with two levels. Importantly, the treatment (example: diet) is randomly assigned to cage and all mice in the cage have this treatment. The five mice per cage are <strong>subsampled replicates</strong> (<strong>subsamples</strong>). As in experiments 1 and 2, each cage is a batch. Each cage has a unique set of factors that contribute to the error variance of the measures of the response in that cage. All response measures within a cage share the component of the error variance unique to that cage and, as a consequence, the error (residuals) within a cage are more similar to each other than they are to the residuals among cages.</li>
</ol>
<p><img src="images/lmm-pseudoreplication.png" width="63%" /></p>
<p>In each of these experiments, there is systematic variation at multiple levels: among treatments due to treatment effects and among batches due to <strong>batch effects</strong>. Batches come in lots of flavors, including experiment, cage, flask, plate, slide, donor, and individual. The among-batch variation is the <strong>random effect</strong>. An assumption of modeling random effects is that the batches are a random sample of the batches that could have been sampled. This is often not strictly true as batches are often <strong>convenience samples</strong> (example: the human donors of the Type 2 diabetes beta cells are those that were in the hospital).</p>
<p>The variation among batches/lack of independence within batches has different consequences on the uncertainty of the estimate of a treatment effect. The batches in Experiment 1 contain all treatment combinations. The researcher is interested in the treatment effect but not the variation due to differences among the batches. The batches are nuissance factors that add additional variance to the response, with the consequence that estimates of treatment effects are less precise, unless the variance due to the batches is explicitly modeled. <strong>Modeling a batch that contains some or all treatment combinations will increase precision and power</strong>.</p>
<p>Batches that contain at least two treatment combinations are known as <strong>blocks</strong>. A block that contains all treatment combinations is a <strong>complete block</strong>. A block that contains fewer than all combinations is an <strong>incomplete block</strong>. Including block structure in the design is known as <strong>blocking</strong>. Adding a blocking factor to a statistical model is used to increase the precision of an estimated treatment effect. Experiment 1 is an example of a <strong>randomized complete block</strong> design.</p>
<p>In Experiment 2, there are multiple measures per mouse and the design is a <strong>randomized complete block with subsampling</strong>. The subsampling is not the kind of replication that can be used to infer the among treatment effect because the treatment assignment was not at the level of the subsamples. The <strong>treatment replicates</strong> (the cage) are the blocks, because <em>it was at this level that treatment assignment was randomized</em>. Nevertheless, there are inference advantages to subsampling if modeled, But, a statistical analysis of all measures from a subsampled design without modeling the correlated error due to the subsampling is a kind of <a href="https://en.wikipedia.org/wiki/Pseudoreplication" target="_blank">pseudoreplication</a>. Pseudoreplication results in incorrectly small standard errors and <em>p</em>-values and increased rates of false discovery.</p>
<p>In Experiment 3, the treatment is randomized <em>to</em> batch, so each batch contains only a single treatment level. In these <strong>segregated</strong> experimental designs, the variation among batches that arises from non-treatment related differences among batches <strong>confounds</strong> the variation among batches due to a true treatment effect. An extreme example of this would be an experiment with only a single cage with control conditions and a single cage with treatment conditions. Imagine 1) the true effect of the treatment is zero and 2) an aggressive mouse in the control cage stimulates the stress response in the other mice and this stress response has a large effect on the value of the response variable measured by the researchers. The researcher is fooled into thinking that the treatment caused the difference in the response. Again, mice are subsampled replicates while treatment replicates are at the level of the cage, because it was at this level that treatment assignment was randomized. This means the researcher has a single, treatment replicate (or, <span class="math inline">\(n=1\)</span>), regardless of the number of mice (subsamples) in each cage. A statistical analysis of all measures from a subsampled design without modeling the correlated error due to the subsampling is a kind of <a href="https://en.wikipedia.org/wiki/Pseudoreplication" target="_blank">pseudoreplication</a>. Pseudoreplication results in incorrectly small standard errors and <em>p</em>-values and increased rates of false discovery.</p>
<div id="lmm-demo" class="section level2" number="18.1">
<h2><span class="header-section-number">18.1</span> Example 1 – A random intercepts and slopes explainer (demo1)</h2>
<p>To introduce linear mixed models, I’m using data from <a href="lmm.html#lmm-example4">Experiment 1g</a> below. The design is <span class="math inline">\(2 \times 2\)</span> factorial with 4-5 mice per treatment combination. To simplify the explanation of <strong>random intercepts</strong> and <strong>random slopes</strong> in linear models with added random factors (<strong>linear mixed models</strong>), I flatten the analysis to a single treatment factor (<span class="math inline">\(\texttt{treatment}\)</span>) with four levels (“Control,” “Tr1,” “Tr2,” "Tr3). The response is percent germinal centers (<span class="math inline">\(\texttt{gc}\)</span>) in <a href="https://en.wikipedia.org/wiki/Lymphatic_system#Secondary_lymphoid_organs" target="_blank">secondary lymphoid tissue</a>. The experiment was replicated 4 times. Each replication is a batch. This batch information is in the variable <span class="math inline">\(\texttt{experiment_id}\)</span>. Further detail isn’t necessary at this point.</p>
<p>Figure <a href="lmm.html#fig:lmm-explainer1a">18.1</a>A is a response plot of the linear model <code>lm(gc ~ treatment)</code> fit to the whole data set, ignoring the fact that the data were collected in batches. This is the <strong>complete pooling</strong> fit. Figure <a href="lmm.html#fig:lmm-explainer1a">18.1</a>B is a response plot of the linear model <code>lm(gc ~ treatment)</code> fit to the means of each treatment combination from each experiment. This is the <strong>means pooling</strong> fit.</p>
<div class="figure"><span id="fig:lmm-explainer1a"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-explainer1a-1.png" alt="A. Response plot of the linear model gc ~ treatment fit to all exp1g data. B. Response plot of the linear model gc ~ treatment fit to the experiment means of the exp1g data." width="1066.66666666667" />
<p class="caption">
Figure 18.1: A. Response plot of the linear model gc ~ treatment fit to all exp1g data. B. Response plot of the linear model gc ~ treatment fit to the experiment means of the exp1g data.
</p>
</div>
<div id="batched-measurements-result-in-clustered-residuals" class="section level3" number="18.1.1">
<h3><span class="header-section-number">18.1.1</span> Batched measurements result in clustered residuals</h3>
<p>Figure <a href="lmm.html#fig:lmm-explainer1b">18.2</a>A is a plot of the residuals of the complete-pooling fit against <span class="math inline">\(\texttt{experiment_id}\)</span>. The residuals are <strong>clustered</strong> by experiment. All residuals from experiment 1 are positive. All residuals from experiment 2 are negative. Residuals from experiment 3 are generally positive. Residuals from experiment 4 seem pretty random. This clustering by experiment is the same in the plot of the residuals of the means-pooling fit against <span class="math inline">\(\texttt{experiment_id}\)</span> (Figure <a href="lmm.html#fig:lmm-explainer1b">18.2</a>B). The residuals are not independent in either fit. If you asked me to guess the sign of a residual and gave me the information that the measure was from experiment 1, I’d be correct 100% of the time. If the residuals were independent, I’d be correct, on average, 50% of the time. Independent residuals are randomly scattered about zero for within each experiment (Figure <a href="lmm.html#fig:lmm-explainer1b">18.2</a>C).</p>
<div class="figure"><span id="fig:lmm-explainer1b"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-explainer1b-1.png" alt="A. Residuals of the model fit to all demo1 data. B. Residuals of the model fit to the mean demo1 data." width="1066.66666666667" />
<p class="caption">
Figure 18.2: A. Residuals of the model fit to all demo1 data. B. Residuals of the model fit to the mean demo1 data.
</p>
</div>
</div>
<div id="lmm-corerr" class="section level3" number="18.1.2">
<h3><span class="header-section-number">18.1.2</span> Clustered residuals result in correlated error</h3>
<p>An assumption of inference from a linear model is independence – each response is independently drawn from a distribution of random values. In Experiment 1g, the experiments are batches and the batched data results in correlated error unless modeled. One way to see this correlated error is to use the residuals from the means-pooled fit.</p>
<ol style="list-style-type: decimal">
<li>aggregate the data by computing the means for each treatment level within each experiment.</li>
<li>fit the fixed effect model (the model without added random factors) to the aggregated data</li>
<li>compute the residuals from the model</li>
<li>cast (or spread) the residuals for each treatment into its own column. This creates a 4 rows (experiment) <span class="math inline">\(\times\)</span> 4 columns (treatments) matrix of residuals.</li>
<li>Compute the correlations among the four treatment combination columns. This is the correlated error due to the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-demo1-cor-error-table">Table 18.1: </span>Residuals of fixed affect model fit to aggregated data. The residuals are split into each treatment.
</caption>
<thead>
<tr>
<th style="text-align:left;">
experiment_id
</th>
<th style="text-align:right;">
Control
</th>
<th style="text-align:right;">
Tr1
</th>
<th style="text-align:right;">
Tr2
</th>
<th style="text-align:right;">
Tr3
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
exp_1
</td>
<td style="text-align:right;">
2.578125
</td>
<td style="text-align:right;">
1.64075
</td>
<td style="text-align:right;">
2.2425
</td>
<td style="text-align:right;">
3.027125
</td>
</tr>
<tr>
<td style="text-align:left;">
exp_2
</td>
<td style="text-align:right;">
-2.229375
</td>
<td style="text-align:right;">
-2.91625
</td>
<td style="text-align:right;">
-5.7475
</td>
<td style="text-align:right;">
-4.435375
</td>
</tr>
<tr>
<td style="text-align:left;">
exp_3
</td>
<td style="text-align:right;">
-0.615375
</td>
<td style="text-align:right;">
1.09975
</td>
<td style="text-align:right;">
3.4625
</td>
<td style="text-align:right;">
3.697125
</td>
</tr>
<tr>
<td style="text-align:left;">
exp_4
</td>
<td style="text-align:right;">
0.266625
</td>
<td style="text-align:right;">
0.17575
</td>
<td style="text-align:right;">
0.0425
</td>
<td style="text-align:right;">
-2.288875
</td>
</tr>
</tbody>
</table>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-demo1-cor-error-1.png" width="576" /></p>
<p>I’ve used <code>GGally::ggpairs</code> to compute and display the correlations as a matrix. The lower triangle of matrix elements contains the scatterplot of the residuals for the treatment combination defined by the row and column headers. The upper triangle of elements contains the Pearson correlation. With only four experiment residuals per treatment combination, large correlations will be common. But all correlations are large, positive values. The asterisks indicate values that would be an unexpected surprise under a null model of no correlation.</p>
<p>We could explicitly model correlated error with a <strong>linear model for correlated error</strong> using the <code>nlme::gls</code> function, using a model for the correlated error that matches our knowledge of how the data were generated (from experiment batches). In this chapter, we implicitly model the correlated error using a linear model with added random factors – a <strong>linear mixed model</strong>. What we explicitly model in a linear mixed model is hierarchical levels of variance.</p>
</div>
<div id="in-blocked-designs-clustered-residuals-adds-a-variance-component-that-masks-treatment-effects" class="section level3" number="18.1.3">
<h3><span class="header-section-number">18.1.3</span> In blocked designs, clustered residuals adds a variance component that masks treatment effects</h3>
<p>The variance among the experiments within a treatment is much greater than the variance among the treatment means. A consequence of this is, the experiment effect masks the effect of treatment. We can manually unmask this by</p>
<ol style="list-style-type: decimal">
<li>compute the experiment means across all treatment combinations.</li>
<li>create a gc variable without variation among experiment means (“adjusted for experiment_id”).</li>
</ol>
<div class="figure"><span id="fig:lmm-demo1-manual-lmm-plot"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-demo1-manual-lmm-plot-1.png" alt="Adjusting for variance among experiments. The black, dashed line is the grand-mean response. In the left panel, the colored, dashed lines are the mean gc for each experiment, ignoring treatment. In the right panel, the individual values have been shifted (adjusted) by centering the experiment means. This has the effect of reducing the error variance -- the spread of the values around the treatment means (large black dots)." width="576" />
<p class="caption">
Figure 18.3: Adjusting for variance among experiments. The black, dashed line is the grand-mean response. In the left panel, the colored, dashed lines are the mean gc for each experiment, ignoring treatment. In the right panel, the individual values have been shifted (adjusted) by centering the experiment means. This has the effect of reducing the error variance – the spread of the values around the treatment means (large black dots).
</p>
</div>
<p>In Figure <a href="lmm.html#fig:lmm-demo1-manual-lmm-plot">18.3</a>, the black dots are the modeled means of each treatment combination. The small colored dots are the measured values of the response for each <span class="math inline">\(\texttt{experiment_id}\)</span> in the left panel and the experiment-adjusted values in the right panel. The black, dashed line is the grand-mean response. The colored, dashed lines are the means of all responses in each experiment. These means are equal in the right panel (and covered by the black line) because the variation among the means has been adjusted away. What is left is error variation uncontaminated by <span class="math inline">\(\texttt{experiment_id}\)</span>.</p>
<p>In Experiment 1g, <span class="math inline">\(\texttt{experiment_id}\)</span> is a nuissance variable – it adds to the noise. In the exercise above, the effects of the treatment variables are adjusted for the elevation of batch effects on the overal batch mean. Linear mixed models are more sophisticated than this. In a linear mixed model, the effects of the treatment variables are adjusted for the elevations of batch effects on the intercept and batch effects on the slopes (or some combination of these). These are the <strong>random intercepts</strong> and <strong>random slopes</strong>.</p>
</div>
<div id="linear-mixed-models-are-linear-models-with-added-random-factors" class="section level3" number="18.1.4">
<h3><span class="header-section-number">18.1.4</span> Linear mixed models are linear models with added random factors</h3>
<p>A linear model adds some combination of <strong>random intercepts</strong> and <strong>random slopes</strong> to a linear model.</p>
<p><span class="math display" id="eq:lmm-demo1-m1">\[
\begin{equation}
\texttt{gc}_{jk} = (\beta_{0} + \gamma_{0j}) + (\beta_{k} + \gamma_{kj}) \texttt{treatment}_{k} + \varepsilon 
\tag{18.1}
\end{equation}
\]</span></p>
<p>A <strong>random intercept</strong> for experiment <em>j</em> is the sum of the fixed intercept (<span class="math inline">\(\beta_0\)</span>) and a <strong>random intercept effect</strong> (<span class="math inline">\(\gamma_{0j}\)</span>). I’ve embedded these within parentheses to show how these combine into the random intercept. A <strong>random slope</strong> for batch <em>j</em> is the sum of the fixed slope (<span class="math inline">\(\beta_k\)</span>) for the non-reference level <span class="math inline">\(k\)</span> and a <strong>random slope effect</strong> (<span class="math inline">\(\gamma_{kj}\)</span>). I’ve embedded these within parentheses to show how these combine into the random slopes.</p>
<p>There is a different <span class="math inline">\(\gamma_{0j}\)</span> for each experiment. There is a different <span class="math inline">\(\gamma_{kj}\)</span> for each combination of non-reference level and experiment. The <span class="math inline">\(\gamma_0j\)</span> and <span class="math inline">\(\gamma_kj\)</span> are modeled as if the values for each experiment is a random draw from an infinite number of experiments. This is why <span class="math inline">\(\gamma_{0j}\)</span> and <span class="math inline">\(\gamma_{kj}\)</span> are <strong>random effects</strong>. In contrast, <span class="math inline">\(\beta_0\)</span> and the three <span class="math inline">\(\beta_k\)</span> for the non-reference treatment levels are the same for all experiments – this is why <span class="math inline">\(\beta_0\)</span> is known as <strong>fixed effects</strong> (technically, <span class="math inline">\(\beta_0\)</span> is a mean and not an effect).</p>
<p>Model <a href="lmm.html#eq:lmm-demo1-m1">(18.1)</a> is fit to the Example 1 data using <code>lme4::lmer()</code></p>
<div class="sourceCode" id="cb525"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb525-1"><a href="lmm.html#cb525-1" aria-hidden="true" tabindex="-1"></a>demo1_m1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> treatment <span class="sc">+</span></span>
<span id="cb525-2"><a href="lmm.html#cb525-2" aria-hidden="true" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb525-3"><a href="lmm.html#cb525-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li><code>(treatment | experiment_id)</code> specifies a random intercept for all levels of <span class="math inline">\(\textt{experiment_id}\)</span> and
a random slope for all combinations the levels of <span class="math inline">\(\textt{experiment_id}\)</span> and the non-reference levels of <span class="math inline">\(\textt{treatment}\)</span></li>
</ol>
</div>
<div id="what-the-random-effects-are" class="section level3" number="18.1.5">
<h3><span class="header-section-number">18.1.5</span> What the random effects are</h3>
<p><strong>Random intercepts model batch effects in the reference treatment level</strong>. <a href="lmm.html#fig:lmm-demo1-explainer-2c">18.4</a>A illustrates random intercepts and random intercept effects. The large, colored dots are the modeled means of each experiment for each treatment combination. For the reference treatment level (“Control”), each mean is the sum of the estimated fixed intercept (<span class="math inline">\(b_0\)</span>), shown by the dashed gray line, and the estimated random intercept effect (<span class="math inline">\(\g_{0j}\)</span>) for experiment <em>j</em>. The random intercept effects are the vertical, colored lines.</p>
<p><strong>Random slopes model the effect of treatment on batch effects in the non-reference treatment levels</strong>. <a href="lmm.html#fig:lmm-demo1-explainer-2c">18.4</a>B illustrates random slopes and random slope effects, focusing on the slopes for the 2nd non-reference treatment level (“Tr2”). The angled black line is the estimated fixed slope <span class="math inline">\(b_2\)</span> for this level. The colored lines are the random slopes for each experiment. The pale, gray dots are where the modeled means at the Tr2 level would be if there were no random slope effect – as if we took the large colored dots at “Control” and rigidly shifted them up the black line to “Tr2.” The estimated random slope effects <span class="math inline">\(\mathrm{g}_{2j}\)</span> are the difference between these large, gray dots and the modeled means.</p>
<div class="figure"><span id="fig:lmm-demo1-explainer-2c"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-demo1-explainer-2c-1.png" alt="What random intercepts and slopes are. (A) A random intercept for batch $j$ is the difference between the fixed intercept and the modeled mean for batch $j$ in the reference treatment level. The random intercepts for experiments 1 ($\mathrm{g}_{0.1}$) and 2 ($\mathrm{g}_{0.2}$) are shown with brackets. (B) The fixed slope for the &quot;Tr2&quot; is illustrated with a bracket. The large, grey dots are the expected values for each batch (experiment_id) in the &quot;Tr2&quot; treatment if the random slope effects are zero. A random slope for batch $j$ is the difference between the expected value for batch j and the modeled mean for batch $j$. The random slope for experiment 2 in &quot;Tr2&quot; ($\mathrm{g}_{2.2}$) is shown with a bracket." width="1066.66666666667" />
<p class="caption">
Figure 18.4: What random intercepts and slopes are. (A) A random intercept for batch <span class="math inline">\(j\)</span> is the difference between the fixed intercept and the modeled mean for batch <span class="math inline">\(j\)</span> in the reference treatment level. The random intercepts for experiments 1 (<span class="math inline">\(\mathrm{g}_{0.1}\)</span>) and 2 (<span class="math inline">\(\mathrm{g}_{0.2}\)</span>) are shown with brackets. (B) The fixed slope for the “Tr2” is illustrated with a bracket. The large, grey dots are the expected values for each batch (experiment_id) in the “Tr2” treatment if the random slope effects are zero. A random slope for batch <span class="math inline">\(j\)</span> is the difference between the expected value for batch j and the modeled mean for batch <span class="math inline">\(j\)</span>. The random slope for experiment 2 in “Tr2” (<span class="math inline">\(\mathrm{g}_{2.2}\)</span>) is shown with a bracket.
</p>
</div>
</div>
<div id="in-a-blocked-design-a-linear-model-with-added-random-effects-increases-precision-of-treatment-effects" class="section level3" number="18.1.6">
<h3><span class="header-section-number">18.1.6</span> In a blocked design, a linear model with added random effects increases precision of treatment effects</h3>
<p><span class="math display" id="eq:lmm-demo1-m0">\[
\begin{equation}
\texttt{gc}_{jk} = \beta_{0} + \beta_{k} \texttt{treatment}_{k} +  (\gamma_{0j} + \gamma_{kj}\texttt{treatment}_{k} + \varepsilon) 
\tag{18.2}
\end{equation}
\]</span></p>
<p>If the random intercepts and random slopes aren’t modeled, this among-experiment variance is shifted to the error variance because the intercept effects and slope effects aren’t estimated but absorbed by the error – everything in Model <a href="lmm.html#eq:lmm-demo1-m0">(18.2)</a> will be estimated by the residuals. As a consequence, the estimate of <span class="math inline">\(\sigma\)</span> (the square root of the error variance) for the linear mixed model is smaller than that for the linear model with only fixed effects.</p>
<div class="sourceCode" id="cb526"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb526-1"><a href="lmm.html#cb526-1" aria-hidden="true" tabindex="-1"></a><span class="co"># sigma for the lmm</span></span>
<span id="cb526-2"><a href="lmm.html#cb526-2" aria-hidden="true" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> treatment <span class="sc">+</span></span>
<span id="cb526-3"><a href="lmm.html#cb526-3" aria-hidden="true" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb526-4"><a href="lmm.html#cb526-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span>
<span id="cb526-5"><a href="lmm.html#cb526-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m1)<span class="sc">$</span>sigma</span></code></pre></div>
<pre><code>## [1] 1.931339</code></pre>
<div class="sourceCode" id="cb528"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb528-1"><a href="lmm.html#cb528-1" aria-hidden="true" tabindex="-1"></a><span class="co"># sigma for the fixed lm</span></span>
<span id="cb528-2"><a href="lmm.html#cb528-2" aria-hidden="true" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(gc <span class="sc">~</span> treatment,</span>
<span id="cb528-3"><a href="lmm.html#cb528-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span>
<span id="cb528-4"><a href="lmm.html#cb528-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m2)<span class="sc">$</span>sigma</span></code></pre></div>
<pre><code>## [1] 3.376402</code></pre>
<p>The consequence of the smaller estimate of <span class="math inline">\(\sigma\)</span> in the linear mixed model on inference (confidence intervals and <em>p</em>-values) depends on the number of subsamples, the <strong>variance of the random effects</strong> relative to the variance of the residual error, and the <strong>correlation among the random effects</strong>.</p>
</div>
<div id="lmm-varcorr" class="section level3" number="18.1.7">
<h3><span class="header-section-number">18.1.7</span> The correlation among random intercepts and slopes</h3>
<p>Again, here is the linear mixed model fit to the experiment 1g data.</p>
<p><span class="math display">\[
\begin{equation}
\texttt{gc}_{jk} = (\beta_{0} + \gamma_{0j}) + (\beta_{k} + \gamma_{kj}) \texttt{treatment}_{k} + \varepsilon 
\end{equation}
\]</span></p>
<p>Think about how this model <strong>generates data</strong>. We have four experiments, so we randomly draw four <span class="math inline">\(\gamma_{0j}\)</span> from a normal distribution with some variance <span class="math inline">\(\sigma_{0}^2\)</span>. And, for each non-reference treatment, we randomly draw four <span class="math inline">\(\gamma_{0k}\)</span> from a normal distribution with some variance <span class="math inline">\(\sigma_{k}^2\)</span>. This gives us a matrix of four columns (one random intercept and three random slopes) and four rows (four experiments).</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
intercept
</th>
<th style="text-align:left;">
slope 1
</th>
<th style="text-align:left;">
slope 2
</th>
<th style="text-align:left;">
slope 3
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{0.1}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{1.1}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{2.1}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{3.1}\)</span>
</td>
</tr>
<tr>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{0.2}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{1.2}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{2.2}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{3.2}\)</span>
</td>
</tr>
<tr>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{0.3}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{1.3}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{2.3}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{3.3}\)</span>
</td>
</tr>
<tr>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{0.4}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{1.4}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{2.4}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{3.4}\)</span>
</td>
</tr>
</tbody>
</table>
<p>To randomly sample these values, the model needs not only the variances (<span class="math inline">\(\sigma_k^2\)</span>) for each column (random effect) <em>but also</em> a correlation for each pair of columns. These correlations are the off-diagonal elements of the correlation matrix of random effects.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<tbody>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{0}\)</span>, <span class="math inline">\(\gamma_{1}\)</span>)
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{0}\)</span>, <span class="math inline">\(\gamma_{2}\)</span>)
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{0}\)</span>, <span class="math inline">\(\gamma_{3}\)</span>)
</td>
</tr>
<tr>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{1}\)</span>, <span class="math inline">\(\gamma_{0}\)</span>)
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{1}\)</span>, <span class="math inline">\(\gamma_{2}\)</span>)
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{1}\)</span>, <span class="math inline">\(\gamma_{3}\)</span>)
</td>
</tr>
<tr>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{2}\)</span>, <span class="math inline">\(\gamma_{0}\)</span>)
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{2}\)</span>, <span class="math inline">\(\gamma_{1}\)</span>)
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{2}\)</span>, <span class="math inline">\(\gamma_{3}\)</span>)
</td>
</tr>
<tr>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{3}\)</span>, <span class="math inline">\(\gamma_{0}\)</span>)
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{3}\)</span>, <span class="math inline">\(\gamma_{1}\)</span>)
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{3}\)</span>, <span class="math inline">\(\gamma_{2}\)</span>)
</td>
<td style="text-align:left;">
1
</td>
</tr>
</tbody>
</table>
<p>In the models fit in this text, a researcher doesn’t specify these variances and correlations. Instead, these are parameters estimated by the model. Here is a summary of the estimates of the variances of the random effects and of the correlations among the random effects for the linear mixed model fit to the experiment 1g data.</p>
<pre><code>##  Groups        Name         Std.Dev. Corr                
##  experiment_id (Intercept)  1.7710                       
##                treatmentTr1 0.9642   -0.025              
##                treatmentTr2 2.9026    0.458  0.876       
##                treatmentTr3 2.9910    0.300  0.879  0.938
##  Residual                   1.9313</code></pre>
<p>The first four values in the column “Std.Dev.” are the square roots of the estimated variances for the random effects given in the column “Name.” The last value in column “Std.Dev.” is the square roots of the estimate of <span class="math inline">\(\sigma^2\)</span> (the error variance). The (lower) triangular matrix of values under “Corr” are the estimates of the correlations among the random effects. The variance of the random effects and the correlation among the random effects creates the <a href="lmm.html#lmm-corerr">correlated error described above</a> but do not confuse these different correlations (this is easy to confuse, you are not alone).</p>
<p>A compact way to view these variances and correlations is a matrix with the random effect standard deviations on the diagonal and the correlations on the off-diagonal. I’ll refer to this as the <em>VarCorr</em> matrix after the <code>lme4</code> function used to get the values.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-233">Table 18.2: </span>The Varcorr matrix. Standard deviations of random effects on the diagonal. Correlations of random effects on the off-diagonal.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
</th>
<th style="text-align:right;">
</th>
<th style="text-align:right;">
</th>
<th style="text-align:right;">
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
1.77
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentTr1
</td>
<td style="text-align:right;">
-0.03
</td>
<td style="text-align:right;">
0.96
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentTr2
</td>
<td style="text-align:right;">
0.46
</td>
<td style="text-align:right;">
0.88
</td>
<td style="text-align:right;">
2.90
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentTr3
</td>
<td style="text-align:right;">
0.30
</td>
<td style="text-align:right;">
0.88
</td>
<td style="text-align:right;">
0.94
</td>
<td style="text-align:right;">
2.99
</td>
</tr>
</tbody>
</table>
<p>It’s probably not worth trying to understand the experimental reason underneath the correlations among the random effects. But, researchers might want to sleuth out why a lab is getting high random intercept and slope variances, relative to the error variances, as these could indicate potential sources of improvement in lab protocols.</p>
</div>
<div id="clustered-residuals-create-heterogeneity-among-treatments" class="section level3" number="18.1.8">
<h3><span class="header-section-number">18.1.8</span> Clustered residuals create heterogeneity among treatments</h3>
<p>The variances of the four treatments are</p>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
Var
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Control
</td>
<td style="text-align:right;">
7.8
</td>
</tr>
<tr>
<td style="text-align:left;">
Tr1
</td>
<td style="text-align:right;">
5.2
</td>
</tr>
<tr>
<td style="text-align:left;">
Tr2
</td>
<td style="text-align:right;">
16.2
</td>
</tr>
<tr>
<td style="text-align:left;">
Tr3
</td>
<td style="text-align:right;">
16.7
</td>
</tr>
</tbody>
</table>
<p>In chapter xxx, I stated that heterogeneity of variances can arise because of clustered data. Why does clustered data generate heterogeneity? Let’s keep peeking at the linear mixed model fit to the experiment 1g data.</p>
<p><span class="math display">\[
\begin{equation}
\texttt{gc}_{jk} = (\beta_{0} + \gamma_{0j}) + (\beta_{k} + \gamma_{kj}) \texttt{treatment}_{k} + \varepsilon 
\end{equation}
\]</span></p>
<p>In a linear model with fixed effects only, the expected variance for any treatment for any treatment is <span class="math inline">\(\sigma^2\)</span>. But if the data are batched, the expected variances include components due to the batch and these batch components depend on the treatment. This creates heterogeneity.</p>
<p>To understand this, first some rules of expected variance. The random variable <span class="math inline">\(\texttt{C}\)</span> is the sum of two random variables <span class="math inline">\(\texttt{A}\)</span> and <span class="math inline">\(\texttt{B}\)</span>. The variances of these variables are <span class="math inline">\(\sigma_{C}^2\)</span>, <span class="math inline">\(\sigma_{A}^2\)</span>, and <span class="math inline">\(\sigma_{B}^2\)</span>.</p>
<ol style="list-style-type: decimal">
<li>The expected variance of <span class="math inline">\(\texttt{C}\)</span> if <span class="math inline">\(\texttt{A}\)</span> and <span class="math inline">\(\texttt{B}\)</span> are independent (uncorrelated) is <span class="math inline">\(\sigma_{C}^2 = \sigma_{A}^2 + \sigma_{B}^2\)</span> (<a href="https://en.wikipedia.org/wiki/Pythagorean_theorem" target="_blank">this equation should look familiar</a>).</li>
<li>The expected variance of <span class="math inline">\(\texttt{C}\)</span> if <span class="math inline">\(\texttt{A}\)</span> and <span class="math inline">\(\texttt{B}\)</span> are not independent (correlated) is <span class="math inline">\(\sigma_{C}^2 = \sigma_{A}^2 + \sigma_{B}^2 + 2\sigma_{A}\ \sigma_{B}\ \rho_{A,B}\)</span> where <span class="math inline">\(\rho_{A,B}\)</span> is the expected correlation between <span class="math inline">\(\texttt{A}\)</span> and <span class="math inline">\(\texttt{B}\)</span> (<a href="https://en.wikipedia.org/wiki/Law_of_cosines" target="_blank">this equation might also look familiar</a>).</li>
</ol>
<p>Here is some code to better know expected variances of the sum of two correlated random variables.</p>
<div class="sourceCode" id="cb531"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb531-1"><a href="lmm.html#cb531-1" aria-hidden="true" tabindex="-1"></a><span class="co"># copy, paste, and explore</span></span>
<span id="cb531-2"><a href="lmm.html#cb531-2" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">4</span></span>
<span id="cb531-3"><a href="lmm.html#cb531-3" aria-hidden="true" tabindex="-1"></a>rho <span class="ot">&lt;-</span> <span class="fl">0.6</span> <span class="co"># change this to any value between -1 and 1</span></span>
<span id="cb531-4"><a href="lmm.html#cb531-4" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">abs</span>(rho))</span>
<span id="cb531-5"><a href="lmm.html#cb531-5" aria-hidden="true" tabindex="-1"></a>z <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n)</span>
<span id="cb531-6"><a href="lmm.html#cb531-6" aria-hidden="true" tabindex="-1"></a>A <span class="ot">&lt;-</span> b<span class="sc">*</span>z <span class="sc">+</span> <span class="fu">sqrt</span>(<span class="dv">1</span><span class="sc">-</span>b<span class="sc">^</span><span class="dv">2</span>)<span class="sc">*</span><span class="fu">rnorm</span>(n)</span>
<span id="cb531-7"><a href="lmm.html#cb531-7" aria-hidden="true" tabindex="-1"></a>B <span class="ot">&lt;-</span> <span class="fu">sign</span>(rho)<span class="sc">*</span>b<span class="sc">*</span>z <span class="sc">+</span> <span class="fu">sqrt</span>(<span class="dv">1</span><span class="sc">-</span>b<span class="sc">^</span><span class="dv">2</span>)<span class="sc">*</span><span class="fu">rnorm</span>(n)</span>
<span id="cb531-8"><a href="lmm.html#cb531-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(A,B) <span class="co"># should be close to rho</span></span>
<span id="cb531-9"><a href="lmm.html#cb531-9" aria-hidden="true" tabindex="-1"></a>C <span class="ot">&lt;-</span> A <span class="sc">+</span> B</span>
<span id="cb531-10"><a href="lmm.html#cb531-10" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(A)<span class="sc">^</span><span class="dv">2</span> <span class="co"># should be close to 1</span></span>
<span id="cb531-11"><a href="lmm.html#cb531-11" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(B)<span class="sc">^</span><span class="dv">2</span> <span class="co"># should be close to 1</span></span>
<span id="cb531-12"><a href="lmm.html#cb531-12" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(C)<span class="sc">^</span><span class="dv">2</span> <span class="co"># should be close to 1^2 + 1^2 + 2*1*1*rho</span></span>
<span id="cb531-13"><a href="lmm.html#cb531-13" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(A)<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="fu">sd</span>(B)<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="dv">2</span><span class="sc">*</span><span class="fu">sd</span>(A)<span class="sc">*</span><span class="fu">sd</span>(B)<span class="sc">*</span><span class="fu">cor</span>(A,B) <span class="co"># should equal previous line</span></span></code></pre></div>
<p>Using these rules and the standard deviations of the random effects given above we can computed the expected variances of the treatment groups given the fit model.</p>
<ol style="list-style-type: decimal">
<li>For the variance of the reference (“Control”) group, we need to add to <span class="math inline">\(\sigma^2\)</span> the variance of the random intercept using rule #1 (the residuals are not correlated with random intercepts or slopes). The modeled variance is a less than the actual variance of the Control group.</li>
</ol>
<div class="sourceCode" id="cb532"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb532-1"><a href="lmm.html#cb532-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(m1)<span class="sc">$</span>sigma<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="fl">1.7710</span><span class="sc">^</span><span class="dv">2</span></span></code></pre></div>
<pre><code>## [1] 6.866512</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>For the variance of a non-reference group, we need to add to <span class="math inline">\(\sigma^2\)</span> the variance of the random intercept and the variance of the random slope for the treatment and the component due to the correlation between the random slope for the treatment and the random intercept. For Tr2, this is</li>
</ol>
<div class="sourceCode" id="cb534"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb534-1"><a href="lmm.html#cb534-1" aria-hidden="true" tabindex="-1"></a><span class="co"># error + intercept + slope + cor(intercept, slope)</span></span>
<span id="cb534-2"><a href="lmm.html#cb534-2" aria-hidden="true" tabindex="-1"></a>(<span class="fu">summary</span>(m1)<span class="sc">$</span>sigma<span class="sc">^</span><span class="dv">2</span>) <span class="sc">+</span> (<span class="fl">1.7710</span><span class="sc">^</span><span class="dv">2</span>) <span class="sc">+</span> (<span class="fl">2.9026</span><span class="sc">^</span><span class="dv">2</span>) <span class="sc">+</span> (<span class="dv">2</span> <span class="sc">*</span> <span class="fl">1.7710</span> <span class="sc">*</span> <span class="fl">2.9026</span> <span class="sc">*</span> <span class="fl">0.458</span>)</span></code></pre></div>
<pre><code>## [1] 20.0003</code></pre>
<p>which is a bit higher than the measured variance.</p>
</div>
<div id="linear-mixed-models-are-flexible" class="section level3" number="18.1.9">
<h3><span class="header-section-number">18.1.9</span> Linear mixed models are flexible</h3>
<p>One more look at the linear mixed model fit to the experiment 1g data.</p>
<p><span class="math display">\[
\begin{equation}
\texttt{gc}_{jk} = (\beta_{0} + \gamma_{0j}) + (\beta_{k} + \gamma_{kj}) \texttt{treatment}_{k} + \varepsilon 
\end{equation}
\]</span></p>
<p>The linear mixed model specifies both a random intercept and a random slope but a researcher might limit the model to the random intercept only, or less commonly, the random slope only. Or a researcher might replace the random slope with a second random intercept that captures variance in the batch by treatment combinations like a random slope. Or a researcher might model the structure (correlated error and heterogeneity of variances) in the residuals <em>in addition to</em> adding random factors to the model.</p>
</div>
<div id="a-random-intercept-only-model" class="section level3" number="18.1.10">
<h3><span class="header-section-number">18.1.10</span> A random intercept only model</h3>
<div class="sourceCode" id="cb536"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb536-1"><a href="lmm.html#cb536-1" aria-hidden="true" tabindex="-1"></a>demo1_m2 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> treatment <span class="sc">+</span></span>
<span id="cb536-2"><a href="lmm.html#cb536-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id),</span>
<span id="cb536-3"><a href="lmm.html#cb536-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Model <code>demo1_m2</code> specifies only a random intercept for each of the levels of <span class="math inline">\(\texttt{experiment_id}\)</span>. The exclusion of the random slopes ia a kind of <strong>model simplification</strong>.</li>
<li>In experiments without subsampling, random slopes cannot be added to the model because there is no variation with a treatment by batch combination.</li>
<li>In experiments with subsampling, a researcher might exclude a random slope term for several reasons, including</li>
</ol>
<ul>
<li>it is the culture in many subfields to only include a random intercept (no, this is not a good reason)</li>
<li>the computation of the model fit returned a convergence warning</li>
<li>model comparison suggested that a model with the random slope was too complex given the data. A useful statistic for comparing models with different random effects specifications is the <strong>AIC</strong>, which is introduced in Section <a href="lmm.html#lmm-demo-aic">18.1.12</a> below.</li>
</ul>
</div>
<div id="a-model-including-an-interaction-intercept" class="section level3" number="18.1.11">
<h3><span class="header-section-number">18.1.11</span> A model including an interaction intercept</h3>
<div class="sourceCode" id="cb537"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb537-1"><a href="lmm.html#cb537-1" aria-hidden="true" tabindex="-1"></a>demo1_m3 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> treatment <span class="sc">+</span></span>
<span id="cb537-2"><a href="lmm.html#cb537-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb537-3"><a href="lmm.html#cb537-3" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>treatment),</span>
<span id="cb537-4"><a href="lmm.html#cb537-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li><code>(1 | experiment_id:treatment)</code> models a random intercept for all combinations of the levels of <span class="math inline">\(\texttt{experiment_id}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>. This <strong>interaction intercept</strong> is an alternative to a random slope for modeling treatment-specific batch effects.</li>
</ol>
</div>
<div id="lmm-demo-aic" class="section level3" number="18.1.12">
<h3><span class="header-section-number">18.1.12</span> AIC and model selection – which model to report?</h3>
<p>Three different linear mixed models were fit to the Example 1 data: <code>demo1_m1</code>, <code>demo1_m2</code>, and <code>demo1_m3</code>. Which model do we report? A useful stastistic for this decision is a statistic known as the AIC.</p>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-demo-acc">Table 18.3: </span>AIC of the two alternative models fit to Example 1 data
</caption>
<thead>
<tr>
<th style="text-align:left;">
Model
</th>
<th style="text-align:right;">
AIC
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
demo1_m1
</td>
<td style="text-align:right;">
344.1529
</td>
</tr>
<tr>
<td style="text-align:left;">
demo1_m2
</td>
<td style="text-align:right;">
344.0765
</td>
</tr>
<tr>
<td style="text-align:left;">
demo1_m3
</td>
<td style="text-align:right;">
337.5332
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>AIC (<a href="https://en.wikipedia.org/wiki/Akaike_information_criterion" target="_blank">Akaike Information Criterion</a>) is a relative measure of model quality. Compare this to <span class="math inline">\(R^2\)</span>, which is an absolute measure of goodness of fit. The AIC formula has two parts, one is a kind of goodness of fit (like <span class="math inline">\(R^2\)</span>) and the other is a penalty based on the number of parameters in the model. As the goodness of fit increases, the AIC goes down. As the number of parameters increases, the AIC goes up. The model with the lowest AIC is the highest quality model. The actual value of AIC, unlike <span class="math inline">\(R^2\)</span>, does not have any absolute meaning; it is only meaningful relative to the AICs computed from fits to different models to the same data.</li>
<li>The AIC of the three models suggests that we report Model <code>demo1_m3</code>.</li>
</ol>
</div>
<div id="the-specification-of-random-effects-matters" class="section level3" number="18.1.13">
<h3><span class="header-section-number">18.1.13</span> The specification of random effects matters</h3>
<p>Inference will often be very different between these two models as they make very different</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-239">Table 18.4: </span>Contrasts from the three linear mixed models fit to the Example 1 data.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>intercepts + slopes (m1)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr1 - Control
</td>
<td style="text-align:right;">
2.54
</td>
<td style="text-align:right;">
0.804
</td>
<td style="text-align:right;">
2.978
</td>
<td style="text-align:right;">
-0.03
</td>
<td style="text-align:right;">
5.11
</td>
<td style="text-align:right;">
3.16
</td>
<td style="text-align:right;">
0.05150
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr2 - Control
</td>
<td style="text-align:right;">
5.61
</td>
<td style="text-align:right;">
1.591
</td>
<td style="text-align:right;">
2.997
</td>
<td style="text-align:right;">
0.54
</td>
<td style="text-align:right;">
10.67
</td>
<td style="text-align:right;">
3.52
</td>
<td style="text-align:right;">
0.03885
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr3 - Control
</td>
<td style="text-align:right;">
2.85
</td>
<td style="text-align:right;">
1.631
</td>
<td style="text-align:right;">
2.997
</td>
<td style="text-align:right;">
-2.35
</td>
<td style="text-align:right;">
8.04
</td>
<td style="text-align:right;">
1.74
</td>
<td style="text-align:right;">
0.17949
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>intercepts only (m2)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr1 - Control
</td>
<td style="text-align:right;">
2.64
</td>
<td style="text-align:right;">
0.755
</td>
<td style="text-align:right;">
66.005
</td>
<td style="text-align:right;">
1.13
</td>
<td style="text-align:right;">
4.15
</td>
<td style="text-align:right;">
3.49
</td>
<td style="text-align:right;">
0.00086
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr2 - Control
</td>
<td style="text-align:right;">
5.65
</td>
<td style="text-align:right;">
0.766
</td>
<td style="text-align:right;">
66.014
</td>
<td style="text-align:right;">
4.12
</td>
<td style="text-align:right;">
7.18
</td>
<td style="text-align:right;">
7.38
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr3 - Control
</td>
<td style="text-align:right;">
2.94
</td>
<td style="text-align:right;">
0.765
</td>
<td style="text-align:right;">
66.000
</td>
<td style="text-align:right;">
1.41
</td>
<td style="text-align:right;">
4.47
</td>
<td style="text-align:right;">
3.85
</td>
<td style="text-align:right;">
0.00027
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>intercept + interaction intercept (m3)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr1 - Control
</td>
<td style="text-align:right;">
2.54
</td>
<td style="text-align:right;">
1.214
</td>
<td style="text-align:right;">
8.912
</td>
<td style="text-align:right;">
-0.21
</td>
<td style="text-align:right;">
5.29
</td>
<td style="text-align:right;">
2.09
</td>
<td style="text-align:right;">
0.06616
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr2 - Control
</td>
<td style="text-align:right;">
5.60
</td>
<td style="text-align:right;">
1.219
</td>
<td style="text-align:right;">
9.061
</td>
<td style="text-align:right;">
2.84
</td>
<td style="text-align:right;">
8.35
</td>
<td style="text-align:right;">
4.59
</td>
<td style="text-align:right;">
0.00128
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr3 - Control
</td>
<td style="text-align:right;">
2.87
</td>
<td style="text-align:right;">
1.219
</td>
<td style="text-align:right;">
9.052
</td>
<td style="text-align:right;">
0.12
</td>
<td style="text-align:right;">
5.63
</td>
<td style="text-align:right;">
2.36
</td>
<td style="text-align:right;">
0.04269
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Inference from the intercept only model (<code>demo1_m2</code>) is almost certainly too optimistic based on simulations that show that intercept only models can be highly anti-conservative (too narrow confidence intervals and too small <em>p</em>-values).</li>
<li>Humans have evolved to make up rational explanations – do not convince yourself that the model with the smallest <em>p</em>-values is the scientifically most rational model.</li>
</ol>
</div>
<div id="mixed-effect-and-repeated-measures-anova" class="section level3" number="18.1.14">
<h3><span class="header-section-number">18.1.14</span> Mixed Effect and Repeated Measures ANOVA</h3>
<p>Two-way mixed-effect ANOVA (some fields would call this a repeated measures ANOVA) is equivalent to Model <code>demo1_m3</code> <em>in this case</em>. More generally, the two are equal in balanced designs – the same number of subsamples in all treatment x batch combinations.</p>
<div class="sourceCode" id="cb538"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb538-1"><a href="lmm.html#cb538-1" aria-hidden="true" tabindex="-1"></a>demo1_m4 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(gc <span class="sc">~</span> treatment <span class="sc">+</span> </span>
<span id="cb538-2"><a href="lmm.html#cb538-2" aria-hidden="true" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb538-3"><a href="lmm.html#cb538-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span></code></pre></div>
<p>Notes
1. The function <code>afex::aov4</code> is used for specifying ANOVA models that are special cases of linear mixed models.
2. The model formula in Model <code>demo1_m2</code> looks exactly like that the random intercepts and slopes model (Model <code>demo1_m1</code>) <em>but these are not the same</em>.
3. One can use either a univariate or multivariate model for mixed or repeated measures ANOVA – see section <a href="lmm.html#lmm-exp6g-rmanova-multi">18.2.11</a>.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-241">Table 18.5: </span>Contrasts from the two-way mixed effect ANOVA model demo1_m4.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Tr1 - Control
</td>
<td style="text-align:right;">
2.50
</td>
<td style="text-align:right;">
1.193
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
-0.19
</td>
<td style="text-align:right;">
5.20
</td>
<td style="text-align:right;">
2.10
</td>
<td style="text-align:right;">
0.06521
</td>
</tr>
<tr>
<td style="text-align:left;">
Tr2 - Control
</td>
<td style="text-align:right;">
5.58
</td>
<td style="text-align:right;">
1.193
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
8.28
</td>
<td style="text-align:right;">
4.67
</td>
<td style="text-align:right;">
0.00116
</td>
</tr>
<tr>
<td style="text-align:left;">
Tr3 - Control
</td>
<td style="text-align:right;">
2.84
</td>
<td style="text-align:right;">
1.193
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
0.14
</td>
<td style="text-align:right;">
5.54
</td>
<td style="text-align:right;">
2.38
</td>
<td style="text-align:right;">
0.04103
</td>
</tr>
</tbody>
</table>
</div>
<div id="pseudoreplication" class="section level3" number="18.1.15">
<h3><span class="header-section-number">18.1.15</span> Pseudoreplication</h3>
</div>
</div>
<div id="lmm-example2" class="section level2" number="18.2">
<h2><span class="header-section-number">18.2</span> Example 2 – experiments without subsampling replication (exp6g)</h2>
<p>This example introduces linear mixed models for batches that contain all treatment levels of a single factor but no subsampling replication. In this example, the batch is the individual mouse (<span class="math inline">\(\texttt{mouse_id}\)</span>). There are four measures of the response variable on each mouse, one measure per treatment level. When there is no subsampling replication, we cannot add a random slope to the model because there is only a single observation at each treatment level and a slope would fit the point at the reference level and the point at the non-reference level perfectly. However, we can explicitly model variation in the correlated error and heterogeneity in the variances among treatments as an alternative to modeling a random slope.</p>
<p><a href="https://www.nature.com/articles/s41586-020-2388-4" target="_blank">Reversing a model of Parkinson’s disease with in situ converted nigral neurons</a></p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/7521455/" target="_blank">Public source</a></p>
<p>Source figure: <a href="https://www.nature.com/articles/s41586-020-2388-4#Fig6" target="_blank">Fig. 6g</a></p>
<p>Source data: <a href="https://www.nature.com/articles/s41586-020-2388-4#MOESM11" target="_blank">Source Data Fig. 6</a></p>
<div id="understand-the-data" class="section level3" number="18.2.1">
<h3><span class="header-section-number">18.2.1</span> Understand the data</h3>
<p>In this study, the researchers investigate the effectiveness of knocking down the protein <a href="https://en.wikipedia.org/wiki/PTBP1" target="_blank">PTBP1</a> to induce astrocytes to convert to neurons in a motor processing region of the brain. Experimental lesions of this region of the brain is a model of <a href="https://en.wikipedia.org/wiki/Parkinson%27s_disease" target="_blank">Parkinson’s disease</a>. In Experiment 6g, the researchers</p>
<ol style="list-style-type: decimal">
<li>Generated a lesion in the motor processing region using 6-hydroxydopamine (6-OHDA). The lesion disrupts the ability to control the <strong>contralateral</strong> (opposite side) forelimb.</li>
<li>One month after the lesion, measured the percent of ipsilateral (same side) forepaw touches (the forelimb extending out and touching the surface) in a test of exploration in a new environment (the “cylinder test”). The expected percent in an intact mouse is 50%. In a lesioned mouse, the percent should be much greater than 50% since there is less control of the contralateral limb. The measure at this point is in the treatment “Lesion.” This is the positive control.</li>
<li>Converted astrocytes in the lesion to functional neurons by knocking down PTBP1.</li>
<li>Two months after knockdown, gave the mouse saline and remeasured percent ipsilateral touches in a cylinder test. If the knockdown worked as expected, there should be closer to 50% ipsilateral touches. The measure at this point is in the treatment “Saline.” The comparison with Lesion is a focal test.</li>
<li>Inhibited neuron action in the converted neurons using clozapine-N-oxide (CNO), which suppresses neuron electrical activity. Then, remeasured percent ipsilateral touches in a cylinder test. If the CNO worked as expected, there should be much greater than 50% ipsilateral touches since there should be re-loss of control of the contralateral limb. The measure at this point is in the treatment “CNO.” The comparison with Saline is a focal test.</li>
<li>Allowed three days for the CNO to degrade, then, remeasured percent ipsilateral touches in a cylinder test. If the CNO degraded as expected, the converted neurons should be functional and there should be closer to 50% ipsilateral touches. The measure at this point is in the treatment “Post_CNO.” The comparison with CNO is a focal test.</li>
</ol>
<p>The design is <span class="math inline">\(4 \times 1\)</span> – a single treatment with four levels (“Lesion,” “Saline,” “CNO,” “Post_CNO”)</p>
<p>The planned contrasts are</p>
<ol style="list-style-type: decimal">
<li>Saline - Lesion. This measures the effect of the knockdown and conversion of astrocytes to functional neurons.</li>
<li>CNO - Saline. This measures the effect of inhibiting the converted neurons to test if it was these and not some other neurons that account for the effect in contrast 1.</li>
<li>Post_CNO - CNO. This is probing the same expectation as contrast 2.</li>
</ol>
</div>
<div id="model-fit-and-inference" class="section level3" number="18.2.2">
<h3><span class="header-section-number">18.2.2</span> Model fit and inference</h3>
<div id="fit-the-model-8" class="section level4" number="18.2.2.1">
<h4><span class="header-section-number">18.2.2.1</span> Fit the model</h4>
<div class="sourceCode" id="cb539"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb539-1"><a href="lmm.html#cb539-1" aria-hidden="true" tabindex="-1"></a>exp6g_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(touch <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>mouse_id), <span class="at">data =</span> exp6g)</span>
<span id="cb539-2"><a href="lmm.html#cb539-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb539-3"><a href="lmm.html#cb539-3" aria-hidden="true" tabindex="-1"></a><span class="co"># alt model</span></span>
<span id="cb539-4"><a href="lmm.html#cb539-4" aria-hidden="true" tabindex="-1"></a>exp6g_m1b <span class="ot">&lt;-</span> <span class="fu">lme</span>(touch <span class="sc">~</span> treatment,</span>
<span id="cb539-5"><a href="lmm.html#cb539-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">random =</span> <span class="sc">~</span><span class="dv">1</span><span class="sc">|</span>mouse_id,</span>
<span id="cb539-6"><a href="lmm.html#cb539-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> mouse_id),</span>
<span id="cb539-7"><a href="lmm.html#cb539-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> treatment),</span>
<span id="cb539-8"><a href="lmm.html#cb539-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp6g)</span>
<span id="cb539-9"><a href="lmm.html#cb539-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb539-10"><a href="lmm.html#cb539-10" aria-hidden="true" tabindex="-1"></a><span class="fu">AIC</span>(exp6g_m1a, exp6g_m1b)</span></code></pre></div>
<pre><code>##           df      AIC
## exp6g_m1a  6 191.3367
## exp6g_m1b 15 198.3888</code></pre>
<div class="sourceCode" id="cb541"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb541-1"><a href="lmm.html#cb541-1" aria-hidden="true" tabindex="-1"></a><span class="co"># report model a</span></span>
<span id="cb541-2"><a href="lmm.html#cb541-2" aria-hidden="true" tabindex="-1"></a>exp6g_m1 <span class="ot">&lt;-</span> exp6g_m1a</span></code></pre></div>
<p>exp6g_m1b overparameterizes, report exp6g_m1a (see <a href="lmm.html#lmm-exp6g-alt">Alternative models for exp6g</a> below)</p>
</div>
<div id="inference-from-the-model-4" class="section level4" number="18.2.2.2">
<h4><span class="header-section-number">18.2.2.2</span> Inference from the model</h4>
<div class="sourceCode" id="cb542"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb542-1"><a href="lmm.html#cb542-1" aria-hidden="true" tabindex="-1"></a>exp6g_m1_coef <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">coef</span>(<span class="fu">summary</span>(exp6g_m1)),</span>
<span id="cb542-2"><a href="lmm.html#cb542-2" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">confint</span>(exp6g_m1)[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>),])</span>
<span id="cb542-3"><a href="lmm.html#cb542-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb542-4"><a href="lmm.html#cb542-4" aria-hidden="true" tabindex="-1"></a><span class="co"># exp5c_m1_coef %&gt;%</span></span>
<span id="cb542-5"><a href="lmm.html#cb542-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   kable(digits = c(2,3,1,1,4,2,2)) %&gt;%</span></span>
<span id="cb542-6"><a href="lmm.html#cb542-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   kable_styling()</span></span></code></pre></div>
<div class="sourceCode" id="cb543"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb543-1"><a href="lmm.html#cb543-1" aria-hidden="true" tabindex="-1"></a>exp6g_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp6g_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>))</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
emmean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Lesion
</td>
<td style="text-align:right;">
83.3
</td>
<td style="text-align:right;">
3.44
</td>
<td style="text-align:right;">
18.9
</td>
<td style="text-align:right;">
76.1
</td>
<td style="text-align:right;">
90.47
</td>
</tr>
<tr>
<td style="text-align:left;">
Saline
</td>
<td style="text-align:right;">
62.8
</td>
<td style="text-align:right;">
3.44
</td>
<td style="text-align:right;">
18.9
</td>
<td style="text-align:right;">
55.5
</td>
<td style="text-align:right;">
69.96
</td>
</tr>
<tr>
<td style="text-align:left;">
CNO
</td>
<td style="text-align:right;">
81.4
</td>
<td style="text-align:right;">
3.44
</td>
<td style="text-align:right;">
18.9
</td>
<td style="text-align:right;">
74.2
</td>
<td style="text-align:right;">
88.63
</td>
</tr>
<tr>
<td style="text-align:left;">
Post_CNO
</td>
<td style="text-align:right;">
58.8
</td>
<td style="text-align:right;">
3.44
</td>
<td style="text-align:right;">
18.9
</td>
<td style="text-align:right;">
51.6
</td>
<td style="text-align:right;">
66.00
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb544"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb544-1"><a href="lmm.html#cb544-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exp6g_m1_emm # print in console to get row numbers</span></span>
<span id="cb544-2"><a href="lmm.html#cb544-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb544-3"><a href="lmm.html#cb544-3" aria-hidden="true" tabindex="-1"></a>lesion <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb544-4"><a href="lmm.html#cb544-4" aria-hidden="true" tabindex="-1"></a>saline <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb544-5"><a href="lmm.html#cb544-5" aria-hidden="true" tabindex="-1"></a>cno <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb544-6"><a href="lmm.html#cb544-6" aria-hidden="true" tabindex="-1"></a>post_cno <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb544-7"><a href="lmm.html#cb544-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb544-8"><a href="lmm.html#cb544-8" aria-hidden="true" tabindex="-1"></a>exp6g_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp6g_m1_emm,</span>
<span id="cb544-9"><a href="lmm.html#cb544-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb544-10"><a href="lmm.html#cb544-10" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;Saline - Lesion&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(saline <span class="sc">-</span> lesion),</span>
<span id="cb544-11"><a href="lmm.html#cb544-11" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;CNO - Saline&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(cno <span class="sc">-</span> saline),</span>
<span id="cb544-12"><a href="lmm.html#cb544-12" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;Post_CNO - CNO&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(post_cno <span class="sc">-</span> cno)</span>
<span id="cb544-13"><a href="lmm.html#cb544-13" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb544-14"><a href="lmm.html#cb544-14" aria-hidden="true" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb544-15"><a href="lmm.html#cb544-15" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb544-16"><a href="lmm.html#cb544-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Saline - Lesion
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-29.07
</td>
<td style="text-align:right;">
-11.96
</td>
<td style="text-align:right;">
-5.04
</td>
<td style="text-align:right;">
0.00009
</td>
</tr>
<tr>
<td style="text-align:left;">
CNO - Saline
</td>
<td style="text-align:right;">
18.67
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
10.12
</td>
<td style="text-align:right;">
27.23
</td>
<td style="text-align:right;">
4.59
</td>
<td style="text-align:right;">
0.00023
</td>
</tr>
<tr>
<td style="text-align:left;">
Post_CNO - CNO
</td>
<td style="text-align:right;">
-22.63
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-31.18
</td>
<td style="text-align:right;">
-14.08
</td>
<td style="text-align:right;">
-5.56
</td>
<td style="text-align:right;">
0.00003
</td>
</tr>
</tbody>
</table>
</div>
<div id="lmm-exp5c-plotthemodel" class="section level4" number="18.2.2.3">
<h4><span class="header-section-number">18.2.2.3</span> Plot the model</h4>
<div class="sourceCode" id="cb545"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb545-1"><a href="lmm.html#cb545-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot_the_model</span>(exp6g_m1,</span>
<span id="cb545-2"><a href="lmm.html#cb545-2" aria-hidden="true" tabindex="-1"></a>                 exp6g_m1_emm,</span>
<span id="cb545-3"><a href="lmm.html#cb545-3" aria-hidden="true" tabindex="-1"></a>                 exp6g_m1_planned,</span>
<span id="cb545-4"><a href="lmm.html#cb545-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y_label =</span> <span class="st">&quot;Percent ipsilateral touch&quot;</span>,</span>
<span id="cb545-5"><a href="lmm.html#cb545-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">effect_label =</span> <span class="st">&quot;Difference in % ipsilateral touch&quot;</span>,</span>
<span id="cb545-6"><a href="lmm.html#cb545-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">palette =</span> pal_okabe_ito_blue,</span>
<span id="cb545-7"><a href="lmm.html#cb545-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="fl">0.5</span>,<span class="dv">1</span>))</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-6g-m1-plot-1.png" width="576" /></p>
</div>
<div id="alternaplot-the-model" class="section level4" number="18.2.2.4">
<h4><span class="header-section-number">18.2.2.4</span> Alternaplot the model</h4>
<div class="sourceCode" id="cb546"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb546-1"><a href="lmm.html#cb546-1" aria-hidden="true" tabindex="-1"></a>gg <span class="ot">&lt;-</span> <span class="fu">ggplot_the_response</span>(exp6g_m1,</span>
<span id="cb546-2"><a href="lmm.html#cb546-2" aria-hidden="true" tabindex="-1"></a>                 exp6g_m1_emm,</span>
<span id="cb546-3"><a href="lmm.html#cb546-3" aria-hidden="true" tabindex="-1"></a>                 exp6g_m1_planned,</span>
<span id="cb546-4"><a href="lmm.html#cb546-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">dots =</span> <span class="st">&quot;jitter&quot;</span>,</span>
<span id="cb546-5"><a href="lmm.html#cb546-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y_label =</span> <span class="st">&quot;Percent ipsilateral touch&quot;</span>,</span>
<span id="cb546-6"><a href="lmm.html#cb546-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">palette =</span> pal_okabe_ito_blue) <span class="sc">+</span></span>
<span id="cb546-7"><a href="lmm.html#cb546-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb546-8"><a href="lmm.html#cb546-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_line(aes(group = mouse_id),</span></span>
<span id="cb546-9"><a href="lmm.html#cb546-9" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           position = position_jitter(seed = 1),</span></span>
<span id="cb546-10"><a href="lmm.html#cb546-10" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           color = &quot;gray&quot;) +</span></span>
<span id="cb546-11"><a href="lmm.html#cb546-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb546-12"><a href="lmm.html#cb546-12" aria-hidden="true" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb546-13"><a href="lmm.html#cb546-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb546-14"><a href="lmm.html#cb546-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb546-15"><a href="lmm.html#cb546-15" aria-hidden="true" tabindex="-1"></a>gg</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-6g-m1-alternaplot-1.png" width="576" /></p>
</div>
</div>
<div id="the-model-exp6g_m1-adds-a-random-intercept-but-not-a-random-slope" class="section level3" number="18.2.3">
<h3><span class="header-section-number">18.2.3</span> The model exp6g_m1 adds a random intercept but not a random slope</h3>
<p>The model fit to the exp6g data is</p>
<p><span class="math display">\[
\begin{equation}
\texttt{touch}_{jk} = (\beta_{0} + \gamma_{0j}) + (\beta_{k} + \gamma_{jk}) \texttt{treatment}_{k} + \varepsilon 
\end{equation}
\]</span></p>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Again, in experiments without subsampling, we cannot add a random slope to the model (for each mouse, there is a single observation at each treatment level so a slope would fit the two points perfectly).</li>
</ol>
</div>
<div id="the-fixed-effect-coefficients-of-model-exp6g_m1" class="section level3" number="18.2.4">
<h3><span class="header-section-number">18.2.4</span> The fixed effect coefficients of model exp6g_m1</h3>
<p>The fixed effect coefficients of model <code>exp6g_m1</code> are</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
83.27
</td>
<td style="text-align:right;">
3.441
</td>
<td style="text-align:right;">
18.9
</td>
<td style="text-align:right;">
24.20
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
76.72
</td>
<td style="text-align:right;">
89.81
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentSaline
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18.0
</td>
<td style="text-align:right;">
-5.04
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
-28.25
</td>
<td style="text-align:right;">
-12.78
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentCNO
</td>
<td style="text-align:right;">
-1.84
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18.0
</td>
<td style="text-align:right;">
-0.45
</td>
<td style="text-align:right;">
0.6565
</td>
<td style="text-align:right;">
-9.58
</td>
<td style="text-align:right;">
5.90
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentPost_CNO
</td>
<td style="text-align:right;">
-24.47
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18.0
</td>
<td style="text-align:right;">
-6.01
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
-32.21
</td>
<td style="text-align:right;">
-16.73
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The interpretation of the fixed effectcs coefficients have the usual interpretation (see <a href="oneway.html#oneway-what-coefs-are">The coefficients of a linear model using dummy coding have a useful interpretation</a>). Figure <a href="lmm.html#fig:lmm-exp6g-explainer-1">18.5</a> is a reminder.</li>
</ol>
<div class="figure"><span id="fig:lmm-exp6g-explainer-1"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp6g-explainer-1-1.png" alt="Fixed effects estimated by exp6g_m1. $b_0$ is the modeled mean of the Lesian treatment. $b_1$ is the difference (Saline - Lesian). $b_2$ is the difference (CNO - Lesian). $b_3$ is the difference (Post_CNO - Lesian)." width="672" />
<p class="caption">
Figure 18.5: Fixed effects estimated by exp6g_m1. <span class="math inline">\(b_0\)</span> is the modeled mean of the Lesian treatment. <span class="math inline">\(b_1\)</span> is the difference (Saline - Lesian). <span class="math inline">\(b_2\)</span> is the difference (CNO - Lesian). <span class="math inline">\(b_3\)</span> is the difference (Post_CNO - Lesian).
</p>
</div>
</div>
<div id="the-random-intercept-coefficients-of-exp6g_m1" class="section level3" number="18.2.5">
<h3><span class="header-section-number">18.2.5</span> The random intercept coefficients of exp6g_m1</h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp6g-m1-random-coef">Table 18.6: </span>Random intercept coefficients (g_0j) for exp6g_m1. For each mouse, the random intercept is the sum of the fixed intercept (<span class="math inline">\(b_0\)</span>) and the random intercept effect <span class="math inline">\(g_{0j}\)</span>.
</caption>
<thead>
<tr>
<th style="text-align:left;">
mouse_id
</th>
<th style="text-align:right;">
random intercept
</th>
<th style="text-align:right;">
b_0
</th>
<th style="text-align:right;">
g_0j
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
mouse_1
</td>
<td style="text-align:right;">
82.874
</td>
<td style="text-align:right;">
83.268
</td>
<td style="text-align:right;">
-0.393
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_2
</td>
<td style="text-align:right;">
83.640
</td>
<td style="text-align:right;">
83.268
</td>
<td style="text-align:right;">
0.372
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_3
</td>
<td style="text-align:right;">
77.499
</td>
<td style="text-align:right;">
83.268
</td>
<td style="text-align:right;">
-5.769
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_4
</td>
<td style="text-align:right;">
79.943
</td>
<td style="text-align:right;">
83.268
</td>
<td style="text-align:right;">
-3.325
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_5
</td>
<td style="text-align:right;">
89.286
</td>
<td style="text-align:right;">
83.268
</td>
<td style="text-align:right;">
6.018
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_6
</td>
<td style="text-align:right;">
82.732
</td>
<td style="text-align:right;">
83.268
</td>
<td style="text-align:right;">
-0.536
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_7
</td>
<td style="text-align:right;">
86.901
</td>
<td style="text-align:right;">
83.268
</td>
<td style="text-align:right;">
3.633
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The random intercept effect <span class="math inline">\(g_{0j}\)</span> is the difference between the modeled mean for mouse <em>j</em> and the mean of the reference treatment. Unlike the fixed intercept (<span class="math inline">\(b_0\)</span>), <span class="math inline">\(g_{0j}\)</span> is an effect.</li>
<li>Figure <a href="#fig:lmm-exp1g-explainer-2"><strong>??</strong></a> illustrates the random intercept coefficients <span class="math inline">\(g_{0j}\)</span> for model <code>exp6g_m1</code>.</li>
<li>The random intercept coefficients are often (but not always) treated as a source of nuisance variation – that is, the coefficients are not generally of interest and the values are not typically reported.</li>
</ol>
<div class="figure"><span id="fig:lmm-exp6g-explainer-2"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp6g-explainer-2-1.png" alt="Random intercept effects for model exp6g_m1. The pale, colored dots are the measured percent ipsilateral touch values for each mouse for each treatment. The dashed, gray lines are the modeled means for each treatment. The dashed grey line for the reference level (&quot;Lesion&quot;) is the fixed intercept. The dark, colored dots at the reference level are the random intercepts. The value of each random intercept is the sum of the fixed intercept and the random intercept effect for that mouse. The vertical, colored line segments at the reference level are the random intercept effects $g_{0j}$. The length of the segment is the residual from the dashed, gray line to the pale dot. The random intercept effects for mice 1 and 2 are too short to see." width="672" />
<p class="caption">
Figure 18.6: Random intercept effects for model exp6g_m1. The pale, colored dots are the measured percent ipsilateral touch values for each mouse for each treatment. The dashed, gray lines are the modeled means for each treatment. The dashed grey line for the reference level (“Lesion”) is the fixed intercept. The dark, colored dots at the reference level are the random intercepts. The value of each random intercept is the sum of the fixed intercept and the random intercept effect for that mouse. The vertical, colored line segments at the reference level are the random intercept effects <span class="math inline">\(g_{0j}\)</span>. The length of the segment is the residual from the dashed, gray line to the pale dot. The random intercept effects for mice 1 and 2 are too short to see.
</p>
</div>
</div>
<div id="the-random-and-residual-variance-and-the-intraclass-correlation-of-model-exp6g_m1" class="section level3" number="18.2.6">
<h3><span class="header-section-number">18.2.6</span> The random and residual variance and the intraclass correlation of model exp6g_m1</h3>
<pre><code>##  Groups   Name        Std.Dev.
##  mouse_id (Intercept) 4.9882  
##  Residual             7.6155</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The first element in “Std.Dev.” is the estimate of <span class="math inline">\(\sqrt{\sigma^2_{0j}}\)</span>, the standard deviation among the donors due to the random effect of <span class="math inline">\(\texttt{mouse_id}\)</span>.</li>
<li>The second element in “Std.Dev.” is <span class="math inline">\(\sqrt{\sigma^2_{0}}\)</span>, the estimate of the standard deviation of the error variance (<span class="math inline">\(\varepsilon^2\)</span>). Remember that the residuals of the model are the estimates of <span class="math inline">\(\varepsilon\)</span>.</li>
<li>The ratio <span class="math inline">\(\frac{\sigma^2_{0j}}{\sigma^2_{0j} + \sigma^2_{0}}\)</span> (the ratio of the among-block variance to total random variance) is known as the <a href="https://en.wikipedia.org/wiki/Intraclass_correlation" target="_blank">intraclass correlation</a>. This correlation is an estimate of the correlated error due to the by-mouse clustering if the model were fit without the added random intercept. For <code>exp6g_m1</code>, this correlation is 0.3.</li>
<li>The intraclass correlation ranges between 0 and 1 and makes a pretty good qualitative indicator of repeatability.</li>
</ol>
</div>
<div id="the-linear-mixed-model-exp6g_m1-increases-precision-of-treatment-effects-relative-to-a-fixed-effects-model" class="section level3" number="18.2.7">
<h3><span class="header-section-number">18.2.7</span> The linear mixed model exp6g_m1 increases precision of treatment effects, relative to a fixed effects model</h3>
<p>Let’s compare the effects estimated by the linear mixed model <code>exp6g_m1</code> with a linear model that ignores donor (a <strong>fixed effects</strong> model).</p>
<div class="sourceCode" id="cb548"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb548-1"><a href="lmm.html#cb548-1" aria-hidden="true" tabindex="-1"></a>exp6g_m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(touch <span class="sc">~</span> treatment, <span class="at">data =</span> exp6g)</span></code></pre></div>
<div class="figure"><span id="fig:lmm-exp6g-why2"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp6g-why2-1.png" alt="A. Inference from a linear mixed model with blocking factor (mouse_id) added as a random intercept. B. Inference from a fixed effects model." width="576" />
<p class="caption">
Figure 18.7: A. Inference from a linear mixed model with blocking factor (mouse_id) added as a random intercept. B. Inference from a fixed effects model.
</p>
</div>
<p>Figure <a href="lmm.html#fig:lmm-exp6g-why2">18.7</a>A is a plot of the effects from the linear mixed model <code>exp6g_m1</code> that models the added variance due to mouse. Figure <a href="lmm.html#fig:lmm-exp6g-why2">18.7</a>B is a plot of the effects from the fixed effect model <code>exp6g_m2</code> that ignores the added variance due to mouse. The 95% confidence intervals of the treatment effects in model <code>exp6g_m1</code> are slightly smaller than those in model <code>exp6g_m2</code>. Adding <span class="math inline">\(\texttt{mouse_id}\)</span> as a random factor to the linear model increases the precision of the estimate of the treatment effects by eliminating the among-mouse component of variance from the error variance.</p>
<p>The error variance (the estimate of $^2) in the linear mixed model and the fixed effects model is</p>
<div class="sourceCode" id="cb549"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb549-1"><a href="lmm.html#cb549-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(exp6g_m1)<span class="sc">$</span>sigma<span class="sc">^</span><span class="dv">2</span></span></code></pre></div>
<pre><code>## [1] 57.99549</code></pre>
<div class="sourceCode" id="cb551"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb551-1"><a href="lmm.html#cb551-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(exp6g_m2)<span class="sc">$</span>sigma<span class="sc">^</span><span class="dv">2</span></span></code></pre></div>
<pre><code>## [1] 82.8777</code></pre>
<p>The error variance is 43% higher in the fixed effects model. In the linear mixed model, the variance lost from the error was shifted to the random intercept. We can track this shift with the table of the two variance components of the linear mixed model, shown above, and here again (the values are the square roots of the variances).</p>
<div class="sourceCode" id="cb553"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb553-1"><a href="lmm.html#cb553-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(exp6g_m1)</span></code></pre></div>
<pre><code>##  Groups   Name        Std.Dev.
##  mouse_id (Intercept) 4.9882  
##  Residual             7.6155</code></pre>
<p>The sum of the two variance components of the linear mixed model is equal to the error variance of the fixed effects model:</p>
<div class="sourceCode" id="cb555"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb555-1"><a href="lmm.html#cb555-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">as.data.frame</span>(<span class="fu">VarCorr</span>(exp6g_m1))<span class="sc">$</span>vcov)</span></code></pre></div>
<pre><code>## [1] 82.8777</code></pre>
</div>
<div id="lmm-exp6g-alt" class="section level3" number="18.2.8">
<h3><span class="header-section-number">18.2.8</span> Alternative models for exp6g</h3>
<p>Linear mixed models are very flexible, a topic which is too advanced for this text. Here I want to focus on an alternative model because of its relevance to <strong>repeated measures ANOVA</strong>.</p>
<div class="sourceCode" id="cb557"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb557-1"><a href="lmm.html#cb557-1" aria-hidden="true" tabindex="-1"></a>exp6g_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(touch <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> mouse_id),</span>
<span id="cb557-2"><a href="lmm.html#cb557-2" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp6g)</span>
<span id="cb557-3"><a href="lmm.html#cb557-3" aria-hidden="true" tabindex="-1"></a>exp6g_m1b <span class="ot">&lt;-</span> <span class="fu">lme</span>(touch <span class="sc">~</span> treatment,</span>
<span id="cb557-4"><a href="lmm.html#cb557-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">random =</span> <span class="sc">~</span><span class="dv">1</span><span class="sc">|</span>mouse_id,</span>
<span id="cb557-5"><a href="lmm.html#cb557-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> mouse_id),</span>
<span id="cb557-6"><a href="lmm.html#cb557-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> treatment),</span>
<span id="cb557-7"><a href="lmm.html#cb557-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp6g)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Two models are fit. The two models have the same fixed effect but differ in how they model the random effect and the pattern of correlations in the residuals. The difference in specification determines the error variance and degrees of freedom for computing uncertainty. I’ll return to inference and the problem of “which model to choose” in a moment. First, how do these models differ?</li>
<li>Model <code>exp6g_m1a</code> is the same model analyzed and explained above. This model specifies a random intercept for each level of <span class="math inline">\(\texttt{mouse_id}\)</span>.</li>
<li>Model <code>exp6g_m1b</code> has the same random effects as <code>exp6g_m1a</code> but adds two additional arguments, a <code>correlation</code> argument that explicitly models correlated error in the residuals and a <code>weights</code> argument that models heterogeneity in the residuals.</li>
<li>Consider the error (the residuals) of the fixed effect model <code>touch ~ treatment</code> cast into a matrix with the residuals for each treatment in its own column. This matrix of residuals will be a <span class="math inline">\(7 \times 4\)</span> (7 mice, 4 treatments) matrix that looks like the table of residuals in Example 1 (Table <a href="lmm.html#tab:lmm-demo1-cor-error-table">18.1</a>).</li>
<li>Model <code>exp6g_m1a</code> implicitly models <strong>compound symmetric</strong> correlated error. This means</li>
</ol>
<ul>
<li>the correlation between every pair of columns of the residual matrix is the same.</li>
<li>the variances of each of the columns of the residual matrix is the same.</li>
<li>Model <code>exp6g_m1a</code> assumes zero correlation among the columns of a residual matrix <em>from Model exp6g_m1a</em>. This is because the source of the correlated residuals from the fixed model has been modeled by the random intercept.</li>
</ul>
<ol start="5" style="list-style-type: decimal">
<li>Model <code>exp6g_m1b</code> does not assume <strong>compound symmetric</strong> correlated error.</li>
</ol>
<ul>
<li>the argument <code>correlation = corSymm(form = ~ 1 | mouse_id)</code> in model <code>exp6g_m1b</code> explicitly models different correlations for all pairs of treatment combinations – that is, <strong>unstructured</strong> correlated error.</li>
<li>the argument <code>weights = varIdent(form = ~ 1 | treatment)</code> in model <code>exp6g_m1b</code> explicitly models heterogeneity in the residuals among the levels of $.</li>
</ul>
<ol start="6" style="list-style-type: decimal">
<li>Model <code>exp6g_m1b</code> makes fewer assumptions for inference. The trade-off is the estimation of more parameters and the potential of <a href="https://en.wikipedia.org/wiki/Overfitting" target="_blank">overfitting</a>.</li>
<li>A univariate model of a <strong>repeated measures ANOVA</strong> fit to the <code>exp6g</code> data is equivalent to model <code>exp6g_m1a</code>. See section <a href="lmm.html#lmm-exp6g-rmanova">18.2.10</a>.</li>
<li>A multivariate model of a <strong>repeated measures ANOVA</strong> fit to the <code>exp6g</code> data is equivalent to model <code>exp6g_m1b</code>. See section <a href="lmm.html#lmm-exp6g-rmanova-multi">18.2.11</a>.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp6g-alternatives-planned">Table 18.7: </span>Planned contrasts from the two alternative models.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp6g_m1a</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Saline - Lesion
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-29.07
</td>
<td style="text-align:right;">
-11.96
</td>
<td style="text-align:right;">
-5.04
</td>
<td style="text-align:right;">
0.000085
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
CNO - Saline
</td>
<td style="text-align:right;">
18.67
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
10.12
</td>
<td style="text-align:right;">
27.23
</td>
<td style="text-align:right;">
4.59
</td>
<td style="text-align:right;">
0.000228
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Post_CNO - CNO
</td>
<td style="text-align:right;">
-22.63
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-31.18
</td>
<td style="text-align:right;">
-14.08
</td>
<td style="text-align:right;">
-5.56
</td>
<td style="text-align:right;">
0.000028
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp6g_m1b</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Saline - Lesion
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
3.172
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-27.18
</td>
<td style="text-align:right;">
-13.85
</td>
<td style="text-align:right;">
-6.47
</td>
<td style="text-align:right;">
0.000004
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
CNO - Saline
</td>
<td style="text-align:right;">
18.67
</td>
<td style="text-align:right;">
3.057
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
12.25
</td>
<td style="text-align:right;">
25.10
</td>
<td style="text-align:right;">
6.11
</td>
<td style="text-align:right;">
0.000009
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Post_CNO - CNO
</td>
<td style="text-align:right;">
-22.63
</td>
<td style="text-align:right;">
5.610
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-34.42
</td>
<td style="text-align:right;">
-10.84
</td>
<td style="text-align:right;">
-4.03
</td>
<td style="text-align:right;">
0.000779
</td>
</tr>
</tbody>
</table>
<ol start="9" style="list-style-type: decimal">
<li>Planned comparisons of the two models are given in the tables above. The effect estimates are the same but inference differs quantitatively among the models (but not qualitatively) because of how each model models the error. Which model do we report? One way to evaluate the models is a statistic known as the AIC.</li>
</ol>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp6-aic">Table 18.8: </span>AIC of the two alternative models fit to exp6g data
</caption>
<thead>
<tr>
<th style="text-align:left;">
Model
</th>
<th style="text-align:right;">
AIC
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
exp6g_m1a
</td>
<td style="text-align:right;">
191.3367
</td>
</tr>
<tr>
<td style="text-align:left;">
exp6g_m1b
</td>
<td style="text-align:right;">
198.3888
</td>
</tr>
</tbody>
</table>
<ol start="10" style="list-style-type: decimal">
<li>AIC (<a href="https://en.wikipedia.org/wiki/Akaike_information_criterion" target="_blank">Akaike Information Criterion</a>) is a relative measure of model quality. Compare this to <span class="math inline">\(R^2\)</span>, which is an absolute measure of goodness of fit. The AIC formula has two parts, one is a kind of goodness of fit (like <span class="math inline">\(R^2\)</span>) and the other is a penalty based on the number of parameters in the model. As the goodness of fit increases, the AIC goes down. As the number of parameters increases, the AIC goes up. The model with the lowest AIC is the highest quality model. The actual number, however does not have any absolute meaning; it is only meaningful relative to the AICs computed from fits to different models using the same data.</li>
<li>The AICs of models <code>exp6g_m1a</code> and models <code>exp6g_m1b</code> suggest that model <code>exp6g_m1b</code> is too complex given the data. This has relevance for the repeated measures ANOVA analysis in the next section.</li>
</ol>
</div>
<div id="paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models" class="section level3" number="18.2.9">
<h3><span class="header-section-number">18.2.9</span> Paired t-tests and repeated measures ANOVA are special cases of linear mixed models</h3>
<p>A paired <em>t</em>-test is a special case of linear mixed model fit to data from a randomized complete block design with no subsampling and only two treatment levels (see <a href="violations.html#oneway-paired-t">Lack of independence</a> in the Violations chapter). A repeated measures ANOVA is a special case of linear mixed model fit to data from a randomized complete block design with no subsampling and more than two treatment levels (see <a href="violations.html#oneway-paired-t">Lack of independence</a> in the Violations chapter).</p>
<p>Experiment 6g is a randomized complete block design with no subsampling and four treatments. A “which test” key in a traditional, experimental statistics textbook would guide a researcher to analyze these data using <a href="https://en.wikipedia.org/wiki/Repeated_measures_design#Repeated_measures_ANOVA" target="_blank">repeated measures ANOVA</a>. There are two methods of repeated measures anova, the univariate model and the multivariate model.</p>
</div>
<div id="lmm-exp6g-rmanova" class="section level3" number="18.2.10">
<h3><span class="header-section-number">18.2.10</span> Classical (“univariate model”) repeated measures ANOVA of exp6g</h3>
<div class="sourceCode" id="cb558"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb558-1"><a href="lmm.html#cb558-1" aria-hidden="true" tabindex="-1"></a>exp6g_aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(touch <span class="sc">~</span> treatment <span class="sc">+</span></span>
<span id="cb558-2"><a href="lmm.html#cb558-2" aria-hidden="true" tabindex="-1"></a>                    (treatment <span class="sc">|</span> mouse_id),</span>
<span id="cb558-3"><a href="lmm.html#cb558-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> exp6g)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp6g-uni-rmanova-planned">Table 18.9: </span>Planned contrasts for the linear mixed model exp6g_m1a and the univariate model of the repeated measures ANOVA.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp6g_m1a</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Saline - Lesion
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-29.07
</td>
<td style="text-align:right;">
-11.96
</td>
<td style="text-align:right;">
-5.04
</td>
<td style="text-align:right;">
0.000085
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
CNO - Saline
</td>
<td style="text-align:right;">
18.67
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
10.12
</td>
<td style="text-align:right;">
27.23
</td>
<td style="text-align:right;">
4.59
</td>
<td style="text-align:right;">
0.000228
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Post_CNO - CNO
</td>
<td style="text-align:right;">
-22.63
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-31.18
</td>
<td style="text-align:right;">
-14.08
</td>
<td style="text-align:right;">
-5.56
</td>
<td style="text-align:right;">
0.000028
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>univariate RM-ANOVA</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Saline - Lesion
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-29.07
</td>
<td style="text-align:right;">
-11.96
</td>
<td style="text-align:right;">
-5.04
</td>
<td style="text-align:right;">
0.000085
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
CNO - Saline
</td>
<td style="text-align:right;">
18.67
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
10.12
</td>
<td style="text-align:right;">
27.23
</td>
<td style="text-align:right;">
4.59
</td>
<td style="text-align:right;">
0.000228
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Post_CNO - CNO
</td>
<td style="text-align:right;">
-22.63
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-31.18
</td>
<td style="text-align:right;">
-14.08
</td>
<td style="text-align:right;">
-5.56
</td>
<td style="text-align:right;">
0.000028
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>In addition to the assumptions of the linear model outlined in the Violations chapter, the classical (“univariate model”) repeated measures ANOVA assumes <a href="https://en.wikipedia.org/wiki/Mauchly%27s_sphericity_test#Sphericity" target="_blank">sphericity</a>, which is the equality of the variances of <em>all pairwise differences</em> among treatment combinations.</li>
<li>The univariate model of the classic RM ANOVA is equivalent to Model <code>exp6g_m1a</code>.</li>
</ol>
</div>
<div id="lmm-exp6g-rmanova-multi" class="section level3" number="18.2.11">
<h3><span class="header-section-number">18.2.11</span> “Multivariate model” repeated measures ANOVA</h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp6g-multi-rmanova">Table 18.10: </span>Planned contrasts for the linear mixed model exp6g_m1b and the multivariate model of the repeated measures ANOVA.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp6g_m1b</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Saline - Lesion
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
3.172
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-27.18
</td>
<td style="text-align:right;">
-13.85
</td>
<td style="text-align:right;">
-6.47
</td>
<td style="text-align:right;">
0.000004
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
CNO - Saline
</td>
<td style="text-align:right;">
18.67
</td>
<td style="text-align:right;">
3.057
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
12.25
</td>
<td style="text-align:right;">
25.10
</td>
<td style="text-align:right;">
6.11
</td>
<td style="text-align:right;">
0.000009
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Post_CNO - CNO
</td>
<td style="text-align:right;">
-22.63
</td>
<td style="text-align:right;">
5.610
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-34.42
</td>
<td style="text-align:right;">
-10.84
</td>
<td style="text-align:right;">
-4.03
</td>
<td style="text-align:right;">
0.000779
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>multivariate RM-ANOVA</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Saline - Lesion
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
3.172
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
-28.28
</td>
<td style="text-align:right;">
-12.75
</td>
<td style="text-align:right;">
-6.47
</td>
<td style="text-align:right;">
0.000649
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
CNO - Saline
</td>
<td style="text-align:right;">
18.67
</td>
<td style="text-align:right;">
3.057
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
11.19
</td>
<td style="text-align:right;">
26.15
</td>
<td style="text-align:right;">
6.11
</td>
<td style="text-align:right;">
0.000877
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Post_CNO - CNO
</td>
<td style="text-align:right;">
-22.63
</td>
<td style="text-align:right;">
5.610
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
-36.36
</td>
<td style="text-align:right;">
-8.90
</td>
<td style="text-align:right;">
-4.03
</td>
<td style="text-align:right;">
0.006853
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The sphericity assumption of classical repeated measures ANOVA is relaxed if the model is fit using the “multivariate model.”</li>
<li>The multivariate model repeated measures ANOVA is a linear model with a multivariate response and not a linear mixed model. In the multivariate model, each treatment combination is a different response variable and there is a single row for each level of the random factor (<span class="math inline">\(\texttt{mouse_id}\)</span> in Experiment 6g). The multivariate model is a different way of handling the correlated error that occurs when conceiving of the design as univariate.</li>
<li>The multivariate repeated measures ANOVA is equivalent to the linear mixed model <code>exp6g_m1b</code>. The SEs of the contrasts are the same but the degrees of freedom differ. Because of the increased df of the linear mixed model, the CIs are narrower and the <em>p</em>-value is smaller – that is the linear mixed model is less conservative than the repeated measures ANOVA. There is no correct degrees of freedom for a linear mixed model like this and emmeans outputs one way to compute these (with options for others, none of which are equivalent to those from the multivariate model RM-ANOVA).</li>
</ol>
</div>
<div id="linear-mixed-models-vs-repeated-measures-anova" class="section level3" number="18.2.12">
<h3><span class="header-section-number">18.2.12</span> Linear mixed models vs repeated measures ANOVA</h3>
<p>Many modern textbooks encourage researchers to use linear mixed models instead of repeated measures ANOVA for randomized complete block designs (with or without subsampling) because</p>
<ol style="list-style-type: decimal">
<li>linear mixed models do not exclude random units (subject/mouse/donor/cage) with missing measures of one of the treatment combinations. <a href="violations.html#violations-rmanova">Example 2 (diHOME exp2a) – A repeated measures ANOVA is a special case of a linear mixed model</a> is an example of this kind of missing data.</li>
<li>if there is subsampling, linear mixed models do not <strong>aggregate</strong> the random-unit data, that is, linear mixed models do not simply compute the means and ignore the variance of the sample within each random unit.</li>
<li>linear mixed models allow for modeling additional sources of correlated error, that is, an experiment may have two or more random factor variables (for example, donor and experiment).</li>
<li>linear mixed models allow a researcher to model different patterns of correlated error. This is especially important in longitudinal experiments.</li>
<li>linear mixed models can be generalized to model sampling from non-normal distributions – these are generalized linear mixed models.</li>
</ol>
</div>
<div id="modeling-textttmouse_id-as-a-fixed-effect" class="section level3" number="18.2.13">
<h3><span class="header-section-number">18.2.13</span> Modeling <span class="math inline">\(\texttt{mouse_id}\)</span> as a fixed effect</h3>
<p><span class="math display" id="eq:lmm-exp6g-m3">\[
\begin{equation}
\texttt{touch} \sim \texttt{treatment + mouse_id}
\end{equation}
\tag{18.3}
\]</span></p>
<p>Model <a href="lmm.html#eq:lmm-exp6g-m3">(18.3)</a> (using R formula syntax) is a linear model with the block <span class="math inline">\(\texttt{mouse_id}\)</span> added as a fixed covariate instead of a random intercept. The coefficients of the fit model are</p>
<div class="sourceCode" id="cb559"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb559-1"><a href="lmm.html#cb559-1" aria-hidden="true" tabindex="-1"></a>exp6g_m3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(touch <span class="sc">~</span> treatment <span class="sc">+</span> mouse_id,</span>
<span id="cb559-2"><a href="lmm.html#cb559-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> exp6g)</span>
<span id="cb559-3"><a href="lmm.html#cb559-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb559-4"><a href="lmm.html#cb559-4" aria-hidden="true" tabindex="-1"></a>exp6g_m3_coef <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">coef</span>(<span class="fu">summary</span>(exp6g_m3)),</span>
<span id="cb559-5"><a href="lmm.html#cb559-5" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">confint</span>(exp6g_m3))</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
82.65
</td>
<td style="text-align:right;">
4.551
</td>
<td style="text-align:right;">
18.2
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
73.08
</td>
<td style="text-align:right;">
92.21
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentSaline
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
-5.0
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
-29.07
</td>
<td style="text-align:right;">
-11.96
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentCNO
</td>
<td style="text-align:right;">
-1.84
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
-0.5
</td>
<td style="text-align:right;">
0.6565
</td>
<td style="text-align:right;">
-10.39
</td>
<td style="text-align:right;">
6.71
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentPost_CNO
</td>
<td style="text-align:right;">
-24.47
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
-6.0
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
-33.02
</td>
<td style="text-align:right;">
-15.92
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_idmouse_2
</td>
<td style="text-align:right;">
1.21
</td>
<td style="text-align:right;">
5.385
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:right;">
0.8245
</td>
<td style="text-align:right;">
-10.10
</td>
<td style="text-align:right;">
12.52
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_idmouse_3
</td>
<td style="text-align:right;">
-8.51
</td>
<td style="text-align:right;">
5.385
</td>
<td style="text-align:right;">
-1.6
</td>
<td style="text-align:right;">
0.1315
</td>
<td style="text-align:right;">
-19.82
</td>
<td style="text-align:right;">
2.81
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_idmouse_4
</td>
<td style="text-align:right;">
-4.64
</td>
<td style="text-align:right;">
5.385
</td>
<td style="text-align:right;">
-0.9
</td>
<td style="text-align:right;">
0.4003
</td>
<td style="text-align:right;">
-15.95
</td>
<td style="text-align:right;">
6.67
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_idmouse_5
</td>
<td style="text-align:right;">
10.15
</td>
<td style="text-align:right;">
5.385
</td>
<td style="text-align:right;">
1.9
</td>
<td style="text-align:right;">
0.0758
</td>
<td style="text-align:right;">
-1.17
</td>
<td style="text-align:right;">
21.46
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_idmouse_6
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
5.385
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:right;">
0.9671
</td>
<td style="text-align:right;">
-11.54
</td>
<td style="text-align:right;">
11.09
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_idmouse_7
</td>
<td style="text-align:right;">
6.37
</td>
<td style="text-align:right;">
5.385
</td>
<td style="text-align:right;">
1.2
</td>
<td style="text-align:right;">
0.2521
</td>
<td style="text-align:right;">
-4.94
</td>
<td style="text-align:right;">
17.69
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The coefficients include a slope for the seven non-reference levels of mouse_id We typically don’t care about these.</li>
<li>In a randomized complete block design with only 1 replicate of each treatment combination per block, like that in Experiment 6g, inference about treatment effects is exactly that same between this model and the random intercept model <code>exp6g_m1.</code> Compare the SE, CI and <em>p</em>-value for the coefficients of the treatment effects to those from the linear mixed model <code>exp6g_m1</code>.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
83.27
</td>
<td style="text-align:right;">
3.441
</td>
<td style="text-align:right;">
18.9
</td>
<td style="text-align:right;">
24.20
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
76.72
</td>
<td style="text-align:right;">
89.81
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentSaline
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18.0
</td>
<td style="text-align:right;">
-5.04
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
-28.25
</td>
<td style="text-align:right;">
-12.78
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentCNO
</td>
<td style="text-align:right;">
-1.84
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18.0
</td>
<td style="text-align:right;">
-0.45
</td>
<td style="text-align:right;">
0.6565
</td>
<td style="text-align:right;">
-9.58
</td>
<td style="text-align:right;">
5.90
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentPost_CNO
</td>
<td style="text-align:right;">
-24.47
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18.0
</td>
<td style="text-align:right;">
-6.01
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
-32.21
</td>
<td style="text-align:right;">
-16.73
</td>
</tr>
</tbody>
</table>
<ol start="3" style="list-style-type: decimal">
<li>In a randomized complete block design with only 1 replicate of each treatment combination per block, like that in Experiment 6g, inference about <em>treatment means</em> differs between this model and the random intercept model – the SE of the means of the fixed effect model are smaller than the SE of the means of the linear mixed model. Compare the SE and CI of the <code>(Intercept)</code> (the mean of the reference treatment combination) between the fixed effect and linear mixed model.</li>
<li>The equivalence of inference in the treatment effect between the fixed effect and linear mixed model holds only for balanced randomized complete block designs – where all blocks contain all treatment combinations and the subsampling replicate size is the same for all treatment combinations for all blocks. This means, outside of special cases like Experiment 6g, the choice between adding a blocking variable as a random or fixed factor depends on assumptions about the model.</li>
</ol>
</div>
</div>
<div id="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c" class="section level2" number="18.3">
<h2><span class="header-section-number">18.3</span> Example 3 – Factorial experiments and no subsampling replicates (exp5c)</h2>
<p>Example 3 is similar to example 2 in that there is no subsampling replication and we cannot add random slopes to the linear mixed model. Example 3 differs in that the design is factorial – there are two, crossed fixed factors. Consequently, there are several alternative models with different sets of random intercepts. The reported model includes two random intercepts, one of which models differences in batch effects among treatment levels (treatment by batch interactions). This <strong>interaction intercept</strong> is an alternative to random slope for modeling treatment by batch interactions.</p>
<p><a href="https://www.nature.com/articles/s41467-019-13869-w" target="_blank">Transcriptomic profiling of skeletal muscle adaptations to exercise and inactivity</a></p>
<p>Source figure: <a href="https://www.nature.com/articles/s41467-019-13869-w#Fig5" target="_blank">Fig. 5c</a></p>
<p>Source data: <a href="https://www.nature.com/articles/s41467-019-13869-w#Sec27" target="_blank">Source Data Fig. 5</a></p>
<div id="understand-the-data-1" class="section level3" number="18.3.1">
<h3><span class="header-section-number">18.3.1</span> Understand the data</h3>
<p>The data for Example 1 are from Figure 5c. Six muscle source cells were used to start six independent cultures. Cells from each culture were treated with either a negative control (“Scr”) or a siRNA (“siNR4A3”) that “silences” expression of the NR4A3 gene product by <a href="https://en.wikipedia.org/wiki/Small_interfering_RNA" target="_blank">cleaving the mRNA</a>. Glucose uptake in the two cell types was measured at rest (“Basal”) and during electrical pulse stimulation (“EPS”).</p>
<p>The design is a <span class="math inline">\(2 \times 2\)</span> <a href="https://en.wikipedia.org/wiki/Blocking_(statistics)#Randomized_block_design" target="_blank">Randomized complete block with no subsampling</a>. There are two factors each with two levels: <span class="math inline">\(\texttt{treatment}\)</span> (“Scr,” “siNR4A3”) and <span class="math inline">\(\texttt{activity}\)</span> (“Basal,” “EPS”). Each source cell is a block. All four treatment combinations were measured once per block.</p>
</div>
<div id="examine-the-data-1" class="section level3" number="18.3.2">
<h3><span class="header-section-number">18.3.2</span> Examine the data</h3>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-examine-1.png" width="576" /></p>
<p>The plot shows a strong donor effect.</p>
</div>
<div id="lmm-exp5c-m1" class="section level3" number="18.3.3">
<h3><span class="header-section-number">18.3.3</span> Model fit and inference</h3>
<div class="sourceCode" id="cb560"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb560-1"><a href="lmm.html#cb560-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb560-2"><a href="lmm.html#cb560-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb560-3"><a href="lmm.html#cb560-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb560-4"><a href="lmm.html#cb560-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb560-5"><a href="lmm.html#cb560-5" aria-hidden="true" tabindex="-1"></a>exp5c_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb560-6"><a href="lmm.html#cb560-6" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb560-7"><a href="lmm.html#cb560-7" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment) <span class="sc">+</span></span>
<span id="cb560-8"><a href="lmm.html#cb560-8" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>activity),</span>
<span id="cb560-9"><a href="lmm.html#cb560-9" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb560-10"><a href="lmm.html#cb560-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb560-11"><a href="lmm.html#cb560-11" aria-hidden="true" tabindex="-1"></a>exp5c_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb560-12"><a href="lmm.html#cb560-12" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb560-13"><a href="lmm.html#cb560-13" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment),</span>
<span id="cb560-14"><a href="lmm.html#cb560-14" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb560-15"><a href="lmm.html#cb560-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb560-16"><a href="lmm.html#cb560-16" aria-hidden="true" tabindex="-1"></a>exp5c_m1d <span class="ot">&lt;-</span> <span class="fu">lme</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb560-17"><a href="lmm.html#cb560-17" aria-hidden="true" tabindex="-1"></a>                 <span class="at">random =</span>  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor,</span>
<span id="cb560-18"><a href="lmm.html#cb560-18" aria-hidden="true" tabindex="-1"></a>                 <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb560-19"><a href="lmm.html#cb560-19" aria-hidden="true" tabindex="-1"></a>                 <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span><span class="sc">|</span>t.by.a),</span>
<span id="cb560-20"><a href="lmm.html#cb560-20" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb560-21"><a href="lmm.html#cb560-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb560-22"><a href="lmm.html#cb560-22" aria-hidden="true" tabindex="-1"></a><span class="co"># check AIC</span></span>
<span id="cb560-23"><a href="lmm.html#cb560-23" aria-hidden="true" tabindex="-1"></a><span class="fu">AIC</span>(exp5c_m1a, exp5c_m1b, exp5c_m1c, exp5c_m1d)</span></code></pre></div>
<pre><code>##           df       AIC
## exp5c_m1a  6  8.153813
## exp5c_m1b  8 10.052676
## exp5c_m1c  7  8.052676
## exp5c_m1d 15 14.763024</code></pre>
<div class="sourceCode" id="cb562"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb562-1"><a href="lmm.html#cb562-1" aria-hidden="true" tabindex="-1"></a><span class="co"># check VarCorr model c</span></span>
<span id="cb562-2"><a href="lmm.html#cb562-2" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(exp5c_m1c) <span class="co"># fine</span></span></code></pre></div>
<pre><code>##  Groups          Name        Std.Dev.
##  donor:treatment (Intercept) 0.089519
##  donor           (Intercept) 0.352097
##  Residual                    0.090685</code></pre>
<div class="sourceCode" id="cb564"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb564-1"><a href="lmm.html#cb564-1" aria-hidden="true" tabindex="-1"></a><span class="co"># report 1c (based on AIC and VarCorr check)</span></span>
<span id="cb564-2"><a href="lmm.html#cb564-2" aria-hidden="true" tabindex="-1"></a>exp5c_m1 <span class="ot">&lt;-</span> exp5c_m1c</span></code></pre></div>
<p>exp5c_m1b (equivalent to univariate repeated measures ANOVA) is singular fit. don’t use. Trivial difference in AIC between exp5c_m1a and exp5c_m1c. Nothing in VarCorr with exp5c_m1c raises red flags. Report exp5c_m1c.</p>
<div id="check-the-model-2" class="section level4" number="18.3.3.1">
<h4><span class="header-section-number">18.3.3.1</span> Check the model</h4>
<div class="sourceCode" id="cb565"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb565-1"><a href="lmm.html#cb565-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcheck_the_model</span>(exp5c_m1)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c_m1-check-1.png" width="576" /></p>
<p>fine.</p>
</div>
<div id="inference-from-the-model-5" class="section level4" number="18.3.3.2">
<h4><span class="header-section-number">18.3.3.2</span> Inference from the model</h4>
<div class="sourceCode" id="cb566"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb566-1"><a href="lmm.html#cb566-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1_coef <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">coef</span>(<span class="fu">summary</span>(exp5c_m1)),</span>
<span id="cb566-2"><a href="lmm.html#cb566-2" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">confint</span>(exp5c_m1)[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>),])</span></code></pre></div>
<div class="sourceCode" id="cb567"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb567-1"><a href="lmm.html#cb567-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1_coef <span class="sc">%&gt;%</span></span>
<span id="cb567-2"><a href="lmm.html#cb567-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb567-3"><a href="lmm.html#cb567-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
1.34
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
8.8
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
1.02
</td>
<td style="text-align:right;">
1.66
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
8.5
</td>
<td style="text-align:right;">
1.1
</td>
<td style="text-align:right;">
0.2994
</td>
<td style="text-align:right;">
-0.07
</td>
<td style="text-align:right;">
0.23
</td>
</tr>
<tr>
<td style="text-align:left;">
activityEPS
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
2.9
</td>
<td style="text-align:right;">
0.0163
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.25
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3:activityEPS
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
-3.6
</td>
<td style="text-align:right;">
0.0048
</td>
<td style="text-align:right;">
-0.41
</td>
<td style="text-align:right;">
-0.12
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb568"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb568-1"><a href="lmm.html#cb568-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp5c_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>))</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:left;">
activity
</th>
<th style="text-align:right;">
emmean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Scr
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:right;">
1.34
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
0.96
</td>
<td style="text-align:right;">
1.72
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:right;">
1.42
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
1.04
</td>
<td style="text-align:right;">
1.80
</td>
</tr>
<tr>
<td style="text-align:left;">
Scr
</td>
<td style="text-align:left;">
EPS
</td>
<td style="text-align:right;">
1.49
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
1.11
</td>
<td style="text-align:right;">
1.87
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3
</td>
<td style="text-align:left;">
EPS
</td>
<td style="text-align:right;">
1.31
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
0.93
</td>
<td style="text-align:right;">
1.68
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb569"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb569-1"><a href="lmm.html#cb569-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exp5c_emm # print in console to get row numbers</span></span>
<span id="cb569-2"><a href="lmm.html#cb569-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb569-3"><a href="lmm.html#cb569-3" aria-hidden="true" tabindex="-1"></a>scr_basal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb569-4"><a href="lmm.html#cb569-4" aria-hidden="true" tabindex="-1"></a>siNR4A3_basal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb569-5"><a href="lmm.html#cb569-5" aria-hidden="true" tabindex="-1"></a>scr_eps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb569-6"><a href="lmm.html#cb569-6" aria-hidden="true" tabindex="-1"></a>siNR4A3_eps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb569-7"><a href="lmm.html#cb569-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb569-8"><a href="lmm.html#cb569-8" aria-hidden="true" tabindex="-1"></a>exp5c_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp5c_m1_emm,</span>
<span id="cb569-9"><a href="lmm.html#cb569-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb569-10"><a href="lmm.html#cb569-10" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;(Scr EPS) - (Scr Basal)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(scr_eps <span class="sc">-</span> scr_basal),</span>
<span id="cb569-11"><a href="lmm.html#cb569-11" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;(siNR4A3 EPS) - (siNR4A3 Basal)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(siNR4A3_eps <span class="sc">-</span> siNR4A3_basal),</span>
<span id="cb569-12"><a href="lmm.html#cb569-12" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;Interaction&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(siNR4A3_eps <span class="sc">-</span> siNR4A3_basal) <span class="sc">-</span></span>
<span id="cb569-13"><a href="lmm.html#cb569-13" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">c</span>(scr_eps <span class="sc">-</span> scr_basal)</span>
<span id="cb569-14"><a href="lmm.html#cb569-14" aria-hidden="true" tabindex="-1"></a>                           </span>
<span id="cb569-15"><a href="lmm.html#cb569-15" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb569-16"><a href="lmm.html#cb569-16" aria-hidden="true" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb569-17"><a href="lmm.html#cb569-17" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb569-18"><a href="lmm.html#cb569-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.016
</td>
</tr>
<tr>
<td style="text-align:left;">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
-2.22
</td>
<td style="text-align:right;">
0.051
</td>
</tr>
<tr>
<td style="text-align:left;">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.43
</td>
<td style="text-align:right;">
-0.10
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.005
</td>
</tr>
</tbody>
</table>
</div>
<div id="lmm-exp5c-plotthemodel" class="section level4" number="18.3.3.3">
<h4><span class="header-section-number">18.3.3.3</span> Plot the model</h4>
<div class="sourceCode" id="cb570"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb570-1"><a href="lmm.html#cb570-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot_the_model</span>(exp5c_m1,</span>
<span id="cb570-2"><a href="lmm.html#cb570-2" aria-hidden="true" tabindex="-1"></a>                 exp5c_m1_emm,</span>
<span id="cb570-3"><a href="lmm.html#cb570-3" aria-hidden="true" tabindex="-1"></a>                 exp5c_m1_planned,</span>
<span id="cb570-4"><a href="lmm.html#cb570-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y_label =</span> <span class="st">&quot;Glucose uptake</span><span class="sc">\n</span><span class="st">(pmol per min)&quot;</span>,</span>
<span id="cb570-5"><a href="lmm.html#cb570-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">effect_label =</span> <span class="st">&quot;Difference in Glucose uptake</span><span class="sc">\n</span><span class="st">(pmol per min)&quot;</span>,</span>
<span id="cb570-6"><a href="lmm.html#cb570-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">palette =</span> pal_okabe_ito_blue,</span>
<span id="cb570-7"><a href="lmm.html#cb570-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="fl">0.5</span>,<span class="dv">1</span>))</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-plot-1.png" width="576" /></p>
</div>
<div id="alternaplot-the-model-1" class="section level4" number="18.3.3.4">
<h4><span class="header-section-number">18.3.3.4</span> Alternaplot the model</h4>
<div class="sourceCode" id="cb571"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb571-1"><a href="lmm.html#cb571-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp5c_m1_emm,</span>
<span id="cb571-2"><a href="lmm.html#cb571-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb571-3"><a href="lmm.html#cb571-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb571-4"><a href="lmm.html#cb571-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb571-5"><a href="lmm.html#cb571-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb571-6"><a href="lmm.html#cb571-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb571-7"><a href="lmm.html#cb571-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb571-8"><a href="lmm.html#cb571-8" aria-hidden="true" tabindex="-1"></a><span class="co"># get coefficients of model</span></span>
<span id="cb571-9"><a href="lmm.html#cb571-9" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(exp5c_m1))[, <span class="st">&quot;Estimate&quot;</span>]</span>
<span id="cb571-10"><a href="lmm.html#cb571-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb571-11"><a href="lmm.html#cb571-11" aria-hidden="true" tabindex="-1"></a><span class="co"># get interaction p</span></span>
<span id="cb571-12"><a href="lmm.html#cb571-12" aria-hidden="true" tabindex="-1"></a>p_ixn <span class="ot">&lt;-</span> exp5c_m1_planned[<span class="dv">3</span>, <span class="st">&quot;p.value&quot;</span>] <span class="co"># check!</span></span>
<span id="cb571-13"><a href="lmm.html#cb571-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb571-14"><a href="lmm.html#cb571-14" aria-hidden="true" tabindex="-1"></a>dodge_width <span class="ot">&lt;-</span> <span class="fl">0.4</span></span>
<span id="cb571-15"><a href="lmm.html#cb571-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb571-16"><a href="lmm.html#cb571-16" aria-hidden="true" tabindex="-1"></a>gg <span class="ot">&lt;-</span> <span class="fu">ggplot_the_response</span>(</span>
<span id="cb571-17"><a href="lmm.html#cb571-17" aria-hidden="true" tabindex="-1"></a>  exp5c_m1,</span>
<span id="cb571-18"><a href="lmm.html#cb571-18" aria-hidden="true" tabindex="-1"></a>  exp5c_m1_emm,</span>
<span id="cb571-19"><a href="lmm.html#cb571-19" aria-hidden="true" tabindex="-1"></a>  exp5c_m1_pairs[<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">4</span>),], <span class="co"># only for comparisons within treatment</span></span>
<span id="cb571-20"><a href="lmm.html#cb571-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> pal_okabe_ito_blue,</span>
<span id="cb571-21"><a href="lmm.html#cb571-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend_position =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb571-22"><a href="lmm.html#cb571-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_label =</span> <span class="st">&quot;Glucose uptake</span><span class="sc">\n</span><span class="st">(pmol per min)&quot;</span></span>
<span id="cb571-23"><a href="lmm.html#cb571-23" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb571-24"><a href="lmm.html#cb571-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">x =</span> <span class="dv">2</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="dv">2</span> <span class="sc">-</span> <span class="fl">0.1</span>,</span>
<span id="cb571-25"><a href="lmm.html#cb571-25" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb571-26"><a href="lmm.html#cb571-26" aria-hidden="true" tabindex="-1"></a>               <span class="at">xend =</span> <span class="dv">2</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="dv">2</span> <span class="sc">+</span> <span class="fl">0.1</span>,</span>
<span id="cb571-27"><a href="lmm.html#cb571-27" aria-hidden="true" tabindex="-1"></a>               <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb571-28"><a href="lmm.html#cb571-28" aria-hidden="true" tabindex="-1"></a>               <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb571-29"><a href="lmm.html#cb571-29" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;gray&quot;</span>) <span class="sc">+</span></span>
<span id="cb571-30"><a href="lmm.html#cb571-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bracket</span>(</span>
<span id="cb571-31"><a href="lmm.html#cb571-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fl">2.35</span>,</span>
<span id="cb571-32"><a href="lmm.html#cb571-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb571-33"><a href="lmm.html#cb571-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>] <span class="sc">+</span> b[<span class="dv">4</span>],</span>
<span id="cb571-34"><a href="lmm.html#cb571-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;ixn p = &quot;</span>,</span>
<span id="cb571-35"><a href="lmm.html#cb571-35" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">fmt_p_value_rmd</span>(p_ixn)),</span>
<span id="cb571-36"><a href="lmm.html#cb571-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.size =</span> <span class="dv">3</span>,</span>
<span id="cb571-37"><a href="lmm.html#cb571-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.hjust =</span> <span class="dv">0</span>,</span>
<span id="cb571-38"><a href="lmm.html#cb571-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb571-39"><a href="lmm.html#cb571-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb571-40"><a href="lmm.html#cb571-40" aria-hidden="true" tabindex="-1"></a>gg</span></code></pre></div>
<div class="figure"><span id="fig:lmm-exp5c-alternaplot"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-alternaplot-1.png" alt="Dashed gray line is expected additive mean of &quot;siNR4A3 EPS&quot;" width="576" />
<p class="caption">
Figure 18.8: Dashed gray line is expected additive mean of “siNR4A3 EPS”
</p>
</div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Many researchers might look at the wide confidence intervals relative to the short distance between the means and think “no effect.” The confidence intervals are correct, they simply are <em>not</em> meant to be tools for inferring anything about differences in means. This is one of many reasons why plots of means and error bars can be misleading for inference, despite the ubiquity of their use for communicating results. And, its why I prefer the <a href="lmm.html#lmm-exp5c-plotthemodel">effects-and-response plots</a>, which explicitly communicate correct inference about effects.</li>
</ol>
</div>
</div>
<div id="why-we-care-about-modeling-batch-in-exp5c" class="section level3" number="18.3.4">
<h3><span class="header-section-number">18.3.4</span> Why we care about modeling batch in exp5c</h3>
<p>Figure <a href="lmm.html#fig:lmm-exp5c-why">18.9</a> shows the modeled means of the four treatment combinations and the individual values colored by donor. It is pretty easy to see that the glucose uptake values for donors 4 and 5 are well above the mean for all four treatments. And, the values for donors 1, 2, and 3 are well below the mean for all four treatments. The values for donor 6 are near the mean for all four treatments.</p>
<div class="figure"><span id="fig:lmm-exp5c-why"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-why-1.png" alt="Why we care about blocking. The black dots are the modeled means of each treatment combination. The colored dots are the measured values of the response for each donor. The position of a donor relative to the mean is easy to see with these data." width="672" />
<p class="caption">
Figure 18.9: Why we care about blocking. The black dots are the modeled means of each treatment combination. The colored dots are the measured values of the response for each donor. The position of a donor relative to the mean is easy to see with these data.
</p>
</div>
<p>Let’s compare the effects estimated by the linear mixed model <code>exp5c_m1</code> with a linear model that ignores donor.</p>
<div class="sourceCode" id="cb572"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb572-1"><a href="lmm.html#cb572-1" aria-hidden="true" tabindex="-1"></a>exp5c_m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb572-2"><a href="lmm.html#cb572-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">data =</span> exp5c)</span></code></pre></div>
<div class="figure"><span id="fig:lmm-exp5c-why2"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-why2-1.png" alt="A. Inference from a linear mixed model with blocking factor (donor) added as a random intercept. B. Inference from a fixed effects model." width="576" />
<p class="caption">
Figure 18.10: A. Inference from a linear mixed model with blocking factor (donor) added as a random intercept. B. Inference from a fixed effects model.
</p>
</div>
<p>Figure <a href="lmm.html#fig:lmm-exp5c-why2">18.10</a>A is a plot of the effects from the linear mixed model that models the correlated error due to donor. Figure <a href="lmm.html#fig:lmm-exp5c-why2">18.10</a>B is a plot of the effects from the fixed effect model that ignores the correlated error due to donor. Adding <span class="math inline">\(\texttt{donor}\)</span> as a factor to the linear model increases the precision of the estimate of the treatment effects by eliminating the among-donor component of variance from the error variance.</p>
</div>
<div id="the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts" class="section level3" number="18.3.5">
<h3><span class="header-section-number">18.3.5</span> The linear mixed model exp5c_m1 adds two random intercepts</h3>
<p>There are <em>two</em> random intercepts in the linear mixed model <code>exp5c_m1</code> fit to the exp5c data (<code>exp5c_m1</code> is copied from <code>exp5c_m1c</code> in <a href="lmm.html#lmm-exp5c-m1">Model fit and inference</a>).</p>
<p><span class="math display" id="eq:lmm-exp5c-m1">\[
\begin{align}
\texttt{glucose_uptake}_j = \ &amp;(\beta_0 + \gamma_{0j} + \gamma_{0jk}) + \beta_1 (\texttt{treatment}_\texttt{siNR4A3}) + \beta_2 (\texttt{activity}_\texttt{EPS}) \ + \\
&amp;\beta_3 (\texttt{treatment}_\texttt{siNR4A3}:\texttt{activity}_\texttt{EPS}) + \varepsilon
\end{align}
\tag{18.4}
\]</span></p>
<ol style="list-style-type: decimal">
<li>The random intercept effect <span class="math inline">\(\gamma_{0j}\)</span> models donor variation in the reference level. <em>j</em> indexes donor <em>j</em>. A coefficient is estimated for each donor.</li>
<li>The random intercept effect <span class="math inline">\(\gamma_{0jk}\)</span> models the variation due to the <span class="math inline">\(\texttt{donor}\)</span> by <span class="math inline">\(\texttt{treatment}\)</span> combinations. A coefficient is estimated for each of the 6 (donors) <span class="math inline">\(\times\)</span> 2 (levels of <span class="math inline">\(\texttt{treatment}\)</span>). <em>k</em> indexes treatment level <em>k</em>. This <strong>interaction intercept</strong> is an alternative to random slope for modeling treatment by batch interactions. Unlike a random slope, there is a <span class="math inline">\(\gamma_{0jk}\)</span> coefficient for each donor in the reference level.</li>
</ol>
</div>
<div id="the-fixed-effect-coefficients-of-model-exp5c_m1" class="section level3" number="18.3.6">
<h3><span class="header-section-number">18.3.6</span> The fixed effect coefficients of model exp5c_m1</h3>
<p>The fixed effect coefficients of model <code>exp5c_m1</code> are</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
1.34
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
8.77
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
1.02
</td>
<td style="text-align:right;">
1.66
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
8.5
</td>
<td style="text-align:right;">
1.11
</td>
<td style="text-align:right;">
0.2994
</td>
<td style="text-align:right;">
-0.07
</td>
<td style="text-align:right;">
0.23
</td>
</tr>
<tr>
<td style="text-align:left;">
activityEPS
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.0163
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.25
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3:activityEPS
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.0048
</td>
<td style="text-align:right;">
-0.41
</td>
<td style="text-align:right;">
-0.12
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The interpretation of the fixed effectcs coefficients have the <a href="factorial.html#twoway-what-coefs-are">usual interpretation for a factorial linear model</a>. Figure <a href="lmm.html#fig:lmm-exp5c-explainer-1">18.11</a> is a reminder.</li>
</ol>
<div class="figure"><span id="fig:lmm-exp5c-explainer-1"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-explainer-1-1.png" alt="Fixed effects estimated by exp5c_m1. The light gray point is the expected value of the GPR174- F treatment if genotype and sex were additive." width="672" />
<p class="caption">
Figure 18.11: Fixed effects estimated by exp5c_m1. The light gray point is the expected value of the GPR174- F treatment if genotype and sex were additive.
</p>
</div>
</div>
<div id="the-random-effect-coefficients-of-model-exp5c_m1" class="section level3" number="18.3.7">
<h3><span class="header-section-number">18.3.7</span> The random effect coefficients of model exp5c_m1</h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-m1-random-coef">Table 18.11: </span>Random intercept effects for each donor (g_0j) and for each donor:treatment combination (g_0jk) for exp5c_m1. There are two random intercepts. random_intercept_j is the sum of the fixed intercept (b_0) and g_0j. random_intercept_jk is the sum of the fixed intercept (b_0) and g_0jk.
</caption>
<thead>
<tr>
<th style="text-align:left;">
donor:treatment
</th>
<th style="text-align:right;">
random_intercept_j
</th>
<th style="text-align:right;">
random_intercept_jk
</th>
<th style="text-align:right;">
b_0
</th>
<th style="text-align:right;">
g_0j
</th>
<th style="text-align:right;">
g_0jk
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
donor_1:Scr
</td>
<td style="text-align:right;">
1.018
</td>
<td style="text-align:right;">
1.349
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.323
</td>
<td style="text-align:right;">
0.008
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_1:siNR4A3
</td>
<td style="text-align:right;">
1.018
</td>
<td style="text-align:right;">
1.312
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.323
</td>
<td style="text-align:right;">
-0.029
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_2:Scr
</td>
<td style="text-align:right;">
1.033
</td>
<td style="text-align:right;">
1.283
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.307
</td>
<td style="text-align:right;">
-0.058
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_2:siNR4A3
</td>
<td style="text-align:right;">
1.033
</td>
<td style="text-align:right;">
1.379
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.307
</td>
<td style="text-align:right;">
0.038
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_3:Scr
</td>
<td style="text-align:right;">
1.168
</td>
<td style="text-align:right;">
1.310
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.173
</td>
<td style="text-align:right;">
-0.031
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_3:siNR4A3
</td>
<td style="text-align:right;">
1.168
</td>
<td style="text-align:right;">
1.360
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.173
</td>
<td style="text-align:right;">
0.020
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_4:Scr
</td>
<td style="text-align:right;">
1.785
</td>
<td style="text-align:right;">
1.302
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
0.445
</td>
<td style="text-align:right;">
-0.039
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_4:siNR4A3
</td>
<td style="text-align:right;">
1.785
</td>
<td style="text-align:right;">
1.408
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
0.445
</td>
<td style="text-align:right;">
0.068
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_5:Scr
</td>
<td style="text-align:right;">
1.744
</td>
<td style="text-align:right;">
1.433
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
0.403
</td>
<td style="text-align:right;">
0.092
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_5:siNR4A3
</td>
<td style="text-align:right;">
1.744
</td>
<td style="text-align:right;">
1.275
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
0.403
</td>
<td style="text-align:right;">
-0.066
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_6:Scr
</td>
<td style="text-align:right;">
1.296
</td>
<td style="text-align:right;">
1.368
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.045
</td>
<td style="text-align:right;">
0.028
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_6:siNR4A3
</td>
<td style="text-align:right;">
1.296
</td>
<td style="text-align:right;">
1.310
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.045
</td>
<td style="text-align:right;">
-0.031
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The random intercept “random_intercept_j” is the sum of the fixed intercept <span class="math inline">\(b_0\)</span> and the random intercept effect <span class="math inline">\(g_{0j}\)</span> for donor <em>j</em>. The <span class="math inline">\(g_{0j}\)</span> estimate the <span class="math inline">\(\gamma_{0j}\)</span> in Model <a href="lmm.html#eq:lmm-exp5c-m1">(18.4)</a>. Note that <span class="math inline">\(g_{0j}\)</span> is the same for both treatment levels within a donor. The illustration of these random effects is similar to that for Example 2 <code>exp6g_m1</code> (Figure <a href="lmm.html#fig:lmm-exp6g-explainer-2">18.6</a>.</li>
<li>The random intercept “random_intercept_jk” is the sum of the fixed intercept <span class="math inline">\(b_0\)</span> and the random intercept effect <span class="math inline">\(g_{0jk}\)</span> for donor <em>j</em> in treatment level <em>k</em> (with the reference level equal to 1). The <span class="math inline">\(g_{0jk}\)</span> estimate the <span class="math inline">\(\gamma_{0jk}\)</span> in Model <a href="lmm.html#eq:lmm-exp5c-m1">(18.4)</a>. The <span class="math inline">\(g_{0jk}\)</span> differ for each combination of <span class="math inline">\(\texttt{donor}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>.</li>
<li>The donor:treatment random intercepts (“random_intercept_jk”) are shown with the dark, colored dots in Figure <a href="lmm.html#fig:lmm-exp5c-explainer-2">18.12</a>A. The distance from the dark colored dot to the dashed, gray line (the modeled mean for the treatment:activity combination) is the random intercept effect estimate of <span class="math inline">\(\gamma_{0jk}\)</span>.</li>
<li>The set of random intercept effects for the donor:treatment combinations (<span class="math inline">\(g_{0jk}\)</span>) are the same between the two activity levels. This is seen in Figure <a href="lmm.html#fig:lmm-exp5c-explainer-2">18.12</a>A - compare the pattern of dark, colored dots in the two EPS treatments to the two Basal treatments.</li>
<li>The combined random intercept effects (<span class="math inline">\(g_{0j} + g_{0jk}\)</span>) are shown in Figure <a href="lmm.html#fig:lmm-exp5c-explainer-2">18.12</a>B as the distance between the dark, colored dots and the dashed, gray lines.</li>
<li>There is some communication decision-making in Figure <a href="lmm.html#fig:lmm-exp5c-explainer-2">18.12</a>. The actual y-values of the dark, colored dots are the random intercept effect plus the treatment-combination modeled mean. Thus, in A, the dark colored dots are equal to “random_intercept_jk” only at the reference level (since the reference model mean is the fixed intercept).</li>
</ol>
<div class="figure"><span id="fig:lmm-exp5c-explainer-2"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-explainer-2-1.png" alt="Random intercepts for model exp5c_m1. The pale, colored dots are the measured glucose uptake values for each donor at each treatment combination. The dashed, gray lines are the modeled means for each treatment combination. (A) The distance from a dark colored dot to the dashed, gray line is the random intercept effect $g_{0jk}$. (B) The distance from a dark colored dot to the dashed, gray line is the combined random intercept effect $g_{0j} + g_{0jk}$." width="1066.66666666667" />
<p class="caption">
Figure 18.12: Random intercepts for model exp5c_m1. The pale, colored dots are the measured glucose uptake values for each donor at each treatment combination. The dashed, gray lines are the modeled means for each treatment combination. (A) The distance from a dark colored dot to the dashed, gray line is the random intercept effect <span class="math inline">\(g_{0jk}\)</span>. (B) The distance from a dark colored dot to the dashed, gray line is the combined random intercept effect <span class="math inline">\(g_{0j} + g_{0jk}\)</span>.
</p>
</div>
</div>
<div id="lmm-exp5c-alt" class="section level3" number="18.3.8">
<h3><span class="header-section-number">18.3.8</span> Alternative models for exp5c</h3>
<div class="sourceCode" id="cb573"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb573-1"><a href="lmm.html#cb573-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb573-2"><a href="lmm.html#cb573-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb573-3"><a href="lmm.html#cb573-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb573-4"><a href="lmm.html#cb573-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb573-5"><a href="lmm.html#cb573-5" aria-hidden="true" tabindex="-1"></a>exp5c_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb573-6"><a href="lmm.html#cb573-6" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb573-7"><a href="lmm.html#cb573-7" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment) <span class="sc">+</span></span>
<span id="cb573-8"><a href="lmm.html#cb573-8" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>activity),</span>
<span id="cb573-9"><a href="lmm.html#cb573-9" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span></code></pre></div>
<pre><code>## boundary (singular) fit: see ?isSingular</code></pre>
<div class="sourceCode" id="cb575"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb575-1"><a href="lmm.html#cb575-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb575-2"><a href="lmm.html#cb575-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb575-3"><a href="lmm.html#cb575-3" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment),</span>
<span id="cb575-4"><a href="lmm.html#cb575-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb575-5"><a href="lmm.html#cb575-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb575-6"><a href="lmm.html#cb575-6" aria-hidden="true" tabindex="-1"></a>exp5c_m1d <span class="ot">&lt;-</span> <span class="fu">lme</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb575-7"><a href="lmm.html#cb575-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">random =</span>  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor,</span>
<span id="cb575-8"><a href="lmm.html#cb575-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb575-9"><a href="lmm.html#cb575-9" aria-hidden="true" tabindex="-1"></a>                 <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span><span class="sc">|</span>t.by.a),</span>
<span id="cb575-10"><a href="lmm.html#cb575-10" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Four models are fit. The models specify the same fixed effects but different random effects or patterns of correlated error. Again, we care about the fixed effects – this is the point of the experiment – but the specification of the random effects determines the error variance and degrees of freedom for computing uncertainty.</li>
<li>Model <code>exp5c_m1a</code> specifies a random intercept for each level of <span class="math inline">\(\texttt{donor}\)</span>.</li>
<li>Model <code>exp5c_m1b</code> adds two additional random intercepts to Model <code>exp5c_m1a</code>. The code <code>(1 | donor:treatment)</code> adds an intercept for each combination of <span class="math inline">\(\texttt{donor}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>. The code <code>(1 | donor:activity)</code> adds an intercept for each combination of <span class="math inline">\(\texttt{donor}\)</span> and <span class="math inline">\(\texttt{activity}\)</span>. This model is the equivalent of the univariate model of a <strong>repeated measures ANOVA</strong> of these data (see <a href="lmm.html#lmm-rmanova-5c">Classical (“univariate model”) repeated measures ANOVA</a> below).</li>
<li>Model <code>exp5c_m1c</code> adds only one additional random intercept (<code>(1 | donor:treatment)</code>). It is a simplification of Model <code>exp5c_m1c</code>. This is the model reported and explained above.</li>
<li>Model <code>exp5c_m1d</code> has the same random effects as <code>exp5c_m1a</code> but models unstructured correlated error and heterogeneity of the error, as described for Example 2: <a href="lmm.html#lmm-exp6g-alt">Alternative models for exp6g</a>. This model is equivalent to the multivariate model of a <strong>repeated measures ANOVA</strong> of these data (see <a href="lmm.html#lmm-rmanova-5c-multi">“Multivariate model” repeated measures ANOVA</a> below).</li>
<li><code>lmer</code> returns a message “boundary (singular) fit: see ?isSingular” for <code>exp5c_m1b</code>. The model <em>was</em> fit but the message means that we should be cautious about (or simply avoid) interpreting any inferential statistics (SEs, CIs, <em>p</em>-values). Looking at the estimates of the estimated parameters of the random effects in the table below (the variances of the effects and the covariances among the effects), Model <code>exp5c_m1b</code> estimates zero variance for the <span class="math inline">\(\texttt{donor:activity}\)</span> random intercept. This suggests that we could simplify <code>exp5c_m1b</code> by removing <code>(1 | donor:activity)</code> from the model – this is <code>exp5c_m1c</code>. Think about this warning. If we fit using repeated measures ANOVA (next section), we won’t get a warning message but we still may be <a href="https://en.wikipedia.org/wiki/Overfitting" target="_blank">overfitting</a>.</li>
</ol>
<div class="sourceCode" id="cb576"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb576-1"><a href="lmm.html#cb576-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(exp5c_m1b)</span></code></pre></div>
<pre><code>##  Groups          Name        Std.Dev.
##  donor:activity  (Intercept) 0.000000
##  donor:treatment (Intercept) 0.089519
##  donor           (Intercept) 0.352092
##  Residual                    0.090686</code></pre>
<ol start="7" style="list-style-type: decimal">
<li>Planned comparisons of the four alternative models are given in Table <a href="lmm.html#tab:lmm-exp5c-alternatives-planned">18.12</a>. The effect estimates are the same but inference differs slightly among the models because of how each model partitions error variance to either random effects or the residuals.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-alternatives-planned">Table 18.12: </span>Planned contrasts from the four alternative models.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1a</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
2.25
</td>
<td style="text-align:right;">
0.040
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.26
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
-1.73
</td>
<td style="text-align:right;">
0.105
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.47
</td>
<td style="text-align:right;">
-0.06
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
0.013
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1b</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.016
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
-2.22
</td>
<td style="text-align:right;">
0.051
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.46
</td>
<td style="text-align:right;">
-0.08
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.015
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1c</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.016
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
-2.22
</td>
<td style="text-align:right;">
0.051
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.43
</td>
<td style="text-align:right;">
-0.10
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.005
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1d</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.07
</td>
<td style="text-align:right;">
0.24
</td>
<td style="text-align:right;">
3.82
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.25
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
-1.85
</td>
<td style="text-align:right;">
0.083
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.44
</td>
<td style="text-align:right;">
-0.09
</td>
<td style="text-align:right;">
-3.22
</td>
<td style="text-align:right;">
0.006
</td>
</tr>
</tbody>
</table>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-aic">Table 18.13: </span>AIC of the four alternative models firt to exp5c data
</caption>
<thead>
<tr>
<th style="text-align:left;">
Model
</th>
<th style="text-align:right;">
AIC
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
exp5c_m1a
</td>
<td style="text-align:right;">
8.153813
</td>
</tr>
<tr>
<td style="text-align:left;">
exp5c_m1b
</td>
<td style="text-align:right;">
10.052675
</td>
</tr>
<tr>
<td style="text-align:left;">
exp5c_m1c
</td>
<td style="text-align:right;">
8.052675
</td>
</tr>
<tr>
<td style="text-align:left;">
exp5c_m1d
</td>
<td style="text-align:right;">
14.763024
</td>
</tr>
</tbody>
</table>
<ol start="6" style="list-style-type: decimal">
<li>The use of AIC for selecting among models with different random effects was described above. The AICs of the models suggest that models <code>exp5c_m1b</code> and <code>exp5c_m1d</code> are too complex given the data. The AICs of models <code>exp5c_m1a</code> and <code>exp5c_m1c</code> are effectively the same. I reported <code>exp5c_m1c</code> simply to allow me to explain the <code>donor:treatment</code> random effect.</li>
</ol>
</div>
<div id="lmm-rmanova-5c" class="section level3" number="18.3.9">
<h3><span class="header-section-number">18.3.9</span> Classical (“univariate model”) repeated measures ANOVA</h3>
<div class="sourceCode" id="cb578"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb578-1"><a href="lmm.html#cb578-1" aria-hidden="true" tabindex="-1"></a>exp5c_aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb578-2"><a href="lmm.html#cb578-2" aria-hidden="true" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb578-3"><a href="lmm.html#cb578-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> exp5c)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-uni-rmanova-planned">Table 18.14: </span>Planned contrasts for the linear mixed model exp5c_m1c and the univariate model of the repeated measures ANOVA.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1c</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10.000
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.016
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10.000
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
-2.22
</td>
<td style="text-align:right;">
0.051
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
10.000
</td>
<td style="text-align:right;">
-0.43
</td>
<td style="text-align:right;">
-0.10
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.005
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>univariate RM-ANOVA</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
9.386
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.017
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
9.386
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
-2.22
</td>
<td style="text-align:right;">
0.053
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
5.000
</td>
<td style="text-align:right;">
-0.48
</td>
<td style="text-align:right;">
-0.05
</td>
<td style="text-align:right;">
-3.22
</td>
<td style="text-align:right;">
0.023
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>In addition to the assumptions of the linear model outlined in the Violations chapter, the classical (“univariate model”) repeated measures ANOVA assumes <a href="https://en.wikipedia.org/wiki/Mauchly%27s_sphericity_test#Sphericity" target="_blank">sphericity</a>, which is the equality of the variances of all pairwise differences among treatment combinations.</li>
<li>The univariate model of the repeated measures ANOVA is equivalent to Model <code>exp5c_m1c</code>. Inference (SEs, CIs, <em>p</em>-values) differs slightly in this here because the estimated variance for the <span class="math inline">\(\texttt{donor:activity}\)</span> random intercept of Model <code>exp5c_m1c</code> is zero. In cases where all variances are greater than zero <em>and the design is balanced</em> (no block is missing a treatment combination), the two tables would be identical.</li>
</ol>
</div>
<div id="lmm-rmanova-5c-multi" class="section level3" number="18.3.10">
<h3><span class="header-section-number">18.3.10</span> “Multivariate model” repeated measures ANOVA of exp5c</h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-multi-rmanova">Table 18.15: </span>Planned contrasts for the linear mixed model exp5c_m1d and the multivariate model of the repeated measures ANOVA.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1d</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.07
</td>
<td style="text-align:right;">
0.24
</td>
<td style="text-align:right;">
3.82
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.25
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
-1.85
</td>
<td style="text-align:right;">
0.083
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.44
</td>
<td style="text-align:right;">
-0.09
</td>
<td style="text-align:right;">
-3.22
</td>
<td style="text-align:right;">
0.006
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>multivariate RM-ANOVA</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.25
</td>
<td style="text-align:right;">
3.82
</td>
<td style="text-align:right;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.28
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
-1.85
</td>
<td style="text-align:right;">
0.123
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.48
</td>
<td style="text-align:right;">
-0.05
</td>
<td style="text-align:right;">
-3.22
</td>
<td style="text-align:right;">
0.023
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The sphericity assumption of the repeated measures ANOVA is relaxed if the model is fit using the “multivariate model.”</li>
<li>The multivariate model repeated measures ANOVA is a linear model with a multivariate response and not a linear mixed model. In the multivariate model, each treatment combination is a different response variable and there is a single row for each level of the random factor (<span class="math inline">\(\texttt{donor}\)</span> in Experiment 5c). The multivariate model is a different way of handling the correlated error that occurs when conceiving of the design as univariate.</li>
<li>The multivariate repeated measures ANOVA can be specified using the linear mixed model <code>exp5c_m1d</code>. Compared to the contrasts from the multivariate model of the repeated measures ANOVA above, the SEs of the contrasts are the same but the degrees of freedom differ. Because of the increased df of the linear mixed model, the CIs are narrower and the <em>p</em>-value is smaller – that is the linear mixed model is less conservative than the repeated measures ANOVA. There is no correct degrees of freedom for a linear mixed model like this and emmeans outputs one way to compute these (with options for others, none of which are equivalent to those from the multivariate model RM-ANOVA).</li>
</ol>
</div>
<div id="modeling-textttdonor-as-a-fixed-effect" class="section level3" number="18.3.11">
<h3><span class="header-section-number">18.3.11</span> Modeling <span class="math inline">\(\texttt{donor}\)</span> as a fixed effect</h3>
<p><span class="math display" id="eq:lmm-exp5c-m3">\[
\begin{equation}
\texttt{glucose_uptake} \sim \texttt{treatment * activity + donor}
\end{equation}
\tag{18.5}
\]</span></p>
<p>Model <a href="lmm.html#eq:lmm-exp5c-m3">(18.5)</a> (using R formula syntax) is a linear model with the block <span class="math inline">\(\texttt{donor}\)</span> added as a fixed covariate instead of a random intercept.</p>
<div class="sourceCode" id="cb579"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb579-1"><a href="lmm.html#cb579-1" aria-hidden="true" tabindex="-1"></a>exp5c_m3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span> donor,</span>
<span id="cb579-2"><a href="lmm.html#cb579-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> exp5c)</span></code></pre></div>
<p>The coefficients of the fit model are</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.071
</td>
<td style="text-align:right;">
14.0
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
0.85
</td>
<td style="text-align:right;">
1.15
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
1.2
</td>
<td style="text-align:right;">
0.2453
</td>
<td style="text-align:right;">
-0.06
</td>
<td style="text-align:right;">
0.22
</td>
</tr>
<tr>
<td style="text-align:left;">
activityEPS
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
2.2
</td>
<td style="text-align:right;">
0.0402
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.29
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_2
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:right;">
0.8444
</td>
<td style="text-align:right;">
-0.16
</td>
<td style="text-align:right;">
0.19
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_3
</td>
<td style="text-align:right;">
0.16
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
1.9
</td>
<td style="text-align:right;">
0.0757
</td>
<td style="text-align:right;">
-0.02
</td>
<td style="text-align:right;">
0.33
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_4
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
9.8
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
0.63
</td>
<td style="text-align:right;">
0.98
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_5
</td>
<td style="text-align:right;">
0.76
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
9.2
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
0.59
</td>
<td style="text-align:right;">
0.94
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_6
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
3.5
</td>
<td style="text-align:right;">
0.0030
</td>
<td style="text-align:right;">
0.12
</td>
<td style="text-align:right;">
0.47
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3:activityEPS
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
-2.8
</td>
<td style="text-align:right;">
0.0132
</td>
<td style="text-align:right;">
-0.47
</td>
<td style="text-align:right;">
-0.06
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The coefficients include a slope for the five non-reference levels of donor. We typically don’t care about these.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-250">Table 18.16: </span>Planned contrasts for the linear mixed model exp5c_m1a and the fixed effect model exp5c_m3.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1a</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
2.25
</td>
<td style="text-align:right;">
0.040
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.26
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
-1.73
</td>
<td style="text-align:right;">
0.105
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.47
</td>
<td style="text-align:right;">
-0.06
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
0.013
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m3</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
2.25
</td>
<td style="text-align:right;">
0.040
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.26
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
-1.73
</td>
<td style="text-align:right;">
0.105
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.47
</td>
<td style="text-align:right;">
-0.06
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
0.013
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>In a randomized complete block design with only 1 replicate of each treatment combination per block, like that in Experiment 5c, inference about treatment effects is exactly that same between this fixed effect model and the random intercept model <code>exp5c_m1a.</code></li>
<li>The equivalence of inference in the treatment effect between the fixed effect and linear mixed model holds only for balanced randomized complete block designs – where all blocks contain all treatment combinations and the subsampling replicate size is the same for all treatment combinations for all blocks. This means, outside of special cases like Experiment 5c, the choice between adding a blocking variable as a random or fixed factor depends on assumptions about the model.</li>
</ol>
</div>
</div>
<div id="lmm-example4" class="section level2" number="18.4">
<h2><span class="header-section-number">18.4</span> Example 4 – Experiments with subsampling replication (exp1g)</h2>
<p>This example is from a design with batches (independent experiments) that contain all treatment levels of a single factor <em>and</em> subsampling replication. These data were used to introduce linear mixed models in Example 1. The design of the experiment is <span class="math inline">\(2 \times 2\)</span> factorial. Example 1 flattened the analysis to simplify explanation of random intercepts and random slopes. Here, the data are analyzed with a factorial model.</p>
<p><a href="https://www.nature.com/articles/s41586-019-1873-0" target="_blank">A GPR174–CCL21 module imparts sexual dimorphism to humoral immunity</a></p>
<p><a href="https://pubmed.ncbi.nlm.nih.gov/31875850/" target="_blank">Public source</a></p>
<p>Source figure: <a href="https://www.nature.com/articles/s41586-019-1873-0/figures/1" target="_blank">Fig. 1g</a></p>
<p>Source data: <a href="https://www.nature.com/articles/s41586-019-1873-0#MOESM3" target="_blank">Source Data Fig. 1</a></p>
<div id="understand-the-data-2" class="section level3" number="18.4.1">
<h3><span class="header-section-number">18.4.1</span> Understand the data</h3>
<p>The researchers in this paper are interested in discovering mechanisms causing the lower <a href="https://en.wikipedia.org/wiki/Humoral_immunity" target="_blank">antibody-mediated immune response</a> in males relative to females. The data in Fig. 1 are from a set of experiments on mice to investigate how the G-protein coupled receptor protein GPR174 regulates formation of the <a href="https://en.wikipedia.org/wiki/Germinal_center" target="_blank">B-cell germinal center in secondary lymph tissue</a>. GPR174 is a X-linked gene.</p>
<p><strong>Response variable</strong> <span class="math inline">\(\texttt{gc}\)</span> – germinal center size (%). The units are the percent of cells expressing germinal center markers.</p>
<p><strong>Factor 1</strong> – <span class="math inline">\(\texttt{sex}\)</span> (“M,” “F”). Male (“M”) is the reference level.</p>
<p><strong>Factor 2</strong> – <span class="math inline">\(\texttt{chromosome}\)</span> (“Gpr174+,” “Gpr174-”). “Gpr174-” is a GPR174 knockout. The wildtype (“Gpr174+”) condition is the reference level.</p>
<p><strong>Design</strong> – <span class="math inline">\(2 \times 2\)</span>, that is, two crossed factors each with two levels. This results in four groups, each with a unique combination of the levels from each factor. “M Gpr174+” is the control. “M Gpr174+” is the knockout genotype in males (“knockout added”). “F Gpr174+” is the wildtype female (“X chromosome added”). “F Gpr174-” is the knockout female (“knockout and X chromosome added.”</p>
</div>
<div id="examine-the-data-2" class="section level3" number="18.4.2">
<h3><span class="header-section-number">18.4.2</span> Examine the data</h3>
<div class="sourceCode" id="cb580"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb580-1"><a href="lmm.html#cb580-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> exp1g,</span>
<span id="cb580-2"><a href="lmm.html#cb580-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb580-3"><a href="lmm.html#cb580-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> gc,</span>
<span id="cb580-4"><a href="lmm.html#cb580-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> experiment_id)) <span class="sc">+</span></span>
<span id="cb580-5"><a href="lmm.html#cb580-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.4</span>))</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp1g-examine-1.png" width="576" /></p>
</div>
<div id="fit-the-model-9" class="section level3" number="18.4.3">
<h3><span class="header-section-number">18.4.3</span> Fit the model</h3>
<div class="sourceCode" id="cb581"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb581-1"><a href="lmm.html#cb581-1" aria-hidden="true" tabindex="-1"></a><span class="co"># three slope parameters</span></span>
<span id="cb581-2"><a href="lmm.html#cb581-2" aria-hidden="true" tabindex="-1"></a>exp1g_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb581-3"><a href="lmm.html#cb581-3" aria-hidden="true" tabindex="-1"></a>                   (genotype <span class="sc">*</span> sex <span class="sc">|</span> experiment_id),</span>
<span id="cb581-4"><a href="lmm.html#cb581-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb581-5"><a href="lmm.html#cb581-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb581-6"><a href="lmm.html#cb581-6" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(exp1g_m1a) <span class="co"># looks fine</span></span></code></pre></div>
<pre><code>##  Groups        Name                 Std.Dev. Corr                
##  experiment_id (Intercept)          1.77082                      
##                genotypeGpr174-      0.96192  -0.023              
##                sexF                 2.90154   0.459  0.878       
##                genotypeGpr174-:sexF 1.33169  -0.314 -0.625 -0.706
##  Residual                           1.93155</code></pre>
<div class="sourceCode" id="cb583"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb583-1"><a href="lmm.html#cb583-1" aria-hidden="true" tabindex="-1"></a><span class="co"># one slope parameter but capturing all treatment combinations</span></span>
<span id="cb583-2"><a href="lmm.html#cb583-2" aria-hidden="true" tabindex="-1"></a>exp1g_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb583-3"><a href="lmm.html#cb583-3" aria-hidden="true" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb583-4"><a href="lmm.html#cb583-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb583-5"><a href="lmm.html#cb583-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb583-6"><a href="lmm.html#cb583-6" aria-hidden="true" tabindex="-1"></a><span class="co"># intercept interactions</span></span>
<span id="cb583-7"><a href="lmm.html#cb583-7" aria-hidden="true" tabindex="-1"></a>exp1g_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb583-8"><a href="lmm.html#cb583-8" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb583-9"><a href="lmm.html#cb583-9" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype) <span class="sc">+</span></span>
<span id="cb583-10"><a href="lmm.html#cb583-10" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb583-11"><a href="lmm.html#cb583-11" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype<span class="sc">:</span>sex),</span>
<span id="cb583-12"><a href="lmm.html#cb583-12" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb583-13"><a href="lmm.html#cb583-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb583-14"><a href="lmm.html#cb583-14" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(exp1g_m1c) <span class="co"># id:genotype is low</span></span></code></pre></div>
<pre><code>##  Groups                     Name        Std.Dev. 
##  experiment_id:genotype:sex (Intercept) 0.4046670
##  experiment_id:sex          (Intercept) 1.6927499
##  experiment_id:genotype     (Intercept) 0.0001216
##  experiment_id              (Intercept) 2.5487069
##  Residual                               1.9792428</code></pre>
<div class="sourceCode" id="cb585"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb585-1"><a href="lmm.html#cb585-1" aria-hidden="true" tabindex="-1"></a><span class="co"># drop id:genotype which has low variance</span></span>
<span id="cb585-2"><a href="lmm.html#cb585-2" aria-hidden="true" tabindex="-1"></a>exp1g_m1d <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb585-3"><a href="lmm.html#cb585-3" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb585-4"><a href="lmm.html#cb585-4" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb585-5"><a href="lmm.html#cb585-5" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex<span class="sc">:</span>genotype),</span>
<span id="cb585-6"><a href="lmm.html#cb585-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb585-7"><a href="lmm.html#cb585-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb585-8"><a href="lmm.html#cb585-8" aria-hidden="true" tabindex="-1"></a><span class="fu">AIC</span>(exp1g_m1a, exp1g_m1b, exp1g_m1c, exp1g_m1d)</span></code></pre></div>
<pre><code>##           df      AIC
## exp1g_m1a 15 344.1579
## exp1g_m1b 15 344.1529
## exp1g_m1c  9 337.7602
## exp1g_m1d  8 335.7602</code></pre>
<div class="sourceCode" id="cb587"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb587-1"><a href="lmm.html#cb587-1" aria-hidden="true" tabindex="-1"></a><span class="co"># go with exp1g_m1d.</span></span>
<span id="cb587-2"><a href="lmm.html#cb587-2" aria-hidden="true" tabindex="-1"></a>exp1g_m1 <span class="ot">&lt;-</span> exp1g_m1d</span></code></pre></div>
</div>
<div id="inference-from-the-model-6" class="section level3" number="18.4.4">
<h3><span class="header-section-number">18.4.4</span> Inference from the model</h3>
<div class="sourceCode" id="cb588"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb588-1"><a href="lmm.html#cb588-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1_coef <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(exp1g_m1))</span></code></pre></div>
<div class="sourceCode" id="cb589"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb589-1"><a href="lmm.html#cb589-1" aria-hidden="true" tabindex="-1"></a><span class="co"># order of factors reversed in specs because I want sex to be</span></span>
<span id="cb589-2"><a href="lmm.html#cb589-2" aria-hidden="true" tabindex="-1"></a><span class="co"># main x-axis variable in plot</span></span>
<span id="cb589-3"><a href="lmm.html#cb589-3" aria-hidden="true" tabindex="-1"></a>exp1g_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp1g_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;genotype&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb590"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb590-1"><a href="lmm.html#cb590-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exp1g_m1_emm # print in console to get row numbers</span></span>
<span id="cb590-2"><a href="lmm.html#cb590-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb590-3"><a href="lmm.html#cb590-3" aria-hidden="true" tabindex="-1"></a>wt_m <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb590-4"><a href="lmm.html#cb590-4" aria-hidden="true" tabindex="-1"></a>wt_f <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb590-5"><a href="lmm.html#cb590-5" aria-hidden="true" tabindex="-1"></a>ko_m <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb590-6"><a href="lmm.html#cb590-6" aria-hidden="true" tabindex="-1"></a>ko_f <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb590-7"><a href="lmm.html#cb590-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-8"><a href="lmm.html#cb590-8" aria-hidden="true" tabindex="-1"></a><span class="co"># simple effects within males and females + interaction </span></span>
<span id="cb590-9"><a href="lmm.html#cb590-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. (ko_m - wt_m) </span></span>
<span id="cb590-10"><a href="lmm.html#cb590-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. (ko_f - wt_f)</span></span>
<span id="cb590-11"><a href="lmm.html#cb590-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-12"><a href="lmm.html#cb590-12" aria-hidden="true" tabindex="-1"></a>exp1g_contrasts <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb590-13"><a href="lmm.html#cb590-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;KO/M - WT/M&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(ko_m <span class="sc">-</span> wt_m),</span>
<span id="cb590-14"><a href="lmm.html#cb590-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;KO/F - WT/F&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(ko_f <span class="sc">-</span> wt_f),</span>
<span id="cb590-15"><a href="lmm.html#cb590-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Interaction&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(ko_f <span class="sc">-</span> wt_f) <span class="sc">-</span></span>
<span id="cb590-16"><a href="lmm.html#cb590-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(ko_m <span class="sc">-</span> wt_m)</span>
<span id="cb590-17"><a href="lmm.html#cb590-17" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb590-18"><a href="lmm.html#cb590-18" aria-hidden="true" tabindex="-1"></a>exp1g_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp1g_m1_emm,</span>
<span id="cb590-19"><a href="lmm.html#cb590-19" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> exp1g_contrasts,</span>
<span id="cb590-20"><a href="lmm.html#cb590-20" aria-hidden="true" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb590-21"><a href="lmm.html#cb590-21" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb590-22"><a href="lmm.html#cb590-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
KO/M - WT/M
</td>
<td style="text-align:right;">
2.57
</td>
<td style="text-align:right;">
0.714
</td>
<td style="text-align:right;">
5.784
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
4.33
</td>
<td style="text-align:right;">
3.60
</td>
<td style="text-align:right;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;">
KO/F - WT/F
</td>
<td style="text-align:right;">
-2.74
</td>
<td style="text-align:right;">
0.722
</td>
<td style="text-align:right;">
6.100
</td>
<td style="text-align:right;">
-4.50
</td>
<td style="text-align:right;">
-0.98
</td>
<td style="text-align:right;">
-3.79
</td>
<td style="text-align:right;">
0.009
</td>
</tr>
<tr>
<td style="text-align:left;">
Interaction
</td>
<td style="text-align:right;">
-5.30
</td>
<td style="text-align:right;">
1.016
</td>
<td style="text-align:right;">
5.942
</td>
<td style="text-align:right;">
-7.80
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
-5.22
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The direction of the estimated effect is opposite in males and females</li>
</ol>
</div>
<div id="plot-the-model-5" class="section level3" number="18.4.5">
<h3><span class="header-section-number">18.4.5</span> Plot the model</h3>
<div class="sourceCode" id="cb591"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb591-1"><a href="lmm.html#cb591-1" aria-hidden="true" tabindex="-1"></a>gg <span class="ot">&lt;-</span> <span class="fu">ggplot_the_model</span>(exp1g_m1,</span>
<span id="cb591-2"><a href="lmm.html#cb591-2" aria-hidden="true" tabindex="-1"></a>                       exp1g_m1_emm,</span>
<span id="cb591-3"><a href="lmm.html#cb591-3" aria-hidden="true" tabindex="-1"></a>                       exp1g_m1_planned,</span>
<span id="cb591-4"><a href="lmm.html#cb591-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">y_label =</span> <span class="st">&quot;Germinal Center (%)&quot;</span>,</span>
<span id="cb591-5"><a href="lmm.html#cb591-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">effect_label =</span> <span class="st">&quot;Effect (%)&quot;</span>,</span>
<span id="cb591-6"><a href="lmm.html#cb591-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">palette =</span> pal_okabe_ito_blue,</span>
<span id="cb591-7"><a href="lmm.html#cb591-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="fl">0.5</span>,<span class="dv">1</span>))</span>
<span id="cb591-8"><a href="lmm.html#cb591-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb591-9"><a href="lmm.html#cb591-9" aria-hidden="true" tabindex="-1"></a>gg</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp1g-plot-the-model-1.png" width="576" /></p>
</div>
<div id="alternaplot-the-model-2" class="section level3" number="18.4.6">
<h3><span class="header-section-number">18.4.6</span> Alternaplot the model</h3>
<div class="sourceCode" id="cb592"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb592-1"><a href="lmm.html#cb592-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp1g_m1_emm,</span>
<span id="cb592-2"><a href="lmm.html#cb592-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb592-3"><a href="lmm.html#cb592-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb592-4"><a href="lmm.html#cb592-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb592-5"><a href="lmm.html#cb592-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb592-6"><a href="lmm.html#cb592-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb592-7"><a href="lmm.html#cb592-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb592-8"><a href="lmm.html#cb592-8" aria-hidden="true" tabindex="-1"></a><span class="co"># get coefficients of model</span></span>
<span id="cb592-9"><a href="lmm.html#cb592-9" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(exp1g_m1))[, <span class="st">&quot;Estimate&quot;</span>]</span>
<span id="cb592-10"><a href="lmm.html#cb592-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb592-11"><a href="lmm.html#cb592-11" aria-hidden="true" tabindex="-1"></a><span class="co"># get interaction p</span></span>
<span id="cb592-12"><a href="lmm.html#cb592-12" aria-hidden="true" tabindex="-1"></a>p_ixn <span class="ot">&lt;-</span> exp1g_m1_planned[<span class="dv">3</span>, <span class="st">&quot;p.value&quot;</span>] <span class="co"># check!</span></span>
<span id="cb592-13"><a href="lmm.html#cb592-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb592-14"><a href="lmm.html#cb592-14" aria-hidden="true" tabindex="-1"></a>dodge_width <span class="ot">&lt;-</span> <span class="fl">0.4</span></span>
<span id="cb592-15"><a href="lmm.html#cb592-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb592-16"><a href="lmm.html#cb592-16" aria-hidden="true" tabindex="-1"></a>gg <span class="ot">&lt;-</span> <span class="fu">ggplot_the_response</span>(</span>
<span id="cb592-17"><a href="lmm.html#cb592-17" aria-hidden="true" tabindex="-1"></a>  exp1g_m1,</span>
<span id="cb592-18"><a href="lmm.html#cb592-18" aria-hidden="true" tabindex="-1"></a>  exp1g_m1_emm,</span>
<span id="cb592-19"><a href="lmm.html#cb592-19" aria-hidden="true" tabindex="-1"></a>  exp1g_m1_pairs[<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>,], <span class="co"># only for comparisons within sex</span></span>
<span id="cb592-20"><a href="lmm.html#cb592-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> pal_okabe_ito_blue,</span>
<span id="cb592-21"><a href="lmm.html#cb592-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend_position =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb592-22"><a href="lmm.html#cb592-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_label =</span> <span class="st">&quot;Germinal Center (%)&quot;</span></span>
<span id="cb592-23"><a href="lmm.html#cb592-23" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb592-24"><a href="lmm.html#cb592-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">x =</span> <span class="dv">2</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="dv">2</span> <span class="sc">-</span> <span class="fl">0.1</span>,</span>
<span id="cb592-25"><a href="lmm.html#cb592-25" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb592-26"><a href="lmm.html#cb592-26" aria-hidden="true" tabindex="-1"></a>               <span class="at">xend =</span> <span class="dv">2</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="dv">2</span> <span class="sc">+</span> <span class="fl">0.1</span>,</span>
<span id="cb592-27"><a href="lmm.html#cb592-27" aria-hidden="true" tabindex="-1"></a>               <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb592-28"><a href="lmm.html#cb592-28" aria-hidden="true" tabindex="-1"></a>               <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb592-29"><a href="lmm.html#cb592-29" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;gray&quot;</span>) <span class="sc">+</span></span>
<span id="cb592-30"><a href="lmm.html#cb592-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bracket</span>(</span>
<span id="cb592-31"><a href="lmm.html#cb592-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fl">2.35</span>,</span>
<span id="cb592-32"><a href="lmm.html#cb592-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb592-33"><a href="lmm.html#cb592-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>] <span class="sc">+</span> b[<span class="dv">4</span>],</span>
<span id="cb592-34"><a href="lmm.html#cb592-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;ixn p = &quot;</span>,</span>
<span id="cb592-35"><a href="lmm.html#cb592-35" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">fmt_p_value_rmd</span>(p_ixn)),</span>
<span id="cb592-36"><a href="lmm.html#cb592-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.size =</span> <span class="dv">3</span>,</span>
<span id="cb592-37"><a href="lmm.html#cb592-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.hjust =</span> <span class="dv">0</span>,</span>
<span id="cb592-38"><a href="lmm.html#cb592-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb592-39"><a href="lmm.html#cb592-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb592-40"><a href="lmm.html#cb592-40" aria-hidden="true" tabindex="-1"></a>gg</span></code></pre></div>
<div class="figure"><span id="fig:lmm-exp1g-alternaplot"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp1g-alternaplot-1.png" alt="Dashed gray line is expected additive mean of Gpr174- F" width="576" />
<p class="caption">
Figure 18.13: Dashed gray line is expected additive mean of Gpr174- F
</p>
</div>
</div>
<div id="understanding-the-alternative-models" class="section level3" number="18.4.7">
<h3><span class="header-section-number">18.4.7</span> Understanding the alternative models</h3>
<div class="sourceCode" id="cb593"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb593-1"><a href="lmm.html#cb593-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb593-2"><a href="lmm.html#cb593-2" aria-hidden="true" tabindex="-1"></a>                   (genotype <span class="sc">*</span> sex <span class="sc">|</span> experiment_id),</span>
<span id="cb593-3"><a href="lmm.html#cb593-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb593-4"><a href="lmm.html#cb593-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb593-5"><a href="lmm.html#cb593-5" aria-hidden="true" tabindex="-1"></a>exp1g_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb593-6"><a href="lmm.html#cb593-6" aria-hidden="true" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb593-7"><a href="lmm.html#cb593-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb593-8"><a href="lmm.html#cb593-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb593-9"><a href="lmm.html#cb593-9" aria-hidden="true" tabindex="-1"></a>exp1g_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb593-10"><a href="lmm.html#cb593-10" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb593-11"><a href="lmm.html#cb593-11" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype) <span class="sc">+</span></span>
<span id="cb593-12"><a href="lmm.html#cb593-12" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb593-13"><a href="lmm.html#cb593-13" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype<span class="sc">:</span>sex),</span>
<span id="cb593-14"><a href="lmm.html#cb593-14" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb593-15"><a href="lmm.html#cb593-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb593-16"><a href="lmm.html#cb593-16" aria-hidden="true" tabindex="-1"></a>exp1g_m1d <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb593-17"><a href="lmm.html#cb593-17" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb593-18"><a href="lmm.html#cb593-18" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb593-19"><a href="lmm.html#cb593-19" aria-hidden="true" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype<span class="sc">:</span>sex),</span>
<span id="cb593-20"><a href="lmm.html#cb593-20" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>All four models model the same fixed effects. The models differ only in how they model the random effects.</li>
<li>Model <code>exp1g_m1a</code> fits one random intercept, two random slopes, and one random interaction.</li>
</ol>
<ul>
<li><span class="math inline">\(\gamma_{0j}\)</span> – a random intercept modeling batch effects of <span class="math inline">\(\texttt{experiment_id}\)</span> on the intercept</li>
<li><span class="math inline">\(\gamma_{1j}\)</span> – a random slope modeling the effect of the non-reference level of “Gpr174-” on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is a <span class="math inline">\(\texttt{experiment_id} \times \texttt{genotype}\)</span> interaction.</li>
<li><span class="math inline">\(\gamma_{2j}\)</span> – a random slope modeling the effect of “F” (female) on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is a <span class="math inline">\(\texttt{experiment_id} \times \texttt{sex}\)</span> interaction.</li>
<li><span class="math inline">\(\gamma_{3j}\)</span> – a random slope modeling the effect of the interaction effect of “Gpr174-” and “F” on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is a <span class="math inline">\(\texttt{experiment_id} \times \texttt{sex} \times \texttt{genotype}\)</span> interaction.</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Model <code>exp1g_m1b</code> fits one random intercept and three random slopes</li>
</ol>
<ul>
<li><span class="math inline">\(\gamma_{0j}\)</span> – a random intercept modeling batch effects of <span class="math inline">\(\texttt{experiment_id}\)</span> on the intercept. This is modeling the same thing as <span class="math inline">\(\gamma_{0j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{1j}\)</span> – a random slope modeling the effect of “Gpr174- M” on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is modeling the same thing as <span class="math inline">\(\gamma_{1j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{2j}\)</span> – a random slope modeling the effect of “Gpr174+ F” on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is modeling the same thing as <span class="math inline">\(\gamma_{2j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{3j}\)</span> – a random slope modeling the effect of “M Gpr174- F” on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is modeling the added variance accounted for by the random interaction <span class="math inline">\(\gamma_{2j}\)</span> in Model <code>exp1g_m1a</code> but in a different way.</li>
</ul>
<ol start="4" style="list-style-type: decimal">
<li>Model <code>exp1g_m1c</code> fits four random intercepts</li>
</ol>
<ul>
<li><span class="math inline">\(\gamma_{0j}\)</span> – a random intercept modeling batch effects of <span class="math inline">\(\texttt{experiment_id}\)</span> on the intercept. This is modeling the same thing as <span class="math inline">\(\gamma_{0j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{0jk}\)</span> – a random intercept modeling the effects of the combination of <span class="math inline">\(\texttt{experiment_id}\)</span> and <span class="math inline">\(\texttt{genotype}\)</span>. This is very similar to the variance modeled by <span class="math inline">\(\gamma_{1j}\)</span> in Model <code>exp1g_m1a</code> except the draws from <span class="math inline">\(\gamma_{0jk}\)</span> are independent (uncorrelated) of draws from the other random intercepts.</li>
<li><span class="math inline">\(\gamma_{0jl}\)</span> – a random intercept modeling the effects of the combination of <span class="math inline">\(\texttt{experiment_id}\)</span> and <span class="math inline">\(\texttt{sex}\)</span>. This is very similar to the variance modeled by <span class="math inline">\(\gamma_{2j}\)</span> in Model <code>exp1g_m1a</code> except the draws from <span class="math inline">\(\gamma_{0jl}\)</span> are independent (uncorrelated) of draws from the other random intercepts (review <a href="lmm.html#lmm-varcorr">The correlation among random intercepts and slopes</a> if this doesn’t make sense).</li>
<li><span class="math inline">\(\gamma_{0jkl}\)</span> – a random intercept modeling the effects of the combination of <span class="math inline">\(\texttt{experiment_id}\)</span>, <span class="math inline">\(\texttt{genotype}\)</span> and <span class="math inline">\(\texttt{sex}\)</span>. This is very similar to the variance modeled by <span class="math inline">\(\gamma_{3j}\)</span> in Model <code>exp1g_m1a</code> except the draws from <span class="math inline">\(\gamma_{0jkl}\)</span> are independent (uncorrelated) of draws from the other random intercepts.</li>
</ul>
<ol start="5" style="list-style-type: decimal">
<li>Model <code>exp1g_m1d</code> fits the same random intercepts as Model <code>exp1g_m1c</code> but excludes
<span class="math inline">\(\gamma_{0jl}\)</span> (the random intercept for the experiment_id by genotyp combination. This was excluded because of the low variance of this component in the fit model.</li>
</ol>
</div>
<div id="the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b" class="section level3" number="18.4.8">
<h3><span class="header-section-number">18.4.8</span> The VarCorr matrix of models exp1g_m1a and exp1g_m1b</h3>
<p>The random effect similarity of models <code>exp1g_m1a</code> and <code>exp1g_m1b</code> can be seen in the estimated variance components and correlations among the random effects.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-253">Table 18.17: </span>The Varcorr matrix. Standard deviations of random effects on the diagonal. Correlations of random effects on the off-diagonal.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
</th>
<th style="text-align:right;">
</th>
<th style="text-align:right;">
</th>
<th style="text-align:right;">
</th>
</tr>
</thead>
<tbody>
<tr grouplength="4">
<td colspan="5" style="border-bottom: 1px solid;">
<strong>exp1g_m1a</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Intercept)
</td>
<td style="text-align:right;">
1.7708
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
genotypeGpr174-
</td>
<td style="text-align:right;">
-0.0232
</td>
<td style="text-align:right;">
0.9619
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
sexF
</td>
<td style="text-align:right;">
0.4589
</td>
<td style="text-align:right;">
0.8776
</td>
<td style="text-align:right;">
2.9015
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
genotypeGpr174-:sexF
</td>
<td style="text-align:right;">
-0.3138
</td>
<td style="text-align:right;">
-0.6246
</td>
<td style="text-align:right;">
-0.7056
</td>
<td style="text-align:right;">
1.3317
</td>
</tr>
<tr grouplength="4">
<td colspan="5" style="border-bottom: 1px solid;">
<strong>exp1g_m1b</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Intercept)
</td>
<td style="text-align:right;">
1.7710
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
treatmentGpr174- M
</td>
<td style="text-align:right;">
-0.0251
</td>
<td style="text-align:right;">
0.9640
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
treatmentGpr174+ F
</td>
<td style="text-align:right;">
0.4585
</td>
<td style="text-align:right;">
0.8761
</td>
<td style="text-align:right;">
2.9022
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
treatmentGpr174- F
</td>
<td style="text-align:right;">
0.2999
</td>
<td style="text-align:right;">
0.8791
</td>
<td style="text-align:right;">
0.9384
</td>
<td style="text-align:right;">
2.9905
</td>
</tr>
</tbody>
</table>
</div>
<div id="the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means" class="section level3" number="18.4.9">
<h3><span class="header-section-number">18.4.9</span> The linear mixed model has more precision and power than the fixed effect model of batch means</h3>
<div class="sourceCode" id="cb594"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb594-1"><a href="lmm.html#cb594-1" aria-hidden="true" tabindex="-1"></a><span class="co"># means pooling model</span></span>
<span id="cb594-2"><a href="lmm.html#cb594-2" aria-hidden="true" tabindex="-1"></a>exp1g_m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(gc <span class="sc">~</span> sex <span class="sc">*</span> genotype,</span>
<span id="cb594-3"><a href="lmm.html#cb594-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp1g_means)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp1g-lm2-planned">Table 18.18: </span>Planned contrasts for the linear mixed model exp1g_m1 and the fixed effects model of experiment means exp1g_m2.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp1g_m1 (lmm)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
KO/M - WT/M
</td>
<td style="text-align:right;">
2.57
</td>
<td style="text-align:right;">
0.714
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
4.33
</td>
<td style="text-align:right;">
3.60
</td>
<td style="text-align:right;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
KO/F - WT/F
</td>
<td style="text-align:right;">
-2.74
</td>
<td style="text-align:right;">
0.722
</td>
<td style="text-align:right;">
6.1
</td>
<td style="text-align:right;">
-4.50
</td>
<td style="text-align:right;">
-0.98
</td>
<td style="text-align:right;">
-3.79
</td>
<td style="text-align:right;">
0.009
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-5.30
</td>
<td style="text-align:right;">
1.016
</td>
<td style="text-align:right;">
5.9
</td>
<td style="text-align:right;">
-7.80
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
-5.22
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp1g_m2 (lm means pooling)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
KO/M - WT/M
</td>
<td style="text-align:right;">
2.50
</td>
<td style="text-align:right;">
2.257
</td>
<td style="text-align:right;">
12.0
</td>
<td style="text-align:right;">
-2.41
</td>
<td style="text-align:right;">
7.42
</td>
<td style="text-align:right;">
1.11
</td>
<td style="text-align:right;">
0.289
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
KO/F - WT/F
</td>
<td style="text-align:right;">
-2.73
</td>
<td style="text-align:right;">
2.257
</td>
<td style="text-align:right;">
12.0
</td>
<td style="text-align:right;">
-7.65
</td>
<td style="text-align:right;">
2.18
</td>
<td style="text-align:right;">
-1.21
</td>
<td style="text-align:right;">
0.249
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-5.24
</td>
<td style="text-align:right;">
3.192
</td>
<td style="text-align:right;">
12.0
</td>
<td style="text-align:right;">
-12.20
</td>
<td style="text-align:right;">
1.72
</td>
<td style="text-align:right;">
-1.64
</td>
<td style="text-align:right;">
0.127
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>A fixed effects model fit to batch-means pooled data is strongly conservative and will result in less discovery.</li>
<li>Means pooling does not make the data independent in a randomized complete block design. A linear mixed model of batch-means pooled data is a <strong>mixed-effect ANOVA</strong> (Next section. Also see Section <a href="issues.html#issues-exp4d">15.1.1</a> in the Issues chapter).</li>
</ol>
</div>
<div id="fixed-effect-models-and-pseudoreplication" class="section level3" number="18.4.10">
<h3><span class="header-section-number">18.4.10</span> Fixed effect models and pseudoreplication</h3>
<div class="sourceCode" id="cb595"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb595-1"><a href="lmm.html#cb595-1" aria-hidden="true" tabindex="-1"></a><span class="co"># complete pooling model</span></span>
<span id="cb595-2"><a href="lmm.html#cb595-2" aria-hidden="true" tabindex="-1"></a>exp1g_m3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(gc <span class="sc">~</span> sex <span class="sc">*</span> genotype,</span>
<span id="cb595-3"><a href="lmm.html#cb595-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp1g)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
(#tab:lmm-exp1g_m3-compare)Planned contrasts for the linear mixed model exp1g_m1 and the fixed effects model exp1g_m3 with complete pooling.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp1g_m1 (lmm)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
KO/M - WT/M
</td>
<td style="text-align:right;">
2.57
</td>
<td style="text-align:right;">
0.714
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
4.33
</td>
<td style="text-align:right;">
3.60
</td>
<td style="text-align:right;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
KO/F - WT/F
</td>
<td style="text-align:right;">
-2.74
</td>
<td style="text-align:right;">
0.722
</td>
<td style="text-align:right;">
6.1
</td>
<td style="text-align:right;">
-4.50
</td>
<td style="text-align:right;">
-0.98
</td>
<td style="text-align:right;">
-3.79
</td>
<td style="text-align:right;">
0.009
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-5.30
</td>
<td style="text-align:right;">
1.016
</td>
<td style="text-align:right;">
5.9
</td>
<td style="text-align:right;">
-7.80
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
-5.22
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp1g_m3 (lm complete pooling)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
KO/M - WT/M
</td>
<td style="text-align:right;">
2.44
</td>
<td style="text-align:right;">
1.111
</td>
<td style="text-align:right;">
69.0
</td>
<td style="text-align:right;">
0.22
</td>
<td style="text-align:right;">
4.65
</td>
<td style="text-align:right;">
2.20
</td>
<td style="text-align:right;">
0.032
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
KO/F - WT/F
</td>
<td style="text-align:right;">
-2.53
</td>
<td style="text-align:right;">
1.125
</td>
<td style="text-align:right;">
69.0
</td>
<td style="text-align:right;">
-4.77
</td>
<td style="text-align:right;">
-0.28
</td>
<td style="text-align:right;">
-2.25
</td>
<td style="text-align:right;">
0.028
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-4.97
</td>
<td style="text-align:right;">
1.581
</td>
<td style="text-align:right;">
69.0
</td>
<td style="text-align:right;">
-8.12
</td>
<td style="text-align:right;">
-1.81
</td>
<td style="text-align:right;">
-3.14
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Complete pooling is strongly anti-conservative and will result in increased false discovery. Complete pooling is a type of pseudoreplication – the subsamples have been analyzed as if they are independent replicates. Subsamples are not independent.</li>
<li>For the Experiment 1g data, the 95% confidence intervals are wider and the <em>p</em>-values are larger in the complete pool model <code>exp1g_m3</code> compared to the linear mixed model <code>exp1g_m1</code>. This is an unusual result.</li>
</ol>
</div>
<div id="mixed-effect-anova" class="section level3" number="18.4.11">
<h3><span class="header-section-number">18.4.11</span> Mixed-effect ANOVA</h3>
<div class="sourceCode" id="cb596"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb596-1"><a href="lmm.html#cb596-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1_aov <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(gc <span class="sc">~</span> sex <span class="sc">*</span> genotype <span class="sc">+</span></span>
<span id="cb596-2"><a href="lmm.html#cb596-2" aria-hidden="true" tabindex="-1"></a>                  (sex <span class="sc">*</span> genotype <span class="sc">|</span> experiment_id),</span>
<span id="cb596-3"><a href="lmm.html#cb596-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> exp1g)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The formula as the same format as that in Example 3 for repeated measures ANOVA on RCB designs with no subsampling. In biology, this is often called a <strong>mixed effect ANOVA</strong> with two fixed factors and one random factor.</li>
<li>The data are aggregated prior to fitting the model – this means the subsamples are averaged within each batch by treatment combination.</li>
<li>The mixed-effect ANOVA is equivalent to the linear mixed model <code>exp1g_m1c</code> if the all batch by treatment combinations have the same number of subsamples. The design is not balanced in this example.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
(#tab:lmm-exp1g_m1_aov-pairs)Planned contrasts from mixed ANOVA compared to lmm equivalent of mixed ANOVA and lowest AIC lmm.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp1g_m1_aov (mixed ANOVA)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
KO/M - WT/M
</td>
<td style="text-align:right;">
2.50
</td>
<td style="text-align:right;">
0.711
</td>
<td style="text-align:right;">
5.834
</td>
<td style="text-align:right;">
0.75
</td>
<td style="text-align:right;">
4.26
</td>
<td style="text-align:right;">
3.52
</td>
<td style="text-align:right;">
0.013
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
KO/F - WT/F
</td>
<td style="text-align:right;">
-2.73
</td>
<td style="text-align:right;">
0.711
</td>
<td style="text-align:right;">
5.834
</td>
<td style="text-align:right;">
-4.49
</td>
<td style="text-align:right;">
-0.98
</td>
<td style="text-align:right;">
-3.85
</td>
<td style="text-align:right;">
0.009
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-5.24
</td>
<td style="text-align:right;">
1.087
</td>
<td style="text-align:right;">
3.000
</td>
<td style="text-align:right;">
-8.70
</td>
<td style="text-align:right;">
-1.78
</td>
<td style="text-align:right;">
-4.82
</td>
<td style="text-align:right;">
0.017
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp1g_m1_c (lmm equivalent of mixed ANOVA)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
KO/M - WT/M
</td>
<td style="text-align:right;">
2.57
</td>
<td style="text-align:right;">
0.715
</td>
<td style="text-align:right;">
5.784
</td>
<td style="text-align:right;">
0.80
</td>
<td style="text-align:right;">
4.33
</td>
<td style="text-align:right;">
3.59
</td>
<td style="text-align:right;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
KO/F - WT/F
</td>
<td style="text-align:right;">
-2.74
</td>
<td style="text-align:right;">
0.724
</td>
<td style="text-align:right;">
6.100
</td>
<td style="text-align:right;">
-4.50
</td>
<td style="text-align:right;">
-0.97
</td>
<td style="text-align:right;">
-3.78
</td>
<td style="text-align:right;">
0.009
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-5.30
</td>
<td style="text-align:right;">
1.017
</td>
<td style="text-align:right;">
2.972
</td>
<td style="text-align:right;">
-8.56
</td>
<td style="text-align:right;">
-2.05
</td>
<td style="text-align:right;">
-5.21
</td>
<td style="text-align:right;">
0.014
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp1g_md (lmm with min AIC)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
KO/M - WT/M
</td>
<td style="text-align:right;">
2.57
</td>
<td style="text-align:right;">
0.714
</td>
<td style="text-align:right;">
5.784
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
4.33
</td>
<td style="text-align:right;">
3.60
</td>
<td style="text-align:right;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
KO/F - WT/F
</td>
<td style="text-align:right;">
-2.74
</td>
<td style="text-align:right;">
0.722
</td>
<td style="text-align:right;">
6.100
</td>
<td style="text-align:right;">
-4.50
</td>
<td style="text-align:right;">
-0.98
</td>
<td style="text-align:right;">
-3.79
</td>
<td style="text-align:right;">
0.009
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-5.30
</td>
<td style="text-align:right;">
1.016
</td>
<td style="text-align:right;">
5.942
</td>
<td style="text-align:right;">
-7.80
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
-5.22
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="working-in-r-5" class="section level2" number="18.5">
<h2><span class="header-section-number">18.5</span> Working in R</h2>
<div id="fitting-linear-mixed-models-with-lme4" class="section level3" number="18.5.1">
<h3><span class="header-section-number">18.5.1</span> Fitting linear mixed models with lme4</h3>
</div>
<div id="repeated-measures-anova-randomized-complete-block-with-no-subsampling" class="section level3" number="18.5.2">
<h3><span class="header-section-number">18.5.2</span> Repeated measures ANOVA (randomized complete block with no subsampling)</h3>
<div class="sourceCode" id="cb597"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb597-1"><a href="lmm.html#cb597-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this is the rm-ANOVA</span></span>
<span id="cb597-2"><a href="lmm.html#cb597-2" aria-hidden="true" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb597-3"><a href="lmm.html#cb597-3" aria-hidden="true" tabindex="-1"></a>              (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb597-4"><a href="lmm.html#cb597-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">data =</span> exp5c)</span>
<span id="cb597-5"><a href="lmm.html#cb597-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-6"><a href="lmm.html#cb597-6" aria-hidden="true" tabindex="-1"></a><span class="co"># lmm equivalent</span></span>
<span id="cb597-7"><a href="lmm.html#cb597-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-8"><a href="lmm.html#cb597-8" aria-hidden="true" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb597-9"><a href="lmm.html#cb597-9" aria-hidden="true" tabindex="-1"></a>         (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb597-10"><a href="lmm.html#cb597-10" aria-hidden="true" tabindex="-1"></a>         (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment) <span class="sc">+</span></span>
<span id="cb597-11"><a href="lmm.html#cb597-11" aria-hidden="true" tabindex="-1"></a>         (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>activity),</span>
<span id="cb597-12"><a href="lmm.html#cb597-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">data =</span> exp5c)</span>
<span id="cb597-13"><a href="lmm.html#cb597-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-14"><a href="lmm.html#cb597-14" aria-hidden="true" tabindex="-1"></a><span class="co"># random intercept and slope model that is *not* equivalent</span></span>
<span id="cb597-15"><a href="lmm.html#cb597-15" aria-hidden="true" tabindex="-1"></a><span class="co"># this isn&#39;t solvable because there is no subsampling</span></span>
<span id="cb597-16"><a href="lmm.html#cb597-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-17"><a href="lmm.html#cb597-17" aria-hidden="true" tabindex="-1"></a>m3 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb597-18"><a href="lmm.html#cb597-18" aria-hidden="true" tabindex="-1"></a>              (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb597-19"><a href="lmm.html#cb597-19" aria-hidden="true" tabindex="-1"></a>            <span class="at">data =</span> exp5c)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li><code>afex</code> computes the repeated measures anova model using both <code>aov</code> (classical univariate repeated measures ANOVA) and using <code>lm</code> with multiple response variables (the multivariate repeated measures ANOVA). As of this writing, the output from <code>aov</code> is the default.</li>
<li>Given only one measure for each donor within a <span class="math inline">\(\texttt{treatment} \times \texttt{activity}\)</span> combination, the linear mixed model <code>m2</code> is equivalent to the univariate repeated measures model m1.</li>
<li>The model formula in <code>m1</code> looks like that in the linear mixed model <code>m3</code> but the two models are not equivalent.</li>
</ol>
<div id="univariate-vs.-multivariate-repeated-measures-anova" class="section level4" number="18.5.2.1">
<h4><span class="header-section-number">18.5.2.1</span> univariate vs. multivariate repeated measures ANOVA</h4>
<p>Use the multivariate model unless you want to replicate the result of someone who used a univariate model.</p>
<div class="sourceCode" id="cb598"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb598-1"><a href="lmm.html#cb598-1" aria-hidden="true" tabindex="-1"></a><span class="co"># aov_4 computes both uni and multi but default output is uni</span></span>
<span id="cb598-2"><a href="lmm.html#cb598-2" aria-hidden="true" tabindex="-1"></a>exp5c_aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb598-3"><a href="lmm.html#cb598-3" aria-hidden="true" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb598-4"><a href="lmm.html#cb598-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> exp5c)</span>
<span id="cb598-5"><a href="lmm.html#cb598-5" aria-hidden="true" tabindex="-1"></a><span class="co"># exclude uni, so output is multi</span></span>
<span id="cb598-6"><a href="lmm.html#cb598-6" aria-hidden="true" tabindex="-1"></a>exp5c_aov2 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb598-7"><a href="lmm.html#cb598-7" aria-hidden="true" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb598-8"><a href="lmm.html#cb598-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> exp5c,</span>
<span id="cb598-9"><a href="lmm.html#cb598-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">include_aov =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The <code>include_aov = FALSE</code> argument forces output from <code>aov_4</code> to use the multivariate model.</li>
</ol>
<p><strong>contrasts from univariate model</strong></p>
<div class="sourceCode" id="cb599"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb599-1"><a href="lmm.html#cb599-1" aria-hidden="true" tabindex="-1"></a><span class="co"># these 2 are the same</span></span>
<span id="cb599-2"><a href="lmm.html#cb599-2" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb599-3"><a href="lmm.html#cb599-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb599-4"><a href="lmm.html#cb599-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb599-5"><a href="lmm.html#cb599-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb599-6"><a href="lmm.html#cb599-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>##  activity treatment contrast      estimate     SE   df t.ratio p.value
##  Basal    .         siNR4A3 - Scr   0.0813 0.0759 8.60  1.071  0.3132 
##  EPS      .         siNR4A3 - Scr  -0.1858 0.0759 8.60 -2.448  0.0380 
##  .        Scr       EPS - Basal     0.1510 0.0524 9.39  2.884  0.0173 
##  .        siNR4A3   EPS - Basal    -0.1161 0.0524 9.39 -2.218  0.0525</code></pre>
<div class="sourceCode" id="cb601"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb601-1"><a href="lmm.html#cb601-1" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov1<span class="sc">$</span>aov, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb601-2"><a href="lmm.html#cb601-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb601-3"><a href="lmm.html#cb601-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb601-4"><a href="lmm.html#cb601-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb601-5"><a href="lmm.html#cb601-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>##  activity treatment contrast      estimate     SE   df t.ratio p.value
##  Basal    .         siNR4A3 - Scr   0.0813 0.0759 8.60  1.071  0.3132 
##  EPS      .         siNR4A3 - Scr  -0.1858 0.0759 8.60 -2.448  0.0380 
##  .        Scr       EPS - Basal     0.1510 0.0524 9.39  2.884  0.0173 
##  .        siNR4A3   EPS - Basal    -0.1161 0.0524 9.39 -2.218  0.0525</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Passing <code>exp5c_aov1</code> to <code>emmeans</code> will return the contrasts from the univariate model by deault. Make this explicit by passing only the univarite fit <code>exp5c_aov1$aov</code>.</li>
</ol>
<p><strong>contrasts from multivariate model</strong></p>
<div class="sourceCode" id="cb603"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb603-1"><a href="lmm.html#cb603-1" aria-hidden="true" tabindex="-1"></a><span class="co"># these 2 are the same</span></span>
<span id="cb603-2"><a href="lmm.html#cb603-2" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov1,</span>
<span id="cb603-3"><a href="lmm.html#cb603-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>),</span>
<span id="cb603-4"><a href="lmm.html#cb603-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">model =</span> <span class="st">&quot;multivariate&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb603-5"><a href="lmm.html#cb603-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb603-6"><a href="lmm.html#cb603-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb603-7"><a href="lmm.html#cb603-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb603-8"><a href="lmm.html#cb603-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>##  activity treatment contrast      estimate     SE df t.ratio p.value
##  Basal    .         siNR4A3 - Scr   0.0813 0.1012  5  0.804  0.4581 
##  EPS      .         siNR4A3 - Scr  -0.1858 0.0359  5 -5.180  0.0035 
##  .        Scr       EPS - Basal     0.1510 0.0395  5  3.821  0.0124 
##  .        siNR4A3   EPS - Basal    -0.1161 0.0626  5 -1.855  0.1228</code></pre>
<div class="sourceCode" id="cb605"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb605-1"><a href="lmm.html#cb605-1" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov2,</span>
<span id="cb605-2"><a href="lmm.html#cb605-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb605-3"><a href="lmm.html#cb605-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb605-4"><a href="lmm.html#cb605-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb605-5"><a href="lmm.html#cb605-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb605-6"><a href="lmm.html#cb605-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>##  activity treatment contrast      estimate     SE df t.ratio p.value
##  Basal    .         siNR4A3 - Scr   0.0813 0.1012  5  0.804  0.4581 
##  EPS      .         siNR4A3 - Scr  -0.1858 0.0359  5 -5.180  0.0035 
##  .        Scr       EPS - Basal     0.1510 0.0395  5  3.821  0.0124 
##  .        siNR4A3   EPS - Basal    -0.1161 0.0626  5 -1.855  0.1228</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>If our rmANOVA model is fit with the default specification (<code>exp5c_aov1</code>), we can get the multivariate output using the <code>model = "multivariate"</code> argument within <code>emmeans</code> (not the <code>contrast</code> function!). Or, if our fit excluded the univariate model (<code>exp5c_aov2</code>), then we don’t need the <code>model</code> argument.</li>
</ol>
</div>
<div id="the-anova-table" class="section level4" number="18.5.2.2">
<h4><span class="header-section-number">18.5.2.2</span> The ANOVA table</h4>
<div class="sourceCode" id="cb607"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb607-1"><a href="lmm.html#cb607-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nice</span>(exp5c_aov1, <span class="at">correction =</span> <span class="st">&quot;GG&quot;</span>)</span></code></pre></div>
<pre><code>## Anova Table (Type 3 tests)
## 
## Response: glucose_uptake
##               Effect   df  MSE       F   ges p.value
## 1          treatment 1, 5 0.02    0.68  .006     .45
## 2           activity 1, 5 0.01    0.30 .0006     .61
## 3 treatment:activity 1, 5 0.01 10.37 *   .04     .02
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;+&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb609"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb609-1"><a href="lmm.html#cb609-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nice</span>(exp5c_aov1, <span class="at">correction =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>## Anova Table (Type 3 tests)
## 
## Response: glucose_uptake
##               Effect   df  MSE       F   ges p.value
## 1          treatment 1, 5 0.02    0.68  .006     .45
## 2           activity 1, 5 0.01    0.30 .0006     .61
## 3 treatment:activity 1, 5 0.01 10.37 *   .04     .02
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;+&#39; 0.1 &#39; &#39; 1</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The Greenhouse-Geiger (“GG”) correction is the default. While the <code>correction = "GG"</code> argument is not needed, it makes the script more transparent.</li>
<li>For <em>these data</em> the Greenhouse-Geiger correction doesn’t make a difference in the table.</li>
</ol>
</div>
</div>
</div>
<div id="hidden-code-5" class="section level2" number="18.6">
<h2><span class="header-section-number">18.6</span> Hidden code</h2>
<div id="import-exp5c" class="section level3" number="18.6.1">
<h3><span class="header-section-number">18.6.1</span> Import exp5c</h3>
<div class="sourceCode" id="cb611"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb611-1"><a href="lmm.html#cb611-1" aria-hidden="true" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">&quot;Transcriptomic profiling of skeletal muscle adaptations to exercise and inactivity&quot;</span></span>
<span id="cb611-2"><a href="lmm.html#cb611-2" aria-hidden="true" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">&quot;41467_2019_13869_MOESM6_ESM.xlsx&quot;</span></span>
<span id="cb611-3"><a href="lmm.html#cb611-3" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb611-4"><a href="lmm.html#cb611-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb611-5"><a href="lmm.html#cb611-5" aria-hidden="true" tabindex="-1"></a>exp5c_wide <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb611-6"><a href="lmm.html#cb611-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sheet =</span> <span class="st">&quot;Fig5c&quot;</span>,</span>
<span id="cb611-7"><a href="lmm.html#cb611-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">range =</span> <span class="st">&quot;A2:M3&quot;</span>,</span>
<span id="cb611-8"><a href="lmm.html#cb611-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb611-9"><a href="lmm.html#cb611-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>() <span class="sc">%&gt;%</span></span>
<span id="cb611-10"><a href="lmm.html#cb611-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transpose</span>(<span class="at">make.names =</span> <span class="dv">1</span>)</span>
<span id="cb611-11"><a href="lmm.html#cb611-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb611-12"><a href="lmm.html#cb611-12" aria-hidden="true" tabindex="-1"></a>activity_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;EPS&quot;</span>)</span>
<span id="cb611-13"><a href="lmm.html#cb611-13" aria-hidden="true" tabindex="-1"></a>treatment_levels <span class="ot">&lt;-</span> <span class="fu">names</span>(exp5c_wide)</span>
<span id="cb611-14"><a href="lmm.html#cb611-14" aria-hidden="true" tabindex="-1"></a>exp5c_wide[, activity <span class="sc">:</span><span class="er">=</span> <span class="fu">rep</span>(activity_levels, <span class="at">each =</span> <span class="dv">6</span>)]</span>
<span id="cb611-15"><a href="lmm.html#cb611-15" aria-hidden="true" tabindex="-1"></a>exp5c_wide[, activity <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(activity, <span class="at">levels =</span> activity_levels)]</span>
<span id="cb611-16"><a href="lmm.html#cb611-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb611-17"><a href="lmm.html#cb611-17" aria-hidden="true" tabindex="-1"></a>exp5c <span class="ot">&lt;-</span> <span class="fu">melt</span>(exp5c_wide,</span>
<span id="cb611-18"><a href="lmm.html#cb611-18" aria-hidden="true" tabindex="-1"></a>              <span class="at">id.vars =</span> <span class="st">&quot;activity&quot;</span>,</span>
<span id="cb611-19"><a href="lmm.html#cb611-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">variable.name =</span> <span class="st">&quot;treatment&quot;</span>,</span>
<span id="cb611-20"><a href="lmm.html#cb611-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">value.name =</span> <span class="st">&quot;glucose_uptake&quot;</span>)</span>
<span id="cb611-21"><a href="lmm.html#cb611-21" aria-hidden="true" tabindex="-1"></a>exp5c[, treatment <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(treatment, <span class="at">levels =</span> treatment_levels)]</span>
<span id="cb611-22"><a href="lmm.html#cb611-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb611-23"><a href="lmm.html#cb611-23" aria-hidden="true" tabindex="-1"></a>exp5c[, donor <span class="sc">:</span><span class="er">=</span> <span class="fu">rep</span>(<span class="fu">paste0</span>(<span class="st">&quot;donor_&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>), <span class="dv">4</span>)]</span></code></pre></div>
</div>
<div id="import-exp1g" class="section level3" number="18.6.2">
<h3><span class="header-section-number">18.6.2</span> Import exp1g</h3>
<div class="sourceCode" id="cb612"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb612-1"><a href="lmm.html#cb612-1" aria-hidden="true" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">&quot;A GPR174–CCL21 module imparts sexual dimorphism to humoral immunity&quot;</span></span>
<span id="cb612-2"><a href="lmm.html#cb612-2" aria-hidden="true" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">&quot;41586_2019_1873_MOESM3_ESM.xlsx&quot;</span></span>
<span id="cb612-3"><a href="lmm.html#cb612-3" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb612-4"><a href="lmm.html#cb612-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb612-5"><a href="lmm.html#cb612-5" aria-hidden="true" tabindex="-1"></a>exp1g_wide <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb612-6"><a href="lmm.html#cb612-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sheet =</span> <span class="st">&quot;Fig 1g&quot;</span>,</span>
<span id="cb612-7"><a href="lmm.html#cb612-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">range =</span> <span class="st">&quot;B4:E25&quot;</span>,</span>
<span id="cb612-8"><a href="lmm.html#cb612-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_types =</span> <span class="fu">c</span>(<span class="st">&quot;numeric&quot;</span>),</span>
<span id="cb612-9"><a href="lmm.html#cb612-9" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb612-10"><a href="lmm.html#cb612-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span>
<span id="cb612-11"><a href="lmm.html#cb612-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb612-12"><a href="lmm.html#cb612-12" aria-hidden="true" tabindex="-1"></a>genotype_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Gpr174+&quot;</span>, <span class="st">&quot;Gpr174-&quot;</span>)</span>
<span id="cb612-13"><a href="lmm.html#cb612-13" aria-hidden="true" tabindex="-1"></a>sex_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;M&quot;</span>, <span class="st">&quot;F&quot;</span>)</span>
<span id="cb612-14"><a href="lmm.html#cb612-14" aria-hidden="true" tabindex="-1"></a>g.by.s_levels <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">expand.grid</span>(genotype_levels, sex_levels))</span>
<span id="cb612-15"><a href="lmm.html#cb612-15" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(exp1g_wide) <span class="ot">&lt;-</span> g.by.s_levels</span>
<span id="cb612-16"><a href="lmm.html#cb612-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb612-17"><a href="lmm.html#cb612-17" aria-hidden="true" tabindex="-1"></a>exp_levels <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;exp_&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)</span>
<span id="cb612-18"><a href="lmm.html#cb612-18" aria-hidden="true" tabindex="-1"></a>exp1g_wide[, experiment_id <span class="sc">:</span><span class="er">=</span> <span class="fu">rep</span>(exp_levels, <span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">6</span>,<span class="dv">5</span>))] <span class="co">#check!</span></span>
<span id="cb612-19"><a href="lmm.html#cb612-19" aria-hidden="true" tabindex="-1"></a>exp1g_wide[, experiment_id <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(experiment_id)] <span class="co">#check!</span></span>
<span id="cb612-20"><a href="lmm.html#cb612-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb612-21"><a href="lmm.html#cb612-21" aria-hidden="true" tabindex="-1"></a>exp1g <span class="ot">&lt;-</span> <span class="fu">melt</span>(exp1g_wide,</span>
<span id="cb612-22"><a href="lmm.html#cb612-22" aria-hidden="true" tabindex="-1"></a>              <span class="at">id.vars =</span> <span class="st">&quot;experiment_id&quot;</span>,</span>
<span id="cb612-23"><a href="lmm.html#cb612-23" aria-hidden="true" tabindex="-1"></a>              <span class="at">measure.vars =</span> g.by.s_levels,</span>
<span id="cb612-24"><a href="lmm.html#cb612-24" aria-hidden="true" tabindex="-1"></a>              <span class="at">variable.name =</span> <span class="st">&quot;treatment&quot;</span>,</span>
<span id="cb612-25"><a href="lmm.html#cb612-25" aria-hidden="true" tabindex="-1"></a>              <span class="at">value.name =</span> <span class="st">&quot;gc&quot;</span>) <span class="sc">%&gt;%</span> <span class="co"># cell count</span></span>
<span id="cb612-26"><a href="lmm.html#cb612-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb612-27"><a href="lmm.html#cb612-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb612-28"><a href="lmm.html#cb612-28" aria-hidden="true" tabindex="-1"></a>exp1g[, <span class="fu">c</span>(<span class="st">&quot;genotype&quot;</span>, <span class="st">&quot;sex&quot;</span>)<span class="sc">:</span><span class="er">=</span> <span class="fu">tstrsplit</span>(treatment,</span>
<span id="cb612-29"><a href="lmm.html#cb612-29" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">&quot; &quot;</span>,</span>
<span id="cb612-30"><a href="lmm.html#cb612-30" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">fixed =</span> <span class="cn">TRUE</span>)]</span>
<span id="cb612-31"><a href="lmm.html#cb612-31" aria-hidden="true" tabindex="-1"></a>exp1g[, genotype <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(genotype,</span>
<span id="cb612-32"><a href="lmm.html#cb612-32" aria-hidden="true" tabindex="-1"></a>                           <span class="at">levels =</span> genotype_levels)]</span>
<span id="cb612-33"><a href="lmm.html#cb612-33" aria-hidden="true" tabindex="-1"></a>exp1g[, sex <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(sex,</span>
<span id="cb612-34"><a href="lmm.html#cb612-34" aria-hidden="true" tabindex="-1"></a>                           <span class="at">levels =</span> sex_levels)]</span>
<span id="cb612-35"><a href="lmm.html#cb612-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb612-36"><a href="lmm.html#cb612-36" aria-hidden="true" tabindex="-1"></a>exp1g_means <span class="ot">&lt;-</span> exp1g[, .(<span class="at">gc =</span> <span class="fu">mean</span>(gc)),</span>
<span id="cb612-37"><a href="lmm.html#cb612-37" aria-hidden="true" tabindex="-1"></a>                     by <span class="ot">=</span> .(treatment, genotype, sex, experiment_id)]</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="part-vii-expanding-the-linear-model.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="pre-post.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/chapters/55-lmm01-blocking.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Walker-elementary-statistical-modeling-draft.pdf", "Walker-elementary-statistical-modeling-draft.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
