<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 16 Models with random factors – Blocking and pseudoreplication | Elements of Statistical Modeling for Experimental Biology</title>
  <meta name="description" content="A first course in statistical modeling for experimental biology researchers" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 16 Models with random factors – Blocking and pseudoreplication | Elements of Statistical Modeling for Experimental Biology" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A first course in statistical modeling for experimental biology researchers" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 16 Models with random factors – Blocking and pseudoreplication | Elements of Statistical Modeling for Experimental Biology" />
  
  <meta name="twitter:description" content="A first course in statistical modeling for experimental biology researchers" />
  

<meta name="author" content="Copyright 2018 Jeffrey A. Walker" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="part-vii-expanding-the-linear-model.html"/>
<link rel="next" href="pre-post.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="libs/d3-3.3.8/d3.min.js"></script>
<script src="libs/dagre-0.4.0/dagre-d3.min.js"></script>
<link href="libs/mermaid-0.3.0/dist/mermaid.css" rel="stylesheet" />
<script src="libs/mermaid-0.3.0/dist/mermaid.slim.min.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/chromatography-0.1/chromatography.js"></script>
<script src="libs/DiagrammeR-binding-1.0.6.1/DiagrammeR.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Elements of Applied Biostatistics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#math"><i class="fa fa-check"></i><b>0.1</b> Math</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#r-and-programming"><i class="fa fa-check"></i><b>0.2</b> R and programming</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-i-getting-started.html"><a href="part-i-getting-started.html"><i class="fa fa-check"></i>Part I: Getting Started</a></li>
<li class="chapter" data-level="1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html"><i class="fa fa-check"></i><b>1</b> Getting Started – R Projects and R Markdown</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#r-vs-r-studio"><i class="fa fa-check"></i><b>1.1</b> R vs R Studio</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#download-and-install-r-and-r-studio"><i class="fa fa-check"></i><b>1.2</b> Download and install R and R studio</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#open-r-studio-and-modify-the-workspace-preference"><i class="fa fa-check"></i><b>1.3</b> Open R Studio and modify the workspace preference</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#if-you-didnt-modify-the-workspace-preferences-from-the-previous-section-go-back-and-do-it"><i class="fa fa-check"></i><b>1.4</b> If you didn’t modify the workspace preferences from the previous section, go back and do it</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#r-markdown-in-a-nutshell"><i class="fa fa-check"></i><b>1.5</b> R Markdown in a nutshell</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#install-r-markdown"><i class="fa fa-check"></i><b>1.6</b> Install R Markdown</a></li>
<li class="chapter" data-level="1.7" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#importing-packages"><i class="fa fa-check"></i><b>1.7</b> Importing Packages</a></li>
<li class="chapter" data-level="1.8" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#setup-create-project"><i class="fa fa-check"></i><b>1.8</b> Create an R Studio Project for this textbook</a></li>
<li class="chapter" data-level="1.9" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#working-on-a-project-in-a-nutshell"><i class="fa fa-check"></i><b>1.9</b> Working on a project, in a nutshell</a></li>
<li class="chapter" data-level="1.10" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-an-r-markdown-file-for-this-chapter"><i class="fa fa-check"></i><b>1.10</b> Create an R Markdown file for this Chapter</a>
<ul>
<li class="chapter" data-level="1.10.1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#modify-the-yaml-header"><i class="fa fa-check"></i><b>1.10.1</b> Modify the yaml header</a></li>
<li class="chapter" data-level="1.10.2" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#modify-the-setup-chunk"><i class="fa fa-check"></i><b>1.10.2</b> Modify the “setup” chunk</a></li>
<li class="chapter" data-level="1.10.3" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-a-fake-data-chunk"><i class="fa fa-check"></i><b>1.10.3</b> Create a “fake-data” chunk</a></li>
<li class="chapter" data-level="1.10.4" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-a-plot-chunk"><i class="fa fa-check"></i><b>1.10.4</b> Create a “plot” chunk</a></li>
<li class="chapter" data-level="1.10.5" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#knit"><i class="fa fa-check"></i><b>1.10.5</b> Knit</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="analyzing-experimental-data-with-a-linear-model.html"><a href="analyzing-experimental-data-with-a-linear-model.html"><i class="fa fa-check"></i><b>2</b> Analyzing experimental data with a linear model</a>
<ul>
<li class="chapter" data-level="2.1" data-path="analyzing-experimental-data-with-a-linear-model.html"><a href="analyzing-experimental-data-with-a-linear-model.html#this-text-is-about-the-estimation-of-treatment-effects-and-the-uncertainty-in-our-estimates-using-linear-models.-this-raises-the-question-what-is-an-effect"><i class="fa fa-check"></i><b>2.1</b> This text is about the estimation of treatment effects and the uncertainty in our estimates using linear models. This, raises the question, what is “an effect?”</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="background-physiology-to-the-experiments-in-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="background-physiology-to-the-experiments-in-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><i class="fa fa-check"></i>Background physiology to the experiments in Figure 2 of “ASK1 inhibits browning of white adipose tissue in obesity”</a></li>
<li class="chapter" data-level="" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><i class="fa fa-check"></i>Analyses for Figure 2 of “ASK1 inhibits browning of white adipose tissue in obesity”</a>
<ul>
<li class="chapter" data-level="2.2" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#setup"><i class="fa fa-check"></i><b>2.2</b> Setup</a></li>
<li class="chapter" data-level="2.3" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#data-source"><i class="fa fa-check"></i><b>2.3</b> Data source</a></li>
<li class="chapter" data-level="2.4" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#control-the-color-palette"><i class="fa fa-check"></i><b>2.4</b> control the color palette</a></li>
<li class="chapter" data-level="2.5" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#useful-functions"><i class="fa fa-check"></i><b>2.5</b> useful functions</a></li>
<li class="chapter" data-level="2.6" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2b-effect-of-ask1-deletion-on-growth-body-weight"><i class="fa fa-check"></i><b>2.6</b> figure 2b – effect of ASK1 deletion on growth (body weight)</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2b-import"><i class="fa fa-check"></i><b>2.6.1</b> figure 2b – import</a></li>
<li class="chapter" data-level="2.6.2" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2b-exploratory-plots"><i class="fa fa-check"></i><b>2.6.2</b> figure 2b – exploratory plots</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2c-effect-of-ask1-deletion-on-final-body-weight"><i class="fa fa-check"></i><b>2.7</b> Figure 2c – Effect of ASK1 deletion on final body weight</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2c-import"><i class="fa fa-check"></i><b>2.7.1</b> Figure 2c – import</a></li>
<li class="chapter" data-level="2.7.2" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2c-check-own-computation-of-weight-change-v-imported-value"><i class="fa fa-check"></i><b>2.7.2</b> Figure 2c – check own computation of weight change v imported value</a></li>
<li class="chapter" data-level="2.7.3" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2c-exploratory-plots"><i class="fa fa-check"></i><b>2.7.3</b> Figure 2c – exploratory plots</a></li>
<li class="chapter" data-level="2.7.4" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2c-fit-the-model-m1-lm"><i class="fa fa-check"></i><b>2.7.4</b> Figure 2c – fit the model: m1 (lm)</a></li>
<li class="chapter" data-level="2.7.5" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2c-check-the-model-m1"><i class="fa fa-check"></i><b>2.7.5</b> Figure 2c – check the model: m1</a></li>
<li class="chapter" data-level="2.7.6" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2c-fit-the-model-m2-gamma-glm"><i class="fa fa-check"></i><b>2.7.6</b> Figure 2c – fit the model: m2 (gamma glm)</a></li>
<li class="chapter" data-level="2.7.7" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2c-check-the-model-m2"><i class="fa fa-check"></i><b>2.7.7</b> Figure 2c – check the model, m2</a></li>
<li class="chapter" data-level="2.7.8" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2c-inference-from-the-model"><i class="fa fa-check"></i><b>2.7.8</b> Figure 2c – inference from the model</a></li>
<li class="chapter" data-level="2.7.9" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2c-plot-the-model"><i class="fa fa-check"></i><b>2.7.9</b> Figure 2c – plot the model</a></li>
<li class="chapter" data-level="2.7.10" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2c-report"><i class="fa fa-check"></i><b>2.7.10</b> Figure 2c – report</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve"><i class="fa fa-check"></i><b>2.8</b> Figure 2d – Effect of ASK1 KO on glucose tolerance (whole curve)</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2d-import"><i class="fa fa-check"></i><b>2.8.1</b> Figure 2d – Import</a></li>
<li class="chapter" data-level="2.8.2" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2d-exploratory-plots"><i class="fa fa-check"></i><b>2.8.2</b> Figure 2d – exploratory plots</a></li>
<li class="chapter" data-level="2.8.3" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2d-fit-the-model"><i class="fa fa-check"></i><b>2.8.3</b> Figure 2d – fit the model</a></li>
<li class="chapter" data-level="2.8.4" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2d-check-the-model"><i class="fa fa-check"></i><b>2.8.4</b> Figure 2d – check the model</a></li>
<li class="chapter" data-level="2.8.5" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2d-inference"><i class="fa fa-check"></i><b>2.8.5</b> Figure 2d – inference</a></li>
<li class="chapter" data-level="2.8.6" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2d-plot-the-model"><i class="fa fa-check"></i><b>2.8.6</b> Figure 2d – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure"><i class="fa fa-check"></i><b>2.9</b> Figure 2e – Effect of ASK1 deletion on glucose tolerance (summary measure)</a>
<ul>
<li class="chapter" data-level="2.9.1" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2e-message-the-data"><i class="fa fa-check"></i><b>2.9.1</b> Figure 2e – message the data</a></li>
<li class="chapter" data-level="2.9.2" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2e-exploratory-plots"><i class="fa fa-check"></i><b>2.9.2</b> Figure 2e – exploratory plots</a></li>
<li class="chapter" data-level="2.9.3" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2e-fit-the-model"><i class="fa fa-check"></i><b>2.9.3</b> Figure 2e – fit the model</a></li>
<li class="chapter" data-level="2.9.4" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2e-check-the-model"><i class="fa fa-check"></i><b>2.9.4</b> Figure 2e – check the model</a></li>
<li class="chapter" data-level="2.9.5" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2e-inference-from-the-model"><i class="fa fa-check"></i><b>2.9.5</b> Figure 2e – inference from the model</a></li>
<li class="chapter" data-level="2.9.6" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2e-plot-the-model"><i class="fa fa-check"></i><b>2.9.6</b> Figure 2e – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="2.10" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate"><i class="fa fa-check"></i><b>2.10</b> Figure 2f – Effect of ASK1 deletion on glucose infusion rate</a>
<ul>
<li class="chapter" data-level="2.10.1" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2f-import"><i class="fa fa-check"></i><b>2.10.1</b> Figure 2f – import</a></li>
<li class="chapter" data-level="2.10.2" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2f-exploratory-plots"><i class="fa fa-check"></i><b>2.10.2</b> Figure 2f – exploratory plots</a></li>
<li class="chapter" data-level="2.10.3" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2f-fit-the-model"><i class="fa fa-check"></i><b>2.10.3</b> Figure 2f – fit the model</a></li>
<li class="chapter" data-level="2.10.4" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2f-check-the-model"><i class="fa fa-check"></i><b>2.10.4</b> Figure 2f – check the model</a></li>
<li class="chapter" data-level="2.10.5" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2f-inference"><i class="fa fa-check"></i><b>2.10.5</b> Figure 2f – inference</a></li>
<li class="chapter" data-level="2.10.6" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2f-plot-the-model"><i class="fa fa-check"></i><b>2.10.6</b> Figure 2f – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="2.11" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake"><i class="fa fa-check"></i><b>2.11</b> Figure 2g – Effect of ASK1 deletion on tissue-specific glucose uptake</a>
<ul>
<li class="chapter" data-level="2.11.1" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2g-import"><i class="fa fa-check"></i><b>2.11.1</b> Figure 2g – import</a></li>
<li class="chapter" data-level="2.11.2" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2g-exploratory-plots"><i class="fa fa-check"></i><b>2.11.2</b> Figure 2g – exploratory plots</a></li>
<li class="chapter" data-level="2.11.3" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2g-fit-the-model"><i class="fa fa-check"></i><b>2.11.3</b> Figure 2g – fit the model</a></li>
<li class="chapter" data-level="2.11.4" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2g-check-the-model"><i class="fa fa-check"></i><b>2.11.4</b> Figure 2g – check the model</a></li>
<li class="chapter" data-level="2.11.5" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2g-inference"><i class="fa fa-check"></i><b>2.11.5</b> Figure 2g – inference</a></li>
<li class="chapter" data-level="2.11.6" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2g-plot-the-model"><i class="fa fa-check"></i><b>2.11.6</b> Figure 2g – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="2.12" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2h"><i class="fa fa-check"></i><b>2.12</b> Figure 2h</a></li>
<li class="chapter" data-level="2.13" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2i-effect-of-ask1-deletion-on-liver-tg"><i class="fa fa-check"></i><b>2.13</b> Figure 2i – Effect of ASK1 deletion on liver TG</a>
<ul>
<li class="chapter" data-level="2.13.1" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2i-fit-the-model"><i class="fa fa-check"></i><b>2.13.1</b> Figure 2i – fit the model</a></li>
<li class="chapter" data-level="2.13.2" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2i-check-the-model"><i class="fa fa-check"></i><b>2.13.2</b> Figure 2i – check the model</a></li>
<li class="chapter" data-level="2.13.3" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2i-inference"><i class="fa fa-check"></i><b>2.13.3</b> Figure 2i – inference</a></li>
<li class="chapter" data-level="2.13.4" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2i-plot-the-model"><i class="fa fa-check"></i><b>2.13.4</b> Figure 2i – plot the model</a></li>
<li class="chapter" data-level="2.13.5" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2i-report-the-model"><i class="fa fa-check"></i><b>2.13.5</b> Figure 2i – report the model</a></li>
</ul></li>
<li class="chapter" data-level="2.14" data-path="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html"><a href="analyses-for-figure-2-of-ask1-inhibits-browning-of-white-adipose-tissue-in-obesity.html#figure-2j"><i class="fa fa-check"></i><b>2.14</b> Figure 2j</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-iii-r-fundamentals.html"><a href="part-iii-r-fundamentals.html"><i class="fa fa-check"></i>Part III: R fundamentals</a></li>
<li class="chapter" data-level="3" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html"><i class="fa fa-check"></i><b>3</b> Data – Reading, Wrangling, and Writing</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#learning-from-this-chapter"><i class="fa fa-check"></i><b>3.1</b> Learning from this chapter</a></li>
<li class="chapter" data-level="3.2" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#data-working-in-r"><i class="fa fa-check"></i><b>3.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#importing-data"><i class="fa fa-check"></i><b>3.2.1</b> Importing data</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#data-wrangling"><i class="fa fa-check"></i><b>3.3</b> Data wrangling</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#reshaping-data-wide-to-long"><i class="fa fa-check"></i><b>3.3.1</b> Reshaping data – Wide to long</a></li>
<li class="chapter" data-level="3.3.2" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#reshaping-data-transpose-turning-the-columns-into-rows"><i class="fa fa-check"></i><b>3.3.2</b> Reshaping data – Transpose (turning the columns into rows)</a></li>
<li class="chapter" data-level="3.3.3" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#combining-data"><i class="fa fa-check"></i><b>3.3.3</b> Combining data</a></li>
<li class="chapter" data-level="3.3.4" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#subsetting-data"><i class="fa fa-check"></i><b>3.3.4</b> Subsetting data</a></li>
<li class="chapter" data-level="3.3.5" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#wrangling-columns"><i class="fa fa-check"></i><b>3.3.5</b> Wrangling columns</a></li>
<li class="chapter" data-level="3.3.6" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#missing-data"><i class="fa fa-check"></i><b>3.3.6</b> Missing data</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#saving-data"><i class="fa fa-check"></i><b>3.4</b> Saving data</a></li>
<li class="chapter" data-level="3.5" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#exercises"><i class="fa fa-check"></i><b>3.5</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="plotting-models.html"><a href="plotting-models.html"><i class="fa fa-check"></i><b>4</b> Plotting Models</a>
<ul>
<li class="chapter" data-level="4.1" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plots-show-the-model-and-the-data"><i class="fa fa-check"></i><b>4.1</b> Pretty good plots show the model and the data</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plot-component-1-modeled-effects-plot"><i class="fa fa-check"></i><b>4.1.1</b> Pretty good plot component 1: Modeled effects plot</a></li>
<li class="chapter" data-level="4.1.2" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plot-component-2-modeled-mean-and-ci-plot"><i class="fa fa-check"></i><b>4.1.2</b> Pretty good plot component 2: Modeled mean and CI plot</a></li>
<li class="chapter" data-level="4.1.3" data-path="plotting-models.html"><a href="plotting-models.html#combining-effects-and-modeled-mean-and-ci-plots-an-effects-and-response-plot."><i class="fa fa-check"></i><b>4.1.3</b> Combining Effects and Modeled mean and CI plots – an Effects and response plot.</a></li>
<li class="chapter" data-level="4.1.4" data-path="plotting-models.html"><a href="plotting-models.html#some-comments-on-plot-components"><i class="fa fa-check"></i><b>4.1.4</b> Some comments on plot components</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="plotting-models.html"><a href="plotting-models.html#working-in-r"><i class="fa fa-check"></i><b>4.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="plotting-models.html"><a href="plotting-models.html#source-data"><i class="fa fa-check"></i><b>4.2.1</b> Source data</a></li>
<li class="chapter" data-level="4.2.2" data-path="plotting-models.html"><a href="plotting-models.html#how-to-plot-the-model"><i class="fa fa-check"></i><b>4.2.2</b> How to plot the model</a></li>
<li class="chapter" data-level="4.2.3" data-path="plotting-models.html"><a href="plotting-models.html#be-sure-ggplot_the_model-is-in-your-r-folder"><i class="fa fa-check"></i><b>4.2.3</b> Be sure ggplot_the_model is in your R folder</a></li>
<li class="chapter" data-level="4.2.4" data-path="plotting-models.html"><a href="plotting-models.html#how-to-use-the-plot-the-model-functions"><i class="fa fa-check"></i><b>4.2.4</b> How to use the Plot the Model functions</a></li>
<li class="chapter" data-level="4.2.5" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-a-response-plot-using-ggpubr"><i class="fa fa-check"></i><b>4.2.5</b> How to generate a Response Plot using ggpubr</a></li>
<li class="chapter" data-level="4.2.6" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-a-response-plot-with-a-grid-of-treatments-using-ggplot2"><i class="fa fa-check"></i><b>4.2.6</b> How to generate a Response Plot with a grid of treatments using ggplot2</a></li>
<li class="chapter" data-level="4.2.7" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-an-effects-plot"><i class="fa fa-check"></i><b>4.2.7</b> How to generate an Effects Plot</a></li>
<li class="chapter" data-level="4.2.8" data-path="plotting-models.html"><a href="plotting-models.html#how-to-combine-the-response-and-effects-plots"><i class="fa fa-check"></i><b>4.2.8</b> How to combine the response and effects plots</a></li>
<li class="chapter" data-level="4.2.9" data-path="plotting-models.html"><a href="plotting-models.html#how-to-add-the-interaction-effect-to-response-and-effects-plots"><i class="fa fa-check"></i><b>4.2.9</b> How to add the interaction effect to response and effects plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-iv-some-fundamentals-of-statistical-modeling.html"><a href="part-iv-some-fundamentals-of-statistical-modeling.html"><i class="fa fa-check"></i>Part IV: Some Fundamentals of Statistical Modeling</a></li>
<li class="chapter" data-level="5" data-path="uncertainty.html"><a href="uncertainty.html"><i class="fa fa-check"></i><b>5</b> Variability and Uncertainty (Standard Deviations, Standard Errors, Confidence Intervals)</a>
<ul>
<li class="chapter" data-level="5.1" data-path="uncertainty.html"><a href="uncertainty.html#the-sample-standard-deviation-vs.-the-standard-error-of-the-mean"><i class="fa fa-check"></i><b>5.1</b> The sample standard deviation vs. the standard error of the mean</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="uncertainty.html"><a href="uncertainty.html#sample-standard-deviation"><i class="fa fa-check"></i><b>5.1.1</b> Sample standard deviation</a></li>
<li class="chapter" data-level="5.1.2" data-path="uncertainty.html"><a href="uncertainty.html#standard-error-of-the-mean"><i class="fa fa-check"></i><b>5.1.2</b> Standard error of the mean</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="uncertainty.html"><a href="uncertainty.html#using-google-sheets-to-generate-fake-data-to-explore-the-standard-error"><i class="fa fa-check"></i><b>5.2</b> Using Google Sheets to generate fake data to explore the standard error</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="uncertainty.html"><a href="uncertainty.html#steps"><i class="fa fa-check"></i><b>5.2.1</b> Steps</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="uncertainty.html"><a href="uncertainty.html#using-r-to-generate-fake-data-to-explore-the-standard-error"><i class="fa fa-check"></i><b>5.3</b> Using R to generate fake data to explore the standard error</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="uncertainty.html"><a href="uncertainty.html#part-i"><i class="fa fa-check"></i><b>5.3.1</b> part I</a></li>
<li class="chapter" data-level="5.3.2" data-path="uncertainty.html"><a href="uncertainty.html#part-ii---means"><i class="fa fa-check"></i><b>5.3.2</b> part II - means</a></li>
<li class="chapter" data-level="5.3.3" data-path="uncertainty.html"><a href="uncertainty.html#part-iii---how-do-sd-and-se-change-as-sample-size-n-increases"><i class="fa fa-check"></i><b>5.3.3</b> part III - how do SD and SE change as sample size (n) increases?</a></li>
<li class="chapter" data-level="5.3.4" data-path="uncertainty.html"><a href="uncertainty.html#part-iv-generating-fake-data-with-for-loops"><i class="fa fa-check"></i><b>5.3.4</b> Part IV – Generating fake data with for-loops</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="uncertainty.html"><a href="uncertainty.html#bootstrap"><i class="fa fa-check"></i><b>5.4</b> Bootstrapped standard errors</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="uncertainty.html"><a href="uncertainty.html#an-example-of-bootstrapped-standard-errors-using-vole-data"><i class="fa fa-check"></i><b>5.4.1</b> An example of bootstrapped standard errors using vole data</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="uncertainty.html"><a href="uncertainty.html#confidence-interval"><i class="fa fa-check"></i><b>5.5</b> Confidence Interval</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="uncertainty.html"><a href="uncertainty.html#interpretation-of-a-confidence-interval"><i class="fa fa-check"></i><b>5.5.1</b> Interpretation of a confidence interval</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="p-values.html"><a href="p-values.html"><i class="fa fa-check"></i><b>6</b> P-values</a>
<ul>
<li class="chapter" data-level="6.1" data-path="p-values.html"><a href="p-values.html#a-p-value-is-the-probability-of-sampling-a-value-as-or-more-extreme-than-the-test-statistic-if-sampling-from-a-null-distribution"><i class="fa fa-check"></i><b>6.1</b> A <em>p</em>-value is the probability of sampling a value as or more extreme than the test statistic if sampling from a null distribution</a></li>
<li class="chapter" data-level="6.2" data-path="p-values.html"><a href="p-values.html#pump-your-intuition-creating-a-null-distribution"><i class="fa fa-check"></i><b>6.2</b> Pump your intuition – Creating a null distribution</a></li>
<li class="chapter" data-level="6.3" data-path="p-values.html"><a href="p-values.html#a-null-distribution-of-t-values-the-t-distribution"><i class="fa fa-check"></i><b>6.3</b> A null distribution of <em>t</em>-values – the <em>t</em> distribution</a></li>
<li class="chapter" data-level="6.4" data-path="p-values.html"><a href="p-values.html#p-values-from-the-perspective-of-permutation"><i class="fa fa-check"></i><b>6.4</b> P-values from the perspective of permutation</a></li>
<li class="chapter" data-level="6.5" data-path="p-values.html"><a href="p-values.html#parametric-vs.-non-parametric-statistics"><i class="fa fa-check"></i><b>6.5</b> Parametric vs. non-parametric statistics</a></li>
<li class="chapter" data-level="6.6" data-path="p-values.html"><a href="p-values.html#frequentist-probability-and-the-interpretation-of-p-values"><i class="fa fa-check"></i><b>6.6</b> frequentist probability and the interpretation of p-values</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="p-values.html"><a href="p-values.html#background"><i class="fa fa-check"></i><b>6.6.1</b> Background</a></li>
<li class="chapter" data-level="6.6.2" data-path="p-values.html"><a href="p-values.html#this-book-covers-frequentist-approaches-to-statistical-modeling-and-when-a-probability-arises-such-as-the-p-value-of-a-test-statistic-this-will-be-a-frequentist-probability."><i class="fa fa-check"></i><b>6.6.2</b> This book covers frequentist approaches to statistical modeling and when a probability arises, such as the <em>p</em>-value of a test statistic, this will be a frequentist probability.</a></li>
<li class="chapter" data-level="6.6.3" data-path="p-values.html"><a href="p-values.html#two-interpretations-of-the-p-value"><i class="fa fa-check"></i><b>6.6.3</b> Two interpretations of the <em>p</em>-value</a></li>
<li class="chapter" data-level="6.6.4" data-path="p-values.html"><a href="p-values.html#nhst"><i class="fa fa-check"></i><b>6.6.4</b> NHST</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="p-values.html"><a href="p-values.html#some-major-misconceptions-of-the-p-value"><i class="fa fa-check"></i><b>6.7</b> Some major misconceptions of the <em>p</em>-value</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="p-values.html"><a href="p-values.html#misconception-p-is-the-probability-that-the-null-is-true-and-1-p-is-probability-that-the-alternative-is-true"><i class="fa fa-check"></i><b>6.7.1</b> Misconception: <em>p</em> is the probability that the null is true <em>and</em> <span class="math inline">\(1-p\)</span> is probability that the alternative is true</a></li>
<li class="chapter" data-level="6.7.2" data-path="p-values.html"><a href="p-values.html#misconception-a-p-value-is-repeatable"><i class="fa fa-check"></i><b>6.7.2</b> Misconception: a <em>p</em>-value is repeatable</a></li>
<li class="chapter" data-level="6.7.3" data-path="p-values.html"><a href="p-values.html#misconception-0.05-is-the-lifetime-rate-of-false-discoveries"><i class="fa fa-check"></i><b>6.7.3</b> Misconception: 0.05 is the lifetime rate of false discoveries</a></li>
<li class="chapter" data-level="6.7.4" data-path="p-values.html"><a href="p-values.html#misconception-a-low-p-value-indicates-an-important-effect"><i class="fa fa-check"></i><b>6.7.4</b> Misconception: a low <em>p</em>-value indicates an important effect</a></li>
<li class="chapter" data-level="6.7.5" data-path="p-values.html"><a href="p-values.html#misconception-a-low-p-value-indicates-high-model-fit-or-high-predictive-capacity"><i class="fa fa-check"></i><b>6.7.5</b> Misconception: a low <em>p</em>-value indicates high model fit or high predictive capacity</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="p-values.html"><a href="p-values.html#what-the-p-value-does-not-mean"><i class="fa fa-check"></i><b>6.8</b> What the <em>p</em>-value does not mean</a></li>
<li class="chapter" data-level="6.9" data-path="p-values.html"><a href="p-values.html#recommendations"><i class="fa fa-check"></i><b>6.9</b> Recommendations</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="p-values.html"><a href="p-values.html#primary-sources-for-recommendations"><i class="fa fa-check"></i><b>6.9.1</b> Primary sources for recommendations</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="p-values.html"><a href="p-values.html#problems"><i class="fa fa-check"></i><b>6.10</b> Problems</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="errors-in-inference.html"><a href="errors-in-inference.html"><i class="fa fa-check"></i><b>7</b> Errors in inference</a>
<ul>
<li class="chapter" data-level="7.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#classical-nhst-concepts-of-wrong"><i class="fa fa-check"></i><b>7.1</b> Classical NHST concepts of wrong</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-i-error"><i class="fa fa-check"></i><b>7.1.1</b> Type I error</a></li>
<li class="chapter" data-level="7.1.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#power"><i class="fa fa-check"></i><b>7.1.2</b> Power</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#a-non-neyman-pearson-concept-of-power"><i class="fa fa-check"></i><b>7.2</b> A non-Neyman-Pearson concept of power</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#estimation-error"><i class="fa fa-check"></i><b>7.2.1</b> Estimation error</a></li>
<li class="chapter" data-level="7.2.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#coverage"><i class="fa fa-check"></i><b>7.2.2</b> Coverage</a></li>
<li class="chapter" data-level="7.2.3" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-s-error"><i class="fa fa-check"></i><b>7.2.3</b> Type S error</a></li>
<li class="chapter" data-level="7.2.4" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-m-error"><i class="fa fa-check"></i><b>7.2.4</b> Type M error</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-v-introduction-to-linear-models.html"><a href="part-v-introduction-to-linear-models.html"><i class="fa fa-check"></i>Part V: Introduction to Linear Models</a></li>
<li class="chapter" data-level="8" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html"><i class="fa fa-check"></i><b>8</b> An introduction to linear models</a>
<ul>
<li class="chapter" data-level="8.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#two-specifications-of-a-linear-model"><i class="fa fa-check"></i><b>8.1</b> Two specifications of a linear model</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#the-error-draw-specification"><i class="fa fa-check"></i><b>8.1.1</b> The “error draw” specification</a></li>
<li class="chapter" data-level="8.1.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#the-conditional-draw-specification"><i class="fa fa-check"></i><b>8.1.2</b> The “conditional draw” specification</a></li>
<li class="chapter" data-level="8.1.3" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#comparing-the-error-draw-and-conditional-draw-ways-of-specifying-the-linear-model"><i class="fa fa-check"></i><b>8.1.3</b> Comparing the error-draw and conditional-draw ways of specifying the linear model</a></li>
<li class="chapter" data-level="8.1.4" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#anova-notation-of-a-linear-model"><i class="fa fa-check"></i><b>8.1.4</b> ANOVA notation of a linear model</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#a-linear-model-can-be-fit-to-data-with-continuous-discrete-or-categorical-x-variables"><i class="fa fa-check"></i><b>8.2</b> A linear model can be fit to data with continuous, discrete, or categorical <span class="math inline">\(X\)</span> variables</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#fitting-linear-models-to-experimental-data-in-which-the-x-variable-is-continuous-or-discrete"><i class="fa fa-check"></i><b>8.2.1</b> Fitting linear models to experimental data in which the <span class="math inline">\(X\)</span> variable is continuous or discrete</a></li>
<li class="chapter" data-level="8.2.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#slope"><i class="fa fa-check"></i><b>8.2.2</b> Fitting linear models to experimental data in which the <span class="math inline">\(X\)</span> variable is categorical</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#statistical-models-are-used-for-prediction-explanation-and-description"><i class="fa fa-check"></i><b>8.3</b> Statistical models are used for prediction, explanation, and description</a></li>
<li class="chapter" data-level="8.4" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#what-do-we-call-the-x-and-y-variables"><i class="fa fa-check"></i><b>8.4</b> What do we call the <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span> variables?</a></li>
<li class="chapter" data-level="8.5" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#modeling-strategy"><i class="fa fa-check"></i><b>8.5</b> Modeling strategy</a></li>
<li class="chapter" data-level="8.6" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#predictions-from-the-model"><i class="fa fa-check"></i><b>8.6</b> Predictions from the model</a></li>
<li class="chapter" data-level="8.7" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#inference-from-the-model"><i class="fa fa-check"></i><b>8.7</b> Inference from the model</a>
<ul>
<li class="chapter" data-level="8.7.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#assumptions-for-inference-with-a-statistical-model"><i class="fa fa-check"></i><b>8.7.1</b> Assumptions for inference with a statistical model</a></li>
<li class="chapter" data-level="8.7.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#specific-assumptions-for-inference-with-a-linear-model"><i class="fa fa-check"></i><b>8.7.2</b> Specific assumptions for inference with a linear model</a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#linear-modelregression-model-orstatistical-model"><i class="fa fa-check"></i><b>8.8</b> “linear model,”regression model“, or”statistical model"?</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="regression.html"><a href="regression.html"><i class="fa fa-check"></i><b>9</b> Linear models with a single, continuous <em>X</em> (“regression”)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="regression.html"><a href="regression.html#a-linear-model-with-a-single-continuous-x-is-classical-regression"><i class="fa fa-check"></i><b>9.1</b> A linear model with a single, continuous <em>X</em> is classical “regression”</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="regression.html"><a href="regression.html#analysis-of-green-down-data"><i class="fa fa-check"></i><b>9.1.1</b> Analysis of “green-down” data</a></li>
<li class="chapter" data-level="9.1.2" data-path="regression.html"><a href="regression.html#learning-from-the-green-down-example"><i class="fa fa-check"></i><b>9.1.2</b> Learning from the green-down example</a></li>
<li class="chapter" data-level="9.1.3" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-explanation-causal-models"><i class="fa fa-check"></i><b>9.1.3</b> Using a regression model for “explanation” – causal models</a></li>
<li class="chapter" data-level="9.1.4" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-prediction-prediction-models"><i class="fa fa-check"></i><b>9.1.4</b> Using a regression model for prediction – prediction models</a></li>
<li class="chapter" data-level="9.1.5" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-creating-a-new-response-variable-comparing-slopes-of-longitudinal-data"><i class="fa fa-check"></i><b>9.1.5</b> Using a regression model for creating a new response variable – comparing slopes of longitudinal data</a></li>
<li class="chapter" data-level="9.1.6" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-for-calibration"><i class="fa fa-check"></i><b>9.1.6</b> Using a regression model for for calibration</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="regression.html"><a href="regression.html#working-in-r-1"><i class="fa fa-check"></i><b>9.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="plotting-models.html"><a href="plotting-models.html#fit-the-model"><i class="fa fa-check"></i><b>9.2.1</b> Fitting the linear model</a></li>
<li class="chapter" data-level="9.2.2" data-path="regression.html"><a href="regression.html#getting-to-know-the-linear-model-the-summary-function"><i class="fa fa-check"></i><b>9.2.2</b> Getting to know the linear model: the <code>summary</code> function</a></li>
<li class="chapter" data-level="9.2.3" data-path="regression.html"><a href="regression.html#inference-the-coefficient-table"><i class="fa fa-check"></i><b>9.2.3</b> Inference – the coefficient table</a></li>
<li class="chapter" data-level="9.2.4" data-path="regression.html"><a href="regression.html#how-good-is-our-model-model-checking"><i class="fa fa-check"></i><b>9.2.4</b> How good is our model? – Model checking</a></li>
<li class="chapter" data-level="9.2.5" data-path="regression.html"><a href="regression.html#plotting-models-with-continuous-x"><i class="fa fa-check"></i><b>9.2.5</b> Plotting models with continuous <em>X</em></a></li>
<li class="chapter" data-level="9.2.6" data-path="regression.html"><a href="regression.html#creating-a-table-of-predicted-values-and-95-prediction-intervals"><i class="fa fa-check"></i><b>9.2.6</b> Creating a table of predicted values and 95% prediction intervals</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="regression.html"><a href="regression.html#hidden-code"><i class="fa fa-check"></i><b>9.3</b> Hidden code</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="regression.html"><a href="regression.html#import-and-plot-of-fig2c-ecosystem-warming-experimental-data"><i class="fa fa-check"></i><b>9.3.1</b> Import and plot of fig2c (ecosystem warming experimental) data</a></li>
<li class="chapter" data-level="9.3.2" data-path="regression.html"><a href="regression.html#import-and-plot-efig_3d-ecosysem-warming-observational-data"><i class="fa fa-check"></i><b>9.3.2</b> Import and plot efig_3d (Ecosysem warming observational) data</a></li>
<li class="chapter" data-level="9.3.3" data-path="regression.html"><a href="regression.html#import-and-plot-of-fig1f-methionine-restriction-data"><i class="fa fa-check"></i><b>9.3.3</b> Import and plot of fig1f (methionine restriction) data</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="regression.html"><a href="regression.html#try-it"><i class="fa fa-check"></i><b>9.4</b> Try it</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="regression.html"><a href="regression.html#a-prediction-model-from-the-literature"><i class="fa fa-check"></i><b>9.4.1</b> A prediction model from the literature</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="regression.html"><a href="regression.html#intuition-pumps"><i class="fa fa-check"></i><b>9.5</b> Intuition pumps</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="regression.html"><a href="regression.html#correlation-and-r2"><i class="fa fa-check"></i><b>9.5.1</b> Correlation and $R^2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="oneway.html"><a href="oneway.html"><i class="fa fa-check"></i><b>10</b> Linear models with a single, categorical <em>X</em> (“t-tests” and “ANOVA”)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="oneway.html"><a href="oneway.html#oneway-data"><i class="fa fa-check"></i><b>10.1</b> A linear model with a single, categorical <em>X</em> variable estimates the effects of the levels of <em>X</em> on the response</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="oneway.html"><a href="oneway.html#example-1-fig3d-two-treatment-levels-groups"><i class="fa fa-check"></i><b>10.1.1</b> Example 1 (fig3d) – two treatment levels (“groups”)</a></li>
<li class="chapter" data-level="10.1.2" data-path="oneway.html"><a href="oneway.html#understanding-the-analysis-with-two-treatment-levels"><i class="fa fa-check"></i><b>10.1.2</b> Understanding the analysis with two treatment levels</a></li>
<li class="chapter" data-level="10.1.3" data-path="oneway.html"><a href="oneway.html#example-2-three-treatment-levels-groups"><i class="fa fa-check"></i><b>10.1.3</b> Example 2 – three treatment levels (“groups”)</a></li>
<li class="chapter" data-level="10.1.4" data-path="oneway.html"><a href="oneway.html#understanding-the-analysis-with-three-or-more-treatment-levels"><i class="fa fa-check"></i><b>10.1.4</b> Understanding the analysis with three (or more) treatment levels</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="oneway.html"><a href="oneway.html#working-in-r-2"><i class="fa fa-check"></i><b>10.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="oneway.html"><a href="oneway.html#fit-the-model-1"><i class="fa fa-check"></i><b>10.2.1</b> Fit the model</a></li>
<li class="chapter" data-level="10.2.2" data-path="oneway.html"><a href="oneway.html#controlling-the-output-in-tables-using-the-coefficient-table-as-an-example"><i class="fa fa-check"></i><b>10.2.2</b> Controlling the output in tables using the coefficient table as an example</a></li>
<li class="chapter" data-level="10.2.3" data-path="oneway.html"><a href="oneway.html#using-the-emmeans-function"><i class="fa fa-check"></i><b>10.2.3</b> Using the emmeans function</a></li>
<li class="chapter" data-level="10.2.4" data-path="oneway.html"><a href="oneway.html#using-the-contrast-function"><i class="fa fa-check"></i><b>10.2.4</b> Using the contrast function</a></li>
<li class="chapter" data-level="10.2.5" data-path="oneway.html"><a href="oneway.html#how-to-generate-anova-tables"><i class="fa fa-check"></i><b>10.2.5</b> How to generate ANOVA tables</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="oneway.html"><a href="oneway.html#hidden-code-1"><i class="fa fa-check"></i><b>10.3</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="oneway.html"><a href="oneway.html#importing-and-wrangling-the-fig_3d-data-for-example-1"><i class="fa fa-check"></i><b>10.3.1</b> Importing and wrangling the fig_3d data for example 1</a></li>
<li class="chapter" data-level="10.3.2" data-path="oneway.html"><a href="oneway.html#importing-and-wrangling-the-fig2a-data-for-example-2"><i class="fa fa-check"></i><b>10.3.2</b> Importing and wrangling the fig2a data for example 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="model-checking.html"><a href="model-checking.html"><i class="fa fa-check"></i><b>11</b> Model Checking</a>
<ul>
<li class="chapter" data-level="11.1" data-path="model-checking.html"><a href="model-checking.html#all-statistical-analyses-should-be-followed-by-model-checking"><i class="fa fa-check"></i><b>11.1</b> All statistical analyses should be followed by model checking</a></li>
<li class="chapter" data-level="11.2" data-path="model-checking.html"><a href="model-checking.html#linear-model-assumptions"><i class="fa fa-check"></i><b>11.2</b> Linear model assumptions</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="model-checking.html"><a href="model-checking.html#a-bit-about-iid"><i class="fa fa-check"></i><b>11.2.1</b> A bit about IID</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="model-checking.html"><a href="model-checking.html#diagnostic-plots-use-the-residuals-from-the-model-fit"><i class="fa fa-check"></i><b>11.3</b> Diagnostic plots use the residuals from the model fit</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="model-checking.html"><a href="model-checking.html#residuals"><i class="fa fa-check"></i><b>11.3.1</b> Residuals</a></li>
<li class="chapter" data-level="11.3.2" data-path="model-checking.html"><a href="model-checking.html#normal-qq"><i class="fa fa-check"></i><b>11.3.2</b> A Normal Q-Q plot is used to check for characteristic departures from Normality</a></li>
<li class="chapter" data-level="11.3.3" data-path="model-checking.html"><a href="model-checking.html#mapping-qq-plot-departures-from-normality"><i class="fa fa-check"></i><b>11.3.3</b> Mapping QQ-plot departures from Normality</a></li>
<li class="chapter" data-level="11.3.4" data-path="model-checking.html"><a href="model-checking.html#model-checking-homoskedasticity"><i class="fa fa-check"></i><b>11.3.4</b> Model checking homoskedasticity</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="model-checking.html"><a href="model-checking.html#using-r"><i class="fa fa-check"></i><b>11.4</b> Using R</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="model-checking.html"><a href="model-checking.html#normal-q-q-plots"><i class="fa fa-check"></i><b>11.4.1</b> Normal Q-Q plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="violations.html"><a href="violations.html"><i class="fa fa-check"></i><b>12</b> Violations of independence, homogeneity, or Normality</a>
<ul>
<li class="chapter" data-level="12.1" data-path="violations.html"><a href="violations.html#oneway-paired-t"><i class="fa fa-check"></i><b>12.1</b> Lack of independence</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="violations.html"><a href="violations.html#example-1-exp1b-a-paired-t-test-is-a-special-case-of-a-linear-mixed-model"><i class="fa fa-check"></i><b>12.1.1</b> Example 1 (exp1b) – a paired t-test is a special case of a linear mixed model</a></li>
<li class="chapter" data-level="12.1.2" data-path="violations.html"><a href="violations.html#violations-rmanova"><i class="fa fa-check"></i><b>12.1.2</b> Example 2 (diHOME exp2a) – A repeated measures ANOVA is a special case of a linear mixed model</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="violations.html"><a href="violations.html#oneway-welch"><i class="fa fa-check"></i><b>12.2</b> Heterogeneity of variances</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="violations.html"><a href="violations.html#when-groups-of-the-focal-test-have-variance"><i class="fa fa-check"></i><b>12.2.1</b> When groups of the focal test have &gt;&gt; variance</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="violations.html"><a href="violations.html#the-conditional-response-isnt-normal"><i class="fa fa-check"></i><b>12.3</b> The conditional response isn’t Normal</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="violations.html"><a href="violations.html#violations-6f"><i class="fa fa-check"></i><b>12.3.1</b> Example 1 (fig6f) – Linear models for non-normal count data</a></li>
<li class="chapter" data-level="12.3.2" data-path="violations.html"><a href="violations.html#my-data-arent-normal-what-is-the-best-practice"><i class="fa fa-check"></i><b>12.3.2</b> My data aren’t normal, what is the best practice?</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="violations.html"><a href="violations.html#hidden-code-2"><i class="fa fa-check"></i><b>12.4</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-exp1b-data"><i class="fa fa-check"></i><b>12.4.1</b> Importing and wrangling the exp1b data</a></li>
<li class="chapter" data-level="12.4.2" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-exp2a-data"><i class="fa fa-check"></i><b>12.4.2</b> Importing and wrangling the exp2a data</a></li>
<li class="chapter" data-level="12.4.3" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-fig6f-data"><i class="fa fa-check"></i><b>12.4.3</b> Importing and wrangling the fig6f data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="issues.html"><a href="issues.html"><i class="fa fa-check"></i><b>13</b> Issues in inference</a>
<ul>
<li class="chapter" data-level="13.1" data-path="issues.html"><a href="issues.html#replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means"><i class="fa fa-check"></i><b>13.1</b> Replicated experiments – include <span class="math inline">\(\texttt{Experiment}\)</span> as a random factor (better than one-way ANOVA of means)</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="issues.html"><a href="issues.html#issues-exp4d"><i class="fa fa-check"></i><b>13.1.1</b> Multiple experiments Example 1 (wound healing Exp4d)</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="issues.html"><a href="issues.html#issues-pre-post"><i class="fa fa-check"></i><b>13.2</b> Comparing change from baseline (pre-post)</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="issues.html"><a href="issues.html#pre-post-example-1-dpp4-fig4c"><i class="fa fa-check"></i><b>13.2.1</b> Pre-post example 1 (DPP4 fig4c)</a></li>
<li class="chapter" data-level="13.2.2" data-path="issues.html"><a href="issues.html#what-if-the-data-in-example-1-were-from-from-an-experiment-where-the-treatment-was-applied-prior-to-the-baseline-measure"><i class="fa fa-check"></i><b>13.2.2</b> What if the data in example 1 were from from an experiment where the treatment was applied prior to the baseline measure?</a></li>
<li class="chapter" data-level="13.2.3" data-path="issues.html"><a href="issues.html#pre-post-example-2-xx-males-fig1c"><i class="fa fa-check"></i><b>13.2.3</b> Pre-post example 2 (XX males fig1c)</a></li>
<li class="chapter" data-level="13.2.4" data-path="issues.html"><a href="issues.html#issues-regression-to-mean"><i class="fa fa-check"></i><b>13.2.4</b> Regression to the mean</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="issues.html"><a href="issues.html#longitudinal-designs-with-more-than-one-post-baseline-measure"><i class="fa fa-check"></i><b>13.3</b> Longitudinal designs with more than one-post baseline measure</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="issues.html"><a href="issues.html#issues-auc"><i class="fa fa-check"></i><b>13.3.1</b> Area under the curve (AUC)</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="issues.html"><a href="issues.html#comparing-ratios"><i class="fa fa-check"></i><b>13.4</b> Comparing ratios</a>
<ul>
<li class="chapter" data-level="13.4.1" data-path="issues.html"><a href="issues.html#the-ratio-is-a-density"><i class="fa fa-check"></i><b>13.4.1</b> The ratio is a density</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="issues.html"><a href="issues.html#dont-do-this-stuff"><i class="fa fa-check"></i><b>13.5</b> Don’t do this stuff</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="issues.html"><a href="issues.html#normalize-the-response-so-that-all-control-values-are-equal-to-1."><i class="fa fa-check"></i><b>13.5.1</b> Normalize the response so that all control values are equal to 1.</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="issues.html"><a href="issues.html#researcher-degrees-of-freedom"><i class="fa fa-check"></i><b>13.6</b> Researcher degrees of freedom</a></li>
<li class="chapter" data-level="13.7" data-path="issues.html"><a href="issues.html#hidden-code-3"><i class="fa fa-check"></i><b>13.7</b> Hidden code</a>
<ul>
<li class="chapter" data-level="13.7.1" data-path="issues.html"><a href="issues.html#import-exp4d-vimentin-cell-count-data-replicate-experiments-example"><i class="fa fa-check"></i><b>13.7.1</b> Import exp4d vimentin cell count data (replicate experiments example)</a></li>
<li class="chapter" data-level="13.7.2" data-path="issues.html"><a href="issues.html#import-fig4c-data"><i class="fa fa-check"></i><b>13.7.2</b> Import Fig4c data</a></li>
<li class="chapter" data-level="13.7.3" data-path="issues.html"><a href="issues.html#xx-males-fig1c"><i class="fa fa-check"></i><b>13.7.3</b> XX males fig1c</a></li>
<li class="chapter" data-level="13.7.4" data-path="issues.html"><a href="issues.html#generation-of-fake-data-to-illustrate-regression-to-the-mean"><i class="fa fa-check"></i><b>13.7.4</b> Generation of fake data to illustrate regression to the mean</a></li>
<li class="chapter" data-level="13.7.5" data-path="issues.html"><a href="issues.html#import-fig3f"><i class="fa fa-check"></i><b>13.7.5</b> Import fig3f</a></li>
</ul></li>
</ul></li>
<li><a href="part-vi-more-than-one-x-multivariable-models.html#part-vi-more-than-one-x-multivariable-models">Part VI: More than one <span class="math inline">\(X\)</span> – Multivariable Models</a></li>
<li class="chapter" data-level="14" data-path="covariates.html"><a href="covariates.html"><i class="fa fa-check"></i><b>14</b> Linear models with added covariates (“ANCOVA”)</a>
<ul>
<li class="chapter" data-level="14.1" data-path="covariates.html"><a href="covariates.html#adding-covariates-can-increases-the-precision-of-the-effect-of-interest"><i class="fa fa-check"></i><b>14.1</b> Adding covariates can increases the precision of the effect of interest</a></li>
<li class="chapter" data-level="14.2" data-path="covariates.html"><a href="covariates.html#understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data"><i class="fa fa-check"></i><b>14.2</b> Understanding a linear model with an added covariate – heart necrosis data</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="covariates.html"><a href="covariates.html#fit-the-model-5"><i class="fa fa-check"></i><b>14.2.1</b> Fit the model</a></li>
<li class="chapter" data-level="14.2.2" data-path="covariates.html"><a href="covariates.html#plot-the-model-1"><i class="fa fa-check"></i><b>14.2.2</b> Plot the model</a></li>
<li class="chapter" data-level="14.2.3" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-model-coefficients"><i class="fa fa-check"></i><b>14.2.3</b> Interpretation of the model coefficients</a></li>
<li class="chapter" data-level="14.2.4" data-path="covariates.html"><a href="covariates.html#everything-adds-up"><i class="fa fa-check"></i><b>14.2.4</b> Everything adds up</a></li>
<li class="chapter" data-level="14.2.5" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-estimated-marginal-means"><i class="fa fa-check"></i><b>14.2.5</b> Interpretation of the estimated marginal means</a></li>
<li class="chapter" data-level="14.2.6" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-contrasts"><i class="fa fa-check"></i><b>14.2.6</b> Interpretation of the contrasts</a></li>
<li class="chapter" data-level="14.2.7" data-path="covariates.html"><a href="covariates.html#adding-the-covariate-improves-inference"><i class="fa fa-check"></i><b>14.2.7</b> Adding the covariate improves inference</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="covariates.html"><a href="covariates.html#understanding-interaction-effects-with-covariates"><i class="fa fa-check"></i><b>14.3</b> Understanding interaction effects with covariates</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="covariates.html"><a href="covariates.html#fit-the-model-6"><i class="fa fa-check"></i><b>14.3.1</b> Fit the model</a></li>
<li class="chapter" data-level="14.3.2" data-path="covariates.html"><a href="covariates.html#plot-the-model-with-interaction-effect"><i class="fa fa-check"></i><b>14.3.2</b> Plot the model with interaction effect</a></li>
<li class="chapter" data-level="14.3.3" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-model-coefficients-1"><i class="fa fa-check"></i><b>14.3.3</b> Interpretation of the model coefficients</a></li>
<li class="chapter" data-level="14.3.4" data-path="covariates.html"><a href="covariates.html#what-is-the-effect-of-a-treatment-if-interactions-are-modeled-it-depends."><i class="fa fa-check"></i><b>14.3.4</b> What is the effect of a treatment, if interactions are modeled? – it depends.</a></li>
<li class="chapter" data-level="14.3.5" data-path="covariates.html"><a href="covariates.html#which-model-do-we-use-mathcalm_1-or-mathcalm_2"><i class="fa fa-check"></i><b>14.3.5</b> Which model do we use, <span class="math inline">\(\mathcal{M}_1\)</span> or <span class="math inline">\(\mathcal{M}_2\)</span>?</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="covariates.html"><a href="covariates.html#understanding-ancova-tables"><i class="fa fa-check"></i><b>14.4</b> Understanding ANCOVA tables</a></li>
<li class="chapter" data-level="14.5" data-path="covariates.html"><a href="covariates.html#working-in-r-3"><i class="fa fa-check"></i><b>14.5</b> Working in R</a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="covariates.html"><a href="covariates.html#importing-the-heart-necrosis-data"><i class="fa fa-check"></i><b>14.5.1</b> Importing the heart necrosis data</a></li>
<li class="chapter" data-level="14.5.2" data-path="covariates.html"><a href="covariates.html#fitting-the-model"><i class="fa fa-check"></i><b>14.5.2</b> Fitting the model</a></li>
<li class="chapter" data-level="14.5.3" data-path="covariates.html"><a href="covariates.html#using-the-emmeans-function-1"><i class="fa fa-check"></i><b>14.5.3</b> Using the emmeans function</a></li>
<li class="chapter" data-level="14.5.4" data-path="covariates.html"><a href="covariates.html#ancova-tables"><i class="fa fa-check"></i><b>14.5.4</b> ANCOVA tables</a></li>
<li class="chapter" data-level="14.5.5" data-path="covariates.html"><a href="covariates.html#plotting-the-model"><i class="fa fa-check"></i><b>14.5.5</b> Plotting the model</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="covariates.html"><a href="covariates.html#best-practices"><i class="fa fa-check"></i><b>14.6</b> Best practices</a>
<ul>
<li class="chapter" data-level="14.6.1" data-path="covariates.html"><a href="covariates.html#do-not-use-a-ratio-of-partwhole-as-a-response-variable-instead-add-the-denominator-as-a-covariate"><i class="fa fa-check"></i><b>14.6.1</b> Do not use a ratio of part:whole as a response variable – instead add the denominator as a covariate</a></li>
<li class="chapter" data-level="14.6.2" data-path="covariates.html"><a href="covariates.html#do-not-use-change-from-baseline-as-a-response-variable-instead-add-the-baseline-measure-as-a-covariate"><i class="fa fa-check"></i><b>14.6.2</b> Do not use change from baseline as a response variable – instead add the baseline measure as a covariate</a></li>
<li class="chapter" data-level="14.6.3" data-path="covariates.html"><a href="covariates.html#do-not-test-for-balance-of-baseline-measures"><i class="fa fa-check"></i><b>14.6.3</b> Do not “test for balance” of baseline measures</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="covariates.html"><a href="covariates.html#best-practices-2-use-a-covariate-instead-of-normalizing-a-response"><i class="fa fa-check"></i><b>14.7</b> Best practices 2: Use a covariate instead of normalizing a response</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="factorial.html"><a href="factorial.html"><i class="fa fa-check"></i><b>15</b> Linear models with two categorical <span class="math inline">\(X\)</span> – Factorial linear models (“two-way ANOVA”)</a>
<ul>
<li class="chapter" data-level="15.1" data-path="factorial.html"><a href="factorial.html#a-linear-model-with-crossed-factors-estimates-interaction-effects"><i class="fa fa-check"></i><b>15.1</b> A linear model with crossed factors estimates interaction effects</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="factorial.html"><a href="factorial.html#an-interaction-is-a-difference-in-simple-effects"><i class="fa fa-check"></i><b>15.1.1</b> An interaction is a difference in simple effects</a></li>
<li class="chapter" data-level="15.1.2" data-path="factorial.html"><a href="factorial.html#a-linear-model-with-crossed-factors-includes-interaction-effects"><i class="fa fa-check"></i><b>15.1.2</b> A linear model with crossed factors includes interaction effects</a></li>
<li class="chapter" data-level="15.1.3" data-path="factorial.html"><a href="factorial.html#factorial-experiments-are-frequently-analyzed-as-flattened-linear-models-in-the-experimental-biology-literature"><i class="fa fa-check"></i><b>15.1.3</b> factorial experiments are frequently analyzed as flattened linear models in the experimental biology literature</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="factorial.html"><a href="factorial.html#example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data"><i class="fa fa-check"></i><b>15.2</b> Example 1 – Estimation of a treatment effect relative to a control effect (“Something different”) (Experiment 2j glucose uptake data)</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="factorial.html"><a href="factorial.html#twoway-understand1"><i class="fa fa-check"></i><b>15.2.1</b> Understand the experimental design</a></li>
<li class="chapter" data-level="15.2.2" data-path="factorial.html"><a href="factorial.html#fit-the-linear-model"><i class="fa fa-check"></i><b>15.2.2</b> Fit the linear model</a></li>
<li class="chapter" data-level="15.2.3" data-path="factorial.html"><a href="factorial.html#twoway-exp2j-inf"><i class="fa fa-check"></i><b>15.2.3</b> Inference</a></li>
<li class="chapter" data-level="15.2.4" data-path="factorial.html"><a href="factorial.html#plot-the-model-2"><i class="fa fa-check"></i><b>15.2.4</b> Plot the model</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-1"><i class="fa fa-check"></i><b>15.3</b> Understanding the linear model with crossed factors 1</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="factorial.html"><a href="factorial.html#twoway-what-coefs-are"><i class="fa fa-check"></i><b>15.3.1</b> What the coefficients are</a></li>
<li class="chapter" data-level="15.3.2" data-path="factorial.html"><a href="factorial.html#the-interaction-effect-is-something-different"><i class="fa fa-check"></i><b>15.3.2</b> The interaction effect is something different</a></li>
<li class="chapter" data-level="15.3.3" data-path="factorial.html"><a href="factorial.html#why-we-want-to-compare-the-treatment-effect-to-a-control-effect"><i class="fa fa-check"></i><b>15.3.3</b> Why we want to compare the treatment effect to a control effect</a></li>
<li class="chapter" data-level="15.3.4" data-path="factorial.html"><a href="factorial.html#the-order-of-the-factors-in-the-model-tells-the-same-story-differently"><i class="fa fa-check"></i><b>15.3.4</b> The order of the factors in the model tells the same story differently</a></li>
<li class="chapter" data-level="15.3.5" data-path="factorial.html"><a href="factorial.html#power-for-the-interaction-effect-is-less-than-that-for-simple-effects"><i class="fa fa-check"></i><b>15.3.5</b> Power for the interaction effect is less than that for simple effects</a></li>
<li class="chapter" data-level="15.3.6" data-path="factorial.html"><a href="factorial.html#planned-comparisons-vs.-post-hoc-tests"><i class="fa fa-check"></i><b>15.3.6</b> Planned comparisons vs. post-hoc tests</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="factorial.html"><a href="factorial.html#example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data"><i class="fa fa-check"></i><b>15.4</b> Example 2: Estimation of the effect of background condition on an effect (“it depends”) (Experiment 3e lesian area data)</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="factorial.html"><a href="factorial.html#understand-the-experimental-design"><i class="fa fa-check"></i><b>15.4.1</b> Understand the experimental design</a></li>
<li class="chapter" data-level="15.4.2" data-path="factorial.html"><a href="factorial.html#fit-the-linear-model-1"><i class="fa fa-check"></i><b>15.4.2</b> Fit the linear model</a></li>
<li class="chapter" data-level="15.4.3" data-path="factorial.html"><a href="factorial.html#check-the-model-1"><i class="fa fa-check"></i><b>15.4.3</b> Check the model</a></li>
<li class="chapter" data-level="15.4.4" data-path="factorial.html"><a href="factorial.html#inference-from-the-model-2"><i class="fa fa-check"></i><b>15.4.4</b> Inference from the model</a></li>
<li class="chapter" data-level="15.4.5" data-path="factorial.html"><a href="factorial.html#plot-the-model-3"><i class="fa fa-check"></i><b>15.4.5</b> Plot the model</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-2"><i class="fa fa-check"></i><b>15.5</b> Understanding the linear model with crossed factors 2</a>
<ul>
<li class="chapter" data-level="15.5.1" data-path="factorial.html"><a href="factorial.html#twoway-marginal-means"><i class="fa fa-check"></i><b>15.5.1</b> Conditional and marginal means</a></li>
<li class="chapter" data-level="15.5.2" data-path="factorial.html"><a href="factorial.html#simple-conditional-effects"><i class="fa fa-check"></i><b>15.5.2</b> Simple (conditional) effects</a></li>
<li class="chapter" data-level="15.5.3" data-path="factorial.html"><a href="factorial.html#twoway-marginal-effects"><i class="fa fa-check"></i><b>15.5.3</b> Marginal effects</a></li>
<li class="chapter" data-level="15.5.4" data-path="factorial.html"><a href="factorial.html#the-additive-model"><i class="fa fa-check"></i><b>15.5.4</b> The additive model</a></li>
<li class="chapter" data-level="15.5.5" data-path="factorial.html"><a href="factorial.html#twoway-reduce"><i class="fa fa-check"></i><b>15.5.5</b> Reduce models for the right reason</a></li>
<li class="chapter" data-level="15.5.6" data-path="factorial.html"><a href="factorial.html#the-marginal-means-of-an-additive-linear-model-with-two-factors-can-be-weird"><i class="fa fa-check"></i><b>15.5.6</b> The marginal means of an additive linear model with two factors can be weird</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="factorial.html"><a href="factorial.html#example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data"><i class="fa fa-check"></i><b>15.6</b> Example 3: Estimation of synergy (“More than the sum of the parts”) (Experiment 1c JA data)</a>
<ul>
<li class="chapter" data-level="15.6.1" data-path="factorial.html"><a href="factorial.html#examine-the-data"><i class="fa fa-check"></i><b>15.6.1</b> Examine the data</a></li>
<li class="chapter" data-level="15.6.2" data-path="factorial.html"><a href="factorial.html#fit-the-model-7"><i class="fa fa-check"></i><b>15.6.2</b> Fit the model</a></li>
<li class="chapter" data-level="15.6.3" data-path="factorial.html"><a href="factorial.html#model-check"><i class="fa fa-check"></i><b>15.6.3</b> Model check</a></li>
<li class="chapter" data-level="15.6.4" data-path="factorial.html"><a href="factorial.html#inference-from-the-model-3"><i class="fa fa-check"></i><b>15.6.4</b> Inference from the model</a></li>
<li class="chapter" data-level="15.6.5" data-path="factorial.html"><a href="factorial.html#plot-the-model-4"><i class="fa fa-check"></i><b>15.6.5</b> Plot the model</a></li>
<li class="chapter" data-level="15.6.6" data-path="factorial.html"><a href="factorial.html#alternative-plot"><i class="fa fa-check"></i><b>15.6.6</b> Alternative plot</a></li>
</ul></li>
<li class="chapter" data-level="15.7" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-3"><i class="fa fa-check"></i><b>15.7</b> Understanding the linear model with crossed factors 3</a>
<ul>
<li class="chapter" data-level="15.7.1" data-path="factorial.html"><a href="factorial.html#thinking-about-the-coefficients-of-the-linear-model"><i class="fa fa-check"></i><b>15.7.1</b> Thinking about the coefficients of the linear model</a></li>
</ul></li>
<li class="chapter" data-level="15.8" data-path="factorial.html"><a href="factorial.html#issues-in-inference"><i class="fa fa-check"></i><b>15.8</b> Issues in inference</a>
<ul>
<li class="chapter" data-level="15.8.1" data-path="factorial.html"><a href="factorial.html#for-pairwise-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model"><i class="fa fa-check"></i><b>15.8.1</b> For pairwise contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</a></li>
<li class="chapter" data-level="15.8.2" data-path="factorial.html"><a href="factorial.html#for-interaction-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model"><i class="fa fa-check"></i><b>15.8.2</b> For interaction contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</a></li>
<li class="chapter" data-level="15.8.3" data-path="factorial.html"><a href="factorial.html#twoway-multiple-tests"><i class="fa fa-check"></i><b>15.8.3</b> Adjusting <em>p</em>-values for multiple tests</a></li>
</ul></li>
<li class="chapter" data-level="15.9" data-path="factorial.html"><a href="factorial.html#two-way-anova"><i class="fa fa-check"></i><b>15.9</b> Two-way ANOVA</a>
<ul>
<li class="chapter" data-level="15.9.1" data-path="factorial.html"><a href="factorial.html#how-to-read-a-two-way-anova-table"><i class="fa fa-check"></i><b>15.9.1</b> How to read a two-way ANOVA table</a></li>
<li class="chapter" data-level="15.9.2" data-path="factorial.html"><a href="factorial.html#what-do-the-main-effects-in-an-anova-table-mean"><i class="fa fa-check"></i><b>15.9.2</b> What do the main effects in an ANOVA table mean?</a></li>
</ul></li>
<li class="chapter" data-level="15.10" data-path="factorial.html"><a href="factorial.html#more-issues-in-inference"><i class="fa fa-check"></i><b>15.10</b> More issues in inference</a>
<ul>
<li class="chapter" data-level="15.10.1" data-path="factorial.html"><a href="factorial.html#longitudinal-experiments-include-time-as-a-random-factor-better-than-repeated-measures-anova"><i class="fa fa-check"></i><b>15.10.1</b> Longitudinal experiments – include Time as a random factor (better than repeated measures ANOVA)</a></li>
</ul></li>
<li class="chapter" data-level="15.11" data-path="factorial.html"><a href="factorial.html#working-in-r-4"><i class="fa fa-check"></i><b>15.11</b> Working in R</a>
<ul>
<li class="chapter" data-level="15.11.1" data-path="factorial.html"><a href="factorial.html#model-formula"><i class="fa fa-check"></i><b>15.11.1</b> Model formula</a></li>
<li class="chapter" data-level="15.11.2" data-path="factorial.html"><a href="factorial.html#using-the-emmeans-function-2"><i class="fa fa-check"></i><b>15.11.2</b> Using the emmeans function</a></li>
<li class="chapter" data-level="15.11.3" data-path="factorial.html"><a href="factorial.html#contrasts"><i class="fa fa-check"></i><b>15.11.3</b> Contrasts</a></li>
<li class="chapter" data-level="15.11.4" data-path="factorial.html"><a href="factorial.html#practice-safe-anova"><i class="fa fa-check"></i><b>15.11.4</b> Practice safe ANOVA</a></li>
<li class="chapter" data-level="15.11.5" data-path="factorial.html"><a href="factorial.html#better-to-avoid-these"><i class="fa fa-check"></i><b>15.11.5</b> Better to avoid these</a></li>
</ul></li>
<li class="chapter" data-level="15.12" data-path="factorial.html"><a href="factorial.html#hidden-code-4"><i class="fa fa-check"></i><b>15.12</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="15.12.1" data-path="factorial.html"><a href="factorial.html#import-exp2j-example-1"><i class="fa fa-check"></i><b>15.12.1</b> Import exp2j (Example 1)</a></li>
<li class="chapter" data-level="15.12.2" data-path="factorial.html"><a href="factorial.html#import-exp3e-lesian-area-data-example-2"><i class="fa fa-check"></i><b>15.12.2</b> Import exp3e lesian area data (Example 2)</a></li>
<li class="chapter" data-level="15.12.3" data-path="factorial.html"><a href="factorial.html#import-exp1c-ja-data-example-3"><i class="fa fa-check"></i><b>15.12.3</b> Import Exp1c JA data (Example 3)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-vii-expanding-the-linear-model.html"><a href="part-vii-expanding-the-linear-model.html"><i class="fa fa-check"></i>Part VII – Expanding the Linear Model</a></li>
<li class="chapter" data-level="16" data-path="lmm.html"><a href="lmm.html"><i class="fa fa-check"></i><b>16</b> Models with random factors – Blocking and pseudoreplication</a>
<ul>
<li class="chapter" data-level="16.1" data-path="lmm.html"><a href="lmm.html#most-experiments-in-bench-biology-generate-batched-data"><i class="fa fa-check"></i><b>16.1</b> Most experiments in bench biology generate batched data</a></li>
<li class="chapter" data-level="16.2" data-path="lmm.html"><a href="lmm.html#example-1-experiment-5c-random-intercepts-experiments-with-only-one-replicate-of-each-treatment-per-block"><i class="fa fa-check"></i><b>16.2</b> Example 1 (Experiment 5c) – Random intercepts – experiments with only one replicate of each treatment per block</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="lmm.html"><a href="lmm.html#understand-the-data"><i class="fa fa-check"></i><b>16.2.1</b> Understand the data</a></li>
<li class="chapter" data-level="16.2.2" data-path="lmm.html"><a href="lmm.html#examine-the-data-1"><i class="fa fa-check"></i><b>16.2.2</b> Examine the data</a></li>
<li class="chapter" data-level="16.2.3" data-path="lmm.html"><a href="lmm.html#fit-the-model-8"><i class="fa fa-check"></i><b>16.2.3</b> Fit the model</a></li>
<li class="chapter" data-level="16.2.4" data-path="lmm.html"><a href="lmm.html#check-the-model-2"><i class="fa fa-check"></i><b>16.2.4</b> Check the model</a></li>
<li class="chapter" data-level="16.2.5" data-path="lmm.html"><a href="lmm.html#inference-from-the-model-4"><i class="fa fa-check"></i><b>16.2.5</b> Inference from the model</a></li>
<li class="chapter" data-level="16.2.6" data-path="lmm.html"><a href="lmm.html#lmm-exp5c-plotthemodel"><i class="fa fa-check"></i><b>16.2.6</b> Plot the model</a></li>
<li class="chapter" data-level="16.2.7" data-path="lmm.html"><a href="lmm.html#alternaplot-the-model"><i class="fa fa-check"></i><b>16.2.7</b> Alternaplot the model</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="lmm.html"><a href="lmm.html#understanding-the-linear-mixed-model-1"><i class="fa fa-check"></i><b>16.3</b> Understanding the linear mixed model 1</a>
<ul>
<li class="chapter" data-level="16.3.1" data-path="lmm.html"><a href="lmm.html#why-we-care-about-blocking"><i class="fa fa-check"></i><b>16.3.1</b> Why we care about blocking</a></li>
<li class="chapter" data-level="16.3.2" data-path="lmm.html"><a href="lmm.html#the-linear-mixed-model-with-a-random-intercept"><i class="fa fa-check"></i><b>16.3.2</b> The linear mixed model with a random intercept</a></li>
<li class="chapter" data-level="16.3.3" data-path="lmm.html"><a href="lmm.html#the-coefficients-of-the-linear-mixed-model-with-random-intercept"><i class="fa fa-check"></i><b>16.3.3</b> The coefficients of the linear mixed model with random intercept</a></li>
<li class="chapter" data-level="16.3.4" data-path="lmm.html"><a href="lmm.html#alternative-models"><i class="fa fa-check"></i><b>16.3.4</b> Alternative models</a></li>
<li class="chapter" data-level="16.3.5" data-path="lmm.html"><a href="lmm.html#paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models"><i class="fa fa-check"></i><b>16.3.5</b> Paired t-tests and repeated measures ANOVA are special cases of linear mixed models</a></li>
<li class="chapter" data-level="16.3.6" data-path="lmm.html"><a href="lmm.html#modeling-textttdonor-as-a-fixed-effect"><i class="fa fa-check"></i><b>16.3.6</b> Modeling <span class="math inline">\(\texttt{donor}\)</span> as a fixed effect</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="lmm.html"><a href="lmm.html#example-2-experiment-1g-random-intercepts-and-slopes-experiments-with-more-than-one-replicate-of-each-treatment-per-block"><i class="fa fa-check"></i><b>16.4</b> Example 2 (Experiment 1g) – Random intercepts and slopes – experiments with more than one replicate of each treatment per block</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="lmm.html"><a href="lmm.html#understand-the-data-1"><i class="fa fa-check"></i><b>16.4.1</b> Understand the data</a></li>
<li class="chapter" data-level="16.4.2" data-path="lmm.html"><a href="lmm.html#examine-the-data-2"><i class="fa fa-check"></i><b>16.4.2</b> Examine the data</a></li>
<li class="chapter" data-level="16.4.3" data-path="lmm.html"><a href="lmm.html#fit-the-model-9"><i class="fa fa-check"></i><b>16.4.3</b> Fit the model</a></li>
<li class="chapter" data-level="16.4.4" data-path="lmm.html"><a href="lmm.html#inference-from-the-model-5"><i class="fa fa-check"></i><b>16.4.4</b> Inference from the model</a></li>
<li class="chapter" data-level="16.4.5" data-path="lmm.html"><a href="lmm.html#plot-the-model-5"><i class="fa fa-check"></i><b>16.4.5</b> Plot the model</a></li>
<li class="chapter" data-level="16.4.6" data-path="lmm.html"><a href="lmm.html#alternaplot-the-model-1"><i class="fa fa-check"></i><b>16.4.6</b> Alternaplot the model</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="lmm.html"><a href="lmm.html#understanding-linear-mixed-models-2"><i class="fa fa-check"></i><b>16.5</b> Understanding linear mixed models 2 –</a>
<ul>
<li class="chapter" data-level="16.5.1" data-path="lmm.html"><a href="lmm.html#the-problem"><i class="fa fa-check"></i><b>16.5.1</b> The problem</a></li>
<li class="chapter" data-level="16.5.2" data-path="lmm.html"><a href="lmm.html#understanding-the-model-formula"><i class="fa fa-check"></i><b>16.5.2</b> Understanding the model formula</a></li>
<li class="chapter" data-level="16.5.3" data-path="lmm.html"><a href="lmm.html#the-random-coefficients"><i class="fa fa-check"></i><b>16.5.3</b> The random coefficients</a></li>
<li class="chapter" data-level="16.5.4" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-have-parameters-for-random-intercept-effects-and-random-slope-effects"><i class="fa fa-check"></i><b>16.5.4</b> Linear mixed models have parameters for random intercept effects and random slope effects</a></li>
<li class="chapter" data-level="16.5.5" data-path="lmm.html"><a href="lmm.html#what-the-random-effects-are"><i class="fa fa-check"></i><b>16.5.5</b> What the random effects are</a></li>
<li class="chapter" data-level="16.5.6" data-path="lmm.html"><a href="lmm.html#alternative-models-1"><i class="fa fa-check"></i><b>16.5.6</b> Alternative models</a></li>
</ul></li>
<li class="chapter" data-level="16.6" data-path="lmm.html"><a href="lmm.html#working-in-r-5"><i class="fa fa-check"></i><b>16.6</b> Working in R</a>
<ul>
<li class="chapter" data-level="16.6.1" data-path="lmm.html"><a href="lmm.html#repeated-measures-anova-randomized-complete-block-with-no-subsampling"><i class="fa fa-check"></i><b>16.6.1</b> Repeated measures ANOVA (randomized complete block with no subsampling)</a></li>
</ul></li>
<li class="chapter" data-level="16.7" data-path="lmm.html"><a href="lmm.html#hidden-code-5"><i class="fa fa-check"></i><b>16.7</b> Hidden code</a>
<ul>
<li class="chapter" data-level="16.7.1" data-path="lmm.html"><a href="lmm.html#import-exp5c"><i class="fa fa-check"></i><b>16.7.1</b> Import exp5c</a></li>
<li class="chapter" data-level="16.7.2" data-path="lmm.html"><a href="lmm.html#import-exp1g"><i class="fa fa-check"></i><b>16.7.2</b> Import exp1g</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="pre-post.html"><a href="pre-post.html"><i class="fa fa-check"></i><b>17</b> Linear models for longitudinal experiments – I. pre-post designs</a>
<ul>
<li class="chapter" data-level="17.1" data-path="pre-post.html"><a href="pre-post.html#best-practice-models"><i class="fa fa-check"></i><b>17.1</b> Best practice models</a></li>
<li class="chapter" data-level="17.2" data-path="pre-post.html"><a href="pre-post.html#common-alternatives-that-are-not-recommended"><i class="fa fa-check"></i><b>17.2</b> Common alternatives that are not recommended</a></li>
<li class="chapter" data-level="17.3" data-path="pre-post.html"><a href="pre-post.html#advanced-models"><i class="fa fa-check"></i><b>17.3</b> Advanced models</a></li>
<li class="chapter" data-level="17.4" data-path="pre-post.html"><a href="pre-post.html#understanding-the-alternative-models"><i class="fa fa-check"></i><b>17.4</b> Understanding the alternative models</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="pre-post.html"><a href="pre-post.html#m1-linear-model-with-the-baseline-measure-as-the-covariate-ancova-model"><i class="fa fa-check"></i><b>17.4.1</b> (M1) Linear model with the baseline measure as the covariate (ANCOVA model)</a></li>
<li class="chapter" data-level="17.4.2" data-path="pre-post.html"><a href="pre-post.html#m2-linear-model-of-the-change-score-change-score-model"><i class="fa fa-check"></i><b>17.4.2</b> (M2) Linear model of the change score (change-score model)</a></li>
<li class="chapter" data-level="17.4.3" data-path="pre-post.html"><a href="pre-post.html#m3-linear-model-of-post-baseline-values-without-the-baseline-as-a-covariate-post-model"><i class="fa fa-check"></i><b>17.4.3</b> (M3) Linear model of post-baseline values without the baseline as a covariate (post model)</a></li>
<li class="chapter" data-level="17.4.4" data-path="pre-post.html"><a href="pre-post.html#m4-linear-model-with-factorial-fixed-effects-fixed-effects-model"><i class="fa fa-check"></i><b>17.4.4</b> (M4) Linear model with factorial fixed effects (fixed-effects model)</a></li>
<li class="chapter" data-level="17.4.5" data-path="pre-post.html"><a href="pre-post.html#m5-repeated-measures-anova"><i class="fa fa-check"></i><b>17.4.5</b> (M5) Repeated measures ANOVA</a></li>
<li class="chapter" data-level="17.4.6" data-path="pre-post.html"><a href="pre-post.html#m6-linear-mixed-model"><i class="fa fa-check"></i><b>17.4.6</b> (M6) Linear mixed model</a></li>
<li class="chapter" data-level="17.4.7" data-path="pre-post.html"><a href="pre-post.html#m7-linear-model-with-correlated-error"><i class="fa fa-check"></i><b>17.4.7</b> (M7) Linear model with correlated error</a></li>
<li class="chapter" data-level="17.4.8" data-path="pre-post.html"><a href="pre-post.html#m8-constrained-fixed-effects-model-with-correlated-error-clda-model"><i class="fa fa-check"></i><b>17.4.8</b> (M8) Constrained fixed effects model with correlated error (cLDA model)</a></li>
<li class="chapter" data-level="17.4.9" data-path="pre-post.html"><a href="pre-post.html#comparison-table"><i class="fa fa-check"></i><b>17.4.9</b> Comparison table</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="pre-post.html"><a href="pre-post.html#example-1-a-single-post-baseline-measure-pre-post-design"><i class="fa fa-check"></i><b>17.5</b> Example 1 – a single post-baseline measure (pre-post design)</a></li>
<li class="chapter" data-level="17.6" data-path="pre-post.html"><a href="pre-post.html#working-in-r-6"><i class="fa fa-check"></i><b>17.6</b> Working in R</a></li>
<li class="chapter" data-level="17.7" data-path="pre-post.html"><a href="pre-post.html#hidden-code-6"><i class="fa fa-check"></i><b>17.7</b> Hidden code</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="pre-post.html"><a href="pre-post.html#import-and-wrangle-mouse-sociability-data"><i class="fa fa-check"></i><b>17.7.1</b> Import and wrangle mouse sociability data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="counts.html"><a href="counts.html"><i class="fa fa-check"></i><b>18</b> Linear models for count data – Generalized Linear Models I</a>
<ul>
<li class="chapter" data-level="18.1" data-path="counts.html"><a href="counts.html#the-generalized-linear-model"><i class="fa fa-check"></i><b>18.1</b> The generalized linear model</a></li>
<li class="chapter" data-level="18.2" data-path="counts.html"><a href="counts.html#count-data-example-number-of-trematode-worm-larvae-in-eyes-of-threespine-stickleback-fish"><i class="fa fa-check"></i><b>18.2</b> Count data example – number of trematode worm larvae in eyes of threespine stickleback fish</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="counts.html"><a href="counts.html#modeling-strategy-1"><i class="fa fa-check"></i><b>18.2.1</b> Modeling strategy</a></li>
<li class="chapter" data-level="18.2.2" data-path="counts.html"><a href="counts.html#checking-the-model-i-a-normal-q-q-plot"><i class="fa fa-check"></i><b>18.2.2</b> Checking the model I – a Normal Q-Q plot</a></li>
<li class="chapter" data-level="18.2.3" data-path="counts.html"><a href="counts.html#checking-the-model-ii-scale-location-plot-for-checking-homoskedasticity"><i class="fa fa-check"></i><b>18.2.3</b> Checking the model II – scale-location plot for checking homoskedasticity</a></li>
<li class="chapter" data-level="18.2.4" data-path="counts.html"><a href="counts.html#two-distributions-for-count-data-poisson-and-negative-binomial"><i class="fa fa-check"></i><b>18.2.4</b> Two distributions for count data – Poisson and Negative Binomial</a></li>
<li class="chapter" data-level="18.2.5" data-path="counts.html"><a href="counts.html#fitting-a-glm-with-a-poisson-distribution-to-the-worm-data"><i class="fa fa-check"></i><b>18.2.5</b> Fitting a GLM with a Poisson distribution to the worm data</a></li>
<li class="chapter" data-level="18.2.6" data-path="counts.html"><a href="counts.html#model-checking-fits-to-count-data"><i class="fa fa-check"></i><b>18.2.6</b> Model checking fits to count data</a></li>
<li class="chapter" data-level="18.2.7" data-path="counts.html"><a href="counts.html#fitting-a-glm-with-a-negative-binomial-distribution-to-the-worm-data"><i class="fa fa-check"></i><b>18.2.7</b> Fitting a GLM with a Negative Binomial distribution to the worm data</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="counts.html"><a href="counts.html#working-in-r-7"><i class="fa fa-check"></i><b>18.3</b> Working in R</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="counts.html"><a href="counts.html#fitting-a-glm-to-count-data"><i class="fa fa-check"></i><b>18.3.1</b> Fitting a GLM to count data</a></li>
<li class="chapter" data-level="18.3.2" data-path="counts.html"><a href="counts.html#fitting-a-generalized-linear-mixed-model-glmm-to-count-data"><i class="fa fa-check"></i><b>18.3.2</b> Fitting a generalized linear mixed model (GLMM) to count data</a></li>
<li class="chapter" data-level="18.3.3" data-path="counts.html"><a href="counts.html#fitting-a-generalized-linear-model-to-continouus-data"><i class="fa fa-check"></i><b>18.3.3</b> Fitting a generalized linear model to continouus data</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="counts.html"><a href="counts.html#problems-1"><i class="fa fa-check"></i><b>18.4</b> Problems</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="gls.html"><a href="gls.html"><i class="fa fa-check"></i><b>19</b> Linear models with heterogenous variance</a></li>
<li class="chapter" data-level="20" data-path="count-sim.html"><a href="count-sim.html"><i class="fa fa-check"></i><b>20</b> Simulations – Count data (alternatives to a t-test)</a>
<ul>
<li class="chapter" data-level="20.1" data-path="count-sim.html"><a href="count-sim.html#use-data-similar-to-figure-6f-from-example-1"><i class="fa fa-check"></i><b>20.1</b> Use data similar to Figure 6f from Example 1</a></li>
<li class="chapter" data-level="20.2" data-path="count-sim.html"><a href="count-sim.html#functions"><i class="fa fa-check"></i><b>20.2</b> Functions</a></li>
<li class="chapter" data-level="20.3" data-path="count-sim.html"><a href="count-sim.html#simulations"><i class="fa fa-check"></i><b>20.3</b> Simulations</a>
<ul>
<li class="chapter" data-level="20.3.1" data-path="count-sim.html"><a href="count-sim.html#type-i-pseudo-normal-distribution"><i class="fa fa-check"></i><b>20.3.1</b> Type I, Pseudo-Normal distribution</a></li>
<li class="chapter" data-level="20.3.2" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-equal-n"><i class="fa fa-check"></i><b>20.3.2</b> Type I, neg binom, equal n</a></li>
<li class="chapter" data-level="20.3.3" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-equal-n-small-theta"><i class="fa fa-check"></i><b>20.3.3</b> Type I, neg binom, equal n, small theta</a></li>
<li class="chapter" data-level="20.3.4" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-unequal-n"><i class="fa fa-check"></i><b>20.3.4</b> Type I, neg binom, unequal n</a></li>
<li class="chapter" data-level="20.3.5" data-path="count-sim.html"><a href="count-sim.html#power-pseudo-normal-distribution-equal-n"><i class="fa fa-check"></i><b>20.3.5</b> Power, Pseudo-Normal distribution, equal n</a></li>
<li class="chapter" data-level="20.3.6" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-equal-n"><i class="fa fa-check"></i><b>20.3.6</b> Power, neg binom, equal n</a></li>
<li class="chapter" data-level="20.3.7" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-small-theta"><i class="fa fa-check"></i><b>20.3.7</b> Power, neg binom, small theta</a></li>
<li class="chapter" data-level="20.3.8" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-unequal-n"><i class="fa fa-check"></i><b>20.3.8</b> Power, neg binom, unequal n</a></li>
<li class="chapter" data-level="20.3.9" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-unequal-n-unequal-theta"><i class="fa fa-check"></i><b>20.3.9</b> Power, neg binom, unequal n, unequal theta</a></li>
<li class="chapter" data-level="20.3.10" data-path="count-sim.html"><a href="count-sim.html#type-1-neg-binom-equal-n-unequal-theta"><i class="fa fa-check"></i><b>20.3.10</b> Type 1, neg binom, equal n, unequal theta</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="count-sim.html"><a href="count-sim.html#save-it-read-it"><i class="fa fa-check"></i><b>20.4</b> Save it, Read it</a></li>
<li class="chapter" data-level="20.5" data-path="count-sim.html"><a href="count-sim.html#analysis"><i class="fa fa-check"></i><b>20.5</b> Analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html"><i class="fa fa-check"></i>Appendix 1: Getting Started with R</a>
<ul>
<li class="chapter" data-level="20.6" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#get-your-computer-ready"><i class="fa fa-check"></i><b>20.6</b> Get your computer ready</a>
<ul>
<li class="chapter" data-level="20.6.1" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#start-here"><i class="fa fa-check"></i><b>20.6.1</b> Start here</a></li>
<li class="chapter" data-level="20.6.2" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r"><i class="fa fa-check"></i><b>20.6.2</b> Install R</a></li>
<li class="chapter" data-level="20.6.3" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r-studio"><i class="fa fa-check"></i><b>20.6.3</b> Install R Studio</a></li>
<li class="chapter" data-level="20.6.4" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r-markdown-1"><i class="fa fa-check"></i><b>20.6.4</b> Install R Markdown</a></li>
<li class="chapter" data-level="20.6.5" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#optional-alternative-latex-installations"><i class="fa fa-check"></i><b>20.6.5</b> (optional) Alternative LaTeX installations</a></li>
</ul></li>
<li class="chapter" data-level="20.7" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#start-learning-r-studio"><i class="fa fa-check"></i><b>20.7</b> Start learning R Studio</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix-2-online-resources-for-getting-started-with-statistical-modeling-in-r.html"><a href="appendix-2-online-resources-for-getting-started-with-statistical-modeling-in-r.html"><i class="fa fa-check"></i>Appendix 2: Online Resources for Getting Started with Statistical Modeling in R</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Elements of Statistical Modeling for Experimental Biology</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="lmm" class="section level1" number="16">
<h1><span class="header-section-number">Chapter 16</span> Models with random factors – Blocking and pseudoreplication</h1>
<div id="most-experiments-in-bench-biology-generate-batched-data" class="section level2" number="16.1">
<h2><span class="header-section-number">16.1</span> Most experiments in bench biology generate batched data</h2>
<p>This chapter is about linear models of which a <strong>paired t-test</strong> and <strong>repeated measures ANOVA</strong> are special cases. Both paired <em>t</em>-tests and repeated measures ANOVA involve experiments in which the response was measured multiple times on the same thing – maybe the left and right side of a mouse, multiple mice with cell from the same donor, or a mouse over multiple weeks. Most data in experimental biology is composed of subsets – or <strong>batches</strong> – of data that were measured from the “same thing.” Batched data results in <strong>correlated error</strong> that muddles statistical inference unless the correlated error is modeled. In some designs, modeling the correlated error increases precision and power, and ultimately decreases false discovery. In other designs, failure to model the correlated error results in incorrectly high precision and incorrectly low <em>p</em>-values, leading to increased rates of false discovery. I think it’s fair to infer from the experimental biology literature, that experimental biologists don’t recognize the ubiquitousness of batched data and correlated error. This is probably the biggest issue in inference in the field (far more of an issue than say, a <em>t</em>-test on non-normal data).</p>
<p>What do I mean by “batch” and how can correlated error both increase and decrease false discovery? Consider these experiments:</p>
<ol style="list-style-type: decimal">
<li>The experiment in Figure <a href="#fig:lmm-biological-replicates"><strong>??</strong></a> is a factorial design with two factors, <span class="math inline">\(\texttt{genotype}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>, each with two levels. One mice of each treatment combination is randomly assigned to a cage. There are five replicates of each cage. The five replicate mice per treatment combination are <strong>biological replicates</strong>. Each cage is a batch. Each cage has a unique set of factors that contribute to the error variance of the measure of the response. Each cage shares a cage-specific history of temperature, humidity, food, light, interactions with animal facilities staff and behavioral interactions among the mice. All response measures within a cage share the component of the error variance unique to that cage and, as a consequence, the error (residuals) within a cage are more similar to each other than they are to the residuals between cages.</li>
</ol>
<p><img src="images/lmm-biological-replicates-435-273.png" width="218" /></p>
<ol start="2" style="list-style-type: decimal">
<li>The experiment in Figure <a href="#fig:lmm-technical-replicates"><strong>??</strong></a> is a factorial design with two factors, <span class="math inline">\(\texttt{genotype}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>, each with two levels. One mice of each treatment combination is randomly assigned to a cage. There are five replicates of each cage. The researchers take three measures of the response variable per mouse. The three measures are <strong>subsampled replicates</strong> (most often referred to as <strong>subsamples</strong>). The subsamples could be <strong>technical replicates</strong> if multiple measures are taken from the same prep or <strong>repeated measures</strong> if the the multiple measures are taken at different time points. In addition to each cage being a batch, each mouse is a batch. Each mouse has a unique set of factors that contribute to the error variance of the measures of the response in that mouse. All response measures within a mouse share the component of the error variance unique to that mouse and, as a consequence, the error (residuals) within a mouse are more similar to each other than they are to the residuals between mice</li>
</ol>
<p><img src="images/lmm-technical-replicates-435-273.png" width="218" /></p>
<ol start="3" style="list-style-type: decimal">
<li>The experiment in Figure <a href="#fig:lmm-segregated"><strong>??</strong></a> has a single factor <span class="math inline">\(\texttt{treatment}\)</span> with two levels. Importantly, the treatment (example: diet) is randomly assigned to cage and all mice in the cage have this treatment. The five mice per cage are <strong>subsampled replicates</strong> (<strong>subsamples</strong>). As in experiments 1 and 2, each cage is a batch. Each cage has a unique set of factors that contribute to the error variance of the measures of the response in that cage. All response measures within a cage share the component of the error variance unique to that cage and, as a consequence, the error (residuals) within a cage are more similar to each other than they are to the residuals among cages.</li>
</ol>
<p><img src="images/lmm-pseudoreplication.png" width="255" /></p>
<p>In each of these experiments, there is systematic variation at multiple levels: among treatments due to treatment effects and among batches due to <strong>batch effects</strong>. Batches come in lots of flavors, including experiment, cage, flask, plate, slide, donor, and individual. The among-batch variation is the <strong>random effect</strong>. An assumption of modeling random effects is that the batches are a random sample of the batches that could have been sampled. This is often not strictly true as batches are often <strong>convenience samples</strong> (example: the human donors of the Type 2 diabetes beta cells are those that were in the hospital).</p>
<p>The variation among batches/lack of independence within batches has different consequences on the uncertainty of the estimate of a treatment effect. The batches in Experiment 1 contain all treatment combinations. The researcher is interested in the treatment effect but not the variation due to differences among the batches. The batches are nuissance factors that add additional variance to the response, with the consequence that estimates of treatment effects are less precise, unless the variance due to the batches is explicitly modeled. <strong>Modeling a batch that contains some or all treatment combinations will increase precision and power</strong>.</p>
<p>Batches that contain at least two treatment combinations are known as <strong>blocks</strong>. A block that contains all treatment combinations is a <strong>complete block</strong>. A block that contains fewer than all combinations is an <strong>incomplete block</strong>. Including block structure in the design is known as <strong>blocking</strong>. Adding a blocking factor to a statistical model is used to increase the precision of an estimated treatment effect. Experiment 1 is an example of a <strong>randomized complete block</strong> design. “Complete” means that each block has all treatment levels or combinations of levels if there is more than one factor.</p>
<p>In Experiment 2, there are multiple measures per mouse and the design is a <strong>randomized complete block with subsampling</strong>. The subsampling is not the kind of replication that can be used to infer the among treatment effect because the treatment assignment was not at the level of the subsamples. The <strong>treatment replicates</strong> (the cage) are the blocks, because <em>it was at this level that treatment assignment was randomized</em>. Nevertheless, there are inference advantages to subsampling if modeled, But, a statistical analysis of all measures from a subsampled design without modeling the correlated error due to the subsampling is a kind of <a href="https://en.wikipedia.org/wiki/Pseudoreplication" target="_blank">pseudoreplication</a>. Pseudoreplication results in incorrectly small standard errors and <em>p</em>-values and increased rates of false discovery.</p>
<p>In Experiment 3, the treatment is randomized <em>to</em> batch, so each batch contains only a single treatment level. In these <strong>segregated</strong> experimental designs, the variation among batches that arises from non-treatment related differences among batches <strong>confounds</strong> the variation among batches due to a true treatment effect. An extreme example of this would be an experiment with only a single cage with control conditions and a single cage with treatment conditions. Imagine 1) the true effect of the treatment is zero and 2) an aggressive mouse in the control cage stimulates the stress response in the other mice and this stress response has a large effect on the value of the response variable measured by the researchers. The researcher is fooled into thinking that the treatment caused the difference in the response. Again, mice are subsampled replicates while treatment replicates are at the level of the cage, because it was at this level that treatment assignment was randomized. This means the researcher has a single, treatment replicate (or, <span class="math inline">\(n=1\)</span>), regardless of the number of mice (subsamples) in each cage. A statistical analysis of all measures from a subsampled design without modeling the correlated error due to the subsampling is a kind of <a href="https://en.wikipedia.org/wiki/Pseudoreplication" target="_blank">pseudoreplication</a>. Pseudoreplication results in incorrectly small standard errors and <em>p</em>-values and increased rates of false discovery.</p>
</div>
<div id="example-1-experiment-5c-random-intercepts-experiments-with-only-one-replicate-of-each-treatment-per-block" class="section level2" number="16.2">
<h2><span class="header-section-number">16.2</span> Example 1 (Experiment 5c) – Random intercepts – experiments with only one replicate of each treatment per block</h2>
<p>source: <a href="https://www.nature.com/articles/s41467-019-13869-w" target="_blank">Transcriptomic profiling of skeletal muscle adaptations to exercise and inactivity</a></p>
<div id="understand-the-data" class="section level3" number="16.2.1">
<h3><span class="header-section-number">16.2.1</span> Understand the data</h3>
<p>The data for Example 1 are from Figure 5c. Six muscle source cells were used to start six independent cultures. Cells from each culture were treated with either a negative control (“Scr”) or a siRNA (“siNR4A3”) that “silences” expression of the NR4A3 gene product by <a href="https://en.wikipedia.org/wiki/Small_interfering_RNA" target="_blank">cleaving the mRNA</a>. Glucose uptake in the two cell types was measured at rest (“Basal”) and during electrical pulse stimulation (“EPS”).</p>
<p>The design is a <span class="math inline">\(2 \times 2\)</span> <a href="https://en.wikipedia.org/wiki/Blocking_(statistics)#Randomized_block_design" target="_blank">Randomized complete block with no subsampling</a>. There are two factors each with two levels: <span class="math inline">\(\texttt{treatment}\)</span> (“Scr,” “siNR4A3”) and <span class="math inline">\(\texttt{activity}\)</span> (“Basal,” “EPS”). Each source cell is a block. All four treatment combinations were measured once per block.</p>
</div>
<div id="examine-the-data-1" class="section level3" number="16.2.2">
<h3><span class="header-section-number">16.2.2</span> Examine the data</h3>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-examine-1.png" width="576" /></p>
<p>The plot shows a strong donor effect.</p>
</div>
<div id="fit-the-model-8" class="section level3" number="16.2.3">
<h3><span class="header-section-number">16.2.3</span> Fit the model</h3>
<div class="sourceCode" id="cb527"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb527-1"><a href="lmm.html#cb527-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb527-2"><a href="lmm.html#cb527-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb527-3"><a href="lmm.html#cb527-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span></code></pre></div>
</div>
<div id="check-the-model-2" class="section level3" number="16.2.4">
<h3><span class="header-section-number">16.2.4</span> Check the model</h3>
<div class="sourceCode" id="cb528"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb528-1"><a href="lmm.html#cb528-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggcheck_the_model</span>(exp5c_m1)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c_m1-check-1.png" width="576" /></p>
<p>fine.</p>
</div>
<div id="inference-from-the-model-4" class="section level3" number="16.2.5">
<h3><span class="header-section-number">16.2.5</span> Inference from the model</h3>
<div class="sourceCode" id="cb529"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb529-1"><a href="lmm.html#cb529-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1_coef <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">coef</span>(<span class="fu">summary</span>(exp5c_m1)),</span>
<span id="cb529-2"><a href="lmm.html#cb529-2" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">confint</span>(exp5c_m1)[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>),])</span>
<span id="cb529-3"><a href="lmm.html#cb529-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb529-4"><a href="lmm.html#cb529-4" aria-hidden="true" tabindex="-1"></a><span class="co"># exp5c_m1_coef %&gt;%</span></span>
<span id="cb529-5"><a href="lmm.html#cb529-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   kable(digits = c(2,3,1,1,4,2,2)) %&gt;%</span></span>
<span id="cb529-6"><a href="lmm.html#cb529-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   kable_styling()</span></span></code></pre></div>
<div class="sourceCode" id="cb530"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb530-1"><a href="lmm.html#cb530-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp5c_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>))</span>
<span id="cb530-2"><a href="lmm.html#cb530-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb530-3"><a href="lmm.html#cb530-3" aria-hidden="true" tabindex="-1"></a>exp5c_m1_emm <span class="sc">%&gt;%</span></span>
<span id="cb530-4"><a href="lmm.html#cb530-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb530-5"><a href="lmm.html#cb530-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:left;">
activity
</th>
<th style="text-align:right;">
emmean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Scr
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:right;">
1.34
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
0.96
</td>
<td style="text-align:right;">
1.72
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:right;">
1.42
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
1.04
</td>
<td style="text-align:right;">
1.80
</td>
</tr>
<tr>
<td style="text-align:left;">
Scr
</td>
<td style="text-align:left;">
EPS
</td>
<td style="text-align:right;">
1.49
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
1.11
</td>
<td style="text-align:right;">
1.87
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3
</td>
<td style="text-align:left;">
EPS
</td>
<td style="text-align:right;">
1.31
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
0.93
</td>
<td style="text-align:right;">
1.68
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb531"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb531-1"><a href="lmm.html#cb531-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exp5c_emm # print in console to get row numbers</span></span>
<span id="cb531-2"><a href="lmm.html#cb531-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb531-3"><a href="lmm.html#cb531-3" aria-hidden="true" tabindex="-1"></a>scr_basal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb531-4"><a href="lmm.html#cb531-4" aria-hidden="true" tabindex="-1"></a>siNR4A3_basal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb531-5"><a href="lmm.html#cb531-5" aria-hidden="true" tabindex="-1"></a>scr_eps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb531-6"><a href="lmm.html#cb531-6" aria-hidden="true" tabindex="-1"></a>siNR4A3_eps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb531-7"><a href="lmm.html#cb531-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb531-8"><a href="lmm.html#cb531-8" aria-hidden="true" tabindex="-1"></a>exp5c_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp5c_m1_emm,</span>
<span id="cb531-9"><a href="lmm.html#cb531-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb531-10"><a href="lmm.html#cb531-10" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;(Scr EPS) - (Scr Basal)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(scr_eps <span class="sc">-</span> scr_basal),</span>
<span id="cb531-11"><a href="lmm.html#cb531-11" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;(siNR4A3 EPS) - (siNR4A3 Basal)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(siNR4A3_eps <span class="sc">-</span> siNR4A3_basal),</span>
<span id="cb531-12"><a href="lmm.html#cb531-12" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;Interaction&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(siNR4A3_eps <span class="sc">-</span> siNR4A3_basal) <span class="sc">-</span></span>
<span id="cb531-13"><a href="lmm.html#cb531-13" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">c</span>(scr_eps <span class="sc">-</span> scr_basal)</span>
<span id="cb531-14"><a href="lmm.html#cb531-14" aria-hidden="true" tabindex="-1"></a>                           </span>
<span id="cb531-15"><a href="lmm.html#cb531-15" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb531-16"><a href="lmm.html#cb531-16" aria-hidden="true" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb531-17"><a href="lmm.html#cb531-17" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb531-18"><a href="lmm.html#cb531-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb531-19"><a href="lmm.html#cb531-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb531-20"><a href="lmm.html#cb531-20" aria-hidden="true" tabindex="-1"></a>exp5c_m1_planned <span class="sc">%&gt;%</span></span>
<span id="cb531-21"><a href="lmm.html#cb531-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb531-22"><a href="lmm.html#cb531-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
2.25
</td>
<td style="text-align:right;">
0.040
</td>
</tr>
<tr>
<td style="text-align:left;">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.26
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
-1.73
</td>
<td style="text-align:right;">
0.105
</td>
</tr>
<tr>
<td style="text-align:left;">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.47
</td>
<td style="text-align:right;">
-0.06
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
0.013
</td>
</tr>
</tbody>
</table>
</div>
<div id="lmm-exp5c-plotthemodel" class="section level3" number="16.2.6">
<h3><span class="header-section-number">16.2.6</span> Plot the model</h3>
<div class="sourceCode" id="cb532"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb532-1"><a href="lmm.html#cb532-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot_the_model</span>(exp5c_m1,</span>
<span id="cb532-2"><a href="lmm.html#cb532-2" aria-hidden="true" tabindex="-1"></a>                 exp5c_m1_emm,</span>
<span id="cb532-3"><a href="lmm.html#cb532-3" aria-hidden="true" tabindex="-1"></a>                 exp5c_m1_planned,</span>
<span id="cb532-4"><a href="lmm.html#cb532-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y_label =</span> <span class="st">&quot;Glucose uptake</span><span class="sc">\n</span><span class="st">(pmol per min)&quot;</span>,</span>
<span id="cb532-5"><a href="lmm.html#cb532-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">effect_label =</span> <span class="st">&quot;Difference in Glucose uptake</span><span class="sc">\n</span><span class="st">(pmol per min)&quot;</span>,</span>
<span id="cb532-6"><a href="lmm.html#cb532-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">palette =</span> pal_okabe_ito_blue,</span>
<span id="cb532-7"><a href="lmm.html#cb532-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="fl">0.5</span>,<span class="dv">1</span>))</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-plot-1.png" width="576" /></p>
</div>
<div id="alternaplot-the-model" class="section level3" number="16.2.7">
<h3><span class="header-section-number">16.2.7</span> Alternaplot the model</h3>
<div class="sourceCode" id="cb533"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb533-1"><a href="lmm.html#cb533-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp5c_m1_emm,</span>
<span id="cb533-2"><a href="lmm.html#cb533-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb533-3"><a href="lmm.html#cb533-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb533-4"><a href="lmm.html#cb533-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb533-5"><a href="lmm.html#cb533-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb533-6"><a href="lmm.html#cb533-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb533-7"><a href="lmm.html#cb533-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-8"><a href="lmm.html#cb533-8" aria-hidden="true" tabindex="-1"></a><span class="co"># get coefficients of model</span></span>
<span id="cb533-9"><a href="lmm.html#cb533-9" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(exp5c_m1))[, <span class="st">&quot;Estimate&quot;</span>]</span>
<span id="cb533-10"><a href="lmm.html#cb533-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-11"><a href="lmm.html#cb533-11" aria-hidden="true" tabindex="-1"></a><span class="co"># get interaction p</span></span>
<span id="cb533-12"><a href="lmm.html#cb533-12" aria-hidden="true" tabindex="-1"></a>p_ixn <span class="ot">&lt;-</span> exp5c_m1_planned[<span class="dv">3</span>, <span class="st">&quot;p.value&quot;</span>] <span class="co"># check!</span></span>
<span id="cb533-13"><a href="lmm.html#cb533-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-14"><a href="lmm.html#cb533-14" aria-hidden="true" tabindex="-1"></a>dodge_width <span class="ot">&lt;-</span> <span class="fl">0.4</span></span>
<span id="cb533-15"><a href="lmm.html#cb533-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-16"><a href="lmm.html#cb533-16" aria-hidden="true" tabindex="-1"></a>gg <span class="ot">&lt;-</span> <span class="fu">ggplot_the_response</span>(</span>
<span id="cb533-17"><a href="lmm.html#cb533-17" aria-hidden="true" tabindex="-1"></a>  exp5c_m1,</span>
<span id="cb533-18"><a href="lmm.html#cb533-18" aria-hidden="true" tabindex="-1"></a>  exp5c_m1_emm,</span>
<span id="cb533-19"><a href="lmm.html#cb533-19" aria-hidden="true" tabindex="-1"></a>  exp5c_m1_pairs[<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">4</span>),], <span class="co"># only for comparisons within treatment</span></span>
<span id="cb533-20"><a href="lmm.html#cb533-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> pal_okabe_ito_blue,</span>
<span id="cb533-21"><a href="lmm.html#cb533-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend_position =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb533-22"><a href="lmm.html#cb533-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_label =</span> <span class="st">&quot;Glucose uptake</span><span class="sc">\n</span><span class="st">(pmol per min)&quot;</span></span>
<span id="cb533-23"><a href="lmm.html#cb533-23" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb533-24"><a href="lmm.html#cb533-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">x =</span> <span class="dv">2</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="dv">2</span> <span class="sc">-</span> <span class="fl">0.1</span>,</span>
<span id="cb533-25"><a href="lmm.html#cb533-25" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb533-26"><a href="lmm.html#cb533-26" aria-hidden="true" tabindex="-1"></a>               <span class="at">xend =</span> <span class="dv">2</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="dv">2</span> <span class="sc">+</span> <span class="fl">0.1</span>,</span>
<span id="cb533-27"><a href="lmm.html#cb533-27" aria-hidden="true" tabindex="-1"></a>               <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb533-28"><a href="lmm.html#cb533-28" aria-hidden="true" tabindex="-1"></a>               <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb533-29"><a href="lmm.html#cb533-29" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;gray&quot;</span>) <span class="sc">+</span></span>
<span id="cb533-30"><a href="lmm.html#cb533-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bracket</span>(</span>
<span id="cb533-31"><a href="lmm.html#cb533-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fl">2.35</span>,</span>
<span id="cb533-32"><a href="lmm.html#cb533-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb533-33"><a href="lmm.html#cb533-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>] <span class="sc">+</span> b[<span class="dv">4</span>],</span>
<span id="cb533-34"><a href="lmm.html#cb533-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;ixn p = &quot;</span>,</span>
<span id="cb533-35"><a href="lmm.html#cb533-35" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">fmt_p_value_rmd</span>(p_ixn)),</span>
<span id="cb533-36"><a href="lmm.html#cb533-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.size =</span> <span class="dv">3</span>,</span>
<span id="cb533-37"><a href="lmm.html#cb533-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.hjust =</span> <span class="dv">0</span>,</span>
<span id="cb533-38"><a href="lmm.html#cb533-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb533-39"><a href="lmm.html#cb533-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-40"><a href="lmm.html#cb533-40" aria-hidden="true" tabindex="-1"></a>gg</span></code></pre></div>
<div class="figure"><span id="fig:lmm-exp5c-alternaplot"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-alternaplot-1.png" alt="Dashed gray line is expected additive mean of &quot;siNR4A3 EPS&quot;" width="576" />
<p class="caption">
Figure 16.1: Dashed gray line is expected additive mean of “siNR4A3 EPS”
</p>
</div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Many researchers might look at the long confidence intervals relative to the short distance between the means and think “no effect.” The confidence intervals are correct, they simply are <em>not</em> meant to be tools for inferring anything about differences in means. This is one of many reasons why plots of means and error bars can be misleading for inference, despite the ubiquity of their use for communicating results. And, its why I prefer the <a href="lmm.html#lmm-exp5c-plotthemodel">effects-and-response plots</a>, which explicitly communicate correct inference about effects.</li>
</ol>
</div>
</div>
<div id="understanding-the-linear-mixed-model-1" class="section level2" number="16.3">
<h2><span class="header-section-number">16.3</span> Understanding the linear mixed model 1</h2>
<div id="why-we-care-about-blocking" class="section level3" number="16.3.1">
<h3><span class="header-section-number">16.3.1</span> Why we care about blocking</h3>
<p>Figure <a href="lmm.html#fig:lmm-exp5c-why">16.2</a> shows the modeled means of the four treatment combinations and the individual values colored by donor. It is pretty easy to see that the glucose uptake values for donors 4 and 5 are well above the mean for all four treatments. And, the values for donors 1, 2, and 3 are well below the mean for all four treatments. The values for donor 6 are near the mean for all four treatments.</p>
<div class="figure"><span id="fig:lmm-exp5c-why"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-why-1.png" alt="Why we care about blocking. The black dots are the modeled means of each treatment combination. The colored dots are the measured values of the response for each donor. The position of a donor relative to the mean is easy to see with these data." width="672" />
<p class="caption">
Figure 16.2: Why we care about blocking. The black dots are the modeled means of each treatment combination. The colored dots are the measured values of the response for each donor. The position of a donor relative to the mean is easy to see with these data.
</p>
</div>
<p>There are two ways of thinking about this clustering of <span class="math inline">\(\texttt{donor}\)</span>.</p>
<div id="lmm-exp5c-cor" class="section level4" number="16.3.1.1">
<h4><span class="header-section-number">16.3.1.1</span> Correlated error</h4>
<p>An assumption of inference from a linear model is independence – each response is independently drawn from a distribution of random values. In Experiment 5c, the donors are batches and the batched data results in correlated error unless modeled. One way to see this correlated error is to use the residuals from a linear model without the <span class="math inline">\(\texttt{donor}\)</span> factor.</p>
<ol style="list-style-type: decimal">
<li>fit the model without random factor</li>
<li>compute the residuals from the model</li>
<li>cast (or spread) the residuals for each treatment combination into its own column. This creates a 6 rows (donor) <span class="math inline">\(\times\)</span> 4 columns (treatment combination) matrix of residuals.</li>
<li>Compute the correlations among the four treatment combination columns. This is the correlated error due to the batch effect of <span class="math inline">\(\texttt{donor}\)</span>.</li>
</ol>
<div class="sourceCode" id="cb534"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb534-1"><a href="lmm.html#cb534-1" aria-hidden="true" tabindex="-1"></a><span class="co"># the fixed effect model</span></span>
<span id="cb534-2"><a href="lmm.html#cb534-2" aria-hidden="true" tabindex="-1"></a>m0 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> activity <span class="sc">*</span> treatment, <span class="at">data =</span> exp5c)</span>
<span id="cb534-3"><a href="lmm.html#cb534-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb534-4"><a href="lmm.html#cb534-4" aria-hidden="true" tabindex="-1"></a><span class="co"># create a column of residuals</span></span>
<span id="cb534-5"><a href="lmm.html#cb534-5" aria-hidden="true" tabindex="-1"></a>exp5c[, m0_res <span class="sc">:</span><span class="er">=</span> <span class="fu">residuals</span>(m0)]</span>
<span id="cb534-6"><a href="lmm.html#cb534-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb534-7"><a href="lmm.html#cb534-7" aria-hidden="true" tabindex="-1"></a><span class="co"># cast into a new data.table</span></span>
<span id="cb534-8"><a href="lmm.html#cb534-8" aria-hidden="true" tabindex="-1"></a>exp5c_wide <span class="ot">&lt;-</span> <span class="fu">dcast</span>(exp5c,</span>
<span id="cb534-9"><a href="lmm.html#cb534-9" aria-hidden="true" tabindex="-1"></a>                    donor <span class="sc">~</span> t.by.a,</span>
<span id="cb534-10"><a href="lmm.html#cb534-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">value.var =</span> <span class="st">&quot;m0_res&quot;</span>)</span>
<span id="cb534-11"><a href="lmm.html#cb534-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb534-12"><a href="lmm.html#cb534-12" aria-hidden="true" tabindex="-1"></a><span class="co"># cor(exp5c_wide[, .SD, .SDcols = names(exp5c_wide)[2:5]]) %&gt;%</span></span>
<span id="cb534-13"><a href="lmm.html#cb534-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   round(2)</span></span>
<span id="cb534-14"><a href="lmm.html#cb534-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb534-15"><a href="lmm.html#cb534-15" aria-hidden="true" tabindex="-1"></a>GGally<span class="sc">::</span><span class="fu">ggpairs</span>(exp5c_wide[, <span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">5</span>)], <span class="at">progress =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-cor-error-1.png" width="576" />
I’ve used <code>GGally::ggpairs</code> to compute and display the correlations as a matrix. The lower triangle of matrix elements contains the scatterplot of the residuals for the treatment combination defined by the row and column headers. The upper triangle of elements contains the Pearson correlation. With only six donor residuals per treatment combination, large correlations will be common. But all correlations are large, positive values. The asterisks indicate values that would be a unexpected surprise under a null model of no correlation.</p>
<p>We could explicitly model correlated error with a <strong>linear model for correlated error</strong> using the <code>nlme::gls</code> function, using a model for the correlated error that matches our knowledge of how the data were generated (from donor batches). Here, we implicitly model the correlated error using a linear model with added random factors – a <strong>linear mixed model</strong>. What we explicitly model in a linear mixed model is the second way of thinking about the clustering of <span class="math inline">\(\textt{donor}\)</span> in Figure <a href="lmm.html#fig:lmm-exp5c-why">16.2</a>.</p>
</div>
<div id="different-sources-of-error-variance" class="section level4" number="16.3.1.2">
<h4><span class="header-section-number">16.3.1.2</span> Different sources of error variance</h4>
<p>The variance among the donors within a treatment is much greater than the variance among the treatment means. A consequence of this is, the donor effect masks the effect of treatment. We can manually unmask this by</p>
<ol style="list-style-type: decimal">
<li>compute the donor means across all treatment combinations. Here, this is done using the expected (<code>predict</code>) values from the linear model with formula <code>glucose_uptake ~ donor</code></li>
<li>create a glucose_uptake variable without variation among donor means (“adjusted for donor”). Here, I’ve added the grand mean of glucose_uptake to the adjusted variable to shift the means of each treatment combination back up to the values in Figure <a href="lmm.html#fig:lmm-exp5c-why">16.2</a>.</li>
</ol>
<div class="sourceCode" id="cb535"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb535-1"><a href="lmm.html#cb535-1" aria-hidden="true" tabindex="-1"></a><span class="co"># step 1</span></span>
<span id="cb535-2"><a href="lmm.html#cb535-2" aria-hidden="true" tabindex="-1"></a>m0 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> donor, <span class="at">data =</span> exp5c)</span>
<span id="cb535-3"><a href="lmm.html#cb535-3" aria-hidden="true" tabindex="-1"></a>exp5c[, donor_mean <span class="sc">:</span><span class="er">=</span> <span class="fu">predict</span>(m0)]</span>
<span id="cb535-4"><a href="lmm.html#cb535-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb535-5"><a href="lmm.html#cb535-5" aria-hidden="true" tabindex="-1"></a><span class="co"># step 2</span></span>
<span id="cb535-6"><a href="lmm.html#cb535-6" aria-hidden="true" tabindex="-1"></a>exp5c[, glucose_uptake_donor_adj <span class="sc">:</span><span class="er">=</span> glucose_uptake <span class="sc">-</span> donor_mean <span class="sc">+</span> <span class="fu">mean</span>(glucose_uptake)]</span>
<span id="cb535-7"><a href="lmm.html#cb535-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb535-8"><a href="lmm.html#cb535-8" aria-hidden="true" tabindex="-1"></a>exp5c_long <span class="ot">&lt;-</span> <span class="fu">melt</span>(exp5c,</span>
<span id="cb535-9"><a href="lmm.html#cb535-9" aria-hidden="true" tabindex="-1"></a>                   <span class="at">id.vars =</span> <span class="fu">c</span>(<span class="st">&quot;t.by.a&quot;</span>, <span class="st">&quot;donor&quot;</span>),</span>
<span id="cb535-10"><a href="lmm.html#cb535-10" aria-hidden="true" tabindex="-1"></a>                   <span class="at">measure.vars =</span> <span class="fu">c</span>(<span class="st">&quot;glucose_uptake&quot;</span>,</span>
<span id="cb535-11"><a href="lmm.html#cb535-11" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">&quot;glucose_uptake_donor_adj&quot;</span>),</span>
<span id="cb535-12"><a href="lmm.html#cb535-12" aria-hidden="true" tabindex="-1"></a>                   <span class="at">variable.name =</span> <span class="st">&quot;measure&quot;</span>,</span>
<span id="cb535-13"><a href="lmm.html#cb535-13" aria-hidden="true" tabindex="-1"></a>                   <span class="at">value.name =</span> <span class="st">&quot;glucose_uptake&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:lmm-exp5c-manual-lmm-plot"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-manual-lmm-plot-1.png" alt="Why we care about blocking II. See text for explaination" width="576" />
<p class="caption">
Figure 16.3: Why we care about blocking II. See text for explaination
</p>
</div>
<p>In Figure <a href="lmm.html#fig:lmm-exp5c-manual-lmm-plot">16.3</a>, the black dots are the modeled means of each treatment combination. The light colored dots are the measured values of the response for each donor. The dark colored dots are the donor-adjusted values of the response for each donor. If you look closely, the mean for each donor across the treatment combinations are the same. What is left is error variation uncontaminated by donor.</p>
</div>
<div id="the-linear-model-with-added-random-intercept-increases-precision-and-power" class="section level4" number="16.3.1.3">
<h4><span class="header-section-number">16.3.1.3</span> The linear model with added random intercept increases precision and power</h4>
<p>Let’s compare the effects estimated by the linear mixed model <code>exp5c_m1</code> with a linear model that ignores donor.</p>
<div class="sourceCode" id="cb536"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb536-1"><a href="lmm.html#cb536-1" aria-hidden="true" tabindex="-1"></a>exp5c_m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb536-2"><a href="lmm.html#cb536-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">data =</span> exp5c)</span></code></pre></div>
<div class="figure"><span id="fig:lmm-exp5c-why2"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-why2-1.png" alt="A. Inference from a linear mixed model with blocking factor (donor) added as a random intercept. B. Inference from a fixed effects model." width="576" />
<p class="caption">
Figure 16.4: A. Inference from a linear mixed model with blocking factor (donor) added as a random intercept. B. Inference from a fixed effects model.
</p>
</div>
<p>Figure <a href="lmm.html#fig:lmm-exp5c-why2">16.4</a>A is a plot of the effects from the linear mixed model that models the correlated error due to donor. Figure <a href="lmm.html#fig:lmm-exp5c-why2">16.4</a>B is a plot of the effects from the fixed effect model that ignores the correlated error due to donor. Adding <span class="math inline">\(\texttt{donor}\)</span> as a factor to the linear model increases the precision of the estimate of the treatment effects by eliminating the among-donor component of variance from the error variance.</p>
</div>
</div>
<div id="the-linear-mixed-model-with-a-random-intercept" class="section level3" number="16.3.2">
<h3><span class="header-section-number">16.3.2</span> The linear mixed model with a random intercept</h3>
<p>In the linear mixed model <code>exp5c_m1</code>, the donor factor is added as a <strong>random intercept</strong> to the <strong>fixed</strong> factors. The linear mixed model fit to <code>exp5c</code> data is</p>
<p><span class="math display">\[
\begin{align}
\texttt{glucose_uptake}_j = \ &amp;(\beta_0 + \gamma_{0j}) + \beta_1 (\texttt{treatment}_\texttt{siNR4A3}) + \beta_2 (\texttt{activity}_\texttt{EPS}) \ + \\
&amp;\beta_3 (\texttt{treatment}_\texttt{siNR4A3}:\texttt{activity}_\texttt{EPS}) + \varepsilon
\end{align}
\]</span></p>
<p>A <strong>random intercept</strong> for block (<span class="math inline">\(\textt{donor}\)</span>) <em>j</em> is the sum of the fixed intercept (<span class="math inline">\(\beta_0\)</span>) and a <strong>random intercept coeficient</strong> (<span class="math inline">\(\gamma_{0j}\)</span>). I’ve embedded these within parentheses to show how these combine into the random intercept that is unique for each block. There is a different <span class="math inline">\(\gamma_{0j}\)</span> for each donor. In contrast, <span class="math inline">\(\beta_0\)</span>, <span class="math inline">\(\beta_1\)</span>, <span class="math inline">\(\beta_2\)</span>, and <span class="math inline">\(\beta_3\)</span> are the same for all donors – this is why they are <strong>fixed effects</strong> (technically, <span class="math inline">\(\beta_0\)</span> is a mean and not an effect). <span class="math inline">\(\gamma_0j\)</span> is modeled as if the value for each donor is a random draw from an infinite number of donors. This is why <span class="math inline">\(\gamma_{0j}\)</span> is a <strong>random effect</strong>.</p>
<p>Compare the linear mixed model <code>exp5c_m1</code> to the linear model ignoring <span class="math inline">\(\texttt{donor}\)</span> (<code>exp5c_m2</code>).</p>
<p><span class="math display">\[
\begin{align}
\texttt{glucose_uptake}_j = \ &amp;\beta_0 + \beta_1 (\texttt{treatment}_\texttt{siNR4A3}) + \beta_2 (\texttt{activity}_\texttt{EPS}) \ + \\
&amp;\beta_3 (\texttt{treatment}_\texttt{siNR4A3}:\texttt{activity}_\texttt{EPS}) + (\gamma_{0j} + \varepsilon)
\end{align}
\]</span>
The <span class="math inline">\(\gamma_{0j}\)</span> term is combined with the error term (<span class="math inline">\(\varepsilon\)</span>) but, since model <code>exp5c_m1</code> doesn’t estimate <span class="math inline">\(\gamma_{0j}\)</span>, the residuals (<span class="math inline">\(e\)</span>) estimate the combination <span class="math inline">\(\gamma_{0j} + \varepsilon\)</span>. The variance of the residuals of model <code>exp5c_m2</code> (the estimate of <span class="math inline">\(\sigma^2\)</span>) is large relative to that in the linear mixed model because it is the sum of the variances of <span class="math inline">\(\gamma_{0j}\)</span> and <span class="math inline">\(\varepsilon\)</span>.</p>
<p>The linear model with the blocking factor added as a random intercept is implemented in R using the <code>lme4::lmer</code> function.</p>
<div class="sourceCode" id="cb537"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb537-1"><a href="lmm.html#cb537-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb537-2"><a href="lmm.html#cb537-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">data =</span> exp5c)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The <code>lmer()</code> function takes the usual R formula with the response variable on the left side of the <span class="math inline">\(\sim\)</span> and the treatment variables and covariates on the right side of the <span class="math inline">\(\sim\)</span>.</li>
<li>What is new: The general formula for adding random factors to a linear model using <code>lmer</code> is <span class="math inline">\(\operatorname{(fixed \ factors | random \ factors)}\)</span>. The <span class="math inline">\(\texttt{1}\)</span> tells R to fit the random factor <span class="math inline">\(\texttt{donor}\)</span> as a random intercept.</li>
<li>What is old: R automatically expands <span class="math inline">\(\texttt{treatment * activity}\)</span> to <span class="math inline">\(\texttt{1 + treatment + activity + treatment:activity}\)</span>. The initial <span class="math inline">\(\texttt{1}\)</span> tells the model to fit the intercept (<span class="math inline">\(\beta_0\)</span>). The <span class="math inline">\(\texttt{treatment}\)</span> term tells the model to estimate a coefficient for each non-reference levels of <span class="math inline">\(\texttt{treatment}\)</span>. The <span class="math inline">\(\texttt{activity}\)</span> term tells the model to estimate a coefficient for each non-reference levels of <span class="math inline">\(\texttt{activity}\)</span>. The <span class="math inline">\(\texttt{treatment:activity}\)</span> term tells the model to estimate a coefficient for every combination (interaction) of the non-reference levels of <span class="math inline">\(\texttt{treatment}\)</span> and <span class="math inline">\(\texttt{activity}\)</span>.</li>
</ol>
</div>
<div id="the-coefficients-of-the-linear-mixed-model-with-random-intercept" class="section level3" number="16.3.3">
<h3><span class="header-section-number">16.3.3</span> The coefficients of the linear mixed model with random intercept</h3>
<div id="the-fixed-effect-coefficients" class="section level4" number="16.3.3.1">
<h4><span class="header-section-number">16.3.3.1</span> The fixed effect coefficients</h4>
<p>The fixed effect coefficients of model <code>exp5c_m1</code> are</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
1.34
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
8.77
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
1.02
</td>
<td style="text-align:right;">
1.66
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15.0
</td>
<td style="text-align:right;">
1.21
</td>
<td style="text-align:right;">
0.2453
</td>
<td style="text-align:right;">
-0.05
</td>
<td style="text-align:right;">
0.21
</td>
</tr>
<tr>
<td style="text-align:left;">
activityEPS
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15.0
</td>
<td style="text-align:right;">
2.25
</td>
<td style="text-align:right;">
0.0402
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.28
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3:activityEPS
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
15.0
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
0.0132
</td>
<td style="text-align:right;">
-0.45
</td>
<td style="text-align:right;">
-0.09
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The interpretation of the fixed effectcs coefficients have the <a href="factorial.html#twoway-what-coefs-are">usual interpretation for a factorial linear model</a>. Figure @ref(fig:lmm-exp5c-explainer_1) is a reminder.</li>
</ol>
<div class="figure">
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-explainer_1-1.png" alt="Fixed effects estimated by exp5c_m1. The light gray point is the expected value of the GPR174- F treatment if genotype and sex were additive." width="672" />
<p class="caption">
(#fig:lmm-exp5c-explainer_1)Fixed effects estimated by exp5c_m1. The light gray point is the expected value of the GPR174- F treatment if genotype and sex were additive.
</p>
</div>
</div>
<div id="the-random-intercept-coefficients" class="section level4" number="16.3.3.2">
<h4><span class="header-section-number">16.3.3.2</span> The random intercept coefficients</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-m1-random-coef">Table 16.1: </span>Random intercept coefficients (b_0j) for exp5c_m1. For each donor, the random intercept is the sum of the fixed intercept (b_0) and the random intercept coefficient.
</caption>
<thead>
<tr>
<th style="text-align:left;">
donor
</th>
<th style="text-align:right;">
random intercept
</th>
<th style="text-align:right;">
b_0
</th>
<th style="text-align:right;">
b_0j
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
donor_1
</td>
<td style="text-align:right;">
1.011
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.330
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_2
</td>
<td style="text-align:right;">
1.027
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.314
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_3
</td>
<td style="text-align:right;">
1.164
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.177
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_4
</td>
<td style="text-align:right;">
1.795
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
0.454
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_5
</td>
<td style="text-align:right;">
1.753
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
0.412
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_6
</td>
<td style="text-align:right;">
1.295
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.046
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The random intercept coefficient <span class="math inline">\(b_0j\)</span> is the difference between the modeled mean for donor <em>j</em> and the mean of the reference treatment combination. Unlike the fixed intercept (<span class="math inline">\(b_0\)</span>), <span class="math inline">\(b_0j\)</span> is an effect.</li>
<li>Figure <a href="lmm.html#fig:lmm-exp1g-explainer-2">16.7</a> illustrates the random intercept coefficients <span class="math inline">\(b_0j\)</span> for model <code>exp5c_m1</code>.</li>
<li>The random intercept coefficients are often (but not always) treated as a source of nuisance variation – that is, the coefficients are not generally of interest and the values are not typically reported.</li>
</ol>
<div class="figure"><span id="fig:lmm-exp5c-explainer-2"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-explainer-2-1.png" alt="What random intercepts are. The large, colored dots are the modeled mean for each treatment (genotype by sex) by experiment_id combination. The dashed grey line is the fixed intercept -- it is the modeled mean of the reference level (&quot;GPR174+ M&quot;). The large, colored dots at the reference level are the random intercepts. The value of each random intercept is the sum of the fixed intercept and the random intercept effect for that experiment_id. Each random intercept effect is the residual from (or, distance from colored dot to) the dashed, gray line, shown by the vertical, colored lines" width="672" />
<p class="caption">
Figure 16.5: What random intercepts are. The large, colored dots are the modeled mean for each treatment (genotype by sex) by experiment_id combination. The dashed grey line is the fixed intercept – it is the modeled mean of the reference level (“GPR174+ M”). The large, colored dots at the reference level are the random intercepts. The value of each random intercept is the sum of the fixed intercept and the random intercept effect for that experiment_id. Each random intercept effect is the residual from (or, distance from colored dot to) the dashed, gray line, shown by the vertical, colored lines
</p>
</div>
</div>
<div id="the-random-and-residual-variance-and-the-intraclass-correlation" class="section level4" number="16.3.3.3">
<h4><span class="header-section-number">16.3.3.3</span> The random and residual variance and the intraclass correlation</h4>
<pre><code>##  Groups   Name        Std.Dev.
##  donor    (Intercept) 0.35587 
##  Residual             0.11647</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The first element in “Std.Dev.” is the estimate of <span class="math inline">\(\sqrt{\sigma^2_{0j}}\)</span>, the standard deviation among the donors due to the random effect of <span class="math inline">\(\textt{donor}\)</span>.</li>
<li>The second element in “Std.Dev.” is <span class="math inline">\(\sqrt{\sigma^2_{0}}\)</span>, the estimate of the standard deviation of the error variance (<span class="math inline">\(\varepsilon^2\)</span>). Remember that the residuals of the model are the estimates of <span class="math inline">\(\varepsilon\)</span>.</li>
<li>The ratio <span class="math inline">\(\frac{\sigma^2_{0j}}{\sigma^2_{0j} + \sigma^2_{0}}\)</span>, the ratio of the among-block to total random variance, is known as the <a href="https://en.wikipedia.org/wiki/Intraclass_correlation" target="_blank">intraclass correlation</a>. This correlation is an estimate of the correlated error due to the by-donor clustering if the model were fit without the added random intercept. For <code>exp5c_m1</code>, this correlation is 0.9.</li>
<li>The intraclass correlation ranges between 0 and 1 and makes a pretty good qualitative indicator of repeatability.</li>
</ol>
</div>
</div>
<div id="alternative-models" class="section level3" number="16.3.4">
<h3><span class="header-section-number">16.3.4</span> Alternative models</h3>
<p>Linear mixed models are very flexible, a topic which is too advanced for this text. Here I want to focus on two alternative models because of their relevance to <strong>repeated measures ANOVA</strong>.</p>
<div class="sourceCode" id="cb539"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb539-1"><a href="lmm.html#cb539-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb539-2"><a href="lmm.html#cb539-2" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb539-3"><a href="lmm.html#cb539-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb539-4"><a href="lmm.html#cb539-4" aria-hidden="true" tabindex="-1"></a>exp5c_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb539-5"><a href="lmm.html#cb539-5" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb539-6"><a href="lmm.html#cb539-6" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment) <span class="sc">+</span></span>
<span id="cb539-7"><a href="lmm.html#cb539-7" aria-hidden="true" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>activity),</span>
<span id="cb539-8"><a href="lmm.html#cb539-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span></code></pre></div>
<pre><code>## boundary (singular) fit: see ?isSingular</code></pre>
<div class="sourceCode" id="cb541"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb541-1"><a href="lmm.html#cb541-1" aria-hidden="true" tabindex="-1"></a>exp5c_m1c <span class="ot">&lt;-</span> <span class="fu">lme</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb541-2"><a href="lmm.html#cb541-2" aria-hidden="true" tabindex="-1"></a>                 <span class="at">random =</span>  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor,</span>
<span id="cb541-3"><a href="lmm.html#cb541-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb541-4"><a href="lmm.html#cb541-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Three models are fit. The models specify the same fixed effects but different random effects. I’ll return to inference and the problem of “which model to choose” in a moment. First, how do these models differ?</li>
<li>Model <code>exp5c_m1a</code> is the same model analyzed and explained above. This model specifies a random intercept for each level of <span class="math inline">\(\texttt{donor}\)</span>.</li>
<li>Model <code>exp5c_m1b</code> adds two additional random intercepts to Model <code>exp5c_m1</code>. The code <code>(1 | donor:treatment)</code> adds an intercept for each combination of <span class="math inline">\(\texttt{donor}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>. The code <code>(1 | donor:activity)</code> adds an intercept for each combination of <span class="math inline">\(\texttt{donor}\)</span> and <span class="math inline">\(\texttt{activity}\)</span>. This model is the equivalent of the univariate model of a <strong>repeated measures ANOVA</strong> of these data (see <a href="lmm.html#lmm-rmanova">Classical (“univariate model”) repeated measures ANOVA</a> below).</li>
<li>Model <code>exp5c_m1c</code> has the same random effects as <code>exp5c_m1</code> but adds the argument <code>correlation = corSymm(form = ~ 1 | donor)</code>. Model <code>exp5c_m1</code> assumes the correlated error for all pairs of treatment combinations is the same (see <a href="lmm.html#lmm-exp5c-cor">Correlated error</a> above) – one component of a condition called <strong>compound symmetry</strong>. Model <code>exp5c_m1c</code> makes no assumption about the structure of these correlations (an <strong>unstructured</strong> covariance matrix). The trade-off is the estimation of more parameters and the potential of <a href="https://en.wikipedia.org/wiki/Overfitting" target="_blank">overfitting</a>. This model is similar to the multivariate model of a <strong>repeated measures ANOVA</strong> of these data (see <a href="lmm.html#lmm-rmanova-multi">“Multivariate model” repeated measures ANOVA</a> below).</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-alternatives-planned">Table 16.2: </span>Model exp5c_m1a planned comparisons.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
2.25
</td>
<td style="text-align:right;">
0.040
</td>
</tr>
<tr>
<td style="text-align:left;">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.26
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
-1.73
</td>
<td style="text-align:right;">
0.105
</td>
</tr>
<tr>
<td style="text-align:left;">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.47
</td>
<td style="text-align:right;">
-0.06
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
0.013
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-alternatives-planned">Table 16.2: </span>Model exp5c_m1b planned comparisons.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.016
</td>
</tr>
<tr>
<td style="text-align:left;">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
-2.22
</td>
<td style="text-align:right;">
0.051
</td>
</tr>
<tr>
<td style="text-align:left;">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.46
</td>
<td style="text-align:right;">
-0.08
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.015
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-alternatives-planned">Table 16.2: </span>Model exp5c_m1c planned comparisons.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.07
</td>
<td style="text-align:right;">
0.24
</td>
<td style="text-align:right;">
3.81
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
<tr>
<td style="text-align:left;">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.061
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.25
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
-1.89
</td>
<td style="text-align:right;">
0.078
</td>
</tr>
<tr>
<td style="text-align:left;">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.079
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.44
</td>
<td style="text-align:right;">
-0.10
</td>
<td style="text-align:right;">
-3.37
</td>
<td style="text-align:right;">
0.004
</td>
</tr>
</tbody>
</table>
<ol start="5" style="list-style-type: decimal">
<li>Planned comparisons of the three alternative models are given in the tables above. The effect estimates are the same but inference differs slightly among the models because of how each model partitions error variance to either random effects or the residuals. Which model do we report? One way to evaluate the models is a statistic known as the AIC.</li>
</ol>
<div class="sourceCode" id="cb542"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb542-1"><a href="lmm.html#cb542-1" aria-hidden="true" tabindex="-1"></a><span class="fu">AIC</span>(exp5c_m1a)</span></code></pre></div>
<pre><code>## [1] 8.153813</code></pre>
<div class="sourceCode" id="cb544"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb544-1"><a href="lmm.html#cb544-1" aria-hidden="true" tabindex="-1"></a><span class="fu">AIC</span>(exp5c_m1b)</span></code></pre></div>
<pre><code>## [1] 10.05268</code></pre>
<div class="sourceCode" id="cb546"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb546-1"><a href="lmm.html#cb546-1" aria-hidden="true" tabindex="-1"></a><span class="fu">AIC</span>(exp5c_m1c)</span></code></pre></div>
<pre><code>## [1] 9.393386</code></pre>
<ol start="6" style="list-style-type: decimal">
<li>AIC is the <a href="https://en.wikipedia.org/wiki/Akaike_information_criterion" target="_blank">Akaike Information Criterion</a> is a relative measure of model quality. Compare this to <span class="math inline">\(R^2\)</span>, which is an absolute measure of goodness of fit. The AIC formula has two parts, one is a goodness of fit (like <span class="math inline">\(R^2\)</span>) and the other is a penalty based on the number of parameters in the model. As the goodness of fit increases, the AIC goes down. As the number of parameters increases, the AIC goes up. The model with the lowest AIC is the highest quality model. The actual number, however does not have any absolute meaning; it is only meaningful relative to the AICs computed from fits to different models using the same data. The AICs of the two models suggest that models <code>exp5c_m1b</code> and <code>exp5c_m1c</code> are too complex given the data. This has relevance for the repeated measures ANOVA analysis in the next section.</li>
<li><code>lmer</code> returns a message “boundary (singular) fit: see ?isSingular” for <code>exp5c_m1b</code>. The model <em>was</em> fit but the message means that we should be cautious about(or simply avoid) interpreting any inferential statistics (SEs, CIs, <em>p</em>-values). Looking at the estimates of the estimated parameters of the random effects in the table below (the variances of the effects and the covariances among the effects), Model <code>exp5c_m1b</code> estimates zero variance for the <span class="math inline">\(\texttt{donor:activity}\)</span> random intercept. This suggests that we could simplify <code>exp5c_m1b</code> by removing <code>(1 | donor:activity)</code> from the model. Think about this. If we fit using repeated measures ANOVA (next section), we won’t get a warning message but we still may be <a href="https://en.wikipedia.org/wiki/Overfitting" target="_blank">overfitting</a>.</li>
</ol>
<div class="sourceCode" id="cb548"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb548-1"><a href="lmm.html#cb548-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(exp5c_m1b)</span></code></pre></div>
<pre><code>##  Groups          Name        Std.Dev.
##  donor:activity  (Intercept) 0.000000
##  donor:treatment (Intercept) 0.089519
##  donor           (Intercept) 0.352092
##  Residual                    0.090686</code></pre>
</div>
<div id="paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models" class="section level3" number="16.3.5">
<h3><span class="header-section-number">16.3.5</span> Paired t-tests and repeated measures ANOVA are special cases of linear mixed models</h3>
<p>A paired <em>t</em>-test is a special case of linear mixed model fit to data from a randomized complete block design with no subsampling and only two treatment levels (see <a href="violations.html#oneway-paired-t">Lack of independence</a> in the Violations chapter). A repeated measures ANOVA is a special case of linear mixed model fit to data from a randomized complete block design with no subsampling and more than two treatment levels (see <a href="violations.html#oneway-paired-t">Lack of independence</a> in the Violations chapter).</p>
<p>Experiment 5c is a randomized complete block design with no subsampling and four treatment combinations, but the treatment combinations are the <span class="math inline">\(2 \times 2\)</span> combinations of the two factors <span class="math inline">\(\texttt{treatment}\)</span> and <span class="math inline">\(\texttt{activity}\)</span> instead of the <span class="math inline">\(4 \times 1\)</span> combinations of a single treatment factor with four levels.</p>
<p>The repeated measures ANOVA is computed using the <code>afex::aov_4</code> function.</p>
<div id="lmm-rmanova" class="section level4" number="16.3.5.1">
<h4><span class="header-section-number">16.3.5.1</span> Classical (“univariate model”) repeated measures ANOVA</h4>
<div class="sourceCode" id="cb550"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb550-1"><a href="lmm.html#cb550-1" aria-hidden="true" tabindex="-1"></a>exp5c_aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb550-2"><a href="lmm.html#cb550-2" aria-hidden="true" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb550-3"><a href="lmm.html#cb550-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> exp5c)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
siNR4A3 Basal - Scr Basal
</td>
<td style="text-align:right;">
0.081
</td>
<td style="text-align:right;">
0.076
</td>
<td style="text-align:right;">
8.6
</td>
<td style="text-align:right;">
-0.092
</td>
<td style="text-align:right;">
0.254
</td>
<td style="text-align:right;">
1.1
</td>
<td style="text-align:right;">
0.3132
</td>
</tr>
<tr>
<td style="text-align:left;">
Scr EPS - Scr Basal
</td>
<td style="text-align:right;">
0.151
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
9.4
</td>
<td style="text-align:right;">
0.033
</td>
<td style="text-align:right;">
0.269
</td>
<td style="text-align:right;">
2.9
</td>
<td style="text-align:right;">
0.0173
</td>
</tr>
<tr>
<td style="text-align:left;">
Scr EPS - siNR4A3 Basal
</td>
<td style="text-align:right;">
0.070
</td>
<td style="text-align:right;">
0.071
</td>
<td style="text-align:right;">
7.4
</td>
<td style="text-align:right;">
-0.097
</td>
<td style="text-align:right;">
0.236
</td>
<td style="text-align:right;">
1.0
</td>
<td style="text-align:right;">
0.3583
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3 EPS - Scr Basal
</td>
<td style="text-align:right;">
-0.035
</td>
<td style="text-align:right;">
0.071
</td>
<td style="text-align:right;">
7.4
</td>
<td style="text-align:right;">
-0.201
</td>
<td style="text-align:right;">
0.132
</td>
<td style="text-align:right;">
-0.5
</td>
<td style="text-align:right;">
0.6387
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3 EPS - siNR4A3 Basal
</td>
<td style="text-align:right;">
-0.116
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
9.4
</td>
<td style="text-align:right;">
-0.234
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
-2.2
</td>
<td style="text-align:right;">
0.0525
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3 EPS - Scr EPS
</td>
<td style="text-align:right;">
-0.186
</td>
<td style="text-align:right;">
0.076
</td>
<td style="text-align:right;">
8.6
</td>
<td style="text-align:right;">
-0.359
</td>
<td style="text-align:right;">
-0.013
</td>
<td style="text-align:right;">
-2.4
</td>
<td style="text-align:right;">
0.0380
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>In addition to the assumptions of the linear model outlined in the Violations chapter, the classical (“univariate model”) repeated measures ANOVA assumes <a href="https://en.wikipedia.org/wiki/Mauchly%27s_sphericity_test#Sphericity" target="_blank">sphericity</a>, which is the equality of the variances of all pairwise differences among treatment combinations.</li>
<li>The linear model that is equivalent to the univariate model of the classic RM ANOVA is <code>exp5c_m1b</code> from above. Here are all pairwise estimated effects of Model <code>exp5c_m1b</code>.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
siNR4A3 Basal - Scr Basal
</td>
<td style="text-align:right;">
0.081
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
-0.088
</td>
<td style="text-align:right;">
0.251
</td>
<td style="text-align:right;">
1.1
</td>
<td style="text-align:right;">
0.3010
</td>
</tr>
<tr>
<td style="text-align:left;">
Scr EPS - Scr Basal
</td>
<td style="text-align:right;">
0.151
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.034
</td>
<td style="text-align:right;">
0.268
</td>
<td style="text-align:right;">
2.9
</td>
<td style="text-align:right;">
0.0163
</td>
</tr>
<tr>
<td style="text-align:left;">
Scr EPS - siNR4A3 Basal
</td>
<td style="text-align:right;">
0.070
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
-0.100
</td>
<td style="text-align:right;">
0.239
</td>
<td style="text-align:right;">
0.9
</td>
<td style="text-align:right;">
0.3711
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3 EPS - Scr Basal
</td>
<td style="text-align:right;">
-0.035
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
-0.204
</td>
<td style="text-align:right;">
0.135
</td>
<td style="text-align:right;">
-0.5
</td>
<td style="text-align:right;">
0.6485
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3 EPS - siNR4A3 Basal
</td>
<td style="text-align:right;">
-0.116
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.233
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
-2.2
</td>
<td style="text-align:right;">
0.0509
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3 EPS - Scr EPS
</td>
<td style="text-align:right;">
-0.186
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
-0.355
</td>
<td style="text-align:right;">
-0.016
</td>
<td style="text-align:right;">
-2.5
</td>
<td style="text-align:right;">
0.0353
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The classical (univariate) repeated measures ANOVA model and the linear mixed model <code>exp5c_m1b</code> <em>are</em> equivalent but the inference (SEs, CIs, <em>p</em>-values) differ slightly in this example. This is because of the estimated variance for the <span class="math inline">\(\texttt{donor:activity}\)</span> random intercept is zero. In cases where all variances are greater than zero <em>and the design is balanced</em> (no block is missing a treatment combination), the two tables would be identical.</li>
</ol>
</div>
<div id="lmm-rmanova-multi" class="section level4" number="16.3.5.2">
<h4><span class="header-section-number">16.3.5.2</span> “Multivariate model” repeated measures ANOVA</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
siNR4A3 Basal - Scr Basal
</td>
<td style="text-align:right;">
0.081
</td>
<td style="text-align:right;">
0.101
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.179
</td>
<td style="text-align:right;">
0.341
</td>
<td style="text-align:right;">
0.80
</td>
<td style="text-align:right;">
0.4581
</td>
</tr>
<tr>
<td style="text-align:left;">
Scr EPS - Scr Basal
</td>
<td style="text-align:right;">
0.151
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.049
</td>
<td style="text-align:right;">
0.253
</td>
<td style="text-align:right;">
3.82
</td>
<td style="text-align:right;">
0.0124
</td>
</tr>
<tr>
<td style="text-align:left;">
Scr EPS - siNR4A3 Basal
</td>
<td style="text-align:right;">
0.070
</td>
<td style="text-align:right;">
0.085
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.149
</td>
<td style="text-align:right;">
0.288
</td>
<td style="text-align:right;">
0.82
</td>
<td style="text-align:right;">
0.4501
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3 EPS - Scr Basal
</td>
<td style="text-align:right;">
-0.035
</td>
<td style="text-align:right;">
0.054
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.173
</td>
<td style="text-align:right;">
0.103
</td>
<td style="text-align:right;">
-0.65
</td>
<td style="text-align:right;">
0.5450
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3 EPS - siNR4A3 Basal
</td>
<td style="text-align:right;">
-0.116
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.277
</td>
<td style="text-align:right;">
0.045
</td>
<td style="text-align:right;">
-1.85
</td>
<td style="text-align:right;">
0.1228
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3 EPS - Scr EPS
</td>
<td style="text-align:right;">
-0.186
</td>
<td style="text-align:right;">
0.036
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.278
</td>
<td style="text-align:right;">
-0.094
</td>
<td style="text-align:right;">
-5.18
</td>
<td style="text-align:right;">
0.0035
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The sphericity assumption is relaxed if the model is fit using the “multivariate model.”</li>
<li>The multivariate model repeated measures ANOVA is a linear model with a multivariate response and not a linear mixed model. In the multivariate model, each treatment combination is a different response variable and there is a single row for each level of the random factor (<span class="math inline">\(\texttt{donor}\)</span> in Experiment 5c). The multivariate model is a different way of handling the correlated error that occurs when conceiving of the design as univariate.</li>
<li><code>exp5c_m1c</code> is a linear mixed model that is similar to the multivariate model of the repeated measures ANOVA. Here are all pairwise estimated effects of Model <code>exp5c_m1c</code>.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
siNR4A3 Basal - Scr Basal
</td>
<td style="text-align:right;">
0.081
</td>
<td style="text-align:right;">
0.098
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.129
</td>
<td style="text-align:right;">
0.291
</td>
<td style="text-align:right;">
0.83
</td>
<td style="text-align:right;">
0.4219
</td>
</tr>
<tr>
<td style="text-align:left;">
Scr EPS - Scr Basal
</td>
<td style="text-align:right;">
0.151
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.066
</td>
<td style="text-align:right;">
0.236
</td>
<td style="text-align:right;">
3.81
</td>
<td style="text-align:right;">
0.0017
</td>
</tr>
<tr>
<td style="text-align:left;">
Scr EPS - siNR4A3 Basal
</td>
<td style="text-align:right;">
0.070
</td>
<td style="text-align:right;">
0.085
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.111
</td>
<td style="text-align:right;">
0.251
</td>
<td style="text-align:right;">
0.82
</td>
<td style="text-align:right;">
0.4246
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3 EPS - Scr Basal
</td>
<td style="text-align:right;">
-0.035
</td>
<td style="text-align:right;">
0.053
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.149
</td>
<td style="text-align:right;">
0.079
</td>
<td style="text-align:right;">
-0.65
</td>
<td style="text-align:right;">
0.5240
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3 EPS - siNR4A3 Basal
</td>
<td style="text-align:right;">
-0.116
</td>
<td style="text-align:right;">
0.061
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.247
</td>
<td style="text-align:right;">
0.015
</td>
<td style="text-align:right;">
-1.89
</td>
<td style="text-align:right;">
0.0779
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3 EPS - Scr EPS
</td>
<td style="text-align:right;">
-0.186
</td>
<td style="text-align:right;">
0.036
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.263
</td>
<td style="text-align:right;">
-0.108
</td>
<td style="text-align:right;">
-5.12
</td>
<td style="text-align:right;">
0.0001
</td>
</tr>
</tbody>
</table>
</div>
<div id="linear-mixed-models-vs-repeated-measures-anova" class="section level4" number="16.3.5.3">
<h4><span class="header-section-number">16.3.5.3</span> Linear mixed models vs repeated measures ANOVA</h4>
<p>Many modern textbooks encourage researchers to use linear mixed models instead of repeated measures ANOVA for randomized complete block designs (with or without subsampling) because</p>
<ol style="list-style-type: decimal">
<li>linear mixed models do not exclude random units (subject/mouse/donor/cage) with missing measures of one of the treatment combinations. <a href="violations.html#violations-rmanova">Example 2 (diHOME exp2a) – A repeated measures ANOVA is a special case of a linear mixed model</a> is an example of this kind of missing data.</li>
<li>if there is subsampling, linear mixed models do not <strong>aggregate</strong> the random-unit data, that is, linear mixed models do not simply compute the means and ignore the variance of the sample within each random unit.</li>
<li>linear mixed models allow for modeling additional sources of correlated error, that is, an experiment may have two or more random factor variables (for example, donor and experiment).</li>
<li>linear mixed models allow a researcher to model different patterns of correlated error. This is especially important in longitudinal experiments.</li>
<li>linear mixed models can be generalized to model sampling from non-normal distributions – these are generalized linear mixed models.</li>
</ol>
</div>
</div>
<div id="modeling-textttdonor-as-a-fixed-effect" class="section level3" number="16.3.6">
<h3><span class="header-section-number">16.3.6</span> Modeling <span class="math inline">\(\texttt{donor}\)</span> as a fixed effect</h3>
<p><span class="math display" id="eq:lmm-exp5c-m3">\[
\begin{equation}
\texttt{glucose_uptake} \sim \texttt{treatment * activity + donor}
\end{equation}
\tag{16.1}
\]</span></p>
<p>Model <a href="lmm.html#eq:lmm-exp5c-m3">(16.1)</a> (using R formula syntax) is a linear model with the block <span class="math inline">\(\texttt{donor}\)</span> added as a fixed covariate instead of a random intercept. The coefficients of the fit model are</p>
<div class="sourceCode" id="cb551"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb551-1"><a href="lmm.html#cb551-1" aria-hidden="true" tabindex="-1"></a>exp5c_m3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span> donor,</span>
<span id="cb551-2"><a href="lmm.html#cb551-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> exp5c)</span>
<span id="cb551-3"><a href="lmm.html#cb551-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb551-4"><a href="lmm.html#cb551-4" aria-hidden="true" tabindex="-1"></a>exp5c_m3_coef <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">coef</span>(<span class="fu">summary</span>(exp5c_m3)),</span>
<span id="cb551-5"><a href="lmm.html#cb551-5" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">confint</span>(exp5c_m3))</span>
<span id="cb551-6"><a href="lmm.html#cb551-6" aria-hidden="true" tabindex="-1"></a>exp5c_m3_coef <span class="sc">%&gt;%</span></span>
<span id="cb551-7"><a href="lmm.html#cb551-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb551-8"><a href="lmm.html#cb551-8" aria-hidden="true" tabindex="-1"></a>  kable_styling</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.071
</td>
<td style="text-align:right;">
14.0
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
0.85
</td>
<td style="text-align:right;">
1.15
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
1.2
</td>
<td style="text-align:right;">
0.2453
</td>
<td style="text-align:right;">
-0.06
</td>
<td style="text-align:right;">
0.22
</td>
</tr>
<tr>
<td style="text-align:left;">
activityEPS
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
2.2
</td>
<td style="text-align:right;">
0.0402
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.29
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_2
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:right;">
0.8444
</td>
<td style="text-align:right;">
-0.16
</td>
<td style="text-align:right;">
0.19
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_3
</td>
<td style="text-align:right;">
0.16
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
1.9
</td>
<td style="text-align:right;">
0.0757
</td>
<td style="text-align:right;">
-0.02
</td>
<td style="text-align:right;">
0.33
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_4
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
9.8
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
0.63
</td>
<td style="text-align:right;">
0.98
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_5
</td>
<td style="text-align:right;">
0.76
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
9.2
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
0.59
</td>
<td style="text-align:right;">
0.94
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_6
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
3.5
</td>
<td style="text-align:right;">
0.0030
</td>
<td style="text-align:right;">
0.12
</td>
<td style="text-align:right;">
0.47
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3:activityEPS
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
-2.8
</td>
<td style="text-align:right;">
0.0132
</td>
<td style="text-align:right;">
-0.47
</td>
<td style="text-align:right;">
-0.06
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The coefficients include a slope for the five non-reference levels of donor. We typically don’t care about these.</li>
<li>In a randomized complete block design with only 1 replicate of each treatment combination per block, like that in Experiment 5c, inference about treatment effects is exactly that same between this model and the random intercept model <code>exp5c_m1.</code> Compare the SE, CI and <em>p</em>-value for the coefficients of <code>treatmentsiNR4A3</code>, <code>activityEPS</code> and <code>treatmentsiNR4A3:activityEPS</code> to those from the linear mixed model.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
1.34
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
8.77
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
1.02
</td>
<td style="text-align:right;">
1.66
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15.0
</td>
<td style="text-align:right;">
1.21
</td>
<td style="text-align:right;">
0.2453
</td>
<td style="text-align:right;">
-0.05
</td>
<td style="text-align:right;">
0.21
</td>
</tr>
<tr>
<td style="text-align:left;">
activityEPS
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15.0
</td>
<td style="text-align:right;">
2.25
</td>
<td style="text-align:right;">
0.0402
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.28
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3:activityEPS
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
15.0
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
0.0132
</td>
<td style="text-align:right;">
-0.45
</td>
<td style="text-align:right;">
-0.09
</td>
</tr>
</tbody>
</table>
<ol start="3" style="list-style-type: decimal">
<li>In a randomized complete block design with only 1 replicate of each treatment combination per block, like that in Experiment 5c, inference about treatment means differs between this model and the random intercept model – the SE of the means of the fixed effect model are smaller than the SE of the means of the linear mixed model. Compare the SE and CI of the <code>(Intercept)</code> (the mean of the reference treatment combination) between the fixed effect and linear mixed model.</li>
<li>The equivalence of inference in the treatment effect between the fixed effect and linear mixed model holds only for balanced randomized complete block designs – where all blocks contain all treatment combinations and the subsampling replicate size is the same for all treatment combinations for all blocks. This means, outside of special cases like Experiment 5c, the choice between adding a blocking variable as a random or fixed factor depends on assumptions about the model.</li>
</ol>
</div>
</div>
<div id="example-2-experiment-1g-random-intercepts-and-slopes-experiments-with-more-than-one-replicate-of-each-treatment-per-block" class="section level2" number="16.4">
<h2><span class="header-section-number">16.4</span> Example 2 (Experiment 1g) – Random intercepts and slopes – experiments with more than one replicate of each treatment per block</h2>
<div id="understand-the-data-1" class="section level3" number="16.4.1">
<h3><span class="header-section-number">16.4.1</span> Understand the data</h3>
</div>
<div id="examine-the-data-2" class="section level3" number="16.4.2">
<h3><span class="header-section-number">16.4.2</span> Examine the data</h3>
<div class="sourceCode" id="cb552"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb552-1"><a href="lmm.html#cb552-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> exp1g,</span>
<span id="cb552-2"><a href="lmm.html#cb552-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb552-3"><a href="lmm.html#cb552-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> gc,</span>
<span id="cb552-4"><a href="lmm.html#cb552-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> experiment_id)) <span class="sc">+</span></span>
<span id="cb552-5"><a href="lmm.html#cb552-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.4</span>))</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp1g-examine-1.png" width="576" /></p>
</div>
<div id="fit-the-model-9" class="section level3" number="16.4.3">
<h3><span class="header-section-number">16.4.3</span> Fit the model</h3>
<div class="sourceCode" id="cb553"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb553-1"><a href="lmm.html#cb553-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb553-2"><a href="lmm.html#cb553-2" aria-hidden="true" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb553-3"><a href="lmm.html#cb553-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb553-4"><a href="lmm.html#cb553-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb553-5"><a href="lmm.html#cb553-5" aria-hidden="true" tabindex="-1"></a>exp1g_m2 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb553-6"><a href="lmm.html#cb553-6" aria-hidden="true" tabindex="-1"></a>                   (genotype <span class="sc">*</span> sex <span class="sc">|</span> experiment_id),</span>
<span id="cb553-7"><a href="lmm.html#cb553-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span></code></pre></div>
</div>
<div id="inference-from-the-model-5" class="section level3" number="16.4.4">
<h3><span class="header-section-number">16.4.4</span> Inference from the model</h3>
<div class="sourceCode" id="cb554"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb554-1"><a href="lmm.html#cb554-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1_coef <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(exp1g_m1))</span>
<span id="cb554-2"><a href="lmm.html#cb554-2" aria-hidden="true" tabindex="-1"></a>exp1g_m1_coef <span class="sc">%&gt;%</span></span>
<span id="cb554-3"><a href="lmm.html#cb554-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb554-4"><a href="lmm.html#cb554-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
8.443397
</td>
<td style="text-align:right;">
0.9964048
</td>
<td style="text-align:right;">
2.889370
</td>
<td style="text-align:right;">
8.473862
</td>
<td style="text-align:right;">
0.0039725
</td>
</tr>
<tr>
<td style="text-align:left;">
genotypeGpr174-
</td>
<td style="text-align:right;">
2.539656
</td>
<td style="text-align:right;">
0.7981940
</td>
<td style="text-align:right;">
4.740452
</td>
<td style="text-align:right;">
3.181753
</td>
<td style="text-align:right;">
0.0263828
</td>
</tr>
<tr>
<td style="text-align:left;">
sexF
</td>
<td style="text-align:right;">
5.607069
</td>
<td style="text-align:right;">
1.5884257
</td>
<td style="text-align:right;">
3.056924
</td>
<td style="text-align:right;">
3.529954
</td>
<td style="text-align:right;">
0.0375009
</td>
</tr>
<tr>
<td style="text-align:left;">
genotypeGpr174-:sexF
</td>
<td style="text-align:right;">
-5.301251
</td>
<td style="text-align:right;">
1.1325708
</td>
<td style="text-align:right;">
3.284669
</td>
<td style="text-align:right;">
-4.680723
</td>
<td style="text-align:right;">
0.0150149
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb555"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb555-1"><a href="lmm.html#cb555-1" aria-hidden="true" tabindex="-1"></a>exp1g_m2_coef <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(exp1g_m2))</span>
<span id="cb555-2"><a href="lmm.html#cb555-2" aria-hidden="true" tabindex="-1"></a>exp1g_m2_coef <span class="sc">%&gt;%</span></span>
<span id="cb555-3"><a href="lmm.html#cb555-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb555-4"><a href="lmm.html#cb555-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
8.442879
</td>
<td style="text-align:right;">
0.9963285
</td>
<td style="text-align:right;">
2.889096
</td>
<td style="text-align:right;">
8.473991
</td>
<td style="text-align:right;">
0.0039737
</td>
</tr>
<tr>
<td style="text-align:left;">
genotypeGpr174-
</td>
<td style="text-align:right;">
2.540958
</td>
<td style="text-align:right;">
0.7975946
</td>
<td style="text-align:right;">
4.742326
</td>
<td style="text-align:right;">
3.185776
</td>
<td style="text-align:right;">
0.0262539
</td>
</tr>
<tr>
<td style="text-align:left;">
sexF
</td>
<td style="text-align:right;">
5.607919
</td>
<td style="text-align:right;">
1.5881335
</td>
<td style="text-align:right;">
3.056336
</td>
<td style="text-align:right;">
3.531138
</td>
<td style="text-align:right;">
0.0374801
</td>
</tr>
<tr>
<td style="text-align:left;">
genotypeGpr174-:sexF
</td>
<td style="text-align:right;">
-5.303019
</td>
<td style="text-align:right;">
1.1251652
</td>
<td style="text-align:right;">
3.306757
</td>
<td style="text-align:right;">
-4.713103
</td>
<td style="text-align:right;">
0.0144880
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb556"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb556-1"><a href="lmm.html#cb556-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1_varcor <span class="ot">&lt;-</span> <span class="fu">VarCorr</span>(exp1g_m1)</span>
<span id="cb556-2"><a href="lmm.html#cb556-2" aria-hidden="true" tabindex="-1"></a>exp1g_m2_varcor <span class="ot">&lt;-</span> <span class="fu">VarCorr</span>(exp1g_m2)</span>
<span id="cb556-3"><a href="lmm.html#cb556-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb556-4"><a href="lmm.html#cb556-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb556-5"><a href="lmm.html#cb556-5" aria-hidden="true" tabindex="-1"></a>exp1g_m1_varcor</span></code></pre></div>
<pre><code>##  Groups        Name               Std.Dev. Corr                
##  experiment_id (Intercept)        1.77104                      
##                treatmentGpr174- M 0.96403  -0.025              
##                treatmentGpr174+ F 2.90225   0.458  0.876       
##                treatmentGpr174- F 2.99053   0.300  0.879  0.938
##  Residual                         1.93135</code></pre>
<div class="sourceCode" id="cb558"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb558-1"><a href="lmm.html#cb558-1" aria-hidden="true" tabindex="-1"></a>exp1g_m2_varcor</span></code></pre></div>
<pre><code>##  Groups        Name                 Std.Dev. Corr                
##  experiment_id (Intercept)          1.77082                      
##                genotypeGpr174-      0.96192  -0.023              
##                sexF                 2.90154   0.459  0.878       
##                genotypeGpr174-:sexF 1.33169  -0.314 -0.625 -0.706
##  Residual                           1.93155</code></pre>
<div class="sourceCode" id="cb560"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb560-1"><a href="lmm.html#cb560-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp1g_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;genotype&quot;</span>))</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>the two factors passed to <code>specs</code> are reversed from the fit model so that <span class="math inline">\(\texttt{sex}\)</span> is ploted on the x-axis below. This ordering has no effect on inference.</li>
</ol>
<div class="sourceCode" id="cb561"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb561-1"><a href="lmm.html#cb561-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exp1g_m1_emm # print in console to get row numbers</span></span>
<span id="cb561-2"><a href="lmm.html#cb561-2" aria-hidden="true" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb561-3"><a href="lmm.html#cb561-3" aria-hidden="true" tabindex="-1"></a>wt_m <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb561-4"><a href="lmm.html#cb561-4" aria-hidden="true" tabindex="-1"></a>wt_f <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb561-5"><a href="lmm.html#cb561-5" aria-hidden="true" tabindex="-1"></a>ko_m <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb561-6"><a href="lmm.html#cb561-6" aria-hidden="true" tabindex="-1"></a>ko_f <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb561-7"><a href="lmm.html#cb561-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb561-8"><a href="lmm.html#cb561-8" aria-hidden="true" tabindex="-1"></a><span class="co"># simple effects within males and females + interaction </span></span>
<span id="cb561-9"><a href="lmm.html#cb561-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. (ko_m - wt_m) </span></span>
<span id="cb561-10"><a href="lmm.html#cb561-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. (ko_f - wt_f)</span></span>
<span id="cb561-11"><a href="lmm.html#cb561-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb561-12"><a href="lmm.html#cb561-12" aria-hidden="true" tabindex="-1"></a>exp1g_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp1g_m1_emm,</span>
<span id="cb561-13"><a href="lmm.html#cb561-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb561-14"><a href="lmm.html#cb561-14" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;KO/M - WT/M&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(ko_m <span class="sc">-</span> wt_m),</span>
<span id="cb561-15"><a href="lmm.html#cb561-15" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;KO/F - WT/F&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(ko_f <span class="sc">-</span> wt_f),</span>
<span id="cb561-16"><a href="lmm.html#cb561-16" aria-hidden="true" tabindex="-1"></a>                         <span class="st">&quot;Interaction&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(ko_f <span class="sc">-</span> wt_f) <span class="sc">-</span></span>
<span id="cb561-17"><a href="lmm.html#cb561-17" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">c</span>(ko_m <span class="sc">-</span> wt_m)</span>
<span id="cb561-18"><a href="lmm.html#cb561-18" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb561-19"><a href="lmm.html#cb561-19" aria-hidden="true" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb561-20"><a href="lmm.html#cb561-20" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb561-21"><a href="lmm.html#cb561-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb561-22"><a href="lmm.html#cb561-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb561-23"><a href="lmm.html#cb561-23" aria-hidden="true" tabindex="-1"></a>exp1g_m1_planned <span class="sc">%&gt;%</span></span>
<span id="cb561-24"><a href="lmm.html#cb561-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb561-25"><a href="lmm.html#cb561-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
KO/M - WT/M
</td>
<td style="text-align:right;">
2.54
</td>
<td style="text-align:right;">
0.804
</td>
<td style="text-align:right;">
2.978
</td>
<td style="text-align:right;">
-0.03
</td>
<td style="text-align:right;">
5.11
</td>
<td style="text-align:right;">
3.16
</td>
<td style="text-align:right;">
0.051
</td>
</tr>
<tr>
<td style="text-align:left;">
KO/F - WT/F
</td>
<td style="text-align:right;">
-2.76
</td>
<td style="text-align:right;">
0.833
</td>
<td style="text-align:right;">
2.985
</td>
<td style="text-align:right;">
-5.42
</td>
<td style="text-align:right;">
-0.10
</td>
<td style="text-align:right;">
-3.32
</td>
<td style="text-align:right;">
0.046
</td>
</tr>
<tr>
<td style="text-align:left;">
Interaction
</td>
<td style="text-align:right;">
-5.30
</td>
<td style="text-align:right;">
1.140
</td>
<td style="text-align:right;">
2.982
</td>
<td style="text-align:right;">
-8.94
</td>
<td style="text-align:right;">
-1.66
</td>
<td style="text-align:right;">
-4.65
</td>
<td style="text-align:right;">
0.019
</td>
</tr>
</tbody>
</table>
</div>
<div id="plot-the-model-5" class="section level3" number="16.4.5">
<h3><span class="header-section-number">16.4.5</span> Plot the model</h3>
<div class="sourceCode" id="cb562"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb562-1"><a href="lmm.html#cb562-1" aria-hidden="true" tabindex="-1"></a>gg <span class="ot">&lt;-</span> <span class="fu">ggplot_the_model</span>(exp1g_m1,</span>
<span id="cb562-2"><a href="lmm.html#cb562-2" aria-hidden="true" tabindex="-1"></a>                       exp1g_m1_emm,</span>
<span id="cb562-3"><a href="lmm.html#cb562-3" aria-hidden="true" tabindex="-1"></a>                       exp1g_m1_planned,</span>
<span id="cb562-4"><a href="lmm.html#cb562-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">y_label =</span> <span class="st">&quot;Germinal Center (%)&quot;</span>,</span>
<span id="cb562-5"><a href="lmm.html#cb562-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">effect_label =</span> <span class="st">&quot;Effect (%)&quot;</span>,</span>
<span id="cb562-6"><a href="lmm.html#cb562-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">palette =</span> pal_okabe_ito_blue,</span>
<span id="cb562-7"><a href="lmm.html#cb562-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="fl">0.5</span>,<span class="dv">1</span>))</span>
<span id="cb562-8"><a href="lmm.html#cb562-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb562-9"><a href="lmm.html#cb562-9" aria-hidden="true" tabindex="-1"></a>gg</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp1g-plot-the-model-1.png" width="576" /></p>
</div>
<div id="alternaplot-the-model-1" class="section level3" number="16.4.6">
<h3><span class="header-section-number">16.4.6</span> Alternaplot the model</h3>
<div class="sourceCode" id="cb563"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb563-1"><a href="lmm.html#cb563-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1_pairs <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp1g_m1_emm,</span>
<span id="cb563-2"><a href="lmm.html#cb563-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb563-3"><a href="lmm.html#cb563-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb563-4"><a href="lmm.html#cb563-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb563-5"><a href="lmm.html#cb563-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb563-6"><a href="lmm.html#cb563-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb563-7"><a href="lmm.html#cb563-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb563-8"><a href="lmm.html#cb563-8" aria-hidden="true" tabindex="-1"></a><span class="co"># get coefficients of model</span></span>
<span id="cb563-9"><a href="lmm.html#cb563-9" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(exp1g_m1))[, <span class="st">&quot;Estimate&quot;</span>]</span>
<span id="cb563-10"><a href="lmm.html#cb563-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb563-11"><a href="lmm.html#cb563-11" aria-hidden="true" tabindex="-1"></a><span class="co"># get interaction p</span></span>
<span id="cb563-12"><a href="lmm.html#cb563-12" aria-hidden="true" tabindex="-1"></a>p_ixn <span class="ot">&lt;-</span> exp1g_m1_planned[<span class="dv">3</span>, <span class="st">&quot;p.value&quot;</span>] <span class="co"># check!</span></span>
<span id="cb563-13"><a href="lmm.html#cb563-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb563-14"><a href="lmm.html#cb563-14" aria-hidden="true" tabindex="-1"></a>dodge_width <span class="ot">&lt;-</span> <span class="fl">0.4</span></span>
<span id="cb563-15"><a href="lmm.html#cb563-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb563-16"><a href="lmm.html#cb563-16" aria-hidden="true" tabindex="-1"></a>gg <span class="ot">&lt;-</span> <span class="fu">ggplot_the_response</span>(</span>
<span id="cb563-17"><a href="lmm.html#cb563-17" aria-hidden="true" tabindex="-1"></a>  exp1g_m1,</span>
<span id="cb563-18"><a href="lmm.html#cb563-18" aria-hidden="true" tabindex="-1"></a>  exp1g_m1_emm,</span>
<span id="cb563-19"><a href="lmm.html#cb563-19" aria-hidden="true" tabindex="-1"></a>  exp1g_m1_pairs[<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">4</span>),], <span class="co"># only for comparisons within sex</span></span>
<span id="cb563-20"><a href="lmm.html#cb563-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">palette =</span> pal_okabe_ito_blue,</span>
<span id="cb563-21"><a href="lmm.html#cb563-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend_position =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb563-22"><a href="lmm.html#cb563-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_label =</span> <span class="st">&quot;Germinal Center (%)&quot;</span></span>
<span id="cb563-23"><a href="lmm.html#cb563-23" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb563-24"><a href="lmm.html#cb563-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">x =</span> <span class="dv">2</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="dv">2</span> <span class="sc">-</span> <span class="fl">0.1</span>,</span>
<span id="cb563-25"><a href="lmm.html#cb563-25" aria-hidden="true" tabindex="-1"></a>               <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb563-26"><a href="lmm.html#cb563-26" aria-hidden="true" tabindex="-1"></a>               <span class="at">xend =</span> <span class="dv">2</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="dv">2</span> <span class="sc">+</span> <span class="fl">0.1</span>,</span>
<span id="cb563-27"><a href="lmm.html#cb563-27" aria-hidden="true" tabindex="-1"></a>               <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb563-28"><a href="lmm.html#cb563-28" aria-hidden="true" tabindex="-1"></a>               <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb563-29"><a href="lmm.html#cb563-29" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;gray&quot;</span>) <span class="sc">+</span></span>
<span id="cb563-30"><a href="lmm.html#cb563-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bracket</span>(</span>
<span id="cb563-31"><a href="lmm.html#cb563-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fl">2.35</span>,</span>
<span id="cb563-32"><a href="lmm.html#cb563-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb563-33"><a href="lmm.html#cb563-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>] <span class="sc">+</span> b[<span class="dv">4</span>],</span>
<span id="cb563-34"><a href="lmm.html#cb563-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;ixn p = &quot;</span>,</span>
<span id="cb563-35"><a href="lmm.html#cb563-35" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">fmt_p_value_rmd</span>(p_ixn)),</span>
<span id="cb563-36"><a href="lmm.html#cb563-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.size =</span> <span class="dv">3</span>,</span>
<span id="cb563-37"><a href="lmm.html#cb563-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">text.hjust =</span> <span class="dv">0</span>,</span>
<span id="cb563-38"><a href="lmm.html#cb563-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb563-39"><a href="lmm.html#cb563-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb563-40"><a href="lmm.html#cb563-40" aria-hidden="true" tabindex="-1"></a>gg</span></code></pre></div>
<div class="figure"><span id="fig:lmm-exp1g-alternaplot"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp1g-alternaplot-1.png" alt="Dashed gray line is expected additive mean of Gpr174- F" width="576" />
<p class="caption">
Figure 16.6: Dashed gray line is expected additive mean of Gpr174- F
</p>
</div>
</div>
</div>
<div id="understanding-linear-mixed-models-2" class="section level2" number="16.5">
<h2><span class="header-section-number">16.5</span> Understanding linear mixed models 2 –</h2>
<div id="the-problem" class="section level3" number="16.5.1">
<h3><span class="header-section-number">16.5.1</span> The problem</h3>
<p>show picture of variation not accounted for by random intercept. This adds correlated error. This increases unmodeled error variance. Two basic models</p>
<p>random intercept + slope
random main intercepts plus random interaction intercepts</p>
</div>
<div id="understanding-the-model-formula" class="section level3" number="16.5.2">
<h3><span class="header-section-number">16.5.2</span> Understanding the model formula</h3>
<div class="sourceCode" id="cb564"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb564-1"><a href="lmm.html#cb564-1" aria-hidden="true" tabindex="-1"></a>exp1g_m1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb564-2"><a href="lmm.html#cb564-2" aria-hidden="true" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb564-3"><a href="lmm.html#cb564-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb564-4"><a href="lmm.html#cb564-4" aria-hidden="true" tabindex="-1"></a>exp1g_m2 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb564-5"><a href="lmm.html#cb564-5" aria-hidden="true" tabindex="-1"></a>                   (genotype <span class="sc">*</span> sex <span class="sc">|</span> experiment_id),</span>
<span id="cb564-6"><a href="lmm.html#cb564-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The design is randomized complete block with subsampling. This means that we have multiple replicates of all combinations of <span class="math inline">\(\texttt{genotype} \times \texttt{sex}\)</span> in each block (<span class="math inline">\(\texttt{experiment_id}\)</span>).</li>
</ol>
</div>
<div id="the-random-coefficients" class="section level3" number="16.5.3">
<h3><span class="header-section-number">16.5.3</span> The random coefficients</h3>
</div>
<div id="linear-mixed-models-have-parameters-for-random-intercept-effects-and-random-slope-effects" class="section level3" number="16.5.4">
<h3><span class="header-section-number">16.5.4</span> Linear mixed models have parameters for random intercept effects and random slope effects</h3>
<p>The linear mixed model fit to <code>exp1g</code> data is</p>
<p><span class="math display">\[
\begin{align}
\texttt{gc}_j = \ &amp;\beta_0 + \beta_1 (\texttt{genotype}_\texttt{KO}) + \beta_2 (\texttt{sex}_\texttt{F}) + \beta_3 (\texttt{genotype}_\texttt{KO}:\texttt{sex}_\texttt{F}) \ + \\
&amp;\gamma_{0j} + \gamma_{1j}(\texttt{genotype}_\texttt{KO}:\texttt{sex}_\texttt{M}) +  \gamma_{2j}(\texttt{genotype}_\texttt{WT}:\texttt{sex}_\texttt{F}) \ + \\  &amp;\gamma_{3j}(\texttt{genotype}_\texttt{KO}:\texttt{sex}_\texttt{F}) + \varepsilon
\end{align}
\]</span>
The <span class="math inline">\(\beta\)</span> are the <strong>fixed effects</strong>, although <span class="math inline">\(\beta_\)</span> is not an effect but a mean. The <span class="math inline">\(\gamma\)</span> (the lower case, greek letter “gamma”) are the <strong>random effects</strong>. All of these, including the <span class="math inline">\(\gamma_0\)</span>, are effects (differences between means). The <em>j</em> in the <span class="math inline">\(\gamma\)</span> references the <em>j</em>th experiment – this means that there is a different <span class="math inline">\(\gamma_0\)</span>, <span class="math inline">\(\gamma_1\)</span>, <span class="math inline">\(\gamma_2\)</span>, and <span class="math inline">\(\gamma_3\)</span> for each experiment. In contrast, <span class="math inline">\(\beta_0\)</span>, <span class="math inline">\(\beta_1\)</span>, <span class="math inline">\(\beta_2\)</span>, and <span class="math inline">\(\beta_3\)</span> are the same for all experiments – this is why they are “fixed.” The <span class="math inline">\(\gamma\)</span> are modeled as if each experiment is a random draw from an infinite number of experiments. This is why the <span class="math inline">\(\gamma\)</span> are “random.”</p>
<p>A reminder of what these fixed effects are is illustrated in Figure @ref(fig:lmm-exp1g-explainer_1). <a href="factorial.html#factorial">A more thorough explanation is in the chapter Linear models with two categorical <span class="math inline">\(X\)</span> – Factorial linear models (“two-way ANOVA”)</a></p>
<p>A <strong>random intercept</strong> for experiment <em>j</em> is the sum of the fixed intercept (<span class="math inline">\(\beta_0\)</span>) and a <strong>random intercept effect</strong> <span class="math inline">\(\gamma_0j\)</span> (Figure <a href="lmm.html#fig:lmm-exp1g-explainer-2">16.7</a>).</p>
<div class="figure">
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp1g-explainer_1-1.png" alt="Fixed effects estimated by exp1g_m1. The light gray point is the expected value of the GPR174- F treatment if genotype and sex were additive." width="672" />
<p class="caption">
(#fig:lmm-exp1g-explainer_1)Fixed effects estimated by exp1g_m1. The light gray point is the expected value of the GPR174- F treatment if genotype and sex were additive.
</p>
</div>
</div>
<div id="what-the-random-effects-are" class="section level3" number="16.5.5">
<h3><span class="header-section-number">16.5.5</span> What the random effects are</h3>
<p>A <strong>random slope</strong> is illustrated in Figure <a href="lmm.html#fig:lmm-exp1g-explainer-3">16.8</a> for the treatment combination “GPR174+ F.” The random slope for experiment <em>j</em> is the sum of the fixed slope (<span class="math inline">\(\beta_2\)</span>) and a <strong>random slope effect</strong> <span class="math inline">\(\gamma_2j\)</span>. This makes <span class="math inline">\(\gamma_2j\)</span> equal to the residual of the modeled mean of the “GPR174+ F” treatment for experiment <em>j</em> and the expected mean if there were no random slope. This expected mean is the fixed intercept plus the fixed effect for the “GPR174+ F” treatment plus the random intercept effect for experiment <em>j</em>, or <span class="math inline">\(\beta_0 + \beta_1 + \gamma_{0j}\)</span>. These expected means (if each <span class="math inline">\(\gamma_{0j}=0\)</span>) are illustrated by the large gray dots in Figure <a href="lmm.html#fig:lmm-exp1g-explainer-3">16.8</a>. The random slopes can also be visualized by the lines connecting the modeled means for each experiment from the reference to the “GPR174+ F” treatment (Figure <a href="lmm.html#fig:lmm-exp1g-explainer-3">16.8</a>).</p>
<div class="figure"><span id="fig:lmm-exp1g-explainer-2"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp1g-explainer-2-1.png" alt="What random intercepts are. The large, colored dots are the modeled mean for each treatment (genotype by sex) by experiment_id combination. The dashed grey line is the fixed intercept -- it is the modeled mean of the reference level (&quot;GPR174+ M&quot;). The large, colored dots at the reference level are the random intercepts. The value of each random intercept is the sum of the fixed intercept and the random intercept effect for that experiment_id. Each random intercept effect is the residual from (or, distance from colored dot to) the dashed, gray line, shown by the vertical, colored lines" width="672" />
<p class="caption">
Figure 16.7: What random intercepts are. The large, colored dots are the modeled mean for each treatment (genotype by sex) by experiment_id combination. The dashed grey line is the fixed intercept – it is the modeled mean of the reference level (“GPR174+ M”). The large, colored dots at the reference level are the random intercepts. The value of each random intercept is the sum of the fixed intercept and the random intercept effect for that experiment_id. Each random intercept effect is the residual from (or, distance from colored dot to) the dashed, gray line, shown by the vertical, colored lines
</p>
</div>
<div class="figure"><span id="fig:lmm-exp1g-explainer-3"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp1g-explainer-3-1.png" alt="What random slopes are. The large, black dots are the modeled means for the reference (&quot;GPR174+ M&quot;) and &quot;GPR174+ F&quot; treatment combinations. The difference in the y-value of these two points is the fixed effect (or slope) $b_2$, which is the effect of &quot;GPR174+ F&quot; relative to the reference. This effect can be visualized as the slope of the black line connecting the two black dots. The large, gray dots are the expected modeled means of &quot;GPR174+ F&quot; treatment combination for each experiment_id if there were no random slope. The differences between each large, colored dot and large, gray dot from the same experiment_id is the random slope effect for the &quot;GPR174+ F&quot; treatment combination for that experiment_id. The value of this effect for an experiment_id is equivalent to the difference of the slope of the colored line (the random slope) for the experiment_id and the black line (the fixed slope)." width="672" />
<p class="caption">
Figure 16.8: What random slopes are. The large, black dots are the modeled means for the reference (“GPR174+ M”) and “GPR174+ F” treatment combinations. The difference in the y-value of these two points is the fixed effect (or slope) <span class="math inline">\(b_2\)</span>, which is the effect of “GPR174+ F” relative to the reference. This effect can be visualized as the slope of the black line connecting the two black dots. The large, gray dots are the expected modeled means of “GPR174+ F” treatment combination for each experiment_id if there were no random slope. The differences between each large, colored dot and large, gray dot from the same experiment_id is the random slope effect for the “GPR174+ F” treatment combination for that experiment_id. The value of this effect for an experiment_id is equivalent to the difference of the slope of the colored line (the random slope) for the experiment_id and the black line (the fixed slope).
</p>
</div>
</div>
<div id="alternative-models-1" class="section level3" number="16.5.6">
<h3><span class="header-section-number">16.5.6</span> Alternative models</h3>
<div id="uncorrelated" class="section level4" number="16.5.6.1">
<h4><span class="header-section-number">16.5.6.1</span> Uncorrelated</h4>
</div>
</div>
</div>
<div id="working-in-r-5" class="section level2" number="16.6">
<h2><span class="header-section-number">16.6</span> Working in R</h2>
<div id="repeated-measures-anova-randomized-complete-block-with-no-subsampling" class="section level3" number="16.6.1">
<h3><span class="header-section-number">16.6.1</span> Repeated measures ANOVA (randomized complete block with no subsampling)</h3>
<div class="sourceCode" id="cb565"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb565-1"><a href="lmm.html#cb565-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this is the rm-ANOVA</span></span>
<span id="cb565-2"><a href="lmm.html#cb565-2" aria-hidden="true" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb565-3"><a href="lmm.html#cb565-3" aria-hidden="true" tabindex="-1"></a>              (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb565-4"><a href="lmm.html#cb565-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">data =</span> exp5c)</span>
<span id="cb565-5"><a href="lmm.html#cb565-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb565-6"><a href="lmm.html#cb565-6" aria-hidden="true" tabindex="-1"></a><span class="co"># lmm equivalent</span></span>
<span id="cb565-7"><a href="lmm.html#cb565-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb565-8"><a href="lmm.html#cb565-8" aria-hidden="true" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb565-9"><a href="lmm.html#cb565-9" aria-hidden="true" tabindex="-1"></a>         (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb565-10"><a href="lmm.html#cb565-10" aria-hidden="true" tabindex="-1"></a>         (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment) <span class="sc">+</span></span>
<span id="cb565-11"><a href="lmm.html#cb565-11" aria-hidden="true" tabindex="-1"></a>         (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>activity),</span>
<span id="cb565-12"><a href="lmm.html#cb565-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">data =</span> exp5c)</span>
<span id="cb565-13"><a href="lmm.html#cb565-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb565-14"><a href="lmm.html#cb565-14" aria-hidden="true" tabindex="-1"></a><span class="co"># random intercept and slope model that is *not* equivalent</span></span>
<span id="cb565-15"><a href="lmm.html#cb565-15" aria-hidden="true" tabindex="-1"></a><span class="co"># this isn&#39;t solvable because there is no subsampling</span></span>
<span id="cb565-16"><a href="lmm.html#cb565-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb565-17"><a href="lmm.html#cb565-17" aria-hidden="true" tabindex="-1"></a>m3 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb565-18"><a href="lmm.html#cb565-18" aria-hidden="true" tabindex="-1"></a>              (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb565-19"><a href="lmm.html#cb565-19" aria-hidden="true" tabindex="-1"></a>            <span class="at">data =</span> exp5c)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li><code>afex</code> computes the repeated measures anova model using both <code>aov</code> (classical univariate repeated measures ANOVA) and using <code>lm</code> with multiple response variables (the multivariate repeated measures ANOVA). As of this writing, the output from <code>aov</code> is the default.</li>
<li>Given only one measure for each donor within a <span class="math inline">\(\texttt{treatment} \times \texttt{activity}\)</span> combination, the linear mixed model <code>m2</code> is equivalent to the univariate repeated measures model m1.</li>
<li>The model formula in <code>m1</code> looks like that in the linear mixed model <code>m3</code> but the two models are not equivalent.</li>
</ol>
<div id="univariate-vs.-multivariate-repeated-measures-anova" class="section level4" number="16.6.1.1">
<h4><span class="header-section-number">16.6.1.1</span> univariate vs. multivariate repeated measures ANOVA</h4>
<p>Use the multivariate model unless you want to replicate the result of someone who used a univariate model.</p>
<div class="sourceCode" id="cb566"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb566-1"><a href="lmm.html#cb566-1" aria-hidden="true" tabindex="-1"></a><span class="co"># aov_4 computes both uni and multi but default output is uni</span></span>
<span id="cb566-2"><a href="lmm.html#cb566-2" aria-hidden="true" tabindex="-1"></a>exp5c_aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb566-3"><a href="lmm.html#cb566-3" aria-hidden="true" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb566-4"><a href="lmm.html#cb566-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> exp5c)</span>
<span id="cb566-5"><a href="lmm.html#cb566-5" aria-hidden="true" tabindex="-1"></a><span class="co"># exclude uni, so output is multi</span></span>
<span id="cb566-6"><a href="lmm.html#cb566-6" aria-hidden="true" tabindex="-1"></a>exp5c_aov2 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb566-7"><a href="lmm.html#cb566-7" aria-hidden="true" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb566-8"><a href="lmm.html#cb566-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> exp5c,</span>
<span id="cb566-9"><a href="lmm.html#cb566-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">include_aov =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The <code>include_aov = FALSE</code> argument forces output from <code>aov_4</code> to use the multivariate model.</li>
</ol>
<p><strong>contrasts from univariate model</strong></p>
<div class="sourceCode" id="cb567"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb567-1"><a href="lmm.html#cb567-1" aria-hidden="true" tabindex="-1"></a><span class="co"># these 2 are the same</span></span>
<span id="cb567-2"><a href="lmm.html#cb567-2" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb567-3"><a href="lmm.html#cb567-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb567-4"><a href="lmm.html#cb567-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb567-5"><a href="lmm.html#cb567-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb567-6"><a href="lmm.html#cb567-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>##  activity treatment contrast      estimate     SE   df t.ratio p.value
##  Basal    .         siNR4A3 - Scr   0.0813 0.0759 8.60  1.071  0.3132 
##  EPS      .         siNR4A3 - Scr  -0.1858 0.0759 8.60 -2.448  0.0380 
##  .        Scr       EPS - Basal     0.1510 0.0524 9.39  2.884  0.0173 
##  .        siNR4A3   EPS - Basal    -0.1161 0.0524 9.39 -2.218  0.0525</code></pre>
<div class="sourceCode" id="cb569"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb569-1"><a href="lmm.html#cb569-1" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov1<span class="sc">$</span>aov, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb569-2"><a href="lmm.html#cb569-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb569-3"><a href="lmm.html#cb569-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb569-4"><a href="lmm.html#cb569-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb569-5"><a href="lmm.html#cb569-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>##  activity treatment contrast      estimate     SE   df t.ratio p.value
##  Basal    .         siNR4A3 - Scr   0.0813 0.0759 8.60  1.071  0.3132 
##  EPS      .         siNR4A3 - Scr  -0.1858 0.0759 8.60 -2.448  0.0380 
##  .        Scr       EPS - Basal     0.1510 0.0524 9.39  2.884  0.0173 
##  .        siNR4A3   EPS - Basal    -0.1161 0.0524 9.39 -2.218  0.0525</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Passing <code>exp5c_aov1</code> to <code>emmeans</code> will return the contrasts from the univariate model by deault. Make this explicit by passing only the univarite fit <code>exp5c_aov1$aov</code>.</li>
</ol>
<p><strong>contrasts from multivariate model</strong></p>
<div class="sourceCode" id="cb571"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb571-1"><a href="lmm.html#cb571-1" aria-hidden="true" tabindex="-1"></a><span class="co"># these 2 are the same</span></span>
<span id="cb571-2"><a href="lmm.html#cb571-2" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov1,</span>
<span id="cb571-3"><a href="lmm.html#cb571-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>),</span>
<span id="cb571-4"><a href="lmm.html#cb571-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">model =</span> <span class="st">&quot;multivariate&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb571-5"><a href="lmm.html#cb571-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb571-6"><a href="lmm.html#cb571-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb571-7"><a href="lmm.html#cb571-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb571-8"><a href="lmm.html#cb571-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>##  activity treatment contrast      estimate     SE df t.ratio p.value
##  Basal    .         siNR4A3 - Scr   0.0813 0.1012  5  0.804  0.4581 
##  EPS      .         siNR4A3 - Scr  -0.1858 0.0359  5 -5.180  0.0035 
##  .        Scr       EPS - Basal     0.1510 0.0395  5  3.821  0.0124 
##  .        siNR4A3   EPS - Basal    -0.1161 0.0626  5 -1.855  0.1228</code></pre>
<div class="sourceCode" id="cb573"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb573-1"><a href="lmm.html#cb573-1" aria-hidden="true" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov2,</span>
<span id="cb573-2"><a href="lmm.html#cb573-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb573-3"><a href="lmm.html#cb573-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb573-4"><a href="lmm.html#cb573-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb573-5"><a href="lmm.html#cb573-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb573-6"><a href="lmm.html#cb573-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>##  activity treatment contrast      estimate     SE df t.ratio p.value
##  Basal    .         siNR4A3 - Scr   0.0813 0.1012  5  0.804  0.4581 
##  EPS      .         siNR4A3 - Scr  -0.1858 0.0359  5 -5.180  0.0035 
##  .        Scr       EPS - Basal     0.1510 0.0395  5  3.821  0.0124 
##  .        siNR4A3   EPS - Basal    -0.1161 0.0626  5 -1.855  0.1228</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>If our rmANOVA model is fit with the default specification (<code>exp5c_aov1</code>), we can get the multivariate output using the <code>model = "multivariate"</code> argument within <code>emmeans</code> (not the <code>contrast</code> function!). Or, if our fit excluded the univariate model (<code>exp5c_aov2</code>), then we don’t need the <code>model</code> argument.</li>
</ol>
</div>
<div id="the-anova-table" class="section level4" number="16.6.1.2">
<h4><span class="header-section-number">16.6.1.2</span> The ANOVA table</h4>
<div class="sourceCode" id="cb575"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb575-1"><a href="lmm.html#cb575-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nice</span>(exp5c_aov1, <span class="at">correction =</span> <span class="st">&quot;GG&quot;</span>)</span></code></pre></div>
<pre><code>## Anova Table (Type 3 tests)
## 
## Response: glucose_uptake
##               Effect   df  MSE       F   ges p.value
## 1          treatment 1, 5 0.02    0.68  .006     .45
## 2           activity 1, 5 0.01    0.30 .0006     .61
## 3 treatment:activity 1, 5 0.01 10.37 *   .04     .02
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;+&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb577"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb577-1"><a href="lmm.html#cb577-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nice</span>(exp5c_aov1, <span class="at">correction =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>## Anova Table (Type 3 tests)
## 
## Response: glucose_uptake
##               Effect   df  MSE       F   ges p.value
## 1          treatment 1, 5 0.02    0.68  .006     .45
## 2           activity 1, 5 0.01    0.30 .0006     .61
## 3 treatment:activity 1, 5 0.01 10.37 *   .04     .02
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;+&#39; 0.1 &#39; &#39; 1</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The Greenhouse-Geiger (“GG”) correction is the default. While the <code>correction = "GG"</code> argument is not needed, it makes the script more transparent.</li>
<li>For <em>these data</em> the Greenhouse-Geiger correction doesn’t make a difference in the table.</li>
</ol>
</div>
</div>
</div>
<div id="hidden-code-5" class="section level2" number="16.7">
<h2><span class="header-section-number">16.7</span> Hidden code</h2>
<div id="import-exp5c" class="section level3" number="16.7.1">
<h3><span class="header-section-number">16.7.1</span> Import exp5c</h3>
<div class="sourceCode" id="cb579"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb579-1"><a href="lmm.html#cb579-1" aria-hidden="true" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">&quot;Transcriptomic profiling of skeletal muscle adaptations to exercise and inactivity&quot;</span></span>
<span id="cb579-2"><a href="lmm.html#cb579-2" aria-hidden="true" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">&quot;41467_2019_13869_MOESM6_ESM.xlsx&quot;</span></span>
<span id="cb579-3"><a href="lmm.html#cb579-3" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb579-4"><a href="lmm.html#cb579-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb579-5"><a href="lmm.html#cb579-5" aria-hidden="true" tabindex="-1"></a>exp5c_wide <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb579-6"><a href="lmm.html#cb579-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sheet =</span> <span class="st">&quot;Fig5c&quot;</span>,</span>
<span id="cb579-7"><a href="lmm.html#cb579-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">range =</span> <span class="st">&quot;A2:M3&quot;</span>,</span>
<span id="cb579-8"><a href="lmm.html#cb579-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb579-9"><a href="lmm.html#cb579-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>() <span class="sc">%&gt;%</span></span>
<span id="cb579-10"><a href="lmm.html#cb579-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transpose</span>(<span class="at">make.names =</span> <span class="dv">1</span>)</span>
<span id="cb579-11"><a href="lmm.html#cb579-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb579-12"><a href="lmm.html#cb579-12" aria-hidden="true" tabindex="-1"></a>activity_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;EPS&quot;</span>)</span>
<span id="cb579-13"><a href="lmm.html#cb579-13" aria-hidden="true" tabindex="-1"></a>treatment_levels <span class="ot">&lt;-</span> <span class="fu">names</span>(exp5c_wide)</span>
<span id="cb579-14"><a href="lmm.html#cb579-14" aria-hidden="true" tabindex="-1"></a>exp5c_wide[, activity <span class="sc">:</span><span class="er">=</span> <span class="fu">rep</span>(activity_levels, <span class="at">each =</span> <span class="dv">6</span>)]</span>
<span id="cb579-15"><a href="lmm.html#cb579-15" aria-hidden="true" tabindex="-1"></a>exp5c_wide[, activity <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(activity, <span class="at">levels =</span> activity_levels)]</span>
<span id="cb579-16"><a href="lmm.html#cb579-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb579-17"><a href="lmm.html#cb579-17" aria-hidden="true" tabindex="-1"></a>exp5c <span class="ot">&lt;-</span> <span class="fu">melt</span>(exp5c_wide,</span>
<span id="cb579-18"><a href="lmm.html#cb579-18" aria-hidden="true" tabindex="-1"></a>              <span class="at">id.vars =</span> <span class="st">&quot;activity&quot;</span>,</span>
<span id="cb579-19"><a href="lmm.html#cb579-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">variable.name =</span> <span class="st">&quot;treatment&quot;</span>,</span>
<span id="cb579-20"><a href="lmm.html#cb579-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">value.name =</span> <span class="st">&quot;glucose_uptake&quot;</span>)</span>
<span id="cb579-21"><a href="lmm.html#cb579-21" aria-hidden="true" tabindex="-1"></a>exp5c[, treatment <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(treatment, <span class="at">levels =</span> treatment_levels)]</span>
<span id="cb579-22"><a href="lmm.html#cb579-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb579-23"><a href="lmm.html#cb579-23" aria-hidden="true" tabindex="-1"></a>exp5c[, donor <span class="sc">:</span><span class="er">=</span> <span class="fu">rep</span>(<span class="fu">paste0</span>(<span class="st">&quot;donor_&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>), <span class="dv">4</span>)]</span></code></pre></div>
</div>
<div id="import-exp1g" class="section level3" number="16.7.2">
<h3><span class="header-section-number">16.7.2</span> Import exp1g</h3>
<div class="sourceCode" id="cb580"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb580-1"><a href="lmm.html#cb580-1" aria-hidden="true" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">&quot;A GPR174–CCL21 module imparts sexual dimorphism to humoral immunity&quot;</span></span>
<span id="cb580-2"><a href="lmm.html#cb580-2" aria-hidden="true" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">&quot;41586_2019_1873_MOESM3_ESM.xlsx&quot;</span></span>
<span id="cb580-3"><a href="lmm.html#cb580-3" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb580-4"><a href="lmm.html#cb580-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb580-5"><a href="lmm.html#cb580-5" aria-hidden="true" tabindex="-1"></a>exp1g_wide <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb580-6"><a href="lmm.html#cb580-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sheet =</span> <span class="st">&quot;Fig 1g&quot;</span>,</span>
<span id="cb580-7"><a href="lmm.html#cb580-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">range =</span> <span class="st">&quot;B4:E25&quot;</span>,</span>
<span id="cb580-8"><a href="lmm.html#cb580-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_types =</span> <span class="fu">c</span>(<span class="st">&quot;numeric&quot;</span>),</span>
<span id="cb580-9"><a href="lmm.html#cb580-9" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb580-10"><a href="lmm.html#cb580-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.table</span>()</span>
<span id="cb580-11"><a href="lmm.html#cb580-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb580-12"><a href="lmm.html#cb580-12" aria-hidden="true" tabindex="-1"></a>genotype_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Gpr174+&quot;</span>, <span class="st">&quot;Gpr174-&quot;</span>)</span>
<span id="cb580-13"><a href="lmm.html#cb580-13" aria-hidden="true" tabindex="-1"></a>sex_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;M&quot;</span>, <span class="st">&quot;F&quot;</span>)</span>
<span id="cb580-14"><a href="lmm.html#cb580-14" aria-hidden="true" tabindex="-1"></a>g.by.s_levels <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">expand.grid</span>(genotype_levels, sex_levels))</span>
<span id="cb580-15"><a href="lmm.html#cb580-15" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(exp1g_wide) <span class="ot">&lt;-</span> g.by.s_levels</span>
<span id="cb580-16"><a href="lmm.html#cb580-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb580-17"><a href="lmm.html#cb580-17" aria-hidden="true" tabindex="-1"></a>exp_levels <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;exp_&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)</span>
<span id="cb580-18"><a href="lmm.html#cb580-18" aria-hidden="true" tabindex="-1"></a>exp1g_wide[, experiment_id <span class="sc">:</span><span class="er">=</span> <span class="fu">rep</span>(exp_levels, <span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">6</span>,<span class="dv">5</span>))] <span class="co">#check!</span></span>
<span id="cb580-19"><a href="lmm.html#cb580-19" aria-hidden="true" tabindex="-1"></a>exp1g_wide[, experiment_id <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(experiment_id)] <span class="co">#check!</span></span>
<span id="cb580-20"><a href="lmm.html#cb580-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb580-21"><a href="lmm.html#cb580-21" aria-hidden="true" tabindex="-1"></a>exp1g <span class="ot">&lt;-</span> <span class="fu">melt</span>(exp1g_wide,</span>
<span id="cb580-22"><a href="lmm.html#cb580-22" aria-hidden="true" tabindex="-1"></a>              <span class="at">id.vars =</span> <span class="st">&quot;experiment_id&quot;</span>,</span>
<span id="cb580-23"><a href="lmm.html#cb580-23" aria-hidden="true" tabindex="-1"></a>              <span class="at">measure.vars =</span> g.by.s_levels,</span>
<span id="cb580-24"><a href="lmm.html#cb580-24" aria-hidden="true" tabindex="-1"></a>              <span class="at">variable.name =</span> <span class="st">&quot;treatment&quot;</span>,</span>
<span id="cb580-25"><a href="lmm.html#cb580-25" aria-hidden="true" tabindex="-1"></a>              <span class="at">value.name =</span> <span class="st">&quot;gc&quot;</span>) <span class="sc">%&gt;%</span> <span class="co"># cell count</span></span>
<span id="cb580-26"><a href="lmm.html#cb580-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb580-27"><a href="lmm.html#cb580-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb580-28"><a href="lmm.html#cb580-28" aria-hidden="true" tabindex="-1"></a>exp1g[, <span class="fu">c</span>(<span class="st">&quot;genotype&quot;</span>, <span class="st">&quot;sex&quot;</span>)<span class="sc">:</span><span class="er">=</span> <span class="fu">tstrsplit</span>(treatment,</span>
<span id="cb580-29"><a href="lmm.html#cb580-29" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">&quot; &quot;</span>,</span>
<span id="cb580-30"><a href="lmm.html#cb580-30" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">fixed =</span> <span class="cn">TRUE</span>)]</span>
<span id="cb580-31"><a href="lmm.html#cb580-31" aria-hidden="true" tabindex="-1"></a>exp1g[, genotype <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(genotype,</span>
<span id="cb580-32"><a href="lmm.html#cb580-32" aria-hidden="true" tabindex="-1"></a>                           <span class="at">levels =</span> genotype_levels)]</span>
<span id="cb580-33"><a href="lmm.html#cb580-33" aria-hidden="true" tabindex="-1"></a>exp1g[, sex <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(sex,</span>
<span id="cb580-34"><a href="lmm.html#cb580-34" aria-hidden="true" tabindex="-1"></a>                           <span class="at">levels =</span> sex_levels)]</span>
<span id="cb580-35"><a href="lmm.html#cb580-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb580-36"><a href="lmm.html#cb580-36" aria-hidden="true" tabindex="-1"></a>exp1g_means <span class="ot">&lt;-</span> exp1g[, .(<span class="at">gc =</span> <span class="fu">mean</span>(gc)),</span>
<span id="cb580-37"><a href="lmm.html#cb580-37" aria-hidden="true" tabindex="-1"></a>                     by <span class="ot">=</span> .(treatment, genotype, sex, experiment_id)]</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="part-vii-expanding-the-linear-model.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="pre-post.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/chapters/55-lmm01-blocking.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Walker-elementary-statistical-modeling-draft.pdf", "Walker-elementary-statistical-modeling-draft.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
