<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 16 Models with random factors – linear mixed models | Statistics for the Experimental Biologist</title>
  <meta name="description" content="An introduction to statistical modeling for experimental biology researchers" />
  <meta name="generator" content="bookdown 0.32 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 16 Models with random factors – linear mixed models | Statistics for the Experimental Biologist" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="An introduction to statistical modeling for experimental biology researchers" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 16 Models with random factors – linear mixed models | Statistics for the Experimental Biologist" />
  
  <meta name="twitter:description" content="An introduction to statistical modeling for experimental biology researchers" />
  

<meta name="author" content="Copyright 2018 Jeffrey A. Walker" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="part-vi-expanding-the-linear-model.html"/>
<link rel="next" href="pre-post.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Statistics for the Experimental Biologist</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#why-bother-with-linear-regession-models-arent-t-tests-and-anova-good-enough"><i class="fa fa-check"></i>Why bother with linear (regession!) models – aren’t <em>t</em>-tests and ANOVA good enough?</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#what-is-unusual-about-this-book"><i class="fa fa-check"></i>What is unusual about this book?</a></li>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#motivations"><i class="fa fa-check"></i><b>0.1</b> Motivations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="a-table-of-models.html"><a href="a-table-of-models.html"><i class="fa fa-check"></i>A Table of Models</a>
<ul>
<li class="chapter" data-level="" data-path="a-table-of-models.html"><a href="a-table-of-models.html#including-mapping-between-linear-models-and-classical-tests"><i class="fa fa-check"></i>including mapping between linear models and classical tests</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="ask1-intro.html"><a href="ask1-intro.html"><i class="fa fa-check"></i><b>1</b> Analyzing experimental data with a linear model</a>
<ul>
<li class="chapter" data-level="1.1" data-path="ask1-intro.html"><a href="ask1-intro.html#this-text-is-about-using-linear-models-to-estimate-treatment-effects-and-the-uncertainty-in-our-estimates.-this-raises-the-question-what-is-an-effect"><i class="fa fa-check"></i><b>1.1</b> This text is about using linear models to estimate treatment effects and the uncertainty in our estimates. This, raises the question, what is “an effect”?</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-i-getting-started.html"><a href="part-i-getting-started.html"><i class="fa fa-check"></i>Part I: Getting Started</a></li>
<li class="chapter" data-level="2" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html"><i class="fa fa-check"></i><b>2</b> Getting Started – R Projects and R Markdown</a>
<ul>
<li class="chapter" data-level="2.1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#r-vs-r-studio"><i class="fa fa-check"></i><b>2.1</b> R vs R Studio</a></li>
<li class="chapter" data-level="2.2" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#download-and-install-r-and-r-studio"><i class="fa fa-check"></i><b>2.2</b> Download and install R and R studio</a></li>
<li class="chapter" data-level="2.3" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#open-r-studio-and-modify-the-workspace-preference"><i class="fa fa-check"></i><b>2.3</b> Open R Studio and modify the workspace preference</a></li>
<li class="chapter" data-level="2.4" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#if-you-didnt-modify-the-workspace-preferences-from-the-previous-section-go-back-and-do-it"><i class="fa fa-check"></i><b>2.4</b> If you didn’t modify the workspace preferences from the previous section, go back and do it</a></li>
<li class="chapter" data-level="2.5" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#r-markdown-in-a-nutshell"><i class="fa fa-check"></i><b>2.5</b> R Markdown in a nutshell</a></li>
<li class="chapter" data-level="2.6" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#install-r-markdown"><i class="fa fa-check"></i><b>2.6</b> Install R Markdown</a></li>
<li class="chapter" data-level="2.7" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#importing-packages"><i class="fa fa-check"></i><b>2.7</b> Importing Packages</a></li>
<li class="chapter" data-level="2.8" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#setup-create-project"><i class="fa fa-check"></i><b>2.8</b> Create an R Studio Project for this textbook</a></li>
<li class="chapter" data-level="2.9" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#working-on-a-project-in-a-nutshell"><i class="fa fa-check"></i><b>2.9</b> Working on a project, in a nutshell</a></li>
<li class="chapter" data-level="2.10" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#setup-create-rmd"><i class="fa fa-check"></i><b>2.10</b> Create and setup an R Markdown document (Rmd)</a>
<ul>
<li class="chapter" data-level="2.10.1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#modify-the-yaml-header"><i class="fa fa-check"></i><b>2.10.1</b> Modify the yaml header</a></li>
<li class="chapter" data-level="2.10.2" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#modify-the-setup-chunk"><i class="fa fa-check"></i><b>2.10.2</b> Modify the “setup” chunk</a></li>
</ul></li>
<li class="chapter" data-level="2.11" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#lets-play-around-with-an-r-markdown-file"><i class="fa fa-check"></i><b>2.11</b> Let’s play around with an R Markdown file</a>
<ul>
<li class="chapter" data-level="2.11.1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-a-fake-data-chunk"><i class="fa fa-check"></i><b>2.11.1</b> Create a “fake-data” chunk</a></li>
<li class="chapter" data-level="2.11.2" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-a-plot-chunk"><i class="fa fa-check"></i><b>2.11.2</b> Create a “plot” chunk</a></li>
<li class="chapter" data-level="2.11.3" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#knit-the-rmd"><i class="fa fa-check"></i><b>2.11.3</b> Knit the Rmd</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-ii-r-fundamentals.html"><a href="part-ii-r-fundamentals.html"><i class="fa fa-check"></i>Part II: R fundamentals</a></li>
<li class="chapter" data-level="3" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>3</b> Data – Reading, Wrangling, and Writing</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data.html"><a href="data.html#data-long"><i class="fa fa-check"></i><b>3.1</b> Long data format – the way data should be</a></li>
<li class="chapter" data-level="3.2" data-path="data.html"><a href="data.html#data-here"><i class="fa fa-check"></i><b>3.2</b> Use the here function to construct the file path</a></li>
<li class="chapter" data-level="3.3" data-path="data.html"><a href="data.html#learning-from-this-chapter"><i class="fa fa-check"></i><b>3.3</b> Learning from this chapter</a></li>
<li class="chapter" data-level="3.4" data-path="data.html"><a href="data.html#data-working-in-r"><i class="fa fa-check"></i><b>3.4</b> Working in R</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="data.html"><a href="data.html#importing-data"><i class="fa fa-check"></i><b>3.4.1</b> Importing data</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="data.html"><a href="data.html#data-wrangling"><i class="fa fa-check"></i><b>3.5</b> Data wrangling</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="data.html"><a href="data.html#reshaping-data-wide-to-long"><i class="fa fa-check"></i><b>3.5.1</b> Reshaping data – Wide to long</a></li>
<li class="chapter" data-level="3.5.2" data-path="data.html"><a href="data.html#reshaping-data-transpose-turning-the-columns-into-rows"><i class="fa fa-check"></i><b>3.5.2</b> Reshaping data – Transpose (turning the columns into rows)</a></li>
<li class="chapter" data-level="3.5.3" data-path="data.html"><a href="data.html#combining-data"><i class="fa fa-check"></i><b>3.5.3</b> Combining data</a></li>
<li class="chapter" data-level="3.5.4" data-path="data.html"><a href="data.html#data-subset"><i class="fa fa-check"></i><b>3.5.4</b> Subsetting data</a></li>
<li class="chapter" data-level="3.5.5" data-path="data.html"><a href="data.html#wrangling-columns"><i class="fa fa-check"></i><b>3.5.5</b> Wrangling columns</a></li>
<li class="chapter" data-level="3.5.6" data-path="data.html"><a href="data.html#missing-data"><i class="fa fa-check"></i><b>3.5.6</b> Missing data</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="data.html"><a href="data.html#saving-data"><i class="fa fa-check"></i><b>3.6</b> Saving data</a></li>
<li class="chapter" data-level="3.7" data-path="data.html"><a href="data.html#exercises"><i class="fa fa-check"></i><b>3.7</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="plotting-models.html"><a href="plotting-models.html"><i class="fa fa-check"></i><b>4</b> Plotting Models</a>
<ul>
<li class="chapter" data-level="4.1" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plots-show-the-model-and-the-data"><i class="fa fa-check"></i><b>4.1</b> Pretty good plots show the model and the data</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plot-component-1-modeled-effects-plot"><i class="fa fa-check"></i><b>4.1.1</b> Pretty good plot component 1: Modeled effects plot</a></li>
<li class="chapter" data-level="4.1.2" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plot-component-2-modeled-mean-and-ci-plot"><i class="fa fa-check"></i><b>4.1.2</b> Pretty good plot component 2: Modeled mean and CI plot</a></li>
<li class="chapter" data-level="4.1.3" data-path="plotting-models.html"><a href="plotting-models.html#combining-effects-and-modeled-mean-and-ci-plots-an-effects-and-response-plot."><i class="fa fa-check"></i><b>4.1.3</b> Combining Effects and Modeled mean and CI plots – an Effects and response plot.</a></li>
<li class="chapter" data-level="4.1.4" data-path="plotting-models.html"><a href="plotting-models.html#some-comments-on-plot-components"><i class="fa fa-check"></i><b>4.1.4</b> Some comments on plot components</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="plotting-models.html"><a href="plotting-models.html#working-in-r"><i class="fa fa-check"></i><b>4.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="plotting-models.html"><a href="plotting-models.html#source-data"><i class="fa fa-check"></i><b>4.2.1</b> Source data</a></li>
<li class="chapter" data-level="4.2.2" data-path="plotting-models.html"><a href="plotting-models.html#how-to-plot-the-model"><i class="fa fa-check"></i><b>4.2.2</b> How to plot the model</a></li>
<li class="chapter" data-level="4.2.3" data-path="plotting-models.html"><a href="plotting-models.html#be-sure-ggplot_the_model-is-in-your-r-folder"><i class="fa fa-check"></i><b>4.2.3</b> Be sure ggplot_the_model is in your R folder</a></li>
<li class="chapter" data-level="4.2.4" data-path="plotting-models.html"><a href="plotting-models.html#how-to-use-the-plot-the-model-functions"><i class="fa fa-check"></i><b>4.2.4</b> How to use the Plot the Model functions</a></li>
<li class="chapter" data-level="4.2.5" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-a-response-plot-using-ggpubr"><i class="fa fa-check"></i><b>4.2.5</b> How to generate a Response Plot using ggpubr</a></li>
<li class="chapter" data-level="4.2.6" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-a-response-plot-with-a-grid-of-treatments-using-ggplot2"><i class="fa fa-check"></i><b>4.2.6</b> How to generate a Response Plot with a grid of treatments using ggplot2</a></li>
<li class="chapter" data-level="4.2.7" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-an-effects-plot"><i class="fa fa-check"></i><b>4.2.7</b> How to generate an Effects Plot</a></li>
<li class="chapter" data-level="4.2.8" data-path="plotting-models.html"><a href="plotting-models.html#how-to-combine-the-response-and-effects-plots"><i class="fa fa-check"></i><b>4.2.8</b> How to combine the response and effects plots</a></li>
<li class="chapter" data-level="4.2.9" data-path="plotting-models.html"><a href="plotting-models.html#how-to-add-the-interaction-effect-to-response-and-effects-plots"><i class="fa fa-check"></i><b>4.2.9</b> How to add the interaction effect to response and effects plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-iii-some-fundamentals-of-statistical-modeling.html"><a href="part-iii-some-fundamentals-of-statistical-modeling.html"><i class="fa fa-check"></i>Part III: Some Fundamentals of Statistical Modeling</a></li>
<li class="chapter" data-level="5" data-path="uncertainty.html"><a href="uncertainty.html"><i class="fa fa-check"></i><b>5</b> Variability and Uncertainty (Standard Deviations, Standard Errors, and Confidence Intervals)</a>
<ul>
<li class="chapter" data-level="5.1" data-path="uncertainty.html"><a href="uncertainty.html#standard-errors-are-used-to-compute-p-values-and-confidence-intervals"><i class="fa fa-check"></i><b>5.1</b> Standard errors are used to compute <em>p</em>-values and confidence intervals</a></li>
<li class="chapter" data-level="5.2" data-path="uncertainty.html"><a href="uncertainty.html#background"><i class="fa fa-check"></i><b>5.2</b> Background</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="uncertainty.html"><a href="uncertainty.html#sample-standard-deviation"><i class="fa fa-check"></i><b>5.2.1</b> Sample standard deviation</a></li>
<li class="chapter" data-level="5.2.2" data-path="uncertainty.html"><a href="uncertainty.html#standard-error-of-the-mean"><i class="fa fa-check"></i><b>5.2.2</b> Standard error of the mean</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="uncertainty.html"><a href="uncertainty.html#simulations-using-fake-data-as-an-intuition-pump"><i class="fa fa-check"></i><b>5.3</b> Simulations – using fake data as an intuition pump</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="uncertainty.html"><a href="uncertainty.html#using-google-sheets-to-generate-fake-data-to-explore-the-standard-error"><i class="fa fa-check"></i><b>5.3.1</b> Using Google Sheets to generate fake data to explore the standard error</a></li>
<li class="chapter" data-level="5.3.2" data-path="uncertainty.html"><a href="uncertainty.html#using-r-to-generate-fake-data-to-explore-the-standard-error"><i class="fa fa-check"></i><b>5.3.2</b> Using R to generate fake data to explore the standard error</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="uncertainty.html"><a href="uncertainty.html#bootstrap"><i class="fa fa-check"></i><b>5.4</b> Bootstrapped standard errors</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="uncertainty.html"><a href="uncertainty.html#an-example-of-bootstrapped-standard-errors"><i class="fa fa-check"></i><b>5.4.1</b> An example of bootstrapped standard errors</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="uncertainty.html"><a href="uncertainty.html#confidence-intervals"><i class="fa fa-check"></i><b>5.5</b> Confidence Intervals</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="uncertainty.html"><a href="uncertainty.html#just-the-math"><i class="fa fa-check"></i><b>5.5.1</b> Just the math</a></li>
<li class="chapter" data-level="5.5.2" data-path="uncertainty.html"><a href="uncertainty.html#interpretation-of-a-confidence-interval"><i class="fa fa-check"></i><b>5.5.2</b> Interpretation of a confidence interval</a></li>
<li class="chapter" data-level="5.5.3" data-path="uncertainty.html"><a href="uncertainty.html#bootstrap-confidence-interval"><i class="fa fa-check"></i><b>5.5.3</b> Bootstrap confidence interval</a></li>
<li class="chapter" data-level="5.5.4" data-path="uncertainty.html"><a href="uncertainty.html#a-plot-of-a-parametric-ci-vs.-bootstrap-ci-of-the-means"><i class="fa fa-check"></i><b>5.5.4</b> A plot of a “parametric” CI vs. bootstrap CI of the means</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="uncertainty.html"><a href="uncertainty.html#standard-error-of-a-difference-between-means"><i class="fa fa-check"></i><b>5.6</b> Standard error of a difference between means</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="uncertainty.html"><a href="uncertainty.html#the-standard-error-of-a-difference-between-means-is-the-standard-deviation-of-the-sampling-distribution-of-the-difference"><i class="fa fa-check"></i><b>5.6.1</b> The standard error of a difference between means is the standard deviation of the sampling distribution of the difference</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="uncertainty.html"><a href="uncertainty.html#confidence-limits-of-a-difference-between-means"><i class="fa fa-check"></i><b>5.7</b> Confidence limits of a difference between means</a>
<ul>
<li class="chapter" data-level="5.7.1" data-path="uncertainty.html"><a href="uncertainty.html#of-95-cis-of-the-difference-include-the-true-difference"><i class="fa fa-check"></i><b>5.7.1</b> 95% of 95% CIs of the difference include the true difference</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="uncertainty.html"><a href="uncertainty.html#hidden-code"><i class="fa fa-check"></i><b>5.8</b> Hidden code</a>
<ul>
<li class="chapter" data-level="5.8.1" data-path="uncertainty.html"><a href="uncertainty.html#import-exp3b"><i class="fa fa-check"></i><b>5.8.1</b> Import exp3b</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="p-values.html"><a href="p-values.html"><i class="fa fa-check"></i><b>6</b> P-values</a>
<ul>
<li class="chapter" data-level="6.1" data-path="p-values.html"><a href="p-values.html#a-p-value-is-the-probability-of-sampling-a-value-as-or-more-extreme-than-the-test-statistic-if-sampling-from-a-null-distribution"><i class="fa fa-check"></i><b>6.1</b> A <em>p</em>-value is the probability of sampling a value as or more extreme than the test statistic if sampling from a null distribution</a></li>
<li class="chapter" data-level="6.2" data-path="p-values.html"><a href="p-values.html#pump-your-intuition-creating-a-null-distribution"><i class="fa fa-check"></i><b>6.2</b> Pump your intuition – Creating a null distribution</a></li>
<li class="chapter" data-level="6.3" data-path="p-values.html"><a href="p-values.html#a-null-distribution-of-t-values-the-t-distribution"><i class="fa fa-check"></i><b>6.3</b> A null distribution of <em>t</em>-values – the <em>t</em> distribution</a></li>
<li class="chapter" data-level="6.4" data-path="p-values.html"><a href="p-values.html#p-values-from-the-perspective-of-permutation"><i class="fa fa-check"></i><b>6.4</b> P-values from the perspective of permutation</a></li>
<li class="chapter" data-level="6.5" data-path="p-values.html"><a href="p-values.html#parametric-vs.-non-parametric-statistics"><i class="fa fa-check"></i><b>6.5</b> Parametric vs. non-parametric statistics</a></li>
<li class="chapter" data-level="6.6" data-path="p-values.html"><a href="p-values.html#frequentist-probability-and-the-interpretation-of-p-values"><i class="fa fa-check"></i><b>6.6</b> frequentist probability and the interpretation of p-values</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="p-values.html"><a href="p-values.html#background-1"><i class="fa fa-check"></i><b>6.6.1</b> Background</a></li>
<li class="chapter" data-level="6.6.2" data-path="p-values.html"><a href="p-values.html#this-book-covers-frequentist-approaches-to-statistical-modeling-and-when-a-probability-arises-such-as-the-p-value-of-a-test-statistic-this-will-be-a-frequentist-probability."><i class="fa fa-check"></i><b>6.6.2</b> This book covers frequentist approaches to statistical modeling and when a probability arises, such as the <em>p</em>-value of a test statistic, this will be a frequentist probability.</a></li>
<li class="chapter" data-level="6.6.3" data-path="p-values.html"><a href="p-values.html#two-interpretations-of-the-p-value"><i class="fa fa-check"></i><b>6.6.3</b> Two interpretations of the <em>p</em>-value</a></li>
<li class="chapter" data-level="6.6.4" data-path="p-values.html"><a href="p-values.html#nhst"><i class="fa fa-check"></i><b>6.6.4</b> NHST</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="p-values.html"><a href="p-values.html#some-major-misconceptions-of-the-p-value"><i class="fa fa-check"></i><b>6.7</b> Some major misconceptions of the <em>p</em>-value</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="p-values.html"><a href="p-values.html#misconception-p-0.05-means-there-is-no-effect-of-treatment"><i class="fa fa-check"></i><b>6.7.1</b> Misconception: <span class="math inline">\(p &gt; 0.05\)</span> means there is no effect of treatment</a></li>
<li class="chapter" data-level="6.7.2" data-path="p-values.html"><a href="p-values.html#misconception-a-p-value-is-repeatable"><i class="fa fa-check"></i><b>6.7.2</b> Misconception: a <em>p</em>-value is repeatable</a></li>
<li class="chapter" data-level="6.7.3" data-path="p-values.html"><a href="p-values.html#misconception-0.05-is-the-lifetime-rate-of-false-discoveries"><i class="fa fa-check"></i><b>6.7.3</b> Misconception: 0.05 is the lifetime rate of false discoveries</a></li>
<li class="chapter" data-level="6.7.4" data-path="p-values.html"><a href="p-values.html#misconception-a-low-p-value-indicates-an-important-effect"><i class="fa fa-check"></i><b>6.7.4</b> Misconception: a low <em>p</em>-value indicates an important effect</a></li>
<li class="chapter" data-level="6.7.5" data-path="p-values.html"><a href="p-values.html#misconception-a-low-p-value-indicates-high-model-fit-or-high-predictive-capacity"><i class="fa fa-check"></i><b>6.7.5</b> Misconception: a low <em>p</em>-value indicates high model fit or high predictive capacity</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="p-values.html"><a href="p-values.html#what-the-p-value-does-not-mean"><i class="fa fa-check"></i><b>6.8</b> What the <em>p</em>-value does not mean</a></li>
<li class="chapter" data-level="6.9" data-path="p-values.html"><a href="p-values.html#on-using-p-values-in-experimental-biology"><i class="fa fa-check"></i><b>6.9</b> On using <em>p</em>-values in experimental biology</a></li>
<li class="chapter" data-level="6.10" data-path="p-values.html"><a href="p-values.html#better-reproducibility"><i class="fa fa-check"></i><b>6.10</b> Better reproducibility</a></li>
<li class="chapter" data-level="6.11" data-path="p-values.html"><a href="p-values.html#multiple-testing-and-controlling-for-false-positives"><i class="fa fa-check"></i><b>6.11</b> Multiple testing and controlling for false positives</a>
<ul>
<li class="chapter" data-level="6.11.1" data-path="p-values.html"><a href="p-values.html#controlling-the-family-wise-error-rate-when-all-tests-are-testing-the-same-hypothesis"><i class="fa fa-check"></i><b>6.11.1</b> Controlling the family-wise error rate when all tests are testing the same hypothesis</a></li>
</ul></li>
<li class="chapter" data-level="6.12" data-path="p-values.html"><a href="p-values.html#recommendations"><i class="fa fa-check"></i><b>6.12</b> Recommendations</a>
<ul>
<li class="chapter" data-level="6.12.1" data-path="p-values.html"><a href="p-values.html#primary-sources-for-recommendations"><i class="fa fa-check"></i><b>6.12.1</b> Primary sources for recommendations</a></li>
</ul></li>
<li class="chapter" data-level="6.13" data-path="p-values.html"><a href="p-values.html#problems"><i class="fa fa-check"></i><b>6.13</b> Problems</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="errors-in-inference.html"><a href="errors-in-inference.html"><i class="fa fa-check"></i><b>7</b> Errors in inference</a>
<ul>
<li class="chapter" data-level="7.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#classical-nhst-concepts-of-wrong"><i class="fa fa-check"></i><b>7.1</b> Classical NHST concepts of wrong</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-i-error"><i class="fa fa-check"></i><b>7.1.1</b> Type I error</a></li>
<li class="chapter" data-level="7.1.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#power"><i class="fa fa-check"></i><b>7.1.2</b> Power</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#a-non-neyman-pearson-concept-of-power"><i class="fa fa-check"></i><b>7.2</b> A non-Neyman-Pearson concept of power</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#estimation-error"><i class="fa fa-check"></i><b>7.2.1</b> Estimation error</a></li>
<li class="chapter" data-level="7.2.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#coverage"><i class="fa fa-check"></i><b>7.2.2</b> Coverage</a></li>
<li class="chapter" data-level="7.2.3" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-s-error"><i class="fa fa-check"></i><b>7.2.3</b> Type S error</a></li>
<li class="chapter" data-level="7.2.4" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-m-error"><i class="fa fa-check"></i><b>7.2.4</b> Type M error</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-iv-introduction-to-linear-models.html"><a href="part-iv-introduction-to-linear-models.html"><i class="fa fa-check"></i>Part IV: Introduction to Linear Models</a></li>
<li class="chapter" data-level="8" data-path="intro-linear-models.html"><a href="intro-linear-models.html"><i class="fa fa-check"></i><b>8</b> An introduction to linear models</a>
<ul>
<li class="chapter" data-level="8.1" data-path="intro-linear-models.html"><a href="intro-linear-models.html#lm-specifications"><i class="fa fa-check"></i><b>8.1</b> Two specifications of a linear model</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="intro-linear-models.html"><a href="intro-linear-models.html#the-error-draw-specification"><i class="fa fa-check"></i><b>8.1.1</b> The “error draw” specification</a></li>
<li class="chapter" data-level="8.1.2" data-path="intro-linear-models.html"><a href="intro-linear-models.html#the-conditional-draw-specification"><i class="fa fa-check"></i><b>8.1.2</b> The “conditional draw” specification</a></li>
<li class="chapter" data-level="8.1.3" data-path="intro-linear-models.html"><a href="intro-linear-models.html#comparing-the-error-draw-and-conditional-draw-ways-of-specifying-the-linear-model"><i class="fa fa-check"></i><b>8.1.3</b> Comparing the error-draw and conditional-draw ways of specifying the linear model</a></li>
<li class="chapter" data-level="8.1.4" data-path="intro-linear-models.html"><a href="intro-linear-models.html#anova-notation-of-a-linear-model"><i class="fa fa-check"></i><b>8.1.4</b> ANOVA notation of a linear model</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="intro-linear-models.html"><a href="intro-linear-models.html#a-linear-model-can-be-fit-to-data-with-continuous-discrete-or-categorical-x-variables"><i class="fa fa-check"></i><b>8.2</b> A linear model can be fit to data with continuous, discrete, or categorical <span class="math inline">\(X\)</span> variables</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="intro-linear-models.html"><a href="intro-linear-models.html#fitting-linear-models-to-experimental-data-in-which-the-x-variable-is-continuous-or-discrete"><i class="fa fa-check"></i><b>8.2.1</b> Fitting linear models to experimental data in which the <span class="math inline">\(X\)</span> variable is continuous or discrete</a></li>
<li class="chapter" data-level="8.2.2" data-path="intro-linear-models.html"><a href="intro-linear-models.html#slope"><i class="fa fa-check"></i><b>8.2.2</b> Fitting linear models to experimental data in which the <span class="math inline">\(X\)</span> variable is categorical</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="intro-linear-models.html"><a href="intro-linear-models.html#statistical-models-are-used-for-prediction-explanation-and-description"><i class="fa fa-check"></i><b>8.3</b> Statistical models are used for prediction, explanation, and description</a></li>
<li class="chapter" data-level="8.4" data-path="intro-linear-models.html"><a href="intro-linear-models.html#what-is-the-interpretation-of-a-regression-coefficient"><i class="fa fa-check"></i><b>8.4</b> What is the interpretation of a regression coefficient?</a></li>
<li class="chapter" data-level="8.5" data-path="intro-linear-models.html"><a href="intro-linear-models.html#what-do-we-call-the-x-and-y-variables"><i class="fa fa-check"></i><b>8.5</b> What do we call the <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span> variables?</a></li>
<li class="chapter" data-level="8.6" data-path="intro-linear-models.html"><a href="intro-linear-models.html#modeling-strategy"><i class="fa fa-check"></i><b>8.6</b> Modeling strategy</a></li>
<li class="chapter" data-level="8.7" data-path="intro-linear-models.html"><a href="intro-linear-models.html#predictions-from-the-model"><i class="fa fa-check"></i><b>8.7</b> Predictions from the model</a></li>
<li class="chapter" data-level="8.8" data-path="intro-linear-models.html"><a href="intro-linear-models.html#inference-from-the-model"><i class="fa fa-check"></i><b>8.8</b> Inference from the model</a>
<ul>
<li class="chapter" data-level="8.8.1" data-path="intro-linear-models.html"><a href="intro-linear-models.html#assumptions-for-inference-with-a-statistical-model"><i class="fa fa-check"></i><b>8.8.1</b> Assumptions for inference with a statistical model</a></li>
<li class="chapter" data-level="8.8.2" data-path="intro-linear-models.html"><a href="intro-linear-models.html#specific-assumptions-for-inference-with-a-linear-model"><i class="fa fa-check"></i><b>8.8.2</b> Specific assumptions for inference with a linear model</a></li>
</ul></li>
<li class="chapter" data-level="8.9" data-path="intro-linear-models.html"><a href="intro-linear-models.html#linear-modelregression-model-or-statistical-model"><i class="fa fa-check"></i><b>8.9</b> “linear model,”regression model”, or “statistical model”?</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="regression.html"><a href="regression.html"><i class="fa fa-check"></i><b>9</b> Linear models with a single, continuous <em>X</em> (“regression”)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="regression.html"><a href="regression.html#a-linear-model-with-a-single-continuous-x-is-classical-regression"><i class="fa fa-check"></i><b>9.1</b> A linear model with a single, continuous <em>X</em> is classical “regression”</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="regression.html"><a href="regression.html#analysis-of-green-down-data"><i class="fa fa-check"></i><b>9.1.1</b> Analysis of “green-down” data</a></li>
<li class="chapter" data-level="9.1.2" data-path="regression.html"><a href="regression.html#learning-from-the-green-down-example"><i class="fa fa-check"></i><b>9.1.2</b> Learning from the green-down example</a></li>
<li class="chapter" data-level="9.1.3" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-explanation-causal-models"><i class="fa fa-check"></i><b>9.1.3</b> Using a regression model for “explanation” – causal models</a></li>
<li class="chapter" data-level="9.1.4" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-prediction-prediction-models"><i class="fa fa-check"></i><b>9.1.4</b> Using a regression model for prediction – prediction models</a></li>
<li class="chapter" data-level="9.1.5" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-creating-a-new-response-variable-comparing-slopes-of-longitudinal-data"><i class="fa fa-check"></i><b>9.1.5</b> Using a regression model for creating a new response variable – comparing slopes of longitudinal data</a></li>
<li class="chapter" data-level="9.1.6" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-for-calibration"><i class="fa fa-check"></i><b>9.1.6</b> Using a regression model for for calibration</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="regression.html"><a href="regression.html#working-in-r-1"><i class="fa fa-check"></i><b>9.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="plotting-models.html"><a href="plotting-models.html#fit-the-model"><i class="fa fa-check"></i><b>9.2.1</b> Fitting the linear model</a></li>
<li class="chapter" data-level="9.2.2" data-path="regression.html"><a href="regression.html#getting-to-know-the-linear-model-the-summary-function"><i class="fa fa-check"></i><b>9.2.2</b> Getting to know the linear model: the <code>summary</code> function</a></li>
<li class="chapter" data-level="9.2.3" data-path="regression.html"><a href="regression.html#inference-the-coefficient-table"><i class="fa fa-check"></i><b>9.2.3</b> Inference – the coefficient table</a></li>
<li class="chapter" data-level="9.2.4" data-path="regression.html"><a href="regression.html#how-good-is-our-model-model-checking"><i class="fa fa-check"></i><b>9.2.4</b> How good is our model? – Model checking</a></li>
<li class="chapter" data-level="9.2.5" data-path="regression.html"><a href="regression.html#plotting-models-with-continuous-x"><i class="fa fa-check"></i><b>9.2.5</b> Plotting models with continuous <em>X</em></a></li>
<li class="chapter" data-level="9.2.6" data-path="regression.html"><a href="regression.html#creating-a-table-of-predicted-values-and-95-prediction-intervals"><i class="fa fa-check"></i><b>9.2.6</b> Creating a table of predicted values and 95% prediction intervals</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="regression.html"><a href="regression.html#hidden-code-1"><i class="fa fa-check"></i><b>9.3</b> Hidden code</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="regression.html"><a href="regression.html#import-and-plot-of-fig2c-ecosystem-warming-experimental-data"><i class="fa fa-check"></i><b>9.3.1</b> Import and plot of fig2c (ecosystem warming experimental) data</a></li>
<li class="chapter" data-level="9.3.2" data-path="regression.html"><a href="regression.html#import-and-plot-efig_3d-ecosysem-warming-observational-data"><i class="fa fa-check"></i><b>9.3.2</b> Import and plot efig_3d (Ecosysem warming observational) data</a></li>
<li class="chapter" data-level="9.3.3" data-path="regression.html"><a href="regression.html#import-and-plot-of-fig1f-methionine-restriction-data"><i class="fa fa-check"></i><b>9.3.3</b> Import and plot of fig1f (methionine restriction) data</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="regression.html"><a href="regression.html#try-it"><i class="fa fa-check"></i><b>9.4</b> Try it</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="regression.html"><a href="regression.html#a-prediction-model-from-the-literature"><i class="fa fa-check"></i><b>9.4.1</b> A prediction model from the literature</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="regression.html"><a href="regression.html#intuition-pumps"><i class="fa fa-check"></i><b>9.5</b> Intuition pumps</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="regression.html"><a href="regression.html#correlation-and-r2"><i class="fa fa-check"></i><b>9.5.1</b> Correlation and $R^2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="oneway.html"><a href="oneway.html"><i class="fa fa-check"></i><b>10</b> Linear models with a single, categorical <em>X</em> (“t-tests” and “ANOVA”)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="oneway.html"><a href="oneway.html#oneway-data"><i class="fa fa-check"></i><b>10.1</b> A linear model with a single factor (categorical <em>X</em>variable) estimates the effects of the levels of factor on the response</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="oneway.html"><a href="oneway.html#oneway-example1"><i class="fa fa-check"></i><b>10.1.1</b> Example 1 (fig3d) – two treatment levels (“groups”)</a></li>
<li class="chapter" data-level="10.1.2" data-path="oneway.html"><a href="oneway.html#understanding-the-analysis-with-two-treatment-levels"><i class="fa fa-check"></i><b>10.1.2</b> Understanding the analysis with two treatment levels</a></li>
<li class="chapter" data-level="10.1.3" data-path="oneway.html"><a href="oneway.html#oneway-example2"><i class="fa fa-check"></i><b>10.1.3</b> Example 2 – three treatment levels (“groups”)</a></li>
<li class="chapter" data-level="10.1.4" data-path="oneway.html"><a href="oneway.html#understanding-the-analysis-with-three-or-more-treatment-levels"><i class="fa fa-check"></i><b>10.1.4</b> Understanding the analysis with three (or more) treatment levels</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="oneway.html"><a href="oneway.html#working-in-r-2"><i class="fa fa-check"></i><b>10.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="oneway.html"><a href="oneway.html#fit-the-model-1"><i class="fa fa-check"></i><b>10.2.1</b> Fit the model</a></li>
<li class="chapter" data-level="10.2.2" data-path="oneway.html"><a href="oneway.html#controlling-the-output-in-tables-using-the-coefficient-table-as-an-example"><i class="fa fa-check"></i><b>10.2.2</b> Controlling the output in tables using the coefficient table as an example</a></li>
<li class="chapter" data-level="10.2.3" data-path="oneway.html"><a href="oneway.html#using-the-emmeans-function"><i class="fa fa-check"></i><b>10.2.3</b> Using the emmeans function</a></li>
<li class="chapter" data-level="10.2.4" data-path="oneway.html"><a href="oneway.html#using-the-contrast-function"><i class="fa fa-check"></i><b>10.2.4</b> Using the contrast function</a></li>
<li class="chapter" data-level="10.2.5" data-path="oneway.html"><a href="oneway.html#how-to-generate-anova-tables"><i class="fa fa-check"></i><b>10.2.5</b> How to generate ANOVA tables</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="oneway.html"><a href="oneway.html#hidden-code-2"><i class="fa fa-check"></i><b>10.3</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="oneway.html"><a href="oneway.html#importing-and-wrangling-the-fig3d-data-for-example-1"><i class="fa fa-check"></i><b>10.3.1</b> Importing and wrangling the fig3d data for example 1</a></li>
<li class="chapter" data-level="10.3.2" data-path="oneway.html"><a href="oneway.html#importing-and-wrangling-the-fig2a-data-for-example-2"><i class="fa fa-check"></i><b>10.3.2</b> Importing and wrangling the fig2a data for example 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="model-checking.html"><a href="model-checking.html"><i class="fa fa-check"></i><b>11</b> Model Checking</a>
<ul>
<li class="chapter" data-level="11.1" data-path="model-checking.html"><a href="model-checking.html#check-check"><i class="fa fa-check"></i><b>11.1</b> All statistical analyses should be followed by model checking</a></li>
<li class="chapter" data-level="11.2" data-path="model-checking.html"><a href="model-checking.html#linear-model-assumptions"><i class="fa fa-check"></i><b>11.2</b> Linear model assumptions</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="model-checking.html"><a href="model-checking.html#a-bit-about-iid"><i class="fa fa-check"></i><b>11.2.1</b> A bit about IID</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="model-checking.html"><a href="model-checking.html#diagnostic-plots-use-the-residuals-from-the-model-fit"><i class="fa fa-check"></i><b>11.3</b> Diagnostic plots use the residuals from the model fit</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="model-checking.html"><a href="model-checking.html#residuals"><i class="fa fa-check"></i><b>11.3.1</b> Residuals</a></li>
<li class="chapter" data-level="11.3.2" data-path="model-checking.html"><a href="model-checking.html#normal-qq"><i class="fa fa-check"></i><b>11.3.2</b> A Normal Q-Q plot is used to check for characteristic departures from Normality</a></li>
<li class="chapter" data-level="11.3.3" data-path="model-checking.html"><a href="model-checking.html#mapping-qq-plot-departures-from-normality"><i class="fa fa-check"></i><b>11.3.3</b> Mapping QQ-plot departures from Normality</a></li>
<li class="chapter" data-level="11.3.4" data-path="model-checking.html"><a href="model-checking.html#model-checking-spread-level"><i class="fa fa-check"></i><b>11.3.4</b> Model checking homoskedasticity</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="model-checking.html"><a href="model-checking.html#using-r"><i class="fa fa-check"></i><b>11.4</b> Using R</a></li>
<li class="chapter" data-level="11.5" data-path="model-checking.html"><a href="model-checking.html#hidden-code-3"><i class="fa fa-check"></i><b>11.5</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="model-checking.html"><a href="model-checking.html#normal-q-q-plots"><i class="fa fa-check"></i><b>11.5.1</b> Normal Q-Q plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="violations.html"><a href="violations.html"><i class="fa fa-check"></i><b>12</b> Violations of independence, homogeneity, or Normality</a>
<ul>
<li class="chapter" data-level="12.1" data-path="violations.html"><a href="violations.html#oneway-paired-t"><i class="fa fa-check"></i><b>12.1</b> Lack of independence</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="violations.html"><a href="violations.html#violations-paired-t"><i class="fa fa-check"></i><b>12.1.1</b> Example 1 (exp1b) – a paired t-test is a special case of a linear mixed model</a></li>
<li class="chapter" data-level="12.1.2" data-path="violations.html"><a href="violations.html#violations-rmanova"><i class="fa fa-check"></i><b>12.1.2</b> Example 2 (diHOME exp2a) – A repeated measures ANOVA is a special case of a linear mixed model</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="violations.html"><a href="violations.html#oneway-welch"><i class="fa fa-check"></i><b>12.2</b> Heterogeneity of variances</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="violations.html"><a href="violations.html#when-groups-of-the-focal-test-have-variance"><i class="fa fa-check"></i><b>12.2.1</b> When groups of the focal test have &gt;&gt; variance</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="violations.html"><a href="violations.html#the-conditional-response-isnt-normal"><i class="fa fa-check"></i><b>12.3</b> The conditional response isn’t Normal</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="violations.html"><a href="violations.html#violations-6f"><i class="fa fa-check"></i><b>12.3.1</b> Example 1 (fig6f) – Linear models for non-normal count data</a></li>
<li class="chapter" data-level="12.3.2" data-path="violations.html"><a href="violations.html#my-data-arent-normal-what-is-the-best-practice"><i class="fa fa-check"></i><b>12.3.2</b> My data aren’t normal, what is the best practice?</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="violations.html"><a href="violations.html#hidden-code-4"><i class="fa fa-check"></i><b>12.4</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-exp1b-data"><i class="fa fa-check"></i><b>12.4.1</b> Importing and wrangling the exp1b data</a></li>
<li class="chapter" data-level="12.4.2" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-exp2a-data"><i class="fa fa-check"></i><b>12.4.2</b> Importing and wrangling the exp2a data</a></li>
<li class="chapter" data-level="12.4.3" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-fig6f-data"><i class="fa fa-check"></i><b>12.4.3</b> Importing and wrangling the fig6f data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="issues.html"><a href="issues.html"><i class="fa fa-check"></i><b>13</b> Issues in inference</a>
<ul>
<li class="chapter" data-level="13.1" data-path="issues.html"><a href="issues.html#replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means"><i class="fa fa-check"></i><b>13.1</b> Replicated experiments – include <span class="math inline">\(\texttt{Experiment}\)</span> as a random factor (better than one-way ANOVA of means)</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="issues.html"><a href="issues.html#issues-exp4d"><i class="fa fa-check"></i><b>13.1.1</b> Multiple experiments Example 1 (wound healing Exp4d)</a></li>
<li class="chapter" data-level="13.1.2" data-path="issues.html"><a href="issues.html#issues-combined-experiments"><i class="fa fa-check"></i><b>13.1.2</b> Models for combining replicated experiments</a></li>
<li class="chapter" data-level="13.1.3" data-path="issues.html"><a href="issues.html#understanding-model-exp4d_m1"><i class="fa fa-check"></i><b>13.1.3</b> Understanding Model <code>exp4d_m1</code></a></li>
<li class="chapter" data-level="13.1.4" data-path="issues.html"><a href="issues.html#the-univariate-model-is-equivalent-to-a-linear-mixed-model-of-the-aggregated-data-model-exp4d_m2"><i class="fa fa-check"></i><b>13.1.4</b> The univariate model is equivalent to a linear mixed model of the aggregated data (Model exp4d_m2)</a></li>
<li class="chapter" data-level="13.1.5" data-path="issues.html"><a href="issues.html#a-linear-mixed-model-of-the-full-data"><i class="fa fa-check"></i><b>13.1.5</b> A linear mixed model of the full data</a></li>
<li class="chapter" data-level="13.1.6" data-path="issues.html"><a href="issues.html#analysis-of-the-experiment-means-has-less-precision-and-power"><i class="fa fa-check"></i><b>13.1.6</b> Analysis of the experiment means has less precision and power</a></li>
<li class="chapter" data-level="13.1.7" data-path="issues.html"><a href="issues.html#dont-do-this-a-t-testfixed-effect-anova-of-the-full-data"><i class="fa fa-check"></i><b>13.1.7</b> Don’t do this – a t-test/fixed-effect ANOVA of the full data</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="issues.html"><a href="issues.html#issues-pre-post"><i class="fa fa-check"></i><b>13.2</b> Comparing change from baseline (pre-post)</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="issues.html"><a href="issues.html#pre-post-example-1-dpp4-fig4c"><i class="fa fa-check"></i><b>13.2.1</b> Pre-post example 1 (DPP4 fig4c)</a></li>
<li class="chapter" data-level="13.2.2" data-path="issues.html"><a href="issues.html#what-if-the-data-in-example-1-were-from-from-an-experiment-where-the-treatment-was-applied-prior-to-the-baseline-measure"><i class="fa fa-check"></i><b>13.2.2</b> What if the data in example 1 were from from an experiment where the treatment was applied prior to the baseline measure?</a></li>
<li class="chapter" data-level="13.2.3" data-path="issues.html"><a href="issues.html#pre-post-example-2-xx-males-fig1c"><i class="fa fa-check"></i><b>13.2.3</b> Pre-post example 2 (XX males fig1c)</a></li>
<li class="chapter" data-level="13.2.4" data-path="issues.html"><a href="issues.html#issues-regression-to-mean"><i class="fa fa-check"></i><b>13.2.4</b> Regression to the mean</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="issues.html"><a href="issues.html#longitudinal-designs-with-more-than-one-post-baseline-measure"><i class="fa fa-check"></i><b>13.3</b> Longitudinal designs with more than one-post baseline measure</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="issues.html"><a href="issues.html#issues-auc"><i class="fa fa-check"></i><b>13.3.1</b> Area under the curve (AUC)</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="issues.html"><a href="issues.html#normalization-the-analysis-of-ratios"><i class="fa fa-check"></i><b>13.4</b> Normalization – the analysis of ratios</a>
<ul>
<li class="chapter" data-level="13.4.1" data-path="issues.html"><a href="issues.html#kinds-of-ratios-in-experimental-biology"><i class="fa fa-check"></i><b>13.4.1</b> Kinds of ratios in experimental biology</a></li>
<li class="chapter" data-level="13.4.2" data-path="issues.html"><a href="issues.html#issues-offset"><i class="fa fa-check"></i><b>13.4.2</b> Example 1 – The ratio is a density (number of something per area)</a></li>
<li class="chapter" data-level="13.4.3" data-path="issues.html"><a href="issues.html#issues-size"><i class="fa fa-check"></i><b>13.4.3</b> Example 2 – The ratio is normalizing for size differences</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="issues.html"><a href="issues.html#dont-do-this-stuff"><i class="fa fa-check"></i><b>13.5</b> Don’t do this stuff</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="issues.html"><a href="issues.html#normalize-the-response-so-that-all-control-values-are-equal-to-1."><i class="fa fa-check"></i><b>13.5.1</b> Normalize the response so that all control values are equal to 1.</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="issues.html"><a href="issues.html#a-difference-in-significance-is-not-necessarily-significant"><i class="fa fa-check"></i><b>13.6</b> A difference in significance is not necessarily significant</a></li>
<li class="chapter" data-level="13.7" data-path="issues.html"><a href="issues.html#researcher-degrees-of-freedom"><i class="fa fa-check"></i><b>13.7</b> Researcher degrees of freedom</a></li>
<li class="chapter" data-level="13.8" data-path="issues.html"><a href="issues.html#hidden-code-5"><i class="fa fa-check"></i><b>13.8</b> Hidden code</a>
<ul>
<li class="chapter" data-level="13.8.1" data-path="issues.html"><a href="issues.html#import-exp4d-vimentin-cell-count-data-replicate-experiments-example"><i class="fa fa-check"></i><b>13.8.1</b> Import exp4d vimentin cell count data (replicate experiments example)</a></li>
<li class="chapter" data-level="13.8.2" data-path="issues.html"><a href="issues.html#import-fig4c-data"><i class="fa fa-check"></i><b>13.8.2</b> Import Fig4c data</a></li>
<li class="chapter" data-level="13.8.3" data-path="issues.html"><a href="issues.html#xx-males-fig1c"><i class="fa fa-check"></i><b>13.8.3</b> XX males fig1c</a></li>
<li class="chapter" data-level="13.8.4" data-path="issues.html"><a href="issues.html#generation-of-fake-data-to-illustrate-regression-to-the-mean"><i class="fa fa-check"></i><b>13.8.4</b> Generation of fake data to illustrate regression to the mean</a></li>
<li class="chapter" data-level="13.8.5" data-path="issues.html"><a href="issues.html#import-fig3f"><i class="fa fa-check"></i><b>13.8.5</b> Import fig3f</a></li>
<li class="chapter" data-level="13.8.6" data-path="issues.html"><a href="issues.html#issues-import-exp3b"><i class="fa fa-check"></i><b>13.8.6</b> Import exp3b</a></li>
<li class="chapter" data-level="13.8.7" data-path="issues.html"><a href="issues.html#issues-exp3b-plot"><i class="fa fa-check"></i><b>13.8.7</b> Plot the model of exp3b (glm offset data)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-v-more-than-one-x-multivariable-models.html"><a href="part-v-more-than-one-x-multivariable-models.html"><i class="fa fa-check"></i>Part V: More than one <span class="math inline">\(X\)</span> – Multivariable Models</a></li>
<li class="chapter" data-level="14" data-path="covariates.html"><a href="covariates.html"><i class="fa fa-check"></i><b>14</b> Linear models with added covariates (“ANCOVA”)</a>
<ul>
<li class="chapter" data-level="14.1" data-path="covariates.html"><a href="covariates.html#adding-covariates-can-increases-the-precision-of-the-effect-of-interest"><i class="fa fa-check"></i><b>14.1</b> Adding covariates can increases the precision of the effect of interest</a></li>
<li class="chapter" data-level="14.2" data-path="covariates.html"><a href="covariates.html#understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data"><i class="fa fa-check"></i><b>14.2</b> Understanding a linear model with an added covariate – heart necrosis data</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="covariates.html"><a href="covariates.html#fit-the-model-5"><i class="fa fa-check"></i><b>14.2.1</b> Fit the model</a></li>
<li class="chapter" data-level="14.2.2" data-path="covariates.html"><a href="covariates.html#plot-the-model-2"><i class="fa fa-check"></i><b>14.2.2</b> Plot the model</a></li>
<li class="chapter" data-level="14.2.3" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-model-coefficients"><i class="fa fa-check"></i><b>14.2.3</b> Interpretation of the model coefficients</a></li>
<li class="chapter" data-level="14.2.4" data-path="covariates.html"><a href="covariates.html#everything-adds-up"><i class="fa fa-check"></i><b>14.2.4</b> Everything adds up</a></li>
<li class="chapter" data-level="14.2.5" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-estimated-marginal-means"><i class="fa fa-check"></i><b>14.2.5</b> Interpretation of the estimated marginal means</a></li>
<li class="chapter" data-level="14.2.6" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-contrasts"><i class="fa fa-check"></i><b>14.2.6</b> Interpretation of the contrasts</a></li>
<li class="chapter" data-level="14.2.7" data-path="covariates.html"><a href="covariates.html#adding-the-covariate-improves-inference"><i class="fa fa-check"></i><b>14.2.7</b> Adding the covariate improves inference</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="covariates.html"><a href="covariates.html#understanding-interaction-effects-with-covariates"><i class="fa fa-check"></i><b>14.3</b> Understanding interaction effects with covariates</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="covariates.html"><a href="covariates.html#fit-the-model-6"><i class="fa fa-check"></i><b>14.3.1</b> Fit the model</a></li>
<li class="chapter" data-level="14.3.2" data-path="covariates.html"><a href="covariates.html#plot-the-model-with-interaction-effect"><i class="fa fa-check"></i><b>14.3.2</b> Plot the model with interaction effect</a></li>
<li class="chapter" data-level="14.3.3" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-model-coefficients-1"><i class="fa fa-check"></i><b>14.3.3</b> Interpretation of the model coefficients</a></li>
<li class="chapter" data-level="14.3.4" data-path="covariates.html"><a href="covariates.html#what-is-the-effect-of-a-treatment-if-interactions-are-modeled-it-depends."><i class="fa fa-check"></i><b>14.3.4</b> What is the effect of a treatment, if interactions are modeled? – it depends.</a></li>
<li class="chapter" data-level="14.3.5" data-path="covariates.html"><a href="covariates.html#which-model-do-we-use-mathcalm_1-or-mathcalm_2"><i class="fa fa-check"></i><b>14.3.5</b> Which model do we use, <span class="math inline">\(\mathcal{M}_1\)</span> or <span class="math inline">\(\mathcal{M}_2\)</span>?</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="covariates.html"><a href="covariates.html#understanding-ancova-tables"><i class="fa fa-check"></i><b>14.4</b> Understanding ANCOVA tables</a></li>
<li class="chapter" data-level="14.5" data-path="covariates.html"><a href="covariates.html#working-in-r-3"><i class="fa fa-check"></i><b>14.5</b> Working in R</a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="covariates.html"><a href="covariates.html#importing-the-heart-necrosis-data"><i class="fa fa-check"></i><b>14.5.1</b> Importing the heart necrosis data</a></li>
<li class="chapter" data-level="14.5.2" data-path="covariates.html"><a href="covariates.html#fitting-the-model"><i class="fa fa-check"></i><b>14.5.2</b> Fitting the model</a></li>
<li class="chapter" data-level="14.5.3" data-path="covariates.html"><a href="covariates.html#using-the-emmeans-function-1"><i class="fa fa-check"></i><b>14.5.3</b> Using the emmeans function</a></li>
<li class="chapter" data-level="14.5.4" data-path="covariates.html"><a href="covariates.html#ancova-tables"><i class="fa fa-check"></i><b>14.5.4</b> ANCOVA tables</a></li>
<li class="chapter" data-level="14.5.5" data-path="covariates.html"><a href="covariates.html#plotting-the-model"><i class="fa fa-check"></i><b>14.5.5</b> Plotting the model</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="covariates.html"><a href="covariates.html#best-practices"><i class="fa fa-check"></i><b>14.6</b> Best practices</a>
<ul>
<li class="chapter" data-level="14.6.1" data-path="covariates.html"><a href="covariates.html#do-not-use-a-ratio-of-partwhole-as-a-response-variable-instead-add-the-denominator-as-a-covariate"><i class="fa fa-check"></i><b>14.6.1</b> Do not use a ratio of part:whole as a response variable – instead add the denominator as a covariate</a></li>
<li class="chapter" data-level="14.6.2" data-path="covariates.html"><a href="covariates.html#do-not-use-change-from-baseline-as-a-response-variable-instead-add-the-baseline-measure-as-a-covariate"><i class="fa fa-check"></i><b>14.6.2</b> Do not use change from baseline as a response variable – instead add the baseline measure as a covariate</a></li>
<li class="chapter" data-level="14.6.3" data-path="covariates.html"><a href="covariates.html#do-not-test-for-balance-of-baseline-measures"><i class="fa fa-check"></i><b>14.6.3</b> Do not “test for balance” of baseline measures</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="covariates.html"><a href="covariates.html#best-practices-2-use-a-covariate-instead-of-normalizing-a-response"><i class="fa fa-check"></i><b>14.7</b> Best practices 2: Use a covariate instead of normalizing a response</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="factorial.html"><a href="factorial.html"><i class="fa fa-check"></i><b>15</b> Linear models with two categorical <span class="math inline">\(X\)</span> – Factorial linear models (“two-way ANOVA”)</a>
<ul>
<li class="chapter" data-level="15.1" data-path="factorial.html"><a href="factorial.html#a-linear-model-with-crossed-factors-estimates-interaction-effects"><i class="fa fa-check"></i><b>15.1</b> A linear model with crossed factors estimates interaction effects</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="factorial.html"><a href="factorial.html#an-interaction-is-a-difference-in-simple-effects"><i class="fa fa-check"></i><b>15.1.1</b> An interaction is a difference in simple effects</a></li>
<li class="chapter" data-level="15.1.2" data-path="factorial.html"><a href="factorial.html#a-linear-model-with-crossed-factors-includes-interaction-effects"><i class="fa fa-check"></i><b>15.1.2</b> A linear model with crossed factors includes interaction effects</a></li>
<li class="chapter" data-level="15.1.3" data-path="factorial.html"><a href="factorial.html#factorial-experiments-are-frequently-analyzed-as-flattened-linear-models-in-the-experimental-biology-literature"><i class="fa fa-check"></i><b>15.1.3</b> factorial experiments are frequently analyzed as flattened linear models in the experimental biology literature</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="factorial.html"><a href="factorial.html#example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data"><i class="fa fa-check"></i><b>15.2</b> Example 1 – Estimation of a treatment effect relative to a control effect (“Something different”) (Experiment 2j glucose uptake data)</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="factorial.html"><a href="factorial.html#twoway-understand1"><i class="fa fa-check"></i><b>15.2.1</b> Understand the experimental design</a></li>
<li class="chapter" data-level="15.2.2" data-path="factorial.html"><a href="factorial.html#fit-the-linear-model"><i class="fa fa-check"></i><b>15.2.2</b> Fit the linear model</a></li>
<li class="chapter" data-level="15.2.3" data-path="factorial.html"><a href="factorial.html#twoway-exp2j-inf"><i class="fa fa-check"></i><b>15.2.3</b> Inference</a></li>
<li class="chapter" data-level="15.2.4" data-path="factorial.html"><a href="factorial.html#plot-the-model-3"><i class="fa fa-check"></i><b>15.2.4</b> Plot the model</a></li>
<li class="chapter" data-level="15.2.5" data-path="factorial.html"><a href="factorial.html#alternaplot-the-model"><i class="fa fa-check"></i><b>15.2.5</b> alternaPlot the model</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-1"><i class="fa fa-check"></i><b>15.3</b> Understanding the linear model with crossed factors 1</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="factorial.html"><a href="factorial.html#twoway-what-coefs-are"><i class="fa fa-check"></i><b>15.3.1</b> What the coefficients are</a></li>
<li class="chapter" data-level="15.3.2" data-path="factorial.html"><a href="factorial.html#the-interaction-effect-is-something-different"><i class="fa fa-check"></i><b>15.3.2</b> The interaction effect is something different</a></li>
<li class="chapter" data-level="15.3.3" data-path="factorial.html"><a href="factorial.html#why-we-want-to-compare-the-treatment-effect-to-a-control-effect"><i class="fa fa-check"></i><b>15.3.3</b> Why we want to compare the treatment effect to a control effect</a></li>
<li class="chapter" data-level="15.3.4" data-path="factorial.html"><a href="factorial.html#the-order-of-the-factors-in-the-model-tells-the-same-story-differently"><i class="fa fa-check"></i><b>15.3.4</b> The order of the factors in the model tells the same story differently</a></li>
<li class="chapter" data-level="15.3.5" data-path="factorial.html"><a href="factorial.html#power-for-the-interaction-effect-is-less-than-that-for-simple-effects"><i class="fa fa-check"></i><b>15.3.5</b> Power for the interaction effect is less than that for simple effects</a></li>
<li class="chapter" data-level="15.3.6" data-path="factorial.html"><a href="factorial.html#planned-comparisons-vs.-post-hoc-tests"><i class="fa fa-check"></i><b>15.3.6</b> Planned comparisons vs. post-hoc tests</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="factorial.html"><a href="factorial.html#example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data"><i class="fa fa-check"></i><b>15.4</b> Example 2: Estimation of the effect of background condition on an effect (“it depends”) (Experiment 3e lesian area data)</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="factorial.html"><a href="factorial.html#understand-the-experimental-design"><i class="fa fa-check"></i><b>15.4.1</b> Understand the experimental design</a></li>
<li class="chapter" data-level="15.4.2" data-path="factorial.html"><a href="factorial.html#fit-the-linear-model-1"><i class="fa fa-check"></i><b>15.4.2</b> Fit the linear model</a></li>
<li class="chapter" data-level="15.4.3" data-path="factorial.html"><a href="factorial.html#check-the-model-1"><i class="fa fa-check"></i><b>15.4.3</b> Check the model</a></li>
<li class="chapter" data-level="15.4.4" data-path="factorial.html"><a href="factorial.html#inference-from-the-model-2"><i class="fa fa-check"></i><b>15.4.4</b> Inference from the model</a></li>
<li class="chapter" data-level="15.4.5" data-path="factorial.html"><a href="factorial.html#plot-the-model-4"><i class="fa fa-check"></i><b>15.4.5</b> Plot the model</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-2"><i class="fa fa-check"></i><b>15.5</b> Understanding the linear model with crossed factors 2</a>
<ul>
<li class="chapter" data-level="15.5.1" data-path="factorial.html"><a href="factorial.html#twoway-marginal-means"><i class="fa fa-check"></i><b>15.5.1</b> Conditional and marginal means</a></li>
<li class="chapter" data-level="15.5.2" data-path="factorial.html"><a href="factorial.html#simple-conditional-effects"><i class="fa fa-check"></i><b>15.5.2</b> Simple (conditional) effects</a></li>
<li class="chapter" data-level="15.5.3" data-path="factorial.html"><a href="factorial.html#twoway-marginal-effects"><i class="fa fa-check"></i><b>15.5.3</b> Marginal effects</a></li>
<li class="chapter" data-level="15.5.4" data-path="factorial.html"><a href="factorial.html#the-additive-model"><i class="fa fa-check"></i><b>15.5.4</b> The additive model</a></li>
<li class="chapter" data-level="15.5.5" data-path="factorial.html"><a href="factorial.html#twoway-reduce"><i class="fa fa-check"></i><b>15.5.5</b> Reduce models for the right reason</a></li>
<li class="chapter" data-level="15.5.6" data-path="factorial.html"><a href="factorial.html#the-marginal-means-of-an-additive-linear-model-with-two-factors-can-be-weird"><i class="fa fa-check"></i><b>15.5.6</b> The marginal means of an additive linear model with two factors can be weird</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="factorial.html"><a href="factorial.html#example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data"><i class="fa fa-check"></i><b>15.6</b> Example 3: Estimation of synergy (“More than the sum of the parts”) (Experiment 1c JA data)</a>
<ul>
<li class="chapter" data-level="15.6.1" data-path="factorial.html"><a href="factorial.html#examine-the-data"><i class="fa fa-check"></i><b>15.6.1</b> Examine the data</a></li>
<li class="chapter" data-level="15.6.2" data-path="factorial.html"><a href="factorial.html#fit-the-model-7"><i class="fa fa-check"></i><b>15.6.2</b> Fit the model</a></li>
<li class="chapter" data-level="15.6.3" data-path="factorial.html"><a href="factorial.html#model-check"><i class="fa fa-check"></i><b>15.6.3</b> Model check</a></li>
<li class="chapter" data-level="15.6.4" data-path="factorial.html"><a href="factorial.html#inference-from-the-model-3"><i class="fa fa-check"></i><b>15.6.4</b> Inference from the model</a></li>
<li class="chapter" data-level="15.6.5" data-path="factorial.html"><a href="factorial.html#plot-the-model-5"><i class="fa fa-check"></i><b>15.6.5</b> Plot the model</a></li>
<li class="chapter" data-level="15.6.6" data-path="factorial.html"><a href="factorial.html#alternative-plot"><i class="fa fa-check"></i><b>15.6.6</b> Alternative plot</a></li>
</ul></li>
<li class="chapter" data-level="15.7" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-3"><i class="fa fa-check"></i><b>15.7</b> Understanding the linear model with crossed factors 3</a>
<ul>
<li class="chapter" data-level="15.7.1" data-path="factorial.html"><a href="factorial.html#thinking-about-the-coefficients-of-the-linear-model"><i class="fa fa-check"></i><b>15.7.1</b> Thinking about the coefficients of the linear model</a></li>
</ul></li>
<li class="chapter" data-level="15.8" data-path="factorial.html"><a href="factorial.html#issues-in-inference"><i class="fa fa-check"></i><b>15.8</b> Issues in inference</a>
<ul>
<li class="chapter" data-level="15.8.1" data-path="factorial.html"><a href="factorial.html#for-pairwise-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model"><i class="fa fa-check"></i><b>15.8.1</b> For pairwise contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</a></li>
<li class="chapter" data-level="15.8.2" data-path="factorial.html"><a href="factorial.html#for-interaction-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model"><i class="fa fa-check"></i><b>15.8.2</b> For interaction contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</a></li>
<li class="chapter" data-level="15.8.3" data-path="factorial.html"><a href="factorial.html#twoway-multiple-tests"><i class="fa fa-check"></i><b>15.8.3</b> Adjusting <em>p</em>-values for multiple tests</a></li>
</ul></li>
<li class="chapter" data-level="15.9" data-path="factorial.html"><a href="factorial.html#two-way-anova"><i class="fa fa-check"></i><b>15.9</b> Two-way ANOVA</a>
<ul>
<li class="chapter" data-level="15.9.1" data-path="factorial.html"><a href="factorial.html#how-to-read-a-two-way-anova-table"><i class="fa fa-check"></i><b>15.9.1</b> How to read a two-way ANOVA table</a></li>
<li class="chapter" data-level="15.9.2" data-path="factorial.html"><a href="factorial.html#what-do-the-main-effects-in-an-anova-table-mean"><i class="fa fa-check"></i><b>15.9.2</b> What do the main effects in an ANOVA table mean?</a></li>
</ul></li>
<li class="chapter" data-level="15.10" data-path="factorial.html"><a href="factorial.html#more-issues-in-inference"><i class="fa fa-check"></i><b>15.10</b> More issues in inference</a>
<ul>
<li class="chapter" data-level="15.10.1" data-path="factorial.html"><a href="factorial.html#longitudinal-experiments-include-time-as-a-random-factor-better-than-repeated-measures-anova"><i class="fa fa-check"></i><b>15.10.1</b> Longitudinal experiments – include Time as a random factor (better than repeated measures ANOVA)</a></li>
</ul></li>
<li class="chapter" data-level="15.11" data-path="factorial.html"><a href="factorial.html#working-in-r-4"><i class="fa fa-check"></i><b>15.11</b> Working in R</a>
<ul>
<li class="chapter" data-level="15.11.1" data-path="factorial.html"><a href="factorial.html#model-formula"><i class="fa fa-check"></i><b>15.11.1</b> Model formula</a></li>
<li class="chapter" data-level="15.11.2" data-path="factorial.html"><a href="factorial.html#using-the-emmeans-function-2"><i class="fa fa-check"></i><b>15.11.2</b> Using the emmeans function</a></li>
<li class="chapter" data-level="15.11.3" data-path="factorial.html"><a href="factorial.html#contrasts"><i class="fa fa-check"></i><b>15.11.3</b> Contrasts</a></li>
<li class="chapter" data-level="15.11.4" data-path="factorial.html"><a href="factorial.html#practice-safe-anova"><i class="fa fa-check"></i><b>15.11.4</b> Practice safe ANOVA</a></li>
<li class="chapter" data-level="15.11.5" data-path="factorial.html"><a href="factorial.html#better-to-avoid-these"><i class="fa fa-check"></i><b>15.11.5</b> Better to avoid these</a></li>
</ul></li>
<li class="chapter" data-level="15.12" data-path="factorial.html"><a href="factorial.html#hidden-code-6"><i class="fa fa-check"></i><b>15.12</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="15.12.1" data-path="factorial.html"><a href="factorial.html#import-exp2j-example-1"><i class="fa fa-check"></i><b>15.12.1</b> Import exp2j (Example 1)</a></li>
<li class="chapter" data-level="15.12.2" data-path="factorial.html"><a href="factorial.html#import-exp3e-lesian-area-data-example-2"><i class="fa fa-check"></i><b>15.12.2</b> Import exp3e lesian area data (Example 2)</a></li>
<li class="chapter" data-level="15.12.3" data-path="factorial.html"><a href="factorial.html#import-exp1c-ja-data-example-3"><i class="fa fa-check"></i><b>15.12.3</b> Import Exp1c JA data (Example 3)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-vi-expanding-the-linear-model.html"><a href="part-vi-expanding-the-linear-model.html"><i class="fa fa-check"></i>Part VI – Expanding the Linear Model</a></li>
<li class="chapter" data-level="16" data-path="lmm.html"><a href="lmm.html"><i class="fa fa-check"></i><b>16</b> Models with random factors – linear mixed models</a>
<ul>
<li class="chapter" data-level="16.1" data-path="lmm.html"><a href="lmm.html#chapter-summary"><i class="fa fa-check"></i><b>16.1</b> Chapter summary</a></li>
<li class="chapter" data-level="16.2" data-path="lmm.html"><a href="lmm.html#introduction"><i class="fa fa-check"></i><b>16.2</b> Introduction</a></li>
<li class="chapter" data-level="16.3" data-path="lmm.html"><a href="lmm.html#experimental-designs-in-experimental-bench-biology"><i class="fa fa-check"></i><b>16.3</b> Experimental designs in experimental bench biology</a>
<ul>
<li class="chapter" data-level="16.3.1" data-path="lmm.html"><a href="lmm.html#notation-for-models"><i class="fa fa-check"></i><b>16.3.1</b> Notation for models</a></li>
<li class="chapter" data-level="16.3.2" data-path="lmm.html"><a href="lmm.html#completely-randomized-design-crd"><i class="fa fa-check"></i><b>16.3.2</b> Completely Randomized Design (CRD)</a></li>
<li class="chapter" data-level="16.3.3" data-path="lmm.html"><a href="lmm.html#completely-randomized-design-with-subsampling-crds-or-nested-design"><i class="fa fa-check"></i><b>16.3.3</b> Completely Randomized Design with Subsampling (CRDS) or “Nested Design”</a></li>
<li class="chapter" data-level="16.3.4" data-path="lmm.html"><a href="lmm.html#randomized-complete-block-design-rcbd"><i class="fa fa-check"></i><b>16.3.4</b> Randomized Complete Block Design (RCBD)</a></li>
<li class="chapter" data-level="16.3.5" data-path="lmm.html"><a href="lmm.html#randomized-split-plot-design-rspd"><i class="fa fa-check"></i><b>16.3.5</b> Randomized Split Plot Design (RSPD)</a></li>
<li class="chapter" data-level="16.3.6" data-path="lmm.html"><a href="lmm.html#generalized-randomized-complete-block-design-grcbd"><i class="fa fa-check"></i><b>16.3.6</b> Generalized Randomized Complete Block Design (GRCBD)</a></li>
<li class="chapter" data-level="16.3.7" data-path="lmm.html"><a href="lmm.html#nested-randomized-complete-block-design-nrcbd"><i class="fa fa-check"></i><b>16.3.7</b> Nested Randomized Complete Block Design (NRCBD)</a></li>
<li class="chapter" data-level="16.3.8" data-path="lmm.html"><a href="lmm.html#longitudinal-randomized-complete-block-design-lrcbd"><i class="fa fa-check"></i><b>16.3.8</b> Longitudinal Randomized Complete Block Design (LRCBD)</a></li>
<li class="chapter" data-level="16.3.9" data-path="lmm.html"><a href="lmm.html#variations-due-to-multiple-measures-of-the-response-variable"><i class="fa fa-check"></i><b>16.3.9</b> Variations due to multiple measures of the response variable</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="lmm.html"><a href="lmm.html#building-the-linear-mixed-model-for-clustered-data"><i class="fa fa-check"></i><b>16.4</b> Building the linear (mixed) model for clustered data</a></li>
<li class="chapter" data-level="16.5" data-path="lmm.html"><a href="lmm.html#lmm-demo"><i class="fa fa-check"></i><b>16.5</b> Example 1 – A random intercepts and slopes explainer (demo1)</a>
<ul>
<li class="chapter" data-level="16.5.1" data-path="lmm.html"><a href="lmm.html#batched-measurements-result-in-clustered-residuals"><i class="fa fa-check"></i><b>16.5.1</b> Batched measurements result in clustered residuals</a></li>
<li class="chapter" data-level="16.5.2" data-path="lmm.html"><a href="lmm.html#lmm-corerr"><i class="fa fa-check"></i><b>16.5.2</b> Clustered residuals result in correlated error</a></li>
<li class="chapter" data-level="16.5.3" data-path="lmm.html"><a href="lmm.html#in-blocked-designs-clustered-residuals-adds-a-variance-component-that-masks-treatment-effects"><i class="fa fa-check"></i><b>16.5.3</b> In blocked designs, clustered residuals adds a variance component that masks treatment effects</a></li>
<li class="chapter" data-level="16.5.4" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-are-linear-models-with-added-random-factors"><i class="fa fa-check"></i><b>16.5.4</b> Linear mixed models are linear models with added random factors</a></li>
<li class="chapter" data-level="16.5.5" data-path="lmm.html"><a href="lmm.html#what-the-random-effects-are"><i class="fa fa-check"></i><b>16.5.5</b> What the random effects are</a></li>
<li class="chapter" data-level="16.5.6" data-path="lmm.html"><a href="lmm.html#in-a-blocked-design-a-linear-model-with-added-random-effects-increases-precision-of-treatment-effects"><i class="fa fa-check"></i><b>16.5.6</b> In a blocked design, a linear model with added random effects increases precision of treatment effects</a></li>
<li class="chapter" data-level="16.5.7" data-path="lmm.html"><a href="lmm.html#lmm-varcorr"><i class="fa fa-check"></i><b>16.5.7</b> The correlation among random intercepts and slopes</a></li>
<li class="chapter" data-level="16.5.8" data-path="lmm.html"><a href="lmm.html#clustered-residuals-create-heterogeneity-among-treatments"><i class="fa fa-check"></i><b>16.5.8</b> Clustered residuals create heterogeneity among treatments</a></li>
<li class="chapter" data-level="16.5.9" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-are-flexible"><i class="fa fa-check"></i><b>16.5.9</b> Linear mixed models are flexible</a></li>
<li class="chapter" data-level="16.5.10" data-path="lmm.html"><a href="lmm.html#a-random-intercept-only-model"><i class="fa fa-check"></i><b>16.5.10</b> A random intercept only model</a></li>
<li class="chapter" data-level="16.5.11" data-path="lmm.html"><a href="lmm.html#a-model-including-an-interaction-intercept"><i class="fa fa-check"></i><b>16.5.11</b> A model including an interaction intercept</a></li>
<li class="chapter" data-level="16.5.12" data-path="lmm.html"><a href="lmm.html#lmm-demo-aic"><i class="fa fa-check"></i><b>16.5.12</b> AIC and model selection – which model to report?</a></li>
<li class="chapter" data-level="16.5.13" data-path="lmm.html"><a href="lmm.html#the-specification-of-random-effects-matters"><i class="fa fa-check"></i><b>16.5.13</b> The specification of random effects matters</a></li>
<li class="chapter" data-level="16.5.14" data-path="lmm.html"><a href="lmm.html#mixed-effect-and-repeated-measures-anova"><i class="fa fa-check"></i><b>16.5.14</b> Mixed Effect and Repeated Measures ANOVA</a></li>
<li class="chapter" data-level="16.5.15" data-path="lmm.html"><a href="lmm.html#pseudoreplication"><i class="fa fa-check"></i><b>16.5.15</b> Pseudoreplication</a></li>
</ul></li>
<li class="chapter" data-level="16.6" data-path="lmm.html"><a href="lmm.html#lmm-example2"><i class="fa fa-check"></i><b>16.6</b> Example 2 – experiments without subsampling replication (exp6g)</a>
<ul>
<li class="chapter" data-level="16.6.1" data-path="lmm.html"><a href="lmm.html#understand-the-data"><i class="fa fa-check"></i><b>16.6.1</b> Understand the data</a></li>
<li class="chapter" data-level="16.6.2" data-path="lmm.html"><a href="lmm.html#model-fit-and-inference"><i class="fa fa-check"></i><b>16.6.2</b> Model fit and inference</a></li>
<li class="chapter" data-level="16.6.3" data-path="lmm.html"><a href="lmm.html#the-model-exp6g_m1-adds-a-random-intercept-but-not-a-random-slope"><i class="fa fa-check"></i><b>16.6.3</b> The model exp6g_m1 adds a random intercept but not a random slope</a></li>
<li class="chapter" data-level="16.6.4" data-path="lmm.html"><a href="lmm.html#the-fixed-effect-coefficients-of-model-exp6g_m1"><i class="fa fa-check"></i><b>16.6.4</b> The fixed effect coefficients of model exp6g_m1</a></li>
<li class="chapter" data-level="16.6.5" data-path="lmm.html"><a href="lmm.html#the-random-intercept-coefficients-of-exp6g_m1"><i class="fa fa-check"></i><b>16.6.5</b> The random intercept coefficients of exp6g_m1</a></li>
<li class="chapter" data-level="16.6.6" data-path="lmm.html"><a href="lmm.html#the-random-and-residual-variance-and-the-intraclass-correlation-of-model-exp6g_m1"><i class="fa fa-check"></i><b>16.6.6</b> The random and residual variance and the intraclass correlation of model exp6g_m1</a></li>
<li class="chapter" data-level="16.6.7" data-path="lmm.html"><a href="lmm.html#the-linear-mixed-model-exp6g_m1-increases-precision-of-treatment-effects-relative-to-a-fixed-effects-model"><i class="fa fa-check"></i><b>16.6.7</b> The linear mixed model exp6g_m1 increases precision of treatment effects, relative to a fixed effects model</a></li>
<li class="chapter" data-level="16.6.8" data-path="lmm.html"><a href="lmm.html#lmm-exp6g-alt"><i class="fa fa-check"></i><b>16.6.8</b> Alternative models for exp6g</a></li>
<li class="chapter" data-level="16.6.9" data-path="lmm.html"><a href="lmm.html#paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models"><i class="fa fa-check"></i><b>16.6.9</b> Paired t-tests and repeated measures ANOVA are special cases of linear mixed models</a></li>
<li class="chapter" data-level="16.6.10" data-path="lmm.html"><a href="lmm.html#lmm-exp6g-rmanova"><i class="fa fa-check"></i><b>16.6.10</b> Classical (“univariate model”) repeated measures ANOVA of exp6g</a></li>
<li class="chapter" data-level="16.6.11" data-path="lmm.html"><a href="lmm.html#lmm-exp6g-rmanova-multi"><i class="fa fa-check"></i><b>16.6.11</b> “Multivariate model” repeated measures ANOVA</a></li>
<li class="chapter" data-level="16.6.12" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-vs-repeated-measures-anova"><i class="fa fa-check"></i><b>16.6.12</b> Linear mixed models vs repeated measures ANOVA</a></li>
<li class="chapter" data-level="16.6.13" data-path="lmm.html"><a href="lmm.html#modeling-textttmouse_id-as-a-fixed-effect"><i class="fa fa-check"></i><b>16.6.13</b> Modeling <span class="math inline">\(\texttt{mouse_id}\)</span> as a fixed effect</a></li>
</ul></li>
<li class="chapter" data-level="16.7" data-path="lmm.html"><a href="lmm.html#example-3-factorial-experiments-and-no-subsampling-replicates-exp5c"><i class="fa fa-check"></i><b>16.7</b> Example 3 – Factorial experiments and no subsampling replicates (exp5c)</a>
<ul>
<li class="chapter" data-level="16.7.1" data-path="lmm.html"><a href="lmm.html#understand-the-data-1"><i class="fa fa-check"></i><b>16.7.1</b> Understand the data</a></li>
<li class="chapter" data-level="16.7.2" data-path="lmm.html"><a href="lmm.html#examine-the-data-1"><i class="fa fa-check"></i><b>16.7.2</b> Examine the data</a></li>
<li class="chapter" data-level="16.7.3" data-path="lmm.html"><a href="lmm.html#lmm-exp5c-m1"><i class="fa fa-check"></i><b>16.7.3</b> Model fit and inference</a></li>
<li class="chapter" data-level="16.7.4" data-path="lmm.html"><a href="lmm.html#why-we-care-about-modeling-batch-in-exp5c"><i class="fa fa-check"></i><b>16.7.4</b> Why we care about modeling batch in exp5c</a></li>
<li class="chapter" data-level="16.7.5" data-path="lmm.html"><a href="lmm.html#the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts"><i class="fa fa-check"></i><b>16.7.5</b> The linear mixed model exp5c_m1 adds two random intercepts</a></li>
<li class="chapter" data-level="16.7.6" data-path="lmm.html"><a href="lmm.html#the-fixed-effect-coefficients-of-model-exp5c_m1"><i class="fa fa-check"></i><b>16.7.6</b> The fixed effect coefficients of model exp5c_m1</a></li>
<li class="chapter" data-level="16.7.7" data-path="lmm.html"><a href="lmm.html#the-random-effect-coefficients-of-model-exp5c_m1"><i class="fa fa-check"></i><b>16.7.7</b> The random effect coefficients of model exp5c_m1</a></li>
<li class="chapter" data-level="16.7.8" data-path="lmm.html"><a href="lmm.html#lmm-exp5c-alt"><i class="fa fa-check"></i><b>16.7.8</b> Alternative models for exp5c</a></li>
<li class="chapter" data-level="16.7.9" data-path="lmm.html"><a href="lmm.html#lmm-rmanova-5c"><i class="fa fa-check"></i><b>16.7.9</b> Classical (“univariate model”) repeated measures ANOVA</a></li>
<li class="chapter" data-level="16.7.10" data-path="lmm.html"><a href="lmm.html#lmm-rmanova-5c-multi"><i class="fa fa-check"></i><b>16.7.10</b> “Multivariate model” repeated measures ANOVA of exp5c</a></li>
<li class="chapter" data-level="16.7.11" data-path="lmm.html"><a href="lmm.html#modeling-textttdonor-as-a-fixed-effect"><i class="fa fa-check"></i><b>16.7.11</b> Modeling <span class="math inline">\(\texttt{donor}\)</span> as a fixed effect</a></li>
</ul></li>
<li class="chapter" data-level="16.8" data-path="lmm.html"><a href="lmm.html#lmm-example4"><i class="fa fa-check"></i><b>16.8</b> Example 4 – Experiments with subsampling replication (exp1g)</a>
<ul>
<li class="chapter" data-level="16.8.1" data-path="lmm.html"><a href="lmm.html#understand-the-data-2"><i class="fa fa-check"></i><b>16.8.1</b> Understand the data</a></li>
<li class="chapter" data-level="16.8.2" data-path="lmm.html"><a href="lmm.html#examine-the-data-2"><i class="fa fa-check"></i><b>16.8.2</b> Examine the data</a></li>
<li class="chapter" data-level="16.8.3" data-path="lmm.html"><a href="lmm.html#fit-the-model-9"><i class="fa fa-check"></i><b>16.8.3</b> Fit the model</a></li>
<li class="chapter" data-level="16.8.4" data-path="lmm.html"><a href="lmm.html#inference-from-the-model-6"><i class="fa fa-check"></i><b>16.8.4</b> Inference from the model</a></li>
<li class="chapter" data-level="16.8.5" data-path="lmm.html"><a href="lmm.html#plot-the-model-6"><i class="fa fa-check"></i><b>16.8.5</b> Plot the model</a></li>
<li class="chapter" data-level="16.8.6" data-path="lmm.html"><a href="lmm.html#alternaplot-the-model-3"><i class="fa fa-check"></i><b>16.8.6</b> Alternaplot the model</a></li>
<li class="chapter" data-level="16.8.7" data-path="lmm.html"><a href="lmm.html#understanding-the-alternative-models"><i class="fa fa-check"></i><b>16.8.7</b> Understanding the alternative models</a></li>
<li class="chapter" data-level="16.8.8" data-path="lmm.html"><a href="lmm.html#the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b"><i class="fa fa-check"></i><b>16.8.8</b> The VarCorr matrix of models exp1g_m1a and exp1g_m1b</a></li>
<li class="chapter" data-level="16.8.9" data-path="lmm.html"><a href="lmm.html#the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means"><i class="fa fa-check"></i><b>16.8.9</b> The linear mixed model has more precision and power than the fixed effect model of batch means</a></li>
<li class="chapter" data-level="16.8.10" data-path="lmm.html"><a href="lmm.html#fixed-effect-models-and-pseudoreplication"><i class="fa fa-check"></i><b>16.8.10</b> Fixed effect models and pseudoreplication</a></li>
<li class="chapter" data-level="16.8.11" data-path="lmm.html"><a href="lmm.html#mixed-effect-anova"><i class="fa fa-check"></i><b>16.8.11</b> Mixed-effect ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="16.9" data-path="lmm.html"><a href="lmm.html#statistical-models-for-experimental-designs"><i class="fa fa-check"></i><b>16.9</b> Statistical models for experimental designs</a>
<ul>
<li class="chapter" data-level="16.9.1" data-path="lmm.html"><a href="lmm.html#models-for-completely-randomized-designs-crd"><i class="fa fa-check"></i><b>16.9.1</b> Models for Completely Randomized Designs (CRD)</a></li>
<li class="chapter" data-level="16.9.2" data-path="lmm.html"><a href="lmm.html#models-for-batched-data-crds-rcbd-rspd-grcbd-nrcbd"><i class="fa fa-check"></i><b>16.9.2</b> Models for batched data (CRDS, RCBD, RSPD, GRCBD, NRCBD)</a></li>
</ul></li>
<li class="chapter" data-level="16.10" data-path="lmm.html"><a href="lmm.html#which-model-and-why"><i class="fa fa-check"></i><b>16.10</b> Which model and why?</a>
<ul>
<li class="chapter" data-level="16.10.1" data-path="lmm.html"><a href="lmm.html#crd"><i class="fa fa-check"></i><b>16.10.1</b> CRD</a></li>
<li class="chapter" data-level="16.10.2" data-path="lmm.html"><a href="lmm.html#crds"><i class="fa fa-check"></i><b>16.10.2</b> CRDS</a></li>
<li class="chapter" data-level="16.10.3" data-path="lmm.html"><a href="lmm.html#rcbd"><i class="fa fa-check"></i><b>16.10.3</b> RCBD</a></li>
<li class="chapter" data-level="16.10.4" data-path="lmm.html"><a href="lmm.html#rspd"><i class="fa fa-check"></i><b>16.10.4</b> RSPD</a></li>
<li class="chapter" data-level="16.10.5" data-path="lmm.html"><a href="lmm.html#rcbds"><i class="fa fa-check"></i><b>16.10.5</b> RCBDS</a></li>
<li class="chapter" data-level="16.10.6" data-path="lmm.html"><a href="lmm.html#grcbd"><i class="fa fa-check"></i><b>16.10.6</b> GRCBD</a></li>
<li class="chapter" data-level="16.10.7" data-path="lmm.html"><a href="lmm.html#grcbds"><i class="fa fa-check"></i><b>16.10.7</b> GRCBDS</a></li>
</ul></li>
<li class="chapter" data-level="16.11" data-path="lmm.html"><a href="lmm.html#working-in-r-5"><i class="fa fa-check"></i><b>16.11</b> Working in R</a>
<ul>
<li class="chapter" data-level="16.11.1" data-path="lmm.html"><a href="lmm.html#plotting-models-fit-to-batched-data"><i class="fa fa-check"></i><b>16.11.1</b> Plotting models fit to batched data</a></li>
<li class="chapter" data-level="16.11.2" data-path="lmm.html"><a href="lmm.html#repeated-measures-anova-randomized-complete-block-with-no-subsampling"><i class="fa fa-check"></i><b>16.11.2</b> Repeated measures ANOVA (randomized complete block with no subsampling)</a></li>
</ul></li>
<li class="chapter" data-level="16.12" data-path="lmm.html"><a href="lmm.html#hidden-code-7"><i class="fa fa-check"></i><b>16.12</b> Hidden code</a>
<ul>
<li class="chapter" data-level="16.12.1" data-path="lmm.html"><a href="lmm.html#import-exp5c"><i class="fa fa-check"></i><b>16.12.1</b> Import exp5c</a></li>
<li class="chapter" data-level="16.12.2" data-path="lmm.html"><a href="lmm.html#import-exp1g"><i class="fa fa-check"></i><b>16.12.2</b> Import exp1g</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="pre-post.html"><a href="pre-post.html"><i class="fa fa-check"></i><b>17</b> Linear models for longitudinal experiments – I. pre-post designs</a>
<ul>
<li class="chapter" data-level="17.1" data-path="pre-post.html"><a href="pre-post.html#best-practice-models"><i class="fa fa-check"></i><b>17.1</b> Best practice models</a></li>
<li class="chapter" data-level="17.2" data-path="pre-post.html"><a href="pre-post.html#common-alternatives-that-are-not-recommended"><i class="fa fa-check"></i><b>17.2</b> Common alternatives that are not recommended</a></li>
<li class="chapter" data-level="17.3" data-path="pre-post.html"><a href="pre-post.html#advanced-models"><i class="fa fa-check"></i><b>17.3</b> Advanced models</a></li>
<li class="chapter" data-level="17.4" data-path="pre-post.html"><a href="pre-post.html#understanding-the-alternative-models-1"><i class="fa fa-check"></i><b>17.4</b> Understanding the alternative models</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="pre-post.html"><a href="pre-post.html#m1-linear-model-with-the-baseline-measure-as-the-covariate-ancova-model"><i class="fa fa-check"></i><b>17.4.1</b> (M1) Linear model with the baseline measure as the covariate (ANCOVA model)</a></li>
<li class="chapter" data-level="17.4.2" data-path="pre-post.html"><a href="pre-post.html#m2-linear-model-of-the-change-score-change-score-model"><i class="fa fa-check"></i><b>17.4.2</b> (M2) Linear model of the change score (change-score model)</a></li>
<li class="chapter" data-level="17.4.3" data-path="pre-post.html"><a href="pre-post.html#m3-linear-model-of-post-baseline-values-without-the-baseline-as-a-covariate-post-model"><i class="fa fa-check"></i><b>17.4.3</b> (M3) Linear model of post-baseline values without the baseline as a covariate (post model)</a></li>
<li class="chapter" data-level="17.4.4" data-path="pre-post.html"><a href="pre-post.html#m4-linear-model-with-factorial-fixed-effects-fixed-effects-model"><i class="fa fa-check"></i><b>17.4.4</b> (M4) Linear model with factorial fixed effects (fixed-effects model)</a></li>
<li class="chapter" data-level="17.4.5" data-path="pre-post.html"><a href="pre-post.html#m5-repeated-measures-anova"><i class="fa fa-check"></i><b>17.4.5</b> (M5) Repeated measures ANOVA</a></li>
<li class="chapter" data-level="17.4.6" data-path="pre-post.html"><a href="pre-post.html#m6-linear-mixed-model"><i class="fa fa-check"></i><b>17.4.6</b> (M6) Linear mixed model</a></li>
<li class="chapter" data-level="17.4.7" data-path="pre-post.html"><a href="pre-post.html#m7-linear-model-with-correlated-error"><i class="fa fa-check"></i><b>17.4.7</b> (M7) Linear model with correlated error</a></li>
<li class="chapter" data-level="17.4.8" data-path="pre-post.html"><a href="pre-post.html#m8-constrained-fixed-effects-model-with-correlated-error-clda-model"><i class="fa fa-check"></i><b>17.4.8</b> (M8) Constrained fixed effects model with correlated error (cLDA model)</a></li>
<li class="chapter" data-level="17.4.9" data-path="pre-post.html"><a href="pre-post.html#comparison-table"><i class="fa fa-check"></i><b>17.4.9</b> Comparison table</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="pre-post.html"><a href="pre-post.html#example-1-a-single-post-baseline-measure-pre-post-design"><i class="fa fa-check"></i><b>17.5</b> Example 1 – a single post-baseline measure (pre-post design)</a></li>
<li class="chapter" data-level="17.6" data-path="pre-post.html"><a href="pre-post.html#working-in-r-6"><i class="fa fa-check"></i><b>17.6</b> Working in R</a></li>
<li class="chapter" data-level="17.7" data-path="pre-post.html"><a href="pre-post.html#hidden-code-8"><i class="fa fa-check"></i><b>17.7</b> Hidden code</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="pre-post.html"><a href="pre-post.html#import-and-wrangle-mouse-sociability-data"><i class="fa fa-check"></i><b>17.7.1</b> Import and wrangle mouse sociability data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="counts.html"><a href="counts.html"><i class="fa fa-check"></i><b>18</b> Linear models for counts, binary responses, skewed responses, and ratios – Generalized Linear Models</a>
<ul>
<li class="chapter" data-level="18.1" data-path="counts.html"><a href="counts.html#glm-counts"><i class="fa fa-check"></i><b>18.1</b> Introducing Generalized Linear Models using count data examples</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="counts.html"><a href="counts.html#the-generalized-linear-model-glm"><i class="fa fa-check"></i><b>18.1.1</b> The Generalized Linear Model (GLM)</a></li>
<li class="chapter" data-level="18.1.2" data-path="counts.html"><a href="counts.html#kinds-of-data-that-are-modeled-by-a-glm"><i class="fa fa-check"></i><b>18.1.2</b> Kinds of data that are modeled by a GLM</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="counts.html"><a href="counts.html#glm-1"><i class="fa fa-check"></i><b>18.2</b> Example 1 – GLM models for count responses (“angiogenic sprouts” fig3a)</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="counts.html"><a href="counts.html#understand-the-data-3"><i class="fa fa-check"></i><b>18.2.1</b> Understand the data</a></li>
<li class="chapter" data-level="18.2.2" data-path="counts.html"><a href="counts.html#model-fit-and-inference-1"><i class="fa fa-check"></i><b>18.2.2</b> Model fit and inference</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="counts.html"><a href="counts.html#understanding-example-1"><i class="fa fa-check"></i><b>18.3</b> Understanding Example 1</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="counts.html"><a href="counts.html#modeling-strategy-1"><i class="fa fa-check"></i><b>18.3.1</b> Modeling strategy</a></li>
<li class="chapter" data-level="18.3.2" data-path="counts.html"><a href="counts.html#model-checking-fits-to-count-data"><i class="fa fa-check"></i><b>18.3.2</b> Model checking fits to count data</a></li>
<li class="chapter" data-level="18.3.3" data-path="counts.html"><a href="counts.html#biological-count-data-are-rarely-fit-well-by-a-poisson-glm.-instead-fit-a-quasi-poisson-or-negative-binomial-glm-model."><i class="fa fa-check"></i><b>18.3.3</b> Biological count data are rarely fit well by a Poisson GLM. Instead, fit a quasi-poisson or negative binomial GLM model.</a></li>
<li class="chapter" data-level="18.3.4" data-path="counts.html"><a href="counts.html#a-glm-is-a-linear-model-on-the-link-scale"><i class="fa fa-check"></i><b>18.3.4</b> A GLM is a linear model on the link scale</a></li>
<li class="chapter" data-level="18.3.5" data-path="counts.html"><a href="counts.html#coeffecients-of-a-generalized-linear-model-with-a-log-link-function-are-on-the-link-scale."><i class="fa fa-check"></i><b>18.3.5</b> Coeffecients of a Generalized Linear Model with a log-link function are on the link scale.</a></li>
<li class="chapter" data-level="18.3.6" data-path="counts.html"><a href="counts.html#modeled-means-in-the-emmeans-table-of-a-generalized-linear-model-can-be-on-the-link-scale-or-response-scale-report-the-response-scale"><i class="fa fa-check"></i><b>18.3.6</b> Modeled means in the emmeans table of a Generalized Linear Model can be on the link scale or response scale – Report the response scale</a></li>
<li class="chapter" data-level="18.3.7" data-path="counts.html"><a href="counts.html#some-consequences-of-fitting-a-linear-model-to-count-data"><i class="fa fa-check"></i><b>18.3.7</b> Some consequences of fitting a linear model to count data</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="counts.html"><a href="counts.html#example-2-use-a-glm-with-an-offset-instead-of-a-ratio-of-some-measurement-per-total-dna-damage-data-fig3b-glm_offset"><i class="fa fa-check"></i><b>18.4</b> Example 2 – Use a GLM with an offset instead of a ratio of some measurement per total (“dna damage” data fig3b) #glm_offset</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="counts.html"><a href="counts.html#fig3b-dna-damage-data"><i class="fa fa-check"></i><b>18.4.1</b> fig3b (“dna damage”) data</a></li>
<li class="chapter" data-level="18.4.2" data-path="counts.html"><a href="counts.html#understand-the-data-4"><i class="fa fa-check"></i><b>18.4.2</b> Understand the data</a></li>
<li class="chapter" data-level="18.4.3" data-path="counts.html"><a href="counts.html#model-fit-and-inference-2"><i class="fa fa-check"></i><b>18.4.3</b> Model fit and inference</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="counts.html"><a href="counts.html#understanding-example-2"><i class="fa fa-check"></i><b>18.5</b> Understanding Example 2</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="counts.html"><a href="counts.html#an-offset-is-an-added-covariate-with-a-coefficient-fixed-at-1"><i class="fa fa-check"></i><b>18.5.1</b> An offset is an added covariate with a coefficient fixed at 1</a></li>
<li class="chapter" data-level="18.5.2" data-path="counts.html"><a href="counts.html#glm-rate"><i class="fa fa-check"></i><b>18.5.2</b> A count GLM with an offset models the area-normalized means</a></li>
<li class="chapter" data-level="18.5.3" data-path="counts.html"><a href="counts.html#compare-an-offset-to-an-added-covariate-with-an-estimated-coefficient"><i class="fa fa-check"></i><b>18.5.3</b> Compare an offset to an added covariate with an estimated coefficient</a></li>
<li class="chapter" data-level="18.5.4" data-path="counts.html"><a href="counts.html#issues-with-plotting"><i class="fa fa-check"></i><b>18.5.4</b> Issues with plotting</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="counts.html"><a href="counts.html#example-3-glm-models-for-binary-responses"><i class="fa fa-check"></i><b>18.6</b> Example 3 – GLM models for binary responses</a></li>
<li class="chapter" data-level="18.7" data-path="counts.html"><a href="counts.html#working-in-r-7"><i class="fa fa-check"></i><b>18.7</b> Working in R</a>
<ul>
<li class="chapter" data-level="18.7.1" data-path="counts.html"><a href="counts.html#fitting-glms-to-count-data"><i class="fa fa-check"></i><b>18.7.1</b> Fitting GLMs to count data</a></li>
<li class="chapter" data-level="18.7.2" data-path="counts.html"><a href="counts.html#fitting-a-glm-to-a-continuous-conditional-response-with-right-skew."><i class="fa fa-check"></i><b>18.7.2</b> Fitting a GLM to a continuous conditional response with right skew.</a></li>
<li class="chapter" data-level="18.7.3" data-path="counts.html"><a href="counts.html#fitting-a-glm-to-a-binary-success-or-failure-presence-or-absence-survived-or-died-response"><i class="fa fa-check"></i><b>18.7.3</b> Fitting a GLM to a binary (success or failure, presence or absence, survived or died) response</a></li>
<li class="chapter" data-level="18.7.4" data-path="counts.html"><a href="counts.html#fitting-generalized-linear-mixed-models"><i class="fa fa-check"></i><b>18.7.4</b> Fitting Generalized Linear Mixed Models</a></li>
</ul></li>
<li class="chapter" data-level="18.8" data-path="counts.html"><a href="counts.html#model-checking-glms"><i class="fa fa-check"></i><b>18.8</b> Model checking GLMs</a></li>
<li class="chapter" data-level="18.9" data-path="counts.html"><a href="counts.html#hidden-code-9"><i class="fa fa-check"></i><b>18.9</b> Hidden code</a>
<ul>
<li class="chapter" data-level="18.9.1" data-path="counts.html"><a href="counts.html#import-example-1-data-fig3a-angiogenic-sprouts"><i class="fa fa-check"></i><b>18.9.1</b> Import Example 1 data (fig3a – “angiogenic sprouts”)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="gls.html"><a href="gls.html"><i class="fa fa-check"></i><b>19</b> Linear models with heterogenous variance</a></li>
<li class="chapter" data-level="" data-path="appendix-an-example-set-of-analyses-of-experimental-data-with-linear-models.html"><a href="appendix-an-example-set-of-analyses-of-experimental-data-with-linear-models.html"><i class="fa fa-check"></i>Appendix: An example set of analyses of experimental data with linear models</a></li>
<li class="chapter" data-level="20" data-path="ask1-bio.html"><a href="ask1-bio.html"><i class="fa fa-check"></i><b>20</b> Background physiology to the experiments in Figure 2 of “ASK1 inhibits browning of white adipose tissue in obesity”</a></li>
<li class="chapter" data-level="21" data-path="ask1-report.html"><a href="ask1-report.html"><i class="fa fa-check"></i><b>21</b> Analyses for Figure 2 of “ASK1 inhibits browning of white adipose tissue in obesity”</a>
<ul>
<li class="chapter" data-level="21.1" data-path="ask1-report.html"><a href="ask1-report.html#setup"><i class="fa fa-check"></i><b>21.1</b> Setup</a></li>
<li class="chapter" data-level="21.2" data-path="ask1-report.html"><a href="ask1-report.html#data-source"><i class="fa fa-check"></i><b>21.2</b> Data source</a></li>
<li class="chapter" data-level="21.3" data-path="ask1-report.html"><a href="ask1-report.html#control-the-color-palette"><i class="fa fa-check"></i><b>21.3</b> control the color palette</a></li>
<li class="chapter" data-level="21.4" data-path="ask1-report.html"><a href="ask1-report.html#useful-functions"><i class="fa fa-check"></i><b>21.4</b> useful functions</a></li>
<li class="chapter" data-level="21.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2b-effect-of-ask1-deletion-on-growth-body-weight"><i class="fa fa-check"></i><b>21.5</b> figure 2b – effect of ASK1 deletion on growth (body weight)</a>
<ul>
<li class="chapter" data-level="21.5.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2b-import"><i class="fa fa-check"></i><b>21.5.1</b> figure 2b – import</a></li>
<li class="chapter" data-level="21.5.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2b-exploratory-plots"><i class="fa fa-check"></i><b>21.5.2</b> figure 2b – exploratory plots</a></li>
</ul></li>
<li class="chapter" data-level="21.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-effect-of-ask1-deletion-on-final-body-weight"><i class="fa fa-check"></i><b>21.6</b> Figure 2c – Effect of ASK1 deletion on final body weight</a>
<ul>
<li class="chapter" data-level="21.6.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-import"><i class="fa fa-check"></i><b>21.6.1</b> Figure 2c – import</a></li>
<li class="chapter" data-level="21.6.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-check-own-computation-of-weight-change-v-imported-value"><i class="fa fa-check"></i><b>21.6.2</b> Figure 2c – check own computation of weight change v imported value</a></li>
<li class="chapter" data-level="21.6.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-exploratory-plots"><i class="fa fa-check"></i><b>21.6.3</b> Figure 2c – exploratory plots</a></li>
<li class="chapter" data-level="21.6.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-fit-the-model-m1-lm"><i class="fa fa-check"></i><b>21.6.4</b> Figure 2c – fit the model: m1 (lm)</a></li>
<li class="chapter" data-level="21.6.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-check-the-model-m1"><i class="fa fa-check"></i><b>21.6.5</b> Figure 2c – check the model: m1</a></li>
<li class="chapter" data-level="21.6.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-fit-the-model-m2-gamma-glm"><i class="fa fa-check"></i><b>21.6.6</b> Figure 2c – fit the model: m2 (gamma glm)</a></li>
<li class="chapter" data-level="21.6.7" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-check-the-model-m2"><i class="fa fa-check"></i><b>21.6.7</b> Figure 2c – check the model, m2</a></li>
<li class="chapter" data-level="21.6.8" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-inference-from-the-model"><i class="fa fa-check"></i><b>21.6.8</b> Figure 2c – inference from the model</a></li>
<li class="chapter" data-level="21.6.9" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-plot-the-model"><i class="fa fa-check"></i><b>21.6.9</b> Figure 2c – plot the model</a></li>
<li class="chapter" data-level="21.6.10" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-report"><i class="fa fa-check"></i><b>21.6.10</b> Figure 2c – report</a></li>
</ul></li>
<li class="chapter" data-level="21.7" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve"><i class="fa fa-check"></i><b>21.7</b> Figure 2d – Effect of ASK1 KO on glucose tolerance (whole curve)</a>
<ul>
<li class="chapter" data-level="21.7.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-import"><i class="fa fa-check"></i><b>21.7.1</b> Figure 2d – Import</a></li>
<li class="chapter" data-level="21.7.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-exploratory-plots"><i class="fa fa-check"></i><b>21.7.2</b> Figure 2d – exploratory plots</a></li>
<li class="chapter" data-level="21.7.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-fit-the-model"><i class="fa fa-check"></i><b>21.7.3</b> Figure 2d – fit the model</a></li>
<li class="chapter" data-level="21.7.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-check-the-model"><i class="fa fa-check"></i><b>21.7.4</b> Figure 2d – check the model</a></li>
<li class="chapter" data-level="21.7.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-inference"><i class="fa fa-check"></i><b>21.7.5</b> Figure 2d – inference</a></li>
<li class="chapter" data-level="21.7.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-plot-the-model"><i class="fa fa-check"></i><b>21.7.6</b> Figure 2d – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="21.8" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure"><i class="fa fa-check"></i><b>21.8</b> Figure 2e – Effect of ASK1 deletion on glucose tolerance (summary measure)</a>
<ul>
<li class="chapter" data-level="21.8.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-message-the-data"><i class="fa fa-check"></i><b>21.8.1</b> Figure 2e – message the data</a></li>
<li class="chapter" data-level="21.8.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-exploratory-plots"><i class="fa fa-check"></i><b>21.8.2</b> Figure 2e – exploratory plots</a></li>
<li class="chapter" data-level="21.8.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-fit-the-model"><i class="fa fa-check"></i><b>21.8.3</b> Figure 2e – fit the model</a></li>
<li class="chapter" data-level="21.8.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-check-the-model"><i class="fa fa-check"></i><b>21.8.4</b> Figure 2e – check the model</a></li>
<li class="chapter" data-level="21.8.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-inference-from-the-model"><i class="fa fa-check"></i><b>21.8.5</b> Figure 2e – inference from the model</a></li>
<li class="chapter" data-level="21.8.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-plot-the-model"><i class="fa fa-check"></i><b>21.8.6</b> Figure 2e – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="21.9" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate"><i class="fa fa-check"></i><b>21.9</b> Figure 2f – Effect of ASK1 deletion on glucose infusion rate</a>
<ul>
<li class="chapter" data-level="21.9.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-import"><i class="fa fa-check"></i><b>21.9.1</b> Figure 2f – import</a></li>
<li class="chapter" data-level="21.9.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-exploratory-plots"><i class="fa fa-check"></i><b>21.9.2</b> Figure 2f – exploratory plots</a></li>
<li class="chapter" data-level="21.9.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-fit-the-model"><i class="fa fa-check"></i><b>21.9.3</b> Figure 2f – fit the model</a></li>
<li class="chapter" data-level="21.9.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-check-the-model"><i class="fa fa-check"></i><b>21.9.4</b> Figure 2f – check the model</a></li>
<li class="chapter" data-level="21.9.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-inference"><i class="fa fa-check"></i><b>21.9.5</b> Figure 2f – inference</a></li>
<li class="chapter" data-level="21.9.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-plot-the-model"><i class="fa fa-check"></i><b>21.9.6</b> Figure 2f – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="21.10" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake"><i class="fa fa-check"></i><b>21.10</b> Figure 2g – Effect of ASK1 deletion on tissue-specific glucose uptake</a>
<ul>
<li class="chapter" data-level="21.10.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-import"><i class="fa fa-check"></i><b>21.10.1</b> Figure 2g – import</a></li>
<li class="chapter" data-level="21.10.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-exploratory-plots"><i class="fa fa-check"></i><b>21.10.2</b> Figure 2g – exploratory plots</a></li>
<li class="chapter" data-level="21.10.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-fit-the-model"><i class="fa fa-check"></i><b>21.10.3</b> Figure 2g – fit the model</a></li>
<li class="chapter" data-level="21.10.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-check-the-model"><i class="fa fa-check"></i><b>21.10.4</b> Figure 2g – check the model</a></li>
<li class="chapter" data-level="21.10.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-inference"><i class="fa fa-check"></i><b>21.10.5</b> Figure 2g – inference</a></li>
<li class="chapter" data-level="21.10.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-plot-the-model"><i class="fa fa-check"></i><b>21.10.6</b> Figure 2g – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="21.11" data-path="ask1-report.html"><a href="ask1-report.html#figure-2h"><i class="fa fa-check"></i><b>21.11</b> Figure 2h</a></li>
<li class="chapter" data-level="21.12" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-effect-of-ask1-deletion-on-liver-tg"><i class="fa fa-check"></i><b>21.12</b> Figure 2i – Effect of ASK1 deletion on liver TG</a>
<ul>
<li class="chapter" data-level="21.12.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-fit-the-model"><i class="fa fa-check"></i><b>21.12.1</b> Figure 2i – fit the model</a></li>
<li class="chapter" data-level="21.12.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-check-the-model"><i class="fa fa-check"></i><b>21.12.2</b> Figure 2i – check the model</a></li>
<li class="chapter" data-level="21.12.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-inference"><i class="fa fa-check"></i><b>21.12.3</b> Figure 2i – inference</a></li>
<li class="chapter" data-level="21.12.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-plot-the-model"><i class="fa fa-check"></i><b>21.12.4</b> Figure 2i – plot the model</a></li>
<li class="chapter" data-level="21.12.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-report-the-model"><i class="fa fa-check"></i><b>21.12.5</b> Figure 2i – report the model</a></li>
</ul></li>
<li class="chapter" data-level="21.13" data-path="ask1-report.html"><a href="ask1-report.html#figure-2j"><i class="fa fa-check"></i><b>21.13</b> Figure 2j</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="count-sim.html"><a href="count-sim.html"><i class="fa fa-check"></i><b>22</b> Simulations – Count data (alternatives to a t-test)</a>
<ul>
<li class="chapter" data-level="22.1" data-path="count-sim.html"><a href="count-sim.html#use-data-similar-to-figure-6f-from-example-1"><i class="fa fa-check"></i><b>22.1</b> Use data similar to Figure 6f from Example 1</a></li>
<li class="chapter" data-level="22.2" data-path="count-sim.html"><a href="count-sim.html#functions"><i class="fa fa-check"></i><b>22.2</b> Functions</a></li>
<li class="chapter" data-level="22.3" data-path="count-sim.html"><a href="count-sim.html#simulations"><i class="fa fa-check"></i><b>22.3</b> Simulations</a>
<ul>
<li class="chapter" data-level="22.3.1" data-path="count-sim.html"><a href="count-sim.html#type-i-pseudo-normal-distribution"><i class="fa fa-check"></i><b>22.3.1</b> Type I, Pseudo-Normal distribution</a></li>
<li class="chapter" data-level="22.3.2" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-equal-n"><i class="fa fa-check"></i><b>22.3.2</b> Type I, neg binom, equal n</a></li>
<li class="chapter" data-level="22.3.3" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-equal-n-small-theta"><i class="fa fa-check"></i><b>22.3.3</b> Type I, neg binom, equal n, small theta</a></li>
<li class="chapter" data-level="22.3.4" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-unequal-n"><i class="fa fa-check"></i><b>22.3.4</b> Type I, neg binom, unequal n</a></li>
<li class="chapter" data-level="22.3.5" data-path="count-sim.html"><a href="count-sim.html#power-pseudo-normal-distribution-equal-n"><i class="fa fa-check"></i><b>22.3.5</b> Power, Pseudo-Normal distribution, equal n</a></li>
<li class="chapter" data-level="22.3.6" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-equal-n"><i class="fa fa-check"></i><b>22.3.6</b> Power, neg binom, equal n</a></li>
<li class="chapter" data-level="22.3.7" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-small-theta"><i class="fa fa-check"></i><b>22.3.7</b> Power, neg binom, small theta</a></li>
<li class="chapter" data-level="22.3.8" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-unequal-n"><i class="fa fa-check"></i><b>22.3.8</b> Power, neg binom, unequal n</a></li>
<li class="chapter" data-level="22.3.9" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-unequal-n-unequal-theta"><i class="fa fa-check"></i><b>22.3.9</b> Power, neg binom, unequal n, unequal theta</a></li>
<li class="chapter" data-level="22.3.10" data-path="count-sim.html"><a href="count-sim.html#type-1-neg-binom-equal-n-unequal-theta"><i class="fa fa-check"></i><b>22.3.10</b> Type 1, neg binom, equal n, unequal theta</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="count-sim.html"><a href="count-sim.html#save-it-read-it"><i class="fa fa-check"></i><b>22.4</b> Save it, Read it</a></li>
<li class="chapter" data-level="22.5" data-path="count-sim.html"><a href="count-sim.html#analysis"><i class="fa fa-check"></i><b>22.5</b> Analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html"><i class="fa fa-check"></i>Appendix 1: Getting Started with R</a>
<ul>
<li class="chapter" data-level="22.6" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#get-your-computer-ready"><i class="fa fa-check"></i><b>22.6</b> Get your computer ready</a>
<ul>
<li class="chapter" data-level="22.6.1" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#start-here"><i class="fa fa-check"></i><b>22.6.1</b> Start here</a></li>
<li class="chapter" data-level="22.6.2" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r"><i class="fa fa-check"></i><b>22.6.2</b> Install R</a></li>
<li class="chapter" data-level="22.6.3" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r-studio"><i class="fa fa-check"></i><b>22.6.3</b> Install R Studio</a></li>
<li class="chapter" data-level="22.6.4" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r-markdown-1"><i class="fa fa-check"></i><b>22.6.4</b> Install R Markdown</a></li>
<li class="chapter" data-level="22.6.5" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#optional-alternative-latex-installations"><i class="fa fa-check"></i><b>22.6.5</b> (optional) Alternative LaTeX installations</a></li>
</ul></li>
<li class="chapter" data-level="22.7" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#start-learning-r-studio"><i class="fa fa-check"></i><b>22.7</b> Start learning R Studio</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix-2-online-resources-for-getting-started-with-statistical-modeling-in-r.html"><a href="appendix-2-online-resources-for-getting-started-with-statistical-modeling-in-r.html"><i class="fa fa-check"></i>Appendix 2: Online Resources for Getting Started with Statistical Modeling in R</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistics for the Experimental Biologist</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="lmm" class="section level1 hasAnchor" number="16">
<h1><span class="header-section-number">Chapter 16</span> Models with random factors – linear mixed models<a href="lmm.html#lmm" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="chapter-summary" class="section level2 hasAnchor" number="16.1">
<h2><span class="header-section-number">16.1</span> Chapter summary<a href="lmm.html#chapter-summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li>Technical replicates do not increase sample size. If the diameter of a single muscle fiber is measured in each of five WT and five KO mice, the sample size is five. If the diameters of ten muscle fibers are measured in each of the mice, the sample size is still five. If the researcher analyzes the data as if the ten measures per mouse are independent experimental units, the standard errors and <em>p</em>-value will be falsely small. This is <strong>pseudoreplication</strong> and leads to increased rate of false discovery. Pseudoreplication is very, very common in the experimental biology literature.</li>
<li>Paired <em>t</em>-test are commonly used for experiments in which both treatment levels of an experimental factor are
Blocked designs are extremely common in experimental biology, but largely unrecognized.</li>
</ol>
</div>
<div id="introduction" class="section level2 hasAnchor" number="16.2">
<h2><span class="header-section-number">16.2</span> Introduction<a href="lmm.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This chapter is about models for <strong>correlated error</strong>, including linear models with added <strong>random factors</strong>, which are known as <strong>linear mixed models</strong>. In classical hypothesis testing, a <strong>paired t-test</strong>, <strong>repeated measures ANOVA</strong>, and <strong>mixed-effect ANOVA</strong> are equivalent to specific cases of linear mixed models. Linear mixed models are used for analyzing data composed of subsets – or <strong>batches</strong> – of data that were measured from the “same thing”, such as multiple measures within a mouse, or multiple mice within a litter. Batched data results in <strong>correlated error</strong>, which violates a key assumption of linear models (and their “which test” equivalents) and muddles statistical inference unless the correlated error is modeled, explicitly or implicitly. In some experimental designs (<strong>blocked designs</strong>), failure to model the correlated error reduces precision and power, contributing to reduced rates of discovery or confirmation. In other designs (<strong>nested designs</strong>), failure to model the correlated error results in falsely high precision and low <em>p</em>-values, leading to increased rates of false discovery. The falsely high precision is due to <strong>pseudoreplication</strong>. I think it’s fair to infer from the experimental biology literature, that experimental biologists don’t recognize the ubiquitousness of batched data and correlated error. This is probably the biggest issue in inference in the field (far more of an issue than say, a <em>t</em>-test on non-normal data).</p>
<p>What do I mean by “batch” and how can correlated error both increase and decrease false discovery? Consider an experiment to measure pancreatic islet area in response to two experimental factors: <span class="math inline">\(\texttt{genotype}\)</span> (WT, KO) and <span class="math inline">\(\texttt{treatment}\)</span> (presence/absence of some drug believed to be an agonist of the knocked out protein). While it may seem like the data from this experiment should be analyzed using the ANOVA option in GraphPad Prism (or, as advocated in this book, a general linear model that is equivalent to the ANOVA), the best practice statistical model actually depends on the <strong>experimental design</strong>. Experimental design matters because different designs introduce different patterns of <strong>correlated error</strong> due to shared genetics and environment. Recall that inference from a linear model (including <em>t</em>-tests and ANOVA) assumes <em>independence</em> (Chapter xxx) – that is, each response value has no relationship to any other value, other than that due to treatment. Lack of independence results in patterns of correlation among the residuals, or correlated error.</p>
<p>Something like the first experiment below (Design 1) is the necessary design to use the statistics that have been covered in this book to this point, without extreme violation of the independence assumption. But many (most?) experiments in experimental bench biology do not look like the design in Design 1 below. Instead, many (most?) experiments are variants of Designs 2-4, all of which have extreme violations of the independence assumption. Interestingly, some of these violations result in conservative statistics and reduced, true discovery rate while others result in liberal statistics and increased, false discovery rate.</p>
<p>Design 1. The design in Figure @ref(fig:lmm-biological-replicates_1) is a factorial design with two factors, <span class="math inline">\(\texttt{genotype}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>, each with two levels. Twenty mice of the same sex, each from a unique litter with a different dam and sire, are randomly sampled and assigned to one <span class="math inline">\(genotype \times treatment\)</span> combination (five mice per combination). All mice are housed individually (20 cages). The pancreatic tissue from all mice is prepared in a single batch and a single islet is measured from each mouse, all using the same microscope, which is recalibrated before each measure. The entire experiment is carried out at the same time and each component (tissue preparation, measuring) is carried out by the same person (these could be different people for each component). This is a <strong>Completely Randomized Design</strong> (CRD). The five replicate mice per treatment are <strong>treatment replicates</strong> (often called <strong>biological replicates</strong> in experimental biology. A CRD does not have batched data.</p>
<p><img src="images/experimental_designs/lmm-biological-replicates%20500%20W.png" width="63%" /></p>
<p>Design 2. In the design in Figure <a href="#fig:lmm-biological-replicates"><strong>??</strong></a>, four littermates are randomly sampled from five litters, each with a different dam and sire. Within each litter, mice are randomly assigned to each of the four treatment combinations (one per combination). Each litter is randomly assigned to cage with only a single litter per cage. All other aspects of this design are as in Design 1. This is a <strong>Randomized Complete Block Design</strong>. The five replicate mice per treatment are the treatment replicates. Each litter/cage combination is a type of batch called a <strong>block</strong>. A blocked design typically functions to reduce noise in the model fit (this increases power) and to reduce the number of litters and cages needed for an experiment. The four measures of Islet Area within a litter/cage (one per mouse) are not independent of each other. Each cage has four mice from the same litter and these mice share genetic and maternal factors that contribute to mouse anatomy and physiology that are not shared by mice in other litters. Additionally, each cage has a unique set of environmental factors that contribute to the error variance of the measure of the response. Each cage shares a cage-specific history of temperature, humidity, food, light, interactions with animal facilities staff, and behavioral interactions among the mice. All response measures within a litter/cage share the component of the error variance unique to that litter/cage and, as a consequence, the error (residuals) within a litter/cage are more similar to each other than they are to the residuals among litters/cages.</p>
<p><img src="images/experimental_designs/lmm-technical-replicates%20500%20W.png" width="63%" /></p>
<p>Design 3. The design in Figure <a href="#fig:lmm-technical-replicates"><strong>??</strong></a> is exactly like that in Design 2, except that the researchers take three measures of iselet area per mouse. The three measures are <strong>subsampled replicates</strong>. Experimental biologists often call these <strong>technical replicates</strong>, especially when the multiple measures are taken from the same preparation. Subsampling is a kind of <strong>nested design</strong> in which one variable is nested within (as opposed to crossed with) another variable. Here, the subsampled variable (subsample_id) is nested within the mouse_id variable. In addition to each litter/cage being a batch, each mouse is a batch. Each mouse has a unique set of factors that contribute to the error variance of the measures of the response in that mouse. All response measures within a mouse share the component of the error variance unique to that mouse and, as a consequence, the error (residuals) within a mouse are more similar to each other than they are to the residuals between mice</p>
<p><img src="images/lmm-pseudoreplication.png" width="63%" /></p>
<p>Design 4. The design in Figure <a href="#fig:lmm-segregated"><strong>??</strong></a> is a variation of Design 2, but the five treatment replicates of each combination are housed together in the same cage. In this design, each litter is a batch and each cage is a batch but these are different batches, unlike Design 2.</p>
<p>In each of these experiments, there is systematic variation at multiple levels: among treatments due to treatment effects and among batches due to <strong>batch effects</strong>. Batches come in lots of flavors, including experiment, litter, cage, flask, plate, slide, donor, and individual. The among-treatment differences in means are the <strong>fixed effects</strong>. The among-batch differences are the <strong>random effect</strong>. An assumption of modeling random effects is that the batches are a random sample of the batches that could have been sampled. This is often not strictly true as batches are often <strong>convenience samples</strong> (example: the human donors of the Type 2 diabetes beta cells are those that were in the hospital).</p>
<p>The variation among batches/lack of independence within batches has different consequences on the uncertainty of the estimate of a treatment effect. The batches in Experiment 1 contain all treatment combinations. The researcher is interested in the treatment effect but not the variation due to differences among the batches. The batches are nuissance factors that add additional variance to the response, with the consequence that estimates of treatment effects are less precise, unless the variance due to the batches is explicitly modeled. <strong>Modeling a batch that contains some or all treatment combinations will increase precision and power</strong>.</p>
<p>Batches that contain at least two treatment combinations are known as <strong>blocks</strong>. A block that contains all treatment combinations is a <strong>complete block</strong>. A block that contains fewer than all combinations is an <strong>incomplete block</strong>. Including block structure in the design is known as <strong>blocking</strong>. Blocks are non-experimental factors. Adding a blocking factor to a statistical model is used to increase the precision of an estimated treatment effect. Design 2 is an example of a <strong>randomized complete block design</strong>.</p>
<p>In Design 3, there are multiple measures per mouse and the design is a <strong>randomized complete block design with subsampling</strong>. The subsampling is not the kind of replication that can be used to infer the among treatment effect because the treatment assignment was not at the level of the subsamples. The <strong>treatment replicates</strong> are the litters/cages, because <em>it was at this level that treatment assignment was randomized</em>. A statistical analysis of all measures from a subsampled design without modeling the correlated error due to the subsampling is a kind of <a href="https://en.wikipedia.org/wiki/Pseudoreplication" target="_blank">pseudoreplication</a>. Pseudoreplication results in falsely precise standard errors and false small <em>p</em>-values and, consequently, increased rates of false discovery.</p>
<p>In Design 4, the treatment is randomized <em>to</em> batch, so each batch contains only a single treatment level. In these <strong>segregated</strong> experimental designs, the variation among batches that arises from non-treatment related differences among batches <strong>confounds</strong> the variation among batches due to a true treatment effect. Design 4 is an extreme example of this – there is only a single cage with a specific treatment combination. Imagine 1) the true effect of a treatment combination is zero and 2) an aggressive mouse in the control cage stimulates the stress response in the other mice and this stress response has a large effect on the value of the response variable measured by the researchers. The researcher is fooled into thinking that the treatment caused the difference in the response.</p>
<p>In all of these designs, it is important for the researcher to identify the <strong>experimental unit</strong> and the <strong>measurement</strong> unit. The experimental unit is the entity that was randomly assigned the treatment. In designs 1 – 3, the experimental unit is the mouse. In experiment 4, the experimental unit is the cage. The measurement unit is the entity that was measured. In designs 1, 2, and 4, the measurement unit is the mouse. In design 3, the measurement unit is specific islet that was measured.</p>
<div style="background-color:#cccccc; text-align:left; vertical-align: middle; padding:20px 47px;">
<strong>Pseudoreplication</strong>. In pseudoreplication, the degrees of freedom used to compute the test statistic and the <em>p</em>-value are inflated given the experimental design and research question. An example: A researcher wants to investigate the effect of some protein on mitochondrial biogenesis and designs an experiment with a wildtype (WT) and a conditional knockout (KO) mouse. Mitochondrial counts from twenty cells in one WT mouse and one KO mouse are measured and the researcher uses a <em>t</em>-test to compare counts. The sample size used to compute the standard error in the denominator of the <em>t</em>-value is 20. The <em>t</em>-distribution used to compute the <em>p</em>-value uses 38 df (20 measures times two groups minus two estimated parameters). This is wrong. The df are inflated and the estimate of the standard error of the difference (the denominator of the t-value) is falsely small. The correct sample size for this design is 1 and the correct df is zero. The sample size and df are inflated for this design because the treatment was randomized to mouse and not to cell. Mouse is the experimental unit – the number of experimental units is what gives the degrees of freedom. The df are correct for inference about the two individuals (how compatible are the data and a model of sampling from the same individual?), but not for inference about the effect of genotype. We cannot infer anything about genotype with a sample size of 1, even with 20 measures per mouse, because any effect of treatment is completely confounded with other differences between the two mice.
</div>
</div>
<div id="experimental-designs-in-experimental-bench-biology" class="section level2 hasAnchor" number="16.3">
<h2><span class="header-section-number">16.3</span> Experimental designs in experimental bench biology<a href="lmm.html#experimental-designs-in-experimental-bench-biology" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Given the basic principles above, let’s consider the kinds of experimental designs seen in experimental bench biology (Fig. <a href="#lmm-designs"><strong>??</strong></a>).</p>
<div class="figure"><span style="display:block;" id="fig:lmm-designs-1"></span>
<img src="images/experimental_designs/designs_1.png" alt="Experimental designs in experimental bench biology. Images created with BioRender.com" width="1526" />
<p class="caption">
Figure 16.1: Experimental designs in experimental bench biology. Images created with BioRender.com
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:lmm-designs-2"></span>
<img src="images/experimental_designs/designs_2.png" alt="Experimental designs in experimental bench biology. Images created with BioRender.com" width="1011" />
<p class="caption">
Figure 16.2: Experimental designs in experimental bench biology. Images created with BioRender.com
</p>
</div>
<div id="notation-for-models" class="section level3 hasAnchor" number="16.3.1">
<h3><span class="header-section-number">16.3.1</span> Notation for models<a href="lmm.html#notation-for-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>i = 1..t (treatments)</p>
<p>j = 1..b (blocks)</p>
<p>k = 1..r (experimental replications within a block or within a CRD with no block structure)</p>
<p>m = 1..s (subsamples or technical replicates)</p>
</div>
<div id="completely-randomized-design-crd" class="section level3 hasAnchor" number="16.3.2">
<h3><span class="header-section-number">16.3.2</span> Completely Randomized Design (CRD)<a href="lmm.html#completely-randomized-design-crd" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <strong>Completely Randomized Design</strong> experiment in Figure <a href="#fig:lmm-designs"><strong>??</strong></a>A has a single factor, <span class="math inline">\(\texttt{treatment}\)</span> with two levels (“Cn” and “Tr”). Five mice are randomly assigned to each treatment level. Each mouse is bred from a different litter and housed in a separate cage. The researchers measure a single value of the response variable from each mouse. The five replicate mice per treatment are the <strong>treatment (biological) replicates</strong>. The design is completely randomized because there is no subgrouping due to batches. What kinds of subgrouping does this design avoid?</p>
<p>By using a single mouse per litter, there are no litter batches and subsets of mice don’t share <strong>litter effects</strong> – common litter responses to the Cn or Tr treatments. Each litter has a unique set of factors that contribute to the error variance of the measure of the response. Siblings from the same dam and sire share more genetic variation than non-siblings and this shared genetic variation contributes to phenotypes (including the response to treatment) that are more likely to be similar to each other than to non-siblings. Siblings from the same litter share the same history of maternal factors (<strong>maternal effects</strong>, including <strong>epigenetic effects</strong>) specific to the pregnancy and even the history of events leading up to the pregnancy. This shared non-genetic and epigenetic variation contributes to phenotypes (including the response to treatment) that are more likely to be similar to each other than to non-siblings. All response measures within a litter share the genetic, maternal environmental, and epigenetic components of the error variance unique to that litter and, as a consequence, the error (residuals) within a litter are more similar to each other than they are to the residuals between litters.</p>
<p>By housing each mouse in it’s own cage, there is no cage batch and subsets of mice don’t share <strong>cage effects</strong> – common cage responses to the Cn or Tr treatments. As stated earlier, each cage has a unique set of factors that contribute to the error variance of the measure of the response. Each cage shares a cage-specific history of temperature, humidity, food, light, interactions with animal facilities staff and behavioral interactions among the mice. All response measures within a cage share the component of the error variance unique to that cage and, as a consequence, the error (residuals) within a cage are more similar to each other than they are to the residuals between cages.</p>
<p>Examples:</p>
<ol style="list-style-type: decimal">
<li>Ten mice from separate litters are sampled. Five mice are randomly assigned to control. Five mice are randomly assigned to treatment. A single measure per mouse is taken. Mouse is the experimental unit. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=0\)</span>, <span class="math inline">\(r=5\)</span>, and <span class="math inline">\(s=1\)</span>.</li>
<li>Ten cell cultures are created. Five cultures are randomly assigned to control and five to treatment. A single measure per culture is taken. Culture is the experimental unit. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=0\)</span>, <span class="math inline">\(r=5\)</span>, and <span class="math inline">\(s=1\)</span>.</li>
</ol>
div style=“background-color:#cccccc; text-align:left; vertical-align: middle; padding:20px 47px;”&gt;
While most data from experiments in bench biology are analyzed as if the experimental design is a CRD, a good question is, what fraction of these actually are CRD? We know that many (most?) mouse experiments will have batched measures and correlated responses because most experiments are conducted with multiple mice per litter and/or cage (or equivalents in other model systems). Many cell culture data come from multiple replicates of the whole experiment – the experiment functions as a block. And time series experiments include multiple measures on the same experimental unit over time. Except for time series experiments, researchers in experimental bench biology are using almost exclusively statistical tests that assume independence of errors (the tests appropriate for CRDs). How this mismatch between experimental design and statistical practice affects the rate of false and true discovery in cell and molecular biology is entirely unknown.
</div>
</div>
<div id="completely-randomized-design-with-subsampling-crds-or-nested-design" class="section level3 hasAnchor" number="16.3.3">
<h3><span class="header-section-number">16.3.3</span> Completely Randomized Design with Subsampling (CRDS) or “Nested Design”<a href="lmm.html#completely-randomized-design-with-subsampling-crds-or-nested-design" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <strong>Completely Randomized Design with Subsampling</strong> experiment in Fig. <a href="#fig:lmm-designs"><strong>??</strong></a>B is exactly like the CRD except that the researchers measure multiple values of the response variable from each mouse <em>under the same condition</em> (that is, not different in treatment or time). The multiple measures are <strong>subsampled (technical) replicates</strong>.</p>
<ol style="list-style-type: decimal">
<li>Do not confuse subsampled replicates with measures of the response under different conditions in the same mouse, for example a measure from one brain slice under the control treatment and a measure from a second brain slice under the drug treatment. This example is a kind of Randomized Complete Block Design, which is outlined next and the core design in this chapter.</li>
<li>Do not confuse subsampled replicates with a measures of the response at different times in the same mouse, for example, the plasma glucose levels at baseline and at five post-baseline time points. This example is a kind of longitudinal design, which is outlined below and more thoroughly in the chapter <a href="pre-post.html#pre-post">Linear models for longitudinal experiments</a>.</li>
<li>Do not confuse subsampled replicates with measures of different response variables from the same mouse, for example measures of the weights of five different skeletal muscles. This example is a kind of <strong>multiple response</strong> which is addressed xxx.</li>
</ol>
<p>The technical replicates are a kind of <strong>pseudoreplication</strong>. The general linear model <code>y ~ treatment</code> fit to these data, including t-tests and traditional ANOVA, will have falsely high precision and falsely low <em>p</em>-values.</p>
<p>Examples:</p>
<ol style="list-style-type: decimal">
<li>Ten mice from separate litters are sampled. Five mice are randomly assigned to control. Five mice are randomly assigned to treatment. Multiple measures per mouse are taken. Example: five measures of Islet Area are measured in each pancreas. Mouse is the experimental unit. Each of the measures is a technical replicate because the treatment is not randomly assigned to each islet but to the whole mouse. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=0\)</span>, <span class="math inline">\(r=5\)</span>, and <span class="math inline">\(s=5\)</span>.</li>
<li>Ten cell cultures are created. Five cultures are randomly assigned to control and five to treatment. Multiple measures per culture are taken. Example: Mitochondrial counts are measured from five cells in each culture. Culture is the experimental unit. Each of the counts is a technical replicate because the treatment is not randomly assigned to each cell but to the whole culture. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=0\)</span>, <span class="math inline">\(r=5\)</span>, and <span class="math inline">\(s=5\)</span>.</li>
</ol>
<p>Notes:</p>
<ol style="list-style-type: decimal">
<li>Subsampling can occur at multiple levels. Example: Ten mice from separate litters are sampled. Five mice are randomly assigned to control. Five mice are randomly assigned to treatment. Five neurons in a slice of brain tissue are identified in each mouse. From each neuron, the length of five dendrite spines are measured. The five measures of spine length are “nested within” neuron and the five neurons are “nested within” mouse. Nested subsampling can quickly lead to massive pseudoreplication and false discovery.</li>
</ol>
</div>
<div id="randomized-complete-block-design-rcbd" class="section level3 hasAnchor" number="16.3.4">
<h3><span class="header-section-number">16.3.4</span> Randomized Complete Block Design (RCBD)<a href="lmm.html#randomized-complete-block-design-rcbd" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <strong>Randomized Complete Block Design</strong> experiment in Figure <a href="#fig:lmm-designs"><strong>??</strong></a>C is similar to the CRD except that all treatment combinations (two here) are randomly assigned to sibling mice within a litter. Here, two mice from each litter are randomly selected and one is randomly assigned to “Cn” and the other to “Tr”. Each litter is randomly assigned to a unique cage. The researchers measure a single value of the response variable from each mouse. The five replicate mice per treatment are the treatment replicates. The litters (or cage) are the blocks. In this design, litter and cage effects are confounded but this has no consequence on the statistical model and inference unless the researchers want to explicitly estimate these effects separately. Compared to the CRD, this design requires fewer resources (five litters instead of ten, five cages instead of ten). Compared to the general linear model fit to data from the CRD (<code>y ~ treatment</code>), including t-tests and traditional ANOVA, the linear mixed model fit to the RCBD has increased precision and power. While many researchers seem to be designing experiments similar to this (“littermate controls”), most are failing to fit a statistical model that accounts for the batching and taking advantage of the increased precision and power.</p>
<p>Examples:</p>
<ol style="list-style-type: decimal">
<li>Ten mice are sampled. In each mouse, one forelimb is assigned to control and the other forelimb is assigned to treatment. Only a single measure on each side is taken. Limb is the experimental unit. Mouse is a block. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=10\)</span>, <span class="math inline">\(r=1\)</span>, and <span class="math inline">\(s=1\)</span>.</li>
<li>Ten litters are sampled. In each litter, one sib is assigned to control and the other sib is assigned to treatment. Only a single measure on each sib is taken. Mouse is the experimental unit. Litter is a block. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=10\)</span>, <span class="math inline">\(r=1\)</span>, and <span class="math inline">\(s=1\)</span>.</li>
<li>Two mice, each from a separate litter are sampled. One is randomly assigned to control and the other to treatment. Only a single measure of the response variable is taken per mouse. The experiment is replicated five times (five different days, each with a newly made set of reagents and machine calibrations). Mouse is the experimental unit. Experiment is a block. <span class="math inline">\(t=2\)</span>, <span class="math inline">\(b=5\)</span>, <span class="math inline">\(r=1\)</span>, and <span class="math inline">\(s=1\)</span>.</li>
</ol>
</div>
<div id="randomized-split-plot-design-rspd" class="section level3 hasAnchor" number="16.3.5">
<h3><span class="header-section-number">16.3.5</span> Randomized Split Plot Design (RSPD)<a href="lmm.html#randomized-split-plot-design-rspd" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <strong>Randomized Split Plot Design</strong> experiment in Figure <a href="#fig:lmm-designs"><strong>??</strong></a>D is similar to the RCBD except that there is now a second experimental factor that is crossed with the first experimental factor. An individual mouse acts as single experimental unit for one factor (here, <span class="math inline">\(\texttt{genotype}\)</span> with levels “WT” and “KO”) but acts as a block for the second experimental factor (here, <span class="math inline">\(\texttt{treatment}\)</span> with levels “Cn” and “Tr”). The first factor (<span class="math inline">\(\texttt{genotype}\)</span>) is the <strong>main plot</strong> – the levels of the factor are randomly assigned to the main plots. The second factor (<span class="math inline">\(\texttt{treatment}\)</span>) is the <strong>subplot</strong> – the levels of this factor are randomly assigned to the subplots. <span class="math inline">\(\texttt{Litter}\)</span> is a replicated block. Also in Figure <a href="#fig:lmm-designs"><strong>??</strong></a>D, a 2 x 2 RCBD with the same two experimental factors is shown for comparison. In the 2 x 2 RCBD, four mice per block (litter) are each randomly assigned one of the 2 x 2 combinations of <span class="math inline">\(\texttt{genotype}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>).</p>
</div>
<div id="generalized-randomized-complete-block-design-grcbd" class="section level3 hasAnchor" number="16.3.6">
<h3><span class="header-section-number">16.3.6</span> Generalized Randomized Complete Block Design (GRCBD)<a href="lmm.html#generalized-randomized-complete-block-design-grcbd" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <strong>Generalized Randomized Complete Block Design</strong> experiment in Figure <a href="#fig:lmm-designs"><strong>??</strong></a>E is similar to the RCBD except that two treatment replicates per block (litter/cage) are assigned.</p>
<p><strong>Important and somewhat not intuitive</strong> Because the treatment replicates within a litter share common error variance, these do not act like independent replicates. One consequence of this is, the sample size (<span class="math inline">\(n\)</span>) is five and not ten (<span class="math inline">\(litters \times treatment replicates\)</span>). The general linear model <code>y ~ treatment</code> fit to these data, including t-tests and traditional ANOVA, will generally have falsely high precision and falsely low <em>p</em>-values.</p>
</div>
<div id="nested-randomized-complete-block-design-nrcbd" class="section level3 hasAnchor" number="16.3.7">
<h3><span class="header-section-number">16.3.7</span> Nested Randomized Complete Block Design (NRCBD)<a href="lmm.html#nested-randomized-complete-block-design-nrcbd" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <strong>Nested Randomized Complete Block Design</strong> experiment in Figure <a href="#fig:lmm-designs"><strong>??</strong></a>F is similar to the RCBD except that there are now replicated experiments. This is a nested block design with litter (a block) nested within experiment (a block).</p>
</div>
<div id="longitudinal-randomized-complete-block-design-lrcbd" class="section level3 hasAnchor" number="16.3.8">
<h3><span class="header-section-number">16.3.8</span> Longitudinal Randomized Complete Block Design (LRCBD)<a href="lmm.html#longitudinal-randomized-complete-block-design-lrcbd" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <strong>Longitudinal Randomized Complete Block Design</strong> experiment in Figure <a href="#fig:lmm-designs"><strong>??</strong></a>G is similar to the RCBD except that there are multiple measures of the response variable taken, each taken at a different time point, including <strong>baseline</strong> (time zero).</p>
</div>
<div id="variations-due-to-multiple-measures-of-the-response-variable" class="section level3 hasAnchor" number="16.3.9">
<h3><span class="header-section-number">16.3.9</span> Variations due to multiple measures of the response variable<a href="lmm.html#variations-due-to-multiple-measures-of-the-response-variable" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Similar to the CRDS above, the other basic designs can include subsampling, resulting in, for example, RCBDS or RSPDS. If there is subsampling within subsampled units, then we can designated these with “SS”, for example RCBDSS.</p>
<p>Similar to the LRCBD above, the other basic designs can include longitudinal sampling, resulting in, for example, LCRD or LRSPD.</p>
</div>
</div>
<div id="building-the-linear-mixed-model-for-clustered-data" class="section level2 hasAnchor" number="16.4">
<h2><span class="header-section-number">16.4</span> Building the linear (mixed) model for clustered data<a href="lmm.html#building-the-linear-mixed-model-for-clustered-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Notation</p>
<ol style="list-style-type: decimal">
<li>i = 1..t (treatments)</li>
<li>j = 1..b (blocks)</li>
<li>k = 1..r (experimental replications within a block)</li>
<li>m = 1..s (subsamples or technical replicates)</li>
</ol>
<p><span class="math display">\[y_i = \beta_0 + \beta_i treatment_i + (\gamma_j block_j) + (\gamma_{ij} block_j treatment_i) + \varepsilon\]</span></p>
</div>
<div id="lmm-demo" class="section level2 hasAnchor" number="16.5">
<h2><span class="header-section-number">16.5</span> Example 1 – A random intercepts and slopes explainer (demo1)<a href="lmm.html#lmm-demo" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To introduce linear mixed models, I’m using data from <a href="lmm.html#lmm-example4">Experiment 1g</a> below. The design is <span class="math inline">\(2 \times 2\)</span> factorial with 4-5 mice per treatment combination. To simplify the explanation of <strong>random intercepts</strong> and <strong>random slopes</strong> in linear models with added random factors (<strong>linear mixed models</strong>), I flatten the analysis to a single treatment factor (<span class="math inline">\(\texttt{treatment}\)</span>) with four levels (“Control”, “Tr1”, “Tr2”, “Tr3). The response is percent germinal centers (<span class="math inline">\(\texttt{gc}\)</span>) in <a href="https://en.wikipedia.org/wiki/Lymphatic_system#Secondary_lymphoid_organs" target="_blank">secondary lymphoid tissue</a>. The experiment was replicated 4 times. Each replication is a batch. This batch information is in the variable <span class="math inline">\(\texttt{experiment_id}\)</span>. Further detail isn’t necessary at this point.</p>
<p>Figure <a href="lmm.html#fig:lmm-explainer1a">16.3</a>A is a response plot of the linear model <code>lm(gc ~ treatment)</code> fit to the whole data set, ignoring the fact that the data were collected in batches. This is the <strong>complete pooling</strong> fit. Figure <a href="lmm.html#fig:lmm-explainer1a">16.3</a>B is a response plot of the linear model <code>lm(gc ~ treatment)</code> fit to the means of each treatment combination from each experiment. This is the <strong>means pooling</strong> fit.</p>
<div class="figure"><span style="display:block;" id="fig:lmm-explainer1a"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-explainer1a-1.png" alt="A. Response plot of the linear model gc ~ treatment fit to all exp1g data. B. Response plot of the linear model gc ~ treatment fit to the experiment means of the exp1g data." width="1066.66666666667" />
<p class="caption">
Figure 16.3: A. Response plot of the linear model gc ~ treatment fit to all exp1g data. B. Response plot of the linear model gc ~ treatment fit to the experiment means of the exp1g data.
</p>
</div>
<div id="batched-measurements-result-in-clustered-residuals" class="section level3 hasAnchor" number="16.5.1">
<h3><span class="header-section-number">16.5.1</span> Batched measurements result in clustered residuals<a href="lmm.html#batched-measurements-result-in-clustered-residuals" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Figure <a href="lmm.html#fig:lmm-explainer1b">16.4</a>A is a plot of the residuals of the complete-pooling fit against <span class="math inline">\(\texttt{experiment_id}\)</span>. The residuals are <strong>clustered</strong> by experiment. All residuals from experiment 1 are positive. All residuals from experiment 2 are negative. Residuals from experiment 3 are generally positive. Residuals from experiment 4 seem pretty random. This clustering by experiment is the same in the plot of the residuals of the means-pooling fit against <span class="math inline">\(\texttt{experiment_id}\)</span> (Figure <a href="lmm.html#fig:lmm-explainer1b">16.4</a>B). The residuals are not independent in either fit. If you asked me to guess the sign of a residual and gave me the information that the measure was from experiment 1, I’d be correct 100% of the time. If the residuals were independent, I’d be correct, on average, 50% of the time. Independent residuals are randomly scattered about zero for within each experiment (Figure <a href="lmm.html#fig:lmm-explainer1b">16.4</a>C).</p>
<div class="figure"><span style="display:block;" id="fig:lmm-explainer1b"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-explainer1b-1.png" alt="A. Residuals of the model fit to all demo1 data. B. Residuals of the model fit to the mean demo1 data." width="1066.66666666667" />
<p class="caption">
Figure 16.4: A. Residuals of the model fit to all demo1 data. B. Residuals of the model fit to the mean demo1 data.
</p>
</div>
</div>
<div id="lmm-corerr" class="section level3 hasAnchor" number="16.5.2">
<h3><span class="header-section-number">16.5.2</span> Clustered residuals result in correlated error<a href="lmm.html#lmm-corerr" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>An assumption of inference from a linear model is independence – each response is independently drawn from a distribution of random values. In Experiment 1g, the experiments are batches and the batched data results in correlated error unless modeled. One way to see this correlated error is to use the residuals from the means-pooled fit.</p>
<ol style="list-style-type: decimal">
<li>aggregate the data by computing the means for each treatment level within each experiment.</li>
<li>fit the fixed effect model (the model without added random factors) to the aggregated data</li>
<li>compute the residuals from the model</li>
<li>cast (or spread) the residuals for each treatment into its own column. This creates a 4 rows (experiment) <span class="math inline">\(\times\)</span> 4 columns (treatments) matrix of residuals.</li>
<li>Compute the correlations among the four treatment combination columns. This is the correlated error due to the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-demo1-cor-error-table">Table 16.1: </span>Residuals of fixed affect model fit to aggregated data. The residuals are split into each treatment.
</caption>
<thead>
<tr>
<th style="text-align:left;">
experiment_id
</th>
<th style="text-align:right;">
Control
</th>
<th style="text-align:right;">
Tr1
</th>
<th style="text-align:right;">
Tr2
</th>
<th style="text-align:right;">
Tr3
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
exp_1
</td>
<td style="text-align:right;">
2.578125
</td>
<td style="text-align:right;">
1.64075
</td>
<td style="text-align:right;">
2.2425
</td>
<td style="text-align:right;">
3.027125
</td>
</tr>
<tr>
<td style="text-align:left;">
exp_2
</td>
<td style="text-align:right;">
-2.229375
</td>
<td style="text-align:right;">
-2.91625
</td>
<td style="text-align:right;">
-5.7475
</td>
<td style="text-align:right;">
-4.435375
</td>
</tr>
<tr>
<td style="text-align:left;">
exp_3
</td>
<td style="text-align:right;">
-0.615375
</td>
<td style="text-align:right;">
1.09975
</td>
<td style="text-align:right;">
3.4625
</td>
<td style="text-align:right;">
3.697125
</td>
</tr>
<tr>
<td style="text-align:left;">
exp_4
</td>
<td style="text-align:right;">
0.266625
</td>
<td style="text-align:right;">
0.17575
</td>
<td style="text-align:right;">
0.0425
</td>
<td style="text-align:right;">
-2.288875
</td>
</tr>
</tbody>
</table>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-demo1-cor-error-1.png" width="576" /></p>
<p>I’ve used <code>GGally::ggpairs</code> to compute and display the correlations as a matrix. The lower triangle of matrix elements contains the scatterplot of the residuals for the treatment combination defined by the row and column headers. The upper triangle of elements contains the Pearson correlation. With only four experiment residuals per treatment combination, large correlations will be common. But all correlations are large, positive values. The asterisks indicate values that would be an unexpected surprise under a null model of no correlation.</p>
<p>We could explicitly model correlated error with a <strong>linear model for correlated error</strong> using the <code>nlme::gls</code> function, using a model for the correlated error that matches our knowledge of how the data were generated (from experiment batches). In this chapter, we implicitly model the correlated error using a linear model with added random factors – a <strong>linear mixed model</strong>. What we explicitly model in a linear mixed model is hierarchical levels of variance.</p>
</div>
<div id="in-blocked-designs-clustered-residuals-adds-a-variance-component-that-masks-treatment-effects" class="section level3 hasAnchor" number="16.5.3">
<h3><span class="header-section-number">16.5.3</span> In blocked designs, clustered residuals adds a variance component that masks treatment effects<a href="lmm.html#in-blocked-designs-clustered-residuals-adds-a-variance-component-that-masks-treatment-effects" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The variance among the experiments within a treatment is much greater than the variance among the treatment means. A consequence of this is, the experiment effect masks the effect of treatment. We can manually unmask this by</p>
<ol style="list-style-type: decimal">
<li>compute the experiment means across all treatment combinations.</li>
<li>create a gc variable without variation among experiment means (“adjusted for experiment_id”).</li>
</ol>
<div class="figure"><span style="display:block;" id="fig:lmm-demo1-manual-lmm-plot"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-demo1-manual-lmm-plot-1.png" alt="Adjusting for variance among experiments. The black, dashed line is the grand-mean response. In the left panel, the colored, dashed lines are the mean gc for each experiment, ignoring treatment. In the right panel, the individual values have been shifted (adjusted) by centering the experiment means. This has the effect of reducing the error variance -- the spread of the values around the treatment means (large black dots)." width="576" />
<p class="caption">
Figure 16.5: Adjusting for variance among experiments. The black, dashed line is the grand-mean response. In the left panel, the colored, dashed lines are the mean gc for each experiment, ignoring treatment. In the right panel, the individual values have been shifted (adjusted) by centering the experiment means. This has the effect of reducing the error variance – the spread of the values around the treatment means (large black dots).
</p>
</div>
<p>In Figure <a href="lmm.html#fig:lmm-demo1-manual-lmm-plot">16.5</a>, the black dots are the modeled means of each treatment combination. The small colored dots are the measured values of the response for each <span class="math inline">\(\texttt{experiment_id}\)</span> in the left panel and the experiment-adjusted values in the right panel. The black, dashed line is the grand-mean response. The colored, dashed lines are the means of all responses in each experiment. These means are equal in the right panel (and covered by the black line) because the variation among the means has been adjusted away. What is left is error variation uncontaminated by <span class="math inline">\(\texttt{experiment_id}\)</span>.</p>
<p>In Experiment 1g, <span class="math inline">\(\texttt{experiment_id}\)</span> is a nuissance variable – it adds to the noise. In the exercise above, the effects of the treatment variables are adjusted for the elevation of batch effects on the overal batch mean. Linear mixed models are more sophisticated than this. In a linear mixed model, the effects of the treatment variables are adjusted for the elevations of batch effects on the intercept and batch effects on the slopes (or some combination of these). These are the <strong>random intercepts</strong> and <strong>random slopes</strong>.</p>
</div>
<div id="linear-mixed-models-are-linear-models-with-added-random-factors" class="section level3 hasAnchor" number="16.5.4">
<h3><span class="header-section-number">16.5.4</span> Linear mixed models are linear models with added random factors<a href="lmm.html#linear-mixed-models-are-linear-models-with-added-random-factors" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A linear model adds some combination of <strong>random intercepts</strong> and <strong>random slopes</strong> to a linear model.</p>
<p><span class="math display" id="eq:lmm-demo1-m1">\[
\begin{equation}
\texttt{gc}_{jk} = (\beta_{0} + \gamma_{0j}) + (\beta_{k} + \gamma_{kj}) \texttt{treatment}_{k} + \varepsilon
\tag{16.1}
\end{equation}
\]</span></p>
<p>A <strong>random intercept</strong> for experiment <em>j</em> is the sum of the fixed intercept (<span class="math inline">\(\beta_0\)</span>) and a <strong>random intercept effect</strong> (<span class="math inline">\(\gamma_{0j}\)</span>). I’ve embedded these within parentheses to show how these combine into the random intercept. A <strong>random slope</strong> for batch <em>j</em> is the sum of the fixed slope (<span class="math inline">\(\beta_k\)</span>) for the non-reference level <span class="math inline">\(k\)</span> and a <strong>random slope effect</strong> (<span class="math inline">\(\gamma_{kj}\)</span>). I’ve embedded these within parentheses to show how these combine into the random slopes.</p>
<p>There is a different <span class="math inline">\(\gamma_{0j}\)</span> for each experiment. There is a different <span class="math inline">\(\gamma_{kj}\)</span> for each combination of non-reference level and experiment. The <span class="math inline">\(\gamma_0j\)</span> and <span class="math inline">\(\gamma_kj\)</span> are modeled as if the values for each experiment is a random draw from an infinite number of experiments. This is why <span class="math inline">\(\gamma_{0j}\)</span> and <span class="math inline">\(\gamma_{kj}\)</span> are <strong>random effects</strong>. In contrast, <span class="math inline">\(\beta_0\)</span> and the three <span class="math inline">\(\beta_k\)</span> for the non-reference treatment levels are the same for all experiments – this is why <span class="math inline">\(\beta_0\)</span> is known as <strong>fixed effects</strong> (technically, <span class="math inline">\(\beta_0\)</span> is a mean and not an effect).</p>
<p>Model <a href="lmm.html#eq:lmm-demo1-m1">(16.1)</a> is fit to the Example 1 data using <code>lme4::lmer()</code></p>
<div class="sourceCode" id="cb499"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb499-1"><a href="lmm.html#cb499-1" tabindex="-1"></a>demo1_m1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> treatment <span class="sc">+</span></span>
<span id="cb499-2"><a href="lmm.html#cb499-2" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb499-3"><a href="lmm.html#cb499-3" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li><code>(treatment | experiment_id)</code> specifies a random intercept for all levels of <span class="math inline">\(\texttt{experiment_id}\)</span> and
a random slope for all combinations the levels of <span class="math inline">\(\texttt{experiment_id}\)</span> and the non-reference levels of <span class="math inline">\(\texttt{treatment}\)</span></li>
</ol>
</div>
<div id="what-the-random-effects-are" class="section level3 hasAnchor" number="16.5.5">
<h3><span class="header-section-number">16.5.5</span> What the random effects are<a href="lmm.html#what-the-random-effects-are" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Random intercepts model batch effects in the reference treatment level</strong>. <a href="lmm.html#fig:lmm-demo1-explainer-2c">16.6</a>A illustrates random intercepts and random intercept effects. The large, colored dots are the modeled means of each experiment for each treatment combination. For the reference treatment level (“Control”), each mean is the sum of the estimated fixed intercept (<span class="math inline">\(b_0\)</span>), shown by the dashed gray line, and the estimated random intercept effect (<span class="math inline">\(g_{0j}\)</span>) for experiment <em>j</em>. The random intercept effects are the vertical, colored lines.</p>
<p><strong>Random slopes model the effect of treatment on batch effects in the non-reference treatment levels</strong>. <a href="lmm.html#fig:lmm-demo1-explainer-2c">16.6</a>B illustrates random slopes and random slope effects, focusing on the slopes for the 2nd non-reference treatment level (“Tr2”). The angled black line is the estimated fixed slope <span class="math inline">\(b_2\)</span> for this level. The colored lines are the random slopes for each experiment. The pale, gray dots are where the modeled means at the Tr2 level would be if there were no random slope effect – as if we took the large colored dots at “Control” and rigidly shifted them up the black line to “Tr2”. The estimated random slope effects <span class="math inline">\(\mathrm{g}_{2j}\)</span> are the difference between these large, gray dots and the modeled means.</p>
<div class="figure"><span style="display:block;" id="fig:lmm-demo1-explainer-2c"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-demo1-explainer-2c-1.png" alt="What random intercepts and slopes are. (A) A random intercept for batch $j$ is the difference between the fixed intercept and the modeled mean for batch $j$ in the reference treatment level. The random intercepts for experiments 1 ($\mathrm{g}_{0.1}$) and 2 ($\mathrm{g}_{0.2}$) are shown with brackets. (B) The fixed slope for the &quot;Tr2&quot; is illustrated with a bracket. The large, grey dots are the expected values for each batch (experiment_id) in the &quot;Tr2&quot; treatment if the random slope effects are zero. A random slope for batch $j$ is the difference between the expected value for batch j and the modeled mean for batch $j$. The random slope for experiment 2 in &quot;Tr2&quot; ($\mathrm{g}_{2.2}$) is shown with a bracket." width="1066.66666666667" />
<p class="caption">
Figure 16.6: What random intercepts and slopes are. (A) A random intercept for batch <span class="math inline">\(j\)</span> is the difference between the fixed intercept and the modeled mean for batch <span class="math inline">\(j\)</span> in the reference treatment level. The random intercepts for experiments 1 (<span class="math inline">\(\mathrm{g}_{0.1}\)</span>) and 2 (<span class="math inline">\(\mathrm{g}_{0.2}\)</span>) are shown with brackets. (B) The fixed slope for the “Tr2” is illustrated with a bracket. The large, grey dots are the expected values for each batch (experiment_id) in the “Tr2” treatment if the random slope effects are zero. A random slope for batch <span class="math inline">\(j\)</span> is the difference between the expected value for batch j and the modeled mean for batch <span class="math inline">\(j\)</span>. The random slope for experiment 2 in “Tr2” (<span class="math inline">\(\mathrm{g}_{2.2}\)</span>) is shown with a bracket.
</p>
</div>
</div>
<div id="in-a-blocked-design-a-linear-model-with-added-random-effects-increases-precision-of-treatment-effects" class="section level3 hasAnchor" number="16.5.6">
<h3><span class="header-section-number">16.5.6</span> In a blocked design, a linear model with added random effects increases precision of treatment effects<a href="lmm.html#in-a-blocked-design-a-linear-model-with-added-random-effects-increases-precision-of-treatment-effects" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><span class="math display" id="eq:lmm-demo1-m0">\[
\begin{equation}
\texttt{gc}_{jk} = \beta_{0} + \beta_{k} \texttt{treatment}_{k} +  (\gamma_{0j} + \gamma_{kj}\texttt{treatment}_{k} + \varepsilon)
\tag{16.2}
\end{equation}
\]</span></p>
<p>If the random intercepts and random slopes aren’t modeled, this among-experiment variance is shifted to the error variance because the intercept effects and slope effects aren’t estimated but absorbed by the error – everything in Model <a href="lmm.html#eq:lmm-demo1-m0">(16.2)</a> will be estimated by the residuals. As a consequence, the estimate of <span class="math inline">\(\sigma\)</span> (the square root of the error variance) for the linear mixed model is smaller than that for the linear model with only fixed effects.</p>
<div class="sourceCode" id="cb500"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb500-1"><a href="lmm.html#cb500-1" tabindex="-1"></a><span class="co"># sigma for the lmm</span></span>
<span id="cb500-2"><a href="lmm.html#cb500-2" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> treatment <span class="sc">+</span></span>
<span id="cb500-3"><a href="lmm.html#cb500-3" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb500-4"><a href="lmm.html#cb500-4" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span>
<span id="cb500-5"><a href="lmm.html#cb500-5" tabindex="-1"></a><span class="fu">summary</span>(m1)<span class="sc">$</span>sigma</span></code></pre></div>
<pre><code>## [1] 1.931339</code></pre>
<div class="sourceCode" id="cb502"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb502-1"><a href="lmm.html#cb502-1" tabindex="-1"></a><span class="co"># sigma for the fixed lm</span></span>
<span id="cb502-2"><a href="lmm.html#cb502-2" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(gc <span class="sc">~</span> treatment,</span>
<span id="cb502-3"><a href="lmm.html#cb502-3" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span>
<span id="cb502-4"><a href="lmm.html#cb502-4" tabindex="-1"></a><span class="fu">summary</span>(m2)<span class="sc">$</span>sigma</span></code></pre></div>
<pre><code>## [1] 3.376402</code></pre>
<p>The consequence of the smaller estimate of <span class="math inline">\(\sigma\)</span> in the linear mixed model on inference (confidence intervals and <em>p</em>-values) depends on the number of subsamples, the <strong>variance of the random effects</strong> relative to the variance of the residual error, and the <strong>correlation among the random effects</strong>.</p>
</div>
<div id="lmm-varcorr" class="section level3 hasAnchor" number="16.5.7">
<h3><span class="header-section-number">16.5.7</span> The correlation among random intercepts and slopes<a href="lmm.html#lmm-varcorr" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Again, here is the linear mixed model fit to the experiment 1g data.</p>
<p><span class="math display">\[
\begin{equation}
\texttt{gc}_{jk} = (\beta_{0} + \gamma_{0j}) + (\beta_{k} + \gamma_{kj}) \texttt{treatment}_{k} + \varepsilon
\end{equation}
\]</span></p>
<p>Think about how this model <strong>generates data</strong>. We have four experiments, so we randomly draw four <span class="math inline">\(\gamma_{0j}\)</span> from a normal distribution with some variance <span class="math inline">\(\sigma_{0}^2\)</span>. And, for each non-reference treatment, we randomly draw four <span class="math inline">\(\gamma_{0k}\)</span> from a normal distribution with some variance <span class="math inline">\(\sigma_{k}^2\)</span>. This gives us a matrix of four columns (one random intercept and three random slopes) and four rows (four experiments).</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
intercept
</th>
<th style="text-align:left;">
slope 1
</th>
<th style="text-align:left;">
slope 2
</th>
<th style="text-align:left;">
slope 3
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{0.1}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{1.1}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{2.1}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{3.1}\)</span>
</td>
</tr>
<tr>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{0.2}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{1.2}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{2.2}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{3.2}\)</span>
</td>
</tr>
<tr>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{0.3}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{1.3}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{2.3}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{3.3}\)</span>
</td>
</tr>
<tr>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{0.4}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{1.4}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{2.4}\)</span>
</td>
<td style="text-align:left;">
<span class="math inline">\(\gamma_{3.4}\)</span>
</td>
</tr>
</tbody>
</table>
<p>To randomly sample these values, the model needs not only the variances (<span class="math inline">\(\sigma_k^2\)</span>) for each column (random effect) <em>but also</em> a correlation for each pair of columns. These correlations are the off-diagonal elements of the correlation matrix of random effects.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<tbody>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{0}\)</span>, <span class="math inline">\(\gamma_{1}\)</span>)
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{0}\)</span>, <span class="math inline">\(\gamma_{2}\)</span>)
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{0}\)</span>, <span class="math inline">\(\gamma_{3}\)</span>)
</td>
</tr>
<tr>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{1}\)</span>, <span class="math inline">\(\gamma_{0}\)</span>)
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{1}\)</span>, <span class="math inline">\(\gamma_{2}\)</span>)
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{1}\)</span>, <span class="math inline">\(\gamma_{3}\)</span>)
</td>
</tr>
<tr>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{2}\)</span>, <span class="math inline">\(\gamma_{0}\)</span>)
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{2}\)</span>, <span class="math inline">\(\gamma_{1}\)</span>)
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{2}\)</span>, <span class="math inline">\(\gamma_{3}\)</span>)
</td>
</tr>
<tr>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{3}\)</span>, <span class="math inline">\(\gamma_{0}\)</span>)
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{3}\)</span>, <span class="math inline">\(\gamma_{1}\)</span>)
</td>
<td style="text-align:left;">
COR(<span class="math inline">\(\gamma_{3}\)</span>, <span class="math inline">\(\gamma_{2}\)</span>)
</td>
<td style="text-align:left;">
1
</td>
</tr>
</tbody>
</table>
<p>In the models fit in this text, a researcher doesn’t specify these variances and correlations. Instead, these are parameters estimated by the model. Here is a summary of the estimates of the variances of the random effects and of the correlations among the random effects for the linear mixed model fit to the experiment 1g data.</p>
<pre><code>##  Groups        Name         Std.Dev. Corr                
##  experiment_id (Intercept)  1.7710                       
##                treatmentTr1 0.9642   -0.025              
##                treatmentTr2 2.9026    0.458  0.876       
##                treatmentTr3 2.9910    0.300  0.879  0.938
##  Residual                   1.9313</code></pre>
<p>The first four values in the column “Std.Dev.” are the square roots of the estimated variances for the random effects given in the column “Name”. The last value in column “Std.Dev.” is the square roots of the estimate of <span class="math inline">\(\sigma^2\)</span> (the error variance). The (lower) triangular matrix of values under “Corr” are the estimates of the correlations among the random effects. The variance of the random effects and the correlation among the random effects creates the <a href="lmm.html#lmm-corerr">correlated error described above</a> but do not confuse these different correlations (this is easy to confuse, you are not alone).</p>
<p>A compact way to view these variances and correlations is a matrix with the random effect standard deviations on the diagonal and the correlations on the off-diagonal. I’ll refer to this as the <em>VarCorr</em> matrix after the <code>lme4</code> function used to get the values.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-205">Table 16.2: </span>The Varcorr matrix. Standard deviations of random effects on the diagonal. Correlations of random effects on the off-diagonal.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
</th>
<th style="text-align:right;">
</th>
<th style="text-align:right;">
</th>
<th style="text-align:right;">
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
1.77
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentTr1
</td>
<td style="text-align:right;">
-0.03
</td>
<td style="text-align:right;">
0.96
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentTr2
</td>
<td style="text-align:right;">
0.46
</td>
<td style="text-align:right;">
0.88
</td>
<td style="text-align:right;">
2.90
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentTr3
</td>
<td style="text-align:right;">
0.30
</td>
<td style="text-align:right;">
0.88
</td>
<td style="text-align:right;">
0.94
</td>
<td style="text-align:right;">
2.99
</td>
</tr>
</tbody>
</table>
<p>It’s probably not worth trying to understand the experimental reason underneath the correlations among the random effects. But, researchers might want to sleuth out why a lab is getting high random intercept and slope variances, relative to the error variances, as these could indicate potential sources of improvement in lab protocols.</p>
</div>
<div id="clustered-residuals-create-heterogeneity-among-treatments" class="section level3 hasAnchor" number="16.5.8">
<h3><span class="header-section-number">16.5.8</span> Clustered residuals create heterogeneity among treatments<a href="lmm.html#clustered-residuals-create-heterogeneity-among-treatments" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The variances of the four treatments are</p>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
Var
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Control
</td>
<td style="text-align:right;">
7.8
</td>
</tr>
<tr>
<td style="text-align:left;">
Tr1
</td>
<td style="text-align:right;">
5.2
</td>
</tr>
<tr>
<td style="text-align:left;">
Tr2
</td>
<td style="text-align:right;">
16.2
</td>
</tr>
<tr>
<td style="text-align:left;">
Tr3
</td>
<td style="text-align:right;">
16.7
</td>
</tr>
</tbody>
</table>
<p>In chapter xxx, I stated that heterogeneity of variances can arise because of clustered data. Why does clustered data generate heterogeneity? Let’s keep peeking at the linear mixed model fit to the experiment 1g data.</p>
<p><span class="math display">\[
\begin{equation}
\texttt{gc}_{jk} = (\beta_{0} + \gamma_{0j}) + (\beta_{k} + \gamma_{kj}) \texttt{treatment}_{k} + \varepsilon
\end{equation}
\]</span></p>
<p>In a linear model with fixed effects only, the expected variance for any treatment for any treatment is <span class="math inline">\(\sigma^2\)</span>. But if the data are batched, the expected variances include components due to the batch and these batch components depend on the treatment. This creates heterogeneity.</p>
<p>To understand this, first some rules of expected variance. The random variable <span class="math inline">\(\texttt{C}\)</span> is the sum of two random variables <span class="math inline">\(\texttt{A}\)</span> and <span class="math inline">\(\texttt{B}\)</span>. The variances of these variables are <span class="math inline">\(\sigma_{C}^2\)</span>, <span class="math inline">\(\sigma_{A}^2\)</span>, and <span class="math inline">\(\sigma_{B}^2\)</span>.</p>
<ol style="list-style-type: decimal">
<li>The expected variance of <span class="math inline">\(\texttt{C}\)</span> if <span class="math inline">\(\texttt{A}\)</span> and <span class="math inline">\(\texttt{B}\)</span> are independent (uncorrelated) is <span class="math inline">\(\sigma_{C}^2 = \sigma_{A}^2 + \sigma_{B}^2\)</span> (<a href="https://en.wikipedia.org/wiki/Pythagorean_theorem" target="_blank">this equation should look familiar</a>).</li>
<li>The expected variance of <span class="math inline">\(\texttt{C}\)</span> if <span class="math inline">\(\texttt{A}\)</span> and <span class="math inline">\(\texttt{B}\)</span> are not independent (correlated) is <span class="math inline">\(\sigma_{C}^2 = \sigma_{A}^2 + \sigma_{B}^2 + 2\sigma_{A}\ \sigma_{B}\ \rho_{A,B}\)</span> where <span class="math inline">\(\rho_{A,B}\)</span> is the expected correlation between <span class="math inline">\(\texttt{A}\)</span> and <span class="math inline">\(\texttt{B}\)</span> (<a href="https://en.wikipedia.org/wiki/Law_of_cosines" target="_blank">this equation might also look familiar</a>).</li>
</ol>
<p>Here is some code to better know expected variances of the sum of two correlated random variables.</p>
<div class="sourceCode" id="cb505"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb505-1"><a href="lmm.html#cb505-1" tabindex="-1"></a><span class="co"># copy, paste, and explore</span></span>
<span id="cb505-2"><a href="lmm.html#cb505-2" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">4</span></span>
<span id="cb505-3"><a href="lmm.html#cb505-3" tabindex="-1"></a>rho <span class="ot">&lt;-</span> <span class="fl">0.6</span> <span class="co"># change this to any value between -1 and 1</span></span>
<span id="cb505-4"><a href="lmm.html#cb505-4" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">abs</span>(rho))</span>
<span id="cb505-5"><a href="lmm.html#cb505-5" tabindex="-1"></a>z <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n)</span>
<span id="cb505-6"><a href="lmm.html#cb505-6" tabindex="-1"></a>A <span class="ot">&lt;-</span> b<span class="sc">*</span>z <span class="sc">+</span> <span class="fu">sqrt</span>(<span class="dv">1</span><span class="sc">-</span>b<span class="sc">^</span><span class="dv">2</span>)<span class="sc">*</span><span class="fu">rnorm</span>(n)</span>
<span id="cb505-7"><a href="lmm.html#cb505-7" tabindex="-1"></a>B <span class="ot">&lt;-</span> <span class="fu">sign</span>(rho)<span class="sc">*</span>b<span class="sc">*</span>z <span class="sc">+</span> <span class="fu">sqrt</span>(<span class="dv">1</span><span class="sc">-</span>b<span class="sc">^</span><span class="dv">2</span>)<span class="sc">*</span><span class="fu">rnorm</span>(n)</span>
<span id="cb505-8"><a href="lmm.html#cb505-8" tabindex="-1"></a><span class="fu">cor</span>(A,B) <span class="co"># should be close to rho</span></span>
<span id="cb505-9"><a href="lmm.html#cb505-9" tabindex="-1"></a>C <span class="ot">&lt;-</span> A <span class="sc">+</span> B</span>
<span id="cb505-10"><a href="lmm.html#cb505-10" tabindex="-1"></a><span class="fu">sd</span>(A)<span class="sc">^</span><span class="dv">2</span> <span class="co"># should be close to 1</span></span>
<span id="cb505-11"><a href="lmm.html#cb505-11" tabindex="-1"></a><span class="fu">sd</span>(B)<span class="sc">^</span><span class="dv">2</span> <span class="co"># should be close to 1</span></span>
<span id="cb505-12"><a href="lmm.html#cb505-12" tabindex="-1"></a><span class="fu">sd</span>(C)<span class="sc">^</span><span class="dv">2</span> <span class="co"># should be close to 1^2 + 1^2 + 2*1*1*rho</span></span>
<span id="cb505-13"><a href="lmm.html#cb505-13" tabindex="-1"></a><span class="fu">sd</span>(A)<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="fu">sd</span>(B)<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="dv">2</span><span class="sc">*</span><span class="fu">sd</span>(A)<span class="sc">*</span><span class="fu">sd</span>(B)<span class="sc">*</span><span class="fu">cor</span>(A,B) <span class="co"># should equal previous line</span></span></code></pre></div>
<p>Using these rules and the standard deviations of the random effects given above we can compute the expected variances of the treatment groups given the fit model.</p>
<ol style="list-style-type: decimal">
<li>For the variance of the reference (“Control”) group, we need to add to <span class="math inline">\(\sigma^2\)</span> the variance of the random intercept using rule #1 (the residuals are not correlated with random intercepts or slopes). The modeled variance is a less than the actual variance of the Control group.</li>
</ol>
<div class="sourceCode" id="cb506"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb506-1"><a href="lmm.html#cb506-1" tabindex="-1"></a><span class="fu">summary</span>(m1)<span class="sc">$</span>sigma<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> <span class="fl">1.7710</span><span class="sc">^</span><span class="dv">2</span></span></code></pre></div>
<pre><code>## [1] 6.866512</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>For the variance of a non-reference group, we need to add to <span class="math inline">\(\sigma^2\)</span> the variance of the random intercept and the variance of the random slope for the treatment and the component due to the correlation between the random slope for the treatment and the random intercept. For Tr2, this is</li>
</ol>
<div class="sourceCode" id="cb508"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb508-1"><a href="lmm.html#cb508-1" tabindex="-1"></a><span class="co"># error + intercept + slope + cor(intercept, slope)</span></span>
<span id="cb508-2"><a href="lmm.html#cb508-2" tabindex="-1"></a>(<span class="fu">summary</span>(m1)<span class="sc">$</span>sigma<span class="sc">^</span><span class="dv">2</span>) <span class="sc">+</span> (<span class="fl">1.7710</span><span class="sc">^</span><span class="dv">2</span>) <span class="sc">+</span> (<span class="fl">2.9026</span><span class="sc">^</span><span class="dv">2</span>) <span class="sc">+</span> (<span class="dv">2</span> <span class="sc">*</span> <span class="fl">1.7710</span> <span class="sc">*</span> <span class="fl">2.9026</span> <span class="sc">*</span> <span class="fl">0.458</span>)</span></code></pre></div>
<pre><code>## [1] 20.0003</code></pre>
<p>which is a bit higher than the measured variance.</p>
</div>
<div id="linear-mixed-models-are-flexible" class="section level3 hasAnchor" number="16.5.9">
<h3><span class="header-section-number">16.5.9</span> Linear mixed models are flexible<a href="lmm.html#linear-mixed-models-are-flexible" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>One more look at the linear mixed model fit to the experiment 1g data.</p>
<p><span class="math display">\[
\begin{equation}
\texttt{gc}_{jk} = (\beta_{0} + \gamma_{0j}) + (\beta_{k} + \gamma_{kj}) \texttt{treatment}_{k} + \varepsilon
\end{equation}
\]</span></p>
<p>The linear mixed model specifies both a random intercept and a random slope but a researcher might limit the model to the random intercept only, or less commonly, the random slope only. Or a researcher might replace the random slope with a second random intercept that captures variance in the batch by treatment combinations like a random slope. Or a researcher might model the structure (correlated error and heterogeneity of variances) in the residuals <em>in addition to</em> adding random factors to the model.</p>
</div>
<div id="a-random-intercept-only-model" class="section level3 hasAnchor" number="16.5.10">
<h3><span class="header-section-number">16.5.10</span> A random intercept only model<a href="lmm.html#a-random-intercept-only-model" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb510"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb510-1"><a href="lmm.html#cb510-1" tabindex="-1"></a>demo1_m2 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> treatment <span class="sc">+</span></span>
<span id="cb510-2"><a href="lmm.html#cb510-2" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id),</span>
<span id="cb510-3"><a href="lmm.html#cb510-3" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Model <code>demo1_m2</code> specifies only a random intercept for each of the levels of <span class="math inline">\(\texttt{experiment_id}\)</span>. The exclusion of the random slopes ia a kind of <strong>model simplification</strong>.</li>
<li>In experiments without subsampling, random slopes cannot be added to the model because there is no variation with a treatment by batch combination.</li>
<li>In experiments with subsampling, a researcher might exclude a random slope term for several reasons, including</li>
</ol>
<ul>
<li>it is the culture in many subfields to only include a random intercept (no, this is not a good reason)</li>
<li>the computation of the model fit returned a convergence warning</li>
<li>model comparison suggested that a model with the random slope was too complex given the data. A useful statistic for comparing models with different random effects specifications is the <strong>AIC</strong>, which is introduced in Section <a href="lmm.html#lmm-demo-aic">16.5.12</a> below.</li>
</ul>
</div>
<div id="a-model-including-an-interaction-intercept" class="section level3 hasAnchor" number="16.5.11">
<h3><span class="header-section-number">16.5.11</span> A model including an interaction intercept<a href="lmm.html#a-model-including-an-interaction-intercept" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb511"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb511-1"><a href="lmm.html#cb511-1" tabindex="-1"></a>demo1_m3 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> treatment <span class="sc">+</span></span>
<span id="cb511-2"><a href="lmm.html#cb511-2" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb511-3"><a href="lmm.html#cb511-3" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>treatment),</span>
<span id="cb511-4"><a href="lmm.html#cb511-4" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li><code>(1 | experiment_id:treatment)</code> models a random intercept for all combinations of the levels of <span class="math inline">\(\texttt{experiment_id}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>. This <strong>interaction intercept</strong> is an alternative to a random slope for modeling treatment-specific batch effects.</li>
</ol>
</div>
<div id="lmm-demo-aic" class="section level3 hasAnchor" number="16.5.12">
<h3><span class="header-section-number">16.5.12</span> AIC and model selection – which model to report?<a href="lmm.html#lmm-demo-aic" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Three different linear mixed models were fit to the Example 1 data: <code>demo1_m1</code>, <code>demo1_m2</code>, and <code>demo1_m3</code>. Which model do we report? A useful stastistic for this decision is a statistic known as the AIC.</p>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-demo-acc">Table 16.3: </span>AIC of the two alternative models fit to Example 1 data
</caption>
<thead>
<tr>
<th style="text-align:left;">
Model
</th>
<th style="text-align:right;">
AIC
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
demo1_m1
</td>
<td style="text-align:right;">
344.1529
</td>
</tr>
<tr>
<td style="text-align:left;">
demo1_m2
</td>
<td style="text-align:right;">
344.0765
</td>
</tr>
<tr>
<td style="text-align:left;">
demo1_m3
</td>
<td style="text-align:right;">
337.5332
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>AIC (<a href="https://en.wikipedia.org/wiki/Akaike_information_criterion" target="_blank">Akaike Information Criterion</a>) is a relative measure of model quality. Compare this to <span class="math inline">\(R^2\)</span>, which is an absolute measure of goodness of fit. The AIC formula has two parts, one is a kind of goodness of fit (like <span class="math inline">\(R^2\)</span>) and the other is a penalty based on the number of parameters in the model. As the goodness of fit increases, the AIC goes down. As the number of parameters increases, the AIC goes up. The model with the lowest AIC is the highest quality model. The actual value of AIC, unlike <span class="math inline">\(R^2\)</span>, does not have any absolute meaning; it is only meaningful relative to the AICs computed from fits to different models to the same data.</li>
<li>The AIC of the three models suggests that we report Model <code>demo1_m3</code>.</li>
</ol>
</div>
<div id="the-specification-of-random-effects-matters" class="section level3 hasAnchor" number="16.5.13">
<h3><span class="header-section-number">16.5.13</span> The specification of random effects matters<a href="lmm.html#the-specification-of-random-effects-matters" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Inference will often be very different between these two models as they make very different</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-211">Table 16.4: </span>Contrasts from the three linear mixed models fit to the Example 1 data.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>intercepts + slopes (m1)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr1 - Control
</td>
<td style="text-align:right;">
2.54
</td>
<td style="text-align:right;">
0.804
</td>
<td style="text-align:right;">
2.978
</td>
<td style="text-align:right;">
-0.03
</td>
<td style="text-align:right;">
5.11
</td>
<td style="text-align:right;">
3.16
</td>
<td style="text-align:right;">
0.05150
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr2 - Control
</td>
<td style="text-align:right;">
5.61
</td>
<td style="text-align:right;">
1.591
</td>
<td style="text-align:right;">
2.997
</td>
<td style="text-align:right;">
0.54
</td>
<td style="text-align:right;">
10.67
</td>
<td style="text-align:right;">
3.52
</td>
<td style="text-align:right;">
0.03885
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr3 - Control
</td>
<td style="text-align:right;">
2.85
</td>
<td style="text-align:right;">
1.631
</td>
<td style="text-align:right;">
2.997
</td>
<td style="text-align:right;">
-2.35
</td>
<td style="text-align:right;">
8.04
</td>
<td style="text-align:right;">
1.74
</td>
<td style="text-align:right;">
0.17949
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>intercepts only (m2)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr1 - Control
</td>
<td style="text-align:right;">
2.64
</td>
<td style="text-align:right;">
0.755
</td>
<td style="text-align:right;">
66.005
</td>
<td style="text-align:right;">
1.13
</td>
<td style="text-align:right;">
4.15
</td>
<td style="text-align:right;">
3.49
</td>
<td style="text-align:right;">
0.00086
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr2 - Control
</td>
<td style="text-align:right;">
5.65
</td>
<td style="text-align:right;">
0.766
</td>
<td style="text-align:right;">
66.014
</td>
<td style="text-align:right;">
4.12
</td>
<td style="text-align:right;">
7.18
</td>
<td style="text-align:right;">
7.38
</td>
<td style="text-align:right;">
0.00000
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr3 - Control
</td>
<td style="text-align:right;">
2.94
</td>
<td style="text-align:right;">
0.765
</td>
<td style="text-align:right;">
66.000
</td>
<td style="text-align:right;">
1.41
</td>
<td style="text-align:right;">
4.47
</td>
<td style="text-align:right;">
3.85
</td>
<td style="text-align:right;">
0.00027
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>intercept + interaction intercept (m3)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr1 - Control
</td>
<td style="text-align:right;">
2.54
</td>
<td style="text-align:right;">
1.214
</td>
<td style="text-align:right;">
8.912
</td>
<td style="text-align:right;">
-0.21
</td>
<td style="text-align:right;">
5.29
</td>
<td style="text-align:right;">
2.09
</td>
<td style="text-align:right;">
0.06616
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr2 - Control
</td>
<td style="text-align:right;">
5.60
</td>
<td style="text-align:right;">
1.219
</td>
<td style="text-align:right;">
9.061
</td>
<td style="text-align:right;">
2.84
</td>
<td style="text-align:right;">
8.35
</td>
<td style="text-align:right;">
4.59
</td>
<td style="text-align:right;">
0.00128
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr3 - Control
</td>
<td style="text-align:right;">
2.87
</td>
<td style="text-align:right;">
1.219
</td>
<td style="text-align:right;">
9.052
</td>
<td style="text-align:right;">
0.12
</td>
<td style="text-align:right;">
5.63
</td>
<td style="text-align:right;">
2.36
</td>
<td style="text-align:right;">
0.04269
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Inference from the intercept only model (<code>demo1_m2</code>) is almost certainly too optimistic based on simulations that show that intercept only models can be highly anti-conservative (too narrow confidence intervals and too small <em>p</em>-values).</li>
<li>Humans have evolved to make up rational explanations – do not convince yourself that the model with the smallest <em>p</em>-values is the scientifically most rational model.</li>
</ol>
</div>
<div id="mixed-effect-and-repeated-measures-anova" class="section level3 hasAnchor" number="16.5.14">
<h3><span class="header-section-number">16.5.14</span> Mixed Effect and Repeated Measures ANOVA<a href="lmm.html#mixed-effect-and-repeated-measures-anova" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Two-way mixed-effect ANOVA (some fields would call this a repeated measures ANOVA) is equivalent to Model <code>demo1_m3</code> <em>in this case</em>. More generally, the two are equal in balanced designs – the same number of subsamples in all treatment x batch combinations.</p>
<div class="sourceCode" id="cb512"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb512-1"><a href="lmm.html#cb512-1" tabindex="-1"></a>demo1_m4 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(gc <span class="sc">~</span> treatment <span class="sc">+</span> </span>
<span id="cb512-2"><a href="lmm.html#cb512-2" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb512-3"><a href="lmm.html#cb512-3" tabindex="-1"></a>                 <span class="at">data =</span> demo1)</span></code></pre></div>
<p>Notes
1. The function <code>afex::aov4</code> is used for specifying ANOVA models that are special cases of linear mixed models.
2. The model formula in Model <code>demo1_m2</code> looks exactly like that the random intercepts and slopes model (Model <code>demo1_m1</code>) <em>but these are not the same</em>.
3. One can use either a univariate or multivariate model for mixed or repeated measures ANOVA – see section <a href="lmm.html#lmm-exp6g-rmanova-multi">16.6.11</a>.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-213">Table 16.5: </span>Contrasts from the two-way mixed effect ANOVA model demo1_m4.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Tr1 - Control
</td>
<td style="text-align:right;">
2.50
</td>
<td style="text-align:right;">
0.599
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0.60
</td>
<td style="text-align:right;">
4.41
</td>
<td style="text-align:right;">
4.18
</td>
<td style="text-align:right;">
0.02488
</td>
</tr>
<tr>
<td style="text-align:left;">
Tr2 - Control
</td>
<td style="text-align:right;">
5.58
</td>
<td style="text-align:right;">
1.559
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
10.54
</td>
<td style="text-align:right;">
3.58
</td>
<td style="text-align:right;">
0.03734
</td>
</tr>
<tr>
<td style="text-align:left;">
Tr3 - Control
</td>
<td style="text-align:right;">
2.84
</td>
<td style="text-align:right;">
1.586
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
-2.20
</td>
<td style="text-align:right;">
7.89
</td>
<td style="text-align:right;">
1.79
</td>
<td style="text-align:right;">
0.17095
</td>
</tr>
</tbody>
</table>
</div>
<div id="pseudoreplication" class="section level3 hasAnchor" number="16.5.15">
<h3><span class="header-section-number">16.5.15</span> Pseudoreplication<a href="lmm.html#pseudoreplication" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
</div>
<div id="lmm-example2" class="section level2 hasAnchor" number="16.6">
<h2><span class="header-section-number">16.6</span> Example 2 – experiments without subsampling replication (exp6g)<a href="lmm.html#lmm-example2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This example introduces linear mixed models for batches that contain all treatment levels of a single factor but no subsampling replication. In this example, the batch is the individual mouse (<span class="math inline">\(\texttt{mouse_id}\)</span>). There are four measures of the response variable on each mouse, one measure per treatment level. When there is no subsampling replication, we cannot add a random slope to the model because there is only a single observation at each treatment level and a slope would fit the point at the reference level and the point at the non-reference level perfectly. However, we can explicitly model variation in the correlated error and heterogeneity in the variances among treatments as an alternative to modeling a random slope.</p>
<p><a href="https://www.nature.com/articles/s41586-020-2388-4" target="_blank">Reversing a model of Parkinson’s disease with in situ converted nigral neurons</a></p>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/7521455/" target="_blank">Public source</a></p>
<p>Source figure: <a href="https://www.nature.com/articles/s41586-020-2388-4#Fig6" target="_blank">Fig. 6g</a></p>
<p>Source data: <a href="https://www.nature.com/articles/s41586-020-2388-4#MOESM11" target="_blank">Source Data Fig. 6</a></p>
<div id="understand-the-data" class="section level3 hasAnchor" number="16.6.1">
<h3><span class="header-section-number">16.6.1</span> Understand the data<a href="lmm.html#understand-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In this study, the researchers investigated the effectiveness of knocking down the protein <a href="https://en.wikipedia.org/wiki/PTBP1" target="_blank">PTBP1</a> to induce astrocytes to convert to neurons in a motor processing region of the brain. Experimental lesions of this region of the brain is a model of <a href="https://en.wikipedia.org/wiki/Parkinson%27s_disease" target="_blank">Parkinson’s disease</a>. In Experiment 6g, the researchers</p>
<ol style="list-style-type: decimal">
<li>Generated a lesion in the motor processing region using 6-hydroxydopamine (6-OHDA). The lesion disrupts the ability to control the <strong>contralateral</strong> (opposite side) forelimb.</li>
<li>One month after the lesion, measured the percent of ipsilateral (same side) forepaw touches (the forelimb extending out and touching the surface) in a test of exploration in a new environment (the “cylinder test”). The expected percent in an intact mouse is 50%. In a lesioned mouse, the percent should be much greater than 50% since there is less control of the contralateral limb. The measure at this point is in the treatment “Lesion”. This is the positive control.</li>
<li>Converted astrocytes in the lesion to functional neurons by knocking down PTBP1.</li>
<li>Two months after knockdown, gave the mouse saline and remeasured percent ipsilateral touches in a cylinder test. If the knockdown worked as expected, there should be closer to 50% ipsilateral touches. The measure at this point is in the treatment “Saline”. The comparison with Lesion is a focal test.</li>
<li>Inhibited neuron action in the converted neurons using clozapine-N-oxide (CNO), which suppresses neuron electrical activity. Then, remeasured percent ipsilateral touches in a cylinder test. If the CNO worked as expected, there should be much greater than 50% ipsilateral touches since there should be re-loss of control of the contralateral limb. The measure at this point is in the treatment “CNO”. The comparison with Saline is a focal test.</li>
<li>Allowed three days for the CNO to degrade, then, remeasured percent ipsilateral touches in a cylinder test. If the CNO degraded as expected, the converted neurons should be functional and there should be closer to 50% ipsilateral touches. The measure at this point is in the treatment “Post_CNO”. The comparison with CNO is a focal test.</li>
</ol>
<p>The design is <span class="math inline">\(4 \times 1\)</span> – a single treatment with four levels (“Lesion”, “Saline”, “CNO”, “Post_CNO”)</p>
<p>The planned contrasts are</p>
<ol style="list-style-type: decimal">
<li>Saline - Lesion. This measures the effect of the knockdown and conversion of astrocytes to functional neurons.</li>
<li>CNO - Saline. This measures the effect of inhibiting the converted neurons to test if it was these and not some other neurons that account for the effect in contrast 1.</li>
<li>Post_CNO - CNO. This is probing the same expectation as contrast 2.</li>
</ol>
</div>
<div id="model-fit-and-inference" class="section level3 hasAnchor" number="16.6.2">
<h3><span class="header-section-number">16.6.2</span> Model fit and inference<a href="lmm.html#model-fit-and-inference" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="fit-the-model-8" class="section level4 hasAnchor" number="16.6.2.1">
<h4><span class="header-section-number">16.6.2.1</span> Fit the model<a href="lmm.html#fit-the-model-8" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb513"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb513-1"><a href="lmm.html#cb513-1" tabindex="-1"></a>exp6g_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(touch <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>mouse_id), <span class="at">data =</span> exp6g)</span>
<span id="cb513-2"><a href="lmm.html#cb513-2" tabindex="-1"></a></span>
<span id="cb513-3"><a href="lmm.html#cb513-3" tabindex="-1"></a><span class="co"># alt model</span></span>
<span id="cb513-4"><a href="lmm.html#cb513-4" tabindex="-1"></a>exp6g_m1b <span class="ot">&lt;-</span> <span class="fu">lme</span>(touch <span class="sc">~</span> treatment,</span>
<span id="cb513-5"><a href="lmm.html#cb513-5" tabindex="-1"></a>                <span class="at">random =</span> <span class="sc">~</span><span class="dv">1</span><span class="sc">|</span>mouse_id,</span>
<span id="cb513-6"><a href="lmm.html#cb513-6" tabindex="-1"></a>                <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> mouse_id),</span>
<span id="cb513-7"><a href="lmm.html#cb513-7" tabindex="-1"></a>                <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> treatment),</span>
<span id="cb513-8"><a href="lmm.html#cb513-8" tabindex="-1"></a>                <span class="at">data =</span> exp6g)</span>
<span id="cb513-9"><a href="lmm.html#cb513-9" tabindex="-1"></a></span>
<span id="cb513-10"><a href="lmm.html#cb513-10" tabindex="-1"></a><span class="fu">AIC</span>(exp6g_m1a, exp6g_m1b)</span></code></pre></div>
<pre><code>##           df      AIC
## exp6g_m1a  6 191.3367
## exp6g_m1b 15 198.3888</code></pre>
<div class="sourceCode" id="cb515"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb515-1"><a href="lmm.html#cb515-1" tabindex="-1"></a><span class="co"># report model a</span></span>
<span id="cb515-2"><a href="lmm.html#cb515-2" tabindex="-1"></a>exp6g_m1 <span class="ot">&lt;-</span> exp6g_m1a</span></code></pre></div>
<p>exp6g_m1b overparameterizes, report exp6g_m1a (see <a href="lmm.html#lmm-exp6g-alt">Alternative models for exp6g</a> below)</p>
</div>
<div id="inference-from-the-model-4" class="section level4 hasAnchor" number="16.6.2.2">
<h4><span class="header-section-number">16.6.2.2</span> Inference from the model<a href="lmm.html#inference-from-the-model-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb516"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb516-1"><a href="lmm.html#cb516-1" tabindex="-1"></a>exp6g_m1_coef <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">coef</span>(<span class="fu">summary</span>(exp6g_m1)),</span>
<span id="cb516-2"><a href="lmm.html#cb516-2" tabindex="-1"></a>                       <span class="fu">confint</span>(exp6g_m1)[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>),])</span>
<span id="cb516-3"><a href="lmm.html#cb516-3" tabindex="-1"></a></span>
<span id="cb516-4"><a href="lmm.html#cb516-4" tabindex="-1"></a><span class="co"># exp5c_m1_coef %&gt;%</span></span>
<span id="cb516-5"><a href="lmm.html#cb516-5" tabindex="-1"></a><span class="co">#   kable(digits = c(2,3,1,1,4,2,2)) %&gt;%</span></span>
<span id="cb516-6"><a href="lmm.html#cb516-6" tabindex="-1"></a><span class="co">#   kable_styling()</span></span></code></pre></div>
<div class="sourceCode" id="cb517"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb517-1"><a href="lmm.html#cb517-1" tabindex="-1"></a>exp6g_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp6g_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>))</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:right;">
emmean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Lesion
</td>
<td style="text-align:right;">
83.3
</td>
<td style="text-align:right;">
3.44
</td>
<td style="text-align:right;">
18.9
</td>
<td style="text-align:right;">
76.1
</td>
<td style="text-align:right;">
90.47
</td>
</tr>
<tr>
<td style="text-align:left;">
Saline
</td>
<td style="text-align:right;">
62.8
</td>
<td style="text-align:right;">
3.44
</td>
<td style="text-align:right;">
18.9
</td>
<td style="text-align:right;">
55.5
</td>
<td style="text-align:right;">
69.96
</td>
</tr>
<tr>
<td style="text-align:left;">
CNO
</td>
<td style="text-align:right;">
81.4
</td>
<td style="text-align:right;">
3.44
</td>
<td style="text-align:right;">
18.9
</td>
<td style="text-align:right;">
74.2
</td>
<td style="text-align:right;">
88.63
</td>
</tr>
<tr>
<td style="text-align:left;">
Post_CNO
</td>
<td style="text-align:right;">
58.8
</td>
<td style="text-align:right;">
3.44
</td>
<td style="text-align:right;">
18.9
</td>
<td style="text-align:right;">
51.6
</td>
<td style="text-align:right;">
66.00
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb518"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb518-1"><a href="lmm.html#cb518-1" tabindex="-1"></a><span class="co"># exp6g_m1_emm # print in console to get row numbers</span></span>
<span id="cb518-2"><a href="lmm.html#cb518-2" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb518-3"><a href="lmm.html#cb518-3" tabindex="-1"></a>lesion <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb518-4"><a href="lmm.html#cb518-4" tabindex="-1"></a>saline <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb518-5"><a href="lmm.html#cb518-5" tabindex="-1"></a>cno <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb518-6"><a href="lmm.html#cb518-6" tabindex="-1"></a>post_cno <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb518-7"><a href="lmm.html#cb518-7" tabindex="-1"></a></span>
<span id="cb518-8"><a href="lmm.html#cb518-8" tabindex="-1"></a>exp6g_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp6g_m1_emm,</span>
<span id="cb518-9"><a href="lmm.html#cb518-9" tabindex="-1"></a>                       <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb518-10"><a href="lmm.html#cb518-10" tabindex="-1"></a>                         <span class="st">&quot;Saline - Lesion&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(saline <span class="sc">-</span> lesion),</span>
<span id="cb518-11"><a href="lmm.html#cb518-11" tabindex="-1"></a>                         <span class="st">&quot;CNO - Saline&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(cno <span class="sc">-</span> saline),</span>
<span id="cb518-12"><a href="lmm.html#cb518-12" tabindex="-1"></a>                         <span class="st">&quot;Post_CNO - CNO&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(post_cno <span class="sc">-</span> cno)</span>
<span id="cb518-13"><a href="lmm.html#cb518-13" tabindex="-1"></a>                       ),</span>
<span id="cb518-14"><a href="lmm.html#cb518-14" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb518-15"><a href="lmm.html#cb518-15" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb518-16"><a href="lmm.html#cb518-16" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Saline - Lesion
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-29.07
</td>
<td style="text-align:right;">
-11.96
</td>
<td style="text-align:right;">
-5.04
</td>
<td style="text-align:right;">
0.00009
</td>
</tr>
<tr>
<td style="text-align:left;">
CNO - Saline
</td>
<td style="text-align:right;">
18.67
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
10.12
</td>
<td style="text-align:right;">
27.23
</td>
<td style="text-align:right;">
4.59
</td>
<td style="text-align:right;">
0.00023
</td>
</tr>
<tr>
<td style="text-align:left;">
Post_CNO - CNO
</td>
<td style="text-align:right;">
-22.63
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-31.18
</td>
<td style="text-align:right;">
-14.08
</td>
<td style="text-align:right;">
-5.56
</td>
<td style="text-align:right;">
0.00003
</td>
</tr>
</tbody>
</table>
</div>
<div id="lmm-exp6g-plotthemodel" class="section level4 hasAnchor" number="16.6.2.3">
<h4><span class="header-section-number">16.6.2.3</span> Plot the model<a href="lmm.html#lmm-exp6g-plotthemodel" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-217"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/unnamed-chunk-217-1.png" alt="Treatment effect on ipsilateral touch, as percent of all touches. Gray dots connected by lines are individual mice." width="576" />
<p class="caption">
Figure 16.7: Treatment effect on ipsilateral touch, as percent of all touches. Gray dots connected by lines are individual mice.
</p>
</div>
</div>
<div id="alternaplot-the-model-1" class="section level4 hasAnchor" number="16.6.2.4">
<h4><span class="header-section-number">16.6.2.4</span> Alternaplot the model<a href="lmm.html#alternaplot-the-model-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-218"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/unnamed-chunk-218-1.png" alt="Ipsilateral touch (percent of all touches) response to different treatments. Gray dots connected by lines are individual mice." width="576" />
<p class="caption">
Figure 16.8: Ipsilateral touch (percent of all touches) response to different treatments. Gray dots connected by lines are individual mice.
</p>
</div>
</div>
</div>
<div id="the-model-exp6g_m1-adds-a-random-intercept-but-not-a-random-slope" class="section level3 hasAnchor" number="16.6.3">
<h3><span class="header-section-number">16.6.3</span> The model exp6g_m1 adds a random intercept but not a random slope<a href="lmm.html#the-model-exp6g_m1-adds-a-random-intercept-but-not-a-random-slope" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The model fit to the exp6g data is</p>
<p><span class="math display">\[
\begin{equation}
\texttt{touch}_{jk} = (\beta_{0} + \gamma_{0j}) + (\beta_{k} + \gamma_{jk}) \texttt{treatment}_{k} + \varepsilon
\end{equation}
\]</span></p>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Again, in experiments without subsampling, we cannot add a random slope to the model (for each mouse, there is a single observation at each treatment level so a slope would fit the two points perfectly).</li>
</ol>
</div>
<div id="the-fixed-effect-coefficients-of-model-exp6g_m1" class="section level3 hasAnchor" number="16.6.4">
<h3><span class="header-section-number">16.6.4</span> The fixed effect coefficients of model exp6g_m1<a href="lmm.html#the-fixed-effect-coefficients-of-model-exp6g_m1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The fixed effect coefficients of model <code>exp6g_m1</code> are</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
83.27
</td>
<td style="text-align:right;">
3.441
</td>
<td style="text-align:right;">
18.9
</td>
<td style="text-align:right;">
24.20
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
76.72
</td>
<td style="text-align:right;">
89.81
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentSaline
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18.0
</td>
<td style="text-align:right;">
-5.04
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
-28.25
</td>
<td style="text-align:right;">
-12.78
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentCNO
</td>
<td style="text-align:right;">
-1.84
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18.0
</td>
<td style="text-align:right;">
-0.45
</td>
<td style="text-align:right;">
0.6565
</td>
<td style="text-align:right;">
-9.58
</td>
<td style="text-align:right;">
5.90
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentPost_CNO
</td>
<td style="text-align:right;">
-24.47
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18.0
</td>
<td style="text-align:right;">
-6.01
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
-32.21
</td>
<td style="text-align:right;">
-16.73
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The interpretation of the fixed effectcs coefficients have the usual interpretation (see <a href="oneway.html#oneway-what-coefs-are">The coefficients of a linear model using dummy coding have a useful interpretation</a>). Figure <a href="lmm.html#fig:lmm-exp6g-explainer-1">16.9</a> is a reminder.</li>
</ol>
<div class="figure"><span style="display:block;" id="fig:lmm-exp6g-explainer-1"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp6g-explainer-1-1.png" alt="Fixed effects estimated by exp6g_m1. $b_0$ is the modeled mean of the Lesian treatment. $b_1$ is the difference (Saline - Lesian). $b_2$ is the difference (CNO - Lesian). $b_3$ is the difference (Post_CNO - Lesian)." width="672" />
<p class="caption">
Figure 16.9: Fixed effects estimated by exp6g_m1. <span class="math inline">\(b_0\)</span> is the modeled mean of the Lesian treatment. <span class="math inline">\(b_1\)</span> is the difference (Saline - Lesian). <span class="math inline">\(b_2\)</span> is the difference (CNO - Lesian). <span class="math inline">\(b_3\)</span> is the difference (Post_CNO - Lesian).
</p>
</div>
</div>
<div id="the-random-intercept-coefficients-of-exp6g_m1" class="section level3 hasAnchor" number="16.6.5">
<h3><span class="header-section-number">16.6.5</span> The random intercept coefficients of exp6g_m1<a href="lmm.html#the-random-intercept-coefficients-of-exp6g_m1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp6g-m1-random-coef">Table 16.6: </span>Random intercept coefficients (g_0j) for exp6g_m1. For each mouse, the random intercept is the sum of the fixed intercept (<span class="math inline">\(b_0\)</span>) and the random intercept effect <span class="math inline">\(g_{0j}\)</span>.
</caption>
<thead>
<tr>
<th style="text-align:left;">
mouse_id
</th>
<th style="text-align:right;">
random intercept
</th>
<th style="text-align:right;">
b_0
</th>
<th style="text-align:right;">
g_0j
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
mouse_1
</td>
<td style="text-align:right;">
82.874
</td>
<td style="text-align:right;">
83.268
</td>
<td style="text-align:right;">
-0.393
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_2
</td>
<td style="text-align:right;">
83.640
</td>
<td style="text-align:right;">
83.268
</td>
<td style="text-align:right;">
0.372
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_3
</td>
<td style="text-align:right;">
77.499
</td>
<td style="text-align:right;">
83.268
</td>
<td style="text-align:right;">
-5.769
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_4
</td>
<td style="text-align:right;">
79.943
</td>
<td style="text-align:right;">
83.268
</td>
<td style="text-align:right;">
-3.325
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_5
</td>
<td style="text-align:right;">
89.286
</td>
<td style="text-align:right;">
83.268
</td>
<td style="text-align:right;">
6.018
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_6
</td>
<td style="text-align:right;">
82.732
</td>
<td style="text-align:right;">
83.268
</td>
<td style="text-align:right;">
-0.536
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_7
</td>
<td style="text-align:right;">
86.901
</td>
<td style="text-align:right;">
83.268
</td>
<td style="text-align:right;">
3.633
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The random intercept effect <span class="math inline">\(g_{0j}\)</span> is the difference between the modeled mean for mouse <em>j</em> and the mean of the reference treatment. Unlike the fixed intercept (<span class="math inline">\(b_0\)</span>), <span class="math inline">\(g_{0j}\)</span> is an effect.</li>
<li>Figure <a href="#fig:lmm-exp1g-explainer-2"><strong>??</strong></a> illustrates the random intercept coefficients <span class="math inline">\(g_{0j}\)</span> for model <code>exp6g_m1</code>.</li>
<li>The random intercept coefficients are often (but not always) treated as a source of nuisance variation – that is, the coefficients are not generally of interest and the values are not typically reported.</li>
</ol>
<div class="figure"><span style="display:block;" id="fig:lmm-exp6g-explainer-2"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp6g-explainer-2-1.png" alt="Random intercept effects for model exp6g_m1. The pale, colored dots are the measured percent ipsilateral touch values for each mouse for each treatment. The dashed, gray lines are the modeled means for each treatment. The dashed grey line for the reference level (&quot;Lesion&quot;) is the fixed intercept. The dark, colored dots at the reference level are the random intercepts. The value of each random intercept is the sum of the fixed intercept and the random intercept effect for that mouse. The vertical, colored line segments at the reference level are the random intercept effects $g_{0j}$. The length of the segment is the residual from the dashed, gray line to the pale dot. The random intercept effects for mice 1 and 2 are too short to see." width="672" />
<p class="caption">
Figure 16.10: Random intercept effects for model exp6g_m1. The pale, colored dots are the measured percent ipsilateral touch values for each mouse for each treatment. The dashed, gray lines are the modeled means for each treatment. The dashed grey line for the reference level (“Lesion”) is the fixed intercept. The dark, colored dots at the reference level are the random intercepts. The value of each random intercept is the sum of the fixed intercept and the random intercept effect for that mouse. The vertical, colored line segments at the reference level are the random intercept effects <span class="math inline">\(g_{0j}\)</span>. The length of the segment is the residual from the dashed, gray line to the pale dot. The random intercept effects for mice 1 and 2 are too short to see.
</p>
</div>
</div>
<div id="the-random-and-residual-variance-and-the-intraclass-correlation-of-model-exp6g_m1" class="section level3 hasAnchor" number="16.6.6">
<h3><span class="header-section-number">16.6.6</span> The random and residual variance and the intraclass correlation of model exp6g_m1<a href="lmm.html#the-random-and-residual-variance-and-the-intraclass-correlation-of-model-exp6g_m1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre><code>##  Groups   Name        Std.Dev.
##  mouse_id (Intercept) 4.9882  
##  Residual             7.6155</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The first element in “Std.Dev.” is the estimate of <span class="math inline">\(\sqrt{\sigma^2_{0j}}\)</span>, the standard deviation among the donors due to the random effect of <span class="math inline">\(\texttt{mouse_id}\)</span>.</li>
<li>The second element in “Std.Dev.” is <span class="math inline">\(\sqrt{\sigma^2_{0}}\)</span>, the estimate of the standard deviation of the error variance (<span class="math inline">\(\varepsilon^2\)</span>). Remember that the residuals of the model are the estimates of <span class="math inline">\(\varepsilon\)</span>.</li>
<li>The ratio <span class="math inline">\(\frac{\sigma^2_{0j}}{\sigma^2_{0j} + \sigma^2_{0}}\)</span> (the ratio of the among-block variance to total random variance) is known as the <a href="https://en.wikipedia.org/wiki/Intraclass_correlation" target="_blank">intraclass correlation</a>. This correlation is an estimate of the correlated error due to the by-mouse clustering if the model were fit without the added random intercept. For <code>exp6g_m1</code>, this correlation is 0.3.</li>
<li>The intraclass correlation ranges between 0 and 1 and makes a pretty good qualitative indicator of repeatability.</li>
</ol>
</div>
<div id="the-linear-mixed-model-exp6g_m1-increases-precision-of-treatment-effects-relative-to-a-fixed-effects-model" class="section level3 hasAnchor" number="16.6.7">
<h3><span class="header-section-number">16.6.7</span> The linear mixed model exp6g_m1 increases precision of treatment effects, relative to a fixed effects model<a href="lmm.html#the-linear-mixed-model-exp6g_m1-increases-precision-of-treatment-effects-relative-to-a-fixed-effects-model" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s compare the effects estimated by the linear mixed model <code>exp6g_m1</code> with a linear model that ignores donor (a <strong>fixed effects</strong> model).</p>
<div class="sourceCode" id="cb520"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb520-1"><a href="lmm.html#cb520-1" tabindex="-1"></a>exp6g_m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(touch <span class="sc">~</span> treatment, <span class="at">data =</span> exp6g)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:lmm-exp6g-why2"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp6g-why2-1.png" alt="A. Inference from a linear mixed model with blocking factor (mouse_id) added as a random intercept. B. Inference from a fixed effects model." width="576" />
<p class="caption">
Figure 16.11: A. Inference from a linear mixed model with blocking factor (mouse_id) added as a random intercept. B. Inference from a fixed effects model.
</p>
</div>
<p>Figure <a href="lmm.html#fig:lmm-exp6g-why2">16.11</a>A is a plot of the effects from the linear mixed model <code>exp6g_m1</code> that models the added variance due to mouse. Figure <a href="lmm.html#fig:lmm-exp6g-why2">16.11</a>B is a plot of the effects from the fixed effect model <code>exp6g_m2</code> that ignores the added variance due to mouse. The 95% confidence intervals of the treatment effects in model <code>exp6g_m1</code> are slightly smaller than those in model <code>exp6g_m2</code>. Adding <span class="math inline">\(\texttt{mouse_id}\)</span> as a random factor to the linear model increases the precision of the estimate of the treatment effects by eliminating the among-mouse component of variance from the error variance.</p>
<p>The error variance (the estimate of $^2) in the linear mixed model and the fixed effects model is</p>
<div class="sourceCode" id="cb521"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb521-1"><a href="lmm.html#cb521-1" tabindex="-1"></a><span class="fu">summary</span>(exp6g_m1)<span class="sc">$</span>sigma<span class="sc">^</span><span class="dv">2</span></span></code></pre></div>
<pre><code>## [1] 57.99549</code></pre>
<div class="sourceCode" id="cb523"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb523-1"><a href="lmm.html#cb523-1" tabindex="-1"></a><span class="fu">summary</span>(exp6g_m2)<span class="sc">$</span>sigma<span class="sc">^</span><span class="dv">2</span></span></code></pre></div>
<pre><code>## [1] 82.8777</code></pre>
<p>The error variance is 43% higher in the fixed effects model. In the linear mixed model, the variance lost from the error was shifted to the random intercept. We can track this shift with the table of the two variance components of the linear mixed model, shown above, and here again (the values are the square roots of the variances).</p>
<div class="sourceCode" id="cb525"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb525-1"><a href="lmm.html#cb525-1" tabindex="-1"></a><span class="fu">VarCorr</span>(exp6g_m1)</span></code></pre></div>
<pre><code>##  Groups   Name        Std.Dev.
##  mouse_id (Intercept) 4.9882  
##  Residual             7.6155</code></pre>
<p>The sum of the two variance components of the linear mixed model is equal to the error variance of the fixed effects model:</p>
<div class="sourceCode" id="cb527"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb527-1"><a href="lmm.html#cb527-1" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">as.data.frame</span>(<span class="fu">VarCorr</span>(exp6g_m1))<span class="sc">$</span>vcov)</span></code></pre></div>
<pre><code>## [1] 82.8777</code></pre>
</div>
<div id="lmm-exp6g-alt" class="section level3 hasAnchor" number="16.6.8">
<h3><span class="header-section-number">16.6.8</span> Alternative models for exp6g<a href="lmm.html#lmm-exp6g-alt" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Linear mixed models are very flexible, a topic which is too advanced for this text. Here I want to focus on an alternative model because of its relevance to <strong>repeated measures ANOVA</strong>.</p>
<div class="sourceCode" id="cb529"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb529-1"><a href="lmm.html#cb529-1" tabindex="-1"></a>exp6g_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(touch <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> mouse_id),</span>
<span id="cb529-2"><a href="lmm.html#cb529-2" tabindex="-1"></a>                 <span class="at">data =</span> exp6g)</span>
<span id="cb529-3"><a href="lmm.html#cb529-3" tabindex="-1"></a>exp6g_m1b <span class="ot">&lt;-</span> <span class="fu">lme</span>(touch <span class="sc">~</span> treatment,</span>
<span id="cb529-4"><a href="lmm.html#cb529-4" tabindex="-1"></a>                <span class="at">random =</span> <span class="sc">~</span><span class="dv">1</span><span class="sc">|</span>mouse_id,</span>
<span id="cb529-5"><a href="lmm.html#cb529-5" tabindex="-1"></a>                <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> mouse_id),</span>
<span id="cb529-6"><a href="lmm.html#cb529-6" tabindex="-1"></a>                <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> treatment),</span>
<span id="cb529-7"><a href="lmm.html#cb529-7" tabindex="-1"></a>                <span class="at">data =</span> exp6g)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Two models are fit. The two models have the same fixed effect but differ in how they model the random effect and the pattern of correlations in the residuals. The difference in specification determines the error variance and degrees of freedom for computing uncertainty. I’ll return to inference and the problem of “which model to choose” in a moment. First, how do these models differ?</li>
<li>Model <code>exp6g_m1a</code> is the same model analyzed and explained above. This model specifies a random intercept for each level of <span class="math inline">\(\texttt{mouse_id}\)</span>.</li>
<li>Model <code>exp6g_m1b</code> has the same random effects as <code>exp6g_m1a</code> but adds two additional arguments, a <code>correlation</code> argument that explicitly models correlated error in the residuals and a <code>weights</code> argument that models heterogeneity in the residuals.</li>
<li>Consider the error (the residuals) of the fixed effect model <code>touch ~ treatment</code> cast into a matrix with the residuals for each treatment in its own column. This matrix of residuals will be a <span class="math inline">\(7 \times 4\)</span> (7 mice, 4 treatments) matrix that looks like the table of residuals in Example 1 (Table <a href="lmm.html#tab:lmm-demo1-cor-error-table">16.1</a>).</li>
<li>Model <code>exp6g_m1a</code> implicitly models <strong>compound symmetric</strong> correlated error. This means</li>
</ol>
<ul>
<li>the correlation between every pair of columns of the residual matrix is the same.</li>
<li>the variances of each of the columns of the residual matrix is the same.</li>
<li>Model <code>exp6g_m1a</code> assumes zero correlation among the columns of a residual matrix <em>from Model exp6g_m1a</em>. This is because the source of the correlated residuals from the fixed model has been modeled by the random intercept.</li>
</ul>
<ol start="5" style="list-style-type: decimal">
<li>Model <code>exp6g_m1b</code> does not assume <strong>compound symmetric</strong> correlated error.</li>
</ol>
<ul>
<li>the argument <code>correlation = corSymm(form = ~ 1 | mouse_id)</code> in model <code>exp6g_m1b</code> explicitly models different correlations for all pairs of treatment combinations – that is, <strong>unstructured</strong> correlated error.</li>
<li>the argument <code>weights = varIdent(form = ~ 1 | treatment)</code> in model <code>exp6g_m1b</code> explicitly models heterogeneity in the residuals among the levels of $.</li>
</ul>
<ol start="6" style="list-style-type: decimal">
<li>Model <code>exp6g_m1b</code> makes fewer assumptions for inference. The trade-off is the estimation of more parameters and the potential of <a href="https://en.wikipedia.org/wiki/Overfitting" target="_blank">overfitting</a>.</li>
<li>A univariate model of a <strong>repeated measures ANOVA</strong> fit to the <code>exp6g</code> data is equivalent to model <code>exp6g_m1a</code>. See section <a href="lmm.html#lmm-exp6g-rmanova">16.6.10</a>.</li>
<li>A multivariate model of a <strong>repeated measures ANOVA</strong> fit to the <code>exp6g</code> data is equivalent to model <code>exp6g_m1b</code>. See section <a href="lmm.html#lmm-exp6g-rmanova-multi">16.6.11</a>.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp6g-alternatives-planned">Table 16.7: </span>Planned contrasts from the two alternative models.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp6g_m1a</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Saline - Lesion
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-29.07
</td>
<td style="text-align:right;">
-11.96
</td>
<td style="text-align:right;">
-5.04
</td>
<td style="text-align:right;">
0.000085
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
CNO - Saline
</td>
<td style="text-align:right;">
18.67
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
10.12
</td>
<td style="text-align:right;">
27.23
</td>
<td style="text-align:right;">
4.59
</td>
<td style="text-align:right;">
0.000228
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Post_CNO - CNO
</td>
<td style="text-align:right;">
-22.63
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-31.18
</td>
<td style="text-align:right;">
-14.08
</td>
<td style="text-align:right;">
-5.56
</td>
<td style="text-align:right;">
0.000028
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp6g_m1b</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Saline - Lesion
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
3.172
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-27.18
</td>
<td style="text-align:right;">
-13.85
</td>
<td style="text-align:right;">
-6.47
</td>
<td style="text-align:right;">
0.000004
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
CNO - Saline
</td>
<td style="text-align:right;">
18.67
</td>
<td style="text-align:right;">
3.057
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
12.25
</td>
<td style="text-align:right;">
25.10
</td>
<td style="text-align:right;">
6.11
</td>
<td style="text-align:right;">
0.000009
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Post_CNO - CNO
</td>
<td style="text-align:right;">
-22.63
</td>
<td style="text-align:right;">
5.610
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-34.42
</td>
<td style="text-align:right;">
-10.84
</td>
<td style="text-align:right;">
-4.03
</td>
<td style="text-align:right;">
0.000779
</td>
</tr>
</tbody>
</table>
<ol start="9" style="list-style-type: decimal">
<li>Planned comparisons of the two models are given in the tables above. The effect estimates are the same but inference differs quantitatively among the models (but not qualitatively) because of how each model models the error. Which model do we report? One way to evaluate the models is a statistic known as the AIC.</li>
</ol>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp6-aic">Table 16.8: </span>AIC of the two alternative models fit to exp6g data
</caption>
<thead>
<tr>
<th style="text-align:left;">
Model
</th>
<th style="text-align:right;">
AIC
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
exp6g_m1a
</td>
<td style="text-align:right;">
191.3367
</td>
</tr>
<tr>
<td style="text-align:left;">
exp6g_m1b
</td>
<td style="text-align:right;">
198.3888
</td>
</tr>
</tbody>
</table>
<ol start="10" style="list-style-type: decimal">
<li>AIC (<a href="https://en.wikipedia.org/wiki/Akaike_information_criterion" target="_blank">Akaike Information Criterion</a>) is a relative measure of model quality. Compare this to <span class="math inline">\(R^2\)</span>, which is an absolute measure of goodness of fit. The AIC formula has two parts, one is a kind of goodness of fit (like <span class="math inline">\(R^2\)</span>) and the other is a penalty based on the number of parameters in the model. As the goodness of fit increases, the AIC goes down. As the number of parameters increases, the AIC goes up. The model with the lowest AIC is the highest quality model. The actual number, however does not have any absolute meaning; it is only meaningful relative to the AICs computed from fits to different models using the same data.</li>
<li>The AICs of models <code>exp6g_m1a</code> and models <code>exp6g_m1b</code> suggest that model <code>exp6g_m1b</code> is too complex given the data. This has relevance for the repeated measures ANOVA analysis in the next section.</li>
</ol>
</div>
<div id="paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models" class="section level3 hasAnchor" number="16.6.9">
<h3><span class="header-section-number">16.6.9</span> Paired t-tests and repeated measures ANOVA are special cases of linear mixed models<a href="lmm.html#paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A paired <em>t</em>-test is a special case of linear mixed model fit to data from a randomized complete block design with no subsampling and only two treatment levels (see <a href="violations.html#oneway-paired-t">Lack of independence</a> in the Violations chapter). A repeated measures ANOVA is a special case of linear mixed model fit to data from a randomized complete block design with no subsampling and more than two treatment levels (see <a href="violations.html#oneway-paired-t">Lack of independence</a> in the Violations chapter).</p>
<p>Experiment 6g is a randomized complete block design with no subsampling and four treatments. A “which test” key in a traditional, experimental statistics textbook would guide a researcher to analyze these data using <a href="https://en.wikipedia.org/wiki/Repeated_measures_design#Repeated_measures_ANOVA" target="_blank">repeated measures ANOVA</a>. There are two methods of repeated measures anova, the univariate model and the multivariate model.</p>
</div>
<div id="lmm-exp6g-rmanova" class="section level3 hasAnchor" number="16.6.10">
<h3><span class="header-section-number">16.6.10</span> Classical (“univariate model”) repeated measures ANOVA of exp6g<a href="lmm.html#lmm-exp6g-rmanova" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb530"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb530-1"><a href="lmm.html#cb530-1" tabindex="-1"></a>exp6g_aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(touch <span class="sc">~</span> treatment <span class="sc">+</span></span>
<span id="cb530-2"><a href="lmm.html#cb530-2" tabindex="-1"></a>                    (treatment <span class="sc">|</span> mouse_id),</span>
<span id="cb530-3"><a href="lmm.html#cb530-3" tabindex="-1"></a>                  <span class="at">data =</span> exp6g)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp6g-uni-rmanova-planned">Table 16.9: </span>Planned contrasts for the linear mixed model exp6g_m1a and the univariate model of the repeated measures ANOVA.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp6g_m1a</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Saline - Lesion
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-29.07
</td>
<td style="text-align:right;">
-11.96
</td>
<td style="text-align:right;">
-5.04
</td>
<td style="text-align:right;">
0.000085
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
CNO - Saline
</td>
<td style="text-align:right;">
18.67
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
10.12
</td>
<td style="text-align:right;">
27.23
</td>
<td style="text-align:right;">
4.59
</td>
<td style="text-align:right;">
0.000228
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Post_CNO - CNO
</td>
<td style="text-align:right;">
-22.63
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-31.18
</td>
<td style="text-align:right;">
-14.08
</td>
<td style="text-align:right;">
-5.56
</td>
<td style="text-align:right;">
0.000028
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>univariate RM-ANOVA</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Saline - Lesion
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
3.172
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
-28.28
</td>
<td style="text-align:right;">
-12.75
</td>
<td style="text-align:right;">
-6.47
</td>
<td style="text-align:right;">
0.000649
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
CNO - Saline
</td>
<td style="text-align:right;">
18.67
</td>
<td style="text-align:right;">
3.057
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
11.19
</td>
<td style="text-align:right;">
26.15
</td>
<td style="text-align:right;">
6.11
</td>
<td style="text-align:right;">
0.000877
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Post_CNO - CNO
</td>
<td style="text-align:right;">
-22.63
</td>
<td style="text-align:right;">
5.610
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
-36.36
</td>
<td style="text-align:right;">
-8.90
</td>
<td style="text-align:right;">
-4.03
</td>
<td style="text-align:right;">
0.006853
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>In addition to the assumptions of the linear model outlined in the Violations chapter, the classical (“univariate model”) repeated measures ANOVA assumes <a href="https://en.wikipedia.org/wiki/Mauchly%27s_sphericity_test#Sphericity" target="_blank">sphericity</a>, which is the equality of the variances of <em>all pairwise differences</em> among treatment combinations.</li>
<li>The univariate model of the classic RM ANOVA is equivalent to Model <code>exp6g_m1a</code>.</li>
</ol>
</div>
<div id="lmm-exp6g-rmanova-multi" class="section level3 hasAnchor" number="16.6.11">
<h3><span class="header-section-number">16.6.11</span> “Multivariate model” repeated measures ANOVA<a href="lmm.html#lmm-exp6g-rmanova-multi" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp6g-multi-rmanova">Table 16.10: </span>Planned contrasts for the linear mixed model exp6g_m1b and the multivariate model of the repeated measures ANOVA.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp6g_m1b</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Saline - Lesion
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
3.172
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-27.18
</td>
<td style="text-align:right;">
-13.85
</td>
<td style="text-align:right;">
-6.47
</td>
<td style="text-align:right;">
0.000004
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
CNO - Saline
</td>
<td style="text-align:right;">
18.67
</td>
<td style="text-align:right;">
3.057
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
12.25
</td>
<td style="text-align:right;">
25.10
</td>
<td style="text-align:right;">
6.11
</td>
<td style="text-align:right;">
0.000009
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Post_CNO - CNO
</td>
<td style="text-align:right;">
-22.63
</td>
<td style="text-align:right;">
5.610
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
-34.42
</td>
<td style="text-align:right;">
-10.84
</td>
<td style="text-align:right;">
-4.03
</td>
<td style="text-align:right;">
0.000779
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>multivariate RM-ANOVA</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Saline - Lesion
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
3.172
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
-28.28
</td>
<td style="text-align:right;">
-12.75
</td>
<td style="text-align:right;">
-6.47
</td>
<td style="text-align:right;">
0.000649
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
CNO - Saline
</td>
<td style="text-align:right;">
18.67
</td>
<td style="text-align:right;">
3.057
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
11.19
</td>
<td style="text-align:right;">
26.15
</td>
<td style="text-align:right;">
6.11
</td>
<td style="text-align:right;">
0.000877
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Post_CNO - CNO
</td>
<td style="text-align:right;">
-22.63
</td>
<td style="text-align:right;">
5.610
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
-36.36
</td>
<td style="text-align:right;">
-8.90
</td>
<td style="text-align:right;">
-4.03
</td>
<td style="text-align:right;">
0.006853
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The sphericity assumption of classical repeated measures ANOVA is relaxed if the model is fit using the “multivariate model”.</li>
<li>The multivariate model repeated measures ANOVA is a linear model with a multivariate response and not a linear mixed model. In the multivariate model, each treatment combination is a different response variable and there is a single row for each level of the random factor (<span class="math inline">\(\texttt{mouse_id}\)</span> in Experiment 6g). The multivariate model is a different way of handling the correlated error that occurs when conceiving of the design as univariate.</li>
<li>The multivariate repeated measures ANOVA is equivalent to the linear mixed model <code>exp6g_m1b</code>. The SEs of the contrasts are the same but the degrees of freedom differ. Because of the increased df of the linear mixed model, the CIs are narrower and the <em>p</em>-value is smaller – that is the linear mixed model is less conservative than the repeated measures ANOVA. There is no correct degrees of freedom for a linear mixed model like this and emmeans outputs one way to compute these (with options for others, none of which are equivalent to those from the multivariate model RM-ANOVA).</li>
</ol>
</div>
<div id="linear-mixed-models-vs-repeated-measures-anova" class="section level3 hasAnchor" number="16.6.12">
<h3><span class="header-section-number">16.6.12</span> Linear mixed models vs repeated measures ANOVA<a href="lmm.html#linear-mixed-models-vs-repeated-measures-anova" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Many modern textbooks encourage researchers to use linear mixed models instead of repeated measures ANOVA for randomized complete block designs (with or without subsampling) because</p>
<ol style="list-style-type: decimal">
<li>linear mixed models do not exclude random units (subject/mouse/donor/cage) with missing measures of one of the treatment combinations. <a href="violations.html#violations-rmanova">Example 2 (diHOME exp2a) – A repeated measures ANOVA is a special case of a linear mixed model</a> is an example of this kind of missing data.</li>
<li>if there is subsampling, linear mixed models do not <strong>aggregate</strong> the random-unit data, that is, linear mixed models do not simply compute the means and ignore the variance of the sample within each random unit.</li>
<li>linear mixed models allow for modeling additional sources of correlated error, that is, an experiment may have two or more random factor variables (for example, donor and experiment).</li>
<li>linear mixed models allow a researcher to model different patterns of correlated error. This is especially important in longitudinal experiments.</li>
<li>linear mixed models can be generalized to model sampling from non-normal distributions – these are generalized linear mixed models.</li>
</ol>
</div>
<div id="modeling-textttmouse_id-as-a-fixed-effect" class="section level3 hasAnchor" number="16.6.13">
<h3><span class="header-section-number">16.6.13</span> Modeling <span class="math inline">\(\texttt{mouse_id}\)</span> as a fixed effect<a href="lmm.html#modeling-textttmouse_id-as-a-fixed-effect" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><span class="math display" id="eq:lmm-exp6g-m3">\[
\begin{equation}
\texttt{touch} \sim \texttt{treatment + mouse_id}
\end{equation}
\tag{16.3}
\]</span></p>
<p>Model <a href="lmm.html#eq:lmm-exp6g-m3">(16.3)</a> (using R formula syntax) is a linear model with the block <span class="math inline">\(\texttt{mouse_id}\)</span> added as a fixed covariate instead of a random intercept. The coefficients of the fit model are</p>
<div class="sourceCode" id="cb531"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb531-1"><a href="lmm.html#cb531-1" tabindex="-1"></a>exp6g_m3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(touch <span class="sc">~</span> treatment <span class="sc">+</span> mouse_id,</span>
<span id="cb531-2"><a href="lmm.html#cb531-2" tabindex="-1"></a>               <span class="at">data =</span> exp6g)</span>
<span id="cb531-3"><a href="lmm.html#cb531-3" tabindex="-1"></a></span>
<span id="cb531-4"><a href="lmm.html#cb531-4" tabindex="-1"></a>exp6g_m3_coef <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">coef</span>(<span class="fu">summary</span>(exp6g_m3)),</span>
<span id="cb531-5"><a href="lmm.html#cb531-5" tabindex="-1"></a>                       <span class="fu">confint</span>(exp6g_m3))</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
82.65
</td>
<td style="text-align:right;">
4.551
</td>
<td style="text-align:right;">
18.2
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
73.08
</td>
<td style="text-align:right;">
92.21
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentSaline
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
-5.0
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
-29.07
</td>
<td style="text-align:right;">
-11.96
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentCNO
</td>
<td style="text-align:right;">
-1.84
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
-0.5
</td>
<td style="text-align:right;">
0.6565
</td>
<td style="text-align:right;">
-10.39
</td>
<td style="text-align:right;">
6.71
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentPost_CNO
</td>
<td style="text-align:right;">
-24.47
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
-6.0
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
-33.02
</td>
<td style="text-align:right;">
-15.92
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_idmouse_2
</td>
<td style="text-align:right;">
1.21
</td>
<td style="text-align:right;">
5.385
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:right;">
0.8245
</td>
<td style="text-align:right;">
-10.10
</td>
<td style="text-align:right;">
12.52
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_idmouse_3
</td>
<td style="text-align:right;">
-8.51
</td>
<td style="text-align:right;">
5.385
</td>
<td style="text-align:right;">
-1.6
</td>
<td style="text-align:right;">
0.1315
</td>
<td style="text-align:right;">
-19.82
</td>
<td style="text-align:right;">
2.81
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_idmouse_4
</td>
<td style="text-align:right;">
-4.64
</td>
<td style="text-align:right;">
5.385
</td>
<td style="text-align:right;">
-0.9
</td>
<td style="text-align:right;">
0.4003
</td>
<td style="text-align:right;">
-15.95
</td>
<td style="text-align:right;">
6.67
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_idmouse_5
</td>
<td style="text-align:right;">
10.15
</td>
<td style="text-align:right;">
5.385
</td>
<td style="text-align:right;">
1.9
</td>
<td style="text-align:right;">
0.0758
</td>
<td style="text-align:right;">
-1.17
</td>
<td style="text-align:right;">
21.46
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_idmouse_6
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
5.385
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:right;">
0.9671
</td>
<td style="text-align:right;">
-11.54
</td>
<td style="text-align:right;">
11.09
</td>
</tr>
<tr>
<td style="text-align:left;">
mouse_idmouse_7
</td>
<td style="text-align:right;">
6.37
</td>
<td style="text-align:right;">
5.385
</td>
<td style="text-align:right;">
1.2
</td>
<td style="text-align:right;">
0.2521
</td>
<td style="text-align:right;">
-4.94
</td>
<td style="text-align:right;">
17.69
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The coefficients include a slope for the seven non-reference levels of mouse_id We typically don’t care about these.</li>
<li>In a randomized complete block design with only 1 replicate of each treatment combination per block, like that in Experiment 6g, inference about treatment effects is exactly that same between this model and the random intercept model <code>exp6g_m1.</code> Compare the SE, CI and <em>p</em>-value for the coefficients of the treatment effects to those from the linear mixed model <code>exp6g_m1</code>.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
83.27
</td>
<td style="text-align:right;">
3.441
</td>
<td style="text-align:right;">
18.9
</td>
<td style="text-align:right;">
24.20
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
76.72
</td>
<td style="text-align:right;">
89.81
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentSaline
</td>
<td style="text-align:right;">
-20.52
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18.0
</td>
<td style="text-align:right;">
-5.04
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
-28.25
</td>
<td style="text-align:right;">
-12.78
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentCNO
</td>
<td style="text-align:right;">
-1.84
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18.0
</td>
<td style="text-align:right;">
-0.45
</td>
<td style="text-align:right;">
0.6565
</td>
<td style="text-align:right;">
-9.58
</td>
<td style="text-align:right;">
5.90
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentPost_CNO
</td>
<td style="text-align:right;">
-24.47
</td>
<td style="text-align:right;">
4.071
</td>
<td style="text-align:right;">
18.0
</td>
<td style="text-align:right;">
-6.01
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
-32.21
</td>
<td style="text-align:right;">
-16.73
</td>
</tr>
</tbody>
</table>
<ol start="3" style="list-style-type: decimal">
<li>In a randomized complete block design with only 1 replicate of each treatment combination per block, like that in Experiment 6g, inference about <em>treatment means</em> differs between this model and the random intercept model – the SE of the means of the fixed effect model are smaller than the SE of the means of the linear mixed model. Compare the SE and CI of the <code>(Intercept)</code> (the mean of the reference treatment combination) between the fixed effect and linear mixed model.</li>
<li>The equivalence of inference in the treatment effect between the fixed effect and linear mixed model holds only for balanced randomized complete block designs – where all blocks contain all treatment combinations and the subsampling replicate size is the same for all treatment combinations for all blocks. This means, outside of special cases like Experiment 6g, the choice between adding a blocking variable as a random or fixed factor depends on assumptions about the model.</li>
</ol>
</div>
</div>
<div id="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c" class="section level2 hasAnchor" number="16.7">
<h2><span class="header-section-number">16.7</span> Example 3 – Factorial experiments and no subsampling replicates (exp5c)<a href="lmm.html#example-3-factorial-experiments-and-no-subsampling-replicates-exp5c" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Example 3 is similar to example 2 in that there is no subsampling replication and we cannot add random slopes to the linear mixed model. Example 3 differs in that the design is factorial – there are two, crossed fixed factors. Consequently, there are several alternative models with different sets of random intercepts. The reported model includes two random intercepts, one of which models differences in batch effects among treatment levels (treatment by batch interactions). This <strong>interaction intercept</strong> is an alternative to random slope for modeling treatment by batch interactions.</p>
<p><a href="https://www.nature.com/articles/s41467-019-13869-w" target="_blank">Transcriptomic profiling of skeletal muscle adaptations to exercise and inactivity</a></p>
<p>Source figure: <a href="https://www.nature.com/articles/s41467-019-13869-w#Fig5" target="_blank">Fig. 5c</a></p>
<p>Source data: <a href="https://www.nature.com/articles/s41467-019-13869-w#Sec27" target="_blank">Source Data Fig. 5</a></p>
<div id="understand-the-data-1" class="section level3 hasAnchor" number="16.7.1">
<h3><span class="header-section-number">16.7.1</span> Understand the data<a href="lmm.html#understand-the-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The data for Example 1 are from Figure 5c. Six muscle source cells were used to start six independent cultures. Cells from each culture were treated with either a negative control (“Scr”) or a siRNA (“siNR4A3”) that “silences” expression of the NR4A3 gene product by <a href="https://en.wikipedia.org/wiki/Small_interfering_RNA" target="_blank">cleaving the mRNA</a>. Glucose uptake in the two cell types was measured at rest (“Basal”) and during electrical pulse stimulation (“EPS”).</p>
<p>The design is a <span class="math inline">\(2 \times 2\)</span> <a href="https://en.wikipedia.org/wiki/Blocking_(statistics)#Randomized_block_design" target="_blank">Randomized complete block with no subsampling</a>. There are two factors each with two levels: <span class="math inline">\(\texttt{treatment}\)</span> (“Scr”, “siNR4A3”) and <span class="math inline">\(\texttt{activity}\)</span> (“Basal”, “EPS”). Each source cell is a block. All four treatment combinations were measured once per block.</p>
</div>
<div id="examine-the-data-1" class="section level3 hasAnchor" number="16.7.2">
<h3><span class="header-section-number">16.7.2</span> Examine the data<a href="lmm.html#examine-the-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-examine-1.png" width="576" /></p>
<p>The plot shows a strong donor effect.</p>
</div>
<div id="lmm-exp5c-m1" class="section level3 hasAnchor" number="16.7.3">
<h3><span class="header-section-number">16.7.3</span> Model fit and inference<a href="lmm.html#lmm-exp5c-m1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb532"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb532-1"><a href="lmm.html#cb532-1" tabindex="-1"></a>exp5c_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb532-2"><a href="lmm.html#cb532-2" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb532-3"><a href="lmm.html#cb532-3" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb532-4"><a href="lmm.html#cb532-4" tabindex="-1"></a></span>
<span id="cb532-5"><a href="lmm.html#cb532-5" tabindex="-1"></a>exp5c_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb532-6"><a href="lmm.html#cb532-6" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb532-7"><a href="lmm.html#cb532-7" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment) <span class="sc">+</span></span>
<span id="cb532-8"><a href="lmm.html#cb532-8" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>activity),</span>
<span id="cb532-9"><a href="lmm.html#cb532-9" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb532-10"><a href="lmm.html#cb532-10" tabindex="-1"></a></span>
<span id="cb532-11"><a href="lmm.html#cb532-11" tabindex="-1"></a>exp5c_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb532-12"><a href="lmm.html#cb532-12" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb532-13"><a href="lmm.html#cb532-13" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment),</span>
<span id="cb532-14"><a href="lmm.html#cb532-14" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb532-15"><a href="lmm.html#cb532-15" tabindex="-1"></a></span>
<span id="cb532-16"><a href="lmm.html#cb532-16" tabindex="-1"></a>exp5c_m1d <span class="ot">&lt;-</span> <span class="fu">lme</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb532-17"><a href="lmm.html#cb532-17" tabindex="-1"></a>                 <span class="at">random =</span>  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor,</span>
<span id="cb532-18"><a href="lmm.html#cb532-18" tabindex="-1"></a>                 <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb532-19"><a href="lmm.html#cb532-19" tabindex="-1"></a>                 <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span><span class="sc">|</span>t.by.a),</span>
<span id="cb532-20"><a href="lmm.html#cb532-20" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb532-21"><a href="lmm.html#cb532-21" tabindex="-1"></a></span>
<span id="cb532-22"><a href="lmm.html#cb532-22" tabindex="-1"></a><span class="co"># check AIC</span></span>
<span id="cb532-23"><a href="lmm.html#cb532-23" tabindex="-1"></a><span class="fu">AIC</span>(exp5c_m1a, exp5c_m1b, exp5c_m1c, exp5c_m1d)</span></code></pre></div>
<pre><code>##           df       AIC
## exp5c_m1a  6  8.153813
## exp5c_m1b  8 10.052676
## exp5c_m1c  7  8.052676
## exp5c_m1d 15 14.763024</code></pre>
<div class="sourceCode" id="cb534"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb534-1"><a href="lmm.html#cb534-1" tabindex="-1"></a><span class="co"># check VarCorr model c</span></span>
<span id="cb534-2"><a href="lmm.html#cb534-2" tabindex="-1"></a><span class="fu">VarCorr</span>(exp5c_m1c) <span class="co"># fine</span></span></code></pre></div>
<pre><code>##  Groups          Name        Std.Dev.
##  donor:treatment (Intercept) 0.089519
##  donor           (Intercept) 0.352097
##  Residual                    0.090685</code></pre>
<div class="sourceCode" id="cb536"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb536-1"><a href="lmm.html#cb536-1" tabindex="-1"></a><span class="co"># report 1c (based on AIC and VarCorr check)</span></span>
<span id="cb536-2"><a href="lmm.html#cb536-2" tabindex="-1"></a>exp5c_m1 <span class="ot">&lt;-</span> exp5c_m1c</span></code></pre></div>
<p>exp5c_m1b (equivalent to univariate repeated measures ANOVA) is singular fit. don’t use. Trivial difference in AIC between exp5c_m1a and exp5c_m1c. Nothing in VarCorr with exp5c_m1c raises red flags. Report exp5c_m1c.</p>
<div id="check-the-model-2" class="section level4 hasAnchor" number="16.7.3.1">
<h4><span class="header-section-number">16.7.3.1</span> Check the model<a href="lmm.html#check-the-model-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb537"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb537-1"><a href="lmm.html#cb537-1" tabindex="-1"></a><span class="fu">ggcheck_the_model</span>(exp5c_m1)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c_m1-check-1.png" width="576" /></p>
<p>fine.</p>
</div>
<div id="inference-from-the-model-5" class="section level4 hasAnchor" number="16.7.3.2">
<h4><span class="header-section-number">16.7.3.2</span> Inference from the model<a href="lmm.html#inference-from-the-model-5" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb538"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb538-1"><a href="lmm.html#cb538-1" tabindex="-1"></a>exp5c_m1_coef <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">coef</span>(<span class="fu">summary</span>(exp5c_m1)),</span>
<span id="cb538-2"><a href="lmm.html#cb538-2" tabindex="-1"></a>                       <span class="fu">confint</span>(exp5c_m1)[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>),])</span></code></pre></div>
<div class="sourceCode" id="cb539"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb539-1"><a href="lmm.html#cb539-1" tabindex="-1"></a>exp5c_m1_coef <span class="sc">%&gt;%</span></span>
<span id="cb539-2"><a href="lmm.html#cb539-2" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb539-3"><a href="lmm.html#cb539-3" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
1.34
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
8.8
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
1.02
</td>
<td style="text-align:right;">
1.66
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
8.5
</td>
<td style="text-align:right;">
1.1
</td>
<td style="text-align:right;">
0.2994
</td>
<td style="text-align:right;">
-0.07
</td>
<td style="text-align:right;">
0.23
</td>
</tr>
<tr>
<td style="text-align:left;">
activityEPS
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
2.9
</td>
<td style="text-align:right;">
0.0163
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.25
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3:activityEPS
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
-3.6
</td>
<td style="text-align:right;">
0.0048
</td>
<td style="text-align:right;">
-0.41
</td>
<td style="text-align:right;">
-0.12
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb540"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb540-1"><a href="lmm.html#cb540-1" tabindex="-1"></a>exp5c_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp5c_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>))</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:left;">
activity
</th>
<th style="text-align:right;">
emmean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Scr
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:right;">
1.34
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
0.96
</td>
<td style="text-align:right;">
1.72
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:right;">
1.42
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
1.04
</td>
<td style="text-align:right;">
1.80
</td>
</tr>
<tr>
<td style="text-align:left;">
Scr
</td>
<td style="text-align:left;">
EPS
</td>
<td style="text-align:right;">
1.49
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
1.11
</td>
<td style="text-align:right;">
1.87
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3
</td>
<td style="text-align:left;">
EPS
</td>
<td style="text-align:right;">
1.31
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
0.93
</td>
<td style="text-align:right;">
1.68
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb541"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb541-1"><a href="lmm.html#cb541-1" tabindex="-1"></a><span class="co"># exp5c_emm # print in console to get row numbers</span></span>
<span id="cb541-2"><a href="lmm.html#cb541-2" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb541-3"><a href="lmm.html#cb541-3" tabindex="-1"></a>scr_basal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb541-4"><a href="lmm.html#cb541-4" tabindex="-1"></a>siNR4A3_basal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb541-5"><a href="lmm.html#cb541-5" tabindex="-1"></a>scr_eps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb541-6"><a href="lmm.html#cb541-6" tabindex="-1"></a>siNR4A3_eps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb541-7"><a href="lmm.html#cb541-7" tabindex="-1"></a></span>
<span id="cb541-8"><a href="lmm.html#cb541-8" tabindex="-1"></a>exp5c_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp5c_m1_emm,</span>
<span id="cb541-9"><a href="lmm.html#cb541-9" tabindex="-1"></a>                       <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb541-10"><a href="lmm.html#cb541-10" tabindex="-1"></a>                         <span class="st">&quot;(Scr EPS) - (Scr Basal)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(scr_eps <span class="sc">-</span> scr_basal),</span>
<span id="cb541-11"><a href="lmm.html#cb541-11" tabindex="-1"></a>                         <span class="st">&quot;(siNR4A3 EPS) - (siNR4A3 Basal)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(siNR4A3_eps <span class="sc">-</span> siNR4A3_basal),</span>
<span id="cb541-12"><a href="lmm.html#cb541-12" tabindex="-1"></a>                         <span class="st">&quot;Interaction&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(siNR4A3_eps <span class="sc">-</span> siNR4A3_basal) <span class="sc">-</span></span>
<span id="cb541-13"><a href="lmm.html#cb541-13" tabindex="-1"></a>                           <span class="fu">c</span>(scr_eps <span class="sc">-</span> scr_basal)</span>
<span id="cb541-14"><a href="lmm.html#cb541-14" tabindex="-1"></a>                           </span>
<span id="cb541-15"><a href="lmm.html#cb541-15" tabindex="-1"></a>                       ),</span>
<span id="cb541-16"><a href="lmm.html#cb541-16" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb541-17"><a href="lmm.html#cb541-17" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb541-18"><a href="lmm.html#cb541-18" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.016
</td>
</tr>
<tr>
<td style="text-align:left;">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
-2.22
</td>
<td style="text-align:right;">
0.051
</td>
</tr>
<tr>
<td style="text-align:left;">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.43
</td>
<td style="text-align:right;">
-0.10
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.005
</td>
</tr>
</tbody>
</table>
</div>
<div id="lmm-exp5c-plotthemodel" class="section level4 hasAnchor" number="16.7.3.3">
<h4><span class="header-section-number">16.7.3.3</span> Plot the model<a href="lmm.html#lmm-exp5c-plotthemodel" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-228"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/unnamed-chunk-228-1.png" alt="Treatment effect on glucose uptake. Gray dots connected by lines are individual donors." width="576" />
<p class="caption">
Figure 16.12: Treatment effect on glucose uptake. Gray dots connected by lines are individual donors.
</p>
</div>
</div>
<div id="alternaplot-the-model-2" class="section level4 hasAnchor" number="16.7.3.4">
<h4><span class="header-section-number">16.7.3.4</span> Alternaplot the model<a href="lmm.html#alternaplot-the-model-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="figure"><span style="display:block;" id="fig:lmm-exp5c-alternaplot"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-alternaplot-1.png" alt="Glucose response to different treatments. Gray dots connected by lines are individual donors. Dashed gray line is expected additive mean of &quot;siNR4A3 EPS&quot;." width="576" />
<p class="caption">
Figure 16.13: Glucose response to different treatments. Gray dots connected by lines are individual donors. Dashed gray line is expected additive mean of “siNR4A3 EPS”.
</p>
</div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Many researchers might look at the wide confidence intervals relative to the short distance between the means and think “no effect”. The confidence intervals are correct, they simply are <em>not</em> meant to be tools for inferring anything about differences in means. This is one of many reasons why plots of means and error bars can be misleading for inference, despite the ubiquity of their use for communicating results. And, its why I prefer the <a href="lmm.html#lmm-exp5c-plotthemodel">effects-and-response plots</a>, which explicitly communicate correct inference about effects.</li>
</ol>
</div>
</div>
<div id="why-we-care-about-modeling-batch-in-exp5c" class="section level3 hasAnchor" number="16.7.4">
<h3><span class="header-section-number">16.7.4</span> Why we care about modeling batch in exp5c<a href="lmm.html#why-we-care-about-modeling-batch-in-exp5c" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Figure <a href="lmm.html#fig:lmm-exp5c-why">16.14</a> shows the modeled means of the four treatment combinations and the individual values colored by donor. It is pretty easy to see that the glucose uptake values for donors 4 and 5 are well above the mean for all four treatments. And, the values for donors 1, 2, and 3 are well below the mean for all four treatments. The values for donor 6 are near the mean for all four treatments.</p>
<div class="figure"><span style="display:block;" id="fig:lmm-exp5c-why"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-why-1.png" alt="Why we care about blocking. The black dots are the modeled means of each treatment combination. The colored dots are the measured values of the response for each donor. The position of a donor relative to the mean is easy to see with these data." width="672" />
<p class="caption">
Figure 16.14: Why we care about blocking. The black dots are the modeled means of each treatment combination. The colored dots are the measured values of the response for each donor. The position of a donor relative to the mean is easy to see with these data.
</p>
</div>
<p>Let’s compare the effects estimated by the linear mixed model <code>exp5c_m1</code> with a linear model that ignores donor.</p>
<div class="sourceCode" id="cb542"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb542-1"><a href="lmm.html#cb542-1" tabindex="-1"></a>exp5c_m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb542-2"><a href="lmm.html#cb542-2" tabindex="-1"></a>           <span class="at">data =</span> exp5c)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:lmm-exp5c-why2"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-why2-1.png" alt="A. Inference from a linear mixed model with blocking factor (donor) added as a random intercept. B. Inference from a fixed effects model." width="576" />
<p class="caption">
Figure 16.15: A. Inference from a linear mixed model with blocking factor (donor) added as a random intercept. B. Inference from a fixed effects model.
</p>
</div>
<p>Figure <a href="lmm.html#fig:lmm-exp5c-why2">16.15</a>A is a plot of the effects from the linear mixed model that models the correlated error due to donor. Figure <a href="lmm.html#fig:lmm-exp5c-why2">16.15</a>B is a plot of the effects from the fixed effect model that ignores the correlated error due to donor. Adding <span class="math inline">\(\texttt{donor}\)</span> as a factor to the linear model increases the precision of the estimate of the treatment effects by eliminating the among-donor component of variance from the error variance.</p>
</div>
<div id="the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts" class="section level3 hasAnchor" number="16.7.5">
<h3><span class="header-section-number">16.7.5</span> The linear mixed model exp5c_m1 adds two random intercepts<a href="lmm.html#the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are <em>two</em> random intercepts in the linear mixed model <code>exp5c_m1</code> fit to the exp5c data (<code>exp5c_m1</code> is copied from <code>exp5c_m1c</code> in <a href="lmm.html#lmm-exp5c-m1">Model fit and inference</a>).</p>
<p><span class="math display" id="eq:lmm-exp5c-m1">\[
\begin{align}
\texttt{glucose_uptake}_j = \ &amp;(\beta_0 + \gamma_{0j} + \gamma_{0jk}) + \beta_1 (\texttt{treatment}_\texttt{siNR4A3}) + \beta_2 (\texttt{activity}_\texttt{EPS}) \ + \\
&amp;\beta_3 (\texttt{treatment}_\texttt{siNR4A3}:\texttt{activity}_\texttt{EPS}) + \varepsilon
\end{align}
\tag{16.4}
\]</span></p>
<ol style="list-style-type: decimal">
<li>The random intercept effect <span class="math inline">\(\gamma_{0j}\)</span> models donor variation in the reference level. <em>j</em> indexes donor <em>j</em>. A coefficient is estimated for each donor.</li>
<li>The random intercept effect <span class="math inline">\(\gamma_{0jk}\)</span> models the variation due to the <span class="math inline">\(\texttt{donor}\)</span> by <span class="math inline">\(\texttt{treatment}\)</span> combinations. A coefficient is estimated for each of the 6 (donors) <span class="math inline">\(\times\)</span> 2 (levels of <span class="math inline">\(\texttt{treatment}\)</span>). <em>k</em> indexes treatment level <em>k</em>. This <strong>interaction intercept</strong> is an alternative to random slope for modeling treatment by batch interactions. Unlike a random slope, there is a <span class="math inline">\(\gamma_{0jk}\)</span> coefficient for each donor in the reference level.</li>
</ol>
</div>
<div id="the-fixed-effect-coefficients-of-model-exp5c_m1" class="section level3 hasAnchor" number="16.7.6">
<h3><span class="header-section-number">16.7.6</span> The fixed effect coefficients of model exp5c_m1<a href="lmm.html#the-fixed-effect-coefficients-of-model-exp5c_m1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The fixed effect coefficients of model <code>exp5c_m1</code> are</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
1.34
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
8.77
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
1.02
</td>
<td style="text-align:right;">
1.66
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
8.5
</td>
<td style="text-align:right;">
1.11
</td>
<td style="text-align:right;">
0.2994
</td>
<td style="text-align:right;">
-0.07
</td>
<td style="text-align:right;">
0.23
</td>
</tr>
<tr>
<td style="text-align:left;">
activityEPS
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.0163
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.25
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3:activityEPS
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.0048
</td>
<td style="text-align:right;">
-0.41
</td>
<td style="text-align:right;">
-0.12
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The interpretation of the fixed effectcs coefficients have the <a href="factorial.html#twoway-what-coefs-are">usual interpretation for a factorial linear model</a>. Figure <a href="lmm.html#fig:lmm-exp5c-explainer-1">16.16</a> is a reminder.</li>
</ol>
<div class="figure"><span style="display:block;" id="fig:lmm-exp5c-explainer-1"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-explainer-1-1.png" alt="Fixed effects estimated by exp5c_m1. The light gray point is the expected value of the GPR174- F treatment if genotype and sex were additive." width="672" />
<p class="caption">
Figure 16.16: Fixed effects estimated by exp5c_m1. The light gray point is the expected value of the GPR174- F treatment if genotype and sex were additive.
</p>
</div>
</div>
<div id="the-random-effect-coefficients-of-model-exp5c_m1" class="section level3 hasAnchor" number="16.7.7">
<h3><span class="header-section-number">16.7.7</span> The random effect coefficients of model exp5c_m1<a href="lmm.html#the-random-effect-coefficients-of-model-exp5c_m1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-m1-random-coef">Table 16.11: </span>Random intercept effects for each donor (g_0j) and for each donor:treatment combination (g_0jk) for exp5c_m1. There are two random intercepts. random_intercept_j is the sum of the fixed intercept (b_0) and g_0j. random_intercept_jk is the sum of the fixed intercept (b_0) and g_0jk.
</caption>
<thead>
<tr>
<th style="text-align:left;">
donor:treatment
</th>
<th style="text-align:right;">
random_intercept_j
</th>
<th style="text-align:right;">
random_intercept_jk
</th>
<th style="text-align:right;">
b_0
</th>
<th style="text-align:right;">
g_0j
</th>
<th style="text-align:right;">
g_0jk
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
donor_1:Scr
</td>
<td style="text-align:right;">
1.018
</td>
<td style="text-align:right;">
1.349
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.323
</td>
<td style="text-align:right;">
0.008
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_1:siNR4A3
</td>
<td style="text-align:right;">
1.018
</td>
<td style="text-align:right;">
1.312
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.323
</td>
<td style="text-align:right;">
-0.029
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_2:Scr
</td>
<td style="text-align:right;">
1.033
</td>
<td style="text-align:right;">
1.283
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.307
</td>
<td style="text-align:right;">
-0.058
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_2:siNR4A3
</td>
<td style="text-align:right;">
1.033
</td>
<td style="text-align:right;">
1.379
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.307
</td>
<td style="text-align:right;">
0.038
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_3:Scr
</td>
<td style="text-align:right;">
1.168
</td>
<td style="text-align:right;">
1.310
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.173
</td>
<td style="text-align:right;">
-0.031
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_3:siNR4A3
</td>
<td style="text-align:right;">
1.168
</td>
<td style="text-align:right;">
1.360
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.173
</td>
<td style="text-align:right;">
0.020
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_4:Scr
</td>
<td style="text-align:right;">
1.785
</td>
<td style="text-align:right;">
1.302
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
0.445
</td>
<td style="text-align:right;">
-0.039
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_4:siNR4A3
</td>
<td style="text-align:right;">
1.785
</td>
<td style="text-align:right;">
1.408
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
0.445
</td>
<td style="text-align:right;">
0.068
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_5:Scr
</td>
<td style="text-align:right;">
1.744
</td>
<td style="text-align:right;">
1.433
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
0.403
</td>
<td style="text-align:right;">
0.092
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_5:siNR4A3
</td>
<td style="text-align:right;">
1.744
</td>
<td style="text-align:right;">
1.275
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
0.403
</td>
<td style="text-align:right;">
-0.066
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_6:Scr
</td>
<td style="text-align:right;">
1.296
</td>
<td style="text-align:right;">
1.368
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.045
</td>
<td style="text-align:right;">
0.028
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_6:siNR4A3
</td>
<td style="text-align:right;">
1.296
</td>
<td style="text-align:right;">
1.310
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.045
</td>
<td style="text-align:right;">
-0.031
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The random intercept “random_intercept_j” is the sum of the fixed intercept <span class="math inline">\(b_0\)</span> and the random intercept effect <span class="math inline">\(g_{0j}\)</span> for donor <em>j</em>. The <span class="math inline">\(g_{0j}\)</span> estimate the <span class="math inline">\(\gamma_{0j}\)</span> in Model <a href="lmm.html#eq:lmm-exp5c-m1">(16.4)</a>. Note that <span class="math inline">\(g_{0j}\)</span> is the same for both treatment levels within a donor. The illustration of these random effects is similar to that for Example 2 <code>exp6g_m1</code> (Figure <a href="lmm.html#fig:lmm-exp6g-explainer-2">16.10</a>.</li>
<li>The random intercept “random_intercept_jk” is the sum of the fixed intercept <span class="math inline">\(b_0\)</span> and the random intercept effect <span class="math inline">\(g_{0jk}\)</span> for donor <em>j</em> in treatment level <em>k</em> (with the reference level equal to 1). The <span class="math inline">\(g_{0jk}\)</span> estimate the <span class="math inline">\(\gamma_{0jk}\)</span> in Model <a href="lmm.html#eq:lmm-exp5c-m1">(16.4)</a>. The <span class="math inline">\(g_{0jk}\)</span> differ for each combination of <span class="math inline">\(\texttt{donor}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>.</li>
<li>The donor:treatment random intercepts (“random_intercept_jk”) are shown with the dark, colored dots in Figure <a href="lmm.html#fig:lmm-exp5c-explainer-2">16.17</a>A. The distance from the dark colored dot to the dashed, gray line (the modeled mean for the treatment:activity combination) is the random intercept effect estimate of <span class="math inline">\(\gamma_{0jk}\)</span>.</li>
<li>The set of random intercept effects for the donor:treatment combinations (<span class="math inline">\(g_{0jk}\)</span>) are the same between the two activity levels. This is seen in Figure <a href="lmm.html#fig:lmm-exp5c-explainer-2">16.17</a>A - compare the pattern of dark, colored dots in the two EPS treatments to the two Basal treatments.</li>
<li>The combined random intercept effects (<span class="math inline">\(g_{0j} + g_{0jk}\)</span>) are shown in Figure <a href="lmm.html#fig:lmm-exp5c-explainer-2">16.17</a>B as the distance between the dark, colored dots and the dashed, gray lines.</li>
<li>There is some communication decision-making in Figure <a href="lmm.html#fig:lmm-exp5c-explainer-2">16.17</a>. The actual y-values of the dark, colored dots are the random intercept effect plus the treatment-combination modeled mean. Thus, in A, the dark colored dots are equal to “random_intercept_jk” only at the reference level (since the reference model mean is the fixed intercept).</li>
</ol>
<div class="figure"><span style="display:block;" id="fig:lmm-exp5c-explainer-2"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-explainer-2-1.png" alt="Random intercepts for model exp5c_m1. The pale, colored dots are the measured glucose uptake values for each donor at each treatment combination. The dashed, gray lines are the modeled means for each treatment combination. (A) The distance from a dark colored dot to the dashed, gray line is the random intercept effect $g_{0jk}$. (B) The distance from a dark colored dot to the dashed, gray line is the combined random intercept effect $g_{0j} + g_{0jk}$." width="1066.66666666667" />
<p class="caption">
Figure 16.17: Random intercepts for model exp5c_m1. The pale, colored dots are the measured glucose uptake values for each donor at each treatment combination. The dashed, gray lines are the modeled means for each treatment combination. (A) The distance from a dark colored dot to the dashed, gray line is the random intercept effect <span class="math inline">\(g_{0jk}\)</span>. (B) The distance from a dark colored dot to the dashed, gray line is the combined random intercept effect <span class="math inline">\(g_{0j} + g_{0jk}\)</span>.
</p>
</div>
</div>
<div id="lmm-exp5c-alt" class="section level3 hasAnchor" number="16.7.8">
<h3><span class="header-section-number">16.7.8</span> Alternative models for exp5c<a href="lmm.html#lmm-exp5c-alt" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb543"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb543-1"><a href="lmm.html#cb543-1" tabindex="-1"></a>exp5c_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb543-2"><a href="lmm.html#cb543-2" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb543-3"><a href="lmm.html#cb543-3" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb543-4"><a href="lmm.html#cb543-4" tabindex="-1"></a></span>
<span id="cb543-5"><a href="lmm.html#cb543-5" tabindex="-1"></a>exp5c_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb543-6"><a href="lmm.html#cb543-6" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb543-7"><a href="lmm.html#cb543-7" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment) <span class="sc">+</span></span>
<span id="cb543-8"><a href="lmm.html#cb543-8" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>activity),</span>
<span id="cb543-9"><a href="lmm.html#cb543-9" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span></code></pre></div>
<pre><code>## boundary (singular) fit: see help(&#39;isSingular&#39;)</code></pre>
<div class="sourceCode" id="cb545"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb545-1"><a href="lmm.html#cb545-1" tabindex="-1"></a>exp5c_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb545-2"><a href="lmm.html#cb545-2" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb545-3"><a href="lmm.html#cb545-3" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment),</span>
<span id="cb545-4"><a href="lmm.html#cb545-4" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb545-5"><a href="lmm.html#cb545-5" tabindex="-1"></a></span>
<span id="cb545-6"><a href="lmm.html#cb545-6" tabindex="-1"></a>exp5c_m1d <span class="ot">&lt;-</span> <span class="fu">lme</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb545-7"><a href="lmm.html#cb545-7" tabindex="-1"></a>                 <span class="at">random =</span>  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor,</span>
<span id="cb545-8"><a href="lmm.html#cb545-8" tabindex="-1"></a>                 <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb545-9"><a href="lmm.html#cb545-9" tabindex="-1"></a>                 <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span><span class="sc">|</span>t.by.a),</span>
<span id="cb545-10"><a href="lmm.html#cb545-10" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Four models are fit. The models specify the same fixed effects but different random effects or patterns of correlated error. Again, we care about the fixed effects – this is the point of the experiment – but the specification of the random effects determines the error variance and degrees of freedom for computing uncertainty.</li>
<li>Model <code>exp5c_m1a</code> specifies a random intercept for each level of <span class="math inline">\(\texttt{donor}\)</span>.</li>
<li>Model <code>exp5c_m1b</code> adds two additional random intercepts to Model <code>exp5c_m1a</code>. The code <code>(1 | donor:treatment)</code> adds an intercept for each combination of <span class="math inline">\(\texttt{donor}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>. The code <code>(1 | donor:activity)</code> adds an intercept for each combination of <span class="math inline">\(\texttt{donor}\)</span> and <span class="math inline">\(\texttt{activity}\)</span>. This model is the equivalent of the univariate model of a <strong>repeated measures ANOVA</strong> of these data (see <a href="lmm.html#lmm-rmanova-5c">Classical (“univariate model”) repeated measures ANOVA</a> below).</li>
<li>Model <code>exp5c_m1c</code> adds only one additional random intercept (<code>(1 | donor:treatment)</code>). It is a simplification of Model <code>exp5c_m1c</code>. This is the model reported and explained above.</li>
<li>Model <code>exp5c_m1d</code> has the same random effects as <code>exp5c_m1a</code> but models unstructured correlated error and heterogeneity of the error, as described for Example 2: <a href="lmm.html#lmm-exp6g-alt">Alternative models for exp6g</a>. This model is equivalent to the multivariate model of a <strong>repeated measures ANOVA</strong> of these data (see <a href="lmm.html#lmm-rmanova-5c-multi">“Multivariate model” repeated measures ANOVA</a> below).</li>
<li><code>lmer</code> returns a message “boundary (singular) fit: see ?isSingular” for <code>exp5c_m1b</code>. The model <em>was</em> fit but the message means that we should be cautious about (or simply avoid) interpreting any inferential statistics (SEs, CIs, <em>p</em>-values). Looking at the estimates of the estimated parameters of the random effects in the table below (the variances of the effects and the covariances among the effects), Model <code>exp5c_m1b</code> estimates zero variance for the <span class="math inline">\(\texttt{donor:activity}\)</span> random intercept. This suggests that we could simplify <code>exp5c_m1b</code> by removing <code>(1 | donor:activity)</code> from the model – this is <code>exp5c_m1c</code>. Think about this warning. If we fit using repeated measures ANOVA (next section), we won’t get a warning message but we still may be <a href="https://en.wikipedia.org/wiki/Overfitting" target="_blank">overfitting</a>.</li>
</ol>
<div class="sourceCode" id="cb546"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb546-1"><a href="lmm.html#cb546-1" tabindex="-1"></a><span class="fu">VarCorr</span>(exp5c_m1b)</span></code></pre></div>
<pre><code>##  Groups          Name        Std.Dev.
##  donor:activity  (Intercept) 0.000000
##  donor:treatment (Intercept) 0.089519
##  donor           (Intercept) 0.352092
##  Residual                    0.090686</code></pre>
<ol start="7" style="list-style-type: decimal">
<li>Planned comparisons of the four alternative models are given in Table <a href="lmm.html#tab:lmm-exp5c-alternatives-planned">16.12</a>. The effect estimates are the same but inference differs slightly among the models because of how each model partitions error variance to either random effects or the residuals.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-alternatives-planned">Table 16.12: </span>Planned contrasts from the four alternative models.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1a</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
2.25
</td>
<td style="text-align:right;">
0.040
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.26
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
-1.73
</td>
<td style="text-align:right;">
0.105
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.47
</td>
<td style="text-align:right;">
-0.06
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
0.013
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1b</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.016
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
-2.22
</td>
<td style="text-align:right;">
0.051
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.46
</td>
<td style="text-align:right;">
-0.08
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.015
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1c</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.016
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
-2.22
</td>
<td style="text-align:right;">
0.051
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.43
</td>
<td style="text-align:right;">
-0.10
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.005
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1d</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.07
</td>
<td style="text-align:right;">
0.24
</td>
<td style="text-align:right;">
3.82
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.25
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
-1.85
</td>
<td style="text-align:right;">
0.083
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.44
</td>
<td style="text-align:right;">
-0.09
</td>
<td style="text-align:right;">
-3.22
</td>
<td style="text-align:right;">
0.006
</td>
</tr>
</tbody>
</table>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-aic">Table 16.13: </span>AIC of the four alternative models firt to exp5c data
</caption>
<thead>
<tr>
<th style="text-align:left;">
Model
</th>
<th style="text-align:right;">
AIC
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
exp5c_m1a
</td>
<td style="text-align:right;">
8.153813
</td>
</tr>
<tr>
<td style="text-align:left;">
exp5c_m1b
</td>
<td style="text-align:right;">
10.052675
</td>
</tr>
<tr>
<td style="text-align:left;">
exp5c_m1c
</td>
<td style="text-align:right;">
8.052675
</td>
</tr>
<tr>
<td style="text-align:left;">
exp5c_m1d
</td>
<td style="text-align:right;">
14.763024
</td>
</tr>
</tbody>
</table>
<ol start="6" style="list-style-type: decimal">
<li>The use of AIC for selecting among models with different random effects was described above. The AICs of the models suggest that models <code>exp5c_m1b</code> and <code>exp5c_m1d</code> are too complex given the data. The AICs of models <code>exp5c_m1a</code> and <code>exp5c_m1c</code> are effectively the same. I reported <code>exp5c_m1c</code> simply to allow me to explain the <code>donor:treatment</code> random effect.</li>
</ol>
</div>
<div id="lmm-rmanova-5c" class="section level3 hasAnchor" number="16.7.9">
<h3><span class="header-section-number">16.7.9</span> Classical (“univariate model”) repeated measures ANOVA<a href="lmm.html#lmm-rmanova-5c" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb548"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb548-1"><a href="lmm.html#cb548-1" tabindex="-1"></a>exp5c_aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb548-2"><a href="lmm.html#cb548-2" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb548-3"><a href="lmm.html#cb548-3" tabindex="-1"></a>                  <span class="at">data =</span> exp5c)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-uni-rmanova-planned">Table 16.14: </span>Planned contrasts for the linear mixed model exp5c_m1c and the univariate model of the repeated measures ANOVA.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1c</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.016
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
-2.22
</td>
<td style="text-align:right;">
0.051
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.43
</td>
<td style="text-align:right;">
-0.10
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.005
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>univariate RM-ANOVA</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.25
</td>
<td style="text-align:right;">
3.82
</td>
<td style="text-align:right;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.28
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
-1.85
</td>
<td style="text-align:right;">
0.123
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.48
</td>
<td style="text-align:right;">
-0.05
</td>
<td style="text-align:right;">
-3.22
</td>
<td style="text-align:right;">
0.023
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>In addition to the assumptions of the linear model outlined in the Violations chapter, the classical (“univariate model”) repeated measures ANOVA assumes <a href="https://en.wikipedia.org/wiki/Mauchly%27s_sphericity_test#Sphericity" target="_blank">sphericity</a>, which is the equality of the variances of all pairwise differences among treatment combinations.</li>
<li>The univariate model of the repeated measures ANOVA is equivalent to Model <code>exp5c_m1c</code>. Inference (SEs, CIs, <em>p</em>-values) differs slightly in this here because the estimated variance for the <span class="math inline">\(\texttt{donor:activity}\)</span> random intercept of Model <code>exp5c_m1c</code> is zero. In cases where all variances are greater than zero <em>and the design is balanced</em> (no block is missing a treatment combination), the two tables would be identical.</li>
</ol>
</div>
<div id="lmm-rmanova-5c-multi" class="section level3 hasAnchor" number="16.7.10">
<h3><span class="header-section-number">16.7.10</span> “Multivariate model” repeated measures ANOVA of exp5c<a href="lmm.html#lmm-rmanova-5c-multi" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-multi-rmanova">Table 16.15: </span>Planned contrasts for the linear mixed model exp5c_m1d and the multivariate model of the repeated measures ANOVA.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1d</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.07
</td>
<td style="text-align:right;">
0.24
</td>
<td style="text-align:right;">
3.82
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.25
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
-1.85
</td>
<td style="text-align:right;">
0.083
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.44
</td>
<td style="text-align:right;">
-0.09
</td>
<td style="text-align:right;">
-3.22
</td>
<td style="text-align:right;">
0.006
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>multivariate RM-ANOVA</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.25
</td>
<td style="text-align:right;">
3.82
</td>
<td style="text-align:right;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.28
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
-1.85
</td>
<td style="text-align:right;">
0.123
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.48
</td>
<td style="text-align:right;">
-0.05
</td>
<td style="text-align:right;">
-3.22
</td>
<td style="text-align:right;">
0.023
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The sphericity assumption of the repeated measures ANOVA is relaxed if the model is fit using the “multivariate model”.</li>
<li>The multivariate model repeated measures ANOVA is a linear model with a multivariate response and not a linear mixed model. In the multivariate model, each treatment combination is a different response variable and there is a single row for each level of the random factor (<span class="math inline">\(\texttt{donor}\)</span> in Experiment 5c). The multivariate model is a different way of handling the correlated error that occurs when conceiving of the design as univariate.</li>
<li>The multivariate repeated measures ANOVA can be specified using the linear mixed model <code>exp5c_m1d</code>. Compared to the contrasts from the multivariate model of the repeated measures ANOVA above, the SEs of the contrasts are the same but the degrees of freedom differ. Because of the increased df of the linear mixed model, the CIs are narrower and the <em>p</em>-value is smaller – that is the linear mixed model is less conservative than the repeated measures ANOVA. There is no correct degrees of freedom for a linear mixed model like this and emmeans outputs one way to compute these (with options for others, none of which are equivalent to those from the multivariate model RM-ANOVA).</li>
</ol>
</div>
<div id="modeling-textttdonor-as-a-fixed-effect" class="section level3 hasAnchor" number="16.7.11">
<h3><span class="header-section-number">16.7.11</span> Modeling <span class="math inline">\(\texttt{donor}\)</span> as a fixed effect<a href="lmm.html#modeling-textttdonor-as-a-fixed-effect" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><span class="math display" id="eq:lmm-exp5c-m3">\[
\begin{equation}
\texttt{glucose_uptake} \sim \texttt{treatment * activity + donor}
\end{equation}
\tag{16.5}
\]</span></p>
<p>Model <a href="lmm.html#eq:lmm-exp5c-m3">(16.5)</a> (using R formula syntax) is a linear model with the block <span class="math inline">\(\texttt{donor}\)</span> added as a fixed covariate instead of a random intercept.</p>
<div class="sourceCode" id="cb549"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb549-1"><a href="lmm.html#cb549-1" tabindex="-1"></a>exp5c_m3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span> donor,</span>
<span id="cb549-2"><a href="lmm.html#cb549-2" tabindex="-1"></a>               <span class="at">data =</span> exp5c)</span></code></pre></div>
<p>The coefficients of the fit model are</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.071
</td>
<td style="text-align:right;">
14.0
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
0.85
</td>
<td style="text-align:right;">
1.15
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
1.2
</td>
<td style="text-align:right;">
0.2453
</td>
<td style="text-align:right;">
-0.06
</td>
<td style="text-align:right;">
0.22
</td>
</tr>
<tr>
<td style="text-align:left;">
activityEPS
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
2.2
</td>
<td style="text-align:right;">
0.0402
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.29
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_2
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:right;">
0.8444
</td>
<td style="text-align:right;">
-0.16
</td>
<td style="text-align:right;">
0.19
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_3
</td>
<td style="text-align:right;">
0.16
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
1.9
</td>
<td style="text-align:right;">
0.0757
</td>
<td style="text-align:right;">
-0.02
</td>
<td style="text-align:right;">
0.33
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_4
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
9.8
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
0.63
</td>
<td style="text-align:right;">
0.98
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_5
</td>
<td style="text-align:right;">
0.76
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
9.2
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
0.59
</td>
<td style="text-align:right;">
0.94
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_6
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
3.5
</td>
<td style="text-align:right;">
0.0030
</td>
<td style="text-align:right;">
0.12
</td>
<td style="text-align:right;">
0.47
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3:activityEPS
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
-2.8
</td>
<td style="text-align:right;">
0.0132
</td>
<td style="text-align:right;">
-0.47
</td>
<td style="text-align:right;">
-0.06
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The coefficients include a slope for the five non-reference levels of donor. We typically don’t care about these.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-231">Table 16.16: </span>Planned contrasts for the linear mixed model exp5c_m1a and the fixed effect model exp5c_m3.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1a</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
2.25
</td>
<td style="text-align:right;">
0.040
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.26
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
-1.73
</td>
<td style="text-align:right;">
0.105
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.47
</td>
<td style="text-align:right;">
-0.06
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
0.013
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m3</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
2.25
</td>
<td style="text-align:right;">
0.040
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.26
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
-1.73
</td>
<td style="text-align:right;">
0.105
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.47
</td>
<td style="text-align:right;">
-0.06
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
0.013
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>In a randomized complete block design with only 1 replicate of each treatment combination per block, like that in Experiment 5c, inference about treatment effects is exactly that same between this fixed effect model and the random intercept model <code>exp5c_m1a.</code></li>
<li>The equivalence of inference in the treatment effect between the fixed effect and linear mixed model holds only for balanced randomized complete block designs – where all blocks contain all treatment combinations and the subsampling replicate size is the same for all treatment combinations for all blocks. This means, outside of special cases like Experiment 5c, the choice between adding a blocking variable as a random or fixed factor depends on assumptions about the model.</li>
</ol>
</div>
</div>
<div id="lmm-example4" class="section level2 hasAnchor" number="16.8">
<h2><span class="header-section-number">16.8</span> Example 4 – Experiments with subsampling replication (exp1g)<a href="lmm.html#lmm-example4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This example is from a design with batches (independent experiments) that contain all treatment levels of a single factor <em>and</em> subsampling replication. These data were used to introduce linear mixed models in Example 1. The design of the experiment is <span class="math inline">\(2 \times 2\)</span> factorial. Example 1 flattened the analysis to simplify explanation of random intercepts and random slopes. Here, the data are analyzed with a factorial model.</p>
<p><a href="https://www.nature.com/articles/s41586-019-1873-0" target="_blank">A GPR174–CCL21 module imparts sexual dimorphism to humoral immunity</a></p>
<p><a href="https://pubmed.ncbi.nlm.nih.gov/31875850/" target="_blank">Public source</a></p>
<p>Source figure: <a href="https://www.nature.com/articles/s41586-019-1873-0/figures/1" target="_blank">Fig. 1g</a></p>
<p>Source data: <a href="https://www.nature.com/articles/s41586-019-1873-0#MOESM3" target="_blank">Source Data Fig. 1</a></p>
<div id="understand-the-data-2" class="section level3 hasAnchor" number="16.8.1">
<h3><span class="header-section-number">16.8.1</span> Understand the data<a href="lmm.html#understand-the-data-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The researchers in this paper are interested in discovering mechanisms causing the lower <a href="https://en.wikipedia.org/wiki/Humoral_immunity" target="_blank">antibody-mediated immune response</a> in males relative to females. The data in Fig. 1 are from a set of experiments on mice to investigate how the G-protein coupled receptor protein GPR174 regulates formation of the <a href="https://en.wikipedia.org/wiki/Germinal_center" target="_blank">B-cell germinal center in secondary lymph tissue</a>. GPR174 is a X-linked gene.</p>
<p><strong>Response variable</strong> <span class="math inline">\(\texttt{gc}\)</span> – germinal center size (%). The units are the percent of cells expressing germinal center markers.</p>
<p><strong>Factor 1</strong> – <span class="math inline">\(\texttt{sex}\)</span> (“M”, “F”). Male (“M”) is the reference level.</p>
<p><strong>Factor 2</strong> – <span class="math inline">\(\texttt{chromosome}\)</span> (“Gpr174+”, “Gpr174-”). “Gpr174-” is a GPR174 knockout. The wildtype (“Gpr174+”) condition is the reference level.</p>
<p><strong>Design</strong> – <span class="math inline">\(2 \times 2\)</span>, that is, two crossed factors each with two levels. This results in four groups, each with a unique combination of the levels from each factor. “M Gpr174+” is the control. “M Gpr174+” is the knockout genotype in males (“knockout added”). “F Gpr174+” is the wildtype female (“X chromosome added”). “F Gpr174-” is the knockout female (“knockout and X chromosome added”.</p>
</div>
<div id="examine-the-data-2" class="section level3 hasAnchor" number="16.8.2">
<h3><span class="header-section-number">16.8.2</span> Examine the data<a href="lmm.html#examine-the-data-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb550"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb550-1"><a href="lmm.html#cb550-1" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> exp1g,</span>
<span id="cb550-2"><a href="lmm.html#cb550-2" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb550-3"><a href="lmm.html#cb550-3" tabindex="-1"></a>           <span class="at">y =</span> gc,</span>
<span id="cb550-4"><a href="lmm.html#cb550-4" tabindex="-1"></a>           <span class="at">color =</span> experiment_id)) <span class="sc">+</span></span>
<span id="cb550-5"><a href="lmm.html#cb550-5" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.4</span>))</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp1g-examine-1.png" width="576" /></p>
</div>
<div id="fit-the-model-9" class="section level3 hasAnchor" number="16.8.3">
<h3><span class="header-section-number">16.8.3</span> Fit the model<a href="lmm.html#fit-the-model-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb551"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb551-1"><a href="lmm.html#cb551-1" tabindex="-1"></a><span class="co"># three slope parameters</span></span>
<span id="cb551-2"><a href="lmm.html#cb551-2" tabindex="-1"></a>exp1g_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb551-3"><a href="lmm.html#cb551-3" tabindex="-1"></a>                   (genotype <span class="sc">*</span> sex <span class="sc">|</span> experiment_id),</span>
<span id="cb551-4"><a href="lmm.html#cb551-4" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb551-5"><a href="lmm.html#cb551-5" tabindex="-1"></a></span>
<span id="cb551-6"><a href="lmm.html#cb551-6" tabindex="-1"></a><span class="fu">VarCorr</span>(exp1g_m1a) <span class="co"># looks fine</span></span></code></pre></div>
<pre><code>##  Groups        Name                 Std.Dev. Corr                
##  experiment_id (Intercept)          1.77082                      
##                genotypeGpr174-      0.96192  -0.023              
##                sexF                 2.90154   0.459  0.878       
##                genotypeGpr174-:sexF 1.33169  -0.314 -0.625 -0.706
##  Residual                           1.93155</code></pre>
<div class="sourceCode" id="cb553"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb553-1"><a href="lmm.html#cb553-1" tabindex="-1"></a><span class="co"># one slope parameter but capturing all treatment combinations</span></span>
<span id="cb553-2"><a href="lmm.html#cb553-2" tabindex="-1"></a>exp1g_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb553-3"><a href="lmm.html#cb553-3" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb553-4"><a href="lmm.html#cb553-4" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb553-5"><a href="lmm.html#cb553-5" tabindex="-1"></a></span>
<span id="cb553-6"><a href="lmm.html#cb553-6" tabindex="-1"></a><span class="co"># intercept interactions</span></span>
<span id="cb553-7"><a href="lmm.html#cb553-7" tabindex="-1"></a>exp1g_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb553-8"><a href="lmm.html#cb553-8" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb553-9"><a href="lmm.html#cb553-9" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype) <span class="sc">+</span></span>
<span id="cb553-10"><a href="lmm.html#cb553-10" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb553-11"><a href="lmm.html#cb553-11" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype<span class="sc">:</span>sex),</span>
<span id="cb553-12"><a href="lmm.html#cb553-12" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb553-13"><a href="lmm.html#cb553-13" tabindex="-1"></a></span>
<span id="cb553-14"><a href="lmm.html#cb553-14" tabindex="-1"></a><span class="fu">VarCorr</span>(exp1g_m1c) <span class="co"># id:genotype is low</span></span></code></pre></div>
<pre><code>##  Groups                     Name        Std.Dev. 
##  experiment_id:genotype:sex (Intercept) 0.4046670
##  experiment_id:sex          (Intercept) 1.6927499
##  experiment_id:genotype     (Intercept) 0.0001216
##  experiment_id              (Intercept) 2.5487069
##  Residual                               1.9792428</code></pre>
<div class="sourceCode" id="cb555"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb555-1"><a href="lmm.html#cb555-1" tabindex="-1"></a><span class="co"># drop id:genotype which has low variance</span></span>
<span id="cb555-2"><a href="lmm.html#cb555-2" tabindex="-1"></a>exp1g_m1d <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb555-3"><a href="lmm.html#cb555-3" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb555-4"><a href="lmm.html#cb555-4" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb555-5"><a href="lmm.html#cb555-5" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex<span class="sc">:</span>genotype),</span>
<span id="cb555-6"><a href="lmm.html#cb555-6" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb555-7"><a href="lmm.html#cb555-7" tabindex="-1"></a></span>
<span id="cb555-8"><a href="lmm.html#cb555-8" tabindex="-1"></a><span class="fu">AIC</span>(exp1g_m1a, exp1g_m1b, exp1g_m1c, exp1g_m1d)</span></code></pre></div>
<pre><code>##           df      AIC
## exp1g_m1a 15 344.1579
## exp1g_m1b 15 344.1529
## exp1g_m1c  9 337.7602
## exp1g_m1d  8 335.7602</code></pre>
<div class="sourceCode" id="cb557"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb557-1"><a href="lmm.html#cb557-1" tabindex="-1"></a><span class="co"># go with exp1g_m1d.</span></span>
<span id="cb557-2"><a href="lmm.html#cb557-2" tabindex="-1"></a>exp1g_m1 <span class="ot">&lt;-</span> exp1g_m1d</span></code></pre></div>
</div>
<div id="inference-from-the-model-6" class="section level3 hasAnchor" number="16.8.4">
<h3><span class="header-section-number">16.8.4</span> Inference from the model<a href="lmm.html#inference-from-the-model-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb558"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb558-1"><a href="lmm.html#cb558-1" tabindex="-1"></a>exp1g_m1_coef <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(exp1g_m1))</span></code></pre></div>
<div class="sourceCode" id="cb559"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb559-1"><a href="lmm.html#cb559-1" tabindex="-1"></a>exp1g_m1_coef</span></code></pre></div>
<pre><code>##                       Estimate Std. Error       df   t value    Pr(&gt;|t|)
## (Intercept)           8.435718   1.612602 4.638595  5.231123 0.004202015
## genotypeGpr174-       2.568365   0.712122 5.760582  3.606637 0.012094327
## sexF                  5.583860   1.397388 4.024010  3.995927 0.015995349
## genotypeGpr174-:sexF -5.304752   1.013552 5.918159 -5.233823 0.002033657</code></pre>
<div class="sourceCode" id="cb561"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb561-1"><a href="lmm.html#cb561-1" tabindex="-1"></a><span class="co"># order of factors reversed in specs because I want sex to be</span></span>
<span id="cb561-2"><a href="lmm.html#cb561-2" tabindex="-1"></a><span class="co"># main x-axis variable in plot</span></span>
<span id="cb561-3"><a href="lmm.html#cb561-3" tabindex="-1"></a>exp1g_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp1g_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;genotype&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb562"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb562-1"><a href="lmm.html#cb562-1" tabindex="-1"></a><span class="co"># exp1g_m1_emm # print in console to get row numbers</span></span>
<span id="cb562-2"><a href="lmm.html#cb562-2" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb562-3"><a href="lmm.html#cb562-3" tabindex="-1"></a>wt_m <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb562-4"><a href="lmm.html#cb562-4" tabindex="-1"></a>wt_f <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb562-5"><a href="lmm.html#cb562-5" tabindex="-1"></a>ko_m <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb562-6"><a href="lmm.html#cb562-6" tabindex="-1"></a>ko_f <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb562-7"><a href="lmm.html#cb562-7" tabindex="-1"></a></span>
<span id="cb562-8"><a href="lmm.html#cb562-8" tabindex="-1"></a><span class="co"># simple effects within males and females + interaction </span></span>
<span id="cb562-9"><a href="lmm.html#cb562-9" tabindex="-1"></a><span class="co"># 1. (ko_m - wt_m) </span></span>
<span id="cb562-10"><a href="lmm.html#cb562-10" tabindex="-1"></a><span class="co"># 2. (ko_f - wt_f)</span></span>
<span id="cb562-11"><a href="lmm.html#cb562-11" tabindex="-1"></a></span>
<span id="cb562-12"><a href="lmm.html#cb562-12" tabindex="-1"></a>exp1g_contrasts <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb562-13"><a href="lmm.html#cb562-13" tabindex="-1"></a>  <span class="st">&quot;(Gpr174- M) - (Gpr174+ M)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(ko_m <span class="sc">-</span> wt_m),</span>
<span id="cb562-14"><a href="lmm.html#cb562-14" tabindex="-1"></a>  <span class="st">&quot;(Gpr174- F) - (Gpr174+ F)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(ko_f <span class="sc">-</span> wt_f),</span>
<span id="cb562-15"><a href="lmm.html#cb562-15" tabindex="-1"></a>  <span class="st">&quot;Interaction&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(ko_f <span class="sc">-</span> wt_f) <span class="sc">-</span></span>
<span id="cb562-16"><a href="lmm.html#cb562-16" tabindex="-1"></a>    <span class="fu">c</span>(ko_m <span class="sc">-</span> wt_m)</span>
<span id="cb562-17"><a href="lmm.html#cb562-17" tabindex="-1"></a>)</span>
<span id="cb562-18"><a href="lmm.html#cb562-18" tabindex="-1"></a>exp1g_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp1g_m1_emm,</span>
<span id="cb562-19"><a href="lmm.html#cb562-19" tabindex="-1"></a>                       <span class="at">method =</span> exp1g_contrasts,</span>
<span id="cb562-20"><a href="lmm.html#cb562-20" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb562-21"><a href="lmm.html#cb562-21" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb562-22"><a href="lmm.html#cb562-22" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Gpr174- M) - (Gpr174+ M)
</td>
<td style="text-align:right;">
2.57
</td>
<td style="text-align:right;">
0.714
</td>
<td style="text-align:right;">
5.784
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
4.33
</td>
<td style="text-align:right;">
3.60
</td>
<td style="text-align:right;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;">
(Gpr174- F) - (Gpr174+ F)
</td>
<td style="text-align:right;">
-2.74
</td>
<td style="text-align:right;">
0.722
</td>
<td style="text-align:right;">
6.100
</td>
<td style="text-align:right;">
-4.50
</td>
<td style="text-align:right;">
-0.98
</td>
<td style="text-align:right;">
-3.79
</td>
<td style="text-align:right;">
0.009
</td>
</tr>
<tr>
<td style="text-align:left;">
Interaction
</td>
<td style="text-align:right;">
-5.30
</td>
<td style="text-align:right;">
1.016
</td>
<td style="text-align:right;">
5.942
</td>
<td style="text-align:right;">
-7.80
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
-5.22
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The direction of the estimated effect is opposite in males and females</li>
</ol>
</div>
<div id="plot-the-model-6" class="section level3 hasAnchor" number="16.8.5">
<h3><span class="header-section-number">16.8.5</span> Plot the model<a href="lmm.html#plot-the-model-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-236"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/unnamed-chunk-236-1.png" alt="Treatment effect on germinal center (GC) formation. Small, pale, colored dots are independent experiments. Intermediate size colored dots are experiment means." width="576" />
<p class="caption">
Figure 16.18: Treatment effect on germinal center (GC) formation. Small, pale, colored dots are independent experiments. Intermediate size colored dots are experiment means.
</p>
</div>
</div>
<div id="alternaplot-the-model-3" class="section level3 hasAnchor" number="16.8.6">
<h3><span class="header-section-number">16.8.6</span> Alternaplot the model<a href="lmm.html#alternaplot-the-model-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure"><span style="display:block;" id="fig:lmm-exp1g-alternaplot"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp1g-alternaplot-1.png" alt="Germinal center (GC) formation in response to treatment. Small, pale, colored dots are independent experiments. Intermediate size colored dots are experiment means. Dashed gray line is expected additive mean of Gpr174- F" width="576" />
<p class="caption">
Figure 16.19: Germinal center (GC) formation in response to treatment. Small, pale, colored dots are independent experiments. Intermediate size colored dots are experiment means. Dashed gray line is expected additive mean of Gpr174- F
</p>
</div>
</div>
<div id="understanding-the-alternative-models" class="section level3 hasAnchor" number="16.8.7">
<h3><span class="header-section-number">16.8.7</span> Understanding the alternative models<a href="lmm.html#understanding-the-alternative-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb563"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb563-1"><a href="lmm.html#cb563-1" tabindex="-1"></a>exp1g_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb563-2"><a href="lmm.html#cb563-2" tabindex="-1"></a>                   (genotype <span class="sc">*</span> sex <span class="sc">|</span> experiment_id),</span>
<span id="cb563-3"><a href="lmm.html#cb563-3" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb563-4"><a href="lmm.html#cb563-4" tabindex="-1"></a></span>
<span id="cb563-5"><a href="lmm.html#cb563-5" tabindex="-1"></a>exp1g_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb563-6"><a href="lmm.html#cb563-6" tabindex="-1"></a>                   (treatment <span class="sc">|</span> experiment_id),</span>
<span id="cb563-7"><a href="lmm.html#cb563-7" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb563-8"><a href="lmm.html#cb563-8" tabindex="-1"></a></span>
<span id="cb563-9"><a href="lmm.html#cb563-9" tabindex="-1"></a>exp1g_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb563-10"><a href="lmm.html#cb563-10" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb563-11"><a href="lmm.html#cb563-11" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype) <span class="sc">+</span></span>
<span id="cb563-12"><a href="lmm.html#cb563-12" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb563-13"><a href="lmm.html#cb563-13" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype<span class="sc">:</span>sex),</span>
<span id="cb563-14"><a href="lmm.html#cb563-14" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span>
<span id="cb563-15"><a href="lmm.html#cb563-15" tabindex="-1"></a></span>
<span id="cb563-16"><a href="lmm.html#cb563-16" tabindex="-1"></a>exp1g_m1d <span class="ot">&lt;-</span> <span class="fu">lmer</span>(gc <span class="sc">~</span> genotype <span class="sc">*</span> sex <span class="sc">+</span></span>
<span id="cb563-17"><a href="lmm.html#cb563-17" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> experiment_id) <span class="sc">+</span></span>
<span id="cb563-18"><a href="lmm.html#cb563-18" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>sex) <span class="sc">+</span></span>
<span id="cb563-19"><a href="lmm.html#cb563-19" tabindex="-1"></a>                    (<span class="dv">1</span> <span class="sc">|</span> experiment_id<span class="sc">:</span>genotype<span class="sc">:</span>sex),</span>
<span id="cb563-20"><a href="lmm.html#cb563-20" tabindex="-1"></a>                 <span class="at">data =</span> exp1g)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>All four models model the same fixed effects. The models differ only in how they model the random effects.</li>
<li>Model <code>exp1g_m1a</code> fits one random intercept, two random slopes, and one random interaction.</li>
</ol>
<ul>
<li><span class="math inline">\(\gamma_{0j}\)</span> – a random intercept modeling batch effects of <span class="math inline">\(\texttt{experiment_id}\)</span> on the intercept</li>
<li><span class="math inline">\(\gamma_{1j}\)</span> – a random slope modeling the effect of the non-reference level of “Gpr174-” on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is a <span class="math inline">\(\texttt{experiment_id} \times \texttt{genotype}\)</span> interaction.</li>
<li><span class="math inline">\(\gamma_{2j}\)</span> – a random slope modeling the effect of “F” (female) on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is a <span class="math inline">\(\texttt{experiment_id} \times \texttt{sex}\)</span> interaction.</li>
<li><span class="math inline">\(\gamma_{3j}\)</span> – a random slope modeling the effect of the interaction effect of “Gpr174-” and “F” on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is a <span class="math inline">\(\texttt{experiment_id} \times \texttt{sex} \times \texttt{genotype}\)</span> interaction.</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Model <code>exp1g_m1b</code> fits one random intercept and three random slopes</li>
</ol>
<ul>
<li><span class="math inline">\(\gamma_{0j}\)</span> – a random intercept modeling batch effects of <span class="math inline">\(\texttt{experiment_id}\)</span> on the intercept. This is modeling the same thing as <span class="math inline">\(\gamma_{0j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{1j}\)</span> – a random slope modeling the effect of “Gpr174- M” on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is modeling the same thing as <span class="math inline">\(\gamma_{1j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{2j}\)</span> – a random slope modeling the effect of “Gpr174+ F” on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is modeling the same thing as <span class="math inline">\(\gamma_{2j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{3j}\)</span> – a random slope modeling the effect of “M Gpr174- F” on the batch effect of <span class="math inline">\(\texttt{experiment_id}\)</span>. This is modeling the added variance accounted for by the random interaction <span class="math inline">\(\gamma_{2j}\)</span> in Model <code>exp1g_m1a</code> but in a different way.</li>
</ul>
<ol start="4" style="list-style-type: decimal">
<li>Model <code>exp1g_m1c</code> fits four random intercepts</li>
</ol>
<ul>
<li><span class="math inline">\(\gamma_{0j}\)</span> – a random intercept modeling batch effects of <span class="math inline">\(\texttt{experiment_id}\)</span> on the intercept. This is modeling the same thing as <span class="math inline">\(\gamma_{0j}\)</span> in Model <code>exp1g_m1a</code>.</li>
<li><span class="math inline">\(\gamma_{0jk}\)</span> – a random intercept modeling the effects of the combination of <span class="math inline">\(\texttt{experiment_id}\)</span> and <span class="math inline">\(\texttt{genotype}\)</span>. This is very similar to the variance modeled by <span class="math inline">\(\gamma_{1j}\)</span> in Model <code>exp1g_m1a</code> except the draws from <span class="math inline">\(\gamma_{0jk}\)</span> are independent (uncorrelated) of draws from the other random intercepts.</li>
<li><span class="math inline">\(\gamma_{0jl}\)</span> – a random intercept modeling the effects of the combination of <span class="math inline">\(\texttt{experiment_id}\)</span> and <span class="math inline">\(\texttt{sex}\)</span>. This is very similar to the variance modeled by <span class="math inline">\(\gamma_{2j}\)</span> in Model <code>exp1g_m1a</code> except the draws from <span class="math inline">\(\gamma_{0jl}\)</span> are independent (uncorrelated) of draws from the other random intercepts (review <a href="lmm.html#lmm-varcorr">The correlation among random intercepts and slopes</a> if this doesn’t make sense).</li>
<li><span class="math inline">\(\gamma_{0jkl}\)</span> – a random intercept modeling the effects of the combination of <span class="math inline">\(\texttt{experiment_id}\)</span>, <span class="math inline">\(\texttt{genotype}\)</span> and <span class="math inline">\(\texttt{sex}\)</span>. This is very similar to the variance modeled by <span class="math inline">\(\gamma_{3j}\)</span> in Model <code>exp1g_m1a</code> except the draws from <span class="math inline">\(\gamma_{0jkl}\)</span> are independent (uncorrelated) of draws from the other random intercepts.</li>
</ul>
<ol start="5" style="list-style-type: decimal">
<li>Model <code>exp1g_m1d</code> fits the same random intercepts as Model <code>exp1g_m1c</code> but excludes
<span class="math inline">\(\gamma_{0jl}\)</span> (the random intercept for the experiment_id by genotyp combination. This was excluded because of the low variance of this component in the fit model.</li>
</ol>
</div>
<div id="the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b" class="section level3 hasAnchor" number="16.8.8">
<h3><span class="header-section-number">16.8.8</span> The VarCorr matrix of models exp1g_m1a and exp1g_m1b<a href="lmm.html#the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The random effect similarity of models <code>exp1g_m1a</code> and <code>exp1g_m1b</code> can be seen in the estimated variance components and correlations among the random effects.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-238">Table 16.17: </span>The Varcorr matrix. Standard deviations of random effects on the diagonal. Correlations of random effects on the off-diagonal.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
</th>
<th style="text-align:right;">
</th>
<th style="text-align:right;">
</th>
<th style="text-align:right;">
</th>
</tr>
</thead>
<tbody>
<tr grouplength="4">
<td colspan="5" style="border-bottom: 1px solid;">
<strong>exp1g_m1a</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Intercept)
</td>
<td style="text-align:right;">
1.7708
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
genotypeGpr174-
</td>
<td style="text-align:right;">
-0.0232
</td>
<td style="text-align:right;">
0.9619
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
sexF
</td>
<td style="text-align:right;">
0.4589
</td>
<td style="text-align:right;">
0.8776
</td>
<td style="text-align:right;">
2.9015
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
genotypeGpr174-:sexF
</td>
<td style="text-align:right;">
-0.3138
</td>
<td style="text-align:right;">
-0.6246
</td>
<td style="text-align:right;">
-0.7056
</td>
<td style="text-align:right;">
1.3317
</td>
</tr>
<tr grouplength="4">
<td colspan="5" style="border-bottom: 1px solid;">
<strong>exp1g_m1b</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Intercept)
</td>
<td style="text-align:right;">
1.7710
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
treatmentGpr174- M
</td>
<td style="text-align:right;">
-0.0251
</td>
<td style="text-align:right;">
0.9640
</td>
<td style="text-align:right;">
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
treatmentGpr174+ F
</td>
<td style="text-align:right;">
0.4585
</td>
<td style="text-align:right;">
0.8761
</td>
<td style="text-align:right;">
2.9022
</td>
<td style="text-align:right;">
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
treatmentGpr174- F
</td>
<td style="text-align:right;">
0.2999
</td>
<td style="text-align:right;">
0.8791
</td>
<td style="text-align:right;">
0.9384
</td>
<td style="text-align:right;">
2.9905
</td>
</tr>
</tbody>
</table>
</div>
<div id="the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means" class="section level3 hasAnchor" number="16.8.9">
<h3><span class="header-section-number">16.8.9</span> The linear mixed model has more precision and power than the fixed effect model of batch means<a href="lmm.html#the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb564"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb564-1"><a href="lmm.html#cb564-1" tabindex="-1"></a><span class="co"># means pooling model</span></span>
<span id="cb564-2"><a href="lmm.html#cb564-2" tabindex="-1"></a>exp1g_m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(gc <span class="sc">~</span> sex <span class="sc">*</span> genotype,</span>
<span id="cb564-3"><a href="lmm.html#cb564-3" tabindex="-1"></a>                <span class="at">data =</span> exp1g_means)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp1g-lm2-planned">Table 16.18: </span>Planned contrasts for the linear mixed model exp1g_m1 and the fixed effects model of experiment means exp1g_m2.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp1g_m1 (lmm)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Gpr174- M) - (Gpr174+ M)
</td>
<td style="text-align:right;">
2.57
</td>
<td style="text-align:right;">
0.714
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
4.33
</td>
<td style="text-align:right;">
3.60
</td>
<td style="text-align:right;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Gpr174- F) - (Gpr174+ F)
</td>
<td style="text-align:right;">
-2.74
</td>
<td style="text-align:right;">
0.722
</td>
<td style="text-align:right;">
6.1
</td>
<td style="text-align:right;">
-4.50
</td>
<td style="text-align:right;">
-0.98
</td>
<td style="text-align:right;">
-3.79
</td>
<td style="text-align:right;">
0.009
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-5.30
</td>
<td style="text-align:right;">
1.016
</td>
<td style="text-align:right;">
5.9
</td>
<td style="text-align:right;">
-7.80
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
-5.22
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp1g_m2 (lm means pooling)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Gpr174- M) - (Gpr174+ M)
</td>
<td style="text-align:right;">
2.50
</td>
<td style="text-align:right;">
2.257
</td>
<td style="text-align:right;">
12.0
</td>
<td style="text-align:right;">
-2.41
</td>
<td style="text-align:right;">
7.42
</td>
<td style="text-align:right;">
1.11
</td>
<td style="text-align:right;">
0.289
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Gpr174- F) - (Gpr174+ F)
</td>
<td style="text-align:right;">
-2.73
</td>
<td style="text-align:right;">
2.257
</td>
<td style="text-align:right;">
12.0
</td>
<td style="text-align:right;">
-7.65
</td>
<td style="text-align:right;">
2.18
</td>
<td style="text-align:right;">
-1.21
</td>
<td style="text-align:right;">
0.249
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-5.24
</td>
<td style="text-align:right;">
3.192
</td>
<td style="text-align:right;">
12.0
</td>
<td style="text-align:right;">
-12.20
</td>
<td style="text-align:right;">
1.72
</td>
<td style="text-align:right;">
-1.64
</td>
<td style="text-align:right;">
0.127
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>A fixed effects model fit to batch-means pooled data is strongly conservative and will result in less discovery.</li>
<li>Means pooling does not make the data independent in a randomized complete block design. A linear mixed model of batch-means pooled data is a <strong>mixed-effect ANOVA</strong> (Next section. Also see Section <a href="issues.html#issues-exp4d">13.1.1</a> in the Issues chapter).</li>
</ol>
</div>
<div id="fixed-effect-models-and-pseudoreplication" class="section level3 hasAnchor" number="16.8.10">
<h3><span class="header-section-number">16.8.10</span> Fixed effect models and pseudoreplication<a href="lmm.html#fixed-effect-models-and-pseudoreplication" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb565"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb565-1"><a href="lmm.html#cb565-1" tabindex="-1"></a><span class="co"># complete pooling model</span></span>
<span id="cb565-2"><a href="lmm.html#cb565-2" tabindex="-1"></a>exp1g_m3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(gc <span class="sc">~</span> sex <span class="sc">*</span> genotype,</span>
<span id="cb565-3"><a href="lmm.html#cb565-3" tabindex="-1"></a>                <span class="at">data =</span> exp1g)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
(#tab:lmm-exp1g_m3-compare)Planned contrasts for the linear mixed model exp1g_m1 and the fixed effects model exp1g_m3 with complete pooling.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp1g_m1 (lmm)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Gpr174- M) - (Gpr174+ M)
</td>
<td style="text-align:right;">
2.57
</td>
<td style="text-align:right;">
0.714
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
4.33
</td>
<td style="text-align:right;">
3.60
</td>
<td style="text-align:right;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Gpr174- F) - (Gpr174+ F)
</td>
<td style="text-align:right;">
-2.74
</td>
<td style="text-align:right;">
0.722
</td>
<td style="text-align:right;">
6.1
</td>
<td style="text-align:right;">
-4.50
</td>
<td style="text-align:right;">
-0.98
</td>
<td style="text-align:right;">
-3.79
</td>
<td style="text-align:right;">
0.009
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-5.30
</td>
<td style="text-align:right;">
1.016
</td>
<td style="text-align:right;">
5.9
</td>
<td style="text-align:right;">
-7.80
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
-5.22
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp1g_m3 (lm complete pooling)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Gpr174- M) - (Gpr174+ M)
</td>
<td style="text-align:right;">
2.44
</td>
<td style="text-align:right;">
1.111
</td>
<td style="text-align:right;">
69.0
</td>
<td style="text-align:right;">
0.22
</td>
<td style="text-align:right;">
4.65
</td>
<td style="text-align:right;">
2.20
</td>
<td style="text-align:right;">
0.032
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Gpr174- F) - (Gpr174+ F)
</td>
<td style="text-align:right;">
-2.53
</td>
<td style="text-align:right;">
1.125
</td>
<td style="text-align:right;">
69.0
</td>
<td style="text-align:right;">
-4.77
</td>
<td style="text-align:right;">
-0.28
</td>
<td style="text-align:right;">
-2.25
</td>
<td style="text-align:right;">
0.028
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-4.97
</td>
<td style="text-align:right;">
1.581
</td>
<td style="text-align:right;">
69.0
</td>
<td style="text-align:right;">
-8.12
</td>
<td style="text-align:right;">
-1.81
</td>
<td style="text-align:right;">
-3.14
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Complete pooling is strongly anti-conservative and will result in increased false discovery. Complete pooling is a type of pseudoreplication – the subsamples have been analyzed as if they are independent replicates. Subsamples are not independent.</li>
<li>For the Experiment 1g data, the 95% confidence intervals are wider and the <em>p</em>-values are larger in the complete pool model <code>exp1g_m3</code> compared to the linear mixed model <code>exp1g_m1</code>. This is an unusual result.</li>
</ol>
</div>
<div id="mixed-effect-anova" class="section level3 hasAnchor" number="16.8.11">
<h3><span class="header-section-number">16.8.11</span> Mixed-effect ANOVA<a href="lmm.html#mixed-effect-anova" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb566"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb566-1"><a href="lmm.html#cb566-1" tabindex="-1"></a>exp1g_m1_aov <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(gc <span class="sc">~</span> sex <span class="sc">*</span> genotype <span class="sc">+</span></span>
<span id="cb566-2"><a href="lmm.html#cb566-2" tabindex="-1"></a>                  (sex <span class="sc">*</span> genotype <span class="sc">|</span> experiment_id),</span>
<span id="cb566-3"><a href="lmm.html#cb566-3" tabindex="-1"></a>                <span class="at">data =</span> exp1g)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The formula has the same format as that in Example 3 for repeated measures ANOVA on RCB designs with no subsampling. In biology, this is often called a <strong>mixed effect ANOVA</strong> with two fixed factors and one random factor.</li>
<li>The data are aggregated prior to fitting the model – this means the subsamples are averaged within each batch by treatment combination.</li>
<li>The mixed-effect ANOVA is equivalent to the linear mixed model <code>exp1g_m1c</code> if there are the same number of replicates in each treatment combination and the same number of subsamples in all treatment by <span class="math inline">\(\texttt{experiment_id}\)</span> combinations. The design is not balanced in this example.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
(#tab:lmm-exp1g_m1_aov-pairs)Planned contrasts from mixed ANOVA compared to lmm equivalent of mixed ANOVA and lowest AIC lmm.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp1g_m1_aov (mixed ANOVA)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Gpr174- M) - (Gpr174+ M)
</td>
<td style="text-align:right;">
2.50
</td>
<td style="text-align:right;">
0.599
</td>
<td style="text-align:right;">
3.000
</td>
<td style="text-align:right;">
0.60
</td>
<td style="text-align:right;">
4.41
</td>
<td style="text-align:right;">
4.18
</td>
<td style="text-align:right;">
0.025
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Gpr174- F) - (Gpr174+ F)
</td>
<td style="text-align:right;">
-2.73
</td>
<td style="text-align:right;">
0.808
</td>
<td style="text-align:right;">
3.000
</td>
<td style="text-align:right;">
-5.30
</td>
<td style="text-align:right;">
-0.16
</td>
<td style="text-align:right;">
-3.39
</td>
<td style="text-align:right;">
0.043
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-5.24
</td>
<td style="text-align:right;">
1.087
</td>
<td style="text-align:right;">
3.000
</td>
<td style="text-align:right;">
-8.70
</td>
<td style="text-align:right;">
-1.78
</td>
<td style="text-align:right;">
-4.82
</td>
<td style="text-align:right;">
0.017
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp1g_m1_c (lmm equivalent of mixed ANOVA)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Gpr174- M) - (Gpr174+ M)
</td>
<td style="text-align:right;">
2.57
</td>
<td style="text-align:right;">
0.715
</td>
<td style="text-align:right;">
5.784
</td>
<td style="text-align:right;">
0.80
</td>
<td style="text-align:right;">
4.33
</td>
<td style="text-align:right;">
3.59
</td>
<td style="text-align:right;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Gpr174- F) - (Gpr174+ F)
</td>
<td style="text-align:right;">
-2.74
</td>
<td style="text-align:right;">
0.724
</td>
<td style="text-align:right;">
6.100
</td>
<td style="text-align:right;">
-4.50
</td>
<td style="text-align:right;">
-0.97
</td>
<td style="text-align:right;">
-3.78
</td>
<td style="text-align:right;">
0.009
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-5.30
</td>
<td style="text-align:right;">
1.017
</td>
<td style="text-align:right;">
2.972
</td>
<td style="text-align:right;">
-8.56
</td>
<td style="text-align:right;">
-2.05
</td>
<td style="text-align:right;">
-5.21
</td>
<td style="text-align:right;">
0.014
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp1g_md (lmm with min AIC)</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Gpr174- M) - (Gpr174+ M)
</td>
<td style="text-align:right;">
2.57
</td>
<td style="text-align:right;">
0.714
</td>
<td style="text-align:right;">
5.784
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
4.33
</td>
<td style="text-align:right;">
3.60
</td>
<td style="text-align:right;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Gpr174- F) - (Gpr174+ F)
</td>
<td style="text-align:right;">
-2.74
</td>
<td style="text-align:right;">
0.722
</td>
<td style="text-align:right;">
6.100
</td>
<td style="text-align:right;">
-4.50
</td>
<td style="text-align:right;">
-0.98
</td>
<td style="text-align:right;">
-3.79
</td>
<td style="text-align:right;">
0.009
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-5.30
</td>
<td style="text-align:right;">
1.016
</td>
<td style="text-align:right;">
5.942
</td>
<td style="text-align:right;">
-7.80
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
-5.22
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="statistical-models-for-experimental-designs" class="section level2 hasAnchor" number="16.9">
<h2><span class="header-section-number">16.9</span> Statistical models for experimental designs<a href="lmm.html#statistical-models-for-experimental-designs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="models-for-completely-randomized-designs-crd" class="section level3 hasAnchor" number="16.9.1">
<h3><span class="header-section-number">16.9.1</span> Models for Completely Randomized Designs (CRD)<a href="lmm.html#models-for-completely-randomized-designs-crd" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb567"><pre class="sourceCode r fold-show"><code class="sourceCode r"><span id="cb567-1"><a href="lmm.html#cb567-1" tabindex="-1"></a>lm0 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> treatment,</span>
<span id="cb567-2"><a href="lmm.html#cb567-2" tabindex="-1"></a>             <span class="at">data =</span> figx)</span></code></pre></div>
</div>
<div id="models-for-batched-data-crds-rcbd-rspd-grcbd-nrcbd" class="section level3 hasAnchor" number="16.9.2">
<h3><span class="header-section-number">16.9.2</span> Models for batched data (CRDS, RCBD, RSPD, GRCBD, NRCBD)<a href="lmm.html#models-for-batched-data-crds-rcbd-rspd-grcbd-nrcbd" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="linear-models" class="section level4 hasAnchor" number="16.9.2.1">
<h4><span class="header-section-number">16.9.2.1</span> Linear models<a href="lmm.html#linear-models" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="fixed-effect-model" class="section level5 hasAnchor" number="16.9.2.1.1">
<h5><span class="header-section-number">16.9.2.1.1</span> fixed effect model<a href="lmm.html#fixed-effect-model" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb568"><pre class="sourceCode r fold-show"><code class="sourceCode r"><span id="cb568-1"><a href="lmm.html#cb568-1" tabindex="-1"></a>lm1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> block,</span>
<span id="cb568-2"><a href="lmm.html#cb568-2" tabindex="-1"></a>             <span class="at">data =</span> figx)</span></code></pre></div>
</div>
<div id="linear-model-of-batch-means" class="section level5 hasAnchor" number="16.9.2.1.2">
<h5><span class="header-section-number">16.9.2.1.2</span> linear model of batch means<a href="lmm.html#linear-model-of-batch-means" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb569"><pre class="sourceCode r fold-show"><code class="sourceCode r"><span id="cb569-1"><a href="lmm.html#cb569-1" tabindex="-1"></a>figx_means <span class="ot">&lt;-</span> figx[, .(<span class="at">y =</span> <span class="fu">mean</span>(y)),</span>
<span id="cb569-2"><a href="lmm.html#cb569-2" tabindex="-1"></a>               by <span class="ot">=</span> .(treatment, batch)]</span>
<span id="cb569-3"><a href="lmm.html#cb569-3" tabindex="-1"></a>lm2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> treatment,</span>
<span id="cb569-4"><a href="lmm.html#cb569-4" tabindex="-1"></a>             <span class="at">data =</span> figx_means)</span></code></pre></div>
</div>
</div>
<div id="linear-mixed-models-random-effects-models" class="section level4 hasAnchor" number="16.9.2.2">
<h4><span class="header-section-number">16.9.2.2</span> linear mixed models (“random effects” models)<a href="lmm.html#linear-mixed-models-random-effects-models" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="random-intercept-model" class="section level5 hasAnchor" number="16.9.2.2.1">
<h5><span class="header-section-number">16.9.2.2.1</span> Random intercept model<a href="lmm.html#random-intercept-model" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb570"><pre class="sourceCode r fold-show"><code class="sourceCode r"><span id="cb570-1"><a href="lmm.html#cb570-1" tabindex="-1"></a>lmm1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block),</span>
<span id="cb570-2"><a href="lmm.html#cb570-2" tabindex="-1"></a>              <span class="at">data =</span> figx)</span></code></pre></div>
</div>
<div id="lmm-for-correlated-error" class="section level5 hasAnchor" number="16.9.2.2.2">
<h5><span class="header-section-number">16.9.2.2.2</span> lmm for correlated error<a href="lmm.html#lmm-for-correlated-error" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb571"><pre class="sourceCode r fold-show"><code class="sourceCode r"><span id="cb571-1"><a href="lmm.html#cb571-1" tabindex="-1"></a>lmm2 <span class="ot">&lt;-</span> <span class="fu">lme</span>(y <span class="sc">~</span> treatment,</span>
<span id="cb571-2"><a href="lmm.html#cb571-2" tabindex="-1"></a>             <span class="at">random =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> block,</span>
<span id="cb571-3"><a href="lmm.html#cb571-3" tabindex="-1"></a>             <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> block),</span>
<span id="cb571-4"><a href="lmm.html#cb571-4" tabindex="-1"></a>             <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> treatment),</span>
<span id="cb571-5"><a href="lmm.html#cb571-5" tabindex="-1"></a>             <span class="at">data =</span> figx)</span></code></pre></div>
</div>
<div id="random-intercept-and-slope-model" class="section level5 hasAnchor" number="16.9.2.2.3">
<h5><span class="header-section-number">16.9.2.2.3</span> random intercept and slope model<a href="lmm.html#random-intercept-and-slope-model" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb572"><pre class="sourceCode r fold-show"><code class="sourceCode r"><span id="cb572-1"><a href="lmm.html#cb572-1" tabindex="-1"></a>lmm3 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (treatment <span class="sc">|</span> block),</span>
<span id="cb572-2"><a href="lmm.html#cb572-2" tabindex="-1"></a>               <span class="at">data =</span> figx)</span></code></pre></div>
</div>
<div id="random-interaction-intercept-model" class="section level5 hasAnchor" number="16.9.2.2.4">
<h5><span class="header-section-number">16.9.2.2.4</span> random interaction intercept model<a href="lmm.html#random-interaction-intercept-model" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb573"><pre class="sourceCode r fold-show"><code class="sourceCode r"><span id="cb573-1"><a href="lmm.html#cb573-1" tabindex="-1"></a>lmm4 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>treatment),</span>
<span id="cb573-2"><a href="lmm.html#cb573-2" tabindex="-1"></a>               <span class="at">data =</span> figx)</span></code></pre></div>
</div>
<div id="random-interaction-model-with-subsampling" class="section level5 hasAnchor" number="16.9.2.2.5">
<h5><span class="header-section-number">16.9.2.2.5</span> random interaction model with subsampling<a href="lmm.html#random-interaction-model-with-subsampling" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb574"><pre class="sourceCode r fold-show"><code class="sourceCode r"><span id="cb574-1"><a href="lmm.html#cb574-1" tabindex="-1"></a>lmm5 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>treatment) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>treatment<span class="sc">:</span>replicate),</span>
<span id="cb574-2"><a href="lmm.html#cb574-2" tabindex="-1"></a>               <span class="at">data =</span> figx)</span></code></pre></div>
</div>
<div id="crd-split-plot-model" class="section level5 hasAnchor" number="16.9.2.2.6">
<h5><span class="header-section-number">16.9.2.2.6</span> CRD split plot model<a href="lmm.html#crd-split-plot-model" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb575"><pre class="sourceCode r fold-show"><code class="sourceCode r"><span id="cb575-1"><a href="lmm.html#cb575-1" tabindex="-1"></a><span class="co"># tr1 is the main plot</span></span>
<span id="cb575-2"><a href="lmm.html#cb575-2" tabindex="-1"></a><span class="co"># tr2 is the subplot</span></span>
<span id="cb575-3"><a href="lmm.html#cb575-3" tabindex="-1"></a><span class="co"># main_plot is tr1:rep, where rep is the rep id</span></span>
<span id="cb575-4"><a href="lmm.html#cb575-4" tabindex="-1"></a>lmm6 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> tr1 <span class="sc">*</span> tr2 <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> main_plot),</span>
<span id="cb575-5"><a href="lmm.html#cb575-5" tabindex="-1"></a>               <span class="at">data =</span> figx)</span></code></pre></div>
</div>
<div id="rcbd-split-plot-model" class="section level5 hasAnchor" number="16.9.2.2.7">
<h5><span class="header-section-number">16.9.2.2.7</span> RCBD split plot model<a href="lmm.html#rcbd-split-plot-model" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb576"><pre class="sourceCode r fold-show"><code class="sourceCode r"><span id="cb576-1"><a href="lmm.html#cb576-1" tabindex="-1"></a><span class="co"># tr1 is the main plot</span></span>
<span id="cb576-2"><a href="lmm.html#cb576-2" tabindex="-1"></a><span class="co"># tr2 is the subplot</span></span>
<span id="cb576-3"><a href="lmm.html#cb576-3" tabindex="-1"></a><span class="co"># main_plot is tr1:block</span></span>
<span id="cb576-4"><a href="lmm.html#cb576-4" tabindex="-1"></a>lmm7 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> tr1 <span class="sc">*</span> tr2 <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>tr1),</span>
<span id="cb576-5"><a href="lmm.html#cb576-5" tabindex="-1"></a>               <span class="at">data =</span> figx)</span>
<span id="cb576-6"><a href="lmm.html#cb576-6" tabindex="-1"></a></span>
<span id="cb576-7"><a href="lmm.html#cb576-7" tabindex="-1"></a>lmm7 <span class="ot">&lt;-</span>  <span class="fu">lmer</span>(y <span class="sc">~</span> tr1 <span class="sc">*</span> tr2 <span class="sc">+</span> block <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>tr1),</span>
<span id="cb576-8"><a href="lmm.html#cb576-8" tabindex="-1"></a>               <span class="at">data =</span> figx)</span>
<span id="cb576-9"><a href="lmm.html#cb576-9" tabindex="-1"></a><span class="co"># the two versions are numerically equivalent</span></span></code></pre></div>
</div>
</div>
<div id="anova-models-mixed-models-repeated-measures-anova" class="section level4 hasAnchor" number="16.9.2.3">
<h4><span class="header-section-number">16.9.2.3</span> ANOVA models (“mixed models”, “repeated measures ANOVA”)<a href="lmm.html#anova-models-mixed-models-repeated-measures-anova" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="multivariate-repeated-measures-anova" class="section level5 hasAnchor" number="16.9.2.3.1">
<h5><span class="header-section-number">16.9.2.3.1</span> Multivariate repeated measures ANOVA<a href="lmm.html#multivariate-repeated-measures-anova" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb577"><pre class="sourceCode r fold-show"><code class="sourceCode r"><span id="cb577-1"><a href="lmm.html#cb577-1" tabindex="-1"></a>aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (treatment <span class="sc">|</span> block),</span>
<span id="cb577-2"><a href="lmm.html#cb577-2" tabindex="-1"></a>             <span class="at">data =</span> figx)</span></code></pre></div>
</div>
<div id="univariate-repeated-measures-anova" class="section level5 hasAnchor" number="16.9.2.3.2">
<h5><span class="header-section-number">16.9.2.3.2</span> Univariate repeated measures ANOVA<a href="lmm.html#univariate-repeated-measures-anova" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb578"><pre class="sourceCode r fold-show"><code class="sourceCode r"><span id="cb578-1"><a href="lmm.html#cb578-1" tabindex="-1"></a>aov2 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(y <span class="sc">~</span> treatment <span class="sc">+</span> (treatment <span class="sc">|</span> block),</span>
<span id="cb578-2"><a href="lmm.html#cb578-2" tabindex="-1"></a>           <span class="at">include_aov =</span> <span class="cn">TRUE</span>,</span>
<span id="cb578-3"><a href="lmm.html#cb578-3" tabindex="-1"></a>             <span class="at">data =</span> figx)</span></code></pre></div>
</div>
<div id="pairwise-paired-t-tests" class="section level5 hasAnchor" number="16.9.2.3.3">
<h5><span class="header-section-number">16.9.2.3.3</span> Pairwise paired, t-tests<a href="lmm.html#pairwise-paired-t-tests" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb579"><pre class="sourceCode r fold-show"><code class="sourceCode r"><span id="cb579-1"><a href="lmm.html#cb579-1" tabindex="-1"></a>pptt <span class="ot">&lt;-</span> <span class="fu">pairwise_t_tests</span>(<span class="at">y_col =</span> <span class="st">&quot;y&quot;</span>,</span>
<span id="cb579-2"><a href="lmm.html#cb579-2" tabindex="-1"></a>                         <span class="at">g_col =</span> <span class="st">&quot;treatment&quot;</span>,</span>
<span id="cb579-3"><a href="lmm.html#cb579-3" tabindex="-1"></a>                         <span class="at">id_col =</span> <span class="st">&quot;block&quot;</span>,</span>
<span id="cb579-4"><a href="lmm.html#cb579-4" tabindex="-1"></a>                         <span class="at">data =</span> figx)</span></code></pre></div>
</div>
</div>
</div>
</div>
<div id="which-model-and-why" class="section level2 hasAnchor" number="16.10">
<h2><span class="header-section-number">16.10</span> Which model and why?<a href="lmm.html#which-model-and-why" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="crd" class="section level3 hasAnchor" number="16.10.1">
<h3><span class="header-section-number">16.10.1</span> CRD<a href="lmm.html#crd" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>No batch to model!</li>
</ol>
</div>
<div id="crds" class="section level3 hasAnchor" number="16.10.2">
<h3><span class="header-section-number">16.10.2</span> CRDS<a href="lmm.html#crds" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>lmm1 – random intercept model</li>
<li>lm2 – linear model of batch means</li>
</ol>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>These are numerically equivalent and should result in same estimates, SE, CIs, and p-values</li>
<li>These are equivalent to a <strong>Nested t-test</strong> or <strong>Nested ANOVA</strong></li>
</ol>
</div>
<div id="rcbd" class="section level3 hasAnchor" number="16.10.3">
<h3><span class="header-section-number">16.10.3</span> RCBD<a href="lmm.html#rcbd" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Reasonable models:</p>
<ol style="list-style-type: decimal">
<li>lmm1 – random intercept model</li>
<li>lmm2 – lmm for correlated error</li>
<li>lm1 – fixed effect model</li>
<li>aov1 – multivariate RM-ANOVA</li>
<li>aov2 – univariate RM-ANOVA</li>
<li>pptt – pairwise, paired t-test</li>
</ol>
<p>(Cannot use lmm3 or lmm4 because there is no replication of each block:treatment combination.)</p>
<p>Assumptions:</p>
<ol style="list-style-type: decimal">
<li>lmm1, lm1, and aov2 assume
<ul>
<li><strong>compound symmetry</strong>.</li>
<li><strong>sphericity</strong>.</li>
</ul></li>
<li>lmm2, aov1, and pptt do not assume compound symmetry and sphericity.</li>
</ol>
<p>Notes:</p>
<ol style="list-style-type: decimal">
<li>lmm1 is the standard. If the number of treatments = 2, then lmm1 is eqivalent with a paired t-test.</li>
<li>If the design is balanced (all blocks have a single value for both treatments) AND the number of treatments = 2 then <em>all six methods are numerically equivalent</em>.</li>
<li>If the design is balanced AND the number of treatments &gt; 2, then
<ul>
<li>lmm1, lm1, and aov2 are numerically equivalent</li>
<li>lmm2, aov1, pptt result in the same estimates and SE but lmm2 has more df, so the CIs, and p-values of lmm2 are less conservative.</li>
</ul></li>
<li>If the number of treatments = 2 AND the design is not balanced (at least one block is missing value for one treatment level) then the linear models (lm1, aov1, aov2, pptt) are equivalent but the linear mixed models (lmm1, lmm2) differ from each other and from the linear models.</li>
<li>If a treatment within a block is missing, the whole block is deleted in the RM-ANOVA models. This reduces power (the loss of power depends partly on the number of missing values).</li>
<li>If a treatment within a block is missing, the block is deleted only in the comparisons including the missing treatment in the pairwise, paired t-tests. This makes pptt more powerful than aov1.</li>
<li>the lmm models are very flexible – covariates can be added or these can be used as generalized lmms for non-normal distributions.</li>
<li>the lmm models and especially lmm2 sometimes (often?) fail to converge, especially with small samples (small number of blocks) or if the among-block component of variance is small.</li>
</ol>
<p>Performance:</p>
<div class="sourceCode" id="cb580"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb580-1"><a href="lmm.html#cb580-1" tabindex="-1"></a>seed_starter <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb580-2"><a href="lmm.html#cb580-2" tabindex="-1"></a>n_sims <span class="ot">&lt;-</span> <span class="dv">1000</span></span>
<span id="cb580-3"><a href="lmm.html#cb580-3" tabindex="-1"></a>do_sim <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb580-4"><a href="lmm.html#cb580-4" tabindex="-1"></a></span>
<span id="cb580-5"><a href="lmm.html#cb580-5" tabindex="-1"></a>  data_from <span class="ot">&lt;-</span> <span class="st">&quot;simulations/grcbds/data&quot;</span></span>
<span id="cb580-6"><a href="lmm.html#cb580-6" tabindex="-1"></a>  outfile_name <span class="ot">&lt;-</span> <span class="st">&quot;rcbd.Rds&quot;</span></span>
<span id="cb580-7"><a href="lmm.html#cb580-7" tabindex="-1"></a>  save_file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_from, outfile_name)</span>
<span id="cb580-8"><a href="lmm.html#cb580-8" tabindex="-1"></a></span>
<span id="cb580-9"><a href="lmm.html#cb580-9" tabindex="-1"></a></span>
<span id="cb580-10"><a href="lmm.html#cb580-10" tabindex="-1"></a>param_sets <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb580-11"><a href="lmm.html#cb580-11" tabindex="-1"></a>treatment_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Cn&quot;</span>, <span class="st">&quot;Tr1&quot;</span>, <span class="st">&quot;Tr2&quot;</span>)</span>
<span id="cb580-12"><a href="lmm.html#cb580-12" tabindex="-1"></a>model_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;lm0&quot;</span>, <span class="st">&quot;lm1&quot;</span>, <span class="st">&quot;aov1&quot;</span>, <span class="st">&quot;aov2&quot;</span>, <span class="st">&quot;pptt&quot;</span>, <span class="st">&quot;lmm2&quot;</span>)</span>
<span id="cb580-13"><a href="lmm.html#cb580-13" tabindex="-1"></a>  </span>
<span id="cb580-14"><a href="lmm.html#cb580-14" tabindex="-1"></a>p1_mat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="cn">NA</span>), <span class="at">nrow =</span> n_sims, <span class="at">ncol =</span> <span class="fu">length</span>(model_levels))</span>
<span id="cb580-15"><a href="lmm.html#cb580-15" tabindex="-1"></a>p2_mat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="cn">NA</span>), <span class="at">nrow =</span> n_sims, <span class="at">ncol =</span> <span class="fu">length</span>(model_levels))</span>
<span id="cb580-16"><a href="lmm.html#cb580-16" tabindex="-1"></a>p3_mat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="cn">NA</span>), <span class="at">nrow =</span> n_sims, <span class="at">ncol =</span> <span class="fu">length</span>(model_levels))</span>
<span id="cb580-17"><a href="lmm.html#cb580-17" tabindex="-1"></a>r_mat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="cn">NA</span>), <span class="at">nrow =</span> n_sims, <span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb580-18"><a href="lmm.html#cb580-18" tabindex="-1"></a></span>
<span id="cb580-19"><a href="lmm.html#cb580-19" tabindex="-1"></a><span class="co"># parameters</span></span>
<span id="cb580-20"><a href="lmm.html#cb580-20" tabindex="-1"></a>beta_i <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">0</span>, <span class="dv">0</span>)</span>
<span id="cb580-21"><a href="lmm.html#cb580-21" tabindex="-1"></a></span>
<span id="cb580-22"><a href="lmm.html#cb580-22" tabindex="-1"></a><span class="co"># each row is different param_set</span></span>
<span id="cb580-23"><a href="lmm.html#cb580-23" tabindex="-1"></a>sigma_exp.block.treat_matrix <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(</span>
<span id="cb580-24"><a href="lmm.html#cb580-24" tabindex="-1"></a>  <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>, <span class="fl">0.2</span>),</span>
<span id="cb580-25"><a href="lmm.html#cb580-25" tabindex="-1"></a>  <span class="fu">c</span>(<span class="fl">0.5</span>, .<span class="dv">25</span>, <span class="fl">0.05</span>),</span>
<span id="cb580-26"><a href="lmm.html#cb580-26" tabindex="-1"></a>  <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>),</span>
<span id="cb580-27"><a href="lmm.html#cb580-27" tabindex="-1"></a>  <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>)</span>
<span id="cb580-28"><a href="lmm.html#cb580-28" tabindex="-1"></a>), <span class="at">nrow =</span> <span class="dv">3</span>, <span class="at">ncol =</span> <span class="dv">4</span>) <span class="sc">%&gt;%</span></span>
<span id="cb580-29"><a href="lmm.html#cb580-29" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb580-30"><a href="lmm.html#cb580-30" tabindex="-1"></a><span class="fu">colnames</span>(sigma_exp.block.treat_matrix) <span class="ot">&lt;-</span> treatment_levels</span>
<span id="cb580-31"><a href="lmm.html#cb580-31" tabindex="-1"></a></span>
<span id="cb580-32"><a href="lmm.html#cb580-32" tabindex="-1"></a><span class="co"># each value is different param_set</span></span>
<span id="cb580-33"><a href="lmm.html#cb580-33" tabindex="-1"></a>sigma_exp.block_list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">0.25</span>, <span class="fl">0.8</span>, <span class="dv">0</span>)</span>
<span id="cb580-34"><a href="lmm.html#cb580-34" tabindex="-1"></a></span>
<span id="cb580-35"><a href="lmm.html#cb580-35" tabindex="-1"></a><span class="co"># each value is different param_set</span></span>
<span id="cb580-36"><a href="lmm.html#cb580-36" tabindex="-1"></a>sigma_ss_list <span class="ot">&lt;-</span> <span class="fu">c</span>(.<span class="dv">1</span>, .<span class="dv">4</span>, .<span class="dv">4</span>, <span class="dv">1</span>)</span>
<span id="cb580-37"><a href="lmm.html#cb580-37" tabindex="-1"></a></span>
<span id="cb580-38"><a href="lmm.html#cb580-38" tabindex="-1"></a>param_set_matrix <span class="ot">&lt;-</span> <span class="fu">data.table</span>(</span>
<span id="cb580-39"><a href="lmm.html#cb580-39" tabindex="-1"></a>  <span class="at">sim_id =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(sigma_exp.block_list),</span>
<span id="cb580-40"><a href="lmm.html#cb580-40" tabindex="-1"></a>  <span class="at">sigma_block =</span> sigma_exp.block_list,</span>
<span id="cb580-41"><a href="lmm.html#cb580-41" tabindex="-1"></a>  sigma_exp.block.treat_matrix,</span>
<span id="cb580-42"><a href="lmm.html#cb580-42" tabindex="-1"></a>  <span class="at">sigma_ss =</span> sigma_ss_list</span>
<span id="cb580-43"><a href="lmm.html#cb580-43" tabindex="-1"></a>)</span>
<span id="cb580-44"><a href="lmm.html#cb580-44" tabindex="-1"></a></span>
<span id="cb580-45"><a href="lmm.html#cb580-45" tabindex="-1"></a>res_table <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="cn">NA</span>), <span class="at">nrow =</span> <span class="fu">nrow</span>(param_set_matrix), <span class="at">ncol =</span> <span class="dv">15</span>)</span>
<span id="cb580-46"><a href="lmm.html#cb580-46" tabindex="-1"></a>sim_table <span class="ot">&lt;-</span> <span class="fu">data.table</span>(<span class="cn">NULL</span>)</span>
<span id="cb580-47"><a href="lmm.html#cb580-47" tabindex="-1"></a>big_table <span class="ot">&lt;-</span> <span class="fu">data.table</span>(<span class="cn">NULL</span>)</span>
<span id="cb580-48"><a href="lmm.html#cb580-48" tabindex="-1"></a></span>
<span id="cb580-49"><a href="lmm.html#cb580-49" tabindex="-1"></a><span class="cf">if</span>(do_sim <span class="sc">==</span> <span class="cn">TRUE</span>){</span>
<span id="cb580-50"><a href="lmm.html#cb580-50" tabindex="-1"></a>  <span class="cf">for</span>(param_set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(param_set_matrix)){</span>
<span id="cb580-51"><a href="lmm.html#cb580-51" tabindex="-1"></a>    sigma_exp.block_i <span class="ot">&lt;-</span> param_set_matrix[param_set, sigma_block]</span>
<span id="cb580-52"><a href="lmm.html#cb580-52" tabindex="-1"></a>    sigma_exp.block.treat_i <span class="ot">&lt;-</span> param_set_matrix[param_set, .SD, .SDcols <span class="ot">=</span> treatment_levels] <span class="sc">%&gt;%</span> <span class="fu">as.numeric</span>()</span>
<span id="cb580-53"><a href="lmm.html#cb580-53" tabindex="-1"></a>    sigma_ss_i <span class="ot">&lt;-</span> param_set_matrix[param_set, sigma_ss]</span>
<span id="cb580-54"><a href="lmm.html#cb580-54" tabindex="-1"></a>    sigma_rep_i <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb580-55"><a href="lmm.html#cb580-55" tabindex="-1"></a>    sigma_exp.treat_i <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb580-56"><a href="lmm.html#cb580-56" tabindex="-1"></a>    sigma_exp_i <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb580-57"><a href="lmm.html#cb580-57" tabindex="-1"></a>    n_treat_i <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb580-58"><a href="lmm.html#cb580-58" tabindex="-1"></a>    n_block_i <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb580-59"><a href="lmm.html#cb580-59" tabindex="-1"></a>    n_rep_i <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb580-60"><a href="lmm.html#cb580-60" tabindex="-1"></a>    n_ss_i <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb580-61"><a href="lmm.html#cb580-61" tabindex="-1"></a>    fake_rcbd <span class="ot">&lt;-</span> <span class="fu">simulator</span>(</span>
<span id="cb580-62"><a href="lmm.html#cb580-62" tabindex="-1"></a>      <span class="at">seed_i =</span> seed_starter,</span>
<span id="cb580-63"><a href="lmm.html#cb580-63" tabindex="-1"></a>      <span class="at">n_sim =</span> n_sims,</span>
<span id="cb580-64"><a href="lmm.html#cb580-64" tabindex="-1"></a>      <span class="at">n_treat =</span> n_treat_i, <span class="co"># number of treatment levels</span></span>
<span id="cb580-65"><a href="lmm.html#cb580-65" tabindex="-1"></a>      <span class="at">n_block =</span> n_block_i, <span class="co"># number of litters (blocks)</span></span>
<span id="cb580-66"><a href="lmm.html#cb580-66" tabindex="-1"></a>      <span class="at">n_rep =</span> n_rep_i, <span class="co"># number of mice per litter:treatment</span></span>
<span id="cb580-67"><a href="lmm.html#cb580-67" tabindex="-1"></a>      <span class="at">n_ss =</span> n_ss_i, <span class="co"># number of subsamples</span></span>
<span id="cb580-68"><a href="lmm.html#cb580-68" tabindex="-1"></a>      <span class="at">design =</span> <span class="st">&quot;rcbd&quot;</span>,</span>
<span id="cb580-69"><a href="lmm.html#cb580-69" tabindex="-1"></a>      <span class="at">correlated_slopes =</span> <span class="cn">FALSE</span>, <span class="co"># TRUE uses random int/slope model to generate data</span></span>
<span id="cb580-70"><a href="lmm.html#cb580-70" tabindex="-1"></a>      <span class="at">beta =</span> beta_i,</span>
<span id="cb580-71"><a href="lmm.html#cb580-71" tabindex="-1"></a>      <span class="at">gamma =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>), <span class="co"># sd of random intercept and slopes for non-ref</span></span>
<span id="cb580-72"><a href="lmm.html#cb580-72" tabindex="-1"></a>      <span class="at">rho =</span> <span class="fl">0.0</span>, <span class="co"># r between random intercept and slopes</span></span>
<span id="cb580-73"><a href="lmm.html#cb580-73" tabindex="-1"></a>      <span class="at">sigma_exp =</span> sigma_exp_i, <span class="co"># sd among experiments</span></span>
<span id="cb580-74"><a href="lmm.html#cb580-74" tabindex="-1"></a>      <span class="at">sigma_exp.block =</span> sigma_exp.block_i, <span class="co"># sd among exp:block (or block if n_exp = 1)</span></span>
<span id="cb580-75"><a href="lmm.html#cb580-75" tabindex="-1"></a>      <span class="at">sigma_exp.treat =</span> sigma_exp.treat_i, <span class="co"># sd among exp:treat</span></span>
<span id="cb580-76"><a href="lmm.html#cb580-76" tabindex="-1"></a>      <span class="at">sigma_exp.block.treat =</span> sigma_exp.block.treat_i, <span class="co"># sd among exp:block:treat</span></span>
<span id="cb580-77"><a href="lmm.html#cb580-77" tabindex="-1"></a>      <span class="at">sigma_rep =</span> sigma_rep_i,</span>
<span id="cb580-78"><a href="lmm.html#cb580-78" tabindex="-1"></a>      <span class="at">sigma_ss =</span> sigma_ss_i, <span class="co"># sd among subsamples within replication of treatment:block</span></span>
<span id="cb580-79"><a href="lmm.html#cb580-79" tabindex="-1"></a>      <span class="co"># sigma_exp = 0, # sd among experiments</span></span>
<span id="cb580-80"><a href="lmm.html#cb580-80" tabindex="-1"></a>      <span class="co"># sigma_exp.block = 0.1, # sd among exp:block (or block if n_exp = 1)</span></span>
<span id="cb580-81"><a href="lmm.html#cb580-81" tabindex="-1"></a>      <span class="co"># sigma_exp.treat = 0, # sd among exp:treat</span></span>
<span id="cb580-82"><a href="lmm.html#cb580-82" tabindex="-1"></a>      <span class="co"># sigma_exp.block.treat = c(0.1, 0.1, 0.1), # sd among exp:block:treat</span></span>
<span id="cb580-83"><a href="lmm.html#cb580-83" tabindex="-1"></a>      <span class="co"># sigma_rep = 0.1,</span></span>
<span id="cb580-84"><a href="lmm.html#cb580-84" tabindex="-1"></a>      <span class="co"># sigma_ss = 1, # sd among subsamples within replication of treatment:block</span></span>
<span id="cb580-85"><a href="lmm.html#cb580-85" tabindex="-1"></a>      <span class="at">block_name =</span> <span class="st">&quot;experiment&quot;</span>,</span>
<span id="cb580-86"><a href="lmm.html#cb580-86" tabindex="-1"></a>      <span class="at">rep_name =</span> <span class="st">&quot;mouse&quot;</span>,</span>
<span id="cb580-87"><a href="lmm.html#cb580-87" tabindex="-1"></a>      <span class="at">ss_name =</span> <span class="st">&quot;ss&quot;</span>,</span>
<span id="cb580-88"><a href="lmm.html#cb580-88" tabindex="-1"></a>    )</span>
<span id="cb580-89"><a href="lmm.html#cb580-89" tabindex="-1"></a>    </span>
<span id="cb580-90"><a href="lmm.html#cb580-90" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_sims){</span>
<span id="cb580-91"><a href="lmm.html#cb580-91" tabindex="-1"></a>      </span>
<span id="cb580-92"><a href="lmm.html#cb580-92" tabindex="-1"></a>      <span class="co"># models</span></span>
<span id="cb580-93"><a href="lmm.html#cb580-93" tabindex="-1"></a>      form_lm0 <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">paste0</span>(<span class="st">&quot;sim_&quot;</span>, i), <span class="st">&quot;~ treatment&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb580-94"><a href="lmm.html#cb580-94" tabindex="-1"></a>        <span class="fu">formula</span>()</span>
<span id="cb580-95"><a href="lmm.html#cb580-95" tabindex="-1"></a>      form_lm1 <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">paste0</span>(<span class="st">&quot;sim_&quot;</span>, i), <span class="st">&quot;~ treatment + experiment&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb580-96"><a href="lmm.html#cb580-96" tabindex="-1"></a>        <span class="fu">formula</span>()</span>
<span id="cb580-97"><a href="lmm.html#cb580-97" tabindex="-1"></a>      form_aov <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">paste0</span>(<span class="st">&quot;sim_&quot;</span>, i), <span class="st">&quot;~ treatment + (treatment | experiment)&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb580-98"><a href="lmm.html#cb580-98" tabindex="-1"></a>        <span class="fu">formula</span>()</span>
<span id="cb580-99"><a href="lmm.html#cb580-99" tabindex="-1"></a>      lm0 <span class="ot">&lt;-</span> <span class="fu">lm</span>(form_lm0, <span class="at">data =</span> fake_rcbd)</span>
<span id="cb580-100"><a href="lmm.html#cb580-100" tabindex="-1"></a>      lm1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(form_lm1, <span class="at">data =</span> fake_rcbd)</span>
<span id="cb580-101"><a href="lmm.html#cb580-101" tabindex="-1"></a>      lmm2 <span class="ot">&lt;-</span> <span class="fu">lme</span>(form_lm0,</span>
<span id="cb580-102"><a href="lmm.html#cb580-102" tabindex="-1"></a>                  <span class="at">random =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> experiment,</span>
<span id="cb580-103"><a href="lmm.html#cb580-103" tabindex="-1"></a>                  <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> experiment),</span>
<span id="cb580-104"><a href="lmm.html#cb580-104" tabindex="-1"></a>                  <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> treatment),</span>
<span id="cb580-105"><a href="lmm.html#cb580-105" tabindex="-1"></a>                  <span class="at">data =</span> fake_rcbd) <span class="sc">%&gt;%</span></span>
<span id="cb580-106"><a href="lmm.html#cb580-106" tabindex="-1"></a>        <span class="fu">try</span>()</span>
<span id="cb580-107"><a href="lmm.html#cb580-107" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">inherits</span>(lmm2, <span class="st">&quot;try-error&quot;</span>)){</span>
<span id="cb580-108"><a href="lmm.html#cb580-108" tabindex="-1"></a>        lmm2 <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb580-109"><a href="lmm.html#cb580-109" tabindex="-1"></a>      }</span>
<span id="cb580-110"><a href="lmm.html#cb580-110" tabindex="-1"></a>      </span>
<span id="cb580-111"><a href="lmm.html#cb580-111" tabindex="-1"></a>      <span class="co"># aov2 used for both multivariate (aov1) and univariate (aov2)</span></span>
<span id="cb580-112"><a href="lmm.html#cb580-112" tabindex="-1"></a>      aov2 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(form_aov,</span>
<span id="cb580-113"><a href="lmm.html#cb580-113" tabindex="-1"></a>                    <span class="at">fun_aggregate =</span> mean,</span>
<span id="cb580-114"><a href="lmm.html#cb580-114" tabindex="-1"></a>                    <span class="at">include_aov =</span> <span class="cn">TRUE</span>,</span>
<span id="cb580-115"><a href="lmm.html#cb580-115" tabindex="-1"></a>                    <span class="at">data =</span> fake_rcbd)</span>
<span id="cb580-116"><a href="lmm.html#cb580-116" tabindex="-1"></a>      fake_rcbd_means <span class="ot">&lt;-</span> fake_rcbd[, .(<span class="at">sim_1 =</span> <span class="fu">mean</span>(<span class="fu">get</span>(<span class="fu">paste0</span>(<span class="st">&quot;sim_&quot;</span>, i)))),</span>
<span id="cb580-117"><a href="lmm.html#cb580-117" tabindex="-1"></a>                                     by <span class="ot">=</span> .(treatment, experiment)]</span>
<span id="cb580-118"><a href="lmm.html#cb580-118" tabindex="-1"></a>      </span>
<span id="cb580-119"><a href="lmm.html#cb580-119" tabindex="-1"></a>      pptt_pairs <span class="ot">&lt;-</span> <span class="fu">pairwise_t_tests</span>(<span class="at">y_col =</span> <span class="st">&quot;sim_1&quot;</span>,</span>
<span id="cb580-120"><a href="lmm.html#cb580-120" tabindex="-1"></a>                              <span class="at">g_col =</span> <span class="st">&quot;treatment&quot;</span>,</span>
<span id="cb580-121"><a href="lmm.html#cb580-121" tabindex="-1"></a>                              <span class="at">id_col =</span> <span class="st">&quot;experiment&quot;</span>,</span>
<span id="cb580-122"><a href="lmm.html#cb580-122" tabindex="-1"></a>                              <span class="at">data =</span> fake_rcbd_means)</span>
<span id="cb580-123"><a href="lmm.html#cb580-123" tabindex="-1"></a>      </span>
<span id="cb580-124"><a href="lmm.html#cb580-124" tabindex="-1"></a>      lm0_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(lm0, <span class="at">specs =</span> <span class="st">&quot;treatment&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb580-125"><a href="lmm.html#cb580-125" tabindex="-1"></a>        <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>, <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb580-126"><a href="lmm.html#cb580-126" tabindex="-1"></a>      lm1_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(lm1, <span class="at">specs =</span> <span class="st">&quot;treatment&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb580-127"><a href="lmm.html#cb580-127" tabindex="-1"></a>        <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>, <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb580-128"><a href="lmm.html#cb580-128" tabindex="-1"></a>      aov1_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(aov2, <span class="at">specs =</span> <span class="st">&quot;treatment&quot;</span>, <span class="at">model =</span> <span class="st">&quot;multivariate&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb580-129"><a href="lmm.html#cb580-129" tabindex="-1"></a>        <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>, <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb580-130"><a href="lmm.html#cb580-130" tabindex="-1"></a>      aov2_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(aov2, <span class="at">specs =</span> <span class="st">&quot;treatment&quot;</span>, <span class="at">model =</span> <span class="st">&quot;univariate&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb580-131"><a href="lmm.html#cb580-131" tabindex="-1"></a>        <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>, <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb580-132"><a href="lmm.html#cb580-132" tabindex="-1"></a>      </span>
<span id="cb580-133"><a href="lmm.html#cb580-133" tabindex="-1"></a>      fake_rcbd[, y_res <span class="sc">:=</span> <span class="fu">residuals</span>(lm0)]</span>
<span id="cb580-134"><a href="lmm.html#cb580-134" tabindex="-1"></a>      fake_rcbd_wide <span class="ot">&lt;-</span> <span class="fu">dcast</span>(fake_rcbd,</span>
<span id="cb580-135"><a href="lmm.html#cb580-135" tabindex="-1"></a>                               experiment <span class="sc">+</span> rep <span class="sc">~</span> treatment,</span>
<span id="cb580-136"><a href="lmm.html#cb580-136" tabindex="-1"></a>                               <span class="at">value.var =</span> <span class="st">&quot;y_res&quot;</span>)</span>
<span id="cb580-137"><a href="lmm.html#cb580-137" tabindex="-1"></a>      residual_correlations <span class="ot">&lt;-</span> <span class="fu">cor</span>(fake_rcbd_wide[, .SD, <span class="at">.SDcols =</span> treatment_levels])</span>
<span id="cb580-138"><a href="lmm.html#cb580-138" tabindex="-1"></a>      r_mat[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> residual_correlations[<span class="dv">2</span>,<span class="dv">1</span>]</span>
<span id="cb580-139"><a href="lmm.html#cb580-139" tabindex="-1"></a>      r_mat[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> residual_correlations[<span class="dv">3</span>,<span class="dv">1</span>]</span>
<span id="cb580-140"><a href="lmm.html#cb580-140" tabindex="-1"></a>      r_mat[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> residual_correlations[<span class="dv">3</span>,<span class="dv">2</span>]</span>
<span id="cb580-141"><a href="lmm.html#cb580-141" tabindex="-1"></a>      </span>
<span id="cb580-142"><a href="lmm.html#cb580-142" tabindex="-1"></a>      lm0_p <span class="ot">&lt;-</span> <span class="fu">summary</span>(lm0_pairs)[, <span class="st">&quot;p.value&quot;</span>]</span>
<span id="cb580-143"><a href="lmm.html#cb580-143" tabindex="-1"></a>      lm1_p <span class="ot">&lt;-</span> <span class="fu">summary</span>(lm1_pairs)[, <span class="st">&quot;p.value&quot;</span>]</span>
<span id="cb580-144"><a href="lmm.html#cb580-144" tabindex="-1"></a>      aov1_p <span class="ot">&lt;-</span> <span class="fu">summary</span>(aov1_pairs)[, <span class="st">&quot;p.value&quot;</span>]</span>
<span id="cb580-145"><a href="lmm.html#cb580-145" tabindex="-1"></a>      aov2_p <span class="ot">&lt;-</span> <span class="fu">summary</span>(aov2_pairs)[, <span class="st">&quot;p.value&quot;</span>]</span>
<span id="cb580-146"><a href="lmm.html#cb580-146" tabindex="-1"></a>      pptt_p <span class="ot">&lt;-</span> pptt_pairs[, p.value]</span>
<span id="cb580-147"><a href="lmm.html#cb580-147" tabindex="-1"></a>      </span>
<span id="cb580-148"><a href="lmm.html#cb580-148" tabindex="-1"></a>      p1_mat[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> lm0_p[<span class="dv">1</span>]</span>
<span id="cb580-149"><a href="lmm.html#cb580-149" tabindex="-1"></a>      p1_mat[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> lm1_p[<span class="dv">1</span>]</span>
<span id="cb580-150"><a href="lmm.html#cb580-150" tabindex="-1"></a>      p1_mat[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> aov1_p[<span class="dv">1</span>]</span>
<span id="cb580-151"><a href="lmm.html#cb580-151" tabindex="-1"></a>      p1_mat[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> aov2_p[<span class="dv">1</span>]</span>
<span id="cb580-152"><a href="lmm.html#cb580-152" tabindex="-1"></a>      p1_mat[i, <span class="dv">5</span>] <span class="ot">&lt;-</span> pptt_p[<span class="dv">1</span>]</span>
<span id="cb580-153"><a href="lmm.html#cb580-153" tabindex="-1"></a>      </span>
<span id="cb580-154"><a href="lmm.html#cb580-154" tabindex="-1"></a>      p2_mat[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> lm0_p[<span class="dv">2</span>]</span>
<span id="cb580-155"><a href="lmm.html#cb580-155" tabindex="-1"></a>      p2_mat[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> lm1_p[<span class="dv">2</span>]</span>
<span id="cb580-156"><a href="lmm.html#cb580-156" tabindex="-1"></a>      p2_mat[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> aov1_p[<span class="dv">2</span>]</span>
<span id="cb580-157"><a href="lmm.html#cb580-157" tabindex="-1"></a>      p2_mat[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> aov2_p[<span class="dv">2</span>]</span>
<span id="cb580-158"><a href="lmm.html#cb580-158" tabindex="-1"></a>      p2_mat[i, <span class="dv">5</span>] <span class="ot">&lt;-</span> pptt_p[<span class="dv">2</span>]</span>
<span id="cb580-159"><a href="lmm.html#cb580-159" tabindex="-1"></a>      </span>
<span id="cb580-160"><a href="lmm.html#cb580-160" tabindex="-1"></a>      p3_mat[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> lm0_p[<span class="dv">3</span>]</span>
<span id="cb580-161"><a href="lmm.html#cb580-161" tabindex="-1"></a>      p3_mat[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> lm1_p[<span class="dv">3</span>]</span>
<span id="cb580-162"><a href="lmm.html#cb580-162" tabindex="-1"></a>      p3_mat[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> aov1_p[<span class="dv">3</span>]</span>
<span id="cb580-163"><a href="lmm.html#cb580-163" tabindex="-1"></a>      p3_mat[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> aov2_p[<span class="dv">3</span>]</span>
<span id="cb580-164"><a href="lmm.html#cb580-164" tabindex="-1"></a>      p3_mat[i, <span class="dv">5</span>] <span class="ot">&lt;-</span> pptt_p[<span class="dv">3</span>]</span>
<span id="cb580-165"><a href="lmm.html#cb580-165" tabindex="-1"></a>      </span>
<span id="cb580-166"><a href="lmm.html#cb580-166" tabindex="-1"></a>      <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(lmm2)){</span>
<span id="cb580-167"><a href="lmm.html#cb580-167" tabindex="-1"></a>        lmm2_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(lmm2, <span class="at">specs =</span> <span class="st">&quot;treatment&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb580-168"><a href="lmm.html#cb580-168" tabindex="-1"></a>          <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>, <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb580-169"><a href="lmm.html#cb580-169" tabindex="-1"></a>        p1_mat[i, <span class="dv">6</span>] <span class="ot">&lt;-</span> <span class="fu">summary</span>(lmm2_pairs)[<span class="dv">1</span>, <span class="st">&quot;p.value&quot;</span>]</span>
<span id="cb580-170"><a href="lmm.html#cb580-170" tabindex="-1"></a>        p2_mat[i, <span class="dv">6</span>] <span class="ot">&lt;-</span> <span class="fu">summary</span>(lmm2_pairs)[<span class="dv">2</span>, <span class="st">&quot;p.value&quot;</span>]</span>
<span id="cb580-171"><a href="lmm.html#cb580-171" tabindex="-1"></a>        p3_mat[i, <span class="dv">6</span>] <span class="ot">&lt;-</span> <span class="fu">summary</span>(lmm2_pairs)[<span class="dv">3</span>, <span class="st">&quot;p.value&quot;</span>]</span>
<span id="cb580-172"><a href="lmm.html#cb580-172" tabindex="-1"></a>      }<span class="cf">else</span>{</span>
<span id="cb580-173"><a href="lmm.html#cb580-173" tabindex="-1"></a>        p1_mat[i, <span class="dv">6</span>] <span class="ot">&lt;-</span> lm1_p[<span class="dv">1</span>]</span>
<span id="cb580-174"><a href="lmm.html#cb580-174" tabindex="-1"></a>        p2_mat[i, <span class="dv">6</span>] <span class="ot">&lt;-</span> lm1_p[<span class="dv">2</span>]</span>
<span id="cb580-175"><a href="lmm.html#cb580-175" tabindex="-1"></a>        p3_mat[i, <span class="dv">6</span>] <span class="ot">&lt;-</span> lm1_p[<span class="dv">3</span>]</span>
<span id="cb580-176"><a href="lmm.html#cb580-176" tabindex="-1"></a>      }</span>
<span id="cb580-177"><a href="lmm.html#cb580-177" tabindex="-1"></a></span>
<span id="cb580-178"><a href="lmm.html#cb580-178" tabindex="-1"></a>      </span>
<span id="cb580-179"><a href="lmm.html#cb580-179" tabindex="-1"></a>    }</span>
<span id="cb580-180"><a href="lmm.html#cb580-180" tabindex="-1"></a>    </span>
<span id="cb580-181"><a href="lmm.html#cb580-181" tabindex="-1"></a>    sim_table <span class="ot">&lt;-</span> <span class="fu">cbind</span>(r_mat, p1_mat, p2_mat, p3_mat)</span>
<span id="cb580-182"><a href="lmm.html#cb580-182" tabindex="-1"></a>    sim_table_names <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">rep</span>(model_levels, <span class="dv">3</span>),</span>
<span id="cb580-183"><a href="lmm.html#cb580-183" tabindex="-1"></a>          <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&quot;contrast_1&quot;</span>, <span class="st">&quot;contrast_2&quot;</span>, <span class="st">&quot;contrast_3&quot;</span>), <span class="at">each =</span> <span class="fu">length</span>(model_levels)),</span>
<span id="cb580-184"><a href="lmm.html#cb580-184" tabindex="-1"></a>          <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb580-185"><a href="lmm.html#cb580-185" tabindex="-1"></a>    <span class="fu">colnames</span>(sim_table) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;r21&quot;</span>, <span class="st">&quot;r31&quot;</span>, <span class="st">&quot;r32&quot;</span>, sim_table_names)</span>
<span id="cb580-186"><a href="lmm.html#cb580-186" tabindex="-1"></a>    </span>
<span id="cb580-187"><a href="lmm.html#cb580-187" tabindex="-1"></a>    big_table <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb580-188"><a href="lmm.html#cb580-188" tabindex="-1"></a>      big_table,</span>
<span id="cb580-189"><a href="lmm.html#cb580-189" tabindex="-1"></a>      <span class="fu">data.table</span>(</span>
<span id="cb580-190"><a href="lmm.html#cb580-190" tabindex="-1"></a>        <span class="at">sim_id =</span> param_set,</span>
<span id="cb580-191"><a href="lmm.html#cb580-191" tabindex="-1"></a>        <span class="at">n_treat =</span> n_treat_i,</span>
<span id="cb580-192"><a href="lmm.html#cb580-192" tabindex="-1"></a>        <span class="at">n_block =</span> n_block_i,</span>
<span id="cb580-193"><a href="lmm.html#cb580-193" tabindex="-1"></a>        <span class="at">n_rep =</span> n_rep_i,</span>
<span id="cb580-194"><a href="lmm.html#cb580-194" tabindex="-1"></a>        <span class="at">n_ss =</span> n_ss_i,</span>
<span id="cb580-195"><a href="lmm.html#cb580-195" tabindex="-1"></a>        <span class="at">sigma_block =</span> sigma_exp.block_i,</span>
<span id="cb580-196"><a href="lmm.html#cb580-196" tabindex="-1"></a>        <span class="at">sigma_block.treat_1 =</span> sigma_exp.block.treat_i[<span class="dv">1</span>],</span>
<span id="cb580-197"><a href="lmm.html#cb580-197" tabindex="-1"></a>        <span class="at">sigma_block.treat_2 =</span> sigma_exp.block.treat_i[<span class="dv">2</span>],</span>
<span id="cb580-198"><a href="lmm.html#cb580-198" tabindex="-1"></a>        <span class="at">sigma_block.treat_3 =</span> sigma_exp.block.treat_i[<span class="dv">3</span>],</span>
<span id="cb580-199"><a href="lmm.html#cb580-199" tabindex="-1"></a>        <span class="at">sigma_exp.treat =</span> sigma_exp.treat_i,</span>
<span id="cb580-200"><a href="lmm.html#cb580-200" tabindex="-1"></a>        <span class="at">sigma_rep =</span> sigma_rep_i,</span>
<span id="cb580-201"><a href="lmm.html#cb580-201" tabindex="-1"></a>        <span class="at">sigma_exp =</span> sigma_exp_i,</span>
<span id="cb580-202"><a href="lmm.html#cb580-202" tabindex="-1"></a>        <span class="at">sigma_ss =</span> sigma_ss_i,</span>
<span id="cb580-203"><a href="lmm.html#cb580-203" tabindex="-1"></a>        sim_table</span>
<span id="cb580-204"><a href="lmm.html#cb580-204" tabindex="-1"></a>      )</span>
<span id="cb580-205"><a href="lmm.html#cb580-205" tabindex="-1"></a>    )</span>
<span id="cb580-206"><a href="lmm.html#cb580-206" tabindex="-1"></a>  } <span class="co"># end sim</span></span>
<span id="cb580-207"><a href="lmm.html#cb580-207" tabindex="-1"></a>  </span>
<span id="cb580-208"><a href="lmm.html#cb580-208" tabindex="-1"></a>  <span class="fu">saveRDS</span>(<span class="at">object =</span> big_table, <span class="at">file =</span> save_file_path)</span>
<span id="cb580-209"><a href="lmm.html#cb580-209" tabindex="-1"></a>  </span>
<span id="cb580-210"><a href="lmm.html#cb580-210" tabindex="-1"></a>}<span class="cf">else</span>{</span>
<span id="cb580-211"><a href="lmm.html#cb580-211" tabindex="-1"></a>  big_table <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(save_file_path)</span>
<span id="cb580-212"><a href="lmm.html#cb580-212" tabindex="-1"></a>} <span class="co"># end do_sim</span></span>
<span id="cb580-213"><a href="lmm.html#cb580-213" tabindex="-1"></a></span>
<span id="cb580-214"><a href="lmm.html#cb580-214" tabindex="-1"></a></span>
<span id="cb580-215"><a href="lmm.html#cb580-215" tabindex="-1"></a>out_table <span class="ot">&lt;-</span> <span class="fu">data.table</span>(<span class="cn">NULL</span>)</span>
<span id="cb580-216"><a href="lmm.html#cb580-216" tabindex="-1"></a><span class="cf">for</span>(param_set <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(param_set_matrix)){</span>
<span id="cb580-217"><a href="lmm.html#cb580-217" tabindex="-1"></a>  r_mat <span class="ot">&lt;-</span> big_table[sim_id <span class="sc">==</span> param_set, .SD, .SDcols <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;r21&quot;</span>, <span class="st">&quot;r31&quot;</span>, <span class="st">&quot;r32&quot;</span>)]</span>
<span id="cb580-218"><a href="lmm.html#cb580-218" tabindex="-1"></a>  p1_mat <span class="ot">&lt;-</span> big_table[sim_id <span class="sc">==</span> param_set, .SD, .SDcols <span class="ot">=</span> <span class="fu">paste0</span>(model_levels, <span class="st">&quot;_contrast_1&quot;</span>)]</span>
<span id="cb580-219"><a href="lmm.html#cb580-219" tabindex="-1"></a>  p2_mat <span class="ot">&lt;-</span> big_table[sim_id <span class="sc">==</span> param_set, .SD, .SDcols <span class="ot">=</span> <span class="fu">paste0</span>(model_levels, <span class="st">&quot;_contrast_2&quot;</span>)]</span>
<span id="cb580-220"><a href="lmm.html#cb580-220" tabindex="-1"></a>  p3_mat <span class="ot">&lt;-</span> big_table[sim_id <span class="sc">==</span> param_set, .SD, .SDcols <span class="ot">=</span> <span class="fu">paste0</span>(model_levels, <span class="st">&quot;_contrast_3&quot;</span>)]</span>
<span id="cb580-221"><a href="lmm.html#cb580-221" tabindex="-1"></a>  out_table <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb580-222"><a href="lmm.html#cb580-222" tabindex="-1"></a>    out_table,</span>
<span id="cb580-223"><a href="lmm.html#cb580-223" tabindex="-1"></a>    <span class="fu">data.table</span>(</span>
<span id="cb580-224"><a href="lmm.html#cb580-224" tabindex="-1"></a>      <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">&quot;Cn, Tr1&quot;</span>, <span class="st">&quot;Cn, Tr2&quot;</span>, <span class="st">&quot;Tr1, Tr2&quot;</span>),</span>
<span id="cb580-225"><a href="lmm.html#cb580-225" tabindex="-1"></a>      <span class="at">cor =</span> <span class="fu">apply</span>(r_mat, <span class="dv">2</span>, mean),</span>
<span id="cb580-226"><a href="lmm.html#cb580-226" tabindex="-1"></a>      <span class="fu">rbind</span>(<span class="fu">apply</span>(p1_mat, <span class="dv">2</span>, pless),</span>
<span id="cb580-227"><a href="lmm.html#cb580-227" tabindex="-1"></a>            <span class="fu">apply</span>(p2_mat, <span class="dv">2</span>, pless),</span>
<span id="cb580-228"><a href="lmm.html#cb580-228" tabindex="-1"></a>            <span class="fu">apply</span>(p3_mat, <span class="dv">2</span>, pless))</span>
<span id="cb580-229"><a href="lmm.html#cb580-229" tabindex="-1"></a>    ))</span>
<span id="cb580-230"><a href="lmm.html#cb580-230" tabindex="-1"></a>}</span>
<span id="cb580-231"><a href="lmm.html#cb580-231" tabindex="-1"></a></span>
<span id="cb580-232"><a href="lmm.html#cb580-232" tabindex="-1"></a><span class="fu">setnames</span>(out_table,</span>
<span id="cb580-233"><a href="lmm.html#cb580-233" tabindex="-1"></a>         <span class="fu">names</span>(out_table),</span>
<span id="cb580-234"><a href="lmm.html#cb580-234" tabindex="-1"></a>         <span class="fu">c</span>(<span class="st">&quot;Contrast&quot;</span>, <span class="st">&quot;Cor Error&quot;</span>, model_levels))</span>
<span id="cb580-235"><a href="lmm.html#cb580-235" tabindex="-1"></a>out_table <span class="sc">%&gt;%</span></span>
<span id="cb580-236"><a href="lmm.html#cb580-236" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">4</span>),</span>
<span id="cb580-237"><a href="lmm.html#cb580-237" tabindex="-1"></a>        <span class="at">caption =</span> <span class="st">&quot;Type I error rate the RCBD statistical models under different levels of correlation structure among the observations. The Cor Error column is the average correlated error of the residuals fit by a simple linear model y ~ treatment. Model abbreviations as in text.&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb580-238"><a href="lmm.html#cb580-238" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb580-239"><a href="lmm.html#cb580-239" tabindex="-1"></a>  <span class="fu">pack_rows</span>(<span class="st">&quot;sim 1 - high correlated error&quot;</span>, <span class="dv">1</span>, <span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb580-240"><a href="lmm.html#cb580-240" tabindex="-1"></a>  <span class="fu">pack_rows</span>(<span class="st">&quot;sim 2 - low correlated error&quot;</span>, <span class="dv">4</span>, <span class="dv">6</span>) <span class="sc">%&gt;%</span></span>
<span id="cb580-241"><a href="lmm.html#cb580-241" tabindex="-1"></a>  <span class="fu">pack_rows</span>(<span class="st">&quot;sim 3 - equal correlated error&quot;</span>, <span class="dv">7</span>, <span class="dv">9</span>) <span class="sc">%&gt;%</span></span>
<span id="cb580-242"><a href="lmm.html#cb580-242" tabindex="-1"></a>  <span class="fu">pack_rows</span>(<span class="st">&quot;sim 4 - zero correlated error&quot;</span>, <span class="dv">10</span>, <span class="dv">12</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:rcbd-performance-sim">Table 16.19: </span>Type I error rate the RCBD statistical models under different levels of correlation structure among the observations. The Cor Error column is the average correlated error of the residuals fit by a simple linear model y ~ treatment. Model abbreviations as in text.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Contrast
</th>
<th style="text-align:right;">
Cor Error
</th>
<th style="text-align:right;">
lm0
</th>
<th style="text-align:right;">
lm1
</th>
<th style="text-align:right;">
aov1
</th>
<th style="text-align:right;">
aov2
</th>
<th style="text-align:right;">
pptt
</th>
<th style="text-align:right;">
lmm2
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>sim 1 - high correlated error</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Cn, Tr1
</td>
<td style="text-align:right;">
0.299
</td>
<td style="text-align:right;">
0.047
</td>
<td style="text-align:right;">
0.110
</td>
<td style="text-align:right;">
0.058
</td>
<td style="text-align:right;">
0.110
</td>
<td style="text-align:right;">
0.058
</td>
<td style="text-align:right;">
0.068
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Cn, Tr2
</td>
<td style="text-align:right;">
0.412
</td>
<td style="text-align:right;">
0.027
</td>
<td style="text-align:right;">
0.090
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
0.090
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
0.070
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr1, Tr2
</td>
<td style="text-align:right;">
0.662
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.005
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
0.005
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
0.070
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>sim 2 - low correlated error</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Cn, Tr1
</td>
<td style="text-align:right;">
0.165
</td>
<td style="text-align:right;">
0.044
</td>
<td style="text-align:right;">
0.072
</td>
<td style="text-align:right;">
0.055
</td>
<td style="text-align:right;">
0.072
</td>
<td style="text-align:right;">
0.055
</td>
<td style="text-align:right;">
0.063
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Cn, Tr2
</td>
<td style="text-align:right;">
0.164
</td>
<td style="text-align:right;">
0.051
</td>
<td style="text-align:right;">
0.077
</td>
<td style="text-align:right;">
0.060
</td>
<td style="text-align:right;">
0.077
</td>
<td style="text-align:right;">
0.060
</td>
<td style="text-align:right;">
0.076
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr1, Tr2
</td>
<td style="text-align:right;">
0.238
</td>
<td style="text-align:right;">
0.016
</td>
<td style="text-align:right;">
0.028
</td>
<td style="text-align:right;">
0.062
</td>
<td style="text-align:right;">
0.028
</td>
<td style="text-align:right;">
0.062
</td>
<td style="text-align:right;">
0.079
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>sim 3 - equal correlated error</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Cn, Tr1
</td>
<td style="text-align:right;">
0.771
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
0.057
</td>
<td style="text-align:right;">
0.058
</td>
<td style="text-align:right;">
0.057
</td>
<td style="text-align:right;">
0.058
</td>
<td style="text-align:right;">
0.069
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Cn, Tr2
</td>
<td style="text-align:right;">
0.773
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
0.049
</td>
<td style="text-align:right;">
0.056
</td>
<td style="text-align:right;">
0.049
</td>
<td style="text-align:right;">
0.056
</td>
<td style="text-align:right;">
0.067
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr1, Tr2
</td>
<td style="text-align:right;">
0.769
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.056
</td>
<td style="text-align:right;">
0.056
</td>
<td style="text-align:right;">
0.056
</td>
<td style="text-align:right;">
0.056
</td>
<td style="text-align:right;">
0.074
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>sim 4 - zero correlated error</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Cn, Tr1
</td>
<td style="text-align:right;">
0.025
</td>
<td style="text-align:right;">
0.050
</td>
<td style="text-align:right;">
0.047
</td>
<td style="text-align:right;">
0.042
</td>
<td style="text-align:right;">
0.047
</td>
<td style="text-align:right;">
0.042
</td>
<td style="text-align:right;">
0.061
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Cn, Tr2
</td>
<td style="text-align:right;">
0.010
</td>
<td style="text-align:right;">
0.054
</td>
<td style="text-align:right;">
0.054
</td>
<td style="text-align:right;">
0.046
</td>
<td style="text-align:right;">
0.054
</td>
<td style="text-align:right;">
0.046
</td>
<td style="text-align:right;">
0.061
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Tr1, Tr2
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.056
</td>
<td style="text-align:right;">
0.060
</td>
<td style="text-align:right;">
0.053
</td>
<td style="text-align:right;">
0.060
</td>
<td style="text-align:right;">
0.053
</td>
<td style="text-align:right;">
0.076
</td>
</tr>
</tbody>
</table>
<ol style="list-style-type: decimal">
<li>aov1, pptt, and lmm2 have well-behaved type I error rates (lmm2 is a little anti-conservative) regardless of correlation structure – False discoveries should be well controlled.</li>
<li>lm1, lmm1, and aov2 have poorly-behaved type I error rates when there is a block:treatment interaction (sim 1 and 2), resulting in heterogenous correlation structure (remember that this is moot when there are only two treatment levels) – this consequence is small when the block and block:treatment variance is small (small correlations in residuals). When there is block:treatment interaction, these models are conservative in contrasts associated with large residual correlation, resulting in low power, and anti-conservative in contrasts associated with small residual correlations. resulting in high false discovery rates.</li>
</ol>
<p>Best Practices:</p>
<ol style="list-style-type: decimal">
<li>If the number of treatments = 2 and
<ul>
<li>the design is balanced: it doesn’t matter which method you use.</li>
<li>the design is not balanced: use lmm1 or lmm2</li>
</ul></li>
<li>If the number of treatments &gt; 2 and
<ul>
<li>the design is balanced: use aov1/pptt (these are numerically equivalent). lmm1 is the standard but it assumes equal correlated error among treatment levels and equal standard errors among contrasts. lmm2 explicitly models heterogeneity of correlations and variances but 1) the model often fails and 2) has slightly anti-conservative Type I error.</li>
<li>the design is not balanced: lmm2 is attractive if the model runs. pptt is attractive if there are few missing values.</li>
</ul></li>
<li>If there are covariates, then start with lmm2. Use lmeControl if lmm2 fails. If this fails, use pairwise lmm1, which is the same as pairwise t-tests but allows covariates to be added to the model. Note that the covariate will be modeled independently in each pair, which effectively models the consequence of a treatment:covariate interaction.</li>
</ol>
</div>
<div id="rspd" class="section level3 hasAnchor" number="16.10.4">
<h3><span class="header-section-number">16.10.4</span> RSPD<a href="lmm.html#rspd" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Reasonable models:</p>
<ol style="list-style-type: decimal">
<li>lmm4 – random interaction intercept model</li>
</ol>
<p>Assumptions:</p>
<ol style="list-style-type: decimal">
<li></li>
</ol>
<p>Notes:</p>
<ol style="list-style-type: decimal">
<li></li>
</ol>
<p>Performance:</p>
<div class="sourceCode" id="cb581"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb581-1"><a href="lmm.html#cb581-1" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">&quot;simulations/grcbds/data&quot;</span></span>
<span id="cb581-2"><a href="lmm.html#cb581-2" tabindex="-1"></a></span>
<span id="cb581-3"><a href="lmm.html#cb581-3" tabindex="-1"></a>doit <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb581-4"><a href="lmm.html#cb581-4" tabindex="-1"></a>writeit <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb581-5"><a href="lmm.html#cb581-5" tabindex="-1"></a>readit <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb581-6"><a href="lmm.html#cb581-6" tabindex="-1"></a>sim_id_i <span class="ot">&lt;-</span> <span class="dv">7</span></span>
<span id="cb581-7"><a href="lmm.html#cb581-7" tabindex="-1"></a></span>
<span id="cb581-8"><a href="lmm.html#cb581-8" tabindex="-1"></a>n_sim <span class="ot">&lt;-</span> <span class="dv">1000</span></span>
<span id="cb581-9"><a href="lmm.html#cb581-9" tabindex="-1"></a>n_main_plot <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="co"># types of main plot, each assigned single Tr1</span></span>
<span id="cb581-10"><a href="lmm.html#cb581-10" tabindex="-1"></a>n_sub_plot <span class="ot">&lt;-</span> <span class="dv">3</span> <span class="co"># type os sub plot, each assigned single Tr2</span></span>
<span id="cb581-11"><a href="lmm.html#cb581-11" tabindex="-1"></a>n_block <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb581-12"><a href="lmm.html#cb581-12" tabindex="-1"></a>n_ss <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb581-13"><a href="lmm.html#cb581-13" tabindex="-1"></a>N <span class="ot">&lt;-</span> n_main_plot <span class="sc">*</span> n_sub_plot <span class="sc">*</span> n_block <span class="sc">*</span> n_ss</span>
<span id="cb581-14"><a href="lmm.html#cb581-14" tabindex="-1"></a></span>
<span id="cb581-15"><a href="lmm.html#cb581-15" tabindex="-1"></a>beta <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>)</span>
<span id="cb581-16"><a href="lmm.html#cb581-16" tabindex="-1"></a>sigma_block <span class="ot">&lt;-</span> <span class="fl">0.8</span></span>
<span id="cb581-17"><a href="lmm.html#cb581-17" tabindex="-1"></a>sigma_ss <span class="ot">&lt;-</span> <span class="fl">0.5</span></span>
<span id="cb581-18"><a href="lmm.html#cb581-18" tabindex="-1"></a></span>
<span id="cb581-19"><a href="lmm.html#cb581-19" tabindex="-1"></a><span class="cf">if</span>(sim_id_i <span class="sc">==</span> <span class="dv">1</span>){</span>
<span id="cb581-20"><a href="lmm.html#cb581-20" tabindex="-1"></a>  <span class="co"># sim 1</span></span>
<span id="cb581-21"><a href="lmm.html#cb581-21" tabindex="-1"></a>  sigma_block_main <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.6</span>) <span class="co"># 0.4, 0, 0.4, 0</span></span>
<span id="cb581-22"><a href="lmm.html#cb581-22" tabindex="-1"></a>  sigma_block_sub <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.0</span>, <span class="fl">0.0</span>, <span class="fl">0.0</span>) <span class="co"># 0, 0.4, 0.4, 0</span></span>
<span id="cb581-23"><a href="lmm.html#cb581-23" tabindex="-1"></a>}</span>
<span id="cb581-24"><a href="lmm.html#cb581-24" tabindex="-1"></a><span class="cf">if</span>(sim_id_i <span class="sc">==</span> <span class="dv">2</span>){</span>
<span id="cb581-25"><a href="lmm.html#cb581-25" tabindex="-1"></a>  <span class="co"># sim 2</span></span>
<span id="cb581-26"><a href="lmm.html#cb581-26" tabindex="-1"></a>  sigma_block_main <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.0</span>, <span class="fl">0.0</span>) <span class="co"># 0.4, 0, 0.4, 0</span></span>
<span id="cb581-27"><a href="lmm.html#cb581-27" tabindex="-1"></a>  sigma_block_sub <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.6</span>, <span class="fl">0.6</span>) <span class="co"># 0, 0.4, 0.4, 0</span></span>
<span id="cb581-28"><a href="lmm.html#cb581-28" tabindex="-1"></a>}</span>
<span id="cb581-29"><a href="lmm.html#cb581-29" tabindex="-1"></a><span class="cf">if</span>(sim_id_i <span class="sc">==</span> <span class="dv">3</span>){</span>
<span id="cb581-30"><a href="lmm.html#cb581-30" tabindex="-1"></a>  <span class="co"># sim 3</span></span>
<span id="cb581-31"><a href="lmm.html#cb581-31" tabindex="-1"></a>  sigma_block_main <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.6</span>) <span class="co"># 0.4, 0, 0.4, 0</span></span>
<span id="cb581-32"><a href="lmm.html#cb581-32" tabindex="-1"></a>  sigma_block_sub <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.4</span>, <span class="fl">0.4</span>, <span class="fl">0.4</span>) <span class="co"># 0, 0.4, 0.4, 0</span></span>
<span id="cb581-33"><a href="lmm.html#cb581-33" tabindex="-1"></a>}</span>
<span id="cb581-34"><a href="lmm.html#cb581-34" tabindex="-1"></a><span class="cf">if</span>(sim_id_i <span class="sc">==</span> <span class="dv">4</span>){</span>
<span id="cb581-35"><a href="lmm.html#cb581-35" tabindex="-1"></a>  <span class="co"># sim 4</span></span>
<span id="cb581-36"><a href="lmm.html#cb581-36" tabindex="-1"></a>  sigma_block_main <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.0</span>, <span class="fl">0.0</span>) <span class="co"># 0.4, 0, 0.4, 0</span></span>
<span id="cb581-37"><a href="lmm.html#cb581-37" tabindex="-1"></a>  sigma_block_sub <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.0</span>, <span class="fl">0.0</span>, <span class="fl">0.0</span>) <span class="co"># 0, 0.4, 0.4, 0</span></span>
<span id="cb581-38"><a href="lmm.html#cb581-38" tabindex="-1"></a>}</span>
<span id="cb581-39"><a href="lmm.html#cb581-39" tabindex="-1"></a><span class="cf">if</span>(sim_id_i <span class="sc">==</span> <span class="dv">5</span>){</span>
<span id="cb581-40"><a href="lmm.html#cb581-40" tabindex="-1"></a>  <span class="co"># sim 5</span></span>
<span id="cb581-41"><a href="lmm.html#cb581-41" tabindex="-1"></a>  sigma_block_main <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.2</span>) <span class="co"># 0.4, 0, 0.4, 0</span></span>
<span id="cb581-42"><a href="lmm.html#cb581-42" tabindex="-1"></a>  sigma_block_sub <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.0</span>, <span class="fl">0.0</span>, <span class="fl">0.0</span>) <span class="co"># 0, 0.4, 0.4, 0</span></span>
<span id="cb581-43"><a href="lmm.html#cb581-43" tabindex="-1"></a>}</span>
<span id="cb581-44"><a href="lmm.html#cb581-44" tabindex="-1"></a><span class="cf">if</span>(sim_id_i <span class="sc">==</span> <span class="dv">6</span>){</span>
<span id="cb581-45"><a href="lmm.html#cb581-45" tabindex="-1"></a>  <span class="co"># sim 6</span></span>
<span id="cb581-46"><a href="lmm.html#cb581-46" tabindex="-1"></a>  sigma_block_main <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.0</span>, <span class="fl">0.0</span>) <span class="co"># 0.4, 0, 0.4, 0</span></span>
<span id="cb581-47"><a href="lmm.html#cb581-47" tabindex="-1"></a>  sigma_block_sub <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.4</span>, <span class="fl">0.2</span>) <span class="co"># 0, 0.4, 0.4, 0</span></span>
<span id="cb581-48"><a href="lmm.html#cb581-48" tabindex="-1"></a>}</span>
<span id="cb581-49"><a href="lmm.html#cb581-49" tabindex="-1"></a><span class="cf">if</span>(sim_id_i <span class="sc">==</span> <span class="dv">7</span>){</span>
<span id="cb581-50"><a href="lmm.html#cb581-50" tabindex="-1"></a>  <span class="co"># sim 7</span></span>
<span id="cb581-51"><a href="lmm.html#cb581-51" tabindex="-1"></a>  sigma_block_main <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.2</span>) <span class="co"># 0.4, 0, 0.4, 0</span></span>
<span id="cb581-52"><a href="lmm.html#cb581-52" tabindex="-1"></a>  sigma_block_sub <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.4</span>, <span class="fl">0.2</span>) <span class="co"># 0, 0.4, 0.4, 0</span></span>
<span id="cb581-53"><a href="lmm.html#cb581-53" tabindex="-1"></a>}</span>
<span id="cb581-54"><a href="lmm.html#cb581-54" tabindex="-1"></a></span>
<span id="cb581-55"><a href="lmm.html#cb581-55" tabindex="-1"></a>tr1_main_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;WT&quot;</span>, <span class="st">&quot;KO&quot;</span>)</span>
<span id="cb581-56"><a href="lmm.html#cb581-56" tabindex="-1"></a>tr2_sub_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Cn&quot;</span>, <span class="st">&quot;Tr1&quot;</span>, <span class="st">&quot;Tr2&quot;</span>)</span>
<span id="cb581-57"><a href="lmm.html#cb581-57" tabindex="-1"></a></span>
<span id="cb581-58"><a href="lmm.html#cb581-58" tabindex="-1"></a><span class="cf">if</span>(doit <span class="sc">==</span> <span class="cn">TRUE</span>){</span>
<span id="cb581-59"><a href="lmm.html#cb581-59" tabindex="-1"></a>  fake_data <span class="ot">&lt;-</span> <span class="fu">data.table</span>(</span>
<span id="cb581-60"><a href="lmm.html#cb581-60" tabindex="-1"></a>    <span class="at">tr1_main =</span> <span class="fu">rep</span>(<span class="fu">rep</span>(tr1_main_levels, <span class="at">each =</span> n_sub_plot), n_block),</span>
<span id="cb581-61"><a href="lmm.html#cb581-61" tabindex="-1"></a>    <span class="at">tr2_sub =</span> <span class="fu">rep</span>(<span class="fu">rep</span>(tr2_sub_levels, n_main_plot), n_block),</span>
<span id="cb581-62"><a href="lmm.html#cb581-62" tabindex="-1"></a>    <span class="at">main_plot =</span> <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span>(n_main_plot <span class="sc">*</span> n_block), <span class="at">each =</span> n_sub_plot),</span>
<span id="cb581-63"><a href="lmm.html#cb581-63" tabindex="-1"></a>    <span class="at">block =</span> <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span>n_block, <span class="at">each =</span> n_sub_plot<span class="sc">*</span>n_main_plot)</span>
<span id="cb581-64"><a href="lmm.html#cb581-64" tabindex="-1"></a>  )</span>
<span id="cb581-65"><a href="lmm.html#cb581-65" tabindex="-1"></a>  </span>
<span id="cb581-66"><a href="lmm.html#cb581-66" tabindex="-1"></a>  fake_data[, tr1_main <span class="sc">:=</span> <span class="fu">factor</span>(tr1_main, <span class="at">levels =</span> tr1_main_levels)]</span>
<span id="cb581-67"><a href="lmm.html#cb581-67" tabindex="-1"></a>  fake_data[, tr2_sub <span class="sc">:=</span> <span class="fu">factor</span>(tr2_sub, <span class="at">levels =</span> tr2_sub_levels)]</span>
<span id="cb581-68"><a href="lmm.html#cb581-68" tabindex="-1"></a>  fake_data[, block <span class="sc">:=</span> <span class="fu">factor</span>(block)]</span>
<span id="cb581-69"><a href="lmm.html#cb581-69" tabindex="-1"></a>  </span>
<span id="cb581-70"><a href="lmm.html#cb581-70" tabindex="-1"></a>  <span class="co"># fixed effect matrix</span></span>
<span id="cb581-71"><a href="lmm.html#cb581-71" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> tr1_main <span class="sc">*</span> tr2_sub, <span class="at">data =</span> fake_data)</span>
<span id="cb581-72"><a href="lmm.html#cb581-72" tabindex="-1"></a>  </span>
<span id="cb581-73"><a href="lmm.html#cb581-73" tabindex="-1"></a>  </span>
<span id="cb581-74"><a href="lmm.html#cb581-74" tabindex="-1"></a>  <span class="co"># random effect matrix</span></span>
<span id="cb581-75"><a href="lmm.html#cb581-75" tabindex="-1"></a>  Z1a <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> block, <span class="at">data =</span> fake_data)</span>
<span id="cb581-76"><a href="lmm.html#cb581-76" tabindex="-1"></a>  Z1b <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> block<span class="sc">:</span>tr1_main, <span class="at">data =</span> fake_data)</span>
<span id="cb581-77"><a href="lmm.html#cb581-77" tabindex="-1"></a>  Z1c <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> block<span class="sc">:</span>tr2_sub, <span class="at">data =</span> fake_data)</span>
<span id="cb581-78"><a href="lmm.html#cb581-78" tabindex="-1"></a>  Z1 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(Z1a, Z1b, Z1c)</span>
<span id="cb581-79"><a href="lmm.html#cb581-79" tabindex="-1"></a>  </span>
<span id="cb581-80"><a href="lmm.html#cb581-80" tabindex="-1"></a>  y_mat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="cn">NA</span>), <span class="at">nrow =</span> N, <span class="at">ncol =</span> n_sim)</span>
<span id="cb581-81"><a href="lmm.html#cb581-81" tabindex="-1"></a>  </span>
<span id="cb581-82"><a href="lmm.html#cb581-82" tabindex="-1"></a>  <span class="co"># get n_sim sets of fake data</span></span>
<span id="cb581-83"><a href="lmm.html#cb581-83" tabindex="-1"></a>  <span class="cf">for</span>(sim_i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_sim){</span>
<span id="cb581-84"><a href="lmm.html#cb581-84" tabindex="-1"></a>    seed_starter <span class="ot">&lt;-</span> seed_starter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb581-85"><a href="lmm.html#cb581-85" tabindex="-1"></a>    <span class="fu">set.seed</span>(seed_starter)</span>
<span id="cb581-86"><a href="lmm.html#cb581-86" tabindex="-1"></a>    </span>
<span id="cb581-87"><a href="lmm.html#cb581-87" tabindex="-1"></a>    u1a <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n_block, <span class="at">mean =</span> <span class="dv">0</span>, <span class="co"># block</span></span>
<span id="cb581-88"><a href="lmm.html#cb581-88" tabindex="-1"></a>                 <span class="at">sd =</span> sigma_block)</span>
<span id="cb581-89"><a href="lmm.html#cb581-89" tabindex="-1"></a>    u1b <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n_block<span class="sc">*</span>n_main_plot, <span class="co"># block:tr1_main</span></span>
<span id="cb581-90"><a href="lmm.html#cb581-90" tabindex="-1"></a>                 <span class="at">mean =</span> <span class="fu">rep</span>(<span class="dv">0</span>, n_block<span class="sc">*</span>n_main_plot),</span>
<span id="cb581-91"><a href="lmm.html#cb581-91" tabindex="-1"></a>                 <span class="at">sd =</span> <span class="fu">rep</span>(sigma_block_main, <span class="at">each =</span> n_block))</span>
<span id="cb581-92"><a href="lmm.html#cb581-92" tabindex="-1"></a>    u1c <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n_block<span class="sc">*</span>n_sub_plot,  <span class="co"># block:tr2_sub</span></span>
<span id="cb581-93"><a href="lmm.html#cb581-93" tabindex="-1"></a>                 <span class="at">mean =</span> <span class="fu">rep</span>(<span class="dv">0</span>, n_block<span class="sc">*</span>n_sub_plot),</span>
<span id="cb581-94"><a href="lmm.html#cb581-94" tabindex="-1"></a>                 <span class="at">sd =</span> <span class="fu">rep</span>(sigma_block_sub, <span class="at">each =</span> n_block))         </span>
<span id="cb581-95"><a href="lmm.html#cb581-95" tabindex="-1"></a>    u1 <span class="ot">&lt;-</span> <span class="fu">c</span>(u1a, u1b, u1c)</span>
<span id="cb581-96"><a href="lmm.html#cb581-96" tabindex="-1"></a>    e <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(N, <span class="dv">0</span>, sigma_ss)</span>
<span id="cb581-97"><a href="lmm.html#cb581-97" tabindex="-1"></a>    </span>
<span id="cb581-98"><a href="lmm.html#cb581-98" tabindex="-1"></a>    y_mat[, sim_i] <span class="ot">&lt;-</span>  X <span class="sc">%*%</span> beta <span class="sc">+</span> Z1 <span class="sc">%*%</span> u1 <span class="sc">+</span> e</span>
<span id="cb581-99"><a href="lmm.html#cb581-99" tabindex="-1"></a>    </span>
<span id="cb581-100"><a href="lmm.html#cb581-100" tabindex="-1"></a>  }  </span>
<span id="cb581-101"><a href="lmm.html#cb581-101" tabindex="-1"></a>    <span class="fu">colnames</span>(y_mat) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;sim_&quot;</span>, <span class="dv">1</span><span class="sc">:</span>n_sim)</span>
<span id="cb581-102"><a href="lmm.html#cb581-102" tabindex="-1"></a>    fake_data <span class="ot">&lt;-</span> <span class="fu">cbind</span>(fake_data, y_mat)</span>
<span id="cb581-103"><a href="lmm.html#cb581-103" tabindex="-1"></a>  </span>
<span id="cb581-104"><a href="lmm.html#cb581-104" tabindex="-1"></a>  </span>
<span id="cb581-105"><a href="lmm.html#cb581-105" tabindex="-1"></a>  <span class="co"># run models on fake data</span></span>
<span id="cb581-106"><a href="lmm.html#cb581-106" tabindex="-1"></a>  n_pairs <span class="ot">&lt;-</span> (n_main_plot <span class="sc">*</span> n_sub_plot<span class="sc">*</span>(n_sub_plot<span class="dv">-1</span>)<span class="sc">/</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb581-107"><a href="lmm.html#cb581-107" tabindex="-1"></a>    (n_sub_plot <span class="sc">*</span> n_main_plot<span class="sc">*</span>(n_main_plot<span class="dv">-1</span>)<span class="sc">/</span><span class="dv">2</span>)</span>
<span id="cb581-108"><a href="lmm.html#cb581-108" tabindex="-1"></a>  lm0_p <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="cn">NA</span>), <span class="at">nrow =</span> n_sim, <span class="at">ncol =</span> n_pairs)</span>
<span id="cb581-109"><a href="lmm.html#cb581-109" tabindex="-1"></a>  lm1_p <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="cn">NA</span>), <span class="at">nrow =</span> n_sim, <span class="at">ncol =</span> n_pairs)</span>
<span id="cb581-110"><a href="lmm.html#cb581-110" tabindex="-1"></a>  lmm1_p <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="cn">NA</span>), <span class="at">nrow =</span> n_sim, <span class="at">ncol =</span> n_pairs)</span>
<span id="cb581-111"><a href="lmm.html#cb581-111" tabindex="-1"></a>  lmm3_p <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="cn">NA</span>), <span class="at">nrow =</span> n_sim, <span class="at">ncol =</span> n_pairs)</span>
<span id="cb581-112"><a href="lmm.html#cb581-112" tabindex="-1"></a>  lmm4_p <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="cn">NA</span>), <span class="at">nrow =</span> n_sim, <span class="at">ncol =</span> n_pairs)</span>
<span id="cb581-113"><a href="lmm.html#cb581-113" tabindex="-1"></a>  lmm5_p <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="cn">NA</span>), <span class="at">nrow =</span> n_sim, <span class="at">ncol =</span> n_pairs)</span>
<span id="cb581-114"><a href="lmm.html#cb581-114" tabindex="-1"></a>  lmm6_p <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="cn">NA</span>), <span class="at">nrow =</span> n_sim, <span class="at">ncol =</span> n_pairs)</span>
<span id="cb581-115"><a href="lmm.html#cb581-115" tabindex="-1"></a>  r_mat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">as.numeric</span>(<span class="cn">NA</span>), <span class="at">nrow =</span> n_sim, <span class="at">ncol =</span> n_pairs)</span>
<span id="cb581-116"><a href="lmm.html#cb581-116" tabindex="-1"></a>  lmm3_check <span class="ot">&lt;-</span> <span class="fu">array</span>(<span class="at">data =</span> <span class="cn">NA</span>)</span>
<span id="cb581-117"><a href="lmm.html#cb581-117" tabindex="-1"></a>  lmm4_check <span class="ot">&lt;-</span> <span class="fu">array</span>(<span class="at">data =</span> <span class="cn">NA</span>)</span>
<span id="cb581-118"><a href="lmm.html#cb581-118" tabindex="-1"></a>  lmm5_check <span class="ot">&lt;-</span> <span class="fu">array</span>(<span class="at">data =</span> <span class="cn">NA</span>)</span>
<span id="cb581-119"><a href="lmm.html#cb581-119" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_sim){</span>
<span id="cb581-120"><a href="lmm.html#cb581-120" tabindex="-1"></a>    fake_i <span class="ot">&lt;-</span> fake_data[, .SD, .SDcols <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;tr1_main&quot;</span>,</span>
<span id="cb581-121"><a href="lmm.html#cb581-121" tabindex="-1"></a>                                           <span class="st">&quot;tr2_sub&quot;</span>, <span class="st">&quot;block&quot;</span>, <span class="fu">paste0</span>(<span class="st">&quot;sim_&quot;</span>, i))]</span>
<span id="cb581-122"><a href="lmm.html#cb581-122" tabindex="-1"></a>    <span class="fu">setnames</span>(fake_i, <span class="fu">paste0</span>(<span class="st">&quot;sim_&quot;</span>, i), <span class="st">&quot;y&quot;</span>)</span>
<span id="cb581-123"><a href="lmm.html#cb581-123" tabindex="-1"></a>    </span>
<span id="cb581-124"><a href="lmm.html#cb581-124" tabindex="-1"></a>    lm0 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> tr1_main <span class="sc">*</span> tr2_sub,</span>
<span id="cb581-125"><a href="lmm.html#cb581-125" tabindex="-1"></a>              <span class="at">data =</span> fake_i)</span>
<span id="cb581-126"><a href="lmm.html#cb581-126" tabindex="-1"></a>    lm1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> tr1_main <span class="sc">*</span> tr2_sub <span class="sc">+</span> block <span class="sc">+</span> tr1_main<span class="sc">:</span>block,</span>
<span id="cb581-127"><a href="lmm.html#cb581-127" tabindex="-1"></a>              <span class="at">data =</span> fake_i)</span>
<span id="cb581-128"><a href="lmm.html#cb581-128" tabindex="-1"></a>    lmm1 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(y <span class="sc">~</span> tr1_main <span class="sc">*</span> tr2_sub <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block),</span>
<span id="cb581-129"><a href="lmm.html#cb581-129" tabindex="-1"></a>                 <span class="at">data =</span> fake_i)</span>
<span id="cb581-130"><a href="lmm.html#cb581-130" tabindex="-1"></a>    lmm3 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(y <span class="sc">~</span> tr1_main <span class="sc">*</span> tr2_sub <span class="sc">+</span> (tr1_main<span class="sc">|</span>block),</span>
<span id="cb581-131"><a href="lmm.html#cb581-131" tabindex="-1"></a>                 <span class="at">data =</span> fake_i)</span>
<span id="cb581-132"><a href="lmm.html#cb581-132" tabindex="-1"></a>    lmm4 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(y <span class="sc">~</span> tr1_main <span class="sc">*</span> tr2_sub <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>tr1_main),</span>
<span id="cb581-133"><a href="lmm.html#cb581-133" tabindex="-1"></a>                 <span class="at">data =</span> fake_i)</span>
<span id="cb581-134"><a href="lmm.html#cb581-134" tabindex="-1"></a>    lmm5 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(y <span class="sc">~</span> tr1_main <span class="sc">*</span> tr2_sub <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>tr2_sub),</span>
<span id="cb581-135"><a href="lmm.html#cb581-135" tabindex="-1"></a>                 <span class="at">data =</span> fake_i)</span>
<span id="cb581-136"><a href="lmm.html#cb581-136" tabindex="-1"></a>    lmm6 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(y <span class="sc">~</span> tr1_main <span class="sc">*</span> tr2_sub <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>tr2_sub) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> block<span class="sc">:</span>tr1_main),</span>
<span id="cb581-137"><a href="lmm.html#cb581-137" tabindex="-1"></a>                 <span class="at">data =</span> fake_i)</span>
<span id="cb581-138"><a href="lmm.html#cb581-138" tabindex="-1"></a>    <span class="co"># lmm5 &lt;- lmer(y ~ tr1_main * tr2_sub + block + (1 | block:tr1_main),</span></span>
<span id="cb581-139"><a href="lmm.html#cb581-139" tabindex="-1"></a>    <span class="co">#              data = fake_data)</span></span>
<span id="cb581-140"><a href="lmm.html#cb581-140" tabindex="-1"></a>    </span>
<span id="cb581-141"><a href="lmm.html#cb581-141" tabindex="-1"></a>    <span class="co"># aov1 &lt;- aov_4(y ~ tr1_main * tr2_sub + (tr1_main | block),</span></span>
<span id="cb581-142"><a href="lmm.html#cb581-142" tabindex="-1"></a>    <span class="co">#               data = fake_i)</span></span>
<span id="cb581-143"><a href="lmm.html#cb581-143" tabindex="-1"></a>    <span class="co"># </span></span>
<span id="cb581-144"><a href="lmm.html#cb581-144" tabindex="-1"></a>    lm0_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(lm0, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;tr1_main&quot;</span>, <span class="st">&quot;tr2_sub&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb581-145"><a href="lmm.html#cb581-145" tabindex="-1"></a>      <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb581-146"><a href="lmm.html#cb581-146" tabindex="-1"></a>               <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb581-147"><a href="lmm.html#cb581-147" tabindex="-1"></a>               <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb581-148"><a href="lmm.html#cb581-148" tabindex="-1"></a>               <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb581-149"><a href="lmm.html#cb581-149" tabindex="-1"></a>      <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb581-150"><a href="lmm.html#cb581-150" tabindex="-1"></a>    lm0_p[i,] <span class="ot">&lt;-</span> lm0_pairs[, <span class="st">&quot;p.value&quot;</span>]</span>
<span id="cb581-151"><a href="lmm.html#cb581-151" tabindex="-1"></a>    </span>
<span id="cb581-152"><a href="lmm.html#cb581-152" tabindex="-1"></a>    lm1_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(lm1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;tr1_main&quot;</span>, <span class="st">&quot;tr2_sub&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb581-153"><a href="lmm.html#cb581-153" tabindex="-1"></a>      <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb581-154"><a href="lmm.html#cb581-154" tabindex="-1"></a>               <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb581-155"><a href="lmm.html#cb581-155" tabindex="-1"></a>               <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb581-156"><a href="lmm.html#cb581-156" tabindex="-1"></a>               <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb581-157"><a href="lmm.html#cb581-157" tabindex="-1"></a>      <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb581-158"><a href="lmm.html#cb581-158" tabindex="-1"></a>    lm1_p[i,] <span class="ot">&lt;-</span> lm1_pairs[, <span class="st">&quot;p.value&quot;</span>]</span>
<span id="cb581-159"><a href="lmm.html#cb581-159" tabindex="-1"></a>    </span>
<span id="cb581-160"><a href="lmm.html#cb581-160" tabindex="-1"></a>    lmm1_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(lmm1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;tr1_main&quot;</span>, <span class="st">&quot;tr2_sub&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb581-161"><a href="lmm.html#cb581-161" tabindex="-1"></a>      <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb581-162"><a href="lmm.html#cb581-162" tabindex="-1"></a>               <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb581-163"><a href="lmm.html#cb581-163" tabindex="-1"></a>               <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb581-164"><a href="lmm.html#cb581-164" tabindex="-1"></a>               <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb581-165"><a href="lmm.html#cb581-165" tabindex="-1"></a>      <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb581-166"><a href="lmm.html#cb581-166" tabindex="-1"></a>    lmm1_p[i,] <span class="ot">&lt;-</span> lmm1_pairs[, <span class="st">&quot;p.value&quot;</span>]</span>
<span id="cb581-167"><a href="lmm.html#cb581-167" tabindex="-1"></a>    </span>
<span id="cb581-168"><a href="lmm.html#cb581-168" tabindex="-1"></a>    lmm3_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(lmm3, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;tr1_main&quot;</span>, <span class="st">&quot;tr2_sub&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb581-169"><a href="lmm.html#cb581-169" tabindex="-1"></a>      <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb581-170"><a href="lmm.html#cb581-170" tabindex="-1"></a>               <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb581-171"><a href="lmm.html#cb581-171" tabindex="-1"></a>               <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb581-172"><a href="lmm.html#cb581-172" tabindex="-1"></a>               <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb581-173"><a href="lmm.html#cb581-173" tabindex="-1"></a>      <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb581-174"><a href="lmm.html#cb581-174" tabindex="-1"></a>    lmm3_p[i,] <span class="ot">&lt;-</span> lmm3_pairs[, <span class="st">&quot;p.value&quot;</span>]</span>
<span id="cb581-175"><a href="lmm.html#cb581-175" tabindex="-1"></a></span>
<span id="cb581-176"><a href="lmm.html#cb581-176" tabindex="-1"></a>    lmm4_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(lmm4, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;tr1_main&quot;</span>, <span class="st">&quot;tr2_sub&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb581-177"><a href="lmm.html#cb581-177" tabindex="-1"></a>      <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb581-178"><a href="lmm.html#cb581-178" tabindex="-1"></a>               <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb581-179"><a href="lmm.html#cb581-179" tabindex="-1"></a>               <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb581-180"><a href="lmm.html#cb581-180" tabindex="-1"></a>               <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb581-181"><a href="lmm.html#cb581-181" tabindex="-1"></a>      <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb581-182"><a href="lmm.html#cb581-182" tabindex="-1"></a>    lmm4_p[i,] <span class="ot">&lt;-</span> lmm4_pairs[, <span class="st">&quot;p.value&quot;</span>]</span>
<span id="cb581-183"><a href="lmm.html#cb581-183" tabindex="-1"></a>    </span>
<span id="cb581-184"><a href="lmm.html#cb581-184" tabindex="-1"></a>    lmm5_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(lmm5, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;tr1_main&quot;</span>, <span class="st">&quot;tr2_sub&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb581-185"><a href="lmm.html#cb581-185" tabindex="-1"></a>      <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb581-186"><a href="lmm.html#cb581-186" tabindex="-1"></a>               <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb581-187"><a href="lmm.html#cb581-187" tabindex="-1"></a>               <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb581-188"><a href="lmm.html#cb581-188" tabindex="-1"></a>               <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb581-189"><a href="lmm.html#cb581-189" tabindex="-1"></a>      <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb581-190"><a href="lmm.html#cb581-190" tabindex="-1"></a>    lmm5_p[i,] <span class="ot">&lt;-</span> lmm5_pairs[, <span class="st">&quot;p.value&quot;</span>]</span>
<span id="cb581-191"><a href="lmm.html#cb581-191" tabindex="-1"></a>    </span>
<span id="cb581-192"><a href="lmm.html#cb581-192" tabindex="-1"></a>    lmm6_pairs <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(lmm6, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;tr1_main&quot;</span>, <span class="st">&quot;tr2_sub&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb581-193"><a href="lmm.html#cb581-193" tabindex="-1"></a>      <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb581-194"><a href="lmm.html#cb581-194" tabindex="-1"></a>               <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb581-195"><a href="lmm.html#cb581-195" tabindex="-1"></a>               <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb581-196"><a href="lmm.html#cb581-196" tabindex="-1"></a>               <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb581-197"><a href="lmm.html#cb581-197" tabindex="-1"></a>      <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span>
<span id="cb581-198"><a href="lmm.html#cb581-198" tabindex="-1"></a>    lmm6_p[i,] <span class="ot">&lt;-</span> lmm6_pairs[, <span class="st">&quot;p.value&quot;</span>]</span>
<span id="cb581-199"><a href="lmm.html#cb581-199" tabindex="-1"></a></span>
<span id="cb581-200"><a href="lmm.html#cb581-200" tabindex="-1"></a>    <span class="co"># get correlations</span></span>
<span id="cb581-201"><a href="lmm.html#cb581-201" tabindex="-1"></a>    fake_i[, y_res <span class="sc">:=</span> <span class="fu">residuals</span>(lm0)]</span>
<span id="cb581-202"><a href="lmm.html#cb581-202" tabindex="-1"></a>    fake_i_wide <span class="ot">&lt;-</span> <span class="fu">dcast</span>(fake_i,</span>
<span id="cb581-203"><a href="lmm.html#cb581-203" tabindex="-1"></a>                         block <span class="sc">~</span> tr1_main <span class="sc">+</span> tr2_sub,</span>
<span id="cb581-204"><a href="lmm.html#cb581-204" tabindex="-1"></a>                         <span class="at">value.var =</span> <span class="st">&quot;y_res&quot;</span>)</span>
<span id="cb581-205"><a href="lmm.html#cb581-205" tabindex="-1"></a>    treatment_levels <span class="ot">&lt;-</span> <span class="fu">names</span>(fake_i_wide)[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb581-206"><a href="lmm.html#cb581-206" tabindex="-1"></a>    residual_correlations <span class="ot">&lt;-</span> <span class="fu">cor</span>(fake_i_wide[, .SD, <span class="at">.SDcols =</span> treatment_levels])</span>
<span id="cb581-207"><a href="lmm.html#cb581-207" tabindex="-1"></a>    r_mat[i, <span class="dv">1</span>] <span class="ot">&lt;-</span> residual_correlations[<span class="dv">4</span>,<span class="dv">1</span>]</span>
<span id="cb581-208"><a href="lmm.html#cb581-208" tabindex="-1"></a>    r_mat[i, <span class="dv">2</span>] <span class="ot">&lt;-</span> residual_correlations[<span class="dv">5</span>,<span class="dv">2</span>]</span>
<span id="cb581-209"><a href="lmm.html#cb581-209" tabindex="-1"></a>    r_mat[i, <span class="dv">3</span>] <span class="ot">&lt;-</span> residual_correlations[<span class="dv">6</span>,<span class="dv">3</span>]</span>
<span id="cb581-210"><a href="lmm.html#cb581-210" tabindex="-1"></a>    r_mat[i, <span class="dv">4</span>] <span class="ot">&lt;-</span> residual_correlations[<span class="dv">2</span>,<span class="dv">1</span>]</span>
<span id="cb581-211"><a href="lmm.html#cb581-211" tabindex="-1"></a>    r_mat[i, <span class="dv">5</span>] <span class="ot">&lt;-</span> residual_correlations[<span class="dv">3</span>,<span class="dv">1</span>]</span>
<span id="cb581-212"><a href="lmm.html#cb581-212" tabindex="-1"></a>    r_mat[i, <span class="dv">6</span>] <span class="ot">&lt;-</span> residual_correlations[<span class="dv">3</span>,<span class="dv">2</span>]</span>
<span id="cb581-213"><a href="lmm.html#cb581-213" tabindex="-1"></a>    r_mat[i, <span class="dv">7</span>] <span class="ot">&lt;-</span> residual_correlations[<span class="dv">5</span>,<span class="dv">4</span>]</span>
<span id="cb581-214"><a href="lmm.html#cb581-214" tabindex="-1"></a>    r_mat[i, <span class="dv">8</span>] <span class="ot">&lt;-</span> residual_correlations[<span class="dv">6</span>,<span class="dv">4</span>]</span>
<span id="cb581-215"><a href="lmm.html#cb581-215" tabindex="-1"></a>    r_mat[i, <span class="dv">9</span>] <span class="ot">&lt;-</span> residual_correlations[<span class="dv">6</span>,<span class="dv">5</span>]</span>
<span id="cb581-216"><a href="lmm.html#cb581-216" tabindex="-1"></a>    </span>
<span id="cb581-217"><a href="lmm.html#cb581-217" tabindex="-1"></a>    <span class="co"># get checks</span></span>
<span id="cb581-218"><a href="lmm.html#cb581-218" tabindex="-1"></a>    <span class="co">#  lmm3_check[i] &lt;- is.null(lmer_check(lmm3))</span></span>
<span id="cb581-219"><a href="lmm.html#cb581-219" tabindex="-1"></a>    lmm4_check[i] <span class="ot">&lt;-</span> <span class="fu">is.null</span>(<span class="fu">lmer_check</span>(lmm4))</span>
<span id="cb581-220"><a href="lmm.html#cb581-220" tabindex="-1"></a>    lmm5_check[i] <span class="ot">&lt;-</span> <span class="fu">is.null</span>(<span class="fu">lmer_check</span>(lmm5))</span>
<span id="cb581-221"><a href="lmm.html#cb581-221" tabindex="-1"></a>    </span>
<span id="cb581-222"><a href="lmm.html#cb581-222" tabindex="-1"></a>  }</span>
<span id="cb581-223"><a href="lmm.html#cb581-223" tabindex="-1"></a>  </span>
<span id="cb581-224"><a href="lmm.html#cb581-224" tabindex="-1"></a>  <span class="co"># lmm3_pairs</span></span>
<span id="cb581-225"><a href="lmm.html#cb581-225" tabindex="-1"></a>  <span class="co"># lmm4_pairs</span></span>
<span id="cb581-226"><a href="lmm.html#cb581-226" tabindex="-1"></a>  </span>
<span id="cb581-227"><a href="lmm.html#cb581-227" tabindex="-1"></a>  big_table <span class="ot">&lt;-</span> <span class="fu">data.table</span>(<span class="cn">NULL</span>)</span>
<span id="cb581-228"><a href="lmm.html#cb581-228" tabindex="-1"></a>  model_labels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;lm0&quot;</span>, <span class="st">&quot;lm1&quot;</span>, <span class="st">&quot;lmm1&quot;</span>, <span class="st">&quot;lmm3&quot;</span>, <span class="st">&quot;lmm4&quot;</span>, <span class="st">&quot;lmm5&quot;</span>, <span class="st">&quot;lmm6&quot;</span>, <span class="st">&quot;r_mat&quot;</span>)</span>
<span id="cb581-229"><a href="lmm.html#cb581-229" tabindex="-1"></a>  models <span class="ot">&lt;-</span> <span class="fu">list</span>(lm0_p, lm1_p, lmm1_p, lmm3_p, lmm4_p, lmm5_p, lmm6_p, r_mat)</span>
<span id="cb581-230"><a href="lmm.html#cb581-230" tabindex="-1"></a>  <span class="cf">for</span>(model_i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(models)){</span>
<span id="cb581-231"><a href="lmm.html#cb581-231" tabindex="-1"></a>   big_table <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb581-232"><a href="lmm.html#cb581-232" tabindex="-1"></a>     big_table,</span>
<span id="cb581-233"><a href="lmm.html#cb581-233" tabindex="-1"></a>    <span class="fu">data.table</span>(</span>
<span id="cb581-234"><a href="lmm.html#cb581-234" tabindex="-1"></a>      <span class="at">sim_id =</span> sim_id_i,</span>
<span id="cb581-235"><a href="lmm.html#cb581-235" tabindex="-1"></a>      <span class="at">sig_block =</span> sigma_block,</span>
<span id="cb581-236"><a href="lmm.html#cb581-236" tabindex="-1"></a>      <span class="at">sig_ss =</span> sigma_ss,</span>
<span id="cb581-237"><a href="lmm.html#cb581-237" tabindex="-1"></a>      <span class="at">sig_main =</span> sigma_block_main,</span>
<span id="cb581-238"><a href="lmm.html#cb581-238" tabindex="-1"></a>      <span class="at">sig_sub =</span> sigma_block_sub,</span>
<span id="cb581-239"><a href="lmm.html#cb581-239" tabindex="-1"></a>      <span class="at">model =</span> model_labels[model_i],</span>
<span id="cb581-240"><a href="lmm.html#cb581-240" tabindex="-1"></a>      models[[model_i]]</span>
<span id="cb581-241"><a href="lmm.html#cb581-241" tabindex="-1"></a>    ))</span>
<span id="cb581-242"><a href="lmm.html#cb581-242" tabindex="-1"></a>  }</span>
<span id="cb581-243"><a href="lmm.html#cb581-243" tabindex="-1"></a></span>
<span id="cb581-244"><a href="lmm.html#cb581-244" tabindex="-1"></a>  outfile_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;rspd-&quot;</span>, sim_id_i, <span class="st">&quot;.Rds&quot;</span>)</span>
<span id="cb581-245"><a href="lmm.html#cb581-245" tabindex="-1"></a>  save_file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_from, outfile_name)</span>
<span id="cb581-246"><a href="lmm.html#cb581-246" tabindex="-1"></a>  </span>
<span id="cb581-247"><a href="lmm.html#cb581-247" tabindex="-1"></a>  <span class="cf">if</span>(writeit <span class="sc">==</span> <span class="cn">TRUE</span>){</span>
<span id="cb581-248"><a href="lmm.html#cb581-248" tabindex="-1"></a>    <span class="fu">saveRDS</span>(<span class="at">object =</span> big_table, <span class="at">file =</span> save_file_path)</span>
<span id="cb581-249"><a href="lmm.html#cb581-249" tabindex="-1"></a>  }</span>
<span id="cb581-250"><a href="lmm.html#cb581-250" tabindex="-1"></a>  </span>
<span id="cb581-251"><a href="lmm.html#cb581-251" tabindex="-1"></a>  <span class="fu">data.table</span>(</span>
<span id="cb581-252"><a href="lmm.html#cb581-252" tabindex="-1"></a>    <span class="at">r =</span> <span class="fu">apply</span>(r_mat, <span class="dv">2</span>, mean),</span>
<span id="cb581-253"><a href="lmm.html#cb581-253" tabindex="-1"></a>    <span class="at">lm0 =</span> <span class="fu">apply</span>(lm0_p, <span class="dv">2</span>, pless),</span>
<span id="cb581-254"><a href="lmm.html#cb581-254" tabindex="-1"></a>    <span class="at">lm1 =</span> <span class="fu">apply</span>(lm1_p, <span class="dv">2</span>, pless),</span>
<span id="cb581-255"><a href="lmm.html#cb581-255" tabindex="-1"></a>    <span class="at">lmm1 =</span> <span class="fu">apply</span>(lmm1_p, <span class="dv">2</span>, pless),</span>
<span id="cb581-256"><a href="lmm.html#cb581-256" tabindex="-1"></a>    <span class="at">lmm3 =</span> <span class="fu">apply</span>(lmm3_p, <span class="dv">2</span>, pless),</span>
<span id="cb581-257"><a href="lmm.html#cb581-257" tabindex="-1"></a>    <span class="at">lmm4 =</span> <span class="fu">apply</span>(lmm4_p, <span class="dv">2</span>, pless),</span>
<span id="cb581-258"><a href="lmm.html#cb581-258" tabindex="-1"></a>    <span class="at">lmm5 =</span> <span class="fu">apply</span>(lmm5_p, <span class="dv">2</span>, pless),</span>
<span id="cb581-259"><a href="lmm.html#cb581-259" tabindex="-1"></a>    <span class="at">lmm6 =</span> <span class="fu">apply</span>(lmm5_p, <span class="dv">2</span>, pless)</span>
<span id="cb581-260"><a href="lmm.html#cb581-260" tabindex="-1"></a>  )</span>
<span id="cb581-261"><a href="lmm.html#cb581-261" tabindex="-1"></a>  </span>
<span id="cb581-262"><a href="lmm.html#cb581-262" tabindex="-1"></a>  <span class="co"># data.table(</span></span>
<span id="cb581-263"><a href="lmm.html#cb581-263" tabindex="-1"></a>  <span class="co">#   all = apply(lmm3_p, 2, pless),</span></span>
<span id="cb581-264"><a href="lmm.html#cb581-264" tabindex="-1"></a>  <span class="co">#   fail = apply(lmm3_p[lmm3_check == FALSE, ], 2, pless),</span></span>
<span id="cb581-265"><a href="lmm.html#cb581-265" tabindex="-1"></a>  <span class="co">#   pass = apply(lmm3_p[lmm3_check == TRUE,], 2, pless)</span></span>
<span id="cb581-266"><a href="lmm.html#cb581-266" tabindex="-1"></a>  <span class="co"># )</span></span>
<span id="cb581-267"><a href="lmm.html#cb581-267" tabindex="-1"></a>  <span class="co"># </span></span>
<span id="cb581-268"><a href="lmm.html#cb581-268" tabindex="-1"></a>  <span class="co"># data.table(</span></span>
<span id="cb581-269"><a href="lmm.html#cb581-269" tabindex="-1"></a>  <span class="co">#   all = apply(lmm4_p, 2, pless),</span></span>
<span id="cb581-270"><a href="lmm.html#cb581-270" tabindex="-1"></a>  <span class="co">#   fail = apply(lmm4_p[lmm3_check == FALSE, ], 2, pless),</span></span>
<span id="cb581-271"><a href="lmm.html#cb581-271" tabindex="-1"></a>  <span class="co">#   pass = apply(lmm4_p[lmm3_check == TRUE,], 2, pless)</span></span>
<span id="cb581-272"><a href="lmm.html#cb581-272" tabindex="-1"></a>  <span class="co"># )</span></span>
<span id="cb581-273"><a href="lmm.html#cb581-273" tabindex="-1"></a>  </span>
<span id="cb581-274"><a href="lmm.html#cb581-274" tabindex="-1"></a>}</span>
<span id="cb581-275"><a href="lmm.html#cb581-275" tabindex="-1"></a></span>
<span id="cb581-276"><a href="lmm.html#cb581-276" tabindex="-1"></a></span>
<span id="cb581-277"><a href="lmm.html#cb581-277" tabindex="-1"></a><span class="cf">if</span>(readit <span class="sc">==</span> <span class="cn">TRUE</span>){</span>
<span id="cb581-278"><a href="lmm.html#cb581-278" tabindex="-1"></a>  <span class="co"># combine 4 data sets and summarize</span></span>
<span id="cb581-279"><a href="lmm.html#cb581-279" tabindex="-1"></a>  res_out <span class="ot">&lt;-</span> <span class="fu">data.table</span>(<span class="cn">NULL</span>)</span>
<span id="cb581-280"><a href="lmm.html#cb581-280" tabindex="-1"></a>  params <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;sim_id&quot;</span>, <span class="st">&quot;sig_block&quot;</span>, <span class="st">&quot;sig_ss&quot;</span>, <span class="st">&quot;sig_main&quot;</span>, <span class="st">&quot;sig_sub&quot;</span>)</span>
<span id="cb581-281"><a href="lmm.html#cb581-281" tabindex="-1"></a>  p_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;V1&quot;</span>, <span class="st">&quot;V2&quot;</span>, <span class="st">&quot;V3&quot;</span>, <span class="st">&quot;V4&quot;</span>, <span class="st">&quot;V5&quot;</span>, <span class="st">&quot;V6&quot;</span>, <span class="st">&quot;V7&quot;</span>, <span class="st">&quot;V8&quot;</span>, <span class="st">&quot;V9&quot;</span>)</span>
<span id="cb581-282"><a href="lmm.html#cb581-282" tabindex="-1"></a>  <span class="cf">for</span>(sim_id_i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>){</span>
<span id="cb581-283"><a href="lmm.html#cb581-283" tabindex="-1"></a>    outfile_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;rspd-&quot;</span>, sim_id_i, <span class="st">&quot;.Rds&quot;</span>)</span>
<span id="cb581-284"><a href="lmm.html#cb581-284" tabindex="-1"></a>    save_file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_from, outfile_name)</span>
<span id="cb581-285"><a href="lmm.html#cb581-285" tabindex="-1"></a>    big_table <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(save_file_path)</span>
<span id="cb581-286"><a href="lmm.html#cb581-286" tabindex="-1"></a>    r_mat <span class="ot">&lt;-</span> big_table[model <span class="sc">==</span> <span class="st">&quot;r_mat&quot;</span>, ]</span>
<span id="cb581-287"><a href="lmm.html#cb581-287" tabindex="-1"></a>    lm0_p <span class="ot">&lt;-</span> big_table[model <span class="sc">==</span> <span class="st">&quot;lm0&quot;</span>, ]</span>
<span id="cb581-288"><a href="lmm.html#cb581-288" tabindex="-1"></a>    lm1_p <span class="ot">&lt;-</span> big_table[model <span class="sc">==</span> <span class="st">&quot;lm1&quot;</span>, ]</span>
<span id="cb581-289"><a href="lmm.html#cb581-289" tabindex="-1"></a>    lmm1_p <span class="ot">&lt;-</span> big_table[model <span class="sc">==</span> <span class="st">&quot;lmm1&quot;</span>, ]</span>
<span id="cb581-290"><a href="lmm.html#cb581-290" tabindex="-1"></a>    lmm3_p <span class="ot">&lt;-</span> big_table[model <span class="sc">==</span> <span class="st">&quot;lmm3&quot;</span>, ]</span>
<span id="cb581-291"><a href="lmm.html#cb581-291" tabindex="-1"></a>    lmm4_p <span class="ot">&lt;-</span> big_table[model <span class="sc">==</span> <span class="st">&quot;lmm4&quot;</span>, ]</span>
<span id="cb581-292"><a href="lmm.html#cb581-292" tabindex="-1"></a>    lmm5_p <span class="ot">&lt;-</span> big_table[model <span class="sc">==</span> <span class="st">&quot;lmm5&quot;</span>, ]</span>
<span id="cb581-293"><a href="lmm.html#cb581-293" tabindex="-1"></a>    lmm6_p <span class="ot">&lt;-</span> big_table[model <span class="sc">==</span> <span class="st">&quot;lmm6&quot;</span>, ]</span>
<span id="cb581-294"><a href="lmm.html#cb581-294" tabindex="-1"></a>    out_table <span class="ot">&lt;-</span> <span class="fu">data.table</span>(</span>
<span id="cb581-295"><a href="lmm.html#cb581-295" tabindex="-1"></a>      <span class="at">contrast =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&quot;main&quot;</span>, <span class="st">&quot;sub&quot;</span>), <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">6</span>)),</span>
<span id="cb581-296"><a href="lmm.html#cb581-296" tabindex="-1"></a>      <span class="at">r =</span> <span class="fu">apply</span>(r_mat[, .SD, <span class="at">.SDcols =</span> p_cols], <span class="dv">2</span>, mean),</span>
<span id="cb581-297"><a href="lmm.html#cb581-297" tabindex="-1"></a>      <span class="at">lm0 =</span> <span class="fu">apply</span>(lm0_p[, .SD, <span class="at">.SDcols =</span> p_cols], <span class="dv">2</span>, pless),</span>
<span id="cb581-298"><a href="lmm.html#cb581-298" tabindex="-1"></a>      <span class="at">lm1 =</span> <span class="fu">apply</span>(lm1_p[, .SD, <span class="at">.SDcols =</span> p_cols], <span class="dv">2</span>, pless),</span>
<span id="cb581-299"><a href="lmm.html#cb581-299" tabindex="-1"></a>      <span class="at">lmm1 =</span> <span class="fu">apply</span>(lmm1_p[, .SD, <span class="at">.SDcols =</span> p_cols], <span class="dv">2</span>, pless),</span>
<span id="cb581-300"><a href="lmm.html#cb581-300" tabindex="-1"></a>      <span class="at">lmm3 =</span> <span class="fu">apply</span>(lmm3_p[, .SD, <span class="at">.SDcols =</span> p_cols], <span class="dv">2</span>, pless),</span>
<span id="cb581-301"><a href="lmm.html#cb581-301" tabindex="-1"></a>      <span class="at">lmm4 =</span> <span class="fu">apply</span>(lmm4_p[, .SD, <span class="at">.SDcols =</span> p_cols], <span class="dv">2</span>, pless),</span>
<span id="cb581-302"><a href="lmm.html#cb581-302" tabindex="-1"></a>      <span class="at">lmm5 =</span> <span class="fu">apply</span>(lmm5_p[, .SD, <span class="at">.SDcols =</span> p_cols], <span class="dv">2</span>, pless),</span>
<span id="cb581-303"><a href="lmm.html#cb581-303" tabindex="-1"></a>      <span class="at">lmm6 =</span> <span class="fu">apply</span>(lmm6_p[, .SD, <span class="at">.SDcols =</span> p_cols], <span class="dv">2</span>, pless)</span>
<span id="cb581-304"><a href="lmm.html#cb581-304" tabindex="-1"></a>    )</span>
<span id="cb581-305"><a href="lmm.html#cb581-305" tabindex="-1"></a>    summary_i <span class="ot">&lt;-</span> out_table[, .(<span class="at">r =</span> <span class="fu">mean</span>(r),</span>
<span id="cb581-306"><a href="lmm.html#cb581-306" tabindex="-1"></a>                               <span class="at">lm0 =</span> <span class="fu">mean</span>(lm0),</span>
<span id="cb581-307"><a href="lmm.html#cb581-307" tabindex="-1"></a>                               <span class="at">lm1 =</span> <span class="fu">mean</span>(lm1),</span>
<span id="cb581-308"><a href="lmm.html#cb581-308" tabindex="-1"></a>                               <span class="at">lmm1 =</span> <span class="fu">mean</span>(lmm1),</span>
<span id="cb581-309"><a href="lmm.html#cb581-309" tabindex="-1"></a>                               <span class="at">lmm3 =</span> <span class="fu">mean</span>(lmm3),</span>
<span id="cb581-310"><a href="lmm.html#cb581-310" tabindex="-1"></a>                               <span class="at">lmm4 =</span> <span class="fu">mean</span>(lmm4),</span>
<span id="cb581-311"><a href="lmm.html#cb581-311" tabindex="-1"></a>                               <span class="at">lmm5 =</span> <span class="fu">mean</span>(lmm5),</span>
<span id="cb581-312"><a href="lmm.html#cb581-312" tabindex="-1"></a>                               <span class="at">lmm6 =</span> <span class="fu">mean</span>(lmm6)),</span>
<span id="cb581-313"><a href="lmm.html#cb581-313" tabindex="-1"></a>                           by <span class="ot">=</span> .(contrast)]</span>
<span id="cb581-314"><a href="lmm.html#cb581-314" tabindex="-1"></a>    res_out <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb581-315"><a href="lmm.html#cb581-315" tabindex="-1"></a>      res_out,</span>
<span id="cb581-316"><a href="lmm.html#cb581-316" tabindex="-1"></a>      <span class="fu">data.table</span>(</span>
<span id="cb581-317"><a href="lmm.html#cb581-317" tabindex="-1"></a>        <span class="at">sim_id  =</span> big_table[<span class="dv">1</span>, sim_id],</span>
<span id="cb581-318"><a href="lmm.html#cb581-318" tabindex="-1"></a>        <span class="at">sig_block  =</span> big_table[<span class="dv">1</span>, sig_block],</span>
<span id="cb581-319"><a href="lmm.html#cb581-319" tabindex="-1"></a>        <span class="at">sig_ss  =</span> big_table[<span class="dv">1</span>, sig_ss],</span>
<span id="cb581-320"><a href="lmm.html#cb581-320" tabindex="-1"></a>        <span class="at">sig_main  =</span> big_table[<span class="dv">1</span>, sig_main],</span>
<span id="cb581-321"><a href="lmm.html#cb581-321" tabindex="-1"></a>        <span class="at">sig_sub  =</span> big_table[<span class="dv">1</span>, sig_sub],</span>
<span id="cb581-322"><a href="lmm.html#cb581-322" tabindex="-1"></a>        summary_i</span>
<span id="cb581-323"><a href="lmm.html#cb581-323" tabindex="-1"></a>      )</span>
<span id="cb581-324"><a href="lmm.html#cb581-324" tabindex="-1"></a>    )</span>
<span id="cb581-325"><a href="lmm.html#cb581-325" tabindex="-1"></a>    </span>
<span id="cb581-326"><a href="lmm.html#cb581-326" tabindex="-1"></a>  }</span>
<span id="cb581-327"><a href="lmm.html#cb581-327" tabindex="-1"></a>  </span>
<span id="cb581-328"><a href="lmm.html#cb581-328" tabindex="-1"></a>  res_out <span class="sc">%&gt;%</span></span>
<span id="cb581-329"><a href="lmm.html#cb581-329" tabindex="-1"></a>    <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">3</span>),</span>
<span id="cb581-330"><a href="lmm.html#cb581-330" tabindex="-1"></a>          <span class="at">caption =</span> <span class="st">&quot;Type I error rate for the RSPD statistical models under different models of random variance. The design is 2 (main plot: WT, KO) x 3 (subplot: Cn, Tr1, Tr2) with ten blocks. All four simulations have a component of variance due to block. Sim 1 includes a block:main plot component, Sim 2  includes a block:sub plot component, Sim 3 includes both block:main and block:sub components. Type I error rates for the 9 simple effect contrasts were averaged within the two sets: contrast = main is the aggregate of the single main plot contrast (KO - WT) at each level of the subplot factor. contrast = sub is the aggregate of the three subplot contrasts (Tr1 - Cn, Tr2 - Cn, Tr2 - Tr1) at each level of the main plot factor. The Cor Error column is the average correlated error of the residuals fit by a simple linear model y ~ treatment. Model abbreviations as in text.&quot;</span></span>
<span id="cb581-331"><a href="lmm.html#cb581-331" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb581-332"><a href="lmm.html#cb581-332" tabindex="-1"></a>    <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb581-333"><a href="lmm.html#cb581-333" tabindex="-1"></a>    <span class="fu">pack_rows</span>(<span class="st">&quot;sim 1 - block:main plot&quot;</span>, <span class="dv">1</span>, <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb581-334"><a href="lmm.html#cb581-334" tabindex="-1"></a>    <span class="fu">pack_rows</span>(<span class="st">&quot;sim 2 - block:sub plot&quot;</span>, <span class="dv">3</span>, <span class="dv">4</span>) <span class="sc">%&gt;%</span></span>
<span id="cb581-335"><a href="lmm.html#cb581-335" tabindex="-1"></a>    <span class="fu">pack_rows</span>(<span class="st">&quot;sim 3 - block:main + block:sub&quot;</span>, <span class="dv">5</span>, <span class="dv">6</span>) <span class="sc">%&gt;%</span></span>
<span id="cb581-336"><a href="lmm.html#cb581-336" tabindex="-1"></a>    <span class="fu">pack_rows</span>(<span class="st">&quot;sim 4 - block only&quot;</span>, <span class="dv">7</span>, <span class="dv">8</span>)</span>
<span id="cb581-337"><a href="lmm.html#cb581-337" tabindex="-1"></a>}</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:rspd-sim">Table 16.20: </span>Type I error rate for the RSPD statistical models under different models of random variance. The design is 2 (main plot: WT, KO) x 3 (subplot: Cn, Tr1, Tr2) with ten blocks. All four simulations have a component of variance due to block. Sim 1 includes a block:main plot component, Sim 2 includes a block:sub plot component, Sim 3 includes both block:main and block:sub components. Type I error rates for the 9 simple effect contrasts were averaged within the two sets: contrast = main is the aggregate of the single main plot contrast (KO - WT) at each level of the subplot factor. contrast = sub is the aggregate of the three subplot contrasts (Tr1 - Cn, Tr2 - Cn, Tr2 - Tr1) at each level of the main plot factor. The Cor Error column is the average correlated error of the residuals fit by a simple linear model y ~ treatment. Model abbreviations as in text.
</caption>
<thead>
<tr>
<th style="text-align:right;">
sim_id
</th>
<th style="text-align:right;">
sig_block
</th>
<th style="text-align:right;">
sig_ss
</th>
<th style="text-align:right;">
sig_main
</th>
<th style="text-align:right;">
sig_sub
</th>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
r
</th>
<th style="text-align:right;">
lm0
</th>
<th style="text-align:right;">
lm1
</th>
<th style="text-align:right;">
lmm1
</th>
<th style="text-align:right;">
lmm3
</th>
<th style="text-align:right;">
lmm4
</th>
<th style="text-align:right;">
lmm5
</th>
<th style="text-align:right;">
lmm6
</th>
</tr>
</thead>
<tbody>
<tr grouplength="2">
<td colspan="14" style="border-bottom: 1px solid;">
<strong>sim 1 - block:main plot</strong>
</td>
</tr>
<tr>
<td style="text-align:right;padding-left: 2em;" indentlevel="1">
1
</td>
<td style="text-align:right;">
0.8
</td>
<td style="text-align:right;">
0.5
</td>
<td style="text-align:right;">
0.6
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:left;">
main
</td>
<td style="text-align:right;">
0.494
</td>
<td style="text-align:right;">
0.010
</td>
<td style="text-align:right;">
0.199
</td>
<td style="text-align:right;">
0.087
</td>
<td style="text-align:right;">
0.051
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
0.054
</td>
</tr>
<tr>
<td style="text-align:right;padding-left: 2em;" indentlevel="1">
1
</td>
<td style="text-align:right;">
0.8
</td>
<td style="text-align:right;">
0.5
</td>
<td style="text-align:right;">
0.6
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:left;">
sub
</td>
<td style="text-align:right;">
0.783
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
0.011
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
0.011
</td>
<td style="text-align:right;">
0.052
</td>
</tr>
<tr grouplength="2">
<td colspan="14" style="border-bottom: 1px solid;">
<strong>sim 2 - block:sub plot</strong>
</td>
</tr>
<tr>
<td style="text-align:right;padding-left: 2em;" indentlevel="1">
2
</td>
<td style="text-align:right;">
0.8
</td>
<td style="text-align:right;">
0.5
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:right;">
0.6
</td>
<td style="text-align:left;">
main
</td>
<td style="text-align:right;">
0.782
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.003
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.003
</td>
<td style="text-align:right;">
0.042
</td>
<td style="text-align:right;">
0.041
</td>
</tr>
<tr>
<td style="text-align:right;padding-left: 2em;" indentlevel="1">
2
</td>
<td style="text-align:right;">
0.8
</td>
<td style="text-align:right;">
0.5
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:right;">
0.6
</td>
<td style="text-align:left;">
sub
</td>
<td style="text-align:right;">
0.497
</td>
<td style="text-align:right;">
0.008
</td>
<td style="text-align:right;">
0.053
</td>
<td style="text-align:right;">
0.066
</td>
<td style="text-align:right;">
0.068
</td>
<td style="text-align:right;">
0.065
</td>
<td style="text-align:right;">
0.050
</td>
<td style="text-align:right;">
0.053
</td>
</tr>
<tr grouplength="2">
<td colspan="14" style="border-bottom: 1px solid;">
<strong>sim 3 - block:main + block:sub</strong>
</td>
</tr>
<tr>
<td style="text-align:right;padding-left: 2em;" indentlevel="1">
3
</td>
<td style="text-align:right;">
0.8
</td>
<td style="text-align:right;">
0.5
</td>
<td style="text-align:right;">
0.6
</td>
<td style="text-align:right;">
0.4
</td>
<td style="text-align:left;">
main
</td>
<td style="text-align:right;">
0.549
</td>
<td style="text-align:right;">
0.007
</td>
<td style="text-align:right;">
0.111
</td>
<td style="text-align:right;">
0.062
</td>
<td style="text-align:right;">
0.038
</td>
<td style="text-align:right;">
0.039
</td>
<td style="text-align:right;">
0.068
</td>
<td style="text-align:right;">
0.054
</td>
</tr>
<tr>
<td style="text-align:right;padding-left: 2em;" indentlevel="1">
3
</td>
<td style="text-align:right;">
0.8
</td>
<td style="text-align:right;">
0.5
</td>
<td style="text-align:right;">
0.6
</td>
<td style="text-align:right;">
0.4
</td>
<td style="text-align:left;">
sub
</td>
<td style="text-align:right;">
0.689
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
0.056
</td>
<td style="text-align:right;">
0.026
</td>
<td style="text-align:right;">
0.057
</td>
<td style="text-align:right;">
0.056
</td>
<td style="text-align:right;">
0.024
</td>
<td style="text-align:right;">
0.054
</td>
</tr>
<tr grouplength="2">
<td colspan="14" style="border-bottom: 1px solid;">
<strong>sim 4 - block only</strong>
</td>
</tr>
<tr>
<td style="text-align:right;padding-left: 2em;" indentlevel="1">
4
</td>
<td style="text-align:right;">
0.8
</td>
<td style="text-align:right;">
0.5
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:left;">
main
</td>
<td style="text-align:right;">
0.707
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.049
</td>
<td style="text-align:right;">
0.049
</td>
<td style="text-align:right;">
0.044
</td>
<td style="text-align:right;">
0.045
</td>
<td style="text-align:right;">
0.056
</td>
<td style="text-align:right;">
0.054
</td>
</tr>
<tr>
<td style="text-align:right;padding-left: 2em;" indentlevel="1">
4
</td>
<td style="text-align:right;">
0.8
</td>
<td style="text-align:right;">
0.5
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:right;">
0.0
</td>
<td style="text-align:left;">
sub
</td>
<td style="text-align:right;">
0.704
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.057
</td>
<td style="text-align:right;">
0.055
</td>
<td style="text-align:right;">
0.061
</td>
<td style="text-align:right;">
0.060
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
0.059
</td>
</tr>
</tbody>
</table>
</div>
<div id="rcbds" class="section level3 hasAnchor" number="16.10.5">
<h3><span class="header-section-number">16.10.5</span> RCBDS<a href="lmm.html#rcbds" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>lmm3 – random intercept and slopes model</li>
<li>lmm4 – random interaction intercept model</li>
<li>lmm1 – random intercept model on batch means</li>
<li>lmm2 – lmm for correlated error on batch means</li>
<li>aov1 – multivariate RM-ANOVA</li>
<li>aov2 – univariate RM-ANOVA</li>
<li>pptt – pairwise, paired t-test</li>
</ol>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>lmm3 is the “maximal model” – it fits the most parameters and has the fewest assumptions. lmm1 on the full data is the minimal model. I’m not recommending this at all as this is pseudoreplication and will result in very anti-conservative inference. lmm4 is less conservative than lmm3 (and makes more assumptions).</li>
<li>lmm1 and lmm2 <em>here</em> are the models for RCBD designs but using the aggregated data - that is the subsampled replicates within a batch (block:treatment combination) are averaged.</li>
<li>If design is balanced
<ul>
<li>lmm3 and aov1 result in same estimates, SE, CIs, and p-values</li>
<li>lmm4, lmm1, and aov2 result in same estimates, SE, CIs, and p-values</li>
<li>lmm2 has same estimates and SE as lmm3/aov1 but more df so less conservative</li>
</ul></li>
<li>If a treatment within a block is missing, the whole block is deleted in the RM-ANOVA models. This reduces power</li>
<li>the lmm models are very flexible – covariates can be added or these can be used as generalized lmms for non-normal distributions.</li>
<li>the lmm models, and especially lmm2 and lmm3, sometimes (often?) fail to converge, especially with small samples (small number of blocks) or if within block correlation is small</li>
<li>lmm1, lmm4, and aov2 will be less conservative (more power but at higher type I error/false positive rate)</li>
</ol>
<p>Best practices</p>
<ol style="list-style-type: decimal">
<li>If balanced and no covariates then use lmm3/lmm2/aov1 if discovery is expensive (want to avoid false positives) or lmm4/aov2/lmm1 if discovery is cheap (can afford false positives). lmm2 is less conservative than lmm3/aov1. Use lmm3 if interested in the variance at different levels and/or covariance structure.</li>
<li>If unbalanced, or if there are covariates, then start with lmm3 or lmm2. Use lmeControl if lmm2 fails. If these fail, go to 1.</li>
</ol>
</div>
<div id="grcbd" class="section level3 hasAnchor" number="16.10.6">
<h3><span class="header-section-number">16.10.6</span> GRCBD<a href="lmm.html#grcbd" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>lmm3 – random intercept and slopes model</li>
<li>lmm4 – random interaction intercept model</li>
<li>lmm1 – random intercept model on block:treatment means</li>
<li>lmm2 – lmm for correlated error on block:treatment means</li>
<li>aov1 – multivariate RM-ANOVA</li>
<li>aov2 – univariate RM-ANOVA</li>
<li>lm1 – fixed effect model on all data</li>
<li>pptt – pairwise, paired t-test on block:treatment means</li>
</ol>
</div>
<div id="grcbds" class="section level3 hasAnchor" number="16.10.7">
<h3><span class="header-section-number">16.10.7</span> GRCBDS<a href="lmm.html#grcbds" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>lmm3 – random intercept and slopes model</li>
<li>lmm4 – random interaction intercept model</li>
<li>lmm1 – random intercept model on block:treatment means</li>
<li>lmm2 – lmm for correlated error on block:treatment means</li>
<li>aov1 – multivariate RM-ANOVA</li>
<li>aov2 – univariate RM-ANOVA</li>
<li>lm1 – fixed effect model on all block:treatment:rep means</li>
<li>pptt – pairwise, paired t-test on block:treatment:rep means</li>
</ol>
</div>
</div>
<div id="working-in-r-5" class="section level2 hasAnchor" number="16.11">
<h2><span class="header-section-number">16.11</span> Working in R<a href="lmm.html#working-in-r-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="plotting-models-fit-to-batched-data" class="section level3 hasAnchor" number="16.11.1">
<h3><span class="header-section-number">16.11.1</span> Plotting models fit to batched data<a href="lmm.html#plotting-models-fit-to-batched-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="models-without-subsampling---experiment-6g" class="section level4 hasAnchor" number="16.11.1.1">
<h4><span class="header-section-number">16.11.1.1</span> Models without subsampling - Experiment 6g<a href="lmm.html#models-without-subsampling---experiment-6g" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Data wrangling necessary for plot:</p>
<div class="sourceCode" id="cb582"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb582-1"><a href="lmm.html#cb582-1" tabindex="-1"></a><span class="co"># convert contrast table to a data.table</span></span>
<span id="cb582-2"><a href="lmm.html#cb582-2" tabindex="-1"></a>exp6g_m1_planned_dt <span class="ot">&lt;-</span> <span class="fu">data.table</span>(exp6g_m1_planned)</span>
<span id="cb582-3"><a href="lmm.html#cb582-3" tabindex="-1"></a></span>
<span id="cb582-4"><a href="lmm.html#cb582-4" tabindex="-1"></a><span class="co"># create a pretty p-value column</span></span>
<span id="cb582-5"><a href="lmm.html#cb582-5" tabindex="-1"></a>exp6g_m1_planned_dt[, pretty_p <span class="sc">:=</span> <span class="fu">pvalString</span>(p.value)]</span>
<span id="cb582-6"><a href="lmm.html#cb582-6" tabindex="-1"></a></span>
<span id="cb582-7"><a href="lmm.html#cb582-7" tabindex="-1"></a><span class="co"># add group1 and group2 columns to exp1g_m1_planned</span></span>
<span id="cb582-8"><a href="lmm.html#cb582-8" tabindex="-1"></a>exp6g_m1_planned_dt[, group1 <span class="sc">:=</span> <span class="fu">c</span>(<span class="st">&quot;Saline&quot;</span>, <span class="st">&quot;CNO&quot;</span>, <span class="st">&quot;Post_CNO&quot;</span>)]</span>
<span id="cb582-9"><a href="lmm.html#cb582-9" tabindex="-1"></a>exp6g_m1_planned_dt[, group2 <span class="sc">:=</span> <span class="fu">c</span>(<span class="st">&quot;Lesion&quot;</span>, <span class="st">&quot;Saline&quot;</span>, <span class="st">&quot;CNO&quot;</span>)]</span></code></pre></div>
<p>Experiments are colored:</p>
<div class="sourceCode" id="cb583"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb583-1"><a href="lmm.html#cb583-1" tabindex="-1"></a>gg1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> exp6g,</span>
<span id="cb583-2"><a href="lmm.html#cb583-2" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb583-3"><a href="lmm.html#cb583-3" tabindex="-1"></a>                  <span class="at">y =</span> touch,</span>
<span id="cb583-4"><a href="lmm.html#cb583-4" tabindex="-1"></a>                  <span class="at">color =</span> mouse_id)) <span class="sc">+</span></span>
<span id="cb583-5"><a href="lmm.html#cb583-5" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb583-6"><a href="lmm.html#cb583-6" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> mouse_id),</span>
<span id="cb583-7"><a href="lmm.html#cb583-7" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.2</span>),</span>
<span id="cb583-8"><a href="lmm.html#cb583-8" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">&quot;gray80&quot;</span>) <span class="sc">+</span></span>
<span id="cb583-9"><a href="lmm.html#cb583-9" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> <span class="fu">summary</span>(exp6g_m1_emm),</span>
<span id="cb583-10"><a href="lmm.html#cb583-10" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> emmean),</span>
<span id="cb583-11"><a href="lmm.html#cb583-11" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb583-12"><a href="lmm.html#cb583-12" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb583-13"><a href="lmm.html#cb583-13" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> <span class="fu">summary</span>(exp6g_m1_emm),</span>
<span id="cb583-14"><a href="lmm.html#cb583-14" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> emmean,</span>
<span id="cb583-15"><a href="lmm.html#cb583-15" tabindex="-1"></a>                 <span class="at">ymin =</span> lower.CL,</span>
<span id="cb583-16"><a href="lmm.html#cb583-16" tabindex="-1"></a>                 <span class="at">ymax =</span> upper.CL),</span>
<span id="cb583-17"><a href="lmm.html#cb583-17" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb583-18"><a href="lmm.html#cb583-18" tabindex="-1"></a>             <span class="at">width =</span> .<span class="dv">05</span>) <span class="sc">+</span></span>
<span id="cb583-19"><a href="lmm.html#cb583-19" tabindex="-1"></a></span>
<span id="cb583-20"><a href="lmm.html#cb583-20" tabindex="-1"></a>  <span class="co"># pvalue brackets</span></span>
<span id="cb583-21"><a href="lmm.html#cb583-21" tabindex="-1"></a>  <span class="fu">stat_pvalue_manual</span>(exp6g_m1_planned_dt,</span>
<span id="cb583-22"><a href="lmm.html#cb583-22" tabindex="-1"></a>                     <span class="at">label =</span> <span class="st">&quot;pretty_p&quot;</span>,</span>
<span id="cb583-23"><a href="lmm.html#cb583-23" tabindex="-1"></a>                     <span class="at">y.position =</span> <span class="fu">c</span>(<span class="dv">99</span>,<span class="dv">97</span>,<span class="dv">95</span>),</span>
<span id="cb583-24"><a href="lmm.html#cb583-24" tabindex="-1"></a>                     <span class="at">size =</span> <span class="fl">2.5</span>,</span>
<span id="cb583-25"><a href="lmm.html#cb583-25" tabindex="-1"></a>                     <span class="at">tip.length =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb583-26"><a href="lmm.html#cb583-26" tabindex="-1"></a></span>
<span id="cb583-27"><a href="lmm.html#cb583-27" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Percent ipsilateral touch&quot;</span>) <span class="sc">+</span></span>
<span id="cb583-28"><a href="lmm.html#cb583-28" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal_okabe_ito_blue) <span class="sc">+</span></span>
<span id="cb583-29"><a href="lmm.html#cb583-29" tabindex="-1"></a>  <span class="fu">theme_pubr</span>() <span class="sc">+</span></span>
<span id="cb583-30"><a href="lmm.html#cb583-30" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb583-31"><a href="lmm.html#cb583-31" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), <span class="co"># no x-axis title</span></span>
<span id="cb583-32"><a href="lmm.html#cb583-32" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb583-33"><a href="lmm.html#cb583-33" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb583-34"><a href="lmm.html#cb583-34" tabindex="-1"></a>  <span class="cn">NULL</span></span></code></pre></div>
<p>Treatments are colored:</p>
<div class="sourceCode" id="cb584"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb584-1"><a href="lmm.html#cb584-1" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> exp6g,</span>
<span id="cb584-2"><a href="lmm.html#cb584-2" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb584-3"><a href="lmm.html#cb584-3" tabindex="-1"></a>                  <span class="at">y =</span> touch)) <span class="sc">+</span></span>
<span id="cb584-4"><a href="lmm.html#cb584-4" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">group =</span> mouse_id),</span>
<span id="cb584-5"><a href="lmm.html#cb584-5" tabindex="-1"></a>             <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.2</span>),</span>
<span id="cb584-6"><a href="lmm.html#cb584-6" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">&quot;gray&quot;</span>) <span class="sc">+</span></span>
<span id="cb584-7"><a href="lmm.html#cb584-7" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> mouse_id),</span>
<span id="cb584-8"><a href="lmm.html#cb584-8" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.2</span>),</span>
<span id="cb584-9"><a href="lmm.html#cb584-9" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">&quot;gray80&quot;</span>) <span class="sc">+</span></span>
<span id="cb584-10"><a href="lmm.html#cb584-10" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> <span class="fu">summary</span>(exp6g_m1_emm),</span>
<span id="cb584-11"><a href="lmm.html#cb584-11" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> emmean,</span>
<span id="cb584-12"><a href="lmm.html#cb584-12" tabindex="-1"></a>             <span class="at">color =</span> treatment),</span>
<span id="cb584-13"><a href="lmm.html#cb584-13" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb584-14"><a href="lmm.html#cb584-14" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> <span class="fu">summary</span>(exp6g_m1_emm),</span>
<span id="cb584-15"><a href="lmm.html#cb584-15" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> emmean,</span>
<span id="cb584-16"><a href="lmm.html#cb584-16" tabindex="-1"></a>                 <span class="at">ymin =</span> lower.CL,</span>
<span id="cb584-17"><a href="lmm.html#cb584-17" tabindex="-1"></a>                 <span class="at">ymax =</span> upper.CL,</span>
<span id="cb584-18"><a href="lmm.html#cb584-18" tabindex="-1"></a>             <span class="at">color =</span> treatment),</span>
<span id="cb584-19"><a href="lmm.html#cb584-19" tabindex="-1"></a>             <span class="at">width =</span> .<span class="dv">05</span>) <span class="sc">+</span></span>
<span id="cb584-20"><a href="lmm.html#cb584-20" tabindex="-1"></a>  </span>
<span id="cb584-21"><a href="lmm.html#cb584-21" tabindex="-1"></a>  <span class="co"># pvalue brackets</span></span>
<span id="cb584-22"><a href="lmm.html#cb584-22" tabindex="-1"></a>  <span class="fu">stat_pvalue_manual</span>(exp6g_m1_planned_dt,</span>
<span id="cb584-23"><a href="lmm.html#cb584-23" tabindex="-1"></a>                     <span class="at">label =</span> <span class="st">&quot;pretty_p&quot;</span>,</span>
<span id="cb584-24"><a href="lmm.html#cb584-24" tabindex="-1"></a>                     <span class="at">y.position =</span> <span class="fu">c</span>(<span class="dv">99</span>,<span class="dv">97</span>,<span class="dv">95</span>),</span>
<span id="cb584-25"><a href="lmm.html#cb584-25" tabindex="-1"></a>                     <span class="at">size =</span> <span class="fl">2.5</span>,</span>
<span id="cb584-26"><a href="lmm.html#cb584-26" tabindex="-1"></a>                     <span class="at">tip.length =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb584-27"><a href="lmm.html#cb584-27" tabindex="-1"></a></span>
<span id="cb584-28"><a href="lmm.html#cb584-28" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Percent ipsilateral touch&quot;</span>) <span class="sc">+</span></span>
<span id="cb584-29"><a href="lmm.html#cb584-29" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal_okabe_ito_blue) <span class="sc">+</span></span>
<span id="cb584-30"><a href="lmm.html#cb584-30" tabindex="-1"></a>  <span class="fu">theme_pubr</span>() <span class="sc">+</span></span>
<span id="cb584-31"><a href="lmm.html#cb584-31" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb584-32"><a href="lmm.html#cb584-32" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), <span class="co"># no x-axis title</span></span>
<span id="cb584-33"><a href="lmm.html#cb584-33" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb584-34"><a href="lmm.html#cb584-34" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb584-35"><a href="lmm.html#cb584-35" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb584-36"><a href="lmm.html#cb584-36" tabindex="-1"></a></span>
<span id="cb584-37"><a href="lmm.html#cb584-37" tabindex="-1"></a><span class="co">#gg2</span></span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/unnamed-chunk-255-1.png" width="1066.66666666667" /></p>
</div>
<div id="models-with-subsampling---experiment-1g" class="section level4 hasAnchor" number="16.11.1.2">
<h4><span class="header-section-number">16.11.1.2</span> Models with subsampling - Experiment 1g<a href="lmm.html#models-with-subsampling---experiment-1g" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Data wrangling necessary for plot:</p>
<div class="sourceCode" id="cb585"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb585-1"><a href="lmm.html#cb585-1" tabindex="-1"></a><span class="co"># add treatment column to emmeans table</span></span>
<span id="cb585-2"><a href="lmm.html#cb585-2" tabindex="-1"></a>exp1g_m1_emm_dt <span class="ot">&lt;-</span> <span class="fu">summary</span>(exp1g_m1_emm) <span class="sc">%&gt;%</span></span>
<span id="cb585-3"><a href="lmm.html#cb585-3" tabindex="-1"></a>  data.table</span>
<span id="cb585-4"><a href="lmm.html#cb585-4" tabindex="-1"></a>exp1g_m1_emm_dt[, treatment <span class="sc">:=</span> <span class="fu">paste</span>(genotype, sex)]</span>
<span id="cb585-5"><a href="lmm.html#cb585-5" tabindex="-1"></a>exp1g_m1_emm_dt[, treatment <span class="sc">:=</span> <span class="fu">factor</span>(treatment,</span>
<span id="cb585-6"><a href="lmm.html#cb585-6" tabindex="-1"></a>                                      <span class="at">levels =</span> <span class="fu">levels</span>(exp1g<span class="sc">$</span>treatment))]</span>
<span id="cb585-7"><a href="lmm.html#cb585-7" tabindex="-1"></a></span>
<span id="cb585-8"><a href="lmm.html#cb585-8" tabindex="-1"></a><span class="co"># create table of means for each treatment * experiment_id combination</span></span>
<span id="cb585-9"><a href="lmm.html#cb585-9" tabindex="-1"></a>exp1g[, group_mean <span class="sc">:=</span> <span class="fu">predict</span>(exp1g_m1)]</span>
<span id="cb585-10"><a href="lmm.html#cb585-10" tabindex="-1"></a>exp1g_m1_emm2 <span class="ot">&lt;-</span> exp1g[, .(<span class="at">group_mean =</span> <span class="fu">mean</span>(group_mean)),</span>
<span id="cb585-11"><a href="lmm.html#cb585-11" tabindex="-1"></a>                       by <span class="ot">=</span> .(treatment, experiment_id)]</span>
<span id="cb585-12"><a href="lmm.html#cb585-12" tabindex="-1"></a></span>
<span id="cb585-13"><a href="lmm.html#cb585-13" tabindex="-1"></a><span class="co"># add group1 and group2 columns to exp1g_m1_planned</span></span>
<span id="cb585-14"><a href="lmm.html#cb585-14" tabindex="-1"></a>exp1g_m1_planned_dt <span class="ot">&lt;-</span> <span class="fu">data.table</span>(exp1g_m1_planned)</span>
<span id="cb585-15"><a href="lmm.html#cb585-15" tabindex="-1"></a>exp1g_m1_planned_dt[, pretty_p <span class="sc">:=</span> <span class="fu">pvalString</span>(p.value)]</span>
<span id="cb585-16"><a href="lmm.html#cb585-16" tabindex="-1"></a>exp1g_m1_planned_dt[, group1 <span class="sc">:=</span> <span class="fu">c</span>(<span class="st">&quot;Gpr174- M&quot;</span>, <span class="st">&quot;Gpr174- F&quot;</span>, <span class="st">&quot;&quot;</span>)]</span>
<span id="cb585-17"><a href="lmm.html#cb585-17" tabindex="-1"></a>exp1g_m1_planned_dt[, group2 <span class="sc">:=</span> <span class="fu">c</span>(<span class="st">&quot;Gpr174+ M&quot;</span>, <span class="st">&quot;Gpr174+ F&quot;</span>, <span class="st">&quot;&quot;</span>)]</span></code></pre></div>
<div class="sourceCode" id="cb586"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb586-1"><a href="lmm.html#cb586-1" tabindex="-1"></a><span class="co"># get coefficients of model</span></span>
<span id="cb586-2"><a href="lmm.html#cb586-2" tabindex="-1"></a>b <span class="ot">&lt;-</span> exp1g_m1_coef[, <span class="st">&quot;Estimate&quot;</span>]</span>
<span id="cb586-3"><a href="lmm.html#cb586-3" tabindex="-1"></a></span>
<span id="cb586-4"><a href="lmm.html#cb586-4" tabindex="-1"></a><span class="co"># get interaction p</span></span>
<span id="cb586-5"><a href="lmm.html#cb586-5" tabindex="-1"></a>p_ixn <span class="ot">&lt;-</span> exp1g_m1_planned_dt[contrast <span class="sc">==</span> <span class="st">&quot;Interaction&quot;</span>, pretty_p]</span>
<span id="cb586-6"><a href="lmm.html#cb586-6" tabindex="-1"></a></span>
<span id="cb586-7"><a href="lmm.html#cb586-7" tabindex="-1"></a>exp1g_plot_a <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> exp1g,</span>
<span id="cb586-8"><a href="lmm.html#cb586-8" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb586-9"><a href="lmm.html#cb586-9" tabindex="-1"></a>                  <span class="at">y =</span> gc,</span>
<span id="cb586-10"><a href="lmm.html#cb586-10" tabindex="-1"></a>                  <span class="at">color =</span> experiment_id)) <span class="sc">+</span></span>
<span id="cb586-11"><a href="lmm.html#cb586-11" tabindex="-1"></a>  <span class="co"># modeled experiment by treatment means</span></span>
<span id="cb586-12"><a href="lmm.html#cb586-12" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> exp1g_m1_emm2,</span>
<span id="cb586-13"><a href="lmm.html#cb586-13" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">y =</span> group_mean,</span>
<span id="cb586-14"><a href="lmm.html#cb586-14" tabindex="-1"></a>                <span class="at">color =</span> experiment_id),</span>
<span id="cb586-15"><a href="lmm.html#cb586-15" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.4</span>),</span>
<span id="cb586-16"><a href="lmm.html#cb586-16" tabindex="-1"></a>            <span class="at">alpha =</span> <span class="dv">1</span>,</span>
<span id="cb586-17"><a href="lmm.html#cb586-17" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb586-18"><a href="lmm.html#cb586-18" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> exp1g_m1_emm2,</span>
<span id="cb586-19"><a href="lmm.html#cb586-19" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">y =</span> group_mean,</span>
<span id="cb586-20"><a href="lmm.html#cb586-20" tabindex="-1"></a>                <span class="at">group =</span> experiment_id,</span>
<span id="cb586-21"><a href="lmm.html#cb586-21" tabindex="-1"></a>                <span class="at">color =</span> experiment_id),</span>
<span id="cb586-22"><a href="lmm.html#cb586-22" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.4</span>)) <span class="sc">+</span></span>
<span id="cb586-23"><a href="lmm.html#cb586-23" tabindex="-1"></a>  </span>
<span id="cb586-24"><a href="lmm.html#cb586-24" tabindex="-1"></a>  <span class="co"># raw data</span></span>
<span id="cb586-25"><a href="lmm.html#cb586-25" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.4</span>),</span>
<span id="cb586-26"><a href="lmm.html#cb586-26" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.3</span></span>
<span id="cb586-27"><a href="lmm.html#cb586-27" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb586-28"><a href="lmm.html#cb586-28" tabindex="-1"></a>  </span>
<span id="cb586-29"><a href="lmm.html#cb586-29" tabindex="-1"></a>  <span class="co"># modeled treatment means</span></span>
<span id="cb586-30"><a href="lmm.html#cb586-30" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> exp1g_m1_emm_dt,</span>
<span id="cb586-31"><a href="lmm.html#cb586-31" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> emmean),</span>
<span id="cb586-32"><a href="lmm.html#cb586-32" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb586-33"><a href="lmm.html#cb586-33" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb586-34"><a href="lmm.html#cb586-34" tabindex="-1"></a>  <span class="co"># geom_errorbar(data = exp1g_m1_emm_dt,</span></span>
<span id="cb586-35"><a href="lmm.html#cb586-35" tabindex="-1"></a>  <span class="co">#            aes(y = emmean,</span></span>
<span id="cb586-36"><a href="lmm.html#cb586-36" tabindex="-1"></a>  <span class="co">#                ymin = lower.CL,</span></span>
<span id="cb586-37"><a href="lmm.html#cb586-37" tabindex="-1"></a>  <span class="co">#                ymax = upper.CL),</span></span>
<span id="cb586-38"><a href="lmm.html#cb586-38" tabindex="-1"></a>  <span class="co">#             color = &quot;black&quot;,</span></span>
<span id="cb586-39"><a href="lmm.html#cb586-39" tabindex="-1"></a>  <span class="co">#           width = .05) +</span></span>
<span id="cb586-40"><a href="lmm.html#cb586-40" tabindex="-1"></a>  </span>
<span id="cb586-41"><a href="lmm.html#cb586-41" tabindex="-1"></a>  <span class="co"># pvalue brackets</span></span>
<span id="cb586-42"><a href="lmm.html#cb586-42" tabindex="-1"></a>  <span class="fu">stat_pvalue_manual</span>(exp1g_m1_planned_dt[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],</span>
<span id="cb586-43"><a href="lmm.html#cb586-43" tabindex="-1"></a>                     <span class="at">label =</span> <span class="st">&quot;pretty_p&quot;</span>,</span>
<span id="cb586-44"><a href="lmm.html#cb586-44" tabindex="-1"></a>                     <span class="at">y.position =</span> <span class="fu">c</span>(<span class="dv">20</span>,<span class="dv">20</span>),</span>
<span id="cb586-45"><a href="lmm.html#cb586-45" tabindex="-1"></a>                     <span class="at">size =</span> <span class="fl">2.5</span>,</span>
<span id="cb586-46"><a href="lmm.html#cb586-46" tabindex="-1"></a>                     <span class="at">tip.length =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb586-47"><a href="lmm.html#cb586-47" tabindex="-1"></a>  </span>
<span id="cb586-48"><a href="lmm.html#cb586-48" tabindex="-1"></a>  <span class="co"># additive line + interaction p bracket</span></span>
<span id="cb586-49"><a href="lmm.html#cb586-49" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">x =</span> <span class="fl">3.85</span>,</span>
<span id="cb586-50"><a href="lmm.html#cb586-50" tabindex="-1"></a>               <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb586-51"><a href="lmm.html#cb586-51" tabindex="-1"></a>               <span class="at">xend =</span> <span class="fl">4.15</span>,</span>
<span id="cb586-52"><a href="lmm.html#cb586-52" tabindex="-1"></a>               <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb586-53"><a href="lmm.html#cb586-53" tabindex="-1"></a>               <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb586-54"><a href="lmm.html#cb586-54" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;gray&quot;</span>) <span class="sc">+</span></span>
<span id="cb586-55"><a href="lmm.html#cb586-55" tabindex="-1"></a>  <span class="fu">geom_bracket</span>(</span>
<span id="cb586-56"><a href="lmm.html#cb586-56" tabindex="-1"></a>    <span class="at">x =</span> <span class="fl">4.2</span>,</span>
<span id="cb586-57"><a href="lmm.html#cb586-57" tabindex="-1"></a>    <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb586-58"><a href="lmm.html#cb586-58" tabindex="-1"></a>    <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>] <span class="sc">+</span> b[<span class="dv">4</span>],</span>
<span id="cb586-59"><a href="lmm.html#cb586-59" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;interaction</span><span class="sc">\n</span><span class="st">p = &quot;</span>, p_ixn),</span>
<span id="cb586-60"><a href="lmm.html#cb586-60" tabindex="-1"></a>    <span class="at">text.size =</span> <span class="dv">3</span>,</span>
<span id="cb586-61"><a href="lmm.html#cb586-61" tabindex="-1"></a>    <span class="at">text.hjust =</span> <span class="dv">0</span>,</span>
<span id="cb586-62"><a href="lmm.html#cb586-62" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb586-63"><a href="lmm.html#cb586-63" tabindex="-1"></a>  </span>
<span id="cb586-64"><a href="lmm.html#cb586-64" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;GC (%)&quot;</span>) <span class="sc">+</span></span>
<span id="cb586-65"><a href="lmm.html#cb586-65" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal_okabe_ito_blue) <span class="sc">+</span></span>
<span id="cb586-66"><a href="lmm.html#cb586-66" tabindex="-1"></a>  <span class="fu">theme_pubr</span>() <span class="sc">+</span></span>
<span id="cb586-67"><a href="lmm.html#cb586-67" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">4.1</span>)) <span class="sc">+</span></span>
<span id="cb586-68"><a href="lmm.html#cb586-68" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb586-69"><a href="lmm.html#cb586-69" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), <span class="co"># no x-axis title</span></span>
<span id="cb586-70"><a href="lmm.html#cb586-70" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb586-71"><a href="lmm.html#cb586-71" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb586-72"><a href="lmm.html#cb586-72" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb586-73"><a href="lmm.html#cb586-73" tabindex="-1"></a>  </span>
<span id="cb586-74"><a href="lmm.html#cb586-74" tabindex="-1"></a>exp1g_plot_a <span class="ot">&lt;-</span> <span class="fu">factor_wrap</span>(exp1g_plot_a)</span>
<span id="cb586-75"><a href="lmm.html#cb586-75" tabindex="-1"></a></span>
<span id="cb586-76"><a href="lmm.html#cb586-76" tabindex="-1"></a><span class="co">#exp1g_plot_a</span></span></code></pre></div>
<div class="sourceCode" id="cb587"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb587-1"><a href="lmm.html#cb587-1" tabindex="-1"></a><span class="co"># get coefficients of model</span></span>
<span id="cb587-2"><a href="lmm.html#cb587-2" tabindex="-1"></a>b <span class="ot">&lt;-</span> exp1g_m1_coef[, <span class="st">&quot;Estimate&quot;</span>]</span>
<span id="cb587-3"><a href="lmm.html#cb587-3" tabindex="-1"></a></span>
<span id="cb587-4"><a href="lmm.html#cb587-4" tabindex="-1"></a><span class="co"># get interaction p</span></span>
<span id="cb587-5"><a href="lmm.html#cb587-5" tabindex="-1"></a>p_ixn <span class="ot">&lt;-</span> exp1g_m1_planned_dt[contrast <span class="sc">==</span> <span class="st">&quot;Interaction&quot;</span>, pretty_p]</span>
<span id="cb587-6"><a href="lmm.html#cb587-6" tabindex="-1"></a></span>
<span id="cb587-7"><a href="lmm.html#cb587-7" tabindex="-1"></a>dodge_width <span class="ot">=</span> <span class="fl">0.6</span></span>
<span id="cb587-8"><a href="lmm.html#cb587-8" tabindex="-1"></a>exp1g_plot_b <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> exp1g,</span>
<span id="cb587-9"><a href="lmm.html#cb587-9" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> treatment,</span>
<span id="cb587-10"><a href="lmm.html#cb587-10" tabindex="-1"></a>                  <span class="at">y =</span> gc,</span>
<span id="cb587-11"><a href="lmm.html#cb587-11" tabindex="-1"></a>                  <span class="at">group =</span> experiment_id)) <span class="sc">+</span></span>
<span id="cb587-12"><a href="lmm.html#cb587-12" tabindex="-1"></a>  <span class="co"># modeled experiment by treatment means</span></span>
<span id="cb587-13"><a href="lmm.html#cb587-13" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> exp1g_m1_emm2,</span>
<span id="cb587-14"><a href="lmm.html#cb587-14" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">y =</span> group_mean,</span>
<span id="cb587-15"><a href="lmm.html#cb587-15" tabindex="-1"></a>                <span class="at">color =</span> experiment_id),</span>
<span id="cb587-16"><a href="lmm.html#cb587-16" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> dodge_width),</span>
<span id="cb587-17"><a href="lmm.html#cb587-17" tabindex="-1"></a>            <span class="at">alpha =</span> <span class="dv">1</span>,</span>
<span id="cb587-18"><a href="lmm.html#cb587-18" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb587-19"><a href="lmm.html#cb587-19" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> exp1g_m1_emm2,</span>
<span id="cb587-20"><a href="lmm.html#cb587-20" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">y =</span> group_mean,</span>
<span id="cb587-21"><a href="lmm.html#cb587-21" tabindex="-1"></a>                <span class="at">group =</span> experiment_id,</span>
<span id="cb587-22"><a href="lmm.html#cb587-22" tabindex="-1"></a>                <span class="at">color =</span> experiment_id),</span>
<span id="cb587-23"><a href="lmm.html#cb587-23" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> dodge_width)) <span class="sc">+</span></span>
<span id="cb587-24"><a href="lmm.html#cb587-24" tabindex="-1"></a>  </span>
<span id="cb587-25"><a href="lmm.html#cb587-25" tabindex="-1"></a>  <span class="co"># raw data</span></span>
<span id="cb587-26"><a href="lmm.html#cb587-26" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> dodge_width),</span>
<span id="cb587-27"><a href="lmm.html#cb587-27" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">&quot;gray80&quot;</span></span>
<span id="cb587-28"><a href="lmm.html#cb587-28" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb587-29"><a href="lmm.html#cb587-29" tabindex="-1"></a>  </span>
<span id="cb587-30"><a href="lmm.html#cb587-30" tabindex="-1"></a>  <span class="co"># modeled treatment means</span></span>
<span id="cb587-31"><a href="lmm.html#cb587-31" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> exp1g_m1_emm_dt,</span>
<span id="cb587-32"><a href="lmm.html#cb587-32" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">y =</span> emmean),</span>
<span id="cb587-33"><a href="lmm.html#cb587-33" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb587-34"><a href="lmm.html#cb587-34" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb587-35"><a href="lmm.html#cb587-35" tabindex="-1"></a>  <span class="co"># geom_errorbar(data = exp1g_m1_emm_dt,</span></span>
<span id="cb587-36"><a href="lmm.html#cb587-36" tabindex="-1"></a>  <span class="co">#            aes(y = emmean,</span></span>
<span id="cb587-37"><a href="lmm.html#cb587-37" tabindex="-1"></a>  <span class="co">#                ymin = lower.CL,</span></span>
<span id="cb587-38"><a href="lmm.html#cb587-38" tabindex="-1"></a>  <span class="co">#                ymax = upper.CL),</span></span>
<span id="cb587-39"><a href="lmm.html#cb587-39" tabindex="-1"></a>  <span class="co">#             color = &quot;black&quot;,</span></span>
<span id="cb587-40"><a href="lmm.html#cb587-40" tabindex="-1"></a>  <span class="co">#           width = .05) +</span></span>
<span id="cb587-41"><a href="lmm.html#cb587-41" tabindex="-1"></a>  </span>
<span id="cb587-42"><a href="lmm.html#cb587-42" tabindex="-1"></a>  <span class="co"># pvalue brackets</span></span>
<span id="cb587-43"><a href="lmm.html#cb587-43" tabindex="-1"></a>  <span class="fu">stat_pvalue_manual</span>(exp1g_m1_planned_dt[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],</span>
<span id="cb587-44"><a href="lmm.html#cb587-44" tabindex="-1"></a>                     <span class="at">label =</span> <span class="st">&quot;pretty_p&quot;</span>,</span>
<span id="cb587-45"><a href="lmm.html#cb587-45" tabindex="-1"></a>                     <span class="at">y.position =</span> <span class="fu">c</span>(<span class="dv">20</span>,<span class="dv">20</span>),</span>
<span id="cb587-46"><a href="lmm.html#cb587-46" tabindex="-1"></a>                     <span class="at">size =</span> <span class="fl">2.5</span>,</span>
<span id="cb587-47"><a href="lmm.html#cb587-47" tabindex="-1"></a>                     <span class="at">tip.length =</span> <span class="fl">0.01</span>) <span class="sc">+</span></span>
<span id="cb587-48"><a href="lmm.html#cb587-48" tabindex="-1"></a>  </span>
<span id="cb587-49"><a href="lmm.html#cb587-49" tabindex="-1"></a>  <span class="co"># additive line + interaction p bracket</span></span>
<span id="cb587-50"><a href="lmm.html#cb587-50" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">x =</span> <span class="dv">4</span> <span class="sc">-</span> dodge_width<span class="sc">/</span><span class="dv">2</span>,</span>
<span id="cb587-51"><a href="lmm.html#cb587-51" tabindex="-1"></a>               <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb587-52"><a href="lmm.html#cb587-52" tabindex="-1"></a>               <span class="at">xend =</span> <span class="dv">4</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="dv">2</span>,</span>
<span id="cb587-53"><a href="lmm.html#cb587-53" tabindex="-1"></a>               <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb587-54"><a href="lmm.html#cb587-54" tabindex="-1"></a>               <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>,</span>
<span id="cb587-55"><a href="lmm.html#cb587-55" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;gray&quot;</span>) <span class="sc">+</span></span>
<span id="cb587-56"><a href="lmm.html#cb587-56" tabindex="-1"></a>  <span class="fu">geom_bracket</span>(</span>
<span id="cb587-57"><a href="lmm.html#cb587-57" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">4</span> <span class="sc">+</span> dodge_width<span class="sc">/</span><span class="fl">1.9</span>,</span>
<span id="cb587-58"><a href="lmm.html#cb587-58" tabindex="-1"></a>    <span class="at">y =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>],</span>
<span id="cb587-59"><a href="lmm.html#cb587-59" tabindex="-1"></a>    <span class="at">yend =</span> b[<span class="dv">1</span>] <span class="sc">+</span> b[<span class="dv">2</span>] <span class="sc">+</span> b[<span class="dv">3</span>] <span class="sc">+</span> b[<span class="dv">4</span>],</span>
<span id="cb587-60"><a href="lmm.html#cb587-60" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;interaction</span><span class="sc">\n</span><span class="st">p = &quot;</span>, p_ixn),</span>
<span id="cb587-61"><a href="lmm.html#cb587-61" tabindex="-1"></a>    <span class="at">text.size =</span> <span class="dv">3</span>,</span>
<span id="cb587-62"><a href="lmm.html#cb587-62" tabindex="-1"></a>    <span class="at">text.hjust =</span> <span class="dv">0</span>,</span>
<span id="cb587-63"><a href="lmm.html#cb587-63" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb587-64"><a href="lmm.html#cb587-64" tabindex="-1"></a>  </span>
<span id="cb587-65"><a href="lmm.html#cb587-65" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;GC (%)&quot;</span>) <span class="sc">+</span></span>
<span id="cb587-66"><a href="lmm.html#cb587-66" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> pal_okabe_ito_blue) <span class="sc">+</span></span>
<span id="cb587-67"><a href="lmm.html#cb587-67" tabindex="-1"></a>  <span class="fu">theme_pubr</span>() <span class="sc">+</span></span>
<span id="cb587-68"><a href="lmm.html#cb587-68" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">4.2</span>)) <span class="sc">+</span></span>
<span id="cb587-69"><a href="lmm.html#cb587-69" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb587-70"><a href="lmm.html#cb587-70" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), <span class="co"># no x-axis title</span></span>
<span id="cb587-71"><a href="lmm.html#cb587-71" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb587-72"><a href="lmm.html#cb587-72" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb587-73"><a href="lmm.html#cb587-73" tabindex="-1"></a>  <span class="cn">NULL</span></span>
<span id="cb587-74"><a href="lmm.html#cb587-74" tabindex="-1"></a>  </span>
<span id="cb587-75"><a href="lmm.html#cb587-75" tabindex="-1"></a>exp1g_plot_b <span class="ot">&lt;-</span> <span class="fu">factor_wrap</span>(exp1g_plot_b)</span>
<span id="cb587-76"><a href="lmm.html#cb587-76" tabindex="-1"></a></span>
<span id="cb587-77"><a href="lmm.html#cb587-77" tabindex="-1"></a><span class="co"># exp1g_plot_b</span></span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/unnamed-chunk-259-1.png" width="1066.66666666667" /></p>
</div>
</div>
<div id="repeated-measures-anova-randomized-complete-block-with-no-subsampling" class="section level3 hasAnchor" number="16.11.2">
<h3><span class="header-section-number">16.11.2</span> Repeated measures ANOVA (randomized complete block with no subsampling)<a href="lmm.html#repeated-measures-anova-randomized-complete-block-with-no-subsampling" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb588"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb588-1"><a href="lmm.html#cb588-1" tabindex="-1"></a><span class="co"># this is the rm-ANOVA</span></span>
<span id="cb588-2"><a href="lmm.html#cb588-2" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb588-3"><a href="lmm.html#cb588-3" tabindex="-1"></a>              (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb588-4"><a href="lmm.html#cb588-4" tabindex="-1"></a>            <span class="at">data =</span> exp5c)</span>
<span id="cb588-5"><a href="lmm.html#cb588-5" tabindex="-1"></a></span>
<span id="cb588-6"><a href="lmm.html#cb588-6" tabindex="-1"></a><span class="co"># lmm equivalent</span></span>
<span id="cb588-7"><a href="lmm.html#cb588-7" tabindex="-1"></a></span>
<span id="cb588-8"><a href="lmm.html#cb588-8" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb588-9"><a href="lmm.html#cb588-9" tabindex="-1"></a>         (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb588-10"><a href="lmm.html#cb588-10" tabindex="-1"></a>         (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment) <span class="sc">+</span></span>
<span id="cb588-11"><a href="lmm.html#cb588-11" tabindex="-1"></a>         (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>activity),</span>
<span id="cb588-12"><a href="lmm.html#cb588-12" tabindex="-1"></a>       <span class="at">data =</span> exp5c)</span>
<span id="cb588-13"><a href="lmm.html#cb588-13" tabindex="-1"></a></span>
<span id="cb588-14"><a href="lmm.html#cb588-14" tabindex="-1"></a><span class="co"># random intercept and slope model that is *not* equivalent</span></span>
<span id="cb588-15"><a href="lmm.html#cb588-15" tabindex="-1"></a><span class="co"># this isn&#39;t solvable because there is no subsampling</span></span>
<span id="cb588-16"><a href="lmm.html#cb588-16" tabindex="-1"></a></span>
<span id="cb588-17"><a href="lmm.html#cb588-17" tabindex="-1"></a>m3 <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb588-18"><a href="lmm.html#cb588-18" tabindex="-1"></a>              (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb588-19"><a href="lmm.html#cb588-19" tabindex="-1"></a>            <span class="at">data =</span> exp5c)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li><code>afex</code> computes the repeated measures anova model using both <code>aov</code> (classical univariate repeated measures ANOVA) and using <code>lm</code> with multiple response variables (the multivariate repeated measures ANOVA). As of this writing, the output from <code>aov</code> is the default.</li>
<li>Given only one measure for each donor within a <span class="math inline">\(\texttt{treatment} \times \texttt{activity}\)</span> combination, the linear mixed model <code>m2</code> is equivalent to the univariate repeated measures model m1.</li>
<li>The model formula in <code>m1</code> looks like that in the linear mixed model <code>m3</code> but the two models are not equivalent.</li>
</ol>
<div id="univariate-vs.-multivariate-repeated-measures-anova" class="section level4 hasAnchor" number="16.11.2.1">
<h4><span class="header-section-number">16.11.2.1</span> univariate vs. multivariate repeated measures ANOVA<a href="lmm.html#univariate-vs.-multivariate-repeated-measures-anova" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Use the multivariate model unless you want to replicate the result of someone who used a univariate model. By default, <code>aov_4</code> computes only the multivariate model (prior to version xxx, the default was to compute both models).</p>
<div class="sourceCode" id="cb589"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb589-1"><a href="lmm.html#cb589-1" tabindex="-1"></a><span class="co"># default mode -- should be multivariate</span></span>
<span id="cb589-2"><a href="lmm.html#cb589-2" tabindex="-1"></a>exp5c_aov <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb589-3"><a href="lmm.html#cb589-3" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb589-4"><a href="lmm.html#cb589-4" tabindex="-1"></a>                  <span class="at">data =</span> exp5c)</span>
<span id="cb589-5"><a href="lmm.html#cb589-5" tabindex="-1"></a></span>
<span id="cb589-6"><a href="lmm.html#cb589-6" tabindex="-1"></a><span class="co"># force aov_4 to compute univariate model</span></span>
<span id="cb589-7"><a href="lmm.html#cb589-7" tabindex="-1"></a>exp5c_aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb589-8"><a href="lmm.html#cb589-8" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb589-9"><a href="lmm.html#cb589-9" tabindex="-1"></a>                  <span class="at">data =</span> exp5c,</span>
<span id="cb589-10"><a href="lmm.html#cb589-10" tabindex="-1"></a>                  <span class="at">include_aov =</span> <span class="cn">TRUE</span>)</span>
<span id="cb589-11"><a href="lmm.html#cb589-11" tabindex="-1"></a></span>
<span id="cb589-12"><a href="lmm.html#cb589-12" tabindex="-1"></a><span class="co"># explicitly exclude the univariate model</span></span>
<span id="cb589-13"><a href="lmm.html#cb589-13" tabindex="-1"></a>exp5c_aov2 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb589-14"><a href="lmm.html#cb589-14" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb589-15"><a href="lmm.html#cb589-15" tabindex="-1"></a>                  <span class="at">data =</span> exp5c,</span>
<span id="cb589-16"><a href="lmm.html#cb589-16" tabindex="-1"></a>                  <span class="at">include_aov =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The <code>include_aov = TRUE</code> argument forces output from <code>aov_4</code> to include the univariate model.</li>
</ol>
<p><strong>contrasts from multivariate model</strong></p>
<div class="sourceCode" id="cb590"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb590-1"><a href="lmm.html#cb590-1" tabindex="-1"></a><span class="co"># These three should give equivalent results</span></span>
<span id="cb590-2"><a href="lmm.html#cb590-2" tabindex="-1"></a></span>
<span id="cb590-3"><a href="lmm.html#cb590-3" tabindex="-1"></a><span class="co"># exp5c_aov was fit with the default -- multivariate model only</span></span>
<span id="cb590-4"><a href="lmm.html#cb590-4" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov,</span>
<span id="cb590-5"><a href="lmm.html#cb590-5" tabindex="-1"></a>        <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb590-6"><a href="lmm.html#cb590-6" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb590-7"><a href="lmm.html#cb590-7" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb590-8"><a href="lmm.html#cb590-8" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb590-9"><a href="lmm.html#cb590-9" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>##  activity treatment contrast      estimate     SE df t.ratio p.value
##  Basal    .         siNR4A3 - Scr   0.0813 0.1012  5   0.804  0.4581
##  EPS      .         siNR4A3 - Scr  -0.1858 0.0359  5  -5.180  0.0035
##  .        Scr       EPS - Basal     0.1510 0.0395  5   3.821  0.0124
##  .        siNR4A3   EPS - Basal    -0.1161 0.0626  5  -1.855  0.1228</code></pre>
<div class="sourceCode" id="cb592"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb592-1"><a href="lmm.html#cb592-1" tabindex="-1"></a><span class="co"># exp5c_aov included the univariate model but the default is still the multivariate model</span></span>
<span id="cb592-2"><a href="lmm.html#cb592-2" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov1,</span>
<span id="cb592-3"><a href="lmm.html#cb592-3" tabindex="-1"></a>        <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb592-4"><a href="lmm.html#cb592-4" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb592-5"><a href="lmm.html#cb592-5" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb592-6"><a href="lmm.html#cb592-6" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb592-7"><a href="lmm.html#cb592-7" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>##  activity treatment contrast      estimate     SE df t.ratio p.value
##  Basal    .         siNR4A3 - Scr   0.0813 0.1012  5   0.804  0.4581
##  EPS      .         siNR4A3 - Scr  -0.1858 0.0359  5  -5.180  0.0035
##  .        Scr       EPS - Basal     0.1510 0.0395  5   3.821  0.0124
##  .        siNR4A3   EPS - Basal    -0.1161 0.0626  5  -1.855  0.1228</code></pre>
<div class="sourceCode" id="cb594"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb594-1"><a href="lmm.html#cb594-1" tabindex="-1"></a><span class="co"># passing `model = &quot;multivariate&quot;` makes the model choice transparent</span></span>
<span id="cb594-2"><a href="lmm.html#cb594-2" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov1,</span>
<span id="cb594-3"><a href="lmm.html#cb594-3" tabindex="-1"></a>        <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>),</span>
<span id="cb594-4"><a href="lmm.html#cb594-4" tabindex="-1"></a>        <span class="at">model =</span> <span class="st">&quot;multivariate&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb594-5"><a href="lmm.html#cb594-5" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb594-6"><a href="lmm.html#cb594-6" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb594-7"><a href="lmm.html#cb594-7" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb594-8"><a href="lmm.html#cb594-8" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>##  activity treatment contrast      estimate     SE df t.ratio p.value
##  Basal    .         siNR4A3 - Scr   0.0813 0.1012  5   0.804  0.4581
##  EPS      .         siNR4A3 - Scr  -0.1858 0.0359  5  -5.180  0.0035
##  .        Scr       EPS - Basal     0.1510 0.0395  5   3.821  0.0124
##  .        siNR4A3   EPS - Basal    -0.1161 0.0626  5  -1.855  0.1228</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>If our rmANOVA model is fit with the default specification (<code>exp5c_aov1</code>), we can get the multivariate output using the <code>model = "multivariate"</code> argument within <code>emmeans</code> (not the <code>contrast</code> function!). Or, if our fit excluded the univariate model (<code>exp5c_aov2</code>), then we don’t need the <code>model</code> argument.</li>
</ol>
<p><strong>contrasts from univariate model</strong></p>
<div class="sourceCode" id="cb596"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb596-1"><a href="lmm.html#cb596-1" tabindex="-1"></a><span class="co"># get the univariate model results using $aov</span></span>
<span id="cb596-2"><a href="lmm.html#cb596-2" tabindex="-1"></a><span class="fu">emmeans</span>(exp5c_aov1<span class="sc">$</span>aov, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb596-3"><a href="lmm.html#cb596-3" tabindex="-1"></a>  <span class="fu">contrast</span>(<span class="at">method =</span> <span class="st">&quot;revpairwise&quot;</span>,</span>
<span id="cb596-4"><a href="lmm.html#cb596-4" tabindex="-1"></a>           <span class="at">simple =</span> <span class="st">&quot;each&quot;</span>,</span>
<span id="cb596-5"><a href="lmm.html#cb596-5" tabindex="-1"></a>           <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb596-6"><a href="lmm.html#cb596-6" tabindex="-1"></a>           <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>##  activity treatment contrast      estimate     SE   df t.ratio p.value
##  Basal    .         siNR4A3 - Scr   0.0813 0.0759 8.60   1.071  0.3132
##  EPS      .         siNR4A3 - Scr  -0.1858 0.0759 8.60  -2.448  0.0380
##  .        Scr       EPS - Basal     0.1510 0.0524 9.39   2.884  0.0173
##  .        siNR4A3   EPS - Basal    -0.1161 0.0524 9.39  -2.218  0.0525</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Passing <code>exp5c_aov1</code> to <code>emmeans</code> will return the contrasts from the multivariate. change this to the univariate model by passing <code>exp5c_aov1$aov</code>.</li>
</ol>
</div>
<div id="the-anova-table" class="section level4 hasAnchor" number="16.11.2.2">
<h4><span class="header-section-number">16.11.2.2</span> The ANOVA table<a href="lmm.html#the-anova-table" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb598"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb598-1"><a href="lmm.html#cb598-1" tabindex="-1"></a><span class="fu">nice</span>(exp5c_aov1, <span class="at">correction =</span> <span class="st">&quot;GG&quot;</span>)</span></code></pre></div>
<pre><code>## Anova Table (Type 3 tests)
## 
## Response: glucose_uptake
##               Effect   df  MSE       F   ges p.value
## 1          treatment 1, 5 0.02    0.68  .006     .45
## 2           activity 1, 5 0.01    0.30 .0006     .61
## 3 treatment:activity 1, 5 0.01 10.37 *   .04     .02
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;+&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb600"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb600-1"><a href="lmm.html#cb600-1" tabindex="-1"></a><span class="fu">nice</span>(exp5c_aov1, <span class="at">correction =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>## Anova Table (Type 3 tests)
## 
## Response: glucose_uptake
##               Effect   df  MSE       F   ges p.value
## 1          treatment 1, 5 0.02    0.68  .006     .45
## 2           activity 1, 5 0.01    0.30 .0006     .61
## 3 treatment:activity 1, 5 0.01 10.37 *   .04     .02
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;+&#39; 0.1 &#39; &#39; 1</code></pre>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The Greenhouse-Geiger (“GG”) correction is the default. While the <code>correction = "GG"</code> argument is not needed, it makes the script more transparent.</li>
<li>For <em>these data</em> the Greenhouse-Geiger correction doesn’t make a difference in the table.</li>
</ol>
</div>
</div>
</div>
<div id="hidden-code-7" class="section level2 hasAnchor" number="16.12">
<h2><span class="header-section-number">16.12</span> Hidden code<a href="lmm.html#hidden-code-7" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="import-exp5c" class="section level3 hasAnchor" number="16.12.1">
<h3><span class="header-section-number">16.12.1</span> Import exp5c<a href="lmm.html#import-exp5c" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb602"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb602-1"><a href="lmm.html#cb602-1" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">&quot;Transcriptomic profiling of skeletal muscle adaptations to exercise and inactivity&quot;</span></span>
<span id="cb602-2"><a href="lmm.html#cb602-2" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">&quot;41467_2019_13869_MOESM6_ESM.xlsx&quot;</span></span>
<span id="cb602-3"><a href="lmm.html#cb602-3" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb602-4"><a href="lmm.html#cb602-4" tabindex="-1"></a></span>
<span id="cb602-5"><a href="lmm.html#cb602-5" tabindex="-1"></a>exp5c_wide <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb602-6"><a href="lmm.html#cb602-6" tabindex="-1"></a>                         <span class="at">sheet =</span> <span class="st">&quot;Fig5c&quot;</span>,</span>
<span id="cb602-7"><a href="lmm.html#cb602-7" tabindex="-1"></a>                         <span class="at">range =</span> <span class="st">&quot;A2:M3&quot;</span>,</span>
<span id="cb602-8"><a href="lmm.html#cb602-8" tabindex="-1"></a>                         <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb602-9"><a href="lmm.html#cb602-9" tabindex="-1"></a>  <span class="fu">data.table</span>() <span class="sc">%&gt;%</span></span>
<span id="cb602-10"><a href="lmm.html#cb602-10" tabindex="-1"></a>  <span class="fu">transpose</span>(<span class="at">make.names =</span> <span class="dv">1</span>)</span>
<span id="cb602-11"><a href="lmm.html#cb602-11" tabindex="-1"></a></span>
<span id="cb602-12"><a href="lmm.html#cb602-12" tabindex="-1"></a>activity_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;EPS&quot;</span>)</span>
<span id="cb602-13"><a href="lmm.html#cb602-13" tabindex="-1"></a>treatment_levels <span class="ot">&lt;-</span> <span class="fu">names</span>(exp5c_wide)</span>
<span id="cb602-14"><a href="lmm.html#cb602-14" tabindex="-1"></a>exp5c_wide[, activity <span class="sc">:=</span> <span class="fu">rep</span>(activity_levels, <span class="at">each =</span> <span class="dv">6</span>)]</span>
<span id="cb602-15"><a href="lmm.html#cb602-15" tabindex="-1"></a>exp5c_wide[, activity <span class="sc">:=</span> <span class="fu">factor</span>(activity, <span class="at">levels =</span> activity_levels)]</span>
<span id="cb602-16"><a href="lmm.html#cb602-16" tabindex="-1"></a></span>
<span id="cb602-17"><a href="lmm.html#cb602-17" tabindex="-1"></a>exp5c <span class="ot">&lt;-</span> <span class="fu">melt</span>(exp5c_wide,</span>
<span id="cb602-18"><a href="lmm.html#cb602-18" tabindex="-1"></a>              <span class="at">id.vars =</span> <span class="st">&quot;activity&quot;</span>,</span>
<span id="cb602-19"><a href="lmm.html#cb602-19" tabindex="-1"></a>              <span class="at">variable.name =</span> <span class="st">&quot;treatment&quot;</span>,</span>
<span id="cb602-20"><a href="lmm.html#cb602-20" tabindex="-1"></a>              <span class="at">value.name =</span> <span class="st">&quot;glucose_uptake&quot;</span>)</span>
<span id="cb602-21"><a href="lmm.html#cb602-21" tabindex="-1"></a>exp5c[, treatment <span class="sc">:=</span> <span class="fu">factor</span>(treatment, <span class="at">levels =</span> treatment_levels)]</span>
<span id="cb602-22"><a href="lmm.html#cb602-22" tabindex="-1"></a></span>
<span id="cb602-23"><a href="lmm.html#cb602-23" tabindex="-1"></a>exp5c[, donor <span class="sc">:=</span> <span class="fu">rep</span>(<span class="fu">paste0</span>(<span class="st">&quot;donor_&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>), <span class="dv">4</span>)]</span></code></pre></div>
</div>
<div id="import-exp1g" class="section level3 hasAnchor" number="16.12.2">
<h3><span class="header-section-number">16.12.2</span> Import exp1g<a href="lmm.html#import-exp1g" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb603"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb603-1"><a href="lmm.html#cb603-1" tabindex="-1"></a>data_from <span class="ot">&lt;-</span> <span class="st">&quot;A GPR174–CCL21 module imparts sexual dimorphism to humoral immunity&quot;</span></span>
<span id="cb603-2"><a href="lmm.html#cb603-2" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">&quot;41586_2019_1873_MOESM3_ESM.xlsx&quot;</span></span>
<span id="cb603-3"><a href="lmm.html#cb603-3" tabindex="-1"></a>file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(data_folder, data_from, file_name)</span>
<span id="cb603-4"><a href="lmm.html#cb603-4" tabindex="-1"></a></span>
<span id="cb603-5"><a href="lmm.html#cb603-5" tabindex="-1"></a>exp1g_wide <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path,</span>
<span id="cb603-6"><a href="lmm.html#cb603-6" tabindex="-1"></a>                         <span class="at">sheet =</span> <span class="st">&quot;Fig 1g&quot;</span>,</span>
<span id="cb603-7"><a href="lmm.html#cb603-7" tabindex="-1"></a>                         <span class="at">range =</span> <span class="st">&quot;B4:E25&quot;</span>,</span>
<span id="cb603-8"><a href="lmm.html#cb603-8" tabindex="-1"></a>                         <span class="at">col_types =</span> <span class="fu">c</span>(<span class="st">&quot;numeric&quot;</span>),</span>
<span id="cb603-9"><a href="lmm.html#cb603-9" tabindex="-1"></a>                         <span class="at">col_names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb603-10"><a href="lmm.html#cb603-10" tabindex="-1"></a>  <span class="fu">data.table</span>()</span>
<span id="cb603-11"><a href="lmm.html#cb603-11" tabindex="-1"></a></span>
<span id="cb603-12"><a href="lmm.html#cb603-12" tabindex="-1"></a>genotype_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Gpr174+&quot;</span>, <span class="st">&quot;Gpr174-&quot;</span>)</span>
<span id="cb603-13"><a href="lmm.html#cb603-13" tabindex="-1"></a>sex_levels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;M&quot;</span>, <span class="st">&quot;F&quot;</span>)</span>
<span id="cb603-14"><a href="lmm.html#cb603-14" tabindex="-1"></a>g.by.s_levels <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">expand.grid</span>(genotype_levels, sex_levels))</span>
<span id="cb603-15"><a href="lmm.html#cb603-15" tabindex="-1"></a><span class="fu">colnames</span>(exp1g_wide) <span class="ot">&lt;-</span> g.by.s_levels</span>
<span id="cb603-16"><a href="lmm.html#cb603-16" tabindex="-1"></a></span>
<span id="cb603-17"><a href="lmm.html#cb603-17" tabindex="-1"></a>exp_levels <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;exp_&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)</span>
<span id="cb603-18"><a href="lmm.html#cb603-18" tabindex="-1"></a>exp1g_wide[, experiment_id <span class="sc">:=</span> <span class="fu">rep</span>(exp_levels, <span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">6</span>,<span class="dv">5</span>))] <span class="co">#check!</span></span>
<span id="cb603-19"><a href="lmm.html#cb603-19" tabindex="-1"></a>exp1g_wide[, experiment_id <span class="sc">:=</span> <span class="fu">factor</span>(experiment_id)] <span class="co">#check!</span></span>
<span id="cb603-20"><a href="lmm.html#cb603-20" tabindex="-1"></a></span>
<span id="cb603-21"><a href="lmm.html#cb603-21" tabindex="-1"></a>exp1g <span class="ot">&lt;-</span> <span class="fu">melt</span>(exp1g_wide,</span>
<span id="cb603-22"><a href="lmm.html#cb603-22" tabindex="-1"></a>              <span class="at">id.vars =</span> <span class="st">&quot;experiment_id&quot;</span>,</span>
<span id="cb603-23"><a href="lmm.html#cb603-23" tabindex="-1"></a>              <span class="at">measure.vars =</span> g.by.s_levels,</span>
<span id="cb603-24"><a href="lmm.html#cb603-24" tabindex="-1"></a>              <span class="at">variable.name =</span> <span class="st">&quot;treatment&quot;</span>,</span>
<span id="cb603-25"><a href="lmm.html#cb603-25" tabindex="-1"></a>              <span class="at">value.name =</span> <span class="st">&quot;gc&quot;</span>) <span class="sc">%&gt;%</span> <span class="co"># cell count</span></span>
<span id="cb603-26"><a href="lmm.html#cb603-26" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb603-27"><a href="lmm.html#cb603-27" tabindex="-1"></a></span>
<span id="cb603-28"><a href="lmm.html#cb603-28" tabindex="-1"></a>exp1g[, <span class="fu">c</span>(<span class="st">&quot;genotype&quot;</span>, <span class="st">&quot;sex&quot;</span>)<span class="sc">:=</span> <span class="fu">tstrsplit</span>(treatment,</span>
<span id="cb603-29"><a href="lmm.html#cb603-29" tabindex="-1"></a>                                             <span class="st">&quot; &quot;</span>,</span>
<span id="cb603-30"><a href="lmm.html#cb603-30" tabindex="-1"></a>                                             <span class="at">fixed =</span> <span class="cn">TRUE</span>)]</span>
<span id="cb603-31"><a href="lmm.html#cb603-31" tabindex="-1"></a>exp1g[, genotype <span class="sc">:=</span> <span class="fu">factor</span>(genotype,</span>
<span id="cb603-32"><a href="lmm.html#cb603-32" tabindex="-1"></a>                           <span class="at">levels =</span> genotype_levels)]</span>
<span id="cb603-33"><a href="lmm.html#cb603-33" tabindex="-1"></a>exp1g[, sex <span class="sc">:=</span> <span class="fu">factor</span>(sex,</span>
<span id="cb603-34"><a href="lmm.html#cb603-34" tabindex="-1"></a>                           <span class="at">levels =</span> sex_levels)]</span>
<span id="cb603-35"><a href="lmm.html#cb603-35" tabindex="-1"></a></span>
<span id="cb603-36"><a href="lmm.html#cb603-36" tabindex="-1"></a>exp1g_means <span class="ot">&lt;-</span> exp1g[, .(<span class="at">gc =</span> <span class="fu">mean</span>(gc)),</span>
<span id="cb603-37"><a href="lmm.html#cb603-37" tabindex="-1"></a>                     by <span class="ot">=</span> .(treatment, genotype, sex, experiment_id)]</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="part-vi-expanding-the-linear-model.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="pre-post.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/chapters/55-lmm01-blocking.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Walker-elementary-statistical-modeling-draft.pdf", "Walker-elementary-statistical-modeling-draft.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
