<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 P-values | Elements of Statistical Modeling for Experimental Biology</title>
  <meta name="description" content="A first course in statistical modeling for experimental biology researchers" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 P-values | Elements of Statistical Modeling for Experimental Biology" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A first course in statistical modeling for experimental biology researchers" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 P-values | Elements of Statistical Modeling for Experimental Biology" />
  
  <meta name="twitter:description" content="A first course in statistical modeling for experimental biology researchers" />
  

<meta name="author" content="Copyright 2018 Jeffrey A. Walker" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="uncertainty.html"/>
<link rel="next" href="errors-in-inference.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="libs/d3-3.3.8/d3.min.js"></script>
<script src="libs/dagre-0.4.0/dagre-d3.min.js"></script>
<link href="libs/mermaid-0.3.0/dist/mermaid.css" rel="stylesheet" />
<script src="libs/mermaid-0.3.0/dist/mermaid.slim.min.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/chromatography-0.1/chromatography.js"></script>
<script src="libs/DiagrammeR-binding-1.0.6.1/DiagrammeR.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Elements of Applied Biostatistics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#math"><i class="fa fa-check"></i><b>0.1</b> Math</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#r-and-programming"><i class="fa fa-check"></i><b>0.2</b> R and programming</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="a-table-of-models.html"><a href="a-table-of-models.html"><i class="fa fa-check"></i>A Table of Models</a>
<ul>
<li class="chapter" data-level="" data-path="a-table-of-models.html"><a href="a-table-of-models.html#including-mapping-between-linear-models-and-classical-tests"><i class="fa fa-check"></i>including mapping between linear models and classical tests</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-i-getting-started.html"><a href="part-i-getting-started.html"><i class="fa fa-check"></i>Part I: Getting Started</a></li>
<li class="chapter" data-level="1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html"><i class="fa fa-check"></i><b>1</b> Getting Started – R Projects and R Markdown</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#r-vs-r-studio"><i class="fa fa-check"></i><b>1.1</b> R vs R Studio</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#download-and-install-r-and-r-studio"><i class="fa fa-check"></i><b>1.2</b> Download and install R and R studio</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#open-r-studio-and-modify-the-workspace-preference"><i class="fa fa-check"></i><b>1.3</b> Open R Studio and modify the workspace preference</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#if-you-didnt-modify-the-workspace-preferences-from-the-previous-section-go-back-and-do-it"><i class="fa fa-check"></i><b>1.4</b> If you didn’t modify the workspace preferences from the previous section, go back and do it</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#r-markdown-in-a-nutshell"><i class="fa fa-check"></i><b>1.5</b> R Markdown in a nutshell</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#install-r-markdown"><i class="fa fa-check"></i><b>1.6</b> Install R Markdown</a></li>
<li class="chapter" data-level="1.7" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#importing-packages"><i class="fa fa-check"></i><b>1.7</b> Importing Packages</a></li>
<li class="chapter" data-level="1.8" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#setup-create-project"><i class="fa fa-check"></i><b>1.8</b> Create an R Studio Project for this textbook</a></li>
<li class="chapter" data-level="1.9" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#working-on-a-project-in-a-nutshell"><i class="fa fa-check"></i><b>1.9</b> Working on a project, in a nutshell</a></li>
<li class="chapter" data-level="1.10" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-an-r-markdown-file-for-this-chapter"><i class="fa fa-check"></i><b>1.10</b> Create an R Markdown file for this Chapter</a>
<ul>
<li class="chapter" data-level="1.10.1" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#modify-the-yaml-header"><i class="fa fa-check"></i><b>1.10.1</b> Modify the yaml header</a></li>
<li class="chapter" data-level="1.10.2" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#modify-the-setup-chunk"><i class="fa fa-check"></i><b>1.10.2</b> Modify the “setup” chunk</a></li>
<li class="chapter" data-level="1.10.3" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-a-fake-data-chunk"><i class="fa fa-check"></i><b>1.10.3</b> Create a “fake-data” chunk</a></li>
<li class="chapter" data-level="1.10.4" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#create-a-plot-chunk"><i class="fa fa-check"></i><b>1.10.4</b> Create a “plot” chunk</a></li>
<li class="chapter" data-level="1.10.5" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html#knit"><i class="fa fa-check"></i><b>1.10.5</b> Knit</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-ii-an-introduction-to-the-analysis-of-experimental-data-with-a-linear-model.html"><a href="part-ii-an-introduction-to-the-analysis-of-experimental-data-with-a-linear-model.html"><i class="fa fa-check"></i>Part II: An introduction to the analysis of experimental data with a linear model</a></li>
<li class="chapter" data-level="2" data-path="analyzing-experimental-data-with-a-linear-model.html"><a href="analyzing-experimental-data-with-a-linear-model.html"><i class="fa fa-check"></i><b>2</b> Analyzing experimental data with a linear model</a>
<ul>
<li class="chapter" data-level="2.1" data-path="analyzing-experimental-data-with-a-linear-model.html"><a href="analyzing-experimental-data-with-a-linear-model.html#this-text-is-about-using-linear-models-to-estimate-treatment-effects-and-the-uncertainty-in-our-estimates.-this-raises-the-question-what-is-an-effect"><i class="fa fa-check"></i><b>2.1</b> This text is about using linear models to estimate treatment effects and the uncertainty in our estimates. This, raises the question, what is “an effect?”</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="ask1-bio.html"><a href="ask1-bio.html"><i class="fa fa-check"></i><b>3</b> Background physiology to the experiments in Figure 2 of “ASK1 inhibits browning of white adipose tissue in obesity”</a></li>
<li class="chapter" data-level="4" data-path="ask1-report.html"><a href="ask1-report.html"><i class="fa fa-check"></i><b>4</b> Analyses for Figure 2 of “ASK1 inhibits browning of white adipose tissue in obesity”</a>
<ul>
<li class="chapter" data-level="4.1" data-path="ask1-report.html"><a href="ask1-report.html#setup"><i class="fa fa-check"></i><b>4.1</b> Setup</a></li>
<li class="chapter" data-level="4.2" data-path="ask1-report.html"><a href="ask1-report.html#data-source"><i class="fa fa-check"></i><b>4.2</b> Data source</a></li>
<li class="chapter" data-level="4.3" data-path="ask1-report.html"><a href="ask1-report.html#control-the-color-palette"><i class="fa fa-check"></i><b>4.3</b> control the color palette</a></li>
<li class="chapter" data-level="4.4" data-path="ask1-report.html"><a href="ask1-report.html#useful-functions"><i class="fa fa-check"></i><b>4.4</b> useful functions</a></li>
<li class="chapter" data-level="4.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2b-effect-of-ask1-deletion-on-growth-body-weight"><i class="fa fa-check"></i><b>4.5</b> figure 2b – effect of ASK1 deletion on growth (body weight)</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2b-import"><i class="fa fa-check"></i><b>4.5.1</b> figure 2b – import</a></li>
<li class="chapter" data-level="4.5.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2b-exploratory-plots"><i class="fa fa-check"></i><b>4.5.2</b> figure 2b – exploratory plots</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-effect-of-ask1-deletion-on-final-body-weight"><i class="fa fa-check"></i><b>4.6</b> Figure 2c – Effect of ASK1 deletion on final body weight</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-import"><i class="fa fa-check"></i><b>4.6.1</b> Figure 2c – import</a></li>
<li class="chapter" data-level="4.6.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-check-own-computation-of-weight-change-v-imported-value"><i class="fa fa-check"></i><b>4.6.2</b> Figure 2c – check own computation of weight change v imported value</a></li>
<li class="chapter" data-level="4.6.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-exploratory-plots"><i class="fa fa-check"></i><b>4.6.3</b> Figure 2c – exploratory plots</a></li>
<li class="chapter" data-level="4.6.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-fit-the-model-m1-lm"><i class="fa fa-check"></i><b>4.6.4</b> Figure 2c – fit the model: m1 (lm)</a></li>
<li class="chapter" data-level="4.6.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-check-the-model-m1"><i class="fa fa-check"></i><b>4.6.5</b> Figure 2c – check the model: m1</a></li>
<li class="chapter" data-level="4.6.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-fit-the-model-m2-gamma-glm"><i class="fa fa-check"></i><b>4.6.6</b> Figure 2c – fit the model: m2 (gamma glm)</a></li>
<li class="chapter" data-level="4.6.7" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-check-the-model-m2"><i class="fa fa-check"></i><b>4.6.7</b> Figure 2c – check the model, m2</a></li>
<li class="chapter" data-level="4.6.8" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-inference-from-the-model"><i class="fa fa-check"></i><b>4.6.8</b> Figure 2c – inference from the model</a></li>
<li class="chapter" data-level="4.6.9" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-plot-the-model"><i class="fa fa-check"></i><b>4.6.9</b> Figure 2c – plot the model</a></li>
<li class="chapter" data-level="4.6.10" data-path="ask1-report.html"><a href="ask1-report.html#figure-2c-report"><i class="fa fa-check"></i><b>4.6.10</b> Figure 2c – report</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve"><i class="fa fa-check"></i><b>4.7</b> Figure 2d – Effect of ASK1 KO on glucose tolerance (whole curve)</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-import"><i class="fa fa-check"></i><b>4.7.1</b> Figure 2d – Import</a></li>
<li class="chapter" data-level="4.7.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-exploratory-plots"><i class="fa fa-check"></i><b>4.7.2</b> Figure 2d – exploratory plots</a></li>
<li class="chapter" data-level="4.7.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-fit-the-model"><i class="fa fa-check"></i><b>4.7.3</b> Figure 2d – fit the model</a></li>
<li class="chapter" data-level="4.7.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-check-the-model"><i class="fa fa-check"></i><b>4.7.4</b> Figure 2d – check the model</a></li>
<li class="chapter" data-level="4.7.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-inference"><i class="fa fa-check"></i><b>4.7.5</b> Figure 2d – inference</a></li>
<li class="chapter" data-level="4.7.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2d-plot-the-model"><i class="fa fa-check"></i><b>4.7.6</b> Figure 2d – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure"><i class="fa fa-check"></i><b>4.8</b> Figure 2e – Effect of ASK1 deletion on glucose tolerance (summary measure)</a>
<ul>
<li class="chapter" data-level="4.8.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-message-the-data"><i class="fa fa-check"></i><b>4.8.1</b> Figure 2e – message the data</a></li>
<li class="chapter" data-level="4.8.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-exploratory-plots"><i class="fa fa-check"></i><b>4.8.2</b> Figure 2e – exploratory plots</a></li>
<li class="chapter" data-level="4.8.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-fit-the-model"><i class="fa fa-check"></i><b>4.8.3</b> Figure 2e – fit the model</a></li>
<li class="chapter" data-level="4.8.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-check-the-model"><i class="fa fa-check"></i><b>4.8.4</b> Figure 2e – check the model</a></li>
<li class="chapter" data-level="4.8.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-inference-from-the-model"><i class="fa fa-check"></i><b>4.8.5</b> Figure 2e – inference from the model</a></li>
<li class="chapter" data-level="4.8.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2e-plot-the-model"><i class="fa fa-check"></i><b>4.8.6</b> Figure 2e – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate"><i class="fa fa-check"></i><b>4.9</b> Figure 2f – Effect of ASK1 deletion on glucose infusion rate</a>
<ul>
<li class="chapter" data-level="4.9.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-import"><i class="fa fa-check"></i><b>4.9.1</b> Figure 2f – import</a></li>
<li class="chapter" data-level="4.9.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-exploratory-plots"><i class="fa fa-check"></i><b>4.9.2</b> Figure 2f – exploratory plots</a></li>
<li class="chapter" data-level="4.9.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-fit-the-model"><i class="fa fa-check"></i><b>4.9.3</b> Figure 2f – fit the model</a></li>
<li class="chapter" data-level="4.9.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-check-the-model"><i class="fa fa-check"></i><b>4.9.4</b> Figure 2f – check the model</a></li>
<li class="chapter" data-level="4.9.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-inference"><i class="fa fa-check"></i><b>4.9.5</b> Figure 2f – inference</a></li>
<li class="chapter" data-level="4.9.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2f-plot-the-model"><i class="fa fa-check"></i><b>4.9.6</b> Figure 2f – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.10" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake"><i class="fa fa-check"></i><b>4.10</b> Figure 2g – Effect of ASK1 deletion on tissue-specific glucose uptake</a>
<ul>
<li class="chapter" data-level="4.10.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-import"><i class="fa fa-check"></i><b>4.10.1</b> Figure 2g – import</a></li>
<li class="chapter" data-level="4.10.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-exploratory-plots"><i class="fa fa-check"></i><b>4.10.2</b> Figure 2g – exploratory plots</a></li>
<li class="chapter" data-level="4.10.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-fit-the-model"><i class="fa fa-check"></i><b>4.10.3</b> Figure 2g – fit the model</a></li>
<li class="chapter" data-level="4.10.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-check-the-model"><i class="fa fa-check"></i><b>4.10.4</b> Figure 2g – check the model</a></li>
<li class="chapter" data-level="4.10.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-inference"><i class="fa fa-check"></i><b>4.10.5</b> Figure 2g – inference</a></li>
<li class="chapter" data-level="4.10.6" data-path="ask1-report.html"><a href="ask1-report.html#figure-2g-plot-the-model"><i class="fa fa-check"></i><b>4.10.6</b> Figure 2g – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="4.11" data-path="ask1-report.html"><a href="ask1-report.html#figure-2h"><i class="fa fa-check"></i><b>4.11</b> Figure 2h</a></li>
<li class="chapter" data-level="4.12" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-effect-of-ask1-deletion-on-liver-tg"><i class="fa fa-check"></i><b>4.12</b> Figure 2i – Effect of ASK1 deletion on liver TG</a>
<ul>
<li class="chapter" data-level="4.12.1" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-fit-the-model"><i class="fa fa-check"></i><b>4.12.1</b> Figure 2i – fit the model</a></li>
<li class="chapter" data-level="4.12.2" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-check-the-model"><i class="fa fa-check"></i><b>4.12.2</b> Figure 2i – check the model</a></li>
<li class="chapter" data-level="4.12.3" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-inference"><i class="fa fa-check"></i><b>4.12.3</b> Figure 2i – inference</a></li>
<li class="chapter" data-level="4.12.4" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-plot-the-model"><i class="fa fa-check"></i><b>4.12.4</b> Figure 2i – plot the model</a></li>
<li class="chapter" data-level="4.12.5" data-path="ask1-report.html"><a href="ask1-report.html#figure-2i-report-the-model"><i class="fa fa-check"></i><b>4.12.5</b> Figure 2i – report the model</a></li>
</ul></li>
<li class="chapter" data-level="4.13" data-path="ask1-report.html"><a href="ask1-report.html#figure-2j"><i class="fa fa-check"></i><b>4.13</b> Figure 2j</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-iii-r-fundamentals.html"><a href="part-iii-r-fundamentals.html"><i class="fa fa-check"></i>Part III: R fundamentals</a></li>
<li class="chapter" data-level="5" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html"><i class="fa fa-check"></i><b>5</b> Data – Reading, Wrangling, and Writing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#learning-from-this-chapter"><i class="fa fa-check"></i><b>5.1</b> Learning from this chapter</a></li>
<li class="chapter" data-level="5.2" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#data-working-in-r"><i class="fa fa-check"></i><b>5.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#importing-data"><i class="fa fa-check"></i><b>5.2.1</b> Importing data</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#data-wrangling"><i class="fa fa-check"></i><b>5.3</b> Data wrangling</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#reshaping-data-wide-to-long"><i class="fa fa-check"></i><b>5.3.1</b> Reshaping data – Wide to long</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#reshaping-data-transpose-turning-the-columns-into-rows"><i class="fa fa-check"></i><b>5.3.2</b> Reshaping data – Transpose (turning the columns into rows)</a></li>
<li class="chapter" data-level="5.3.3" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#combining-data"><i class="fa fa-check"></i><b>5.3.3</b> Combining data</a></li>
<li class="chapter" data-level="5.3.4" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#subsetting-data"><i class="fa fa-check"></i><b>5.3.4</b> Subsetting data</a></li>
<li class="chapter" data-level="5.3.5" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#wrangling-columns"><i class="fa fa-check"></i><b>5.3.5</b> Wrangling columns</a></li>
<li class="chapter" data-level="5.3.6" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#missing-data"><i class="fa fa-check"></i><b>5.3.6</b> Missing data</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#saving-data"><i class="fa fa-check"></i><b>5.4</b> Saving data</a></li>
<li class="chapter" data-level="5.5" data-path="data-reading-wrangling-and-writing.html"><a href="data-reading-wrangling-and-writing.html#exercises"><i class="fa fa-check"></i><b>5.5</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="plotting-models.html"><a href="plotting-models.html"><i class="fa fa-check"></i><b>6</b> Plotting Models</a>
<ul>
<li class="chapter" data-level="6.1" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plots-show-the-model-and-the-data"><i class="fa fa-check"></i><b>6.1</b> Pretty good plots show the model and the data</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plot-component-1-modeled-effects-plot"><i class="fa fa-check"></i><b>6.1.1</b> Pretty good plot component 1: Modeled effects plot</a></li>
<li class="chapter" data-level="6.1.2" data-path="plotting-models.html"><a href="plotting-models.html#pretty-good-plot-component-2-modeled-mean-and-ci-plot"><i class="fa fa-check"></i><b>6.1.2</b> Pretty good plot component 2: Modeled mean and CI plot</a></li>
<li class="chapter" data-level="6.1.3" data-path="plotting-models.html"><a href="plotting-models.html#combining-effects-and-modeled-mean-and-ci-plots-an-effects-and-response-plot."><i class="fa fa-check"></i><b>6.1.3</b> Combining Effects and Modeled mean and CI plots – an Effects and response plot.</a></li>
<li class="chapter" data-level="6.1.4" data-path="plotting-models.html"><a href="plotting-models.html#some-comments-on-plot-components"><i class="fa fa-check"></i><b>6.1.4</b> Some comments on plot components</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="plotting-models.html"><a href="plotting-models.html#working-in-r"><i class="fa fa-check"></i><b>6.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="plotting-models.html"><a href="plotting-models.html#source-data"><i class="fa fa-check"></i><b>6.2.1</b> Source data</a></li>
<li class="chapter" data-level="6.2.2" data-path="plotting-models.html"><a href="plotting-models.html#how-to-plot-the-model"><i class="fa fa-check"></i><b>6.2.2</b> How to plot the model</a></li>
<li class="chapter" data-level="6.2.3" data-path="plotting-models.html"><a href="plotting-models.html#be-sure-ggplot_the_model-is-in-your-r-folder"><i class="fa fa-check"></i><b>6.2.3</b> Be sure ggplot_the_model is in your R folder</a></li>
<li class="chapter" data-level="6.2.4" data-path="plotting-models.html"><a href="plotting-models.html#how-to-use-the-plot-the-model-functions"><i class="fa fa-check"></i><b>6.2.4</b> How to use the Plot the Model functions</a></li>
<li class="chapter" data-level="6.2.5" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-a-response-plot-using-ggpubr"><i class="fa fa-check"></i><b>6.2.5</b> How to generate a Response Plot using ggpubr</a></li>
<li class="chapter" data-level="6.2.6" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-a-response-plot-with-a-grid-of-treatments-using-ggplot2"><i class="fa fa-check"></i><b>6.2.6</b> How to generate a Response Plot with a grid of treatments using ggplot2</a></li>
<li class="chapter" data-level="6.2.7" data-path="plotting-models.html"><a href="plotting-models.html#how-to-generate-an-effects-plot"><i class="fa fa-check"></i><b>6.2.7</b> How to generate an Effects Plot</a></li>
<li class="chapter" data-level="6.2.8" data-path="plotting-models.html"><a href="plotting-models.html#how-to-combine-the-response-and-effects-plots"><i class="fa fa-check"></i><b>6.2.8</b> How to combine the response and effects plots</a></li>
<li class="chapter" data-level="6.2.9" data-path="plotting-models.html"><a href="plotting-models.html#how-to-add-the-interaction-effect-to-response-and-effects-plots"><i class="fa fa-check"></i><b>6.2.9</b> How to add the interaction effect to response and effects plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-iv-some-fundamentals-of-statistical-modeling.html"><a href="part-iv-some-fundamentals-of-statistical-modeling.html"><i class="fa fa-check"></i>Part IV: Some Fundamentals of Statistical Modeling</a></li>
<li class="chapter" data-level="7" data-path="uncertainty.html"><a href="uncertainty.html"><i class="fa fa-check"></i><b>7</b> Variability and Uncertainty (Standard Deviations, Standard Errors, Confidence Intervals)</a>
<ul>
<li class="chapter" data-level="7.1" data-path="uncertainty.html"><a href="uncertainty.html#the-sample-standard-deviation-vs.-the-standard-error-of-the-mean"><i class="fa fa-check"></i><b>7.1</b> The sample standard deviation vs. the standard error of the mean</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="uncertainty.html"><a href="uncertainty.html#sample-standard-deviation"><i class="fa fa-check"></i><b>7.1.1</b> Sample standard deviation</a></li>
<li class="chapter" data-level="7.1.2" data-path="uncertainty.html"><a href="uncertainty.html#standard-error-of-the-mean"><i class="fa fa-check"></i><b>7.1.2</b> Standard error of the mean</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="uncertainty.html"><a href="uncertainty.html#using-google-sheets-to-generate-fake-data-to-explore-the-standard-error"><i class="fa fa-check"></i><b>7.2</b> Using Google Sheets to generate fake data to explore the standard error</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="uncertainty.html"><a href="uncertainty.html#steps"><i class="fa fa-check"></i><b>7.2.1</b> Steps</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="uncertainty.html"><a href="uncertainty.html#using-r-to-generate-fake-data-to-explore-the-standard-error"><i class="fa fa-check"></i><b>7.3</b> Using R to generate fake data to explore the standard error</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="uncertainty.html"><a href="uncertainty.html#part-i"><i class="fa fa-check"></i><b>7.3.1</b> part I</a></li>
<li class="chapter" data-level="7.3.2" data-path="uncertainty.html"><a href="uncertainty.html#part-ii---means"><i class="fa fa-check"></i><b>7.3.2</b> part II - means</a></li>
<li class="chapter" data-level="7.3.3" data-path="uncertainty.html"><a href="uncertainty.html#part-iii---how-do-sd-and-se-change-as-sample-size-n-increases"><i class="fa fa-check"></i><b>7.3.3</b> part III - how do SD and SE change as sample size (n) increases?</a></li>
<li class="chapter" data-level="7.3.4" data-path="uncertainty.html"><a href="uncertainty.html#part-iv-generating-fake-data-with-for-loops"><i class="fa fa-check"></i><b>7.3.4</b> Part IV – Generating fake data with for-loops</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="uncertainty.html"><a href="uncertainty.html#bootstrap"><i class="fa fa-check"></i><b>7.4</b> Bootstrapped standard errors</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="uncertainty.html"><a href="uncertainty.html#an-example-of-bootstrapped-standard-errors-using-vole-data"><i class="fa fa-check"></i><b>7.4.1</b> An example of bootstrapped standard errors using vole data</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="uncertainty.html"><a href="uncertainty.html#confidence-interval"><i class="fa fa-check"></i><b>7.5</b> Confidence Interval</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="uncertainty.html"><a href="uncertainty.html#interpretation-of-a-confidence-interval"><i class="fa fa-check"></i><b>7.5.1</b> Interpretation of a confidence interval</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="p-values.html"><a href="p-values.html"><i class="fa fa-check"></i><b>8</b> P-values</a>
<ul>
<li class="chapter" data-level="8.1" data-path="p-values.html"><a href="p-values.html#a-p-value-is-the-probability-of-sampling-a-value-as-or-more-extreme-than-the-test-statistic-if-sampling-from-a-null-distribution"><i class="fa fa-check"></i><b>8.1</b> A <em>p</em>-value is the probability of sampling a value as or more extreme than the test statistic if sampling from a null distribution</a></li>
<li class="chapter" data-level="8.2" data-path="p-values.html"><a href="p-values.html#pump-your-intuition-creating-a-null-distribution"><i class="fa fa-check"></i><b>8.2</b> Pump your intuition – Creating a null distribution</a></li>
<li class="chapter" data-level="8.3" data-path="p-values.html"><a href="p-values.html#a-null-distribution-of-t-values-the-t-distribution"><i class="fa fa-check"></i><b>8.3</b> A null distribution of <em>t</em>-values – the <em>t</em> distribution</a></li>
<li class="chapter" data-level="8.4" data-path="p-values.html"><a href="p-values.html#p-values-from-the-perspective-of-permutation"><i class="fa fa-check"></i><b>8.4</b> P-values from the perspective of permutation</a></li>
<li class="chapter" data-level="8.5" data-path="p-values.html"><a href="p-values.html#parametric-vs.-non-parametric-statistics"><i class="fa fa-check"></i><b>8.5</b> Parametric vs. non-parametric statistics</a></li>
<li class="chapter" data-level="8.6" data-path="p-values.html"><a href="p-values.html#frequentist-probability-and-the-interpretation-of-p-values"><i class="fa fa-check"></i><b>8.6</b> frequentist probability and the interpretation of p-values</a>
<ul>
<li class="chapter" data-level="8.6.1" data-path="p-values.html"><a href="p-values.html#background"><i class="fa fa-check"></i><b>8.6.1</b> Background</a></li>
<li class="chapter" data-level="8.6.2" data-path="p-values.html"><a href="p-values.html#this-book-covers-frequentist-approaches-to-statistical-modeling-and-when-a-probability-arises-such-as-the-p-value-of-a-test-statistic-this-will-be-a-frequentist-probability."><i class="fa fa-check"></i><b>8.6.2</b> This book covers frequentist approaches to statistical modeling and when a probability arises, such as the <em>p</em>-value of a test statistic, this will be a frequentist probability.</a></li>
<li class="chapter" data-level="8.6.3" data-path="p-values.html"><a href="p-values.html#two-interpretations-of-the-p-value"><i class="fa fa-check"></i><b>8.6.3</b> Two interpretations of the <em>p</em>-value</a></li>
<li class="chapter" data-level="8.6.4" data-path="p-values.html"><a href="p-values.html#nhst"><i class="fa fa-check"></i><b>8.6.4</b> NHST</a></li>
</ul></li>
<li class="chapter" data-level="8.7" data-path="p-values.html"><a href="p-values.html#some-major-misconceptions-of-the-p-value"><i class="fa fa-check"></i><b>8.7</b> Some major misconceptions of the <em>p</em>-value</a>
<ul>
<li class="chapter" data-level="8.7.1" data-path="p-values.html"><a href="p-values.html#misconception-p-is-the-probability-that-the-null-is-true-and-1-p-is-probability-that-the-alternative-is-true"><i class="fa fa-check"></i><b>8.7.1</b> Misconception: <em>p</em> is the probability that the null is true <em>and</em> <span class="math inline">\(1-p\)</span> is probability that the alternative is true</a></li>
<li class="chapter" data-level="8.7.2" data-path="p-values.html"><a href="p-values.html#misconception-a-p-value-is-repeatable"><i class="fa fa-check"></i><b>8.7.2</b> Misconception: a <em>p</em>-value is repeatable</a></li>
<li class="chapter" data-level="8.7.3" data-path="p-values.html"><a href="p-values.html#misconception-0.05-is-the-lifetime-rate-of-false-discoveries"><i class="fa fa-check"></i><b>8.7.3</b> Misconception: 0.05 is the lifetime rate of false discoveries</a></li>
<li class="chapter" data-level="8.7.4" data-path="p-values.html"><a href="p-values.html#misconception-a-low-p-value-indicates-an-important-effect"><i class="fa fa-check"></i><b>8.7.4</b> Misconception: a low <em>p</em>-value indicates an important effect</a></li>
<li class="chapter" data-level="8.7.5" data-path="p-values.html"><a href="p-values.html#misconception-a-low-p-value-indicates-high-model-fit-or-high-predictive-capacity"><i class="fa fa-check"></i><b>8.7.5</b> Misconception: a low <em>p</em>-value indicates high model fit or high predictive capacity</a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="p-values.html"><a href="p-values.html#what-the-p-value-does-not-mean"><i class="fa fa-check"></i><b>8.8</b> What the <em>p</em>-value does not mean</a></li>
<li class="chapter" data-level="8.9" data-path="p-values.html"><a href="p-values.html#recommendations"><i class="fa fa-check"></i><b>8.9</b> Recommendations</a>
<ul>
<li class="chapter" data-level="8.9.1" data-path="p-values.html"><a href="p-values.html#primary-sources-for-recommendations"><i class="fa fa-check"></i><b>8.9.1</b> Primary sources for recommendations</a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="p-values.html"><a href="p-values.html#problems"><i class="fa fa-check"></i><b>8.10</b> Problems</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="errors-in-inference.html"><a href="errors-in-inference.html"><i class="fa fa-check"></i><b>9</b> Errors in inference</a>
<ul>
<li class="chapter" data-level="9.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#classical-nhst-concepts-of-wrong"><i class="fa fa-check"></i><b>9.1</b> Classical NHST concepts of wrong</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-i-error"><i class="fa fa-check"></i><b>9.1.1</b> Type I error</a></li>
<li class="chapter" data-level="9.1.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#power"><i class="fa fa-check"></i><b>9.1.2</b> Power</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#a-non-neyman-pearson-concept-of-power"><i class="fa fa-check"></i><b>9.2</b> A non-Neyman-Pearson concept of power</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="errors-in-inference.html"><a href="errors-in-inference.html#estimation-error"><i class="fa fa-check"></i><b>9.2.1</b> Estimation error</a></li>
<li class="chapter" data-level="9.2.2" data-path="errors-in-inference.html"><a href="errors-in-inference.html#coverage"><i class="fa fa-check"></i><b>9.2.2</b> Coverage</a></li>
<li class="chapter" data-level="9.2.3" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-s-error"><i class="fa fa-check"></i><b>9.2.3</b> Type S error</a></li>
<li class="chapter" data-level="9.2.4" data-path="errors-in-inference.html"><a href="errors-in-inference.html#type-m-error"><i class="fa fa-check"></i><b>9.2.4</b> Type M error</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-v-introduction-to-linear-models.html"><a href="part-v-introduction-to-linear-models.html"><i class="fa fa-check"></i>Part V: Introduction to Linear Models</a></li>
<li class="chapter" data-level="10" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html"><i class="fa fa-check"></i><b>10</b> An introduction to linear models</a>
<ul>
<li class="chapter" data-level="10.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#lm-specifications"><i class="fa fa-check"></i><b>10.1</b> Two specifications of a linear model</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#the-error-draw-specification"><i class="fa fa-check"></i><b>10.1.1</b> The “error draw” specification</a></li>
<li class="chapter" data-level="10.1.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#the-conditional-draw-specification"><i class="fa fa-check"></i><b>10.1.2</b> The “conditional draw” specification</a></li>
<li class="chapter" data-level="10.1.3" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#comparing-the-error-draw-and-conditional-draw-ways-of-specifying-the-linear-model"><i class="fa fa-check"></i><b>10.1.3</b> Comparing the error-draw and conditional-draw ways of specifying the linear model</a></li>
<li class="chapter" data-level="10.1.4" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#anova-notation-of-a-linear-model"><i class="fa fa-check"></i><b>10.1.4</b> ANOVA notation of a linear model</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#a-linear-model-can-be-fit-to-data-with-continuous-discrete-or-categorical-x-variables"><i class="fa fa-check"></i><b>10.2</b> A linear model can be fit to data with continuous, discrete, or categorical <span class="math inline">\(X\)</span> variables</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#fitting-linear-models-to-experimental-data-in-which-the-x-variable-is-continuous-or-discrete"><i class="fa fa-check"></i><b>10.2.1</b> Fitting linear models to experimental data in which the <span class="math inline">\(X\)</span> variable is continuous or discrete</a></li>
<li class="chapter" data-level="10.2.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#slope"><i class="fa fa-check"></i><b>10.2.2</b> Fitting linear models to experimental data in which the <span class="math inline">\(X\)</span> variable is categorical</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#statistical-models-are-used-for-prediction-explanation-and-description"><i class="fa fa-check"></i><b>10.3</b> Statistical models are used for prediction, explanation, and description</a></li>
<li class="chapter" data-level="10.4" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#what-do-we-call-the-x-and-y-variables"><i class="fa fa-check"></i><b>10.4</b> What do we call the <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span> variables?</a></li>
<li class="chapter" data-level="10.5" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#modeling-strategy"><i class="fa fa-check"></i><b>10.5</b> Modeling strategy</a></li>
<li class="chapter" data-level="10.6" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#predictions-from-the-model"><i class="fa fa-check"></i><b>10.6</b> Predictions from the model</a></li>
<li class="chapter" data-level="10.7" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#inference-from-the-model"><i class="fa fa-check"></i><b>10.7</b> Inference from the model</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#assumptions-for-inference-with-a-statistical-model"><i class="fa fa-check"></i><b>10.7.1</b> Assumptions for inference with a statistical model</a></li>
<li class="chapter" data-level="10.7.2" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#specific-assumptions-for-inference-with-a-linear-model"><i class="fa fa-check"></i><b>10.7.2</b> Specific assumptions for inference with a linear model</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="an-introduction-to-linear-models.html"><a href="an-introduction-to-linear-models.html#linear-modelregression-model-orstatistical-model"><i class="fa fa-check"></i><b>10.8</b> “linear model,”regression model“, or”statistical model"?</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="regression.html"><a href="regression.html"><i class="fa fa-check"></i><b>11</b> Linear models with a single, continuous <em>X</em> (“regression”)</a>
<ul>
<li class="chapter" data-level="11.1" data-path="regression.html"><a href="regression.html#a-linear-model-with-a-single-continuous-x-is-classical-regression"><i class="fa fa-check"></i><b>11.1</b> A linear model with a single, continuous <em>X</em> is classical “regression”</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="regression.html"><a href="regression.html#analysis-of-green-down-data"><i class="fa fa-check"></i><b>11.1.1</b> Analysis of “green-down” data</a></li>
<li class="chapter" data-level="11.1.2" data-path="regression.html"><a href="regression.html#learning-from-the-green-down-example"><i class="fa fa-check"></i><b>11.1.2</b> Learning from the green-down example</a></li>
<li class="chapter" data-level="11.1.3" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-explanation-causal-models"><i class="fa fa-check"></i><b>11.1.3</b> Using a regression model for “explanation” – causal models</a></li>
<li class="chapter" data-level="11.1.4" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-prediction-prediction-models"><i class="fa fa-check"></i><b>11.1.4</b> Using a regression model for prediction – prediction models</a></li>
<li class="chapter" data-level="11.1.5" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-creating-a-new-response-variable-comparing-slopes-of-longitudinal-data"><i class="fa fa-check"></i><b>11.1.5</b> Using a regression model for creating a new response variable – comparing slopes of longitudinal data</a></li>
<li class="chapter" data-level="11.1.6" data-path="regression.html"><a href="regression.html#using-a-regression-model-for-for-calibration"><i class="fa fa-check"></i><b>11.1.6</b> Using a regression model for for calibration</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="regression.html"><a href="regression.html#working-in-r-1"><i class="fa fa-check"></i><b>11.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="plotting-models.html"><a href="plotting-models.html#fit-the-model"><i class="fa fa-check"></i><b>11.2.1</b> Fitting the linear model</a></li>
<li class="chapter" data-level="11.2.2" data-path="regression.html"><a href="regression.html#getting-to-know-the-linear-model-the-summary-function"><i class="fa fa-check"></i><b>11.2.2</b> Getting to know the linear model: the <code>summary</code> function</a></li>
<li class="chapter" data-level="11.2.3" data-path="regression.html"><a href="regression.html#inference-the-coefficient-table"><i class="fa fa-check"></i><b>11.2.3</b> Inference – the coefficient table</a></li>
<li class="chapter" data-level="11.2.4" data-path="regression.html"><a href="regression.html#how-good-is-our-model-model-checking"><i class="fa fa-check"></i><b>11.2.4</b> How good is our model? – Model checking</a></li>
<li class="chapter" data-level="11.2.5" data-path="regression.html"><a href="regression.html#plotting-models-with-continuous-x"><i class="fa fa-check"></i><b>11.2.5</b> Plotting models with continuous <em>X</em></a></li>
<li class="chapter" data-level="11.2.6" data-path="regression.html"><a href="regression.html#creating-a-table-of-predicted-values-and-95-prediction-intervals"><i class="fa fa-check"></i><b>11.2.6</b> Creating a table of predicted values and 95% prediction intervals</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="regression.html"><a href="regression.html#hidden-code"><i class="fa fa-check"></i><b>11.3</b> Hidden code</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="regression.html"><a href="regression.html#import-and-plot-of-fig2c-ecosystem-warming-experimental-data"><i class="fa fa-check"></i><b>11.3.1</b> Import and plot of fig2c (ecosystem warming experimental) data</a></li>
<li class="chapter" data-level="11.3.2" data-path="regression.html"><a href="regression.html#import-and-plot-efig_3d-ecosysem-warming-observational-data"><i class="fa fa-check"></i><b>11.3.2</b> Import and plot efig_3d (Ecosysem warming observational) data</a></li>
<li class="chapter" data-level="11.3.3" data-path="regression.html"><a href="regression.html#import-and-plot-of-fig1f-methionine-restriction-data"><i class="fa fa-check"></i><b>11.3.3</b> Import and plot of fig1f (methionine restriction) data</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="regression.html"><a href="regression.html#try-it"><i class="fa fa-check"></i><b>11.4</b> Try it</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="regression.html"><a href="regression.html#a-prediction-model-from-the-literature"><i class="fa fa-check"></i><b>11.4.1</b> A prediction model from the literature</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="regression.html"><a href="regression.html#intuition-pumps"><i class="fa fa-check"></i><b>11.5</b> Intuition pumps</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="regression.html"><a href="regression.html#correlation-and-r2"><i class="fa fa-check"></i><b>11.5.1</b> Correlation and $R^2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="oneway.html"><a href="oneway.html"><i class="fa fa-check"></i><b>12</b> Linear models with a single, categorical <em>X</em> (“t-tests” and “ANOVA”)</a>
<ul>
<li class="chapter" data-level="12.1" data-path="oneway.html"><a href="oneway.html#oneway-data"><i class="fa fa-check"></i><b>12.1</b> A linear model with a single, categorical <em>X</em> variable estimates the effects of the levels of <em>X</em> on the response</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="oneway.html"><a href="oneway.html#oneway-example1"><i class="fa fa-check"></i><b>12.1.1</b> Example 1 (fig3d) – two treatment levels (“groups”)</a></li>
<li class="chapter" data-level="12.1.2" data-path="oneway.html"><a href="oneway.html#understanding-the-analysis-with-two-treatment-levels"><i class="fa fa-check"></i><b>12.1.2</b> Understanding the analysis with two treatment levels</a></li>
<li class="chapter" data-level="12.1.3" data-path="oneway.html"><a href="oneway.html#oneway-example2"><i class="fa fa-check"></i><b>12.1.3</b> Example 2 – three treatment levels (“groups”)</a></li>
<li class="chapter" data-level="12.1.4" data-path="oneway.html"><a href="oneway.html#understanding-the-analysis-with-three-or-more-treatment-levels"><i class="fa fa-check"></i><b>12.1.4</b> Understanding the analysis with three (or more) treatment levels</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="oneway.html"><a href="oneway.html#working-in-r-2"><i class="fa fa-check"></i><b>12.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="oneway.html"><a href="oneway.html#fit-the-model-1"><i class="fa fa-check"></i><b>12.2.1</b> Fit the model</a></li>
<li class="chapter" data-level="12.2.2" data-path="oneway.html"><a href="oneway.html#controlling-the-output-in-tables-using-the-coefficient-table-as-an-example"><i class="fa fa-check"></i><b>12.2.2</b> Controlling the output in tables using the coefficient table as an example</a></li>
<li class="chapter" data-level="12.2.3" data-path="oneway.html"><a href="oneway.html#using-the-emmeans-function"><i class="fa fa-check"></i><b>12.2.3</b> Using the emmeans function</a></li>
<li class="chapter" data-level="12.2.4" data-path="oneway.html"><a href="oneway.html#using-the-contrast-function"><i class="fa fa-check"></i><b>12.2.4</b> Using the contrast function</a></li>
<li class="chapter" data-level="12.2.5" data-path="oneway.html"><a href="oneway.html#how-to-generate-anova-tables"><i class="fa fa-check"></i><b>12.2.5</b> How to generate ANOVA tables</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="oneway.html"><a href="oneway.html#hidden-code-1"><i class="fa fa-check"></i><b>12.3</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="oneway.html"><a href="oneway.html#importing-and-wrangling-the-fig_3d-data-for-example-1"><i class="fa fa-check"></i><b>12.3.1</b> Importing and wrangling the fig_3d data for example 1</a></li>
<li class="chapter" data-level="12.3.2" data-path="oneway.html"><a href="oneway.html#importing-and-wrangling-the-fig2a-data-for-example-2"><i class="fa fa-check"></i><b>12.3.2</b> Importing and wrangling the fig2a data for example 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="model-checking.html"><a href="model-checking.html"><i class="fa fa-check"></i><b>13</b> Model Checking</a>
<ul>
<li class="chapter" data-level="13.1" data-path="model-checking.html"><a href="model-checking.html#check-check"><i class="fa fa-check"></i><b>13.1</b> All statistical analyses should be followed by model checking</a></li>
<li class="chapter" data-level="13.2" data-path="model-checking.html"><a href="model-checking.html#linear-model-assumptions"><i class="fa fa-check"></i><b>13.2</b> Linear model assumptions</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="model-checking.html"><a href="model-checking.html#a-bit-about-iid"><i class="fa fa-check"></i><b>13.2.1</b> A bit about IID</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="model-checking.html"><a href="model-checking.html#diagnostic-plots-use-the-residuals-from-the-model-fit"><i class="fa fa-check"></i><b>13.3</b> Diagnostic plots use the residuals from the model fit</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="model-checking.html"><a href="model-checking.html#residuals"><i class="fa fa-check"></i><b>13.3.1</b> Residuals</a></li>
<li class="chapter" data-level="13.3.2" data-path="model-checking.html"><a href="model-checking.html#normal-qq"><i class="fa fa-check"></i><b>13.3.2</b> A Normal Q-Q plot is used to check for characteristic departures from Normality</a></li>
<li class="chapter" data-level="13.3.3" data-path="model-checking.html"><a href="model-checking.html#mapping-qq-plot-departures-from-normality"><i class="fa fa-check"></i><b>13.3.3</b> Mapping QQ-plot departures from Normality</a></li>
<li class="chapter" data-level="13.3.4" data-path="model-checking.html"><a href="model-checking.html#model-checking-spread-level"><i class="fa fa-check"></i><b>13.3.4</b> Model checking homoskedasticity</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="model-checking.html"><a href="model-checking.html#using-r"><i class="fa fa-check"></i><b>13.4</b> Using R</a></li>
<li class="chapter" data-level="13.5" data-path="model-checking.html"><a href="model-checking.html#hidden-code-2"><i class="fa fa-check"></i><b>13.5</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="model-checking.html"><a href="model-checking.html#normal-q-q-plots"><i class="fa fa-check"></i><b>13.5.1</b> Normal Q-Q plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="violations.html"><a href="violations.html"><i class="fa fa-check"></i><b>14</b> Violations of independence, homogeneity, or Normality</a>
<ul>
<li class="chapter" data-level="14.1" data-path="violations.html"><a href="violations.html#oneway-paired-t"><i class="fa fa-check"></i><b>14.1</b> Lack of independence</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="violations.html"><a href="violations.html#violations-paired-t"><i class="fa fa-check"></i><b>14.1.1</b> Example 1 (exp1b) – a paired t-test is a special case of a linear mixed model</a></li>
<li class="chapter" data-level="14.1.2" data-path="violations.html"><a href="violations.html#violations-rmanova"><i class="fa fa-check"></i><b>14.1.2</b> Example 2 (diHOME exp2a) – A repeated measures ANOVA is a special case of a linear mixed model</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="violations.html"><a href="violations.html#oneway-welch"><i class="fa fa-check"></i><b>14.2</b> Heterogeneity of variances</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="violations.html"><a href="violations.html#when-groups-of-the-focal-test-have-variance"><i class="fa fa-check"></i><b>14.2.1</b> When groups of the focal test have &gt;&gt; variance</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="violations.html"><a href="violations.html#the-conditional-response-isnt-normal"><i class="fa fa-check"></i><b>14.3</b> The conditional response isn’t Normal</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="violations.html"><a href="violations.html#violations-6f"><i class="fa fa-check"></i><b>14.3.1</b> Example 1 (fig6f) – Linear models for non-normal count data</a></li>
<li class="chapter" data-level="14.3.2" data-path="violations.html"><a href="violations.html#my-data-arent-normal-what-is-the-best-practice"><i class="fa fa-check"></i><b>14.3.2</b> My data aren’t normal, what is the best practice?</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="violations.html"><a href="violations.html#hidden-code-3"><i class="fa fa-check"></i><b>14.4</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-exp1b-data"><i class="fa fa-check"></i><b>14.4.1</b> Importing and wrangling the exp1b data</a></li>
<li class="chapter" data-level="14.4.2" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-exp2a-data"><i class="fa fa-check"></i><b>14.4.2</b> Importing and wrangling the exp2a data</a></li>
<li class="chapter" data-level="14.4.3" data-path="violations.html"><a href="violations.html#importing-and-wrangling-the-fig6f-data"><i class="fa fa-check"></i><b>14.4.3</b> Importing and wrangling the fig6f data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="issues.html"><a href="issues.html"><i class="fa fa-check"></i><b>15</b> Issues in inference</a>
<ul>
<li class="chapter" data-level="15.1" data-path="issues.html"><a href="issues.html#replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means"><i class="fa fa-check"></i><b>15.1</b> Replicated experiments – include <span class="math inline">\(\texttt{Experiment}\)</span> as a random factor (better than one-way ANOVA of means)</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="issues.html"><a href="issues.html#issues-exp4d"><i class="fa fa-check"></i><b>15.1.1</b> Multiple experiments Example 1 (wound healing Exp4d)</a></li>
<li class="chapter" data-level="15.1.2" data-path="issues.html"><a href="issues.html#issues-combined-experiments"><i class="fa fa-check"></i><b>15.1.2</b> Models for combining replicated experiments</a></li>
<li class="chapter" data-level="15.1.3" data-path="issues.html"><a href="issues.html#understanding-model-exp4d_m1"><i class="fa fa-check"></i><b>15.1.3</b> Understanding Model <code>exp4d_m1</code></a></li>
<li class="chapter" data-level="15.1.4" data-path="issues.html"><a href="issues.html#the-univariate-model-is-equivalent-to-a-linear-mixed-model-of-the-aggregated-data-model-exp4d_m2"><i class="fa fa-check"></i><b>15.1.4</b> The univariate model is equivalent to a linear mixed model of the aggregated data (Model exp4d_m2)</a></li>
<li class="chapter" data-level="15.1.5" data-path="issues.html"><a href="issues.html#a-linear-mixed-model-of-the-full-data"><i class="fa fa-check"></i><b>15.1.5</b> A linear mixed model of the full data</a></li>
<li class="chapter" data-level="15.1.6" data-path="issues.html"><a href="issues.html#analysis-of-the-experiment-means-has-less-precision-and-power"><i class="fa fa-check"></i><b>15.1.6</b> Analysis of the experiment means has less precision and power</a></li>
<li class="chapter" data-level="15.1.7" data-path="issues.html"><a href="issues.html#dont-do-this-a-t-testfixed-effect-anova-of-the-full-data"><i class="fa fa-check"></i><b>15.1.7</b> Don’t do this – a t-test/fixed-effect ANOVA of the full data</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="issues.html"><a href="issues.html#issues-pre-post"><i class="fa fa-check"></i><b>15.2</b> Comparing change from baseline (pre-post)</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="issues.html"><a href="issues.html#pre-post-example-1-dpp4-fig4c"><i class="fa fa-check"></i><b>15.2.1</b> Pre-post example 1 (DPP4 fig4c)</a></li>
<li class="chapter" data-level="15.2.2" data-path="issues.html"><a href="issues.html#what-if-the-data-in-example-1-were-from-from-an-experiment-where-the-treatment-was-applied-prior-to-the-baseline-measure"><i class="fa fa-check"></i><b>15.2.2</b> What if the data in example 1 were from from an experiment where the treatment was applied prior to the baseline measure?</a></li>
<li class="chapter" data-level="15.2.3" data-path="issues.html"><a href="issues.html#pre-post-example-2-xx-males-fig1c"><i class="fa fa-check"></i><b>15.2.3</b> Pre-post example 2 (XX males fig1c)</a></li>
<li class="chapter" data-level="15.2.4" data-path="issues.html"><a href="issues.html#issues-regression-to-mean"><i class="fa fa-check"></i><b>15.2.4</b> Regression to the mean</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="issues.html"><a href="issues.html#longitudinal-designs-with-more-than-one-post-baseline-measure"><i class="fa fa-check"></i><b>15.3</b> Longitudinal designs with more than one-post baseline measure</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="issues.html"><a href="issues.html#issues-auc"><i class="fa fa-check"></i><b>15.3.1</b> Area under the curve (AUC)</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="issues.html"><a href="issues.html#comparing-ratios"><i class="fa fa-check"></i><b>15.4</b> Comparing ratios</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="issues.html"><a href="issues.html#kinds-of-ratios-in-experimental-biology"><i class="fa fa-check"></i><b>15.4.1</b> Kinds of ratios in experimental biology</a></li>
<li class="chapter" data-level="15.4.2" data-path="issues.html"><a href="issues.html#issues-offset"><i class="fa fa-check"></i><b>15.4.2</b> Example 1 – The ratio is a density (number of something per area)</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="issues.html"><a href="issues.html#dont-do-this-stuff"><i class="fa fa-check"></i><b>15.5</b> Don’t do this stuff</a>
<ul>
<li class="chapter" data-level="15.5.1" data-path="issues.html"><a href="issues.html#normalize-the-response-so-that-all-control-values-are-equal-to-1."><i class="fa fa-check"></i><b>15.5.1</b> Normalize the response so that all control values are equal to 1.</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="issues.html"><a href="issues.html#a-difference-in-significance-is-not-necessarily-significant"><i class="fa fa-check"></i><b>15.6</b> A difference in significance is not necessarily significant</a></li>
<li class="chapter" data-level="15.7" data-path="issues.html"><a href="issues.html#researcher-degrees-of-freedom"><i class="fa fa-check"></i><b>15.7</b> Researcher degrees of freedom</a></li>
<li class="chapter" data-level="15.8" data-path="issues.html"><a href="issues.html#hidden-code-4"><i class="fa fa-check"></i><b>15.8</b> Hidden code</a>
<ul>
<li class="chapter" data-level="15.8.1" data-path="issues.html"><a href="issues.html#import-exp4d-vimentin-cell-count-data-replicate-experiments-example"><i class="fa fa-check"></i><b>15.8.1</b> Import exp4d vimentin cell count data (replicate experiments example)</a></li>
<li class="chapter" data-level="15.8.2" data-path="issues.html"><a href="issues.html#import-fig4c-data"><i class="fa fa-check"></i><b>15.8.2</b> Import Fig4c data</a></li>
<li class="chapter" data-level="15.8.3" data-path="issues.html"><a href="issues.html#xx-males-fig1c"><i class="fa fa-check"></i><b>15.8.3</b> XX males fig1c</a></li>
<li class="chapter" data-level="15.8.4" data-path="issues.html"><a href="issues.html#generation-of-fake-data-to-illustrate-regression-to-the-mean"><i class="fa fa-check"></i><b>15.8.4</b> Generation of fake data to illustrate regression to the mean</a></li>
<li class="chapter" data-level="15.8.5" data-path="issues.html"><a href="issues.html#import-fig3f"><i class="fa fa-check"></i><b>15.8.5</b> Import fig3f</a></li>
</ul></li>
</ul></li>
<li><a href="part-vi-more-than-one-x-multivariable-models.html#part-vi-more-than-one-x-multivariable-models">Part VI: More than one <span class="math inline">\(X\)</span> – Multivariable Models</a></li>
<li class="chapter" data-level="16" data-path="covariates.html"><a href="covariates.html"><i class="fa fa-check"></i><b>16</b> Linear models with added covariates (“ANCOVA”)</a>
<ul>
<li class="chapter" data-level="16.1" data-path="covariates.html"><a href="covariates.html#adding-covariates-can-increases-the-precision-of-the-effect-of-interest"><i class="fa fa-check"></i><b>16.1</b> Adding covariates can increases the precision of the effect of interest</a></li>
<li class="chapter" data-level="16.2" data-path="covariates.html"><a href="covariates.html#understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data"><i class="fa fa-check"></i><b>16.2</b> Understanding a linear model with an added covariate – heart necrosis data</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="covariates.html"><a href="covariates.html#fit-the-model-5"><i class="fa fa-check"></i><b>16.2.1</b> Fit the model</a></li>
<li class="chapter" data-level="16.2.2" data-path="covariates.html"><a href="covariates.html#plot-the-model-2"><i class="fa fa-check"></i><b>16.2.2</b> Plot the model</a></li>
<li class="chapter" data-level="16.2.3" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-model-coefficients"><i class="fa fa-check"></i><b>16.2.3</b> Interpretation of the model coefficients</a></li>
<li class="chapter" data-level="16.2.4" data-path="covariates.html"><a href="covariates.html#everything-adds-up"><i class="fa fa-check"></i><b>16.2.4</b> Everything adds up</a></li>
<li class="chapter" data-level="16.2.5" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-estimated-marginal-means"><i class="fa fa-check"></i><b>16.2.5</b> Interpretation of the estimated marginal means</a></li>
<li class="chapter" data-level="16.2.6" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-contrasts"><i class="fa fa-check"></i><b>16.2.6</b> Interpretation of the contrasts</a></li>
<li class="chapter" data-level="16.2.7" data-path="covariates.html"><a href="covariates.html#adding-the-covariate-improves-inference"><i class="fa fa-check"></i><b>16.2.7</b> Adding the covariate improves inference</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="covariates.html"><a href="covariates.html#understanding-interaction-effects-with-covariates"><i class="fa fa-check"></i><b>16.3</b> Understanding interaction effects with covariates</a>
<ul>
<li class="chapter" data-level="16.3.1" data-path="covariates.html"><a href="covariates.html#fit-the-model-6"><i class="fa fa-check"></i><b>16.3.1</b> Fit the model</a></li>
<li class="chapter" data-level="16.3.2" data-path="covariates.html"><a href="covariates.html#plot-the-model-with-interaction-effect"><i class="fa fa-check"></i><b>16.3.2</b> Plot the model with interaction effect</a></li>
<li class="chapter" data-level="16.3.3" data-path="covariates.html"><a href="covariates.html#interpretation-of-the-model-coefficients-1"><i class="fa fa-check"></i><b>16.3.3</b> Interpretation of the model coefficients</a></li>
<li class="chapter" data-level="16.3.4" data-path="covariates.html"><a href="covariates.html#what-is-the-effect-of-a-treatment-if-interactions-are-modeled-it-depends."><i class="fa fa-check"></i><b>16.3.4</b> What is the effect of a treatment, if interactions are modeled? – it depends.</a></li>
<li class="chapter" data-level="16.3.5" data-path="covariates.html"><a href="covariates.html#which-model-do-we-use-mathcalm_1-or-mathcalm_2"><i class="fa fa-check"></i><b>16.3.5</b> Which model do we use, <span class="math inline">\(\mathcal{M}_1\)</span> or <span class="math inline">\(\mathcal{M}_2\)</span>?</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="covariates.html"><a href="covariates.html#understanding-ancova-tables"><i class="fa fa-check"></i><b>16.4</b> Understanding ANCOVA tables</a></li>
<li class="chapter" data-level="16.5" data-path="covariates.html"><a href="covariates.html#working-in-r-3"><i class="fa fa-check"></i><b>16.5</b> Working in R</a>
<ul>
<li class="chapter" data-level="16.5.1" data-path="covariates.html"><a href="covariates.html#importing-the-heart-necrosis-data"><i class="fa fa-check"></i><b>16.5.1</b> Importing the heart necrosis data</a></li>
<li class="chapter" data-level="16.5.2" data-path="covariates.html"><a href="covariates.html#fitting-the-model"><i class="fa fa-check"></i><b>16.5.2</b> Fitting the model</a></li>
<li class="chapter" data-level="16.5.3" data-path="covariates.html"><a href="covariates.html#using-the-emmeans-function-1"><i class="fa fa-check"></i><b>16.5.3</b> Using the emmeans function</a></li>
<li class="chapter" data-level="16.5.4" data-path="covariates.html"><a href="covariates.html#ancova-tables"><i class="fa fa-check"></i><b>16.5.4</b> ANCOVA tables</a></li>
<li class="chapter" data-level="16.5.5" data-path="covariates.html"><a href="covariates.html#plotting-the-model"><i class="fa fa-check"></i><b>16.5.5</b> Plotting the model</a></li>
</ul></li>
<li class="chapter" data-level="16.6" data-path="covariates.html"><a href="covariates.html#best-practices"><i class="fa fa-check"></i><b>16.6</b> Best practices</a>
<ul>
<li class="chapter" data-level="16.6.1" data-path="covariates.html"><a href="covariates.html#do-not-use-a-ratio-of-partwhole-as-a-response-variable-instead-add-the-denominator-as-a-covariate"><i class="fa fa-check"></i><b>16.6.1</b> Do not use a ratio of part:whole as a response variable – instead add the denominator as a covariate</a></li>
<li class="chapter" data-level="16.6.2" data-path="covariates.html"><a href="covariates.html#do-not-use-change-from-baseline-as-a-response-variable-instead-add-the-baseline-measure-as-a-covariate"><i class="fa fa-check"></i><b>16.6.2</b> Do not use change from baseline as a response variable – instead add the baseline measure as a covariate</a></li>
<li class="chapter" data-level="16.6.3" data-path="covariates.html"><a href="covariates.html#do-not-test-for-balance-of-baseline-measures"><i class="fa fa-check"></i><b>16.6.3</b> Do not “test for balance” of baseline measures</a></li>
</ul></li>
<li class="chapter" data-level="16.7" data-path="covariates.html"><a href="covariates.html#best-practices-2-use-a-covariate-instead-of-normalizing-a-response"><i class="fa fa-check"></i><b>16.7</b> Best practices 2: Use a covariate instead of normalizing a response</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="factorial.html"><a href="factorial.html"><i class="fa fa-check"></i><b>17</b> Linear models with two categorical <span class="math inline">\(X\)</span> – Factorial linear models (“two-way ANOVA”)</a>
<ul>
<li class="chapter" data-level="17.1" data-path="factorial.html"><a href="factorial.html#a-linear-model-with-crossed-factors-estimates-interaction-effects"><i class="fa fa-check"></i><b>17.1</b> A linear model with crossed factors estimates interaction effects</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="factorial.html"><a href="factorial.html#an-interaction-is-a-difference-in-simple-effects"><i class="fa fa-check"></i><b>17.1.1</b> An interaction is a difference in simple effects</a></li>
<li class="chapter" data-level="17.1.2" data-path="factorial.html"><a href="factorial.html#a-linear-model-with-crossed-factors-includes-interaction-effects"><i class="fa fa-check"></i><b>17.1.2</b> A linear model with crossed factors includes interaction effects</a></li>
<li class="chapter" data-level="17.1.3" data-path="factorial.html"><a href="factorial.html#factorial-experiments-are-frequently-analyzed-as-flattened-linear-models-in-the-experimental-biology-literature"><i class="fa fa-check"></i><b>17.1.3</b> factorial experiments are frequently analyzed as flattened linear models in the experimental biology literature</a></li>
</ul></li>
<li class="chapter" data-level="17.2" data-path="factorial.html"><a href="factorial.html#example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data"><i class="fa fa-check"></i><b>17.2</b> Example 1 – Estimation of a treatment effect relative to a control effect (“Something different”) (Experiment 2j glucose uptake data)</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="factorial.html"><a href="factorial.html#twoway-understand1"><i class="fa fa-check"></i><b>17.2.1</b> Understand the experimental design</a></li>
<li class="chapter" data-level="17.2.2" data-path="factorial.html"><a href="factorial.html#fit-the-linear-model"><i class="fa fa-check"></i><b>17.2.2</b> Fit the linear model</a></li>
<li class="chapter" data-level="17.2.3" data-path="factorial.html"><a href="factorial.html#twoway-exp2j-inf"><i class="fa fa-check"></i><b>17.2.3</b> Inference</a></li>
<li class="chapter" data-level="17.2.4" data-path="factorial.html"><a href="factorial.html#plot-the-model-3"><i class="fa fa-check"></i><b>17.2.4</b> Plot the model</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-1"><i class="fa fa-check"></i><b>17.3</b> Understanding the linear model with crossed factors 1</a>
<ul>
<li class="chapter" data-level="17.3.1" data-path="factorial.html"><a href="factorial.html#twoway-what-coefs-are"><i class="fa fa-check"></i><b>17.3.1</b> What the coefficients are</a></li>
<li class="chapter" data-level="17.3.2" data-path="factorial.html"><a href="factorial.html#the-interaction-effect-is-something-different"><i class="fa fa-check"></i><b>17.3.2</b> The interaction effect is something different</a></li>
<li class="chapter" data-level="17.3.3" data-path="factorial.html"><a href="factorial.html#why-we-want-to-compare-the-treatment-effect-to-a-control-effect"><i class="fa fa-check"></i><b>17.3.3</b> Why we want to compare the treatment effect to a control effect</a></li>
<li class="chapter" data-level="17.3.4" data-path="factorial.html"><a href="factorial.html#the-order-of-the-factors-in-the-model-tells-the-same-story-differently"><i class="fa fa-check"></i><b>17.3.4</b> The order of the factors in the model tells the same story differently</a></li>
<li class="chapter" data-level="17.3.5" data-path="factorial.html"><a href="factorial.html#power-for-the-interaction-effect-is-less-than-that-for-simple-effects"><i class="fa fa-check"></i><b>17.3.5</b> Power for the interaction effect is less than that for simple effects</a></li>
<li class="chapter" data-level="17.3.6" data-path="factorial.html"><a href="factorial.html#planned-comparisons-vs.-post-hoc-tests"><i class="fa fa-check"></i><b>17.3.6</b> Planned comparisons vs. post-hoc tests</a></li>
</ul></li>
<li class="chapter" data-level="17.4" data-path="factorial.html"><a href="factorial.html#example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data"><i class="fa fa-check"></i><b>17.4</b> Example 2: Estimation of the effect of background condition on an effect (“it depends”) (Experiment 3e lesian area data)</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="factorial.html"><a href="factorial.html#understand-the-experimental-design"><i class="fa fa-check"></i><b>17.4.1</b> Understand the experimental design</a></li>
<li class="chapter" data-level="17.4.2" data-path="factorial.html"><a href="factorial.html#fit-the-linear-model-1"><i class="fa fa-check"></i><b>17.4.2</b> Fit the linear model</a></li>
<li class="chapter" data-level="17.4.3" data-path="factorial.html"><a href="factorial.html#check-the-model-1"><i class="fa fa-check"></i><b>17.4.3</b> Check the model</a></li>
<li class="chapter" data-level="17.4.4" data-path="factorial.html"><a href="factorial.html#inference-from-the-model-2"><i class="fa fa-check"></i><b>17.4.4</b> Inference from the model</a></li>
<li class="chapter" data-level="17.4.5" data-path="factorial.html"><a href="factorial.html#plot-the-model-4"><i class="fa fa-check"></i><b>17.4.5</b> Plot the model</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-2"><i class="fa fa-check"></i><b>17.5</b> Understanding the linear model with crossed factors 2</a>
<ul>
<li class="chapter" data-level="17.5.1" data-path="factorial.html"><a href="factorial.html#twoway-marginal-means"><i class="fa fa-check"></i><b>17.5.1</b> Conditional and marginal means</a></li>
<li class="chapter" data-level="17.5.2" data-path="factorial.html"><a href="factorial.html#simple-conditional-effects"><i class="fa fa-check"></i><b>17.5.2</b> Simple (conditional) effects</a></li>
<li class="chapter" data-level="17.5.3" data-path="factorial.html"><a href="factorial.html#twoway-marginal-effects"><i class="fa fa-check"></i><b>17.5.3</b> Marginal effects</a></li>
<li class="chapter" data-level="17.5.4" data-path="factorial.html"><a href="factorial.html#the-additive-model"><i class="fa fa-check"></i><b>17.5.4</b> The additive model</a></li>
<li class="chapter" data-level="17.5.5" data-path="factorial.html"><a href="factorial.html#twoway-reduce"><i class="fa fa-check"></i><b>17.5.5</b> Reduce models for the right reason</a></li>
<li class="chapter" data-level="17.5.6" data-path="factorial.html"><a href="factorial.html#the-marginal-means-of-an-additive-linear-model-with-two-factors-can-be-weird"><i class="fa fa-check"></i><b>17.5.6</b> The marginal means of an additive linear model with two factors can be weird</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="factorial.html"><a href="factorial.html#example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data"><i class="fa fa-check"></i><b>17.6</b> Example 3: Estimation of synergy (“More than the sum of the parts”) (Experiment 1c JA data)</a>
<ul>
<li class="chapter" data-level="17.6.1" data-path="factorial.html"><a href="factorial.html#examine-the-data"><i class="fa fa-check"></i><b>17.6.1</b> Examine the data</a></li>
<li class="chapter" data-level="17.6.2" data-path="factorial.html"><a href="factorial.html#fit-the-model-7"><i class="fa fa-check"></i><b>17.6.2</b> Fit the model</a></li>
<li class="chapter" data-level="17.6.3" data-path="factorial.html"><a href="factorial.html#model-check"><i class="fa fa-check"></i><b>17.6.3</b> Model check</a></li>
<li class="chapter" data-level="17.6.4" data-path="factorial.html"><a href="factorial.html#inference-from-the-model-3"><i class="fa fa-check"></i><b>17.6.4</b> Inference from the model</a></li>
<li class="chapter" data-level="17.6.5" data-path="factorial.html"><a href="factorial.html#plot-the-model-5"><i class="fa fa-check"></i><b>17.6.5</b> Plot the model</a></li>
<li class="chapter" data-level="17.6.6" data-path="factorial.html"><a href="factorial.html#alternative-plot"><i class="fa fa-check"></i><b>17.6.6</b> Alternative plot</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="factorial.html"><a href="factorial.html#understanding-the-linear-model-with-crossed-factors-3"><i class="fa fa-check"></i><b>17.7</b> Understanding the linear model with crossed factors 3</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="factorial.html"><a href="factorial.html#thinking-about-the-coefficients-of-the-linear-model"><i class="fa fa-check"></i><b>17.7.1</b> Thinking about the coefficients of the linear model</a></li>
</ul></li>
<li class="chapter" data-level="17.8" data-path="factorial.html"><a href="factorial.html#issues-in-inference"><i class="fa fa-check"></i><b>17.8</b> Issues in inference</a>
<ul>
<li class="chapter" data-level="17.8.1" data-path="factorial.html"><a href="factorial.html#for-pairwise-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model"><i class="fa fa-check"></i><b>17.8.1</b> For pairwise contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</a></li>
<li class="chapter" data-level="17.8.2" data-path="factorial.html"><a href="factorial.html#for-interaction-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model"><i class="fa fa-check"></i><b>17.8.2</b> For interaction contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</a></li>
<li class="chapter" data-level="17.8.3" data-path="factorial.html"><a href="factorial.html#twoway-multiple-tests"><i class="fa fa-check"></i><b>17.8.3</b> Adjusting <em>p</em>-values for multiple tests</a></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="factorial.html"><a href="factorial.html#two-way-anova"><i class="fa fa-check"></i><b>17.9</b> Two-way ANOVA</a>
<ul>
<li class="chapter" data-level="17.9.1" data-path="factorial.html"><a href="factorial.html#how-to-read-a-two-way-anova-table"><i class="fa fa-check"></i><b>17.9.1</b> How to read a two-way ANOVA table</a></li>
<li class="chapter" data-level="17.9.2" data-path="factorial.html"><a href="factorial.html#what-do-the-main-effects-in-an-anova-table-mean"><i class="fa fa-check"></i><b>17.9.2</b> What do the main effects in an ANOVA table mean?</a></li>
</ul></li>
<li class="chapter" data-level="17.10" data-path="factorial.html"><a href="factorial.html#more-issues-in-inference"><i class="fa fa-check"></i><b>17.10</b> More issues in inference</a>
<ul>
<li class="chapter" data-level="17.10.1" data-path="factorial.html"><a href="factorial.html#longitudinal-experiments-include-time-as-a-random-factor-better-than-repeated-measures-anova"><i class="fa fa-check"></i><b>17.10.1</b> Longitudinal experiments – include Time as a random factor (better than repeated measures ANOVA)</a></li>
</ul></li>
<li class="chapter" data-level="17.11" data-path="factorial.html"><a href="factorial.html#working-in-r-4"><i class="fa fa-check"></i><b>17.11</b> Working in R</a>
<ul>
<li class="chapter" data-level="17.11.1" data-path="factorial.html"><a href="factorial.html#model-formula"><i class="fa fa-check"></i><b>17.11.1</b> Model formula</a></li>
<li class="chapter" data-level="17.11.2" data-path="factorial.html"><a href="factorial.html#using-the-emmeans-function-2"><i class="fa fa-check"></i><b>17.11.2</b> Using the emmeans function</a></li>
<li class="chapter" data-level="17.11.3" data-path="factorial.html"><a href="factorial.html#contrasts"><i class="fa fa-check"></i><b>17.11.3</b> Contrasts</a></li>
<li class="chapter" data-level="17.11.4" data-path="factorial.html"><a href="factorial.html#practice-safe-anova"><i class="fa fa-check"></i><b>17.11.4</b> Practice safe ANOVA</a></li>
<li class="chapter" data-level="17.11.5" data-path="factorial.html"><a href="factorial.html#better-to-avoid-these"><i class="fa fa-check"></i><b>17.11.5</b> Better to avoid these</a></li>
</ul></li>
<li class="chapter" data-level="17.12" data-path="factorial.html"><a href="factorial.html#hidden-code-5"><i class="fa fa-check"></i><b>17.12</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="17.12.1" data-path="factorial.html"><a href="factorial.html#import-exp2j-example-1"><i class="fa fa-check"></i><b>17.12.1</b> Import exp2j (Example 1)</a></li>
<li class="chapter" data-level="17.12.2" data-path="factorial.html"><a href="factorial.html#import-exp3e-lesian-area-data-example-2"><i class="fa fa-check"></i><b>17.12.2</b> Import exp3e lesian area data (Example 2)</a></li>
<li class="chapter" data-level="17.12.3" data-path="factorial.html"><a href="factorial.html#import-exp1c-ja-data-example-3"><i class="fa fa-check"></i><b>17.12.3</b> Import Exp1c JA data (Example 3)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-vii-expanding-the-linear-model.html"><a href="part-vii-expanding-the-linear-model.html"><i class="fa fa-check"></i>Part VII – Expanding the Linear Model</a></li>
<li class="chapter" data-level="18" data-path="lmm.html"><a href="lmm.html"><i class="fa fa-check"></i><b>18</b> Models with random factors – linear mixed models</a>
<ul>
<li class="chapter" data-level="18.1" data-path="lmm.html"><a href="lmm.html#lmm-demo"><i class="fa fa-check"></i><b>18.1</b> Example 1 – A random intercepts and slopes explainer (demo1)</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="lmm.html"><a href="lmm.html#batched-measurements-result-in-clustered-residuals"><i class="fa fa-check"></i><b>18.1.1</b> Batched measurements result in clustered residuals</a></li>
<li class="chapter" data-level="18.1.2" data-path="lmm.html"><a href="lmm.html#lmm-corerr"><i class="fa fa-check"></i><b>18.1.2</b> Clustered residuals result in correlated error</a></li>
<li class="chapter" data-level="18.1.3" data-path="lmm.html"><a href="lmm.html#in-blocked-designs-clustered-residuals-adds-a-variance-component-that-masks-treatment-effects"><i class="fa fa-check"></i><b>18.1.3</b> In blocked designs, clustered residuals adds a variance component that masks treatment effects</a></li>
<li class="chapter" data-level="18.1.4" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-are-linear-models-with-added-random-factors"><i class="fa fa-check"></i><b>18.1.4</b> Linear mixed models are linear models with added random factors</a></li>
<li class="chapter" data-level="18.1.5" data-path="lmm.html"><a href="lmm.html#what-the-random-effects-are"><i class="fa fa-check"></i><b>18.1.5</b> What the random effects are</a></li>
<li class="chapter" data-level="18.1.6" data-path="lmm.html"><a href="lmm.html#in-a-blocked-design-a-linear-model-with-added-random-effects-increases-precision-of-treatment-effects"><i class="fa fa-check"></i><b>18.1.6</b> In a blocked design, a linear model with added random effects increases precision of treatment effects</a></li>
<li class="chapter" data-level="18.1.7" data-path="lmm.html"><a href="lmm.html#lmm-varcorr"><i class="fa fa-check"></i><b>18.1.7</b> The correlation among random intercepts and slopes</a></li>
<li class="chapter" data-level="18.1.8" data-path="lmm.html"><a href="lmm.html#clustered-residuals-create-heterogeneity-among-treatments"><i class="fa fa-check"></i><b>18.1.8</b> Clustered residuals create heterogeneity among treatments</a></li>
<li class="chapter" data-level="18.1.9" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-are-flexible"><i class="fa fa-check"></i><b>18.1.9</b> Linear mixed models are flexible</a></li>
<li class="chapter" data-level="18.1.10" data-path="lmm.html"><a href="lmm.html#a-random-intercept-only-model"><i class="fa fa-check"></i><b>18.1.10</b> A random intercept only model</a></li>
<li class="chapter" data-level="18.1.11" data-path="lmm.html"><a href="lmm.html#a-model-including-an-interaction-intercept"><i class="fa fa-check"></i><b>18.1.11</b> A model including an interaction intercept</a></li>
<li class="chapter" data-level="18.1.12" data-path="lmm.html"><a href="lmm.html#lmm-demo-aic"><i class="fa fa-check"></i><b>18.1.12</b> AIC and model selection – which model to report?</a></li>
<li class="chapter" data-level="18.1.13" data-path="lmm.html"><a href="lmm.html#the-specification-of-random-effects-matters"><i class="fa fa-check"></i><b>18.1.13</b> The specification of random effects matters</a></li>
<li class="chapter" data-level="18.1.14" data-path="lmm.html"><a href="lmm.html#mixed-effect-and-repeated-measures-anova"><i class="fa fa-check"></i><b>18.1.14</b> Mixed Effect and Repeated Measures ANOVA</a></li>
<li class="chapter" data-level="18.1.15" data-path="lmm.html"><a href="lmm.html#pseudoreplication"><i class="fa fa-check"></i><b>18.1.15</b> Pseudoreplication</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="lmm.html"><a href="lmm.html#lmm-example2"><i class="fa fa-check"></i><b>18.2</b> Example 2 – experiments without subsampling replication (exp6g)</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="lmm.html"><a href="lmm.html#understand-the-data"><i class="fa fa-check"></i><b>18.2.1</b> Understand the data</a></li>
<li class="chapter" data-level="18.2.2" data-path="lmm.html"><a href="lmm.html#model-fit-and-inference"><i class="fa fa-check"></i><b>18.2.2</b> Model fit and inference</a></li>
<li class="chapter" data-level="18.2.3" data-path="lmm.html"><a href="lmm.html#the-model-exp6g_m1-adds-a-random-intercept-but-not-a-random-slope"><i class="fa fa-check"></i><b>18.2.3</b> The model exp6g_m1 adds a random intercept but not a random slope</a></li>
<li class="chapter" data-level="18.2.4" data-path="lmm.html"><a href="lmm.html#the-fixed-effect-coefficients-of-model-exp6g_m1"><i class="fa fa-check"></i><b>18.2.4</b> The fixed effect coefficients of model exp6g_m1</a></li>
<li class="chapter" data-level="18.2.5" data-path="lmm.html"><a href="lmm.html#the-random-intercept-coefficients-of-exp6g_m1"><i class="fa fa-check"></i><b>18.2.5</b> The random intercept coefficients of exp6g_m1</a></li>
<li class="chapter" data-level="18.2.6" data-path="lmm.html"><a href="lmm.html#the-random-and-residual-variance-and-the-intraclass-correlation-of-model-exp6g_m1"><i class="fa fa-check"></i><b>18.2.6</b> The random and residual variance and the intraclass correlation of model exp6g_m1</a></li>
<li class="chapter" data-level="18.2.7" data-path="lmm.html"><a href="lmm.html#the-linear-mixed-model-exp6g_m1-increases-precision-of-treatment-effects-relative-to-a-fixed-effects-model"><i class="fa fa-check"></i><b>18.2.7</b> The linear mixed model exp6g_m1 increases precision of treatment effects, relative to a fixed effects model</a></li>
<li class="chapter" data-level="18.2.8" data-path="lmm.html"><a href="lmm.html#lmm-exp6g-alt"><i class="fa fa-check"></i><b>18.2.8</b> Alternative models for exp6g</a></li>
<li class="chapter" data-level="18.2.9" data-path="lmm.html"><a href="lmm.html#paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models"><i class="fa fa-check"></i><b>18.2.9</b> Paired t-tests and repeated measures ANOVA are special cases of linear mixed models</a></li>
<li class="chapter" data-level="18.2.10" data-path="lmm.html"><a href="lmm.html#lmm-exp6g-rmanova"><i class="fa fa-check"></i><b>18.2.10</b> Classical (“univariate model”) repeated measures ANOVA of exp6g</a></li>
<li class="chapter" data-level="18.2.11" data-path="lmm.html"><a href="lmm.html#lmm-exp6g-rmanova-multi"><i class="fa fa-check"></i><b>18.2.11</b> “Multivariate model” repeated measures ANOVA</a></li>
<li class="chapter" data-level="18.2.12" data-path="lmm.html"><a href="lmm.html#linear-mixed-models-vs-repeated-measures-anova"><i class="fa fa-check"></i><b>18.2.12</b> Linear mixed models vs repeated measures ANOVA</a></li>
<li class="chapter" data-level="18.2.13" data-path="lmm.html"><a href="lmm.html#modeling-textttmouse_id-as-a-fixed-effect"><i class="fa fa-check"></i><b>18.2.13</b> Modeling <span class="math inline">\(\texttt{mouse_id}\)</span> as a fixed effect</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="lmm.html"><a href="lmm.html#example-3-factorial-experiments-and-no-subsampling-replicates-exp5c"><i class="fa fa-check"></i><b>18.3</b> Example 3 – Factorial experiments and no subsampling replicates (exp5c)</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="lmm.html"><a href="lmm.html#understand-the-data-1"><i class="fa fa-check"></i><b>18.3.1</b> Understand the data</a></li>
<li class="chapter" data-level="18.3.2" data-path="lmm.html"><a href="lmm.html#examine-the-data-1"><i class="fa fa-check"></i><b>18.3.2</b> Examine the data</a></li>
<li class="chapter" data-level="18.3.3" data-path="lmm.html"><a href="lmm.html#lmm-exp5c-m1"><i class="fa fa-check"></i><b>18.3.3</b> Model fit and inference</a></li>
<li class="chapter" data-level="18.3.4" data-path="lmm.html"><a href="lmm.html#why-we-care-about-modeling-batch-in-exp5c"><i class="fa fa-check"></i><b>18.3.4</b> Why we care about modeling batch in exp5c</a></li>
<li class="chapter" data-level="18.3.5" data-path="lmm.html"><a href="lmm.html#the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts"><i class="fa fa-check"></i><b>18.3.5</b> The linear mixed model exp5c_m1 adds two random intercepts</a></li>
<li class="chapter" data-level="18.3.6" data-path="lmm.html"><a href="lmm.html#the-fixed-effect-coefficients-of-model-exp5c_m1"><i class="fa fa-check"></i><b>18.3.6</b> The fixed effect coefficients of model exp5c_m1</a></li>
<li class="chapter" data-level="18.3.7" data-path="lmm.html"><a href="lmm.html#the-random-effect-coefficients-of-model-exp5c_m1"><i class="fa fa-check"></i><b>18.3.7</b> The random effect coefficients of model exp5c_m1</a></li>
<li class="chapter" data-level="18.3.8" data-path="lmm.html"><a href="lmm.html#lmm-exp5c-alt"><i class="fa fa-check"></i><b>18.3.8</b> Alternative models for exp5c</a></li>
<li class="chapter" data-level="18.3.9" data-path="lmm.html"><a href="lmm.html#lmm-rmanova-5c"><i class="fa fa-check"></i><b>18.3.9</b> Classical (“univariate model”) repeated measures ANOVA</a></li>
<li class="chapter" data-level="18.3.10" data-path="lmm.html"><a href="lmm.html#lmm-rmanova-5c-multi"><i class="fa fa-check"></i><b>18.3.10</b> “Multivariate model” repeated measures ANOVA of exp5c</a></li>
<li class="chapter" data-level="18.3.11" data-path="lmm.html"><a href="lmm.html#modeling-textttdonor-as-a-fixed-effect"><i class="fa fa-check"></i><b>18.3.11</b> Modeling <span class="math inline">\(\texttt{donor}\)</span> as a fixed effect</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="lmm.html"><a href="lmm.html#lmm-example4"><i class="fa fa-check"></i><b>18.4</b> Example 4 – Experiments with subsampling replication (exp1g)</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="lmm.html"><a href="lmm.html#understand-the-data-2"><i class="fa fa-check"></i><b>18.4.1</b> Understand the data</a></li>
<li class="chapter" data-level="18.4.2" data-path="lmm.html"><a href="lmm.html#examine-the-data-2"><i class="fa fa-check"></i><b>18.4.2</b> Examine the data</a></li>
<li class="chapter" data-level="18.4.3" data-path="lmm.html"><a href="lmm.html#fit-the-model-9"><i class="fa fa-check"></i><b>18.4.3</b> Fit the model</a></li>
<li class="chapter" data-level="18.4.4" data-path="lmm.html"><a href="lmm.html#inference-from-the-model-6"><i class="fa fa-check"></i><b>18.4.4</b> Inference from the model</a></li>
<li class="chapter" data-level="18.4.5" data-path="lmm.html"><a href="lmm.html#plot-the-model-6"><i class="fa fa-check"></i><b>18.4.5</b> Plot the model</a></li>
<li class="chapter" data-level="18.4.6" data-path="lmm.html"><a href="lmm.html#alternaplot-the-model-2"><i class="fa fa-check"></i><b>18.4.6</b> Alternaplot the model</a></li>
<li class="chapter" data-level="18.4.7" data-path="lmm.html"><a href="lmm.html#understanding-the-alternative-models"><i class="fa fa-check"></i><b>18.4.7</b> Understanding the alternative models</a></li>
<li class="chapter" data-level="18.4.8" data-path="lmm.html"><a href="lmm.html#the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b"><i class="fa fa-check"></i><b>18.4.8</b> The VarCorr matrix of models exp1g_m1a and exp1g_m1b</a></li>
<li class="chapter" data-level="18.4.9" data-path="lmm.html"><a href="lmm.html#the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means"><i class="fa fa-check"></i><b>18.4.9</b> The linear mixed model has more precision and power than the fixed effect model of batch means</a></li>
<li class="chapter" data-level="18.4.10" data-path="lmm.html"><a href="lmm.html#fixed-effect-models-and-pseudoreplication"><i class="fa fa-check"></i><b>18.4.10</b> Fixed effect models and pseudoreplication</a></li>
<li class="chapter" data-level="18.4.11" data-path="lmm.html"><a href="lmm.html#mixed-effect-anova"><i class="fa fa-check"></i><b>18.4.11</b> Mixed-effect ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="lmm.html"><a href="lmm.html#working-in-r-5"><i class="fa fa-check"></i><b>18.5</b> Working in R</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="lmm.html"><a href="lmm.html#fitting-linear-mixed-models"><i class="fa fa-check"></i><b>18.5.1</b> Fitting linear mixed models</a></li>
<li class="chapter" data-level="18.5.2" data-path="lmm.html"><a href="lmm.html#plotting-models-fit-to-batched-data"><i class="fa fa-check"></i><b>18.5.2</b> Plotting models fit to batched data</a></li>
<li class="chapter" data-level="18.5.3" data-path="lmm.html"><a href="lmm.html#repeated-measures-anova-randomized-complete-block-with-no-subsampling"><i class="fa fa-check"></i><b>18.5.3</b> Repeated measures ANOVA (randomized complete block with no subsampling)</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="lmm.html"><a href="lmm.html#hidden-code-6"><i class="fa fa-check"></i><b>18.6</b> Hidden code</a>
<ul>
<li class="chapter" data-level="18.6.1" data-path="lmm.html"><a href="lmm.html#import-exp5c"><i class="fa fa-check"></i><b>18.6.1</b> Import exp5c</a></li>
<li class="chapter" data-level="18.6.2" data-path="lmm.html"><a href="lmm.html#import-exp1g"><i class="fa fa-check"></i><b>18.6.2</b> Import exp1g</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="pre-post.html"><a href="pre-post.html"><i class="fa fa-check"></i><b>19</b> Linear models for longitudinal experiments – I. pre-post designs</a>
<ul>
<li class="chapter" data-level="19.1" data-path="pre-post.html"><a href="pre-post.html#best-practice-models"><i class="fa fa-check"></i><b>19.1</b> Best practice models</a></li>
<li class="chapter" data-level="19.2" data-path="pre-post.html"><a href="pre-post.html#common-alternatives-that-are-not-recommended"><i class="fa fa-check"></i><b>19.2</b> Common alternatives that are not recommended</a></li>
<li class="chapter" data-level="19.3" data-path="pre-post.html"><a href="pre-post.html#advanced-models"><i class="fa fa-check"></i><b>19.3</b> Advanced models</a></li>
<li class="chapter" data-level="19.4" data-path="pre-post.html"><a href="pre-post.html#understanding-the-alternative-models-1"><i class="fa fa-check"></i><b>19.4</b> Understanding the alternative models</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="pre-post.html"><a href="pre-post.html#m1-linear-model-with-the-baseline-measure-as-the-covariate-ancova-model"><i class="fa fa-check"></i><b>19.4.1</b> (M1) Linear model with the baseline measure as the covariate (ANCOVA model)</a></li>
<li class="chapter" data-level="19.4.2" data-path="pre-post.html"><a href="pre-post.html#m2-linear-model-of-the-change-score-change-score-model"><i class="fa fa-check"></i><b>19.4.2</b> (M2) Linear model of the change score (change-score model)</a></li>
<li class="chapter" data-level="19.4.3" data-path="pre-post.html"><a href="pre-post.html#m3-linear-model-of-post-baseline-values-without-the-baseline-as-a-covariate-post-model"><i class="fa fa-check"></i><b>19.4.3</b> (M3) Linear model of post-baseline values without the baseline as a covariate (post model)</a></li>
<li class="chapter" data-level="19.4.4" data-path="pre-post.html"><a href="pre-post.html#m4-linear-model-with-factorial-fixed-effects-fixed-effects-model"><i class="fa fa-check"></i><b>19.4.4</b> (M4) Linear model with factorial fixed effects (fixed-effects model)</a></li>
<li class="chapter" data-level="19.4.5" data-path="pre-post.html"><a href="pre-post.html#m5-repeated-measures-anova"><i class="fa fa-check"></i><b>19.4.5</b> (M5) Repeated measures ANOVA</a></li>
<li class="chapter" data-level="19.4.6" data-path="pre-post.html"><a href="pre-post.html#m6-linear-mixed-model"><i class="fa fa-check"></i><b>19.4.6</b> (M6) Linear mixed model</a></li>
<li class="chapter" data-level="19.4.7" data-path="pre-post.html"><a href="pre-post.html#m7-linear-model-with-correlated-error"><i class="fa fa-check"></i><b>19.4.7</b> (M7) Linear model with correlated error</a></li>
<li class="chapter" data-level="19.4.8" data-path="pre-post.html"><a href="pre-post.html#m8-constrained-fixed-effects-model-with-correlated-error-clda-model"><i class="fa fa-check"></i><b>19.4.8</b> (M8) Constrained fixed effects model with correlated error (cLDA model)</a></li>
<li class="chapter" data-level="19.4.9" data-path="pre-post.html"><a href="pre-post.html#comparison-table"><i class="fa fa-check"></i><b>19.4.9</b> Comparison table</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="pre-post.html"><a href="pre-post.html#example-1-a-single-post-baseline-measure-pre-post-design"><i class="fa fa-check"></i><b>19.5</b> Example 1 – a single post-baseline measure (pre-post design)</a></li>
<li class="chapter" data-level="19.6" data-path="pre-post.html"><a href="pre-post.html#working-in-r-6"><i class="fa fa-check"></i><b>19.6</b> Working in R</a></li>
<li class="chapter" data-level="19.7" data-path="pre-post.html"><a href="pre-post.html#hidden-code-7"><i class="fa fa-check"></i><b>19.7</b> Hidden code</a>
<ul>
<li class="chapter" data-level="19.7.1" data-path="pre-post.html"><a href="pre-post.html#import-and-wrangle-mouse-sociability-data"><i class="fa fa-check"></i><b>19.7.1</b> Import and wrangle mouse sociability data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="counts.html"><a href="counts.html"><i class="fa fa-check"></i><b>20</b> Linear models for counts, binary responses, skewed responses, and ratios – Generalized Linear Models</a>
<ul>
<li class="chapter" data-level="20.1" data-path="counts.html"><a href="counts.html#glm-counts"><i class="fa fa-check"></i><b>20.1</b> Introducing Generalized Linear Models using count data examples</a>
<ul>
<li class="chapter" data-level="20.1.1" data-path="counts.html"><a href="counts.html#the-generalized-linear-model-glm"><i class="fa fa-check"></i><b>20.1.1</b> The Generalized Linear Model (GLM)</a></li>
<li class="chapter" data-level="20.1.2" data-path="counts.html"><a href="counts.html#kinds-of-data-that-are-modeled-by-a-glm"><i class="fa fa-check"></i><b>20.1.2</b> Kinds of data that are modeled by a GLM</a></li>
</ul></li>
<li class="chapter" data-level="20.2" data-path="counts.html"><a href="counts.html#glm-1"><i class="fa fa-check"></i><b>20.2</b> Example 1 – GLM models for count responses (“angiogenic sprouts” exp3a)</a>
<ul>
<li class="chapter" data-level="20.2.1" data-path="counts.html"><a href="counts.html#understand-the-data-3"><i class="fa fa-check"></i><b>20.2.1</b> Understand the data</a></li>
<li class="chapter" data-level="20.2.2" data-path="counts.html"><a href="counts.html#model-fit-and-inference-1"><i class="fa fa-check"></i><b>20.2.2</b> Model fit and inference</a></li>
</ul></li>
<li class="chapter" data-level="20.3" data-path="counts.html"><a href="counts.html#understanding-example-1"><i class="fa fa-check"></i><b>20.3</b> Understanding Example 1</a>
<ul>
<li class="chapter" data-level="20.3.1" data-path="counts.html"><a href="counts.html#modeling-strategy-1"><i class="fa fa-check"></i><b>20.3.1</b> Modeling strategy</a></li>
<li class="chapter" data-level="20.3.2" data-path="counts.html"><a href="counts.html#model-checking-fits-to-count-data"><i class="fa fa-check"></i><b>20.3.2</b> Model checking fits to count data</a></li>
<li class="chapter" data-level="20.3.3" data-path="counts.html"><a href="counts.html#biological-count-data-are-rarely-fit-well-by-a-poisson-glm.-instead-fit-a-quasi-poisson-or-negative-binomial-glm-model."><i class="fa fa-check"></i><b>20.3.3</b> Biological count data are rarely fit well by a Poisson GLM. Instead, fit a quasi-poisson or negative binomial GLM model.</a></li>
<li class="chapter" data-level="20.3.4" data-path="counts.html"><a href="counts.html#a-glm-is-a-linear-model-on-the-link-scale"><i class="fa fa-check"></i><b>20.3.4</b> A GLM is a linear model on the link scale</a></li>
<li class="chapter" data-level="20.3.5" data-path="counts.html"><a href="counts.html#coeffecients-of-a-generalized-linear-model-with-a-log-link-function-are-on-the-link-scale."><i class="fa fa-check"></i><b>20.3.5</b> Coeffecients of a Generalized Linear Model with a log-link function are on the link scale.</a></li>
<li class="chapter" data-level="20.3.6" data-path="counts.html"><a href="counts.html#modeled-means-in-the-emmeans-table-of-a-generalized-linear-model-can-be-on-the-link-scale-or-response-scale-report-the-response-scale"><i class="fa fa-check"></i><b>20.3.6</b> Modeled means in the emmeans table of a Generalized Linear Model can be on the link scale or response scale – Report the response scale</a></li>
<li class="chapter" data-level="20.3.7" data-path="counts.html"><a href="counts.html#some-consequences-of-fitting-a-linear-model-to-count-data"><i class="fa fa-check"></i><b>20.3.7</b> Some consequences of fitting a linear model to count data</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="counts.html"><a href="counts.html#example-2-use-a-glm-with-an-offset-instead-of-a-ratio-of-some-measurement-per-area-dna-damage-data-exp3b"><i class="fa fa-check"></i><b>20.4</b> Example 2 – Use a GLM with an offset instead of a ratio of some measurement per area (“dna damage” data exp3b)</a>
<ul>
<li class="chapter" data-level="20.4.1" data-path="counts.html"><a href="counts.html#exp3b-dna-damage-data"><i class="fa fa-check"></i><b>20.4.1</b> exp3b (“dna damage”) data</a></li>
<li class="chapter" data-level="20.4.2" data-path="counts.html"><a href="counts.html#understand-the-data-4"><i class="fa fa-check"></i><b>20.4.2</b> Understand the data</a></li>
<li class="chapter" data-level="20.4.3" data-path="counts.html"><a href="counts.html#model-fit-and-inference-2"><i class="fa fa-check"></i><b>20.4.3</b> Model fit and inference</a></li>
<li class="chapter" data-level="20.4.4" data-path="counts.html"><a href="counts.html#compare-an-offset-to-an-added-covariate-with-an-estimated-coefficient"><i class="fa fa-check"></i><b>20.4.4</b> Compare an offset to an added covariate with an estimated coefficient</a></li>
</ul></li>
<li class="chapter" data-level="20.5" data-path="counts.html"><a href="counts.html#example-3-glm-models-for-binary-responses"><i class="fa fa-check"></i><b>20.5</b> Example 3 – GLM models for binary responses</a></li>
<li class="chapter" data-level="20.6" data-path="counts.html"><a href="counts.html#working-in-r-7"><i class="fa fa-check"></i><b>20.6</b> Working in R</a>
<ul>
<li class="chapter" data-level="20.6.1" data-path="counts.html"><a href="counts.html#fitting-glms-to-count-data"><i class="fa fa-check"></i><b>20.6.1</b> Fitting GLMs to count data</a></li>
<li class="chapter" data-level="20.6.2" data-path="counts.html"><a href="counts.html#fitting-a-glm-to-a-continuous-conditional-response-with-right-skew."><i class="fa fa-check"></i><b>20.6.2</b> Fitting a GLM to a continuous conditional response with right skew.</a></li>
<li class="chapter" data-level="20.6.3" data-path="counts.html"><a href="counts.html#fitting-a-glm-to-a-binary-success-or-failure-presence-or-absence-survived-or-died-response"><i class="fa fa-check"></i><b>20.6.3</b> Fitting a GLM to a binary (success or failure, presence or absence, survived or died) response</a></li>
<li class="chapter" data-level="20.6.4" data-path="counts.html"><a href="counts.html#fitting-generalized-linear-mixed-models"><i class="fa fa-check"></i><b>20.6.4</b> Fitting Generalized Linear Mixed Models</a></li>
</ul></li>
<li class="chapter" data-level="20.7" data-path="counts.html"><a href="counts.html#model-checking-glms"><i class="fa fa-check"></i><b>20.7</b> Model checking GLMs</a></li>
<li class="chapter" data-level="20.8" data-path="counts.html"><a href="counts.html#hidden-code-8"><i class="fa fa-check"></i><b>20.8</b> Hidden code</a>
<ul>
<li class="chapter" data-level="20.8.1" data-path="counts.html"><a href="counts.html#import-example-1-data-exp3a-angiogenic-sprouts"><i class="fa fa-check"></i><b>20.8.1</b> Import Example 1 data (exp3a – “angiogenic sprouts”)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="gls.html"><a href="gls.html"><i class="fa fa-check"></i><b>21</b> Linear models with heterogenous variance</a></li>
<li class="chapter" data-level="22" data-path="count-sim.html"><a href="count-sim.html"><i class="fa fa-check"></i><b>22</b> Simulations – Count data (alternatives to a t-test)</a>
<ul>
<li class="chapter" data-level="22.1" data-path="count-sim.html"><a href="count-sim.html#use-data-similar-to-figure-6f-from-example-1"><i class="fa fa-check"></i><b>22.1</b> Use data similar to Figure 6f from Example 1</a></li>
<li class="chapter" data-level="22.2" data-path="count-sim.html"><a href="count-sim.html#functions"><i class="fa fa-check"></i><b>22.2</b> Functions</a></li>
<li class="chapter" data-level="22.3" data-path="count-sim.html"><a href="count-sim.html#simulations"><i class="fa fa-check"></i><b>22.3</b> Simulations</a>
<ul>
<li class="chapter" data-level="22.3.1" data-path="count-sim.html"><a href="count-sim.html#type-i-pseudo-normal-distribution"><i class="fa fa-check"></i><b>22.3.1</b> Type I, Pseudo-Normal distribution</a></li>
<li class="chapter" data-level="22.3.2" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-equal-n"><i class="fa fa-check"></i><b>22.3.2</b> Type I, neg binom, equal n</a></li>
<li class="chapter" data-level="22.3.3" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-equal-n-small-theta"><i class="fa fa-check"></i><b>22.3.3</b> Type I, neg binom, equal n, small theta</a></li>
<li class="chapter" data-level="22.3.4" data-path="count-sim.html"><a href="count-sim.html#type-i-neg-binom-unequal-n"><i class="fa fa-check"></i><b>22.3.4</b> Type I, neg binom, unequal n</a></li>
<li class="chapter" data-level="22.3.5" data-path="count-sim.html"><a href="count-sim.html#power-pseudo-normal-distribution-equal-n"><i class="fa fa-check"></i><b>22.3.5</b> Power, Pseudo-Normal distribution, equal n</a></li>
<li class="chapter" data-level="22.3.6" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-equal-n"><i class="fa fa-check"></i><b>22.3.6</b> Power, neg binom, equal n</a></li>
<li class="chapter" data-level="22.3.7" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-small-theta"><i class="fa fa-check"></i><b>22.3.7</b> Power, neg binom, small theta</a></li>
<li class="chapter" data-level="22.3.8" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-unequal-n"><i class="fa fa-check"></i><b>22.3.8</b> Power, neg binom, unequal n</a></li>
<li class="chapter" data-level="22.3.9" data-path="count-sim.html"><a href="count-sim.html#power-neg-binom-unequal-n-unequal-theta"><i class="fa fa-check"></i><b>22.3.9</b> Power, neg binom, unequal n, unequal theta</a></li>
<li class="chapter" data-level="22.3.10" data-path="count-sim.html"><a href="count-sim.html#type-1-neg-binom-equal-n-unequal-theta"><i class="fa fa-check"></i><b>22.3.10</b> Type 1, neg binom, equal n, unequal theta</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="count-sim.html"><a href="count-sim.html#save-it-read-it"><i class="fa fa-check"></i><b>22.4</b> Save it, Read it</a></li>
<li class="chapter" data-level="22.5" data-path="count-sim.html"><a href="count-sim.html#analysis"><i class="fa fa-check"></i><b>22.5</b> Analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html"><i class="fa fa-check"></i>Appendix 1: Getting Started with R</a>
<ul>
<li class="chapter" data-level="22.6" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#get-your-computer-ready"><i class="fa fa-check"></i><b>22.6</b> Get your computer ready</a>
<ul>
<li class="chapter" data-level="22.6.1" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#start-here"><i class="fa fa-check"></i><b>22.6.1</b> Start here</a></li>
<li class="chapter" data-level="22.6.2" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r"><i class="fa fa-check"></i><b>22.6.2</b> Install R</a></li>
<li class="chapter" data-level="22.6.3" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r-studio"><i class="fa fa-check"></i><b>22.6.3</b> Install R Studio</a></li>
<li class="chapter" data-level="22.6.4" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#install-r-markdown-1"><i class="fa fa-check"></i><b>22.6.4</b> Install R Markdown</a></li>
<li class="chapter" data-level="22.6.5" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#optional-alternative-latex-installations"><i class="fa fa-check"></i><b>22.6.5</b> (optional) Alternative LaTeX installations</a></li>
</ul></li>
<li class="chapter" data-level="22.7" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html#start-learning-r-studio"><i class="fa fa-check"></i><b>22.7</b> Start learning R Studio</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix-2-online-resources-for-getting-started-with-statistical-modeling-in-r.html"><a href="appendix-2-online-resources-for-getting-started-with-statistical-modeling-in-r.html"><i class="fa fa-check"></i>Appendix 2: Online Resources for Getting Started with Statistical Modeling in R</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Elements of Statistical Modeling for Experimental Biology</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="p-values" class="section level1" number="8">
<h1><span class="header-section-number">Chapter 8</span> P-values</h1>
<p><em>A general perception of a “replication crisis” may thus reflect failure to recognize that statistical tests not only test hypotheses, but countless assumptions and the entire environment in which research takes place. Because of all the uncertain and unknown assumptions that underpin statistical inferences, we should treat inferential statistics as highly unstable local descriptions of relations between assumptions and data, rather than as providing generalizable inferences about hypotheses or models. And that means we should treat statistical results as being much more incomplete and uncertain than is currently the norm.</em><a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a></p>
<p>A <em>p</em>-value <em>is a measure of the compatibility between observed data and the null model</em>. Here, “compatibility” is a probability, specifically, the probability of sampling a test-statistic as or more extreme than the observed test statistic, if all the assumptions used to compute the <em>p</em>-value are true.</p>
<p>To deconstruct what this means, and the implications of the meaning, let’s use the experiment from Figure 2i in the study on the browning of white adipose tissue in mice that was introduced in Chapter <a href="analyzing-experimental-data-with-a-linear-model.html#analyzing-experimental-data-with-a-linear-model">2</a> (<a href="analyzing-experimental-data-with-a-linear-model.html#analyzing-experimental-data-with-a-linear-model">Analyzing experimental data with a linear model</a>).</p>
<p>Data source: <a href="https://www.nature.com/articles/s41467-020-15483-7">ASK1 inhibits browning of white adipose tissue in obesity</a></p>
<p>A linear model with <code>liver_tg</code> as the response variable and <code>treatment</code> as the single <span class="math inline">\(X\)</span>-varaiable was fit to the data. Figure <a href="#p-value-fig2i-fig"><strong>??</strong></a> is the plot of the modeled means, 95% confidence intervals of the mean, and <em>p</em>-value of the significance test of the effect of treatment on liver triacylglycerol.</p>
<div class="figure"><span id="fig:p-value-fig2i-fig"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/p-value-fig2i-fig-1.png" alt="UCP1 expression, relative to the mean level in the control group. Mean (circle) and 95% confidence interval (line) are shown. Unadjusted p-values are from linear model with sh-RNA treatment and LPS treatment fully crossed." width="576" />
<p class="caption">
Figure 8.1: UCP1 expression, relative to the mean level in the control group. Mean (circle) and 95% confidence interval (line) are shown. Unadjusted p-values are from linear model with sh-RNA treatment and LPS treatment fully crossed.
</p>
</div>
<p>The coefficients of the model, and the standard error, 95% confidence interval, test-statistic, and <em>p</em>-value of each coefficient are shown in Table <a href="#tab:pvalue-coef-table"><strong>??</strong></a> Recall from Chapter <a href="analyzing-experimental-data-with-a-linear-model.html#analyzing-experimental-data-with-a-linear-model">2</a> that, with this model, the coefficient for the intercept term is the mean <code>liver_tg</code> for the control group, which is the estimate of the true mean for a mice with functional ASK1 protein. And the coefficient for the treatmentASK1Δadipo term is the difference in means between the knockout and control group, which is the estimate for the true effect of knocking out ASK1 in the adipose tissue. The <em>p</em>-value for this “effect” term is 0.012. How do we interpret this number?</p>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
61.5
</td>
<td style="text-align:right;">
4.98
</td>
<td style="text-align:right;">
12.3
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
50.4
</td>
<td style="text-align:right;">
72.6
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentASK1Δadipo
</td>
<td style="text-align:right;">
-21.6
</td>
<td style="text-align:right;">
7.05
</td>
<td style="text-align:right;">
-3.1
</td>
<td style="text-align:right;">
0.012
</td>
<td style="text-align:right;">
-37.3
</td>
<td style="text-align:right;">
-5.9
</td>
</tr>
</tbody>
</table>
<div id="a-p-value-is-the-probability-of-sampling-a-value-as-or-more-extreme-than-the-test-statistic-if-sampling-from-a-null-distribution" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> A <em>p</em>-value is the probability of sampling a value as or more extreme than the test statistic if sampling from a null distribution</h2>
<p>The test statistic in the table above is a <em>t</em>-value. For this specific model, this <em>t</em>-value is precisely the <em>t</em>-value one would get if they executed a classical Student <em>t</em>-test on the two groups of liver TG values. Importantly, this is not generally true. For many of the models in this text, a <em>t</em>-value is computed but this is <em>not</em> the <em>t</em>-value that would be computed in a classical <em>t</em>-test.</p>
<p>When we do a <em>t</em>-test, we get a <em>p</em>-value. The probability returned in a <em>t</em>-test is <span class="math inline">\(p = \mathrm{prob}(t \ge t_{obs} | H_0)\)</span>. Read this as “the p-value is the probability of observing a t-value that is greater than or equal to the observed t-value, given the null model is true.” Probability, in this text, is a long run frequency of sampling. The specific probability associated with the effect of treatment on liver TG is the long-run frequency of observing a <em>t</em>-value as big or bigger than the observed <em>t</em>-value (the one you actually got with your data) if the null is true. Let’s parse this into “long run frequency of observing a <em>t</em>-value as big or bigger than the observed <em>t</em>-value” and “null is true.”</p>
<p>A thought experiment: You open a google sheet and insert 12 standard, normal random deviates (so the true mean is zero and the true variance is one) in Column A, rows 1-12. You arbitrarily assign the first six values (rows 1-6) to treatment A and the second six values (rows 7-12) to treatment B. You use the space immediately below these data to compute the mean of treatment A, the mean of treatment B, the difference in means (A - B), and a <em>t</em>-value. Unfortunately, google sheets doesn’t have a <em>t</em>-value function so you’d have to compute this yourself. Or not, since this is a thought experiment. Now “fill right” or copy and paste these functions into 999 new columns. You now have 1000 <em>t</em>-tests. The expected value of the difference in means is zero (why?) but the actual values will form a normal distribution about zero. Most will be close to zero (either in the negative or positive direction) but some will be further from zero. The expected <em>t</em>-value will also be zero (why?) and the distribution of these 1000 <em>t</em>-values will look normal but the tails are a little fuller. This row of <em>t</em>-values is a null distribution, because in generating the data we used the exact same formula for the values assigned to A and the values assigned to B. Now think of a <em>t</em>-value in your head, say 0.72 (remember that <em>t</em>-values will largely range from about -3 to +3 although the theoretical range is <span class="math inline">\(-\infty\)</span> to <span class="math inline">\(+\infty\)</span>. What is the probability of observing a <em>t</em> of 0.72 <em>or bigger</em> if the null is true? Look at the row of <em>t</em>-values! Count the number of <span class="math inline">\(t \ge 0.72\)</span> and then divide by the total number of <em>t</em>-values in the row (1000) and you have a probability computed as a frequency. But remember the frequentist definition is the long run frequency, or the expected frequency at the limit (when you’ve generated not 1000 or even 1,000,000 but an infinite number of columns and <em>t</em>-values).</p>
<p>Some asides to the thought experiment: First, why “as big or bigger” and not just the probability of the value itself? The reason is that the probability of finding the exact <em>t</em> is 1/infinity, which doesn’t do us much good. So instead we compute the probability of finding <em>t</em> as big, or bigger, than our observed <em>t</em>. Second, the <em>t</em>-test probability described above is a “one-tail probability.” Because a difference can be both in the positive direction and the negative direction, we usually want to count all the <span class="math inline">\(t \ge 0.72\)</span> and the <span class="math inline">\(t \le -0.72\)</span> and then add these two counts to compute the frequency of <em>as extreme or more extreme</em> values. This is called a “two-tailed probability” because we find extremes at both tails of the distribution. Third, we don’t really count <span class="math inline">\(t \ge 0.72\)</span> but take advantage of the beautiful mathematical properties of the theoretical <em>t</em> distribution, which allows us to compute the frequentist probability (expected long range frequency) given the <em>t</em>-value and the degrees of freedom using the <em>t</em>-distribution.</p>
<p>Now what do I mean with the phrase “null is true?” Most people equate “null is true” with “no difference in means” but the phrase entails much more than this. Effectively, the phrase is short for “all assumptions used to compute” the <em>p</em>-value (see the Sander Greenland quote at the start of this chapter). A <em>p</em>-value is based on modeling the real data with a theoretical sample in which all the values were randomly sampled from the same distribution and the assignment of the individual values to treatment was random. Random sampling from the same distribution has three important consequences. First, random assignment to treatment group means that the expected means of each group are the same, or put differently, the expected difference in means between the assigned groups is zero. Second, random assignment to treatment <em>also</em> means that the expected variances of the two groups are equal. And third, random sampling means that the values of each point are independent – we cannot predict the value of one point knowing information about any other point. <strong>Here is what is super important about this</strong>: a low <em>p</em>-value is more likely if <em>any</em> one of these consequences is untrue about our data. A low <em>p</em>-value could arise from a difference in true means, or it could arise from a difference in true variances, or it could arise if the <span class="math inline">\(Y\)</span> values are not independent of each other. This is why we need certain assumptions to make a <em>p</em>-value meaningful for empirical data. By assuming independent error and homogenous (equal) variances in our two samples, a low <em>p</em>-value is evidence of unequal means.</p>
<p>Let’s summarize: A pretty good definition of a <em>p</em>-value is: the long-run frequency of observing a test-statistic as large or larger than the observed statistic, if the null were true. A more succinct way to state this is</p>
<p><span class="math display">\[\begin{equation}
p = \mathrm{prob}(t \ge t_o | H_o)
\end{equation}\]</span></p>
<p>where <em>t</em> is a hypothetically sampled <em>t</em>-value from a null distribution, <span class="math inline">\(t_o\)</span> is the observed <em>t</em>-value, and <span class="math inline">\(H_o\)</span> is the null hypothesis. Part of the null hypothesis is the expected value of the parameter estimated is usually (but not always) zero – this can be called the nil null. For example, if there is no ASK1 deletion effect on liver TG levels, then the expected difference between the means of the control and knockout mice is zero. Or,</p>
<p><span class="math display">\[\begin{equation}
\mathrm{E}(\bar{Y}_{knockout} - \bar{Y}_{control} | H_o) = 0.0
\end{equation}\]</span></p>
</div>
<div id="pump-your-intuition-creating-a-null-distribution" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Pump your intuition – Creating a null distribution</h2>
<p>The mean <code>liver_tg</code> in the knockout treatment is 21.6 µmol less than the mean <code>liver_tg</code> in the control treatment. This is the measured effect, or the <strong>observed differences in means</strong>. How confident are we in this effect? Certainly, if the researchers did the experiment with <em>two</em> control (ASK1F/F) treatment groups, they would measure some difference in their means simply because of finite sampling (more specifically, the many, many random effects that contribute to liver TG levels will differ between the two control groups). So let’s reframe the question: are the observed differences unusually large compared to a distribution of differences that would occur if there were no effect? That is, if the ``null were true’’. To answer this, we compare our observed difference to this <strong>null distribution</strong>. This comparison gives the probability (a long-run frequency) of “sampling” a random difference from the null distribution of differences that is as large, or larger, than the observed difference.</p>
<p>What is a null distribution? It is the distribution of a statistic (such as a difference in means, or better, a <em>t</em>-value) if the null were true. Here, I hope to pump your intuition by generating a null distribution that is relevant to the ASK1 liver TG data. See if you can understand the script before reading the explanation below.</p>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="p-values.html#cb205-1" aria-hidden="true" tabindex="-1"></a>seed <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb205-2"><a href="p-values.html#cb205-2" aria-hidden="true" tabindex="-1"></a>n_iter <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">5</span> <span class="co"># number of iterations</span></span>
<span id="cb205-3"><a href="p-values.html#cb205-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-4"><a href="p-values.html#cb205-4" aria-hidden="true" tabindex="-1"></a>mu <span class="ot">&lt;-</span> <span class="fu">mean</span>(fig_2i[treatment <span class="sc">==</span> <span class="st">&quot;ASK1F/F&quot;</span>, liver_tg]) </span>
<span id="cb205-5"><a href="p-values.html#cb205-5" aria-hidden="true" tabindex="-1"></a>sigma <span class="ot">&lt;-</span> <span class="fu">sd</span>(fig_2i[treatment <span class="sc">==</span> <span class="st">&quot;ASK1F/F&quot;</span>, liver_tg])</span>
<span id="cb205-6"><a href="p-values.html#cb205-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-7"><a href="p-values.html#cb205-7" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">nrow</span>((fig_2i[treatment <span class="sc">==</span> <span class="st">&quot;ASK1F/F&quot;</span>,]))</span>
<span id="cb205-8"><a href="p-values.html#cb205-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-9"><a href="p-values.html#cb205-9" aria-hidden="true" tabindex="-1"></a>sample1 <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">rnorm</span>(n<span class="sc">*</span>n_iter, <span class="at">mean=</span>mu, <span class="at">sd=</span>sigma), <span class="at">nrow=</span>n) <span class="co"># 100,000 samples (each size n)</span></span>
<span id="cb205-10"><a href="p-values.html#cb205-10" aria-hidden="true" tabindex="-1"></a>sample2 <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">rnorm</span>(n<span class="sc">*</span>n_iter, <span class="at">mean=</span>mu, <span class="at">sd=</span>sigma), <span class="at">nrow=</span>n) <span class="co"># 100,000 samples</span></span>
<span id="cb205-11"><a href="p-values.html#cb205-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-12"><a href="p-values.html#cb205-12" aria-hidden="true" tabindex="-1"></a>d_null <span class="ot">&lt;-</span> <span class="fu">apply</span>(sample2, <span class="dv">2</span>, mean) <span class="sc">-</span> <span class="fu">apply</span>(sample1, <span class="dv">2</span>, mean)</span>
<span id="cb205-13"><a href="p-values.html#cb205-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-14"><a href="p-values.html#cb205-14" aria-hidden="true" tabindex="-1"></a><span class="fu">qplot</span>(d_null)</span></code></pre></div>
<div class="figure"><span id="fig:pvalue-d-null"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/pvalue-d-null-1.png" alt="Null distribution for the difference in means of two samples from the same, inifinitely large population with a true mean and standard deviation equal to the observed mean and standard deviation of the ASK1 liver TG data." width="576" />
<p class="caption">
Figure 8.2: Null distribution for the difference in means of two samples from the same, inifinitely large population with a true mean and standard deviation equal to the observed mean and standard deviation of the ASK1 liver TG data.
</p>
</div>
<p>What have we done above? We’ve simulated an infinitely large population of mice that have a distribution of liver TG levels similar to that of the mice assigned to the control (ASK1F/F) group. The true mean (<span class="math inline">\(\mu\)</span>) and standard deviation (<span class="math inline">\(\sigma\)</span>) of the simulated TG level are equal to the observed mean and standard deviation of the TG levels of the control mice Then, the script:</p>
<ol style="list-style-type: decimal">
<li>randomly sample 6 values from this population of simulated lifespans and assign to sample1. We sample 6 values because that is the sample size of our control in the experiment.</li>
<li>randomly sample 6 values from this population of simulated lifespans and assign to sample2.</li>
<li>compute the difference <span class="math inline">\(\bar{Y}_{sample2} - \bar{Y}_{sample1}\)</span>.</li>
<li>repeat 1-3 100,000 times, each time saving the difference in means.</li>
<li>plot the distribution of the 100,000 differences using a histogram</li>
</ol>
<p>The distribution of the differences is a null distribution. Notice that the mode of the null distribution is at zero, and the mean (0.01399) is close to zero (if we had set <span class="math inline">\(n\)</span> to infinity, the mean would be precisely zero). <em>The expected difference between the means of two random samples from the same population is, of course, zero</em>. Don’t gloss over this statement if that is not obvious. The tails extend out to a little more than +20 and -20. What this means is that it would be uncommon to randomly sample a value from this distribution of differences <em>as or more extreme</em> than our observed difference, -21.6. By “more extreme,” I mean any value more negative than -21.6 or more postive than 21.6. So it would be uncommon to sample a value from this distribution whose <em>absolute value</em> is as or more extreme than 21.6. How uncommon would this be?</p>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb206-1"><a href="p-values.html#cb206-1" aria-hidden="true" tabindex="-1"></a>diff_obs <span class="ot">&lt;-</span> fig_2i_m1_coef[<span class="st">&quot;treatmentASK1Δadipo&quot;</span>, <span class="st">&quot;Estimate&quot;</span>]</span>
<span id="cb206-2"><a href="p-values.html#cb206-2" aria-hidden="true" tabindex="-1"></a>null_diff_extreme <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">abs</span>(d_null) <span class="sc">&gt;</span> <span class="fu">abs</span>(diff_obs))</span>
<span id="cb206-3"><a href="p-values.html#cb206-3" aria-hidden="true" tabindex="-1"></a>n_extreme <span class="ot">&lt;-</span> <span class="fu">length</span>(null_diff_extreme)</span>
<span id="cb206-4"><a href="p-values.html#cb206-4" aria-hidden="true" tabindex="-1"></a>(<span class="at">p_d_null =</span> n_extreme<span class="sc">/</span>n_iter)</span></code></pre></div>
<pre><code>## [1] 0.00551</code></pre>
<p>In the 100,000 runs, only 551 generated data with an absolute difference as large or larger than 21.6 (an “absolute difference” is the absoute value of the difference). The frequency of differences as large or larger than our observed difference is 0.00551. This frequency is the <em>probability</em> of sampling a difference as or more extreme than the observed difference “under the null.” It is a <em>p</em>-value, but it is not the <em>p</em>-value in the coefficient table. This is because the <em>p</em>-value in the coefficient table is computed from a distribution of <em>t</em>-values, not raw differences. This raises the question, what is a <em>t</em>-distribution, and a <em>t</em>-value, more generally?</p>
</div>
<div id="a-null-distribution-of-t-values-the-t-distribution" class="section level2" number="8.3">
<h2><span class="header-section-number">8.3</span> A null distribution of <em>t</em>-values – the <em>t</em> distribution</h2>
<p>A <em>t</em>-test is a test of differences between two values. These could be</p>
<ol style="list-style-type: decimal">
<li>the difference between the means of two samples (a “two-sample” <em>t</em>-test)</li>
<li>the difference between a mean of a sample and some pre-specified value (a “one-sample” <em>t</em>-test)</li>
<li>the difference between a coefficient from a linear model and zero</li>
</ol>
<p>A <em>t</em>-test compares an observed <em>t</em>-value to a <em>t</em>-distribution. The null distribution introduced above was a distribution of mean differences under the null. A distribution of mean differences under the null is very specific to the mean and standard deviation of the population modeled and the sample size of the experiment. This isn’t generally useful, since it will be unique to every study (at least it wasn’t generally useful prior to the time of fast computers. One could, and some statisticians do, compute <em>p</em>-values using the algorithm above). A <em>t</em>-distribution is a way of transforming a null distribution of mean differences, which is unique to the study, into a distribution that is a function of sample size only.</p>
<p>A <em>t</em>-distribution is a distribution of <em>t</em>-values under the null, where a <em>t</em>-value is a difference <em>standardized by its standard error</em>. For a two-sample <em>t</em>-test, this is</p>
<p><span class="math display" id="eq:t">\[\begin{equation}
t = \frac{\bar{y}_2 - \bar{y}_1}{SE_{\bar{y}_2 - \bar{y}_1}}
\tag{8.1}
\end{equation}\]</span></p>
<p>The numerator is <strong>the effect</strong> while the denominator is the precision of the estimate. Like many test statistics, a <em>t</em>-value is a signal-to-noise ratio – the effect is the signal and the SE of the difference is the noise.</p>
<p>A <em>t</em> distribution looks like a standard, normal distribution, except the tails are heavy, meaning there are more large-ish values than the normal. Like the standard normal distribution, large <em>t</em>-values are unlikely under the null and, therefore, a large <em>t</em> has a low probability – or <em>p</em>-value – under the null.</p>
<p>Looking at the equation for the two-sample <em>t</em>-test above, it is easy to see that three features of an experiment are associated with large <em>t</em> and small <em>p</em>-values: 1) big effect size (the numerator of the equation), 2) small sample standard deviations (which results in small standard errors of the difference, the denominator of equation <a href="p-values.html#eq:t">(8.1)</a>, and 3) large sample size (which results in small standard errors of the difference). As a quick-and-dirty generalization, absolute t-values greater than 3 are uncommon if the null is true.</p>
<p>The <em>p</em>-value for a <em>t</em>-test comes from comparing the observed <em>t</em> to a null <em>t</em> distribution and “counting” the values that are more extreme than the observed <em>t</em>. The <em>p</em>-value is the relative frequency of these more extreme values (relative to the total number of <em>t</em>-values in the distribution). I have “counting” in quotes because nothing is really counted – there are an infinite number of <em>t</em>-values in the <em>t</em>-distribution. Instead, the t-distribution function is integrated to compute the fraction of the total area under the curve with <em>t</em>-values more extreme than the observed value. In a <strong>two-tailed test</strong>, this fraction includes both tails (positive <em>t</em>-values more positive than <span class="math inline">\(|t|_{observed}\)</span> and negative <em>t</em>-values more negative than <span class="math inline">\(-|t|_{observed}\)</span>.</p>
<p>Let’s modify the simulation of a null distribution of mean differences to generate a null distribution of <em>t</em>-values. I show the script, but don’t just cut and paste the code. Spend time thinking about what the each line does. Explore it by copying parts and pasting into console.</p>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="p-values.html#cb208-1" aria-hidden="true" tabindex="-1"></a>seed <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb208-2"><a href="p-values.html#cb208-2" aria-hidden="true" tabindex="-1"></a>n_iter <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">5</span> <span class="co"># number of iterations</span></span>
<span id="cb208-3"><a href="p-values.html#cb208-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-4"><a href="p-values.html#cb208-4" aria-hidden="true" tabindex="-1"></a>mu <span class="ot">&lt;-</span> <span class="fu">mean</span>(fig_2i[treatment <span class="sc">==</span> <span class="st">&quot;ASK1F/F&quot;</span>, liver_tg]) </span>
<span id="cb208-5"><a href="p-values.html#cb208-5" aria-hidden="true" tabindex="-1"></a>sigma <span class="ot">&lt;-</span> <span class="fu">sd</span>(fig_2i[treatment <span class="sc">==</span> <span class="st">&quot;ASK1F/F&quot;</span>, liver_tg])</span>
<span id="cb208-6"><a href="p-values.html#cb208-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-7"><a href="p-values.html#cb208-7" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">nrow</span>((fig_2i[treatment <span class="sc">==</span> <span class="st">&quot;ASK1F/F&quot;</span>,]))</span>
<span id="cb208-8"><a href="p-values.html#cb208-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-9"><a href="p-values.html#cb208-9" aria-hidden="true" tabindex="-1"></a>sample1 <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">rnorm</span>(n<span class="sc">*</span>n_iter, <span class="at">mean=</span>mu, <span class="at">sd=</span>sigma), <span class="at">nrow=</span>n) <span class="co"># 100,000 samples (each size n)</span></span>
<span id="cb208-10"><a href="p-values.html#cb208-10" aria-hidden="true" tabindex="-1"></a>sample2 <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">rnorm</span>(n<span class="sc">*</span>n_iter, <span class="at">mean=</span>mu, <span class="at">sd=</span>sigma), <span class="at">nrow=</span>n) <span class="co"># 100,000 samples</span></span>
<span id="cb208-11"><a href="p-values.html#cb208-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-12"><a href="p-values.html#cb208-12" aria-hidden="true" tabindex="-1"></a><span class="co">#way no. 1 - compute the t-tests manually</span></span>
<span id="cb208-13"><a href="p-values.html#cb208-13" aria-hidden="true" tabindex="-1"></a>mean_diffs <span class="ot">&lt;-</span> <span class="fu">apply</span>(sample2, <span class="dv">2</span>, mean) <span class="sc">-</span> <span class="fu">apply</span>(sample1, <span class="dv">2</span>, mean) <span class="co"># what is the apply function returning?</span></span>
<span id="cb208-14"><a href="p-values.html#cb208-14" aria-hidden="true" tabindex="-1"></a>se_mean_diffs <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">apply</span>(sample2, <span class="dv">2</span>, sd)<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span>n <span class="sc">+</span> <span class="fu">apply</span>(sample1, <span class="dv">2</span>, sd)<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span>n)</span>
<span id="cb208-15"><a href="p-values.html#cb208-15" aria-hidden="true" tabindex="-1"></a>t_dis <span class="ot">&lt;-</span> mean_diffs<span class="sc">/</span>se_mean_diffs</span>
<span id="cb208-16"><a href="p-values.html#cb208-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-17"><a href="p-values.html#cb208-17" aria-hidden="true" tabindex="-1"></a><span class="co">#way no.2 - compute the t-tests using the linear model</span></span>
<span id="cb208-18"><a href="p-values.html#cb208-18" aria-hidden="true" tabindex="-1"></a>fake_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(sample1, sample2)</span>
<span id="cb208-19"><a href="p-values.html#cb208-19" aria-hidden="true" tabindex="-1"></a>treatment <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&quot;control&quot;</span>, <span class="st">&quot;knockout&quot;</span>), <span class="at">each =</span> n)</span>
<span id="cb208-20"><a href="p-values.html#cb208-20" aria-hidden="true" tabindex="-1"></a>t_dis2 <span class="ot">&lt;-</span> <span class="fu">numeric</span>(n_iter)</span>
<span id="cb208-21"><a href="p-values.html#cb208-21" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(iter <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_iter){</span>
<span id="cb208-22"><a href="p-values.html#cb208-22" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> fake_data[, iter]</span>
<span id="cb208-23"><a href="p-values.html#cb208-23" aria-hidden="true" tabindex="-1"></a>  fake_m1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> treatment)</span>
<span id="cb208-24"><a href="p-values.html#cb208-24" aria-hidden="true" tabindex="-1"></a>  t_dis2[iter] <span class="ot">&lt;-</span> <span class="fu">coef</span>(<span class="fu">summary</span>(fake_m1))[<span class="st">&quot;treatmentknockout&quot;</span>, <span class="st">&quot;t value&quot;</span>]</span>
<span id="cb208-25"><a href="p-values.html#cb208-25" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb208-26"><a href="p-values.html#cb208-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-27"><a href="p-values.html#cb208-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-28"><a href="p-values.html#cb208-28" aria-hidden="true" tabindex="-1"></a><span class="co"># plot the null distribution of t-values</span></span>
<span id="cb208-29"><a href="p-values.html#cb208-29" aria-hidden="true" tabindex="-1"></a><span class="fu">qplot</span>(t_dis2)</span></code></pre></div>
<div class="figure"><span id="fig:pvalue-t-null"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/pvalue-t-null-1.png" alt="Null distribution of t-values. The simulation generated 10,000 t-tests with a true null." width="576" />
<p class="caption">
Figure 8.3: Null distribution of t-values. The simulation generated 10,000 t-tests with a true null.
</p>
</div>
<p>Now let’s use this null distribution of <em>t</em>-values to compute a <em>p</em>-value</p>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="p-values.html#cb209-1" aria-hidden="true" tabindex="-1"></a><span class="co"># what is the p-value?</span></span>
<span id="cb209-2"><a href="p-values.html#cb209-2" aria-hidden="true" tabindex="-1"></a><span class="co"># the p-value is the number of t-values in t_dis that are as large</span></span>
<span id="cb209-3"><a href="p-values.html#cb209-3" aria-hidden="true" tabindex="-1"></a><span class="co"># or larger than the observed t. Large, negative t-values</span></span>
<span id="cb209-4"><a href="p-values.html#cb209-4" aria-hidden="true" tabindex="-1"></a><span class="co"># are as unlikely under the null as large, positive t-values.</span></span>
<span id="cb209-5"><a href="p-values.html#cb209-5" aria-hidden="true" tabindex="-1"></a><span class="co"># To account for this, we want to use absolute values in our counts</span></span>
<span id="cb209-6"><a href="p-values.html#cb209-6" aria-hidden="true" tabindex="-1"></a><span class="co"># this is a &quot;two-tail test&quot;</span></span>
<span id="cb209-7"><a href="p-values.html#cb209-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-8"><a href="p-values.html#cb209-8" aria-hidden="true" tabindex="-1"></a><span class="co"># first assign the observed t-value</span></span>
<span id="cb209-9"><a href="p-values.html#cb209-9" aria-hidden="true" tabindex="-1"></a>t_obs <span class="ot">&lt;-</span> fig_2i_m1_coef[<span class="st">&quot;treatmentASK1Δadipo&quot;</span>, <span class="st">&quot;t value&quot;</span>]</span>
<span id="cb209-10"><a href="p-values.html#cb209-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-11"><a href="p-values.html#cb209-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-12"><a href="p-values.html#cb209-12" aria-hidden="true" tabindex="-1"></a><span class="co"># now count the number of t-values in t_dis as big or bigger than this</span></span>
<span id="cb209-13"><a href="p-values.html#cb209-13" aria-hidden="true" tabindex="-1"></a><span class="co"># include the observed value as one of these (so add 1 to the count)</span></span>
<span id="cb209-14"><a href="p-values.html#cb209-14" aria-hidden="true" tabindex="-1"></a>count <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">abs</span>(t_dis) <span class="sc">&gt;=</span> <span class="fu">abs</span>(t_obs))</span>
<span id="cb209-15"><a href="p-values.html#cb209-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-16"><a href="p-values.html#cb209-16" aria-hidden="true" tabindex="-1"></a><span class="co"># the p-value is the frequency of t_dis &gt;= t_obs, so divide</span></span>
<span id="cb209-17"><a href="p-values.html#cb209-17" aria-hidden="true" tabindex="-1"></a><span class="co"># count by the total number of t-values in the distribution.</span></span>
<span id="cb209-18"><a href="p-values.html#cb209-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Again add one since the observed value counts as a sample</span></span>
<span id="cb209-19"><a href="p-values.html#cb209-19" aria-hidden="true" tabindex="-1"></a>(p_ASK1Δadipo <span class="ot">&lt;-</span> count<span class="sc">/</span>(n_iter))</span></code></pre></div>
<pre><code>## [1] 0.01199</code></pre>
<p>Hey that looks pretty good! Compare this to the <em>p</em>-value in the coefficient table above.</p>
<p>A <em>p</em>-value can be computed by counting the number of simulated <em>t</em>-values, <em>including the observed value</em>, that are equal to or more extreme (in either the positive or negative direction) than the observed <em>t</em>. Including the observed <em>t</em>, there are 1199 values that are more extreme than that observed. An approximate measure of <em>p</em> is this count divided by 100,001 (why is 1 added to the denominator?), which is 0.01199. This simulation-based <em>p</em>-value is very (very!) close to that computed from the observed <em>t</em>-test.</p>
</div>
<div id="p-values-from-the-perspective-of-permutation" class="section level2" number="8.4">
<h2><span class="header-section-number">8.4</span> P-values from the perspective of permutation</h2>
<p>A very intuitive way to think about <em>p</em>-values as a frequency is <strong>random permutation</strong>. A permutation is a re-arrangement of items. If there is an effect of ASK1 deletion on liver TG, then the arrangement of the values in the <code>treatment</code> column matters. If there is no effect of ASK1 deletion on liver TG, then the arrangement of the values in the <code>treatment</code> column does not matter.</p>
<p>Think about the structure of the liver TG data: there are two columns, <code>treatment</code>, which contains the assigned treatment, and <code>liver_tg</code>. The values in the <code>treatment</code> column were randomly assigned prior to the start of the experiment. If there is a negative effect of ASK1 deletion on liver TG, then assginment matters – the values in the <code>liver_tg</code> column for the ASK1Δadipo rows will be smaller than, on average, the values in the ASK1F/F rows. That is, a specific value of a<code>liver_tg</code> is what it is because of the value of <code>treatement</code> in the same row. Assignment to ASK1F/F or ASK1Δadipo changes the expected value of <code>liver_tg</code>. But, if there were no true effect, then assignment to ASK1F/F or ASK1Δadipo does not change the expected value of <code>liver_tg</code>. The expected value of every cell in the <code>liver_tg</code> column would be the same regardless of what is in the <code>treatment</code> column.</p>
<p>In our thought experiment, let’s leave the values in the <code>treatment</code> column be, and just randomly re-arrange or permute the values in the <code>liver_tg</code> column. What is the new expected diference in liver TG between the rows assigned to ASK1F/F and the rows assigned to ASK1Δadipo? The expected difference is Zero. Because the <code>liver_tg</code> values were randomly re-arranged, they <em>cannot</em> be caused by treatment assignment.</p>
<p>A permutation is a random re-arrangement of values in a column. Consider the many thousands of permutations of the values in the <code>liver_tg</code> column. A difference in means can be computed from each of these permuations and a distribution of differences can be generated. Is the observed difference extreme relative to the other values in this distribution? This is a permutation test – it compares an observed statistic to a distribution of the statistic computed over many thousands of permutations.</p>
<p>Let’s create a script for a permutation test</p>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb211-1"><a href="p-values.html#cb211-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb211-2"><a href="p-values.html#cb211-2" aria-hidden="true" tabindex="-1"></a>n_iter <span class="ot">&lt;-</span> <span class="dv">5000</span> <span class="co"># number of random permutations</span></span>
<span id="cb211-3"><a href="p-values.html#cb211-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-4"><a href="p-values.html#cb211-4" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> fig_2i[, liver_tg]</span>
<span id="cb211-5"><a href="p-values.html#cb211-5" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> fig_2i[, treatment]</span>
<span id="cb211-6"><a href="p-values.html#cb211-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-7"><a href="p-values.html#cb211-7" aria-hidden="true" tabindex="-1"></a>d_dist_perm <span class="ot">&lt;-</span> <span class="fu">numeric</span>(n_iter)</span>
<span id="cb211-8"><a href="p-values.html#cb211-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-9"><a href="p-values.html#cb211-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(iter <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_iter){</span>
<span id="cb211-10"><a href="p-values.html#cb211-10" aria-hidden="true" tabindex="-1"></a>  xbar1 <span class="ot">&lt;-</span> <span class="fu">mean</span>(y[x <span class="sc">==</span> <span class="st">&quot;ASK1F/F&quot;</span>])</span>
<span id="cb211-11"><a href="p-values.html#cb211-11" aria-hidden="true" tabindex="-1"></a>  xbar2 <span class="ot">&lt;-</span> <span class="fu">mean</span>(y[x <span class="sc">==</span> <span class="st">&quot;ASK1Δadipo&quot;</span>])</span>
<span id="cb211-12"><a href="p-values.html#cb211-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb211-13"><a href="p-values.html#cb211-13" aria-hidden="true" tabindex="-1"></a>  d_dist_perm[iter] <span class="ot">&lt;-</span> xbar2 <span class="sc">-</span> xbar1</span>
<span id="cb211-14"><a href="p-values.html#cb211-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb211-15"><a href="p-values.html#cb211-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># permute y</span></span>
<span id="cb211-16"><a href="p-values.html#cb211-16" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">sample</span>(y, <span class="at">replace=</span><span class="cn">FALSE</span>)</span>
<span id="cb211-17"><a href="p-values.html#cb211-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># note that, when i=1, the first &quot;permutation&quot; is the original arrangement</span></span>
<span id="cb211-18"><a href="p-values.html#cb211-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb211-19"><a href="p-values.html#cb211-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-20"><a href="p-values.html#cb211-20" aria-hidden="true" tabindex="-1"></a><span class="fu">qplot</span>(d_dist_perm)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/pvalue-d-dist-perm-1.png" width="576" /></p>
<p>From this distribution of distances generated by random permuation of the response, we can compute a permutation <em>p</em>-value.</p>
<p><code>{rpvalue-d-dist-perm-p } (p_permute &lt;- sum(abs(d_dist_perm) &gt;= abs(d_dist_perm[1]))/n_iter)</code></p>
</div>
<div id="parametric-vs.-non-parametric-statistics" class="section level2" number="8.5">
<h2><span class="header-section-number">8.5</span> Parametric vs. non-parametric statistics</h2>
<p>A statistic such as the difference in mean liver TG between ASK1Δadipo and ASK1F/F groups does not have “a” <em>p</em>-value. A <em>p</em>-value is the probability of observing an event <em>given a model of how the event was generated</em>. For the <em>p</em>-value in the coefficient table above, the event is sampling a <em>t</em>-value from a modeled <em>t</em> distribution that is as or more extreme than the observed <em>t</em>-value. The model generating the null distribution of <em>t</em>-values includes random sampling from a distribution that is defined by specific parameters (in this case, a mean and a variance), these parameters define the location and shape of the distribution of values that could be sampled. A <em>p</em>-value computed from a distribution that is defined by a set of parameters is a <strong>parametric</strong> <em>p</em>-value.</p>
<p>For the <em>p</em>-value computed using the permutation test, the event is the probability of of computing a difference of means <em>from a randomly permuted set of <span class="math inline">\(Y\)</span></em> as or more extreme than the observed difference of means. The distribution of differences from the permutated <span class="math inline">\(Y\)</span> data sets was not generated by any of the known distributions (normal, Poisson, binomial, etc.) given a specific value of parameters. Consequently, the permutation <em>p</em>-value is <strong>non-parametric</strong>.</p>
<p>The validity of all <em>p</em>-values depends on a set of model assumptions, which differ from model to model. The permutation <em>p</em>-value has fewer assumptions than a parametric <em>p</em>-value because no distribution is assumed (the permutation <em>p</em>-value is <strong>distribution-free</strong>).</p>
</div>
<div id="frequentist-probability-and-the-interpretation-of-p-values" class="section level2" number="8.6">
<h2><span class="header-section-number">8.6</span> frequentist probability and the interpretation of p-values</h2>
<div id="background" class="section level3" number="8.6.1">
<h3><span class="header-section-number">8.6.1</span> Background</h3>
<p>There are at least three different meanings of <strong>probability</strong>.</p>
<ol style="list-style-type: decimal">
<li><p><strong>subjective probability</strong> is the probability that an individual assigns to an event based on prior knowledge and the kinds of information considered reliable evidence. For example, if I asked a sample of students, what is the probability that a 30c homeopathic medicine could clear a <em>Streptococcus</em> infection from your respiratory system, their answers would differ because of variation in their knowledge of basic science, including chemistry and physics, their knowledge of what homeopathic medicines are, and how they weight different kinds of evidence.</p></li>
<li><p><strong>classical probability</strong> is simply one divided by the number of possible unique events. For example, with a six-sided die, there are six possible unique events. The probability of rolling a 2 is <span class="math inline">\(\frac{1}{6}\)</span> and the probability of rolling an odd number is <span class="math inline">\(\frac{1}{2}\)</span>.</p></li>
<li><p><strong>frequentist probability</strong> is based on the concept of <strong>long run frequency</strong>. If I roll a die 10 times, the frequency of rolling a 2 will be approximately <span class="math inline">\(\frac{1}{6}\)</span>. If I roll the die 100 times, the frequency of rolling a two will be closer to <span class="math inline">\(\frac{1}{6}\)</span>. If I roll the die 1000 times, the frequency of rolling the die will be even closer to <span class="math inline">\(\frac{1}{6}\)</span>. So the frequentist definition is the expected frequency given an infinite number of rolls. For events with continous outcomes, a frequentist probability is the long run frquency of <em>observing an outcome equal to or more extreme that that observed</em>.</p></li>
</ol>
</div>
<div id="this-book-covers-frequentist-approaches-to-statistical-modeling-and-when-a-probability-arises-such-as-the-p-value-of-a-test-statistic-this-will-be-a-frequentist-probability." class="section level3" number="8.6.2">
<h3><span class="header-section-number">8.6.2</span> This book covers frequentist approaches to statistical modeling and when a probability arises, such as the <em>p</em>-value of a test statistic, this will be a frequentist probability.</h3>
<p>When we do a <em>t</em>-test, we get a <em>p</em>-value. There are several ways to think about this probability. The most compact way is <span class="math inline">\(P(data | null)\)</span>, which is literally read as the probability of the data given the null (or “conditional” on the null), but is really short for <em>the probability of the data, or something more extreme than the data, given that the null hypothesis is true</em>. The “probability of the data” is kinda vague. More specifically, we mean the probability of some statistic about the data such as the difference in means between group A and group B or the <em>t</em>-value associated with this difference. So, a bit more formally, the probability returned in a <em>t</em>-test is <span class="math inline">\(\mathrm{prob}(t \ge t_{obs} | H_0)\)</span>. This is the long run frequency of observing a <em>t</em>-value as big or bigger than the observed <em>t</em>-value (the one you actually got with your data) if the null is true. Let’s parse this into “long run frequency of observing a <em>t</em>-value as big or bigger than the observed <em>t</em>-value” and “null is true.”</p>
<p>A thought experiment: You open a google sheet and insert 12 standard, normal random deviates (so the true mean is zero and the true variance is one) in Column A, rows 1-12. You arbitrarily assign the first six values (rows 1-6) to treatment A and the second six values (rows 7-12) to treatment B. You use the space immediately below these data to compute the mean of treatment A, the mean of treatment B, the difference in means (A - B), and a <em>t</em>-value. Unfortunately, google sheets doesn’t have a <em>t</em>-value function so you’d have to compute this yourself. Or not, since this is a thought experiment. Now ``fill right’’ or copy and paste these functions into 999 new columns. You now have 1000 <em>t</em> tests. The expected value of the difference in means is zero (why?) but the actual values will form a normal distribution about zero. Most will be close to zero (either in the negative or positive direction) but some will be further from zero. The expected <em>t</em>-value will also be zero (why?) and the distribution of these 1000 <em>t</em>-values will look normal but the tails are a little fuller. This row of <em>t</em>-values is a null distribution, because in generating the data we used the exact same formula for the values assigned to A and the values assigned to B. Now think of a <em>t</em>-value in your head, say 0.72 (remember that <em>t</em>-values will largely range from about -3 to +3 although the theoretical range is <span class="math inline">\(-\infty\)</span> to <span class="math inline">\(+\infty\)</span>. What is the probability of observing a <em>t</em> of 0.72 <em>or bigger</em> if the null is true? Look at the row of <em>t</em>-values! Count the number of <span class="math inline">\(t \ge 0.72\)</span> and then divide by the total number of <em>t</em>-values in the row (1000) and you have a probability computed as a frequency. But remember the frequentist definition is the long run frequency, or the expected frequency at the limit (when you’ve generated not 1000 or even 1,000,000 but an infinite number of columns and <em>t</em>-values).</p>
<p>Some asides to the thought experiment: First, why “as big or bigger” and not just the probability of the value itself? The reason is that the probability of finding the exact <em>t</em> is 1/infinity, which doesn’t do us much good. So instead we compute the probability of finding <em>t</em> as big, or bigger, than our observed <em>t</em>. Second, the <em>t</em>-test probability described above is a “one-tail probability.” Because a difference can be both in the positive direction and the negative direction, we usually want to count all the <span class="math inline">\(t \ge 0.72\)</span> and the <span class="math inline">\(t \le -0.72\)</span> and then add these two counts to compute the frequency of <em>as extreme or more extreme</em> values. This is called a “two-tailed probability” because we find extremes at both tails of the distribution. Third, we don’t really count <span class="math inline">\(t \ge 0.72\)</span> but take advantage of the beautiful mathematical properties of the theoretical <em>t</em> distribution, which allows us to compute the frequentist probability (expected long range frequency) given the <em>t</em>-value and the degrees of freedom using the <em>t</em>-distribution.</p>
<p>Now what do I mean with the phrase “null is true?” Most people equate “null is true” with ``no difference in means’’ but the phrase entails much more than this. Effectively, the phrase means that the <em>p</em>-value is based on modeling the real data with a theoretical sample in which all the points were randomly sampled from the same distribution and that the assignment of the individual points to treatment was random. This model means the theoretical sample has three properties: First, random assignment to treatment after sampling from the same distribution means that the expected means are the same, or put differently, the expected difference in means between the assigned groups is zero. Second, random assignment to treatment after sampling from the same distribution <em>also</em> means that the expected variances of the two groups are equal. And third, random sampling means that the values of each point are independent – we cannot predict the value of one point knowing information about any other point. <strong>Here is what is super important about this</strong>: if we get a really low <em>p</em>-value, any one of these consequences may be untrue about our data, for example it could be that the true means of the two treatment groups really are different, or it could mean it is the variances that differ between the two groups, or it could mean that the data (or technically, the errors) are not independent of each other. This is why we need certain assumptions to make a <em>p</em>-value meaningful for empirical data. By assuming independent error and homogenous (equal) variances in our two samples, a low <em>p</em>-value is evidence of unequal means.</p>
</div>
<div id="two-interpretations-of-the-p-value" class="section level3" number="8.6.3">
<h3><span class="header-section-number">8.6.3</span> Two interpretations of the <em>p</em>-value</h3>
<p>Since we want to be working scientists who want to use <em>p</em>-values as a tool, we need to know how to interpret (or use) the <em>p</em>-value to make reasonable inferences and how to avoid mis-interpreting the <em>p</em>-value and making unreasonable or even incorrect inferences. Ronald Fisher, the inventor of frequentist statistics, developed an interpretation of the <em>p</em>-value that is probably most useful for academic and applied research programs. Neyman and Pearson (Neyman-Pearson) gave the <em>p</em>-value a different interpretation, one that is probably most useful for industrial quality control. Today’s biology researchers use an interpretation that is an odd hybrid of the two, which often leads to silly inference. Regardless, understanding the distinction between Fisher and Neyman-Pearson will inform how we write up our results in a manuscript. I’ll describe these in the context of the two-sample <em>t</em>-test.</p>
<div id="fishers-interpretation" class="section level4" number="8.6.3.1">
<h4><span class="header-section-number">8.6.3.1</span> Fisher’s interpretation</h4>
<p>Fisher was working in the context of an agricultural experiments, the goal of which was to discover better agricultural practices. Does this new fertilizer work better than our old fertilizer? This is the context of much of modern biosciences and clinical medicine. Fisher thought of <em>p</em> as evidence against the null; the smaller the <em>p</em> the stronger the evidence that the two sampling distributions differ, which, in an experimental context, implies a treatment effect. If an experiment results in a large <em>p</em>-value, we can move on and test other fertilizers. If an experiment results in a small <em>p</em>-value, we want to pursue this new fertilizer more. Do more experiments! Fisher never thought of a single experiment as definitive. The decision to move on or pursue is only partly informed by the <em>p</em>-value and Fisher offered no rule about what <em>p</em>-value lies on the threshold of this decision. When pressed, Fisher might say that <span class="math inline">\(p=0.05\)</span> is a reasonable threshold.</p>
</div>
<div id="neyman-pearson-interpretation" class="section level4" number="8.6.3.2">
<h4><span class="header-section-number">8.6.3.2</span> Neyman-Pearson interpretation</h4>
<p>Neyman-Pearson thought of <em>p</em> as the necessary and sufficient information <em>to make a decision</em> between accepting the null (or at least not rejecting the null) or rejecting the null and accepting an alternative hypothesis. This decision balances two sorts of errors: Type I (false positives), which they called <span class="math inline">\(\alpha\)</span>, and Type II (false negatives), which they called <span class="math inline">\(\beta\)</span>. A false positive means the null was rejected but there really is no effect. A false negative means that the null was not rejected but there actually is an effect. <span class="math inline">\(\alpha\)</span> is set by the experimenter and is the long-term frequency (or “rate”) of false positives <strong>when the null is true</strong> that the experimenters are willing to accept. This is easily understood in the context of manufacturing. I’ve just made a batch of beer that I now need to ship. I sample 10 cans and test the quality against a norm. If <span class="math inline">\(p &lt; \alpha\)</span>, we reject the null in favor of the alternative – something may be wrong with the batch, it differs from the norm. We throw the beer away. If <span class="math inline">\(p &gt; \alpha\)</span>, we do not reject the null, nor the beer! We ship it.</p>
<p>After setting <span class="math inline">\(\alpha\)</span>, the experimenter designs the experiment to achieve an acceptable rate of <span class="math inline">\(\beta\)</span>. Since <span class="math inline">\(\beta\)</span> is the false negative rate then <span class="math inline">\(1-\beta\)</span> is the rate of not making a false negative error, that is, the rate of rejecting the null when there really is an effect. This is called the <strong>power</strong> of the experiment. An experiment with high power will have a low probability of a Type II error. An experiment with low power will have a high probability of a Type II error. Power is partly determined by sample size, the bigger the sample the smaller the <em>p</em>-value, all other things equal (think about why in the context of the formula for the <em>t</em>-value). Power is a function of error variance, both the natural variance and the component added because of measurement error (think about why in the context of the formula for the <em>t</em>-value). Power is also a function of <span class="math inline">\(\alpha\)</span>. If we set a low <span class="math inline">\(\alpha\)</span> (say, <span class="math inline">\(\alpha=0.01\)</span>), the test is conservative. We are more likely to fail to reject the null even if the null is false. A researcher can increase power by increasing sample size, using clever strategies to reduce measurement error, or increasing alpha.</p>
<p>An experimenter sets <span class="math inline">\(\alpha\)</span>, computes the sample size needed to achieve a certain level of power (<span class="math inline">\(1-\beta\)</span>), and then does the experiment. A thoughtful researcher will set <span class="math inline">\(\alpha\)</span> after considering and weighing the pros and cons of different levels of <span class="math inline">\(\alpha\)</span>. If false positives have costly consequences (expense, time, deleterious side-effects), then set <span class="math inline">\(\alpha\)</span> to a low value, such as 0.01 or 0.001. For example, if an initial screen has identified a previously unknown candidate that potentially functions in the focal system of the researcher, then a researcher might decide to set a low <span class="math inline">\(\alpha\)</span> (0.001) in the initial tests of this candidate to avoid devoting time, personel, and expense to chasing a phantom (a false-positive candidate). If false positives have trivial consequences, then set <span class="math inline">\(\alpha\)</span> to a high value, such as 0.05, or 0.1, or even 0.2. For example, if the initial tests of a candidate in a functional system are cheap and fast to construct, then a researcher might choose to set a high <span class="math inline">\(\alpha\)</span> for the screen that identifies candidates. False positive candidates don’t cost the lab much effort to identify them as false, but missing positive candidates because of a small <span class="math inline">\(\alpha\)</span> (which results in low power) at the screen stage costs the researcher the discovery of a potentially exciting coponent of the functional system.</p>
<p>In Fisher’s interpretation, there is no <span class="math inline">\(\alpha\)</span>, no <span class="math inline">\(\beta\)</span>, no alternative hypothesis, and no sharp decision rule. Instead, in Fisher, <em>p</em> is a continuous measure of evidence against the null and its value is interpreted subjectively by an informed and knowledgeable expert using additional information to make decisions. Neyman-Pearson rejected Fisher’s conception of <em>p</em> as evidence against the null arguing that a single experimental <em>p</em>-value is too noisy without embedding it into a more formal system of of decision making that maintains long-term type I error rates at <span class="math inline">\(\alpha\)</span>, given a certain power. In Neyman-Pearson, <em>p</em> is compared to a threshold, <span class="math inline">\(\alpha\)</span> and this alone makes the decision. In Neyman-Pearson, <em>p</em> <strong>is not treated as continuous information</strong>. <span class="math inline">\(p=0.00000001\)</span> is no more evidence to use to reject the null than <span class="math inline">\(p=0.049\)</span>.</p>
</div>
</div>
<div id="nhst" class="section level3" number="8.6.4">
<h3><span class="header-section-number">8.6.4</span> NHST</h3>
<p>Most biology researchers today interpret <em>p</em> using a combination of Fisher and Neyman-Pearson concepts in what has become known as Null Hypothesis Significance Testing (NHST).</p>
<ol style="list-style-type: decimal">
<li>Nearly all papers in biology either explicitly state something like “P values &lt; 0.05 were considered to be statistically significant” or implicitly use 0.05 as the “level of significance” (<span class="math inline">\(\alpha\)</span>). Comparing a <em>p</em>-value to a pre-defined <span class="math inline">\(\alpha\)</span> is Neyman-Pearson.</li>
<li>Unlike Neyman-Pearson, there is no evidence that researchers are thoughtfully considering the level of <span class="math inline">\(\alpha\)</span> for each experiment. Instead, researchers mindlessly choose <span class="math inline">\(\alpha=0.05\)</span> because this is what everyone else uses.</li>
<li>Unlike Neyman-Pearson, but somewhat in the spirit of Fisher, researchers, journals, and textbooks, advocate trichotomizing a statistically significant <em>p</em> into “significance bins” – three asterisks for <span class="math inline">\(p &lt; 0.001\)</span>, two asterisks for <span class="math inline">\(0.001 &lt; p &lt; 0.01\)</span>, and one asterisk for <span class="math inline">\(0.01 &lt; p &lt; 0.05\)</span>). This is not Neyman-Pearson. Again, Neyman-Pearson developed a system to control the long-run frequency of Type I error, which is controled by a strict use of <span class="math inline">\(\alpha\)</span>. If an observed <em>p</em>-value is in the *** bin or the * bin is meaningless in a system using Neyman-Pearson. There is only “accept” (<span class="math inline">\(p &gt;= \alpha\)</span>) or “reject” (<span class="math inline">\(p &lt; \alpha\)</span>).</li>
<li>Many researchers report exact <em>p</em>-values when <span class="math inline">\(p &lt; 0.05\)</span> but “n.s.” (not significant) when <span class="math inline">\(p &gt; 0.05\)</span>. Reporting exact <em>p</em>-values is Fisher. Reporting n.s. is Neyman-Pearson.</li>
<li>Many researchers further polychomotomize the <em>p</em>-value space just above 0.05 by using language such as “marginally significant.” If Neyman-Pearson and Fisher got together and spawned a love-child, this would be it.</li>
</ol>
</div>
</div>
<div id="some-major-misconceptions-of-the-p-value" class="section level2" number="8.7">
<h2><span class="header-section-number">8.7</span> Some major misconceptions of the <em>p</em>-value</h2>
<p>Setting the type I error rate <span class="math inline">\(\alpha\)</span> to 0.05 is so pervasive that I’m going to simply use “0.05” instead of “alpha” in discussing misconceptions.</p>
<div id="misconception-p-is-the-probability-that-the-null-is-true-and-1-p-is-probability-that-the-alternative-is-true" class="section level3" number="8.7.1">
<h3><span class="header-section-number">8.7.1</span> Misconception: <em>p</em> is the probability that the null is true <em>and</em> <span class="math inline">\(1-p\)</span> is probability that the alternative is true</h3>
<p>Many researchers believe that if <span class="math inline">\(p &gt; 0.05\)</span> then “there is no effect.” A frequentist hypothesis test cannot show that an effect doesn’t exist, only that the null has a low probablity of producing a test statistic as extreme or more extreme than the observed effect.</p>
<p>Many researchers believe that if <span class="math inline">\(p &lt; 0.05\)</span> then “there is an effect.” Again, a frequentist hypothesis test cannot show that an effect exists, only that the null has a low probablity of producing a test statistic as extreme or more extreme than the observed effect.</p>
<ol style="list-style-type: decimal">
<li>The statement “There is no effect of predators on feeding behavior” is not a valid conclusion of a frequentist hypothesis test.</li>
<li>The statement “We found no effect of predators on feeding behavior” is misleading because a frequentist hypothesis test can neither find an effect nor find no effect.</li>
</ol>
<p>The two errors above are gross misconceptions that are pervasive in the biology literature. A more subtle issue is the belief that a low <em>p</em>-value shows that the researcher’s explanatory hypothesis is correct. For example, researchers believe the result “the prey fish fed 14.2 (95% CI: 9.2, 19.2) minutes shorter in the presence of the predator fish” confirms their hypothesis that prey modulate feeding duration as a function of their ability to assess the risk of predation. Some alternative explanations:</p>
<ol style="list-style-type: decimal">
<li>The predator fish also competes with the prey fish for the prey fish’s food and with less food the prey fish spends less time feeding because it gives up when food density drops below some amount.</li>
<li>The predator fish is introduced to the prey tank by hand and odorant molecules from the researcher’s hands are detected by the prey and the prey reduces feeding duration because of these odorants.</li>
</ol>
<p>Importantly, no single experiment confirms an explanatory hypothesis. Instead, alternative explanations require multiple experiments with different controls to “rigrously probe” the preferred hypothesis.</p>
</div>
<div id="misconception-a-p-value-is-repeatable" class="section level3" number="8.7.2">
<h3><span class="header-section-number">8.7.2</span> Misconception: a <em>p</em>-value is repeatable</h3>
<p>Many researchers believe that a <em>p</em>-value is a precise measure – that if the experiment were replicated, a similar <em>p</em> would result. This belief requires at least two misconceptions. First, if the null were true, then <em>any</em> <em>p</em>-value is equally likely. <span class="math inline">\(p=0.00137\)</span> is just as likely as <span class="math inline">\(p=0.492\)</span>. In other words, if the null were true, the <em>p</em>-value is not replicable at all! Second, the <em>p</em>-value is highly dependent on the sample, and can be highly variable among replications, but there is no true <em>p</em>-value, so there can be no estimate or standard error. Let’s explore these.</p>
<div id="what-is-the-distribution-of-p-values-under-the-null" class="section level4" number="8.7.2.1">
<h4><span class="header-section-number">8.7.2.1</span> What is the distribution of <em>p</em>-values under the null?</h4>
<p>I often ask students, “if the null were true, what is the most likely <em>p</em>-value?” or “if the null were true, what kind of <em>p</em>-values would we expect, that is what is the expected distribution.” A common answer is <span class="math inline">\(p=0.5\)</span> is the most likely value and something like a normal curve, except the tails abruptly stop at 0 and 1, is the expected distribution.</p>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/uniform-1.png" width="576" /></p>
</div>
<div id="the-incredible-inconsistency-of-the-p-value" class="section level4" number="8.7.2.2">
<h4><span class="header-section-number">8.7.2.2</span> The incredible inconsistency of the <em>p</em>-value</h4>
<p>How replicable is the conclusion of an experiment if the <em>p</em>-value for a <em>t</em>-test is 0.03? If our conclusion is based on <span class="math inline">\(p &lt; 0.05\)</span>, then the conclusion is not very replicable. The simulation below shows the results of 15 replicates of an experiment with true power of 40%. There are five “significant” results (one less than expected) but several replicates have very high <em>p</em>-values.</p>
<div class="figure"><span id="fig:dance-fig"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/dance-fig-1.png" alt="Variability of *p*-values when the power is 0.4" width="576" />
<p class="caption">
Figure 8.4: Variability of <em>p</em>-values when the power is 0.4
</p>
</div>
</div>
</div>
<div id="misconception-0.05-is-the-lifetime-rate-of-false-discoveries" class="section level3" number="8.7.3">
<h3><span class="header-section-number">8.7.3</span> Misconception: 0.05 is the lifetime rate of false discoveries</h3>
<p>An important and widespread misconception is that if a researcher consistently uses <span class="math inline">\(\alpha=0.05\)</span>, then the frequency of incorrectly concluding an effect exists, or “discovering” an effect, over the lifetime of the researcher, will be 5%. This is incorrect. <span class="math inline">\(\alpha\)</span> is the rate of false positive if the null hypothesis is true, so our lifetime “false discovery” rate could only be 5% if everything we ever tested has no true effect! More generally, the <strong>false discovery rate</strong> is the frequency of false positives divided by the frequency of positives (the sum of false and true positives). This differs from the the Type I error rate, which is the frequency of false positives divided by the frequency of tests <em>in which the null is true</em>.</p>
<p>Imagine we test</p>
<ol style="list-style-type: decimal">
<li>1000 null hypotheses over a lifetime</li>
<li>60% are true nulls, this means there are 600 true nulls and 400 true effects</li>
<li>alpha is 5%. This means we expect to find <span class="math inline">\(p \le 0.05\)</span> 30 times (<span class="math inline">\(0.05 \times 600\)</span>) when the null is true</li>
<li>power is 25%. This means we expect to find <span class="math inline">\(p \le 0.05\)</span> 100 times (<span class="math inline">\(0.25 \times 400\)</span>) when the null is false</li>
<li>We have made <span class="math inline">\(30 + 100=130\)</span> “discoveries” (all experiments with <span class="math inline">\(p \le 0.05\)</span>), but</li>
<li>30 of the 130, or 23%, are “false discoveries.” This is the false discovery rate.</li>
</ol>
<p>Think about this. If the null is never true, you cannot have a false discovery–every <span class="math inline">\(p \le 0.05\)</span> is a true discovery (the false discovery rate is 0%). And if the null is always true, every <span class="math inline">\(p &lt; 0.05\)</span> is a false discovery (the false discovery rate is 100%).</p>
</div>
<div id="misconception-a-low-p-value-indicates-an-important-effect" class="section level3" number="8.7.4">
<h3><span class="header-section-number">8.7.4</span> Misconception: a low <em>p</em>-value indicates an important effect</h3>
<p>Many researchers write results as if they believe that a small <em>p</em>-value means the effect is big or important. This may misconception may arise because of the ubiquitous use of “significant” to indicate a small p-value and “very” or “extremely” or “wicked” significant to indicate a really small p-value. Regardless, this is a misconception. A small p-value will usually result when there is high power (but can occur even if power is low) and power is a function of effect size, variability (the standard deviation), and sample size. A small <em>p</em> could result from a large effect size but can also result with a small effect size if the sample size is big enough.</p>
<p>This is easy to simulate (see script below). Let’s model the effect of the genotype of a gene on height</p>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="p-values.html#cb212-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb212-2"><a href="p-values.html#cb212-2" aria-hidden="true" tabindex="-1"></a>rho <span class="ot">&lt;-</span> <span class="fl">0.5</span></span>
<span id="cb212-3"><a href="p-values.html#cb212-3" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">4</span></span>
<span id="cb212-4"><a href="p-values.html#cb212-4" aria-hidden="true" tabindex="-1"></a>genotype <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;+/+&quot;</span>, <span class="st">&quot;+/-&quot;</span>, <span class="st">&quot;-/-&quot;</span>)</span>
<span id="cb212-5"><a href="p-values.html#cb212-5" aria-hidden="true" tabindex="-1"></a>Sigma <span class="ot">&lt;-</span> <span class="fu">diag</span>(<span class="dv">2</span>)</span>
<span id="cb212-6"><a href="p-values.html#cb212-6" aria-hidden="true" tabindex="-1"></a>Sigma[<span class="dv">1</span>,<span class="dv">2</span>] <span class="ot">&lt;-</span> Sigma[<span class="dv">2</span>,<span class="dv">1</span>] <span class="ot">&lt;-</span> rho</span>
<span id="cb212-7"><a href="p-values.html#cb212-7" aria-hidden="true" tabindex="-1"></a>X <span class="ot">&lt;-</span> <span class="fu">rmvnorm</span>(n, <span class="at">mean=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">sigma=</span>Sigma)</span>
<span id="cb212-8"><a href="p-values.html#cb212-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(X) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;X1&quot;</span>, <span class="st">&quot;X2&quot;</span>)</span>
<span id="cb212-9"><a href="p-values.html#cb212-9" aria-hidden="true" tabindex="-1"></a>beta <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.05</span>)</span>
<span id="cb212-10"><a href="p-values.html#cb212-10" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> X<span class="sc">%*%</span>beta <span class="sc">+</span> <span class="fu">rnorm</span>(n)</span>
<span id="cb212-11"><a href="p-values.html#cb212-11" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> X)</span>
<span id="cb212-12"><a href="p-values.html#cb212-12" aria-hidden="true" tabindex="-1"></a><span class="fu">coefficients</span>(<span class="fu">summary</span>(fit))</span></code></pre></div>
<pre><code>##                Estimate Std. Error   t value     Pr(&gt;|t|)
## (Intercept) 0.007472959 0.01007946 0.7414046 4.584656e-01
## XX1         0.044304824 0.01154709 3.8368830 1.253725e-04
## XX2         0.048228101 0.01170855 4.1190490 3.835033e-05</code></pre>
</div>
<div id="misconception-a-low-p-value-indicates-high-model-fit-or-high-predictive-capacity" class="section level3" number="8.7.5">
<h3><span class="header-section-number">8.7.5</span> Misconception: a low <em>p</em>-value indicates high model fit or high predictive capacity</h3>
<p>On page 606, of Lock et al “Statistics: Unlocking the Power of Data,” the authors state in item D “The p-value from the ANOVA table is 0.000 so the model as a whole is effective at predicting grade point averages.” This is incorrect. A p-value is not a measure of the predictive capability of a model because the p-value is a function of the signal, noise (unmodeled error), and <em>sample size</em> while predictive ability is a function of just the signal:noise ratio. If the signal:noise ratio is tiny, the predictive ability is small but the p-value can be tiny if the sample size is large. This is easy to simulate (see script below). The whole-model p-value is exceptionally small (0.00001002) but the relative predictive ability, measured by the <span class="math inline">\(R^2\)</span>, is near zero (0.002).</p>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="p-values.html#cb214-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb214-2"><a href="p-values.html#cb214-2" aria-hidden="true" tabindex="-1"></a>rho <span class="ot">&lt;-</span> <span class="fl">0.5</span></span>
<span id="cb214-3"><a href="p-values.html#cb214-3" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">4</span></span>
<span id="cb214-4"><a href="p-values.html#cb214-4" aria-hidden="true" tabindex="-1"></a>Sigma <span class="ot">&lt;-</span> <span class="fu">diag</span>(<span class="dv">2</span>)</span>
<span id="cb214-5"><a href="p-values.html#cb214-5" aria-hidden="true" tabindex="-1"></a>Sigma[<span class="dv">1</span>,<span class="dv">2</span>] <span class="ot">&lt;-</span> Sigma[<span class="dv">2</span>,<span class="dv">1</span>] <span class="ot">&lt;-</span> rho</span>
<span id="cb214-6"><a href="p-values.html#cb214-6" aria-hidden="true" tabindex="-1"></a>X <span class="ot">&lt;-</span> <span class="fu">rmvnorm</span>(n, <span class="at">mean=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">sigma=</span>Sigma)</span>
<span id="cb214-7"><a href="p-values.html#cb214-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(X) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;X1&quot;</span>, <span class="st">&quot;X2&quot;</span>)</span>
<span id="cb214-8"><a href="p-values.html#cb214-8" aria-hidden="true" tabindex="-1"></a>beta <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="sc">-</span><span class="fl">0.05</span>)</span>
<span id="cb214-9"><a href="p-values.html#cb214-9" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> X<span class="sc">%*%</span>beta <span class="sc">+</span> <span class="fu">rnorm</span>(n)</span>
<span id="cb214-10"><a href="p-values.html#cb214-10" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> X)</span>
<span id="cb214-11"><a href="p-values.html#cb214-11" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(fit)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = y ~ X)
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -3.6449 -0.6857  0.0148  0.6756  3.6510 
## 
## Coefficients:
##              Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)  0.007473   0.010079   0.741 0.458466    
## XX1          0.044305   0.011547   3.837 0.000125 ***
## XX2         -0.051772   0.011709  -4.422  9.9e-06 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 1.008 on 9997 degrees of freedom
## Multiple R-squared:  0.0023, Adjusted R-squared:  0.002101 
## F-statistic: 11.52 on 2 and 9997 DF,  p-value: 1.002e-05</code></pre>
</div>
</div>
<div id="what-the-p-value-does-not-mean" class="section level2" number="8.8">
<h2><span class="header-section-number">8.8</span> What the <em>p</em>-value does not mean</h2>
<ol style="list-style-type: decimal">
<li><em>p</em> is not the probability of the null being true. More formally, this probability is <span class="math inline">\(Prob(null | data)\)</span> but our <em>p</em>-value is <span class="math inline">\(P(data | null)\)</span>. These are not the same. <span class="math inline">\(P(null | data)\)</span> is the probability of the null being true given the data. <span class="math inline">\(P(data | null)\)</span> is the probability of our data, or something more extreme than our data, conditional on a true null.</li>
<li><span class="math inline">\(1-p\)</span> is not the probability of the alternative</li>
<li><em>p</em> is not a measure of effect size.</li>
<li><em>p</em> in one experiment is not the same level of evidence against the null as in another experiment</li>
<li><em>p</em> is not a great indicator of which is more likely, H0 or H1.</li>
<li>If one treatment level has <span class="math inline">\(p &lt; 0.05\)</span> and another treatment level has <span class="math inline">\(p &gt; 0.05\)</span>, this is not evidence that the treatment levels have different effects on the outcome.</li>
</ol>
</div>
<div id="recommendations" class="section level2" number="8.9">
<h2><span class="header-section-number">8.9</span> Recommendations</h2>
<ol style="list-style-type: decimal">
<li>Simply report the exact <em>p</em>-value, along with a CI of the estimate.</li>
</ol>
<ul>
<li><em>P</em>-values are noisy, there is little reason to report more than two significant digits (report “<span class="math inline">\(p = 0.011\)</span>” not “<span class="math inline">\(p = 0.0108\)</span>”) although some journals recommend more than two significant digits.</li>
<li>For high <em>p</em>-values, report “<span class="math inline">\(p = 0.23\)</span>” not “<span class="math inline">\(p = n.s.\)</span>”</li>
<li>For small <em>p</em>-values, there is little reason to report more than one significant digit (report “<span class="math inline">\(p = 0.0002\)</span>” not “<span class="math inline">\(p = 0.00018\)</span>”).</li>
<li>For really small p-values, there is little reason to report the exact <em>p</em>-value (report “<span class="math inline">\(p &lt; 0.0001\)</span>” and not “<span class="math inline">\(p = 2.365E-11\)</span>”). Recognize that “really small” is entirely arbitrary. Rafael Irizarry suggested that <em>p</em>-values less than something like the probability of being killed by lightning strike should be reported as “<span class="math inline">\(p &lt; m\)</span>,” where <span class="math inline">\(m\)</span> is the probability of being killed by lightning strike<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>. According Google University, this is 0.00000142 in one year or 0.00033 in one lifetime. This text will use <span class="math inline">\(p &lt; ls\)</span>" for <em>p</em>-values less than 0.0001 – the lifetime probability of being killed by lightning strike in someone that spends too much time in doors analyzing data.</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>If <span class="math inline">\(p &lt; 0.05\)</span> (or some other <span class="math inline">\(\alpha\)</span>) do not report this as “significant” – in fact, avoid the word “significant.” In the english language, “significant” implies big or important. Small <em>p</em>-values can result even with trivially small effects if <span class="math inline">\(n\)</span> is big or sample variation is small. The phrase “ASK1 knockout had a significant effect on reducing liver TG (<span class="math inline">\(p = 0.011\)</span>)” is</li>
</ol>
<ul>
<li>potentially misleading, if we interpret “significant” to mean “having a large effect on the regulation of liver TG,”</li>
<li>wrong, if we interpret “significant” to mean “there is an ASK1 knockout effect.”
A low <em>p</em>-value is evidence that the effect of ASK1 knockout is not zero, but I would wager that knocking out any gene expressed in white adipose cells will have <em>some</em> effect (however small) on liver TG.</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>If a decision needs to be made (“do we devote time, expense, and personel to pursue this further?”), then a <em>p</em>-value is a useful tool. If <em>p</em> is smaller than say 0.001, this is pretty good evidence that the data is not a fluke of sampling, <strong>as long as we are justifiably confident in all the assumptions that went into computing this <em>p</em>-value</strong>. A replicate experiment with a small <em>p</em>-value is better evidence. If <em>p</em> is closer to 0.01 or 0.05, this is only weak evidence of a fluke because of the sampling variability of <em>p</em>. A replicate experiment with a small <em>p</em>-value is <em>much</em> better evidence.</li>
</ol>
<div id="primary-sources-for-recommendations" class="section level3" number="8.9.1">
<h3><span class="header-section-number">8.9.1</span> Primary sources for recommendations</h3>
<ol style="list-style-type: decimal">
<li>"". <a href="https://link.springer.com/article/10.1007/s10654-016-0149-3">Statistical tests, P values, confidence intervals, and power: a guide to misinterpretations</a>.</li>
<li>“Q: Why do so many colleges and grad schools teach p = 0.05? A: Because that’s still what the scientific community and journal editors use. Q: Why do so many people still use p = 0.05? A: Because that’s what they were taught in college or grad school.” – <a href="https://amstat.tandfonline.com/doi/full/10.1080/00031305.2016.1154108">ASA Statement on Statistical Significance and P-Values</a></li>
<li>“We then discuss our own proposal, which is to abandon statistical significance. We recommend dropping the NHST paradigm—and the p-value thresholds intrinsic to it—as the default statistical paradigm for research, publication, and discovery in the biomedical and social sciences.” – <a href="https://www.tandfonline.com/doi/abs/10.1080/00031305.2018.1527253">Abandon Statistical Significance</a></li>
<li>" We conclude, based on our review of the articles in this special issue and the broader literature, that it is time to stop using the term “statistically significant” entirely. Nor should variants such as “significantly different,” “p &lt; 0.05,” and “nonsignificant” survive, whether expressed in words, by asterisks in a table, or in some other way." – <a href="https://www.tandfonline.com/doi/full/10.1080/00031305.2019.1583913">Moving to a World Beyond “p &lt; 0.05”</a></li>
<li>“We agree, and call for the entire concept of statistical significance to be abandoned.”–<a href="https://www.nature.com/articles/d41586-019-00857-9">Scientists rise up against statistical significance</a></li>
</ol>
</div>
</div>
<div id="problems" class="section level2" number="8.10">
<h2><span class="header-section-number">8.10</span> Problems</h2>
<p>Problem 1 – simulate the distribution of <em>p</em> under the null. There are many ways to do this but a straightforard approach is to</p>
<ol style="list-style-type: decimal">
<li>Create a <span class="math inline">\(2n \times m\)</span> matrix of random normal deviates with mean 0 and sd 1</li>
<li>Do a <em>t</em>-test on each column, with the first <span class="math inline">\(n\)</span> values assigned to one group and the remaining <span class="math inline">\(n\)</span> values assigned to the second group. Save the <em>p</em>-value from each.</li>
<li>Plot a histogram of the <em>p</em>-values.</li>
<li>What is the distribution? What is the most likely value of <em>p</em>?</li>
</ol>
<p>Problem 2 – simulate power. Again, many ways to do this but following up on Problem 1.
1. Create a <span class="math inline">\(2n \times m\)</span> matrix of random normal deviates with mean 0 and sd 1
2. Add an effect to the first <span class="math inline">\(n\)</span> values of each column. Things to think about
a. what is a good effect size to add? The effect/sd ratio, known as Cohen’s d, is a relative (or standardized) measure of effect size. Cohen suggest 0.2, 0.5, and 0.8 as small, medium, and large standardized effects.
b. should the same effect be added to each individual? Yes! It is the random component that captures the individual variation in the response.
3. Do a <em>t</em>-test on each column of the matrix, using the first <span class="math inline">\(n\)</span> values in group 1 and the remaining <span class="math inline">\(n\)</span> values in group 2. Save the p-values for each.
4. Compute the power, the relative frequency <span class="math inline">\(p \le 0.05\)</span>.
5. Repeat with different values of <span class="math inline">\(n\)</span>, effect size, and sd, but only vary one at a time. How does power vary with these three parameters?</p>

</div>
</div>
<div class="footnotes">
<hr />
<ol start="3">
<li id="fn3"><p>Amrhein, V., Trafimow, D. and Greenland, S., 2019. Inferential statistics as descriptive statistics: There is no replication crisis if we don’t expect replication. The American Statistician, 73(sup1), pp.262-270.<a href="p-values.html#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p><a href="https://twitter.com/rafalab/status/1310610623898808320" class="uri">https://twitter.com/rafalab/status/1310610623898808320</a><a href="p-values.html#fnref4" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="uncertainty.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="errors-in-inference.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/chapters/17-p-values.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Walker-elementary-statistical-modeling-draft.pdf", "Walker-elementary-statistical-modeling-draft.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
