<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>16.8 Example 3 – Factorial experiments and no subsampling replicates (exp5c) | Statistics for the Experimental Biologist</title>
  <meta name="description" content="An introduction to statistical modeling for experimental biology researchers" />
  <meta name="generator" content="bookdown 0.32 and GitBook 2.6.7" />

  <meta property="og:title" content="16.8 Example 3 – Factorial experiments and no subsampling replicates (exp5c) | Statistics for the Experimental Biologist" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="An introduction to statistical modeling for experimental biology researchers" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="16.8 Example 3 – Factorial experiments and no subsampling replicates (exp5c) | Statistics for the Experimental Biologist" />
  
  <meta name="twitter:description" content="An introduction to statistical modeling for experimental biology researchers" />
  

<meta name="author" content="Copyright 2018 Jeffrey A. Walker" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="lmm-example2.html"/>
<link rel="next" href="lmm-example4.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Statistics for the Experimental Biologist</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="" data-path="linear-models.html"><a href="linear-models.html"><i class="fa fa-check"></i>Linear Models</a></li>
<li class="chapter" data-level="" data-path="arent-t-tests-and-anova-good-enough.html"><a href="arent-t-tests-and-anova-good-enough.html"><i class="fa fa-check"></i>Aren’t <em>t</em>-tests and ANOVA good enough?</a></li>
<li class="chapter" data-level="" data-path="what-is-unusual-about-this-book.html"><a href="what-is-unusual-about-this-book.html"><i class="fa fa-check"></i>What is unusual about this book?</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="a-table-of-models.html"><a href="a-table-of-models.html"><i class="fa fa-check"></i>A Table of Models</a>
<ul>
<li class="chapter" data-level="" data-path="including-mapping-between-linear-models-and-classical-tests.html"><a href="including-mapping-between-linear-models-and-classical-tests.html"><i class="fa fa-check"></i>including mapping between linear models and classical tests</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="ask1-intro.html"><a href="ask1-intro.html"><i class="fa fa-check"></i><b>1</b> Analyzing experimental data with a linear model</a>
<ul>
<li class="chapter" data-level="1.1" data-path="this-text-is-about-using-linear-models-to-estimate-treatment-effects-and-the-uncertainty-in-our-estimates.-this-raises-the-question-what-is-an-effect.html"><a href="this-text-is-about-using-linear-models-to-estimate-treatment-effects-and-the-uncertainty-in-our-estimates.-this-raises-the-question-what-is-an-effect.html"><i class="fa fa-check"></i><b>1.1</b> This text is about using linear models to estimate treatment effects and the uncertainty in our estimates. This, raises the question, what is “an effect”?</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-i-getting-started.html"><a href="part-i-getting-started.html"><i class="fa fa-check"></i>Part I: Getting Started</a></li>
<li class="chapter" data-level="2" data-path="getting-started-r-projects-and-r-markdown.html"><a href="getting-started-r-projects-and-r-markdown.html"><i class="fa fa-check"></i><b>2</b> Getting Started – R Projects and R Markdown</a>
<ul>
<li class="chapter" data-level="2.1" data-path="r-vs-r-studio.html"><a href="r-vs-r-studio.html"><i class="fa fa-check"></i><b>2.1</b> R vs R Studio</a></li>
<li class="chapter" data-level="2.2" data-path="download-and-install-r-and-r-studio.html"><a href="download-and-install-r-and-r-studio.html"><i class="fa fa-check"></i><b>2.2</b> Download and install R and R studio</a></li>
<li class="chapter" data-level="2.3" data-path="open-r-studio-and-modify-the-workspace-preference.html"><a href="open-r-studio-and-modify-the-workspace-preference.html"><i class="fa fa-check"></i><b>2.3</b> Open R Studio and modify the workspace preference</a></li>
<li class="chapter" data-level="2.4" data-path="if-you-didnt-modify-the-workspace-preferences-from-the-previous-section-go-back-and-do-it.html"><a href="if-you-didnt-modify-the-workspace-preferences-from-the-previous-section-go-back-and-do-it.html"><i class="fa fa-check"></i><b>2.4</b> If you didn’t modify the workspace preferences from the previous section, go back and do it</a></li>
<li class="chapter" data-level="2.5" data-path="r-markdown-in-a-nutshell.html"><a href="r-markdown-in-a-nutshell.html"><i class="fa fa-check"></i><b>2.5</b> R Markdown in a nutshell</a></li>
<li class="chapter" data-level="2.6" data-path="install-r-markdown.html"><a href="install-r-markdown.html"><i class="fa fa-check"></i><b>2.6</b> Install R Markdown</a></li>
<li class="chapter" data-level="2.7" data-path="importing-packages.html"><a href="importing-packages.html"><i class="fa fa-check"></i><b>2.7</b> Importing Packages</a></li>
<li class="chapter" data-level="2.8" data-path="setup-create-project.html"><a href="setup-create-project.html"><i class="fa fa-check"></i><b>2.8</b> Create an R Studio Project for this textbook</a></li>
<li class="chapter" data-level="2.9" data-path="working-on-a-project-in-a-nutshell.html"><a href="working-on-a-project-in-a-nutshell.html"><i class="fa fa-check"></i><b>2.9</b> Working on a project, in a nutshell</a></li>
<li class="chapter" data-level="2.10" data-path="setup-create-rmd.html"><a href="setup-create-rmd.html"><i class="fa fa-check"></i><b>2.10</b> Create and setup an R Markdown document (Rmd)</a>
<ul>
<li class="chapter" data-level="2.10.1" data-path="setup-create-rmd.html"><a href="setup-create-rmd.html#modify-the-yaml-header"><i class="fa fa-check"></i><b>2.10.1</b> Modify the yaml header</a></li>
<li class="chapter" data-level="2.10.2" data-path="setup-create-rmd.html"><a href="setup-create-rmd.html#modify-the-setup-chunk"><i class="fa fa-check"></i><b>2.10.2</b> Modify the “setup” chunk</a></li>
</ul></li>
<li class="chapter" data-level="2.11" data-path="lets-play-around-with-an-r-markdown-file.html"><a href="lets-play-around-with-an-r-markdown-file.html"><i class="fa fa-check"></i><b>2.11</b> Let’s play around with an R Markdown file</a>
<ul>
<li class="chapter" data-level="2.11.1" data-path="lets-play-around-with-an-r-markdown-file.html"><a href="lets-play-around-with-an-r-markdown-file.html#create-a-fake-data-chunk"><i class="fa fa-check"></i><b>2.11.1</b> Create a “fake-data” chunk</a></li>
<li class="chapter" data-level="2.11.2" data-path="lets-play-around-with-an-r-markdown-file.html"><a href="lets-play-around-with-an-r-markdown-file.html#create-a-plot-chunk"><i class="fa fa-check"></i><b>2.11.2</b> Create a “plot” chunk</a></li>
<li class="chapter" data-level="2.11.3" data-path="lets-play-around-with-an-r-markdown-file.html"><a href="lets-play-around-with-an-r-markdown-file.html#knit-the-rmd"><i class="fa fa-check"></i><b>2.11.3</b> Knit the Rmd</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-ii-r-fundamentals.html"><a href="part-ii-r-fundamentals.html"><i class="fa fa-check"></i>Part II: R fundamentals</a></li>
<li class="chapter" data-level="3" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>3</b> Data – Reading, Wrangling, and Writing</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-long.html"><a href="data-long.html"><i class="fa fa-check"></i><b>3.1</b> Long data format – the way data should be</a></li>
<li class="chapter" data-level="3.2" data-path="data-here.html"><a href="data-here.html"><i class="fa fa-check"></i><b>3.2</b> Use the here function to construct the file path</a></li>
<li class="chapter" data-level="3.3" data-path="learning-from-this-chapter.html"><a href="learning-from-this-chapter.html"><i class="fa fa-check"></i><b>3.3</b> Learning from this chapter</a></li>
<li class="chapter" data-level="3.4" data-path="data-working-in-r.html"><a href="data-working-in-r.html"><i class="fa fa-check"></i><b>3.4</b> Working in R</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="data-working-in-r.html"><a href="data-working-in-r.html#importing-data"><i class="fa fa-check"></i><b>3.4.1</b> Importing data</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="data-wrangling.html"><a href="data-wrangling.html"><i class="fa fa-check"></i><b>3.5</b> Data wrangling</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="data-wrangling.html"><a href="data-wrangling.html#reshaping-data-wide-to-long"><i class="fa fa-check"></i><b>3.5.1</b> Reshaping data – Wide to long</a></li>
<li class="chapter" data-level="3.5.2" data-path="data-wrangling.html"><a href="data-wrangling.html#reshaping-data-transpose-turning-the-columns-into-rows"><i class="fa fa-check"></i><b>3.5.2</b> Reshaping data – Transpose (turning the columns into rows)</a></li>
<li class="chapter" data-level="3.5.3" data-path="data-wrangling.html"><a href="data-wrangling.html#combining-data"><i class="fa fa-check"></i><b>3.5.3</b> Combining data</a></li>
<li class="chapter" data-level="3.5.4" data-path="data-wrangling.html"><a href="data-wrangling.html#data-subset"><i class="fa fa-check"></i><b>3.5.4</b> Subsetting data</a></li>
<li class="chapter" data-level="3.5.5" data-path="data-wrangling.html"><a href="data-wrangling.html#wrangling-columns"><i class="fa fa-check"></i><b>3.5.5</b> Wrangling columns</a></li>
<li class="chapter" data-level="3.5.6" data-path="data-wrangling.html"><a href="data-wrangling.html#missing-data"><i class="fa fa-check"></i><b>3.5.6</b> Missing data</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="saving-data.html"><a href="saving-data.html"><i class="fa fa-check"></i><b>3.6</b> Saving data</a></li>
<li class="chapter" data-level="3.7" data-path="exercises.html"><a href="exercises.html"><i class="fa fa-check"></i><b>3.7</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="plotting-models.html"><a href="plotting-models.html"><i class="fa fa-check"></i><b>4</b> Plotting Models</a>
<ul>
<li class="chapter" data-level="4.1" data-path="pretty-good-plots-show-the-model-and-the-data.html"><a href="pretty-good-plots-show-the-model-and-the-data.html"><i class="fa fa-check"></i><b>4.1</b> Pretty good plots show the model and the data</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="pretty-good-plots-show-the-model-and-the-data.html"><a href="pretty-good-plots-show-the-model-and-the-data.html#pretty-good-plot-component-1-modeled-effects-plot"><i class="fa fa-check"></i><b>4.1.1</b> Pretty good plot component 1: Modeled effects plot</a></li>
<li class="chapter" data-level="4.1.2" data-path="pretty-good-plots-show-the-model-and-the-data.html"><a href="pretty-good-plots-show-the-model-and-the-data.html#pretty-good-plot-component-2-modeled-mean-and-ci-plot"><i class="fa fa-check"></i><b>4.1.2</b> Pretty good plot component 2: Modeled mean and CI plot</a></li>
<li class="chapter" data-level="4.1.3" data-path="pretty-good-plots-show-the-model-and-the-data.html"><a href="pretty-good-plots-show-the-model-and-the-data.html#combining-effects-and-modeled-mean-and-ci-plots-an-effects-and-response-plot."><i class="fa fa-check"></i><b>4.1.3</b> Combining Effects and Modeled mean and CI plots – an Effects and response plot.</a></li>
<li class="chapter" data-level="4.1.4" data-path="pretty-good-plots-show-the-model-and-the-data.html"><a href="pretty-good-plots-show-the-model-and-the-data.html#some-comments-on-plot-components"><i class="fa fa-check"></i><b>4.1.4</b> Some comments on plot components</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="working-in-r.html"><a href="working-in-r.html"><i class="fa fa-check"></i><b>4.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="working-in-r.html"><a href="working-in-r.html#source-data"><i class="fa fa-check"></i><b>4.2.1</b> Source data</a></li>
<li class="chapter" data-level="4.2.2" data-path="working-in-r.html"><a href="working-in-r.html#how-to-plot-the-model"><i class="fa fa-check"></i><b>4.2.2</b> How to plot the model</a></li>
<li class="chapter" data-level="4.2.3" data-path="working-in-r.html"><a href="working-in-r.html#be-sure-ggplot_the_model-is-in-your-r-folder"><i class="fa fa-check"></i><b>4.2.3</b> Be sure ggplot_the_model is in your R folder</a></li>
<li class="chapter" data-level="4.2.4" data-path="working-in-r.html"><a href="working-in-r.html#how-to-use-the-plot-the-model-functions"><i class="fa fa-check"></i><b>4.2.4</b> How to use the Plot the Model functions</a></li>
<li class="chapter" data-level="4.2.5" data-path="working-in-r.html"><a href="working-in-r.html#how-to-generate-a-response-plot-using-ggpubr"><i class="fa fa-check"></i><b>4.2.5</b> How to generate a Response Plot using ggpubr</a></li>
<li class="chapter" data-level="4.2.6" data-path="working-in-r.html"><a href="working-in-r.html#how-to-generate-a-response-plot-with-a-grid-of-treatments-using-ggplot2"><i class="fa fa-check"></i><b>4.2.6</b> How to generate a Response Plot with a grid of treatments using ggplot2</a></li>
<li class="chapter" data-level="4.2.7" data-path="working-in-r.html"><a href="working-in-r.html#how-to-generate-an-effects-plot"><i class="fa fa-check"></i><b>4.2.7</b> How to generate an Effects Plot</a></li>
<li class="chapter" data-level="4.2.8" data-path="working-in-r.html"><a href="working-in-r.html#how-to-combine-the-response-and-effects-plots"><i class="fa fa-check"></i><b>4.2.8</b> How to combine the response and effects plots</a></li>
<li class="chapter" data-level="4.2.9" data-path="working-in-r.html"><a href="working-in-r.html#how-to-add-the-interaction-effect-to-response-and-effects-plots"><i class="fa fa-check"></i><b>4.2.9</b> How to add the interaction effect to response and effects plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-iii-some-fundamentals-of-statistical-modeling.html"><a href="part-iii-some-fundamentals-of-statistical-modeling.html"><i class="fa fa-check"></i>Part III: Some Fundamentals of Statistical Modeling</a></li>
<li class="chapter" data-level="5" data-path="uncertainty.html"><a href="uncertainty.html"><i class="fa fa-check"></i><b>5</b> Variability and Uncertainty (Standard Deviations, Standard Errors, and Confidence Intervals)</a>
<ul>
<li class="chapter" data-level="5.1" data-path="standard-errors-are-used-to-compute-p-values-and-confidence-intervals.html"><a href="standard-errors-are-used-to-compute-p-values-and-confidence-intervals.html"><i class="fa fa-check"></i><b>5.1</b> Standard errors are used to compute <em>p</em>-values and confidence intervals</a></li>
<li class="chapter" data-level="5.2" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>5.2</b> Background</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="background.html"><a href="background.html#sample-standard-deviation"><i class="fa fa-check"></i><b>5.2.1</b> Sample standard deviation</a></li>
<li class="chapter" data-level="5.2.2" data-path="background.html"><a href="background.html#standard-error-of-the-mean"><i class="fa fa-check"></i><b>5.2.2</b> Standard error of the mean</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="simulations-using-fake-data-as-an-intuition-pump.html"><a href="simulations-using-fake-data-as-an-intuition-pump.html"><i class="fa fa-check"></i><b>5.3</b> Simulations – using fake data as an intuition pump</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="simulations-using-fake-data-as-an-intuition-pump.html"><a href="simulations-using-fake-data-as-an-intuition-pump.html#using-google-sheets-to-generate-fake-data-to-explore-the-standard-error"><i class="fa fa-check"></i><b>5.3.1</b> Using Google Sheets to generate fake data to explore the standard error</a></li>
<li class="chapter" data-level="5.3.2" data-path="simulations-using-fake-data-as-an-intuition-pump.html"><a href="simulations-using-fake-data-as-an-intuition-pump.html#using-r-to-generate-fake-data-to-explore-the-standard-error"><i class="fa fa-check"></i><b>5.3.2</b> Using R to generate fake data to explore the standard error</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="bootstrap.html"><a href="bootstrap.html"><i class="fa fa-check"></i><b>5.4</b> Bootstrapped standard errors</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="bootstrap.html"><a href="bootstrap.html#an-example-of-bootstrapped-standard-errors"><i class="fa fa-check"></i><b>5.4.1</b> An example of bootstrapped standard errors</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="confidence-intervals.html"><a href="confidence-intervals.html"><i class="fa fa-check"></i><b>5.5</b> Confidence Intervals</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="confidence-intervals.html"><a href="confidence-intervals.html#just-the-math"><i class="fa fa-check"></i><b>5.5.1</b> Just the math</a></li>
<li class="chapter" data-level="5.5.2" data-path="confidence-intervals.html"><a href="confidence-intervals.html#interpretation-of-a-confidence-interval"><i class="fa fa-check"></i><b>5.5.2</b> Interpretation of a confidence interval</a></li>
<li class="chapter" data-level="5.5.3" data-path="confidence-intervals.html"><a href="confidence-intervals.html#bootstrap-confidence-interval"><i class="fa fa-check"></i><b>5.5.3</b> Bootstrap confidence interval</a></li>
<li class="chapter" data-level="5.5.4" data-path="confidence-intervals.html"><a href="confidence-intervals.html#a-plot-of-a-parametric-ci-vs.-bootstrap-ci-of-the-means"><i class="fa fa-check"></i><b>5.5.4</b> A plot of a “parametric” CI vs. bootstrap CI of the means</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="standard-error-of-a-difference-between-means.html"><a href="standard-error-of-a-difference-between-means.html"><i class="fa fa-check"></i><b>5.6</b> Standard error of a difference between means</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="standard-error-of-a-difference-between-means.html"><a href="standard-error-of-a-difference-between-means.html#the-standard-error-of-a-difference-between-means-is-the-standard-deviation-of-the-sampling-distribution-of-the-difference"><i class="fa fa-check"></i><b>5.6.1</b> The standard error of a difference between means is the standard deviation of the sampling distribution of the difference</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="confidence-limits-of-a-difference-between-means.html"><a href="confidence-limits-of-a-difference-between-means.html"><i class="fa fa-check"></i><b>5.7</b> Confidence limits of a difference between means</a>
<ul>
<li class="chapter" data-level="5.7.1" data-path="confidence-limits-of-a-difference-between-means.html"><a href="confidence-limits-of-a-difference-between-means.html#of-95-cis-of-the-difference-include-the-true-difference"><i class="fa fa-check"></i><b>5.7.1</b> 95% of 95% CIs of the difference include the true difference</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="hidden-code.html"><a href="hidden-code.html"><i class="fa fa-check"></i><b>5.8</b> Hidden code</a>
<ul>
<li class="chapter" data-level="5.8.1" data-path="hidden-code.html"><a href="hidden-code.html#import-exp3b"><i class="fa fa-check"></i><b>5.8.1</b> Import exp3b</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="p-values.html"><a href="p-values.html"><i class="fa fa-check"></i><b>6</b> P-values</a>
<ul>
<li class="chapter" data-level="6.1" data-path="a-p-value-is-the-probability-of-sampling-a-value-as-or-more-extreme-than-the-test-statistic-if-sampling-from-a-null-distribution.html"><a href="a-p-value-is-the-probability-of-sampling-a-value-as-or-more-extreme-than-the-test-statistic-if-sampling-from-a-null-distribution.html"><i class="fa fa-check"></i><b>6.1</b> A <em>p</em>-value is the probability of sampling a value as or more extreme than the test statistic if sampling from a null distribution</a></li>
<li class="chapter" data-level="6.2" data-path="pump-your-intuition-creating-a-null-distribution.html"><a href="pump-your-intuition-creating-a-null-distribution.html"><i class="fa fa-check"></i><b>6.2</b> Pump your intuition – Creating a null distribution</a></li>
<li class="chapter" data-level="6.3" data-path="a-null-distribution-of-t-values-the-t-distribution.html"><a href="a-null-distribution-of-t-values-the-t-distribution.html"><i class="fa fa-check"></i><b>6.3</b> A null distribution of <em>t</em>-values – the <em>t</em> distribution</a></li>
<li class="chapter" data-level="6.4" data-path="p-values-from-the-perspective-of-permutation.html"><a href="p-values-from-the-perspective-of-permutation.html"><i class="fa fa-check"></i><b>6.4</b> P-values from the perspective of permutation</a></li>
<li class="chapter" data-level="6.5" data-path="parametric-vs.-non-parametric-statistics.html"><a href="parametric-vs.-non-parametric-statistics.html"><i class="fa fa-check"></i><b>6.5</b> Parametric vs. non-parametric statistics</a></li>
<li class="chapter" data-level="6.6" data-path="frequentist-probability-and-the-interpretation-of-p-values.html"><a href="frequentist-probability-and-the-interpretation-of-p-values.html"><i class="fa fa-check"></i><b>6.6</b> frequentist probability and the interpretation of p-values</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="frequentist-probability-and-the-interpretation-of-p-values.html"><a href="frequentist-probability-and-the-interpretation-of-p-values.html#background-1"><i class="fa fa-check"></i><b>6.6.1</b> Background</a></li>
<li class="chapter" data-level="6.6.2" data-path="frequentist-probability-and-the-interpretation-of-p-values.html"><a href="frequentist-probability-and-the-interpretation-of-p-values.html#this-book-covers-frequentist-approaches-to-statistical-modeling-and-when-a-probability-arises-such-as-the-p-value-of-a-test-statistic-this-will-be-a-frequentist-probability."><i class="fa fa-check"></i><b>6.6.2</b> This book covers frequentist approaches to statistical modeling and when a probability arises, such as the <em>p</em>-value of a test statistic, this will be a frequentist probability.</a></li>
<li class="chapter" data-level="6.6.3" data-path="frequentist-probability-and-the-interpretation-of-p-values.html"><a href="frequentist-probability-and-the-interpretation-of-p-values.html#two-interpretations-of-the-p-value"><i class="fa fa-check"></i><b>6.6.3</b> Two interpretations of the <em>p</em>-value</a></li>
<li class="chapter" data-level="6.6.4" data-path="frequentist-probability-and-the-interpretation-of-p-values.html"><a href="frequentist-probability-and-the-interpretation-of-p-values.html#nhst"><i class="fa fa-check"></i><b>6.6.4</b> NHST</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="some-major-misconceptions-of-the-p-value.html"><a href="some-major-misconceptions-of-the-p-value.html"><i class="fa fa-check"></i><b>6.7</b> Some major misconceptions of the <em>p</em>-value</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="some-major-misconceptions-of-the-p-value.html"><a href="some-major-misconceptions-of-the-p-value.html#misconception-p-0.05-means-there-is-no-effect-of-treatment"><i class="fa fa-check"></i><b>6.7.1</b> Misconception: <span class="math inline">\(p &gt; 0.05\)</span> means there is no effect of treatment</a></li>
<li class="chapter" data-level="6.7.2" data-path="some-major-misconceptions-of-the-p-value.html"><a href="some-major-misconceptions-of-the-p-value.html#misconception-a-p-value-is-repeatable"><i class="fa fa-check"></i><b>6.7.2</b> Misconception: a <em>p</em>-value is repeatable</a></li>
<li class="chapter" data-level="6.7.3" data-path="some-major-misconceptions-of-the-p-value.html"><a href="some-major-misconceptions-of-the-p-value.html#misconception-0.05-is-the-lifetime-rate-of-false-discoveries"><i class="fa fa-check"></i><b>6.7.3</b> Misconception: 0.05 is the lifetime rate of false discoveries</a></li>
<li class="chapter" data-level="6.7.4" data-path="some-major-misconceptions-of-the-p-value.html"><a href="some-major-misconceptions-of-the-p-value.html#misconception-a-low-p-value-indicates-an-important-effect"><i class="fa fa-check"></i><b>6.7.4</b> Misconception: a low <em>p</em>-value indicates an important effect</a></li>
<li class="chapter" data-level="6.7.5" data-path="some-major-misconceptions-of-the-p-value.html"><a href="some-major-misconceptions-of-the-p-value.html#misconception-a-low-p-value-indicates-high-model-fit-or-high-predictive-capacity"><i class="fa fa-check"></i><b>6.7.5</b> Misconception: a low <em>p</em>-value indicates high model fit or high predictive capacity</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="what-the-p-value-does-not-mean.html"><a href="what-the-p-value-does-not-mean.html"><i class="fa fa-check"></i><b>6.8</b> What the <em>p</em>-value does not mean</a></li>
<li class="chapter" data-level="6.9" data-path="on-using-p-values-in-experimental-biology.html"><a href="on-using-p-values-in-experimental-biology.html"><i class="fa fa-check"></i><b>6.9</b> On using <em>p</em>-values in experimental biology</a></li>
<li class="chapter" data-level="6.10" data-path="better-reproducibility.html"><a href="better-reproducibility.html"><i class="fa fa-check"></i><b>6.10</b> Better reproducibility</a></li>
<li class="chapter" data-level="6.11" data-path="multiple-testing-and-controlling-for-false-positives.html"><a href="multiple-testing-and-controlling-for-false-positives.html"><i class="fa fa-check"></i><b>6.11</b> Multiple testing and controlling for false positives</a>
<ul>
<li class="chapter" data-level="6.11.1" data-path="multiple-testing-and-controlling-for-false-positives.html"><a href="multiple-testing-and-controlling-for-false-positives.html#controlling-the-family-wise-error-rate-when-all-tests-are-testing-the-same-hypothesis"><i class="fa fa-check"></i><b>6.11.1</b> Controlling the family-wise error rate when all tests are testing the same hypothesis</a></li>
</ul></li>
<li class="chapter" data-level="6.12" data-path="recommendations.html"><a href="recommendations.html"><i class="fa fa-check"></i><b>6.12</b> Recommendations</a>
<ul>
<li class="chapter" data-level="6.12.1" data-path="recommendations.html"><a href="recommendations.html#primary-sources-for-recommendations"><i class="fa fa-check"></i><b>6.12.1</b> Primary sources for recommendations</a></li>
</ul></li>
<li class="chapter" data-level="6.13" data-path="problems.html"><a href="problems.html"><i class="fa fa-check"></i><b>6.13</b> Problems</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="errors-in-inference.html"><a href="errors-in-inference.html"><i class="fa fa-check"></i><b>7</b> Errors in inference</a>
<ul>
<li class="chapter" data-level="7.1" data-path="classical-nhst-concepts-of-wrong.html"><a href="classical-nhst-concepts-of-wrong.html"><i class="fa fa-check"></i><b>7.1</b> Classical NHST concepts of wrong</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="classical-nhst-concepts-of-wrong.html"><a href="classical-nhst-concepts-of-wrong.html#type-i-error"><i class="fa fa-check"></i><b>7.1.1</b> Type I error</a></li>
<li class="chapter" data-level="7.1.2" data-path="classical-nhst-concepts-of-wrong.html"><a href="classical-nhst-concepts-of-wrong.html#power"><i class="fa fa-check"></i><b>7.1.2</b> Power</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="a-non-neyman-pearson-concept-of-power.html"><a href="a-non-neyman-pearson-concept-of-power.html"><i class="fa fa-check"></i><b>7.2</b> A non-Neyman-Pearson concept of power</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="a-non-neyman-pearson-concept-of-power.html"><a href="a-non-neyman-pearson-concept-of-power.html#estimation-error"><i class="fa fa-check"></i><b>7.2.1</b> Estimation error</a></li>
<li class="chapter" data-level="7.2.2" data-path="a-non-neyman-pearson-concept-of-power.html"><a href="a-non-neyman-pearson-concept-of-power.html#coverage"><i class="fa fa-check"></i><b>7.2.2</b> Coverage</a></li>
<li class="chapter" data-level="7.2.3" data-path="a-non-neyman-pearson-concept-of-power.html"><a href="a-non-neyman-pearson-concept-of-power.html#type-s-error"><i class="fa fa-check"></i><b>7.2.3</b> Type S error</a></li>
<li class="chapter" data-level="7.2.4" data-path="a-non-neyman-pearson-concept-of-power.html"><a href="a-non-neyman-pearson-concept-of-power.html#type-m-error"><i class="fa fa-check"></i><b>7.2.4</b> Type M error</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-iv-introduction-to-linear-models.html"><a href="part-iv-introduction-to-linear-models.html"><i class="fa fa-check"></i>Part IV: Introduction to Linear Models</a></li>
<li class="chapter" data-level="8" data-path="intro-linear-models.html"><a href="intro-linear-models.html"><i class="fa fa-check"></i><b>8</b> An introduction to linear models</a>
<ul>
<li class="chapter" data-level="8.1" data-path="lm-specifications.html"><a href="lm-specifications.html"><i class="fa fa-check"></i><b>8.1</b> Two specifications of a linear model</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="lm-specifications.html"><a href="lm-specifications.html#the-error-draw-specification"><i class="fa fa-check"></i><b>8.1.1</b> The “error draw” specification</a></li>
<li class="chapter" data-level="8.1.2" data-path="lm-specifications.html"><a href="lm-specifications.html#the-conditional-draw-specification"><i class="fa fa-check"></i><b>8.1.2</b> The “conditional draw” specification</a></li>
<li class="chapter" data-level="8.1.3" data-path="lm-specifications.html"><a href="lm-specifications.html#comparing-the-error-draw-and-conditional-draw-ways-of-specifying-the-linear-model"><i class="fa fa-check"></i><b>8.1.3</b> Comparing the error-draw and conditional-draw ways of specifying the linear model</a></li>
<li class="chapter" data-level="8.1.4" data-path="lm-specifications.html"><a href="lm-specifications.html#anova-notation-of-a-linear-model"><i class="fa fa-check"></i><b>8.1.4</b> ANOVA notation of a linear model</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="a-linear-model-can-be-fit-to-data-with-continuous-discrete-or-categorical-x-variables.html"><a href="a-linear-model-can-be-fit-to-data-with-continuous-discrete-or-categorical-x-variables.html"><i class="fa fa-check"></i><b>8.2</b> A linear model can be fit to data with continuous, discrete, or categorical <span class="math inline">\(X\)</span> variables</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="a-linear-model-can-be-fit-to-data-with-continuous-discrete-or-categorical-x-variables.html"><a href="a-linear-model-can-be-fit-to-data-with-continuous-discrete-or-categorical-x-variables.html#fitting-linear-models-to-experimental-data-in-which-the-x-variable-is-continuous-or-discrete"><i class="fa fa-check"></i><b>8.2.1</b> Fitting linear models to experimental data in which the <span class="math inline">\(X\)</span> variable is continuous or discrete</a></li>
<li class="chapter" data-level="8.2.2" data-path="a-linear-model-can-be-fit-to-data-with-continuous-discrete-or-categorical-x-variables.html"><a href="a-linear-model-can-be-fit-to-data-with-continuous-discrete-or-categorical-x-variables.html#slope"><i class="fa fa-check"></i><b>8.2.2</b> Fitting linear models to experimental data in which the <span class="math inline">\(X\)</span> variable is categorical</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="statistical-models-are-used-for-prediction-explanation-and-description.html"><a href="statistical-models-are-used-for-prediction-explanation-and-description.html"><i class="fa fa-check"></i><b>8.3</b> Statistical models are used for prediction, explanation, and description</a></li>
<li class="chapter" data-level="8.4" data-path="what-is-the-interpretation-of-a-regression-coefficient.html"><a href="what-is-the-interpretation-of-a-regression-coefficient.html"><i class="fa fa-check"></i><b>8.4</b> What is the interpretation of a regression coefficient?</a></li>
<li class="chapter" data-level="8.5" data-path="what-do-we-call-the-x-and-y-variables.html"><a href="what-do-we-call-the-x-and-y-variables.html"><i class="fa fa-check"></i><b>8.5</b> What do we call the <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span> variables?</a></li>
<li class="chapter" data-level="8.6" data-path="modeling-strategy.html"><a href="modeling-strategy.html"><i class="fa fa-check"></i><b>8.6</b> Modeling strategy</a></li>
<li class="chapter" data-level="8.7" data-path="predictions-from-the-model.html"><a href="predictions-from-the-model.html"><i class="fa fa-check"></i><b>8.7</b> Predictions from the model</a></li>
<li class="chapter" data-level="8.8" data-path="inference-from-the-model.html"><a href="inference-from-the-model.html"><i class="fa fa-check"></i><b>8.8</b> Inference from the model</a>
<ul>
<li class="chapter" data-level="8.8.1" data-path="inference-from-the-model.html"><a href="inference-from-the-model.html#assumptions-for-inference-with-a-statistical-model"><i class="fa fa-check"></i><b>8.8.1</b> Assumptions for inference with a statistical model</a></li>
<li class="chapter" data-level="8.8.2" data-path="inference-from-the-model.html"><a href="inference-from-the-model.html#specific-assumptions-for-inference-with-a-linear-model"><i class="fa fa-check"></i><b>8.8.2</b> Specific assumptions for inference with a linear model</a></li>
</ul></li>
<li class="chapter" data-level="8.9" data-path="linear-modelregression-model-or-statistical-model.html"><a href="linear-modelregression-model-or-statistical-model.html"><i class="fa fa-check"></i><b>8.9</b> “linear model,”regression model”, or “statistical model”?</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="regression.html"><a href="regression.html"><i class="fa fa-check"></i><b>9</b> Linear models with a single, continuous <em>X</em> (“regression”)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="a-linear-model-with-a-single-continuous-x-is-classical-regression.html"><a href="a-linear-model-with-a-single-continuous-x-is-classical-regression.html"><i class="fa fa-check"></i><b>9.1</b> A linear model with a single, continuous <em>X</em> is classical “regression”</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="a-linear-model-with-a-single-continuous-x-is-classical-regression.html"><a href="a-linear-model-with-a-single-continuous-x-is-classical-regression.html#analysis-of-green-down-data"><i class="fa fa-check"></i><b>9.1.1</b> Analysis of “green-down” data</a></li>
<li class="chapter" data-level="9.1.2" data-path="a-linear-model-with-a-single-continuous-x-is-classical-regression.html"><a href="a-linear-model-with-a-single-continuous-x-is-classical-regression.html#learning-from-the-green-down-example"><i class="fa fa-check"></i><b>9.1.2</b> Learning from the green-down example</a></li>
<li class="chapter" data-level="9.1.3" data-path="a-linear-model-with-a-single-continuous-x-is-classical-regression.html"><a href="a-linear-model-with-a-single-continuous-x-is-classical-regression.html#using-a-regression-model-for-explanation-causal-models"><i class="fa fa-check"></i><b>9.1.3</b> Using a regression model for “explanation” – causal models</a></li>
<li class="chapter" data-level="9.1.4" data-path="a-linear-model-with-a-single-continuous-x-is-classical-regression.html"><a href="a-linear-model-with-a-single-continuous-x-is-classical-regression.html#using-a-regression-model-for-prediction-prediction-models"><i class="fa fa-check"></i><b>9.1.4</b> Using a regression model for prediction – prediction models</a></li>
<li class="chapter" data-level="9.1.5" data-path="a-linear-model-with-a-single-continuous-x-is-classical-regression.html"><a href="a-linear-model-with-a-single-continuous-x-is-classical-regression.html#using-a-regression-model-for-creating-a-new-response-variable-comparing-slopes-of-longitudinal-data"><i class="fa fa-check"></i><b>9.1.5</b> Using a regression model for creating a new response variable – comparing slopes of longitudinal data</a></li>
<li class="chapter" data-level="9.1.6" data-path="a-linear-model-with-a-single-continuous-x-is-classical-regression.html"><a href="a-linear-model-with-a-single-continuous-x-is-classical-regression.html#using-a-regression-model-for-for-calibration"><i class="fa fa-check"></i><b>9.1.6</b> Using a regression model for for calibration</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="working-in-r-1.html"><a href="working-in-r-1.html"><i class="fa fa-check"></i><b>9.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="working-in-r.html"><a href="working-in-r.html#fit-the-model"><i class="fa fa-check"></i><b>9.2.1</b> Fitting the linear model</a></li>
<li class="chapter" data-level="9.2.2" data-path="working-in-r-1.html"><a href="working-in-r-1.html#getting-to-know-the-linear-model-the-summary-function"><i class="fa fa-check"></i><b>9.2.2</b> Getting to know the linear model: the <code>summary</code> function</a></li>
<li class="chapter" data-level="9.2.3" data-path="working-in-r-1.html"><a href="working-in-r-1.html#inference-the-coefficient-table"><i class="fa fa-check"></i><b>9.2.3</b> Inference – the coefficient table</a></li>
<li class="chapter" data-level="9.2.4" data-path="working-in-r-1.html"><a href="working-in-r-1.html#how-good-is-our-model-model-checking"><i class="fa fa-check"></i><b>9.2.4</b> How good is our model? – Model checking</a></li>
<li class="chapter" data-level="9.2.5" data-path="working-in-r-1.html"><a href="working-in-r-1.html#plotting-models-with-continuous-x"><i class="fa fa-check"></i><b>9.2.5</b> Plotting models with continuous <em>X</em></a></li>
<li class="chapter" data-level="9.2.6" data-path="working-in-r-1.html"><a href="working-in-r-1.html#creating-a-table-of-predicted-values-and-95-prediction-intervals"><i class="fa fa-check"></i><b>9.2.6</b> Creating a table of predicted values and 95% prediction intervals</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="hidden-code-1.html"><a href="hidden-code-1.html"><i class="fa fa-check"></i><b>9.3</b> Hidden code</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="hidden-code-1.html"><a href="hidden-code-1.html#import-and-plot-of-fig2c-ecosystem-warming-experimental-data"><i class="fa fa-check"></i><b>9.3.1</b> Import and plot of fig2c (ecosystem warming experimental) data</a></li>
<li class="chapter" data-level="9.3.2" data-path="hidden-code-1.html"><a href="hidden-code-1.html#import-and-plot-efig_3d-ecosysem-warming-observational-data"><i class="fa fa-check"></i><b>9.3.2</b> Import and plot efig_3d (Ecosysem warming observational) data</a></li>
<li class="chapter" data-level="9.3.3" data-path="hidden-code-1.html"><a href="hidden-code-1.html#import-and-plot-of-fig1f-methionine-restriction-data"><i class="fa fa-check"></i><b>9.3.3</b> Import and plot of fig1f (methionine restriction) data</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="try-it.html"><a href="try-it.html"><i class="fa fa-check"></i><b>9.4</b> Try it</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="try-it.html"><a href="try-it.html#a-prediction-model-from-the-literature"><i class="fa fa-check"></i><b>9.4.1</b> A prediction model from the literature</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="intuition-pumps.html"><a href="intuition-pumps.html"><i class="fa fa-check"></i><b>9.5</b> Intuition pumps</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="intuition-pumps.html"><a href="intuition-pumps.html#correlation-and-r2"><i class="fa fa-check"></i><b>9.5.1</b> Correlation and $R^2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="oneway.html"><a href="oneway.html"><i class="fa fa-check"></i><b>10</b> Linear models with a single, categorical <em>X</em> (“t-tests” and “ANOVA”)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="oneway-data.html"><a href="oneway-data.html"><i class="fa fa-check"></i><b>10.1</b> A linear model with a single factor (categorical <em>X</em>variable) estimates the effects of the levels of factor on the response</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="oneway-data.html"><a href="oneway-data.html#oneway-example1"><i class="fa fa-check"></i><b>10.1.1</b> Example 1 (fig3d) – two treatment levels (“groups”)</a></li>
<li class="chapter" data-level="10.1.2" data-path="oneway-data.html"><a href="oneway-data.html#understanding-the-analysis-with-two-treatment-levels"><i class="fa fa-check"></i><b>10.1.2</b> Understanding the analysis with two treatment levels</a></li>
<li class="chapter" data-level="10.1.3" data-path="oneway-data.html"><a href="oneway-data.html#oneway-example2"><i class="fa fa-check"></i><b>10.1.3</b> Example 2 – three treatment levels (“groups”)</a></li>
<li class="chapter" data-level="10.1.4" data-path="oneway-data.html"><a href="oneway-data.html#understanding-the-analysis-with-three-or-more-treatment-levels"><i class="fa fa-check"></i><b>10.1.4</b> Understanding the analysis with three (or more) treatment levels</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="working-in-r-2.html"><a href="working-in-r-2.html"><i class="fa fa-check"></i><b>10.2</b> Working in R</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="working-in-r-2.html"><a href="working-in-r-2.html#fit-the-model-1"><i class="fa fa-check"></i><b>10.2.1</b> Fit the model</a></li>
<li class="chapter" data-level="10.2.2" data-path="working-in-r-2.html"><a href="working-in-r-2.html#controlling-the-output-in-tables-using-the-coefficient-table-as-an-example"><i class="fa fa-check"></i><b>10.2.2</b> Controlling the output in tables using the coefficient table as an example</a></li>
<li class="chapter" data-level="10.2.3" data-path="working-in-r-2.html"><a href="working-in-r-2.html#using-the-emmeans-function"><i class="fa fa-check"></i><b>10.2.3</b> Using the emmeans function</a></li>
<li class="chapter" data-level="10.2.4" data-path="working-in-r-2.html"><a href="working-in-r-2.html#using-the-contrast-function"><i class="fa fa-check"></i><b>10.2.4</b> Using the contrast function</a></li>
<li class="chapter" data-level="10.2.5" data-path="working-in-r-2.html"><a href="working-in-r-2.html#how-to-generate-anova-tables"><i class="fa fa-check"></i><b>10.2.5</b> How to generate ANOVA tables</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="hidden-code-2.html"><a href="hidden-code-2.html"><i class="fa fa-check"></i><b>10.3</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="hidden-code-2.html"><a href="hidden-code-2.html#importing-and-wrangling-the-fig3d-data-for-example-1"><i class="fa fa-check"></i><b>10.3.1</b> Importing and wrangling the fig3d data for example 1</a></li>
<li class="chapter" data-level="10.3.2" data-path="hidden-code-2.html"><a href="hidden-code-2.html#importing-and-wrangling-the-fig2a-data-for-example-2"><i class="fa fa-check"></i><b>10.3.2</b> Importing and wrangling the fig2a data for example 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="model-checking.html"><a href="model-checking.html"><i class="fa fa-check"></i><b>11</b> Model Checking</a>
<ul>
<li class="chapter" data-level="11.1" data-path="check-check.html"><a href="check-check.html"><i class="fa fa-check"></i><b>11.1</b> All statistical analyses should be followed by model checking</a></li>
<li class="chapter" data-level="11.2" data-path="linear-model-assumptions.html"><a href="linear-model-assumptions.html"><i class="fa fa-check"></i><b>11.2</b> Linear model assumptions</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="linear-model-assumptions.html"><a href="linear-model-assumptions.html#a-bit-about-iid"><i class="fa fa-check"></i><b>11.2.1</b> A bit about IID</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="diagnostic-plots-use-the-residuals-from-the-model-fit.html"><a href="diagnostic-plots-use-the-residuals-from-the-model-fit.html"><i class="fa fa-check"></i><b>11.3</b> Diagnostic plots use the residuals from the model fit</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="diagnostic-plots-use-the-residuals-from-the-model-fit.html"><a href="diagnostic-plots-use-the-residuals-from-the-model-fit.html#residuals"><i class="fa fa-check"></i><b>11.3.1</b> Residuals</a></li>
<li class="chapter" data-level="11.3.2" data-path="diagnostic-plots-use-the-residuals-from-the-model-fit.html"><a href="diagnostic-plots-use-the-residuals-from-the-model-fit.html#normal-qq"><i class="fa fa-check"></i><b>11.3.2</b> A Normal Q-Q plot is used to check for characteristic departures from Normality</a></li>
<li class="chapter" data-level="11.3.3" data-path="diagnostic-plots-use-the-residuals-from-the-model-fit.html"><a href="diagnostic-plots-use-the-residuals-from-the-model-fit.html#mapping-qq-plot-departures-from-normality"><i class="fa fa-check"></i><b>11.3.3</b> Mapping QQ-plot departures from Normality</a></li>
<li class="chapter" data-level="11.3.4" data-path="diagnostic-plots-use-the-residuals-from-the-model-fit.html"><a href="diagnostic-plots-use-the-residuals-from-the-model-fit.html#model-checking-spread-level"><i class="fa fa-check"></i><b>11.3.4</b> Model checking homoskedasticity</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="using-r.html"><a href="using-r.html"><i class="fa fa-check"></i><b>11.4</b> Using R</a></li>
<li class="chapter" data-level="11.5" data-path="hidden-code-3.html"><a href="hidden-code-3.html"><i class="fa fa-check"></i><b>11.5</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="hidden-code-3.html"><a href="hidden-code-3.html#normal-q-q-plots"><i class="fa fa-check"></i><b>11.5.1</b> Normal Q-Q plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="violations.html"><a href="violations.html"><i class="fa fa-check"></i><b>12</b> Violations of independence, homogeneity, or Normality</a>
<ul>
<li class="chapter" data-level="12.1" data-path="oneway-paired-t.html"><a href="oneway-paired-t.html"><i class="fa fa-check"></i><b>12.1</b> Lack of independence</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="oneway-paired-t.html"><a href="oneway-paired-t.html#violations-paired-t"><i class="fa fa-check"></i><b>12.1.1</b> Example 1 (exp1b) – a paired t-test is a special case of a linear mixed model</a></li>
<li class="chapter" data-level="12.1.2" data-path="oneway-paired-t.html"><a href="oneway-paired-t.html#violations-rmanova"><i class="fa fa-check"></i><b>12.1.2</b> Example 2 (diHOME exp2a) – A repeated measures ANOVA is a special case of a linear mixed model</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="oneway-welch.html"><a href="oneway-welch.html"><i class="fa fa-check"></i><b>12.2</b> Heterogeneity of variances</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="oneway-welch.html"><a href="oneway-welch.html#when-groups-of-the-focal-test-have-variance"><i class="fa fa-check"></i><b>12.2.1</b> When groups of the focal test have &gt;&gt; variance</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="the-conditional-response-isnt-normal.html"><a href="the-conditional-response-isnt-normal.html"><i class="fa fa-check"></i><b>12.3</b> The conditional response isn’t Normal</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="the-conditional-response-isnt-normal.html"><a href="the-conditional-response-isnt-normal.html#violations-6f"><i class="fa fa-check"></i><b>12.3.1</b> Example 1 (fig6f) – Linear models for non-normal count data</a></li>
<li class="chapter" data-level="12.3.2" data-path="the-conditional-response-isnt-normal.html"><a href="the-conditional-response-isnt-normal.html#my-data-arent-normal-what-is-the-best-practice"><i class="fa fa-check"></i><b>12.3.2</b> My data aren’t normal, what is the best practice?</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="hidden-code-4.html"><a href="hidden-code-4.html"><i class="fa fa-check"></i><b>12.4</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="hidden-code-4.html"><a href="hidden-code-4.html#importing-and-wrangling-the-exp1b-data"><i class="fa fa-check"></i><b>12.4.1</b> Importing and wrangling the exp1b data</a></li>
<li class="chapter" data-level="12.4.2" data-path="hidden-code-4.html"><a href="hidden-code-4.html#importing-and-wrangling-the-exp2a-data"><i class="fa fa-check"></i><b>12.4.2</b> Importing and wrangling the exp2a data</a></li>
<li class="chapter" data-level="12.4.3" data-path="hidden-code-4.html"><a href="hidden-code-4.html#importing-and-wrangling-the-fig6f-data"><i class="fa fa-check"></i><b>12.4.3</b> Importing and wrangling the fig6f data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="issues.html"><a href="issues.html"><i class="fa fa-check"></i><b>13</b> Issues in inference</a>
<ul>
<li class="chapter" data-level="13.1" data-path="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html"><a href="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html"><i class="fa fa-check"></i><b>13.1</b> Replicated experiments – include <span class="math inline">\(\texttt{Experiment}\)</span> as a random factor (better than one-way ANOVA of means)</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html"><a href="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html#issues-exp4d"><i class="fa fa-check"></i><b>13.1.1</b> Multiple experiments Example 1 (wound healing Exp4d)</a></li>
<li class="chapter" data-level="13.1.2" data-path="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html"><a href="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html#issues-combined-experiments"><i class="fa fa-check"></i><b>13.1.2</b> Models for combining replicated experiments</a></li>
<li class="chapter" data-level="13.1.3" data-path="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html"><a href="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html#understanding-model-exp4d_m1"><i class="fa fa-check"></i><b>13.1.3</b> Understanding Model <code>exp4d_m1</code></a></li>
<li class="chapter" data-level="13.1.4" data-path="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html"><a href="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html#the-univariate-model-is-equivalent-to-a-linear-mixed-model-of-the-aggregated-data-model-exp4d_m2"><i class="fa fa-check"></i><b>13.1.4</b> The univariate model is equivalent to a linear mixed model of the aggregated data (Model exp4d_m2)</a></li>
<li class="chapter" data-level="13.1.5" data-path="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html"><a href="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html#a-linear-mixed-model-of-the-full-data"><i class="fa fa-check"></i><b>13.1.5</b> A linear mixed model of the full data</a></li>
<li class="chapter" data-level="13.1.6" data-path="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html"><a href="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html#analysis-of-the-experiment-means-has-less-precision-and-power"><i class="fa fa-check"></i><b>13.1.6</b> Analysis of the experiment means has less precision and power</a></li>
<li class="chapter" data-level="13.1.7" data-path="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html"><a href="replicated-experiments-include-textttexperiment-as-a-random-factor-better-than-one-way-anova-of-means.html#dont-do-this-a-t-testfixed-effect-anova-of-the-full-data"><i class="fa fa-check"></i><b>13.1.7</b> Don’t do this – a t-test/fixed-effect ANOVA of the full data</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="issues-pre-post.html"><a href="issues-pre-post.html"><i class="fa fa-check"></i><b>13.2</b> Comparing change from baseline (pre-post)</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="issues-pre-post.html"><a href="issues-pre-post.html#pre-post-example-1-dpp4-fig4c"><i class="fa fa-check"></i><b>13.2.1</b> Pre-post example 1 (DPP4 fig4c)</a></li>
<li class="chapter" data-level="13.2.2" data-path="issues-pre-post.html"><a href="issues-pre-post.html#what-if-the-data-in-example-1-were-from-from-an-experiment-where-the-treatment-was-applied-prior-to-the-baseline-measure"><i class="fa fa-check"></i><b>13.2.2</b> What if the data in example 1 were from from an experiment where the treatment was applied prior to the baseline measure?</a></li>
<li class="chapter" data-level="13.2.3" data-path="issues-pre-post.html"><a href="issues-pre-post.html#pre-post-example-2-xx-males-fig1c"><i class="fa fa-check"></i><b>13.2.3</b> Pre-post example 2 (XX males fig1c)</a></li>
<li class="chapter" data-level="13.2.4" data-path="issues-pre-post.html"><a href="issues-pre-post.html#issues-regression-to-mean"><i class="fa fa-check"></i><b>13.2.4</b> Regression to the mean</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="longitudinal-designs-with-more-than-one-post-baseline-measure.html"><a href="longitudinal-designs-with-more-than-one-post-baseline-measure.html"><i class="fa fa-check"></i><b>13.3</b> Longitudinal designs with more than one-post baseline measure</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="longitudinal-designs-with-more-than-one-post-baseline-measure.html"><a href="longitudinal-designs-with-more-than-one-post-baseline-measure.html#issues-auc"><i class="fa fa-check"></i><b>13.3.1</b> Area under the curve (AUC)</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="normalization-the-analysis-of-ratios.html"><a href="normalization-the-analysis-of-ratios.html"><i class="fa fa-check"></i><b>13.4</b> Normalization – the analysis of ratios</a>
<ul>
<li class="chapter" data-level="13.4.1" data-path="normalization-the-analysis-of-ratios.html"><a href="normalization-the-analysis-of-ratios.html#kinds-of-ratios-in-experimental-biology"><i class="fa fa-check"></i><b>13.4.1</b> Kinds of ratios in experimental biology</a></li>
<li class="chapter" data-level="13.4.2" data-path="normalization-the-analysis-of-ratios.html"><a href="normalization-the-analysis-of-ratios.html#issues-offset"><i class="fa fa-check"></i><b>13.4.2</b> Example 1 – The ratio is a density (number of something per area)</a></li>
<li class="chapter" data-level="13.4.3" data-path="normalization-the-analysis-of-ratios.html"><a href="normalization-the-analysis-of-ratios.html#issues-size"><i class="fa fa-check"></i><b>13.4.3</b> Example 2 – The ratio is normalizing for size differences</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="dont-do-this-stuff.html"><a href="dont-do-this-stuff.html"><i class="fa fa-check"></i><b>13.5</b> Don’t do this stuff</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="dont-do-this-stuff.html"><a href="dont-do-this-stuff.html#normalize-the-response-so-that-all-control-values-are-equal-to-1."><i class="fa fa-check"></i><b>13.5.1</b> Normalize the response so that all control values are equal to 1.</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="a-difference-in-significance-is-not-necessarily-significant.html"><a href="a-difference-in-significance-is-not-necessarily-significant.html"><i class="fa fa-check"></i><b>13.6</b> A difference in significance is not necessarily significant</a></li>
<li class="chapter" data-level="13.7" data-path="researcher-degrees-of-freedom.html"><a href="researcher-degrees-of-freedom.html"><i class="fa fa-check"></i><b>13.7</b> Researcher degrees of freedom</a></li>
<li class="chapter" data-level="13.8" data-path="hidden-code-5.html"><a href="hidden-code-5.html"><i class="fa fa-check"></i><b>13.8</b> Hidden code</a>
<ul>
<li class="chapter" data-level="13.8.1" data-path="hidden-code-5.html"><a href="hidden-code-5.html#import-exp4d-vimentin-cell-count-data-replicate-experiments-example"><i class="fa fa-check"></i><b>13.8.1</b> Import exp4d vimentin cell count data (replicate experiments example)</a></li>
<li class="chapter" data-level="13.8.2" data-path="hidden-code-5.html"><a href="hidden-code-5.html#import-fig4c-data"><i class="fa fa-check"></i><b>13.8.2</b> Import Fig4c data</a></li>
<li class="chapter" data-level="13.8.3" data-path="hidden-code-5.html"><a href="hidden-code-5.html#xx-males-fig1c"><i class="fa fa-check"></i><b>13.8.3</b> XX males fig1c</a></li>
<li class="chapter" data-level="13.8.4" data-path="hidden-code-5.html"><a href="hidden-code-5.html#generation-of-fake-data-to-illustrate-regression-to-the-mean"><i class="fa fa-check"></i><b>13.8.4</b> Generation of fake data to illustrate regression to the mean</a></li>
<li class="chapter" data-level="13.8.5" data-path="hidden-code-5.html"><a href="hidden-code-5.html#import-fig3f"><i class="fa fa-check"></i><b>13.8.5</b> Import fig3f</a></li>
<li class="chapter" data-level="13.8.6" data-path="hidden-code-5.html"><a href="hidden-code-5.html#issues-import-exp3b"><i class="fa fa-check"></i><b>13.8.6</b> Import exp3b</a></li>
<li class="chapter" data-level="13.8.7" data-path="hidden-code-5.html"><a href="hidden-code-5.html#issues-exp3b-plot"><i class="fa fa-check"></i><b>13.8.7</b> Plot the model of exp3b (glm offset data)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-v-more-than-one-x-multivariable-models.html"><a href="part-v-more-than-one-x-multivariable-models.html"><i class="fa fa-check"></i>Part V: More than one <span class="math inline">\(X\)</span> – Multivariable Models</a></li>
<li class="chapter" data-level="14" data-path="covariates.html"><a href="covariates.html"><i class="fa fa-check"></i><b>14</b> Linear models with added covariates (“ANCOVA”)</a>
<ul>
<li class="chapter" data-level="14.1" data-path="adding-covariates-can-increases-the-precision-of-the-effect-of-interest.html"><a href="adding-covariates-can-increases-the-precision-of-the-effect-of-interest.html"><i class="fa fa-check"></i><b>14.1</b> Adding covariates can increases the precision of the effect of interest</a></li>
<li class="chapter" data-level="14.2" data-path="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html"><a href="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html"><i class="fa fa-check"></i><b>14.2</b> Understanding a linear model with an added covariate – heart necrosis data</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html"><a href="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html#fit-the-model-5"><i class="fa fa-check"></i><b>14.2.1</b> Fit the model</a></li>
<li class="chapter" data-level="14.2.2" data-path="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html"><a href="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html#plot-the-model-2"><i class="fa fa-check"></i><b>14.2.2</b> Plot the model</a></li>
<li class="chapter" data-level="14.2.3" data-path="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html"><a href="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html#interpretation-of-the-model-coefficients"><i class="fa fa-check"></i><b>14.2.3</b> Interpretation of the model coefficients</a></li>
<li class="chapter" data-level="14.2.4" data-path="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html"><a href="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html#everything-adds-up"><i class="fa fa-check"></i><b>14.2.4</b> Everything adds up</a></li>
<li class="chapter" data-level="14.2.5" data-path="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html"><a href="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html#interpretation-of-the-estimated-marginal-means"><i class="fa fa-check"></i><b>14.2.5</b> Interpretation of the estimated marginal means</a></li>
<li class="chapter" data-level="14.2.6" data-path="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html"><a href="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html#interpretation-of-the-contrasts"><i class="fa fa-check"></i><b>14.2.6</b> Interpretation of the contrasts</a></li>
<li class="chapter" data-level="14.2.7" data-path="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html"><a href="understanding-a-linear-model-with-an-added-covariate-heart-necrosis-data.html#adding-the-covariate-improves-inference"><i class="fa fa-check"></i><b>14.2.7</b> Adding the covariate improves inference</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="understanding-interaction-effects-with-covariates.html"><a href="understanding-interaction-effects-with-covariates.html"><i class="fa fa-check"></i><b>14.3</b> Understanding interaction effects with covariates</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="understanding-interaction-effects-with-covariates.html"><a href="understanding-interaction-effects-with-covariates.html#fit-the-model-6"><i class="fa fa-check"></i><b>14.3.1</b> Fit the model</a></li>
<li class="chapter" data-level="14.3.2" data-path="understanding-interaction-effects-with-covariates.html"><a href="understanding-interaction-effects-with-covariates.html#plot-the-model-with-interaction-effect"><i class="fa fa-check"></i><b>14.3.2</b> Plot the model with interaction effect</a></li>
<li class="chapter" data-level="14.3.3" data-path="understanding-interaction-effects-with-covariates.html"><a href="understanding-interaction-effects-with-covariates.html#interpretation-of-the-model-coefficients-1"><i class="fa fa-check"></i><b>14.3.3</b> Interpretation of the model coefficients</a></li>
<li class="chapter" data-level="14.3.4" data-path="understanding-interaction-effects-with-covariates.html"><a href="understanding-interaction-effects-with-covariates.html#what-is-the-effect-of-a-treatment-if-interactions-are-modeled-it-depends."><i class="fa fa-check"></i><b>14.3.4</b> What is the effect of a treatment, if interactions are modeled? – it depends.</a></li>
<li class="chapter" data-level="14.3.5" data-path="understanding-interaction-effects-with-covariates.html"><a href="understanding-interaction-effects-with-covariates.html#which-model-do-we-use-mathcalm_1-or-mathcalm_2"><i class="fa fa-check"></i><b>14.3.5</b> Which model do we use, <span class="math inline">\(\mathcal{M}_1\)</span> or <span class="math inline">\(\mathcal{M}_2\)</span>?</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="understanding-ancova-tables.html"><a href="understanding-ancova-tables.html"><i class="fa fa-check"></i><b>14.4</b> Understanding ANCOVA tables</a></li>
<li class="chapter" data-level="14.5" data-path="working-in-r-3.html"><a href="working-in-r-3.html"><i class="fa fa-check"></i><b>14.5</b> Working in R</a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="working-in-r-3.html"><a href="working-in-r-3.html#importing-the-heart-necrosis-data"><i class="fa fa-check"></i><b>14.5.1</b> Importing the heart necrosis data</a></li>
<li class="chapter" data-level="14.5.2" data-path="working-in-r-3.html"><a href="working-in-r-3.html#fitting-the-model"><i class="fa fa-check"></i><b>14.5.2</b> Fitting the model</a></li>
<li class="chapter" data-level="14.5.3" data-path="working-in-r-3.html"><a href="working-in-r-3.html#using-the-emmeans-function-1"><i class="fa fa-check"></i><b>14.5.3</b> Using the emmeans function</a></li>
<li class="chapter" data-level="14.5.4" data-path="working-in-r-3.html"><a href="working-in-r-3.html#ancova-tables"><i class="fa fa-check"></i><b>14.5.4</b> ANCOVA tables</a></li>
<li class="chapter" data-level="14.5.5" data-path="working-in-r-3.html"><a href="working-in-r-3.html#plotting-the-model"><i class="fa fa-check"></i><b>14.5.5</b> Plotting the model</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="best-practices.html"><a href="best-practices.html"><i class="fa fa-check"></i><b>14.6</b> Best practices</a>
<ul>
<li class="chapter" data-level="14.6.1" data-path="best-practices.html"><a href="best-practices.html#do-not-use-a-ratio-of-partwhole-as-a-response-variable-instead-add-the-denominator-as-a-covariate"><i class="fa fa-check"></i><b>14.6.1</b> Do not use a ratio of part:whole as a response variable – instead add the denominator as a covariate</a></li>
<li class="chapter" data-level="14.6.2" data-path="best-practices.html"><a href="best-practices.html#do-not-use-change-from-baseline-as-a-response-variable-instead-add-the-baseline-measure-as-a-covariate"><i class="fa fa-check"></i><b>14.6.2</b> Do not use change from baseline as a response variable – instead add the baseline measure as a covariate</a></li>
<li class="chapter" data-level="14.6.3" data-path="best-practices.html"><a href="best-practices.html#do-not-test-for-balance-of-baseline-measures"><i class="fa fa-check"></i><b>14.6.3</b> Do not “test for balance” of baseline measures</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="best-practices-2-use-a-covariate-instead-of-normalizing-a-response.html"><a href="best-practices-2-use-a-covariate-instead-of-normalizing-a-response.html"><i class="fa fa-check"></i><b>14.7</b> Best practices 2: Use a covariate instead of normalizing a response</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="factorial.html"><a href="factorial.html"><i class="fa fa-check"></i><b>15</b> Linear models with two categorical <span class="math inline">\(X\)</span> – Factorial linear models (“two-way ANOVA”)</a>
<ul>
<li class="chapter" data-level="15.1" data-path="a-linear-model-with-crossed-factors-estimates-interaction-effects.html"><a href="a-linear-model-with-crossed-factors-estimates-interaction-effects.html"><i class="fa fa-check"></i><b>15.1</b> A linear model with crossed factors estimates interaction effects</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="a-linear-model-with-crossed-factors-estimates-interaction-effects.html"><a href="a-linear-model-with-crossed-factors-estimates-interaction-effects.html#an-interaction-is-a-difference-in-simple-effects"><i class="fa fa-check"></i><b>15.1.1</b> An interaction is a difference in simple effects</a></li>
<li class="chapter" data-level="15.1.2" data-path="a-linear-model-with-crossed-factors-estimates-interaction-effects.html"><a href="a-linear-model-with-crossed-factors-estimates-interaction-effects.html#a-linear-model-with-crossed-factors-includes-interaction-effects"><i class="fa fa-check"></i><b>15.1.2</b> A linear model with crossed factors includes interaction effects</a></li>
<li class="chapter" data-level="15.1.3" data-path="a-linear-model-with-crossed-factors-estimates-interaction-effects.html"><a href="a-linear-model-with-crossed-factors-estimates-interaction-effects.html#factorial-experiments-are-frequently-analyzed-as-flattened-linear-models-in-the-experimental-biology-literature"><i class="fa fa-check"></i><b>15.1.3</b> factorial experiments are frequently analyzed as flattened linear models in the experimental biology literature</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data.html"><a href="example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data.html"><i class="fa fa-check"></i><b>15.2</b> Example 1 – Estimation of a treatment effect relative to a control effect (“Something different”) (Experiment 2j glucose uptake data)</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data.html"><a href="example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data.html#twoway-understand1"><i class="fa fa-check"></i><b>15.2.1</b> Understand the experimental design</a></li>
<li class="chapter" data-level="15.2.2" data-path="example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data.html"><a href="example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data.html#fit-the-linear-model"><i class="fa fa-check"></i><b>15.2.2</b> Fit the linear model</a></li>
<li class="chapter" data-level="15.2.3" data-path="example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data.html"><a href="example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data.html#twoway-exp2j-inf"><i class="fa fa-check"></i><b>15.2.3</b> Inference</a></li>
<li class="chapter" data-level="15.2.4" data-path="example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data.html"><a href="example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data.html#plot-the-model-3"><i class="fa fa-check"></i><b>15.2.4</b> Plot the model</a></li>
<li class="chapter" data-level="15.2.5" data-path="example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data.html"><a href="example-1-estimation-of-a-treatment-effect-relative-to-a-control-effect-something-different-experiment-2j-glucose-uptake-data.html#alternaplot-the-model"><i class="fa fa-check"></i><b>15.2.5</b> alternaPlot the model</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="understanding-the-linear-model-with-crossed-factors-1.html"><a href="understanding-the-linear-model-with-crossed-factors-1.html"><i class="fa fa-check"></i><b>15.3</b> Understanding the linear model with crossed factors 1</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="understanding-the-linear-model-with-crossed-factors-1.html"><a href="understanding-the-linear-model-with-crossed-factors-1.html#twoway-what-coefs-are"><i class="fa fa-check"></i><b>15.3.1</b> What the coefficients are</a></li>
<li class="chapter" data-level="15.3.2" data-path="understanding-the-linear-model-with-crossed-factors-1.html"><a href="understanding-the-linear-model-with-crossed-factors-1.html#the-interaction-effect-is-something-different"><i class="fa fa-check"></i><b>15.3.2</b> The interaction effect is something different</a></li>
<li class="chapter" data-level="15.3.3" data-path="understanding-the-linear-model-with-crossed-factors-1.html"><a href="understanding-the-linear-model-with-crossed-factors-1.html#why-we-want-to-compare-the-treatment-effect-to-a-control-effect"><i class="fa fa-check"></i><b>15.3.3</b> Why we want to compare the treatment effect to a control effect</a></li>
<li class="chapter" data-level="15.3.4" data-path="understanding-the-linear-model-with-crossed-factors-1.html"><a href="understanding-the-linear-model-with-crossed-factors-1.html#the-order-of-the-factors-in-the-model-tells-the-same-story-differently"><i class="fa fa-check"></i><b>15.3.4</b> The order of the factors in the model tells the same story differently</a></li>
<li class="chapter" data-level="15.3.5" data-path="understanding-the-linear-model-with-crossed-factors-1.html"><a href="understanding-the-linear-model-with-crossed-factors-1.html#power-for-the-interaction-effect-is-less-than-that-for-simple-effects"><i class="fa fa-check"></i><b>15.3.5</b> Power for the interaction effect is less than that for simple effects</a></li>
<li class="chapter" data-level="15.3.6" data-path="understanding-the-linear-model-with-crossed-factors-1.html"><a href="understanding-the-linear-model-with-crossed-factors-1.html#planned-comparisons-vs.-post-hoc-tests"><i class="fa fa-check"></i><b>15.3.6</b> Planned comparisons vs. post-hoc tests</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data.html"><a href="example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data.html"><i class="fa fa-check"></i><b>15.4</b> Example 2: Estimation of the effect of background condition on an effect (“it depends”) (Experiment 3e lesian area data)</a>
<ul>
<li class="chapter" data-level="15.4.1" data-path="example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data.html"><a href="example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data.html#understand-the-experimental-design"><i class="fa fa-check"></i><b>15.4.1</b> Understand the experimental design</a></li>
<li class="chapter" data-level="15.4.2" data-path="example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data.html"><a href="example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data.html#fit-the-linear-model-1"><i class="fa fa-check"></i><b>15.4.2</b> Fit the linear model</a></li>
<li class="chapter" data-level="15.4.3" data-path="example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data.html"><a href="example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data.html#check-the-model-1"><i class="fa fa-check"></i><b>15.4.3</b> Check the model</a></li>
<li class="chapter" data-level="15.4.4" data-path="example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data.html"><a href="example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data.html#inference-from-the-model-2"><i class="fa fa-check"></i><b>15.4.4</b> Inference from the model</a></li>
<li class="chapter" data-level="15.4.5" data-path="example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data.html"><a href="example-2-estimation-of-the-effect-of-background-condition-on-an-effect-it-depends-experiment-3e-lesian-area-data.html#plot-the-model-4"><i class="fa fa-check"></i><b>15.4.5</b> Plot the model</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="understanding-the-linear-model-with-crossed-factors-2.html"><a href="understanding-the-linear-model-with-crossed-factors-2.html"><i class="fa fa-check"></i><b>15.5</b> Understanding the linear model with crossed factors 2</a>
<ul>
<li class="chapter" data-level="15.5.1" data-path="understanding-the-linear-model-with-crossed-factors-2.html"><a href="understanding-the-linear-model-with-crossed-factors-2.html#twoway-marginal-means"><i class="fa fa-check"></i><b>15.5.1</b> Conditional and marginal means</a></li>
<li class="chapter" data-level="15.5.2" data-path="understanding-the-linear-model-with-crossed-factors-2.html"><a href="understanding-the-linear-model-with-crossed-factors-2.html#simple-conditional-effects"><i class="fa fa-check"></i><b>15.5.2</b> Simple (conditional) effects</a></li>
<li class="chapter" data-level="15.5.3" data-path="understanding-the-linear-model-with-crossed-factors-2.html"><a href="understanding-the-linear-model-with-crossed-factors-2.html#twoway-marginal-effects"><i class="fa fa-check"></i><b>15.5.3</b> Marginal effects</a></li>
<li class="chapter" data-level="15.5.4" data-path="understanding-the-linear-model-with-crossed-factors-2.html"><a href="understanding-the-linear-model-with-crossed-factors-2.html#the-additive-model"><i class="fa fa-check"></i><b>15.5.4</b> The additive model</a></li>
<li class="chapter" data-level="15.5.5" data-path="understanding-the-linear-model-with-crossed-factors-2.html"><a href="understanding-the-linear-model-with-crossed-factors-2.html#twoway-reduce"><i class="fa fa-check"></i><b>15.5.5</b> Reduce models for the right reason</a></li>
<li class="chapter" data-level="15.5.6" data-path="understanding-the-linear-model-with-crossed-factors-2.html"><a href="understanding-the-linear-model-with-crossed-factors-2.html#the-marginal-means-of-an-additive-linear-model-with-two-factors-can-be-weird"><i class="fa fa-check"></i><b>15.5.6</b> The marginal means of an additive linear model with two factors can be weird</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data.html"><a href="example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data.html"><i class="fa fa-check"></i><b>15.6</b> Example 3: Estimation of synergy (“More than the sum of the parts”) (Experiment 1c JA data)</a>
<ul>
<li class="chapter" data-level="15.6.1" data-path="example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data.html"><a href="example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data.html#examine-the-data"><i class="fa fa-check"></i><b>15.6.1</b> Examine the data</a></li>
<li class="chapter" data-level="15.6.2" data-path="example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data.html"><a href="example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data.html#fit-the-model-7"><i class="fa fa-check"></i><b>15.6.2</b> Fit the model</a></li>
<li class="chapter" data-level="15.6.3" data-path="example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data.html"><a href="example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data.html#model-check"><i class="fa fa-check"></i><b>15.6.3</b> Model check</a></li>
<li class="chapter" data-level="15.6.4" data-path="example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data.html"><a href="example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data.html#inference-from-the-model-3"><i class="fa fa-check"></i><b>15.6.4</b> Inference from the model</a></li>
<li class="chapter" data-level="15.6.5" data-path="example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data.html"><a href="example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data.html#plot-the-model-5"><i class="fa fa-check"></i><b>15.6.5</b> Plot the model</a></li>
<li class="chapter" data-level="15.6.6" data-path="example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data.html"><a href="example-3-estimation-of-synergy-more-than-the-sum-of-the-parts-experiment-1c-ja-data.html#alternative-plot"><i class="fa fa-check"></i><b>15.6.6</b> Alternative plot</a></li>
</ul></li>
<li class="chapter" data-level="15.7" data-path="understanding-the-linear-model-with-crossed-factors-3.html"><a href="understanding-the-linear-model-with-crossed-factors-3.html"><i class="fa fa-check"></i><b>15.7</b> Understanding the linear model with crossed factors 3</a>
<ul>
<li class="chapter" data-level="15.7.1" data-path="understanding-the-linear-model-with-crossed-factors-3.html"><a href="understanding-the-linear-model-with-crossed-factors-3.html#thinking-about-the-coefficients-of-the-linear-model"><i class="fa fa-check"></i><b>15.7.1</b> Thinking about the coefficients of the linear model</a></li>
</ul></li>
<li class="chapter" data-level="15.8" data-path="issues-in-inference.html"><a href="issues-in-inference.html"><i class="fa fa-check"></i><b>15.8</b> Issues in inference</a>
<ul>
<li class="chapter" data-level="15.8.1" data-path="issues-in-inference.html"><a href="issues-in-inference.html#for-pairwise-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model"><i class="fa fa-check"></i><b>15.8.1</b> For pairwise contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</a></li>
<li class="chapter" data-level="15.8.2" data-path="issues-in-inference.html"><a href="issues-in-inference.html#for-interaction-contrasts-it-doesnt-matter-if-you-fit-a-factorial-or-a-flattened-linear-model"><i class="fa fa-check"></i><b>15.8.2</b> For interaction contrasts, it doesn’t matter if you fit a factorial or a flattened linear model</a></li>
<li class="chapter" data-level="15.8.3" data-path="issues-in-inference.html"><a href="issues-in-inference.html#twoway-multiple-tests"><i class="fa fa-check"></i><b>15.8.3</b> Adjusting <em>p</em>-values for multiple tests</a></li>
</ul></li>
<li class="chapter" data-level="15.9" data-path="two-way-anova.html"><a href="two-way-anova.html"><i class="fa fa-check"></i><b>15.9</b> Two-way ANOVA</a>
<ul>
<li class="chapter" data-level="15.9.1" data-path="two-way-anova.html"><a href="two-way-anova.html#how-to-read-a-two-way-anova-table"><i class="fa fa-check"></i><b>15.9.1</b> How to read a two-way ANOVA table</a></li>
<li class="chapter" data-level="15.9.2" data-path="two-way-anova.html"><a href="two-way-anova.html#what-do-the-main-effects-in-an-anova-table-mean"><i class="fa fa-check"></i><b>15.9.2</b> What do the main effects in an ANOVA table mean?</a></li>
</ul></li>
<li class="chapter" data-level="15.10" data-path="more-issues-in-inference.html"><a href="more-issues-in-inference.html"><i class="fa fa-check"></i><b>15.10</b> More issues in inference</a>
<ul>
<li class="chapter" data-level="15.10.1" data-path="more-issues-in-inference.html"><a href="more-issues-in-inference.html#longitudinal-experiments-include-time-as-a-random-factor-better-than-repeated-measures-anova"><i class="fa fa-check"></i><b>15.10.1</b> Longitudinal experiments – include Time as a random factor (better than repeated measures ANOVA)</a></li>
</ul></li>
<li class="chapter" data-level="15.11" data-path="working-in-r-4.html"><a href="working-in-r-4.html"><i class="fa fa-check"></i><b>15.11</b> Working in R</a>
<ul>
<li class="chapter" data-level="15.11.1" data-path="working-in-r-4.html"><a href="working-in-r-4.html#model-formula"><i class="fa fa-check"></i><b>15.11.1</b> Model formula</a></li>
<li class="chapter" data-level="15.11.2" data-path="working-in-r-4.html"><a href="working-in-r-4.html#using-the-emmeans-function-2"><i class="fa fa-check"></i><b>15.11.2</b> Using the emmeans function</a></li>
<li class="chapter" data-level="15.11.3" data-path="working-in-r-4.html"><a href="working-in-r-4.html#contrasts"><i class="fa fa-check"></i><b>15.11.3</b> Contrasts</a></li>
<li class="chapter" data-level="15.11.4" data-path="working-in-r-4.html"><a href="working-in-r-4.html#practice-safe-anova"><i class="fa fa-check"></i><b>15.11.4</b> Practice safe ANOVA</a></li>
<li class="chapter" data-level="15.11.5" data-path="working-in-r-4.html"><a href="working-in-r-4.html#better-to-avoid-these"><i class="fa fa-check"></i><b>15.11.5</b> Better to avoid these</a></li>
</ul></li>
<li class="chapter" data-level="15.12" data-path="hidden-code-6.html"><a href="hidden-code-6.html"><i class="fa fa-check"></i><b>15.12</b> Hidden Code</a>
<ul>
<li class="chapter" data-level="15.12.1" data-path="hidden-code-6.html"><a href="hidden-code-6.html#import-exp2j-example-1"><i class="fa fa-check"></i><b>15.12.1</b> Import exp2j (Example 1)</a></li>
<li class="chapter" data-level="15.12.2" data-path="hidden-code-6.html"><a href="hidden-code-6.html#import-exp3e-lesian-area-data-example-2"><i class="fa fa-check"></i><b>15.12.2</b> Import exp3e lesian area data (Example 2)</a></li>
<li class="chapter" data-level="15.12.3" data-path="hidden-code-6.html"><a href="hidden-code-6.html#import-exp1c-ja-data-example-3"><i class="fa fa-check"></i><b>15.12.3</b> Import Exp1c JA data (Example 3)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="part-vi-expanding-the-linear-model.html"><a href="part-vi-expanding-the-linear-model.html"><i class="fa fa-check"></i>Part VI – Expanding the Linear Model</a></li>
<li class="chapter" data-level="16" data-path="lmm.html"><a href="lmm.html"><i class="fa fa-check"></i><b>16</b> Models for non-independence – linear mixed models</a>
<ul>
<li class="chapter" data-level="16.1" data-path="liberal-inference-from-pseudoreplication.html"><a href="liberal-inference-from-pseudoreplication.html"><i class="fa fa-check"></i><b>16.1</b> Liberal inference from pseudoreplication</a></li>
<li class="chapter" data-level="16.2" data-path="conservative-inference-from-failure-to-identify-blocks.html"><a href="conservative-inference-from-failure-to-identify-blocks.html"><i class="fa fa-check"></i><b>16.2</b> Conservative inference from failure to identify blocks</a></li>
<li class="chapter" data-level="16.3" data-path="introduction-to-models-for-non-independent-data-linear-mixed-models.html"><a href="introduction-to-models-for-non-independent-data-linear-mixed-models.html"><i class="fa fa-check"></i><b>16.3</b> Introduction to models for non-independent data (linear mixed models)</a></li>
<li class="chapter" data-level="16.4" data-path="experimental-designs-in-experimental-bench-biology.html"><a href="experimental-designs-in-experimental-bench-biology.html"><i class="fa fa-check"></i><b>16.4</b> Experimental designs in experimental bench biology</a>
<ul>
<li class="chapter" data-level="16.4.1" data-path="experimental-designs-in-experimental-bench-biology.html"><a href="experimental-designs-in-experimental-bench-biology.html#notation-for-models"><i class="fa fa-check"></i><b>16.4.1</b> Notation for models</a></li>
<li class="chapter" data-level="16.4.2" data-path="experimental-designs-in-experimental-bench-biology.html"><a href="experimental-designs-in-experimental-bench-biology.html#completely-randomized-design-crd"><i class="fa fa-check"></i><b>16.4.2</b> Completely Randomized Design (CRD)</a></li>
<li class="chapter" data-level="16.4.3" data-path="experimental-designs-in-experimental-bench-biology.html"><a href="experimental-designs-in-experimental-bench-biology.html#completely-randomized-design-with-subsampling-crds-or-nested-design"><i class="fa fa-check"></i><b>16.4.3</b> Completely Randomized Design with Subsampling (CRDS) or “Nested Design”</a></li>
<li class="chapter" data-level="16.4.4" data-path="experimental-designs-in-experimental-bench-biology.html"><a href="experimental-designs-in-experimental-bench-biology.html#randomized-complete-block-design-rcbd"><i class="fa fa-check"></i><b>16.4.4</b> Randomized Complete Block Design (RCBD)</a></li>
<li class="chapter" data-level="16.4.5" data-path="experimental-designs-in-experimental-bench-biology.html"><a href="experimental-designs-in-experimental-bench-biology.html#randomized-split-plot-design-rspd"><i class="fa fa-check"></i><b>16.4.5</b> Randomized Split Plot Design (RSPD)</a></li>
<li class="chapter" data-level="16.4.6" data-path="experimental-designs-in-experimental-bench-biology.html"><a href="experimental-designs-in-experimental-bench-biology.html#generalized-randomized-complete-block-design-grcbd"><i class="fa fa-check"></i><b>16.4.6</b> Generalized Randomized Complete Block Design (GRCBD)</a></li>
<li class="chapter" data-level="16.4.7" data-path="experimental-designs-in-experimental-bench-biology.html"><a href="experimental-designs-in-experimental-bench-biology.html#nested-randomized-complete-block-design-nrcbd"><i class="fa fa-check"></i><b>16.4.7</b> Nested Randomized Complete Block Design (NRCBD)</a></li>
<li class="chapter" data-level="16.4.8" data-path="experimental-designs-in-experimental-bench-biology.html"><a href="experimental-designs-in-experimental-bench-biology.html#longitudinal-randomized-complete-block-design-lrcbd"><i class="fa fa-check"></i><b>16.4.8</b> Longitudinal Randomized Complete Block Design (LRCBD)</a></li>
<li class="chapter" data-level="16.4.9" data-path="experimental-designs-in-experimental-bench-biology.html"><a href="experimental-designs-in-experimental-bench-biology.html#variations-due-to-multiple-measures-of-the-response-variable"><i class="fa fa-check"></i><b>16.4.9</b> Variations due to multiple measures of the response variable</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="building-the-linear-mixed-model-for-clustered-data.html"><a href="building-the-linear-mixed-model-for-clustered-data.html"><i class="fa fa-check"></i><b>16.5</b> Building the linear (mixed) model for clustered data</a></li>
<li class="chapter" data-level="16.6" data-path="lmm-demo.html"><a href="lmm-demo.html"><i class="fa fa-check"></i><b>16.6</b> Example 1 – A random intercepts and slopes explainer (demo1)</a>
<ul>
<li class="chapter" data-level="16.6.1" data-path="lmm-demo.html"><a href="lmm-demo.html#batched-measurements-result-in-clustered-residuals"><i class="fa fa-check"></i><b>16.6.1</b> Batched measurements result in clustered residuals</a></li>
<li class="chapter" data-level="16.6.2" data-path="lmm-demo.html"><a href="lmm-demo.html#lmm-corerr"><i class="fa fa-check"></i><b>16.6.2</b> Clustered residuals result in correlated error</a></li>
<li class="chapter" data-level="16.6.3" data-path="lmm-demo.html"><a href="lmm-demo.html#in-blocked-designs-clustered-residuals-adds-a-variance-component-that-masks-treatment-effects"><i class="fa fa-check"></i><b>16.6.3</b> In blocked designs, clustered residuals adds a variance component that masks treatment effects</a></li>
<li class="chapter" data-level="16.6.4" data-path="lmm-demo.html"><a href="lmm-demo.html#linear-mixed-models-are-linear-models-with-added-random-factors"><i class="fa fa-check"></i><b>16.6.4</b> Linear mixed models are linear models with added random factors</a></li>
<li class="chapter" data-level="16.6.5" data-path="lmm-demo.html"><a href="lmm-demo.html#what-the-random-effects-are"><i class="fa fa-check"></i><b>16.6.5</b> What the random effects are</a></li>
<li class="chapter" data-level="16.6.6" data-path="lmm-demo.html"><a href="lmm-demo.html#in-a-blocked-design-a-linear-model-with-added-random-effects-increases-precision-of-treatment-effects"><i class="fa fa-check"></i><b>16.6.6</b> In a blocked design, a linear model with added random effects increases precision of treatment effects</a></li>
<li class="chapter" data-level="16.6.7" data-path="lmm-demo.html"><a href="lmm-demo.html#lmm-varcorr"><i class="fa fa-check"></i><b>16.6.7</b> The correlation among random intercepts and slopes</a></li>
<li class="chapter" data-level="16.6.8" data-path="lmm-demo.html"><a href="lmm-demo.html#clustered-residuals-create-heterogeneity-among-treatments"><i class="fa fa-check"></i><b>16.6.8</b> Clustered residuals create heterogeneity among treatments</a></li>
<li class="chapter" data-level="16.6.9" data-path="lmm-demo.html"><a href="lmm-demo.html#linear-mixed-models-are-flexible"><i class="fa fa-check"></i><b>16.6.9</b> Linear mixed models are flexible</a></li>
<li class="chapter" data-level="16.6.10" data-path="lmm-demo.html"><a href="lmm-demo.html#a-random-intercept-only-model"><i class="fa fa-check"></i><b>16.6.10</b> A random intercept only model</a></li>
<li class="chapter" data-level="16.6.11" data-path="lmm-demo.html"><a href="lmm-demo.html#a-model-including-an-interaction-intercept"><i class="fa fa-check"></i><b>16.6.11</b> A model including an interaction intercept</a></li>
<li class="chapter" data-level="16.6.12" data-path="lmm-demo.html"><a href="lmm-demo.html#lmm-demo-aic"><i class="fa fa-check"></i><b>16.6.12</b> AIC and model selection – which model to report?</a></li>
<li class="chapter" data-level="16.6.13" data-path="lmm-demo.html"><a href="lmm-demo.html#the-specification-of-random-effects-matters"><i class="fa fa-check"></i><b>16.6.13</b> The specification of random effects matters</a></li>
<li class="chapter" data-level="16.6.14" data-path="lmm-demo.html"><a href="lmm-demo.html#mixed-effect-and-repeated-measures-anova"><i class="fa fa-check"></i><b>16.6.14</b> Mixed Effect and Repeated Measures ANOVA</a></li>
<li class="chapter" data-level="16.6.15" data-path="lmm-demo.html"><a href="lmm-demo.html#pseudoreplication"><i class="fa fa-check"></i><b>16.6.15</b> Pseudoreplication</a></li>
</ul></li>
<li class="chapter" data-level="16.7" data-path="lmm-example2.html"><a href="lmm-example2.html"><i class="fa fa-check"></i><b>16.7</b> Example 2 – experiments without subsampling replication (exp6g)</a>
<ul>
<li class="chapter" data-level="16.7.1" data-path="lmm-example2.html"><a href="lmm-example2.html#understand-the-data"><i class="fa fa-check"></i><b>16.7.1</b> Understand the data</a></li>
<li class="chapter" data-level="16.7.2" data-path="lmm-example2.html"><a href="lmm-example2.html#model-fit-and-inference"><i class="fa fa-check"></i><b>16.7.2</b> Model fit and inference</a></li>
<li class="chapter" data-level="16.7.3" data-path="lmm-example2.html"><a href="lmm-example2.html#the-model-exp6g_m1-adds-a-random-intercept-but-not-a-random-slope"><i class="fa fa-check"></i><b>16.7.3</b> The model exp6g_m1 adds a random intercept but not a random slope</a></li>
<li class="chapter" data-level="16.7.4" data-path="lmm-example2.html"><a href="lmm-example2.html#the-fixed-effect-coefficients-of-model-exp6g_m1"><i class="fa fa-check"></i><b>16.7.4</b> The fixed effect coefficients of model exp6g_m1</a></li>
<li class="chapter" data-level="16.7.5" data-path="lmm-example2.html"><a href="lmm-example2.html#the-random-intercept-coefficients-of-exp6g_m1"><i class="fa fa-check"></i><b>16.7.5</b> The random intercept coefficients of exp6g_m1</a></li>
<li class="chapter" data-level="16.7.6" data-path="lmm-example2.html"><a href="lmm-example2.html#the-random-and-residual-variance-and-the-intraclass-correlation-of-model-exp6g_m1"><i class="fa fa-check"></i><b>16.7.6</b> The random and residual variance and the intraclass correlation of model exp6g_m1</a></li>
<li class="chapter" data-level="16.7.7" data-path="lmm-example2.html"><a href="lmm-example2.html#the-linear-mixed-model-exp6g_m1-increases-precision-of-treatment-effects-relative-to-a-fixed-effects-model"><i class="fa fa-check"></i><b>16.7.7</b> The linear mixed model exp6g_m1 increases precision of treatment effects, relative to a fixed effects model</a></li>
<li class="chapter" data-level="16.7.8" data-path="lmm-example2.html"><a href="lmm-example2.html#lmm-exp6g-alt"><i class="fa fa-check"></i><b>16.7.8</b> Alternative models for exp6g</a></li>
<li class="chapter" data-level="16.7.9" data-path="lmm-example2.html"><a href="lmm-example2.html#paired-t-tests-and-repeated-measures-anova-are-special-cases-of-linear-mixed-models"><i class="fa fa-check"></i><b>16.7.9</b> Paired t-tests and repeated measures ANOVA are special cases of linear mixed models</a></li>
<li class="chapter" data-level="16.7.10" data-path="lmm-example2.html"><a href="lmm-example2.html#lmm-exp6g-rmanova"><i class="fa fa-check"></i><b>16.7.10</b> Classical (“univariate model”) repeated measures ANOVA of exp6g</a></li>
<li class="chapter" data-level="16.7.11" data-path="lmm-example2.html"><a href="lmm-example2.html#lmm-exp6g-rmanova-multi"><i class="fa fa-check"></i><b>16.7.11</b> “Multivariate model” repeated measures ANOVA</a></li>
<li class="chapter" data-level="16.7.12" data-path="lmm-example2.html"><a href="lmm-example2.html#linear-mixed-models-vs-repeated-measures-anova"><i class="fa fa-check"></i><b>16.7.12</b> Linear mixed models vs repeated measures ANOVA</a></li>
<li class="chapter" data-level="16.7.13" data-path="lmm-example2.html"><a href="lmm-example2.html#modeling-textttmouse_id-as-a-fixed-effect"><i class="fa fa-check"></i><b>16.7.13</b> Modeling <span class="math inline">\(\texttt{mouse_id}\)</span> as a fixed effect</a></li>
</ul></li>
<li class="chapter" data-level="16.8" data-path="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html"><i class="fa fa-check"></i><b>16.8</b> Example 3 – Factorial experiments and no subsampling replicates (exp5c)</a>
<ul>
<li class="chapter" data-level="16.8.1" data-path="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#understand-the-data-1"><i class="fa fa-check"></i><b>16.8.1</b> Understand the data</a></li>
<li class="chapter" data-level="16.8.2" data-path="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#examine-the-data-1"><i class="fa fa-check"></i><b>16.8.2</b> Examine the data</a></li>
<li class="chapter" data-level="16.8.3" data-path="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#lmm-exp5c-m1"><i class="fa fa-check"></i><b>16.8.3</b> Model fit and inference</a></li>
<li class="chapter" data-level="16.8.4" data-path="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#why-we-care-about-modeling-batch-in-exp5c"><i class="fa fa-check"></i><b>16.8.4</b> Why we care about modeling batch in exp5c</a></li>
<li class="chapter" data-level="16.8.5" data-path="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts"><i class="fa fa-check"></i><b>16.8.5</b> The linear mixed model exp5c_m1 adds two random intercepts</a></li>
<li class="chapter" data-level="16.8.6" data-path="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#the-fixed-effect-coefficients-of-model-exp5c_m1"><i class="fa fa-check"></i><b>16.8.6</b> The fixed effect coefficients of model exp5c_m1</a></li>
<li class="chapter" data-level="16.8.7" data-path="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#the-random-effect-coefficients-of-model-exp5c_m1"><i class="fa fa-check"></i><b>16.8.7</b> The random effect coefficients of model exp5c_m1</a></li>
<li class="chapter" data-level="16.8.8" data-path="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#lmm-exp5c-alt"><i class="fa fa-check"></i><b>16.8.8</b> Alternative models for exp5c</a></li>
<li class="chapter" data-level="16.8.9" data-path="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#lmm-rmanova-5c"><i class="fa fa-check"></i><b>16.8.9</b> Classical (“univariate model”) repeated measures ANOVA</a></li>
<li class="chapter" data-level="16.8.10" data-path="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#lmm-rmanova-5c-multi"><i class="fa fa-check"></i><b>16.8.10</b> “Multivariate model” repeated measures ANOVA of exp5c</a></li>
<li class="chapter" data-level="16.8.11" data-path="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#modeling-textttdonor-as-a-fixed-effect"><i class="fa fa-check"></i><b>16.8.11</b> Modeling <span class="math inline">\(\texttt{donor}\)</span> as a fixed effect</a></li>
</ul></li>
<li class="chapter" data-level="16.9" data-path="lmm-example4.html"><a href="lmm-example4.html"><i class="fa fa-check"></i><b>16.9</b> Example 4 – Experiments with subsampling replication (exp1g)</a>
<ul>
<li class="chapter" data-level="16.9.1" data-path="lmm-example4.html"><a href="lmm-example4.html#understand-the-data-2"><i class="fa fa-check"></i><b>16.9.1</b> Understand the data</a></li>
<li class="chapter" data-level="16.9.2" data-path="lmm-example4.html"><a href="lmm-example4.html#examine-the-data-2"><i class="fa fa-check"></i><b>16.9.2</b> Examine the data</a></li>
<li class="chapter" data-level="16.9.3" data-path="lmm-example4.html"><a href="lmm-example4.html#fit-the-model-9"><i class="fa fa-check"></i><b>16.9.3</b> Fit the model</a></li>
<li class="chapter" data-level="16.9.4" data-path="lmm-example4.html"><a href="lmm-example4.html#inference-from-the-model-6"><i class="fa fa-check"></i><b>16.9.4</b> Inference from the model</a></li>
<li class="chapter" data-level="16.9.5" data-path="lmm-example4.html"><a href="lmm-example4.html#plot-the-model-6"><i class="fa fa-check"></i><b>16.9.5</b> Plot the model</a></li>
<li class="chapter" data-level="16.9.6" data-path="lmm-example4.html"><a href="lmm-example4.html#alternaplot-the-model-3"><i class="fa fa-check"></i><b>16.9.6</b> Alternaplot the model</a></li>
<li class="chapter" data-level="16.9.7" data-path="lmm-example4.html"><a href="lmm-example4.html#understanding-the-alternative-models"><i class="fa fa-check"></i><b>16.9.7</b> Understanding the alternative models</a></li>
<li class="chapter" data-level="16.9.8" data-path="lmm-example4.html"><a href="lmm-example4.html#the-varcorr-matrix-of-models-exp1g_m1a-and-exp1g_m1b"><i class="fa fa-check"></i><b>16.9.8</b> The VarCorr matrix of models exp1g_m1a and exp1g_m1b</a></li>
<li class="chapter" data-level="16.9.9" data-path="lmm-example4.html"><a href="lmm-example4.html#the-linear-mixed-model-has-more-precision-and-power-than-the-fixed-effect-model-of-batch-means"><i class="fa fa-check"></i><b>16.9.9</b> The linear mixed model has more precision and power than the fixed effect model of batch means</a></li>
<li class="chapter" data-level="16.9.10" data-path="lmm-example4.html"><a href="lmm-example4.html#fixed-effect-models-and-pseudoreplication"><i class="fa fa-check"></i><b>16.9.10</b> Fixed effect models and pseudoreplication</a></li>
<li class="chapter" data-level="16.9.11" data-path="lmm-example4.html"><a href="lmm-example4.html#mixed-effect-anova"><i class="fa fa-check"></i><b>16.9.11</b> Mixed-effect ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="16.10" data-path="statistical-models-for-experimental-designs.html"><a href="statistical-models-for-experimental-designs.html"><i class="fa fa-check"></i><b>16.10</b> Statistical models for experimental designs</a>
<ul>
<li class="chapter" data-level="16.10.1" data-path="statistical-models-for-experimental-designs.html"><a href="statistical-models-for-experimental-designs.html#models-for-completely-randomized-designs-crd"><i class="fa fa-check"></i><b>16.10.1</b> Models for Completely Randomized Designs (CRD)</a></li>
<li class="chapter" data-level="16.10.2" data-path="statistical-models-for-experimental-designs.html"><a href="statistical-models-for-experimental-designs.html#models-for-batched-data-crds-rcbd-rspd-grcbd-nrcbd"><i class="fa fa-check"></i><b>16.10.2</b> Models for batched data (CRDS, RCBD, RSPD, GRCBD, NRCBD)</a></li>
</ul></li>
<li class="chapter" data-level="16.11" data-path="which-model-and-why.html"><a href="which-model-and-why.html"><i class="fa fa-check"></i><b>16.11</b> Which model and why?</a>
<ul>
<li class="chapter" data-level="16.11.1" data-path="which-model-and-why.html"><a href="which-model-and-why.html#crd"><i class="fa fa-check"></i><b>16.11.1</b> CRD</a></li>
<li class="chapter" data-level="16.11.2" data-path="which-model-and-why.html"><a href="which-model-and-why.html#crds"><i class="fa fa-check"></i><b>16.11.2</b> CRDS</a></li>
<li class="chapter" data-level="16.11.3" data-path="which-model-and-why.html"><a href="which-model-and-why.html#rcbd"><i class="fa fa-check"></i><b>16.11.3</b> RCBD</a></li>
<li class="chapter" data-level="16.11.4" data-path="which-model-and-why.html"><a href="which-model-and-why.html#rspd"><i class="fa fa-check"></i><b>16.11.4</b> RSPD</a></li>
<li class="chapter" data-level="16.11.5" data-path="which-model-and-why.html"><a href="which-model-and-why.html#rcbds"><i class="fa fa-check"></i><b>16.11.5</b> RCBDS</a></li>
<li class="chapter" data-level="16.11.6" data-path="which-model-and-why.html"><a href="which-model-and-why.html#grcbd"><i class="fa fa-check"></i><b>16.11.6</b> GRCBD</a></li>
<li class="chapter" data-level="16.11.7" data-path="which-model-and-why.html"><a href="which-model-and-why.html#grcbds"><i class="fa fa-check"></i><b>16.11.7</b> GRCBDS</a></li>
</ul></li>
<li class="chapter" data-level="16.12" data-path="working-in-r-5.html"><a href="working-in-r-5.html"><i class="fa fa-check"></i><b>16.12</b> Working in R</a>
<ul>
<li class="chapter" data-level="16.12.1" data-path="working-in-r-5.html"><a href="working-in-r-5.html#plotting-models-fit-to-batched-data"><i class="fa fa-check"></i><b>16.12.1</b> Plotting models fit to batched data</a></li>
<li class="chapter" data-level="16.12.2" data-path="working-in-r-5.html"><a href="working-in-r-5.html#repeated-measures-anova-randomized-complete-block-with-no-subsampling"><i class="fa fa-check"></i><b>16.12.2</b> Repeated measures ANOVA (randomized complete block with no subsampling)</a></li>
</ul></li>
<li class="chapter" data-level="16.13" data-path="hidden-code-7.html"><a href="hidden-code-7.html"><i class="fa fa-check"></i><b>16.13</b> Hidden code</a>
<ul>
<li class="chapter" data-level="16.13.1" data-path="hidden-code-7.html"><a href="hidden-code-7.html#import-exp5c"><i class="fa fa-check"></i><b>16.13.1</b> Import exp5c</a></li>
<li class="chapter" data-level="16.13.2" data-path="hidden-code-7.html"><a href="hidden-code-7.html#import-exp1g"><i class="fa fa-check"></i><b>16.13.2</b> Import exp1g</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="pre-post.html"><a href="pre-post.html"><i class="fa fa-check"></i><b>17</b> Linear models for longitudinal experiments – I. pre-post designs</a>
<ul>
<li class="chapter" data-level="17.1" data-path="best-practice-models.html"><a href="best-practice-models.html"><i class="fa fa-check"></i><b>17.1</b> Best practice models</a></li>
<li class="chapter" data-level="17.2" data-path="common-alternatives-that-are-not-recommended.html"><a href="common-alternatives-that-are-not-recommended.html"><i class="fa fa-check"></i><b>17.2</b> Common alternatives that are not recommended</a></li>
<li class="chapter" data-level="17.3" data-path="advanced-models.html"><a href="advanced-models.html"><i class="fa fa-check"></i><b>17.3</b> Advanced models</a></li>
<li class="chapter" data-level="17.4" data-path="understanding-the-alternative-models-1.html"><a href="understanding-the-alternative-models-1.html"><i class="fa fa-check"></i><b>17.4</b> Understanding the alternative models</a>
<ul>
<li class="chapter" data-level="17.4.1" data-path="understanding-the-alternative-models-1.html"><a href="understanding-the-alternative-models-1.html#m1-linear-model-with-the-baseline-measure-as-the-covariate-ancova-model"><i class="fa fa-check"></i><b>17.4.1</b> (M1) Linear model with the baseline measure as the covariate (ANCOVA model)</a></li>
<li class="chapter" data-level="17.4.2" data-path="understanding-the-alternative-models-1.html"><a href="understanding-the-alternative-models-1.html#m2-linear-model-of-the-change-score-change-score-model"><i class="fa fa-check"></i><b>17.4.2</b> (M2) Linear model of the change score (change-score model)</a></li>
<li class="chapter" data-level="17.4.3" data-path="understanding-the-alternative-models-1.html"><a href="understanding-the-alternative-models-1.html#m3-linear-model-of-post-baseline-values-without-the-baseline-as-a-covariate-post-model"><i class="fa fa-check"></i><b>17.4.3</b> (M3) Linear model of post-baseline values without the baseline as a covariate (post model)</a></li>
<li class="chapter" data-level="17.4.4" data-path="understanding-the-alternative-models-1.html"><a href="understanding-the-alternative-models-1.html#m4-linear-model-with-factorial-fixed-effects-fixed-effects-model"><i class="fa fa-check"></i><b>17.4.4</b> (M4) Linear model with factorial fixed effects (fixed-effects model)</a></li>
<li class="chapter" data-level="17.4.5" data-path="understanding-the-alternative-models-1.html"><a href="understanding-the-alternative-models-1.html#m5-repeated-measures-anova"><i class="fa fa-check"></i><b>17.4.5</b> (M5) Repeated measures ANOVA</a></li>
<li class="chapter" data-level="17.4.6" data-path="understanding-the-alternative-models-1.html"><a href="understanding-the-alternative-models-1.html#m6-linear-mixed-model"><i class="fa fa-check"></i><b>17.4.6</b> (M6) Linear mixed model</a></li>
<li class="chapter" data-level="17.4.7" data-path="understanding-the-alternative-models-1.html"><a href="understanding-the-alternative-models-1.html#m7-linear-model-with-correlated-error"><i class="fa fa-check"></i><b>17.4.7</b> (M7) Linear model with correlated error</a></li>
<li class="chapter" data-level="17.4.8" data-path="understanding-the-alternative-models-1.html"><a href="understanding-the-alternative-models-1.html#m8-constrained-fixed-effects-model-with-correlated-error-clda-model"><i class="fa fa-check"></i><b>17.4.8</b> (M8) Constrained fixed effects model with correlated error (cLDA model)</a></li>
<li class="chapter" data-level="17.4.9" data-path="understanding-the-alternative-models-1.html"><a href="understanding-the-alternative-models-1.html#comparison-table"><i class="fa fa-check"></i><b>17.4.9</b> Comparison table</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="example-1-a-single-post-baseline-measure-pre-post-design.html"><a href="example-1-a-single-post-baseline-measure-pre-post-design.html"><i class="fa fa-check"></i><b>17.5</b> Example 1 – a single post-baseline measure (pre-post design)</a></li>
<li class="chapter" data-level="17.6" data-path="working-in-r-6.html"><a href="working-in-r-6.html"><i class="fa fa-check"></i><b>17.6</b> Working in R</a></li>
<li class="chapter" data-level="17.7" data-path="hidden-code-8.html"><a href="hidden-code-8.html"><i class="fa fa-check"></i><b>17.7</b> Hidden code</a>
<ul>
<li class="chapter" data-level="17.7.1" data-path="hidden-code-8.html"><a href="hidden-code-8.html#import-and-wrangle-mouse-sociability-data"><i class="fa fa-check"></i><b>17.7.1</b> Import and wrangle mouse sociability data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="counts.html"><a href="counts.html"><i class="fa fa-check"></i><b>18</b> Linear models for counts, binary responses, skewed responses, and ratios – Generalized Linear Models</a>
<ul>
<li class="chapter" data-level="18.1" data-path="glm-counts.html"><a href="glm-counts.html"><i class="fa fa-check"></i><b>18.1</b> Introducing Generalized Linear Models using count data examples</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="glm-counts.html"><a href="glm-counts.html#the-generalized-linear-model-glm"><i class="fa fa-check"></i><b>18.1.1</b> The Generalized Linear Model (GLM)</a></li>
<li class="chapter" data-level="18.1.2" data-path="glm-counts.html"><a href="glm-counts.html#kinds-of-data-that-are-modeled-by-a-glm"><i class="fa fa-check"></i><b>18.1.2</b> Kinds of data that are modeled by a GLM</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="glm-1.html"><a href="glm-1.html"><i class="fa fa-check"></i><b>18.2</b> Example 1 – GLM models for count responses (“angiogenic sprouts” fig3a)</a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="glm-1.html"><a href="glm-1.html#understand-the-data-3"><i class="fa fa-check"></i><b>18.2.1</b> Understand the data</a></li>
<li class="chapter" data-level="18.2.2" data-path="glm-1.html"><a href="glm-1.html#model-fit-and-inference-1"><i class="fa fa-check"></i><b>18.2.2</b> Model fit and inference</a></li>
</ul></li>
<li class="chapter" data-level="18.3" data-path="understanding-example-1.html"><a href="understanding-example-1.html"><i class="fa fa-check"></i><b>18.3</b> Understanding Example 1</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="understanding-example-1.html"><a href="understanding-example-1.html#modeling-strategy-1"><i class="fa fa-check"></i><b>18.3.1</b> Modeling strategy</a></li>
<li class="chapter" data-level="18.3.2" data-path="understanding-example-1.html"><a href="understanding-example-1.html#model-checking-fits-to-count-data"><i class="fa fa-check"></i><b>18.3.2</b> Model checking fits to count data</a></li>
<li class="chapter" data-level="18.3.3" data-path="understanding-example-1.html"><a href="understanding-example-1.html#biological-count-data-are-rarely-fit-well-by-a-poisson-glm.-instead-fit-a-quasi-poisson-or-negative-binomial-glm-model."><i class="fa fa-check"></i><b>18.3.3</b> Biological count data are rarely fit well by a Poisson GLM. Instead, fit a quasi-poisson or negative binomial GLM model.</a></li>
<li class="chapter" data-level="18.3.4" data-path="understanding-example-1.html"><a href="understanding-example-1.html#a-glm-is-a-linear-model-on-the-link-scale"><i class="fa fa-check"></i><b>18.3.4</b> A GLM is a linear model on the link scale</a></li>
<li class="chapter" data-level="18.3.5" data-path="understanding-example-1.html"><a href="understanding-example-1.html#coeffecients-of-a-generalized-linear-model-with-a-log-link-function-are-on-the-link-scale."><i class="fa fa-check"></i><b>18.3.5</b> Coeffecients of a Generalized Linear Model with a log-link function are on the link scale.</a></li>
<li class="chapter" data-level="18.3.6" data-path="understanding-example-1.html"><a href="understanding-example-1.html#modeled-means-in-the-emmeans-table-of-a-generalized-linear-model-can-be-on-the-link-scale-or-response-scale-report-the-response-scale"><i class="fa fa-check"></i><b>18.3.6</b> Modeled means in the emmeans table of a Generalized Linear Model can be on the link scale or response scale – Report the response scale</a></li>
<li class="chapter" data-level="18.3.7" data-path="understanding-example-1.html"><a href="understanding-example-1.html#some-consequences-of-fitting-a-linear-model-to-count-data"><i class="fa fa-check"></i><b>18.3.7</b> Some consequences of fitting a linear model to count data</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="example-2-use-a-glm-with-an-offset-instead-of-a-ratio-of-some-measurement-per-total-dna-damage-data-fig3b-glm_offset.html"><a href="example-2-use-a-glm-with-an-offset-instead-of-a-ratio-of-some-measurement-per-total-dna-damage-data-fig3b-glm_offset.html"><i class="fa fa-check"></i><b>18.4</b> Example 2 – Use a GLM with an offset instead of a ratio of some measurement per total (“dna damage” data fig3b) #glm_offset</a>
<ul>
<li class="chapter" data-level="18.4.1" data-path="example-2-use-a-glm-with-an-offset-instead-of-a-ratio-of-some-measurement-per-total-dna-damage-data-fig3b-glm_offset.html"><a href="example-2-use-a-glm-with-an-offset-instead-of-a-ratio-of-some-measurement-per-total-dna-damage-data-fig3b-glm_offset.html#fig3b-dna-damage-data"><i class="fa fa-check"></i><b>18.4.1</b> fig3b (“dna damage”) data</a></li>
<li class="chapter" data-level="18.4.2" data-path="example-2-use-a-glm-with-an-offset-instead-of-a-ratio-of-some-measurement-per-total-dna-damage-data-fig3b-glm_offset.html"><a href="example-2-use-a-glm-with-an-offset-instead-of-a-ratio-of-some-measurement-per-total-dna-damage-data-fig3b-glm_offset.html#understand-the-data-4"><i class="fa fa-check"></i><b>18.4.2</b> Understand the data</a></li>
<li class="chapter" data-level="18.4.3" data-path="example-2-use-a-glm-with-an-offset-instead-of-a-ratio-of-some-measurement-per-total-dna-damage-data-fig3b-glm_offset.html"><a href="example-2-use-a-glm-with-an-offset-instead-of-a-ratio-of-some-measurement-per-total-dna-damage-data-fig3b-glm_offset.html#model-fit-and-inference-2"><i class="fa fa-check"></i><b>18.4.3</b> Model fit and inference</a></li>
</ul></li>
<li class="chapter" data-level="18.5" data-path="understanding-example-2.html"><a href="understanding-example-2.html"><i class="fa fa-check"></i><b>18.5</b> Understanding Example 2</a>
<ul>
<li class="chapter" data-level="18.5.1" data-path="understanding-example-2.html"><a href="understanding-example-2.html#an-offset-is-an-added-covariate-with-a-coefficient-fixed-at-1"><i class="fa fa-check"></i><b>18.5.1</b> An offset is an added covariate with a coefficient fixed at 1</a></li>
<li class="chapter" data-level="18.5.2" data-path="understanding-example-2.html"><a href="understanding-example-2.html#glm-rate"><i class="fa fa-check"></i><b>18.5.2</b> A count GLM with an offset models the area-normalized means</a></li>
<li class="chapter" data-level="18.5.3" data-path="understanding-example-2.html"><a href="understanding-example-2.html#compare-an-offset-to-an-added-covariate-with-an-estimated-coefficient"><i class="fa fa-check"></i><b>18.5.3</b> Compare an offset to an added covariate with an estimated coefficient</a></li>
<li class="chapter" data-level="18.5.4" data-path="understanding-example-2.html"><a href="understanding-example-2.html#issues-with-plotting"><i class="fa fa-check"></i><b>18.5.4</b> Issues with plotting</a></li>
</ul></li>
<li class="chapter" data-level="18.6" data-path="example-3-glm-models-for-binary-responses.html"><a href="example-3-glm-models-for-binary-responses.html"><i class="fa fa-check"></i><b>18.6</b> Example 3 – GLM models for binary responses</a></li>
<li class="chapter" data-level="18.7" data-path="working-in-r-7.html"><a href="working-in-r-7.html"><i class="fa fa-check"></i><b>18.7</b> Working in R</a>
<ul>
<li class="chapter" data-level="18.7.1" data-path="working-in-r-7.html"><a href="working-in-r-7.html#fitting-glms-to-count-data"><i class="fa fa-check"></i><b>18.7.1</b> Fitting GLMs to count data</a></li>
<li class="chapter" data-level="18.7.2" data-path="working-in-r-7.html"><a href="working-in-r-7.html#fitting-a-glm-to-a-continuous-conditional-response-with-right-skew."><i class="fa fa-check"></i><b>18.7.2</b> Fitting a GLM to a continuous conditional response with right skew.</a></li>
<li class="chapter" data-level="18.7.3" data-path="working-in-r-7.html"><a href="working-in-r-7.html#fitting-a-glm-to-a-binary-success-or-failure-presence-or-absence-survived-or-died-response"><i class="fa fa-check"></i><b>18.7.3</b> Fitting a GLM to a binary (success or failure, presence or absence, survived or died) response</a></li>
<li class="chapter" data-level="18.7.4" data-path="working-in-r-7.html"><a href="working-in-r-7.html#fitting-generalized-linear-mixed-models"><i class="fa fa-check"></i><b>18.7.4</b> Fitting Generalized Linear Mixed Models</a></li>
</ul></li>
<li class="chapter" data-level="18.8" data-path="model-checking-glms.html"><a href="model-checking-glms.html"><i class="fa fa-check"></i><b>18.8</b> Model checking GLMs</a></li>
<li class="chapter" data-level="18.9" data-path="hidden-code-9.html"><a href="hidden-code-9.html"><i class="fa fa-check"></i><b>18.9</b> Hidden code</a>
<ul>
<li class="chapter" data-level="18.9.1" data-path="hidden-code-9.html"><a href="hidden-code-9.html#import-example-1-data-fig3a-angiogenic-sprouts"><i class="fa fa-check"></i><b>18.9.1</b> Import Example 1 data (fig3a – “angiogenic sprouts”)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="gls.html"><a href="gls.html"><i class="fa fa-check"></i><b>19</b> Linear models with heterogenous variance</a></li>
<li class="chapter" data-level="" data-path="appendix-an-example-set-of-analyses-of-experimental-data-with-linear-models.html"><a href="appendix-an-example-set-of-analyses-of-experimental-data-with-linear-models.html"><i class="fa fa-check"></i>Appendix: An example set of analyses of experimental data with linear models</a></li>
<li class="chapter" data-level="20" data-path="ask1-bio.html"><a href="ask1-bio.html"><i class="fa fa-check"></i><b>20</b> Background physiology to the experiments in Figure 2 of “ASK1 inhibits browning of white adipose tissue in obesity”</a></li>
<li class="chapter" data-level="21" data-path="ask1-report.html"><a href="ask1-report.html"><i class="fa fa-check"></i><b>21</b> Analyses for Figure 2 of “ASK1 inhibits browning of white adipose tissue in obesity”</a>
<ul>
<li class="chapter" data-level="21.1" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i><b>21.1</b> Setup</a></li>
<li class="chapter" data-level="21.2" data-path="data-source.html"><a href="data-source.html"><i class="fa fa-check"></i><b>21.2</b> Data source</a></li>
<li class="chapter" data-level="21.3" data-path="control-the-color-palette.html"><a href="control-the-color-palette.html"><i class="fa fa-check"></i><b>21.3</b> control the color palette</a></li>
<li class="chapter" data-level="21.4" data-path="useful-functions.html"><a href="useful-functions.html"><i class="fa fa-check"></i><b>21.4</b> useful functions</a></li>
<li class="chapter" data-level="21.5" data-path="figure-2b-effect-of-ask1-deletion-on-growth-body-weight.html"><a href="figure-2b-effect-of-ask1-deletion-on-growth-body-weight.html"><i class="fa fa-check"></i><b>21.5</b> figure 2b – effect of ASK1 deletion on growth (body weight)</a>
<ul>
<li class="chapter" data-level="21.5.1" data-path="figure-2b-effect-of-ask1-deletion-on-growth-body-weight.html"><a href="figure-2b-effect-of-ask1-deletion-on-growth-body-weight.html#figure-2b-import"><i class="fa fa-check"></i><b>21.5.1</b> figure 2b – import</a></li>
<li class="chapter" data-level="21.5.2" data-path="figure-2b-effect-of-ask1-deletion-on-growth-body-weight.html"><a href="figure-2b-effect-of-ask1-deletion-on-growth-body-weight.html#figure-2b-exploratory-plots"><i class="fa fa-check"></i><b>21.5.2</b> figure 2b – exploratory plots</a></li>
</ul></li>
<li class="chapter" data-level="21.6" data-path="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html"><a href="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html"><i class="fa fa-check"></i><b>21.6</b> Figure 2c – Effect of ASK1 deletion on final body weight</a>
<ul>
<li class="chapter" data-level="21.6.1" data-path="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html"><a href="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html#figure-2c-import"><i class="fa fa-check"></i><b>21.6.1</b> Figure 2c – import</a></li>
<li class="chapter" data-level="21.6.2" data-path="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html"><a href="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html#figure-2c-check-own-computation-of-weight-change-v-imported-value"><i class="fa fa-check"></i><b>21.6.2</b> Figure 2c – check own computation of weight change v imported value</a></li>
<li class="chapter" data-level="21.6.3" data-path="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html"><a href="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html#figure-2c-exploratory-plots"><i class="fa fa-check"></i><b>21.6.3</b> Figure 2c – exploratory plots</a></li>
<li class="chapter" data-level="21.6.4" data-path="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html"><a href="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html#figure-2c-fit-the-model-m1-lm"><i class="fa fa-check"></i><b>21.6.4</b> Figure 2c – fit the model: m1 (lm)</a></li>
<li class="chapter" data-level="21.6.5" data-path="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html"><a href="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html#figure-2c-check-the-model-m1"><i class="fa fa-check"></i><b>21.6.5</b> Figure 2c – check the model: m1</a></li>
<li class="chapter" data-level="21.6.6" data-path="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html"><a href="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html#figure-2c-fit-the-model-m2-gamma-glm"><i class="fa fa-check"></i><b>21.6.6</b> Figure 2c – fit the model: m2 (gamma glm)</a></li>
<li class="chapter" data-level="21.6.7" data-path="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html"><a href="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html#figure-2c-check-the-model-m2"><i class="fa fa-check"></i><b>21.6.7</b> Figure 2c – check the model, m2</a></li>
<li class="chapter" data-level="21.6.8" data-path="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html"><a href="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html#figure-2c-inference-from-the-model"><i class="fa fa-check"></i><b>21.6.8</b> Figure 2c – inference from the model</a></li>
<li class="chapter" data-level="21.6.9" data-path="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html"><a href="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html#figure-2c-plot-the-model"><i class="fa fa-check"></i><b>21.6.9</b> Figure 2c – plot the model</a></li>
<li class="chapter" data-level="21.6.10" data-path="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html"><a href="figure-2c-effect-of-ask1-deletion-on-final-body-weight.html#figure-2c-report"><i class="fa fa-check"></i><b>21.6.10</b> Figure 2c – report</a></li>
</ul></li>
<li class="chapter" data-level="21.7" data-path="figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve.html"><a href="figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve.html"><i class="fa fa-check"></i><b>21.7</b> Figure 2d – Effect of ASK1 KO on glucose tolerance (whole curve)</a>
<ul>
<li class="chapter" data-level="21.7.1" data-path="figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve.html"><a href="figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve.html#figure-2d-import"><i class="fa fa-check"></i><b>21.7.1</b> Figure 2d – Import</a></li>
<li class="chapter" data-level="21.7.2" data-path="figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve.html"><a href="figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve.html#figure-2d-exploratory-plots"><i class="fa fa-check"></i><b>21.7.2</b> Figure 2d – exploratory plots</a></li>
<li class="chapter" data-level="21.7.3" data-path="figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve.html"><a href="figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve.html#figure-2d-fit-the-model"><i class="fa fa-check"></i><b>21.7.3</b> Figure 2d – fit the model</a></li>
<li class="chapter" data-level="21.7.4" data-path="figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve.html"><a href="figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve.html#figure-2d-check-the-model"><i class="fa fa-check"></i><b>21.7.4</b> Figure 2d – check the model</a></li>
<li class="chapter" data-level="21.7.5" data-path="figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve.html"><a href="figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve.html#figure-2d-inference"><i class="fa fa-check"></i><b>21.7.5</b> Figure 2d – inference</a></li>
<li class="chapter" data-level="21.7.6" data-path="figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve.html"><a href="figure-2d-effect-of-ask1-ko-on-glucose-tolerance-whole-curve.html#figure-2d-plot-the-model"><i class="fa fa-check"></i><b>21.7.6</b> Figure 2d – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="21.8" data-path="figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure.html"><a href="figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure.html"><i class="fa fa-check"></i><b>21.8</b> Figure 2e – Effect of ASK1 deletion on glucose tolerance (summary measure)</a>
<ul>
<li class="chapter" data-level="21.8.1" data-path="figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure.html"><a href="figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure.html#figure-2e-message-the-data"><i class="fa fa-check"></i><b>21.8.1</b> Figure 2e – message the data</a></li>
<li class="chapter" data-level="21.8.2" data-path="figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure.html"><a href="figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure.html#figure-2e-exploratory-plots"><i class="fa fa-check"></i><b>21.8.2</b> Figure 2e – exploratory plots</a></li>
<li class="chapter" data-level="21.8.3" data-path="figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure.html"><a href="figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure.html#figure-2e-fit-the-model"><i class="fa fa-check"></i><b>21.8.3</b> Figure 2e – fit the model</a></li>
<li class="chapter" data-level="21.8.4" data-path="figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure.html"><a href="figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure.html#figure-2e-check-the-model"><i class="fa fa-check"></i><b>21.8.4</b> Figure 2e – check the model</a></li>
<li class="chapter" data-level="21.8.5" data-path="figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure.html"><a href="figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure.html#figure-2e-inference-from-the-model"><i class="fa fa-check"></i><b>21.8.5</b> Figure 2e – inference from the model</a></li>
<li class="chapter" data-level="21.8.6" data-path="figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure.html"><a href="figure-2e-effect-of-ask1-deletion-on-glucose-tolerance-summary-measure.html#figure-2e-plot-the-model"><i class="fa fa-check"></i><b>21.8.6</b> Figure 2e – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="21.9" data-path="figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate.html"><a href="figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate.html"><i class="fa fa-check"></i><b>21.9</b> Figure 2f – Effect of ASK1 deletion on glucose infusion rate</a>
<ul>
<li class="chapter" data-level="21.9.1" data-path="figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate.html"><a href="figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate.html#figure-2f-import"><i class="fa fa-check"></i><b>21.9.1</b> Figure 2f – import</a></li>
<li class="chapter" data-level="21.9.2" data-path="figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate.html"><a href="figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate.html#figure-2f-exploratory-plots"><i class="fa fa-check"></i><b>21.9.2</b> Figure 2f – exploratory plots</a></li>
<li class="chapter" data-level="21.9.3" data-path="figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate.html"><a href="figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate.html#figure-2f-fit-the-model"><i class="fa fa-check"></i><b>21.9.3</b> Figure 2f – fit the model</a></li>
<li class="chapter" data-level="21.9.4" data-path="figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate.html"><a href="figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate.html#figure-2f-check-the-model"><i class="fa fa-check"></i><b>21.9.4</b> Figure 2f – check the model</a></li>
<li class="chapter" data-level="21.9.5" data-path="figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate.html"><a href="figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate.html#figure-2f-inference"><i class="fa fa-check"></i><b>21.9.5</b> Figure 2f – inference</a></li>
<li class="chapter" data-level="21.9.6" data-path="figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate.html"><a href="figure-2f-effect-of-ask1-deletion-on-glucose-infusion-rate.html#figure-2f-plot-the-model"><i class="fa fa-check"></i><b>21.9.6</b> Figure 2f – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="21.10" data-path="figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake.html"><a href="figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake.html"><i class="fa fa-check"></i><b>21.10</b> Figure 2g – Effect of ASK1 deletion on tissue-specific glucose uptake</a>
<ul>
<li class="chapter" data-level="21.10.1" data-path="figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake.html"><a href="figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake.html#figure-2g-import"><i class="fa fa-check"></i><b>21.10.1</b> Figure 2g – import</a></li>
<li class="chapter" data-level="21.10.2" data-path="figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake.html"><a href="figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake.html#figure-2g-exploratory-plots"><i class="fa fa-check"></i><b>21.10.2</b> Figure 2g – exploratory plots</a></li>
<li class="chapter" data-level="21.10.3" data-path="figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake.html"><a href="figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake.html#figure-2g-fit-the-model"><i class="fa fa-check"></i><b>21.10.3</b> Figure 2g – fit the model</a></li>
<li class="chapter" data-level="21.10.4" data-path="figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake.html"><a href="figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake.html#figure-2g-check-the-model"><i class="fa fa-check"></i><b>21.10.4</b> Figure 2g – check the model</a></li>
<li class="chapter" data-level="21.10.5" data-path="figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake.html"><a href="figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake.html#figure-2g-inference"><i class="fa fa-check"></i><b>21.10.5</b> Figure 2g – inference</a></li>
<li class="chapter" data-level="21.10.6" data-path="figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake.html"><a href="figure-2g-effect-of-ask1-deletion-on-tissue-specific-glucose-uptake.html#figure-2g-plot-the-model"><i class="fa fa-check"></i><b>21.10.6</b> Figure 2g – plot the model</a></li>
</ul></li>
<li class="chapter" data-level="21.11" data-path="figure-2h.html"><a href="figure-2h.html"><i class="fa fa-check"></i><b>21.11</b> Figure 2h</a></li>
<li class="chapter" data-level="21.12" data-path="figure-2i-effect-of-ask1-deletion-on-liver-tg.html"><a href="figure-2i-effect-of-ask1-deletion-on-liver-tg.html"><i class="fa fa-check"></i><b>21.12</b> Figure 2i – Effect of ASK1 deletion on liver TG</a>
<ul>
<li class="chapter" data-level="21.12.1" data-path="figure-2i-effect-of-ask1-deletion-on-liver-tg.html"><a href="figure-2i-effect-of-ask1-deletion-on-liver-tg.html#figure-2i-fit-the-model"><i class="fa fa-check"></i><b>21.12.1</b> Figure 2i – fit the model</a></li>
<li class="chapter" data-level="21.12.2" data-path="figure-2i-effect-of-ask1-deletion-on-liver-tg.html"><a href="figure-2i-effect-of-ask1-deletion-on-liver-tg.html#figure-2i-check-the-model"><i class="fa fa-check"></i><b>21.12.2</b> Figure 2i – check the model</a></li>
<li class="chapter" data-level="21.12.3" data-path="figure-2i-effect-of-ask1-deletion-on-liver-tg.html"><a href="figure-2i-effect-of-ask1-deletion-on-liver-tg.html#figure-2i-inference"><i class="fa fa-check"></i><b>21.12.3</b> Figure 2i – inference</a></li>
<li class="chapter" data-level="21.12.4" data-path="figure-2i-effect-of-ask1-deletion-on-liver-tg.html"><a href="figure-2i-effect-of-ask1-deletion-on-liver-tg.html#figure-2i-plot-the-model"><i class="fa fa-check"></i><b>21.12.4</b> Figure 2i – plot the model</a></li>
<li class="chapter" data-level="21.12.5" data-path="figure-2i-effect-of-ask1-deletion-on-liver-tg.html"><a href="figure-2i-effect-of-ask1-deletion-on-liver-tg.html#figure-2i-report-the-model"><i class="fa fa-check"></i><b>21.12.5</b> Figure 2i – report the model</a></li>
</ul></li>
<li class="chapter" data-level="21.13" data-path="figure-2j.html"><a href="figure-2j.html"><i class="fa fa-check"></i><b>21.13</b> Figure 2j</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="count-sim.html"><a href="count-sim.html"><i class="fa fa-check"></i><b>22</b> Simulations – Count data (alternatives to a t-test)</a>
<ul>
<li class="chapter" data-level="22.1" data-path="use-data-similar-to-figure-6f-from-example-1.html"><a href="use-data-similar-to-figure-6f-from-example-1.html"><i class="fa fa-check"></i><b>22.1</b> Use data similar to Figure 6f from Example 1</a></li>
<li class="chapter" data-level="22.2" data-path="functions.html"><a href="functions.html"><i class="fa fa-check"></i><b>22.2</b> Functions</a></li>
<li class="chapter" data-level="22.3" data-path="simulations.html"><a href="simulations.html"><i class="fa fa-check"></i><b>22.3</b> Simulations</a>
<ul>
<li class="chapter" data-level="22.3.1" data-path="simulations.html"><a href="simulations.html#type-i-pseudo-normal-distribution"><i class="fa fa-check"></i><b>22.3.1</b> Type I, Pseudo-Normal distribution</a></li>
<li class="chapter" data-level="22.3.2" data-path="simulations.html"><a href="simulations.html#type-i-neg-binom-equal-n"><i class="fa fa-check"></i><b>22.3.2</b> Type I, neg binom, equal n</a></li>
<li class="chapter" data-level="22.3.3" data-path="simulations.html"><a href="simulations.html#type-i-neg-binom-equal-n-small-theta"><i class="fa fa-check"></i><b>22.3.3</b> Type I, neg binom, equal n, small theta</a></li>
<li class="chapter" data-level="22.3.4" data-path="simulations.html"><a href="simulations.html#type-i-neg-binom-unequal-n"><i class="fa fa-check"></i><b>22.3.4</b> Type I, neg binom, unequal n</a></li>
<li class="chapter" data-level="22.3.5" data-path="simulations.html"><a href="simulations.html#power-pseudo-normal-distribution-equal-n"><i class="fa fa-check"></i><b>22.3.5</b> Power, Pseudo-Normal distribution, equal n</a></li>
<li class="chapter" data-level="22.3.6" data-path="simulations.html"><a href="simulations.html#power-neg-binom-equal-n"><i class="fa fa-check"></i><b>22.3.6</b> Power, neg binom, equal n</a></li>
<li class="chapter" data-level="22.3.7" data-path="simulations.html"><a href="simulations.html#power-neg-binom-small-theta"><i class="fa fa-check"></i><b>22.3.7</b> Power, neg binom, small theta</a></li>
<li class="chapter" data-level="22.3.8" data-path="simulations.html"><a href="simulations.html#power-neg-binom-unequal-n"><i class="fa fa-check"></i><b>22.3.8</b> Power, neg binom, unequal n</a></li>
<li class="chapter" data-level="22.3.9" data-path="simulations.html"><a href="simulations.html#power-neg-binom-unequal-n-unequal-theta"><i class="fa fa-check"></i><b>22.3.9</b> Power, neg binom, unequal n, unequal theta</a></li>
<li class="chapter" data-level="22.3.10" data-path="simulations.html"><a href="simulations.html#type-1-neg-binom-equal-n-unequal-theta"><i class="fa fa-check"></i><b>22.3.10</b> Type 1, neg binom, equal n, unequal theta</a></li>
</ul></li>
<li class="chapter" data-level="22.4" data-path="save-it-read-it.html"><a href="save-it-read-it.html"><i class="fa fa-check"></i><b>22.4</b> Save it, Read it</a></li>
<li class="chapter" data-level="22.5" data-path="analysis.html"><a href="analysis.html"><i class="fa fa-check"></i><b>22.5</b> Analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix-1-getting-started-with-r.html"><a href="appendix-1-getting-started-with-r.html"><i class="fa fa-check"></i>Appendix 1: Getting Started with R</a>
<ul>
<li class="chapter" data-level="22.6" data-path="get-your-computer-ready.html"><a href="get-your-computer-ready.html"><i class="fa fa-check"></i><b>22.6</b> Get your computer ready</a>
<ul>
<li class="chapter" data-level="22.6.1" data-path="get-your-computer-ready.html"><a href="get-your-computer-ready.html#start-here"><i class="fa fa-check"></i><b>22.6.1</b> Start here</a></li>
<li class="chapter" data-level="22.6.2" data-path="get-your-computer-ready.html"><a href="get-your-computer-ready.html#install-r"><i class="fa fa-check"></i><b>22.6.2</b> Install R</a></li>
<li class="chapter" data-level="22.6.3" data-path="get-your-computer-ready.html"><a href="get-your-computer-ready.html#install-r-studio"><i class="fa fa-check"></i><b>22.6.3</b> Install R Studio</a></li>
<li class="chapter" data-level="22.6.4" data-path="get-your-computer-ready.html"><a href="get-your-computer-ready.html#install-r-markdown-1"><i class="fa fa-check"></i><b>22.6.4</b> Install R Markdown</a></li>
<li class="chapter" data-level="22.6.5" data-path="get-your-computer-ready.html"><a href="get-your-computer-ready.html#optional-alternative-latex-installations"><i class="fa fa-check"></i><b>22.6.5</b> (optional) Alternative LaTeX installations</a></li>
</ul></li>
<li class="chapter" data-level="22.7" data-path="start-learning-r-studio.html"><a href="start-learning-r-studio.html"><i class="fa fa-check"></i><b>22.7</b> Start learning R Studio</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix-2-online-resources-for-getting-started-with-statistical-modeling-in-r.html"><a href="appendix-2-online-resources-for-getting-started-with-statistical-modeling-in-r.html"><i class="fa fa-check"></i>Appendix 2: Online Resources for Getting Started with Statistical Modeling in R</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistics for the Experimental Biologist</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c" class="section level2 hasAnchor" number="16.8">
<h2><span class="header-section-number">16.8</span> Example 3 – Factorial experiments and no subsampling replicates (exp5c)<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#example-3-factorial-experiments-and-no-subsampling-replicates-exp5c" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Example 3 is similar to example 2 in that there is no subsampling replication and we cannot add random slopes to the linear mixed model. Example 3 differs in that the design is factorial – there are two, crossed fixed factors. Consequently, there are several alternative models with different sets of random intercepts. The reported model includes two random intercepts, one of which models differences in batch effects among treatment levels (treatment by batch interactions). This <strong>interaction intercept</strong> is an alternative to random slope for modeling treatment by batch interactions.</p>
<p><a href="https://www.nature.com/articles/s41467-019-13869-w" target="_blank">Transcriptomic profiling of skeletal muscle adaptations to exercise and inactivity</a></p>
<p>Source figure: <a href="https://www.nature.com/articles/s41467-019-13869-w#Fig5" target="_blank">Fig. 5c</a></p>
<p>Source data: <a href="https://www.nature.com/articles/s41467-019-13869-w#Sec27" target="_blank">Source Data Fig. 5</a></p>
<div id="understand-the-data-1" class="section level3 hasAnchor" number="16.8.1">
<h3><span class="header-section-number">16.8.1</span> Understand the data<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#understand-the-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The data for Example 1 are from Figure 5c. Six muscle source cells were used to start six independent cultures. Cells from each culture were treated with either a negative control (“Scr”) or a siRNA (“siNR4A3”) that “silences” expression of the NR4A3 gene product by <a href="https://en.wikipedia.org/wiki/Small_interfering_RNA" target="_blank">cleaving the mRNA</a>. Glucose uptake in the two cell types was measured at rest (“Basal”) and during electrical pulse stimulation (“EPS”).</p>
<p>The design is a <span class="math inline">\(2 \times 2\)</span> <a href="https://en.wikipedia.org/wiki/Blocking_(statistics)#Randomized_block_design" target="_blank">Randomized complete block with no subsampling</a>. There are two factors each with two levels: <span class="math inline">\(\texttt{treatment}\)</span> (“Scr”, “siNR4A3”) and <span class="math inline">\(\texttt{activity}\)</span> (“Basal”, “EPS”). Each source cell is a block. All four treatment combinations were measured once per block.</p>
</div>
<div id="examine-the-data-1" class="section level3 hasAnchor" number="16.8.2">
<h3><span class="header-section-number">16.8.2</span> Examine the data<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#examine-the-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-examine-1.png" width="576" /></p>
<p>The plot shows a strong donor effect.</p>
</div>
<div id="lmm-exp5c-m1" class="section level3 hasAnchor" number="16.8.3">
<h3><span class="header-section-number">16.8.3</span> Model fit and inference<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#lmm-exp5c-m1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb532"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb532-1"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-1" tabindex="-1"></a>exp5c_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb532-2"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-2" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb532-3"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-3" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb532-4"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-4" tabindex="-1"></a></span>
<span id="cb532-5"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-5" tabindex="-1"></a>exp5c_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb532-6"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-6" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb532-7"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-7" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment) <span class="sc">+</span></span>
<span id="cb532-8"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-8" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>activity),</span>
<span id="cb532-9"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-9" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb532-10"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-10" tabindex="-1"></a></span>
<span id="cb532-11"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-11" tabindex="-1"></a>exp5c_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb532-12"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-12" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb532-13"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-13" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment),</span>
<span id="cb532-14"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-14" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb532-15"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-15" tabindex="-1"></a></span>
<span id="cb532-16"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-16" tabindex="-1"></a>exp5c_m1d <span class="ot">&lt;-</span> <span class="fu">lme</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb532-17"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-17" tabindex="-1"></a>                 <span class="at">random =</span>  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor,</span>
<span id="cb532-18"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-18" tabindex="-1"></a>                 <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb532-19"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-19" tabindex="-1"></a>                 <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span><span class="sc">|</span>t.by.a),</span>
<span id="cb532-20"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-20" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb532-21"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-21" tabindex="-1"></a></span>
<span id="cb532-22"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-22" tabindex="-1"></a><span class="co"># check AIC</span></span>
<span id="cb532-23"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb532-23" tabindex="-1"></a><span class="fu">AIC</span>(exp5c_m1a, exp5c_m1b, exp5c_m1c, exp5c_m1d)</span></code></pre></div>
<pre><code>##           df       AIC
## exp5c_m1a  6  8.153813
## exp5c_m1b  8 10.052676
## exp5c_m1c  7  8.052676
## exp5c_m1d 15 14.763024</code></pre>
<div class="sourceCode" id="cb534"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb534-1"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb534-1" tabindex="-1"></a><span class="co"># check VarCorr model c</span></span>
<span id="cb534-2"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb534-2" tabindex="-1"></a><span class="fu">VarCorr</span>(exp5c_m1c) <span class="co"># fine</span></span></code></pre></div>
<pre><code>##  Groups          Name        Std.Dev.
##  donor:treatment (Intercept) 0.089519
##  donor           (Intercept) 0.352097
##  Residual                    0.090685</code></pre>
<div class="sourceCode" id="cb536"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb536-1"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb536-1" tabindex="-1"></a><span class="co"># report 1c (based on AIC and VarCorr check)</span></span>
<span id="cb536-2"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb536-2" tabindex="-1"></a>exp5c_m1 <span class="ot">&lt;-</span> exp5c_m1c</span></code></pre></div>
<p>exp5c_m1b (equivalent to univariate repeated measures ANOVA) is singular fit. don’t use. Trivial difference in AIC between exp5c_m1a and exp5c_m1c. Nothing in VarCorr with exp5c_m1c raises red flags. Report exp5c_m1c.</p>
<div id="check-the-model-2" class="section level4 hasAnchor" number="16.8.3.1">
<h4><span class="header-section-number">16.8.3.1</span> Check the model<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#check-the-model-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb537"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb537-1"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb537-1" tabindex="-1"></a><span class="fu">ggcheck_the_model</span>(exp5c_m1)</span></code></pre></div>
<p><img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c_m1-check-1.png" width="576" /></p>
<p>fine.</p>
</div>
<div id="inference-from-the-model-5" class="section level4 hasAnchor" number="16.8.3.2">
<h4><span class="header-section-number">16.8.3.2</span> Inference from the model<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#inference-from-the-model-5" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb538"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb538-1"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb538-1" tabindex="-1"></a>exp5c_m1_coef <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">coef</span>(<span class="fu">summary</span>(exp5c_m1)),</span>
<span id="cb538-2"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb538-2" tabindex="-1"></a>                       <span class="fu">confint</span>(exp5c_m1)[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>),])</span></code></pre></div>
<div class="sourceCode" id="cb539"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb539-1"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb539-1" tabindex="-1"></a>exp5c_m1_coef <span class="sc">%&gt;%</span></span>
<span id="cb539-2"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb539-2" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb539-3"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb539-3" tabindex="-1"></a>  <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
1.34
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
8.8
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
1.02
</td>
<td style="text-align:right;">
1.66
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
8.5
</td>
<td style="text-align:right;">
1.1
</td>
<td style="text-align:right;">
0.2994
</td>
<td style="text-align:right;">
-0.07
</td>
<td style="text-align:right;">
0.23
</td>
</tr>
<tr>
<td style="text-align:left;">
activityEPS
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
2.9
</td>
<td style="text-align:right;">
0.0163
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.25
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3:activityEPS
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
-3.6
</td>
<td style="text-align:right;">
0.0048
</td>
<td style="text-align:right;">
-0.41
</td>
<td style="text-align:right;">
-0.12
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb540"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb540-1"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb540-1" tabindex="-1"></a>exp5c_m1_emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(exp5c_m1, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">&quot;treatment&quot;</span>, <span class="st">&quot;activity&quot;</span>))</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
treatment
</th>
<th style="text-align:left;">
activity
</th>
<th style="text-align:right;">
emmean
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Scr
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:right;">
1.34
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
0.96
</td>
<td style="text-align:right;">
1.72
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3
</td>
<td style="text-align:left;">
Basal
</td>
<td style="text-align:right;">
1.42
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
1.04
</td>
<td style="text-align:right;">
1.80
</td>
</tr>
<tr>
<td style="text-align:left;">
Scr
</td>
<td style="text-align:left;">
EPS
</td>
<td style="text-align:right;">
1.49
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
1.11
</td>
<td style="text-align:right;">
1.87
</td>
</tr>
<tr>
<td style="text-align:left;">
siNR4A3
</td>
<td style="text-align:left;">
EPS
</td>
<td style="text-align:right;">
1.31
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
0.93
</td>
<td style="text-align:right;">
1.68
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb541"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb541-1"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-1" tabindex="-1"></a><span class="co"># exp5c_emm # print in console to get row numbers</span></span>
<span id="cb541-2"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-2" tabindex="-1"></a><span class="co"># set the mean as the row number from the emmeans table</span></span>
<span id="cb541-3"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-3" tabindex="-1"></a>scr_basal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb541-4"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-4" tabindex="-1"></a>siNR4A3_basal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb541-5"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-5" tabindex="-1"></a>scr_eps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb541-6"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-6" tabindex="-1"></a>siNR4A3_eps <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb541-7"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-7" tabindex="-1"></a></span>
<span id="cb541-8"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-8" tabindex="-1"></a>exp5c_m1_planned <span class="ot">&lt;-</span> <span class="fu">contrast</span>(exp5c_m1_emm,</span>
<span id="cb541-9"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-9" tabindex="-1"></a>                       <span class="at">method =</span> <span class="fu">list</span>(</span>
<span id="cb541-10"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-10" tabindex="-1"></a>                         <span class="st">&quot;(Scr EPS) - (Scr Basal)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(scr_eps <span class="sc">-</span> scr_basal),</span>
<span id="cb541-11"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-11" tabindex="-1"></a>                         <span class="st">&quot;(siNR4A3 EPS) - (siNR4A3 Basal)&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(siNR4A3_eps <span class="sc">-</span> siNR4A3_basal),</span>
<span id="cb541-12"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-12" tabindex="-1"></a>                         <span class="st">&quot;Interaction&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(siNR4A3_eps <span class="sc">-</span> siNR4A3_basal) <span class="sc">-</span></span>
<span id="cb541-13"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-13" tabindex="-1"></a>                           <span class="fu">c</span>(scr_eps <span class="sc">-</span> scr_basal)</span>
<span id="cb541-14"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-14" tabindex="-1"></a>                           </span>
<span id="cb541-15"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-15" tabindex="-1"></a>                       ),</span>
<span id="cb541-16"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-16" tabindex="-1"></a>                       <span class="at">adjust =</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb541-17"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-17" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb541-18"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb541-18" tabindex="-1"></a>  <span class="fu">summary</span>(<span class="at">infer =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.016
</td>
</tr>
<tr>
<td style="text-align:left;">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
-2.22
</td>
<td style="text-align:right;">
0.051
</td>
</tr>
<tr>
<td style="text-align:left;">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.43
</td>
<td style="text-align:right;">
-0.10
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.005
</td>
</tr>
</tbody>
</table>
</div>
<div id="lmm-exp5c-plotthemodel" class="section level4 hasAnchor" number="16.8.3.3">
<h4><span class="header-section-number">16.8.3.3</span> Plot the model<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#lmm-exp5c-plotthemodel" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-229"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/unnamed-chunk-229-1.png" alt="Treatment effect on glucose uptake. Gray dots connected by lines are individual donors." width="576" />
<p class="caption">
Figure 16.12: Treatment effect on glucose uptake. Gray dots connected by lines are individual donors.
</p>
</div>
</div>
<div id="alternaplot-the-model-2" class="section level4 hasAnchor" number="16.8.3.4">
<h4><span class="header-section-number">16.8.3.4</span> Alternaplot the model<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#alternaplot-the-model-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="figure"><span style="display:block;" id="fig:lmm-exp5c-alternaplot"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-alternaplot-1.png" alt="Glucose response to different treatments. Gray dots connected by lines are individual donors. Dashed gray line is expected additive mean of &quot;siNR4A3 EPS&quot;." width="576" />
<p class="caption">
Figure 16.13: Glucose response to different treatments. Gray dots connected by lines are individual donors. Dashed gray line is expected additive mean of “siNR4A3 EPS”.
</p>
</div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Many researchers might look at the wide confidence intervals relative to the short distance between the means and think “no effect”. The confidence intervals are correct, they simply are <em>not</em> meant to be tools for inferring anything about differences in means. This is one of many reasons why plots of means and error bars can be misleading for inference, despite the ubiquity of their use for communicating results. And, its why I prefer the <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#lmm-exp5c-plotthemodel">effects-and-response plots</a>, which explicitly communicate correct inference about effects.</li>
</ol>
</div>
</div>
<div id="why-we-care-about-modeling-batch-in-exp5c" class="section level3 hasAnchor" number="16.8.4">
<h3><span class="header-section-number">16.8.4</span> Why we care about modeling batch in exp5c<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#why-we-care-about-modeling-batch-in-exp5c" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Figure <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#fig:lmm-exp5c-why">16.14</a> shows the modeled means of the four treatment combinations and the individual values colored by donor. It is pretty easy to see that the glucose uptake values for donors 4 and 5 are well above the mean for all four treatments. And, the values for donors 1, 2, and 3 are well below the mean for all four treatments. The values for donor 6 are near the mean for all four treatments.</p>
<div class="figure"><span style="display:block;" id="fig:lmm-exp5c-why"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-why-1.png" alt="Why we care about blocking. The black dots are the modeled means of each treatment combination. The colored dots are the measured values of the response for each donor. The position of a donor relative to the mean is easy to see with these data." width="672" />
<p class="caption">
Figure 16.14: Why we care about blocking. The black dots are the modeled means of each treatment combination. The colored dots are the measured values of the response for each donor. The position of a donor relative to the mean is easy to see with these data.
</p>
</div>
<p>Let’s compare the effects estimated by the linear mixed model <code>exp5c_m1</code> with a linear model that ignores donor.</p>
<div class="sourceCode" id="cb542"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb542-1"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb542-1" tabindex="-1"></a>exp5c_m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb542-2"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb542-2" tabindex="-1"></a>           <span class="at">data =</span> exp5c)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:lmm-exp5c-why2"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-why2-1.png" alt="A. Inference from a linear mixed model with blocking factor (donor) added as a random intercept. B. Inference from a fixed effects model." width="576" />
<p class="caption">
Figure 16.15: A. Inference from a linear mixed model with blocking factor (donor) added as a random intercept. B. Inference from a fixed effects model.
</p>
</div>
<p>Figure <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#fig:lmm-exp5c-why2">16.15</a>A is a plot of the effects from the linear mixed model that models the correlated error due to donor. Figure <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#fig:lmm-exp5c-why2">16.15</a>B is a plot of the effects from the fixed effect model that ignores the correlated error due to donor. Adding <span class="math inline">\(\texttt{donor}\)</span> as a factor to the linear model increases the precision of the estimate of the treatment effects by eliminating the among-donor component of variance from the error variance.</p>
</div>
<div id="the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts" class="section level3 hasAnchor" number="16.8.5">
<h3><span class="header-section-number">16.8.5</span> The linear mixed model exp5c_m1 adds two random intercepts<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#the-linear-mixed-model-exp5c_m1-adds-two-random-intercepts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are <em>two</em> random intercepts in the linear mixed model <code>exp5c_m1</code> fit to the exp5c data (<code>exp5c_m1</code> is copied from <code>exp5c_m1c</code> in <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#lmm-exp5c-m1">Model fit and inference</a>).</p>
<p><span class="math display" id="eq:lmm-exp5c-m1">\[
\begin{align}
\texttt{glucose_uptake}_j = \ &amp;(\beta_0 + \gamma_{0j} + \gamma_{0jk}) + \beta_1 (\texttt{treatment}_\texttt{siNR4A3}) + \beta_2 (\texttt{activity}_\texttt{EPS}) \ + \\
&amp;\beta_3 (\texttt{treatment}_\texttt{siNR4A3}:\texttt{activity}_\texttt{EPS}) + \varepsilon
\end{align}
\tag{16.4}
\]</span></p>
<ol style="list-style-type: decimal">
<li>The random intercept effect <span class="math inline">\(\gamma_{0j}\)</span> models donor variation in the reference level. <em>j</em> indexes donor <em>j</em>. A coefficient is estimated for each donor.</li>
<li>The random intercept effect <span class="math inline">\(\gamma_{0jk}\)</span> models the variation due to the <span class="math inline">\(\texttt{donor}\)</span> by <span class="math inline">\(\texttt{treatment}\)</span> combinations. A coefficient is estimated for each of the 6 (donors) <span class="math inline">\(\times\)</span> 2 (levels of <span class="math inline">\(\texttt{treatment}\)</span>). <em>k</em> indexes treatment level <em>k</em>. This <strong>interaction intercept</strong> is an alternative to random slope for modeling treatment by batch interactions. Unlike a random slope, there is a <span class="math inline">\(\gamma_{0jk}\)</span> coefficient for each donor in the reference level.</li>
</ol>
</div>
<div id="the-fixed-effect-coefficients-of-model-exp5c_m1" class="section level3 hasAnchor" number="16.8.6">
<h3><span class="header-section-number">16.8.6</span> The fixed effect coefficients of model exp5c_m1<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#the-fixed-effect-coefficients-of-model-exp5c_m1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The fixed effect coefficients of model <code>exp5c_m1</code> are</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
1.34
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
5.8
</td>
<td style="text-align:right;">
8.77
</td>
<td style="text-align:right;">
0.0001
</td>
<td style="text-align:right;">
1.02
</td>
<td style="text-align:right;">
1.66
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
8.5
</td>
<td style="text-align:right;">
1.11
</td>
<td style="text-align:right;">
0.2994
</td>
<td style="text-align:right;">
-0.07
</td>
<td style="text-align:right;">
0.23
</td>
</tr>
<tr>
<td style="text-align:left;">
activityEPS
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.0163
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.25
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3:activityEPS
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
10.0
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.0048
</td>
<td style="text-align:right;">
-0.41
</td>
<td style="text-align:right;">
-0.12
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The interpretation of the fixed effectcs coefficients have the <a href="understanding-the-linear-model-with-crossed-factors-1.html#twoway-what-coefs-are">usual interpretation for a factorial linear model</a>. Figure <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#fig:lmm-exp5c-explainer-1">16.16</a> is a reminder.</li>
</ol>
<div class="figure"><span style="display:block;" id="fig:lmm-exp5c-explainer-1"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-explainer-1-1.png" alt="Fixed effects estimated by exp5c_m1. The light gray point is the expected value of the GPR174- F treatment if genotype and sex were additive." width="672" />
<p class="caption">
Figure 16.16: Fixed effects estimated by exp5c_m1. The light gray point is the expected value of the GPR174- F treatment if genotype and sex were additive.
</p>
</div>
</div>
<div id="the-random-effect-coefficients-of-model-exp5c_m1" class="section level3 hasAnchor" number="16.8.7">
<h3><span class="header-section-number">16.8.7</span> The random effect coefficients of model exp5c_m1<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#the-random-effect-coefficients-of-model-exp5c_m1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-m1-random-coef">Table 16.11: </span>Random intercept effects for each donor (g_0j) and for each donor:treatment combination (g_0jk) for exp5c_m1. There are two random intercepts. random_intercept_j is the sum of the fixed intercept (b_0) and g_0j. random_intercept_jk is the sum of the fixed intercept (b_0) and g_0jk.
</caption>
<thead>
<tr>
<th style="text-align:left;">
donor:treatment
</th>
<th style="text-align:right;">
random_intercept_j
</th>
<th style="text-align:right;">
random_intercept_jk
</th>
<th style="text-align:right;">
b_0
</th>
<th style="text-align:right;">
g_0j
</th>
<th style="text-align:right;">
g_0jk
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
donor_1:Scr
</td>
<td style="text-align:right;">
1.018
</td>
<td style="text-align:right;">
1.349
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.323
</td>
<td style="text-align:right;">
0.008
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_1:siNR4A3
</td>
<td style="text-align:right;">
1.018
</td>
<td style="text-align:right;">
1.312
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.323
</td>
<td style="text-align:right;">
-0.029
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_2:Scr
</td>
<td style="text-align:right;">
1.033
</td>
<td style="text-align:right;">
1.283
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.307
</td>
<td style="text-align:right;">
-0.058
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_2:siNR4A3
</td>
<td style="text-align:right;">
1.033
</td>
<td style="text-align:right;">
1.379
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.307
</td>
<td style="text-align:right;">
0.038
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_3:Scr
</td>
<td style="text-align:right;">
1.168
</td>
<td style="text-align:right;">
1.310
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.173
</td>
<td style="text-align:right;">
-0.031
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_3:siNR4A3
</td>
<td style="text-align:right;">
1.168
</td>
<td style="text-align:right;">
1.360
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.173
</td>
<td style="text-align:right;">
0.020
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_4:Scr
</td>
<td style="text-align:right;">
1.785
</td>
<td style="text-align:right;">
1.302
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
0.445
</td>
<td style="text-align:right;">
-0.039
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_4:siNR4A3
</td>
<td style="text-align:right;">
1.785
</td>
<td style="text-align:right;">
1.408
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
0.445
</td>
<td style="text-align:right;">
0.068
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_5:Scr
</td>
<td style="text-align:right;">
1.744
</td>
<td style="text-align:right;">
1.433
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
0.403
</td>
<td style="text-align:right;">
0.092
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_5:siNR4A3
</td>
<td style="text-align:right;">
1.744
</td>
<td style="text-align:right;">
1.275
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
0.403
</td>
<td style="text-align:right;">
-0.066
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_6:Scr
</td>
<td style="text-align:right;">
1.296
</td>
<td style="text-align:right;">
1.368
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.045
</td>
<td style="text-align:right;">
0.028
</td>
</tr>
<tr>
<td style="text-align:left;">
donor_6:siNR4A3
</td>
<td style="text-align:right;">
1.296
</td>
<td style="text-align:right;">
1.310
</td>
<td style="text-align:right;">
1.341
</td>
<td style="text-align:right;">
-0.045
</td>
<td style="text-align:right;">
-0.031
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The random intercept “random_intercept_j” is the sum of the fixed intercept <span class="math inline">\(b_0\)</span> and the random intercept effect <span class="math inline">\(g_{0j}\)</span> for donor <em>j</em>. The <span class="math inline">\(g_{0j}\)</span> estimate the <span class="math inline">\(\gamma_{0j}\)</span> in Model <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#eq:lmm-exp5c-m1">(16.4)</a>. Note that <span class="math inline">\(g_{0j}\)</span> is the same for both treatment levels within a donor. The illustration of these random effects is similar to that for Example 2 <code>exp6g_m1</code> (Figure <a href="lmm-example2.html#fig:lmm-exp6g-explainer-2">16.10</a>.</li>
<li>The random intercept “random_intercept_jk” is the sum of the fixed intercept <span class="math inline">\(b_0\)</span> and the random intercept effect <span class="math inline">\(g_{0jk}\)</span> for donor <em>j</em> in treatment level <em>k</em> (with the reference level equal to 1). The <span class="math inline">\(g_{0jk}\)</span> estimate the <span class="math inline">\(\gamma_{0jk}\)</span> in Model <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#eq:lmm-exp5c-m1">(16.4)</a>. The <span class="math inline">\(g_{0jk}\)</span> differ for each combination of <span class="math inline">\(\texttt{donor}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>.</li>
<li>The donor:treatment random intercepts (“random_intercept_jk”) are shown with the dark, colored dots in Figure <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#fig:lmm-exp5c-explainer-2">16.17</a>A. The distance from the dark colored dot to the dashed, gray line (the modeled mean for the treatment:activity combination) is the random intercept effect estimate of <span class="math inline">\(\gamma_{0jk}\)</span>.</li>
<li>The set of random intercept effects for the donor:treatment combinations (<span class="math inline">\(g_{0jk}\)</span>) are the same between the two activity levels. This is seen in Figure <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#fig:lmm-exp5c-explainer-2">16.17</a>A - compare the pattern of dark, colored dots in the two EPS treatments to the two Basal treatments.</li>
<li>The combined random intercept effects (<span class="math inline">\(g_{0j} + g_{0jk}\)</span>) are shown in Figure <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#fig:lmm-exp5c-explainer-2">16.17</a>B as the distance between the dark, colored dots and the dashed, gray lines.</li>
<li>There is some communication decision-making in Figure <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#fig:lmm-exp5c-explainer-2">16.17</a>. The actual y-values of the dark, colored dots are the random intercept effect plus the treatment-combination modeled mean. Thus, in A, the dark colored dots are equal to “random_intercept_jk” only at the reference level (since the reference model mean is the fixed intercept).</li>
</ol>
<div class="figure"><span style="display:block;" id="fig:lmm-exp5c-explainer-2"></span>
<img src="Walker-elementary-statistical-modeling-draft_files/figure-html/lmm-exp5c-explainer-2-1.png" alt="Random intercepts for model exp5c_m1. The pale, colored dots are the measured glucose uptake values for each donor at each treatment combination. The dashed, gray lines are the modeled means for each treatment combination. (A) The distance from a dark colored dot to the dashed, gray line is the random intercept effect $g_{0jk}$. (B) The distance from a dark colored dot to the dashed, gray line is the combined random intercept effect $g_{0j} + g_{0jk}$." width="1066.66666666667" />
<p class="caption">
Figure 16.17: Random intercepts for model exp5c_m1. The pale, colored dots are the measured glucose uptake values for each donor at each treatment combination. The dashed, gray lines are the modeled means for each treatment combination. (A) The distance from a dark colored dot to the dashed, gray line is the random intercept effect <span class="math inline">\(g_{0jk}\)</span>. (B) The distance from a dark colored dot to the dashed, gray line is the combined random intercept effect <span class="math inline">\(g_{0j} + g_{0jk}\)</span>.
</p>
</div>
</div>
<div id="lmm-exp5c-alt" class="section level3 hasAnchor" number="16.8.8">
<h3><span class="header-section-number">16.8.8</span> Alternative models for exp5c<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#lmm-exp5c-alt" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb543"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb543-1"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb543-1" tabindex="-1"></a>exp5c_m1a <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb543-2"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb543-2" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb543-3"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb543-3" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb543-4"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb543-4" tabindex="-1"></a></span>
<span id="cb543-5"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb543-5" tabindex="-1"></a>exp5c_m1b <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb543-6"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb543-6" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb543-7"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb543-7" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment) <span class="sc">+</span></span>
<span id="cb543-8"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb543-8" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>activity),</span>
<span id="cb543-9"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb543-9" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span></code></pre></div>
<pre><code>## boundary (singular) fit: see help(&#39;isSingular&#39;)</code></pre>
<div class="sourceCode" id="cb545"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb545-1"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb545-1" tabindex="-1"></a>exp5c_m1c <span class="ot">&lt;-</span> <span class="fu">lmer</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb545-2"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb545-2" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor) <span class="sc">+</span></span>
<span id="cb545-3"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb545-3" tabindex="-1"></a>                   (<span class="dv">1</span> <span class="sc">|</span> donor<span class="sc">:</span>treatment),</span>
<span id="cb545-4"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb545-4" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span>
<span id="cb545-5"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb545-5" tabindex="-1"></a></span>
<span id="cb545-6"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb545-6" tabindex="-1"></a>exp5c_m1d <span class="ot">&lt;-</span> <span class="fu">lme</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity,</span>
<span id="cb545-7"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb545-7" tabindex="-1"></a>                 <span class="at">random =</span>  <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor,</span>
<span id="cb545-8"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb545-8" tabindex="-1"></a>                 <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> donor),</span>
<span id="cb545-9"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb545-9" tabindex="-1"></a>                 <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span><span class="sc">|</span>t.by.a),</span>
<span id="cb545-10"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb545-10" tabindex="-1"></a>                 <span class="at">data =</span> exp5c)</span></code></pre></div>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>Four models are fit. The models specify the same fixed effects but different random effects or patterns of correlated error. Again, we care about the fixed effects – this is the point of the experiment – but the specification of the random effects determines the error variance and degrees of freedom for computing uncertainty.</li>
<li>Model <code>exp5c_m1a</code> specifies a random intercept for each level of <span class="math inline">\(\texttt{donor}\)</span>.</li>
<li>Model <code>exp5c_m1b</code> adds two additional random intercepts to Model <code>exp5c_m1a</code>. The code <code>(1 | donor:treatment)</code> adds an intercept for each combination of <span class="math inline">\(\texttt{donor}\)</span> and <span class="math inline">\(\texttt{treatment}\)</span>. The code <code>(1 | donor:activity)</code> adds an intercept for each combination of <span class="math inline">\(\texttt{donor}\)</span> and <span class="math inline">\(\texttt{activity}\)</span>. This model is the equivalent of the univariate model of a <strong>repeated measures ANOVA</strong> of these data (see <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#lmm-rmanova-5c">Classical (“univariate model”) repeated measures ANOVA</a> below).</li>
<li>Model <code>exp5c_m1c</code> adds only one additional random intercept (<code>(1 | donor:treatment)</code>). It is a simplification of Model <code>exp5c_m1c</code>. This is the model reported and explained above.</li>
<li>Model <code>exp5c_m1d</code> has the same random effects as <code>exp5c_m1a</code> but models unstructured correlated error and heterogeneity of the error, as described for Example 2: <a href="lmm-example2.html#lmm-exp6g-alt">Alternative models for exp6g</a>. This model is equivalent to the multivariate model of a <strong>repeated measures ANOVA</strong> of these data (see <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#lmm-rmanova-5c-multi">“Multivariate model” repeated measures ANOVA</a> below).</li>
<li><code>lmer</code> returns a message “boundary (singular) fit: see ?isSingular” for <code>exp5c_m1b</code>. The model <em>was</em> fit but the message means that we should be cautious about (or simply avoid) interpreting any inferential statistics (SEs, CIs, <em>p</em>-values). Looking at the estimates of the estimated parameters of the random effects in the table below (the variances of the effects and the covariances among the effects), Model <code>exp5c_m1b</code> estimates zero variance for the <span class="math inline">\(\texttt{donor:activity}\)</span> random intercept. This suggests that we could simplify <code>exp5c_m1b</code> by removing <code>(1 | donor:activity)</code> from the model – this is <code>exp5c_m1c</code>. Think about this warning. If we fit using repeated measures ANOVA (next section), we won’t get a warning message but we still may be <a href="https://en.wikipedia.org/wiki/Overfitting" target="_blank">overfitting</a>.</li>
</ol>
<div class="sourceCode" id="cb546"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb546-1"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb546-1" tabindex="-1"></a><span class="fu">VarCorr</span>(exp5c_m1b)</span></code></pre></div>
<pre><code>##  Groups          Name        Std.Dev.
##  donor:activity  (Intercept) 0.000000
##  donor:treatment (Intercept) 0.089519
##  donor           (Intercept) 0.352092
##  Residual                    0.090686</code></pre>
<ol start="7" style="list-style-type: decimal">
<li>Planned comparisons of the four alternative models are given in Table <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#tab:lmm-exp5c-alternatives-planned">16.12</a>. The effect estimates are the same but inference differs slightly among the models because of how each model partitions error variance to either random effects or the residuals.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-alternatives-planned">Table 16.12: </span>Planned contrasts from the four alternative models.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1a</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
2.25
</td>
<td style="text-align:right;">
0.040
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.26
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
-1.73
</td>
<td style="text-align:right;">
0.105
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.47
</td>
<td style="text-align:right;">
-0.06
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
0.013
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1b</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.016
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
-2.22
</td>
<td style="text-align:right;">
0.051
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.46
</td>
<td style="text-align:right;">
-0.08
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.015
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1c</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.016
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
-2.22
</td>
<td style="text-align:right;">
0.051
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.43
</td>
<td style="text-align:right;">
-0.10
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.005
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1d</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.07
</td>
<td style="text-align:right;">
0.24
</td>
<td style="text-align:right;">
3.82
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.25
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
-1.85
</td>
<td style="text-align:right;">
0.083
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.44
</td>
<td style="text-align:right;">
-0.09
</td>
<td style="text-align:right;">
-3.22
</td>
<td style="text-align:right;">
0.006
</td>
</tr>
</tbody>
</table>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-aic">Table 16.13: </span>AIC of the four alternative models firt to exp5c data
</caption>
<thead>
<tr>
<th style="text-align:left;">
Model
</th>
<th style="text-align:right;">
AIC
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
exp5c_m1a
</td>
<td style="text-align:right;">
8.153813
</td>
</tr>
<tr>
<td style="text-align:left;">
exp5c_m1b
</td>
<td style="text-align:right;">
10.052675
</td>
</tr>
<tr>
<td style="text-align:left;">
exp5c_m1c
</td>
<td style="text-align:right;">
8.052675
</td>
</tr>
<tr>
<td style="text-align:left;">
exp5c_m1d
</td>
<td style="text-align:right;">
14.763024
</td>
</tr>
</tbody>
</table>
<ol start="6" style="list-style-type: decimal">
<li>The use of AIC for selecting among models with different random effects was described above. The AICs of the models suggest that models <code>exp5c_m1b</code> and <code>exp5c_m1d</code> are too complex given the data. The AICs of models <code>exp5c_m1a</code> and <code>exp5c_m1c</code> are effectively the same. I reported <code>exp5c_m1c</code> simply to allow me to explain the <code>donor:treatment</code> random effect.</li>
</ol>
</div>
<div id="lmm-rmanova-5c" class="section level3 hasAnchor" number="16.8.9">
<h3><span class="header-section-number">16.8.9</span> Classical (“univariate model”) repeated measures ANOVA<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#lmm-rmanova-5c" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb548"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb548-1"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb548-1" tabindex="-1"></a>exp5c_aov1 <span class="ot">&lt;-</span> <span class="fu">aov_4</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span></span>
<span id="cb548-2"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb548-2" tabindex="-1"></a>                    (treatment <span class="sc">*</span> activity <span class="sc">|</span> donor),</span>
<span id="cb548-3"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb548-3" tabindex="-1"></a>                  <span class="at">data =</span> exp5c)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-uni-rmanova-planned">Table 16.14: </span>Planned contrasts for the linear mixed model exp5c_m1c and the univariate model of the repeated measures ANOVA.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1c</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0.016
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.052
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.23
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
-2.22
</td>
<td style="text-align:right;">
0.051
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.43
</td>
<td style="text-align:right;">
-0.10
</td>
<td style="text-align:right;">
-3.61
</td>
<td style="text-align:right;">
0.005
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>univariate RM-ANOVA</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.25
</td>
<td style="text-align:right;">
3.82
</td>
<td style="text-align:right;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.28
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
-1.85
</td>
<td style="text-align:right;">
0.123
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.48
</td>
<td style="text-align:right;">
-0.05
</td>
<td style="text-align:right;">
-3.22
</td>
<td style="text-align:right;">
0.023
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>In addition to the assumptions of the linear model outlined in the Violations chapter, the classical (“univariate model”) repeated measures ANOVA assumes <a href="https://en.wikipedia.org/wiki/Mauchly%27s_sphericity_test#Sphericity" target="_blank">sphericity</a>, which is the equality of the variances of all pairwise differences among treatment combinations.</li>
<li>The univariate model of the repeated measures ANOVA is equivalent to Model <code>exp5c_m1c</code>. Inference (SEs, CIs, <em>p</em>-values) differs slightly in this here because the estimated variance for the <span class="math inline">\(\texttt{donor:activity}\)</span> random intercept of Model <code>exp5c_m1c</code> is zero. In cases where all variances are greater than zero <em>and the design is balanced</em> (no block is missing a treatment combination), the two tables would be identical.</li>
</ol>
</div>
<div id="lmm-rmanova-5c-multi" class="section level3 hasAnchor" number="16.8.10">
<h3><span class="header-section-number">16.8.10</span> “Multivariate model” repeated measures ANOVA of exp5c<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#lmm-rmanova-5c-multi" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:lmm-exp5c-multi-rmanova">Table 16.15: </span>Planned contrasts for the linear mixed model exp5c_m1d and the multivariate model of the repeated measures ANOVA.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1d</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.07
</td>
<td style="text-align:right;">
0.24
</td>
<td style="text-align:right;">
3.82
</td>
<td style="text-align:right;">
0.002
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.25
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
-1.85
</td>
<td style="text-align:right;">
0.083
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.44
</td>
<td style="text-align:right;">
-0.09
</td>
<td style="text-align:right;">
-3.22
</td>
<td style="text-align:right;">
0.006
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>multivariate RM-ANOVA</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.040
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.25
</td>
<td style="text-align:right;">
3.82
</td>
<td style="text-align:right;">
0.012
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.28
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
-1.85
</td>
<td style="text-align:right;">
0.123
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
-0.48
</td>
<td style="text-align:right;">
-0.05
</td>
<td style="text-align:right;">
-3.22
</td>
<td style="text-align:right;">
0.023
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The sphericity assumption of the repeated measures ANOVA is relaxed if the model is fit using the “multivariate model”.</li>
<li>The multivariate model repeated measures ANOVA is a linear model with a multivariate response and not a linear mixed model. In the multivariate model, each treatment combination is a different response variable and there is a single row for each level of the random factor (<span class="math inline">\(\texttt{donor}\)</span> in Experiment 5c). The multivariate model is a different way of handling the correlated error that occurs when conceiving of the design as univariate.</li>
<li>The multivariate repeated measures ANOVA can be specified using the linear mixed model <code>exp5c_m1d</code>. Compared to the contrasts from the multivariate model of the repeated measures ANOVA above, the SEs of the contrasts are the same but the degrees of freedom differ. Because of the increased df of the linear mixed model, the CIs are narrower and the <em>p</em>-value is smaller – that is the linear mixed model is less conservative than the repeated measures ANOVA. There is no correct degrees of freedom for a linear mixed model like this and emmeans outputs one way to compute these (with options for others, none of which are equivalent to those from the multivariate model RM-ANOVA).</li>
</ol>
</div>
<div id="modeling-textttdonor-as-a-fixed-effect" class="section level3 hasAnchor" number="16.8.11">
<h3><span class="header-section-number">16.8.11</span> Modeling <span class="math inline">\(\texttt{donor}\)</span> as a fixed effect<a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#modeling-textttdonor-as-a-fixed-effect" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><span class="math display" id="eq:lmm-exp5c-m3">\[
\begin{equation}
\texttt{glucose_uptake} \sim \texttt{treatment * activity + donor}
\end{equation}
\tag{16.5}
\]</span></p>
<p>Model <a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#eq:lmm-exp5c-m3">(16.5)</a> (using R formula syntax) is a linear model with the block <span class="math inline">\(\texttt{donor}\)</span> added as a fixed covariate instead of a random intercept.</p>
<div class="sourceCode" id="cb549"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb549-1"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb549-1" tabindex="-1"></a>exp5c_m3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(glucose_uptake <span class="sc">~</span> treatment <span class="sc">*</span> activity <span class="sc">+</span> donor,</span>
<span id="cb549-2"><a href="example-3-factorial-experiments-and-no-subsampling-replicates-exp5c.html#cb549-2" tabindex="-1"></a>               <span class="at">data =</span> exp5c)</span></code></pre></div>
<p>The coefficients of the fit model are</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Std. Error
</th>
<th style="text-align:right;">
t value
</th>
<th style="text-align:right;">
Pr(&gt;|t|)
</th>
<th style="text-align:right;">
2.5 %
</th>
<th style="text-align:right;">
97.5 %
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
(Intercept)
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.071
</td>
<td style="text-align:right;">
14.0
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
0.85
</td>
<td style="text-align:right;">
1.15
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
1.2
</td>
<td style="text-align:right;">
0.2453
</td>
<td style="text-align:right;">
-0.06
</td>
<td style="text-align:right;">
0.22
</td>
</tr>
<tr>
<td style="text-align:left;">
activityEPS
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
2.2
</td>
<td style="text-align:right;">
0.0402
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.29
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_2
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
0.2
</td>
<td style="text-align:right;">
0.8444
</td>
<td style="text-align:right;">
-0.16
</td>
<td style="text-align:right;">
0.19
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_3
</td>
<td style="text-align:right;">
0.16
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
1.9
</td>
<td style="text-align:right;">
0.0757
</td>
<td style="text-align:right;">
-0.02
</td>
<td style="text-align:right;">
0.33
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_4
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
9.8
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
0.63
</td>
<td style="text-align:right;">
0.98
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_5
</td>
<td style="text-align:right;">
0.76
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
9.2
</td>
<td style="text-align:right;">
0.0000
</td>
<td style="text-align:right;">
0.59
</td>
<td style="text-align:right;">
0.94
</td>
</tr>
<tr>
<td style="text-align:left;">
donordonor_6
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
3.5
</td>
<td style="text-align:right;">
0.0030
</td>
<td style="text-align:right;">
0.12
</td>
<td style="text-align:right;">
0.47
</td>
</tr>
<tr>
<td style="text-align:left;">
treatmentsiNR4A3:activityEPS
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
-2.8
</td>
<td style="text-align:right;">
0.0132
</td>
<td style="text-align:right;">
-0.47
</td>
<td style="text-align:right;">
-0.06
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>The coefficients include a slope for the five non-reference levels of donor. We typically don’t care about these.</li>
</ol>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-232">Table 16.16: </span>Planned contrasts for the linear mixed model exp5c_m1a and the fixed effect model exp5c_m3.
</caption>
<thead>
<tr>
<th style="text-align:left;">
contrast
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
SE
</th>
<th style="text-align:right;">
df
</th>
<th style="text-align:right;">
lower.CL
</th>
<th style="text-align:right;">
upper.CL
</th>
<th style="text-align:right;">
t.ratio
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m1a</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
2.25
</td>
<td style="text-align:right;">
0.040
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.26
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
-1.73
</td>
<td style="text-align:right;">
0.105
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.47
</td>
<td style="text-align:right;">
-0.06
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
0.013
</td>
</tr>
<tr grouplength="3">
<td colspan="8" style="border-bottom: 1px solid;">
<strong>exp5c_m3</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(Scr EPS) - (Scr Basal)
</td>
<td style="text-align:right;">
0.15
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
2.25
</td>
<td style="text-align:right;">
0.040
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
(siNR4A3 EPS) - (siNR4A3 Basal)
</td>
<td style="text-align:right;">
-0.12
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.26
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
-1.73
</td>
<td style="text-align:right;">
0.105
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
Interaction
</td>
<td style="text-align:right;">
-0.27
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-0.47
</td>
<td style="text-align:right;">
-0.06
</td>
<td style="text-align:right;">
-2.81
</td>
<td style="text-align:right;">
0.013
</td>
</tr>
</tbody>
</table>
<p>Notes</p>
<ol style="list-style-type: decimal">
<li>In a randomized complete block design with only 1 replicate of each treatment combination per block, like that in Experiment 5c, inference about treatment effects is exactly that same between this fixed effect model and the random intercept model <code>exp5c_m1a.</code></li>
<li>The equivalence of inference in the treatment effect between the fixed effect and linear mixed model holds only for balanced randomized complete block designs – where all blocks contain all treatment combinations and the subsampling replicate size is the same for all treatment combinations for all blocks. This means, outside of special cases like Experiment 5c, the choice between adding a blocking variable as a random or fixed factor depends on assumptions about the model.</li>
</ol>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="lmm-example2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="lmm-example4.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/chapters/55-lmm01-blocking.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Walker-elementary-statistical-modeling-draft.pdf", "Walker-elementary-statistical-modeling-draft.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
